// @pulumi/azure-native/types/enums/synapse/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspacePublicNetworkAccess = exports.TransparentDataEncryptionStatus = exports.StorageAccountType = exports.SkuSize = exports.SkuName = exports.SensitivityLabelRank = exports.ResourceIdentityType = exports.PrincipalType = exports.NodeSizeFamily = exports.NodeSize = exports.Kind = exports.IotHubDataFormat = exports.IntegrationRuntimeType = exports.IntegrationRuntimeSsisCatalogPricingTier = exports.IntegrationRuntimeLicenseType = exports.IntegrationRuntimeEntityReferenceType = exports.IntegrationRuntimeEdition = exports.EventHubDataFormat = exports.EventGridDataFormat = exports.DefaultPrincipalsModificationKind = exports.DatabasePrincipalRole = exports.DataFlowComputeType = exports.DataConnectionKind = exports.CreateMode = exports.ConfigurationType = exports.Compression = exports.ClusterPrincipalRole = exports.BlobStorageEventType = exports.v20210601preview = exports.v20210601 = exports.v20210501 = exports.v20210401preview = void 0;
// Export sub-modules:
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.BlobStorageEventType = {
    Microsoft_Storage_BlobCreated: "Microsoft.Storage.BlobCreated",
    Microsoft_Storage_BlobRenamed: "Microsoft.Storage.BlobRenamed",
};
exports.ClusterPrincipalRole = {
    AllDatabasesAdmin: "AllDatabasesAdmin",
    AllDatabasesViewer: "AllDatabasesViewer",
};
exports.Compression = {
    None: "None",
    GZip: "GZip",
};
exports.ConfigurationType = {
    File: "File",
    Artifact: "Artifact",
};
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    Recovery: "Recovery",
    Restore: "Restore",
};
exports.DataConnectionKind = {
    EventHub: "EventHub",
    EventGrid: "EventGrid",
    IotHub: "IotHub",
};
exports.DataFlowComputeType = {
    General: "General",
    MemoryOptimized: "MemoryOptimized",
    ComputeOptimized: "ComputeOptimized",
};
exports.DatabasePrincipalRole = {
    Admin: "Admin",
    Ingestor: "Ingestor",
    Monitor: "Monitor",
    User: "User",
    UnrestrictedViewer: "UnrestrictedViewer",
    Viewer: "Viewer",
};
exports.DefaultPrincipalsModificationKind = {
    Union: "Union",
    Replace: "Replace",
    None: "None",
};
exports.EventGridDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.EventHubDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.IntegrationRuntimeEdition = {
    Standard: "Standard",
    Enterprise: "Enterprise",
};
exports.IntegrationRuntimeEntityReferenceType = {
    IntegrationRuntimeReference: "IntegrationRuntimeReference",
    LinkedServiceReference: "LinkedServiceReference",
};
exports.IntegrationRuntimeLicenseType = {
    BasePrice: "BasePrice",
    LicenseIncluded: "LicenseIncluded",
};
exports.IntegrationRuntimeSsisCatalogPricingTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    PremiumRS: "PremiumRS",
};
exports.IntegrationRuntimeType = {
    Managed: "Managed",
    SelfHosted: "SelfHosted",
};
exports.IotHubDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.Kind = {
    ReadWrite: "ReadWrite",
    ReadOnlyFollowing: "ReadOnlyFollowing",
};
exports.NodeSize = {
    None: "None",
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
    XLarge: "XLarge",
    XXLarge: "XXLarge",
    XXXLarge: "XXXLarge",
};
exports.NodeSizeFamily = {
    None: "None",
    MemoryOptimized: "MemoryOptimized",
    HardwareAcceleratedFPGA: "HardwareAcceleratedFPGA",
    HardwareAcceleratedGPU: "HardwareAcceleratedGPU",
};
exports.PrincipalType = {
    App: "App",
    Group: "Group",
    User: "User",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.SensitivityLabelRank = {
    None: "None",
    Low: "Low",
    Medium: "Medium",
    High: "High",
    Critical: "Critical",
};
exports.SkuName = {
    Compute_optimized: "Compute optimized",
    Storage_optimized: "Storage optimized",
};
exports.SkuSize = {
    Extra_small: "Extra small",
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
};
exports.StorageAccountType = {
    GRS: "GRS",
    LRS: "LRS",
};
exports.TransparentDataEncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.WorkspacePublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
