// @pulumi/azure-native/types/enums/appplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.TriggerType = exports.StorageType = exports.SessionAffinity = exports.ProbeActionType = exports.ManagedIdentityType = exports.HTTPSchemeType = exports.GitImplementation = exports.GatewayRouteConfigProtocol = exports.GatewayCertificateVerification = exports.DevToolPortalFeatureState = exports.ConfigurationServiceGeneration = exports.ConfigServerEnabledState = exports.BindingType = exports.BackendProtocol = exports.ApmType = exports.v20240501preview = exports.v20240101preview = exports.v20231201 = exports.v20231101preview = exports.v20230901preview = exports.v20230701preview = exports.v20230501preview = void 0;
// Export sub-modules:
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
exports.ApmType = {
    ApplicationInsights: "ApplicationInsights",
    AppDynamics: "AppDynamics",
    Dynatrace: "Dynatrace",
    NewRelic: "NewRelic",
    ElasticAPM: "ElasticAPM",
};
exports.BackendProtocol = {
    GRPC: "GRPC",
    Default: "Default",
};
exports.BindingType = {
    ApplicationInsights: "ApplicationInsights",
    ApacheSkyWalking: "ApacheSkyWalking",
    AppDynamics: "AppDynamics",
    Dynatrace: "Dynatrace",
    NewRelic: "NewRelic",
    ElasticAPM: "ElasticAPM",
    CACertificates: "CACertificates",
};
exports.ConfigServerEnabledState = {
    /**
     * Enable the config server.
     */
    Enabled: "Enabled",
    /**
     * Disable the config server.
     */
    Disabled: "Disabled",
};
exports.ConfigurationServiceGeneration = {
    Gen1: "Gen1",
    Gen2: "Gen2",
};
exports.DevToolPortalFeatureState = {
    /**
     * Enable the plugin in Dev Tool Portal.
     */
    Enabled: "Enabled",
    /**
     * Disable the plugin in Dev Tool Portal.
     */
    Disabled: "Disabled",
};
exports.GatewayCertificateVerification = {
    /**
     * Enable certificate verification in Spring Cloud Gateway.
     */
    Enabled: "Enabled",
    /**
     * Disable certificate verification in Spring Cloud Gateway.
     */
    Disabled: "Disabled",
};
exports.GatewayRouteConfigProtocol = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.GitImplementation = {
    Go_git: "go-git",
    Libgit2: "libgit2",
};
exports.HTTPSchemeType = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ProbeActionType = {
    HTTPGetAction: "HTTPGetAction",
    TCPSocketAction: "TCPSocketAction",
    ExecAction: "ExecAction",
};
exports.SessionAffinity = {
    Cookie: "Cookie",
    None: "None",
};
exports.StorageType = {
    StorageAccount: "StorageAccount",
};
exports.TriggerType = {
    Manual: "Manual",
};
exports.Type = {
    AzureFileVolume: "AzureFileVolume",
};
