// @pulumi/azure-native/types/enums/labservices/v20220801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.SkuTier = exports.ShutdownOnIdleMode = exports.ResourceIdentityType = exports.RecurrenceFrequency = exports.EnableState = exports.CreateOption = exports.ConnectionType = void 0;
exports.ConnectionType = {
    Public: "Public",
    Private: "Private",
    None: "None",
};
exports.CreateOption = {
    /**
     * An image is used to create all lab user virtual machines. When this option is set, no template VM will be created.
     */
    Image: "Image",
    /**
     * A template VM will be used to create all lab user virtual machines.
     */
    TemplateVM: "TemplateVM",
};
exports.EnableState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RecurrenceFrequency = {
    /**
     * Schedule will run every days.
     */
    Daily: "Daily",
    /**
     * Schedule will run every week on days specified in weekDays.
     */
    Weekly: "Weekly",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.ShutdownOnIdleMode = {
    /**
     * The VM won't be shut down when it is idle.
     */
    None: "None",
    /**
     * The VM will be considered as idle when there is no keyboard or mouse input.
     */
    UserAbsence: "UserAbsence",
    /**
     * The VM will be considered as idle when user is absent and the resource (CPU and disk) consumption is low.
     */
    LowUsage: "LowUsage",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.WeekDay = {
    /**
     * Schedule will run on Sunday
     */
    Sunday: "Sunday",
    /**
     * Schedule will run on Monday
     */
    Monday: "Monday",
    /**
     * Schedule will run on Tuesday
     */
    Tuesday: "Tuesday",
    /**
     * Schedule will run on Wednesday
     */
    Wednesday: "Wednesday",
    /**
     * Schedule will run on Thursday
     */
    Thursday: "Thursday",
    /**
     * Schedule will run on Friday
     */
    Friday: "Friday",
    /**
     * Schedule will run on Saturday
     */
    Saturday: "Saturday",
};
