// @pulumi/azure-native/types/enums/resources/v20240301/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.OnErrorDeploymentType = exports.ExtendedLocationType = exports.ExpressionEvaluationOptionsScopeType = exports.DeploymentStacksDeleteDetachEnum = exports.DeploymentMode = exports.DenySettingsMode = void 0;
exports.DenySettingsMode = {
    /**
     * Authorized users are able to read and modify the resources, but cannot delete.
     */
    DenyDelete: "denyDelete",
    /**
     * Authorized users can read from a resource, but cannot modify or delete it.
     */
    DenyWriteAndDelete: "denyWriteAndDelete",
    /**
     * No denyAssignments have been applied.
     */
    None: "none",
};
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.DeploymentStacksDeleteDetachEnum = {
    Delete: "delete",
    Detach: "detach",
};
exports.ExpressionEvaluationOptionsScopeType = {
    NotSpecified: "NotSpecified",
    Outer: "Outer",
    Inner: "Inner",
};
exports.ExtendedLocationType = {
    EdgeZone: "EdgeZone",
};
exports.OnErrorDeploymentType = {
    LastSuccessful: "LastSuccessful",
    SpecificDeployment: "SpecificDeployment",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
