// @pulumi/azure-native/azurestackhci/v20230701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetwork = exports.VirtualMachineInstance = exports.VirtualHardDisk = exports.StorageContainer = exports.NetworkInterface = exports.MarketplaceGalleryImage = exports.GuestAgent = exports.getVirtualNetworkOutput = exports.getVirtualNetwork = exports.getVirtualMachineInstanceOutput = exports.getVirtualMachineInstance = exports.getVirtualHardDiskOutput = exports.getVirtualHardDisk = exports.getStorageContainerOutput = exports.getStorageContainer = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getMarketplaceGalleryImageOutput = exports.getMarketplaceGalleryImage = exports.getGuestAgentOutput = exports.getGuestAgent = exports.getGalleryImageOutput = exports.getGalleryImage = exports.GalleryImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGuestAgent = null;
exports.getGuestAgentOutput = null;
utilities.lazyLoad(exports, ["getGuestAgent", "getGuestAgentOutput"], () => require("./getGuestAgent"));
exports.getMarketplaceGalleryImage = null;
exports.getMarketplaceGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getMarketplaceGalleryImage", "getMarketplaceGalleryImageOutput"], () => require("./getMarketplaceGalleryImage"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getStorageContainer = null;
exports.getStorageContainerOutput = null;
utilities.lazyLoad(exports, ["getStorageContainer", "getStorageContainerOutput"], () => require("./getStorageContainer"));
exports.getVirtualHardDisk = null;
exports.getVirtualHardDiskOutput = null;
utilities.lazyLoad(exports, ["getVirtualHardDisk", "getVirtualHardDiskOutput"], () => require("./getVirtualHardDisk"));
exports.getVirtualMachineInstance = null;
exports.getVirtualMachineInstanceOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineInstance", "getVirtualMachineInstanceOutput"], () => require("./getVirtualMachineInstance"));
exports.getVirtualNetwork = null;
exports.getVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetwork", "getVirtualNetworkOutput"], () => require("./getVirtualNetwork"));
exports.GuestAgent = null;
utilities.lazyLoad(exports, ["GuestAgent"], () => require("./guestAgent"));
exports.MarketplaceGalleryImage = null;
utilities.lazyLoad(exports, ["MarketplaceGalleryImage"], () => require("./marketplaceGalleryImage"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.StorageContainer = null;
utilities.lazyLoad(exports, ["StorageContainer"], () => require("./storageContainer"));
exports.VirtualHardDisk = null;
utilities.lazyLoad(exports, ["VirtualHardDisk"], () => require("./virtualHardDisk"));
exports.VirtualMachineInstance = null;
utilities.lazyLoad(exports, ["VirtualMachineInstance"], () => require("./virtualMachineInstance"));
exports.VirtualNetwork = null;
utilities.lazyLoad(exports, ["VirtualNetwork"], () => require("./virtualNetwork"));
// Export enums:
__exportStar(require("../../types/enums/azurestackhci/v20230701preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestackhci/v20230701preview:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:GuestAgent":
                return new exports.GuestAgent(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:MarketplaceGalleryImage":
                return new exports.MarketplaceGalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:StorageContainer":
                return new exports.StorageContainer(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:VirtualHardDisk":
                return new exports.VirtualHardDisk(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:VirtualMachineInstance":
                return new exports.VirtualMachineInstance(name, undefined, { urn });
            case "azure-native:azurestackhci/v20230701preview:VirtualNetwork":
                return new exports.VirtualNetwork(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestackhci/v20230701preview", _module);
