// @pulumi/azure-native/web/v20201001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getWebAppPublicCertificateSlotOutput = exports.getWebAppPublicCertificateSlot = exports.getWebAppPublicCertificateOutput = exports.getWebAppPublicCertificate = exports.getWebAppPrivateEndpointConnectionOutput = exports.getWebAppPrivateEndpointConnection = exports.getWebAppPremierAddOnSlotOutput = exports.getWebAppPremierAddOnSlot = exports.getWebAppPremierAddOnOutput = exports.getWebAppPremierAddOn = exports.getWebAppInstanceFunctionSlotOutput = exports.getWebAppInstanceFunctionSlot = exports.getWebAppHybridConnectionSlotOutput = exports.getWebAppHybridConnectionSlot = exports.getWebAppHybridConnectionOutput = exports.getWebAppHybridConnection = exports.getWebAppHostNameBindingSlotOutput = exports.getWebAppHostNameBindingSlot = exports.getWebAppHostNameBindingOutput = exports.getWebAppHostNameBinding = exports.getWebAppFunctionOutput = exports.getWebAppFunction = exports.getWebAppFtpAllowedOutput = exports.getWebAppFtpAllowed = exports.getWebAppDomainOwnershipIdentifierSlotOutput = exports.getWebAppDomainOwnershipIdentifierSlot = exports.getWebAppDomainOwnershipIdentifierOutput = exports.getWebAppDomainOwnershipIdentifier = exports.getWebAppDiagnosticLogsConfigurationSlotOutput = exports.getWebAppDiagnosticLogsConfigurationSlot = exports.getWebAppDiagnosticLogsConfigurationOutput = exports.getWebAppDiagnosticLogsConfiguration = exports.getWebAppDeploymentSlotOutput = exports.getWebAppDeploymentSlot = exports.getWebAppDeploymentOutput = exports.getWebAppDeployment = exports.getWebAppOutput = exports.getWebApp = exports.getStaticSiteOutput = exports.getStaticSite = exports.getCertificateOutput = exports.getCertificate = exports.getAppServicePlanOutput = exports.getAppServicePlan = exports.getAppServiceEnvironmentOutput = exports.getAppServiceEnvironment = exports.Certificate = exports.AppServicePlanRouteForVnet = exports.AppServicePlan = exports.AppServiceEnvironment = void 0;
exports.listWebAppAzureStorageAccountsSlotOutput = exports.listWebAppAzureStorageAccountsSlot = exports.listWebAppAzureStorageAccountsOutput = exports.listWebAppAzureStorageAccounts = exports.listWebAppAuthSettingsSlotOutput = exports.listWebAppAuthSettingsSlot = exports.listWebAppAuthSettingsOutput = exports.listWebAppAuthSettings = exports.listWebAppApplicationSettingsSlotOutput = exports.listWebAppApplicationSettingsSlot = exports.listWebAppApplicationSettingsOutput = exports.listWebAppApplicationSettings = exports.listStaticSiteUsersOutput = exports.listStaticSiteUsers = exports.listStaticSiteSecretsOutput = exports.listStaticSiteSecrets = exports.listStaticSiteFunctionAppSettingsOutput = exports.listStaticSiteFunctionAppSettings = exports.listStaticSiteBuildFunctionAppSettingsOutput = exports.listStaticSiteBuildFunctionAppSettings = exports.listSiteIdentifiersAssignedToHostNameOutput = exports.listSiteIdentifiersAssignedToHostName = exports.listAppServicePlanHybridConnectionKeysOutput = exports.listAppServicePlanHybridConnectionKeys = exports.getWebAppVnetConnectionSlotOutput = exports.getWebAppVnetConnectionSlot = exports.getWebAppVnetConnectionOutput = exports.getWebAppVnetConnection = exports.getWebAppSwiftVirtualNetworkConnectionSlotOutput = exports.getWebAppSwiftVirtualNetworkConnectionSlot = exports.getWebAppSwiftVirtualNetworkConnectionOutput = exports.getWebAppSwiftVirtualNetworkConnection = exports.getWebAppSourceControlSlotOutput = exports.getWebAppSourceControlSlot = exports.getWebAppSourceControlOutput = exports.getWebAppSourceControl = exports.getWebAppSlotConfigurationNamesOutput = exports.getWebAppSlotConfigurationNames = exports.getWebAppSlotOutput = exports.getWebAppSlot = exports.getWebAppSiteExtensionSlotOutput = exports.getWebAppSiteExtensionSlot = exports.getWebAppSiteExtensionOutput = exports.getWebAppSiteExtension = exports.getWebAppScmAllowedOutput = exports.getWebAppScmAllowed = exports.getWebAppRelayServiceConnectionSlotOutput = exports.getWebAppRelayServiceConnectionSlot = exports.getWebAppRelayServiceConnectionOutput = exports.getWebAppRelayServiceConnection = void 0;
exports.WebAppAuthSettingsSlot = exports.WebAppAuthSettings = exports.WebAppApplicationSettingsSlot = exports.WebAppApplicationSettings = exports.WebApp = exports.StaticSite = exports.listWebAppSyncFunctionTriggersSlotOutput = exports.listWebAppSyncFunctionTriggersSlot = exports.listWebAppSyncFunctionTriggersOutput = exports.listWebAppSyncFunctionTriggers = exports.listWebAppSitePushSettingsSlotOutput = exports.listWebAppSitePushSettingsSlot = exports.listWebAppSitePushSettingsOutput = exports.listWebAppSitePushSettings = exports.listWebAppSiteBackupsSlotOutput = exports.listWebAppSiteBackupsSlot = exports.listWebAppSiteBackupsOutput = exports.listWebAppSiteBackups = exports.listWebAppPublishingCredentialsSlotOutput = exports.listWebAppPublishingCredentialsSlot = exports.listWebAppPublishingCredentialsOutput = exports.listWebAppPublishingCredentials = exports.listWebAppMetadataSlotOutput = exports.listWebAppMetadataSlot = exports.listWebAppMetadataOutput = exports.listWebAppMetadata = exports.listWebAppHostKeysSlotOutput = exports.listWebAppHostKeysSlot = exports.listWebAppHostKeysOutput = exports.listWebAppHostKeys = exports.listWebAppFunctionSecretsSlotOutput = exports.listWebAppFunctionSecretsSlot = exports.listWebAppFunctionSecretsOutput = exports.listWebAppFunctionSecrets = exports.listWebAppFunctionKeysSlotOutput = exports.listWebAppFunctionKeysSlot = exports.listWebAppFunctionKeysOutput = exports.listWebAppFunctionKeys = exports.listWebAppConnectionStringsSlotOutput = exports.listWebAppConnectionStringsSlot = exports.listWebAppConnectionStringsOutput = exports.listWebAppConnectionStrings = exports.listWebAppBackupStatusSecretsSlotOutput = exports.listWebAppBackupStatusSecretsSlot = exports.listWebAppBackupStatusSecretsOutput = exports.listWebAppBackupStatusSecrets = exports.listWebAppBackupConfigurationSlotOutput = exports.listWebAppBackupConfigurationSlot = exports.listWebAppBackupConfigurationOutput = exports.listWebAppBackupConfiguration = void 0;
exports.WebAppVnetConnectionSlot = exports.WebAppVnetConnection = exports.WebAppSwiftVirtualNetworkConnectionSlot = exports.WebAppSwiftVirtualNetworkConnection = exports.WebAppSourceControlSlot = exports.WebAppSourceControl = exports.WebAppSlotConfigurationNames = exports.WebAppSlot = exports.WebAppSitePushSettingsSlot = exports.WebAppSitePushSettings = exports.WebAppSiteExtensionSlot = exports.WebAppSiteExtension = exports.WebAppScmAllowed = exports.WebAppRelayServiceConnectionSlot = exports.WebAppRelayServiceConnection = exports.WebAppPublicCertificateSlot = exports.WebAppPublicCertificate = exports.WebAppPrivateEndpointConnection = exports.WebAppPremierAddOnSlot = exports.WebAppPremierAddOn = exports.WebAppMetadataSlot = exports.WebAppMetadata = exports.WebAppInstanceFunctionSlot = exports.WebAppHybridConnectionSlot = exports.WebAppHybridConnection = exports.WebAppHostNameBindingSlot = exports.WebAppHostNameBinding = exports.WebAppFunction = exports.WebAppFtpAllowed = exports.WebAppDomainOwnershipIdentifierSlot = exports.WebAppDomainOwnershipIdentifier = exports.WebAppDiagnosticLogsConfigurationSlot = exports.WebAppDiagnosticLogsConfiguration = exports.WebAppDeploymentSlot = exports.WebAppDeployment = exports.WebAppConnectionStringsSlot = exports.WebAppConnectionStrings = exports.WebAppBackupConfigurationSlot = exports.WebAppBackupConfiguration = exports.WebAppAzureStorageAccountsSlot = exports.WebAppAzureStorageAccounts = exports.WebAppAuthSettingsV2Slot = exports.WebAppAuthSettingsV2 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AppServiceEnvironment = null;
utilities.lazyLoad(exports, ["AppServiceEnvironment"], () => require("./appServiceEnvironment"));
exports.AppServicePlan = null;
utilities.lazyLoad(exports, ["AppServicePlan"], () => require("./appServicePlan"));
exports.AppServicePlanRouteForVnet = null;
utilities.lazyLoad(exports, ["AppServicePlanRouteForVnet"], () => require("./appServicePlanRouteForVnet"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.getAppServiceEnvironment = null;
exports.getAppServiceEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getAppServiceEnvironment", "getAppServiceEnvironmentOutput"], () => require("./getAppServiceEnvironment"));
exports.getAppServicePlan = null;
exports.getAppServicePlanOutput = null;
utilities.lazyLoad(exports, ["getAppServicePlan", "getAppServicePlanOutput"], () => require("./getAppServicePlan"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getStaticSite = null;
exports.getStaticSiteOutput = null;
utilities.lazyLoad(exports, ["getStaticSite", "getStaticSiteOutput"], () => require("./getStaticSite"));
exports.getWebApp = null;
exports.getWebAppOutput = null;
utilities.lazyLoad(exports, ["getWebApp", "getWebAppOutput"], () => require("./getWebApp"));
exports.getWebAppDeployment = null;
exports.getWebAppDeploymentOutput = null;
utilities.lazyLoad(exports, ["getWebAppDeployment", "getWebAppDeploymentOutput"], () => require("./getWebAppDeployment"));
exports.getWebAppDeploymentSlot = null;
exports.getWebAppDeploymentSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDeploymentSlot", "getWebAppDeploymentSlotOutput"], () => require("./getWebAppDeploymentSlot"));
exports.getWebAppDiagnosticLogsConfiguration = null;
exports.getWebAppDiagnosticLogsConfigurationOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfiguration", "getWebAppDiagnosticLogsConfigurationOutput"], () => require("./getWebAppDiagnosticLogsConfiguration"));
exports.getWebAppDiagnosticLogsConfigurationSlot = null;
exports.getWebAppDiagnosticLogsConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfigurationSlot", "getWebAppDiagnosticLogsConfigurationSlotOutput"], () => require("./getWebAppDiagnosticLogsConfigurationSlot"));
exports.getWebAppDomainOwnershipIdentifier = null;
exports.getWebAppDomainOwnershipIdentifierOutput = null;
utilities.lazyLoad(exports, ["getWebAppDomainOwnershipIdentifier", "getWebAppDomainOwnershipIdentifierOutput"], () => require("./getWebAppDomainOwnershipIdentifier"));
exports.getWebAppDomainOwnershipIdentifierSlot = null;
exports.getWebAppDomainOwnershipIdentifierSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDomainOwnershipIdentifierSlot", "getWebAppDomainOwnershipIdentifierSlotOutput"], () => require("./getWebAppDomainOwnershipIdentifierSlot"));
exports.getWebAppFtpAllowed = null;
exports.getWebAppFtpAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppFtpAllowed", "getWebAppFtpAllowedOutput"], () => require("./getWebAppFtpAllowed"));
exports.getWebAppFunction = null;
exports.getWebAppFunctionOutput = null;
utilities.lazyLoad(exports, ["getWebAppFunction", "getWebAppFunctionOutput"], () => require("./getWebAppFunction"));
exports.getWebAppHostNameBinding = null;
exports.getWebAppHostNameBindingOutput = null;
utilities.lazyLoad(exports, ["getWebAppHostNameBinding", "getWebAppHostNameBindingOutput"], () => require("./getWebAppHostNameBinding"));
exports.getWebAppHostNameBindingSlot = null;
exports.getWebAppHostNameBindingSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppHostNameBindingSlot", "getWebAppHostNameBindingSlotOutput"], () => require("./getWebAppHostNameBindingSlot"));
exports.getWebAppHybridConnection = null;
exports.getWebAppHybridConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebAppHybridConnection", "getWebAppHybridConnectionOutput"], () => require("./getWebAppHybridConnection"));
exports.getWebAppHybridConnectionSlot = null;
exports.getWebAppHybridConnectionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppHybridConnectionSlot", "getWebAppHybridConnectionSlotOutput"], () => require("./getWebAppHybridConnectionSlot"));
exports.getWebAppInstanceFunctionSlot = null;
exports.getWebAppInstanceFunctionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppInstanceFunctionSlot", "getWebAppInstanceFunctionSlotOutput"], () => require("./getWebAppInstanceFunctionSlot"));
exports.getWebAppPremierAddOn = null;
exports.getWebAppPremierAddOnOutput = null;
utilities.lazyLoad(exports, ["getWebAppPremierAddOn", "getWebAppPremierAddOnOutput"], () => require("./getWebAppPremierAddOn"));
exports.getWebAppPremierAddOnSlot = null;
exports.getWebAppPremierAddOnSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppPremierAddOnSlot", "getWebAppPremierAddOnSlotOutput"], () => require("./getWebAppPremierAddOnSlot"));
exports.getWebAppPrivateEndpointConnection = null;
exports.getWebAppPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebAppPrivateEndpointConnection", "getWebAppPrivateEndpointConnectionOutput"], () => require("./getWebAppPrivateEndpointConnection"));
exports.getWebAppPublicCertificate = null;
exports.getWebAppPublicCertificateOutput = null;
utilities.lazyLoad(exports, ["getWebAppPublicCertificate", "getWebAppPublicCertificateOutput"], () => require("./getWebAppPublicCertificate"));
exports.getWebAppPublicCertificateSlot = null;
exports.getWebAppPublicCertificateSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppPublicCertificateSlot", "getWebAppPublicCertificateSlotOutput"], () => require("./getWebAppPublicCertificateSlot"));
exports.getWebAppRelayServiceConnection = null;
exports.getWebAppRelayServiceConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebAppRelayServiceConnection", "getWebAppRelayServiceConnectionOutput"], () => require("./getWebAppRelayServiceConnection"));
exports.getWebAppRelayServiceConnectionSlot = null;
exports.getWebAppRelayServiceConnectionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppRelayServiceConnectionSlot", "getWebAppRelayServiceConnectionSlotOutput"], () => require("./getWebAppRelayServiceConnectionSlot"));
exports.getWebAppScmAllowed = null;
exports.getWebAppScmAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppScmAllowed", "getWebAppScmAllowedOutput"], () => require("./getWebAppScmAllowed"));
exports.getWebAppSiteExtension = null;
exports.getWebAppSiteExtensionOutput = null;
utilities.lazyLoad(exports, ["getWebAppSiteExtension", "getWebAppSiteExtensionOutput"], () => require("./getWebAppSiteExtension"));
exports.getWebAppSiteExtensionSlot = null;
exports.getWebAppSiteExtensionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSiteExtensionSlot", "getWebAppSiteExtensionSlotOutput"], () => require("./getWebAppSiteExtensionSlot"));
exports.getWebAppSlot = null;
exports.getWebAppSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSlot", "getWebAppSlotOutput"], () => require("./getWebAppSlot"));
exports.getWebAppSlotConfigurationNames = null;
exports.getWebAppSlotConfigurationNamesOutput = null;
utilities.lazyLoad(exports, ["getWebAppSlotConfigurationNames", "getWebAppSlotConfigurationNamesOutput"], () => require("./getWebAppSlotConfigurationNames"));
exports.getWebAppSourceControl = null;
exports.getWebAppSourceControlOutput = null;
utilities.lazyLoad(exports, ["getWebAppSourceControl", "getWebAppSourceControlOutput"], () => require("./getWebAppSourceControl"));
exports.getWebAppSourceControlSlot = null;
exports.getWebAppSourceControlSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSourceControlSlot", "getWebAppSourceControlSlotOutput"], () => require("./getWebAppSourceControlSlot"));
exports.getWebAppSwiftVirtualNetworkConnection = null;
exports.getWebAppSwiftVirtualNetworkConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebAppSwiftVirtualNetworkConnection", "getWebAppSwiftVirtualNetworkConnectionOutput"], () => require("./getWebAppSwiftVirtualNetworkConnection"));
exports.getWebAppSwiftVirtualNetworkConnectionSlot = null;
exports.getWebAppSwiftVirtualNetworkConnectionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSwiftVirtualNetworkConnectionSlot", "getWebAppSwiftVirtualNetworkConnectionSlotOutput"], () => require("./getWebAppSwiftVirtualNetworkConnectionSlot"));
exports.getWebAppVnetConnection = null;
exports.getWebAppVnetConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebAppVnetConnection", "getWebAppVnetConnectionOutput"], () => require("./getWebAppVnetConnection"));
exports.getWebAppVnetConnectionSlot = null;
exports.getWebAppVnetConnectionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppVnetConnectionSlot", "getWebAppVnetConnectionSlotOutput"], () => require("./getWebAppVnetConnectionSlot"));
exports.listAppServicePlanHybridConnectionKeys = null;
exports.listAppServicePlanHybridConnectionKeysOutput = null;
utilities.lazyLoad(exports, ["listAppServicePlanHybridConnectionKeys", "listAppServicePlanHybridConnectionKeysOutput"], () => require("./listAppServicePlanHybridConnectionKeys"));
exports.listSiteIdentifiersAssignedToHostName = null;
exports.listSiteIdentifiersAssignedToHostNameOutput = null;
utilities.lazyLoad(exports, ["listSiteIdentifiersAssignedToHostName", "listSiteIdentifiersAssignedToHostNameOutput"], () => require("./listSiteIdentifiersAssignedToHostName"));
exports.listStaticSiteBuildFunctionAppSettings = null;
exports.listStaticSiteBuildFunctionAppSettingsOutput = null;
utilities.lazyLoad(exports, ["listStaticSiteBuildFunctionAppSettings", "listStaticSiteBuildFunctionAppSettingsOutput"], () => require("./listStaticSiteBuildFunctionAppSettings"));
exports.listStaticSiteFunctionAppSettings = null;
exports.listStaticSiteFunctionAppSettingsOutput = null;
utilities.lazyLoad(exports, ["listStaticSiteFunctionAppSettings", "listStaticSiteFunctionAppSettingsOutput"], () => require("./listStaticSiteFunctionAppSettings"));
exports.listStaticSiteSecrets = null;
exports.listStaticSiteSecretsOutput = null;
utilities.lazyLoad(exports, ["listStaticSiteSecrets", "listStaticSiteSecretsOutput"], () => require("./listStaticSiteSecrets"));
exports.listStaticSiteUsers = null;
exports.listStaticSiteUsersOutput = null;
utilities.lazyLoad(exports, ["listStaticSiteUsers", "listStaticSiteUsersOutput"], () => require("./listStaticSiteUsers"));
exports.listWebAppApplicationSettings = null;
exports.listWebAppApplicationSettingsOutput = null;
utilities.lazyLoad(exports, ["listWebAppApplicationSettings", "listWebAppApplicationSettingsOutput"], () => require("./listWebAppApplicationSettings"));
exports.listWebAppApplicationSettingsSlot = null;
exports.listWebAppApplicationSettingsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppApplicationSettingsSlot", "listWebAppApplicationSettingsSlotOutput"], () => require("./listWebAppApplicationSettingsSlot"));
exports.listWebAppAuthSettings = null;
exports.listWebAppAuthSettingsOutput = null;
utilities.lazyLoad(exports, ["listWebAppAuthSettings", "listWebAppAuthSettingsOutput"], () => require("./listWebAppAuthSettings"));
exports.listWebAppAuthSettingsSlot = null;
exports.listWebAppAuthSettingsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppAuthSettingsSlot", "listWebAppAuthSettingsSlotOutput"], () => require("./listWebAppAuthSettingsSlot"));
exports.listWebAppAzureStorageAccounts = null;
exports.listWebAppAzureStorageAccountsOutput = null;
utilities.lazyLoad(exports, ["listWebAppAzureStorageAccounts", "listWebAppAzureStorageAccountsOutput"], () => require("./listWebAppAzureStorageAccounts"));
exports.listWebAppAzureStorageAccountsSlot = null;
exports.listWebAppAzureStorageAccountsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppAzureStorageAccountsSlot", "listWebAppAzureStorageAccountsSlotOutput"], () => require("./listWebAppAzureStorageAccountsSlot"));
exports.listWebAppBackupConfiguration = null;
exports.listWebAppBackupConfigurationOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupConfiguration", "listWebAppBackupConfigurationOutput"], () => require("./listWebAppBackupConfiguration"));
exports.listWebAppBackupConfigurationSlot = null;
exports.listWebAppBackupConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupConfigurationSlot", "listWebAppBackupConfigurationSlotOutput"], () => require("./listWebAppBackupConfigurationSlot"));
exports.listWebAppBackupStatusSecrets = null;
exports.listWebAppBackupStatusSecretsOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecrets", "listWebAppBackupStatusSecretsOutput"], () => require("./listWebAppBackupStatusSecrets"));
exports.listWebAppBackupStatusSecretsSlot = null;
exports.listWebAppBackupStatusSecretsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecretsSlot", "listWebAppBackupStatusSecretsSlotOutput"], () => require("./listWebAppBackupStatusSecretsSlot"));
exports.listWebAppConnectionStrings = null;
exports.listWebAppConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listWebAppConnectionStrings", "listWebAppConnectionStringsOutput"], () => require("./listWebAppConnectionStrings"));
exports.listWebAppConnectionStringsSlot = null;
exports.listWebAppConnectionStringsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppConnectionStringsSlot", "listWebAppConnectionStringsSlotOutput"], () => require("./listWebAppConnectionStringsSlot"));
exports.listWebAppFunctionKeys = null;
exports.listWebAppFunctionKeysOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionKeys", "listWebAppFunctionKeysOutput"], () => require("./listWebAppFunctionKeys"));
exports.listWebAppFunctionKeysSlot = null;
exports.listWebAppFunctionKeysSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionKeysSlot", "listWebAppFunctionKeysSlotOutput"], () => require("./listWebAppFunctionKeysSlot"));
exports.listWebAppFunctionSecrets = null;
exports.listWebAppFunctionSecretsOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionSecrets", "listWebAppFunctionSecretsOutput"], () => require("./listWebAppFunctionSecrets"));
exports.listWebAppFunctionSecretsSlot = null;
exports.listWebAppFunctionSecretsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionSecretsSlot", "listWebAppFunctionSecretsSlotOutput"], () => require("./listWebAppFunctionSecretsSlot"));
exports.listWebAppHostKeys = null;
exports.listWebAppHostKeysOutput = null;
utilities.lazyLoad(exports, ["listWebAppHostKeys", "listWebAppHostKeysOutput"], () => require("./listWebAppHostKeys"));
exports.listWebAppHostKeysSlot = null;
exports.listWebAppHostKeysSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppHostKeysSlot", "listWebAppHostKeysSlotOutput"], () => require("./listWebAppHostKeysSlot"));
exports.listWebAppMetadata = null;
exports.listWebAppMetadataOutput = null;
utilities.lazyLoad(exports, ["listWebAppMetadata", "listWebAppMetadataOutput"], () => require("./listWebAppMetadata"));
exports.listWebAppMetadataSlot = null;
exports.listWebAppMetadataSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppMetadataSlot", "listWebAppMetadataSlotOutput"], () => require("./listWebAppMetadataSlot"));
exports.listWebAppPublishingCredentials = null;
exports.listWebAppPublishingCredentialsOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentials", "listWebAppPublishingCredentialsOutput"], () => require("./listWebAppPublishingCredentials"));
exports.listWebAppPublishingCredentialsSlot = null;
exports.listWebAppPublishingCredentialsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentialsSlot", "listWebAppPublishingCredentialsSlotOutput"], () => require("./listWebAppPublishingCredentialsSlot"));
exports.listWebAppSiteBackups = null;
exports.listWebAppSiteBackupsOutput = null;
utilities.lazyLoad(exports, ["listWebAppSiteBackups", "listWebAppSiteBackupsOutput"], () => require("./listWebAppSiteBackups"));
exports.listWebAppSiteBackupsSlot = null;
exports.listWebAppSiteBackupsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppSiteBackupsSlot", "listWebAppSiteBackupsSlotOutput"], () => require("./listWebAppSiteBackupsSlot"));
exports.listWebAppSitePushSettings = null;
exports.listWebAppSitePushSettingsOutput = null;
utilities.lazyLoad(exports, ["listWebAppSitePushSettings", "listWebAppSitePushSettingsOutput"], () => require("./listWebAppSitePushSettings"));
exports.listWebAppSitePushSettingsSlot = null;
exports.listWebAppSitePushSettingsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppSitePushSettingsSlot", "listWebAppSitePushSettingsSlotOutput"], () => require("./listWebAppSitePushSettingsSlot"));
exports.listWebAppSyncFunctionTriggers = null;
exports.listWebAppSyncFunctionTriggersOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggers", "listWebAppSyncFunctionTriggersOutput"], () => require("./listWebAppSyncFunctionTriggers"));
exports.listWebAppSyncFunctionTriggersSlot = null;
exports.listWebAppSyncFunctionTriggersSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggersSlot", "listWebAppSyncFunctionTriggersSlotOutput"], () => require("./listWebAppSyncFunctionTriggersSlot"));
exports.StaticSite = null;
utilities.lazyLoad(exports, ["StaticSite"], () => require("./staticSite"));
exports.WebApp = null;
utilities.lazyLoad(exports, ["WebApp"], () => require("./webApp"));
exports.WebAppApplicationSettings = null;
utilities.lazyLoad(exports, ["WebAppApplicationSettings"], () => require("./webAppApplicationSettings"));
exports.WebAppApplicationSettingsSlot = null;
utilities.lazyLoad(exports, ["WebAppApplicationSettingsSlot"], () => require("./webAppApplicationSettingsSlot"));
exports.WebAppAuthSettings = null;
utilities.lazyLoad(exports, ["WebAppAuthSettings"], () => require("./webAppAuthSettings"));
exports.WebAppAuthSettingsSlot = null;
utilities.lazyLoad(exports, ["WebAppAuthSettingsSlot"], () => require("./webAppAuthSettingsSlot"));
exports.WebAppAuthSettingsV2 = null;
utilities.lazyLoad(exports, ["WebAppAuthSettingsV2"], () => require("./webAppAuthSettingsV2"));
exports.WebAppAuthSettingsV2Slot = null;
utilities.lazyLoad(exports, ["WebAppAuthSettingsV2Slot"], () => require("./webAppAuthSettingsV2Slot"));
exports.WebAppAzureStorageAccounts = null;
utilities.lazyLoad(exports, ["WebAppAzureStorageAccounts"], () => require("./webAppAzureStorageAccounts"));
exports.WebAppAzureStorageAccountsSlot = null;
utilities.lazyLoad(exports, ["WebAppAzureStorageAccountsSlot"], () => require("./webAppAzureStorageAccountsSlot"));
exports.WebAppBackupConfiguration = null;
utilities.lazyLoad(exports, ["WebAppBackupConfiguration"], () => require("./webAppBackupConfiguration"));
exports.WebAppBackupConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppBackupConfigurationSlot"], () => require("./webAppBackupConfigurationSlot"));
exports.WebAppConnectionStrings = null;
utilities.lazyLoad(exports, ["WebAppConnectionStrings"], () => require("./webAppConnectionStrings"));
exports.WebAppConnectionStringsSlot = null;
utilities.lazyLoad(exports, ["WebAppConnectionStringsSlot"], () => require("./webAppConnectionStringsSlot"));
exports.WebAppDeployment = null;
utilities.lazyLoad(exports, ["WebAppDeployment"], () => require("./webAppDeployment"));
exports.WebAppDeploymentSlot = null;
utilities.lazyLoad(exports, ["WebAppDeploymentSlot"], () => require("./webAppDeploymentSlot"));
exports.WebAppDiagnosticLogsConfiguration = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfiguration"], () => require("./webAppDiagnosticLogsConfiguration"));
exports.WebAppDiagnosticLogsConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfigurationSlot"], () => require("./webAppDiagnosticLogsConfigurationSlot"));
exports.WebAppDomainOwnershipIdentifier = null;
utilities.lazyLoad(exports, ["WebAppDomainOwnershipIdentifier"], () => require("./webAppDomainOwnershipIdentifier"));
exports.WebAppDomainOwnershipIdentifierSlot = null;
utilities.lazyLoad(exports, ["WebAppDomainOwnershipIdentifierSlot"], () => require("./webAppDomainOwnershipIdentifierSlot"));
exports.WebAppFtpAllowed = null;
utilities.lazyLoad(exports, ["WebAppFtpAllowed"], () => require("./webAppFtpAllowed"));
exports.WebAppFunction = null;
utilities.lazyLoad(exports, ["WebAppFunction"], () => require("./webAppFunction"));
exports.WebAppHostNameBinding = null;
utilities.lazyLoad(exports, ["WebAppHostNameBinding"], () => require("./webAppHostNameBinding"));
exports.WebAppHostNameBindingSlot = null;
utilities.lazyLoad(exports, ["WebAppHostNameBindingSlot"], () => require("./webAppHostNameBindingSlot"));
exports.WebAppHybridConnection = null;
utilities.lazyLoad(exports, ["WebAppHybridConnection"], () => require("./webAppHybridConnection"));
exports.WebAppHybridConnectionSlot = null;
utilities.lazyLoad(exports, ["WebAppHybridConnectionSlot"], () => require("./webAppHybridConnectionSlot"));
exports.WebAppInstanceFunctionSlot = null;
utilities.lazyLoad(exports, ["WebAppInstanceFunctionSlot"], () => require("./webAppInstanceFunctionSlot"));
exports.WebAppMetadata = null;
utilities.lazyLoad(exports, ["WebAppMetadata"], () => require("./webAppMetadata"));
exports.WebAppMetadataSlot = null;
utilities.lazyLoad(exports, ["WebAppMetadataSlot"], () => require("./webAppMetadataSlot"));
exports.WebAppPremierAddOn = null;
utilities.lazyLoad(exports, ["WebAppPremierAddOn"], () => require("./webAppPremierAddOn"));
exports.WebAppPremierAddOnSlot = null;
utilities.lazyLoad(exports, ["WebAppPremierAddOnSlot"], () => require("./webAppPremierAddOnSlot"));
exports.WebAppPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["WebAppPrivateEndpointConnection"], () => require("./webAppPrivateEndpointConnection"));
exports.WebAppPublicCertificate = null;
utilities.lazyLoad(exports, ["WebAppPublicCertificate"], () => require("./webAppPublicCertificate"));
exports.WebAppPublicCertificateSlot = null;
utilities.lazyLoad(exports, ["WebAppPublicCertificateSlot"], () => require("./webAppPublicCertificateSlot"));
exports.WebAppRelayServiceConnection = null;
utilities.lazyLoad(exports, ["WebAppRelayServiceConnection"], () => require("./webAppRelayServiceConnection"));
exports.WebAppRelayServiceConnectionSlot = null;
utilities.lazyLoad(exports, ["WebAppRelayServiceConnectionSlot"], () => require("./webAppRelayServiceConnectionSlot"));
exports.WebAppScmAllowed = null;
utilities.lazyLoad(exports, ["WebAppScmAllowed"], () => require("./webAppScmAllowed"));
exports.WebAppSiteExtension = null;
utilities.lazyLoad(exports, ["WebAppSiteExtension"], () => require("./webAppSiteExtension"));
exports.WebAppSiteExtensionSlot = null;
utilities.lazyLoad(exports, ["WebAppSiteExtensionSlot"], () => require("./webAppSiteExtensionSlot"));
exports.WebAppSitePushSettings = null;
utilities.lazyLoad(exports, ["WebAppSitePushSettings"], () => require("./webAppSitePushSettings"));
exports.WebAppSitePushSettingsSlot = null;
utilities.lazyLoad(exports, ["WebAppSitePushSettingsSlot"], () => require("./webAppSitePushSettingsSlot"));
exports.WebAppSlot = null;
utilities.lazyLoad(exports, ["WebAppSlot"], () => require("./webAppSlot"));
exports.WebAppSlotConfigurationNames = null;
utilities.lazyLoad(exports, ["WebAppSlotConfigurationNames"], () => require("./webAppSlotConfigurationNames"));
exports.WebAppSourceControl = null;
utilities.lazyLoad(exports, ["WebAppSourceControl"], () => require("./webAppSourceControl"));
exports.WebAppSourceControlSlot = null;
utilities.lazyLoad(exports, ["WebAppSourceControlSlot"], () => require("./webAppSourceControlSlot"));
exports.WebAppSwiftVirtualNetworkConnection = null;
utilities.lazyLoad(exports, ["WebAppSwiftVirtualNetworkConnection"], () => require("./webAppSwiftVirtualNetworkConnection"));
exports.WebAppSwiftVirtualNetworkConnectionSlot = null;
utilities.lazyLoad(exports, ["WebAppSwiftVirtualNetworkConnectionSlot"], () => require("./webAppSwiftVirtualNetworkConnectionSlot"));
exports.WebAppVnetConnection = null;
utilities.lazyLoad(exports, ["WebAppVnetConnection"], () => require("./webAppVnetConnection"));
exports.WebAppVnetConnectionSlot = null;
utilities.lazyLoad(exports, ["WebAppVnetConnectionSlot"], () => require("./webAppVnetConnectionSlot"));
// Export enums:
__exportStar(require("../../types/enums/web/v20201001"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20201001:AppServiceEnvironment":
                return new exports.AppServiceEnvironment(name, undefined, { urn });
            case "azure-native:web/v20201001:AppServicePlan":
                return new exports.AppServicePlan(name, undefined, { urn });
            case "azure-native:web/v20201001:AppServicePlanRouteForVnet":
                return new exports.AppServicePlanRouteForVnet(name, undefined, { urn });
            case "azure-native:web/v20201001:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:web/v20201001:StaticSite":
                return new exports.StaticSite(name, undefined, { urn });
            case "azure-native:web/v20201001:WebApp":
                return new exports.WebApp(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppApplicationSettings":
                return new exports.WebAppApplicationSettings(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppApplicationSettingsSlot":
                return new exports.WebAppApplicationSettingsSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAuthSettings":
                return new exports.WebAppAuthSettings(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAuthSettingsSlot":
                return new exports.WebAppAuthSettingsSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAuthSettingsV2":
                return new exports.WebAppAuthSettingsV2(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAuthSettingsV2Slot":
                return new exports.WebAppAuthSettingsV2Slot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAzureStorageAccounts":
                return new exports.WebAppAzureStorageAccounts(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppAzureStorageAccountsSlot":
                return new exports.WebAppAzureStorageAccountsSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppBackupConfiguration":
                return new exports.WebAppBackupConfiguration(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppBackupConfigurationSlot":
                return new exports.WebAppBackupConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppConnectionStrings":
                return new exports.WebAppConnectionStrings(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppConnectionStringsSlot":
                return new exports.WebAppConnectionStringsSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDeployment":
                return new exports.WebAppDeployment(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDeploymentSlot":
                return new exports.WebAppDeploymentSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDiagnosticLogsConfiguration":
                return new exports.WebAppDiagnosticLogsConfiguration(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDiagnosticLogsConfigurationSlot":
                return new exports.WebAppDiagnosticLogsConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDomainOwnershipIdentifier":
                return new exports.WebAppDomainOwnershipIdentifier(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppDomainOwnershipIdentifierSlot":
                return new exports.WebAppDomainOwnershipIdentifierSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppFtpAllowed":
                return new exports.WebAppFtpAllowed(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppFunction":
                return new exports.WebAppFunction(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppHostNameBinding":
                return new exports.WebAppHostNameBinding(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppHostNameBindingSlot":
                return new exports.WebAppHostNameBindingSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppHybridConnection":
                return new exports.WebAppHybridConnection(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppHybridConnectionSlot":
                return new exports.WebAppHybridConnectionSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppInstanceFunctionSlot":
                return new exports.WebAppInstanceFunctionSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppMetadata":
                return new exports.WebAppMetadata(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppMetadataSlot":
                return new exports.WebAppMetadataSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppPremierAddOn":
                return new exports.WebAppPremierAddOn(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppPremierAddOnSlot":
                return new exports.WebAppPremierAddOnSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppPrivateEndpointConnection":
                return new exports.WebAppPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppPublicCertificate":
                return new exports.WebAppPublicCertificate(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppPublicCertificateSlot":
                return new exports.WebAppPublicCertificateSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppRelayServiceConnection":
                return new exports.WebAppRelayServiceConnection(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppRelayServiceConnectionSlot":
                return new exports.WebAppRelayServiceConnectionSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppScmAllowed":
                return new exports.WebAppScmAllowed(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSiteExtension":
                return new exports.WebAppSiteExtension(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSiteExtensionSlot":
                return new exports.WebAppSiteExtensionSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSitePushSettings":
                return new exports.WebAppSitePushSettings(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSitePushSettingsSlot":
                return new exports.WebAppSitePushSettingsSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSlot":
                return new exports.WebAppSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSlotConfigurationNames":
                return new exports.WebAppSlotConfigurationNames(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSourceControl":
                return new exports.WebAppSourceControl(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSourceControlSlot":
                return new exports.WebAppSourceControlSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSwiftVirtualNetworkConnection":
                return new exports.WebAppSwiftVirtualNetworkConnection(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppSwiftVirtualNetworkConnectionSlot":
                return new exports.WebAppSwiftVirtualNetworkConnectionSlot(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppVnetConnection":
                return new exports.WebAppVnetConnection(name, undefined, { urn });
            case "azure-native:web/v20201001:WebAppVnetConnectionSlot":
                return new exports.WebAppVnetConnectionSlot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20201001", _module);
