// @pulumi/azure-native/securityinsights/v20230901preview/gcpdataConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GCPDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents Google Cloud Platform data connector.
 */
class GCPDataConnector extends pulumi.CustomResource {
    /**
     * Get an existing GCPDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GCPDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GCPDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GCPDataConnector.__pulumiType;
    }
    /**
     * Create a GCPDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.auth === undefined) && !opts.urn) {
                throw new Error("Missing required property 'auth'");
            }
            if ((!args || args.connectorDefinitionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectorDefinitionName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.request === undefined) && !opts.urn) {
                throw new Error("Missing required property 'request'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["auth"] = args ? args.auth : undefined;
            resourceInputs["connectorDefinitionName"] = args ? args.connectorDefinitionName : undefined;
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["dcrConfig"] = args ? args.dcrConfig : undefined;
            resourceInputs["kind"] = "GCP";
            resourceInputs["request"] = args ? args.request : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["auth"] = undefined /*out*/;
            resourceInputs["connectorDefinitionName"] = undefined /*out*/;
            resourceInputs["dcrConfig"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["request"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20200101:GCPDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20211001:GCPDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220501preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220601preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220701preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220801:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220801preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20220901preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20221001preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20221101:GCPDataConnector" }, { type: "azure-native:securityinsights/v20221101preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20221201preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230201:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230201preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230301preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230401preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230501preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230601preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230701preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20230801preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20231001preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20231101:GCPDataConnector" }, { type: "azure-native:securityinsights/v20231201preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20240101preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20240301:GCPDataConnector" }, { type: "azure-native:securityinsights/v20240401preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20240901:GCPDataConnector" }, { type: "azure-native:securityinsights/v20241001preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20250101preview:GCPDataConnector" }, { type: "azure-native:securityinsights/v20250301:GCPDataConnector" }, { type: "azure-native:securityinsights:GCPDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GCPDataConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GCPDataConnector = GCPDataConnector;
/** @internal */
GCPDataConnector.__pulumiType = 'azure-native:securityinsights/v20230901preview:GCPDataConnector';
