// @pulumi/azure-native/security/v20210701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityConnector = exports.getSecurityConnectorOutput = exports.getSecurityConnector = exports.getCustomEntityStoreAssignmentOutput = exports.getCustomEntityStoreAssignment = exports.getCustomAssessmentAutomationOutput = exports.getCustomAssessmentAutomation = exports.CustomEntityStoreAssignment = exports.CustomAssessmentAutomation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CustomAssessmentAutomation = null;
utilities.lazyLoad(exports, ["CustomAssessmentAutomation"], () => require("./customAssessmentAutomation"));
exports.CustomEntityStoreAssignment = null;
utilities.lazyLoad(exports, ["CustomEntityStoreAssignment"], () => require("./customEntityStoreAssignment"));
exports.getCustomAssessmentAutomation = null;
exports.getCustomAssessmentAutomationOutput = null;
utilities.lazyLoad(exports, ["getCustomAssessmentAutomation", "getCustomAssessmentAutomationOutput"], () => require("./getCustomAssessmentAutomation"));
exports.getCustomEntityStoreAssignment = null;
exports.getCustomEntityStoreAssignmentOutput = null;
utilities.lazyLoad(exports, ["getCustomEntityStoreAssignment", "getCustomEntityStoreAssignmentOutput"], () => require("./getCustomEntityStoreAssignment"));
exports.getSecurityConnector = null;
exports.getSecurityConnectorOutput = null;
utilities.lazyLoad(exports, ["getSecurityConnector", "getSecurityConnectorOutput"], () => require("./getSecurityConnector"));
exports.SecurityConnector = null;
utilities.lazyLoad(exports, ["SecurityConnector"], () => require("./securityConnector"));
// Export enums:
__exportStar(require("../../types/enums/security/v20210701preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20210701preview:CustomAssessmentAutomation":
                return new exports.CustomAssessmentAutomation(name, undefined, { urn });
            case "azure-native:security/v20210701preview:CustomEntityStoreAssignment":
                return new exports.CustomEntityStoreAssignment(name, undefined, { urn });
            case "azure-native:security/v20210701preview:SecurityConnector":
                return new exports.SecurityConnector(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20210701preview", _module);
