// @pulumi/azure-native/dashboard/v20221001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.ManagedPrivateEndpoint = exports.Grafana = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getManagedPrivateEndpointOutput = exports.getManagedPrivateEndpoint = exports.getGrafanaOutput = exports.getGrafana = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getGrafana = null;
exports.getGrafanaOutput = null;
utilities.lazyLoad(exports, ["getGrafana", "getGrafanaOutput"], () => require("./getGrafana"));
exports.getManagedPrivateEndpoint = null;
exports.getManagedPrivateEndpointOutput = null;
utilities.lazyLoad(exports, ["getManagedPrivateEndpoint", "getManagedPrivateEndpointOutput"], () => require("./getManagedPrivateEndpoint"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.Grafana = null;
utilities.lazyLoad(exports, ["Grafana"], () => require("./grafana"));
exports.ManagedPrivateEndpoint = null;
utilities.lazyLoad(exports, ["ManagedPrivateEndpoint"], () => require("./managedPrivateEndpoint"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../../types/enums/dashboard/v20221001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dashboard/v20221001preview:Grafana":
                return new exports.Grafana(name, undefined, { urn });
            case "azure-native:dashboard/v20221001preview:ManagedPrivateEndpoint":
                return new exports.ManagedPrivateEndpoint(name, undefined, { urn });
            case "azure-native:dashboard/v20221001preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dashboard/v20221001preview", _module);
