// @pulumi/azure-native/hdinsight/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250115preview = exports.v20240801preview = exports.v20240501preview = exports.v20231101preview = exports.v20230815preview = exports.v20230601preview = exports.v20230415preview = exports.v20210601 = exports.PrivateEndpointConnection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getClusterPoolOutput = exports.getClusterPool = exports.getClusterGatewaySettingsOutput = exports.getClusterGatewaySettings = exports.getClusterOutput = exports.getCluster = exports.getApplicationOutput = exports.getApplication = exports.ClusterPool = exports.Cluster = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterPool = null;
utilities.lazyLoad(exports, ["ClusterPool"], () => require("./clusterPool"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterGatewaySettings = null;
exports.getClusterGatewaySettingsOutput = null;
utilities.lazyLoad(exports, ["getClusterGatewaySettings", "getClusterGatewaySettingsOutput"], () => require("./getClusterGatewaySettings"));
exports.getClusterPool = null;
exports.getClusterPoolOutput = null;
utilities.lazyLoad(exports, ["getClusterPool", "getClusterPoolOutput"], () => require("./getClusterPool"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/hdinsight"), exports);
// Export sub-modules:
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20230415preview = require("./v20230415preview");
exports.v20230415preview = v20230415preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230815preview = require("./v20230815preview");
exports.v20230815preview = v20230815preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20250115preview = require("./v20250115preview");
exports.v20250115preview = v20250115preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hdinsight:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:hdinsight:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:hdinsight:ClusterPool":
                return new exports.ClusterPool(name, undefined, { urn });
            case "azure-native:hdinsight:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hdinsight", _module);
