// @pulumi/azure-native/types/enums/appconfiguration/v20240501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateLinkDelegation = exports.IdentityType = exports.CreateMode = exports.ConnectionStatus = exports.AuthenticationMode = void 0;
exports.AuthenticationMode = {
    /**
     * The local authentication mode. Users are not required to have data plane permissions if local authentication is not disabled.
     */
    Local: "Local",
    /**
     * The pass-through authentication mode. User identity will be passed through from Azure Resource Manager (ARM), requiring user to have data plane action permissions (Available via App Configuration Data Owner/ App Configuration Data Reader).
     */
    Pass_through: "Pass-through",
};
exports.ConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.CreateMode = {
    Recover: "Recover",
    Default: "Default",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.PrivateLinkDelegation = {
    /**
     * Azure Resource Manager (ARM) private endpoint is required if the resource requires private link.
     */
    Enabled: "Enabled",
    /**
     * Request is denied if the resource requires private link.
     */
    Disabled: "Disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
