// @pulumi/azure-native/desktopvirtualization/v20220909/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.ScalingPlanPooledSchedule = exports.ScalingPlan = exports.MSIXPackage = exports.HostPool = exports.getWorkspaceOutput = exports.getWorkspace = exports.getScalingPlanPooledScheduleOutput = exports.getScalingPlanPooledSchedule = exports.getScalingPlanOutput = exports.getScalingPlan = exports.getMSIXPackageOutput = exports.getMSIXPackage = exports.getHostPoolRegistrationTokenOutput = exports.getHostPoolRegistrationToken = exports.getHostPoolOutput = exports.getHostPool = exports.getApplicationGroupOutput = exports.getApplicationGroup = exports.getApplicationOutput = exports.getApplication = exports.ApplicationGroup = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationGroup = null;
utilities.lazyLoad(exports, ["ApplicationGroup"], () => require("./applicationGroup"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getApplicationGroup = null;
exports.getApplicationGroupOutput = null;
utilities.lazyLoad(exports, ["getApplicationGroup", "getApplicationGroupOutput"], () => require("./getApplicationGroup"));
exports.getHostPool = null;
exports.getHostPoolOutput = null;
utilities.lazyLoad(exports, ["getHostPool", "getHostPoolOutput"], () => require("./getHostPool"));
exports.getHostPoolRegistrationToken = null;
exports.getHostPoolRegistrationTokenOutput = null;
utilities.lazyLoad(exports, ["getHostPoolRegistrationToken", "getHostPoolRegistrationTokenOutput"], () => require("./getHostPoolRegistrationToken"));
exports.getMSIXPackage = null;
exports.getMSIXPackageOutput = null;
utilities.lazyLoad(exports, ["getMSIXPackage", "getMSIXPackageOutput"], () => require("./getMSIXPackage"));
exports.getScalingPlan = null;
exports.getScalingPlanOutput = null;
utilities.lazyLoad(exports, ["getScalingPlan", "getScalingPlanOutput"], () => require("./getScalingPlan"));
exports.getScalingPlanPooledSchedule = null;
exports.getScalingPlanPooledScheduleOutput = null;
utilities.lazyLoad(exports, ["getScalingPlanPooledSchedule", "getScalingPlanPooledScheduleOutput"], () => require("./getScalingPlanPooledSchedule"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.HostPool = null;
utilities.lazyLoad(exports, ["HostPool"], () => require("./hostPool"));
exports.MSIXPackage = null;
utilities.lazyLoad(exports, ["MSIXPackage"], () => require("./msixpackage"));
exports.ScalingPlan = null;
utilities.lazyLoad(exports, ["ScalingPlan"], () => require("./scalingPlan"));
exports.ScalingPlanPooledSchedule = null;
utilities.lazyLoad(exports, ["ScalingPlanPooledSchedule"], () => require("./scalingPlanPooledSchedule"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../../types/enums/desktopvirtualization/v20220909"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:desktopvirtualization/v20220909:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:ApplicationGroup":
                return new exports.ApplicationGroup(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:HostPool":
                return new exports.HostPool(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:MSIXPackage":
                return new exports.MSIXPackage(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:ScalingPlan":
                return new exports.ScalingPlan(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:ScalingPlanPooledSchedule":
                return new exports.ScalingPlanPooledSchedule(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220909:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "desktopvirtualization/v20220909", _module);
