// @pulumi/azure-native/automanage/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220504 = exports.v20200630preview = exports.getConfigurationProfilesVersionOutput = exports.getConfigurationProfilesVersion = exports.getConfigurationProfilePreferenceOutput = exports.getConfigurationProfilePreference = exports.getConfigurationProfileHCRPAssignmentOutput = exports.getConfigurationProfileHCRPAssignment = exports.getConfigurationProfileHCIAssignmentOutput = exports.getConfigurationProfileHCIAssignment = exports.getConfigurationProfileAssignmentOutput = exports.getConfigurationProfileAssignment = exports.getConfigurationProfileOutput = exports.getConfigurationProfile = exports.getAccountOutput = exports.getAccount = exports.ConfigurationProfilesVersion = exports.ConfigurationProfilePreference = exports.ConfigurationProfileHCRPAssignment = exports.ConfigurationProfileHCIAssignment = exports.ConfigurationProfileAssignment = exports.ConfigurationProfile = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.ConfigurationProfile = null;
utilities.lazyLoad(exports, ["ConfigurationProfile"], () => require("./configurationProfile"));
exports.ConfigurationProfileAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileAssignment"], () => require("./configurationProfileAssignment"));
exports.ConfigurationProfileHCIAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileHCIAssignment"], () => require("./configurationProfileHCIAssignment"));
exports.ConfigurationProfileHCRPAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationProfileHCRPAssignment"], () => require("./configurationProfileHCRPAssignment"));
exports.ConfigurationProfilePreference = null;
utilities.lazyLoad(exports, ["ConfigurationProfilePreference"], () => require("./configurationProfilePreference"));
exports.ConfigurationProfilesVersion = null;
utilities.lazyLoad(exports, ["ConfigurationProfilesVersion"], () => require("./configurationProfilesVersion"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getConfigurationProfile = null;
exports.getConfigurationProfileOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfile", "getConfigurationProfileOutput"], () => require("./getConfigurationProfile"));
exports.getConfigurationProfileAssignment = null;
exports.getConfigurationProfileAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileAssignment", "getConfigurationProfileAssignmentOutput"], () => require("./getConfigurationProfileAssignment"));
exports.getConfigurationProfileHCIAssignment = null;
exports.getConfigurationProfileHCIAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileHCIAssignment", "getConfigurationProfileHCIAssignmentOutput"], () => require("./getConfigurationProfileHCIAssignment"));
exports.getConfigurationProfileHCRPAssignment = null;
exports.getConfigurationProfileHCRPAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfileHCRPAssignment", "getConfigurationProfileHCRPAssignmentOutput"], () => require("./getConfigurationProfileHCRPAssignment"));
exports.getConfigurationProfilePreference = null;
exports.getConfigurationProfilePreferenceOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfilePreference", "getConfigurationProfilePreferenceOutput"], () => require("./getConfigurationProfilePreference"));
exports.getConfigurationProfilesVersion = null;
exports.getConfigurationProfilesVersionOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfilesVersion", "getConfigurationProfilesVersionOutput"], () => require("./getConfigurationProfilesVersion"));
// Export enums:
__exportStar(require("../types/enums/automanage"), exports);
// Export sub-modules:
const v20200630preview = require("./v20200630preview");
exports.v20200630preview = v20200630preview;
const v20220504 = require("./v20220504");
exports.v20220504 = v20220504;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:automanage:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfile":
                return new exports.ConfigurationProfile(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileAssignment":
                return new exports.ConfigurationProfileAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileHCIAssignment":
                return new exports.ConfigurationProfileHCIAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfileHCRPAssignment":
                return new exports.ConfigurationProfileHCRPAssignment(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfilePreference":
                return new exports.ConfigurationProfilePreference(name, undefined, { urn });
            case "azure-native:automanage:ConfigurationProfilesVersion":
                return new exports.ConfigurationProfilesVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "automanage", _module);
