// @pulumi/azure-native/durabletask/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241001preview = exports.TaskHub = exports.Scheduler = exports.getTaskHubOutput = exports.getTaskHub = exports.getSchedulerOutput = exports.getScheduler = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getScheduler = null;
exports.getSchedulerOutput = null;
utilities.lazyLoad(exports, ["getScheduler", "getSchedulerOutput"], () => require("./getScheduler"));
exports.getTaskHub = null;
exports.getTaskHubOutput = null;
utilities.lazyLoad(exports, ["getTaskHub", "getTaskHubOutput"], () => require("./getTaskHub"));
exports.Scheduler = null;
utilities.lazyLoad(exports, ["Scheduler"], () => require("./scheduler"));
exports.TaskHub = null;
utilities.lazyLoad(exports, ["TaskHub"], () => require("./taskHub"));
// Export sub-modules:
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:durabletask:Scheduler":
                return new exports.Scheduler(name, undefined, { urn });
            case "azure-native:durabletask:TaskHub":
                return new exports.TaskHub(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "durabletask", _module);
