// @pulumi/azure-native/documentdb/v20230315preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDBResourceMongoDBCollection = exports.MongoClusterFirewallRule = exports.MongoCluster = exports.listMongoClusterConnectionStringsOutput = exports.listMongoClusterConnectionStrings = exports.listDatabaseAccountKeysOutput = exports.listDatabaseAccountKeys = exports.listDatabaseAccountConnectionStringsOutput = exports.listDatabaseAccountConnectionStrings = exports.GremlinResourceGremlinGraph = exports.GremlinResourceGremlinDatabase = exports.GraphResourceGraph = exports.getTableResourceTableOutput = exports.getTableResourceTable = exports.getSqlResourceSqlUserDefinedFunctionOutput = exports.getSqlResourceSqlUserDefinedFunction = exports.getSqlResourceSqlTriggerOutput = exports.getSqlResourceSqlTrigger = exports.getSqlResourceSqlStoredProcedureOutput = exports.getSqlResourceSqlStoredProcedure = exports.getSqlResourceSqlDatabaseOutput = exports.getSqlResourceSqlDatabase = exports.getSqlResourceSqlContainerOutput = exports.getSqlResourceSqlContainer = exports.getMongoDBResourceMongoDBDatabaseOutput = exports.getMongoDBResourceMongoDBDatabase = exports.getMongoDBResourceMongoDBCollectionOutput = exports.getMongoDBResourceMongoDBCollection = exports.getMongoClusterFirewallRuleOutput = exports.getMongoClusterFirewallRule = exports.getMongoClusterOutput = exports.getMongoCluster = exports.getGremlinResourceGremlinGraphOutput = exports.getGremlinResourceGremlinGraph = exports.getGremlinResourceGremlinDatabaseOutput = exports.getGremlinResourceGremlinDatabase = exports.getGraphResourceGraphOutput = exports.getGraphResourceGraph = exports.getDatabaseAccountOutput = exports.getDatabaseAccount = exports.getCassandraResourceCassandraViewOutput = exports.getCassandraResourceCassandraView = exports.getCassandraResourceCassandraTableOutput = exports.getCassandraResourceCassandraTable = exports.getCassandraResourceCassandraKeyspaceOutput = exports.getCassandraResourceCassandraKeyspace = exports.DatabaseAccount = exports.CassandraResourceCassandraView = exports.CassandraResourceCassandraTable = exports.CassandraResourceCassandraKeyspace = void 0;
exports.TableResourceTable = exports.SqlResourceSqlUserDefinedFunction = exports.SqlResourceSqlTrigger = exports.SqlResourceSqlStoredProcedure = exports.SqlResourceSqlDatabase = exports.SqlResourceSqlContainer = exports.MongoDBResourceMongoDBDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CassandraResourceCassandraKeyspace = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraKeyspace"], () => require("./cassandraResourceCassandraKeyspace"));
exports.CassandraResourceCassandraTable = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraTable"], () => require("./cassandraResourceCassandraTable"));
exports.CassandraResourceCassandraView = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraView"], () => require("./cassandraResourceCassandraView"));
exports.DatabaseAccount = null;
utilities.lazyLoad(exports, ["DatabaseAccount"], () => require("./databaseAccount"));
exports.getCassandraResourceCassandraKeyspace = null;
exports.getCassandraResourceCassandraKeyspaceOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraKeyspace", "getCassandraResourceCassandraKeyspaceOutput"], () => require("./getCassandraResourceCassandraKeyspace"));
exports.getCassandraResourceCassandraTable = null;
exports.getCassandraResourceCassandraTableOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraTable", "getCassandraResourceCassandraTableOutput"], () => require("./getCassandraResourceCassandraTable"));
exports.getCassandraResourceCassandraView = null;
exports.getCassandraResourceCassandraViewOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraView", "getCassandraResourceCassandraViewOutput"], () => require("./getCassandraResourceCassandraView"));
exports.getDatabaseAccount = null;
exports.getDatabaseAccountOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccount", "getDatabaseAccountOutput"], () => require("./getDatabaseAccount"));
exports.getGraphResourceGraph = null;
exports.getGraphResourceGraphOutput = null;
utilities.lazyLoad(exports, ["getGraphResourceGraph", "getGraphResourceGraphOutput"], () => require("./getGraphResourceGraph"));
exports.getGremlinResourceGremlinDatabase = null;
exports.getGremlinResourceGremlinDatabaseOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinDatabase", "getGremlinResourceGremlinDatabaseOutput"], () => require("./getGremlinResourceGremlinDatabase"));
exports.getGremlinResourceGremlinGraph = null;
exports.getGremlinResourceGremlinGraphOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinGraph", "getGremlinResourceGremlinGraphOutput"], () => require("./getGremlinResourceGremlinGraph"));
exports.getMongoCluster = null;
exports.getMongoClusterOutput = null;
utilities.lazyLoad(exports, ["getMongoCluster", "getMongoClusterOutput"], () => require("./getMongoCluster"));
exports.getMongoClusterFirewallRule = null;
exports.getMongoClusterFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getMongoClusterFirewallRule", "getMongoClusterFirewallRuleOutput"], () => require("./getMongoClusterFirewallRule"));
exports.getMongoDBResourceMongoDBCollection = null;
exports.getMongoDBResourceMongoDBCollectionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBCollection", "getMongoDBResourceMongoDBCollectionOutput"], () => require("./getMongoDBResourceMongoDBCollection"));
exports.getMongoDBResourceMongoDBDatabase = null;
exports.getMongoDBResourceMongoDBDatabaseOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBDatabase", "getMongoDBResourceMongoDBDatabaseOutput"], () => require("./getMongoDBResourceMongoDBDatabase"));
exports.getSqlResourceSqlContainer = null;
exports.getSqlResourceSqlContainerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlContainer", "getSqlResourceSqlContainerOutput"], () => require("./getSqlResourceSqlContainer"));
exports.getSqlResourceSqlDatabase = null;
exports.getSqlResourceSqlDatabaseOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlDatabase", "getSqlResourceSqlDatabaseOutput"], () => require("./getSqlResourceSqlDatabase"));
exports.getSqlResourceSqlStoredProcedure = null;
exports.getSqlResourceSqlStoredProcedureOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlStoredProcedure", "getSqlResourceSqlStoredProcedureOutput"], () => require("./getSqlResourceSqlStoredProcedure"));
exports.getSqlResourceSqlTrigger = null;
exports.getSqlResourceSqlTriggerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlTrigger", "getSqlResourceSqlTriggerOutput"], () => require("./getSqlResourceSqlTrigger"));
exports.getSqlResourceSqlUserDefinedFunction = null;
exports.getSqlResourceSqlUserDefinedFunctionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlUserDefinedFunction", "getSqlResourceSqlUserDefinedFunctionOutput"], () => require("./getSqlResourceSqlUserDefinedFunction"));
exports.getTableResourceTable = null;
exports.getTableResourceTableOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTable", "getTableResourceTableOutput"], () => require("./getTableResourceTable"));
exports.GraphResourceGraph = null;
utilities.lazyLoad(exports, ["GraphResourceGraph"], () => require("./graphResourceGraph"));
exports.GremlinResourceGremlinDatabase = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinDatabase"], () => require("./gremlinResourceGremlinDatabase"));
exports.GremlinResourceGremlinGraph = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinGraph"], () => require("./gremlinResourceGremlinGraph"));
exports.listDatabaseAccountConnectionStrings = null;
exports.listDatabaseAccountConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountConnectionStrings", "listDatabaseAccountConnectionStringsOutput"], () => require("./listDatabaseAccountConnectionStrings"));
exports.listDatabaseAccountKeys = null;
exports.listDatabaseAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountKeys", "listDatabaseAccountKeysOutput"], () => require("./listDatabaseAccountKeys"));
exports.listMongoClusterConnectionStrings = null;
exports.listMongoClusterConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listMongoClusterConnectionStrings", "listMongoClusterConnectionStringsOutput"], () => require("./listMongoClusterConnectionStrings"));
exports.MongoCluster = null;
utilities.lazyLoad(exports, ["MongoCluster"], () => require("./mongoCluster"));
exports.MongoClusterFirewallRule = null;
utilities.lazyLoad(exports, ["MongoClusterFirewallRule"], () => require("./mongoClusterFirewallRule"));
exports.MongoDBResourceMongoDBCollection = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBCollection"], () => require("./mongoDBResourceMongoDBCollection"));
exports.MongoDBResourceMongoDBDatabase = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBDatabase"], () => require("./mongoDBResourceMongoDBDatabase"));
exports.SqlResourceSqlContainer = null;
utilities.lazyLoad(exports, ["SqlResourceSqlContainer"], () => require("./sqlResourceSqlContainer"));
exports.SqlResourceSqlDatabase = null;
utilities.lazyLoad(exports, ["SqlResourceSqlDatabase"], () => require("./sqlResourceSqlDatabase"));
exports.SqlResourceSqlStoredProcedure = null;
utilities.lazyLoad(exports, ["SqlResourceSqlStoredProcedure"], () => require("./sqlResourceSqlStoredProcedure"));
exports.SqlResourceSqlTrigger = null;
utilities.lazyLoad(exports, ["SqlResourceSqlTrigger"], () => require("./sqlResourceSqlTrigger"));
exports.SqlResourceSqlUserDefinedFunction = null;
utilities.lazyLoad(exports, ["SqlResourceSqlUserDefinedFunction"], () => require("./sqlResourceSqlUserDefinedFunction"));
exports.TableResourceTable = null;
utilities.lazyLoad(exports, ["TableResourceTable"], () => require("./tableResourceTable"));
// Export enums:
__exportStar(require("../../types/enums/documentdb/v20230315preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:documentdb/v20230315preview:CassandraResourceCassandraKeyspace":
                return new exports.CassandraResourceCassandraKeyspace(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:CassandraResourceCassandraTable":
                return new exports.CassandraResourceCassandraTable(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:CassandraResourceCassandraView":
                return new exports.CassandraResourceCassandraView(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:DatabaseAccount":
                return new exports.DatabaseAccount(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:GraphResourceGraph":
                return new exports.GraphResourceGraph(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:GremlinResourceGremlinDatabase":
                return new exports.GremlinResourceGremlinDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:GremlinResourceGremlinGraph":
                return new exports.GremlinResourceGremlinGraph(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:MongoCluster":
                return new exports.MongoCluster(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:MongoClusterFirewallRule":
                return new exports.MongoClusterFirewallRule(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:MongoDBResourceMongoDBCollection":
                return new exports.MongoDBResourceMongoDBCollection(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:MongoDBResourceMongoDBDatabase":
                return new exports.MongoDBResourceMongoDBDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:SqlResourceSqlContainer":
                return new exports.SqlResourceSqlContainer(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:SqlResourceSqlDatabase":
                return new exports.SqlResourceSqlDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:SqlResourceSqlStoredProcedure":
                return new exports.SqlResourceSqlStoredProcedure(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:SqlResourceSqlTrigger":
                return new exports.SqlResourceSqlTrigger(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:SqlResourceSqlUserDefinedFunction":
                return new exports.SqlResourceSqlUserDefinedFunction(name, undefined, { urn });
            case "azure-native:documentdb/v20230315preview:TableResourceTable":
                return new exports.TableResourceTable(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "documentdb/v20230315preview", _module);
