// @pulumi/gcp/composer/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserWorkloadsSecret = exports.UserWorkloadsConfigMap = exports.getUserWorkloadsSecretOutput = exports.getUserWorkloadsSecret = exports.getUserWorkloadsConfigMapOutput = exports.getUserWorkloadsConfigMap = exports.getImageVersionsOutput = exports.getImageVersions = exports.getEnvironmentOutput = exports.getEnvironment = exports.Environment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getImageVersions = null;
exports.getImageVersionsOutput = null;
utilities.lazyLoad(exports, ["getImageVersions", "getImageVersionsOutput"], () => require("./getImageVersions"));
exports.getUserWorkloadsConfigMap = null;
exports.getUserWorkloadsConfigMapOutput = null;
utilities.lazyLoad(exports, ["getUserWorkloadsConfigMap", "getUserWorkloadsConfigMapOutput"], () => require("./getUserWorkloadsConfigMap"));
exports.getUserWorkloadsSecret = null;
exports.getUserWorkloadsSecretOutput = null;
utilities.lazyLoad(exports, ["getUserWorkloadsSecret", "getUserWorkloadsSecretOutput"], () => require("./getUserWorkloadsSecret"));
exports.UserWorkloadsConfigMap = null;
utilities.lazyLoad(exports, ["UserWorkloadsConfigMap"], () => require("./userWorkloadsConfigMap"));
exports.UserWorkloadsSecret = null;
utilities.lazyLoad(exports, ["UserWorkloadsSecret"], () => require("./userWorkloadsSecret"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:composer/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "gcp:composer/userWorkloadsConfigMap:UserWorkloadsConfigMap":
                return new exports.UserWorkloadsConfigMap(name, undefined, { urn });
            case "gcp:composer/userWorkloadsSecret:UserWorkloadsSecret":
                return new exports.UserWorkloadsSecret(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "composer/environment", _module);
pulumi.runtime.registerResourceModule("gcp", "composer/userWorkloadsConfigMap", _module);
pulumi.runtime.registerResourceModule("gcp", "composer/userWorkloadsSecret", _module);
