// @pulumi/azure-native/types/enums/testbase/v20231101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tier = exports.TestType = exports.SystemAssignedServiceIdentityType = exports.RequestTypes = exports.PackageStudioTabs = exports.IntuneExtractStatus = exports.InteropExecutionMode = exports.ImageSource = exports.ImageSecurityType = exports.ImageOSState = exports.ImageArchitecture = exports.FileUploadResourceType = exports.Engagements = exports.DraftPackageSourceType = exports.CredentialType = exports.ContentType = exports.Architecture = exports.Action = void 0;
exports.Action = {
    Install: "Install",
    Launch: "Launch",
    Close: "Close",
    Uninstall: "Uninstall",
    Custom: "Custom",
    FlowDrivenCustom: "FlowDrivenCustom",
};
exports.Architecture = {
    X86: "x86",
    X64: "x64",
    Arm64: "arm64",
};
exports.ContentType = {
    Inline: "Inline",
    File: "File",
    Path: "Path",
};
exports.CredentialType = {
    /**
     * Username password credential for intune enrollment.
     */
    IntuneAccount: "IntuneAccount",
};
exports.DraftPackageSourceType = {
    Native: "Native",
    IntuneWin: "IntuneWin",
    TestBasePackage: "TestBasePackage",
    GalleryApp: "GalleryApp",
    IntuneEnrollment: "IntuneEnrollment",
};
exports.Engagements = {
    MVI: "MVI",
    MVP: "MVP",
    SUVP: "SUVP",
    MAPP: "MAPP",
    Other: "Other",
};
exports.FileUploadResourceType = {
    /**
     * Upload file for package onboarding.
     */
    Package: "Package",
    /**
     * Upload VHD file for image onboarding.
     */
    VHD: "VHD",
};
exports.ImageArchitecture = {
    /**
     * 64-bit architecture.
     */
    X64: "x64",
};
exports.ImageOSState = {
    /**
     * Sysprep generalization processed.
     */
    Generalized: "Generalized",
    /**
     * Fully kept with user specified settings.
     */
    Specialized: "Specialized",
};
exports.ImageSecurityType = {
    /**
     * Standard security type.
     */
    Standard: "Standard",
    /**
     * Specify higher security level compared to Standard.
     */
    TrustedLaunch: "TrustedLaunch",
};
exports.ImageSource = {
    /**
     * Unknown image source type.
     */
    Unknown: "Unknown",
    /**
     * Specify image onboarding through VHD.
     */
    VHD: "VHD",
};
exports.InteropExecutionMode = {
    /**
     * User application will test with the first party applications. For out-of-box tests, additional test cases for first party applications will also be run.
     */
    FirstPartyAppWithTests: "firstPartyAppWithTests",
    /**
     * User application will test with the first party applications.
     */
    FirstPartyApp: "firstPartyApp",
};
exports.IntuneExtractStatus = {
    Ready: "Ready",
    Uploading: "Uploading",
    UploadFailed: "UploadFailed",
    ExtractFailed: "ExtractFailed",
    NoDependencyApp: "NoDependencyApp",
};
exports.PackageStudioTabs = {
    Unspecified: "Unspecified",
    BasicsTab: "BasicsTab",
    ConfigureTestTab: "ConfigureTestTab",
    EditPackageTab: "EditPackageTab",
    TestMatrixTab: "TestMatrixTab",
    TagsTab: "TagsTab",
    ReviewAndCreateTab: "ReviewAndCreateTab",
};
exports.RequestTypes = {
    PreReleaseAccess: "PreReleaseAccess",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.TestType = {
    OutOfBoxTest: "OutOfBoxTest",
    FunctionalTest: "FunctionalTest",
    FlowDrivenTest: "FlowDrivenTest",
};
exports.Tier = {
    Standard: "Standard",
};
