// @pulumi/azure-native/web/v20160901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAppServicePlanHybridConnectionKeysOutput = exports.listAppServicePlanHybridConnectionKeys = exports.getAppServicePlanOutput = exports.getAppServicePlan = exports.AppServicePlanRouteForVnet = exports.AppServicePlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AppServicePlan = null;
utilities.lazyLoad(exports, ["AppServicePlan"], () => require("./appServicePlan"));
exports.AppServicePlanRouteForVnet = null;
utilities.lazyLoad(exports, ["AppServicePlanRouteForVnet"], () => require("./appServicePlanRouteForVnet"));
exports.getAppServicePlan = null;
exports.getAppServicePlanOutput = null;
utilities.lazyLoad(exports, ["getAppServicePlan", "getAppServicePlanOutput"], () => require("./getAppServicePlan"));
exports.listAppServicePlanHybridConnectionKeys = null;
exports.listAppServicePlanHybridConnectionKeysOutput = null;
utilities.lazyLoad(exports, ["listAppServicePlanHybridConnectionKeys", "listAppServicePlanHybridConnectionKeysOutput"], () => require("./listAppServicePlanHybridConnectionKeys"));
// Export enums:
__exportStar(require("../../types/enums/web/v20160901"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20160901:AppServicePlan":
                return new exports.AppServicePlan(name, undefined, { urn });
            case "azure-native:web/v20160901:AppServicePlanRouteForVnet":
                return new exports.AppServicePlanRouteForVnet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20160901", _module);
