// @pulumi/azure-native/network/v20231101/ddosProtectionPlan.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DdosProtectionPlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A DDoS protection plan in a resource group.
 */
class DdosProtectionPlan extends pulumi.CustomResource {
    /**
     * Get an existing DdosProtectionPlan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DdosProtectionPlan(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DdosProtectionPlan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DdosProtectionPlan.__pulumiType;
    }
    /**
     * Create a DdosProtectionPlan resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["ddosProtectionPlanName"] = args ? args.ddosProtectionPlanName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicIPAddresses"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworks"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicIPAddresses"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworks"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20180201:DdosProtectionPlan" }, { type: "azure-native:network/v20180401:DdosProtectionPlan" }, { type: "azure-native:network/v20180601:DdosProtectionPlan" }, { type: "azure-native:network/v20180701:DdosProtectionPlan" }, { type: "azure-native:network/v20180801:DdosProtectionPlan" }, { type: "azure-native:network/v20181001:DdosProtectionPlan" }, { type: "azure-native:network/v20181101:DdosProtectionPlan" }, { type: "azure-native:network/v20181201:DdosProtectionPlan" }, { type: "azure-native:network/v20190201:DdosProtectionPlan" }, { type: "azure-native:network/v20190401:DdosProtectionPlan" }, { type: "azure-native:network/v20190601:DdosProtectionPlan" }, { type: "azure-native:network/v20190701:DdosProtectionPlan" }, { type: "azure-native:network/v20190801:DdosProtectionPlan" }, { type: "azure-native:network/v20190901:DdosProtectionPlan" }, { type: "azure-native:network/v20191101:DdosProtectionPlan" }, { type: "azure-native:network/v20191201:DdosProtectionPlan" }, { type: "azure-native:network/v20200301:DdosProtectionPlan" }, { type: "azure-native:network/v20200401:DdosProtectionPlan" }, { type: "azure-native:network/v20200501:DdosProtectionPlan" }, { type: "azure-native:network/v20200601:DdosProtectionPlan" }, { type: "azure-native:network/v20200701:DdosProtectionPlan" }, { type: "azure-native:network/v20200801:DdosProtectionPlan" }, { type: "azure-native:network/v20201101:DdosProtectionPlan" }, { type: "azure-native:network/v20210201:DdosProtectionPlan" }, { type: "azure-native:network/v20210301:DdosProtectionPlan" }, { type: "azure-native:network/v20210501:DdosProtectionPlan" }, { type: "azure-native:network/v20210801:DdosProtectionPlan" }, { type: "azure-native:network/v20220101:DdosProtectionPlan" }, { type: "azure-native:network/v20220501:DdosProtectionPlan" }, { type: "azure-native:network/v20220701:DdosProtectionPlan" }, { type: "azure-native:network/v20220901:DdosProtectionPlan" }, { type: "azure-native:network/v20221101:DdosProtectionPlan" }, { type: "azure-native:network/v20230201:DdosProtectionPlan" }, { type: "azure-native:network/v20230401:DdosProtectionPlan" }, { type: "azure-native:network/v20230501:DdosProtectionPlan" }, { type: "azure-native:network/v20230601:DdosProtectionPlan" }, { type: "azure-native:network/v20230901:DdosProtectionPlan" }, { type: "azure-native:network/v20240101:DdosProtectionPlan" }, { type: "azure-native:network/v20240301:DdosProtectionPlan" }, { type: "azure-native:network/v20240501:DdosProtectionPlan" }, { type: "azure-native:network:DdosProtectionPlan" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DdosProtectionPlan.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DdosProtectionPlan = DdosProtectionPlan;
/** @internal */
DdosProtectionPlan.__pulumiType = 'azure-native:network/v20231101:DdosProtectionPlan';
