// @pulumi/azure-native/machinelearningservices/v20230601preview/codeContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * Azure Resource Manager resource envelope.
 */
class CodeContainer extends pulumi.CustomResource {
    /**
     * Get an existing CodeContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CodeContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CodeContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CodeContainer.__pulumiType;
    }
    /**
     * Create a CodeContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.codeContainerProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'codeContainerProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["codeContainerProperties"] = args ? (args.codeContainerProperties ? pulumi.output(args.codeContainerProperties).apply(inputs.machinelearningservices.v20230601preview.codeContainerArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["codeContainerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20220201preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20220501:CodeContainer" }, { type: "azure-native:machinelearningservices/v20220601preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20221001:CodeContainer" }, { type: "azure-native:machinelearningservices/v20221001preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20221201preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20230201preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20230401:CodeContainer" }, { type: "azure-native:machinelearningservices/v20230401preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20230801preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20231001:CodeContainer" }, { type: "azure-native:machinelearningservices/v20240101preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20240401:CodeContainer" }, { type: "azure-native:machinelearningservices/v20240401preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20240701preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20241001:CodeContainer" }, { type: "azure-native:machinelearningservices/v20241001preview:CodeContainer" }, { type: "azure-native:machinelearningservices/v20250101preview:CodeContainer" }, { type: "azure-native:machinelearningservices:CodeContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CodeContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CodeContainer = CodeContainer;
/** @internal */
CodeContainer.__pulumiType = 'azure-native:machinelearningservices/v20230601preview:CodeContainer';
