// @pulumi/azure-native/types/enums/devhub/v20221011preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowRunStatus = exports.ManifestType = exports.ManifestGenerationMode = exports.GenerationManifestType = exports.GenerationLanguage = exports.DockerfileGenerationMode = void 0;
exports.DockerfileGenerationMode = {
    /**
     * Dockerfiles will be generated
     */
    Enabled: "enabled",
    /**
     * Dockerfiles will not be generated
     */
    Disabled: "disabled",
};
exports.GenerationLanguage = {
    /**
     * clojure language
     */
    Clojure: "clojure",
    /**
     * csharp language
     */
    Csharp: "csharp",
    /**
     * erlang language
     */
    Erlang: "erlang",
    /**
     * go language
     */
    Go: "go",
    /**
     * gomodule language
     */
    Gomodule: "gomodule",
    /**
     * gradle language
     */
    Gradle: "gradle",
    /**
     * java language
     */
    Java: "java",
    /**
     * javascript language
     */
    Javascript: "javascript",
    /**
     * php language
     */
    Php: "php",
    /**
     * python language
     */
    Python: "python",
    /**
     * ruby language
     */
    Ruby: "ruby",
    /**
     * rust language
     */
    Rust: "rust",
    /**
     * swift language
     */
    Swift: "swift",
};
exports.GenerationManifestType = {
    /**
     * Helm manifests
     */
    Helm: "helm",
    /**
     * Kubernetes manifests
     */
    Kube: "kube",
};
exports.ManifestGenerationMode = {
    /**
     * Manifests will be generated
     */
    Enabled: "enabled",
    /**
     * Manifests will not be generated
     */
    Disabled: "disabled",
};
exports.ManifestType = {
    /**
     * Repositories using helm
     */
    Helm: "helm",
    /**
     * Repositories using kubernetes manifests
     */
    Kube: "kube",
};
exports.WorkflowRunStatus = {
    /**
     * Workflow run is queued
     */
    Queued: "queued",
    /**
     * Workflow run is inprogress
     */
    Inprogress: "inprogress",
    /**
     * Workflow run is completed
     */
    Completed: "completed",
};
