// @pulumi/azure-native/professionalservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230701preview = exports.ProfessionalServiceSubscriptionLevel = exports.getProfessionalServiceSubscriptionLevelOutput = exports.getProfessionalServiceSubscriptionLevel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getProfessionalServiceSubscriptionLevel = null;
exports.getProfessionalServiceSubscriptionLevelOutput = null;
utilities.lazyLoad(exports, ["getProfessionalServiceSubscriptionLevel", "getProfessionalServiceSubscriptionLevelOutput"], () => require("./getProfessionalServiceSubscriptionLevel"));
exports.ProfessionalServiceSubscriptionLevel = null;
utilities.lazyLoad(exports, ["ProfessionalServiceSubscriptionLevel"], () => require("./professionalServiceSubscriptionLevel"));
// Export sub-modules:
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:professionalservice:ProfessionalServiceSubscriptionLevel":
                return new exports.ProfessionalServiceSubscriptionLevel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "professionalservice", _module);
