// @pulumi/azure-native/workloads/v20211201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WordpressInstance = exports.SAPApplicationServerInstance = exports.PhpWorkload = exports.getWordpressInstanceOutput = exports.getWordpressInstance = exports.getSAPSupportedSkuOutput = exports.getSAPSupportedSku = exports.getSAPSizingRecommendationsOutput = exports.getSAPSizingRecommendations = exports.getSAPDiskConfigurationsOutput = exports.getSAPDiskConfigurations = exports.getSAPAvailabilityZoneDetailsOutput = exports.getSAPAvailabilityZoneDetails = exports.getSAPApplicationServerInstanceOutput = exports.getSAPApplicationServerInstance = exports.getPhpWorkloadOutput = exports.getPhpWorkload = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPhpWorkload = null;
exports.getPhpWorkloadOutput = null;
utilities.lazyLoad(exports, ["getPhpWorkload", "getPhpWorkloadOutput"], () => require("./getPhpWorkload"));
exports.getSAPApplicationServerInstance = null;
exports.getSAPApplicationServerInstanceOutput = null;
utilities.lazyLoad(exports, ["getSAPApplicationServerInstance", "getSAPApplicationServerInstanceOutput"], () => require("./getSAPApplicationServerInstance"));
exports.getSAPAvailabilityZoneDetails = null;
exports.getSAPAvailabilityZoneDetailsOutput = null;
utilities.lazyLoad(exports, ["getSAPAvailabilityZoneDetails", "getSAPAvailabilityZoneDetailsOutput"], () => require("./getSAPAvailabilityZoneDetails"));
exports.getSAPDiskConfigurations = null;
exports.getSAPDiskConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getSAPDiskConfigurations", "getSAPDiskConfigurationsOutput"], () => require("./getSAPDiskConfigurations"));
exports.getSAPSizingRecommendations = null;
exports.getSAPSizingRecommendationsOutput = null;
utilities.lazyLoad(exports, ["getSAPSizingRecommendations", "getSAPSizingRecommendationsOutput"], () => require("./getSAPSizingRecommendations"));
exports.getSAPSupportedSku = null;
exports.getSAPSupportedSkuOutput = null;
utilities.lazyLoad(exports, ["getSAPSupportedSku", "getSAPSupportedSkuOutput"], () => require("./getSAPSupportedSku"));
exports.getWordpressInstance = null;
exports.getWordpressInstanceOutput = null;
utilities.lazyLoad(exports, ["getWordpressInstance", "getWordpressInstanceOutput"], () => require("./getWordpressInstance"));
exports.PhpWorkload = null;
utilities.lazyLoad(exports, ["PhpWorkload"], () => require("./phpWorkload"));
exports.SAPApplicationServerInstance = null;
utilities.lazyLoad(exports, ["SAPApplicationServerInstance"], () => require("./sapapplicationServerInstance"));
exports.WordpressInstance = null;
utilities.lazyLoad(exports, ["WordpressInstance"], () => require("./wordpressInstance"));
// Export enums:
__exportStar(require("../../types/enums/workloads/v20211201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:workloads/v20211201preview:PhpWorkload":
                return new exports.PhpWorkload(name, undefined, { urn });
            case "azure-native:workloads/v20211201preview:SAPApplicationServerInstance":
                return new exports.SAPApplicationServerInstance(name, undefined, { urn });
            case "azure-native:workloads/v20211201preview:WordpressInstance":
                return new exports.WordpressInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "workloads/v20211201preview", _module);
