// @pulumi/azure-native/types/enums/hdinsight/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tier = exports.ResourceProviderConnection = exports.ResourceIdentityType = exports.PrivateLinkServiceConnectionStatus = exports.PrivateLink = exports.PrivateIPAllocationMethod = exports.OSType = exports.JsonWebKeyEncryptionAlgorithm = exports.DirectoryType = exports.DaysOfWeek = exports.v20250115preview = exports.v20240801preview = exports.v20240501preview = exports.v20231101preview = exports.v20230815preview = exports.v20230601preview = exports.v20230415preview = exports.v20210601 = void 0;
// Export sub-modules:
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20230415preview = require("./v20230415preview");
exports.v20230415preview = v20230415preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230815preview = require("./v20230815preview");
exports.v20230815preview = v20230815preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20250115preview = require("./v20250115preview");
exports.v20250115preview = v20250115preview;
exports.DaysOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DirectoryType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.JsonWebKeyEncryptionAlgorithm = {
    RSA_OAEP: "RSA-OAEP",
    RSA_OAEP_256: "RSA-OAEP-256",
    RSA1_5: "RSA1_5",
};
exports.OSType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.PrivateIPAllocationMethod = {
    Dynamic: "dynamic",
    Static: "static",
};
exports.PrivateLink = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.PrivateLinkServiceConnectionStatus = {
    Approved: "Approved",
    Rejected: "Rejected",
    Pending: "Pending",
    Removed: "Removed",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ResourceProviderConnection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.Tier = {
    Standard: "Standard",
    Premium: "Premium",
};
