// @pulumi/azure-native/videoanalyzer/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211101preview = exports.v20210501preview = exports.VideoAnalyzer = exports.Video = exports.PrivateEndpointConnection = exports.PipelineTopology = exports.PipelineJob = exports.LivePipeline = exports.listVideoContentTokenOutput = exports.listVideoContentToken = exports.listEdgeModuleProvisioningTokenOutput = exports.listEdgeModuleProvisioningToken = exports.getVideoAnalyzerOutput = exports.getVideoAnalyzer = exports.getVideoOutput = exports.getVideo = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getPipelineTopologyOutput = exports.getPipelineTopology = exports.getPipelineJobOutput = exports.getPipelineJob = exports.getLivePipelineOutput = exports.getLivePipeline = exports.getEdgeModuleOutput = exports.getEdgeModule = exports.getAccessPolicyOutput = exports.getAccessPolicy = exports.EdgeModule = exports.AccessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessPolicy = null;
utilities.lazyLoad(exports, ["AccessPolicy"], () => require("./accessPolicy"));
exports.EdgeModule = null;
utilities.lazyLoad(exports, ["EdgeModule"], () => require("./edgeModule"));
exports.getAccessPolicy = null;
exports.getAccessPolicyOutput = null;
utilities.lazyLoad(exports, ["getAccessPolicy", "getAccessPolicyOutput"], () => require("./getAccessPolicy"));
exports.getEdgeModule = null;
exports.getEdgeModuleOutput = null;
utilities.lazyLoad(exports, ["getEdgeModule", "getEdgeModuleOutput"], () => require("./getEdgeModule"));
exports.getLivePipeline = null;
exports.getLivePipelineOutput = null;
utilities.lazyLoad(exports, ["getLivePipeline", "getLivePipelineOutput"], () => require("./getLivePipeline"));
exports.getPipelineJob = null;
exports.getPipelineJobOutput = null;
utilities.lazyLoad(exports, ["getPipelineJob", "getPipelineJobOutput"], () => require("./getPipelineJob"));
exports.getPipelineTopology = null;
exports.getPipelineTopologyOutput = null;
utilities.lazyLoad(exports, ["getPipelineTopology", "getPipelineTopologyOutput"], () => require("./getPipelineTopology"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getVideo = null;
exports.getVideoOutput = null;
utilities.lazyLoad(exports, ["getVideo", "getVideoOutput"], () => require("./getVideo"));
exports.getVideoAnalyzer = null;
exports.getVideoAnalyzerOutput = null;
utilities.lazyLoad(exports, ["getVideoAnalyzer", "getVideoAnalyzerOutput"], () => require("./getVideoAnalyzer"));
exports.listEdgeModuleProvisioningToken = null;
exports.listEdgeModuleProvisioningTokenOutput = null;
utilities.lazyLoad(exports, ["listEdgeModuleProvisioningToken", "listEdgeModuleProvisioningTokenOutput"], () => require("./listEdgeModuleProvisioningToken"));
exports.listVideoContentToken = null;
exports.listVideoContentTokenOutput = null;
utilities.lazyLoad(exports, ["listVideoContentToken", "listVideoContentTokenOutput"], () => require("./listVideoContentToken"));
exports.LivePipeline = null;
utilities.lazyLoad(exports, ["LivePipeline"], () => require("./livePipeline"));
exports.PipelineJob = null;
utilities.lazyLoad(exports, ["PipelineJob"], () => require("./pipelineJob"));
exports.PipelineTopology = null;
utilities.lazyLoad(exports, ["PipelineTopology"], () => require("./pipelineTopology"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Video = null;
utilities.lazyLoad(exports, ["Video"], () => require("./video"));
exports.VideoAnalyzer = null;
utilities.lazyLoad(exports, ["VideoAnalyzer"], () => require("./videoAnalyzer"));
// Export enums:
__exportStar(require("../types/enums/videoanalyzer"), exports);
// Export sub-modules:
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:videoanalyzer:AccessPolicy":
                return new exports.AccessPolicy(name, undefined, { urn });
            case "azure-native:videoanalyzer:EdgeModule":
                return new exports.EdgeModule(name, undefined, { urn });
            case "azure-native:videoanalyzer:LivePipeline":
                return new exports.LivePipeline(name, undefined, { urn });
            case "azure-native:videoanalyzer:PipelineJob":
                return new exports.PipelineJob(name, undefined, { urn });
            case "azure-native:videoanalyzer:PipelineTopology":
                return new exports.PipelineTopology(name, undefined, { urn });
            case "azure-native:videoanalyzer:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:videoanalyzer:Video":
                return new exports.Video(name, undefined, { urn });
            case "azure-native:videoanalyzer:VideoAnalyzer":
                return new exports.VideoAnalyzer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "videoanalyzer", _module);
