// @pulumi/azure-native/storage/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listStorageAccountServiceSAS = exports.listStorageAccountSASOutput = exports.listStorageAccountSAS = exports.listStorageAccountKeysOutput = exports.listStorageAccountKeys = exports.listLocalUserKeysOutput = exports.listLocalUserKeys = exports.getTableServicePropertiesOutput = exports.getTableServiceProperties = exports.getTableOutput = exports.getTable = exports.getStorageTaskAssignmentOutput = exports.getStorageTaskAssignment = exports.getStorageAccountOutput = exports.getStorageAccount = exports.getQueueServicePropertiesOutput = exports.getQueueServiceProperties = exports.getQueueOutput = exports.getQueue = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getObjectReplicationPolicyOutput = exports.getObjectReplicationPolicy = exports.getManagementPolicyOutput = exports.getManagementPolicy = exports.getLocalUserOutput = exports.getLocalUser = exports.getFileShareOutput = exports.getFileShare = exports.getFileServicePropertiesOutput = exports.getFileServiceProperties = exports.getEncryptionScopeOutput = exports.getEncryptionScope = exports.getBlobServicePropertiesOutput = exports.getBlobServiceProperties = exports.getBlobInventoryPolicyOutput = exports.getBlobInventoryPolicy = exports.getBlobContainerImmutabilityPolicyOutput = exports.getBlobContainerImmutabilityPolicy = exports.getBlobContainerOutput = exports.getBlobContainer = exports.FileShare = exports.FileServiceProperties = exports.EncryptionScope = exports.BlobServiceProperties = exports.BlobInventoryPolicy = exports.BlobContainerLegalHold = exports.BlobContainerImmutabilityPolicy = exports.BlobContainer = exports.Blob = void 0;
exports.v20240101 = exports.v20230501 = exports.v20230401 = exports.v20230101 = exports.v20220901 = exports.TableServiceProperties = exports.Table = exports.StorageTaskAssignment = exports.StorageAccountStaticWebsite = exports.StorageAccount = exports.QueueServiceProperties = exports.Queue = exports.PrivateEndpointConnection = exports.ObjectReplicationPolicy = exports.ManagementPolicy = exports.LocalUser = exports.listStorageAccountServiceSASOutput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Blob = null;
utilities.lazyLoad(exports, ["Blob"], () => require("./blob"));
exports.BlobContainer = null;
utilities.lazyLoad(exports, ["BlobContainer"], () => require("./blobContainer"));
exports.BlobContainerImmutabilityPolicy = null;
utilities.lazyLoad(exports, ["BlobContainerImmutabilityPolicy"], () => require("./blobContainerImmutabilityPolicy"));
exports.BlobContainerLegalHold = null;
utilities.lazyLoad(exports, ["BlobContainerLegalHold"], () => require("./blobContainerLegalHold"));
exports.BlobInventoryPolicy = null;
utilities.lazyLoad(exports, ["BlobInventoryPolicy"], () => require("./blobInventoryPolicy"));
exports.BlobServiceProperties = null;
utilities.lazyLoad(exports, ["BlobServiceProperties"], () => require("./blobServiceProperties"));
exports.EncryptionScope = null;
utilities.lazyLoad(exports, ["EncryptionScope"], () => require("./encryptionScope"));
exports.FileServiceProperties = null;
utilities.lazyLoad(exports, ["FileServiceProperties"], () => require("./fileServiceProperties"));
exports.FileShare = null;
utilities.lazyLoad(exports, ["FileShare"], () => require("./fileShare"));
exports.getBlobContainer = null;
exports.getBlobContainerOutput = null;
utilities.lazyLoad(exports, ["getBlobContainer", "getBlobContainerOutput"], () => require("./getBlobContainer"));
exports.getBlobContainerImmutabilityPolicy = null;
exports.getBlobContainerImmutabilityPolicyOutput = null;
utilities.lazyLoad(exports, ["getBlobContainerImmutabilityPolicy", "getBlobContainerImmutabilityPolicyOutput"], () => require("./getBlobContainerImmutabilityPolicy"));
exports.getBlobInventoryPolicy = null;
exports.getBlobInventoryPolicyOutput = null;
utilities.lazyLoad(exports, ["getBlobInventoryPolicy", "getBlobInventoryPolicyOutput"], () => require("./getBlobInventoryPolicy"));
exports.getBlobServiceProperties = null;
exports.getBlobServicePropertiesOutput = null;
utilities.lazyLoad(exports, ["getBlobServiceProperties", "getBlobServicePropertiesOutput"], () => require("./getBlobServiceProperties"));
exports.getEncryptionScope = null;
exports.getEncryptionScopeOutput = null;
utilities.lazyLoad(exports, ["getEncryptionScope", "getEncryptionScopeOutput"], () => require("./getEncryptionScope"));
exports.getFileServiceProperties = null;
exports.getFileServicePropertiesOutput = null;
utilities.lazyLoad(exports, ["getFileServiceProperties", "getFileServicePropertiesOutput"], () => require("./getFileServiceProperties"));
exports.getFileShare = null;
exports.getFileShareOutput = null;
utilities.lazyLoad(exports, ["getFileShare", "getFileShareOutput"], () => require("./getFileShare"));
exports.getLocalUser = null;
exports.getLocalUserOutput = null;
utilities.lazyLoad(exports, ["getLocalUser", "getLocalUserOutput"], () => require("./getLocalUser"));
exports.getManagementPolicy = null;
exports.getManagementPolicyOutput = null;
utilities.lazyLoad(exports, ["getManagementPolicy", "getManagementPolicyOutput"], () => require("./getManagementPolicy"));
exports.getObjectReplicationPolicy = null;
exports.getObjectReplicationPolicyOutput = null;
utilities.lazyLoad(exports, ["getObjectReplicationPolicy", "getObjectReplicationPolicyOutput"], () => require("./getObjectReplicationPolicy"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getQueue = null;
exports.getQueueOutput = null;
utilities.lazyLoad(exports, ["getQueue", "getQueueOutput"], () => require("./getQueue"));
exports.getQueueServiceProperties = null;
exports.getQueueServicePropertiesOutput = null;
utilities.lazyLoad(exports, ["getQueueServiceProperties", "getQueueServicePropertiesOutput"], () => require("./getQueueServiceProperties"));
exports.getStorageAccount = null;
exports.getStorageAccountOutput = null;
utilities.lazyLoad(exports, ["getStorageAccount", "getStorageAccountOutput"], () => require("./getStorageAccount"));
exports.getStorageTaskAssignment = null;
exports.getStorageTaskAssignmentOutput = null;
utilities.lazyLoad(exports, ["getStorageTaskAssignment", "getStorageTaskAssignmentOutput"], () => require("./getStorageTaskAssignment"));
exports.getTable = null;
exports.getTableOutput = null;
utilities.lazyLoad(exports, ["getTable", "getTableOutput"], () => require("./getTable"));
exports.getTableServiceProperties = null;
exports.getTableServicePropertiesOutput = null;
utilities.lazyLoad(exports, ["getTableServiceProperties", "getTableServicePropertiesOutput"], () => require("./getTableServiceProperties"));
exports.listLocalUserKeys = null;
exports.listLocalUserKeysOutput = null;
utilities.lazyLoad(exports, ["listLocalUserKeys", "listLocalUserKeysOutput"], () => require("./listLocalUserKeys"));
exports.listStorageAccountKeys = null;
exports.listStorageAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listStorageAccountKeys", "listStorageAccountKeysOutput"], () => require("./listStorageAccountKeys"));
exports.listStorageAccountSAS = null;
exports.listStorageAccountSASOutput = null;
utilities.lazyLoad(exports, ["listStorageAccountSAS", "listStorageAccountSASOutput"], () => require("./listStorageAccountSAS"));
exports.listStorageAccountServiceSAS = null;
exports.listStorageAccountServiceSASOutput = null;
utilities.lazyLoad(exports, ["listStorageAccountServiceSAS", "listStorageAccountServiceSASOutput"], () => require("./listStorageAccountServiceSAS"));
exports.LocalUser = null;
utilities.lazyLoad(exports, ["LocalUser"], () => require("./localUser"));
exports.ManagementPolicy = null;
utilities.lazyLoad(exports, ["ManagementPolicy"], () => require("./managementPolicy"));
exports.ObjectReplicationPolicy = null;
utilities.lazyLoad(exports, ["ObjectReplicationPolicy"], () => require("./objectReplicationPolicy"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueServiceProperties = null;
utilities.lazyLoad(exports, ["QueueServiceProperties"], () => require("./queueServiceProperties"));
exports.StorageAccount = null;
utilities.lazyLoad(exports, ["StorageAccount"], () => require("./storageAccount"));
exports.StorageAccountStaticWebsite = null;
utilities.lazyLoad(exports, ["StorageAccountStaticWebsite"], () => require("./storageAccountStaticWebsite"));
exports.StorageTaskAssignment = null;
utilities.lazyLoad(exports, ["StorageTaskAssignment"], () => require("./storageTaskAssignment"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.TableServiceProperties = null;
utilities.lazyLoad(exports, ["TableServiceProperties"], () => require("./tableServiceProperties"));
// Export enums:
__exportStar(require("../types/enums/storage"), exports);
// Export sub-modules:
const v20220901 = require("./v20220901");
exports.v20220901 = v20220901;
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storage:Blob":
                return new exports.Blob(name, undefined, { urn });
            case "azure-native:storage:BlobContainer":
                return new exports.BlobContainer(name, undefined, { urn });
            case "azure-native:storage:BlobContainerImmutabilityPolicy":
                return new exports.BlobContainerImmutabilityPolicy(name, undefined, { urn });
            case "azure-native:storage:BlobContainerLegalHold":
                return new exports.BlobContainerLegalHold(name, undefined, { urn });
            case "azure-native:storage:BlobInventoryPolicy":
                return new exports.BlobInventoryPolicy(name, undefined, { urn });
            case "azure-native:storage:BlobServiceProperties":
                return new exports.BlobServiceProperties(name, undefined, { urn });
            case "azure-native:storage:EncryptionScope":
                return new exports.EncryptionScope(name, undefined, { urn });
            case "azure-native:storage:FileServiceProperties":
                return new exports.FileServiceProperties(name, undefined, { urn });
            case "azure-native:storage:FileShare":
                return new exports.FileShare(name, undefined, { urn });
            case "azure-native:storage:LocalUser":
                return new exports.LocalUser(name, undefined, { urn });
            case "azure-native:storage:ManagementPolicy":
                return new exports.ManagementPolicy(name, undefined, { urn });
            case "azure-native:storage:ObjectReplicationPolicy":
                return new exports.ObjectReplicationPolicy(name, undefined, { urn });
            case "azure-native:storage:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:storage:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "azure-native:storage:QueueServiceProperties":
                return new exports.QueueServiceProperties(name, undefined, { urn });
            case "azure-native:storage:StorageAccount":
                return new exports.StorageAccount(name, undefined, { urn });
            case "azure-native:storage:StorageAccountStaticWebsite":
                return new exports.StorageAccountStaticWebsite(name, undefined, { urn });
            case "azure-native:storage:StorageTaskAssignment":
                return new exports.StorageTaskAssignment(name, undefined, { urn });
            case "azure-native:storage:Table":
                return new exports.Table(name, undefined, { urn });
            case "azure-native:storage:TableServiceProperties":
                return new exports.TableServiceProperties(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storage", _module);
