// @pulumi/azure-native/types/enums/security/v20240101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingTier = exports.IsEnabled = exports.Enforce = void 0;
exports.Enforce = {
    /**
     * Allows the descendants of this scope to override the pricing configuration set on this scope (allows setting inherited="False")
     */
    False: "False",
    /**
     * Prevents overrides and forces the current scope's pricing configuration to all descendants
     */
    True: "True",
};
exports.IsEnabled = {
    /**
     * Indicates the extension is enabled
     */
    True: "True",
    /**
     * Indicates the extension is disabled
     */
    False: "False",
};
exports.PricingTier = {
    /**
     * Get free Microsoft Defender for Cloud experience with basic security features
     */
    Free: "Free",
    /**
     * Get the standard Microsoft Defender for Cloud experience with advanced security features
     */
    Standard: "Standard",
};
