// @pulumi/azure-native/types/enums/dbformysql/v20230630/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerVersion = exports.ServerSkuTier = exports.ReplicationRole = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.ImportSourceStorageType = exports.HighAvailabilityMode = exports.EnableStatusEnum = exports.DataEncryptionType = exports.CreateMode = exports.ConfigurationSource = exports.AdministratorType = void 0;
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.ConfigurationSource = {
    System_default: "system-default",
    User_override: "user-override",
};
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    Replica: "Replica",
    GeoRestore: "GeoRestore",
};
exports.DataEncryptionType = {
    AzureKeyVault: "AzureKeyVault",
    SystemManaged: "SystemManaged",
};
exports.EnableStatusEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
    SameZone: "SameZone",
};
exports.ImportSourceStorageType = {
    AzureBlob: "AzureBlob",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ReplicationRole = {
    None: "None",
    Source: "Source",
    Replica: "Replica",
};
exports.ServerSkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.ServerVersion = {
    ServerVersion_5_7: "5.7",
    ServerVersion_8_0_21: "8.0.21",
};
