// @pulumi/azure-native/desktopvirtualization/scalingPlanPersonalSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScalingPlanPersonalSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a ScalingPlanPersonalSchedule definition.
 *
 * Uses Azure REST API version 2024-11-01-preview.
 *
 * Other available API versions: 2023-09-05, 2023-10-04-preview, 2023-11-01-preview, 2024-01-16-preview, 2024-03-06-preview, 2024-04-03, 2024-04-08-preview, 2024-08-08-preview.
 */
class ScalingPlanPersonalSchedule extends pulumi.CustomResource {
    /**
     * Get an existing ScalingPlanPersonalSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScalingPlanPersonalSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScalingPlanPersonalSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScalingPlanPersonalSchedule.__pulumiType;
    }
    /**
     * Create a ScalingPlanPersonalSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.daysOfWeek === undefined) && !opts.urn) {
                throw new Error("Missing required property 'daysOfWeek'");
            }
            if ((!args || args.offPeakStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'offPeakStartTime'");
            }
            if ((!args || args.peakStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peakStartTime'");
            }
            if ((!args || args.rampDownStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampDownStartTime'");
            }
            if ((!args || args.rampUpStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampUpStartTime'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scalingPlanName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scalingPlanName'");
            }
            resourceInputs["daysOfWeek"] = args ? args.daysOfWeek : undefined;
            resourceInputs["offPeakActionOnDisconnect"] = args ? args.offPeakActionOnDisconnect : undefined;
            resourceInputs["offPeakActionOnLogoff"] = args ? args.offPeakActionOnLogoff : undefined;
            resourceInputs["offPeakMinutesToWaitOnDisconnect"] = args ? args.offPeakMinutesToWaitOnDisconnect : undefined;
            resourceInputs["offPeakMinutesToWaitOnLogoff"] = args ? args.offPeakMinutesToWaitOnLogoff : undefined;
            resourceInputs["offPeakStartTime"] = args ? args.offPeakStartTime : undefined;
            resourceInputs["offPeakStartVMOnConnect"] = (_a = (args ? args.offPeakStartVMOnConnect : undefined)) !== null && _a !== void 0 ? _a : "Enable";
            resourceInputs["peakActionOnDisconnect"] = args ? args.peakActionOnDisconnect : undefined;
            resourceInputs["peakActionOnLogoff"] = args ? args.peakActionOnLogoff : undefined;
            resourceInputs["peakMinutesToWaitOnDisconnect"] = args ? args.peakMinutesToWaitOnDisconnect : undefined;
            resourceInputs["peakMinutesToWaitOnLogoff"] = args ? args.peakMinutesToWaitOnLogoff : undefined;
            resourceInputs["peakStartTime"] = args ? args.peakStartTime : undefined;
            resourceInputs["peakStartVMOnConnect"] = (_b = (args ? args.peakStartVMOnConnect : undefined)) !== null && _b !== void 0 ? _b : "Enable";
            resourceInputs["rampDownActionOnDisconnect"] = args ? args.rampDownActionOnDisconnect : undefined;
            resourceInputs["rampDownActionOnLogoff"] = args ? args.rampDownActionOnLogoff : undefined;
            resourceInputs["rampDownMinutesToWaitOnDisconnect"] = args ? args.rampDownMinutesToWaitOnDisconnect : undefined;
            resourceInputs["rampDownMinutesToWaitOnLogoff"] = args ? args.rampDownMinutesToWaitOnLogoff : undefined;
            resourceInputs["rampDownStartTime"] = args ? args.rampDownStartTime : undefined;
            resourceInputs["rampDownStartVMOnConnect"] = (_c = (args ? args.rampDownStartVMOnConnect : undefined)) !== null && _c !== void 0 ? _c : "Enable";
            resourceInputs["rampUpActionOnDisconnect"] = args ? args.rampUpActionOnDisconnect : undefined;
            resourceInputs["rampUpActionOnLogoff"] = args ? args.rampUpActionOnLogoff : undefined;
            resourceInputs["rampUpAutoStartHosts"] = args ? args.rampUpAutoStartHosts : undefined;
            resourceInputs["rampUpMinutesToWaitOnDisconnect"] = args ? args.rampUpMinutesToWaitOnDisconnect : undefined;
            resourceInputs["rampUpMinutesToWaitOnLogoff"] = args ? args.rampUpMinutesToWaitOnLogoff : undefined;
            resourceInputs["rampUpStartTime"] = args ? args.rampUpStartTime : undefined;
            resourceInputs["rampUpStartVMOnConnect"] = (_d = (args ? args.rampUpStartVMOnConnect : undefined)) !== null && _d !== void 0 ? _d : "Enable";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scalingPlanName"] = args ? args.scalingPlanName : undefined;
            resourceInputs["scalingPlanScheduleName"] = args ? args.scalingPlanScheduleName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["daysOfWeek"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offPeakActionOnDisconnect"] = undefined /*out*/;
            resourceInputs["offPeakActionOnLogoff"] = undefined /*out*/;
            resourceInputs["offPeakMinutesToWaitOnDisconnect"] = undefined /*out*/;
            resourceInputs["offPeakMinutesToWaitOnLogoff"] = undefined /*out*/;
            resourceInputs["offPeakStartTime"] = undefined /*out*/;
            resourceInputs["offPeakStartVMOnConnect"] = undefined /*out*/;
            resourceInputs["peakActionOnDisconnect"] = undefined /*out*/;
            resourceInputs["peakActionOnLogoff"] = undefined /*out*/;
            resourceInputs["peakMinutesToWaitOnDisconnect"] = undefined /*out*/;
            resourceInputs["peakMinutesToWaitOnLogoff"] = undefined /*out*/;
            resourceInputs["peakStartTime"] = undefined /*out*/;
            resourceInputs["peakStartVMOnConnect"] = undefined /*out*/;
            resourceInputs["rampDownActionOnDisconnect"] = undefined /*out*/;
            resourceInputs["rampDownActionOnLogoff"] = undefined /*out*/;
            resourceInputs["rampDownMinutesToWaitOnDisconnect"] = undefined /*out*/;
            resourceInputs["rampDownMinutesToWaitOnLogoff"] = undefined /*out*/;
            resourceInputs["rampDownStartTime"] = undefined /*out*/;
            resourceInputs["rampDownStartVMOnConnect"] = undefined /*out*/;
            resourceInputs["rampUpActionOnDisconnect"] = undefined /*out*/;
            resourceInputs["rampUpActionOnLogoff"] = undefined /*out*/;
            resourceInputs["rampUpAutoStartHosts"] = undefined /*out*/;
            resourceInputs["rampUpMinutesToWaitOnDisconnect"] = undefined /*out*/;
            resourceInputs["rampUpMinutesToWaitOnLogoff"] = undefined /*out*/;
            resourceInputs["rampUpStartTime"] = undefined /*out*/;
            resourceInputs["rampUpStartVMOnConnect"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:desktopvirtualization/v20230905:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20231004preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20231101preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20240116preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20240306preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20240403:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20240408preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20240808preview:ScalingPlanPersonalSchedule" }, { type: "azure-native:desktopvirtualization/v20241101preview:ScalingPlanPersonalSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScalingPlanPersonalSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ScalingPlanPersonalSchedule = ScalingPlanPersonalSchedule;
/** @internal */
ScalingPlanPersonalSchedule.__pulumiType = 'azure-native:desktopvirtualization:ScalingPlanPersonalSchedule';
