// @pulumi/azure-native/types/enums/apimanagement/v20230501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkType = exports.VersioningScheme = exports.Verbosity = exports.UserState = exports.TranslateRequiredQueryParametersConduct = exports.SubscriptionState = exports.State = exports.SoapApiType = exports.SkuType = exports.SchemaType = exports.SamplingType = exports.PublicNetworkAccess = exports.ProvisioningState = exports.Protocol = exports.ProductState = exports.PrivateEndpointServiceConnectionStatus = exports.PolicyRestrictionRequireBase = exports.PolicyFragmentContentFormat = exports.PolicyContentFormat = exports.OperationNameFormat = exports.OAuth2GrantType = exports.NatGatewayState = exports.LoggerType = exports.LegacyPortalStatus = exports.LegacyApiState = exports.KeyType = exports.IdentityProviderType = exports.HttpCorrelationProtocol = exports.HostnameType = exports.GroupType = exports.GrantType = exports.GatewayListDebugCredentialsContractPurpose = exports.DeveloperPortalStatus = exports.DataMaskingMode = exports.ContentFormat = exports.Confirmation = exports.ClientAuthenticationMethod = exports.CertificateStatus = exports.CertificateSource = exports.BearerTokenSendingMethods = exports.BearerTokenSendingMethod = exports.BackendType = exports.BackendProtocol = exports.AuthorizationType = exports.AuthorizationMethod = exports.AppType = exports.ApimIdentityType = exports.ApiType = exports.AlwaysLog = void 0;
exports.AlwaysLog = {
    /**
     * Always log all erroneous request regardless of sampling settings.
     */
    AllErrors: "allErrors",
};
exports.ApiType = {
    Http: "http",
    Soap: "soap",
    Websocket: "websocket",
    Graphql: "graphql",
    Odata: "odata",
    Grpc: "grpc",
};
exports.ApimIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.AppType = {
    /**
     * User create request was sent by legacy developer portal.
     */
    Portal: "portal",
    /**
     * User create request was sent by new developer portal.
     */
    DeveloperPortal: "developerPortal",
};
exports.AuthorizationMethod = {
    HEAD: "HEAD",
    OPTIONS: "OPTIONS",
    TRACE: "TRACE",
    GET: "GET",
    POST: "POST",
    PUT: "PUT",
    PATCH: "PATCH",
    DELETE: "DELETE",
};
exports.AuthorizationType = {
    /**
     * OAuth2 authorization type
     */
    OAuth2: "OAuth2",
};
exports.BackendProtocol = {
    /**
     * The Backend is a RESTful service.
     */
    Http: "http",
    /**
     * The Backend is a SOAP service.
     */
    Soap: "soap",
};
exports.BackendType = {
    /**
     * supports single backend
     */
    Single: "Single",
    /**
     * supports pool backend
     */
    Pool: "Pool",
};
exports.BearerTokenSendingMethod = {
    AuthorizationHeader: "authorizationHeader",
    Query: "query",
};
exports.BearerTokenSendingMethods = {
    /**
     * Access token will be transmitted in the Authorization header using Bearer schema
     */
    AuthorizationHeader: "authorizationHeader",
    /**
     * Access token will be transmitted as query parameters.
     */
    Query: "query",
};
exports.CertificateSource = {
    Managed: "Managed",
    KeyVault: "KeyVault",
    Custom: "Custom",
    BuiltIn: "BuiltIn",
};
exports.CertificateStatus = {
    Completed: "Completed",
    Failed: "Failed",
    InProgress: "InProgress",
};
exports.ClientAuthenticationMethod = {
    /**
     * Basic Client Authentication method.
     */
    Basic: "Basic",
    /**
     * Body based Authentication method.
     */
    Body: "Body",
};
exports.Confirmation = {
    /**
     * Send an e-mail to the user confirming they have successfully signed up.
     */
    Signup: "signup",
    /**
     * Send an e-mail inviting the user to sign-up and complete registration.
     */
    Invite: "invite",
};
exports.ContentFormat = {
    /**
     * The contents are inline and Content type is a WADL document.
     */
    Wadl_xml: "wadl-xml",
    /**
     * The WADL document is hosted on a publicly accessible internet address.
     */
    Wadl_link_json: "wadl-link-json",
    /**
     * The contents are inline and Content Type is a OpenAPI 2.0 JSON Document.
     */
    Swagger_json: "swagger-json",
    /**
     * The OpenAPI 2.0 JSON document is hosted on a publicly accessible internet address.
     */
    Swagger_link_json: "swagger-link-json",
    /**
     * The contents are inline and the document is a WSDL/Soap document.
     */
    Wsdl: "wsdl",
    /**
     * The WSDL document is hosted on a publicly accessible internet address.
     */
    Wsdl_link: "wsdl-link",
    /**
     * The contents are inline and Content Type is a OpenAPI 3.0 YAML Document.
     */
    Openapi: "openapi",
    /**
     * The contents are inline and Content Type is a OpenAPI 3.0 JSON Document.
     */
    Openapi_json: "openapi+json",
    /**
     * The OpenAPI 3.0 YAML document is hosted on a publicly accessible internet address.
     */
    Openapi_link: "openapi-link",
    /**
     * The OpenAPI 3.0 JSON document is hosted on a publicly accessible internet address.
     */
    Openapi_json_link: "openapi+json-link",
    /**
     * The GraphQL API endpoint hosted on a publicly accessible internet address.
     */
    Graphql_link: "graphql-link",
    /**
     * The contents are inline and Content Type is a OData XML Document.
     */
    Odata: "odata",
    /**
     * The OData metadata document hosted on a publicly accessible internet address.
     */
    Odata_link: "odata-link",
    /**
     * The contents are inline and Content Type is a gRPC protobuf file.
     */
    Grpc: "grpc",
    /**
     * The gRPC protobuf file is hosted on a publicly accessible internet address.
     */
    Grpc_link: "grpc-link",
};
exports.DataMaskingMode = {
    /**
     * Mask the value of an entity.
     */
    Mask: "Mask",
    /**
     * Hide the presence of an entity.
     */
    Hide: "Hide",
};
exports.DeveloperPortalStatus = {
    /**
     * Developer Portal is enabled for the service.
     */
    Enabled: "Enabled",
    /**
     * Developer Portal is disabled for the service.
     */
    Disabled: "Disabled",
};
exports.GatewayListDebugCredentialsContractPurpose = {
    /**
     * The tracing purpose.
     */
    Tracing: "tracing",
};
exports.GrantType = {
    /**
     * Authorization Code Grant flow as described https://tools.ietf.org/html/rfc6749#section-4.1.
     */
    AuthorizationCode: "authorizationCode",
    /**
     * Implicit Code Grant flow as described https://tools.ietf.org/html/rfc6749#section-4.2.
     */
    Implicit: "implicit",
    /**
     * Resource Owner Password Grant flow as described https://tools.ietf.org/html/rfc6749#section-4.3.
     */
    ResourceOwnerPassword: "resourceOwnerPassword",
    /**
     * Client Credentials Grant flow as described https://tools.ietf.org/html/rfc6749#section-4.4.
     */
    ClientCredentials: "clientCredentials",
};
exports.GroupType = {
    Custom: "custom",
    System: "system",
    External: "external",
};
exports.HostnameType = {
    Proxy: "Proxy",
    Portal: "Portal",
    Management: "Management",
    Scm: "Scm",
    DeveloperPortal: "DeveloperPortal",
    ConfigurationApi: "ConfigurationApi",
};
exports.HttpCorrelationProtocol = {
    /**
     * Do not read and inject correlation headers.
     */
    None: "None",
    /**
     * Inject Request-Id and Request-Context headers with request correlation data. See https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/HttpCorrelationProtocol.md.
     */
    Legacy: "Legacy",
    /**
     * Inject Trace Context headers. See https://w3c.github.io/trace-context.
     */
    W3C: "W3C",
};
exports.IdentityProviderType = {
    /**
     * Facebook as Identity provider.
     */
    Facebook: "facebook",
    /**
     * Google as Identity provider.
     */
    Google: "google",
    /**
     * Microsoft Live as Identity provider.
     */
    Microsoft: "microsoft",
    /**
     * Twitter as Identity provider.
     */
    Twitter: "twitter",
    /**
     * Azure Active Directory as Identity provider.
     */
    Aad: "aad",
    /**
     * Azure Active Directory B2C as Identity provider.
     */
    AadB2C: "aadB2C",
};
exports.KeyType = {
    Primary: "primary",
    Secondary: "secondary",
};
exports.LegacyApiState = {
    /**
     * Legacy Configuration API (v1) is enabled for the service and self-hosted gateways can connect to it.
     */
    Enabled: "Enabled",
    /**
     * Legacy Configuration API (v1) is disabled for the service and self-hosted gateways can not connect to it.
     */
    Disabled: "Disabled",
};
exports.LegacyPortalStatus = {
    /**
     * Legacy Portal is enabled for the service.
     */
    Enabled: "Enabled",
    /**
     * Legacy Portal is disabled for the service.
     */
    Disabled: "Disabled",
};
exports.LoggerType = {
    /**
     * Azure Event Hub as log destination.
     */
    AzureEventHub: "azureEventHub",
    /**
     * Azure Application Insights as log destination.
     */
    ApplicationInsights: "applicationInsights",
    /**
     * Azure Monitor
     */
    AzureMonitor: "azureMonitor",
};
exports.NatGatewayState = {
    /**
     * Nat Gateway is enabled for the service.
     */
    Enabled: "Enabled",
    /**
     * Nat Gateway is disabled for the service.
     */
    Disabled: "Disabled",
};
exports.OAuth2GrantType = {
    /**
     * Authorization Code grant
     */
    AuthorizationCode: "AuthorizationCode",
    /**
     * Client Credential grant
     */
    ClientCredentials: "ClientCredentials",
};
exports.OperationNameFormat = {
    /**
     * API_NAME;rev=API_REVISION - OPERATION_NAME
     */
    Name: "Name",
    /**
     * HTTP_VERB URL
     */
    Url: "Url",
};
exports.PolicyContentFormat = {
    /**
     * The contents are inline and Content type is an XML document.
     */
    Xml: "xml",
    /**
     * The policy XML document is hosted on a HTTP endpoint accessible from the API Management service.
     */
    Xml_link: "xml-link",
    /**
     * The contents are inline and Content type is a non XML encoded policy document.
     */
    Rawxml: "rawxml",
    /**
     * The policy document is not XML encoded and is hosted on a HTTP endpoint accessible from the API Management service.
     */
    Rawxml_link: "rawxml-link",
};
exports.PolicyFragmentContentFormat = {
    /**
     * The contents are inline and Content type is an XML document.
     */
    Xml: "xml",
    /**
     * The contents are inline and Content type is a non XML encoded policy document.
     */
    Rawxml: "rawxml",
};
exports.PolicyRestrictionRequireBase = {
    /**
     * The policy is required to have base policy
     */
    True: "true",
    /**
     * The policy does not require to have base policy
     */
    False: "false",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ProductState = {
    NotPublished: "notPublished",
    Published: "published",
};
exports.Protocol = {
    Http: "http",
    Https: "https",
    Ws: "ws",
    Wss: "wss",
};
exports.ProvisioningState = {
    Created: "created",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SamplingType = {
    /**
     * Fixed-rate sampling.
     */
    Fixed: "fixed",
};
exports.SchemaType = {
    /**
     * XML schema type.
     */
    Xml: "xml",
    /**
     * Json schema type.
     */
    Json: "json",
};
exports.SkuType = {
    /**
     * Developer SKU of Api Management.
     */
    Developer: "Developer",
    /**
     * Standard SKU of Api Management.
     */
    Standard: "Standard",
    /**
     * Premium SKU of Api Management.
     */
    Premium: "Premium",
    /**
     * Basic SKU of Api Management.
     */
    Basic: "Basic",
    /**
     * Consumption SKU of Api Management.
     */
    Consumption: "Consumption",
    /**
     * Isolated SKU of Api Management.
     */
    Isolated: "Isolated",
    /**
     * BasicV2 SKU of Api Management.
     */
    BasicV2: "BasicV2",
    /**
     * StandardV2 SKU of Api Management.
     */
    StandardV2: "StandardV2",
};
exports.SoapApiType = {
    /**
     * Imports a SOAP API having a RESTful front end.
     */
    SoapToRest: "http",
    /**
     * Imports the SOAP API having a SOAP front end.
     */
    SoapPassThrough: "soap",
    /**
     * Imports the API having a Websocket front end.
     */
    WebSocket: "websocket",
    /**
     * Imports the API having a GraphQL front end.
     */
    GraphQL: "graphql",
    /**
     * Imports the API having a OData front end.
     */
    OData: "odata",
    /**
     * Imports the API having a gRPC front end.
     */
    GRPC: "grpc",
};
exports.State = {
    /**
     * The issue is proposed.
     */
    Proposed: "proposed",
    /**
     * The issue is opened.
     */
    Open: "open",
    /**
     * The issue was removed.
     */
    Removed: "removed",
    /**
     * The issue is now resolved.
     */
    Resolved: "resolved",
    /**
     * The issue was closed.
     */
    Closed: "closed",
};
exports.SubscriptionState = {
    Suspended: "suspended",
    Active: "active",
    Expired: "expired",
    Submitted: "submitted",
    Rejected: "rejected",
    Cancelled: "cancelled",
};
exports.TranslateRequiredQueryParametersConduct = {
    /**
     * Translates required query parameters to template ones. Is a default value
     */
    Template: "template",
    /**
     * Leaves required query parameters as they are (no translation done).
     */
    Query: "query",
};
exports.UserState = {
    /**
     * User state is active.
     */
    Active: "active",
    /**
     * User is blocked. Blocked users cannot authenticate at developer portal or call API.
     */
    Blocked: "blocked",
    /**
     * User account is pending. Requires identity confirmation before it can be made active.
     */
    Pending: "pending",
    /**
     * User account is closed. All identities and related entities are removed.
     */
    Deleted: "deleted",
};
exports.Verbosity = {
    /**
     * All the traces emitted by trace policies will be sent to the logger attached to this diagnostic instance.
     */
    Verbose: "verbose",
    /**
     * Traces with 'severity' set to 'information' and 'error' will be sent to the logger attached to this diagnostic instance.
     */
    Information: "information",
    /**
     * Only traces with 'severity' set to 'error' will be sent to the logger attached to this diagnostic instance.
     */
    Error: "error",
};
exports.VersioningScheme = {
    /**
     * The API Version is passed in a path segment.
     */
    Segment: "Segment",
    /**
     * The API Version is passed in a query parameter.
     */
    Query: "Query",
    /**
     * The API Version is passed in a HTTP header.
     */
    Header: "Header",
};
exports.VirtualNetworkType = {
    /**
     * The service is not part of any Virtual Network.
     */
    None: "None",
    /**
     * The service is part of Virtual Network and it is accessible from Internet.
     */
    External: "External",
    /**
     * The service is part of Virtual Network and it is only accessible from within the virtual network.
     */
    Internal: "Internal",
};
