// @pulumi/azure-native/types/enums/automation/v20231101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenType = exports.SourceType = exports.SkuNameEnum = exports.ScheduleFrequency = exports.ScheduleDay = exports.RunbookTypeEnum = exports.ResourceIdentityType = exports.EncryptionKeySourceType = exports.ContentSourceType = void 0;
exports.ContentSourceType = {
    EmbeddedContent: "embeddedContent",
    Uri: "uri",
};
exports.EncryptionKeySourceType = {
    Microsoft_Automation: "Microsoft.Automation",
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.RunbookTypeEnum = {
    Script: "Script",
    Graph: "Graph",
    PowerShellWorkflow: "PowerShellWorkflow",
    PowerShell: "PowerShell",
    GraphPowerShellWorkflow: "GraphPowerShellWorkflow",
    GraphPowerShell: "GraphPowerShell",
    Python2: "Python2",
    Python3: "Python3",
    PowerShell72: "PowerShell72",
};
exports.ScheduleDay = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.ScheduleFrequency = {
    OneTime: "OneTime",
    Day: "Day",
    Hour: "Hour",
    Week: "Week",
    Month: "Month",
    /**
     * The minimum allowed interval for Minute schedules is 15 minutes.
     */
    Minute: "Minute",
};
exports.SkuNameEnum = {
    Free: "Free",
    Basic: "Basic",
};
exports.SourceType = {
    VsoGit: "VsoGit",
    VsoTfvc: "VsoTfvc",
    GitHub: "GitHub",
};
exports.TokenType = {
    PersonalAccessToken: "PersonalAccessToken",
    Oauth: "Oauth",
};
