// @pulumi/azure-native/documentdb/v20230915/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GremlinResourceGremlinDatabase = exports.getTableResourceTableOutput = exports.getTableResourceTable = exports.getSqlResourceSqlUserDefinedFunctionOutput = exports.getSqlResourceSqlUserDefinedFunction = exports.getSqlResourceSqlTriggerOutput = exports.getSqlResourceSqlTrigger = exports.getSqlResourceSqlStoredProcedureOutput = exports.getSqlResourceSqlStoredProcedure = exports.getSqlResourceSqlRoleDefinitionOutput = exports.getSqlResourceSqlRoleDefinition = exports.getSqlResourceSqlRoleAssignmentOutput = exports.getSqlResourceSqlRoleAssignment = exports.getSqlResourceSqlDatabaseOutput = exports.getSqlResourceSqlDatabase = exports.getSqlResourceSqlContainerOutput = exports.getSqlResourceSqlContainer = exports.getServiceOutput = exports.getService = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNotebookWorkspaceOutput = exports.getNotebookWorkspace = exports.getMongoDBResourceMongoUserDefinitionOutput = exports.getMongoDBResourceMongoUserDefinition = exports.getMongoDBResourceMongoRoleDefinitionOutput = exports.getMongoDBResourceMongoRoleDefinition = exports.getMongoDBResourceMongoDBDatabaseOutput = exports.getMongoDBResourceMongoDBDatabase = exports.getMongoDBResourceMongoDBCollectionOutput = exports.getMongoDBResourceMongoDBCollection = exports.getGremlinResourceGremlinGraphOutput = exports.getGremlinResourceGremlinGraph = exports.getGremlinResourceGremlinDatabaseOutput = exports.getGremlinResourceGremlinDatabase = exports.getDatabaseAccountOutput = exports.getDatabaseAccount = exports.getCassandraResourceCassandraTableOutput = exports.getCassandraResourceCassandraTable = exports.getCassandraResourceCassandraKeyspaceOutput = exports.getCassandraResourceCassandraKeyspace = exports.getCassandraDataCenterOutput = exports.getCassandraDataCenter = exports.getCassandraClusterOutput = exports.getCassandraCluster = exports.DatabaseAccount = exports.CassandraResourceCassandraTable = exports.CassandraResourceCassandraKeyspace = exports.CassandraDataCenter = exports.CassandraCluster = void 0;
exports.TableResourceTable = exports.SqlResourceSqlUserDefinedFunction = exports.SqlResourceSqlTrigger = exports.SqlResourceSqlStoredProcedure = exports.SqlResourceSqlRoleDefinition = exports.SqlResourceSqlRoleAssignment = exports.SqlResourceSqlDatabase = exports.SqlResourceSqlContainer = exports.Service = exports.PrivateEndpointConnection = exports.NotebookWorkspace = exports.MongoDBResourceMongoUserDefinition = exports.MongoDBResourceMongoRoleDefinition = exports.MongoDBResourceMongoDBDatabase = exports.MongoDBResourceMongoDBCollection = exports.listNotebookWorkspaceConnectionInfoOutput = exports.listNotebookWorkspaceConnectionInfo = exports.listDatabaseAccountKeysOutput = exports.listDatabaseAccountKeys = exports.listDatabaseAccountConnectionStringsOutput = exports.listDatabaseAccountConnectionStrings = exports.GremlinResourceGremlinGraph = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CassandraCluster = null;
utilities.lazyLoad(exports, ["CassandraCluster"], () => require("./cassandraCluster"));
exports.CassandraDataCenter = null;
utilities.lazyLoad(exports, ["CassandraDataCenter"], () => require("./cassandraDataCenter"));
exports.CassandraResourceCassandraKeyspace = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraKeyspace"], () => require("./cassandraResourceCassandraKeyspace"));
exports.CassandraResourceCassandraTable = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraTable"], () => require("./cassandraResourceCassandraTable"));
exports.DatabaseAccount = null;
utilities.lazyLoad(exports, ["DatabaseAccount"], () => require("./databaseAccount"));
exports.getCassandraCluster = null;
exports.getCassandraClusterOutput = null;
utilities.lazyLoad(exports, ["getCassandraCluster", "getCassandraClusterOutput"], () => require("./getCassandraCluster"));
exports.getCassandraDataCenter = null;
exports.getCassandraDataCenterOutput = null;
utilities.lazyLoad(exports, ["getCassandraDataCenter", "getCassandraDataCenterOutput"], () => require("./getCassandraDataCenter"));
exports.getCassandraResourceCassandraKeyspace = null;
exports.getCassandraResourceCassandraKeyspaceOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraKeyspace", "getCassandraResourceCassandraKeyspaceOutput"], () => require("./getCassandraResourceCassandraKeyspace"));
exports.getCassandraResourceCassandraTable = null;
exports.getCassandraResourceCassandraTableOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraTable", "getCassandraResourceCassandraTableOutput"], () => require("./getCassandraResourceCassandraTable"));
exports.getDatabaseAccount = null;
exports.getDatabaseAccountOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccount", "getDatabaseAccountOutput"], () => require("./getDatabaseAccount"));
exports.getGremlinResourceGremlinDatabase = null;
exports.getGremlinResourceGremlinDatabaseOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinDatabase", "getGremlinResourceGremlinDatabaseOutput"], () => require("./getGremlinResourceGremlinDatabase"));
exports.getGremlinResourceGremlinGraph = null;
exports.getGremlinResourceGremlinGraphOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinGraph", "getGremlinResourceGremlinGraphOutput"], () => require("./getGremlinResourceGremlinGraph"));
exports.getMongoDBResourceMongoDBCollection = null;
exports.getMongoDBResourceMongoDBCollectionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBCollection", "getMongoDBResourceMongoDBCollectionOutput"], () => require("./getMongoDBResourceMongoDBCollection"));
exports.getMongoDBResourceMongoDBDatabase = null;
exports.getMongoDBResourceMongoDBDatabaseOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBDatabase", "getMongoDBResourceMongoDBDatabaseOutput"], () => require("./getMongoDBResourceMongoDBDatabase"));
exports.getMongoDBResourceMongoRoleDefinition = null;
exports.getMongoDBResourceMongoRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoRoleDefinition", "getMongoDBResourceMongoRoleDefinitionOutput"], () => require("./getMongoDBResourceMongoRoleDefinition"));
exports.getMongoDBResourceMongoUserDefinition = null;
exports.getMongoDBResourceMongoUserDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoUserDefinition", "getMongoDBResourceMongoUserDefinitionOutput"], () => require("./getMongoDBResourceMongoUserDefinition"));
exports.getNotebookWorkspace = null;
exports.getNotebookWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getNotebookWorkspace", "getNotebookWorkspaceOutput"], () => require("./getNotebookWorkspace"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSqlResourceSqlContainer = null;
exports.getSqlResourceSqlContainerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlContainer", "getSqlResourceSqlContainerOutput"], () => require("./getSqlResourceSqlContainer"));
exports.getSqlResourceSqlDatabase = null;
exports.getSqlResourceSqlDatabaseOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlDatabase", "getSqlResourceSqlDatabaseOutput"], () => require("./getSqlResourceSqlDatabase"));
exports.getSqlResourceSqlRoleAssignment = null;
exports.getSqlResourceSqlRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleAssignment", "getSqlResourceSqlRoleAssignmentOutput"], () => require("./getSqlResourceSqlRoleAssignment"));
exports.getSqlResourceSqlRoleDefinition = null;
exports.getSqlResourceSqlRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleDefinition", "getSqlResourceSqlRoleDefinitionOutput"], () => require("./getSqlResourceSqlRoleDefinition"));
exports.getSqlResourceSqlStoredProcedure = null;
exports.getSqlResourceSqlStoredProcedureOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlStoredProcedure", "getSqlResourceSqlStoredProcedureOutput"], () => require("./getSqlResourceSqlStoredProcedure"));
exports.getSqlResourceSqlTrigger = null;
exports.getSqlResourceSqlTriggerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlTrigger", "getSqlResourceSqlTriggerOutput"], () => require("./getSqlResourceSqlTrigger"));
exports.getSqlResourceSqlUserDefinedFunction = null;
exports.getSqlResourceSqlUserDefinedFunctionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlUserDefinedFunction", "getSqlResourceSqlUserDefinedFunctionOutput"], () => require("./getSqlResourceSqlUserDefinedFunction"));
exports.getTableResourceTable = null;
exports.getTableResourceTableOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTable", "getTableResourceTableOutput"], () => require("./getTableResourceTable"));
exports.GremlinResourceGremlinDatabase = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinDatabase"], () => require("./gremlinResourceGremlinDatabase"));
exports.GremlinResourceGremlinGraph = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinGraph"], () => require("./gremlinResourceGremlinGraph"));
exports.listDatabaseAccountConnectionStrings = null;
exports.listDatabaseAccountConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountConnectionStrings", "listDatabaseAccountConnectionStringsOutput"], () => require("./listDatabaseAccountConnectionStrings"));
exports.listDatabaseAccountKeys = null;
exports.listDatabaseAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountKeys", "listDatabaseAccountKeysOutput"], () => require("./listDatabaseAccountKeys"));
exports.listNotebookWorkspaceConnectionInfo = null;
exports.listNotebookWorkspaceConnectionInfoOutput = null;
utilities.lazyLoad(exports, ["listNotebookWorkspaceConnectionInfo", "listNotebookWorkspaceConnectionInfoOutput"], () => require("./listNotebookWorkspaceConnectionInfo"));
exports.MongoDBResourceMongoDBCollection = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBCollection"], () => require("./mongoDBResourceMongoDBCollection"));
exports.MongoDBResourceMongoDBDatabase = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBDatabase"], () => require("./mongoDBResourceMongoDBDatabase"));
exports.MongoDBResourceMongoRoleDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoRoleDefinition"], () => require("./mongoDBResourceMongoRoleDefinition"));
exports.MongoDBResourceMongoUserDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoUserDefinition"], () => require("./mongoDBResourceMongoUserDefinition"));
exports.NotebookWorkspace = null;
utilities.lazyLoad(exports, ["NotebookWorkspace"], () => require("./notebookWorkspace"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.SqlResourceSqlContainer = null;
utilities.lazyLoad(exports, ["SqlResourceSqlContainer"], () => require("./sqlResourceSqlContainer"));
exports.SqlResourceSqlDatabase = null;
utilities.lazyLoad(exports, ["SqlResourceSqlDatabase"], () => require("./sqlResourceSqlDatabase"));
exports.SqlResourceSqlRoleAssignment = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleAssignment"], () => require("./sqlResourceSqlRoleAssignment"));
exports.SqlResourceSqlRoleDefinition = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleDefinition"], () => require("./sqlResourceSqlRoleDefinition"));
exports.SqlResourceSqlStoredProcedure = null;
utilities.lazyLoad(exports, ["SqlResourceSqlStoredProcedure"], () => require("./sqlResourceSqlStoredProcedure"));
exports.SqlResourceSqlTrigger = null;
utilities.lazyLoad(exports, ["SqlResourceSqlTrigger"], () => require("./sqlResourceSqlTrigger"));
exports.SqlResourceSqlUserDefinedFunction = null;
utilities.lazyLoad(exports, ["SqlResourceSqlUserDefinedFunction"], () => require("./sqlResourceSqlUserDefinedFunction"));
exports.TableResourceTable = null;
utilities.lazyLoad(exports, ["TableResourceTable"], () => require("./tableResourceTable"));
// Export enums:
__exportStar(require("../../types/enums/documentdb/v20230915"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:documentdb/v20230915:CassandraCluster":
                return new exports.CassandraCluster(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:CassandraDataCenter":
                return new exports.CassandraDataCenter(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:CassandraResourceCassandraKeyspace":
                return new exports.CassandraResourceCassandraKeyspace(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:CassandraResourceCassandraTable":
                return new exports.CassandraResourceCassandraTable(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:DatabaseAccount":
                return new exports.DatabaseAccount(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:GremlinResourceGremlinDatabase":
                return new exports.GremlinResourceGremlinDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:GremlinResourceGremlinGraph":
                return new exports.GremlinResourceGremlinGraph(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:MongoDBResourceMongoDBCollection":
                return new exports.MongoDBResourceMongoDBCollection(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:MongoDBResourceMongoDBDatabase":
                return new exports.MongoDBResourceMongoDBDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:MongoDBResourceMongoRoleDefinition":
                return new exports.MongoDBResourceMongoRoleDefinition(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:MongoDBResourceMongoUserDefinition":
                return new exports.MongoDBResourceMongoUserDefinition(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:NotebookWorkspace":
                return new exports.NotebookWorkspace(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlContainer":
                return new exports.SqlResourceSqlContainer(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlDatabase":
                return new exports.SqlResourceSqlDatabase(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlRoleAssignment":
                return new exports.SqlResourceSqlRoleAssignment(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlRoleDefinition":
                return new exports.SqlResourceSqlRoleDefinition(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlStoredProcedure":
                return new exports.SqlResourceSqlStoredProcedure(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlTrigger":
                return new exports.SqlResourceSqlTrigger(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:SqlResourceSqlUserDefinedFunction":
                return new exports.SqlResourceSqlUserDefinedFunction(name, undefined, { urn });
            case "azure-native:documentdb/v20230915:TableResourceTable":
                return new exports.TableResourceTable(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "documentdb/v20230915", _module);
