// @pulumi/azure-native/types/enums/elasticsan/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeCreateOption = exports.StorageTargetType = exports.SkuTier = exports.SkuName = exports.PrivateEndpointServiceConnectionStatus = exports.EncryptionType = exports.Action = exports.v20240601preview = exports.v20240501 = exports.v20230101 = exports.v20221201preview = exports.v20211120preview = void 0;
// Export sub-modules:
const v20211120preview = require("./v20211120preview");
exports.v20211120preview = v20211120preview;
const v20221201preview = require("./v20221201preview");
exports.v20221201preview = v20221201preview;
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
exports.Action = {
    Allow: "Allow",
};
exports.EncryptionType = {
    /**
     * Volume is encrypted at rest with Platform managed key. It is the default encryption type.
     */
    EncryptionAtRestWithPlatformKey: "EncryptionAtRestWithPlatformKey",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Failed: "Failed",
    Rejected: "Rejected",
};
exports.SkuName = {
    /**
     * Premium locally redundant storage
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Premium zone redundant storage
     */
    Premium_ZRS: "Premium_ZRS",
};
exports.SkuTier = {
    /**
     * Premium Tier
     */
    Premium: "Premium",
};
exports.StorageTargetType = {
    Iscsi: "Iscsi",
    None: "None",
};
exports.VolumeCreateOption = {
    None: "None",
};
