// @pulumi/azure-native/types/enums/attestation/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TpmAttestationAuthenticationType = exports.PublicNetworkAccessType = exports.PrivateEndpointServiceConnectionStatus = exports.v20210601preview = exports.v20210601 = void 0;
// Export sub-modules:
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccessType = {
    /**
     * Enables public network connectivity to the Attestation Provider REST APIs.
     */
    Enabled: "Enabled",
    /**
     * Disables public network connectivity to the Attestation Provider REST APIs.
     */
    Disabled: "Disabled",
};
exports.TpmAttestationAuthenticationType = {
    /**
     * Enables the requirement of authentication for TPM Attestation REST APIs.
     */
    Enabled: "Enabled",
    /**
     * Disables the requirement of authentication for TPM Attestation REST APIs.
     */
    Disabled: "Disabled",
};
