// @pulumi/azure-native/impact/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240501preview = exports.WorkloadImpact = exports.Insight = exports.getWorkloadImpactOutput = exports.getWorkloadImpact = exports.getInsightOutput = exports.getInsight = exports.getConnectorOutput = exports.getConnector = exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getInsight = null;
exports.getInsightOutput = null;
utilities.lazyLoad(exports, ["getInsight", "getInsightOutput"], () => require("./getInsight"));
exports.getWorkloadImpact = null;
exports.getWorkloadImpactOutput = null;
utilities.lazyLoad(exports, ["getWorkloadImpact", "getWorkloadImpactOutput"], () => require("./getWorkloadImpact"));
exports.Insight = null;
utilities.lazyLoad(exports, ["Insight"], () => require("./insight"));
exports.WorkloadImpact = null;
utilities.lazyLoad(exports, ["WorkloadImpact"], () => require("./workloadImpact"));
// Export enums:
__exportStar(require("../types/enums/impact"), exports);
// Export sub-modules:
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:impact:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:impact:Insight":
                return new exports.Insight(name, undefined, { urn });
            case "azure-native:impact:WorkloadImpact":
                return new exports.WorkloadImpact(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "impact", _module);
