// @pulumi/azure-native/billing/billingRoleAssignmentByDepartment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingRoleAssignmentByDepartment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The role assignment
 *
 * Uses Azure REST API version 2019-10-01-preview. In version 1.x of the Azure Native provider, it used API version 2019-10-01-preview.
 *
 * Other available API versions: 2024-04-01.
 */
class BillingRoleAssignmentByDepartment extends pulumi.CustomResource {
    /**
     * Get an existing BillingRoleAssignmentByDepartment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BillingRoleAssignmentByDepartment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BillingRoleAssignmentByDepartment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BillingRoleAssignmentByDepartment.__pulumiType;
    }
    /**
     * Create a BillingRoleAssignmentByDepartment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.billingAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'billingAccountName'");
            }
            if ((!args || args.departmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'departmentName'");
            }
            resourceInputs["billingAccountName"] = args ? args.billingAccountName : undefined;
            resourceInputs["billingRoleAssignmentName"] = args ? args.billingRoleAssignmentName : undefined;
            resourceInputs["departmentName"] = args ? args.departmentName : undefined;
            resourceInputs["principalId"] = args ? args.principalId : undefined;
            resourceInputs["principalTenantId"] = args ? args.principalTenantId : undefined;
            resourceInputs["roleDefinitionId"] = args ? args.roleDefinitionId : undefined;
            resourceInputs["userAuthenticationType"] = args ? args.userAuthenticationType : undefined;
            resourceInputs["userEmailAddress"] = args ? args.userEmailAddress : undefined;
            resourceInputs["createdByPrincipalId"] = undefined /*out*/;
            resourceInputs["createdByPrincipalTenantId"] = undefined /*out*/;
            resourceInputs["createdByUserEmailAddress"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdByPrincipalId"] = undefined /*out*/;
            resourceInputs["createdByPrincipalTenantId"] = undefined /*out*/;
            resourceInputs["createdByUserEmailAddress"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalTenantId"] = undefined /*out*/;
            resourceInputs["roleDefinitionId"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userAuthenticationType"] = undefined /*out*/;
            resourceInputs["userEmailAddress"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:billing/v20191001preview:BillingRoleAssignmentByDepartment" }, { type: "azure-native:billing/v20240401:BillingRoleAssignmentByDepartment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BillingRoleAssignmentByDepartment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BillingRoleAssignmentByDepartment = BillingRoleAssignmentByDepartment;
/** @internal */
BillingRoleAssignmentByDepartment.__pulumiType = 'azure-native:billing:BillingRoleAssignmentByDepartment';
