// @pulumi/azure-native/types/enums/app/v20221001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnauthenticatedClientActionV2 = exports.Type = exports.StorageType = exports.SkuName = exports.Scheme = exports.ManagedServiceIdentityType = exports.ManagedEnvironmentOutBoundType = exports.LogLevel = exports.IngressTransportMethod = exports.IngressClientCertificateMode = exports.ForwardProxyConvention = exports.ExtendedLocationTypes = exports.CookieExpirationConvention = exports.ClientCredentialMethod = exports.BindingType = exports.AppProtocol = exports.ActiveRevisionsMode = exports.Action = exports.AccessMode = void 0;
exports.AccessMode = {
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.Action = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.ActiveRevisionsMode = {
    Multiple: "Multiple",
    Single: "Single",
};
exports.AppProtocol = {
    Http: "http",
    Grpc: "grpc",
};
exports.BindingType = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
};
exports.ClientCredentialMethod = {
    ClientSecretPost: "ClientSecretPost",
};
exports.CookieExpirationConvention = {
    FixedTime: "FixedTime",
    IdentityProviderDerived: "IdentityProviderDerived",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.ForwardProxyConvention = {
    NoProxy: "NoProxy",
    Standard: "Standard",
    Custom: "Custom",
};
exports.IngressClientCertificateMode = {
    Ignore: "ignore",
    Accept: "accept",
    Require: "require",
};
exports.IngressTransportMethod = {
    Auto: "auto",
    Http: "http",
    Http2: "http2",
    Tcp: "tcp",
};
exports.LogLevel = {
    Info: "info",
    Debug: "debug",
    Warn: "warn",
    Error: "error",
};
exports.ManagedEnvironmentOutBoundType = {
    LoadBalancer: "LoadBalancer",
    UserDefinedRouting: "UserDefinedRouting",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.Scheme = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.SkuName = {
    /**
     * Consumption SKU of Managed Environment.
     */
    Consumption: "Consumption",
    /**
     * Premium SKU of Managed Environment.
     */
    Premium: "Premium",
};
exports.StorageType = {
    AzureFile: "AzureFile",
    EmptyDir: "EmptyDir",
};
exports.Type = {
    Liveness: "Liveness",
    Readiness: "Readiness",
    Startup: "Startup",
};
exports.UnauthenticatedClientActionV2 = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
    Return401: "Return401",
    Return403: "Return403",
};
