// @pulumi/azure-native/types/enums/dashboard/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.DeterministicOutboundIP = exports.AutoGeneratedDomainNameLabelScope = exports.ApiKey = exports.v20241001 = exports.v20231001preview = exports.v20230901 = exports.v20221001preview = exports.v20220801 = exports.v20210901preview = void 0;
// Export sub-modules:
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20220801 = require("./v20220801");
exports.v20220801 = v20220801;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20241001 = require("./v20241001");
exports.v20241001 = v20241001;
exports.ApiKey = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.AutoGeneratedDomainNameLabelScope = {
    TenantReuse: "TenantReuse",
};
exports.DeterministicOutboundIP = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ZoneRedundancy = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
