// @pulumi/azure-native/security/v20231201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityContact = exports.getSecurityContactOutput = exports.getSecurityContact = exports.getAutomationOutput = exports.getAutomation = exports.Automation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Automation = null;
utilities.lazyLoad(exports, ["Automation"], () => require("./automation"));
exports.getAutomation = null;
exports.getAutomationOutput = null;
utilities.lazyLoad(exports, ["getAutomation", "getAutomationOutput"], () => require("./getAutomation"));
exports.getSecurityContact = null;
exports.getSecurityContactOutput = null;
utilities.lazyLoad(exports, ["getSecurityContact", "getSecurityContactOutput"], () => require("./getSecurityContact"));
exports.SecurityContact = null;
utilities.lazyLoad(exports, ["SecurityContact"], () => require("./securityContact"));
// Export enums:
__exportStar(require("../../types/enums/security/v20231201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20231201preview:Automation":
                return new exports.Automation(name, undefined, { urn });
            case "azure-native:security/v20231201preview:SecurityContact":
                return new exports.SecurityContact(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20231201preview", _module);
