// @pulumi/azure-native/types/enums/policyinsights/v20241001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceDiscoveryMode = exports.ComplianceState = void 0;
exports.ComplianceState = {
    /**
     * The resource is in compliance with the policy.
     */
    Compliant: "Compliant",
    /**
     * The resource is not in compliance with the policy.
     */
    NonCompliant: "NonCompliant",
    /**
     * The compliance state of the resource is not known.
     */
    Unknown: "Unknown",
};
exports.ResourceDiscoveryMode = {
    /**
     * Remediate resources that are already known to be non-compliant.
     */
    ExistingNonCompliant: "ExistingNonCompliant",
    /**
     * Re-evaluate the compliance state of resources and then remediate the resources found to be non-compliant. The resourceIds filter cannot be used in this mode.
     */
    ReEvaluateCompliance: "ReEvaluateCompliance",
};
