// @pulumi/azure-native/types/enums/netapp/v20240501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeLanguage = exports.Type = exports.SmbNonBrowsable = exports.SmbAccessBasedEnumeration = exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.NetworkFeatures = exports.ManagedServiceIdentityType = exports.KeySource = exports.EndpointType = exports.EncryptionType = exports.EncryptionKeySource = exports.EnableSubvolumes = exports.CoolAccessRetrievalPolicy = exports.ChownMode = exports.AvsDataStore = exports.ApplicationType = exports.AcceptGrowCapacityPoolForShortTermCloneSplit = void 0;
exports.AcceptGrowCapacityPoolForShortTermCloneSplit = {
    /**
     * Auto grow capacity pool for short term clone split is accepted.
     */
    Accepted: "Accepted",
    /**
     * Auto grow capacity pool for short term clone split is declined. Short term clone volume creation will not be allowed, to create short term clone volume accept auto grow capacity pool.
     */
    Declined: "Declined",
};
exports.ApplicationType = {
    SAP_HANA: "SAP-HANA",
    ORACLE: "ORACLE",
};
exports.AvsDataStore = {
    /**
     * avsDataStore is enabled
     */
    Enabled: "Enabled",
    /**
     * avsDataStore is disabled
     */
    Disabled: "Disabled",
};
exports.ChownMode = {
    Restricted: "Restricted",
    Unrestricted: "Unrestricted",
};
exports.CoolAccessRetrievalPolicy = {
    Default: "Default",
    OnRead: "OnRead",
    Never: "Never",
};
exports.EnableSubvolumes = {
    /**
     * subvolumes are enabled
     */
    Enabled: "Enabled",
    /**
     * subvolumes are not enabled
     */
    Disabled: "Disabled",
};
exports.EncryptionKeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.EncryptionType = {
    /**
     * EncryptionType Single, volumes will use single encryption at rest
     */
    Single: "Single",
    /**
     * EncryptionType Double, volumes will use double encryption at rest
     */
    Double: "Double",
};
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.KeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NetworkFeatures = {
    /**
     * Basic network features.
     */
    Basic: "Basic",
    /**
     * Standard network features.
     */
    Standard: "Standard",
    /**
     * Updating from Basic to Standard network features.
     */
    Basic_Standard: "Basic_Standard",
    /**
     * Updating from Standard to Basic network features.
     */
    Standard_Basic: "Standard_Basic",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
    /**
     * Zone redundant storage service level
     */
    StandardZRS: "StandardZRS",
};
exports.SmbAccessBasedEnumeration = {
    /**
     * smbAccessBasedEnumeration share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbAccessBasedEnumeration share setting is enabled
     */
    Enabled: "Enabled",
};
exports.SmbNonBrowsable = {
    /**
     * smbNonBrowsable share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbNonBrowsable share setting is enabled
     */
    Enabled: "Enabled",
};
exports.Type = {
    /**
     * Default user quota
     */
    DefaultUserQuota: "DefaultUserQuota",
    /**
     * Default group quota
     */
    DefaultGroupQuota: "DefaultGroupQuota",
    /**
     * Individual user quota
     */
    IndividualUserQuota: "IndividualUserQuota",
    /**
     * Individual group quota
     */
    IndividualGroupQuota: "IndividualGroupQuota",
};
exports.VolumeLanguage = {
    /**
     * Posix with UTF-8
     */
    C_utf_8: "c.utf-8",
    /**
     * UTF-8 with 4 byte character support
     */
    Utf8mb4: "utf8mb4",
    /**
     * Arabic - Deprecated
     */
    Ar: "ar",
    /**
     * Arabic with UTF-8
     */
    Ar_utf_8: "ar.utf-8",
    /**
     * Croatian - Deprecated
     */
    Hr: "hr",
    /**
     * Croatian with UTF-8
     */
    Hr_utf_8: "hr.utf-8",
    /**
     * Czech - Deprecated
     */
    Cs: "cs",
    /**
     * Czech with UTF-8
     */
    Cs_utf_8: "cs.utf-8",
    /**
     * Danish - Deprecated
     */
    Da: "da",
    /**
     * Danish with UTF-8
     */
    Da_utf_8: "da.utf-8",
    /**
     * Dutch - Deprecated
     */
    Nl: "nl",
    /**
     * Dutch with UTF-8
     */
    Nl_utf_8: "nl.utf-8",
    /**
     * English - Deprecated
     */
    En: "en",
    /**
     * English with UTF-8
     */
    En_utf_8: "en.utf-8",
    /**
     * Finnish - Deprecated
     */
    Fi: "fi",
    /**
     * Finnish with UTF-8
     */
    Fi_utf_8: "fi.utf-8",
    /**
     * French - Deprecated
     */
    Fr: "fr",
    /**
     * French with UTF-8
     */
    Fr_utf_8: "fr.utf-8",
    /**
     * German - Deprecated
     */
    De: "de",
    /**
     * German with UTF-8
     */
    De_utf_8: "de.utf-8",
    /**
     * Hebrew - Deprecated
     */
    He: "he",
    /**
     * Hebrew with UTF-8
     */
    He_utf_8: "he.utf-8",
    /**
     * Hungarian - Deprecated
     */
    Hu: "hu",
    /**
     * Hungarian with UTF-8
     */
    Hu_utf_8: "hu.utf-8",
    /**
     * Italian - Deprecated
     */
    It: "it",
    /**
     * Italian with UTF-8
     */
    It_utf_8: "it.utf-8",
    /**
     * Japanese euc-j - Deprecated
     */
    Ja: "ja",
    /**
     * Japanese euc-j with UTF-8
     */
    Ja_utf_8: "ja.utf-8",
    /**
     * Japanese euc-j - Deprecated
     */
    Ja_v1: "ja-v1",
    /**
     * Japanese euc-j with UTF-8
     */
    Ja_v1_utf_8: "ja-v1.utf-8",
    /**
     * Japanese pck
     */
    Ja_jp_pck: "ja-jp.pck",
    /**
     * Japanese pck with UTF-8 - Deprecated
     */
    Ja_jp_pck_utf_8: "ja-jp.pck.utf-8",
    /**
     * Japanese cp932
     */
    Ja_jp_932: "ja-jp.932",
    /**
     * Japanese cp932 with UTF-8 - Deprecated
     */
    Ja_jp_932_utf_8: "ja-jp.932.utf-8",
    /**
     * Japanese pck - sjis
     */
    Ja_jp_pck_v2: "ja-jp.pck-v2",
    /**
     * Japanese pck - sjis with UTF-8 - Deprecated
     */
    Ja_jp_pck_v2_utf_8: "ja-jp.pck-v2.utf-8",
    /**
     * Korean - Deprecated
     */
    Ko: "ko",
    /**
     * Korean with UTF-8
     */
    Ko_utf_8: "ko.utf-8",
    /**
     * Norwegian - Deprecated
     */
    No: "no",
    /**
     * Norwegian with UTF-8
     */
    No_utf_8: "no.utf-8",
    /**
     * Polish - Deprecated
     */
    Pl: "pl",
    /**
     * Polish with UTF-8
     */
    Pl_utf_8: "pl.utf-8",
    /**
     * Portuguese - Deprecated
     */
    Pt: "pt",
    /**
     * Portuguese with UTF-8
     */
    Pt_utf_8: "pt.utf-8",
    /**
     * Posix - Deprecated
     */
    C: "c",
    /**
     * Romanian - Deprecated
     */
    Ro: "ro",
    /**
     * Romanian with UTF-8
     */
    Ro_utf_8: "ro.utf-8",
    /**
     * Russian - Deprecated
     */
    Ru: "ru",
    /**
     * Russian with UTF-8
     */
    Ru_utf_8: "ru.utf-8",
    /**
     * Simplified Chinese - Deprecated
     */
    Zh: "zh",
    /**
     * Simplified Chinese with UTF-8
     */
    Zh_utf_8: "zh.utf-8",
    /**
     * Simplified gbk Chinese
     */
    Zh_gbk: "zh.gbk",
    /**
     * Simplified gbk Chinese with UTF-8 - Deprecated
     */
    Zh_gbk_utf_8: "zh.gbk.utf-8",
    /**
     * Traditional Chinese BIG 5
     */
    Zh_tw_big5: "zh-tw.big5",
    /**
     * Traditional Chinese BIG 5 with UTF-8 - Deprecated
     */
    Zh_tw_big5_utf_8: "zh-tw.big5.utf-8",
    /**
     * Traditional Chinese EUC-TW
     */
    Zh_tw: "zh-tw",
    /**
     * Traditional Chinese EUC-TW with UTF-8 - Deprecated
     */
    Zh_tw_utf_8: "zh-tw.utf-8",
    /**
     * Slovak - Deprecated
     */
    Sk: "sk",
    /**
     * Slovak with UTF-8
     */
    Sk_utf_8: "sk.utf-8",
    /**
     * Slovenian - Deprecated
     */
    Sl: "sl",
    /**
     * Slovenian with UTF-8
     */
    Sl_utf_8: "sl.utf-8",
    /**
     * Spanish - Deprecated
     */
    Es: "es",
    /**
     * Spanish with UTF-8
     */
    Es_utf_8: "es.utf-8",
    /**
     * Swedish - Deprecated
     */
    Sv: "sv",
    /**
     * Swedish with UTF-8
     */
    Sv_utf_8: "sv.utf-8",
    /**
     * Turkish - Deprecated
     */
    Tr: "tr",
    /**
     * Turkish with UTF-8
     */
    Tr_utf_8: "tr.utf-8",
    /**
     * US English - Deprecated
     */
    En_us: "en-us",
    /**
     * US English with UTF-8
     */
    En_us_utf_8: "en-us.utf-8",
};
