// @pulumi/azure-native/machinelearningservices/v20250101preview/inferenceEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InferenceEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
class InferenceEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing InferenceEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InferenceEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InferenceEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InferenceEndpoint.__pulumiType;
    }
    /**
     * Create a InferenceEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.inferenceEndpointProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inferenceEndpointProperties'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["inferenceEndpointProperties"] = args ? (args.inferenceEndpointProperties ? pulumi.output(args.inferenceEndpointProperties).apply(inputs.machinelearningservices.v20250101preview.inferenceEndpointArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["inferenceEndpointProperties"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20230801preview:InferenceEndpoint" }, { type: "azure-native:machinelearningservices/v20240101preview:InferenceEndpoint" }, { type: "azure-native:machinelearningservices/v20240401preview:InferenceEndpoint" }, { type: "azure-native:machinelearningservices/v20241001preview:InferenceEndpoint" }, { type: "azure-native:machinelearningservices:InferenceEndpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InferenceEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InferenceEndpoint = InferenceEndpoint;
/** @internal */
InferenceEndpoint.__pulumiType = 'azure-native:machinelearningservices/v20250101preview:InferenceEndpoint';
