// @pulumi/azure-native/machinelearningservices/v20250101preview/workspace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An object that represents a machine learning workspace.
 */
class Workspace extends pulumi.CustomResource {
    /**
     * Get an existing Workspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Workspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Workspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Workspace.__pulumiType;
    }
    /**
     * Create a Workspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["allowPublicAccessWhenBehindVnet"] = args ? args.allowPublicAccessWhenBehindVnet : undefined;
            resourceInputs["allowRoleAssignmentOnRG"] = args ? args.allowRoleAssignmentOnRG : undefined;
            resourceInputs["applicationInsights"] = args ? args.applicationInsights : undefined;
            resourceInputs["associatedWorkspaces"] = args ? args.associatedWorkspaces : undefined;
            resourceInputs["containerRegistries"] = args ? args.containerRegistries : undefined;
            resourceInputs["containerRegistry"] = args ? args.containerRegistry : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["discoveryUrl"] = args ? args.discoveryUrl : undefined;
            resourceInputs["enableDataIsolation"] = args ? args.enableDataIsolation : undefined;
            resourceInputs["enableServiceSideCMKEncryption"] = args ? args.enableServiceSideCMKEncryption : undefined;
            resourceInputs["enableSimplifiedCmk"] = args ? args.enableSimplifiedCmk : undefined;
            resourceInputs["enableSoftwareBillOfMaterials"] = args ? args.enableSoftwareBillOfMaterials : undefined;
            resourceInputs["encryption"] = args ? args.encryption : undefined;
            resourceInputs["existingWorkspaces"] = args ? args.existingWorkspaces : undefined;
            resourceInputs["featureStoreSettings"] = args ? args.featureStoreSettings : undefined;
            resourceInputs["friendlyName"] = args ? args.friendlyName : undefined;
            resourceInputs["hbiWorkspace"] = args ? args.hbiWorkspace : undefined;
            resourceInputs["hubResourceId"] = args ? args.hubResourceId : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["imageBuildCompute"] = args ? args.imageBuildCompute : undefined;
            resourceInputs["ipAllowlist"] = args ? args.ipAllowlist : undefined;
            resourceInputs["keyVault"] = args ? args.keyVault : undefined;
            resourceInputs["keyVaults"] = args ? args.keyVaults : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managedNetwork"] = args ? args.managedNetwork : undefined;
            resourceInputs["networkAcls"] = args ? args.networkAcls : undefined;
            resourceInputs["primaryUserAssignedIdentity"] = args ? args.primaryUserAssignedIdentity : undefined;
            resourceInputs["provisionNetworkNow"] = args ? args.provisionNetworkNow : undefined;
            resourceInputs["publicNetworkAccess"] = args ? args.publicNetworkAccess : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverlessComputeSettings"] = args ? args.serverlessComputeSettings : undefined;
            resourceInputs["serviceManagedResourcesSettings"] = args ? args.serviceManagedResourcesSettings : undefined;
            resourceInputs["sharedPrivateLinkResources"] = args ? args.sharedPrivateLinkResources : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["softDeleteRetentionInDays"] = args ? args.softDeleteRetentionInDays : undefined;
            resourceInputs["storageAccount"] = args ? args.storageAccount : undefined;
            resourceInputs["storageAccounts"] = args ? args.storageAccounts : undefined;
            resourceInputs["systemDatastoresAuthMode"] = args ? args.systemDatastoresAuthMode : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["v1LegacyMode"] = args ? args.v1LegacyMode : undefined;
            resourceInputs["workspaceHubConfig"] = args ? args.workspaceHubConfig : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["agentsEndpointUri"] = undefined /*out*/;
            resourceInputs["mlFlowTrackingUri"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notebookInfo"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["privateLinkCount"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serviceProvisionedResourceGroup"] = undefined /*out*/;
            resourceInputs["storageHnsEnabled"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workspaceId"] = undefined /*out*/;
        }
        else {
            resourceInputs["agentsEndpointUri"] = undefined /*out*/;
            resourceInputs["allowPublicAccessWhenBehindVnet"] = undefined /*out*/;
            resourceInputs["allowRoleAssignmentOnRG"] = undefined /*out*/;
            resourceInputs["applicationInsights"] = undefined /*out*/;
            resourceInputs["associatedWorkspaces"] = undefined /*out*/;
            resourceInputs["containerRegistries"] = undefined /*out*/;
            resourceInputs["containerRegistry"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["discoveryUrl"] = undefined /*out*/;
            resourceInputs["enableDataIsolation"] = undefined /*out*/;
            resourceInputs["enableServiceSideCMKEncryption"] = undefined /*out*/;
            resourceInputs["enableSimplifiedCmk"] = undefined /*out*/;
            resourceInputs["enableSoftwareBillOfMaterials"] = undefined /*out*/;
            resourceInputs["encryption"] = undefined /*out*/;
            resourceInputs["existingWorkspaces"] = undefined /*out*/;
            resourceInputs["featureStoreSettings"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["hbiWorkspace"] = undefined /*out*/;
            resourceInputs["hubResourceId"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["imageBuildCompute"] = undefined /*out*/;
            resourceInputs["ipAllowlist"] = undefined /*out*/;
            resourceInputs["keyVault"] = undefined /*out*/;
            resourceInputs["keyVaults"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedNetwork"] = undefined /*out*/;
            resourceInputs["mlFlowTrackingUri"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkAcls"] = undefined /*out*/;
            resourceInputs["notebookInfo"] = undefined /*out*/;
            resourceInputs["primaryUserAssignedIdentity"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["privateLinkCount"] = undefined /*out*/;
            resourceInputs["provisionNetworkNow"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicNetworkAccess"] = undefined /*out*/;
            resourceInputs["serverlessComputeSettings"] = undefined /*out*/;
            resourceInputs["serviceManagedResourcesSettings"] = undefined /*out*/;
            resourceInputs["serviceProvisionedResourceGroup"] = undefined /*out*/;
            resourceInputs["sharedPrivateLinkResources"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["softDeleteRetentionInDays"] = undefined /*out*/;
            resourceInputs["storageAccount"] = undefined /*out*/;
            resourceInputs["storageAccounts"] = undefined /*out*/;
            resourceInputs["storageHnsEnabled"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["systemDatastoresAuthMode"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["v1LegacyMode"] = undefined /*out*/;
            resourceInputs["workspaceHubConfig"] = undefined /*out*/;
            resourceInputs["workspaceId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20180301preview:Workspace" }, { type: "azure-native:machinelearningservices/v20181119:Workspace" }, { type: "azure-native:machinelearningservices/v20190501:Workspace" }, { type: "azure-native:machinelearningservices/v20190601:Workspace" }, { type: "azure-native:machinelearningservices/v20191101:Workspace" }, { type: "azure-native:machinelearningservices/v20200101:Workspace" }, { type: "azure-native:machinelearningservices/v20200218preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200301:Workspace" }, { type: "azure-native:machinelearningservices/v20200401:Workspace" }, { type: "azure-native:machinelearningservices/v20200501preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200515preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200601:Workspace" }, { type: "azure-native:machinelearningservices/v20200801:Workspace" }, { type: "azure-native:machinelearningservices/v20200901preview:Workspace" }, { type: "azure-native:machinelearningservices/v20210101:Workspace" }, { type: "azure-native:machinelearningservices/v20210301preview:Workspace" }, { type: "azure-native:machinelearningservices/v20210401:Workspace" }, { type: "azure-native:machinelearningservices/v20210701:Workspace" }, { type: "azure-native:machinelearningservices/v20220101preview:Workspace" }, { type: "azure-native:machinelearningservices/v20220201preview:Workspace" }, { type: "azure-native:machinelearningservices/v20220501:Workspace" }, { type: "azure-native:machinelearningservices/v20220601preview:Workspace" }, { type: "azure-native:machinelearningservices/v20221001:Workspace" }, { type: "azure-native:machinelearningservices/v20221001preview:Workspace" }, { type: "azure-native:machinelearningservices/v20221201preview:Workspace" }, { type: "azure-native:machinelearningservices/v20230201preview:Workspace" }, { type: "azure-native:machinelearningservices/v20230401:Workspace" }, { type: "azure-native:machinelearningservices/v20230401preview:Workspace" }, { type: "azure-native:machinelearningservices/v20230601preview:Workspace" }, { type: "azure-native:machinelearningservices/v20230801preview:Workspace" }, { type: "azure-native:machinelearningservices/v20231001:Workspace" }, { type: "azure-native:machinelearningservices/v20240101preview:Workspace" }, { type: "azure-native:machinelearningservices/v20240401:Workspace" }, { type: "azure-native:machinelearningservices/v20240401preview:Workspace" }, { type: "azure-native:machinelearningservices/v20240701preview:Workspace" }, { type: "azure-native:machinelearningservices/v20241001:Workspace" }, { type: "azure-native:machinelearningservices/v20241001preview:Workspace" }, { type: "azure-native:machinelearningservices:Workspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Workspace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Workspace = Workspace;
/** @internal */
Workspace.__pulumiType = 'azure-native:machinelearningservices/v20250101preview:Workspace';
