// @pulumi/azure-native/web/v20200601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppScmAllowed = exports.WebAppFtpAllowed = exports.WebAppDiagnosticLogsConfigurationSlot = exports.listSiteIdentifiersAssignedToHostNameOutput = exports.listSiteIdentifiersAssignedToHostName = exports.getWebAppScmAllowedOutput = exports.getWebAppScmAllowed = exports.getWebAppFtpAllowedOutput = exports.getWebAppFtpAllowed = exports.getWebAppDiagnosticLogsConfigurationSlotOutput = exports.getWebAppDiagnosticLogsConfigurationSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWebAppDiagnosticLogsConfigurationSlot = null;
exports.getWebAppDiagnosticLogsConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfigurationSlot", "getWebAppDiagnosticLogsConfigurationSlotOutput"], () => require("./getWebAppDiagnosticLogsConfigurationSlot"));
exports.getWebAppFtpAllowed = null;
exports.getWebAppFtpAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppFtpAllowed", "getWebAppFtpAllowedOutput"], () => require("./getWebAppFtpAllowed"));
exports.getWebAppScmAllowed = null;
exports.getWebAppScmAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppScmAllowed", "getWebAppScmAllowedOutput"], () => require("./getWebAppScmAllowed"));
exports.listSiteIdentifiersAssignedToHostName = null;
exports.listSiteIdentifiersAssignedToHostNameOutput = null;
utilities.lazyLoad(exports, ["listSiteIdentifiersAssignedToHostName", "listSiteIdentifiersAssignedToHostNameOutput"], () => require("./listSiteIdentifiersAssignedToHostName"));
exports.WebAppDiagnosticLogsConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfigurationSlot"], () => require("./webAppDiagnosticLogsConfigurationSlot"));
exports.WebAppFtpAllowed = null;
utilities.lazyLoad(exports, ["WebAppFtpAllowed"], () => require("./webAppFtpAllowed"));
exports.WebAppScmAllowed = null;
utilities.lazyLoad(exports, ["WebAppScmAllowed"], () => require("./webAppScmAllowed"));
// Export enums:
__exportStar(require("../../types/enums/web/v20200601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20200601:WebAppDiagnosticLogsConfigurationSlot":
                return new exports.WebAppDiagnosticLogsConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20200601:WebAppFtpAllowed":
                return new exports.WebAppFtpAllowed(name, undefined, { urn });
            case "azure-native:web/v20200601:WebAppScmAllowed":
                return new exports.WebAppScmAllowed(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20200601", _module);
