// @pulumi/azure-native/codesigning/certificateProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Certificate profile resource.
 *
 * Uses Azure REST API version 2024-02-05-preview.
 *
 * Other available API versions: 2024-09-30-preview.
 */
class CertificateProfile extends pulumi.CustomResource {
    /**
     * Get an existing CertificateProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CertificateProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CertificateProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CertificateProfile.__pulumiType;
    }
    /**
     * Create a CertificateProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.profileType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profileType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["identityValidationId"] = args ? args.identityValidationId : undefined;
            resourceInputs["includeCity"] = (_a = (args ? args.includeCity : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["includeCountry"] = (_b = (args ? args.includeCountry : undefined)) !== null && _b !== void 0 ? _b : false;
            resourceInputs["includePostalCode"] = (_c = (args ? args.includePostalCode : undefined)) !== null && _c !== void 0 ? _c : false;
            resourceInputs["includeState"] = (_d = (args ? args.includeState : undefined)) !== null && _d !== void 0 ? _d : false;
            resourceInputs["includeStreetAddress"] = (_e = (args ? args.includeStreetAddress : undefined)) !== null && _e !== void 0 ? _e : false;
            resourceInputs["profileName"] = args ? args.profileName : undefined;
            resourceInputs["profileType"] = args ? args.profileType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["city"] = undefined /*out*/;
            resourceInputs["commonName"] = undefined /*out*/;
            resourceInputs["country"] = undefined /*out*/;
            resourceInputs["enhancedKeyUsage"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["organizationUnit"] = undefined /*out*/;
            resourceInputs["postalCode"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["streetAddress"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["city"] = undefined /*out*/;
            resourceInputs["commonName"] = undefined /*out*/;
            resourceInputs["country"] = undefined /*out*/;
            resourceInputs["enhancedKeyUsage"] = undefined /*out*/;
            resourceInputs["identityValidationId"] = undefined /*out*/;
            resourceInputs["includeCity"] = undefined /*out*/;
            resourceInputs["includeCountry"] = undefined /*out*/;
            resourceInputs["includePostalCode"] = undefined /*out*/;
            resourceInputs["includeState"] = undefined /*out*/;
            resourceInputs["includeStreetAddress"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["organizationUnit"] = undefined /*out*/;
            resourceInputs["postalCode"] = undefined /*out*/;
            resourceInputs["profileType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["streetAddress"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:codesigning/v20240205preview:CertificateProfile" }, { type: "azure-native:codesigning/v20240930preview:CertificateProfile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CertificateProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CertificateProfile = CertificateProfile;
/** @internal */
CertificateProfile.__pulumiType = 'azure-native:codesigning:CertificateProfile';
