// @pulumi/azure-native/codesigning/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240930preview = exports.v20240205preview = exports.getCodeSigningAccountOutput = exports.getCodeSigningAccount = exports.getCertificateProfileOutput = exports.getCertificateProfile = exports.CodeSigningAccount = exports.CertificateProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CertificateProfile = null;
utilities.lazyLoad(exports, ["CertificateProfile"], () => require("./certificateProfile"));
exports.CodeSigningAccount = null;
utilities.lazyLoad(exports, ["CodeSigningAccount"], () => require("./codeSigningAccount"));
exports.getCertificateProfile = null;
exports.getCertificateProfileOutput = null;
utilities.lazyLoad(exports, ["getCertificateProfile", "getCertificateProfileOutput"], () => require("./getCertificateProfile"));
exports.getCodeSigningAccount = null;
exports.getCodeSigningAccountOutput = null;
utilities.lazyLoad(exports, ["getCodeSigningAccount", "getCodeSigningAccountOutput"], () => require("./getCodeSigningAccount"));
// Export enums:
__exportStar(require("../types/enums/codesigning"), exports);
// Export sub-modules:
const v20240205preview = require("./v20240205preview");
exports.v20240205preview = v20240205preview;
const v20240930preview = require("./v20240930preview");
exports.v20240930preview = v20240930preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:codesigning:CertificateProfile":
                return new exports.CertificateProfile(name, undefined, { urn });
            case "azure-native:codesigning:CodeSigningAccount":
                return new exports.CodeSigningAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "codesigning", _module);
