// @pulumi/azure-native/containerregistry/v20170301/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registry = exports.listRegistryCredentialsOutput = exports.listRegistryCredentials = exports.getRegistryOutput = exports.getRegistry = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getRegistry = null;
exports.getRegistryOutput = null;
utilities.lazyLoad(exports, ["getRegistry", "getRegistryOutput"], () => require("./getRegistry"));
exports.listRegistryCredentials = null;
exports.listRegistryCredentialsOutput = null;
utilities.lazyLoad(exports, ["listRegistryCredentials", "listRegistryCredentialsOutput"], () => require("./listRegistryCredentials"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerregistry/v20170301:Registry":
                return new exports.Registry(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerregistry/v20170301", _module);
