// @pulumi/azure-native/keyvault/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241201preview = exports.v20241101 = exports.v20240401preview = exports.v20230701 = exports.v20230201 = exports.Vault = exports.Secret = exports.PrivateEndpointConnection = exports.MHSMPrivateEndpointConnection = exports.ManagedHsm = exports.Key = exports.getVaultOutput = exports.getVault = exports.getSecretOutput = exports.getSecret = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getManagedHsmOutput = exports.getManagedHsm = exports.getMHSMPrivateEndpointConnectionOutput = exports.getMHSMPrivateEndpointConnection = exports.getKeyOutput = exports.getKey = exports.AccessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessPolicy = null;
utilities.lazyLoad(exports, ["AccessPolicy"], () => require("./accessPolicy"));
exports.getKey = null;
exports.getKeyOutput = null;
utilities.lazyLoad(exports, ["getKey", "getKeyOutput"], () => require("./getKey"));
exports.getMHSMPrivateEndpointConnection = null;
exports.getMHSMPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getMHSMPrivateEndpointConnection", "getMHSMPrivateEndpointConnectionOutput"], () => require("./getMHSMPrivateEndpointConnection"));
exports.getManagedHsm = null;
exports.getManagedHsmOutput = null;
utilities.lazyLoad(exports, ["getManagedHsm", "getManagedHsmOutput"], () => require("./getManagedHsm"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getVault = null;
exports.getVaultOutput = null;
utilities.lazyLoad(exports, ["getVault", "getVaultOutput"], () => require("./getVault"));
exports.Key = null;
utilities.lazyLoad(exports, ["Key"], () => require("./key"));
exports.ManagedHsm = null;
utilities.lazyLoad(exports, ["ManagedHsm"], () => require("./managedHsm"));
exports.MHSMPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["MHSMPrivateEndpointConnection"], () => require("./mhsmprivateEndpointConnection"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.Vault = null;
utilities.lazyLoad(exports, ["Vault"], () => require("./vault"));
// Export enums:
__exportStar(require("../types/enums/keyvault"), exports);
// Export sub-modules:
const v20230201 = require("./v20230201");
exports.v20230201 = v20230201;
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
const v20241201preview = require("./v20241201preview");
exports.v20241201preview = v20241201preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:keyvault:AccessPolicy":
                return new exports.AccessPolicy(name, undefined, { urn });
            case "azure-native:keyvault:Key":
                return new exports.Key(name, undefined, { urn });
            case "azure-native:keyvault:MHSMPrivateEndpointConnection":
                return new exports.MHSMPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:keyvault:ManagedHsm":
                return new exports.ManagedHsm(name, undefined, { urn });
            case "azure-native:keyvault:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:keyvault:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "azure-native:keyvault:Vault":
                return new exports.Vault(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "keyvault", _module);
