// @pulumi/azure-native/kusto/v20190907/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReadWriteDatabase = exports.ReadOnlyFollowingDatabase = exports.getReadWriteDatabaseOutput = exports.getReadWriteDatabase = exports.getReadOnlyFollowingDatabaseOutput = exports.getReadOnlyFollowingDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getReadOnlyFollowingDatabase = null;
exports.getReadOnlyFollowingDatabaseOutput = null;
utilities.lazyLoad(exports, ["getReadOnlyFollowingDatabase", "getReadOnlyFollowingDatabaseOutput"], () => require("./getReadOnlyFollowingDatabase"));
exports.getReadWriteDatabase = null;
exports.getReadWriteDatabaseOutput = null;
utilities.lazyLoad(exports, ["getReadWriteDatabase", "getReadWriteDatabaseOutput"], () => require("./getReadWriteDatabase"));
exports.ReadOnlyFollowingDatabase = null;
utilities.lazyLoad(exports, ["ReadOnlyFollowingDatabase"], () => require("./readOnlyFollowingDatabase"));
exports.ReadWriteDatabase = null;
utilities.lazyLoad(exports, ["ReadWriteDatabase"], () => require("./readWriteDatabase"));
// Export enums:
__exportStar(require("../../types/enums/kusto/v20190907"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kusto/v20190907:ReadOnlyFollowingDatabase":
                return new exports.ReadOnlyFollowingDatabase(name, undefined, { urn });
            case "azure-native:kusto/v20190907:ReadWriteDatabase":
                return new exports.ReadWriteDatabase(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kusto/v20190907", _module);
