// @pulumi/azure-native/apimanagement/v20230901preview/tagOperationLink.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagOperationLink = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Tag-operation link details.
 */
class TagOperationLink extends pulumi.CustomResource {
    /**
     * Get an existing TagOperationLink resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TagOperationLink(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TagOperationLink.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TagOperationLink.__pulumiType;
    }
    /**
     * Create a TagOperationLink resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.operationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            if ((!args || args.tagId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tagId'");
            }
            resourceInputs["operationId"] = args ? args.operationId : undefined;
            resourceInputs["operationLinkId"] = args ? args.operationLinkId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["tagId"] = args ? args.tagId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20220901preview:TagOperationLink" }, { type: "azure-native:apimanagement/v20230301preview:TagOperationLink" }, { type: "azure-native:apimanagement/v20230501preview:TagOperationLink" }, { type: "azure-native:apimanagement/v20240501:TagOperationLink" }, { type: "azure-native:apimanagement/v20240601preview:TagOperationLink" }, { type: "azure-native:apimanagement:TagOperationLink" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TagOperationLink.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TagOperationLink = TagOperationLink;
/** @internal */
TagOperationLink.__pulumiType = 'azure-native:apimanagement/v20230901preview:TagOperationLink';
