// @pulumi/azure-native/types/enums/migrate/v20230101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tool = exports.Status = exports.Purpose = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStateStatus = exports.Goal = exports.CleanupState = void 0;
exports.CleanupState = {
    None: "None",
    Started: "Started",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
};
exports.Goal = {
    Servers: "Servers",
    Databases: "Databases",
    DesktopVirtualization: "DesktopVirtualization",
    WebApplications: "WebApplications",
    DataCenter: "DataCenter",
};
exports.PrivateLinkServiceConnectionStateStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    NotSpecified: "NotSpecified",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Purpose = {
    Discovery: "Discovery",
    Assessment: "Assessment",
    Migration: "Migration",
};
exports.Status = {
    Inactive: "Inactive",
    Active: "Active",
};
exports.Tool = {
    ServerDiscovery: "ServerDiscovery",
    ServerAssessment: "ServerAssessment",
    ServerMigration: "ServerMigration",
    Cloudamize: "Cloudamize",
    Turbonomic: "Turbonomic",
    Zerto: "Zerto",
    CorentTech: "CorentTech",
    ServerAssessmentV1: "ServerAssessmentV1",
    ServerMigration_Replication: "ServerMigration_Replication",
    Carbonite: "Carbonite",
    DataMigrationAssistant: "DataMigrationAssistant",
    DatabaseMigrationService: "DatabaseMigrationService",
    Device42: "Device42",
    JetStream: "JetStream",
    RackWare: "RackWare",
    UnifyCloud: "UnifyCloud",
    Flexera: "Flexera",
    ServerDiscovery_Import: "ServerDiscovery_Import",
    Lakeside: "Lakeside",
    AppServiceMigrationAssistant: "AppServiceMigrationAssistant",
    Movere: "Movere",
    CloudSphere: "CloudSphere",
    Modernization: "Modernization",
    ServerMigration_DataReplication: "ServerMigration_DataReplication",
    Unknown: "Unknown",
};
