// @pulumi/azure-native/storagesync/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220901 = exports.v20220601 = exports.SyncGroup = exports.StorageSyncService = exports.ServerEndpoint = exports.RegisteredServer = exports.PrivateEndpointConnection = exports.getSyncGroupOutput = exports.getSyncGroup = exports.getStorageSyncServiceOutput = exports.getStorageSyncService = exports.getServerEndpointOutput = exports.getServerEndpoint = exports.getRegisteredServerOutput = exports.getRegisteredServer = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getCloudEndpointOutput = exports.getCloudEndpoint = exports.CloudEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CloudEndpoint = null;
utilities.lazyLoad(exports, ["CloudEndpoint"], () => require("./cloudEndpoint"));
exports.getCloudEndpoint = null;
exports.getCloudEndpointOutput = null;
utilities.lazyLoad(exports, ["getCloudEndpoint", "getCloudEndpointOutput"], () => require("./getCloudEndpoint"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getRegisteredServer = null;
exports.getRegisteredServerOutput = null;
utilities.lazyLoad(exports, ["getRegisteredServer", "getRegisteredServerOutput"], () => require("./getRegisteredServer"));
exports.getServerEndpoint = null;
exports.getServerEndpointOutput = null;
utilities.lazyLoad(exports, ["getServerEndpoint", "getServerEndpointOutput"], () => require("./getServerEndpoint"));
exports.getStorageSyncService = null;
exports.getStorageSyncServiceOutput = null;
utilities.lazyLoad(exports, ["getStorageSyncService", "getStorageSyncServiceOutput"], () => require("./getStorageSyncService"));
exports.getSyncGroup = null;
exports.getSyncGroupOutput = null;
utilities.lazyLoad(exports, ["getSyncGroup", "getSyncGroupOutput"], () => require("./getSyncGroup"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.RegisteredServer = null;
utilities.lazyLoad(exports, ["RegisteredServer"], () => require("./registeredServer"));
exports.ServerEndpoint = null;
utilities.lazyLoad(exports, ["ServerEndpoint"], () => require("./serverEndpoint"));
exports.StorageSyncService = null;
utilities.lazyLoad(exports, ["StorageSyncService"], () => require("./storageSyncService"));
exports.SyncGroup = null;
utilities.lazyLoad(exports, ["SyncGroup"], () => require("./syncGroup"));
// Export enums:
__exportStar(require("../types/enums/storagesync"), exports);
// Export sub-modules:
const v20220601 = require("./v20220601");
exports.v20220601 = v20220601;
const v20220901 = require("./v20220901");
exports.v20220901 = v20220901;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storagesync:CloudEndpoint":
                return new exports.CloudEndpoint(name, undefined, { urn });
            case "azure-native:storagesync:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:storagesync:RegisteredServer":
                return new exports.RegisteredServer(name, undefined, { urn });
            case "azure-native:storagesync:ServerEndpoint":
                return new exports.ServerEndpoint(name, undefined, { urn });
            case "azure-native:storagesync:StorageSyncService":
                return new exports.StorageSyncService(name, undefined, { urn });
            case "azure-native:storagesync:SyncGroup":
                return new exports.SyncGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storagesync", _module);
