// @pulumi/azure-native/securityinsights/hunt.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hunt = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a Hunt in Azure Security Insights.
 *
 * Uses Azure REST API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-12-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview.
 */
class Hunt extends pulumi.CustomResource {
    /**
     * Get an existing Hunt resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Hunt(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Hunt.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Hunt.__pulumiType;
    }
    /**
     * Create a Hunt resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.description === undefined) && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["attackTactics"] = args ? args.attackTactics : undefined;
            resourceInputs["attackTechniques"] = args ? args.attackTechniques : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["huntId"] = args ? args.huntId : undefined;
            resourceInputs["hypothesisStatus"] = (_a = (args ? args.hypothesisStatus : undefined)) !== null && _a !== void 0 ? _a : "Unknown";
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["owner"] = args ? args.owner : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["status"] = (_b = (args ? args.status : undefined)) !== null && _b !== void 0 ? _b : "New";
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["attackTactics"] = undefined /*out*/;
            resourceInputs["attackTechniques"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["hypothesisStatus"] = undefined /*out*/;
            resourceInputs["labels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["owner"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:Hunt" }, { type: "azure-native:securityinsights/v20230501preview:Hunt" }, { type: "azure-native:securityinsights/v20230601preview:Hunt" }, { type: "azure-native:securityinsights/v20230701preview:Hunt" }, { type: "azure-native:securityinsights/v20230801preview:Hunt" }, { type: "azure-native:securityinsights/v20230901preview:Hunt" }, { type: "azure-native:securityinsights/v20231001preview:Hunt" }, { type: "azure-native:securityinsights/v20231201preview:Hunt" }, { type: "azure-native:securityinsights/v20240101preview:Hunt" }, { type: "azure-native:securityinsights/v20240401preview:Hunt" }, { type: "azure-native:securityinsights/v20241001preview:Hunt" }, { type: "azure-native:securityinsights/v20250101preview:Hunt" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Hunt.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Hunt = Hunt;
/** @internal */
Hunt.__pulumiType = 'azure-native:securityinsights:Hunt';
