// @pulumi/azure-native/securityinsights/workspaceManagerConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceManagerConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The workspace manager configuration
 *
 * Uses Azure REST API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-12-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview.
 */
class WorkspaceManagerConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing WorkspaceManagerConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspaceManagerConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspaceManagerConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceManagerConfiguration.__pulumiType;
    }
    /**
     * Create a WorkspaceManagerConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.mode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mode'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["mode"] = args ? args.mode : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceManagerConfigurationName"] = args ? args.workspaceManagerConfigurationName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20230501preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20230601preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20230701preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20230801preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20230901preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20231001preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20231201preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20240101preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20240401preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20241001preview:WorkspaceManagerConfiguration" }, { type: "azure-native:securityinsights/v20250101preview:WorkspaceManagerConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkspaceManagerConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkspaceManagerConfiguration = WorkspaceManagerConfiguration;
/** @internal */
WorkspaceManagerConfiguration.__pulumiType = 'azure-native:securityinsights:WorkspaceManagerConfiguration';
