// @pulumi/azure-native/machinelearningservices/environmentContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2023-04-01. In version 1.x of the Azure Native provider, it used API version 2021-03-01-preview.
 *
 * Other available API versions: 2021-03-01-preview, 2022-02-01-preview, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-04-01-preview, 2024-07-01-preview, 2024-10-01, 2024-10-01-preview, 2025-01-01-preview.
 */
class EnvironmentContainer extends pulumi.CustomResource {
    /**
     * Get an existing EnvironmentContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnvironmentContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnvironmentContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentContainer.__pulumiType;
    }
    /**
     * Create a EnvironmentContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environmentContainerProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentContainerProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["environmentContainerProperties"] = args ? (args.environmentContainerProperties ? pulumi.output(args.environmentContainerProperties).apply(inputs.machinelearningservices.environmentContainerArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["environmentContainerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20220201preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20220501:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20220601preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20221001:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20221001preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20221201preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20230201preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20230401:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20230401preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20230601preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20230801preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20231001:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20240101preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20240401:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20240401preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20240701preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20241001:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20241001preview:EnvironmentContainer" }, { type: "azure-native:machinelearningservices/v20250101preview:EnvironmentContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EnvironmentContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EnvironmentContainer = EnvironmentContainer;
/** @internal */
EnvironmentContainer.__pulumiType = 'azure-native:machinelearningservices:EnvironmentContainer';
