// @pulumi/azure-native/machinelearningservices/featurestoreEntityVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeaturestoreEntityVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2023-04-01-preview.
 *
 * Other available API versions: 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-04-01-preview, 2024-07-01-preview, 2024-10-01, 2024-10-01-preview, 2025-01-01-preview.
 */
class FeaturestoreEntityVersion extends pulumi.CustomResource {
    /**
     * Get an existing FeaturestoreEntityVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FeaturestoreEntityVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FeaturestoreEntityVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FeaturestoreEntityVersion.__pulumiType;
    }
    /**
     * Create a FeaturestoreEntityVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.featurestoreEntityVersionProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'featurestoreEntityVersionProperties'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["featurestoreEntityVersionProperties"] = args ? (args.featurestoreEntityVersionProperties ? pulumi.output(args.featurestoreEntityVersionProperties).apply(inputs.machinelearningservices.featurestoreEntityVersionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["featurestoreEntityVersionProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20230201preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20230401preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20230601preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20230801preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20231001:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20240101preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20240401:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20240401preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20240701preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20241001:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20241001preview:FeaturestoreEntityVersion" }, { type: "azure-native:machinelearningservices/v20250101preview:FeaturestoreEntityVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FeaturestoreEntityVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FeaturestoreEntityVersion = FeaturestoreEntityVersion;
/** @internal */
FeaturestoreEntityVersion.__pulumiType = 'azure-native:machinelearningservices:FeaturestoreEntityVersion';
