// @pulumi/azure-native/machinelearningservices/labelingJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelingJob = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2023-04-01-preview. In version 1.x of the Azure Native provider, it used API version 2020-09-01-preview.
 *
 * Other available API versions: 2020-09-01-preview, 2021-03-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2024-01-01-preview, 2024-04-01-preview.
 */
class LabelingJob extends pulumi.CustomResource {
    /**
     * Get an existing LabelingJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LabelingJob(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LabelingJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LabelingJob.__pulumiType;
    }
    /**
     * Create a LabelingJob resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labelingJobProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labelingJobProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["labelingJobProperties"] = args ? (args.labelingJobProperties ? pulumi.output(args.labelingJobProperties).apply(inputs.machinelearningservices.labelingJobArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["labelingJobProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20200901preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20210301preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20220601preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20221001preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20221201preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20230201preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20230401preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20230601preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20230801preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20240101preview:LabelingJob" }, { type: "azure-native:machinelearningservices/v20240401preview:LabelingJob" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LabelingJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LabelingJob = LabelingJob;
/** @internal */
LabelingJob.__pulumiType = 'azure-native:machinelearningservices:LabelingJob';
