// @pulumi/azure-native/machinelearningservices/registryEnvironmentVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryEnvironmentVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2023-04-01.
 *
 * Other available API versions: 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-04-01-preview, 2024-07-01-preview, 2024-10-01, 2024-10-01-preview, 2025-01-01-preview.
 */
class RegistryEnvironmentVersion extends pulumi.CustomResource {
    /**
     * Get an existing RegistryEnvironmentVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RegistryEnvironmentVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryEnvironmentVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryEnvironmentVersion.__pulumiType;
    }
    /**
     * Create a RegistryEnvironmentVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentName'");
            }
            if ((!args || args.environmentVersionProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentVersionProperties'");
            }
            if ((!args || args.registryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registryName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["environmentVersionProperties"] = args ? (args.environmentVersionProperties ? pulumi.output(args.environmentVersionProperties).apply(inputs.machinelearningservices.environmentVersionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["registryName"] = args ? args.registryName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["environmentVersionProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20221001preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20221201preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230201preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230401:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230401preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230601preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20230801preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20231001:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240101preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240401:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240401preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20240701preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20241001:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20241001preview:RegistryEnvironmentVersion" }, { type: "azure-native:machinelearningservices/v20250101preview:RegistryEnvironmentVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RegistryEnvironmentVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegistryEnvironmentVersion = RegistryEnvironmentVersion;
/** @internal */
RegistryEnvironmentVersion.__pulumiType = 'azure-native:machinelearningservices:RegistryEnvironmentVersion';
