// @pulumi/azure-native/authorization/v20211201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeAccessReviewScheduleDefinitionById = exports.ScopeAccessReviewHistoryDefinitionById = exports.getScopeAccessReviewScheduleDefinitionByIdOutput = exports.getScopeAccessReviewScheduleDefinitionById = exports.getScopeAccessReviewHistoryDefinitionByIdOutput = exports.getScopeAccessReviewHistoryDefinitionById = exports.getAccessReviewScheduleDefinitionByIdOutput = exports.getAccessReviewScheduleDefinitionById = exports.getAccessReviewHistoryDefinitionByIdOutput = exports.getAccessReviewHistoryDefinitionById = exports.AccessReviewScheduleDefinitionById = exports.AccessReviewHistoryDefinitionById = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AccessReviewHistoryDefinitionById = null;
utilities.lazyLoad(exports, ["AccessReviewHistoryDefinitionById"], () => require("./accessReviewHistoryDefinitionById"));
exports.AccessReviewScheduleDefinitionById = null;
utilities.lazyLoad(exports, ["AccessReviewScheduleDefinitionById"], () => require("./accessReviewScheduleDefinitionById"));
exports.getAccessReviewHistoryDefinitionById = null;
exports.getAccessReviewHistoryDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getAccessReviewHistoryDefinitionById", "getAccessReviewHistoryDefinitionByIdOutput"], () => require("./getAccessReviewHistoryDefinitionById"));
exports.getAccessReviewScheduleDefinitionById = null;
exports.getAccessReviewScheduleDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getAccessReviewScheduleDefinitionById", "getAccessReviewScheduleDefinitionByIdOutput"], () => require("./getAccessReviewScheduleDefinitionById"));
exports.getScopeAccessReviewHistoryDefinitionById = null;
exports.getScopeAccessReviewHistoryDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getScopeAccessReviewHistoryDefinitionById", "getScopeAccessReviewHistoryDefinitionByIdOutput"], () => require("./getScopeAccessReviewHistoryDefinitionById"));
exports.getScopeAccessReviewScheduleDefinitionById = null;
exports.getScopeAccessReviewScheduleDefinitionByIdOutput = null;
utilities.lazyLoad(exports, ["getScopeAccessReviewScheduleDefinitionById", "getScopeAccessReviewScheduleDefinitionByIdOutput"], () => require("./getScopeAccessReviewScheduleDefinitionById"));
exports.ScopeAccessReviewHistoryDefinitionById = null;
utilities.lazyLoad(exports, ["ScopeAccessReviewHistoryDefinitionById"], () => require("./scopeAccessReviewHistoryDefinitionById"));
exports.ScopeAccessReviewScheduleDefinitionById = null;
utilities.lazyLoad(exports, ["ScopeAccessReviewScheduleDefinitionById"], () => require("./scopeAccessReviewScheduleDefinitionById"));
// Export enums:
__exportStar(require("../../types/enums/authorization/v20211201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:authorization/v20211201preview:AccessReviewHistoryDefinitionById":
                return new exports.AccessReviewHistoryDefinitionById(name, undefined, { urn });
            case "azure-native:authorization/v20211201preview:AccessReviewScheduleDefinitionById":
                return new exports.AccessReviewScheduleDefinitionById(name, undefined, { urn });
            case "azure-native:authorization/v20211201preview:ScopeAccessReviewHistoryDefinitionById":
                return new exports.ScopeAccessReviewHistoryDefinitionById(name, undefined, { urn });
            case "azure-native:authorization/v20211201preview:ScopeAccessReviewScheduleDefinitionById":
                return new exports.ScopeAccessReviewScheduleDefinitionById(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "authorization/v20211201preview", _module);
