// @pulumi/azure-native/devices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250201preview = exports.v20230630preview = exports.v20230630 = exports.v20230301preview = exports.v20221212 = exports.v20221115preview = exports.v20220430preview = exports.v20211015 = exports.v20210303preview = exports.v20200901preview = exports.v20200401 = exports.PrivateEndpointConnection = exports.listIotHubResourceKeysForKeyNameOutput = exports.listIotHubResourceKeysForKeyName = exports.listIotHubResourceKeysOutput = exports.listIotHubResourceKeys = exports.listIotDpsResourceKeysForKeyNameOutput = exports.listIotDpsResourceKeysForKeyName = exports.listIotDpsResourceKeysOutput = exports.listIotDpsResourceKeys = exports.IotHubResourceEventHubConsumerGroup = exports.IotHubResource = exports.IotDpsResourcePrivateEndpointConnection = exports.IotDpsResource = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getIotHubResourceEventHubConsumerGroupOutput = exports.getIotHubResourceEventHubConsumerGroup = exports.getIotHubResourceOutput = exports.getIotHubResource = exports.getIotDpsResourcePrivateEndpointConnectionOutput = exports.getIotDpsResourcePrivateEndpointConnection = exports.getIotDpsResourceOutput = exports.getIotDpsResource = exports.getDpsCertificateOutput = exports.getDpsCertificate = exports.getCertificateOutput = exports.getCertificate = exports.DpsCertificate = exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.DpsCertificate = null;
utilities.lazyLoad(exports, ["DpsCertificate"], () => require("./dpsCertificate"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getDpsCertificate = null;
exports.getDpsCertificateOutput = null;
utilities.lazyLoad(exports, ["getDpsCertificate", "getDpsCertificateOutput"], () => require("./getDpsCertificate"));
exports.getIotDpsResource = null;
exports.getIotDpsResourceOutput = null;
utilities.lazyLoad(exports, ["getIotDpsResource", "getIotDpsResourceOutput"], () => require("./getIotDpsResource"));
exports.getIotDpsResourcePrivateEndpointConnection = null;
exports.getIotDpsResourcePrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getIotDpsResourcePrivateEndpointConnection", "getIotDpsResourcePrivateEndpointConnectionOutput"], () => require("./getIotDpsResourcePrivateEndpointConnection"));
exports.getIotHubResource = null;
exports.getIotHubResourceOutput = null;
utilities.lazyLoad(exports, ["getIotHubResource", "getIotHubResourceOutput"], () => require("./getIotHubResource"));
exports.getIotHubResourceEventHubConsumerGroup = null;
exports.getIotHubResourceEventHubConsumerGroupOutput = null;
utilities.lazyLoad(exports, ["getIotHubResourceEventHubConsumerGroup", "getIotHubResourceEventHubConsumerGroupOutput"], () => require("./getIotHubResourceEventHubConsumerGroup"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.IotDpsResource = null;
utilities.lazyLoad(exports, ["IotDpsResource"], () => require("./iotDpsResource"));
exports.IotDpsResourcePrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["IotDpsResourcePrivateEndpointConnection"], () => require("./iotDpsResourcePrivateEndpointConnection"));
exports.IotHubResource = null;
utilities.lazyLoad(exports, ["IotHubResource"], () => require("./iotHubResource"));
exports.IotHubResourceEventHubConsumerGroup = null;
utilities.lazyLoad(exports, ["IotHubResourceEventHubConsumerGroup"], () => require("./iotHubResourceEventHubConsumerGroup"));
exports.listIotDpsResourceKeys = null;
exports.listIotDpsResourceKeysOutput = null;
utilities.lazyLoad(exports, ["listIotDpsResourceKeys", "listIotDpsResourceKeysOutput"], () => require("./listIotDpsResourceKeys"));
exports.listIotDpsResourceKeysForKeyName = null;
exports.listIotDpsResourceKeysForKeyNameOutput = null;
utilities.lazyLoad(exports, ["listIotDpsResourceKeysForKeyName", "listIotDpsResourceKeysForKeyNameOutput"], () => require("./listIotDpsResourceKeysForKeyName"));
exports.listIotHubResourceKeys = null;
exports.listIotHubResourceKeysOutput = null;
utilities.lazyLoad(exports, ["listIotHubResourceKeys", "listIotHubResourceKeysOutput"], () => require("./listIotHubResourceKeys"));
exports.listIotHubResourceKeysForKeyName = null;
exports.listIotHubResourceKeysForKeyNameOutput = null;
utilities.lazyLoad(exports, ["listIotHubResourceKeysForKeyName", "listIotHubResourceKeysForKeyNameOutput"], () => require("./listIotHubResourceKeysForKeyName"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/devices"), exports);
// Export sub-modules:
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
const v20210303preview = require("./v20210303preview");
exports.v20210303preview = v20210303preview;
const v20211015 = require("./v20211015");
exports.v20211015 = v20211015;
const v20220430preview = require("./v20220430preview");
exports.v20220430preview = v20220430preview;
const v20221115preview = require("./v20221115preview");
exports.v20221115preview = v20221115preview;
const v20221212 = require("./v20221212");
exports.v20221212 = v20221212;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230630 = require("./v20230630");
exports.v20230630 = v20230630;
const v20230630preview = require("./v20230630preview");
exports.v20230630preview = v20230630preview;
const v20250201preview = require("./v20250201preview");
exports.v20250201preview = v20250201preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:devices:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:devices:DpsCertificate":
                return new exports.DpsCertificate(name, undefined, { urn });
            case "azure-native:devices:IotDpsResource":
                return new exports.IotDpsResource(name, undefined, { urn });
            case "azure-native:devices:IotDpsResourcePrivateEndpointConnection":
                return new exports.IotDpsResourcePrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:devices:IotHubResource":
                return new exports.IotHubResource(name, undefined, { urn });
            case "azure-native:devices:IotHubResourceEventHubConsumerGroup":
                return new exports.IotHubResourceEventHubConsumerGroup(name, undefined, { urn });
            case "azure-native:devices:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "devices", _module);
