// @pulumi/azure-native/edgeorder/v20211201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderItemByName = exports.listProductFamiliesOutput = exports.listProductFamilies = exports.listConfigurationsOutput = exports.listConfigurations = exports.getOrderItemByNameOutput = exports.getOrderItemByName = exports.getAddressByNameOutput = exports.getAddressByName = exports.AddressByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AddressByName = null;
utilities.lazyLoad(exports, ["AddressByName"], () => require("./addressByName"));
exports.getAddressByName = null;
exports.getAddressByNameOutput = null;
utilities.lazyLoad(exports, ["getAddressByName", "getAddressByNameOutput"], () => require("./getAddressByName"));
exports.getOrderItemByName = null;
exports.getOrderItemByNameOutput = null;
utilities.lazyLoad(exports, ["getOrderItemByName", "getOrderItemByNameOutput"], () => require("./getOrderItemByName"));
exports.listConfigurations = null;
exports.listConfigurationsOutput = null;
utilities.lazyLoad(exports, ["listConfigurations", "listConfigurationsOutput"], () => require("./listConfigurations"));
exports.listProductFamilies = null;
exports.listProductFamiliesOutput = null;
utilities.lazyLoad(exports, ["listProductFamilies", "listProductFamiliesOutput"], () => require("./listProductFamilies"));
exports.OrderItemByName = null;
utilities.lazyLoad(exports, ["OrderItemByName"], () => require("./orderItemByName"));
// Export enums:
__exportStar(require("../../types/enums/edgeorder/v20211201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:edgeorder/v20211201:AddressByName":
                return new exports.AddressByName(name, undefined, { urn });
            case "azure-native:edgeorder/v20211201:OrderItemByName":
                return new exports.OrderItemByName(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "edgeorder/v20211201", _module);
