// @pulumi/azure-native/machinelearningservices/v20230801preview/onlineDeployment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnlineDeployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class OnlineDeployment extends pulumi.CustomResource {
    /**
     * Get an existing OnlineDeployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OnlineDeployment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OnlineDeployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OnlineDeployment.__pulumiType;
    }
    /**
     * Create a OnlineDeployment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.endpointName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointName'");
            }
            if ((!args || args.onlineDeploymentProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'onlineDeploymentProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["deploymentName"] = args ? args.deploymentName : undefined;
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["onlineDeploymentProperties"] = args ? args.onlineDeploymentProperties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["onlineDeploymentProperties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20220201preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20220501:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20220601preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20221001:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20221001preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20221201preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20230201preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20230401:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20230401preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20230601preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20231001:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20240101preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20240401:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20240401preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20240701preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20241001:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20241001preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices/v20250101preview:OnlineDeployment" }, { type: "azure-native:machinelearningservices:OnlineDeployment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OnlineDeployment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OnlineDeployment = OnlineDeployment;
/** @internal */
OnlineDeployment.__pulumiType = 'azure-native:machinelearningservices/v20230801preview:OnlineDeployment';
