// @pulumi/gcp/kms/ekmConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EkmConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * `Ekm Connections` are used to control the connection settings for an `EXTERNAL_VPC` CryptoKey.
 * It is used to connect customer's external key manager to Google Cloud EKM.
 *
 * > **Note:** Ekm Connections cannot be deleted from Google Cloud Platform.
 *
 * To get more information about EkmConnection, see:
 *
 * * [API documentation](https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.ekmConnections)
 * * How-to Guides
 *     * [Creating a Ekm Connection](https://cloud.google.com/kms/docs/create-ekm-connection)
 *
 * ## Example Usage
 *
 * ### Kms Ekm Connection Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example_ekmconnection = new gcp.kms.EkmConnection("example-ekmconnection", {
 *     name: "ekmconnection_example",
 *     location: "us-central1",
 *     keyManagementMode: "MANUAL",
 *     serviceResolvers: [{
 *         serviceDirectoryService: "projects/project_id/locations/us-central1/namespaces/namespace_name/services/service_name",
 *         hostname: "example-ekm.goog",
 *         serverCertificates: [{
 *             rawDer: "==HAwIBCCAr6gAwIBAgIUWR+EV4lqiV7Ql12VY==",
 *         }],
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * EkmConnection can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/ekmConnections/{{name}}`
 *
 * * `{{project}}/{{location}}/{{name}}`
 *
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, EkmConnection can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:kms/ekmConnection:EkmConnection default projects/{{project}}/locations/{{location}}/ekmConnections/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:kms/ekmConnection:EkmConnection default {{project}}/{{location}}/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:kms/ekmConnection:EkmConnection default {{location}}/{{name}}
 * ```
 */
class EkmConnection extends pulumi.CustomResource {
    /**
     * Get an existing EkmConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EkmConnection(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EkmConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EkmConnection.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["cryptoSpacePath"] = state ? state.cryptoSpacePath : undefined;
            resourceInputs["etag"] = state ? state.etag : undefined;
            resourceInputs["keyManagementMode"] = state ? state.keyManagementMode : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["serviceResolvers"] = state ? state.serviceResolvers : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if ((!args || args.serviceResolvers === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceResolvers'");
            }
            resourceInputs["cryptoSpacePath"] = args ? args.cryptoSpacePath : undefined;
            resourceInputs["etag"] = args ? args.etag : undefined;
            resourceInputs["keyManagementMode"] = args ? args.keyManagementMode : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["serviceResolvers"] = args ? args.serviceResolvers : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EkmConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EkmConnection = EkmConnection;
/** @internal */
EkmConnection.__pulumiType = 'gcp:kms/ekmConnection:EkmConnection';
