// @pulumi/gcp/kms/keyHandle.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyHandle = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ### Kms Key Handle Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumi/time";
 *
 * // Create Folder in GCP Organization
 * const autokmsFolder = new gcp.organizations.Folder("autokms_folder", {
 *     displayName: "folder-example",
 *     parent: "organizations/123456789",
 * });
 * // Create the key project
 * const keyProject = new gcp.organizations.Project("key_project", {
 *     projectId: "key-proj",
 *     name: "key-proj",
 *     folderId: autokmsFolder.folderId,
 *     billingAccount: "000000-0000000-0000000-000000",
 * }, {
 *     dependsOn: [autokmsFolder],
 * });
 * // Create the resource project
 * const resourceProject = new gcp.organizations.Project("resource_project", {
 *     projectId: "resources",
 *     name: "resources",
 *     folderId: autokmsFolder.folderId,
 *     billingAccount: "000000-0000000-0000000-000000",
 * }, {
 *     dependsOn: [autokmsFolder],
 * });
 * // Enable the Cloud KMS API
 * const kmsApiService = new gcp.projects.Service("kms_api_service", {
 *     service: "cloudkms.googleapis.com",
 *     project: keyProject.projectId,
 *     disableOnDestroy: false,
 *     disableDependentServices: true,
 * }, {
 *     dependsOn: [keyProject],
 * });
 * // Wait delay after enabling APIs
 * const waitEnableServiceApi = new time.index.Sleep("wait_enable_service_api", {createDuration: "30s"}, {
 *     dependsOn: [kmsApiService],
 * });
 * //Create KMS Service Agent
 * const kmsServiceAgent = new gcp.projects.ServiceIdentity("kms_service_agent", {
 *     service: "cloudkms.googleapis.com",
 *     project: keyProject.number,
 * }, {
 *     dependsOn: [waitEnableServiceApi],
 * });
 * // Wait delay after creating service agent.
 * const waitServiceAgent = new time.index.Sleep("wait_service_agent", {createDuration: "10s"}, {
 *     dependsOn: [kmsServiceAgent],
 * });
 * //Grant the KMS Service Agent the Cloud KMS Admin role
 * const autokeyProjectAdmin = new gcp.projects.IAMMember("autokey_project_admin", {
 *     project: keyProject.projectId,
 *     role: "roles/cloudkms.admin",
 *     member: pulumi.interpolate`serviceAccount:service-${keyProject.number}@gcp-sa-cloudkms.iam.gserviceaccount.com`,
 * }, {
 *     dependsOn: [waitServiceAgent],
 * });
 * // Wait delay after granting IAM permissions
 * const waitSrvAccPermissions = new time.index.Sleep("wait_srv_acc_permissions", {createDuration: "10s"}, {
 *     dependsOn: [autokeyProjectAdmin],
 * });
 * const autokeyConfig = new gcp.kms.AutokeyConfig("autokey_config", {
 *     folder: autokmsFolder.folderId,
 *     keyProject: pulumi.interpolate`projects/${keyProject.projectId}`,
 * }, {
 *     dependsOn: [waitSrvAccPermissions],
 * });
 * // Wait delay for autokey config to take effect
 * const waitAutokeyConfig = new time.index.Sleep("wait_autokey_config", {createDuration: "10s"}, {
 *     dependsOn: [autokeyConfig],
 * });
 * const example_keyhandle = new gcp.kms.KeyHandle("example-keyhandle", {
 *     project: resourceProject.projectId,
 *     name: "example-key-handle",
 *     location: "global",
 *     resourceTypeSelector: "storage.googleapis.com/Bucket",
 * }, {
 *     dependsOn: [waitAutokeyConfig],
 * });
 * ```
 *
 * ## Import
 *
 * KeyHandle can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/keyHandles/{{name}}`
 *
 * * `{{project}}/{{location}}/{{name}}`
 *
 * * `{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, KeyHandle can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:kms/keyHandle:KeyHandle default projects/{{project}}/locations/{{location}}/keyHandles/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:kms/keyHandle:KeyHandle default {{project}}/{{location}}/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:kms/keyHandle:KeyHandle default {{location}}/{{name}}
 * ```
 */
class KeyHandle extends pulumi.CustomResource {
    /**
     * Get an existing KeyHandle resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KeyHandle(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KeyHandle.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KeyHandle.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["kmsKey"] = state ? state.kmsKey : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["resourceTypeSelector"] = state ? state.resourceTypeSelector : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if ((!args || args.resourceTypeSelector === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceTypeSelector'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["resourceTypeSelector"] = args ? args.resourceTypeSelector : undefined;
            resourceInputs["kmsKey"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(KeyHandle.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KeyHandle = KeyHandle;
/** @internal */
KeyHandle.__pulumiType = 'gcp:kms/keyHandle:KeyHandle';
