// @pulumi/azure-native/types/enums/migrate/v20220501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadType = exports.WorkloadDeploymentType = exports.WorkloadDeploymentTarget = exports.TargetStorageProjectionType = exports.TargetStorageAccessType = exports.TargetHydrationStorageProviderType = exports.SecretStoreType = exports.ResourceIdentityTypes = exports.OperatingSystemType = exports.LoadBalancerType = exports.ConfigurationType = exports.AutomationArtifactStatus = void 0;
exports.AutomationArtifactStatus = {
    NotGenerated: "NotGenerated",
    Generated: "Generated",
};
exports.ConfigurationType = {
    IISConnectionString: "IISConnectionString",
    IISAuthentication: "IISAuthentication",
    ApacheTomcatContextResource: "ApacheTomcatContextResource",
};
exports.LoadBalancerType = {
    Private: "Private",
    Public: "Public",
};
exports.OperatingSystemType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ResourceIdentityTypes = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.SecretStoreType = {
    None: "None",
    KubeSecret: "KubeSecret",
    KeyVaultSecret: "KeyVaultSecret",
    AppServiceAppSettings: "AppServiceAppSettings",
};
exports.TargetHydrationStorageProviderType = {
    AzureFileShare: "AzureFileShare",
};
exports.TargetStorageAccessType = {
    Shared: "Shared",
    Exclusive: "Exclusive",
};
exports.TargetStorageProjectionType = {
    ContainerFileSystem: "ContainerFileSystem",
    PersistentVolume: "PersistentVolume",
};
exports.WorkloadDeploymentTarget = {
    AzureKubernetesService: "AzureKubernetesService",
    AzureAppServiceContainer: "AzureAppServiceContainer",
    AzureAppServiceNative: "AzureAppServiceNative",
};
exports.WorkloadDeploymentType = {
    IISAKSWorkloadDeployment: "IISAKSWorkloadDeployment",
    ApacheTomcatAKSWorkloadDeployment: "ApacheTomcatAKSWorkloadDeployment",
};
exports.WorkloadType = {
    IISWorkload: "IISWorkload",
    ApacheTomcatWorkload: "ApacheTomcatWorkload",
};
