// @pulumi/azure-native/types/enums/devopsinfrastructure/v20240404preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountType = exports.ResourcePredictionsProfileType = exports.ProvisioningState = exports.PredictionPreference = exports.OsDiskStorageAccountType = exports.ManagedServiceIdentityType = exports.LogonType = exports.CachingType = exports.AzureDevOpsPermissionType = void 0;
exports.AzureDevOpsPermissionType = {
    /**
     * Pool will inherit permissions from the project or organization.
     */
    Inherit: "Inherit",
    /**
     * Only the pool creator will be an admin of the pool.
     */
    CreatorOnly: "CreatorOnly",
    /**
     * Only the specified accounts will be admins of the pool.
     */
    SpecificAccounts: "SpecificAccounts",
};
exports.CachingType = {
    /**
     * Don't use host caching.
     */
    None: "None",
    /**
     * For workloads that only do read operations.
     */
    ReadOnly: "ReadOnly",
    /**
     * For workloads that do a balance of read and write operations.
     */
    ReadWrite: "ReadWrite",
};
exports.LogonType = {
    /**
     * Run as a service.
     */
    Service: "Service",
    /**
     * Run in interactive mode.
     */
    Interactive: "Interactive",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.OsDiskStorageAccountType = {
    /**
     * Standard OS disk type.
     */
    Standard: "Standard",
    /**
     * Premium OS disk type.
     */
    Premium: "Premium",
    /**
     * Standard SSD OS disk type.
     */
    StandardSSD: "StandardSSD",
};
exports.PredictionPreference = {
    /**
     * Balance between cost and performance.
     */
    Balanced: "Balanced",
    /**
     * Optimizes for cost over performance.
     */
    MostCostEffective: "MostCostEffective",
    /**
     * Halfway through cost and balanced.
     */
    MoreCostEffective: "MoreCostEffective",
    /**
     * Halfway through balanced and performance.
     */
    MorePerformance: "MorePerformance",
    /**
     * Optimizes for performance over cost.
     */
    BestPerformance: "BestPerformance",
};
exports.ProvisioningState = {
    /**
     * Represents a succeeded operation.
     */
    Succeeded: "Succeeded",
    /**
     * Represents a failed operation.
     */
    Failed: "Failed",
    /**
     * Represents a canceled operation.
     */
    Canceled: "Canceled",
    /**
     * Represents a pending operation.
     */
    Provisioning: "Provisioning",
    /**
     * Represents a pending operation.
     */
    Updating: "Updating",
    /**
     * Represents an operation under deletion.
     */
    Deleting: "Deleting",
    /**
     * Represents an accepted operation.
     */
    Accepted: "Accepted",
};
exports.ResourcePredictionsProfileType = {
    /**
     * Customer provides the stand-by agent scheme.
     */
    Manual: "Manual",
    /**
     * The stand-by agent scheme is determined based on historical demand.
     */
    Automatic: "Automatic",
};
exports.StorageAccountType = {
    /**
     * The data disk should use standard locally redundant storage.
     */
    StandardLRS: "Standard_LRS",
    /**
     * The data disk should use premium locally redundant storage.
     */
    PremiumLRS: "Premium_LRS",
    /**
     * The data disk should use standard SSD locally redundant storage.
     */
    StandardSSDLRS: "StandardSSD_LRS",
    /**
     * The data disk should use premium SSD zonal redundant storage.
     */
    PremiumZRS: "Premium_ZRS",
    /**
     * The data disk should use standard SSD zonal redundant storage.
     */
    StandardSSDZRS: "StandardSSD_ZRS",
};
