// @pulumi/azure-native/types/enums/hybridcontainerservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.OsType = exports.NetworkPolicy = exports.Mode = exports.LoadBalancerSku = exports.LicenseType = exports.AutoUpgradeOptions = exports.v20240101 = exports.v20231115preview = exports.v20220901preview = exports.v20220501preview = void 0;
// Export sub-modules:
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20220901preview = require("./v20220901preview");
exports.v20220901preview = v20220901preview;
const v20231115preview = require("./v20231115preview");
exports.v20231115preview = v20231115preview;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
exports.AutoUpgradeOptions = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.LicenseType = {
    Windows_Server: "Windows_Server",
    None: "None",
};
exports.LoadBalancerSku = {
    Unstacked_haproxy: "unstacked-haproxy",
    Stacked_kube_vip: "stacked-kube-vip",
    Stacked_metallb: "stacked-metallb",
    Unmanaged: "unmanaged",
};
exports.Mode = {
    System: "System",
    LB: "LB",
    User: "User",
};
exports.NetworkPolicy = {
    Calico: "calico",
    Flannel: "flannel",
};
exports.OsType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
