// @pulumi/azure-native/cognitiveservices/v20230501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedCommitmentPlan = exports.PrivateEndpointConnection = exports.listAccountKeysOutput = exports.listAccountKeys = exports.getSharedCommitmentPlanOutput = exports.getSharedCommitmentPlan = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getDeploymentOutput = exports.getDeployment = exports.getCommitmentPlanAssociationOutput = exports.getCommitmentPlanAssociation = exports.getCommitmentPlanOutput = exports.getCommitmentPlan = exports.getAccountOutput = exports.getAccount = exports.Deployment = exports.CommitmentPlanAssociation = exports.CommitmentPlan = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.CommitmentPlan = null;
utilities.lazyLoad(exports, ["CommitmentPlan"], () => require("./commitmentPlan"));
exports.CommitmentPlanAssociation = null;
utilities.lazyLoad(exports, ["CommitmentPlanAssociation"], () => require("./commitmentPlanAssociation"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getCommitmentPlan = null;
exports.getCommitmentPlanOutput = null;
utilities.lazyLoad(exports, ["getCommitmentPlan", "getCommitmentPlanOutput"], () => require("./getCommitmentPlan"));
exports.getCommitmentPlanAssociation = null;
exports.getCommitmentPlanAssociationOutput = null;
utilities.lazyLoad(exports, ["getCommitmentPlanAssociation", "getCommitmentPlanAssociationOutput"], () => require("./getCommitmentPlanAssociation"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSharedCommitmentPlan = null;
exports.getSharedCommitmentPlanOutput = null;
utilities.lazyLoad(exports, ["getSharedCommitmentPlan", "getSharedCommitmentPlanOutput"], () => require("./getSharedCommitmentPlan"));
exports.listAccountKeys = null;
exports.listAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listAccountKeys", "listAccountKeysOutput"], () => require("./listAccountKeys"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.SharedCommitmentPlan = null;
utilities.lazyLoad(exports, ["SharedCommitmentPlan"], () => require("./sharedCommitmentPlan"));
// Export enums:
__exportStar(require("../../types/enums/cognitiveservices/v20230501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cognitiveservices/v20230501:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:cognitiveservices/v20230501:CommitmentPlan":
                return new exports.CommitmentPlan(name, undefined, { urn });
            case "azure-native:cognitiveservices/v20230501:CommitmentPlanAssociation":
                return new exports.CommitmentPlanAssociation(name, undefined, { urn });
            case "azure-native:cognitiveservices/v20230501:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:cognitiveservices/v20230501:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:cognitiveservices/v20230501:SharedCommitmentPlan":
                return new exports.SharedCommitmentPlan(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cognitiveservices/v20230501", _module);
