// @pulumi/azure-native/machinelearningservices/v20220201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnlineEndpoint = exports.OnlineDeployment = exports.ModelVersion = exports.ModelContainer = exports.Job = exports.getWorkspaceConnectionOutput = exports.getWorkspaceConnection = exports.getOnlineEndpointOutput = exports.getOnlineEndpoint = exports.getOnlineDeploymentOutput = exports.getOnlineDeployment = exports.getModelVersionOutput = exports.getModelVersion = exports.getModelContainerOutput = exports.getModelContainer = exports.getJobOutput = exports.getJob = exports.getEnvironmentVersionOutput = exports.getEnvironmentVersion = exports.getEnvironmentContainerOutput = exports.getEnvironmentContainer = exports.getDatastoreOutput = exports.getDatastore = exports.getDataVersionOutput = exports.getDataVersion = exports.getDataContainerOutput = exports.getDataContainer = exports.getComponentVersionOutput = exports.getComponentVersion = exports.getComponentContainerOutput = exports.getComponentContainer = exports.getCodeVersionOutput = exports.getCodeVersion = exports.getCodeContainerOutput = exports.getCodeContainer = exports.getBatchEndpointOutput = exports.getBatchEndpoint = exports.getBatchDeploymentOutput = exports.getBatchDeployment = exports.EnvironmentVersion = exports.EnvironmentContainer = exports.Datastore = exports.DataVersion = exports.DataContainer = exports.ComponentVersion = exports.ComponentContainer = exports.CodeVersion = exports.CodeContainer = exports.BatchEndpoint = exports.BatchDeployment = void 0;
exports.WorkspaceConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BatchDeployment = null;
utilities.lazyLoad(exports, ["BatchDeployment"], () => require("./batchDeployment"));
exports.BatchEndpoint = null;
utilities.lazyLoad(exports, ["BatchEndpoint"], () => require("./batchEndpoint"));
exports.CodeContainer = null;
utilities.lazyLoad(exports, ["CodeContainer"], () => require("./codeContainer"));
exports.CodeVersion = null;
utilities.lazyLoad(exports, ["CodeVersion"], () => require("./codeVersion"));
exports.ComponentContainer = null;
utilities.lazyLoad(exports, ["ComponentContainer"], () => require("./componentContainer"));
exports.ComponentVersion = null;
utilities.lazyLoad(exports, ["ComponentVersion"], () => require("./componentVersion"));
exports.DataContainer = null;
utilities.lazyLoad(exports, ["DataContainer"], () => require("./dataContainer"));
exports.DataVersion = null;
utilities.lazyLoad(exports, ["DataVersion"], () => require("./dataVersion"));
exports.Datastore = null;
utilities.lazyLoad(exports, ["Datastore"], () => require("./datastore"));
exports.EnvironmentContainer = null;
utilities.lazyLoad(exports, ["EnvironmentContainer"], () => require("./environmentContainer"));
exports.EnvironmentVersion = null;
utilities.lazyLoad(exports, ["EnvironmentVersion"], () => require("./environmentVersion"));
exports.getBatchDeployment = null;
exports.getBatchDeploymentOutput = null;
utilities.lazyLoad(exports, ["getBatchDeployment", "getBatchDeploymentOutput"], () => require("./getBatchDeployment"));
exports.getBatchEndpoint = null;
exports.getBatchEndpointOutput = null;
utilities.lazyLoad(exports, ["getBatchEndpoint", "getBatchEndpointOutput"], () => require("./getBatchEndpoint"));
exports.getCodeContainer = null;
exports.getCodeContainerOutput = null;
utilities.lazyLoad(exports, ["getCodeContainer", "getCodeContainerOutput"], () => require("./getCodeContainer"));
exports.getCodeVersion = null;
exports.getCodeVersionOutput = null;
utilities.lazyLoad(exports, ["getCodeVersion", "getCodeVersionOutput"], () => require("./getCodeVersion"));
exports.getComponentContainer = null;
exports.getComponentContainerOutput = null;
utilities.lazyLoad(exports, ["getComponentContainer", "getComponentContainerOutput"], () => require("./getComponentContainer"));
exports.getComponentVersion = null;
exports.getComponentVersionOutput = null;
utilities.lazyLoad(exports, ["getComponentVersion", "getComponentVersionOutput"], () => require("./getComponentVersion"));
exports.getDataContainer = null;
exports.getDataContainerOutput = null;
utilities.lazyLoad(exports, ["getDataContainer", "getDataContainerOutput"], () => require("./getDataContainer"));
exports.getDataVersion = null;
exports.getDataVersionOutput = null;
utilities.lazyLoad(exports, ["getDataVersion", "getDataVersionOutput"], () => require("./getDataVersion"));
exports.getDatastore = null;
exports.getDatastoreOutput = null;
utilities.lazyLoad(exports, ["getDatastore", "getDatastoreOutput"], () => require("./getDatastore"));
exports.getEnvironmentContainer = null;
exports.getEnvironmentContainerOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentContainer", "getEnvironmentContainerOutput"], () => require("./getEnvironmentContainer"));
exports.getEnvironmentVersion = null;
exports.getEnvironmentVersionOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentVersion", "getEnvironmentVersionOutput"], () => require("./getEnvironmentVersion"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getModelContainer = null;
exports.getModelContainerOutput = null;
utilities.lazyLoad(exports, ["getModelContainer", "getModelContainerOutput"], () => require("./getModelContainer"));
exports.getModelVersion = null;
exports.getModelVersionOutput = null;
utilities.lazyLoad(exports, ["getModelVersion", "getModelVersionOutput"], () => require("./getModelVersion"));
exports.getOnlineDeployment = null;
exports.getOnlineDeploymentOutput = null;
utilities.lazyLoad(exports, ["getOnlineDeployment", "getOnlineDeploymentOutput"], () => require("./getOnlineDeployment"));
exports.getOnlineEndpoint = null;
exports.getOnlineEndpointOutput = null;
utilities.lazyLoad(exports, ["getOnlineEndpoint", "getOnlineEndpointOutput"], () => require("./getOnlineEndpoint"));
exports.getWorkspaceConnection = null;
exports.getWorkspaceConnectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceConnection", "getWorkspaceConnectionOutput"], () => require("./getWorkspaceConnection"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.ModelContainer = null;
utilities.lazyLoad(exports, ["ModelContainer"], () => require("./modelContainer"));
exports.ModelVersion = null;
utilities.lazyLoad(exports, ["ModelVersion"], () => require("./modelVersion"));
exports.OnlineDeployment = null;
utilities.lazyLoad(exports, ["OnlineDeployment"], () => require("./onlineDeployment"));
exports.OnlineEndpoint = null;
utilities.lazyLoad(exports, ["OnlineEndpoint"], () => require("./onlineEndpoint"));
exports.WorkspaceConnection = null;
utilities.lazyLoad(exports, ["WorkspaceConnection"], () => require("./workspaceConnection"));
// Export enums:
__exportStar(require("../../types/enums/machinelearningservices/v20220201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningservices/v20220201preview:BatchDeployment":
                return new exports.BatchDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:BatchEndpoint":
                return new exports.BatchEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:CodeContainer":
                return new exports.CodeContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:CodeVersion":
                return new exports.CodeVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:ComponentContainer":
                return new exports.ComponentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:ComponentVersion":
                return new exports.ComponentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:DataContainer":
                return new exports.DataContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:DataVersion":
                return new exports.DataVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:Datastore":
                return new exports.Datastore(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:EnvironmentContainer":
                return new exports.EnvironmentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:EnvironmentVersion":
                return new exports.EnvironmentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:ModelContainer":
                return new exports.ModelContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:ModelVersion":
                return new exports.ModelVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:OnlineDeployment":
                return new exports.OnlineDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:OnlineEndpoint":
                return new exports.OnlineEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220201preview:WorkspaceConnection":
                return new exports.WorkspaceConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningservices/v20220201preview", _module);
