// @pulumi/azure-native/network/v20240501/virtualHubBgpConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualHubBgpConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual Appliance Site resource.
 */
class VirtualHubBgpConnection extends pulumi.CustomResource {
    /**
     * Get an existing VirtualHubBgpConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualHubBgpConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualHubBgpConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualHubBgpConnection.__pulumiType;
    }
    /**
     * Create a VirtualHubBgpConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.virtualHubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualHubName'");
            }
            resourceInputs["connectionName"] = args ? args.connectionName : undefined;
            resourceInputs["hubVirtualNetworkConnection"] = args ? args.hubVirtualNetworkConnection : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["peerAsn"] = args ? args.peerAsn : undefined;
            resourceInputs["peerIp"] = args ? args.peerIp : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["virtualHubName"] = args ? args.virtualHubName : undefined;
            resourceInputs["connectionState"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["connectionState"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["hubVirtualNetworkConnection"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peerAsn"] = undefined /*out*/;
            resourceInputs["peerIp"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20200501:VirtualHubBgpConnection" }, { type: "azure-native:network/v20200601:VirtualHubBgpConnection" }, { type: "azure-native:network/v20200701:VirtualHubBgpConnection" }, { type: "azure-native:network/v20200801:VirtualHubBgpConnection" }, { type: "azure-native:network/v20201101:VirtualHubBgpConnection" }, { type: "azure-native:network/v20210201:VirtualHubBgpConnection" }, { type: "azure-native:network/v20210301:VirtualHubBgpConnection" }, { type: "azure-native:network/v20210501:VirtualHubBgpConnection" }, { type: "azure-native:network/v20210801:VirtualHubBgpConnection" }, { type: "azure-native:network/v20220101:VirtualHubBgpConnection" }, { type: "azure-native:network/v20220501:VirtualHubBgpConnection" }, { type: "azure-native:network/v20220701:VirtualHubBgpConnection" }, { type: "azure-native:network/v20220901:VirtualHubBgpConnection" }, { type: "azure-native:network/v20221101:VirtualHubBgpConnection" }, { type: "azure-native:network/v20230201:VirtualHubBgpConnection" }, { type: "azure-native:network/v20230401:VirtualHubBgpConnection" }, { type: "azure-native:network/v20230501:VirtualHubBgpConnection" }, { type: "azure-native:network/v20230601:VirtualHubBgpConnection" }, { type: "azure-native:network/v20230901:VirtualHubBgpConnection" }, { type: "azure-native:network/v20231101:VirtualHubBgpConnection" }, { type: "azure-native:network/v20240101:VirtualHubBgpConnection" }, { type: "azure-native:network/v20240301:VirtualHubBgpConnection" }, { type: "azure-native:network:VirtualHubBgpConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualHubBgpConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualHubBgpConnection = VirtualHubBgpConnection;
/** @internal */
VirtualHubBgpConnection.__pulumiType = 'azure-native:network/v20240501:VirtualHubBgpConnection';
