// @pulumi/azure-native/types/enums/healthcareapis/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceManagedIdentityType = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.Kind = exports.IotIdentityResolutionType = exports.FhirServiceVersion = exports.FhirServiceKind = exports.FhirResourceVersionPolicy = exports.AnalyticsConnectorMappingType = exports.AnalyticsConnectorDataSourceType = exports.AnalyticsConnectorDataDestinationType = exports.v20250301preview = exports.v20240331 = exports.v20240301 = exports.v20231201 = exports.v20231101 = exports.v20230906 = exports.v20230228 = exports.v20221001preview = void 0;
// Export sub-modules:
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230228 = require("./v20230228");
exports.v20230228 = v20230228;
const v20230906 = require("./v20230906");
exports.v20230906 = v20230906;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240331 = require("./v20240331");
exports.v20240331 = v20240331;
const v20250301preview = require("./v20250301preview");
exports.v20250301preview = v20250301preview;
exports.AnalyticsConnectorDataDestinationType = {
    Datalake: "datalake",
};
exports.AnalyticsConnectorDataSourceType = {
    Fhirservice: "fhirservice",
};
exports.AnalyticsConnectorMappingType = {
    FhirToParquet: "fhirToParquet",
};
exports.FhirResourceVersionPolicy = {
    No_version: "no-version",
    Versioned: "versioned",
    Versioned_update: "versioned-update",
};
exports.FhirServiceKind = {
    Fhir_Stu3: "fhir-Stu3",
    Fhir_R4: "fhir-R4",
};
exports.FhirServiceVersion = {
    STU3: "STU3",
    R4: "R4",
};
exports.IotIdentityResolutionType = {
    Create: "Create",
    Lookup: "Lookup",
};
exports.Kind = {
    Fhir: "fhir",
    Fhir_Stu3: "fhir-Stu3",
    Fhir_R4: "fhir-R4",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServiceManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
