// @pulumi/azure-native/resources/v20240301/deploymentStackAtManagementGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentStackAtManagementGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Deployment stack object.
 */
class DeploymentStackAtManagementGroup extends pulumi.CustomResource {
    /**
     * Get an existing DeploymentStackAtManagementGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentStackAtManagementGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeploymentStackAtManagementGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentStackAtManagementGroup.__pulumiType;
    }
    /**
     * Create a DeploymentStackAtManagementGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.actionOnUnmanage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'actionOnUnmanage'");
            }
            if ((!args || args.denySettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'denySettings'");
            }
            if ((!args || args.managementGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managementGroupId'");
            }
            resourceInputs["actionOnUnmanage"] = args ? args.actionOnUnmanage : undefined;
            resourceInputs["bypassStackOutOfSyncError"] = args ? args.bypassStackOutOfSyncError : undefined;
            resourceInputs["debugSetting"] = args ? args.debugSetting : undefined;
            resourceInputs["denySettings"] = args ? args.denySettings : undefined;
            resourceInputs["deploymentScope"] = args ? args.deploymentScope : undefined;
            resourceInputs["deploymentStackName"] = args ? args.deploymentStackName : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managementGroupId"] = args ? args.managementGroupId : undefined;
            resourceInputs["parameters"] = args ? args.parameters : undefined;
            resourceInputs["parametersLink"] = args ? args.parametersLink : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["template"] = args ? args.template : undefined;
            resourceInputs["templateLink"] = args ? args.templateLink : undefined;
            resourceInputs["correlationId"] = undefined /*out*/;
            resourceInputs["deletedResources"] = undefined /*out*/;
            resourceInputs["deploymentId"] = undefined /*out*/;
            resourceInputs["detachedResources"] = undefined /*out*/;
            resourceInputs["duration"] = undefined /*out*/;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["failedResources"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["outputs"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resources"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["actionOnUnmanage"] = undefined /*out*/;
            resourceInputs["correlationId"] = undefined /*out*/;
            resourceInputs["debugSetting"] = undefined /*out*/;
            resourceInputs["deletedResources"] = undefined /*out*/;
            resourceInputs["denySettings"] = undefined /*out*/;
            resourceInputs["deploymentId"] = undefined /*out*/;
            resourceInputs["deploymentScope"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["detachedResources"] = undefined /*out*/;
            resourceInputs["duration"] = undefined /*out*/;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["failedResources"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["outputs"] = undefined /*out*/;
            resourceInputs["parameters"] = undefined /*out*/;
            resourceInputs["parametersLink"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resources"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:resources/v20220801preview:DeploymentStackAtManagementGroup" }, { type: "azure-native:resources:DeploymentStackAtManagementGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DeploymentStackAtManagementGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeploymentStackAtManagementGroup = DeploymentStackAtManagementGroup;
/** @internal */
DeploymentStackAtManagementGroup.__pulumiType = 'azure-native:resources/v20240301:DeploymentStackAtManagementGroup';
