// @pulumi/azure-native/databricks/v20250301preview/vnetPeering.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VNetPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Peerings in a VirtualNetwork resource
 */
class VNetPeering extends pulumi.CustomResource {
    /**
     * Get an existing VNetPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VNetPeering(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VNetPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VNetPeering.__pulumiType;
    }
    /**
     * Create a VNetPeering resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.remoteVirtualNetwork === undefined) && !opts.urn) {
                throw new Error("Missing required property 'remoteVirtualNetwork'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["allowForwardedTraffic"] = args ? args.allowForwardedTraffic : undefined;
            resourceInputs["allowGatewayTransit"] = args ? args.allowGatewayTransit : undefined;
            resourceInputs["allowVirtualNetworkAccess"] = args ? args.allowVirtualNetworkAccess : undefined;
            resourceInputs["databricksAddressSpace"] = args ? args.databricksAddressSpace : undefined;
            resourceInputs["databricksVirtualNetwork"] = args ? args.databricksVirtualNetwork : undefined;
            resourceInputs["peeringName"] = args ? args.peeringName : undefined;
            resourceInputs["remoteAddressSpace"] = args ? args.remoteAddressSpace : undefined;
            resourceInputs["remoteVirtualNetwork"] = args ? args.remoteVirtualNetwork : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["useRemoteGateways"] = args ? args.useRemoteGateways : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peeringState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowForwardedTraffic"] = undefined /*out*/;
            resourceInputs["allowGatewayTransit"] = undefined /*out*/;
            resourceInputs["allowVirtualNetworkAccess"] = undefined /*out*/;
            resourceInputs["databricksAddressSpace"] = undefined /*out*/;
            resourceInputs["databricksVirtualNetwork"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peeringState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["remoteAddressSpace"] = undefined /*out*/;
            resourceInputs["remoteVirtualNetwork"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useRemoteGateways"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databricks/v20180401:VNetPeering" }, { type: "azure-native:databricks/v20180401:vNetPeering" }, { type: "azure-native:databricks/v20210401preview:VNetPeering" }, { type: "azure-native:databricks/v20210401preview:vNetPeering" }, { type: "azure-native:databricks/v20220401preview:VNetPeering" }, { type: "azure-native:databricks/v20220401preview:vNetPeering" }, { type: "azure-native:databricks/v20230201:VNetPeering" }, { type: "azure-native:databricks/v20230201:vNetPeering" }, { type: "azure-native:databricks/v20230915preview:VNetPeering" }, { type: "azure-native:databricks/v20230915preview:vNetPeering" }, { type: "azure-native:databricks/v20240501:VNetPeering" }, { type: "azure-native:databricks/v20240501:vNetPeering" }, { type: "azure-native:databricks/v20240901preview:VNetPeering" }, { type: "azure-native:databricks/v20240901preview:vNetPeering" }, { type: "azure-native:databricks/v20250301preview:vNetPeering" }, { type: "azure-native:databricks:VNetPeering" }, { type: "azure-native:databricks:vNetPeering" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VNetPeering.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VNetPeering = VNetPeering;
/** @internal */
VNetPeering.__pulumiType = 'azure-native:databricks/v20250301preview:VNetPeering';
