// @pulumi/azure-native/testbase/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestBaseAccount = exports.Package = exports.getTestResultVideoDownloadURLOutput = exports.getTestResultVideoDownloadURL = exports.getTestResultDownloadURLOutput = exports.getTestResultDownloadURL = exports.getTestResultConsoleLogDownloadURLOutput = exports.getTestResultConsoleLogDownloadURL = exports.getTestBaseAccountFileUploadUrlOutput = exports.getTestBaseAccountFileUploadUrl = exports.getTestBaseAccountOutput = exports.getTestBaseAccount = exports.getPackageDownloadURLOutput = exports.getPackageDownloadURL = exports.getPackageOutput = exports.getPackage = exports.getFavoriteProcessOutput = exports.getFavoriteProcess = exports.getCustomerEventOutput = exports.getCustomerEvent = exports.getBillingHubServiceUsageOutput = exports.getBillingHubServiceUsage = exports.getBillingHubServiceFreeHourBalanceOutput = exports.getBillingHubServiceFreeHourBalance = exports.FavoriteProcess = exports.CustomerEvent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CustomerEvent = null;
utilities.lazyLoad(exports, ["CustomerEvent"], () => require("./customerEvent"));
exports.FavoriteProcess = null;
utilities.lazyLoad(exports, ["FavoriteProcess"], () => require("./favoriteProcess"));
exports.getBillingHubServiceFreeHourBalance = null;
exports.getBillingHubServiceFreeHourBalanceOutput = null;
utilities.lazyLoad(exports, ["getBillingHubServiceFreeHourBalance", "getBillingHubServiceFreeHourBalanceOutput"], () => require("./getBillingHubServiceFreeHourBalance"));
exports.getBillingHubServiceUsage = null;
exports.getBillingHubServiceUsageOutput = null;
utilities.lazyLoad(exports, ["getBillingHubServiceUsage", "getBillingHubServiceUsageOutput"], () => require("./getBillingHubServiceUsage"));
exports.getCustomerEvent = null;
exports.getCustomerEventOutput = null;
utilities.lazyLoad(exports, ["getCustomerEvent", "getCustomerEventOutput"], () => require("./getCustomerEvent"));
exports.getFavoriteProcess = null;
exports.getFavoriteProcessOutput = null;
utilities.lazyLoad(exports, ["getFavoriteProcess", "getFavoriteProcessOutput"], () => require("./getFavoriteProcess"));
exports.getPackage = null;
exports.getPackageOutput = null;
utilities.lazyLoad(exports, ["getPackage", "getPackageOutput"], () => require("./getPackage"));
exports.getPackageDownloadURL = null;
exports.getPackageDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getPackageDownloadURL", "getPackageDownloadURLOutput"], () => require("./getPackageDownloadURL"));
exports.getTestBaseAccount = null;
exports.getTestBaseAccountOutput = null;
utilities.lazyLoad(exports, ["getTestBaseAccount", "getTestBaseAccountOutput"], () => require("./getTestBaseAccount"));
exports.getTestBaseAccountFileUploadUrl = null;
exports.getTestBaseAccountFileUploadUrlOutput = null;
utilities.lazyLoad(exports, ["getTestBaseAccountFileUploadUrl", "getTestBaseAccountFileUploadUrlOutput"], () => require("./getTestBaseAccountFileUploadUrl"));
exports.getTestResultConsoleLogDownloadURL = null;
exports.getTestResultConsoleLogDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultConsoleLogDownloadURL", "getTestResultConsoleLogDownloadURLOutput"], () => require("./getTestResultConsoleLogDownloadURL"));
exports.getTestResultDownloadURL = null;
exports.getTestResultDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultDownloadURL", "getTestResultDownloadURLOutput"], () => require("./getTestResultDownloadURL"));
exports.getTestResultVideoDownloadURL = null;
exports.getTestResultVideoDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultVideoDownloadURL", "getTestResultVideoDownloadURLOutput"], () => require("./getTestResultVideoDownloadURL"));
exports.Package = null;
utilities.lazyLoad(exports, ["Package"], () => require("./package"));
exports.TestBaseAccount = null;
utilities.lazyLoad(exports, ["TestBaseAccount"], () => require("./testBaseAccount"));
// Export enums:
__exportStar(require("../../types/enums/testbase/v20220401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:testbase/v20220401preview:CustomerEvent":
                return new exports.CustomerEvent(name, undefined, { urn });
            case "azure-native:testbase/v20220401preview:FavoriteProcess":
                return new exports.FavoriteProcess(name, undefined, { urn });
            case "azure-native:testbase/v20220401preview:Package":
                return new exports.Package(name, undefined, { urn });
            case "azure-native:testbase/v20220401preview:TestBaseAccount":
                return new exports.TestBaseAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "testbase/v20220401preview", _module);
