// @pulumi/azure-native/eventhub/v20221001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaRegistry = exports.PrivateEndpointConnection = exports.NamespaceNetworkRuleSet = exports.NamespaceAuthorizationRule = exports.Namespace = exports.listNamespaceKeysOutput = exports.listNamespaceKeys = exports.listEventHubKeysOutput = exports.listEventHubKeys = exports.listDisasterRecoveryConfigKeysOutput = exports.listDisasterRecoveryConfigKeys = exports.getSchemaRegistryOutput = exports.getSchemaRegistry = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNamespaceNetworkRuleSetOutput = exports.getNamespaceNetworkRuleSet = exports.getNamespaceAuthorizationRuleOutput = exports.getNamespaceAuthorizationRule = exports.getNamespaceOutput = exports.getNamespace = exports.getEventHubAuthorizationRuleOutput = exports.getEventHubAuthorizationRule = exports.getEventHubOutput = exports.getEventHub = exports.getDisasterRecoveryConfigOutput = exports.getDisasterRecoveryConfig = exports.getConsumerGroupOutput = exports.getConsumerGroup = exports.getClusterOutput = exports.getCluster = exports.getApplicationGroupOutput = exports.getApplicationGroup = exports.EventHubAuthorizationRule = exports.EventHub = exports.DisasterRecoveryConfig = exports.ConsumerGroup = exports.Cluster = exports.ApplicationGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ApplicationGroup = null;
utilities.lazyLoad(exports, ["ApplicationGroup"], () => require("./applicationGroup"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ConsumerGroup = null;
utilities.lazyLoad(exports, ["ConsumerGroup"], () => require("./consumerGroup"));
exports.DisasterRecoveryConfig = null;
utilities.lazyLoad(exports, ["DisasterRecoveryConfig"], () => require("./disasterRecoveryConfig"));
exports.EventHub = null;
utilities.lazyLoad(exports, ["EventHub"], () => require("./eventHub"));
exports.EventHubAuthorizationRule = null;
utilities.lazyLoad(exports, ["EventHubAuthorizationRule"], () => require("./eventHubAuthorizationRule"));
exports.getApplicationGroup = null;
exports.getApplicationGroupOutput = null;
utilities.lazyLoad(exports, ["getApplicationGroup", "getApplicationGroupOutput"], () => require("./getApplicationGroup"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getConsumerGroup = null;
exports.getConsumerGroupOutput = null;
utilities.lazyLoad(exports, ["getConsumerGroup", "getConsumerGroupOutput"], () => require("./getConsumerGroup"));
exports.getDisasterRecoveryConfig = null;
exports.getDisasterRecoveryConfigOutput = null;
utilities.lazyLoad(exports, ["getDisasterRecoveryConfig", "getDisasterRecoveryConfigOutput"], () => require("./getDisasterRecoveryConfig"));
exports.getEventHub = null;
exports.getEventHubOutput = null;
utilities.lazyLoad(exports, ["getEventHub", "getEventHubOutput"], () => require("./getEventHub"));
exports.getEventHubAuthorizationRule = null;
exports.getEventHubAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getEventHubAuthorizationRule", "getEventHubAuthorizationRuleOutput"], () => require("./getEventHubAuthorizationRule"));
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getNamespaceAuthorizationRule = null;
exports.getNamespaceAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getNamespaceAuthorizationRule", "getNamespaceAuthorizationRuleOutput"], () => require("./getNamespaceAuthorizationRule"));
exports.getNamespaceNetworkRuleSet = null;
exports.getNamespaceNetworkRuleSetOutput = null;
utilities.lazyLoad(exports, ["getNamespaceNetworkRuleSet", "getNamespaceNetworkRuleSetOutput"], () => require("./getNamespaceNetworkRuleSet"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSchemaRegistry = null;
exports.getSchemaRegistryOutput = null;
utilities.lazyLoad(exports, ["getSchemaRegistry", "getSchemaRegistryOutput"], () => require("./getSchemaRegistry"));
exports.listDisasterRecoveryConfigKeys = null;
exports.listDisasterRecoveryConfigKeysOutput = null;
utilities.lazyLoad(exports, ["listDisasterRecoveryConfigKeys", "listDisasterRecoveryConfigKeysOutput"], () => require("./listDisasterRecoveryConfigKeys"));
exports.listEventHubKeys = null;
exports.listEventHubKeysOutput = null;
utilities.lazyLoad(exports, ["listEventHubKeys", "listEventHubKeysOutput"], () => require("./listEventHubKeys"));
exports.listNamespaceKeys = null;
exports.listNamespaceKeysOutput = null;
utilities.lazyLoad(exports, ["listNamespaceKeys", "listNamespaceKeysOutput"], () => require("./listNamespaceKeys"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.NamespaceAuthorizationRule = null;
utilities.lazyLoad(exports, ["NamespaceAuthorizationRule"], () => require("./namespaceAuthorizationRule"));
exports.NamespaceNetworkRuleSet = null;
utilities.lazyLoad(exports, ["NamespaceNetworkRuleSet"], () => require("./namespaceNetworkRuleSet"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.SchemaRegistry = null;
utilities.lazyLoad(exports, ["SchemaRegistry"], () => require("./schemaRegistry"));
// Export enums:
__exportStar(require("../../types/enums/eventhub/v20221001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:eventhub/v20221001preview:ApplicationGroup":
                return new exports.ApplicationGroup(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:ConsumerGroup":
                return new exports.ConsumerGroup(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:DisasterRecoveryConfig":
                return new exports.DisasterRecoveryConfig(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:EventHub":
                return new exports.EventHub(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:EventHubAuthorizationRule":
                return new exports.EventHubAuthorizationRule(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:NamespaceAuthorizationRule":
                return new exports.NamespaceAuthorizationRule(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:NamespaceNetworkRuleSet":
                return new exports.NamespaceNetworkRuleSet(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:eventhub/v20221001preview:SchemaRegistry":
                return new exports.SchemaRegistry(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "eventhub/v20221001preview", _module);
