// @pulumi/azure-native/digitaltwins/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230131 = exports.v20201201 = exports.TimeSeriesDatabaseConnection = exports.PrivateEndpointConnection = exports.getTimeSeriesDatabaseConnectionOutput = exports.getTimeSeriesDatabaseConnection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getDigitalTwinsEndpointOutput = exports.getDigitalTwinsEndpoint = exports.getDigitalTwinOutput = exports.getDigitalTwin = exports.DigitalTwinsEndpoint = exports.DigitalTwin = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DigitalTwin = null;
utilities.lazyLoad(exports, ["DigitalTwin"], () => require("./digitalTwin"));
exports.DigitalTwinsEndpoint = null;
utilities.lazyLoad(exports, ["DigitalTwinsEndpoint"], () => require("./digitalTwinsEndpoint"));
exports.getDigitalTwin = null;
exports.getDigitalTwinOutput = null;
utilities.lazyLoad(exports, ["getDigitalTwin", "getDigitalTwinOutput"], () => require("./getDigitalTwin"));
exports.getDigitalTwinsEndpoint = null;
exports.getDigitalTwinsEndpointOutput = null;
utilities.lazyLoad(exports, ["getDigitalTwinsEndpoint", "getDigitalTwinsEndpointOutput"], () => require("./getDigitalTwinsEndpoint"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getTimeSeriesDatabaseConnection = null;
exports.getTimeSeriesDatabaseConnectionOutput = null;
utilities.lazyLoad(exports, ["getTimeSeriesDatabaseConnection", "getTimeSeriesDatabaseConnectionOutput"], () => require("./getTimeSeriesDatabaseConnection"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.TimeSeriesDatabaseConnection = null;
utilities.lazyLoad(exports, ["TimeSeriesDatabaseConnection"], () => require("./timeSeriesDatabaseConnection"));
// Export enums:
__exportStar(require("../types/enums/digitaltwins"), exports);
// Export sub-modules:
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20230131 = require("./v20230131");
exports.v20230131 = v20230131;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:digitaltwins:DigitalTwin":
                return new exports.DigitalTwin(name, undefined, { urn });
            case "azure-native:digitaltwins:DigitalTwinsEndpoint":
                return new exports.DigitalTwinsEndpoint(name, undefined, { urn });
            case "azure-native:digitaltwins:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:digitaltwins:TimeSeriesDatabaseConnection":
                return new exports.TimeSeriesDatabaseConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "digitaltwins", _module);
