// @pulumi/azure-native/migrate/v20180202/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Project = exports.Group = exports.getProjectKeysOutput = exports.getProjectKeys = exports.getProjectOutput = exports.getProject = exports.getGroupOutput = exports.getGroup = exports.getAssessmentOutput = exports.getAssessment = exports.Assessment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Assessment = null;
utilities.lazyLoad(exports, ["Assessment"], () => require("./assessment"));
exports.getAssessment = null;
exports.getAssessmentOutput = null;
utilities.lazyLoad(exports, ["getAssessment", "getAssessmentOutput"], () => require("./getAssessment"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getProjectKeys = null;
exports.getProjectKeysOutput = null;
utilities.lazyLoad(exports, ["getProjectKeys", "getProjectKeysOutput"], () => require("./getProjectKeys"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
// Export enums:
__exportStar(require("../../types/enums/migrate/v20180202"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate/v20180202:Assessment":
                return new exports.Assessment(name, undefined, { urn });
            case "azure-native:migrate/v20180202:Group":
                return new exports.Group(name, undefined, { urn });
            case "azure-native:migrate/v20180202:Project":
                return new exports.Project(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate/v20180202", _module);
