// @pulumi/azure-native/types/enums/workloads/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadType = exports.WeekOfMonth = exports.VaultType = exports.TieringMode = exports.SslPreference = exports.SslCryptoProvider = exports.ScheduleRunType = exports.SAPSoftwareInstallationType = exports.SAPProductType = exports.SAPHighAvailabilityType = exports.SAPEnvironmentType = exports.SAPDeploymentType = exports.SAPDatabaseType = exports.SAPDatabaseScaleMethod = exports.SAPConfigurationType = exports.RoutingPreference = exports.RetentionScheduleFormat = exports.RetentionDurationType = exports.PolicyType = exports.OSType = exports.NamingPatternType = exports.MonthOfYear = exports.ManagedServiceIdentityType = exports.IAASVMPolicyType = exports.DiskSkuName = exports.DayOfWeek = exports.ConfigurationType = exports.ConditionalOperator = exports.BackupType = exports.AppServicePlanTier = exports.AlertRuleStatus = exports.AlertAutoMitigate = exports.v20240901 = exports.v20240201preview = exports.v20231201preview = exports.v20231001preview = exports.v20230401 = exports.v20221101preview = exports.v20211201preview = void 0;
// Export sub-modules:
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20221101preview = require("./v20221101preview");
exports.v20221101preview = v20221101preview;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240901 = require("./v20240901");
exports.v20240901 = v20240901;
exports.AlertAutoMitigate = {
    /**
     * The alert should be automatically resolved.
     */
    Enable: "Enable",
    /**
     * The alert should not be automatically resolved.
     */
    Disable: "Disable",
};
exports.AlertRuleStatus = {
    /**
     * The scheduled query rule is enabled
     */
    Enabled: "Enabled",
    /**
     * The scheduled query rule is disabled
     */
    Disabled: "Disabled",
};
exports.AppServicePlanTier = {
    /**
     * Elastic Premium plan
     */
    ElasticPremium: "ElasticPremium",
    /**
     * Dedicated Premium V3 plan
     */
    PremiumV3: "PremiumV3",
};
exports.BackupType = {
    VM: "VM",
    SQL: "SQL",
    HANA: "HANA",
};
exports.ConditionalOperator = {
    /**
     * The value is less than the specified value.
     */
    LessThan: "LessThan",
    /**
     * The value is greater than the specified value.
     */
    GreaterThan: "GreaterThan",
    /**
     * The value is equal to the specified value.
     */
    Equal: "Equal",
    /**
     * The value is greater than or equal to the specified value.
     */
    GreaterThanOrEqual: "GreaterThanOrEqual",
    /**
     * The value is less than or equal to the specified value.
     */
    LessThanOrEqual: "LessThanOrEqual",
};
exports.ConfigurationType = {
    Skip: "Skip",
    CreateAndMount: "CreateAndMount",
    Mount: "Mount",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.DiskSkuName = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    StandardSSD_LRS: "StandardSSD_LRS",
    UltraSSD_LRS: "UltraSSD_LRS",
    Premium_ZRS: "Premium_ZRS",
    StandardSSD_ZRS: "StandardSSD_ZRS",
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.IAASVMPolicyType = {
    Invalid: "Invalid",
    V1: "V1",
    V2: "V2",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    UserAssigned: "UserAssigned",
};
exports.MonthOfYear = {
    Invalid: "Invalid",
    January: "January",
    February: "February",
    March: "March",
    April: "April",
    May: "May",
    June: "June",
    July: "July",
    August: "August",
    September: "September",
    October: "October",
    November: "November",
    December: "December",
};
exports.NamingPatternType = {
    FullResourceName: "FullResourceName",
};
exports.OSType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.PolicyType = {
    Invalid: "Invalid",
    Full: "Full",
    Differential: "Differential",
    Log: "Log",
    CopyOnlyFull: "CopyOnlyFull",
    Incremental: "Incremental",
    SnapshotFull: "SnapshotFull",
    SnapshotCopyOnlyFull: "SnapshotCopyOnlyFull",
};
exports.RetentionDurationType = {
    Invalid: "Invalid",
    Days: "Days",
    Weeks: "Weeks",
    Months: "Months",
    Years: "Years",
};
exports.RetentionScheduleFormat = {
    Invalid: "Invalid",
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.RoutingPreference = {
    Default: "Default",
    RouteAll: "RouteAll",
};
exports.SAPConfigurationType = {
    Deployment: "Deployment",
    Discovery: "Discovery",
    DeploymentWithOSConfig: "DeploymentWithOSConfig",
};
exports.SAPDatabaseScaleMethod = {
    /**
     * ScaleUp Hana Database deployment type
     */
    ScaleUp: "ScaleUp",
};
exports.SAPDatabaseType = {
    /**
     * HANA Database type of SAP system.
     */
    HANA: "HANA",
    /**
     * DB2 database type of the SAP system.
     */
    DB2: "DB2",
};
exports.SAPDeploymentType = {
    /**
     * SAP Single server deployment type.
     */
    SingleServer: "SingleServer",
    /**
     * SAP Distributed deployment type.
     */
    ThreeTier: "ThreeTier",
};
exports.SAPEnvironmentType = {
    /**
     * Non Production SAP system.
     */
    NonProd: "NonProd",
    /**
     * Production SAP system.
     */
    Prod: "Prod",
};
exports.SAPHighAvailabilityType = {
    /**
     * HA deployment with availability sets.
     */
    AvailabilitySet: "AvailabilitySet",
    /**
     * HA deployment with availability zones.
     */
    AvailabilityZone: "AvailabilityZone",
};
exports.SAPProductType = {
    /**
     * SAP Product ECC.
     */
    ECC: "ECC",
    /**
     * SAP Product S4HANA.
     */
    S4HANA: "S4HANA",
    /**
     * SAP Products other than the ones listed.
     */
    Other: "Other",
};
exports.SAPSoftwareInstallationType = {
    ServiceInitiated: "ServiceInitiated",
    SAPInstallWithoutOSConfig: "SAPInstallWithoutOSConfig",
    External: "External",
};
exports.ScheduleRunType = {
    Invalid: "Invalid",
    Daily: "Daily",
    Weekly: "Weekly",
    Hourly: "Hourly",
};
exports.SslCryptoProvider = {
    Commoncrypto: "commoncrypto",
    Openssl: "openssl",
};
exports.SslPreference = {
    Disabled: "Disabled",
    RootCertificate: "RootCertificate",
    ServerCertificate: "ServerCertificate",
};
exports.TieringMode = {
    Invalid: "Invalid",
    TierRecommended: "TierRecommended",
    TierAfter: "TierAfter",
    DoNotTier: "DoNotTier",
};
exports.VaultType = {
    Existing: "Existing",
    New: "New",
};
exports.WeekOfMonth = {
    First: "First",
    Second: "Second",
    Third: "Third",
    Fourth: "Fourth",
    Last: "Last",
    Invalid: "Invalid",
};
exports.WorkloadType = {
    Invalid: "Invalid",
    VM: "VM",
    FileFolder: "FileFolder",
    AzureSqlDb: "AzureSqlDb",
    SQLDB: "SQLDB",
    Exchange: "Exchange",
    Sharepoint: "Sharepoint",
    VMwareVM: "VMwareVM",
    SystemState: "SystemState",
    Client: "Client",
    GenericDataSource: "GenericDataSource",
    SQLDataBase: "SQLDataBase",
    AzureFileShare: "AzureFileShare",
    SAPHanaDatabase: "SAPHanaDatabase",
    SAPAseDatabase: "SAPAseDatabase",
    SAPHanaDBInstance: "SAPHanaDBInstance",
};
