// @pulumi/azure-native/security/v20220101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceRule = exports.GovernanceAssignment = exports.getGovernanceRuleOutput = exports.getGovernanceRule = exports.getGovernanceAssignmentOutput = exports.getGovernanceAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getGovernanceAssignment = null;
exports.getGovernanceAssignmentOutput = null;
utilities.lazyLoad(exports, ["getGovernanceAssignment", "getGovernanceAssignmentOutput"], () => require("./getGovernanceAssignment"));
exports.getGovernanceRule = null;
exports.getGovernanceRuleOutput = null;
utilities.lazyLoad(exports, ["getGovernanceRule", "getGovernanceRuleOutput"], () => require("./getGovernanceRule"));
exports.GovernanceAssignment = null;
utilities.lazyLoad(exports, ["GovernanceAssignment"], () => require("./governanceAssignment"));
exports.GovernanceRule = null;
utilities.lazyLoad(exports, ["GovernanceRule"], () => require("./governanceRule"));
// Export enums:
__exportStar(require("../../types/enums/security/v20220101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20220101preview:GovernanceAssignment":
                return new exports.GovernanceAssignment(name, undefined, { urn });
            case "azure-native:security/v20220101preview:GovernanceRule":
                return new exports.GovernanceRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20220101preview", _module);
