// @pulumi/azure-native/types/enums/azuredatatransfer/v20231011preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaStatus = exports.ManagedServiceIdentityType = exports.ListApprovedSchemasDirection = exports.FlowType = exports.FlowStatus = exports.Direction = exports.DataType = void 0;
exports.DataType = {
    Blob: "Blob",
    Table: "Table",
};
exports.Direction = {
    Send: "Send",
    Receive: "Receive",
};
exports.FlowStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FlowType = {
    Unknown: "Unknown",
    Complex: "Complex",
    DevSecOps: "DevSecOps",
    Messaging: "Messaging",
    Mission: "Mission",
    MicrosoftInternal: "MicrosoftInternal",
    BasicFiles: "BasicFiles",
    Data: "Data",
};
exports.ListApprovedSchemasDirection = {
    Send: "Send",
    Receive: "Receive",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.SchemaStatus = {
    New: "New",
    Approved: "Approved",
};
