// @pulumi/azure-native/kubernetesconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241101preview = exports.v20241101 = exports.v20240401preview = exports.v20230501 = exports.v20220701 = exports.v20220402preview = exports.v20220101preview = exports.v20211101preview = exports.v20200701preview = exports.SourceControlConfiguration = exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.getSourceControlConfigurationOutput = exports.getSourceControlConfiguration = exports.getPrivateLinkScopeOutput = exports.getPrivateLinkScope = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getFluxConfigurationOutput = exports.getFluxConfiguration = exports.getExtensionOutput = exports.getExtension = exports.FluxConfiguration = exports.Extension = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.FluxConfiguration = null;
utilities.lazyLoad(exports, ["FluxConfiguration"], () => require("./fluxConfiguration"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getFluxConfiguration = null;
exports.getFluxConfigurationOutput = null;
utilities.lazyLoad(exports, ["getFluxConfiguration", "getFluxConfigurationOutput"], () => require("./getFluxConfiguration"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkScope = null;
exports.getPrivateLinkScopeOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScope", "getPrivateLinkScopeOutput"], () => require("./getPrivateLinkScope"));
exports.getSourceControlConfiguration = null;
exports.getSourceControlConfigurationOutput = null;
utilities.lazyLoad(exports, ["getSourceControlConfiguration", "getSourceControlConfigurationOutput"], () => require("./getSourceControlConfiguration"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkScope = null;
utilities.lazyLoad(exports, ["PrivateLinkScope"], () => require("./privateLinkScope"));
exports.SourceControlConfiguration = null;
utilities.lazyLoad(exports, ["SourceControlConfiguration"], () => require("./sourceControlConfiguration"));
// Export enums:
__exportStar(require("../types/enums/kubernetesconfiguration"), exports);
// Export sub-modules:
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20220402preview = require("./v20220402preview");
exports.v20220402preview = v20220402preview;
const v20220701 = require("./v20220701");
exports.v20220701 = v20220701;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetesconfiguration:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration:FluxConfiguration":
                return new exports.FluxConfiguration(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration:PrivateLinkScope":
                return new exports.PrivateLinkScope(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration:SourceControlConfiguration":
                return new exports.SourceControlConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetesconfiguration", _module);
