// @pulumi/azure-native/types/enums/connectedvmwarevsphere/v20220715preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningAction = exports.PowerOnBootOption = exports.OsType = exports.NICType = exports.InventoryType = exports.IdentityType = exports.IPAddressAllocationMethod = exports.FirmwareType = exports.DiskType = exports.DiskMode = void 0;
exports.DiskMode = {
    Persistent: "persistent",
    Independent_persistent: "independent_persistent",
    Independent_nonpersistent: "independent_nonpersistent",
};
exports.DiskType = {
    Flat: "flat",
    Pmem: "pmem",
    Rawphysical: "rawphysical",
    Rawvirtual: "rawvirtual",
    Sparse: "sparse",
    Sesparse: "sesparse",
    Unknown: "unknown",
};
exports.FirmwareType = {
    Bios: "bios",
    Efi: "efi",
};
exports.IPAddressAllocationMethod = {
    Unset: "unset",
    Dynamic: "dynamic",
    Static: "static",
    Linklayer: "linklayer",
    Random: "random",
    Other: "other",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.InventoryType = {
    ResourcePool: "ResourcePool",
    VirtualMachine: "VirtualMachine",
    VirtualMachineTemplate: "VirtualMachineTemplate",
    VirtualNetwork: "VirtualNetwork",
    Cluster: "Cluster",
    Datastore: "Datastore",
    Host: "Host",
};
exports.NICType = {
    Vmxnet3: "vmxnet3",
    Vmxnet2: "vmxnet2",
    Vmxnet: "vmxnet",
    E1000: "e1000",
    E1000e: "e1000e",
    Pcnet32: "pcnet32",
};
exports.OsType = {
    Windows: "Windows",
    Linux: "Linux",
    Other: "Other",
};
exports.PowerOnBootOption = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ProvisioningAction = {
    Install: "install",
    Uninstall: "uninstall",
    Repair: "repair",
};
