// @pulumi/azure-native/types/enums/eventgrid/v20241215preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicSpacesConfigurationState = exports.TlsVersion = exports.StaticRoutingEnrichmentType = exports.SkuName = exports.Sku = exports.RoutingIdentityType = exports.ResourceProvisioningState = exports.ResourceKind = exports.ReadinessState = exports.PublisherType = exports.PublicNetworkAccess = exports.PersistedConnectionStatus = exports.PermissionType = exports.PartnerTopicRoutingMode = exports.PartnerTopicActivationState = exports.PartnerEndpointType = exports.PartnerDestinationActivationState = exports.PartnerConfigurationProvisioningState = exports.PartnerClientAuthenticationType = exports.MonitorAlertSeverity = exports.IpActionType = exports.InputSchemaMappingType = exports.InputSchema = exports.IdentityType = exports.FilterOperatorType = exports.EventSubscriptionIdentityType = exports.EventInputSchema = exports.EventDeliverySchema = exports.EventDefinitionKind = exports.EndpointType = exports.DeliverySchema = exports.DeliveryMode = exports.DeliveryAttributeMappingType = exports.DeadLetterEndPointType = exports.DataResidencyBoundary = exports.CustomJwtAuthenticationManagedIdentityType = exports.CustomDomainValidationState = exports.CustomDomainIdentityType = exports.ClientState = exports.ClientCertificateValidationScheme = exports.ChannelType = exports.ChannelProvisioningState = exports.AlternativeAuthenticationNameSource = exports.AdvancedFilterOperatorType = void 0;
exports.AdvancedFilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
    NumberInRange: "NumberInRange",
    NumberNotInRange: "NumberNotInRange",
    StringNotBeginsWith: "StringNotBeginsWith",
    StringNotEndsWith: "StringNotEndsWith",
    StringNotContains: "StringNotContains",
    IsNullOrUndefined: "IsNullOrUndefined",
    IsNotNull: "IsNotNull",
};
exports.AlternativeAuthenticationNameSource = {
    ClientCertificateSubject: "ClientCertificateSubject",
    ClientCertificateDns: "ClientCertificateDns",
    ClientCertificateUri: "ClientCertificateUri",
    ClientCertificateIp: "ClientCertificateIp",
    ClientCertificateEmail: "ClientCertificateEmail",
};
exports.ChannelProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
    IdleDueToMirroredPartnerTopicDeletion: "IdleDueToMirroredPartnerTopicDeletion",
    IdleDueToMirroredPartnerDestinationDeletion: "IdleDueToMirroredPartnerDestinationDeletion",
};
exports.ChannelType = {
    PartnerTopic: "PartnerTopic",
    PartnerDestination: "PartnerDestination",
};
exports.ClientCertificateValidationScheme = {
    SubjectMatchesAuthenticationName: "SubjectMatchesAuthenticationName",
    DnsMatchesAuthenticationName: "DnsMatchesAuthenticationName",
    UriMatchesAuthenticationName: "UriMatchesAuthenticationName",
    IpMatchesAuthenticationName: "IpMatchesAuthenticationName",
    EmailMatchesAuthenticationName: "EmailMatchesAuthenticationName",
    ThumbprintMatch: "ThumbprintMatch",
};
exports.ClientState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CustomDomainIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.CustomDomainValidationState = {
    Pending: "Pending",
    Approved: "Approved",
    ErrorRetrievingDnsRecord: "ErrorRetrievingDnsRecord",
};
exports.CustomJwtAuthenticationManagedIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.DataResidencyBoundary = {
    WithinGeopair: "WithinGeopair",
    WithinRegion: "WithinRegion",
};
exports.DeadLetterEndPointType = {
    StorageBlob: "StorageBlob",
};
exports.DeliveryAttributeMappingType = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.DeliveryMode = {
    Queue: "Queue",
    Push: "Push",
};
exports.DeliverySchema = {
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EndpointType = {
    WebHook: "WebHook",
    EventHub: "EventHub",
    StorageQueue: "StorageQueue",
    HybridConnection: "HybridConnection",
    ServiceBusQueue: "ServiceBusQueue",
    ServiceBusTopic: "ServiceBusTopic",
    AzureFunction: "AzureFunction",
    PartnerDestination: "PartnerDestination",
    MonitorAlert: "MonitorAlert",
    NamespaceTopic: "NamespaceTopic",
};
exports.EventDefinitionKind = {
    Inline: "Inline",
};
exports.EventDeliverySchema = {
    EventGridSchema: "EventGridSchema",
    CustomInputSchema: "CustomInputSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventInputSchema = {
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventSubscriptionIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.FilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
    NumberInRange: "NumberInRange",
    NumberNotInRange: "NumberNotInRange",
    StringNotBeginsWith: "StringNotBeginsWith",
    StringNotEndsWith: "StringNotEndsWith",
    StringNotContains: "StringNotContains",
    IsNullOrUndefined: "IsNullOrUndefined",
    IsNotNull: "IsNotNull",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.InputSchema = {
    EventGridSchema: "EventGridSchema",
    CustomEventSchema: "CustomEventSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.InputSchemaMappingType = {
    Json: "Json",
};
exports.IpActionType = {
    Allow: "Allow",
};
exports.MonitorAlertSeverity = {
    Sev0: "Sev0",
    Sev1: "Sev1",
    Sev2: "Sev2",
    Sev3: "Sev3",
    Sev4: "Sev4",
};
exports.PartnerClientAuthenticationType = {
    AzureAD: "AzureAD",
};
exports.PartnerConfigurationProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.PartnerDestinationActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.PartnerEndpointType = {
    WebHook: "WebHook",
};
exports.PartnerTopicActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
    Deactivated: "Deactivated",
};
exports.PartnerTopicRoutingMode = {
    SourceEventAttribute: "SourceEventAttribute",
    ChannelNameHeader: "ChannelNameHeader",
};
exports.PermissionType = {
    Publisher: "Publisher",
    Subscriber: "Subscriber",
};
exports.PersistedConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    SecuredByPerimeter: "SecuredByPerimeter",
};
exports.PublisherType = {
    Custom: "Custom",
};
exports.ReadinessState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.ResourceKind = {
    Azure: "Azure",
    AzureArc: "AzureArc",
};
exports.ResourceProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.RoutingIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.Sku = {
    Basic: "Basic",
    Premium: "Premium",
};
exports.SkuName = {
    Standard: "Standard",
};
exports.StaticRoutingEnrichmentType = {
    String: "String",
};
exports.TlsVersion = {
    TlsVersion_1_0: "1.0",
    TlsVersion_1_1: "1.1",
    TlsVersion_1_2: "1.2",
};
exports.TopicSpacesConfigurationState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
