// @pulumi/azure-native/types/enums/network/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationGatewayFirewallMode = exports.ApplicationGatewayCustomErrorStatusCode = exports.ApplicationGatewayCookieBasedAffinity = exports.ApplicationGatewayClientRevocationOptions = exports.AlwaysServe = exports.AllowedEndpointRecordType = exports.AdminState = exports.AdminRuleKind = exports.AddressPrefixType = exports.ActionType = exports.AccessRuleDirection = exports.Access = exports.v20240601preview = exports.v20240601 = exports.v20240501 = exports.v20240301 = exports.v20240201 = exports.v20240101preview = exports.v20240101 = exports.v20231101 = exports.v20230901 = exports.v20230801preview = exports.v20230701preview = exports.v20230601 = exports.v20230501 = exports.v20230401 = exports.v20230201 = exports.v20220701 = exports.v20220501 = exports.v20220401preview = exports.v20220401 = exports.v20220101 = exports.v20210801 = exports.v20210601 = exports.v20210501preview = exports.v20210301 = exports.v20210201preview = exports.v20210201 = exports.v20200601 = exports.v20200401preview = exports.v20200401 = exports.v20191101 = exports.v20190801 = exports.v20190701 = exports.v20190601 = exports.v20190201 = exports.v20181201 = exports.v20180701 = exports.v20180601 = exports.v20180501 = void 0;
exports.ExpressRouteCircuitSkuTier = exports.ExpressRouteCircuitSkuFamily = exports.ExpressRouteCircuitPeeringState = exports.EnforceCertificateNameCheckEnabledState = exports.EndpointType = exports.EndpointStatus = exports.EndpointMonitorStatus = exports.DynamicCompressionEnabled = exports.DnsSecurityRuleState = exports.DisableBgpRoutePropagation = exports.DhGroup = exports.DestinationPortBehavior = exports.DeleteOptions = exports.DeleteExistingPeering = exports.DeleteExistingNSGs = exports.DdosSettingsProtectionMode = exports.CustomRuleEnabledState = exports.CustomIpPrefixType = exports.CoverageLevel = exports.ConnectivityTopology = exports.ConnectionMonitorTestConfigurationProtocol = exports.ConnectionMonitorEndpointFilterType = exports.ConnectionMonitorEndpointFilterItemType = exports.ConfigurationType = exports.CommissionedState = exports.BlockResponseCode = exports.BastionHostSkuName = exports.BackendEnabledState = exports.AzureFirewallThreatIntelMode = exports.AzureFirewallSkuTier = exports.AzureFirewallSkuName = exports.AzureFirewallRCActionType = exports.AzureFirewallNetworkRuleProtocol = exports.AzureFirewallNatRCActionType = exports.AzureFirewallApplicationRuleProtocolType = exports.AutoLearnPrivateRangesMode = exports.AuthorizationUseStatus = exports.AssociationAccessMode = exports.ApplicationGatewayTier = exports.ApplicationGatewaySslProtocol = exports.ApplicationGatewaySslPolicyType = exports.ApplicationGatewaySslPolicyName = exports.ApplicationGatewaySslCipherSuite = exports.ApplicationGatewaySkuName = exports.ApplicationGatewayRequestRoutingRuleType = exports.ApplicationGatewayRedirectType = exports.ApplicationGatewayProtocol = exports.ApplicationGatewayLoadDistributionAlgorithm = exports.ApplicationGatewayFirewallUserSessionVariable = exports.ApplicationGatewayFirewallRateLimitDuration = void 0;
exports.IpsecIntegrity = exports.IpsecEncryption = exports.IpAllocationType = exports.IpAllocationMethod = exports.IkeIntegrity = exports.IkeEncryption = exports.IPVersion = exports.IPAllocationMethod = exports.HubRoutingPreference = exports.HealthProbeEnabled = exports.HeaderActionType = exports.HTTPConfigurationMethod = exports.GroupConnectivity = exports.Geo = exports.GatewayLoadBalancerTunnelProtocol = exports.GatewayLoadBalancerTunnelInterfaceType = exports.FrontDoorRedirectType = exports.FrontDoorRedirectProtocol = exports.FrontDoorQuery = exports.FrontDoorProtocol = exports.FrontDoorMatchVariable = exports.FrontDoorHealthProbeMethod = exports.FrontDoorForwardingProtocol = exports.FrontDoorEnabledState = exports.ForwardingRuleState = exports.FlowLogFormatType = exports.FirewallPolicySkuTier = exports.FirewallPolicyRuleType = exports.FirewallPolicyRuleNetworkProtocol = exports.FirewallPolicyRuleConditionType = exports.FirewallPolicyRuleConditionNetworkProtocol = exports.FirewallPolicyRuleConditionApplicationProtocolType = exports.FirewallPolicyRuleCollectionType = exports.FirewallPolicyRuleApplicationProtocolType = exports.FirewallPolicyNatRuleCollectionActionType = exports.FirewallPolicyNatRuleActionType = exports.FirewallPolicyIntrusionDetectionStateType = exports.FirewallPolicyIntrusionDetectionProtocol = exports.FirewallPolicyIntrusionDetectionProfileType = exports.FirewallPolicyIDPSQuerySortOrder = exports.FirewallPolicyFilterRuleCollectionActionType = exports.FirewallPolicyFilterRuleActionType = exports.ExtendedLocationTypes = exports.ExpressRoutePortsEncapsulation = exports.ExpressRoutePortsBillingType = exports.ExpressRoutePeeringType = exports.ExpressRoutePeeringState = exports.ExpressRouteLinkMacSecSciState = exports.ExpressRouteLinkMacSecCipher = exports.ExpressRouteLinkAdminState = void 0;
exports.RoutingRuleNextHopType = exports.RoutingRuleEnabledState = exports.RoutingRuleDestinationType = exports.RouteNextHopType = exports.RouteMapMatchCondition = exports.RouteMapActionType = exports.RouteFilterRuleType = exports.ResourceIdentityType = exports.PublicIpAddressDnsSettingsDomainNameLabelScope = exports.PublicIPPrefixSkuTier = exports.PublicIPPrefixSkuName = exports.PublicIPAddressSkuTier = exports.PublicIPAddressSkuName = exports.PublicIPAddressMigrationPhase = exports.ProtocolType = exports.ProfileStatus = exports.ProfileMonitorStatus = exports.ProbeProtocol = exports.PreferredRoutingGateway = exports.PreferredIPVersion = exports.PolicyRequestBodyCheck = exports.PolicyMode = exports.PolicyEnabledState = exports.PfsGroup = exports.PcProtocol = exports.PacketCaptureTargetType = exports.OwaspCrsExclusionEntrySelectorMatchOperator = exports.OwaspCrsExclusionEntryMatchVariable = exports.OutputType = exports.Operator = exports.NextStep = exports.NetworkProtocol = exports.NetworkInterfaceNicType = exports.NetworkInterfaceMigrationPhase = exports.NetworkInterfaceAuxiliarySku = exports.NetworkInterfaceAuxiliaryMode = exports.NetworkIntentPolicyBasedService = exports.NatGatewaySkuName = exports.MonitorProtocol = exports.MatchProcessingBehavior = exports.ManagedRuleSetActionType = exports.ManagedRuleExclusionSelectorMatchOperator = exports.ManagedRuleExclusionMatchVariable = exports.ManagedRuleEnabledState = exports.LoadDistribution = exports.LoadBalancerSkuTier = exports.LoadBalancerSkuName = exports.LoadBalancerOutboundRuleProtocol = exports.LoadBalancerBackendAddressAdminState = exports.IsGlobal = void 0;
exports.WebApplicationFirewallOperator = exports.WebApplicationFirewallMode = exports.WebApplicationFirewallMatchVariable = exports.WebApplicationFirewallEnabledState = exports.WebApplicationFirewallAction = exports.VpnType = exports.VpnPolicyMemberAttributeType = exports.VpnNatRuleType = exports.VpnNatRuleMode = exports.VpnLinkConnectionMode = exports.VpnGatewayTunnelingProtocol = exports.VpnGatewayGeneration = exports.VpnClientProtocol = exports.VpnAuthenticationType = exports.VnetLocalRouteOverrideCriteria = exports.VirtualNetworkPrivateLinkServiceNetworkPolicies = exports.VirtualNetworkPrivateEndpointNetworkPolicies = exports.VirtualNetworkPeeringState = exports.VirtualNetworkPeeringLevel = exports.VirtualNetworkGatewayType = exports.VirtualNetworkGatewaySkuTier = exports.VirtualNetworkGatewaySkuName = exports.VirtualNetworkGatewayConnectionType = exports.VirtualNetworkGatewayConnectionProtocol = exports.VirtualNetworkGatewayConnectionMode = exports.VirtualNetworkEncryptionEnforcement = exports.UserRuleKind = exports.UseHubGateway = exports.TransportProtocol = exports.TransformType = exports.Transform = exports.TrafficViewEnrollmentStatus = exports.TrafficRoutingMethod = exports.State = exports.SkuName = exports.SessionAffinityEnabledState = exports.ServiceProviderProvisioningState = exports.SecurityRuleProtocol = exports.SecurityRuleDirection = exports.SecurityRuleAccess = exports.SecurityProviderName = exports.SecurityConfigurationRuleProtocol = exports.SecurityConfigurationRuleDirection = exports.SecurityConfigurationRuleAccess = exports.ScrubbingRuleEntryState = exports.ScrubbingRuleEntryMatchVariable = exports.ScrubbingRuleEntryMatchOperator = exports.RulesEngineOperator = exports.RulesEngineMatchVariable = exports.RuleType = void 0;
exports.ZoneType = exports.WebApplicationFirewallTransform = exports.WebApplicationFirewallState = exports.WebApplicationFirewallScrubbingState = exports.WebApplicationFirewallRuleType = void 0;
// Export sub-modules:
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20180701 = require("./v20180701");
exports.v20180701 = v20180701;
const v20181201 = require("./v20181201");
exports.v20181201 = v20181201;
const v20190201 = require("./v20190201");
exports.v20190201 = v20190201;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220401 = require("./v20220401");
exports.v20220401 = v20220401;
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20220501 = require("./v20220501");
exports.v20220501 = v20220501;
const v20220701 = require("./v20220701");
exports.v20220701 = v20220701;
const v20230201 = require("./v20230201");
exports.v20230201 = v20230201;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20230601 = require("./v20230601");
exports.v20230601 = v20230601;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240201 = require("./v20240201");
exports.v20240201 = v20240201;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20240601 = require("./v20240601");
exports.v20240601 = v20240601;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
exports.Access = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.AccessRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.ActionType = {
    AnomalyScoring: "AnomalyScoring",
    Allow: "Allow",
    Block: "Block",
    Log: "Log",
};
exports.AddressPrefixType = {
    IPPrefix: "IPPrefix",
    ServiceTag: "ServiceTag",
};
exports.AdminRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
exports.AdminState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AllowedEndpointRecordType = {
    DomainName: "DomainName",
    IPv4Address: "IPv4Address",
    IPv6Address: "IPv6Address",
    Any: "Any",
};
exports.AlwaysServe = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ApplicationGatewayClientRevocationOptions = {
    None: "None",
    OCSP: "OCSP",
};
exports.ApplicationGatewayCookieBasedAffinity = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ApplicationGatewayCustomErrorStatusCode = {
    HttpStatus400: "HttpStatus400",
    HttpStatus403: "HttpStatus403",
    HttpStatus404: "HttpStatus404",
    HttpStatus405: "HttpStatus405",
    HttpStatus408: "HttpStatus408",
    HttpStatus500: "HttpStatus500",
    HttpStatus502: "HttpStatus502",
    HttpStatus503: "HttpStatus503",
    HttpStatus504: "HttpStatus504",
};
exports.ApplicationGatewayFirewallMode = {
    Detection: "Detection",
    Prevention: "Prevention",
};
exports.ApplicationGatewayFirewallRateLimitDuration = {
    OneMin: "OneMin",
    FiveMins: "FiveMins",
};
exports.ApplicationGatewayFirewallUserSessionVariable = {
    ClientAddr: "ClientAddr",
    GeoLocation: "GeoLocation",
    None: "None",
};
exports.ApplicationGatewayLoadDistributionAlgorithm = {
    RoundRobin: "RoundRobin",
    LeastConnections: "LeastConnections",
    IpHash: "IpHash",
};
exports.ApplicationGatewayProtocol = {
    Http: "Http",
    Https: "Https",
    Tcp: "Tcp",
    Tls: "Tls",
};
exports.ApplicationGatewayRedirectType = {
    Permanent: "Permanent",
    Found: "Found",
    SeeOther: "SeeOther",
    Temporary: "Temporary",
};
exports.ApplicationGatewayRequestRoutingRuleType = {
    Basic: "Basic",
    PathBasedRouting: "PathBasedRouting",
};
exports.ApplicationGatewaySkuName = {
    Standard_Small: "Standard_Small",
    Standard_Medium: "Standard_Medium",
    Standard_Large: "Standard_Large",
    WAF_Medium: "WAF_Medium",
    WAF_Large: "WAF_Large",
    Standard_v2: "Standard_v2",
    WAF_v2: "WAF_v2",
};
exports.ApplicationGatewaySslCipherSuite = {
    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384: "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
    TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256: "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA: "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
    TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA: "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
    TLS_DHE_RSA_WITH_AES_256_GCM_SHA384: "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384",
    TLS_DHE_RSA_WITH_AES_128_GCM_SHA256: "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
    TLS_DHE_RSA_WITH_AES_256_CBC_SHA: "TLS_DHE_RSA_WITH_AES_256_CBC_SHA",
    TLS_DHE_RSA_WITH_AES_128_CBC_SHA: "TLS_DHE_RSA_WITH_AES_128_CBC_SHA",
    TLS_RSA_WITH_AES_256_GCM_SHA384: "TLS_RSA_WITH_AES_256_GCM_SHA384",
    TLS_RSA_WITH_AES_128_GCM_SHA256: "TLS_RSA_WITH_AES_128_GCM_SHA256",
    TLS_RSA_WITH_AES_256_CBC_SHA256: "TLS_RSA_WITH_AES_256_CBC_SHA256",
    TLS_RSA_WITH_AES_128_CBC_SHA256: "TLS_RSA_WITH_AES_128_CBC_SHA256",
    TLS_RSA_WITH_AES_256_CBC_SHA: "TLS_RSA_WITH_AES_256_CBC_SHA",
    TLS_RSA_WITH_AES_128_CBC_SHA: "TLS_RSA_WITH_AES_128_CBC_SHA",
    TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384: "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
    TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
    TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384: "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
    TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA: "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA: "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
    TLS_DHE_DSS_WITH_AES_256_CBC_SHA256: "TLS_DHE_DSS_WITH_AES_256_CBC_SHA256",
    TLS_DHE_DSS_WITH_AES_128_CBC_SHA256: "TLS_DHE_DSS_WITH_AES_128_CBC_SHA256",
    TLS_DHE_DSS_WITH_AES_256_CBC_SHA: "TLS_DHE_DSS_WITH_AES_256_CBC_SHA",
    TLS_DHE_DSS_WITH_AES_128_CBC_SHA: "TLS_DHE_DSS_WITH_AES_128_CBC_SHA",
    TLS_RSA_WITH_3DES_EDE_CBC_SHA: "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
    TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA: "TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA",
    TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256: "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
    TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
};
exports.ApplicationGatewaySslPolicyName = {
    AppGwSslPolicy20150501: "AppGwSslPolicy20150501",
    AppGwSslPolicy20170401: "AppGwSslPolicy20170401",
    AppGwSslPolicy20170401S: "AppGwSslPolicy20170401S",
    AppGwSslPolicy20220101: "AppGwSslPolicy20220101",
    AppGwSslPolicy20220101S: "AppGwSslPolicy20220101S",
};
exports.ApplicationGatewaySslPolicyType = {
    Predefined: "Predefined",
    Custom: "Custom",
    CustomV2: "CustomV2",
};
exports.ApplicationGatewaySslProtocol = {
    TLSv1_0: "TLSv1_0",
    TLSv1_1: "TLSv1_1",
    TLSv1_2: "TLSv1_2",
    TLSv1_3: "TLSv1_3",
};
exports.ApplicationGatewayTier = {
    Standard: "Standard",
    WAF: "WAF",
    Standard_v2: "Standard_v2",
    WAF_v2: "WAF_v2",
};
exports.AssociationAccessMode = {
    Learning: "Learning",
    Enforced: "Enforced",
    Audit: "Audit",
};
exports.AuthorizationUseStatus = {
    Available: "Available",
    InUse: "InUse",
};
exports.AutoLearnPrivateRangesMode = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AzureFirewallApplicationRuleProtocolType = {
    Http: "Http",
    Https: "Https",
    Mssql: "Mssql",
};
exports.AzureFirewallNatRCActionType = {
    Snat: "Snat",
    Dnat: "Dnat",
};
exports.AzureFirewallNetworkRuleProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
    ICMP: "ICMP",
};
exports.AzureFirewallRCActionType = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.AzureFirewallSkuName = {
    AZFW_VNet: "AZFW_VNet",
    AZFW_Hub: "AZFW_Hub",
};
exports.AzureFirewallSkuTier = {
    Standard: "Standard",
    Premium: "Premium",
    Basic: "Basic",
};
exports.AzureFirewallThreatIntelMode = {
    Alert: "Alert",
    Deny: "Deny",
    Off: "Off",
};
exports.BackendEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.BastionHostSkuName = {
    Basic: "Basic",
    Standard: "Standard",
};
exports.BlockResponseCode = {
    SERVFAIL: "SERVFAIL",
};
exports.CommissionedState = {
    Provisioning: "Provisioning",
    Provisioned: "Provisioned",
    Commissioning: "Commissioning",
    CommissionedNoInternetAdvertise: "CommissionedNoInternetAdvertise",
    Commissioned: "Commissioned",
    Decommissioning: "Decommissioning",
    Deprovisioning: "Deprovisioning",
    Deprovisioned: "Deprovisioned",
};
exports.ConfigurationType = {
    SecurityAdmin: "SecurityAdmin",
    Connectivity: "Connectivity",
};
exports.ConnectionMonitorEndpointFilterItemType = {
    AgentAddress: "AgentAddress",
};
exports.ConnectionMonitorEndpointFilterType = {
    Include: "Include",
};
exports.ConnectionMonitorTestConfigurationProtocol = {
    Tcp: "Tcp",
    Http: "Http",
    Icmp: "Icmp",
};
exports.ConnectivityTopology = {
    HubAndSpoke: "HubAndSpoke",
    Mesh: "Mesh",
};
exports.CoverageLevel = {
    Default: "Default",
    Low: "Low",
    BelowAverage: "BelowAverage",
    Average: "Average",
    AboveAverage: "AboveAverage",
    Full: "Full",
};
exports.CustomIpPrefixType = {
    Singular: "Singular",
    Parent: "Parent",
    Child: "Child",
};
exports.CustomRuleEnabledState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.DdosSettingsProtectionMode = {
    VirtualNetworkInherited: "VirtualNetworkInherited",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DeleteExistingNSGs = {
    False: "False",
    True: "True",
};
exports.DeleteExistingPeering = {
    False: "False",
    True: "True",
};
exports.DeleteOptions = {
    Delete: "Delete",
    Detach: "Detach",
};
exports.DestinationPortBehavior = {
    None: "None",
    ListenIfAvailable: "ListenIfAvailable",
};
exports.DhGroup = {
    None: "None",
    DHGroup1: "DHGroup1",
    DHGroup2: "DHGroup2",
    DHGroup14: "DHGroup14",
    DHGroup2048: "DHGroup2048",
    ECP256: "ECP256",
    ECP384: "ECP384",
    DHGroup24: "DHGroup24",
};
exports.DisableBgpRoutePropagation = {
    False: "False",
    True: "True",
};
exports.DnsSecurityRuleState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DynamicCompressionEnabled = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EndpointMonitorStatus = {
    CheckingEndpoint: "CheckingEndpoint",
    Online: "Online",
    Degraded: "Degraded",
    Disabled: "Disabled",
    Inactive: "Inactive",
    Stopped: "Stopped",
    Unmonitored: "Unmonitored",
};
exports.EndpointStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EndpointType = {
    AzureVM: "AzureVM",
    AzureVNet: "AzureVNet",
    AzureSubnet: "AzureSubnet",
    ExternalAddress: "ExternalAddress",
    MMAWorkspaceMachine: "MMAWorkspaceMachine",
    MMAWorkspaceNetwork: "MMAWorkspaceNetwork",
    AzureArcVM: "AzureArcVM",
    AzureVMSS: "AzureVMSS",
};
exports.EnforceCertificateNameCheckEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ExpressRouteCircuitPeeringState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ExpressRouteCircuitSkuFamily = {
    UnlimitedData: "UnlimitedData",
    MeteredData: "MeteredData",
};
exports.ExpressRouteCircuitSkuTier = {
    Standard: "Standard",
    Premium: "Premium",
    Basic: "Basic",
    Local: "Local",
};
exports.ExpressRouteLinkAdminState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ExpressRouteLinkMacSecCipher = {
    GcmAes256: "GcmAes256",
    GcmAes128: "GcmAes128",
    GcmAesXpn128: "GcmAesXpn128",
    GcmAesXpn256: "GcmAesXpn256",
};
exports.ExpressRouteLinkMacSecSciState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ExpressRoutePeeringState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ExpressRoutePeeringType = {
    AzurePublicPeering: "AzurePublicPeering",
    AzurePrivatePeering: "AzurePrivatePeering",
    MicrosoftPeering: "MicrosoftPeering",
};
exports.ExpressRoutePortsBillingType = {
    MeteredData: "MeteredData",
    UnlimitedData: "UnlimitedData",
};
exports.ExpressRoutePortsEncapsulation = {
    Dot1Q: "Dot1Q",
    QinQ: "QinQ",
};
exports.ExtendedLocationTypes = {
    EdgeZone: "EdgeZone",
};
exports.FirewallPolicyFilterRuleActionType = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FirewallPolicyFilterRuleCollectionActionType = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FirewallPolicyIDPSQuerySortOrder = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.FirewallPolicyIntrusionDetectionProfileType = {
    Basic: "Basic",
    Standard: "Standard",
    Advanced: "Advanced",
    Extended: "Extended",
};
exports.FirewallPolicyIntrusionDetectionProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    ICMP: "ICMP",
    ANY: "ANY",
};
exports.FirewallPolicyIntrusionDetectionStateType = {
    Off: "Off",
    Alert: "Alert",
    Deny: "Deny",
};
exports.FirewallPolicyNatRuleActionType = {
    DNAT: "DNAT",
};
exports.FirewallPolicyNatRuleCollectionActionType = {
    DNAT: "DNAT",
};
exports.FirewallPolicyRuleApplicationProtocolType = {
    Http: "Http",
    Https: "Https",
};
exports.FirewallPolicyRuleCollectionType = {
    FirewallPolicyNatRuleCollection: "FirewallPolicyNatRuleCollection",
    FirewallPolicyFilterRuleCollection: "FirewallPolicyFilterRuleCollection",
};
exports.FirewallPolicyRuleConditionApplicationProtocolType = {
    Http: "Http",
    Https: "Https",
};
exports.FirewallPolicyRuleConditionNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
    ICMP: "ICMP",
};
exports.FirewallPolicyRuleConditionType = {
    ApplicationRuleCondition: "ApplicationRuleCondition",
    NetworkRuleCondition: "NetworkRuleCondition",
    NatRuleCondition: "NatRuleCondition",
};
exports.FirewallPolicyRuleNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
    ICMP: "ICMP",
};
exports.FirewallPolicyRuleType = {
    FirewallPolicyNatRule: "FirewallPolicyNatRule",
    FirewallPolicyFilterRule: "FirewallPolicyFilterRule",
};
exports.FirewallPolicySkuTier = {
    Standard: "Standard",
    Premium: "Premium",
    Basic: "Basic",
};
exports.FlowLogFormatType = {
    JSON: "JSON",
};
exports.ForwardingRuleState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FrontDoorEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FrontDoorForwardingProtocol = {
    HttpOnly: "HttpOnly",
    HttpsOnly: "HttpsOnly",
    MatchRequest: "MatchRequest",
};
exports.FrontDoorHealthProbeMethod = {
    GET: "GET",
    HEAD: "HEAD",
};
exports.FrontDoorMatchVariable = {
    RemoteAddr: "RemoteAddr",
    RequestMethod: "RequestMethod",
    QueryString: "QueryString",
    PostArgs: "PostArgs",
    RequestUri: "RequestUri",
    RequestHeader: "RequestHeader",
    RequestBody: "RequestBody",
    Cookies: "Cookies",
    SocketAddr: "SocketAddr",
};
exports.FrontDoorProtocol = {
    Http: "Http",
    Https: "Https",
};
exports.FrontDoorQuery = {
    StripNone: "StripNone",
    StripAll: "StripAll",
    StripOnly: "StripOnly",
    StripAllExcept: "StripAllExcept",
};
exports.FrontDoorRedirectProtocol = {
    HttpOnly: "HttpOnly",
    HttpsOnly: "HttpsOnly",
    MatchRequest: "MatchRequest",
};
exports.FrontDoorRedirectType = {
    Moved: "Moved",
    Found: "Found",
    TemporaryRedirect: "TemporaryRedirect",
    PermanentRedirect: "PermanentRedirect",
};
exports.GatewayLoadBalancerTunnelInterfaceType = {
    None: "None",
    Internal: "Internal",
    External: "External",
};
exports.GatewayLoadBalancerTunnelProtocol = {
    None: "None",
    Native: "Native",
    VXLAN: "VXLAN",
};
exports.Geo = {
    GLOBAL: "GLOBAL",
    AFRI: "AFRI",
    APAC: "APAC",
    EURO: "EURO",
    LATAM: "LATAM",
    NAM: "NAM",
    ME: "ME",
    OCEANIA: "OCEANIA",
    AQ: "AQ",
};
exports.GroupConnectivity = {
    None: "None",
    DirectlyConnected: "DirectlyConnected",
};
exports.HTTPConfigurationMethod = {
    Get: "Get",
    Post: "Post",
};
exports.HeaderActionType = {
    Append: "Append",
    Delete: "Delete",
    Overwrite: "Overwrite",
};
exports.HealthProbeEnabled = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HubRoutingPreference = {
    ExpressRoute: "ExpressRoute",
    VpnGateway: "VpnGateway",
    ASPath: "ASPath",
};
exports.IPAllocationMethod = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.IPVersion = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.IkeEncryption = {
    DES: "DES",
    DES3: "DES3",
    AES128: "AES128",
    AES192: "AES192",
    AES256: "AES256",
    GCMAES256: "GCMAES256",
    GCMAES128: "GCMAES128",
};
exports.IkeIntegrity = {
    MD5: "MD5",
    SHA1: "SHA1",
    SHA256: "SHA256",
    SHA384: "SHA384",
    GCMAES256: "GCMAES256",
    GCMAES128: "GCMAES128",
};
exports.IpAllocationMethod = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.IpAllocationType = {
    Undefined: "Undefined",
    Hypernet: "Hypernet",
};
exports.IpsecEncryption = {
    None: "None",
    DES: "DES",
    DES3: "DES3",
    AES128: "AES128",
    AES192: "AES192",
    AES256: "AES256",
    GCMAES128: "GCMAES128",
    GCMAES192: "GCMAES192",
    GCMAES256: "GCMAES256",
};
exports.IpsecIntegrity = {
    MD5: "MD5",
    SHA1: "SHA1",
    SHA256: "SHA256",
    GCMAES128: "GCMAES128",
    GCMAES192: "GCMAES192",
    GCMAES256: "GCMAES256",
};
exports.IsGlobal = {
    False: "False",
    True: "True",
};
exports.LoadBalancerBackendAddressAdminState = {
    None: "None",
    Up: "Up",
    Down: "Down",
};
exports.LoadBalancerOutboundRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    All: "All",
};
exports.LoadBalancerSkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Gateway: "Gateway",
};
exports.LoadBalancerSkuTier = {
    Regional: "Regional",
    Global: "Global",
};
exports.LoadDistribution = {
    Default: "Default",
    SourceIP: "SourceIP",
    SourceIPProtocol: "SourceIPProtocol",
};
exports.ManagedRuleEnabledState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedRuleExclusionMatchVariable = {
    RequestHeaderNames: "RequestHeaderNames",
    RequestCookieNames: "RequestCookieNames",
    QueryStringArgNames: "QueryStringArgNames",
    RequestBodyPostArgNames: "RequestBodyPostArgNames",
    RequestBodyJsonArgNames: "RequestBodyJsonArgNames",
};
exports.ManagedRuleExclusionSelectorMatchOperator = {
    Equals: "Equals",
    Contains: "Contains",
    StartsWith: "StartsWith",
    EndsWith: "EndsWith",
    EqualsAny: "EqualsAny",
};
exports.ManagedRuleSetActionType = {
    Block: "Block",
    Log: "Log",
    Redirect: "Redirect",
};
exports.MatchProcessingBehavior = {
    Continue: "Continue",
    Stop: "Stop",
};
exports.MonitorProtocol = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    TCP: "TCP",
};
exports.NatGatewaySkuName = {
    Standard: "Standard",
};
exports.NetworkIntentPolicyBasedService = {
    None: "None",
    All: "All",
    AllowRulesOnly: "AllowRulesOnly",
};
exports.NetworkInterfaceAuxiliaryMode = {
    None: "None",
    MaxConnections: "MaxConnections",
    Floating: "Floating",
    AcceleratedConnections: "AcceleratedConnections",
};
exports.NetworkInterfaceAuxiliarySku = {
    None: "None",
    A1: "A1",
    A2: "A2",
    A4: "A4",
    A8: "A8",
};
exports.NetworkInterfaceMigrationPhase = {
    None: "None",
    Prepare: "Prepare",
    Commit: "Commit",
    Abort: "Abort",
    Committed: "Committed",
};
exports.NetworkInterfaceNicType = {
    Standard: "Standard",
    Elastic: "Elastic",
};
exports.NetworkProtocol = {
    Any: "Any",
    TCP: "TCP",
    UDP: "UDP",
    ICMP: "ICMP",
};
exports.NextStep = {
    Unknown: "Unknown",
    Continue: "Continue",
    Terminate: "Terminate",
};
exports.Operator = {
    Any: "Any",
    IPMatch: "IPMatch",
    GeoMatch: "GeoMatch",
    Equal: "Equal",
    Contains: "Contains",
    LessThan: "LessThan",
    GreaterThan: "GreaterThan",
    LessThanOrEqual: "LessThanOrEqual",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    BeginsWith: "BeginsWith",
    EndsWith: "EndsWith",
    RegEx: "RegEx",
};
exports.OutputType = {
    Workspace: "Workspace",
};
exports.OwaspCrsExclusionEntryMatchVariable = {
    RequestHeaderNames: "RequestHeaderNames",
    RequestCookieNames: "RequestCookieNames",
    RequestArgNames: "RequestArgNames",
    RequestHeaderKeys: "RequestHeaderKeys",
    RequestHeaderValues: "RequestHeaderValues",
    RequestCookieKeys: "RequestCookieKeys",
    RequestCookieValues: "RequestCookieValues",
    RequestArgKeys: "RequestArgKeys",
    RequestArgValues: "RequestArgValues",
};
exports.OwaspCrsExclusionEntrySelectorMatchOperator = {
    Equals: "Equals",
    Contains: "Contains",
    StartsWith: "StartsWith",
    EndsWith: "EndsWith",
    EqualsAny: "EqualsAny",
};
exports.PacketCaptureTargetType = {
    AzureVM: "AzureVM",
    AzureVMSS: "AzureVMSS",
};
exports.PcProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
};
exports.PfsGroup = {
    None: "None",
    PFS1: "PFS1",
    PFS2: "PFS2",
    PFS2048: "PFS2048",
    ECP256: "ECP256",
    ECP384: "ECP384",
    PFS24: "PFS24",
    PFS14: "PFS14",
    PFSMM: "PFSMM",
};
exports.PolicyEnabledState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.PolicyMode = {
    Prevention: "Prevention",
    Detection: "Detection",
};
exports.PolicyRequestBodyCheck = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.PreferredIPVersion = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.PreferredRoutingGateway = {
    ExpressRoute: "ExpressRoute",
    VpnGateway: "VpnGateway",
    None: "None",
};
exports.ProbeProtocol = {
    Http: "Http",
    Tcp: "Tcp",
    Https: "Https",
};
exports.ProfileMonitorStatus = {
    CheckingEndpoints: "CheckingEndpoints",
    Online: "Online",
    Degraded: "Degraded",
    Disabled: "Disabled",
    Inactive: "Inactive",
};
exports.ProfileStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ProtocolType = {
    DoNotUse: "DoNotUse",
    Icmp: "Icmp",
    Tcp: "Tcp",
    Udp: "Udp",
    Gre: "Gre",
    Esp: "Esp",
    Ah: "Ah",
    Vxlan: "Vxlan",
    All: "All",
};
exports.PublicIPAddressMigrationPhase = {
    None: "None",
    Prepare: "Prepare",
    Commit: "Commit",
    Abort: "Abort",
    Committed: "Committed",
};
exports.PublicIPAddressSkuName = {
    Basic: "Basic",
    Standard: "Standard",
};
exports.PublicIPAddressSkuTier = {
    Regional: "Regional",
    Global: "Global",
};
exports.PublicIPPrefixSkuName = {
    Standard: "Standard",
};
exports.PublicIPPrefixSkuTier = {
    Regional: "Regional",
    Global: "Global",
};
exports.PublicIpAddressDnsSettingsDomainNameLabelScope = {
    TenantReuse: "TenantReuse",
    SubscriptionReuse: "SubscriptionReuse",
    ResourceGroupReuse: "ResourceGroupReuse",
    NoReuse: "NoReuse",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.RouteFilterRuleType = {
    Community: "Community",
};
exports.RouteMapActionType = {
    Unknown: "Unknown",
    Remove: "Remove",
    Add: "Add",
    Replace: "Replace",
    Drop: "Drop",
};
exports.RouteMapMatchCondition = {
    Unknown: "Unknown",
    Contains: "Contains",
    Equals: "Equals",
    NotContains: "NotContains",
    NotEquals: "NotEquals",
};
exports.RouteNextHopType = {
    VirtualNetworkGateway: "VirtualNetworkGateway",
    VnetLocal: "VnetLocal",
    Internet: "Internet",
    VirtualAppliance: "VirtualAppliance",
    None: "None",
};
exports.RoutingRuleDestinationType = {
    AddressPrefix: "AddressPrefix",
    ServiceTag: "ServiceTag",
};
exports.RoutingRuleEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RoutingRuleNextHopType = {
    Internet: "Internet",
    NoNextHop: "NoNextHop",
    VirtualAppliance: "VirtualAppliance",
    VirtualNetworkGateway: "VirtualNetworkGateway",
    VnetLocal: "VnetLocal",
};
exports.RuleType = {
    MatchRule: "MatchRule",
    RateLimitRule: "RateLimitRule",
};
exports.RulesEngineMatchVariable = {
    IsMobile: "IsMobile",
    RemoteAddr: "RemoteAddr",
    RequestMethod: "RequestMethod",
    QueryString: "QueryString",
    PostArgs: "PostArgs",
    RequestUri: "RequestUri",
    RequestPath: "RequestPath",
    RequestFilename: "RequestFilename",
    RequestFilenameExtension: "RequestFilenameExtension",
    RequestHeader: "RequestHeader",
    RequestBody: "RequestBody",
    RequestScheme: "RequestScheme",
};
exports.RulesEngineOperator = {
    Any: "Any",
    IPMatch: "IPMatch",
    GeoMatch: "GeoMatch",
    Equal: "Equal",
    Contains: "Contains",
    LessThan: "LessThan",
    GreaterThan: "GreaterThan",
    LessThanOrEqual: "LessThanOrEqual",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    BeginsWith: "BeginsWith",
    EndsWith: "EndsWith",
};
exports.ScrubbingRuleEntryMatchOperator = {
    Equals: "Equals",
    EqualsAny: "EqualsAny",
};
exports.ScrubbingRuleEntryMatchVariable = {
    RequestHeaderNames: "RequestHeaderNames",
    RequestCookieNames: "RequestCookieNames",
    RequestArgNames: "RequestArgNames",
    RequestPostArgNames: "RequestPostArgNames",
    RequestJSONArgNames: "RequestJSONArgNames",
    RequestIPAddress: "RequestIPAddress",
};
exports.ScrubbingRuleEntryState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SecurityConfigurationRuleAccess = {
    Allow: "Allow",
    Deny: "Deny",
    AlwaysAllow: "AlwaysAllow",
};
exports.SecurityConfigurationRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.SecurityConfigurationRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    Icmp: "Icmp",
    Esp: "Esp",
    Any: "Any",
    Ah: "Ah",
};
exports.SecurityProviderName = {
    ZScaler: "ZScaler",
    IBoss: "IBoss",
    Checkpoint: "Checkpoint",
};
exports.SecurityRuleAccess = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.SecurityRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.SecurityRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    Icmp: "Icmp",
    Esp: "Esp",
    Asterisk: "*",
    Ah: "Ah",
};
exports.ServiceProviderProvisioningState = {
    NotProvisioned: "NotProvisioned",
    Provisioning: "Provisioning",
    Provisioned: "Provisioned",
    Deprovisioning: "Deprovisioning",
};
exports.SessionAffinityEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SkuName = {
    Classic_AzureFrontDoor: "Classic_AzureFrontDoor",
    Standard_AzureFrontDoor: "Standard_AzureFrontDoor",
    Premium_AzureFrontDoor: "Premium_AzureFrontDoor",
};
exports.State = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TrafficRoutingMethod = {
    Performance: "Performance",
    Priority: "Priority",
    Weighted: "Weighted",
    Geographic: "Geographic",
    MultiValue: "MultiValue",
    Subnet: "Subnet",
};
exports.TrafficViewEnrollmentStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Transform = {
    Lowercase: "Lowercase",
    Uppercase: "Uppercase",
    Trim: "Trim",
    UrlDecode: "UrlDecode",
    UrlEncode: "UrlEncode",
    RemoveNulls: "RemoveNulls",
};
exports.TransformType = {
    Lowercase: "Lowercase",
    Uppercase: "Uppercase",
    Trim: "Trim",
    UrlDecode: "UrlDecode",
    UrlEncode: "UrlEncode",
    RemoveNulls: "RemoveNulls",
};
exports.TransportProtocol = {
    Udp: "Udp",
    Tcp: "Tcp",
    All: "All",
};
exports.UseHubGateway = {
    False: "False",
    True: "True",
};
exports.UserRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
exports.VirtualNetworkEncryptionEnforcement = {
    DropUnencrypted: "DropUnencrypted",
    AllowUnencrypted: "AllowUnencrypted",
};
exports.VirtualNetworkGatewayConnectionMode = {
    Default: "Default",
    ResponderOnly: "ResponderOnly",
    InitiatorOnly: "InitiatorOnly",
};
exports.VirtualNetworkGatewayConnectionProtocol = {
    IKEv2: "IKEv2",
    IKEv1: "IKEv1",
};
exports.VirtualNetworkGatewayConnectionType = {
    IPsec: "IPsec",
    Vnet2Vnet: "Vnet2Vnet",
    ExpressRoute: "ExpressRoute",
    VPNClient: "VPNClient",
};
exports.VirtualNetworkGatewaySkuName = {
    Basic: "Basic",
    HighPerformance: "HighPerformance",
    Standard: "Standard",
    UltraPerformance: "UltraPerformance",
    VpnGw1: "VpnGw1",
    VpnGw2: "VpnGw2",
    VpnGw3: "VpnGw3",
    VpnGw4: "VpnGw4",
    VpnGw5: "VpnGw5",
    VpnGw1AZ: "VpnGw1AZ",
    VpnGw2AZ: "VpnGw2AZ",
    VpnGw3AZ: "VpnGw3AZ",
    VpnGw4AZ: "VpnGw4AZ",
    VpnGw5AZ: "VpnGw5AZ",
    ErGw1AZ: "ErGw1AZ",
    ErGw2AZ: "ErGw2AZ",
    ErGw3AZ: "ErGw3AZ",
};
exports.VirtualNetworkGatewaySkuTier = {
    Basic: "Basic",
    HighPerformance: "HighPerformance",
    Standard: "Standard",
    UltraPerformance: "UltraPerformance",
    VpnGw1: "VpnGw1",
    VpnGw2: "VpnGw2",
    VpnGw3: "VpnGw3",
    VpnGw4: "VpnGw4",
    VpnGw5: "VpnGw5",
    VpnGw1AZ: "VpnGw1AZ",
    VpnGw2AZ: "VpnGw2AZ",
    VpnGw3AZ: "VpnGw3AZ",
    VpnGw4AZ: "VpnGw4AZ",
    VpnGw5AZ: "VpnGw5AZ",
    ErGw1AZ: "ErGw1AZ",
    ErGw2AZ: "ErGw2AZ",
    ErGw3AZ: "ErGw3AZ",
};
exports.VirtualNetworkGatewayType = {
    Vpn: "Vpn",
    ExpressRoute: "ExpressRoute",
    LocalGateway: "LocalGateway",
};
exports.VirtualNetworkPeeringLevel = {
    FullyInSync: "FullyInSync",
    RemoteNotInSync: "RemoteNotInSync",
    LocalNotInSync: "LocalNotInSync",
    LocalAndRemoteNotInSync: "LocalAndRemoteNotInSync",
};
exports.VirtualNetworkPeeringState = {
    Initiated: "Initiated",
    Connected: "Connected",
    Disconnected: "Disconnected",
};
exports.VirtualNetworkPrivateEndpointNetworkPolicies = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VirtualNetworkPrivateLinkServiceNetworkPolicies = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VnetLocalRouteOverrideCriteria = {
    Contains: "Contains",
    Equal: "Equal",
};
exports.VpnAuthenticationType = {
    Certificate: "Certificate",
    Radius: "Radius",
    AAD: "AAD",
};
exports.VpnClientProtocol = {
    IkeV2: "IkeV2",
    SSTP: "SSTP",
    OpenVPN: "OpenVPN",
};
exports.VpnGatewayGeneration = {
    None: "None",
    Generation1: "Generation1",
    Generation2: "Generation2",
};
exports.VpnGatewayTunnelingProtocol = {
    IkeV2: "IkeV2",
    OpenVPN: "OpenVPN",
};
exports.VpnLinkConnectionMode = {
    Default: "Default",
    ResponderOnly: "ResponderOnly",
    InitiatorOnly: "InitiatorOnly",
};
exports.VpnNatRuleMode = {
    EgressSnat: "EgressSnat",
    IngressSnat: "IngressSnat",
};
exports.VpnNatRuleType = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.VpnPolicyMemberAttributeType = {
    CertificateGroupId: "CertificateGroupId",
    AADGroupId: "AADGroupId",
    RadiusAzureGroupId: "RadiusAzureGroupId",
};
exports.VpnType = {
    PolicyBased: "PolicyBased",
    RouteBased: "RouteBased",
};
exports.WebApplicationFirewallAction = {
    Allow: "Allow",
    Block: "Block",
    Log: "Log",
};
exports.WebApplicationFirewallEnabledState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.WebApplicationFirewallMatchVariable = {
    RemoteAddr: "RemoteAddr",
    RequestMethod: "RequestMethod",
    QueryString: "QueryString",
    PostArgs: "PostArgs",
    RequestUri: "RequestUri",
    RequestHeaders: "RequestHeaders",
    RequestBody: "RequestBody",
    RequestCookies: "RequestCookies",
};
exports.WebApplicationFirewallMode = {
    Prevention: "Prevention",
    Detection: "Detection",
};
exports.WebApplicationFirewallOperator = {
    IPMatch: "IPMatch",
    Equal: "Equal",
    Contains: "Contains",
    LessThan: "LessThan",
    GreaterThan: "GreaterThan",
    LessThanOrEqual: "LessThanOrEqual",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    BeginsWith: "BeginsWith",
    EndsWith: "EndsWith",
    Regex: "Regex",
    GeoMatch: "GeoMatch",
    Any: "Any",
};
exports.WebApplicationFirewallRuleType = {
    MatchRule: "MatchRule",
    RateLimitRule: "RateLimitRule",
    Invalid: "Invalid",
};
exports.WebApplicationFirewallScrubbingState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.WebApplicationFirewallState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.WebApplicationFirewallTransform = {
    Uppercase: "Uppercase",
    Lowercase: "Lowercase",
    Trim: "Trim",
    UrlDecode: "UrlDecode",
    UrlEncode: "UrlEncode",
    RemoveNulls: "RemoveNulls",
    HtmlEntityDecode: "HtmlEntityDecode",
};
exports.ZoneType = {
    Public: "Public",
    Private: "Private",
};
