// @pulumi/azure-native/iotoperationsorchestrator/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20231004preview = exports.Target = exports.Solution = exports.Instance = exports.getTargetOutput = exports.getTarget = exports.getSolutionOutput = exports.getSolution = exports.getInstanceOutput = exports.getInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getSolution = null;
exports.getSolutionOutput = null;
utilities.lazyLoad(exports, ["getSolution", "getSolutionOutput"], () => require("./getSolution"));
exports.getTarget = null;
exports.getTargetOutput = null;
utilities.lazyLoad(exports, ["getTarget", "getTargetOutput"], () => require("./getTarget"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.Solution = null;
utilities.lazyLoad(exports, ["Solution"], () => require("./solution"));
exports.Target = null;
utilities.lazyLoad(exports, ["Target"], () => require("./target"));
// Export enums:
__exportStar(require("../types/enums/iotoperationsorchestrator"), exports);
// Export sub-modules:
const v20231004preview = require("./v20231004preview");
exports.v20231004preview = v20231004preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:iotoperationsorchestrator:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "azure-native:iotoperationsorchestrator:Solution":
                return new exports.Solution(name, undefined, { urn });
            case "azure-native:iotoperationsorchestrator:Target":
                return new exports.Target(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "iotoperationsorchestrator", _module);
