// @pulumi/azure-native/sql/v20170301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerSecurityAlertPolicy = exports.getServerSecurityAlertPolicyOutput = exports.getServerSecurityAlertPolicy = exports.getBackupLongTermRetentionPolicyOutput = exports.getBackupLongTermRetentionPolicy = exports.BackupLongTermRetentionPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BackupLongTermRetentionPolicy = null;
utilities.lazyLoad(exports, ["BackupLongTermRetentionPolicy"], () => require("./backupLongTermRetentionPolicy"));
exports.getBackupLongTermRetentionPolicy = null;
exports.getBackupLongTermRetentionPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackupLongTermRetentionPolicy", "getBackupLongTermRetentionPolicyOutput"], () => require("./getBackupLongTermRetentionPolicy"));
exports.getServerSecurityAlertPolicy = null;
exports.getServerSecurityAlertPolicyOutput = null;
utilities.lazyLoad(exports, ["getServerSecurityAlertPolicy", "getServerSecurityAlertPolicyOutput"], () => require("./getServerSecurityAlertPolicy"));
exports.ServerSecurityAlertPolicy = null;
utilities.lazyLoad(exports, ["ServerSecurityAlertPolicy"], () => require("./serverSecurityAlertPolicy"));
// Export enums:
__exportStar(require("../../types/enums/sql/v20170301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:sql/v20170301preview:BackupLongTermRetentionPolicy":
                return new exports.BackupLongTermRetentionPolicy(name, undefined, { urn });
            case "azure-native:sql/v20170301preview:ServerSecurityAlertPolicy":
                return new exports.ServerSecurityAlertPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "sql/v20170301preview", _module);
