// @pulumi/azure-native/documentdb/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSqlResourceSqlRoleAssignment = exports.getSqlResourceSqlDatabaseOutput = exports.getSqlResourceSqlDatabase = exports.getSqlResourceSqlContainerOutput = exports.getSqlResourceSqlContainer = exports.getServiceOutput = exports.getService = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNotebookWorkspaceOutput = exports.getNotebookWorkspace = exports.getMongoDBResourceMongoUserDefinitionOutput = exports.getMongoDBResourceMongoUserDefinition = exports.getMongoDBResourceMongoRoleDefinitionOutput = exports.getMongoDBResourceMongoRoleDefinition = exports.getMongoDBResourceMongoDBDatabaseOutput = exports.getMongoDBResourceMongoDBDatabase = exports.getMongoDBResourceMongoDBCollectionOutput = exports.getMongoDBResourceMongoDBCollection = exports.getMongoClusterFirewallRuleOutput = exports.getMongoClusterFirewallRule = exports.getMongoClusterOutput = exports.getMongoCluster = exports.getGremlinResourceGremlinGraphOutput = exports.getGremlinResourceGremlinGraph = exports.getGremlinResourceGremlinDatabaseOutput = exports.getGremlinResourceGremlinDatabase = exports.getGraphResourceGraphOutput = exports.getGraphResourceGraph = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getDatabaseAccountOutput = exports.getDatabaseAccount = exports.getCassandraResourceCassandraViewOutput = exports.getCassandraResourceCassandraView = exports.getCassandraResourceCassandraTableOutput = exports.getCassandraResourceCassandraTable = exports.getCassandraResourceCassandraKeyspaceOutput = exports.getCassandraResourceCassandraKeyspace = exports.getCassandraDataCenterOutput = exports.getCassandraDataCenter = exports.getCassandraClusterOutput = exports.getCassandraCluster = exports.FirewallRule = exports.DatabaseAccount = exports.CassandraResourceCassandraView = exports.CassandraResourceCassandraTable = exports.CassandraResourceCassandraKeyspace = exports.CassandraDataCenter = exports.CassandraCluster = void 0;
exports.ThroughputPool = exports.TableResourceTableRoleDefinition = exports.TableResourceTableRoleAssignment = exports.TableResourceTable = exports.SqlResourceSqlUserDefinedFunction = exports.SqlResourceSqlTrigger = exports.SqlResourceSqlStoredProcedure = exports.SqlResourceSqlRoleDefinition = exports.SqlResourceSqlRoleAssignment = exports.SqlResourceSqlDatabase = exports.SqlResourceSqlContainer = exports.Service = exports.PrivateEndpointConnection = exports.NotebookWorkspace = exports.MongoDBResourceMongoUserDefinition = exports.MongoDBResourceMongoRoleDefinition = exports.MongoDBResourceMongoDBDatabase = exports.MongoDBResourceMongoDBCollection = exports.MongoClusterFirewallRule = exports.MongoCluster = exports.listNotebookWorkspaceConnectionInfoOutput = exports.listNotebookWorkspaceConnectionInfo = exports.listMongoClusterConnectionStringsOutput = exports.listMongoClusterConnectionStrings = exports.listDatabaseAccountKeysOutput = exports.listDatabaseAccountKeys = exports.listDatabaseAccountConnectionStringsOutput = exports.listDatabaseAccountConnectionStrings = exports.GremlinResourceGremlinGraph = exports.GremlinResourceGremlinDatabase = exports.GraphResourceGraph = exports.getThroughputPoolAccountOutput = exports.getThroughputPoolAccount = exports.getThroughputPoolOutput = exports.getThroughputPool = exports.getTableResourceTableRoleDefinitionOutput = exports.getTableResourceTableRoleDefinition = exports.getTableResourceTableRoleAssignmentOutput = exports.getTableResourceTableRoleAssignment = exports.getTableResourceTableOutput = exports.getTableResourceTable = exports.getSqlResourceSqlUserDefinedFunctionOutput = exports.getSqlResourceSqlUserDefinedFunction = exports.getSqlResourceSqlTriggerOutput = exports.getSqlResourceSqlTrigger = exports.getSqlResourceSqlStoredProcedureOutput = exports.getSqlResourceSqlStoredProcedure = exports.getSqlResourceSqlRoleDefinitionOutput = exports.getSqlResourceSqlRoleDefinition = exports.getSqlResourceSqlRoleAssignmentOutput = void 0;
exports.v20241201preview = exports.v20241115 = exports.v20241001preview = exports.v20240901preview = exports.v20240815 = exports.v20240701 = exports.v20240601preview = exports.v20240515preview = exports.v20240515 = exports.v20240301preview = exports.v20240215preview = exports.v20231115preview = exports.v20231115 = exports.v20230915preview = exports.v20230915 = exports.v20230415 = exports.v20230315preview = exports.v20230301preview = exports.v20210701preview = exports.v20210401preview = exports.ThroughputPoolAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CassandraCluster = null;
utilities.lazyLoad(exports, ["CassandraCluster"], () => require("./cassandraCluster"));
exports.CassandraDataCenter = null;
utilities.lazyLoad(exports, ["CassandraDataCenter"], () => require("./cassandraDataCenter"));
exports.CassandraResourceCassandraKeyspace = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraKeyspace"], () => require("./cassandraResourceCassandraKeyspace"));
exports.CassandraResourceCassandraTable = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraTable"], () => require("./cassandraResourceCassandraTable"));
exports.CassandraResourceCassandraView = null;
utilities.lazyLoad(exports, ["CassandraResourceCassandraView"], () => require("./cassandraResourceCassandraView"));
exports.DatabaseAccount = null;
utilities.lazyLoad(exports, ["DatabaseAccount"], () => require("./databaseAccount"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getCassandraCluster = null;
exports.getCassandraClusterOutput = null;
utilities.lazyLoad(exports, ["getCassandraCluster", "getCassandraClusterOutput"], () => require("./getCassandraCluster"));
exports.getCassandraDataCenter = null;
exports.getCassandraDataCenterOutput = null;
utilities.lazyLoad(exports, ["getCassandraDataCenter", "getCassandraDataCenterOutput"], () => require("./getCassandraDataCenter"));
exports.getCassandraResourceCassandraKeyspace = null;
exports.getCassandraResourceCassandraKeyspaceOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraKeyspace", "getCassandraResourceCassandraKeyspaceOutput"], () => require("./getCassandraResourceCassandraKeyspace"));
exports.getCassandraResourceCassandraTable = null;
exports.getCassandraResourceCassandraTableOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraTable", "getCassandraResourceCassandraTableOutput"], () => require("./getCassandraResourceCassandraTable"));
exports.getCassandraResourceCassandraView = null;
exports.getCassandraResourceCassandraViewOutput = null;
utilities.lazyLoad(exports, ["getCassandraResourceCassandraView", "getCassandraResourceCassandraViewOutput"], () => require("./getCassandraResourceCassandraView"));
exports.getDatabaseAccount = null;
exports.getDatabaseAccountOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAccount", "getDatabaseAccountOutput"], () => require("./getDatabaseAccount"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getGraphResourceGraph = null;
exports.getGraphResourceGraphOutput = null;
utilities.lazyLoad(exports, ["getGraphResourceGraph", "getGraphResourceGraphOutput"], () => require("./getGraphResourceGraph"));
exports.getGremlinResourceGremlinDatabase = null;
exports.getGremlinResourceGremlinDatabaseOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinDatabase", "getGremlinResourceGremlinDatabaseOutput"], () => require("./getGremlinResourceGremlinDatabase"));
exports.getGremlinResourceGremlinGraph = null;
exports.getGremlinResourceGremlinGraphOutput = null;
utilities.lazyLoad(exports, ["getGremlinResourceGremlinGraph", "getGremlinResourceGremlinGraphOutput"], () => require("./getGremlinResourceGremlinGraph"));
exports.getMongoCluster = null;
exports.getMongoClusterOutput = null;
utilities.lazyLoad(exports, ["getMongoCluster", "getMongoClusterOutput"], () => require("./getMongoCluster"));
exports.getMongoClusterFirewallRule = null;
exports.getMongoClusterFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getMongoClusterFirewallRule", "getMongoClusterFirewallRuleOutput"], () => require("./getMongoClusterFirewallRule"));
exports.getMongoDBResourceMongoDBCollection = null;
exports.getMongoDBResourceMongoDBCollectionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBCollection", "getMongoDBResourceMongoDBCollectionOutput"], () => require("./getMongoDBResourceMongoDBCollection"));
exports.getMongoDBResourceMongoDBDatabase = null;
exports.getMongoDBResourceMongoDBDatabaseOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoDBDatabase", "getMongoDBResourceMongoDBDatabaseOutput"], () => require("./getMongoDBResourceMongoDBDatabase"));
exports.getMongoDBResourceMongoRoleDefinition = null;
exports.getMongoDBResourceMongoRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoRoleDefinition", "getMongoDBResourceMongoRoleDefinitionOutput"], () => require("./getMongoDBResourceMongoRoleDefinition"));
exports.getMongoDBResourceMongoUserDefinition = null;
exports.getMongoDBResourceMongoUserDefinitionOutput = null;
utilities.lazyLoad(exports, ["getMongoDBResourceMongoUserDefinition", "getMongoDBResourceMongoUserDefinitionOutput"], () => require("./getMongoDBResourceMongoUserDefinition"));
exports.getNotebookWorkspace = null;
exports.getNotebookWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getNotebookWorkspace", "getNotebookWorkspaceOutput"], () => require("./getNotebookWorkspace"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSqlResourceSqlContainer = null;
exports.getSqlResourceSqlContainerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlContainer", "getSqlResourceSqlContainerOutput"], () => require("./getSqlResourceSqlContainer"));
exports.getSqlResourceSqlDatabase = null;
exports.getSqlResourceSqlDatabaseOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlDatabase", "getSqlResourceSqlDatabaseOutput"], () => require("./getSqlResourceSqlDatabase"));
exports.getSqlResourceSqlRoleAssignment = null;
exports.getSqlResourceSqlRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleAssignment", "getSqlResourceSqlRoleAssignmentOutput"], () => require("./getSqlResourceSqlRoleAssignment"));
exports.getSqlResourceSqlRoleDefinition = null;
exports.getSqlResourceSqlRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlRoleDefinition", "getSqlResourceSqlRoleDefinitionOutput"], () => require("./getSqlResourceSqlRoleDefinition"));
exports.getSqlResourceSqlStoredProcedure = null;
exports.getSqlResourceSqlStoredProcedureOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlStoredProcedure", "getSqlResourceSqlStoredProcedureOutput"], () => require("./getSqlResourceSqlStoredProcedure"));
exports.getSqlResourceSqlTrigger = null;
exports.getSqlResourceSqlTriggerOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlTrigger", "getSqlResourceSqlTriggerOutput"], () => require("./getSqlResourceSqlTrigger"));
exports.getSqlResourceSqlUserDefinedFunction = null;
exports.getSqlResourceSqlUserDefinedFunctionOutput = null;
utilities.lazyLoad(exports, ["getSqlResourceSqlUserDefinedFunction", "getSqlResourceSqlUserDefinedFunctionOutput"], () => require("./getSqlResourceSqlUserDefinedFunction"));
exports.getTableResourceTable = null;
exports.getTableResourceTableOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTable", "getTableResourceTableOutput"], () => require("./getTableResourceTable"));
exports.getTableResourceTableRoleAssignment = null;
exports.getTableResourceTableRoleAssignmentOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTableRoleAssignment", "getTableResourceTableRoleAssignmentOutput"], () => require("./getTableResourceTableRoleAssignment"));
exports.getTableResourceTableRoleDefinition = null;
exports.getTableResourceTableRoleDefinitionOutput = null;
utilities.lazyLoad(exports, ["getTableResourceTableRoleDefinition", "getTableResourceTableRoleDefinitionOutput"], () => require("./getTableResourceTableRoleDefinition"));
exports.getThroughputPool = null;
exports.getThroughputPoolOutput = null;
utilities.lazyLoad(exports, ["getThroughputPool", "getThroughputPoolOutput"], () => require("./getThroughputPool"));
exports.getThroughputPoolAccount = null;
exports.getThroughputPoolAccountOutput = null;
utilities.lazyLoad(exports, ["getThroughputPoolAccount", "getThroughputPoolAccountOutput"], () => require("./getThroughputPoolAccount"));
exports.GraphResourceGraph = null;
utilities.lazyLoad(exports, ["GraphResourceGraph"], () => require("./graphResourceGraph"));
exports.GremlinResourceGremlinDatabase = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinDatabase"], () => require("./gremlinResourceGremlinDatabase"));
exports.GremlinResourceGremlinGraph = null;
utilities.lazyLoad(exports, ["GremlinResourceGremlinGraph"], () => require("./gremlinResourceGremlinGraph"));
exports.listDatabaseAccountConnectionStrings = null;
exports.listDatabaseAccountConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountConnectionStrings", "listDatabaseAccountConnectionStringsOutput"], () => require("./listDatabaseAccountConnectionStrings"));
exports.listDatabaseAccountKeys = null;
exports.listDatabaseAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listDatabaseAccountKeys", "listDatabaseAccountKeysOutput"], () => require("./listDatabaseAccountKeys"));
exports.listMongoClusterConnectionStrings = null;
exports.listMongoClusterConnectionStringsOutput = null;
utilities.lazyLoad(exports, ["listMongoClusterConnectionStrings", "listMongoClusterConnectionStringsOutput"], () => require("./listMongoClusterConnectionStrings"));
exports.listNotebookWorkspaceConnectionInfo = null;
exports.listNotebookWorkspaceConnectionInfoOutput = null;
utilities.lazyLoad(exports, ["listNotebookWorkspaceConnectionInfo", "listNotebookWorkspaceConnectionInfoOutput"], () => require("./listNotebookWorkspaceConnectionInfo"));
exports.MongoCluster = null;
utilities.lazyLoad(exports, ["MongoCluster"], () => require("./mongoCluster"));
exports.MongoClusterFirewallRule = null;
utilities.lazyLoad(exports, ["MongoClusterFirewallRule"], () => require("./mongoClusterFirewallRule"));
exports.MongoDBResourceMongoDBCollection = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBCollection"], () => require("./mongoDBResourceMongoDBCollection"));
exports.MongoDBResourceMongoDBDatabase = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoDBDatabase"], () => require("./mongoDBResourceMongoDBDatabase"));
exports.MongoDBResourceMongoRoleDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoRoleDefinition"], () => require("./mongoDBResourceMongoRoleDefinition"));
exports.MongoDBResourceMongoUserDefinition = null;
utilities.lazyLoad(exports, ["MongoDBResourceMongoUserDefinition"], () => require("./mongoDBResourceMongoUserDefinition"));
exports.NotebookWorkspace = null;
utilities.lazyLoad(exports, ["NotebookWorkspace"], () => require("./notebookWorkspace"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.SqlResourceSqlContainer = null;
utilities.lazyLoad(exports, ["SqlResourceSqlContainer"], () => require("./sqlResourceSqlContainer"));
exports.SqlResourceSqlDatabase = null;
utilities.lazyLoad(exports, ["SqlResourceSqlDatabase"], () => require("./sqlResourceSqlDatabase"));
exports.SqlResourceSqlRoleAssignment = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleAssignment"], () => require("./sqlResourceSqlRoleAssignment"));
exports.SqlResourceSqlRoleDefinition = null;
utilities.lazyLoad(exports, ["SqlResourceSqlRoleDefinition"], () => require("./sqlResourceSqlRoleDefinition"));
exports.SqlResourceSqlStoredProcedure = null;
utilities.lazyLoad(exports, ["SqlResourceSqlStoredProcedure"], () => require("./sqlResourceSqlStoredProcedure"));
exports.SqlResourceSqlTrigger = null;
utilities.lazyLoad(exports, ["SqlResourceSqlTrigger"], () => require("./sqlResourceSqlTrigger"));
exports.SqlResourceSqlUserDefinedFunction = null;
utilities.lazyLoad(exports, ["SqlResourceSqlUserDefinedFunction"], () => require("./sqlResourceSqlUserDefinedFunction"));
exports.TableResourceTable = null;
utilities.lazyLoad(exports, ["TableResourceTable"], () => require("./tableResourceTable"));
exports.TableResourceTableRoleAssignment = null;
utilities.lazyLoad(exports, ["TableResourceTableRoleAssignment"], () => require("./tableResourceTableRoleAssignment"));
exports.TableResourceTableRoleDefinition = null;
utilities.lazyLoad(exports, ["TableResourceTableRoleDefinition"], () => require("./tableResourceTableRoleDefinition"));
exports.ThroughputPool = null;
utilities.lazyLoad(exports, ["ThroughputPool"], () => require("./throughputPool"));
exports.ThroughputPoolAccount = null;
utilities.lazyLoad(exports, ["ThroughputPoolAccount"], () => require("./throughputPoolAccount"));
// Export enums:
__exportStar(require("../types/enums/documentdb"), exports);
// Export sub-modules:
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230315preview = require("./v20230315preview");
exports.v20230315preview = v20230315preview;
const v20230415 = require("./v20230415");
exports.v20230415 = v20230415;
const v20230915 = require("./v20230915");
exports.v20230915 = v20230915;
const v20230915preview = require("./v20230915preview");
exports.v20230915preview = v20230915preview;
const v20231115 = require("./v20231115");
exports.v20231115 = v20231115;
const v20231115preview = require("./v20231115preview");
exports.v20231115preview = v20231115preview;
const v20240215preview = require("./v20240215preview");
exports.v20240215preview = v20240215preview;
const v20240301preview = require("./v20240301preview");
exports.v20240301preview = v20240301preview;
const v20240515 = require("./v20240515");
exports.v20240515 = v20240515;
const v20240515preview = require("./v20240515preview");
exports.v20240515preview = v20240515preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
const v20240815 = require("./v20240815");
exports.v20240815 = v20240815;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const v20241115 = require("./v20241115");
exports.v20241115 = v20241115;
const v20241201preview = require("./v20241201preview");
exports.v20241201preview = v20241201preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:documentdb:CassandraCluster":
                return new exports.CassandraCluster(name, undefined, { urn });
            case "azure-native:documentdb:CassandraDataCenter":
                return new exports.CassandraDataCenter(name, undefined, { urn });
            case "azure-native:documentdb:CassandraResourceCassandraKeyspace":
                return new exports.CassandraResourceCassandraKeyspace(name, undefined, { urn });
            case "azure-native:documentdb:CassandraResourceCassandraTable":
                return new exports.CassandraResourceCassandraTable(name, undefined, { urn });
            case "azure-native:documentdb:CassandraResourceCassandraView":
                return new exports.CassandraResourceCassandraView(name, undefined, { urn });
            case "azure-native:documentdb:DatabaseAccount":
                return new exports.DatabaseAccount(name, undefined, { urn });
            case "azure-native:documentdb:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:documentdb:GraphResourceGraph":
                return new exports.GraphResourceGraph(name, undefined, { urn });
            case "azure-native:documentdb:GremlinResourceGremlinDatabase":
                return new exports.GremlinResourceGremlinDatabase(name, undefined, { urn });
            case "azure-native:documentdb:GremlinResourceGremlinGraph":
                return new exports.GremlinResourceGremlinGraph(name, undefined, { urn });
            case "azure-native:documentdb:MongoCluster":
                return new exports.MongoCluster(name, undefined, { urn });
            case "azure-native:documentdb:MongoClusterFirewallRule":
                return new exports.MongoClusterFirewallRule(name, undefined, { urn });
            case "azure-native:documentdb:MongoDBResourceMongoDBCollection":
                return new exports.MongoDBResourceMongoDBCollection(name, undefined, { urn });
            case "azure-native:documentdb:MongoDBResourceMongoDBDatabase":
                return new exports.MongoDBResourceMongoDBDatabase(name, undefined, { urn });
            case "azure-native:documentdb:MongoDBResourceMongoRoleDefinition":
                return new exports.MongoDBResourceMongoRoleDefinition(name, undefined, { urn });
            case "azure-native:documentdb:MongoDBResourceMongoUserDefinition":
                return new exports.MongoDBResourceMongoUserDefinition(name, undefined, { urn });
            case "azure-native:documentdb:NotebookWorkspace":
                return new exports.NotebookWorkspace(name, undefined, { urn });
            case "azure-native:documentdb:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:documentdb:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlContainer":
                return new exports.SqlResourceSqlContainer(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlDatabase":
                return new exports.SqlResourceSqlDatabase(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlRoleAssignment":
                return new exports.SqlResourceSqlRoleAssignment(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlRoleDefinition":
                return new exports.SqlResourceSqlRoleDefinition(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlStoredProcedure":
                return new exports.SqlResourceSqlStoredProcedure(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlTrigger":
                return new exports.SqlResourceSqlTrigger(name, undefined, { urn });
            case "azure-native:documentdb:SqlResourceSqlUserDefinedFunction":
                return new exports.SqlResourceSqlUserDefinedFunction(name, undefined, { urn });
            case "azure-native:documentdb:TableResourceTable":
                return new exports.TableResourceTable(name, undefined, { urn });
            case "azure-native:documentdb:TableResourceTableRoleAssignment":
                return new exports.TableResourceTableRoleAssignment(name, undefined, { urn });
            case "azure-native:documentdb:TableResourceTableRoleDefinition":
                return new exports.TableResourceTableRoleDefinition(name, undefined, { urn });
            case "azure-native:documentdb:ThroughputPool":
                return new exports.ThroughputPool(name, undefined, { urn });
            case "azure-native:documentdb:ThroughputPoolAccount":
                return new exports.ThroughputPoolAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "documentdb", _module);
