// @pulumi/azure-native/documentdb/mongoDBResourceMongoUserDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDBResourceMongoUserDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB User Definition
 *
 * Uses Azure REST API version 2023-04-15. In version 1.x of the Azure Native provider, it used API version 2021-10-15-preview.
 *
 * Other available API versions: 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class MongoDBResourceMongoUserDefinition extends pulumi.CustomResource {
    /**
     * Get an existing MongoDBResourceMongoUserDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MongoDBResourceMongoUserDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MongoDBResourceMongoUserDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MongoDBResourceMongoUserDefinition.__pulumiType;
    }
    /**
     * Create a MongoDBResourceMongoUserDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["customData"] = args ? args.customData : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["mechanisms"] = args ? args.mechanisms : undefined;
            resourceInputs["mongoUserDefinitionId"] = args ? args.mongoUserDefinitionId : undefined;
            resourceInputs["password"] = args ? args.password : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["roles"] = args ? args.roles : undefined;
            resourceInputs["userName"] = args ? args.userName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["customData"] = undefined /*out*/;
            resourceInputs["databaseName"] = undefined /*out*/;
            resourceInputs["mechanisms"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["password"] = undefined /*out*/;
            resourceInputs["roles"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20211015preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20211115preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20220215preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20220515preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20220815:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20220815preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20221115:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20221115preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230301preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230315:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230315preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230415:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230915:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20230915preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20231115:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20231115preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20240215preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20240515:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20240515preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20240815:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20240901preview:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20241115:MongoDBResourceMongoUserDefinition" }, { type: "azure-native:documentdb/v20241201preview:MongoDBResourceMongoUserDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MongoDBResourceMongoUserDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MongoDBResourceMongoUserDefinition = MongoDBResourceMongoUserDefinition;
/** @internal */
MongoDBResourceMongoUserDefinition.__pulumiType = 'azure-native:documentdb:MongoDBResourceMongoUserDefinition';
