// @pulumi/azure-native/documentdb/sqlResourceSqlRoleAssignment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlResourceSqlRoleAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB Role Assignment
 *
 * Uses Azure REST API version 2023-04-15. In version 1.x of the Azure Native provider, it used API version 2021-03-01-preview.
 *
 * Other available API versions: 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-11-15, 2024-12-01-preview.
 */
class SqlResourceSqlRoleAssignment extends pulumi.CustomResource {
    /**
     * Get an existing SqlResourceSqlRoleAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlResourceSqlRoleAssignment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlResourceSqlRoleAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlResourceSqlRoleAssignment.__pulumiType;
    }
    /**
     * Create a SqlResourceSqlRoleAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["principalId"] = args ? args.principalId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["roleAssignmentId"] = args ? args.roleAssignmentId : undefined;
            resourceInputs["roleDefinitionId"] = args ? args.roleDefinitionId : undefined;
            resourceInputs["scope"] = args ? args.scope : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["roleDefinitionId"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb/v20200601preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210301preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210401preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210415:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210515:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210615:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210701preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20211015:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20211015preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20211115preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20220215preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20220515:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20220515preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20220815:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20220815preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20221115:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20221115preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230301preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230315:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230315preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230415:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230915:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20230915preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20231115:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20231115preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20240215preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20240515:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20240515preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20240815:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20240901preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20241115:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20241201preview:SqlResourceSqlRoleAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlResourceSqlRoleAssignment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlResourceSqlRoleAssignment = SqlResourceSqlRoleAssignment;
/** @internal */
SqlResourceSqlRoleAssignment.__pulumiType = 'azure-native:documentdb:SqlResourceSqlRoleAssignment';
