// @pulumi/azure-native/extendedlocation/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210831preview = exports.v20210815 = exports.ResourceSyncRule = exports.getResourceSyncRuleOutput = exports.getResourceSyncRule = exports.getCustomLocationOutput = exports.getCustomLocation = exports.CustomLocation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CustomLocation = null;
utilities.lazyLoad(exports, ["CustomLocation"], () => require("./customLocation"));
exports.getCustomLocation = null;
exports.getCustomLocationOutput = null;
utilities.lazyLoad(exports, ["getCustomLocation", "getCustomLocationOutput"], () => require("./getCustomLocation"));
exports.getResourceSyncRule = null;
exports.getResourceSyncRuleOutput = null;
utilities.lazyLoad(exports, ["getResourceSyncRule", "getResourceSyncRuleOutput"], () => require("./getResourceSyncRule"));
exports.ResourceSyncRule = null;
utilities.lazyLoad(exports, ["ResourceSyncRule"], () => require("./resourceSyncRule"));
// Export enums:
__exportStar(require("../types/enums/extendedlocation"), exports);
// Export sub-modules:
const v20210815 = require("./v20210815");
exports.v20210815 = v20210815;
const v20210831preview = require("./v20210831preview");
exports.v20210831preview = v20210831preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:extendedlocation:CustomLocation":
                return new exports.CustomLocation(name, undefined, { urn });
            case "azure-native:extendedlocation:ResourceSyncRule":
                return new exports.ResourceSyncRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "extendedlocation", _module);
