// @pulumi/azure-native/botservice/v20230915preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = exports.listQnAMakerEndpointKeyOutput = exports.listQnAMakerEndpointKey = exports.listChannelWithKeysOutput = exports.listChannelWithKeys = exports.listBotConnectionWithSecretsOutput = exports.listBotConnectionWithSecrets = exports.listBotConnectionServiceProvidersOutput = exports.listBotConnectionServiceProviders = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getChannelOutput = exports.getChannel = exports.getBotConnectionOutput = exports.getBotConnection = exports.getBotOutput = exports.getBot = exports.Channel = exports.BotConnection = exports.Bot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Bot = null;
utilities.lazyLoad(exports, ["Bot"], () => require("./bot"));
exports.BotConnection = null;
utilities.lazyLoad(exports, ["BotConnection"], () => require("./botConnection"));
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.getBot = null;
exports.getBotOutput = null;
utilities.lazyLoad(exports, ["getBot", "getBotOutput"], () => require("./getBot"));
exports.getBotConnection = null;
exports.getBotConnectionOutput = null;
utilities.lazyLoad(exports, ["getBotConnection", "getBotConnectionOutput"], () => require("./getBotConnection"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.listBotConnectionServiceProviders = null;
exports.listBotConnectionServiceProvidersOutput = null;
utilities.lazyLoad(exports, ["listBotConnectionServiceProviders", "listBotConnectionServiceProvidersOutput"], () => require("./listBotConnectionServiceProviders"));
exports.listBotConnectionWithSecrets = null;
exports.listBotConnectionWithSecretsOutput = null;
utilities.lazyLoad(exports, ["listBotConnectionWithSecrets", "listBotConnectionWithSecretsOutput"], () => require("./listBotConnectionWithSecrets"));
exports.listChannelWithKeys = null;
exports.listChannelWithKeysOutput = null;
utilities.lazyLoad(exports, ["listChannelWithKeys", "listChannelWithKeysOutput"], () => require("./listChannelWithKeys"));
exports.listQnAMakerEndpointKey = null;
exports.listQnAMakerEndpointKeyOutput = null;
utilities.lazyLoad(exports, ["listQnAMakerEndpointKey", "listQnAMakerEndpointKeyOutput"], () => require("./listQnAMakerEndpointKey"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../../types/enums/botservice/v20230915preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:botservice/v20230915preview:Bot":
                return new exports.Bot(name, undefined, { urn });
            case "azure-native:botservice/v20230915preview:BotConnection":
                return new exports.BotConnection(name, undefined, { urn });
            case "azure-native:botservice/v20230915preview:Channel":
                return new exports.Channel(name, undefined, { urn });
            case "azure-native:botservice/v20230915preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "botservice/v20230915preview", _module);
