// @pulumi/azure-native/azurestackhci/v20250201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineInstance = exports.VirtualHardDisk = exports.StorageContainer = exports.SecurityRule = exports.NetworkSecurityGroup = exports.NetworkInterface = exports.MarketplaceGalleryImage = exports.LogicalNetwork = exports.GuestAgent = exports.getVirtualMachineInstanceOutput = exports.getVirtualMachineInstance = exports.getVirtualHardDiskOutput = exports.getVirtualHardDisk = exports.getStorageContainerOutput = exports.getStorageContainer = exports.getSecurityRuleOutput = exports.getSecurityRule = exports.getNetworkSecurityGroupOutput = exports.getNetworkSecurityGroup = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getMarketplaceGalleryImageOutput = exports.getMarketplaceGalleryImage = exports.getLogicalNetworkOutput = exports.getLogicalNetwork = exports.getGuestAgentOutput = exports.getGuestAgent = exports.getGalleryImageOutput = exports.getGalleryImage = exports.GalleryImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGuestAgent = null;
exports.getGuestAgentOutput = null;
utilities.lazyLoad(exports, ["getGuestAgent", "getGuestAgentOutput"], () => require("./getGuestAgent"));
exports.getLogicalNetwork = null;
exports.getLogicalNetworkOutput = null;
utilities.lazyLoad(exports, ["getLogicalNetwork", "getLogicalNetworkOutput"], () => require("./getLogicalNetwork"));
exports.getMarketplaceGalleryImage = null;
exports.getMarketplaceGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getMarketplaceGalleryImage", "getMarketplaceGalleryImageOutput"], () => require("./getMarketplaceGalleryImage"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getNetworkSecurityGroup = null;
exports.getNetworkSecurityGroupOutput = null;
utilities.lazyLoad(exports, ["getNetworkSecurityGroup", "getNetworkSecurityGroupOutput"], () => require("./getNetworkSecurityGroup"));
exports.getSecurityRule = null;
exports.getSecurityRuleOutput = null;
utilities.lazyLoad(exports, ["getSecurityRule", "getSecurityRuleOutput"], () => require("./getSecurityRule"));
exports.getStorageContainer = null;
exports.getStorageContainerOutput = null;
utilities.lazyLoad(exports, ["getStorageContainer", "getStorageContainerOutput"], () => require("./getStorageContainer"));
exports.getVirtualHardDisk = null;
exports.getVirtualHardDiskOutput = null;
utilities.lazyLoad(exports, ["getVirtualHardDisk", "getVirtualHardDiskOutput"], () => require("./getVirtualHardDisk"));
exports.getVirtualMachineInstance = null;
exports.getVirtualMachineInstanceOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineInstance", "getVirtualMachineInstanceOutput"], () => require("./getVirtualMachineInstance"));
exports.GuestAgent = null;
utilities.lazyLoad(exports, ["GuestAgent"], () => require("./guestAgent"));
exports.LogicalNetwork = null;
utilities.lazyLoad(exports, ["LogicalNetwork"], () => require("./logicalNetwork"));
exports.MarketplaceGalleryImage = null;
utilities.lazyLoad(exports, ["MarketplaceGalleryImage"], () => require("./marketplaceGalleryImage"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.NetworkSecurityGroup = null;
utilities.lazyLoad(exports, ["NetworkSecurityGroup"], () => require("./networkSecurityGroup"));
exports.SecurityRule = null;
utilities.lazyLoad(exports, ["SecurityRule"], () => require("./securityRule"));
exports.StorageContainer = null;
utilities.lazyLoad(exports, ["StorageContainer"], () => require("./storageContainer"));
exports.VirtualHardDisk = null;
utilities.lazyLoad(exports, ["VirtualHardDisk"], () => require("./virtualHardDisk"));
exports.VirtualMachineInstance = null;
utilities.lazyLoad(exports, ["VirtualMachineInstance"], () => require("./virtualMachineInstance"));
// Export enums:
__exportStar(require("../../types/enums/azurestackhci/v20250201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestackhci/v20250201preview:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:GuestAgent":
                return new exports.GuestAgent(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:LogicalNetwork":
                return new exports.LogicalNetwork(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:MarketplaceGalleryImage":
                return new exports.MarketplaceGalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:NetworkSecurityGroup":
                return new exports.NetworkSecurityGroup(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:SecurityRule":
                return new exports.SecurityRule(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:StorageContainer":
                return new exports.StorageContainer(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:VirtualHardDisk":
                return new exports.VirtualHardDisk(name, undefined, { urn });
            case "azure-native:azurestackhci/v20250201preview:VirtualMachineInstance":
                return new exports.VirtualMachineInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestackhci/v20250201preview", _module);
