// @pulumi/azure-native/azurelargeinstance/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240801preview = exports.getAzureLargeStorageInstanceOutput = exports.getAzureLargeStorageInstance = exports.getAzureLargeInstanceOutput = exports.getAzureLargeInstance = exports.AzureLargeStorageInstance = exports.AzureLargeInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureLargeInstance = null;
utilities.lazyLoad(exports, ["AzureLargeInstance"], () => require("./azureLargeInstance"));
exports.AzureLargeStorageInstance = null;
utilities.lazyLoad(exports, ["AzureLargeStorageInstance"], () => require("./azureLargeStorageInstance"));
exports.getAzureLargeInstance = null;
exports.getAzureLargeInstanceOutput = null;
utilities.lazyLoad(exports, ["getAzureLargeInstance", "getAzureLargeInstanceOutput"], () => require("./getAzureLargeInstance"));
exports.getAzureLargeStorageInstance = null;
exports.getAzureLargeStorageInstanceOutput = null;
utilities.lazyLoad(exports, ["getAzureLargeStorageInstance", "getAzureLargeStorageInstanceOutput"], () => require("./getAzureLargeStorageInstance"));
// Export enums:
__exportStar(require("../types/enums/azurelargeinstance"), exports);
// Export sub-modules:
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurelargeinstance:AzureLargeInstance":
                return new exports.AzureLargeInstance(name, undefined, { urn });
            case "azure-native:azurelargeinstance:AzureLargeStorageInstance":
                return new exports.AzureLargeStorageInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurelargeinstance", _module);
