// @pulumi/gcp/bigquery/biReservation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BiReservation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a BI Reservation.
 *
 * To get more information about BiReservation, see:
 *
 * * [API documentation](https://cloud.google.com/bigquery/docs/reference/reservations/rest/v1/BiReservation)
 * * How-to Guides
 *     * [Introduction to Reservations](https://cloud.google.com/bigquery/docs/reservations-intro)
 *
 * ## Example Usage
 *
 * ### Bigquery Reservation Bi Reservation Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const reservation = new gcp.bigquery.BiReservation("reservation", {
 *     location: "us-west2",
 *     size: 3000000000,
 * });
 * ```
 *
 * ## Import
 *
 * BiReservation can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/biReservation`
 *
 * * `{{project}}/{{location}}`
 *
 * * `{{location}}`
 *
 * When using the `pulumi import` command, BiReservation can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:bigquery/biReservation:BiReservation default projects/{{project}}/locations/{{location}}/biReservation
 * ```
 *
 * ```sh
 * $ pulumi import gcp:bigquery/biReservation:BiReservation default {{project}}/{{location}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:bigquery/biReservation:BiReservation default {{location}}
 * ```
 */
class BiReservation extends pulumi.CustomResource {
    /**
     * Get an existing BiReservation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BiReservation(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BiReservation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BiReservation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["preferredTables"] = state ? state.preferredTables : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["size"] = state ? state.size : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["preferredTables"] = args ? args.preferredTables : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["size"] = args ? args.size : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BiReservation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BiReservation = BiReservation;
/** @internal */
BiReservation.__pulumiType = 'gcp:bigquery/biReservation:BiReservation';
