// @pulumi/azure-native/insights/v20191017preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.getPrivateLinkScopeOutput = exports.getPrivateLinkScope = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkScope = null;
exports.getPrivateLinkScopeOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScope", "getPrivateLinkScopeOutput"], () => require("./getPrivateLinkScope"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkScope = null;
utilities.lazyLoad(exports, ["PrivateLinkScope"], () => require("./privateLinkScope"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights/v20191017preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:insights/v20191017preview:PrivateLinkScope":
                return new exports.PrivateLinkScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights/v20191017preview", _module);
