// @pulumi/azure-native/apimanagement/v20230501preview/apiWiki.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiWiki = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Wiki properties
 */
class ApiWiki extends pulumi.CustomResource {
    /**
     * Get an existing ApiWiki resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ApiWiki(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApiWiki.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApiWiki.__pulumiType;
    }
    /**
     * Create a ApiWiki resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["apiId"] = args ? args.apiId : undefined;
            resourceInputs["documents"] = args ? args.documents : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["documents"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20220801:ApiWiki" }, { type: "azure-native:apimanagement/v20220901preview:ApiWiki" }, { type: "azure-native:apimanagement/v20230301preview:ApiWiki" }, { type: "azure-native:apimanagement/v20230901preview:ApiWiki" }, { type: "azure-native:apimanagement/v20240501:ApiWiki" }, { type: "azure-native:apimanagement/v20240601preview:ApiWiki" }, { type: "azure-native:apimanagement:ApiWiki" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ApiWiki.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ApiWiki = ApiWiki;
/** @internal */
ApiWiki.__pulumiType = 'azure-native:apimanagement/v20230501preview:ApiWiki';
