// @pulumi/azure-native/types/enums/hybridnetwork/v20240415/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkFunctionNFVIType = exports.Type = exports.TemplateType = exports.SkuName = exports.PublisherScope = exports.NetworkFunctionType = exports.NetworkFunctionConfigurationType = exports.NFVIType = exports.ManagedServiceIdentityType = exports.IdType = exports.ContainerizedNetworkFunctionNFVIType = exports.ConfigurationGroupValueConfigurationType = exports.BackingResourcePublicNetworkAccess = exports.AzureOperatorNexusArtifactType = exports.AzureCoreArtifactType = exports.AzureArcKubernetesArtifactType = exports.ArtifactType = exports.ArtifactStoreType = exports.ArtifactReplicationStrategy = exports.ApplicationEnablement = void 0;
exports.ApplicationEnablement = {
    Unknown: "Unknown",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ArtifactReplicationStrategy = {
    Unknown: "Unknown",
    SingleReplication: "SingleReplication",
};
exports.ArtifactStoreType = {
    Unknown: "Unknown",
    AzureContainerRegistry: "AzureContainerRegistry",
    AzureStorageAccount: "AzureStorageAccount",
};
exports.ArtifactType = {
    Unknown: "Unknown",
    OCIArtifact: "OCIArtifact",
    VhdImageFile: "VhdImageFile",
    ArmTemplate: "ArmTemplate",
    ImageFile: "ImageFile",
};
exports.AzureArcKubernetesArtifactType = {
    Unknown: "Unknown",
    HelmPackage: "HelmPackage",
};
exports.AzureCoreArtifactType = {
    Unknown: "Unknown",
    VhdImageFile: "VhdImageFile",
    ArmTemplate: "ArmTemplate",
};
exports.AzureOperatorNexusArtifactType = {
    Unknown: "Unknown",
    ImageFile: "ImageFile",
    ArmTemplate: "ArmTemplate",
};
exports.BackingResourcePublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ConfigurationGroupValueConfigurationType = {
    Unknown: "Unknown",
    Secret: "Secret",
    Open: "Open",
};
exports.ContainerizedNetworkFunctionNFVIType = {
    Unknown: "Unknown",
    AzureArcKubernetes: "AzureArcKubernetes",
};
exports.IdType = {
    Unknown: "Unknown",
    Open: "Open",
    Secret: "Secret",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NFVIType = {
    Unknown: "Unknown",
    AzureArcKubernetes: "AzureArcKubernetes",
    AzureCore: "AzureCore",
    AzureOperatorNexus: "AzureOperatorNexus",
};
exports.NetworkFunctionConfigurationType = {
    Unknown: "Unknown",
    Secret: "Secret",
    Open: "Open",
};
exports.NetworkFunctionType = {
    Unknown: "Unknown",
    VirtualNetworkFunction: "VirtualNetworkFunction",
    ContainerizedNetworkFunction: "ContainerizedNetworkFunction",
};
exports.PublisherScope = {
    Unknown: "Unknown",
    Private: "Private",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
};
exports.TemplateType = {
    Unknown: "Unknown",
    ArmTemplate: "ArmTemplate",
};
exports.Type = {
    Unknown: "Unknown",
    ArmResourceDefinition: "ArmResourceDefinition",
    NetworkFunctionDefinition: "NetworkFunctionDefinition",
};
exports.VirtualNetworkFunctionNFVIType = {
    Unknown: "Unknown",
    AzureCore: "AzureCore",
    AzureOperatorNexus: "AzureOperatorNexus",
};
