// @pulumi/azure-native/consumption/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240801 = exports.v20231101 = exports.v20230501 = exports.getBudgetOutput = exports.getBudget = exports.Budget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Budget = null;
utilities.lazyLoad(exports, ["Budget"], () => require("./budget"));
exports.getBudget = null;
exports.getBudgetOutput = null;
utilities.lazyLoad(exports, ["getBudget", "getBudgetOutput"], () => require("./getBudget"));
// Export enums:
__exportStar(require("../types/enums/consumption"), exports);
// Export sub-modules:
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240801 = require("./v20240801");
exports.v20240801 = v20240801;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:consumption:Budget":
                return new exports.Budget(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "consumption", _module);
