// @pulumi/azure-native/machinelearningservices/v20240401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDataContainerOutput = exports.getDataContainer = exports.getConnectionRaiPolicyOutput = exports.getConnectionRaiPolicy = exports.getConnectionRaiBlocklistItemOutput = exports.getConnectionRaiBlocklistItem = exports.getConnectionRaiBlocklistOutput = exports.getConnectionRaiBlocklist = exports.getConnectionDeploymentOutput = exports.getConnectionDeployment = exports.getComputeAllowedResizeSizesOutput = exports.getComputeAllowedResizeSizes = exports.getComputeOutput = exports.getCompute = exports.getComponentVersionOutput = exports.getComponentVersion = exports.getComponentContainerOutput = exports.getComponentContainer = exports.getCodeVersionOutput = exports.getCodeVersion = exports.getCodeContainerOutput = exports.getCodeContainer = exports.getCapacityReservationGroupOutput = exports.getCapacityReservationGroup = exports.getBatchEndpointOutput = exports.getBatchEndpoint = exports.getBatchDeploymentOutput = exports.getBatchDeployment = exports.FeaturestoreEntityVersion = exports.FeaturestoreEntityContainerEntity = exports.FeaturesetVersion = exports.FeaturesetContainerEntity = exports.EnvironmentVersion = exports.EnvironmentContainer = exports.EndpointDeployment = exports.Datastore = exports.DataVersion = exports.DataContainer = exports.ConnectionRaiPolicy = exports.ConnectionRaiBlocklistItem = exports.ConnectionRaiBlocklist = exports.ConnectionDeployment = exports.Compute = exports.ComponentVersion = exports.ComponentContainer = exports.CodeVersion = exports.CodeContainer = exports.CapacityReservationGroup = exports.BatchEndpoint = exports.BatchDeployment = void 0;
exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getOnlineEndpointTokenOutput = exports.getOnlineEndpointToken = exports.getOnlineEndpointOutput = exports.getOnlineEndpoint = exports.getOnlineDeploymentLogsOutput = exports.getOnlineDeploymentLogs = exports.getOnlineDeploymentOutput = exports.getOnlineDeployment = exports.getModelVersionOutput = exports.getModelVersion = exports.getModelContainerOutput = exports.getModelContainer = exports.getMarketplaceSubscriptionOutput = exports.getMarketplaceSubscription = exports.getManagedNetworkSettingsRuleOutput = exports.getManagedNetworkSettingsRule = exports.getLabelingJobOutput = exports.getLabelingJob = exports.getJobOutput = exports.getJob = exports.getInferencePoolStatusOutput = exports.getInferencePoolStatus = exports.getInferencePoolOutput = exports.getInferencePool = exports.getInferenceGroupStatusOutput = exports.getInferenceGroupStatus = exports.getInferenceGroupOutput = exports.getInferenceGroup = exports.getInferenceEndpointOutput = exports.getInferenceEndpoint = exports.getFeaturestoreEntityVersionOutput = exports.getFeaturestoreEntityVersion = exports.getFeaturestoreEntityContainerEntityOutput = exports.getFeaturestoreEntityContainerEntity = exports.getFeaturesetVersionOutput = exports.getFeaturesetVersion = exports.getFeaturesetContainerEntityOutput = exports.getFeaturesetContainerEntity = exports.getEnvironmentVersionOutput = exports.getEnvironmentVersion = exports.getEnvironmentContainerOutput = exports.getEnvironmentContainer = exports.getEndpointDeploymentOutput = exports.getEndpointDeployment = exports.getDatastoreOutput = exports.getDatastore = exports.getDataVersionOutput = exports.getDataVersion = void 0;
exports.listOnlineEndpointKeys = exports.listEndpointKeysOutput = exports.listEndpointKeys = exports.listDatastoreSecretsOutput = exports.listDatastoreSecrets = exports.listComputeNodesOutput = exports.listComputeNodes = exports.listComputeKeysOutput = exports.listComputeKeys = exports.listBatchEndpointKeysOutput = exports.listBatchEndpointKeys = exports.LabelingJob = exports.Job = exports.InferencePool = exports.InferenceGroup = exports.InferenceEndpoint = exports.getWorkspaceConnectionOutput = exports.getWorkspaceConnection = exports.getWorkspaceOutput = exports.getWorkspace = exports.getServerlessEndpointStatusOutput = exports.getServerlessEndpointStatus = exports.getServerlessEndpointOutput = exports.getServerlessEndpoint = exports.getScheduleOutput = exports.getSchedule = exports.getRegistryModelVersionOutput = exports.getRegistryModelVersion = exports.getRegistryModelContainerOutput = exports.getRegistryModelContainer = exports.getRegistryEnvironmentVersionOutput = exports.getRegistryEnvironmentVersion = exports.getRegistryEnvironmentContainerOutput = exports.getRegistryEnvironmentContainer = exports.getRegistryDataVersionOutput = exports.getRegistryDataVersion = exports.getRegistryDataContainerOutput = exports.getRegistryDataContainer = exports.getRegistryComponentVersionOutput = exports.getRegistryComponentVersion = exports.getRegistryComponentContainerOutput = exports.getRegistryComponentContainer = exports.getRegistryCodeVersionOutput = exports.getRegistryCodeVersion = exports.getRegistryCodeContainerOutput = exports.getRegistryCodeContainer = exports.getRegistryOutput = exports.getRegistry = exports.getRaiPolicyOutput = exports.getRaiPolicy = void 0;
exports.WorkspaceConnection = exports.Workspace = exports.ServerlessEndpoint = exports.Schedule = exports.RegistryModelVersion = exports.RegistryModelContainer = exports.RegistryEnvironmentVersion = exports.RegistryEnvironmentContainer = exports.RegistryDataVersion = exports.RegistryDataContainer = exports.RegistryComponentVersion = exports.RegistryComponentContainer = exports.RegistryCodeVersion = exports.RegistryCodeContainer = exports.Registry = exports.RaiPolicy = exports.PrivateEndpointConnection = exports.OnlineEndpoint = exports.OnlineDeployment = exports.ModelVersion = exports.ModelContainer = exports.MarketplaceSubscription = exports.ManagedNetworkSettingsRule = exports.listWorkspaceStorageAccountKeysOutput = exports.listWorkspaceStorageAccountKeys = exports.listWorkspaceNotebookKeysOutput = exports.listWorkspaceNotebookKeys = exports.listWorkspaceNotebookAccessTokenOutput = exports.listWorkspaceNotebookAccessToken = exports.listWorkspaceKeysOutput = exports.listWorkspaceKeys = exports.listWorkspaceConnectionSecretsOutput = exports.listWorkspaceConnectionSecrets = exports.listWorkspaceConnectionModelsOutput = exports.listWorkspaceConnectionModels = exports.listServerlessEndpointKeysOutput = exports.listServerlessEndpointKeys = exports.listOnlineEndpointKeysOutput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BatchDeployment = null;
utilities.lazyLoad(exports, ["BatchDeployment"], () => require("./batchDeployment"));
exports.BatchEndpoint = null;
utilities.lazyLoad(exports, ["BatchEndpoint"], () => require("./batchEndpoint"));
exports.CapacityReservationGroup = null;
utilities.lazyLoad(exports, ["CapacityReservationGroup"], () => require("./capacityReservationGroup"));
exports.CodeContainer = null;
utilities.lazyLoad(exports, ["CodeContainer"], () => require("./codeContainer"));
exports.CodeVersion = null;
utilities.lazyLoad(exports, ["CodeVersion"], () => require("./codeVersion"));
exports.ComponentContainer = null;
utilities.lazyLoad(exports, ["ComponentContainer"], () => require("./componentContainer"));
exports.ComponentVersion = null;
utilities.lazyLoad(exports, ["ComponentVersion"], () => require("./componentVersion"));
exports.Compute = null;
utilities.lazyLoad(exports, ["Compute"], () => require("./compute"));
exports.ConnectionDeployment = null;
utilities.lazyLoad(exports, ["ConnectionDeployment"], () => require("./connectionDeployment"));
exports.ConnectionRaiBlocklist = null;
utilities.lazyLoad(exports, ["ConnectionRaiBlocklist"], () => require("./connectionRaiBlocklist"));
exports.ConnectionRaiBlocklistItem = null;
utilities.lazyLoad(exports, ["ConnectionRaiBlocklistItem"], () => require("./connectionRaiBlocklistItem"));
exports.ConnectionRaiPolicy = null;
utilities.lazyLoad(exports, ["ConnectionRaiPolicy"], () => require("./connectionRaiPolicy"));
exports.DataContainer = null;
utilities.lazyLoad(exports, ["DataContainer"], () => require("./dataContainer"));
exports.DataVersion = null;
utilities.lazyLoad(exports, ["DataVersion"], () => require("./dataVersion"));
exports.Datastore = null;
utilities.lazyLoad(exports, ["Datastore"], () => require("./datastore"));
exports.EndpointDeployment = null;
utilities.lazyLoad(exports, ["EndpointDeployment"], () => require("./endpointDeployment"));
exports.EnvironmentContainer = null;
utilities.lazyLoad(exports, ["EnvironmentContainer"], () => require("./environmentContainer"));
exports.EnvironmentVersion = null;
utilities.lazyLoad(exports, ["EnvironmentVersion"], () => require("./environmentVersion"));
exports.FeaturesetContainerEntity = null;
utilities.lazyLoad(exports, ["FeaturesetContainerEntity"], () => require("./featuresetContainerEntity"));
exports.FeaturesetVersion = null;
utilities.lazyLoad(exports, ["FeaturesetVersion"], () => require("./featuresetVersion"));
exports.FeaturestoreEntityContainerEntity = null;
utilities.lazyLoad(exports, ["FeaturestoreEntityContainerEntity"], () => require("./featurestoreEntityContainerEntity"));
exports.FeaturestoreEntityVersion = null;
utilities.lazyLoad(exports, ["FeaturestoreEntityVersion"], () => require("./featurestoreEntityVersion"));
exports.getBatchDeployment = null;
exports.getBatchDeploymentOutput = null;
utilities.lazyLoad(exports, ["getBatchDeployment", "getBatchDeploymentOutput"], () => require("./getBatchDeployment"));
exports.getBatchEndpoint = null;
exports.getBatchEndpointOutput = null;
utilities.lazyLoad(exports, ["getBatchEndpoint", "getBatchEndpointOutput"], () => require("./getBatchEndpoint"));
exports.getCapacityReservationGroup = null;
exports.getCapacityReservationGroupOutput = null;
utilities.lazyLoad(exports, ["getCapacityReservationGroup", "getCapacityReservationGroupOutput"], () => require("./getCapacityReservationGroup"));
exports.getCodeContainer = null;
exports.getCodeContainerOutput = null;
utilities.lazyLoad(exports, ["getCodeContainer", "getCodeContainerOutput"], () => require("./getCodeContainer"));
exports.getCodeVersion = null;
exports.getCodeVersionOutput = null;
utilities.lazyLoad(exports, ["getCodeVersion", "getCodeVersionOutput"], () => require("./getCodeVersion"));
exports.getComponentContainer = null;
exports.getComponentContainerOutput = null;
utilities.lazyLoad(exports, ["getComponentContainer", "getComponentContainerOutput"], () => require("./getComponentContainer"));
exports.getComponentVersion = null;
exports.getComponentVersionOutput = null;
utilities.lazyLoad(exports, ["getComponentVersion", "getComponentVersionOutput"], () => require("./getComponentVersion"));
exports.getCompute = null;
exports.getComputeOutput = null;
utilities.lazyLoad(exports, ["getCompute", "getComputeOutput"], () => require("./getCompute"));
exports.getComputeAllowedResizeSizes = null;
exports.getComputeAllowedResizeSizesOutput = null;
utilities.lazyLoad(exports, ["getComputeAllowedResizeSizes", "getComputeAllowedResizeSizesOutput"], () => require("./getComputeAllowedResizeSizes"));
exports.getConnectionDeployment = null;
exports.getConnectionDeploymentOutput = null;
utilities.lazyLoad(exports, ["getConnectionDeployment", "getConnectionDeploymentOutput"], () => require("./getConnectionDeployment"));
exports.getConnectionRaiBlocklist = null;
exports.getConnectionRaiBlocklistOutput = null;
utilities.lazyLoad(exports, ["getConnectionRaiBlocklist", "getConnectionRaiBlocklistOutput"], () => require("./getConnectionRaiBlocklist"));
exports.getConnectionRaiBlocklistItem = null;
exports.getConnectionRaiBlocklistItemOutput = null;
utilities.lazyLoad(exports, ["getConnectionRaiBlocklistItem", "getConnectionRaiBlocklistItemOutput"], () => require("./getConnectionRaiBlocklistItem"));
exports.getConnectionRaiPolicy = null;
exports.getConnectionRaiPolicyOutput = null;
utilities.lazyLoad(exports, ["getConnectionRaiPolicy", "getConnectionRaiPolicyOutput"], () => require("./getConnectionRaiPolicy"));
exports.getDataContainer = null;
exports.getDataContainerOutput = null;
utilities.lazyLoad(exports, ["getDataContainer", "getDataContainerOutput"], () => require("./getDataContainer"));
exports.getDataVersion = null;
exports.getDataVersionOutput = null;
utilities.lazyLoad(exports, ["getDataVersion", "getDataVersionOutput"], () => require("./getDataVersion"));
exports.getDatastore = null;
exports.getDatastoreOutput = null;
utilities.lazyLoad(exports, ["getDatastore", "getDatastoreOutput"], () => require("./getDatastore"));
exports.getEndpointDeployment = null;
exports.getEndpointDeploymentOutput = null;
utilities.lazyLoad(exports, ["getEndpointDeployment", "getEndpointDeploymentOutput"], () => require("./getEndpointDeployment"));
exports.getEnvironmentContainer = null;
exports.getEnvironmentContainerOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentContainer", "getEnvironmentContainerOutput"], () => require("./getEnvironmentContainer"));
exports.getEnvironmentVersion = null;
exports.getEnvironmentVersionOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentVersion", "getEnvironmentVersionOutput"], () => require("./getEnvironmentVersion"));
exports.getFeaturesetContainerEntity = null;
exports.getFeaturesetContainerEntityOutput = null;
utilities.lazyLoad(exports, ["getFeaturesetContainerEntity", "getFeaturesetContainerEntityOutput"], () => require("./getFeaturesetContainerEntity"));
exports.getFeaturesetVersion = null;
exports.getFeaturesetVersionOutput = null;
utilities.lazyLoad(exports, ["getFeaturesetVersion", "getFeaturesetVersionOutput"], () => require("./getFeaturesetVersion"));
exports.getFeaturestoreEntityContainerEntity = null;
exports.getFeaturestoreEntityContainerEntityOutput = null;
utilities.lazyLoad(exports, ["getFeaturestoreEntityContainerEntity", "getFeaturestoreEntityContainerEntityOutput"], () => require("./getFeaturestoreEntityContainerEntity"));
exports.getFeaturestoreEntityVersion = null;
exports.getFeaturestoreEntityVersionOutput = null;
utilities.lazyLoad(exports, ["getFeaturestoreEntityVersion", "getFeaturestoreEntityVersionOutput"], () => require("./getFeaturestoreEntityVersion"));
exports.getInferenceEndpoint = null;
exports.getInferenceEndpointOutput = null;
utilities.lazyLoad(exports, ["getInferenceEndpoint", "getInferenceEndpointOutput"], () => require("./getInferenceEndpoint"));
exports.getInferenceGroup = null;
exports.getInferenceGroupOutput = null;
utilities.lazyLoad(exports, ["getInferenceGroup", "getInferenceGroupOutput"], () => require("./getInferenceGroup"));
exports.getInferenceGroupStatus = null;
exports.getInferenceGroupStatusOutput = null;
utilities.lazyLoad(exports, ["getInferenceGroupStatus", "getInferenceGroupStatusOutput"], () => require("./getInferenceGroupStatus"));
exports.getInferencePool = null;
exports.getInferencePoolOutput = null;
utilities.lazyLoad(exports, ["getInferencePool", "getInferencePoolOutput"], () => require("./getInferencePool"));
exports.getInferencePoolStatus = null;
exports.getInferencePoolStatusOutput = null;
utilities.lazyLoad(exports, ["getInferencePoolStatus", "getInferencePoolStatusOutput"], () => require("./getInferencePoolStatus"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getLabelingJob = null;
exports.getLabelingJobOutput = null;
utilities.lazyLoad(exports, ["getLabelingJob", "getLabelingJobOutput"], () => require("./getLabelingJob"));
exports.getManagedNetworkSettingsRule = null;
exports.getManagedNetworkSettingsRuleOutput = null;
utilities.lazyLoad(exports, ["getManagedNetworkSettingsRule", "getManagedNetworkSettingsRuleOutput"], () => require("./getManagedNetworkSettingsRule"));
exports.getMarketplaceSubscription = null;
exports.getMarketplaceSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getMarketplaceSubscription", "getMarketplaceSubscriptionOutput"], () => require("./getMarketplaceSubscription"));
exports.getModelContainer = null;
exports.getModelContainerOutput = null;
utilities.lazyLoad(exports, ["getModelContainer", "getModelContainerOutput"], () => require("./getModelContainer"));
exports.getModelVersion = null;
exports.getModelVersionOutput = null;
utilities.lazyLoad(exports, ["getModelVersion", "getModelVersionOutput"], () => require("./getModelVersion"));
exports.getOnlineDeployment = null;
exports.getOnlineDeploymentOutput = null;
utilities.lazyLoad(exports, ["getOnlineDeployment", "getOnlineDeploymentOutput"], () => require("./getOnlineDeployment"));
exports.getOnlineDeploymentLogs = null;
exports.getOnlineDeploymentLogsOutput = null;
utilities.lazyLoad(exports, ["getOnlineDeploymentLogs", "getOnlineDeploymentLogsOutput"], () => require("./getOnlineDeploymentLogs"));
exports.getOnlineEndpoint = null;
exports.getOnlineEndpointOutput = null;
utilities.lazyLoad(exports, ["getOnlineEndpoint", "getOnlineEndpointOutput"], () => require("./getOnlineEndpoint"));
exports.getOnlineEndpointToken = null;
exports.getOnlineEndpointTokenOutput = null;
utilities.lazyLoad(exports, ["getOnlineEndpointToken", "getOnlineEndpointTokenOutput"], () => require("./getOnlineEndpointToken"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getRaiPolicy = null;
exports.getRaiPolicyOutput = null;
utilities.lazyLoad(exports, ["getRaiPolicy", "getRaiPolicyOutput"], () => require("./getRaiPolicy"));
exports.getRegistry = null;
exports.getRegistryOutput = null;
utilities.lazyLoad(exports, ["getRegistry", "getRegistryOutput"], () => require("./getRegistry"));
exports.getRegistryCodeContainer = null;
exports.getRegistryCodeContainerOutput = null;
utilities.lazyLoad(exports, ["getRegistryCodeContainer", "getRegistryCodeContainerOutput"], () => require("./getRegistryCodeContainer"));
exports.getRegistryCodeVersion = null;
exports.getRegistryCodeVersionOutput = null;
utilities.lazyLoad(exports, ["getRegistryCodeVersion", "getRegistryCodeVersionOutput"], () => require("./getRegistryCodeVersion"));
exports.getRegistryComponentContainer = null;
exports.getRegistryComponentContainerOutput = null;
utilities.lazyLoad(exports, ["getRegistryComponentContainer", "getRegistryComponentContainerOutput"], () => require("./getRegistryComponentContainer"));
exports.getRegistryComponentVersion = null;
exports.getRegistryComponentVersionOutput = null;
utilities.lazyLoad(exports, ["getRegistryComponentVersion", "getRegistryComponentVersionOutput"], () => require("./getRegistryComponentVersion"));
exports.getRegistryDataContainer = null;
exports.getRegistryDataContainerOutput = null;
utilities.lazyLoad(exports, ["getRegistryDataContainer", "getRegistryDataContainerOutput"], () => require("./getRegistryDataContainer"));
exports.getRegistryDataVersion = null;
exports.getRegistryDataVersionOutput = null;
utilities.lazyLoad(exports, ["getRegistryDataVersion", "getRegistryDataVersionOutput"], () => require("./getRegistryDataVersion"));
exports.getRegistryEnvironmentContainer = null;
exports.getRegistryEnvironmentContainerOutput = null;
utilities.lazyLoad(exports, ["getRegistryEnvironmentContainer", "getRegistryEnvironmentContainerOutput"], () => require("./getRegistryEnvironmentContainer"));
exports.getRegistryEnvironmentVersion = null;
exports.getRegistryEnvironmentVersionOutput = null;
utilities.lazyLoad(exports, ["getRegistryEnvironmentVersion", "getRegistryEnvironmentVersionOutput"], () => require("./getRegistryEnvironmentVersion"));
exports.getRegistryModelContainer = null;
exports.getRegistryModelContainerOutput = null;
utilities.lazyLoad(exports, ["getRegistryModelContainer", "getRegistryModelContainerOutput"], () => require("./getRegistryModelContainer"));
exports.getRegistryModelVersion = null;
exports.getRegistryModelVersionOutput = null;
utilities.lazyLoad(exports, ["getRegistryModelVersion", "getRegistryModelVersionOutput"], () => require("./getRegistryModelVersion"));
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.getServerlessEndpoint = null;
exports.getServerlessEndpointOutput = null;
utilities.lazyLoad(exports, ["getServerlessEndpoint", "getServerlessEndpointOutput"], () => require("./getServerlessEndpoint"));
exports.getServerlessEndpointStatus = null;
exports.getServerlessEndpointStatusOutput = null;
utilities.lazyLoad(exports, ["getServerlessEndpointStatus", "getServerlessEndpointStatusOutput"], () => require("./getServerlessEndpointStatus"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspaceConnection = null;
exports.getWorkspaceConnectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceConnection", "getWorkspaceConnectionOutput"], () => require("./getWorkspaceConnection"));
exports.InferenceEndpoint = null;
utilities.lazyLoad(exports, ["InferenceEndpoint"], () => require("./inferenceEndpoint"));
exports.InferenceGroup = null;
utilities.lazyLoad(exports, ["InferenceGroup"], () => require("./inferenceGroup"));
exports.InferencePool = null;
utilities.lazyLoad(exports, ["InferencePool"], () => require("./inferencePool"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.LabelingJob = null;
utilities.lazyLoad(exports, ["LabelingJob"], () => require("./labelingJob"));
exports.listBatchEndpointKeys = null;
exports.listBatchEndpointKeysOutput = null;
utilities.lazyLoad(exports, ["listBatchEndpointKeys", "listBatchEndpointKeysOutput"], () => require("./listBatchEndpointKeys"));
exports.listComputeKeys = null;
exports.listComputeKeysOutput = null;
utilities.lazyLoad(exports, ["listComputeKeys", "listComputeKeysOutput"], () => require("./listComputeKeys"));
exports.listComputeNodes = null;
exports.listComputeNodesOutput = null;
utilities.lazyLoad(exports, ["listComputeNodes", "listComputeNodesOutput"], () => require("./listComputeNodes"));
exports.listDatastoreSecrets = null;
exports.listDatastoreSecretsOutput = null;
utilities.lazyLoad(exports, ["listDatastoreSecrets", "listDatastoreSecretsOutput"], () => require("./listDatastoreSecrets"));
exports.listEndpointKeys = null;
exports.listEndpointKeysOutput = null;
utilities.lazyLoad(exports, ["listEndpointKeys", "listEndpointKeysOutput"], () => require("./listEndpointKeys"));
exports.listOnlineEndpointKeys = null;
exports.listOnlineEndpointKeysOutput = null;
utilities.lazyLoad(exports, ["listOnlineEndpointKeys", "listOnlineEndpointKeysOutput"], () => require("./listOnlineEndpointKeys"));
exports.listServerlessEndpointKeys = null;
exports.listServerlessEndpointKeysOutput = null;
utilities.lazyLoad(exports, ["listServerlessEndpointKeys", "listServerlessEndpointKeysOutput"], () => require("./listServerlessEndpointKeys"));
exports.listWorkspaceConnectionModels = null;
exports.listWorkspaceConnectionModelsOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceConnectionModels", "listWorkspaceConnectionModelsOutput"], () => require("./listWorkspaceConnectionModels"));
exports.listWorkspaceConnectionSecrets = null;
exports.listWorkspaceConnectionSecretsOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceConnectionSecrets", "listWorkspaceConnectionSecretsOutput"], () => require("./listWorkspaceConnectionSecrets"));
exports.listWorkspaceKeys = null;
exports.listWorkspaceKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceKeys", "listWorkspaceKeysOutput"], () => require("./listWorkspaceKeys"));
exports.listWorkspaceNotebookAccessToken = null;
exports.listWorkspaceNotebookAccessTokenOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceNotebookAccessToken", "listWorkspaceNotebookAccessTokenOutput"], () => require("./listWorkspaceNotebookAccessToken"));
exports.listWorkspaceNotebookKeys = null;
exports.listWorkspaceNotebookKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceNotebookKeys", "listWorkspaceNotebookKeysOutput"], () => require("./listWorkspaceNotebookKeys"));
exports.listWorkspaceStorageAccountKeys = null;
exports.listWorkspaceStorageAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceStorageAccountKeys", "listWorkspaceStorageAccountKeysOutput"], () => require("./listWorkspaceStorageAccountKeys"));
exports.ManagedNetworkSettingsRule = null;
utilities.lazyLoad(exports, ["ManagedNetworkSettingsRule"], () => require("./managedNetworkSettingsRule"));
exports.MarketplaceSubscription = null;
utilities.lazyLoad(exports, ["MarketplaceSubscription"], () => require("./marketplaceSubscription"));
exports.ModelContainer = null;
utilities.lazyLoad(exports, ["ModelContainer"], () => require("./modelContainer"));
exports.ModelVersion = null;
utilities.lazyLoad(exports, ["ModelVersion"], () => require("./modelVersion"));
exports.OnlineDeployment = null;
utilities.lazyLoad(exports, ["OnlineDeployment"], () => require("./onlineDeployment"));
exports.OnlineEndpoint = null;
utilities.lazyLoad(exports, ["OnlineEndpoint"], () => require("./onlineEndpoint"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.RaiPolicy = null;
utilities.lazyLoad(exports, ["RaiPolicy"], () => require("./raiPolicy"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.RegistryCodeContainer = null;
utilities.lazyLoad(exports, ["RegistryCodeContainer"], () => require("./registryCodeContainer"));
exports.RegistryCodeVersion = null;
utilities.lazyLoad(exports, ["RegistryCodeVersion"], () => require("./registryCodeVersion"));
exports.RegistryComponentContainer = null;
utilities.lazyLoad(exports, ["RegistryComponentContainer"], () => require("./registryComponentContainer"));
exports.RegistryComponentVersion = null;
utilities.lazyLoad(exports, ["RegistryComponentVersion"], () => require("./registryComponentVersion"));
exports.RegistryDataContainer = null;
utilities.lazyLoad(exports, ["RegistryDataContainer"], () => require("./registryDataContainer"));
exports.RegistryDataVersion = null;
utilities.lazyLoad(exports, ["RegistryDataVersion"], () => require("./registryDataVersion"));
exports.RegistryEnvironmentContainer = null;
utilities.lazyLoad(exports, ["RegistryEnvironmentContainer"], () => require("./registryEnvironmentContainer"));
exports.RegistryEnvironmentVersion = null;
utilities.lazyLoad(exports, ["RegistryEnvironmentVersion"], () => require("./registryEnvironmentVersion"));
exports.RegistryModelContainer = null;
utilities.lazyLoad(exports, ["RegistryModelContainer"], () => require("./registryModelContainer"));
exports.RegistryModelVersion = null;
utilities.lazyLoad(exports, ["RegistryModelVersion"], () => require("./registryModelVersion"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
exports.ServerlessEndpoint = null;
utilities.lazyLoad(exports, ["ServerlessEndpoint"], () => require("./serverlessEndpoint"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspaceConnection = null;
utilities.lazyLoad(exports, ["WorkspaceConnection"], () => require("./workspaceConnection"));
// Export enums:
__exportStar(require("../../types/enums/machinelearningservices/v20240401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningservices/v20240401preview:BatchDeployment":
                return new exports.BatchDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:BatchEndpoint":
                return new exports.BatchEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:CapacityReservationGroup":
                return new exports.CapacityReservationGroup(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:CodeContainer":
                return new exports.CodeContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:CodeVersion":
                return new exports.CodeVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ComponentContainer":
                return new exports.ComponentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ComponentVersion":
                return new exports.ComponentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Compute":
                return new exports.Compute(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ConnectionDeployment":
                return new exports.ConnectionDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ConnectionRaiBlocklist":
                return new exports.ConnectionRaiBlocklist(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ConnectionRaiBlocklistItem":
                return new exports.ConnectionRaiBlocklistItem(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ConnectionRaiPolicy":
                return new exports.ConnectionRaiPolicy(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:DataContainer":
                return new exports.DataContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:DataVersion":
                return new exports.DataVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Datastore":
                return new exports.Datastore(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:EndpointDeployment":
                return new exports.EndpointDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:EnvironmentContainer":
                return new exports.EnvironmentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:EnvironmentVersion":
                return new exports.EnvironmentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:FeaturesetContainerEntity":
                return new exports.FeaturesetContainerEntity(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:FeaturesetVersion":
                return new exports.FeaturesetVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:FeaturestoreEntityContainerEntity":
                return new exports.FeaturestoreEntityContainerEntity(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:FeaturestoreEntityVersion":
                return new exports.FeaturestoreEntityVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:InferenceEndpoint":
                return new exports.InferenceEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:InferenceGroup":
                return new exports.InferenceGroup(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:InferencePool":
                return new exports.InferencePool(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:LabelingJob":
                return new exports.LabelingJob(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ManagedNetworkSettingsRule":
                return new exports.ManagedNetworkSettingsRule(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:MarketplaceSubscription":
                return new exports.MarketplaceSubscription(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ModelContainer":
                return new exports.ModelContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ModelVersion":
                return new exports.ModelVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:OnlineDeployment":
                return new exports.OnlineDeployment(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:OnlineEndpoint":
                return new exports.OnlineEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RaiPolicy":
                return new exports.RaiPolicy(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryCodeContainer":
                return new exports.RegistryCodeContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryCodeVersion":
                return new exports.RegistryCodeVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryComponentContainer":
                return new exports.RegistryComponentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryComponentVersion":
                return new exports.RegistryComponentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryDataContainer":
                return new exports.RegistryDataContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryDataVersion":
                return new exports.RegistryDataVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryEnvironmentContainer":
                return new exports.RegistryEnvironmentContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryEnvironmentVersion":
                return new exports.RegistryEnvironmentVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryModelContainer":
                return new exports.RegistryModelContainer(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:RegistryModelVersion":
                return new exports.RegistryModelVersion(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:ServerlessEndpoint":
                return new exports.ServerlessEndpoint(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20240401preview:WorkspaceConnection":
                return new exports.WorkspaceConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningservices/v20240401preview", _module);
