// @pulumi/azure-native/insights/v20210501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionDiagnosticSetting = exports.ManagementGroupDiagnosticSetting = exports.listDiagnosticSettingsCategoryOutput = exports.listDiagnosticSettingsCategory = exports.getSubscriptionDiagnosticSettingOutput = exports.getSubscriptionDiagnosticSetting = exports.getManagementGroupDiagnosticSettingOutput = exports.getManagementGroupDiagnosticSetting = exports.getDiagnosticSettingOutput = exports.getDiagnosticSetting = exports.DiagnosticSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DiagnosticSetting = null;
utilities.lazyLoad(exports, ["DiagnosticSetting"], () => require("./diagnosticSetting"));
exports.getDiagnosticSetting = null;
exports.getDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getDiagnosticSetting", "getDiagnosticSettingOutput"], () => require("./getDiagnosticSetting"));
exports.getManagementGroupDiagnosticSetting = null;
exports.getManagementGroupDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getManagementGroupDiagnosticSetting", "getManagementGroupDiagnosticSettingOutput"], () => require("./getManagementGroupDiagnosticSetting"));
exports.getSubscriptionDiagnosticSetting = null;
exports.getSubscriptionDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionDiagnosticSetting", "getSubscriptionDiagnosticSettingOutput"], () => require("./getSubscriptionDiagnosticSetting"));
exports.listDiagnosticSettingsCategory = null;
exports.listDiagnosticSettingsCategoryOutput = null;
utilities.lazyLoad(exports, ["listDiagnosticSettingsCategory", "listDiagnosticSettingsCategoryOutput"], () => require("./listDiagnosticSettingsCategory"));
exports.ManagementGroupDiagnosticSetting = null;
utilities.lazyLoad(exports, ["ManagementGroupDiagnosticSetting"], () => require("./managementGroupDiagnosticSetting"));
exports.SubscriptionDiagnosticSetting = null;
utilities.lazyLoad(exports, ["SubscriptionDiagnosticSetting"], () => require("./subscriptionDiagnosticSetting"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights/v20210501preview:DiagnosticSetting":
                return new exports.DiagnosticSetting(name, undefined, { urn });
            case "azure-native:insights/v20210501preview:ManagementGroupDiagnosticSetting":
                return new exports.ManagementGroupDiagnosticSetting(name, undefined, { urn });
            case "azure-native:insights/v20210501preview:SubscriptionDiagnosticSetting":
                return new exports.SubscriptionDiagnosticSetting(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights/v20210501preview", _module);
