// @pulumi/azure-native/types/enums/costmanagement/v20230701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeeksOfMonth = exports.TimeframeType = exports.SystemAssignedServiceIdentityType = exports.StatusType = exports.ScheduledActionStatus = exports.ScheduledActionKind = exports.ScheduleFrequency = exports.ReportType = exports.ReportTimeframeType = exports.ReportGranularityType = exports.ReportConfigSortingType = exports.RecurrenceType = exports.QueryColumnType = exports.PivotTypeType = exports.OperatorType = exports.MetricType = exports.KpiTypeType = exports.GranularityType = exports.FunctionType = exports.FormatType = exports.FilterItemNames = exports.FileFormat = exports.ExportType = exports.DestinationType = exports.DaysOfWeek = exports.DataOverwriteBehaviorType = exports.CompressionModeType = exports.ChartType = exports.AccumulatedType = void 0;
exports.AccumulatedType = {
    True: "true",
    False: "false",
};
exports.ChartType = {
    Area: "Area",
    Line: "Line",
    StackedColumn: "StackedColumn",
    GroupedColumn: "GroupedColumn",
    Table: "Table",
};
exports.CompressionModeType = {
    Gzip: "gzip",
    None: "None",
};
exports.DataOverwriteBehaviorType = {
    OverwritePreviousReport: "OverwritePreviousReport",
    CreateNewReport: "CreateNewReport",
};
exports.DaysOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DestinationType = {
    AzureBlob: "AzureBlob",
};
exports.ExportType = {
    Usage: "Usage",
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
    FocusCost: "FocusCost",
    PriceSheet: "PriceSheet",
    ReservationTransactions: "ReservationTransactions",
    ReservationRecommendations: "ReservationRecommendations",
    ReservationDetails: "ReservationDetails",
};
exports.FileFormat = {
    Csv: "Csv",
};
exports.FilterItemNames = {
    ReservationScope: "ReservationScope",
    ResourceType: "ResourceType",
    LookBackPeriod: "LookBackPeriod",
};
exports.FormatType = {
    Csv: "Csv",
};
exports.FunctionType = {
    Sum: "Sum",
};
exports.GranularityType = {
    Daily: "Daily",
    Monthly: "Monthly",
};
exports.KpiTypeType = {
    Forecast: "Forecast",
    Budget: "Budget",
};
exports.MetricType = {
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
    AHUB: "AHUB",
};
exports.OperatorType = {
    In: "In",
    Contains: "Contains",
};
exports.PivotTypeType = {
    Dimension: "Dimension",
    TagKey: "TagKey",
};
exports.QueryColumnType = {
    /**
     * The tag associated with the cost data.
     */
    TagKey: "TagKey",
    /**
     * The dimension of cost data.
     */
    Dimension: "Dimension",
};
exports.RecurrenceType = {
    Daily: "Daily",
    Weekly: "Weekly",
    Monthly: "Monthly",
    Annually: "Annually",
};
exports.ReportConfigSortingType = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.ReportGranularityType = {
    Daily: "Daily",
    Monthly: "Monthly",
};
exports.ReportTimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    YearToDate: "YearToDate",
    Custom: "Custom",
};
exports.ReportType = {
    Usage: "Usage",
};
exports.ScheduleFrequency = {
    /**
     * Cost analysis data will be emailed every day.
     */
    Daily: "Daily",
    /**
     * Cost analysis data will be emailed every week.
     */
    Weekly: "Weekly",
    /**
     * Cost analysis data will be emailed every month.
     */
    Monthly: "Monthly",
};
exports.ScheduledActionKind = {
    /**
     * Cost analysis data will be emailed.
     */
    Email: "Email",
    /**
     * Cost anomaly information will be emailed. Available only on subscription scope at daily frequency. If no anomaly is detected on the resource, an email won't be sent.
     */
    InsightAlert: "InsightAlert",
};
exports.ScheduledActionStatus = {
    /**
     * Scheduled action is saved but will not be run.
     */
    Disabled: "Disabled",
    /**
     * Scheduled action is saved and will be run.
     */
    Enabled: "Enabled",
    /**
     * Scheduled action is expired.
     */
    Expired: "Expired",
};
exports.StatusType = {
    Active: "Active",
    Inactive: "Inactive",
    SystemSuspended: "SystemSuspended",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.TimeframeType = {
    MonthToDate: "MonthToDate",
    BillingMonthToDate: "BillingMonthToDate",
    TheLastMonth: "TheLastMonth",
    TheLastBillingMonth: "TheLastBillingMonth",
    WeekToDate: "WeekToDate",
    Custom: "Custom",
    TheCurrentMonth: "TheCurrentMonth",
};
exports.WeeksOfMonth = {
    First: "First",
    Second: "Second",
    Third: "Third",
    Fourth: "Fourth",
    Last: "Last",
};
