// @pulumi/azure-native/databox/v20240201preview/job.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Job = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * Job Resource.
 */
class Job extends pulumi.CustomResource {
    /**
     * Get an existing Job resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Job(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Job.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Job.__pulumiType;
    }
    /**
     * Create a Job resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            if ((!args || args.transferType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transferType'");
            }
            resourceInputs["deliveryInfo"] = args ? args.deliveryInfo : undefined;
            resourceInputs["deliveryType"] = (_a = (args ? args.deliveryType : undefined)) !== null && _a !== void 0 ? _a : "NonScheduled";
            resourceInputs["details"] = args ? args.details : undefined;
            resourceInputs["identity"] = args ? (args.identity ? pulumi.output(args.identity).apply(inputs.databox.v20240201preview.resourceIdentityArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["jobName"] = args ? args.jobName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["transferType"] = args ? args.transferType : undefined;
            resourceInputs["allDevicesLost"] = undefined /*out*/;
            resourceInputs["cancellationReason"] = undefined /*out*/;
            resourceInputs["delayedStage"] = undefined /*out*/;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["isCancellable"] = undefined /*out*/;
            resourceInputs["isCancellableWithoutFee"] = undefined /*out*/;
            resourceInputs["isDeletable"] = undefined /*out*/;
            resourceInputs["isPrepareToShipEnabled"] = undefined /*out*/;
            resourceInputs["isShippingAddressEditable"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["reverseShippingDetailsUpdate"] = undefined /*out*/;
            resourceInputs["reverseTransportPreferenceUpdate"] = undefined /*out*/;
            resourceInputs["startTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allDevicesLost"] = undefined /*out*/;
            resourceInputs["cancellationReason"] = undefined /*out*/;
            resourceInputs["delayedStage"] = undefined /*out*/;
            resourceInputs["deliveryInfo"] = undefined /*out*/;
            resourceInputs["deliveryType"] = undefined /*out*/;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["isCancellable"] = undefined /*out*/;
            resourceInputs["isCancellableWithoutFee"] = undefined /*out*/;
            resourceInputs["isDeletable"] = undefined /*out*/;
            resourceInputs["isPrepareToShipEnabled"] = undefined /*out*/;
            resourceInputs["isShippingAddressEditable"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["reverseShippingDetailsUpdate"] = undefined /*out*/;
            resourceInputs["reverseTransportPreferenceUpdate"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["startTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["transferType"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databox/v20180101:Job" }, { type: "azure-native:databox/v20190901:Job" }, { type: "azure-native:databox/v20200401:Job" }, { type: "azure-native:databox/v20201101:Job" }, { type: "azure-native:databox/v20210301:Job" }, { type: "azure-native:databox/v20210501:Job" }, { type: "azure-native:databox/v20210801preview:Job" }, { type: "azure-native:databox/v20211201:Job" }, { type: "azure-native:databox/v20220201:Job" }, { type: "azure-native:databox/v20220901:Job" }, { type: "azure-native:databox/v20221001:Job" }, { type: "azure-native:databox/v20221201:Job" }, { type: "azure-native:databox/v20230301:Job" }, { type: "azure-native:databox/v20231201:Job" }, { type: "azure-native:databox/v20240301preview:Job" }, { type: "azure-native:databox/v20250201:Job" }, { type: "azure-native:databox:Job" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Job.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Job = Job;
/** @internal */
Job.__pulumiType = 'azure-native:databox/v20240201preview:Job';
