// @pulumi/azure-native/deviceregistry/v20240901preview/assetEndpointProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetEndpointProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * Asset Endpoint Profile definition.
 */
class AssetEndpointProfile extends pulumi.CustomResource {
    /**
     * Get an existing AssetEndpointProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AssetEndpointProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AssetEndpointProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AssetEndpointProfile.__pulumiType;
    }
    /**
     * Create a AssetEndpointProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.endpointProfileType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointProfileType'");
            }
            if ((!args || args.extendedLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.targetAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetAddress'");
            }
            resourceInputs["additionalConfiguration"] = args ? args.additionalConfiguration : undefined;
            resourceInputs["assetEndpointProfileName"] = args ? args.assetEndpointProfileName : undefined;
            resourceInputs["authentication"] = args ? (args.authentication ? pulumi.output(args.authentication).apply(inputs.deviceregistry.v20240901preview.authenticationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["discoveredAssetEndpointProfileRef"] = args ? args.discoveredAssetEndpointProfileRef : undefined;
            resourceInputs["endpointProfileType"] = args ? args.endpointProfileType : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetAddress"] = args ? args.targetAddress : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uuid"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalConfiguration"] = undefined /*out*/;
            resourceInputs["authentication"] = undefined /*out*/;
            resourceInputs["discoveredAssetEndpointProfileRef"] = undefined /*out*/;
            resourceInputs["endpointProfileType"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetAddress"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uuid"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deviceregistry/v20231101preview:AssetEndpointProfile" }, { type: "azure-native:deviceregistry/v20241101:AssetEndpointProfile" }, { type: "azure-native:deviceregistry:AssetEndpointProfile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AssetEndpointProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AssetEndpointProfile = AssetEndpointProfile;
/** @internal */
AssetEndpointProfile.__pulumiType = 'azure-native:deviceregistry/v20240901preview:AssetEndpointProfile';
