// @pulumi/azure-native/web/v20230101/webAppAuthSettingsV2WithoutSecrets.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAuthSettingsV2WithoutSecrets = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Configuration settings for the Azure App Service Authentication / Authorization V2 feature.
 */
class WebAppAuthSettingsV2WithoutSecrets extends pulumi.CustomResource {
    /**
     * Get an existing WebAppAuthSettingsV2WithoutSecrets resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppAuthSettingsV2WithoutSecrets(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppAuthSettingsV2WithoutSecrets.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppAuthSettingsV2WithoutSecrets.__pulumiType;
    }
    /**
     * Create a WebAppAuthSettingsV2WithoutSecrets resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["globalValidation"] = args ? args.globalValidation : undefined;
            resourceInputs["httpSettings"] = args ? args.httpSettings : undefined;
            resourceInputs["identityProviders"] = args ? args.identityProviders : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["login"] = args ? args.login : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["platform"] = args ? args.platform : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["globalValidation"] = undefined /*out*/;
            resourceInputs["httpSettings"] = undefined /*out*/;
            resourceInputs["identityProviders"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["login"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20200601:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20200901:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20201001:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20201201:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20210101:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20210115:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20210201:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20210301:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20220301:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20220901:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20231201:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web/v20240401:WebAppAuthSettingsV2WithoutSecrets" }, { type: "azure-native:web:WebAppAuthSettingsV2WithoutSecrets" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppAuthSettingsV2WithoutSecrets.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppAuthSettingsV2WithoutSecrets = WebAppAuthSettingsV2WithoutSecrets;
/** @internal */
WebAppAuthSettingsV2WithoutSecrets.__pulumiType = 'azure-native:web/v20230101:WebAppAuthSettingsV2WithoutSecrets';
