// @pulumi/azure-native/machinelearning/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20191001 = exports.v20170101 = exports.v20160501preview = exports.Workspace = exports.WebService = exports.listWorkspaceKeysOutput = exports.listWorkspaceKeys = exports.getWorkspaceOutput = exports.getWorkspace = exports.getWebServiceOutput = exports.getWebService = exports.getCommitmentPlanOutput = exports.getCommitmentPlan = exports.CommitmentPlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CommitmentPlan = null;
utilities.lazyLoad(exports, ["CommitmentPlan"], () => require("./commitmentPlan"));
exports.getCommitmentPlan = null;
exports.getCommitmentPlanOutput = null;
utilities.lazyLoad(exports, ["getCommitmentPlan", "getCommitmentPlanOutput"], () => require("./getCommitmentPlan"));
exports.getWebService = null;
exports.getWebServiceOutput = null;
utilities.lazyLoad(exports, ["getWebService", "getWebServiceOutput"], () => require("./getWebService"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.listWorkspaceKeys = null;
exports.listWorkspaceKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceKeys", "listWorkspaceKeysOutput"], () => require("./listWorkspaceKeys"));
exports.WebService = null;
utilities.lazyLoad(exports, ["WebService"], () => require("./webService"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/machinelearning"), exports);
// Export sub-modules:
const v20160501preview = require("./v20160501preview");
exports.v20160501preview = v20160501preview;
const v20170101 = require("./v20170101");
exports.v20170101 = v20170101;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearning:CommitmentPlan":
                return new exports.CommitmentPlan(name, undefined, { urn });
            case "azure-native:machinelearning:WebService":
                return new exports.WebService(name, undefined, { urn });
            case "azure-native:machinelearning:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearning", _module);
