// @pulumi/azure-native/recoveryservices/v20240430preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vault = exports.ResourceGuardProxy = exports.ProtectionPolicy = exports.ProtectionIntent = exports.ProtectionContainer = exports.ProtectedItem = exports.PrivateEndpointConnection = exports.getVaultOutput = exports.getVault = exports.getResourceGuardProxyOutput = exports.getResourceGuardProxy = exports.getProtectionPolicyOutput = exports.getProtectionPolicy = exports.getProtectionIntentOutput = exports.getProtectionIntent = exports.getProtectionContainerOutput = exports.getProtectionContainer = exports.getProtectedItemOutput = exports.getProtectedItem = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getProtectedItem = null;
exports.getProtectedItemOutput = null;
utilities.lazyLoad(exports, ["getProtectedItem", "getProtectedItemOutput"], () => require("./getProtectedItem"));
exports.getProtectionContainer = null;
exports.getProtectionContainerOutput = null;
utilities.lazyLoad(exports, ["getProtectionContainer", "getProtectionContainerOutput"], () => require("./getProtectionContainer"));
exports.getProtectionIntent = null;
exports.getProtectionIntentOutput = null;
utilities.lazyLoad(exports, ["getProtectionIntent", "getProtectionIntentOutput"], () => require("./getProtectionIntent"));
exports.getProtectionPolicy = null;
exports.getProtectionPolicyOutput = null;
utilities.lazyLoad(exports, ["getProtectionPolicy", "getProtectionPolicyOutput"], () => require("./getProtectionPolicy"));
exports.getResourceGuardProxy = null;
exports.getResourceGuardProxyOutput = null;
utilities.lazyLoad(exports, ["getResourceGuardProxy", "getResourceGuardProxyOutput"], () => require("./getResourceGuardProxy"));
exports.getVault = null;
exports.getVaultOutput = null;
utilities.lazyLoad(exports, ["getVault", "getVaultOutput"], () => require("./getVault"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.ProtectedItem = null;
utilities.lazyLoad(exports, ["ProtectedItem"], () => require("./protectedItem"));
exports.ProtectionContainer = null;
utilities.lazyLoad(exports, ["ProtectionContainer"], () => require("./protectionContainer"));
exports.ProtectionIntent = null;
utilities.lazyLoad(exports, ["ProtectionIntent"], () => require("./protectionIntent"));
exports.ProtectionPolicy = null;
utilities.lazyLoad(exports, ["ProtectionPolicy"], () => require("./protectionPolicy"));
exports.ResourceGuardProxy = null;
utilities.lazyLoad(exports, ["ResourceGuardProxy"], () => require("./resourceGuardProxy"));
exports.Vault = null;
utilities.lazyLoad(exports, ["Vault"], () => require("./vault"));
// Export enums:
__exportStar(require("../../types/enums/recoveryservices/v20240430preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:recoveryservices/v20240430preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:ProtectedItem":
                return new exports.ProtectedItem(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:ProtectionContainer":
                return new exports.ProtectionContainer(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:ProtectionIntent":
                return new exports.ProtectionIntent(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:ProtectionPolicy":
                return new exports.ProtectionPolicy(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:ResourceGuardProxy":
                return new exports.ResourceGuardProxy(name, undefined, { urn });
            case "azure-native:recoveryservices/v20240430preview:Vault":
                return new exports.Vault(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "recoveryservices/v20240430preview", _module);
