// @pulumi/azure-native/recoveryservices/v20240430preview/resourceGuardProxy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceGuardProxy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class ResourceGuardProxy extends pulumi.CustomResource {
    /**
     * Get an existing ResourceGuardProxy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ResourceGuardProxy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceGuardProxy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceGuardProxy.__pulumiType;
    }
    /**
     * Create a ResourceGuardProxy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vaultName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vaultName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceGuardProxyName"] = args ? args.resourceGuardProxyName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vaultName"] = args ? args.vaultName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20210201preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20210701:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20210801:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20211001:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20211201:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220101:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220201:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220301:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220401:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220601preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220901preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20220930preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20221001:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20230101:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20230201:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20230401:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20230601:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20230801:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20240101:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20240201:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20240401:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20240730preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20241001:ResourceGuardProxy" }, { type: "azure-native:recoveryservices/v20241101preview:ResourceGuardProxy" }, { type: "azure-native:recoveryservices:ResourceGuardProxy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ResourceGuardProxy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResourceGuardProxy = ResourceGuardProxy;
/** @internal */
ResourceGuardProxy.__pulumiType = 'azure-native:recoveryservices/v20240430preview:ResourceGuardProxy';
