// @pulumi/azure-native/enterpriseknowledgegraph/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20181203 = exports.getEnterpriseKnowledgeGraphOutput = exports.getEnterpriseKnowledgeGraph = exports.EnterpriseKnowledgeGraph = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EnterpriseKnowledgeGraph = null;
utilities.lazyLoad(exports, ["EnterpriseKnowledgeGraph"], () => require("./enterpriseKnowledgeGraph"));
exports.getEnterpriseKnowledgeGraph = null;
exports.getEnterpriseKnowledgeGraphOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseKnowledgeGraph", "getEnterpriseKnowledgeGraphOutput"], () => require("./getEnterpriseKnowledgeGraph"));
// Export enums:
__exportStar(require("../types/enums/enterpriseknowledgegraph"), exports);
// Export sub-modules:
const v20181203 = require("./v20181203");
exports.v20181203 = v20181203;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:enterpriseknowledgegraph:EnterpriseKnowledgeGraph":
                return new exports.EnterpriseKnowledgeGraph(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "enterpriseknowledgegraph", _module);
