// @pulumi/gcp/networkconnectivity/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Spoke = exports.ServiceConnectionPolicy = exports.RegionalEndpoint = exports.PolicyBasedRoute = exports.InternalRange = exports.Hub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Hub = null;
utilities.lazyLoad(exports, ["Hub"], () => require("./hub"));
exports.InternalRange = null;
utilities.lazyLoad(exports, ["InternalRange"], () => require("./internalRange"));
exports.PolicyBasedRoute = null;
utilities.lazyLoad(exports, ["PolicyBasedRoute"], () => require("./policyBasedRoute"));
exports.RegionalEndpoint = null;
utilities.lazyLoad(exports, ["RegionalEndpoint"], () => require("./regionalEndpoint"));
exports.ServiceConnectionPolicy = null;
utilities.lazyLoad(exports, ["ServiceConnectionPolicy"], () => require("./serviceConnectionPolicy"));
exports.Spoke = null;
utilities.lazyLoad(exports, ["Spoke"], () => require("./spoke"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:networkconnectivity/hub:Hub":
                return new exports.Hub(name, undefined, { urn });
            case "gcp:networkconnectivity/internalRange:InternalRange":
                return new exports.InternalRange(name, undefined, { urn });
            case "gcp:networkconnectivity/policyBasedRoute:PolicyBasedRoute":
                return new exports.PolicyBasedRoute(name, undefined, { urn });
            case "gcp:networkconnectivity/regionalEndpoint:RegionalEndpoint":
                return new exports.RegionalEndpoint(name, undefined, { urn });
            case "gcp:networkconnectivity/serviceConnectionPolicy:ServiceConnectionPolicy":
                return new exports.ServiceConnectionPolicy(name, undefined, { urn });
            case "gcp:networkconnectivity/spoke:Spoke":
                return new exports.Spoke(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/hub", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/internalRange", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/policyBasedRoute", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/regionalEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/serviceConnectionPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networkconnectivity/spoke", _module);
