// @pulumi/azure-native/streamanalytics/v20211001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingJob = exports.Output = exports.Input = exports.getStreamingJobOutput = exports.getStreamingJob = exports.getOutputOutput = exports.getOutput = exports.getInputOutput = exports.getInput = exports.getFunctionOutput = exports.getFunction = exports.Function = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Function = null;
utilities.lazyLoad(exports, ["Function"], () => require("./function"));
exports.getFunction = null;
exports.getFunctionOutput = null;
utilities.lazyLoad(exports, ["getFunction", "getFunctionOutput"], () => require("./getFunction"));
exports.getInput = null;
exports.getInputOutput = null;
utilities.lazyLoad(exports, ["getInput", "getInputOutput"], () => require("./getInput"));
exports.getOutput = null;
exports.getOutputOutput = null;
utilities.lazyLoad(exports, ["getOutput", "getOutputOutput"], () => require("./getOutput"));
exports.getStreamingJob = null;
exports.getStreamingJobOutput = null;
utilities.lazyLoad(exports, ["getStreamingJob", "getStreamingJobOutput"], () => require("./getStreamingJob"));
exports.Input = null;
utilities.lazyLoad(exports, ["Input"], () => require("./input"));
exports.Output = null;
utilities.lazyLoad(exports, ["Output"], () => require("./output"));
exports.StreamingJob = null;
utilities.lazyLoad(exports, ["StreamingJob"], () => require("./streamingJob"));
// Export enums:
__exportStar(require("../../types/enums/streamanalytics/v20211001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:streamanalytics/v20211001preview:Function":
                return new exports.Function(name, undefined, { urn });
            case "azure-native:streamanalytics/v20211001preview:Input":
                return new exports.Input(name, undefined, { urn });
            case "azure-native:streamanalytics/v20211001preview:Output":
                return new exports.Output(name, undefined, { urn });
            case "azure-native:streamanalytics/v20211001preview:StreamingJob":
                return new exports.StreamingJob(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "streamanalytics/v20211001preview", _module);
