// @pulumi/gcp/assuredworkloads/workload.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workload = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AssuredWorkloads Workload resource
 *
 * ## Example Usage
 *
 * ### Basic_workload
 * A basic test of a assuredworkloads api
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const primary = new gcp.assuredworkloads.Workload("primary", {
 *     complianceRegime: "FEDRAMP_MODERATE",
 *     displayName: "{{display}}",
 *     location: "us-west1",
 *     organization: "123456789",
 *     billingAccount: "billingAccounts/000000-0000000-0000000-000000",
 *     kmsSettings: {
 *         nextRotationTime: "9999-10-02T15:01:23Z",
 *         rotationPeriod: "10368000s",
 *     },
 *     provisionedResourcesParent: "folders/519620126891",
 *     resourceSettings: [
 *         {
 *             displayName: "folder-display-name",
 *             resourceType: "CONSUMER_FOLDER",
 *         },
 *         {
 *             resourceType: "ENCRYPTION_KEYS_PROJECT",
 *         },
 *         {
 *             resourceId: "ring",
 *             resourceType: "KEYRING",
 *         },
 *     ],
 *     violationNotificationsEnabled: true,
 *     labels: {
 *         "label-one": "value-one",
 *     },
 * });
 * ```
 * ### Sovereign_controls_workload
 * A Sovereign Controls test of the assuredworkloads api
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const primary = new gcp.assuredworkloads.Workload("primary", {
 *     complianceRegime: "EU_REGIONS_AND_SUPPORT",
 *     displayName: "display",
 *     location: "europe-west9",
 *     organization: "123456789",
 *     billingAccount: "billingAccounts/000000-0000000-0000000-000000",
 *     enableSovereignControls: true,
 *     kmsSettings: {
 *         nextRotationTime: "9999-10-02T15:01:23Z",
 *         rotationPeriod: "10368000s",
 *     },
 *     resourceSettings: [
 *         {
 *             resourceType: "CONSUMER_FOLDER",
 *         },
 *         {
 *             resourceType: "ENCRYPTION_KEYS_PROJECT",
 *         },
 *         {
 *             resourceId: "ring",
 *             resourceType: "KEYRING",
 *         },
 *     ],
 *     labels: {
 *         "label-one": "value-one",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Workload can be imported using any of these accepted formats:
 *
 * * `organizations/{{organization}}/locations/{{location}}/workloads/{{name}}`
 *
 * * `{{organization}}/{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, Workload can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:assuredworkloads/workload:Workload default organizations/{{organization}}/locations/{{location}}/workloads/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:assuredworkloads/workload:Workload default {{organization}}/{{location}}/{{name}}
 * ```
 */
class Workload extends pulumi.CustomResource {
    /**
     * Get an existing Workload resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Workload(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Workload.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Workload.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["billingAccount"] = state ? state.billingAccount : undefined;
            resourceInputs["complianceRegime"] = state ? state.complianceRegime : undefined;
            resourceInputs["complianceStatuses"] = state ? state.complianceStatuses : undefined;
            resourceInputs["compliantButDisallowedServices"] = state ? state.compliantButDisallowedServices : undefined;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["effectiveLabels"] = state ? state.effectiveLabels : undefined;
            resourceInputs["ekmProvisioningResponses"] = state ? state.ekmProvisioningResponses : undefined;
            resourceInputs["enableSovereignControls"] = state ? state.enableSovereignControls : undefined;
            resourceInputs["kajEnrollmentState"] = state ? state.kajEnrollmentState : undefined;
            resourceInputs["kmsSettings"] = state ? state.kmsSettings : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["organization"] = state ? state.organization : undefined;
            resourceInputs["partner"] = state ? state.partner : undefined;
            resourceInputs["partnerPermissions"] = state ? state.partnerPermissions : undefined;
            resourceInputs["provisionedResourcesParent"] = state ? state.provisionedResourcesParent : undefined;
            resourceInputs["pulumiLabels"] = state ? state.pulumiLabels : undefined;
            resourceInputs["resourceSettings"] = state ? state.resourceSettings : undefined;
            resourceInputs["resources"] = state ? state.resources : undefined;
            resourceInputs["saaEnrollmentResponses"] = state ? state.saaEnrollmentResponses : undefined;
            resourceInputs["violationNotificationsEnabled"] = state ? state.violationNotificationsEnabled : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.complianceRegime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'complianceRegime'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'location'");
            }
            if ((!args || args.organization === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            resourceInputs["billingAccount"] = args ? args.billingAccount : undefined;
            resourceInputs["complianceRegime"] = args ? args.complianceRegime : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["enableSovereignControls"] = args ? args.enableSovereignControls : undefined;
            resourceInputs["kmsSettings"] = args ? args.kmsSettings : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["organization"] = args ? args.organization : undefined;
            resourceInputs["partner"] = args ? args.partner : undefined;
            resourceInputs["partnerPermissions"] = args ? args.partnerPermissions : undefined;
            resourceInputs["provisionedResourcesParent"] = args ? args.provisionedResourcesParent : undefined;
            resourceInputs["resourceSettings"] = args ? args.resourceSettings : undefined;
            resourceInputs["violationNotificationsEnabled"] = args ? args.violationNotificationsEnabled : undefined;
            resourceInputs["complianceStatuses"] = undefined /*out*/;
            resourceInputs["compliantButDisallowedServices"] = undefined /*out*/;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["ekmProvisioningResponses"] = undefined /*out*/;
            resourceInputs["kajEnrollmentState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["resources"] = undefined /*out*/;
            resourceInputs["saaEnrollmentResponses"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Workload.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Workload = Workload;
/** @internal */
Workload.__pulumiType = 'gcp:assuredworkloads/workload:Workload';
