// @pulumi/gcp/containeranalysis/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Occurence = exports.NoteIamPolicy = exports.NoteIamMember = exports.NoteIamBinding = exports.Note = exports.getNoteIamPolicyOutput = exports.getNoteIamPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getNoteIamPolicy = null;
exports.getNoteIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getNoteIamPolicy", "getNoteIamPolicyOutput"], () => require("./getNoteIamPolicy"));
exports.Note = null;
utilities.lazyLoad(exports, ["Note"], () => require("./note"));
exports.NoteIamBinding = null;
utilities.lazyLoad(exports, ["NoteIamBinding"], () => require("./noteIamBinding"));
exports.NoteIamMember = null;
utilities.lazyLoad(exports, ["NoteIamMember"], () => require("./noteIamMember"));
exports.NoteIamPolicy = null;
utilities.lazyLoad(exports, ["NoteIamPolicy"], () => require("./noteIamPolicy"));
exports.Occurence = null;
utilities.lazyLoad(exports, ["Occurence"], () => require("./occurence"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:containeranalysis/note:Note":
                return new exports.Note(name, undefined, { urn });
            case "gcp:containeranalysis/noteIamBinding:NoteIamBinding":
                return new exports.NoteIamBinding(name, undefined, { urn });
            case "gcp:containeranalysis/noteIamMember:NoteIamMember":
                return new exports.NoteIamMember(name, undefined, { urn });
            case "gcp:containeranalysis/noteIamPolicy:NoteIamPolicy":
                return new exports.NoteIamPolicy(name, undefined, { urn });
            case "gcp:containeranalysis/occurence:Occurence":
                return new exports.Occurence(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "containeranalysis/note", _module);
pulumi.runtime.registerResourceModule("gcp", "containeranalysis/noteIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "containeranalysis/noteIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "containeranalysis/noteIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "containeranalysis/occurence", _module);
