// @pulumi/azure-native/types/enums/azurestackhci/v20231101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsServerSubscription = exports.UpdateSummariesPropertiesState = exports.UpdateRunPropertiesState = exports.State = exports.SoftwareAssuranceIntent = exports.ServiceName = exports.ManagedServiceIdentityType = exports.DiagnosticLevel = exports.DeploymentMode = exports.ComplianceAssignmentType = exports.AvailabilityType = void 0;
exports.AvailabilityType = {
    Local: "Local",
    Online: "Online",
    Notify: "Notify",
};
exports.ComplianceAssignmentType = {
    /**
     * Report on the state of the machine, but don't make changes.
     */
    Audit: "Audit",
    /**
     * Applied to the machine. If it drifts, the local service inside the machine makes a correction at the next evaluation.
     */
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
};
exports.DeploymentMode = {
    /**
     * Validate deployment settings for cluster.
     */
    Validate: "Validate",
    /**
     * Deploy cluster using deployment settings.
     */
    Deploy: "Deploy",
};
exports.DiagnosticLevel = {
    Off: "Off",
    Basic: "Basic",
    Enhanced: "Enhanced",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.ServiceName = {
    WAC: "WAC",
};
exports.SoftwareAssuranceIntent = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.State = {
    HasPrerequisite: "HasPrerequisite",
    Obsolete: "Obsolete",
    Ready: "Ready",
    NotApplicableBecauseAnotherUpdateIsInProgress: "NotApplicableBecauseAnotherUpdateIsInProgress",
    Preparing: "Preparing",
    Installing: "Installing",
    Installed: "Installed",
    PreparationFailed: "PreparationFailed",
    InstallationFailed: "InstallationFailed",
    Invalid: "Invalid",
    Recalled: "Recalled",
    Downloading: "Downloading",
    DownloadFailed: "DownloadFailed",
    HealthChecking: "HealthChecking",
    HealthCheckFailed: "HealthCheckFailed",
    ReadyToInstall: "ReadyToInstall",
    ScanInProgress: "ScanInProgress",
    ScanFailed: "ScanFailed",
    AdditionalContentRequired: "AdditionalContentRequired",
};
exports.UpdateRunPropertiesState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    InProgress: "InProgress",
    Failed: "Failed",
};
exports.UpdateSummariesPropertiesState = {
    Unknown: "Unknown",
    AppliedSuccessfully: "AppliedSuccessfully",
    UpdateAvailable: "UpdateAvailable",
    UpdateInProgress: "UpdateInProgress",
    UpdateFailed: "UpdateFailed",
    NeedsAttention: "NeedsAttention",
    PreparationInProgress: "PreparationInProgress",
    PreparationFailed: "PreparationFailed",
};
exports.WindowsServerSubscription = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
