// @pulumi/azure-native/types/enums/elasticsan/v20221201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeCreateOption = exports.StorageTargetType = exports.SkuTier = exports.SkuName = exports.PrivateEndpointServiceConnectionStatus = exports.EncryptionType = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.EncryptionType = {
    /**
     * Volume is encrypted at rest with Platform managed key. It is the default encryption type.
     */
    EncryptionAtRestWithPlatformKey: "EncryptionAtRestWithPlatformKey",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Failed: "Failed",
    Rejected: "Rejected",
};
exports.SkuName = {
    /**
     * Premium locally redundant storage
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Premium zone redundant storage
     */
    Premium_ZRS: "Premium_ZRS",
};
exports.SkuTier = {
    /**
     * Premium Tier
     */
    Premium: "Premium",
};
exports.StorageTargetType = {
    Iscsi: "Iscsi",
    None: "None",
};
exports.VolumeCreateOption = {
    None: "None",
};
