// @pulumi/azure-native/synapse/sqlPoolSensitivityLabel.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPoolSensitivityLabel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A sensitivity label.
 *
 * Uses Azure REST API version 2021-06-01. In version 1.x of the Azure Native provider, it used API version 2021-03-01.
 *
 * Other available API versions: 2021-06-01-preview.
 */
class SqlPoolSensitivityLabel extends pulumi.CustomResource {
    /**
     * Get an existing SqlPoolSensitivityLabel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlPoolSensitivityLabel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlPoolSensitivityLabel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlPoolSensitivityLabel.__pulumiType;
    }
    /**
     * Create a SqlPoolSensitivityLabel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.columnName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'columnName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.schemaName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schemaName'");
            }
            if ((!args || args.sqlPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sqlPoolName'");
            }
            if ((!args || args.tableName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tableName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["columnName"] = args ? args.columnName : undefined;
            resourceInputs["informationType"] = args ? args.informationType : undefined;
            resourceInputs["informationTypeId"] = args ? args.informationTypeId : undefined;
            resourceInputs["labelId"] = args ? args.labelId : undefined;
            resourceInputs["labelName"] = args ? args.labelName : undefined;
            resourceInputs["rank"] = args ? args.rank : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["schemaName"] = args ? args.schemaName : undefined;
            resourceInputs["sensitivityLabelSource"] = args ? args.sensitivityLabelSource : undefined;
            resourceInputs["sqlPoolName"] = args ? args.sqlPoolName : undefined;
            resourceInputs["tableName"] = args ? args.tableName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["isDisabled"] = undefined /*out*/;
            resourceInputs["managedBy"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["columnName"] = undefined /*out*/;
            resourceInputs["informationType"] = undefined /*out*/;
            resourceInputs["informationTypeId"] = undefined /*out*/;
            resourceInputs["isDisabled"] = undefined /*out*/;
            resourceInputs["labelId"] = undefined /*out*/;
            resourceInputs["labelName"] = undefined /*out*/;
            resourceInputs["managedBy"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rank"] = undefined /*out*/;
            resourceInputs["schemaName"] = undefined /*out*/;
            resourceInputs["tableName"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:synapse/v20190601preview:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20201201:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20210301:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20210401preview:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20210501:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20210601:SqlPoolSensitivityLabel" }, { type: "azure-native:synapse/v20210601preview:SqlPoolSensitivityLabel" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlPoolSensitivityLabel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlPoolSensitivityLabel = SqlPoolSensitivityLabel;
/** @internal */
SqlPoolSensitivityLabel.__pulumiType = 'azure-native:synapse:SqlPoolSensitivityLabel';
