// @pulumi/azure-native/types/enums/notificationhubs/v20231001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancyPreference = exports.SkuName = exports.ReplicationRegion = exports.PublicNetworkAccess = exports.PrivateLinkConnectionStatus = exports.PrivateEndpointConnectionProvisioningState = exports.OperationProvisioningState = exports.NamespaceType = exports.NamespaceStatus = exports.AccessRights = void 0;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.NamespaceStatus = {
    Created: "Created",
    Creating: "Creating",
    Suspended: "Suspended",
    Deleting: "Deleting",
};
exports.NamespaceType = {
    Messaging: "Messaging",
    NotificationHub: "NotificationHub",
};
exports.OperationProvisioningState = {
    Unknown: "Unknown",
    InProgress: "InProgress",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Pending: "Pending",
    Disabled: "Disabled",
};
exports.PrivateEndpointConnectionProvisioningState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    Creating: "Creating",
    Updating: "Updating",
    UpdatingByProxy: "UpdatingByProxy",
    Deleting: "Deleting",
    DeletingByProxy: "DeletingByProxy",
    Deleted: "Deleted",
};
exports.PrivateLinkConnectionStatus = {
    Disconnected: "Disconnected",
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReplicationRegion = {
    Default: "Default",
    WestUs2: "WestUs2",
    NorthEurope: "NorthEurope",
    AustraliaEast: "AustraliaEast",
    BrazilSouth: "BrazilSouth",
    SouthEastAsia: "SouthEastAsia",
    SouthAfricaNorth: "SouthAfricaNorth",
    None: "None",
};
exports.SkuName = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
};
exports.ZoneRedundancyPreference = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
