// @pulumi/azure-native/azuredata/v20190724preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlServerRegistration = exports.SqlServer = exports.getSqlServerRegistrationOutput = exports.getSqlServerRegistration = exports.getSqlServerOutput = exports.getSqlServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getSqlServer = null;
exports.getSqlServerOutput = null;
utilities.lazyLoad(exports, ["getSqlServer", "getSqlServerOutput"], () => require("./getSqlServer"));
exports.getSqlServerRegistration = null;
exports.getSqlServerRegistrationOutput = null;
utilities.lazyLoad(exports, ["getSqlServerRegistration", "getSqlServerRegistrationOutput"], () => require("./getSqlServerRegistration"));
exports.SqlServer = null;
utilities.lazyLoad(exports, ["SqlServer"], () => require("./sqlServer"));
exports.SqlServerRegistration = null;
utilities.lazyLoad(exports, ["SqlServerRegistration"], () => require("./sqlServerRegistration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azuredata/v20190724preview:SqlServer":
                return new exports.SqlServer(name, undefined, { urn });
            case "azure-native:azuredata/v20190724preview:SqlServerRegistration":
                return new exports.SqlServerRegistration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azuredata/v20190724preview", _module);
