// @pulumi/azure-native/types/enums/cognitiveservices/v20240601preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.RoutingMethods = exports.ResourceIdentityType = exports.RaiPolicyMode = exports.RaiPolicyContentSource = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NetworkRuleAction = exports.KeySource = exports.HostingModel = exports.EncryptionScopeState = exports.DeploymentScaleType = exports.DeploymentModelVersionUpgradeOption = exports.ContentLevel = exports.ByPassSelection = void 0;
exports.ByPassSelection = {
    None: "None",
    AzureServices: "AzureServices",
};
exports.ContentLevel = {
    Low: "Low",
    Medium: "Medium",
    High: "High",
};
exports.DeploymentModelVersionUpgradeOption = {
    OnceNewDefaultVersionAvailable: "OnceNewDefaultVersionAvailable",
    OnceCurrentVersionExpired: "OnceCurrentVersionExpired",
    NoAutoUpgrade: "NoAutoUpgrade",
};
exports.DeploymentScaleType = {
    Standard: "Standard",
    Manual: "Manual",
};
exports.EncryptionScopeState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.HostingModel = {
    Web: "Web",
    ConnectedContainer: "ConnectedContainer",
    DisconnectedContainer: "DisconnectedContainer",
    ProvisionedWeb: "ProvisionedWeb",
};
exports.KeySource = {
    Microsoft_CognitiveServices: "Microsoft.CognitiveServices",
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.NetworkRuleAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RaiPolicyContentSource = {
    Prompt: "Prompt",
    Completion: "Completion",
};
exports.RaiPolicyMode = {
    Default: "Default",
    Deferred: "Deferred",
    Blocking: "Blocking",
    Asynchronous_filter: "Asynchronous_filter",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.RoutingMethods = {
    Priority: "Priority",
    Weighted: "Weighted",
    Performance: "Performance",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    Enterprise: "Enterprise",
};
