// @pulumi/azure-native/databox/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250201 = exports.v20240301preview = exports.v20240201preview = exports.v20231201 = exports.v20230301 = exports.v20221201 = exports.listJobCredentialsOutput = exports.listJobCredentials = exports.Job = exports.getJobOutput = exports.getJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.listJobCredentials = null;
exports.listJobCredentialsOutput = null;
utilities.lazyLoad(exports, ["listJobCredentials", "listJobCredentialsOutput"], () => require("./listJobCredentials"));
// Export enums:
__exportStar(require("../types/enums/databox"), exports);
// Export sub-modules:
const v20221201 = require("./v20221201");
exports.v20221201 = v20221201;
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240301preview = require("./v20240301preview");
exports.v20240301preview = v20240301preview;
const v20250201 = require("./v20250201");
exports.v20250201 = v20250201;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:databox:Job":
                return new exports.Job(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "databox", _module);
