// @pulumi/azure-native/labservices/v20181015/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.listGlobalUserLabsOutput = exports.listGlobalUserLabs = exports.listGlobalUserEnvironmentsOutput = exports.listGlobalUserEnvironments = exports.LabAccount = exports.Lab = exports.getUserOutput = exports.getUser = exports.getLabAccountRegionalAvailabilityOutput = exports.getLabAccountRegionalAvailability = exports.getLabAccountOutput = exports.getLabAccount = exports.getLabOutput = exports.getLab = exports.getGlobalUserPersonalPreferencesOutput = exports.getGlobalUserPersonalPreferences = exports.getGlobalUserOperationStatusOutput = exports.getGlobalUserOperationStatus = exports.getGlobalUserOperationBatchStatusOutput = exports.getGlobalUserOperationBatchStatus = exports.getGlobalUserEnvironmentOutput = exports.getGlobalUserEnvironment = exports.getGalleryImageOutput = exports.getGalleryImage = exports.getEnvironmentSettingOutput = exports.getEnvironmentSetting = exports.getEnvironmentOutput = exports.getEnvironment = exports.GalleryImage = exports.EnvironmentSetting = exports.Environment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.EnvironmentSetting = null;
utilities.lazyLoad(exports, ["EnvironmentSetting"], () => require("./environmentSetting"));
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getEnvironmentSetting = null;
exports.getEnvironmentSettingOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentSetting", "getEnvironmentSettingOutput"], () => require("./getEnvironmentSetting"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGlobalUserEnvironment = null;
exports.getGlobalUserEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getGlobalUserEnvironment", "getGlobalUserEnvironmentOutput"], () => require("./getGlobalUserEnvironment"));
exports.getGlobalUserOperationBatchStatus = null;
exports.getGlobalUserOperationBatchStatusOutput = null;
utilities.lazyLoad(exports, ["getGlobalUserOperationBatchStatus", "getGlobalUserOperationBatchStatusOutput"], () => require("./getGlobalUserOperationBatchStatus"));
exports.getGlobalUserOperationStatus = null;
exports.getGlobalUserOperationStatusOutput = null;
utilities.lazyLoad(exports, ["getGlobalUserOperationStatus", "getGlobalUserOperationStatusOutput"], () => require("./getGlobalUserOperationStatus"));
exports.getGlobalUserPersonalPreferences = null;
exports.getGlobalUserPersonalPreferencesOutput = null;
utilities.lazyLoad(exports, ["getGlobalUserPersonalPreferences", "getGlobalUserPersonalPreferencesOutput"], () => require("./getGlobalUserPersonalPreferences"));
exports.getLab = null;
exports.getLabOutput = null;
utilities.lazyLoad(exports, ["getLab", "getLabOutput"], () => require("./getLab"));
exports.getLabAccount = null;
exports.getLabAccountOutput = null;
utilities.lazyLoad(exports, ["getLabAccount", "getLabAccountOutput"], () => require("./getLabAccount"));
exports.getLabAccountRegionalAvailability = null;
exports.getLabAccountRegionalAvailabilityOutput = null;
utilities.lazyLoad(exports, ["getLabAccountRegionalAvailability", "getLabAccountRegionalAvailabilityOutput"], () => require("./getLabAccountRegionalAvailability"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.Lab = null;
utilities.lazyLoad(exports, ["Lab"], () => require("./lab"));
exports.LabAccount = null;
utilities.lazyLoad(exports, ["LabAccount"], () => require("./labAccount"));
exports.listGlobalUserEnvironments = null;
exports.listGlobalUserEnvironmentsOutput = null;
utilities.lazyLoad(exports, ["listGlobalUserEnvironments", "listGlobalUserEnvironmentsOutput"], () => require("./listGlobalUserEnvironments"));
exports.listGlobalUserLabs = null;
exports.listGlobalUserLabsOutput = null;
utilities.lazyLoad(exports, ["listGlobalUserLabs", "listGlobalUserLabsOutput"], () => require("./listGlobalUserLabs"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../../types/enums/labservices/v20181015"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:labservices/v20181015:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "azure-native:labservices/v20181015:EnvironmentSetting":
                return new exports.EnvironmentSetting(name, undefined, { urn });
            case "azure-native:labservices/v20181015:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:labservices/v20181015:Lab":
                return new exports.Lab(name, undefined, { urn });
            case "azure-native:labservices/v20181015:LabAccount":
                return new exports.LabAccount(name, undefined, { urn });
            case "azure-native:labservices/v20181015:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "labservices/v20181015", _module);
