// @pulumi/azure-native/synapse/v20210501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.SqlPool = exports.getWorkspaceOutput = exports.getWorkspace = exports.getSqlPoolOutput = exports.getSqlPool = exports.getBigDataPoolOutput = exports.getBigDataPool = exports.BigDataPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BigDataPool = null;
utilities.lazyLoad(exports, ["BigDataPool"], () => require("./bigDataPool"));
exports.getBigDataPool = null;
exports.getBigDataPoolOutput = null;
utilities.lazyLoad(exports, ["getBigDataPool", "getBigDataPoolOutput"], () => require("./getBigDataPool"));
exports.getSqlPool = null;
exports.getSqlPoolOutput = null;
utilities.lazyLoad(exports, ["getSqlPool", "getSqlPoolOutput"], () => require("./getSqlPool"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.SqlPool = null;
utilities.lazyLoad(exports, ["SqlPool"], () => require("./sqlPool"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../../types/enums/synapse/v20210501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:synapse/v20210501:BigDataPool":
                return new exports.BigDataPool(name, undefined, { urn });
            case "azure-native:synapse/v20210501:SqlPool":
                return new exports.SqlPool(name, undefined, { urn });
            case "azure-native:synapse/v20210501:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "synapse/v20210501", _module);
