// @pulumi/azure-native/types/enums/costmanagement/v20221005preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingsKind = exports.ReportType = exports.ReportTimeframeType = exports.ReportGranularityType = exports.ReportConfigSortingType = exports.ReportConfigColumnType = exports.PivotTypeType = exports.OperatorType = exports.MetricType = exports.KpiTypeType = exports.FunctionType = exports.ChartType = exports.AccumulatedType = void 0;
exports.AccumulatedType = {
    True: "true",
    False: "false",
};
exports.ChartType = {
    Area: "Area",
    Line: "Line",
    StackedColumn: "StackedColumn",
    GroupedColumn: "GroupedColumn",
    Table: "Table",
};
exports.FunctionType = {
    Sum: "Sum",
};
exports.KpiTypeType = {
    Forecast: "Forecast",
    Budget: "Budget",
};
exports.MetricType = {
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
    AHUB: "AHUB",
};
exports.OperatorType = {
    In: "In",
    Contains: "Contains",
};
exports.PivotTypeType = {
    Dimension: "Dimension",
    TagKey: "TagKey",
};
exports.ReportConfigColumnType = {
    Tag: "Tag",
    Dimension: "Dimension",
};
exports.ReportConfigSortingType = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.ReportGranularityType = {
    Daily: "Daily",
    Monthly: "Monthly",
};
exports.ReportTimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    YearToDate: "YearToDate",
    Custom: "Custom",
};
exports.ReportType = {
    Usage: "Usage",
};
exports.SettingsKind = {
    Taginheritance: "taginheritance",
};
