// @pulumi/azure-native/network/v20240101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifierWorkspace = exports.StaticCidr = exports.SecurityAdminConfiguration = exports.ReachabilityAnalysisRun = exports.ReachabilityAnalysisIntent = exports.NetworkManager = exports.listNetworkManagerDeploymentStatusOutput = exports.listNetworkManagerDeploymentStatus = exports.listIpamPoolAssociatedResourcesOutput = exports.listIpamPoolAssociatedResources = exports.IpamPool = exports.getVerifierWorkspaceOutput = exports.getVerifierWorkspace = exports.getStaticCidrOutput = exports.getStaticCidr = exports.getSecurityAdminConfigurationOutput = exports.getSecurityAdminConfiguration = exports.getReachabilityAnalysisRunOutput = exports.getReachabilityAnalysisRun = exports.getReachabilityAnalysisIntentOutput = exports.getReachabilityAnalysisIntent = exports.getNetworkManagerOutput = exports.getNetworkManager = exports.getIpamPoolUsageOutput = exports.getIpamPoolUsage = exports.getIpamPoolOutput = exports.getIpamPool = exports.getDefaultAdminRuleOutput = exports.getDefaultAdminRule = exports.getAdminRuleCollectionOutput = exports.getAdminRuleCollection = exports.getAdminRuleOutput = exports.getAdminRule = exports.DefaultAdminRule = exports.AdminRuleCollection = exports.AdminRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AdminRule = null;
utilities.lazyLoad(exports, ["AdminRule"], () => require("./adminRule"));
exports.AdminRuleCollection = null;
utilities.lazyLoad(exports, ["AdminRuleCollection"], () => require("./adminRuleCollection"));
exports.DefaultAdminRule = null;
utilities.lazyLoad(exports, ["DefaultAdminRule"], () => require("./defaultAdminRule"));
exports.getAdminRule = null;
exports.getAdminRuleOutput = null;
utilities.lazyLoad(exports, ["getAdminRule", "getAdminRuleOutput"], () => require("./getAdminRule"));
exports.getAdminRuleCollection = null;
exports.getAdminRuleCollectionOutput = null;
utilities.lazyLoad(exports, ["getAdminRuleCollection", "getAdminRuleCollectionOutput"], () => require("./getAdminRuleCollection"));
exports.getDefaultAdminRule = null;
exports.getDefaultAdminRuleOutput = null;
utilities.lazyLoad(exports, ["getDefaultAdminRule", "getDefaultAdminRuleOutput"], () => require("./getDefaultAdminRule"));
exports.getIpamPool = null;
exports.getIpamPoolOutput = null;
utilities.lazyLoad(exports, ["getIpamPool", "getIpamPoolOutput"], () => require("./getIpamPool"));
exports.getIpamPoolUsage = null;
exports.getIpamPoolUsageOutput = null;
utilities.lazyLoad(exports, ["getIpamPoolUsage", "getIpamPoolUsageOutput"], () => require("./getIpamPoolUsage"));
exports.getNetworkManager = null;
exports.getNetworkManagerOutput = null;
utilities.lazyLoad(exports, ["getNetworkManager", "getNetworkManagerOutput"], () => require("./getNetworkManager"));
exports.getReachabilityAnalysisIntent = null;
exports.getReachabilityAnalysisIntentOutput = null;
utilities.lazyLoad(exports, ["getReachabilityAnalysisIntent", "getReachabilityAnalysisIntentOutput"], () => require("./getReachabilityAnalysisIntent"));
exports.getReachabilityAnalysisRun = null;
exports.getReachabilityAnalysisRunOutput = null;
utilities.lazyLoad(exports, ["getReachabilityAnalysisRun", "getReachabilityAnalysisRunOutput"], () => require("./getReachabilityAnalysisRun"));
exports.getSecurityAdminConfiguration = null;
exports.getSecurityAdminConfigurationOutput = null;
utilities.lazyLoad(exports, ["getSecurityAdminConfiguration", "getSecurityAdminConfigurationOutput"], () => require("./getSecurityAdminConfiguration"));
exports.getStaticCidr = null;
exports.getStaticCidrOutput = null;
utilities.lazyLoad(exports, ["getStaticCidr", "getStaticCidrOutput"], () => require("./getStaticCidr"));
exports.getVerifierWorkspace = null;
exports.getVerifierWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getVerifierWorkspace", "getVerifierWorkspaceOutput"], () => require("./getVerifierWorkspace"));
exports.IpamPool = null;
utilities.lazyLoad(exports, ["IpamPool"], () => require("./ipamPool"));
exports.listIpamPoolAssociatedResources = null;
exports.listIpamPoolAssociatedResourcesOutput = null;
utilities.lazyLoad(exports, ["listIpamPoolAssociatedResources", "listIpamPoolAssociatedResourcesOutput"], () => require("./listIpamPoolAssociatedResources"));
exports.listNetworkManagerDeploymentStatus = null;
exports.listNetworkManagerDeploymentStatusOutput = null;
utilities.lazyLoad(exports, ["listNetworkManagerDeploymentStatus", "listNetworkManagerDeploymentStatusOutput"], () => require("./listNetworkManagerDeploymentStatus"));
exports.NetworkManager = null;
utilities.lazyLoad(exports, ["NetworkManager"], () => require("./networkManager"));
exports.ReachabilityAnalysisIntent = null;
utilities.lazyLoad(exports, ["ReachabilityAnalysisIntent"], () => require("./reachabilityAnalysisIntent"));
exports.ReachabilityAnalysisRun = null;
utilities.lazyLoad(exports, ["ReachabilityAnalysisRun"], () => require("./reachabilityAnalysisRun"));
exports.SecurityAdminConfiguration = null;
utilities.lazyLoad(exports, ["SecurityAdminConfiguration"], () => require("./securityAdminConfiguration"));
exports.StaticCidr = null;
utilities.lazyLoad(exports, ["StaticCidr"], () => require("./staticCidr"));
exports.VerifierWorkspace = null;
utilities.lazyLoad(exports, ["VerifierWorkspace"], () => require("./verifierWorkspace"));
// Export enums:
__exportStar(require("../../types/enums/network/v20240101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20240101preview:AdminRule":
                return new exports.AdminRule(name, undefined, { urn });
            case "azure-native:network/v20240101preview:AdminRuleCollection":
                return new exports.AdminRuleCollection(name, undefined, { urn });
            case "azure-native:network/v20240101preview:DefaultAdminRule":
                return new exports.DefaultAdminRule(name, undefined, { urn });
            case "azure-native:network/v20240101preview:IpamPool":
                return new exports.IpamPool(name, undefined, { urn });
            case "azure-native:network/v20240101preview:NetworkManager":
                return new exports.NetworkManager(name, undefined, { urn });
            case "azure-native:network/v20240101preview:ReachabilityAnalysisIntent":
                return new exports.ReachabilityAnalysisIntent(name, undefined, { urn });
            case "azure-native:network/v20240101preview:ReachabilityAnalysisRun":
                return new exports.ReachabilityAnalysisRun(name, undefined, { urn });
            case "azure-native:network/v20240101preview:SecurityAdminConfiguration":
                return new exports.SecurityAdminConfiguration(name, undefined, { urn });
            case "azure-native:network/v20240101preview:StaticCidr":
                return new exports.StaticCidr(name, undefined, { urn });
            case "azure-native:network/v20240101preview:VerifierWorkspace":
                return new exports.VerifierWorkspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20240101preview", _module);
