// @pulumi/azure-native/netapp/v20211001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeGroup = exports.Volume = exports.getVolumeGroupOutput = exports.getVolumeGroup = exports.getVolumeOutput = exports.getVolume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.getVolumeGroup = null;
exports.getVolumeGroupOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroup", "getVolumeGroupOutput"], () => require("./getVolumeGroup"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
exports.VolumeGroup = null;
utilities.lazyLoad(exports, ["VolumeGroup"], () => require("./volumeGroup"));
// Export enums:
__exportStar(require("../../types/enums/netapp/v20211001"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:netapp/v20211001:Volume":
                return new exports.Volume(name, undefined, { urn });
            case "azure-native:netapp/v20211001:VolumeGroup":
                return new exports.VolumeGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "netapp/v20211001", _module);
