// @pulumi/azure-native/azuresphere/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240401 = exports.v20220901preview = exports.Product = exports.listCatalogDevicesOutput = exports.listCatalogDevices = exports.listCatalogDeviceInsightsOutput = exports.listCatalogDeviceInsights = exports.listCatalogDeviceGroupsOutput = exports.listCatalogDeviceGroups = exports.listCatalogDeploymentsOutput = exports.listCatalogDeployments = exports.Image = exports.getProductOutput = exports.getProduct = exports.getImageOutput = exports.getImage = exports.getDeviceGroupOutput = exports.getDeviceGroup = exports.getDeviceOutput = exports.getDevice = exports.getDeploymentOutput = exports.getDeployment = exports.getCatalogOutput = exports.getCatalog = exports.DeviceGroup = exports.Device = exports.Deployment = exports.Catalog = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Catalog = null;
utilities.lazyLoad(exports, ["Catalog"], () => require("./catalog"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.Device = null;
utilities.lazyLoad(exports, ["Device"], () => require("./device"));
exports.DeviceGroup = null;
utilities.lazyLoad(exports, ["DeviceGroup"], () => require("./deviceGroup"));
exports.getCatalog = null;
exports.getCatalogOutput = null;
utilities.lazyLoad(exports, ["getCatalog", "getCatalogOutput"], () => require("./getCatalog"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getDevice = null;
exports.getDeviceOutput = null;
utilities.lazyLoad(exports, ["getDevice", "getDeviceOutput"], () => require("./getDevice"));
exports.getDeviceGroup = null;
exports.getDeviceGroupOutput = null;
utilities.lazyLoad(exports, ["getDeviceGroup", "getDeviceGroupOutput"], () => require("./getDeviceGroup"));
exports.getImage = null;
exports.getImageOutput = null;
utilities.lazyLoad(exports, ["getImage", "getImageOutput"], () => require("./getImage"));
exports.getProduct = null;
exports.getProductOutput = null;
utilities.lazyLoad(exports, ["getProduct", "getProductOutput"], () => require("./getProduct"));
exports.Image = null;
utilities.lazyLoad(exports, ["Image"], () => require("./image"));
exports.listCatalogDeployments = null;
exports.listCatalogDeploymentsOutput = null;
utilities.lazyLoad(exports, ["listCatalogDeployments", "listCatalogDeploymentsOutput"], () => require("./listCatalogDeployments"));
exports.listCatalogDeviceGroups = null;
exports.listCatalogDeviceGroupsOutput = null;
utilities.lazyLoad(exports, ["listCatalogDeviceGroups", "listCatalogDeviceGroupsOutput"], () => require("./listCatalogDeviceGroups"));
exports.listCatalogDeviceInsights = null;
exports.listCatalogDeviceInsightsOutput = null;
utilities.lazyLoad(exports, ["listCatalogDeviceInsights", "listCatalogDeviceInsightsOutput"], () => require("./listCatalogDeviceInsights"));
exports.listCatalogDevices = null;
exports.listCatalogDevicesOutput = null;
utilities.lazyLoad(exports, ["listCatalogDevices", "listCatalogDevicesOutput"], () => require("./listCatalogDevices"));
exports.Product = null;
utilities.lazyLoad(exports, ["Product"], () => require("./product"));
// Export enums:
__exportStar(require("../types/enums/azuresphere"), exports);
// Export sub-modules:
const v20220901preview = require("./v20220901preview");
exports.v20220901preview = v20220901preview;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azuresphere:Catalog":
                return new exports.Catalog(name, undefined, { urn });
            case "azure-native:azuresphere:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:azuresphere:Device":
                return new exports.Device(name, undefined, { urn });
            case "azure-native:azuresphere:DeviceGroup":
                return new exports.DeviceGroup(name, undefined, { urn });
            case "azure-native:azuresphere:Image":
                return new exports.Image(name, undefined, { urn });
            case "azure-native:azuresphere:Product":
                return new exports.Product(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azuresphere", _module);
