// @pulumi/azure-native/managementpartner/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20180201 = exports.Partner = exports.getPartnerOutput = exports.getPartner = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getPartner = null;
exports.getPartnerOutput = null;
utilities.lazyLoad(exports, ["getPartner", "getPartnerOutput"], () => require("./getPartner"));
exports.Partner = null;
utilities.lazyLoad(exports, ["Partner"], () => require("./partner"));
// Export sub-modules:
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managementpartner:Partner":
                return new exports.Partner(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managementpartner", _module);
