// @pulumi/azure-native/iotoperationsmq/dataLakeConnectorTopicMap.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataLakeConnectorTopicMap = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * MQ dataLakeConnector/topicMap resource
 *
 * Uses Azure REST API version 2023-10-04-preview.
 */
class DataLakeConnectorTopicMap extends pulumi.CustomResource {
    /**
     * Get an existing DataLakeConnectorTopicMap resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataLakeConnectorTopicMap(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataLakeConnectorTopicMap.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataLakeConnectorTopicMap.__pulumiType;
    }
    /**
     * Create a DataLakeConnectorTopicMap resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataLakeConnectorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeConnectorName'");
            }
            if ((!args || args.dataLakeConnectorRef === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeConnectorRef'");
            }
            if ((!args || args.extendedLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if ((!args || args.mapping === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mapping'");
            }
            if ((!args || args.mqName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mqName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["dataLakeConnectorName"] = args ? args.dataLakeConnectorName : undefined;
            resourceInputs["dataLakeConnectorRef"] = args ? args.dataLakeConnectorRef : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mapping"] = args ? (args.mapping ? pulumi.output(args.mapping).apply(inputs.iotoperationsmq.dataLakeConnectorMapArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["mqName"] = args ? args.mqName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["topicMapName"] = args ? args.topicMapName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataLakeConnectorRef"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["mapping"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:iotoperationsmq/v20231004preview:DataLakeConnectorTopicMap" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DataLakeConnectorTopicMap.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataLakeConnectorTopicMap = DataLakeConnectorTopicMap;
/** @internal */
DataLakeConnectorTopicMap.__pulumiType = 'azure-native:iotoperationsmq:DataLakeConnectorTopicMap';
