// @pulumi/azure-native/azurestackhci/v20250401preview/virtualHardDisk.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualHardDisk = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The virtual hard disk resource definition.
 */
class VirtualHardDisk extends pulumi.CustomResource {
    /**
     * Get an existing VirtualHardDisk resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualHardDisk(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualHardDisk.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualHardDisk.__pulumiType;
    }
    /**
     * Create a VirtualHardDisk resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["blockSizeBytes"] = args ? args.blockSizeBytes : undefined;
            resourceInputs["containerId"] = args ? args.containerId : undefined;
            resourceInputs["createFromLocal"] = (_a = (args ? args.createFromLocal : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["diskFileFormat"] = args ? args.diskFileFormat : undefined;
            resourceInputs["diskSizeGB"] = args ? args.diskSizeGB : undefined;
            resourceInputs["downloadUrl"] = args ? args.downloadUrl : undefined;
            resourceInputs["dynamic"] = args ? args.dynamic : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["hyperVGeneration"] = args ? args.hyperVGeneration : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["logicalSectorBytes"] = args ? args.logicalSectorBytes : undefined;
            resourceInputs["physicalSectorBytes"] = args ? args.physicalSectorBytes : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualHardDiskName"] = args ? args.virtualHardDiskName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["blockSizeBytes"] = undefined /*out*/;
            resourceInputs["containerId"] = undefined /*out*/;
            resourceInputs["createFromLocal"] = undefined /*out*/;
            resourceInputs["diskFileFormat"] = undefined /*out*/;
            resourceInputs["diskSizeGB"] = undefined /*out*/;
            resourceInputs["downloadUrl"] = undefined /*out*/;
            resourceInputs["dynamic"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["hyperVGeneration"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["logicalSectorBytes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["physicalSectorBytes"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20210701preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20210901preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20221215preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20230701preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20230901preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20240101:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20240201preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20240501preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20240715preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20240801preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20241001preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci/v20250201preview:VirtualHardDisk" }, { type: "azure-native:azurestackhci:VirtualHardDisk" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualHardDisk.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualHardDisk = VirtualHardDisk;
/** @internal */
VirtualHardDisk.__pulumiType = 'azure-native:azurestackhci/v20250401preview:VirtualHardDisk';
