// @pulumi/azure-native/types/enums/databasewatcher/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetAuthenticationType = exports.ManagedServiceIdentityType = exports.KustoOfferingType = exports.AlertRuleCreationProperties = exports.v20250102 = exports.v20241001preview = exports.v20240719preview = exports.v20230901preview = void 0;
// Export sub-modules:
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20240719preview = require("./v20240719preview");
exports.v20240719preview = v20240719preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const v20250102 = require("./v20250102");
exports.v20250102 = v20250102;
exports.AlertRuleCreationProperties = {
    /**
     * The alert rule was created with an action group.
     */
    CreatedWithActionGroup: "CreatedWithActionGroup",
    /**
     * The alert rule was created with no properties.
     */
    None: "None",
};
exports.KustoOfferingType = {
    /**
     * The Azure Data Explorer cluster Kusto offering.
     */
    Adx: "adx",
    /**
     * The free Azure Data Explorer cluster Kusto offering.
     */
    Free: "free",
    /**
     * The Fabric Real-Time Analytics Kusto offering.
     */
    Fabric: "fabric",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.TargetAuthenticationType = {
    /**
     * The Azure Active Directory authentication.
     */
    Aad: "Aad",
    /**
     * The SQL password authentication.
     */
    Sql: "Sql",
};
