// @pulumi/gcp/redis/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Instance = exports.getInstanceOutput = exports.getInstance = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:redis/cluster:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "gcp:redis/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "redis/cluster", _module);
pulumi.runtime.registerResourceModule("gcp", "redis/instance", _module);
