// @pulumi/azure-native/media/v20151001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaService = exports.listMediaServiceKeysOutput = exports.listMediaServiceKeys = exports.getMediaServiceOutput = exports.getMediaService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMediaService = null;
exports.getMediaServiceOutput = null;
utilities.lazyLoad(exports, ["getMediaService", "getMediaServiceOutput"], () => require("./getMediaService"));
exports.listMediaServiceKeys = null;
exports.listMediaServiceKeysOutput = null;
utilities.lazyLoad(exports, ["listMediaServiceKeys", "listMediaServiceKeysOutput"], () => require("./listMediaServiceKeys"));
exports.MediaService = null;
utilities.lazyLoad(exports, ["MediaService"], () => require("./mediaService"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:media/v20151001:MediaService":
                return new exports.MediaService(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "media/v20151001", _module);
