// @pulumi/azure-native/migrate/v20180901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Solution = exports.MigrateProject = exports.getSolutionConfigOutput = exports.getSolutionConfig = exports.getSolutionOutput = exports.getSolution = exports.getMigrateProjectOutput = exports.getMigrateProject = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMigrateProject = null;
exports.getMigrateProjectOutput = null;
utilities.lazyLoad(exports, ["getMigrateProject", "getMigrateProjectOutput"], () => require("./getMigrateProject"));
exports.getSolution = null;
exports.getSolutionOutput = null;
utilities.lazyLoad(exports, ["getSolution", "getSolutionOutput"], () => require("./getSolution"));
exports.getSolutionConfig = null;
exports.getSolutionConfigOutput = null;
utilities.lazyLoad(exports, ["getSolutionConfig", "getSolutionConfigOutput"], () => require("./getSolutionConfig"));
exports.MigrateProject = null;
utilities.lazyLoad(exports, ["MigrateProject"], () => require("./migrateProject"));
exports.Solution = null;
utilities.lazyLoad(exports, ["Solution"], () => require("./solution"));
// Export enums:
__exportStar(require("../../types/enums/migrate/v20180901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate/v20180901preview:MigrateProject":
                return new exports.MigrateProject(name, undefined, { urn });
            case "azure-native:migrate/v20180901preview:Solution":
                return new exports.Solution(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate/v20180901preview", _module);
