// @pulumi/azure-native/insights/v20170501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionDiagnosticSetting = exports.listDiagnosticSettingsCategoryOutput = exports.listDiagnosticSettingsCategory = exports.getSubscriptionDiagnosticSettingOutput = exports.getSubscriptionDiagnosticSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getSubscriptionDiagnosticSetting = null;
exports.getSubscriptionDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionDiagnosticSetting", "getSubscriptionDiagnosticSettingOutput"], () => require("./getSubscriptionDiagnosticSetting"));
exports.listDiagnosticSettingsCategory = null;
exports.listDiagnosticSettingsCategoryOutput = null;
utilities.lazyLoad(exports, ["listDiagnosticSettingsCategory", "listDiagnosticSettingsCategoryOutput"], () => require("./listDiagnosticSettingsCategory"));
exports.SubscriptionDiagnosticSetting = null;
utilities.lazyLoad(exports, ["SubscriptionDiagnosticSetting"], () => require("./subscriptionDiagnosticSetting"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights/v20170501preview:SubscriptionDiagnosticSetting":
                return new exports.SubscriptionDiagnosticSetting(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights/v20170501preview", _module);
