// @pulumi/gcp/vmwareengine/externalAccessRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalAccessRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * External access firewall rules for filtering incoming traffic destined to `ExternalAddress` resources.
 *
 * To get more information about ExternalAccessRule, see:
 *
 * * [API documentation](https://cloud.google.com/vmware-engine/docs/reference/rest/v1/projects.locations.networkPolicies.externalAccessRules)
 *
 * ## Example Usage
 *
 * ### Vmware Engine External Access Rule Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const external_access_rule_nw = new gcp.vmwareengine.Network("external-access-rule-nw", {
 *     name: "sample-nw",
 *     location: "global",
 *     type: "STANDARD",
 *     description: "PC network description.",
 * });
 * const external_access_rule_np = new gcp.vmwareengine.NetworkPolicy("external-access-rule-np", {
 *     location: "us-west1",
 *     name: "sample-np",
 *     edgeServicesCidr: "192.168.30.0/26",
 *     vmwareEngineNetwork: external_access_rule_nw.id,
 * });
 * const vmw_engine_external_access_rule = new gcp.vmwareengine.ExternalAccessRule("vmw-engine-external-access-rule", {
 *     name: "sample-external-access-rule",
 *     parent: external_access_rule_np.id,
 *     priority: 101,
 *     action: "DENY",
 *     ipProtocol: "TCP",
 *     sourceIpRanges: [{
 *         ipAddressRange: "0.0.0.0/0",
 *     }],
 *     sourcePorts: ["80"],
 *     destinationIpRanges: [{
 *         ipAddressRange: "0.0.0.0/0",
 *     }],
 *     destinationPorts: ["433"],
 * });
 * ```
 * ### Vmware Engine External Access Rule Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const external_access_rule_nw = new gcp.vmwareengine.Network("external-access-rule-nw", {
 *     name: "sample-nw",
 *     location: "global",
 *     type: "STANDARD",
 *     description: "PC network description.",
 * });
 * const external_access_rule_pc = new gcp.vmwareengine.PrivateCloud("external-access-rule-pc", {
 *     location: "us-west1-a",
 *     name: "sample-pc",
 *     description: "Sample test PC.",
 *     networkConfig: {
 *         managementCidr: "192.168.50.0/24",
 *         vmwareEngineNetwork: external_access_rule_nw.id,
 *     },
 *     managementCluster: {
 *         clusterId: "sample-mgmt-cluster",
 *         nodeTypeConfigs: [{
 *             nodeTypeId: "standard-72",
 *             nodeCount: 3,
 *         }],
 *     },
 * });
 * const external_access_rule_np = new gcp.vmwareengine.NetworkPolicy("external-access-rule-np", {
 *     location: "us-west1",
 *     name: "sample-np",
 *     edgeServicesCidr: "192.168.30.0/26",
 *     vmwareEngineNetwork: external_access_rule_nw.id,
 * });
 * const external_access_rule_ea = new gcp.vmwareengine.ExternalAddress("external-access-rule-ea", {
 *     name: "sample-ea",
 *     parent: external_access_rule_pc.id,
 *     internalIp: "192.168.0.65",
 * });
 * const vmw_engine_external_access_rule = new gcp.vmwareengine.ExternalAccessRule("vmw-engine-external-access-rule", {
 *     name: "sample-external-access-rule",
 *     parent: external_access_rule_np.id,
 *     description: "Sample Description",
 *     priority: 101,
 *     action: "ALLOW",
 *     ipProtocol: "tcp",
 *     sourceIpRanges: [{
 *         ipAddressRange: "0.0.0.0/0",
 *     }],
 *     sourcePorts: ["80"],
 *     destinationIpRanges: [{
 *         externalAddress: external_access_rule_ea.id,
 *     }],
 *     destinationPorts: ["433"],
 * });
 * ```
 *
 * ## Import
 *
 * ExternalAccessRule can be imported using any of these accepted formats:
 *
 * * `{{parent}}/externalAccessRules/{{name}}`
 *
 * When using the `pulumi import` command, ExternalAccessRule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:vmwareengine/externalAccessRule:ExternalAccessRule default {{parent}}/externalAccessRules/{{name}}
 * ```
 */
class ExternalAccessRule extends pulumi.CustomResource {
    /**
     * Get an existing ExternalAccessRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ExternalAccessRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExternalAccessRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExternalAccessRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["action"] = state ? state.action : undefined;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["destinationIpRanges"] = state ? state.destinationIpRanges : undefined;
            resourceInputs["destinationPorts"] = state ? state.destinationPorts : undefined;
            resourceInputs["ipProtocol"] = state ? state.ipProtocol : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["parent"] = state ? state.parent : undefined;
            resourceInputs["priority"] = state ? state.priority : undefined;
            resourceInputs["sourceIpRanges"] = state ? state.sourceIpRanges : undefined;
            resourceInputs["sourcePorts"] = state ? state.sourcePorts : undefined;
            resourceInputs["state"] = state ? state.state : undefined;
            resourceInputs["uid"] = state ? state.uid : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.destinationIpRanges === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationIpRanges'");
            }
            if ((!args || args.destinationPorts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationPorts'");
            }
            if ((!args || args.ipProtocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipProtocol'");
            }
            if ((!args || args.parent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'parent'");
            }
            if ((!args || args.priority === undefined) && !opts.urn) {
                throw new Error("Missing required property 'priority'");
            }
            if ((!args || args.sourceIpRanges === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceIpRanges'");
            }
            if ((!args || args.sourcePorts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourcePorts'");
            }
            resourceInputs["action"] = args ? args.action : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["destinationIpRanges"] = args ? args.destinationIpRanges : undefined;
            resourceInputs["destinationPorts"] = args ? args.destinationPorts : undefined;
            resourceInputs["ipProtocol"] = args ? args.ipProtocol : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["parent"] = args ? args.parent : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["sourceIpRanges"] = args ? args.sourceIpRanges : undefined;
            resourceInputs["sourcePorts"] = args ? args.sourcePorts : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ExternalAccessRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExternalAccessRule = ExternalAccessRule;
/** @internal */
ExternalAccessRule.__pulumiType = 'gcp:vmwareengine/externalAccessRule:ExternalAccessRule';
