// @pulumi/azure-native/app/v20250101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listConnectedEnvironmentsDaprComponentSecretsOutput = exports.listConnectedEnvironmentsDaprComponentSecrets = exports.Job = exports.JavaComponent = exports.getManagedEnvironmentsStorageOutput = exports.getManagedEnvironmentsStorage = exports.getManagedEnvironmentAuthTokenOutput = exports.getManagedEnvironmentAuthToken = exports.getManagedEnvironmentOutput = exports.getManagedEnvironment = exports.getManagedCertificateOutput = exports.getManagedCertificate = exports.getJobOutput = exports.getJob = exports.getJavaComponentOutput = exports.getJavaComponent = exports.getDaprComponentOutput = exports.getDaprComponent = exports.getCustomDomainVerificationIdOutput = exports.getCustomDomainVerificationId = exports.getContainerAppsSourceControlOutput = exports.getContainerAppsSourceControl = exports.getContainerAppsSessionPoolOutput = exports.getContainerAppsSessionPool = exports.getContainerAppsAuthConfigOutput = exports.getContainerAppsAuthConfig = exports.getContainerAppAuthTokenOutput = exports.getContainerAppAuthToken = exports.getContainerAppOutput = exports.getContainerApp = exports.getConnectedEnvironmentsStorageOutput = exports.getConnectedEnvironmentsStorage = exports.getConnectedEnvironmentsDaprComponentOutput = exports.getConnectedEnvironmentsDaprComponent = exports.getConnectedEnvironmentsCertificateOutput = exports.getConnectedEnvironmentsCertificate = exports.getConnectedEnvironmentOutput = exports.getConnectedEnvironment = exports.getCertificateOutput = exports.getCertificate = exports.DaprComponent = exports.ContainerAppsSourceControl = exports.ContainerAppsSessionPool = exports.ContainerAppsAuthConfig = exports.ContainerApp = exports.ConnectedEnvironmentsStorage = exports.ConnectedEnvironmentsDaprComponent = exports.ConnectedEnvironmentsCertificate = exports.ConnectedEnvironment = exports.Certificate = void 0;
exports.ManagedEnvironmentsStorage = exports.ManagedEnvironment = exports.ManagedCertificate = exports.listJobSecretsOutput = exports.listJobSecrets = exports.listDaprComponentSecretsOutput = exports.listDaprComponentSecrets = exports.listContainerAppSecretsOutput = exports.listContainerAppSecrets = exports.listContainerAppCustomHostNameAnalysisOutput = exports.listContainerAppCustomHostNameAnalysis = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.ConnectedEnvironment = null;
utilities.lazyLoad(exports, ["ConnectedEnvironment"], () => require("./connectedEnvironment"));
exports.ConnectedEnvironmentsCertificate = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsCertificate"], () => require("./connectedEnvironmentsCertificate"));
exports.ConnectedEnvironmentsDaprComponent = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsDaprComponent"], () => require("./connectedEnvironmentsDaprComponent"));
exports.ConnectedEnvironmentsStorage = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsStorage"], () => require("./connectedEnvironmentsStorage"));
exports.ContainerApp = null;
utilities.lazyLoad(exports, ["ContainerApp"], () => require("./containerApp"));
exports.ContainerAppsAuthConfig = null;
utilities.lazyLoad(exports, ["ContainerAppsAuthConfig"], () => require("./containerAppsAuthConfig"));
exports.ContainerAppsSessionPool = null;
utilities.lazyLoad(exports, ["ContainerAppsSessionPool"], () => require("./containerAppsSessionPool"));
exports.ContainerAppsSourceControl = null;
utilities.lazyLoad(exports, ["ContainerAppsSourceControl"], () => require("./containerAppsSourceControl"));
exports.DaprComponent = null;
utilities.lazyLoad(exports, ["DaprComponent"], () => require("./daprComponent"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getConnectedEnvironment = null;
exports.getConnectedEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironment", "getConnectedEnvironmentOutput"], () => require("./getConnectedEnvironment"));
exports.getConnectedEnvironmentsCertificate = null;
exports.getConnectedEnvironmentsCertificateOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsCertificate", "getConnectedEnvironmentsCertificateOutput"], () => require("./getConnectedEnvironmentsCertificate"));
exports.getConnectedEnvironmentsDaprComponent = null;
exports.getConnectedEnvironmentsDaprComponentOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsDaprComponent", "getConnectedEnvironmentsDaprComponentOutput"], () => require("./getConnectedEnvironmentsDaprComponent"));
exports.getConnectedEnvironmentsStorage = null;
exports.getConnectedEnvironmentsStorageOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsStorage", "getConnectedEnvironmentsStorageOutput"], () => require("./getConnectedEnvironmentsStorage"));
exports.getContainerApp = null;
exports.getContainerAppOutput = null;
utilities.lazyLoad(exports, ["getContainerApp", "getContainerAppOutput"], () => require("./getContainerApp"));
exports.getContainerAppAuthToken = null;
exports.getContainerAppAuthTokenOutput = null;
utilities.lazyLoad(exports, ["getContainerAppAuthToken", "getContainerAppAuthTokenOutput"], () => require("./getContainerAppAuthToken"));
exports.getContainerAppsAuthConfig = null;
exports.getContainerAppsAuthConfigOutput = null;
utilities.lazyLoad(exports, ["getContainerAppsAuthConfig", "getContainerAppsAuthConfigOutput"], () => require("./getContainerAppsAuthConfig"));
exports.getContainerAppsSessionPool = null;
exports.getContainerAppsSessionPoolOutput = null;
utilities.lazyLoad(exports, ["getContainerAppsSessionPool", "getContainerAppsSessionPoolOutput"], () => require("./getContainerAppsSessionPool"));
exports.getContainerAppsSourceControl = null;
exports.getContainerAppsSourceControlOutput = null;
utilities.lazyLoad(exports, ["getContainerAppsSourceControl", "getContainerAppsSourceControlOutput"], () => require("./getContainerAppsSourceControl"));
exports.getCustomDomainVerificationId = null;
exports.getCustomDomainVerificationIdOutput = null;
utilities.lazyLoad(exports, ["getCustomDomainVerificationId", "getCustomDomainVerificationIdOutput"], () => require("./getCustomDomainVerificationId"));
exports.getDaprComponent = null;
exports.getDaprComponentOutput = null;
utilities.lazyLoad(exports, ["getDaprComponent", "getDaprComponentOutput"], () => require("./getDaprComponent"));
exports.getJavaComponent = null;
exports.getJavaComponentOutput = null;
utilities.lazyLoad(exports, ["getJavaComponent", "getJavaComponentOutput"], () => require("./getJavaComponent"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getManagedCertificate = null;
exports.getManagedCertificateOutput = null;
utilities.lazyLoad(exports, ["getManagedCertificate", "getManagedCertificateOutput"], () => require("./getManagedCertificate"));
exports.getManagedEnvironment = null;
exports.getManagedEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironment", "getManagedEnvironmentOutput"], () => require("./getManagedEnvironment"));
exports.getManagedEnvironmentAuthToken = null;
exports.getManagedEnvironmentAuthTokenOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironmentAuthToken", "getManagedEnvironmentAuthTokenOutput"], () => require("./getManagedEnvironmentAuthToken"));
exports.getManagedEnvironmentsStorage = null;
exports.getManagedEnvironmentsStorageOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironmentsStorage", "getManagedEnvironmentsStorageOutput"], () => require("./getManagedEnvironmentsStorage"));
exports.JavaComponent = null;
utilities.lazyLoad(exports, ["JavaComponent"], () => require("./javaComponent"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.listConnectedEnvironmentsDaprComponentSecrets = null;
exports.listConnectedEnvironmentsDaprComponentSecretsOutput = null;
utilities.lazyLoad(exports, ["listConnectedEnvironmentsDaprComponentSecrets", "listConnectedEnvironmentsDaprComponentSecretsOutput"], () => require("./listConnectedEnvironmentsDaprComponentSecrets"));
exports.listContainerAppCustomHostNameAnalysis = null;
exports.listContainerAppCustomHostNameAnalysisOutput = null;
utilities.lazyLoad(exports, ["listContainerAppCustomHostNameAnalysis", "listContainerAppCustomHostNameAnalysisOutput"], () => require("./listContainerAppCustomHostNameAnalysis"));
exports.listContainerAppSecrets = null;
exports.listContainerAppSecretsOutput = null;
utilities.lazyLoad(exports, ["listContainerAppSecrets", "listContainerAppSecretsOutput"], () => require("./listContainerAppSecrets"));
exports.listDaprComponentSecrets = null;
exports.listDaprComponentSecretsOutput = null;
utilities.lazyLoad(exports, ["listDaprComponentSecrets", "listDaprComponentSecretsOutput"], () => require("./listDaprComponentSecrets"));
exports.listJobSecrets = null;
exports.listJobSecretsOutput = null;
utilities.lazyLoad(exports, ["listJobSecrets", "listJobSecretsOutput"], () => require("./listJobSecrets"));
exports.ManagedCertificate = null;
utilities.lazyLoad(exports, ["ManagedCertificate"], () => require("./managedCertificate"));
exports.ManagedEnvironment = null;
utilities.lazyLoad(exports, ["ManagedEnvironment"], () => require("./managedEnvironment"));
exports.ManagedEnvironmentsStorage = null;
utilities.lazyLoad(exports, ["ManagedEnvironmentsStorage"], () => require("./managedEnvironmentsStorage"));
// Export enums:
__exportStar(require("../../types/enums/app/v20250101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:app/v20250101:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:app/v20250101:ConnectedEnvironment":
                return new exports.ConnectedEnvironment(name, undefined, { urn });
            case "azure-native:app/v20250101:ConnectedEnvironmentsCertificate":
                return new exports.ConnectedEnvironmentsCertificate(name, undefined, { urn });
            case "azure-native:app/v20250101:ConnectedEnvironmentsDaprComponent":
                return new exports.ConnectedEnvironmentsDaprComponent(name, undefined, { urn });
            case "azure-native:app/v20250101:ConnectedEnvironmentsStorage":
                return new exports.ConnectedEnvironmentsStorage(name, undefined, { urn });
            case "azure-native:app/v20250101:ContainerApp":
                return new exports.ContainerApp(name, undefined, { urn });
            case "azure-native:app/v20250101:ContainerAppsAuthConfig":
                return new exports.ContainerAppsAuthConfig(name, undefined, { urn });
            case "azure-native:app/v20250101:ContainerAppsSessionPool":
                return new exports.ContainerAppsSessionPool(name, undefined, { urn });
            case "azure-native:app/v20250101:ContainerAppsSourceControl":
                return new exports.ContainerAppsSourceControl(name, undefined, { urn });
            case "azure-native:app/v20250101:DaprComponent":
                return new exports.DaprComponent(name, undefined, { urn });
            case "azure-native:app/v20250101:JavaComponent":
                return new exports.JavaComponent(name, undefined, { urn });
            case "azure-native:app/v20250101:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:app/v20250101:ManagedCertificate":
                return new exports.ManagedCertificate(name, undefined, { urn });
            case "azure-native:app/v20250101:ManagedEnvironment":
                return new exports.ManagedEnvironment(name, undefined, { urn });
            case "azure-native:app/v20250101:ManagedEnvironmentsStorage":
                return new exports.ManagedEnvironmentsStorage(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "app/v20250101", _module);
