// @pulumi/azure-native/types/enums/containerstorage/v20230701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Zone = exports.ReclaimPolicy = exports.ManagedServiceIdentityType = exports.ElasticSanSkuName = exports.AzureDiskSkuName = void 0;
exports.AzureDiskSkuName = {
    /**
     * Premium_LRS SKU
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Standard_LRS SKU
     */
    Standard_LRS: "Standard_LRS",
    /**
     * StandardSSD_LRS SKU
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * UltraSSD_LRS SKU
     */
    UltraSSD_LRS: "UltraSSD_LRS",
    /**
     * Premium_ZRS SKU
     */
    Premium_ZRS: "Premium_ZRS",
    /**
     * PremiumV2_LRS SKU
     */
    PremiumV2_LRS: "PremiumV2_LRS",
    /**
     * StandardSSD_ZRS SKU
     */
    StandardSSD_ZRS: "StandardSSD_ZRS",
};
exports.ElasticSanSkuName = {
    /**
     * Premium_LRS SKU
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Premium_ZRS SKU
     */
    Premium_ZRS: "Premium_ZRS",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.ReclaimPolicy = {
    /**
     * Delete resource
     */
    Delete: "Delete",
    /**
     * Retain resource
     */
    Retain: "Retain",
};
exports.Zone = {
    /**
     * First availability zone.
     */
    Zone1: "1",
    /**
     * Second availability zone.
     */
    Zone2: "2",
    /**
     * Third availability zone.
     */
    Zone3: "3",
};
