// @pulumi/azure-native/types/enums/scvmm/v20230401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningAction = exports.LimitCpuForMigration = exports.InventoryType = exports.IdentityType = exports.DynamicMemoryEnabled = exports.CreateDiffDisk = exports.AllocationMethod = void 0;
exports.AllocationMethod = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.CreateDiffDisk = {
    False: "false",
    True: "true",
};
exports.DynamicMemoryEnabled = {
    False: "false",
    True: "true",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.InventoryType = {
    Cloud: "Cloud",
    VirtualNetwork: "VirtualNetwork",
    VirtualMachineTemplate: "VirtualMachineTemplate",
    VirtualMachine: "VirtualMachine",
};
exports.LimitCpuForMigration = {
    False: "false",
    True: "true",
};
exports.ProvisioningAction = {
    Install: "install",
    Uninstall: "uninstall",
    Repair: "repair",
};
