// @pulumi/azure-native/types/enums/maps/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningKey = exports.PrivateEndpointServiceConnectionStatus = exports.Name = exports.Kind = exports.v20240701preview = exports.v20240101preview = exports.v20231201preview = exports.v20230801preview = exports.v20230601 = exports.v20211201preview = exports.v20210201 = void 0;
// Export sub-modules:
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20230601 = require("./v20230601");
exports.v20230601 = v20230601;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
exports.Kind = {
    Gen1: "Gen1",
    Gen2: "Gen2",
};
exports.Name = {
    S0: "S0",
    S1: "S1",
    G2: "G2",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.SigningKey = {
    PrimaryKey: "primaryKey",
    SecondaryKey: "secondaryKey",
};
