// @pulumi/gcp/clouddeploy/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetIamPolicy = exports.TargetIamMember = exports.TargetIamBinding = exports.Target = exports.getTargetIamPolicyOutput = exports.getTargetIamPolicy = exports.getDeliveryPipelineIamPolicyOutput = exports.getDeliveryPipelineIamPolicy = exports.getCustomTargetTypeIamPolicyOutput = exports.getCustomTargetTypeIamPolicy = exports.DeliveryPipelineIamPolicy = exports.DeliveryPipelineIamMember = exports.DeliveryPipelineIamBinding = exports.DeliveryPipeline = exports.CustomTargetTypeIamPolicy = exports.CustomTargetTypeIamMember = exports.CustomTargetTypeIamBinding = exports.CustomTargetType = exports.Automation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Automation = null;
utilities.lazyLoad(exports, ["Automation"], () => require("./automation"));
exports.CustomTargetType = null;
utilities.lazyLoad(exports, ["CustomTargetType"], () => require("./customTargetType"));
exports.CustomTargetTypeIamBinding = null;
utilities.lazyLoad(exports, ["CustomTargetTypeIamBinding"], () => require("./customTargetTypeIamBinding"));
exports.CustomTargetTypeIamMember = null;
utilities.lazyLoad(exports, ["CustomTargetTypeIamMember"], () => require("./customTargetTypeIamMember"));
exports.CustomTargetTypeIamPolicy = null;
utilities.lazyLoad(exports, ["CustomTargetTypeIamPolicy"], () => require("./customTargetTypeIamPolicy"));
exports.DeliveryPipeline = null;
utilities.lazyLoad(exports, ["DeliveryPipeline"], () => require("./deliveryPipeline"));
exports.DeliveryPipelineIamBinding = null;
utilities.lazyLoad(exports, ["DeliveryPipelineIamBinding"], () => require("./deliveryPipelineIamBinding"));
exports.DeliveryPipelineIamMember = null;
utilities.lazyLoad(exports, ["DeliveryPipelineIamMember"], () => require("./deliveryPipelineIamMember"));
exports.DeliveryPipelineIamPolicy = null;
utilities.lazyLoad(exports, ["DeliveryPipelineIamPolicy"], () => require("./deliveryPipelineIamPolicy"));
exports.getCustomTargetTypeIamPolicy = null;
exports.getCustomTargetTypeIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getCustomTargetTypeIamPolicy", "getCustomTargetTypeIamPolicyOutput"], () => require("./getCustomTargetTypeIamPolicy"));
exports.getDeliveryPipelineIamPolicy = null;
exports.getDeliveryPipelineIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getDeliveryPipelineIamPolicy", "getDeliveryPipelineIamPolicyOutput"], () => require("./getDeliveryPipelineIamPolicy"));
exports.getTargetIamPolicy = null;
exports.getTargetIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getTargetIamPolicy", "getTargetIamPolicyOutput"], () => require("./getTargetIamPolicy"));
exports.Target = null;
utilities.lazyLoad(exports, ["Target"], () => require("./target"));
exports.TargetIamBinding = null;
utilities.lazyLoad(exports, ["TargetIamBinding"], () => require("./targetIamBinding"));
exports.TargetIamMember = null;
utilities.lazyLoad(exports, ["TargetIamMember"], () => require("./targetIamMember"));
exports.TargetIamPolicy = null;
utilities.lazyLoad(exports, ["TargetIamPolicy"], () => require("./targetIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:clouddeploy/automation:Automation":
                return new exports.Automation(name, undefined, { urn });
            case "gcp:clouddeploy/customTargetType:CustomTargetType":
                return new exports.CustomTargetType(name, undefined, { urn });
            case "gcp:clouddeploy/customTargetTypeIamBinding:CustomTargetTypeIamBinding":
                return new exports.CustomTargetTypeIamBinding(name, undefined, { urn });
            case "gcp:clouddeploy/customTargetTypeIamMember:CustomTargetTypeIamMember":
                return new exports.CustomTargetTypeIamMember(name, undefined, { urn });
            case "gcp:clouddeploy/customTargetTypeIamPolicy:CustomTargetTypeIamPolicy":
                return new exports.CustomTargetTypeIamPolicy(name, undefined, { urn });
            case "gcp:clouddeploy/deliveryPipeline:DeliveryPipeline":
                return new exports.DeliveryPipeline(name, undefined, { urn });
            case "gcp:clouddeploy/deliveryPipelineIamBinding:DeliveryPipelineIamBinding":
                return new exports.DeliveryPipelineIamBinding(name, undefined, { urn });
            case "gcp:clouddeploy/deliveryPipelineIamMember:DeliveryPipelineIamMember":
                return new exports.DeliveryPipelineIamMember(name, undefined, { urn });
            case "gcp:clouddeploy/deliveryPipelineIamPolicy:DeliveryPipelineIamPolicy":
                return new exports.DeliveryPipelineIamPolicy(name, undefined, { urn });
            case "gcp:clouddeploy/target:Target":
                return new exports.Target(name, undefined, { urn });
            case "gcp:clouddeploy/targetIamBinding:TargetIamBinding":
                return new exports.TargetIamBinding(name, undefined, { urn });
            case "gcp:clouddeploy/targetIamMember:TargetIamMember":
                return new exports.TargetIamMember(name, undefined, { urn });
            case "gcp:clouddeploy/targetIamPolicy:TargetIamPolicy":
                return new exports.TargetIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/automation", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/customTargetType", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/customTargetTypeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/customTargetTypeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/customTargetTypeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/deliveryPipeline", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/deliveryPipelineIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/deliveryPipelineIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/deliveryPipelineIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/target", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/targetIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/targetIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "clouddeploy/targetIamPolicy", _module);
