// @pulumi/azure-native/containerregistry/v20190501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registry = exports.listRegistryCredentialsOutput = exports.listRegistryCredentials = exports.getRegistryOutput = exports.getRegistry = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getRegistry = null;
exports.getRegistryOutput = null;
utilities.lazyLoad(exports, ["getRegistry", "getRegistryOutput"], () => require("./getRegistry"));
exports.listRegistryCredentials = null;
exports.listRegistryCredentialsOutput = null;
utilities.lazyLoad(exports, ["listRegistryCredentials", "listRegistryCredentialsOutput"], () => require("./listRegistryCredentials"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
// Export enums:
__exportStar(require("../../types/enums/containerregistry/v20190501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerregistry/v20190501:Registry":
                return new exports.Registry(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerregistry/v20190501", _module);
