// @pulumi/azure-native/network/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRuleCollection = exports.UserRule = exports.TrafficManagerUserMetricsKey = exports.SecurityUserConfiguration = exports.Profile = exports.NetworkGroup = exports.listNetworkManagerEffectiveSecurityAdminRulesOutput = exports.listNetworkManagerEffectiveSecurityAdminRules = exports.listNetworkManagerEffectiveConnectivityConfigurationsOutput = exports.listNetworkManagerEffectiveConnectivityConfigurations = exports.listListEffectiveVirtualNetworkByNetworkGroupOutput = exports.listListEffectiveVirtualNetworkByNetworkGroup = exports.listEffectiveVirtualNetworkByNetworkManagerOutput = exports.listEffectiveVirtualNetworkByNetworkManager = exports.listActiveSecurityUserRulesOutput = exports.listActiveSecurityUserRules = exports.getUserRuleCollectionOutput = exports.getUserRuleCollection = exports.getUserRuleOutput = exports.getUserRule = exports.getTrafficManagerUserMetricsKeyOutput = exports.getTrafficManagerUserMetricsKey = exports.getSecurityUserConfigurationOutput = exports.getSecurityUserConfiguration = exports.getProfileOutput = exports.getProfile = exports.getNetworkGroupOutput = exports.getNetworkGroup = exports.getEndpointOutput = exports.getEndpoint = exports.getDefaultUserRuleOutput = exports.getDefaultUserRule = exports.Endpoint = exports.DefaultUserRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DefaultUserRule = null;
utilities.lazyLoad(exports, ["DefaultUserRule"], () => require("./defaultUserRule"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getDefaultUserRule = null;
exports.getDefaultUserRuleOutput = null;
utilities.lazyLoad(exports, ["getDefaultUserRule", "getDefaultUserRuleOutput"], () => require("./getDefaultUserRule"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getNetworkGroup = null;
exports.getNetworkGroupOutput = null;
utilities.lazyLoad(exports, ["getNetworkGroup", "getNetworkGroupOutput"], () => require("./getNetworkGroup"));
exports.getProfile = null;
exports.getProfileOutput = null;
utilities.lazyLoad(exports, ["getProfile", "getProfileOutput"], () => require("./getProfile"));
exports.getSecurityUserConfiguration = null;
exports.getSecurityUserConfigurationOutput = null;
utilities.lazyLoad(exports, ["getSecurityUserConfiguration", "getSecurityUserConfigurationOutput"], () => require("./getSecurityUserConfiguration"));
exports.getTrafficManagerUserMetricsKey = null;
exports.getTrafficManagerUserMetricsKeyOutput = null;
utilities.lazyLoad(exports, ["getTrafficManagerUserMetricsKey", "getTrafficManagerUserMetricsKeyOutput"], () => require("./getTrafficManagerUserMetricsKey"));
exports.getUserRule = null;
exports.getUserRuleOutput = null;
utilities.lazyLoad(exports, ["getUserRule", "getUserRuleOutput"], () => require("./getUserRule"));
exports.getUserRuleCollection = null;
exports.getUserRuleCollectionOutput = null;
utilities.lazyLoad(exports, ["getUserRuleCollection", "getUserRuleCollectionOutput"], () => require("./getUserRuleCollection"));
exports.listActiveSecurityUserRules = null;
exports.listActiveSecurityUserRulesOutput = null;
utilities.lazyLoad(exports, ["listActiveSecurityUserRules", "listActiveSecurityUserRulesOutput"], () => require("./listActiveSecurityUserRules"));
exports.listEffectiveVirtualNetworkByNetworkManager = null;
exports.listEffectiveVirtualNetworkByNetworkManagerOutput = null;
utilities.lazyLoad(exports, ["listEffectiveVirtualNetworkByNetworkManager", "listEffectiveVirtualNetworkByNetworkManagerOutput"], () => require("./listEffectiveVirtualNetworkByNetworkManager"));
exports.listListEffectiveVirtualNetworkByNetworkGroup = null;
exports.listListEffectiveVirtualNetworkByNetworkGroupOutput = null;
utilities.lazyLoad(exports, ["listListEffectiveVirtualNetworkByNetworkGroup", "listListEffectiveVirtualNetworkByNetworkGroupOutput"], () => require("./listListEffectiveVirtualNetworkByNetworkGroup"));
exports.listNetworkManagerEffectiveConnectivityConfigurations = null;
exports.listNetworkManagerEffectiveConnectivityConfigurationsOutput = null;
utilities.lazyLoad(exports, ["listNetworkManagerEffectiveConnectivityConfigurations", "listNetworkManagerEffectiveConnectivityConfigurationsOutput"], () => require("./listNetworkManagerEffectiveConnectivityConfigurations"));
exports.listNetworkManagerEffectiveSecurityAdminRules = null;
exports.listNetworkManagerEffectiveSecurityAdminRulesOutput = null;
utilities.lazyLoad(exports, ["listNetworkManagerEffectiveSecurityAdminRules", "listNetworkManagerEffectiveSecurityAdminRulesOutput"], () => require("./listNetworkManagerEffectiveSecurityAdminRules"));
exports.NetworkGroup = null;
utilities.lazyLoad(exports, ["NetworkGroup"], () => require("./networkGroup"));
exports.Profile = null;
utilities.lazyLoad(exports, ["Profile"], () => require("./profile"));
exports.SecurityUserConfiguration = null;
utilities.lazyLoad(exports, ["SecurityUserConfiguration"], () => require("./securityUserConfiguration"));
exports.TrafficManagerUserMetricsKey = null;
utilities.lazyLoad(exports, ["TrafficManagerUserMetricsKey"], () => require("./trafficManagerUserMetricsKey"));
exports.UserRule = null;
utilities.lazyLoad(exports, ["UserRule"], () => require("./userRule"));
exports.UserRuleCollection = null;
utilities.lazyLoad(exports, ["UserRuleCollection"], () => require("./userRuleCollection"));
// Export enums:
__exportStar(require("../../types/enums/network/v20220401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20220401preview:DefaultUserRule":
                return new exports.DefaultUserRule(name, undefined, { urn });
            case "azure-native:network/v20220401preview:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "azure-native:network/v20220401preview:NetworkGroup":
                return new exports.NetworkGroup(name, undefined, { urn });
            case "azure-native:network/v20220401preview:Profile":
                return new exports.Profile(name, undefined, { urn });
            case "azure-native:network/v20220401preview:SecurityUserConfiguration":
                return new exports.SecurityUserConfiguration(name, undefined, { urn });
            case "azure-native:network/v20220401preview:TrafficManagerUserMetricsKey":
                return new exports.TrafficManagerUserMetricsKey(name, undefined, { urn });
            case "azure-native:network/v20220401preview:UserRule":
                return new exports.UserRule(name, undefined, { urn });
            case "azure-native:network/v20220401preview:UserRuleCollection":
                return new exports.UserRuleCollection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20220401preview", _module);
