// @pulumi/azure-native/webpubsub/v20210401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebPubSub = exports.listWebPubSubKeysOutput = exports.listWebPubSubKeys = exports.getWebPubSubOutput = exports.getWebPubSub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWebPubSub = null;
exports.getWebPubSubOutput = null;
utilities.lazyLoad(exports, ["getWebPubSub", "getWebPubSubOutput"], () => require("./getWebPubSub"));
exports.listWebPubSubKeys = null;
exports.listWebPubSubKeysOutput = null;
utilities.lazyLoad(exports, ["listWebPubSubKeys", "listWebPubSubKeysOutput"], () => require("./listWebPubSubKeys"));
exports.WebPubSub = null;
utilities.lazyLoad(exports, ["WebPubSub"], () => require("./webPubSub"));
// Export enums:
__exportStar(require("../../types/enums/webpubsub/v20210401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:webpubsub/v20210401preview:WebPubSub":
                return new exports.WebPubSub(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "webpubsub/v20210401preview", _module);
