// @pulumi/azure-native/containerservice/v20231002preview/trustedAccessRoleBinding.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustedAccessRoleBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Defines binding between a resource and role
 */
class TrustedAccessRoleBinding extends pulumi.CustomResource {
    /**
     * Get an existing TrustedAccessRoleBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TrustedAccessRoleBinding(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TrustedAccessRoleBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrustedAccessRoleBinding.__pulumiType;
    }
    /**
     * Create a TrustedAccessRoleBinding resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            if ((!args || args.roles === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roles'");
            }
            if ((!args || args.sourceResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceResourceId'");
            }
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["roles"] = args ? args.roles : undefined;
            resourceInputs["sourceResourceId"] = args ? args.sourceResourceId : undefined;
            resourceInputs["trustedAccessRoleBindingName"] = args ? args.trustedAccessRoleBindingName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["roles"] = undefined /*out*/;
            resourceInputs["sourceResourceId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerservice/v20220402preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220502preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220602preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220702preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220802preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220803preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20220902preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20221002preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20221102preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230102preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230202preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230302preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230402preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230502preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230602preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230702preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230802preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230901:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20230902preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20231001:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20231101:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20231102preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240101:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240102preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240201:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240202preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240302preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240402preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240501:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240502preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240602preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240701:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240702preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240801:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240901:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20240902preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20241001:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20241002preview:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice/v20250101:TrustedAccessRoleBinding" }, { type: "azure-native:containerservice:TrustedAccessRoleBinding" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TrustedAccessRoleBinding.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrustedAccessRoleBinding = TrustedAccessRoleBinding;
/** @internal */
TrustedAccessRoleBinding.__pulumiType = 'azure-native:containerservice/v20231002preview:TrustedAccessRoleBinding';
