// @pulumi/azure-native/types/enums/datashare/v20210801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.TriggerKind = exports.SynchronizationSettingKind = exports.SynchronizationMode = exports.ShareKind = exports.RecurrenceInterval = exports.OutputType = exports.DataSetMappingKind = exports.DataSetKind = void 0;
exports.DataSetKind = {
    Blob: "Blob",
    Container: "Container",
    BlobFolder: "BlobFolder",
    AdlsGen2FileSystem: "AdlsGen2FileSystem",
    AdlsGen2Folder: "AdlsGen2Folder",
    AdlsGen2File: "AdlsGen2File",
    AdlsGen1Folder: "AdlsGen1Folder",
    AdlsGen1File: "AdlsGen1File",
    KustoCluster: "KustoCluster",
    KustoDatabase: "KustoDatabase",
    KustoTable: "KustoTable",
    SqlDBTable: "SqlDBTable",
    SqlDWTable: "SqlDWTable",
    SynapseWorkspaceSqlPoolTable: "SynapseWorkspaceSqlPoolTable",
};
exports.DataSetMappingKind = {
    Blob: "Blob",
    Container: "Container",
    BlobFolder: "BlobFolder",
    AdlsGen2FileSystem: "AdlsGen2FileSystem",
    AdlsGen2Folder: "AdlsGen2Folder",
    AdlsGen2File: "AdlsGen2File",
    KustoCluster: "KustoCluster",
    KustoDatabase: "KustoDatabase",
    KustoTable: "KustoTable",
    SqlDBTable: "SqlDBTable",
    SqlDWTable: "SqlDWTable",
    SynapseWorkspaceSqlPoolTable: "SynapseWorkspaceSqlPoolTable",
};
exports.OutputType = {
    Csv: "Csv",
    Parquet: "Parquet",
};
exports.RecurrenceInterval = {
    Hour: "Hour",
    Day: "Day",
};
exports.ShareKind = {
    CopyBased: "CopyBased",
    InPlace: "InPlace",
};
exports.SynchronizationMode = {
    Incremental: "Incremental",
    FullSync: "FullSync",
};
exports.SynchronizationSettingKind = {
    ScheduleBased: "ScheduleBased",
};
exports.TriggerKind = {
    ScheduleBased: "ScheduleBased",
};
exports.Type = {
    SystemAssigned: "SystemAssigned",
};
