// @pulumi/azure-native/azuredata/sqlServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A SQL server.
 *
 * Uses Azure REST API version 2019-07-24-preview. In version 1.x of the Azure Native provider, it used API version 2019-07-24-preview.
 */
class SqlServer extends pulumi.CustomResource {
    /**
     * Get an existing SqlServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlServer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlServer.__pulumiType;
    }
    /**
     * Create a SqlServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sqlServerRegistrationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sqlServerRegistrationName'");
            }
            resourceInputs["cores"] = args ? args.cores : undefined;
            resourceInputs["edition"] = args ? args.edition : undefined;
            resourceInputs["propertyBag"] = args ? args.propertyBag : undefined;
            resourceInputs["registrationID"] = args ? args.registrationID : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sqlServerName"] = args ? args.sqlServerName : undefined;
            resourceInputs["sqlServerRegistrationName"] = args ? args.sqlServerRegistrationName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["cores"] = undefined /*out*/;
            resourceInputs["edition"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["propertyBag"] = undefined /*out*/;
            resourceInputs["registrationID"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azuredata/v20170301preview:SqlServer" }, { type: "azure-native:azuredata/v20190724preview:SqlServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlServer = SqlServer;
/** @internal */
SqlServer.__pulumiType = 'azure-native:azuredata:SqlServer';
