// @pulumi/azure-native/resourcehealth/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240201 = exports.v20231001preview = exports.v20230701preview = exports.v20221001preview = exports.v20221001 = exports.listSecurityAdvisoryImpactedResourceByTenantIdAndEventIdOutput = exports.listSecurityAdvisoryImpactedResourceByTenantIdAndEventId = exports.listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventIdOutput = exports.listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventId = void 0;
const utilities = require("../utilities");
exports.listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventId = null;
exports.listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventIdOutput = null;
utilities.lazyLoad(exports, ["listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventId", "listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventIdOutput"], () => require("./listSecurityAdvisoryImpactedResourceBySubscriptionIdAndEventId"));
exports.listSecurityAdvisoryImpactedResourceByTenantIdAndEventId = null;
exports.listSecurityAdvisoryImpactedResourceByTenantIdAndEventIdOutput = null;
utilities.lazyLoad(exports, ["listSecurityAdvisoryImpactedResourceByTenantIdAndEventId", "listSecurityAdvisoryImpactedResourceByTenantIdAndEventIdOutput"], () => require("./listSecurityAdvisoryImpactedResourceByTenantIdAndEventId"));
// Export sub-modules:
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20240201 = require("./v20240201");
exports.v20240201 = v20240201;
