// @pulumi/azure-native/loadtestservice/v20231201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadTestProfileMapping = exports.LoadTestMapping = exports.LoadTest = exports.getLoadTestProfileMappingOutput = exports.getLoadTestProfileMapping = exports.getLoadTestMappingOutput = exports.getLoadTestMapping = exports.getLoadTestOutput = exports.getLoadTest = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLoadTest = null;
exports.getLoadTestOutput = null;
utilities.lazyLoad(exports, ["getLoadTest", "getLoadTestOutput"], () => require("./getLoadTest"));
exports.getLoadTestMapping = null;
exports.getLoadTestMappingOutput = null;
utilities.lazyLoad(exports, ["getLoadTestMapping", "getLoadTestMappingOutput"], () => require("./getLoadTestMapping"));
exports.getLoadTestProfileMapping = null;
exports.getLoadTestProfileMappingOutput = null;
utilities.lazyLoad(exports, ["getLoadTestProfileMapping", "getLoadTestProfileMappingOutput"], () => require("./getLoadTestProfileMapping"));
exports.LoadTest = null;
utilities.lazyLoad(exports, ["LoadTest"], () => require("./loadTest"));
exports.LoadTestMapping = null;
utilities.lazyLoad(exports, ["LoadTestMapping"], () => require("./loadTestMapping"));
exports.LoadTestProfileMapping = null;
utilities.lazyLoad(exports, ["LoadTestProfileMapping"], () => require("./loadTestProfileMapping"));
// Export enums:
__exportStar(require("../../types/enums/loadtestservice/v20231201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:loadtestservice/v20231201preview:LoadTest":
                return new exports.LoadTest(name, undefined, { urn });
            case "azure-native:loadtestservice/v20231201preview:LoadTestMapping":
                return new exports.LoadTestMapping(name, undefined, { urn });
            case "azure-native:loadtestservice/v20231201preview:LoadTestProfileMapping":
                return new exports.LoadTestProfileMapping(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "loadtestservice/v20231201preview", _module);
