// @pulumi/azure-native/types/enums/containerregistry/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.WebhookStatus = exports.WebhookAction = exports.TrustPolicyType = exports.TriggerStatus = exports.TokenStatus = exports.TokenPasswordName = exports.TokenCertificateName = exports.SkuName = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PolicyStatus = exports.PipelineSourceType = exports.PipelineRunTargetType = exports.PipelineRunSourceType = exports.PipelineOptions = exports.PackageSourceType = exports.NetworkRuleBypassOptions = exports.MetadataSearch = exports.LogLevel = exports.ExportPolicyStatus = exports.EncryptionStatus = exports.DefaultAction = exports.CredentialName = exports.ConnectionStatus = exports.ConnectedRegistryMode = exports.AzureADAuthenticationAsArmPolicyStatus = exports.AutoGeneratedDomainNameLabelScope = exports.AuditLogStatus = exports.ActionsRequired = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.ActionsRequired = {
    None: "None",
    Recreate: "Recreate",
};
exports.AuditLogStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AutoGeneratedDomainNameLabelScope = {
    Unsecure: "Unsecure",
    TenantReuse: "TenantReuse",
    SubscriptionReuse: "SubscriptionReuse",
    ResourceGroupReuse: "ResourceGroupReuse",
    NoReuse: "NoReuse",
};
exports.AzureADAuthenticationAsArmPolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ConnectedRegistryMode = {
    ReadWrite: "ReadWrite",
    ReadOnly: "ReadOnly",
    Registry: "Registry",
    Mirror: "Mirror",
};
exports.ConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.CredentialName = {
    Credential1: "Credential1",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EncryptionStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ExportPolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.LogLevel = {
    Debug: "Debug",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
    None: "None",
};
exports.MetadataSearch = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.NetworkRuleBypassOptions = {
    AzureServices: "AzureServices",
    None: "None",
};
exports.PackageSourceType = {
    Remote: "remote",
};
exports.PipelineOptions = {
    OverwriteTags: "OverwriteTags",
    OverwriteBlobs: "OverwriteBlobs",
    DeleteSourceBlobOnSuccess: "DeleteSourceBlobOnSuccess",
    ContinueOnErrors: "ContinueOnErrors",
};
exports.PipelineRunSourceType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineRunTargetType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineSourceType = {
    AzureStorageBlobContainer: "AzureStorageBlobContainer",
};
exports.PolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.SkuName = {
    Classic: "Classic",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TokenCertificateName = {
    Certificate1: "certificate1",
    Certificate2: "certificate2",
};
exports.TokenPasswordName = {
    Password1: "password1",
    Password2: "password2",
};
exports.TokenStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.TriggerStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TrustPolicyType = {
    Notary: "Notary",
};
exports.WebhookAction = {
    Push: "push",
    Delete: "delete",
    Quarantine: "quarantine",
    Chart_push: "chart_push",
    Chart_delete: "chart_delete",
};
exports.WebhookStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ZoneRedundancy = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
