// @pulumi/azure-native/types/enums/streamanalytics/v20211001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateMode = exports.UpdatableUdfRefreshType = exports.SkuName = exports.RefreshType = exports.OutputWatermarkMode = exports.OutputStartMode = exports.OutputErrorPolicy = exports.JsonOutputSerializationFormat = exports.JobType = exports.InputWatermarkMode = exports.EventsOutOfOrderPolicy = exports.EventSerializationType = exports.EventGridEventSchemaType = exports.Encoding = exports.ContentStoragePolicy = exports.CompressionType = exports.CompatibilityLevel = exports.BlobWriteMode = exports.AuthenticationMode = void 0;
exports.AuthenticationMode = {
    Msi: "Msi",
    UserToken: "UserToken",
    ConnectionString: "ConnectionString",
};
exports.BlobWriteMode = {
    Append: "Append",
    Once: "Once",
};
exports.CompatibilityLevel = {
    CompatibilityLevel_1_0: "1.0",
    CompatibilityLevel_1_2: "1.2",
};
exports.CompressionType = {
    None: "None",
    GZip: "GZip",
    Deflate: "Deflate",
};
exports.ContentStoragePolicy = {
    SystemAccount: "SystemAccount",
    JobStorageAccount: "JobStorageAccount",
};
exports.Encoding = {
    UTF8: "UTF8",
};
exports.EventGridEventSchemaType = {
    EventGridEventSchema: "EventGridEventSchema",
    CloudEventSchema: "CloudEventSchema",
};
exports.EventSerializationType = {
    Csv: "Csv",
    Avro: "Avro",
    Json: "Json",
    CustomClr: "CustomClr",
    Parquet: "Parquet",
    Delta: "Delta",
};
exports.EventsOutOfOrderPolicy = {
    Adjust: "Adjust",
    Drop: "Drop",
};
exports.InputWatermarkMode = {
    None: "None",
    ReadWatermark: "ReadWatermark",
};
exports.JobType = {
    Cloud: "Cloud",
    Edge: "Edge",
};
exports.JsonOutputSerializationFormat = {
    LineSeparated: "LineSeparated",
    Array: "Array",
};
exports.OutputErrorPolicy = {
    Stop: "Stop",
    Drop: "Drop",
};
exports.OutputStartMode = {
    JobStartTime: "JobStartTime",
    CustomTime: "CustomTime",
    LastOutputEventTime: "LastOutputEventTime",
};
exports.OutputWatermarkMode = {
    None: "None",
    SendCurrentPartitionWatermark: "SendCurrentPartitionWatermark",
    SendLowestWatermarkAcrossPartitions: "SendLowestWatermarkAcrossPartitions",
};
exports.RefreshType = {
    Static: "Static",
    RefreshPeriodicallyWithFull: "RefreshPeriodicallyWithFull",
    RefreshPeriodicallyWithDelta: "RefreshPeriodicallyWithDelta",
};
exports.SkuName = {
    Standard: "Standard",
};
exports.UpdatableUdfRefreshType = {
    Blocking: "Blocking",
    Nonblocking: "Nonblocking",
};
exports.UpdateMode = {
    Static: "Static",
    Refreshable: "Refreshable",
};
