// @pulumi/azure-native/types/enums/powerplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.State = exports.ResourceIdentityType = exports.PrivateEndpointServiceConnectionStatus = exports.HealthStatus = exports.EnterprisePolicyKind = exports.v20201030preview = void 0;
// Export sub-modules:
const v20201030preview = require("./v20201030preview");
exports.v20201030preview = v20201030preview;
exports.EnterprisePolicyKind = {
    Lockbox: "Lockbox",
    PrivateEndpoint: "PrivateEndpoint",
    Encryption: "Encryption",
    NetworkInjection: "NetworkInjection",
    Identity: "Identity",
};
exports.HealthStatus = {
    Undetermined: "Undetermined",
    Healthy: "Healthy",
    Warning: "Warning",
    Unhealthy: "Unhealthy",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.State = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    NotConfigured: "NotConfigured",
};
