// @pulumi/azure-native/securityinsights/v20220701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listSourceControlRepositoriesOutput = exports.listSourceControlRepositories = exports.getEntityInsightsOutput = exports.getEntityInsights = exports.getEntitiesGetTimelineOutput = exports.getEntitiesGetTimeline = void 0;
const utilities = require("../../utilities");
exports.getEntitiesGetTimeline = null;
exports.getEntitiesGetTimelineOutput = null;
utilities.lazyLoad(exports, ["getEntitiesGetTimeline", "getEntitiesGetTimelineOutput"], () => require("./getEntitiesGetTimeline"));
exports.getEntityInsights = null;
exports.getEntityInsightsOutput = null;
utilities.lazyLoad(exports, ["getEntityInsights", "getEntityInsightsOutput"], () => require("./getEntityInsights"));
exports.listSourceControlRepositories = null;
exports.listSourceControlRepositoriesOutput = null;
utilities.lazyLoad(exports, ["listSourceControlRepositories", "listSourceControlRepositoriesOutput"], () => require("./listSourceControlRepositories"));
// Export enums:
__exportStar(require("../../types/enums/securityinsights/v20220701preview"), exports);
