// @pulumi/azure-native/azurearcdata/activeDirectoryConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActiveDirectoryConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Active directory connector resource
 *
 * Uses Azure REST API version 2023-01-15-preview. In version 1.x of the Azure Native provider, it used API version 2022-03-01-preview.
 *
 * Other available API versions: 2024-01-01, 2024-05-01-preview, 2025-03-01-preview.
 */
class ActiveDirectoryConnector extends pulumi.CustomResource {
    /**
     * Get an existing ActiveDirectoryConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ActiveDirectoryConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ActiveDirectoryConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ActiveDirectoryConnector.__pulumiType;
    }
    /**
     * Create a ActiveDirectoryConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataControllerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataControllerName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["activeDirectoryConnectorName"] = args ? args.activeDirectoryConnectorName : undefined;
            resourceInputs["dataControllerName"] = args ? args.dataControllerName : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(inputs.azurearcdata.activeDirectoryConnectorPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurearcdata/v20220301preview:ActiveDirectoryConnector" }, { type: "azure-native:azurearcdata/v20220615preview:ActiveDirectoryConnector" }, { type: "azure-native:azurearcdata/v20230115preview:ActiveDirectoryConnector" }, { type: "azure-native:azurearcdata/v20240101:ActiveDirectoryConnector" }, { type: "azure-native:azurearcdata/v20240501preview:ActiveDirectoryConnector" }, { type: "azure-native:azurearcdata/v20250301preview:ActiveDirectoryConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ActiveDirectoryConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ActiveDirectoryConnector = ActiveDirectoryConnector;
/** @internal */
ActiveDirectoryConnector.__pulumiType = 'azure-native:azurearcdata:ActiveDirectoryConnector';
