// @pulumi/azure-native/securityinsights/v20241001preview/scheduledAlertRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledAlertRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents scheduled alert rule.
 */
class ScheduledAlertRule extends pulumi.CustomResource {
    /**
     * Get an existing ScheduledAlertRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScheduledAlertRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScheduledAlertRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScheduledAlertRule.__pulumiType;
    }
    /**
     * Create a ScheduledAlertRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.enabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.query === undefined) && !opts.urn) {
                throw new Error("Missing required property 'query'");
            }
            if ((!args || args.queryFrequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryFrequency'");
            }
            if ((!args || args.queryPeriod === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryPeriod'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.severity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'severity'");
            }
            if ((!args || args.suppressionDuration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'suppressionDuration'");
            }
            if ((!args || args.suppressionEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'suppressionEnabled'");
            }
            if ((!args || args.triggerOperator === undefined) && !opts.urn) {
                throw new Error("Missing required property 'triggerOperator'");
            }
            if ((!args || args.triggerThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'triggerThreshold'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["alertDetailsOverride"] = args ? args.alertDetailsOverride : undefined;
            resourceInputs["alertRuleTemplateName"] = args ? args.alertRuleTemplateName : undefined;
            resourceInputs["customDetails"] = args ? args.customDetails : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["entityMappings"] = args ? args.entityMappings : undefined;
            resourceInputs["eventGroupingSettings"] = args ? args.eventGroupingSettings : undefined;
            resourceInputs["incidentConfiguration"] = args ? args.incidentConfiguration : undefined;
            resourceInputs["kind"] = "Scheduled";
            resourceInputs["query"] = args ? args.query : undefined;
            resourceInputs["queryFrequency"] = args ? args.queryFrequency : undefined;
            resourceInputs["queryPeriod"] = args ? args.queryPeriod : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["ruleId"] = args ? args.ruleId : undefined;
            resourceInputs["sentinelEntitiesMappings"] = args ? args.sentinelEntitiesMappings : undefined;
            resourceInputs["severity"] = args ? args.severity : undefined;
            resourceInputs["subTechniques"] = args ? args.subTechniques : undefined;
            resourceInputs["suppressionDuration"] = args ? args.suppressionDuration : undefined;
            resourceInputs["suppressionEnabled"] = args ? args.suppressionEnabled : undefined;
            resourceInputs["tactics"] = args ? args.tactics : undefined;
            resourceInputs["techniques"] = args ? args.techniques : undefined;
            resourceInputs["templateVersion"] = args ? args.templateVersion : undefined;
            resourceInputs["triggerOperator"] = args ? args.triggerOperator : undefined;
            resourceInputs["triggerThreshold"] = args ? args.triggerThreshold : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["alertDetailsOverride"] = undefined /*out*/;
            resourceInputs["alertRuleTemplateName"] = undefined /*out*/;
            resourceInputs["customDetails"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["entityMappings"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["eventGroupingSettings"] = undefined /*out*/;
            resourceInputs["incidentConfiguration"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastModifiedUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["query"] = undefined /*out*/;
            resourceInputs["queryFrequency"] = undefined /*out*/;
            resourceInputs["queryPeriod"] = undefined /*out*/;
            resourceInputs["sentinelEntitiesMappings"] = undefined /*out*/;
            resourceInputs["severity"] = undefined /*out*/;
            resourceInputs["subTechniques"] = undefined /*out*/;
            resourceInputs["suppressionDuration"] = undefined /*out*/;
            resourceInputs["suppressionEnabled"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tactics"] = undefined /*out*/;
            resourceInputs["techniques"] = undefined /*out*/;
            resourceInputs["templateVersion"] = undefined /*out*/;
            resourceInputs["triggerOperator"] = undefined /*out*/;
            resourceInputs["triggerThreshold"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20200101:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20210301preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20210901preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20211001:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20211001preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220401preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220501preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220601preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220701preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220801:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220801preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20220901preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20221001preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20221101:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20221101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20221201preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230201:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230201preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230301preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230401preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230501preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230601preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230701preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230801preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20230901preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20231001preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20231101:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20231201preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20240101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20240301:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20240401preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20240901:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20250101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20250301:ScheduledAlertRule" }, { type: "azure-native:securityinsights:ScheduledAlertRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScheduledAlertRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ScheduledAlertRule = ScheduledAlertRule;
/** @internal */
ScheduledAlertRule.__pulumiType = 'azure-native:securityinsights/v20241001preview:ScheduledAlertRule';
