// @pulumi/azure-native/costmanagement/v20230401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewByScope = exports.View = exports.ScheduledActionByScope = exports.ScheduledAction = exports.getViewByScopeOutput = exports.getViewByScope = exports.getViewOutput = exports.getView = exports.getScheduledActionByScopeOutput = exports.getScheduledActionByScope = exports.getScheduledActionOutput = exports.getScheduledAction = exports.getExportOutput = exports.getExport = exports.getBudgetOutput = exports.getBudget = exports.Export = exports.Budget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Budget = null;
utilities.lazyLoad(exports, ["Budget"], () => require("./budget"));
exports.Export = null;
utilities.lazyLoad(exports, ["Export"], () => require("./export"));
exports.getBudget = null;
exports.getBudgetOutput = null;
utilities.lazyLoad(exports, ["getBudget", "getBudgetOutput"], () => require("./getBudget"));
exports.getExport = null;
exports.getExportOutput = null;
utilities.lazyLoad(exports, ["getExport", "getExportOutput"], () => require("./getExport"));
exports.getScheduledAction = null;
exports.getScheduledActionOutput = null;
utilities.lazyLoad(exports, ["getScheduledAction", "getScheduledActionOutput"], () => require("./getScheduledAction"));
exports.getScheduledActionByScope = null;
exports.getScheduledActionByScopeOutput = null;
utilities.lazyLoad(exports, ["getScheduledActionByScope", "getScheduledActionByScopeOutput"], () => require("./getScheduledActionByScope"));
exports.getView = null;
exports.getViewOutput = null;
utilities.lazyLoad(exports, ["getView", "getViewOutput"], () => require("./getView"));
exports.getViewByScope = null;
exports.getViewByScopeOutput = null;
utilities.lazyLoad(exports, ["getViewByScope", "getViewByScopeOutput"], () => require("./getViewByScope"));
exports.ScheduledAction = null;
utilities.lazyLoad(exports, ["ScheduledAction"], () => require("./scheduledAction"));
exports.ScheduledActionByScope = null;
utilities.lazyLoad(exports, ["ScheduledActionByScope"], () => require("./scheduledActionByScope"));
exports.View = null;
utilities.lazyLoad(exports, ["View"], () => require("./view"));
exports.ViewByScope = null;
utilities.lazyLoad(exports, ["ViewByScope"], () => require("./viewByScope"));
// Export enums:
__exportStar(require("../../types/enums/costmanagement/v20230401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:costmanagement/v20230401preview:Budget":
                return new exports.Budget(name, undefined, { urn });
            case "azure-native:costmanagement/v20230401preview:Export":
                return new exports.Export(name, undefined, { urn });
            case "azure-native:costmanagement/v20230401preview:ScheduledAction":
                return new exports.ScheduledAction(name, undefined, { urn });
            case "azure-native:costmanagement/v20230401preview:ScheduledActionByScope":
                return new exports.ScheduledActionByScope(name, undefined, { urn });
            case "azure-native:costmanagement/v20230401preview:View":
                return new exports.View(name, undefined, { urn });
            case "azure-native:costmanagement/v20230401preview:ViewByScope":
                return new exports.ViewByScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "costmanagement/v20230401preview", _module);
