// @pulumi/azure-native/hybriddata/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20190601 = exports.JobDefinition = exports.getJobDefinitionOutput = exports.getJobDefinition = exports.getDataStoreOutput = exports.getDataStore = exports.getDataManagerOutput = exports.getDataManager = exports.DataStore = exports.DataManager = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataManager = null;
utilities.lazyLoad(exports, ["DataManager"], () => require("./dataManager"));
exports.DataStore = null;
utilities.lazyLoad(exports, ["DataStore"], () => require("./dataStore"));
exports.getDataManager = null;
exports.getDataManagerOutput = null;
utilities.lazyLoad(exports, ["getDataManager", "getDataManagerOutput"], () => require("./getDataManager"));
exports.getDataStore = null;
exports.getDataStoreOutput = null;
utilities.lazyLoad(exports, ["getDataStore", "getDataStoreOutput"], () => require("./getDataStore"));
exports.getJobDefinition = null;
exports.getJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getJobDefinition", "getJobDefinitionOutput"], () => require("./getJobDefinition"));
exports.JobDefinition = null;
utilities.lazyLoad(exports, ["JobDefinition"], () => require("./jobDefinition"));
// Export enums:
__exportStar(require("../types/enums/hybriddata"), exports);
// Export sub-modules:
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybriddata:DataManager":
                return new exports.DataManager(name, undefined, { urn });
            case "azure-native:hybriddata:DataStore":
                return new exports.DataStore(name, undefined, { urn });
            case "azure-native:hybriddata:JobDefinition":
                return new exports.JobDefinition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybriddata", _module);
