// @pulumi/azure-native/management/v20210401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementGroupSubscription = exports.ManagementGroup = exports.HierarchySetting = exports.getManagementGroupSubscriptionOutput = exports.getManagementGroupSubscription = exports.getManagementGroupOutput = exports.getManagementGroup = exports.getHierarchySettingOutput = exports.getHierarchySetting = exports.getEntityOutput = exports.getEntity = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getEntity = null;
exports.getEntityOutput = null;
utilities.lazyLoad(exports, ["getEntity", "getEntityOutput"], () => require("./getEntity"));
exports.getHierarchySetting = null;
exports.getHierarchySettingOutput = null;
utilities.lazyLoad(exports, ["getHierarchySetting", "getHierarchySettingOutput"], () => require("./getHierarchySetting"));
exports.getManagementGroup = null;
exports.getManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getManagementGroup", "getManagementGroupOutput"], () => require("./getManagementGroup"));
exports.getManagementGroupSubscription = null;
exports.getManagementGroupSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getManagementGroupSubscription", "getManagementGroupSubscriptionOutput"], () => require("./getManagementGroupSubscription"));
exports.HierarchySetting = null;
utilities.lazyLoad(exports, ["HierarchySetting"], () => require("./hierarchySetting"));
exports.ManagementGroup = null;
utilities.lazyLoad(exports, ["ManagementGroup"], () => require("./managementGroup"));
exports.ManagementGroupSubscription = null;
utilities.lazyLoad(exports, ["ManagementGroupSubscription"], () => require("./managementGroupSubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:management/v20210401:HierarchySetting":
                return new exports.HierarchySetting(name, undefined, { urn });
            case "azure-native:management/v20210401:ManagementGroup":
                return new exports.ManagementGroup(name, undefined, { urn });
            case "azure-native:management/v20210401:ManagementGroupSubscription":
                return new exports.ManagementGroupSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "management/v20210401", _module);
