// @pulumi/azure-native/types/enums/migrate/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancerType = exports.LicensingProgram = exports.LicenseType = exports.HyperVLicenseType = exports.GroupType = exports.Goal = exports.FttAndRaidLevel = exports.EnvironmentType = exports.DiscoverySource = exports.Currency = exports.ConsolidationType = exports.ConfigurationType = exports.ComputeTier = exports.CleanupState = exports.BusinessCaseCurrency = exports.AzureVmFamily = exports.AzureVmCategory = exports.AzureStorageRedundancy = exports.AzureSqlServiceTier = exports.AzureSqlPurchaseModel = exports.AzureSqlInstanceType = exports.AzureSqlDataBaseType = exports.AzureSecurityOfferingType = exports.AzureReservedInstance = exports.AzurePricingTier = exports.AzureOfferCode = exports.AzureLocation = exports.AzureHybridUseBenefit = exports.AzureEnvironmentType = exports.AzureDiskType = exports.AzureCurrency = exports.AzureAvsNodeType = exports.AutomationArtifactStatus = exports.AsyncCommitModeIntent = exports.AssessmentType = exports.AssessmentStage = exports.AssessmentSizingCriterion = exports.v20240101preview = exports.v20230909preview = exports.v20230801 = exports.v20230501preview = exports.v20230401preview = exports.v20230315 = exports.v20230101 = exports.v20220801 = exports.v20220501preview = exports.v20200501 = exports.v20191001 = exports.v20180901preview = exports.v20180202 = void 0;
exports.ZoneRedundant = exports.WorkloadType = exports.WorkloadDeploymentType = exports.WorkloadDeploymentTarget = exports.VsphereManagementLicenseType = exports.Tool = exports.TimeRange = exports.TargetStorageProjectionType = exports.TargetStorageAccessType = exports.TargetHydrationStorageProviderType = exports.TargetAvailabilityZone = exports.Status = exports.SqlServerLicenseType = exports.SqlServerLicense = exports.SecretStoreType = exports.SavingsOptions = exports.SavingsOption = exports.ResourceIdentityTypes = exports.ResourceIdentityType = exports.ReservedInstance = exports.Purpose = exports.ProvisioningState = exports.ProjectStatus = exports.PrivateEndpointServiceConnectionStatus = exports.PricingTier = exports.Percentile = exports.OsLicense = exports.OptimizationLogic = exports.OperatingSystemType = exports.MultiSubnetIntent = exports.MigrationStrategy = void 0;
// Export sub-modules:
const v20180202 = require("./v20180202");
exports.v20180202 = v20180202;
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20220801 = require("./v20220801");
exports.v20220801 = v20220801;
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
const v20230315 = require("./v20230315");
exports.v20230315 = v20230315;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20230909preview = require("./v20230909preview");
exports.v20230909preview = v20230909preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
exports.AssessmentSizingCriterion = {
    /**
     * Performance Data based Sizing.
     */
    PerformanceBased: "PerformanceBased",
    /**
     * As On Premises or Static Data based Sizing.
     */
    AsOnPremises: "AsOnPremises",
};
exports.AssessmentStage = {
    InProgress: "InProgress",
    UnderReview: "UnderReview",
    Approved: "Approved",
};
exports.AssessmentType = {
    Unknown: "Unknown",
    MachineAssessment: "MachineAssessment",
    AvsAssessment: "AvsAssessment",
    SqlAssessment: "SqlAssessment",
    WebAppAssessment: "WebAppAssessment",
};
exports.AsyncCommitModeIntent = {
    None: "None",
    HighAvailability: "HighAvailability",
    DisasterRecovery: "DisasterRecovery",
};
exports.AutomationArtifactStatus = {
    NotGenerated: "NotGenerated",
    Generated: "Generated",
};
exports.AzureAvsNodeType = {
    Unknown: "Unknown",
    AV36: "AV36",
};
exports.AzureCurrency = {
    Unknown: "Unknown",
    USD: "USD",
    DKK: "DKK",
    CAD: "CAD",
    IDR: "IDR",
    JPY: "JPY",
    KRW: "KRW",
    NZD: "NZD",
    NOK: "NOK",
    RUB: "RUB",
    SAR: "SAR",
    ZAR: "ZAR",
    SEK: "SEK",
    TRY: "TRY",
    GBP: "GBP",
    MXN: "MXN",
    MYR: "MYR",
    INR: "INR",
    HKD: "HKD",
    BRL: "BRL",
    TWD: "TWD",
    EUR: "EUR",
    CHF: "CHF",
    ARS: "ARS",
    AUD: "AUD",
    CNY: "CNY",
};
exports.AzureDiskType = {
    Unknown: "Unknown",
    Standard: "Standard",
    StandardSSD: "StandardSSD",
    Premium: "Premium",
    StandardOrPremium: "StandardOrPremium",
    Ultra: "Ultra",
    PremiumV2: "PremiumV2",
};
exports.AzureEnvironmentType = {
    /**
     * Unknown. Indicates missing data.
     */
    Unknown: "Unknown",
    /**
     * Development or Test Environment.
     */
    DevTest: "DevTest",
    /**
     * Production Environment.
     */
    Production: "Production",
};
exports.AzureHybridUseBenefit = {
    Unknown: "Unknown",
    Yes: "Yes",
    No: "No",
};
exports.AzureLocation = {
    Unknown: "Unknown",
    EastAsia: "EastAsia",
    SoutheastAsia: "SoutheastAsia",
    AustraliaEast: "AustraliaEast",
    AustraliaSoutheast: "AustraliaSoutheast",
    BrazilSouth: "BrazilSouth",
    CanadaCentral: "CanadaCentral",
    CanadaEast: "CanadaEast",
    WestEurope: "WestEurope",
    NorthEurope: "NorthEurope",
    CentralIndia: "CentralIndia",
    SouthIndia: "SouthIndia",
    WestIndia: "WestIndia",
    JapanEast: "JapanEast",
    JapanWest: "JapanWest",
    KoreaCentral: "KoreaCentral",
    KoreaSouth: "KoreaSouth",
    UkWest: "UkWest",
    UkSouth: "UkSouth",
    NorthCentralUs: "NorthCentralUs",
    EastUs: "EastUs",
    WestUs2: "WestUs2",
    SouthCentralUs: "SouthCentralUs",
    CentralUs: "CentralUs",
    EastUs2: "EastUs2",
    WestUs: "WestUs",
    WestCentralUs: "WestCentralUs",
    GermanyCentral: "GermanyCentral",
    GermanyNortheast: "GermanyNortheast",
    ChinaNorth: "ChinaNorth",
    ChinaEast: "ChinaEast",
    USGovArizona: "USGovArizona",
    USGovTexas: "USGovTexas",
    USGovIowa: "USGovIowa",
    USGovVirginia: "USGovVirginia",
    USDoDCentral: "USDoDCentral",
    USDoDEast: "USDoDEast",
    FranceCentral: "FranceCentral",
    AustraliaCentral: "AustraliaCentral",
    SouthAfricaNorth: "SouthAfricaNorth",
    FranceSouth: "FranceSouth",
    AustraliaCentral2: "AustraliaCentral2",
    SouthAfricaWest: "SouthAfricaWest",
    GermanyNorth: "GermanyNorth",
    GermanyWestCentral: "GermanyWestCentral",
    NorwayEast: "NorwayEast",
    NorwayWest: "NorwayWest",
    ChinaEast2: "ChinaEast2",
    ChinaNorth2: "ChinaNorth2",
    SwitzerlandNorth: "SwitzerlandNorth",
    SwitzerlandWest: "SwitzerlandWest",
    UAENorth: "UAENorth",
    UAECentral: "UAECentral",
    UsNatEast: "UsNatEast",
    UsNatWest: "UsNatWest",
    UsSecEast: "UsSecEast",
    UsSecCentral: "UsSecCentral",
    UsSecWest: "UsSecWest",
    SwedenCentral: "SwedenCentral",
    QatarCentral: "QatarCentral",
};
exports.AzureOfferCode = {
    Unknown: "Unknown",
    Msazr0003P: "MSAZR0003P",
    Msazr0044P: "MSAZR0044P",
    Msazr0059P: "MSAZR0059P",
    Msazr0060P: "MSAZR0060P",
    Msazr0062P: "MSAZR0062P",
    Msazr0063P: "MSAZR0063P",
    Msazr0064P: "MSAZR0064P",
    Msazr0029P: "MSAZR0029P",
    Msazr0022P: "MSAZR0022P",
    Msazr0023P: "MSAZR0023P",
    Msazr0148P: "MSAZR0148P",
    Msazr0025P: "MSAZR0025P",
    Msazr0036P: "MSAZR0036P",
    Msazr0120P: "MSAZR0120P",
    Msazr0121P: "MSAZR0121P",
    Msazr0122P: "MSAZR0122P",
    Msazr0123P: "MSAZR0123P",
    Msazr0124P: "MSAZR0124P",
    Msazr0125P: "MSAZR0125P",
    Msazr0126P: "MSAZR0126P",
    Msazr0127P: "MSAZR0127P",
    Msazr0128P: "MSAZR0128P",
    Msazr0129P: "MSAZR0129P",
    Msazr0130P: "MSAZR0130P",
    Msazr0111P: "MSAZR0111P",
    Msazr0144P: "MSAZR0144P",
    Msazr0149P: "MSAZR0149P",
    Msmcazr0044P: "MSMCAZR0044P",
    Msmcazr0059P: "MSMCAZR0059P",
    Msmcazr0060P: "MSMCAZR0060P",
    Msmcazr0063P: "MSMCAZR0063P",
    Msmcazr0120P: "MSMCAZR0120P",
    Msmcazr0121P: "MSMCAZR0121P",
    Msmcazr0125P: "MSMCAZR0125P",
    Msmcazr0128P: "MSMCAZR0128P",
    Msazrde0003P: "MSAZRDE0003P",
    Msazrde0044P: "MSAZRDE0044P",
    Msazrusgov0003P: "MSAZRUSGOV0003P",
    EA: "EA",
    Msazr0243P: "MSAZR0243P",
    SavingsPlan1Year: "SavingsPlan1Year",
    SavingsPlan3Year: "SavingsPlan3Year",
};
exports.AzurePricingTier = {
    Standard: "Standard",
    Basic: "Basic",
};
exports.AzureReservedInstance = {
    None: "None",
    RI1Year: "RI1Year",
    RI3Year: "RI3Year",
};
exports.AzureSecurityOfferingType = {
    NO: "NO",
    MDC: "MDC",
};
exports.AzureSqlDataBaseType = {
    Unknown: "Unknown",
    Automatic: "Automatic",
    SingleDatabase: "SingleDatabase",
    ElasticPool: "ElasticPool",
};
exports.AzureSqlInstanceType = {
    Unknown: "Unknown",
    Automatic: "Automatic",
    SingleInstance: "SingleInstance",
    InstancePools: "InstancePools",
};
exports.AzureSqlPurchaseModel = {
    Unknown: "Unknown",
    VCore: "VCore",
    DTU: "DTU",
};
exports.AzureSqlServiceTier = {
    Unknown: "Unknown",
    Automatic: "Automatic",
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
    HyperScale: "HyperScale",
};
exports.AzureStorageRedundancy = {
    Unknown: "Unknown",
    LocallyRedundant: "LocallyRedundant",
    ZoneRedundant: "ZoneRedundant",
    GeoRedundant: "GeoRedundant",
    ReadAccessGeoRedundant: "ReadAccessGeoRedundant",
};
exports.AzureVmCategory = {
    /**
     * Indicates All categories of VM.
     */
    All: "All",
    /**
     * Compute Optimized.
     */
    ComputeOptimized: "ComputeOptimized",
    /**
     * General Purpose.
     */
    GeneralPurpose: "GeneralPurpose",
    /**
     * GPU Optimized.
     */
    GpuOptimized: "GpuOptimized",
    /**
     * High Performance Compute.
     */
    HighPerformanceCompute: "HighPerformanceCompute",
    /**
     * Memory Optimized.
     */
    MemoryOptimized: "MemoryOptimized",
    /**
     * Storage Optimized.
     */
    StorageOptimized: "StorageOptimized",
    /**
     * Isolated VM.
     */
    Isolated: "Isolated",
};
exports.AzureVmFamily = {
    Unknown: "Unknown",
    BasicA0A4: "Basic_A0_A4",
    StandardA0A7: "Standard_A0_A7",
    StandardA8A11: "Standard_A8_A11",
    Av2Series: "Av2_series",
    DSeries: "D_series",
    Dv2Series: "Dv2_series",
    DSSeries: "DS_series",
    DSv2Series: "DSv2_series",
    FSeries: "F_series",
    FsSeries: "Fs_series",
    GSeries: "G_series",
    GSSeries: "GS_series",
    HSeries: "H_series",
    LsSeries: "Ls_series",
    Dsv3Series: "Dsv3_series",
    Dv3Series: "Dv3_series",
    Fsv2Series: "Fsv2_series",
    Ev3Series: "Ev3_series",
    Esv3Series: "Esv3_series",
    MSeries: "M_series",
    DCSeries: "DC_Series",
    Lsv2Series: "Lsv2_series",
    Ev4Series: "Ev4_series",
    Esv4Series: "Esv4_series",
    Edv4Series: "Edv4_series",
    Edsv4Series: "Edsv4_series",
    Dv4Series: "Dv4_series",
    Dsv4Series: "Dsv4_series",
    Ddv4Series: "Ddv4_series",
    Ddsv4Series: "Ddsv4_series",
    Easv4Series: "Easv4_series",
    Dasv4Series: "Dasv4_series",
    Mv2Series: "Mv2_series",
    Eav4Series: "Eav4_series",
    Dav4Series: "Dav4_series",
    Msv2Series: "Msv2_series",
    Mdsv2Series: "Mdsv2_series",
    Dv5Series: "Dv5_series",
    Dsv5Series: "Dsv5_series",
    Ddv5Series: "Ddv5_series",
    Ddsv5Series: "Ddsv5_series",
    Dasv5Series: "Dasv5_series",
    Dadsv5Series: "Dadsv5_series",
    Ev5Series: "Ev5_series",
    Esv5Series: "Esv5_series",
    Edv5Series: "Edv5_series",
    Edsv5Series: "Edsv5_series",
    Easv5Series: "Easv5_series",
    Eadsv5Series: "Eadsv5_series",
    Ebsv5Series: "Ebsv5_series",
    Ebdsv5Series: "Ebdsv5_series",
};
exports.BusinessCaseCurrency = {
    /**
     * Currency Unknown.
     */
    Unknown: "Unknown",
    /**
     * Currency USD.
     */
    USD: "USD",
    /**
     * Currency DKK.
     */
    DKK: "DKK",
    /**
     * Currency CAD.
     */
    CAD: "CAD",
    /**
     * Currency IDR.
     */
    IDR: "IDR",
    /**
     * Currency JPY.
     */
    JPY: "JPY",
    /**
     * Currency KRW.
     */
    KRW: "KRW",
    /**
     * Currency NZD.
     */
    NZD: "NZD",
    /**
     * Currency NOK.
     */
    NOK: "NOK",
    /**
     * Currency RUB.
     */
    RUB: "RUB",
    /**
     * Currency SAR.
     */
    SAR: "SAR",
    /**
     * Currency ZAR.
     */
    ZAR: "ZAR",
    /**
     * Currency SEK.
     */
    SEK: "SEK",
    /**
     * Currency TRY.
     */
    TRY: "TRY",
    /**
     * Currency GBP.
     */
    GBP: "GBP",
    /**
     * Currency MXN.
     */
    MXN: "MXN",
    /**
     * Currency MYR.
     */
    MYR: "MYR",
    /**
     * Currency INR.
     */
    INR: "INR",
    /**
     * Currency HKD.
     */
    HKD: "HKD",
    /**
     * Currency BRL.
     */
    BRL: "BRL",
    /**
     * Currency TWD.
     */
    TWD: "TWD",
    /**
     * Currency EUR.
     */
    EUR: "EUR",
    /**
     * Currency CHF.
     */
    CHF: "CHF",
    /**
     * Currency ARS.
     */
    ARS: "ARS",
    /**
     * Currency AUD.
     */
    AUD: "AUD",
    /**
     * Currency CNY.
     */
    CNY: "CNY",
};
exports.CleanupState = {
    None: "None",
    Started: "Started",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
};
exports.ComputeTier = {
    Unknown: "Unknown",
    Automatic: "Automatic",
    Provisioned: "Provisioned",
    Serverless: "Serverless",
};
exports.ConfigurationType = {
    IISConnectionString: "IISConnectionString",
    IISAuthentication: "IISAuthentication",
    ApacheTomcatContextResource: "ApacheTomcatContextResource",
};
exports.ConsolidationType = {
    /**
     * Full Consolidation.
     */
    Full: "Full",
    /**
     * As On Source or On Premises Consolidation.
     */
    AsOnSource: "AsOnSource",
};
exports.Currency = {
    Unknown: "Unknown",
    USD: "USD",
    DKK: "DKK",
    CAD: "CAD",
    IDR: "IDR",
    JPY: "JPY",
    KRW: "KRW",
    NZD: "NZD",
    NOK: "NOK",
    RUB: "RUB",
    SAR: "SAR",
    ZAR: "ZAR",
    SEK: "SEK",
    TRY: "TRY",
    GBP: "GBP",
    MXN: "MXN",
    MYR: "MYR",
    INR: "INR",
    HKD: "HKD",
    BRL: "BRL",
    TWD: "TWD",
    EUR: "EUR",
    CHF: "CHF",
    ARS: "ARS",
    AUD: "AUD",
    CNY: "CNY",
};
exports.DiscoverySource = {
    /**
     * Unknown Discovery Source.
     */
    Unknown: "Unknown",
    /**
     * Appliance Discovery Source.
     */
    Appliance: "Appliance",
    /**
     * Import Discovery Source.
     */
    Import: "Import",
};
exports.EnvironmentType = {
    Production: "Production",
    Test: "Test",
};
exports.FttAndRaidLevel = {
    Unknown: "Unknown",
    Ftt1Raid1: "Ftt1Raid1",
    Ftt1Raid5: "Ftt1Raid5",
    Ftt2Raid1: "Ftt2Raid1",
    Ftt2Raid6: "Ftt2Raid6",
    Ftt3Raid1: "Ftt3Raid1",
};
exports.Goal = {
    Servers: "Servers",
    Databases: "Databases",
    DesktopVirtualization: "DesktopVirtualization",
    WebApplications: "WebApplications",
    DataCenter: "DataCenter",
};
exports.GroupType = {
    Default: "Default",
    Import: "Import",
};
exports.HyperVLicenseType = {
    /**
     * Unknown HyperV License.
     */
    Unknown: "Unknown",
    /**
     * Datacentre HyperV License.
     */
    Datacentre: "Datacentre",
    /**
     * Standard HyperV License.
     */
    Standard: "Standard",
};
exports.LicenseType = {
    /**
     * Unknown License.
     */
    Unknown: "Unknown",
    /**
     * VSphereStandard License.
     */
    VSphereStandard: "VSphereStandard",
    /**
     * VSphereEnterprisePlus License.
     */
    VSphereEnterprisePlus: "VSphereEnterprisePlus",
};
exports.LicensingProgram = {
    /**
     * Default value. Indicates Pay As You Go.
     */
    Default: "Default",
    /**
     * Enterprise Agreement.
     */
    EA: "EA",
};
exports.LoadBalancerType = {
    Private: "Private",
    Public: "Public",
};
exports.MigrationStrategy = {
    /**
     * Unknown Migration Strategy.
     */
    Unknown: "Unknown",
    /**
     * Optimize for cost.
     */
    OptimizeForCost: "OptimizeForCost",
    /**
     * IaaS only.
     */
    IaaSOnly: "IaaSOnly",
    /**
     * Optimize for PaaS.
     */
    OptimizeForPaas: "OptimizeForPaas",
    /**
     * Avs only.
     */
    AVSOnly: "AVSOnly",
};
exports.MultiSubnetIntent = {
    None: "None",
    HighAvailability: "HighAvailability",
    DisasterRecovery: "DisasterRecovery",
};
exports.OperatingSystemType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.OptimizationLogic = {
    MinimizeCost: "MinimizeCost",
    ModernizeToPaaS: "ModernizeToPaaS",
    ModernizeToAzureSqlMi: "ModernizeToAzureSqlMi",
    ModernizeToAzureSqlDb: "ModernizeToAzureSqlDb",
};
exports.OsLicense = {
    Unknown: "Unknown",
    Yes: "Yes",
    No: "No",
};
exports.Percentile = {
    /**
     * Percentile 50.
     */
    Percentile50: "Percentile50",
    /**
     * Percentile 90.
     */
    Percentile90: "Percentile90",
    /**
     * Percentile 95.
     */
    Percentile95: "Percentile95",
    /**
     * Percentile 99.
     */
    Percentile99: "Percentile99",
};
exports.PricingTier = {
    /**
     * Standard Pricing Tier.
     */
    Standard: "Standard",
    /**
     * Free Pricing Tier.
     */
    Free: "Free",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ProjectStatus = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.ProvisioningState = {
    /**
     * Resource has been created.
     */
    Succeeded: "Succeeded",
    /**
     * Resource creation failed.
     */
    Failed: "Failed",
    /**
     * Resource creation was canceled.
     */
    Canceled: "Canceled",
    /**
     * Resource is being Provisioned.
     */
    Provisioning: "Provisioning",
    /**
     * Resource is being Updated.
     */
    Updating: "Updating",
    /**
     * Resource is being Deleted.
     */
    Deleting: "Deleting",
    /**
     * Resource is being Accepted.
     */
    Accepted: "Accepted",
};
exports.Purpose = {
    Discovery: "Discovery",
    Assessment: "Assessment",
    Migration: "Migration",
};
exports.ReservedInstance = {
    None: "None",
    RI1Year: "RI1Year",
    RI3Year: "RI3Year",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.ResourceIdentityTypes = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.SavingsOption = {
    /**
     * Unknown Savings Option.
     */
    Unknown: "Unknown",
    /**
     * Reserved Instance 3 Year.
     */
    RI3Year: "RI3Year",
    /**
     * Azure Savings Plan 3 Year.
     */
    SavingsPlan3Year: "SavingsPlan3Year",
};
exports.SavingsOptions = {
    /**
     * Savings Options is not applicable.
     */
    None: "None",
    /**
     * One Year Savings Plan.
     */
    OneYearSavings: "OneYearSavings",
    /**
     * Three Years Savings Plan.
     */
    ThreeYearsSavings: "ThreeYearsSavings",
    /**
     * One Year Reserved Instances.
     */
    OneYearReserved: "OneYearReserved",
    /**
     * Three Years Reserved Instances.
     */
    ThreeYearsReserved: "ThreeYearsReserved",
};
exports.SecretStoreType = {
    None: "None",
    KubeSecret: "KubeSecret",
    KeyVaultSecret: "KeyVaultSecret",
    AppServiceAppSettings: "AppServiceAppSettings",
};
exports.SqlServerLicense = {
    Unknown: "Unknown",
    Yes: "Yes",
    No: "No",
};
exports.SqlServerLicenseType = {
    /**
     * Unknown Sql Server License.
     */
    Unknown: "Unknown",
    /**
     * Enterprise Sql Server License.
     */
    Enterprise: "Enterprise",
    /**
     * Standard Sql Server License.
     */
    Standard: "Standard",
};
exports.Status = {
    Inactive: "Inactive",
    Active: "Active",
};
exports.TargetAvailabilityZone = {
    One: "1",
    Two: "2",
    Three: "3",
    NA: "NA",
};
exports.TargetHydrationStorageProviderType = {
    AzureFileShare: "AzureFileShare",
};
exports.TargetStorageAccessType = {
    Shared: "Shared",
    Exclusive: "Exclusive",
};
exports.TargetStorageProjectionType = {
    ContainerFileSystem: "ContainerFileSystem",
    PersistentVolume: "PersistentVolume",
};
exports.TimeRange = {
    /**
     * Daily.
     */
    Day: "Day",
    /**
     * Weekly.
     */
    Week: "Week",
    /**
     * Monthly.
     */
    Month: "Month",
    /**
     * Custom Time Range.
     */
    Custom: "Custom",
};
exports.Tool = {
    ServerDiscovery: "ServerDiscovery",
    ServerAssessment: "ServerAssessment",
    ServerMigration: "ServerMigration",
    Cloudamize: "Cloudamize",
    Turbonomic: "Turbonomic",
    Zerto: "Zerto",
    CorentTech: "CorentTech",
    ServerAssessmentV1: "ServerAssessmentV1",
    ServerMigration_Replication: "ServerMigration_Replication",
    Carbonite: "Carbonite",
    DataMigrationAssistant: "DataMigrationAssistant",
    DatabaseMigrationService: "DatabaseMigrationService",
    Device42: "Device42",
    JetStream: "JetStream",
    RackWare: "RackWare",
    UnifyCloud: "UnifyCloud",
    Flexera: "Flexera",
    ServerDiscovery_Import: "ServerDiscovery_Import",
    Lakeside: "Lakeside",
    AppServiceMigrationAssistant: "AppServiceMigrationAssistant",
    Movere: "Movere",
    CloudSphere: "CloudSphere",
    Modernization: "Modernization",
    ServerMigration_DataReplication: "ServerMigration_DataReplication",
    Unknown: "Unknown",
};
exports.VsphereManagementLicenseType = {
    /**
     * Unknown License.
     */
    Unknown: "Unknown",
    /**
     * VSphereServerStandard License.
     */
    VSphereServerStandard: "VSphereServerStandard",
};
exports.WorkloadDeploymentTarget = {
    AzureKubernetesService: "AzureKubernetesService",
    AzureAppServiceContainer: "AzureAppServiceContainer",
    AzureAppServiceNative: "AzureAppServiceNative",
};
exports.WorkloadDeploymentType = {
    IISAKSWorkloadDeployment: "IISAKSWorkloadDeployment",
    ApacheTomcatAKSWorkloadDeployment: "ApacheTomcatAKSWorkloadDeployment",
};
exports.WorkloadType = {
    IISWorkload: "IISWorkload",
    ApacheTomcatWorkload: "ApacheTomcatWorkload",
};
exports.ZoneRedundant = {
    Enable: "Enable",
    Disable: "Disable",
};
