// @pulumi/azure-native/servicefabric/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = exports.NodeType = exports.ManagedCluster = exports.listNodeTypeFaultSimulationOutput = exports.listNodeTypeFaultSimulation = exports.listManagedClusterFaultSimulationOutput = exports.listManagedClusterFaultSimulation = exports.getmanagedMaintenanceWindowStatusOutput = exports.getmanagedMaintenanceWindowStatus = exports.getmanagedAzResiliencyStatusOutput = exports.getmanagedAzResiliencyStatus = exports.getServiceOutput = exports.getService = exports.getNodeTypeFaultSimulationOutput = exports.getNodeTypeFaultSimulation = exports.getNodeTypeOutput = exports.getNodeType = exports.getManagedClusterFaultSimulationOutput = exports.getManagedClusterFaultSimulation = exports.getManagedClusterOutput = exports.getManagedCluster = exports.getApplicationTypeVersionOutput = exports.getApplicationTypeVersion = exports.getApplicationTypeOutput = exports.getApplicationType = exports.getApplicationOutput = exports.getApplication = exports.ApplicationTypeVersion = exports.ApplicationType = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationType = null;
utilities.lazyLoad(exports, ["ApplicationType"], () => require("./applicationType"));
exports.ApplicationTypeVersion = null;
utilities.lazyLoad(exports, ["ApplicationTypeVersion"], () => require("./applicationTypeVersion"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getApplicationType = null;
exports.getApplicationTypeOutput = null;
utilities.lazyLoad(exports, ["getApplicationType", "getApplicationTypeOutput"], () => require("./getApplicationType"));
exports.getApplicationTypeVersion = null;
exports.getApplicationTypeVersionOutput = null;
utilities.lazyLoad(exports, ["getApplicationTypeVersion", "getApplicationTypeVersionOutput"], () => require("./getApplicationTypeVersion"));
exports.getManagedCluster = null;
exports.getManagedClusterOutput = null;
utilities.lazyLoad(exports, ["getManagedCluster", "getManagedClusterOutput"], () => require("./getManagedCluster"));
exports.getManagedClusterFaultSimulation = null;
exports.getManagedClusterFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterFaultSimulation", "getManagedClusterFaultSimulationOutput"], () => require("./getManagedClusterFaultSimulation"));
exports.getNodeType = null;
exports.getNodeTypeOutput = null;
utilities.lazyLoad(exports, ["getNodeType", "getNodeTypeOutput"], () => require("./getNodeType"));
exports.getNodeTypeFaultSimulation = null;
exports.getNodeTypeFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["getNodeTypeFaultSimulation", "getNodeTypeFaultSimulationOutput"], () => require("./getNodeTypeFaultSimulation"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getmanagedAzResiliencyStatus = null;
exports.getmanagedAzResiliencyStatusOutput = null;
utilities.lazyLoad(exports, ["getmanagedAzResiliencyStatus", "getmanagedAzResiliencyStatusOutput"], () => require("./getmanagedAzResiliencyStatus"));
exports.getmanagedMaintenanceWindowStatus = null;
exports.getmanagedMaintenanceWindowStatusOutput = null;
utilities.lazyLoad(exports, ["getmanagedMaintenanceWindowStatus", "getmanagedMaintenanceWindowStatusOutput"], () => require("./getmanagedMaintenanceWindowStatus"));
exports.listManagedClusterFaultSimulation = null;
exports.listManagedClusterFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterFaultSimulation", "listManagedClusterFaultSimulationOutput"], () => require("./listManagedClusterFaultSimulation"));
exports.listNodeTypeFaultSimulation = null;
exports.listNodeTypeFaultSimulationOutput = null;
utilities.lazyLoad(exports, ["listNodeTypeFaultSimulation", "listNodeTypeFaultSimulationOutput"], () => require("./listNodeTypeFaultSimulation"));
exports.ManagedCluster = null;
utilities.lazyLoad(exports, ["ManagedCluster"], () => require("./managedCluster"));
exports.NodeType = null;
utilities.lazyLoad(exports, ["NodeType"], () => require("./nodeType"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
// Export enums:
__exportStar(require("../../types/enums/servicefabric/v20241101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicefabric/v20241101preview:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ApplicationType":
                return new exports.ApplicationType(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ApplicationTypeVersion":
                return new exports.ApplicationTypeVersion(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:ManagedCluster":
                return new exports.ManagedCluster(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:NodeType":
                return new exports.NodeType(name, undefined, { urn });
            case "azure-native:servicefabric/v20241101preview:Service":
                return new exports.Service(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicefabric/v20241101preview", _module);
