// @pulumi/azure-native/servicefabric/v20241101preview/managedCluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * The managed cluster resource
 */
class ManagedCluster extends pulumi.CustomResource {
    /**
     * Get an existing ManagedCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedCluster.__pulumiType;
    }
    /**
     * Create a ManagedCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.adminUserName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'adminUserName'");
            }
            if ((!args || args.dnsName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dnsName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["addonFeatures"] = args ? args.addonFeatures : undefined;
            resourceInputs["adminPassword"] = args ? args.adminPassword : undefined;
            resourceInputs["adminUserName"] = args ? args.adminUserName : undefined;
            resourceInputs["allocatedOutboundPorts"] = args ? args.allocatedOutboundPorts : undefined;
            resourceInputs["allowRdpAccess"] = args ? args.allowRdpAccess : undefined;
            resourceInputs["applicationTypeVersionsCleanupPolicy"] = args ? args.applicationTypeVersionsCleanupPolicy : undefined;
            resourceInputs["autoGeneratedDomainNameLabelScope"] = args ? args.autoGeneratedDomainNameLabelScope : undefined;
            resourceInputs["auxiliarySubnets"] = args ? args.auxiliarySubnets : undefined;
            resourceInputs["azureActiveDirectory"] = args ? args.azureActiveDirectory : undefined;
            resourceInputs["clientConnectionPort"] = (_a = (args ? args.clientConnectionPort : undefined)) !== null && _a !== void 0 ? _a : 19000;
            resourceInputs["clients"] = args ? args.clients : undefined;
            resourceInputs["clusterCodeVersion"] = args ? args.clusterCodeVersion : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["clusterUpgradeCadence"] = args ? args.clusterUpgradeCadence : undefined;
            resourceInputs["clusterUpgradeMode"] = (_b = (args ? args.clusterUpgradeMode : undefined)) !== null && _b !== void 0 ? _b : "Automatic";
            resourceInputs["ddosProtectionPlanId"] = args ? args.ddosProtectionPlanId : undefined;
            resourceInputs["dnsName"] = args ? args.dnsName : undefined;
            resourceInputs["enableAutoOSUpgrade"] = args ? args.enableAutoOSUpgrade : undefined;
            resourceInputs["enableHttpGatewayExclusiveAuthMode"] = args ? args.enableHttpGatewayExclusiveAuthMode : undefined;
            resourceInputs["enableIpv6"] = args ? args.enableIpv6 : undefined;
            resourceInputs["enableServicePublicIP"] = args ? args.enableServicePublicIP : undefined;
            resourceInputs["fabricSettings"] = args ? args.fabricSettings : undefined;
            resourceInputs["httpGatewayConnectionPort"] = (_c = (args ? args.httpGatewayConnectionPort : undefined)) !== null && _c !== void 0 ? _c : 19080;
            resourceInputs["httpGatewayTokenAuthConnectionPort"] = args ? args.httpGatewayTokenAuthConnectionPort : undefined;
            resourceInputs["ipTags"] = args ? args.ipTags : undefined;
            resourceInputs["loadBalancingRules"] = args ? args.loadBalancingRules : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkSecurityRules"] = args ? args.networkSecurityRules : undefined;
            resourceInputs["publicIPPrefixId"] = args ? args.publicIPPrefixId : undefined;
            resourceInputs["publicIPv6PrefixId"] = args ? args.publicIPv6PrefixId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceEndpoints"] = args ? args.serviceEndpoints : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["subnetId"] = args ? args.subnetId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["upgradeDescription"] = args ? (args.upgradeDescription ? pulumi.output(args.upgradeDescription).apply(inputs.servicefabric.v20241101preview.clusterUpgradePolicyArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["useCustomVnet"] = args ? args.useCustomVnet : undefined;
            resourceInputs["zonalResiliency"] = (_d = (args ? args.zonalResiliency : undefined)) !== null && _d !== void 0 ? _d : false;
            resourceInputs["zonalUpdateMode"] = args ? args.zonalUpdateMode : undefined;
            resourceInputs["clusterCertificateThumbprints"] = undefined /*out*/;
            resourceInputs["clusterId"] = undefined /*out*/;
            resourceInputs["clusterState"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["ipv4Address"] = undefined /*out*/;
            resourceInputs["ipv6Address"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["addonFeatures"] = undefined /*out*/;
            resourceInputs["adminPassword"] = undefined /*out*/;
            resourceInputs["adminUserName"] = undefined /*out*/;
            resourceInputs["allocatedOutboundPorts"] = undefined /*out*/;
            resourceInputs["allowRdpAccess"] = undefined /*out*/;
            resourceInputs["applicationTypeVersionsCleanupPolicy"] = undefined /*out*/;
            resourceInputs["autoGeneratedDomainNameLabelScope"] = undefined /*out*/;
            resourceInputs["auxiliarySubnets"] = undefined /*out*/;
            resourceInputs["azureActiveDirectory"] = undefined /*out*/;
            resourceInputs["clientConnectionPort"] = undefined /*out*/;
            resourceInputs["clients"] = undefined /*out*/;
            resourceInputs["clusterCertificateThumbprints"] = undefined /*out*/;
            resourceInputs["clusterCodeVersion"] = undefined /*out*/;
            resourceInputs["clusterId"] = undefined /*out*/;
            resourceInputs["clusterState"] = undefined /*out*/;
            resourceInputs["clusterUpgradeCadence"] = undefined /*out*/;
            resourceInputs["clusterUpgradeMode"] = undefined /*out*/;
            resourceInputs["ddosProtectionPlanId"] = undefined /*out*/;
            resourceInputs["dnsName"] = undefined /*out*/;
            resourceInputs["enableAutoOSUpgrade"] = undefined /*out*/;
            resourceInputs["enableHttpGatewayExclusiveAuthMode"] = undefined /*out*/;
            resourceInputs["enableIpv6"] = undefined /*out*/;
            resourceInputs["enableServicePublicIP"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fabricSettings"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["httpGatewayConnectionPort"] = undefined /*out*/;
            resourceInputs["httpGatewayTokenAuthConnectionPort"] = undefined /*out*/;
            resourceInputs["ipTags"] = undefined /*out*/;
            resourceInputs["ipv4Address"] = undefined /*out*/;
            resourceInputs["ipv6Address"] = undefined /*out*/;
            resourceInputs["loadBalancingRules"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkSecurityRules"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicIPPrefixId"] = undefined /*out*/;
            resourceInputs["publicIPv6PrefixId"] = undefined /*out*/;
            resourceInputs["serviceEndpoints"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["subnetId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upgradeDescription"] = undefined /*out*/;
            resourceInputs["useCustomVnet"] = undefined /*out*/;
            resourceInputs["zonalResiliency"] = undefined /*out*/;
            resourceInputs["zonalUpdateMode"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabric/v20200101preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20210101preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20210501:ManagedCluster" }, { type: "azure-native:servicefabric/v20210701preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20210901privatepreview:ManagedCluster" }, { type: "azure-native:servicefabric/v20211101preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20220101:ManagedCluster" }, { type: "azure-native:servicefabric/v20220201preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20220601preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20220801preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20221001preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20230201preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20230301preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20230701preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20230901preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20231101preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20231201preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20240201preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20240401:ManagedCluster" }, { type: "azure-native:servicefabric/v20240601preview:ManagedCluster" }, { type: "azure-native:servicefabric/v20240901preview:ManagedCluster" }, { type: "azure-native:servicefabric:ManagedCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedCluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagedCluster = ManagedCluster;
/** @internal */
ManagedCluster.__pulumiType = 'azure-native:servicefabric/v20241101preview:ManagedCluster';
