// @pulumi/azure-native/servicefabric/v20241101preview/nodeType.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a node type in the cluster, each node type represents sub set of nodes in the cluster.
 */
class NodeType extends pulumi.CustomResource {
    /**
     * Get an existing NodeType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NodeType(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NodeType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NodeType.__pulumiType;
    }
    /**
     * Create a NodeType resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.isPrimary === undefined) && !opts.urn) {
                throw new Error("Missing required property 'isPrimary'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vmInstanceCount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vmInstanceCount'");
            }
            resourceInputs["additionalDataDisks"] = args ? args.additionalDataDisks : undefined;
            resourceInputs["additionalNetworkInterfaceConfigurations"] = args ? args.additionalNetworkInterfaceConfigurations : undefined;
            resourceInputs["applicationPorts"] = args ? args.applicationPorts : undefined;
            resourceInputs["capacities"] = args ? args.capacities : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["computerNamePrefix"] = args ? args.computerNamePrefix : undefined;
            resourceInputs["dataDiskLetter"] = args ? args.dataDiskLetter : undefined;
            resourceInputs["dataDiskSizeGB"] = args ? args.dataDiskSizeGB : undefined;
            resourceInputs["dataDiskType"] = (_a = (args ? args.dataDiskType : undefined)) !== null && _a !== void 0 ? _a : "StandardSSD_LRS";
            resourceInputs["dscpConfigurationId"] = args ? args.dscpConfigurationId : undefined;
            resourceInputs["enableAcceleratedNetworking"] = args ? args.enableAcceleratedNetworking : undefined;
            resourceInputs["enableEncryptionAtHost"] = (_b = (args ? args.enableEncryptionAtHost : undefined)) !== null && _b !== void 0 ? _b : false;
            resourceInputs["enableNodePublicIP"] = args ? args.enableNodePublicIP : undefined;
            resourceInputs["enableNodePublicIPv6"] = args ? args.enableNodePublicIPv6 : undefined;
            resourceInputs["enableOverProvisioning"] = args ? args.enableOverProvisioning : undefined;
            resourceInputs["ephemeralPorts"] = args ? args.ephemeralPorts : undefined;
            resourceInputs["evictionPolicy"] = args ? args.evictionPolicy : undefined;
            resourceInputs["frontendConfigurations"] = args ? args.frontendConfigurations : undefined;
            resourceInputs["hostGroupId"] = args ? args.hostGroupId : undefined;
            resourceInputs["isPrimary"] = args ? args.isPrimary : undefined;
            resourceInputs["isSpotVM"] = args ? args.isSpotVM : undefined;
            resourceInputs["isStateless"] = (_c = (args ? args.isStateless : undefined)) !== null && _c !== void 0 ? _c : false;
            resourceInputs["multiplePlacementGroups"] = (_d = (args ? args.multiplePlacementGroups : undefined)) !== null && _d !== void 0 ? _d : false;
            resourceInputs["natConfigurations"] = args ? args.natConfigurations : undefined;
            resourceInputs["natGatewayId"] = args ? args.natGatewayId : undefined;
            resourceInputs["networkSecurityRules"] = args ? args.networkSecurityRules : undefined;
            resourceInputs["nodeTypeName"] = args ? args.nodeTypeName : undefined;
            resourceInputs["placementProperties"] = args ? args.placementProperties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["secureBootEnabled"] = args ? args.secureBootEnabled : undefined;
            resourceInputs["securityType"] = args ? args.securityType : undefined;
            resourceInputs["serviceArtifactReferenceId"] = args ? args.serviceArtifactReferenceId : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["spotRestoreTimeout"] = args ? args.spotRestoreTimeout : undefined;
            resourceInputs["subnetId"] = args ? args.subnetId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["useDefaultPublicLoadBalancer"] = args ? args.useDefaultPublicLoadBalancer : undefined;
            resourceInputs["useEphemeralOSDisk"] = args ? args.useEphemeralOSDisk : undefined;
            resourceInputs["useTempDataDisk"] = args ? args.useTempDataDisk : undefined;
            resourceInputs["vmApplications"] = args ? args.vmApplications : undefined;
            resourceInputs["vmExtensions"] = args ? args.vmExtensions : undefined;
            resourceInputs["vmImageOffer"] = args ? args.vmImageOffer : undefined;
            resourceInputs["vmImagePlan"] = args ? args.vmImagePlan : undefined;
            resourceInputs["vmImagePublisher"] = args ? args.vmImagePublisher : undefined;
            resourceInputs["vmImageResourceId"] = args ? args.vmImageResourceId : undefined;
            resourceInputs["vmImageSku"] = args ? args.vmImageSku : undefined;
            resourceInputs["vmImageVersion"] = args ? args.vmImageVersion : undefined;
            resourceInputs["vmInstanceCount"] = args ? args.vmInstanceCount : undefined;
            resourceInputs["vmManagedIdentity"] = args ? args.vmManagedIdentity : undefined;
            resourceInputs["vmSecrets"] = args ? args.vmSecrets : undefined;
            resourceInputs["vmSetupActions"] = args ? args.vmSetupActions : undefined;
            resourceInputs["vmSharedGalleryImageId"] = args ? args.vmSharedGalleryImageId : undefined;
            resourceInputs["vmSize"] = args ? args.vmSize : undefined;
            resourceInputs["zones"] = args ? args.zones : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalDataDisks"] = undefined /*out*/;
            resourceInputs["additionalNetworkInterfaceConfigurations"] = undefined /*out*/;
            resourceInputs["applicationPorts"] = undefined /*out*/;
            resourceInputs["capacities"] = undefined /*out*/;
            resourceInputs["computerNamePrefix"] = undefined /*out*/;
            resourceInputs["dataDiskLetter"] = undefined /*out*/;
            resourceInputs["dataDiskSizeGB"] = undefined /*out*/;
            resourceInputs["dataDiskType"] = undefined /*out*/;
            resourceInputs["dscpConfigurationId"] = undefined /*out*/;
            resourceInputs["enableAcceleratedNetworking"] = undefined /*out*/;
            resourceInputs["enableEncryptionAtHost"] = undefined /*out*/;
            resourceInputs["enableNodePublicIP"] = undefined /*out*/;
            resourceInputs["enableNodePublicIPv6"] = undefined /*out*/;
            resourceInputs["enableOverProvisioning"] = undefined /*out*/;
            resourceInputs["ephemeralPorts"] = undefined /*out*/;
            resourceInputs["evictionPolicy"] = undefined /*out*/;
            resourceInputs["frontendConfigurations"] = undefined /*out*/;
            resourceInputs["hostGroupId"] = undefined /*out*/;
            resourceInputs["isPrimary"] = undefined /*out*/;
            resourceInputs["isSpotVM"] = undefined /*out*/;
            resourceInputs["isStateless"] = undefined /*out*/;
            resourceInputs["multiplePlacementGroups"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["natConfigurations"] = undefined /*out*/;
            resourceInputs["natGatewayId"] = undefined /*out*/;
            resourceInputs["networkSecurityRules"] = undefined /*out*/;
            resourceInputs["placementProperties"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["secureBootEnabled"] = undefined /*out*/;
            resourceInputs["securityType"] = undefined /*out*/;
            resourceInputs["serviceArtifactReferenceId"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["spotRestoreTimeout"] = undefined /*out*/;
            resourceInputs["subnetId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useDefaultPublicLoadBalancer"] = undefined /*out*/;
            resourceInputs["useEphemeralOSDisk"] = undefined /*out*/;
            resourceInputs["useTempDataDisk"] = undefined /*out*/;
            resourceInputs["vmApplications"] = undefined /*out*/;
            resourceInputs["vmExtensions"] = undefined /*out*/;
            resourceInputs["vmImageOffer"] = undefined /*out*/;
            resourceInputs["vmImagePlan"] = undefined /*out*/;
            resourceInputs["vmImagePublisher"] = undefined /*out*/;
            resourceInputs["vmImageResourceId"] = undefined /*out*/;
            resourceInputs["vmImageSku"] = undefined /*out*/;
            resourceInputs["vmImageVersion"] = undefined /*out*/;
            resourceInputs["vmInstanceCount"] = undefined /*out*/;
            resourceInputs["vmManagedIdentity"] = undefined /*out*/;
            resourceInputs["vmSecrets"] = undefined /*out*/;
            resourceInputs["vmSetupActions"] = undefined /*out*/;
            resourceInputs["vmSharedGalleryImageId"] = undefined /*out*/;
            resourceInputs["vmSize"] = undefined /*out*/;
            resourceInputs["zones"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabric/v20200101preview:NodeType" }, { type: "azure-native:servicefabric/v20210101preview:NodeType" }, { type: "azure-native:servicefabric/v20210501:NodeType" }, { type: "azure-native:servicefabric/v20210701preview:NodeType" }, { type: "azure-native:servicefabric/v20210901privatepreview:NodeType" }, { type: "azure-native:servicefabric/v20211101preview:NodeType" }, { type: "azure-native:servicefabric/v20220101:NodeType" }, { type: "azure-native:servicefabric/v20220201preview:NodeType" }, { type: "azure-native:servicefabric/v20220601preview:NodeType" }, { type: "azure-native:servicefabric/v20220801preview:NodeType" }, { type: "azure-native:servicefabric/v20221001preview:NodeType" }, { type: "azure-native:servicefabric/v20230201preview:NodeType" }, { type: "azure-native:servicefabric/v20230301preview:NodeType" }, { type: "azure-native:servicefabric/v20230701preview:NodeType" }, { type: "azure-native:servicefabric/v20230901preview:NodeType" }, { type: "azure-native:servicefabric/v20231101preview:NodeType" }, { type: "azure-native:servicefabric/v20231201preview:NodeType" }, { type: "azure-native:servicefabric/v20240201preview:NodeType" }, { type: "azure-native:servicefabric/v20240401:NodeType" }, { type: "azure-native:servicefabric/v20240601preview:NodeType" }, { type: "azure-native:servicefabric/v20240901preview:NodeType" }, { type: "azure-native:servicefabric:NodeType" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NodeType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NodeType = NodeType;
/** @internal */
NodeType.__pulumiType = 'azure-native:servicefabric/v20241101preview:NodeType';
