// @pulumi/azure-native/securityinsights/v20240101preview/microsoftPurviewInformationProtectionDataConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MicrosoftPurviewInformationProtectionDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents Microsoft Purview Information Protection data connector.
 */
class MicrosoftPurviewInformationProtectionDataConnector extends pulumi.CustomResource {
    /**
     * Get an existing MicrosoftPurviewInformationProtectionDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MicrosoftPurviewInformationProtectionDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MicrosoftPurviewInformationProtectionDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MicrosoftPurviewInformationProtectionDataConnector.__pulumiType;
    }
    /**
     * Create a MicrosoftPurviewInformationProtectionDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataTypes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataTypes'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.tenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tenantId'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["dataTypes"] = args ? args.dataTypes : undefined;
            resourceInputs["kind"] = "MicrosoftPurviewInformationProtection";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataTypes"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20200101:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20211001:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220501preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220601preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220701preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220801:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220801preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20220901preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20221001preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20221101:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20221101preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20221201preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230201:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230201preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230301preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230401preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230501preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230601preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230701preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230801preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20230901preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20231001preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20231101:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20231201preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20240301:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20240401preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20240901:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20241001preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20250101preview:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights/v20250301:MicrosoftPurviewInformationProtectionDataConnector" }, { type: "azure-native:securityinsights:MicrosoftPurviewInformationProtectionDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MicrosoftPurviewInformationProtectionDataConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MicrosoftPurviewInformationProtectionDataConnector = MicrosoftPurviewInformationProtectionDataConnector;
/** @internal */
MicrosoftPurviewInformationProtectionDataConnector.__pulumiType = 'azure-native:securityinsights/v20240101preview:MicrosoftPurviewInformationProtectionDataConnector';
