// @pulumi/azure-native/kubernetesruntime/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240301 = exports.v20231001preview = exports.StorageClass = exports.Service = exports.LoadBalancer = exports.getStorageClassOutput = exports.getStorageClass = exports.getServiceOutput = exports.getService = exports.getLoadBalancerOutput = exports.getLoadBalancer = exports.getBgpPeerOutput = exports.getBgpPeer = exports.BgpPeer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BgpPeer = null;
utilities.lazyLoad(exports, ["BgpPeer"], () => require("./bgpPeer"));
exports.getBgpPeer = null;
exports.getBgpPeerOutput = null;
utilities.lazyLoad(exports, ["getBgpPeer", "getBgpPeerOutput"], () => require("./getBgpPeer"));
exports.getLoadBalancer = null;
exports.getLoadBalancerOutput = null;
utilities.lazyLoad(exports, ["getLoadBalancer", "getLoadBalancerOutput"], () => require("./getLoadBalancer"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getStorageClass = null;
exports.getStorageClassOutput = null;
utilities.lazyLoad(exports, ["getStorageClass", "getStorageClassOutput"], () => require("./getStorageClass"));
exports.LoadBalancer = null;
utilities.lazyLoad(exports, ["LoadBalancer"], () => require("./loadBalancer"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.StorageClass = null;
utilities.lazyLoad(exports, ["StorageClass"], () => require("./storageClass"));
// Export enums:
__exportStar(require("../types/enums/kubernetesruntime"), exports);
// Export sub-modules:
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetesruntime:BgpPeer":
                return new exports.BgpPeer(name, undefined, { urn });
            case "azure-native:kubernetesruntime:LoadBalancer":
                return new exports.LoadBalancer(name, undefined, { urn });
            case "azure-native:kubernetesruntime:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:kubernetesruntime:StorageClass":
                return new exports.StorageClass(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetesruntime", _module);
