// @pulumi/azure-native/domainregistration/v20230101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTopLevelDomainAgreementsOutput = exports.listTopLevelDomainAgreements = exports.listDomainRecommendationsOutput = exports.listDomainRecommendations = exports.getDomainOwnershipIdentifierOutput = exports.getDomainOwnershipIdentifier = exports.getDomainOutput = exports.getDomain = exports.DomainOwnershipIdentifier = exports.Domain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.DomainOwnershipIdentifier = null;
utilities.lazyLoad(exports, ["DomainOwnershipIdentifier"], () => require("./domainOwnershipIdentifier"));
exports.getDomain = null;
exports.getDomainOutput = null;
utilities.lazyLoad(exports, ["getDomain", "getDomainOutput"], () => require("./getDomain"));
exports.getDomainOwnershipIdentifier = null;
exports.getDomainOwnershipIdentifierOutput = null;
utilities.lazyLoad(exports, ["getDomainOwnershipIdentifier", "getDomainOwnershipIdentifierOutput"], () => require("./getDomainOwnershipIdentifier"));
exports.listDomainRecommendations = null;
exports.listDomainRecommendationsOutput = null;
utilities.lazyLoad(exports, ["listDomainRecommendations", "listDomainRecommendationsOutput"], () => require("./listDomainRecommendations"));
exports.listTopLevelDomainAgreements = null;
exports.listTopLevelDomainAgreementsOutput = null;
utilities.lazyLoad(exports, ["listTopLevelDomainAgreements", "listTopLevelDomainAgreementsOutput"], () => require("./listTopLevelDomainAgreements"));
// Export enums:
__exportStar(require("../../types/enums/domainregistration/v20230101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:domainregistration/v20230101:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "azure-native:domainregistration/v20230101:DomainOwnershipIdentifier":
                return new exports.DomainOwnershipIdentifier(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "domainregistration/v20230101", _module);
