// @pulumi/azure-native/types/enums/hybridcontainerservice/v20240101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OsType = exports.OSSKU = exports.NetworkPolicy = exports.ExtendedLocationTypes = exports.Expander = exports.AzureHybridBenefit = void 0;
exports.AzureHybridBenefit = {
    True: "True",
    False: "False",
    NotApplicable: "NotApplicable",
};
exports.Expander = {
    /**
     * Selects the node group that will have the least idle CPU (if tied, unused memory) after scale-up. This is useful when you have different classes of nodes, for example, high CPU or high memory nodes, and only want to expand those when there are pending pods that need a lot of those resources.
     */
    Least_waste: "least-waste",
    /**
     * Selects the node group that would be able to schedule the most pods when scaling up. This is useful when you are using nodeSelector to make sure certain pods land on certain nodes. Note that this won't cause the autoscaler to select bigger nodes vs. smaller, as it can add multiple smaller nodes at once.
     */
    Most_pods: "most-pods",
    /**
     * Selects the node group that has the highest priority assigned by the user. It's configuration is described in more details [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/expander/priority/readme.md).
     */
    Priority: "priority",
    /**
     * Used when you don't have a particular need for the node groups to scale differently.
     */
    Random: "random",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.NetworkPolicy = {
    Calico: "calico",
};
exports.OSSKU = {
    /**
     * Use Mariner as the OS for node images.
     */
    CBLMariner: "CBLMariner",
    /**
     * Use Windows2019 as the OS for node images.
     */
    Windows2019: "Windows2019",
    /**
     * Use Windows2022 as the OS for node images.
     */
    Windows2022: "Windows2022",
};
exports.OsType = {
    Linux: "Linux",
    Windows: "Windows",
};
