// @pulumi/azure-native/hybridcontainerservice/v20220501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisionedCluster = exports.HybridIdentityMetadatum = exports.getProvisionedClusterOutput = exports.getProvisionedCluster = exports.getHybridIdentityMetadatumOutput = exports.getHybridIdentityMetadatum = exports.getAgentPoolOutput = exports.getAgentPool = exports.AgentPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getHybridIdentityMetadatum = null;
exports.getHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getHybridIdentityMetadatum", "getHybridIdentityMetadatumOutput"], () => require("./getHybridIdentityMetadatum"));
exports.getProvisionedCluster = null;
exports.getProvisionedClusterOutput = null;
utilities.lazyLoad(exports, ["getProvisionedCluster", "getProvisionedClusterOutput"], () => require("./getProvisionedCluster"));
exports.HybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["HybridIdentityMetadatum"], () => require("./hybridIdentityMetadatum"));
exports.ProvisionedCluster = null;
utilities.lazyLoad(exports, ["ProvisionedCluster"], () => require("./provisionedCluster"));
// Export enums:
__exportStar(require("../../types/enums/hybridcontainerservice/v20220501preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcontainerservice/v20220501preview:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20220501preview:HybridIdentityMetadatum":
                return new exports.HybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20220501preview:ProvisionedCluster":
                return new exports.ProvisionedCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcontainerservice/v20220501preview", _module);
