// @pulumi/azure-native/confidentialledger/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240919preview = exports.v20240709preview = exports.v20230628preview = exports.v20230126preview = exports.v20220513 = exports.ManagedCCF = exports.Ledger = exports.getManagedCCFOutput = exports.getManagedCCF = exports.getLedgerOutput = exports.getLedger = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getLedger = null;
exports.getLedgerOutput = null;
utilities.lazyLoad(exports, ["getLedger", "getLedgerOutput"], () => require("./getLedger"));
exports.getManagedCCF = null;
exports.getManagedCCFOutput = null;
utilities.lazyLoad(exports, ["getManagedCCF", "getManagedCCFOutput"], () => require("./getManagedCCF"));
exports.Ledger = null;
utilities.lazyLoad(exports, ["Ledger"], () => require("./ledger"));
exports.ManagedCCF = null;
utilities.lazyLoad(exports, ["ManagedCCF"], () => require("./managedCCF"));
// Export enums:
__exportStar(require("../types/enums/confidentialledger"), exports);
// Export sub-modules:
const v20220513 = require("./v20220513");
exports.v20220513 = v20220513;
const v20230126preview = require("./v20230126preview");
exports.v20230126preview = v20230126preview;
const v20230628preview = require("./v20230628preview");
exports.v20230628preview = v20230628preview;
const v20240709preview = require("./v20240709preview");
exports.v20240709preview = v20240709preview;
const v20240919preview = require("./v20240919preview");
exports.v20240919preview = v20240919preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:confidentialledger:Ledger":
                return new exports.Ledger(name, undefined, { urn });
            case "azure-native:confidentialledger:ManagedCCF":
                return new exports.ManagedCCF(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "confidentialledger", _module);
