// @pulumi/azure-native/types/enums/fluidrelay/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageSKU = exports.ResourceIdentityType = exports.ProvisioningState = exports.CmkIdentityType = exports.v20220601 = exports.v20210615preview = void 0;
// Export sub-modules:
const v20210615preview = require("./v20210615preview");
exports.v20210615preview = v20210615preview;
const v20220601 = require("./v20220601");
exports.v20220601 = v20220601;
exports.CmkIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.StorageSKU = {
    Standard: "standard",
    Basic: "basic",
};
