// @pulumi/azure-native/types/enums/maps/v20231201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningKey = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.Name = exports.ManagedServiceIdentityType = exports.Kind = exports.InfrastructureEncryption = void 0;
exports.InfrastructureEncryption = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.Kind = {
    Gen2: "Gen2",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.Name = {
    G2: "G2",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.SigningKey = {
    PrimaryKey: "primaryKey",
    SecondaryKey: "secondaryKey",
    ManagedIdentity: "managedIdentity",
};
