// @pulumi/azure-native/community/communityTraining.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunityTraining = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * A CommunityProviderHub resource
 *
 * Uses Azure REST API version 2023-11-01.
 */
class CommunityTraining extends pulumi.CustomResource {
    /**
     * Get an existing CommunityTraining resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CommunityTraining(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CommunityTraining.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CommunityTraining.__pulumiType;
    }
    /**
     * Create a CommunityTraining resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.disasterRecoveryEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'disasterRecoveryEnabled'");
            }
            if ((!args || args.identityConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identityConfiguration'");
            }
            if ((!args || args.portalAdminEmailAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'portalAdminEmailAddress'");
            }
            if ((!args || args.portalName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'portalName'");
            }
            if ((!args || args.portalOwnerEmailAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'portalOwnerEmailAddress'");
            }
            if ((!args || args.portalOwnerOrganizationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'portalOwnerOrganizationName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.zoneRedundancyEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'zoneRedundancyEnabled'");
            }
            resourceInputs["communityTrainingName"] = args ? args.communityTrainingName : undefined;
            resourceInputs["disasterRecoveryEnabled"] = args ? args.disasterRecoveryEnabled : undefined;
            resourceInputs["identityConfiguration"] = args ? (args.identityConfiguration ? pulumi.output(args.identityConfiguration).apply(inputs.community.identityConfigurationPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["portalAdminEmailAddress"] = args ? args.portalAdminEmailAddress : undefined;
            resourceInputs["portalName"] = args ? args.portalName : undefined;
            resourceInputs["portalOwnerEmailAddress"] = args ? args.portalOwnerEmailAddress : undefined;
            resourceInputs["portalOwnerOrganizationName"] = args ? args.portalOwnerOrganizationName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["zoneRedundancyEnabled"] = args ? args.zoneRedundancyEnabled : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["disasterRecoveryEnabled"] = undefined /*out*/;
            resourceInputs["identityConfiguration"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["portalAdminEmailAddress"] = undefined /*out*/;
            resourceInputs["portalName"] = undefined /*out*/;
            resourceInputs["portalOwnerEmailAddress"] = undefined /*out*/;
            resourceInputs["portalOwnerOrganizationName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["zoneRedundancyEnabled"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:community/v20231101:CommunityTraining" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CommunityTraining.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CommunityTraining = CommunityTraining;
/** @internal */
CommunityTraining.__pulumiType = 'azure-native:community:CommunityTraining';
