// @pulumi/azure-native/cdn/v20200901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Route = exports.Profile = exports.getRouteOutput = exports.getRoute = exports.getProfileSupportedOptimizationTypesOutput = exports.getProfileSupportedOptimizationTypes = exports.getProfileOutput = exports.getProfile = exports.getAFDOriginGroupOutput = exports.getAFDOriginGroup = exports.getAFDEndpointOutput = exports.getAFDEndpoint = exports.AFDOriginGroup = exports.AFDEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AFDEndpoint = null;
utilities.lazyLoad(exports, ["AFDEndpoint"], () => require("./afdendpoint"));
exports.AFDOriginGroup = null;
utilities.lazyLoad(exports, ["AFDOriginGroup"], () => require("./afdoriginGroup"));
exports.getAFDEndpoint = null;
exports.getAFDEndpointOutput = null;
utilities.lazyLoad(exports, ["getAFDEndpoint", "getAFDEndpointOutput"], () => require("./getAFDEndpoint"));
exports.getAFDOriginGroup = null;
exports.getAFDOriginGroupOutput = null;
utilities.lazyLoad(exports, ["getAFDOriginGroup", "getAFDOriginGroupOutput"], () => require("./getAFDOriginGroup"));
exports.getProfile = null;
exports.getProfileOutput = null;
utilities.lazyLoad(exports, ["getProfile", "getProfileOutput"], () => require("./getProfile"));
exports.getProfileSupportedOptimizationTypes = null;
exports.getProfileSupportedOptimizationTypesOutput = null;
utilities.lazyLoad(exports, ["getProfileSupportedOptimizationTypes", "getProfileSupportedOptimizationTypesOutput"], () => require("./getProfileSupportedOptimizationTypes"));
exports.getRoute = null;
exports.getRouteOutput = null;
utilities.lazyLoad(exports, ["getRoute", "getRouteOutput"], () => require("./getRoute"));
exports.Profile = null;
utilities.lazyLoad(exports, ["Profile"], () => require("./profile"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
// Export enums:
__exportStar(require("../../types/enums/cdn/v20200901"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cdn/v20200901:AFDEndpoint":
                return new exports.AFDEndpoint(name, undefined, { urn });
            case "azure-native:cdn/v20200901:AFDOriginGroup":
                return new exports.AFDOriginGroup(name, undefined, { urn });
            case "azure-native:cdn/v20200901:Profile":
                return new exports.Profile(name, undefined, { urn });
            case "azure-native:cdn/v20200901:Route":
                return new exports.Route(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cdn/v20200901", _module);
