// @pulumi/azure-native/hybridcloud/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230101preview = exports.getCloudConnectorOutput = exports.getCloudConnector = exports.getCloudConnectionOutput = exports.getCloudConnection = exports.CloudConnector = exports.CloudConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CloudConnection = null;
utilities.lazyLoad(exports, ["CloudConnection"], () => require("./cloudConnection"));
exports.CloudConnector = null;
utilities.lazyLoad(exports, ["CloudConnector"], () => require("./cloudConnector"));
exports.getCloudConnection = null;
exports.getCloudConnectionOutput = null;
utilities.lazyLoad(exports, ["getCloudConnection", "getCloudConnectionOutput"], () => require("./getCloudConnection"));
exports.getCloudConnector = null;
exports.getCloudConnectorOutput = null;
utilities.lazyLoad(exports, ["getCloudConnector", "getCloudConnectorOutput"], () => require("./getCloudConnector"));
// Export enums:
__exportStar(require("../types/enums/hybridcloud"), exports);
// Export sub-modules:
const v20230101preview = require("./v20230101preview");
exports.v20230101preview = v20230101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcloud:CloudConnection":
                return new exports.CloudConnection(name, undefined, { urn });
            case "azure-native:hybridcloud:CloudConnector":
                return new exports.CloudConnector(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcloud", _module);
