// @pulumi/azure-native/intune/v20150114privatepreview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoMAMPolicyByName = exports.getIoMAMPolicyByNameOutput = exports.getIoMAMPolicyByName = exports.getAndroidMAMPolicyByNameOutput = exports.getAndroidMAMPolicyByName = exports.AndroidMAMPolicyByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AndroidMAMPolicyByName = null;
utilities.lazyLoad(exports, ["AndroidMAMPolicyByName"], () => require("./androidMAMPolicyByName"));
exports.getAndroidMAMPolicyByName = null;
exports.getAndroidMAMPolicyByNameOutput = null;
utilities.lazyLoad(exports, ["getAndroidMAMPolicyByName", "getAndroidMAMPolicyByNameOutput"], () => require("./getAndroidMAMPolicyByName"));
exports.getIoMAMPolicyByName = null;
exports.getIoMAMPolicyByNameOutput = null;
utilities.lazyLoad(exports, ["getIoMAMPolicyByName", "getIoMAMPolicyByNameOutput"], () => require("./getIoMAMPolicyByName"));
exports.IoMAMPolicyByName = null;
utilities.lazyLoad(exports, ["IoMAMPolicyByName"], () => require("./ioMAMPolicyByName"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:intune/v20150114privatepreview:AndroidMAMPolicyByName":
                return new exports.AndroidMAMPolicyByName(name, undefined, { urn });
            case "azure-native:intune/v20150114privatepreview:IoMAMPolicyByName":
                return new exports.IoMAMPolicyByName(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "intune/v20150114privatepreview", _module);
