// @pulumi/azure-native/orbital/v20240301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listGroundStationL2ConnectionsOutput = exports.listGroundStationL2Connections = exports.listEdgeSiteL2ConnectionsOutput = exports.listEdgeSiteL2Connections = exports.L2Connection = exports.GroundStation = exports.getL2ConnectionOutput = exports.getL2Connection = exports.getGroundStationOutput = exports.getGroundStation = exports.getEdgeSiteOutput = exports.getEdgeSite = exports.EdgeSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.EdgeSite = null;
utilities.lazyLoad(exports, ["EdgeSite"], () => require("./edgeSite"));
exports.getEdgeSite = null;
exports.getEdgeSiteOutput = null;
utilities.lazyLoad(exports, ["getEdgeSite", "getEdgeSiteOutput"], () => require("./getEdgeSite"));
exports.getGroundStation = null;
exports.getGroundStationOutput = null;
utilities.lazyLoad(exports, ["getGroundStation", "getGroundStationOutput"], () => require("./getGroundStation"));
exports.getL2Connection = null;
exports.getL2ConnectionOutput = null;
utilities.lazyLoad(exports, ["getL2Connection", "getL2ConnectionOutput"], () => require("./getL2Connection"));
exports.GroundStation = null;
utilities.lazyLoad(exports, ["GroundStation"], () => require("./groundStation"));
exports.L2Connection = null;
utilities.lazyLoad(exports, ["L2Connection"], () => require("./l2connection"));
exports.listEdgeSiteL2Connections = null;
exports.listEdgeSiteL2ConnectionsOutput = null;
utilities.lazyLoad(exports, ["listEdgeSiteL2Connections", "listEdgeSiteL2ConnectionsOutput"], () => require("./listEdgeSiteL2Connections"));
exports.listGroundStationL2Connections = null;
exports.listGroundStationL2ConnectionsOutput = null;
utilities.lazyLoad(exports, ["listGroundStationL2Connections", "listGroundStationL2ConnectionsOutput"], () => require("./listGroundStationL2Connections"));
// Export enums:
__exportStar(require("../../types/enums/orbital/v20240301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:orbital/v20240301preview:EdgeSite":
                return new exports.EdgeSite(name, undefined, { urn });
            case "azure-native:orbital/v20240301preview:GroundStation":
                return new exports.GroundStation(name, undefined, { urn });
            case "azure-native:orbital/v20240301preview:L2Connection":
                return new exports.L2Connection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "orbital/v20240301preview", _module);
