// @pulumi/azure-native/confluent/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240701 = exports.v20240213 = exports.v20230822 = exports.v20211201 = exports.v20200301preview = exports.Topic = exports.OrganizationEnvironmentById = exports.OrganizationClusterById = exports.Organization = exports.listOrganizationRegionsOutput = exports.listOrganizationRegions = exports.listAccessUsersOutput = exports.listAccessUsers = exports.listAccessServiceAccountsOutput = exports.listAccessServiceAccounts = exports.listAccessRoleBindingsOutput = exports.listAccessRoleBindings = exports.listAccessRoleBindingNameOutput = exports.listAccessRoleBindingName = exports.listAccessInvitationsOutput = exports.listAccessInvitations = exports.listAccessEnvironmentsOutput = exports.listAccessEnvironments = exports.listAccessClustersOutput = exports.listAccessClusters = exports.getTopicOutput = exports.getTopic = exports.getOrganizationEnvironmentByIdOutput = exports.getOrganizationEnvironmentById = exports.getOrganizationClusterByIdOutput = exports.getOrganizationClusterById = exports.getOrganizationOutput = exports.getOrganization = exports.getConnectorOutput = exports.getConnector = exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getOrganization = null;
exports.getOrganizationOutput = null;
utilities.lazyLoad(exports, ["getOrganization", "getOrganizationOutput"], () => require("./getOrganization"));
exports.getOrganizationClusterById = null;
exports.getOrganizationClusterByIdOutput = null;
utilities.lazyLoad(exports, ["getOrganizationClusterById", "getOrganizationClusterByIdOutput"], () => require("./getOrganizationClusterById"));
exports.getOrganizationEnvironmentById = null;
exports.getOrganizationEnvironmentByIdOutput = null;
utilities.lazyLoad(exports, ["getOrganizationEnvironmentById", "getOrganizationEnvironmentByIdOutput"], () => require("./getOrganizationEnvironmentById"));
exports.getTopic = null;
exports.getTopicOutput = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
exports.listAccessClusters = null;
exports.listAccessClustersOutput = null;
utilities.lazyLoad(exports, ["listAccessClusters", "listAccessClustersOutput"], () => require("./listAccessClusters"));
exports.listAccessEnvironments = null;
exports.listAccessEnvironmentsOutput = null;
utilities.lazyLoad(exports, ["listAccessEnvironments", "listAccessEnvironmentsOutput"], () => require("./listAccessEnvironments"));
exports.listAccessInvitations = null;
exports.listAccessInvitationsOutput = null;
utilities.lazyLoad(exports, ["listAccessInvitations", "listAccessInvitationsOutput"], () => require("./listAccessInvitations"));
exports.listAccessRoleBindingName = null;
exports.listAccessRoleBindingNameOutput = null;
utilities.lazyLoad(exports, ["listAccessRoleBindingName", "listAccessRoleBindingNameOutput"], () => require("./listAccessRoleBindingName"));
exports.listAccessRoleBindings = null;
exports.listAccessRoleBindingsOutput = null;
utilities.lazyLoad(exports, ["listAccessRoleBindings", "listAccessRoleBindingsOutput"], () => require("./listAccessRoleBindings"));
exports.listAccessServiceAccounts = null;
exports.listAccessServiceAccountsOutput = null;
utilities.lazyLoad(exports, ["listAccessServiceAccounts", "listAccessServiceAccountsOutput"], () => require("./listAccessServiceAccounts"));
exports.listAccessUsers = null;
exports.listAccessUsersOutput = null;
utilities.lazyLoad(exports, ["listAccessUsers", "listAccessUsersOutput"], () => require("./listAccessUsers"));
exports.listOrganizationRegions = null;
exports.listOrganizationRegionsOutput = null;
utilities.lazyLoad(exports, ["listOrganizationRegions", "listOrganizationRegionsOutput"], () => require("./listOrganizationRegions"));
exports.Organization = null;
utilities.lazyLoad(exports, ["Organization"], () => require("./organization"));
exports.OrganizationClusterById = null;
utilities.lazyLoad(exports, ["OrganizationClusterById"], () => require("./organizationClusterById"));
exports.OrganizationEnvironmentById = null;
utilities.lazyLoad(exports, ["OrganizationEnvironmentById"], () => require("./organizationEnvironmentById"));
exports.Topic = null;
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
// Export enums:
__exportStar(require("../types/enums/confluent"), exports);
// Export sub-modules:
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20211201 = require("./v20211201");
exports.v20211201 = v20211201;
const v20230822 = require("./v20230822");
exports.v20230822 = v20230822;
const v20240213 = require("./v20240213");
exports.v20240213 = v20240213;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:confluent:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:confluent:Organization":
                return new exports.Organization(name, undefined, { urn });
            case "azure-native:confluent:OrganizationClusterById":
                return new exports.OrganizationClusterById(name, undefined, { urn });
            case "azure-native:confluent:OrganizationEnvironmentById":
                return new exports.OrganizationEnvironmentById(name, undefined, { urn });
            case "azure-native:confluent:Topic":
                return new exports.Topic(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "confluent", _module);
