// @pulumi/gcp/securitycenter/managementOrganizationSecurityHealthAnalyticsCustomModule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementOrganizationSecurityHealthAnalyticsCustomModule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents an instance of a Security Health Analytics custom module, including
 * its full module name, display name, enablement state, and last updated time.
 * You can create a custom module at the organization, folder, or project level.
 * Custom modules that you create at the organization or folder level are inherited
 * by the child folders and projects.
 *
 * To get more information about OrganizationSecurityHealthAnalyticsCustomModule, see:
 *
 * * [API documentation](https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.securityHealthAnalyticsCustomModules)
 * * How-to Guides
 *     * [Overview of custom modules for Security Health Analytics](https://cloud.google.com/security-command-center/docs/custom-modules-sha-overview)
 *
 * ## Example Usage
 *
 * ### Scc Management Organization Security Health Analytics Custom Module Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.securitycenter.ManagementOrganizationSecurityHealthAnalyticsCustomModule("example", {
 *     organization: "123456789",
 *     displayName: "basic_custom_module",
 *     location: "global",
 *     enablementState: "ENABLED",
 *     customConfig: {
 *         predicate: {
 *             expression: "resource.rotationPeriod > duration(\"2592000s\")",
 *         },
 *         resourceSelector: {
 *             resourceTypes: ["cloudkms.googleapis.com/CryptoKey"],
 *         },
 *         description: "The rotation period of the identified cryptokey resource exceeds 30 days.",
 *         recommendation: "Set the rotation period to at most 30 days.",
 *         severity: "MEDIUM",
 *     },
 * });
 * ```
 * ### Scc Management Organization Security Health Analytics Custom Module Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const example = new gcp.securitycenter.ManagementOrganizationSecurityHealthAnalyticsCustomModule("example", {
 *     organization: "123456789",
 *     displayName: "full_custom_module",
 *     location: "global",
 *     enablementState: "ENABLED",
 *     customConfig: {
 *         predicate: {
 *             expression: "resource.rotationPeriod > duration(\"2592000s\")",
 *             title: "Purpose of the expression",
 *             description: "description of the expression",
 *             location: "location of the expression",
 *         },
 *         customOutput: {
 *             properties: [{
 *                 name: "duration",
 *                 valueExpression: {
 *                     expression: "resource.rotationPeriod",
 *                     title: "Purpose of the expression",
 *                     description: "description of the expression",
 *                     location: "location of the expression",
 *                 },
 *             }],
 *         },
 *         resourceSelector: {
 *             resourceTypes: ["cloudkms.googleapis.com/CryptoKey"],
 *         },
 *         severity: "LOW",
 *         description: "Description of the custom module",
 *         recommendation: "Steps to resolve violation",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * OrganizationSecurityHealthAnalyticsCustomModule can be imported using any of these accepted formats:
 *
 * * `organizations/{{organization}}/locations/{{location}}/securityHealthAnalyticsCustomModules/{{name}}`
 *
 * * `{{organization}}/{{location}}/{{name}}`
 *
 * When using the `pulumi import` command, OrganizationSecurityHealthAnalyticsCustomModule can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:securitycenter/managementOrganizationSecurityHealthAnalyticsCustomModule:ManagementOrganizationSecurityHealthAnalyticsCustomModule default organizations/{{organization}}/locations/{{location}}/securityHealthAnalyticsCustomModules/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:securitycenter/managementOrganizationSecurityHealthAnalyticsCustomModule:ManagementOrganizationSecurityHealthAnalyticsCustomModule default {{organization}}/{{location}}/{{name}}
 * ```
 */
class ManagementOrganizationSecurityHealthAnalyticsCustomModule extends pulumi.CustomResource {
    /**
     * Get an existing ManagementOrganizationSecurityHealthAnalyticsCustomModule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ManagementOrganizationSecurityHealthAnalyticsCustomModule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagementOrganizationSecurityHealthAnalyticsCustomModule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementOrganizationSecurityHealthAnalyticsCustomModule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["ancestorModule"] = state ? state.ancestorModule : undefined;
            resourceInputs["customConfig"] = state ? state.customConfig : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["enablementState"] = state ? state.enablementState : undefined;
            resourceInputs["lastEditor"] = state ? state.lastEditor : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["organization"] = state ? state.organization : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.organization === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            resourceInputs["customConfig"] = args ? args.customConfig : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["enablementState"] = args ? args.enablementState : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["organization"] = args ? args.organization : undefined;
            resourceInputs["ancestorModule"] = undefined /*out*/;
            resourceInputs["lastEditor"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ManagementOrganizationSecurityHealthAnalyticsCustomModule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagementOrganizationSecurityHealthAnalyticsCustomModule = ManagementOrganizationSecurityHealthAnalyticsCustomModule;
/** @internal */
ManagementOrganizationSecurityHealthAnalyticsCustomModule.__pulumiType = 'gcp:securitycenter/managementOrganizationSecurityHealthAnalyticsCustomModule:ManagementOrganizationSecurityHealthAnalyticsCustomModule';
