// @pulumi/azure-native/types/enums/documentdb/v20240301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NodeKind = exports.CreateMode = void 0;
exports.CreateMode = {
    /**
     * Create a new mongo cluster.
     */
    Default: "Default",
    /**
     * Create a mongo cluster from a restore point-in-time.
     */
    PointInTimeRestore: "PointInTimeRestore",
};
exports.NodeKind = {
    /**
     * The node is a shard kind.
     */
    Shard: "Shard",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    /**
     * If set, mongo cluster can be accessed through private and public methods.
     */
    Enabled: "Enabled",
    /**
     * If set, the private endpoints are the exclusive access method.
     */
    Disabled: "Disabled",
};
