// @pulumi/azure-native/network/v20180701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpnSite = exports.VpnGateway = exports.VpnConnection = exports.VirtualWAN = exports.VirtualHub = exports.ServiceEndpointPolicyDefinition = exports.ServiceEndpointPolicy = exports.NetworkInterface = exports.getVpnSiteOutput = exports.getVpnSite = exports.getVpnGatewayOutput = exports.getVpnGateway = exports.getVpnConnectionOutput = exports.getVpnConnection = exports.getVirtualWANOutput = exports.getVirtualWAN = exports.getVirtualHubOutput = exports.getVirtualHub = exports.getServiceEndpointPolicyDefinitionOutput = exports.getServiceEndpointPolicyDefinition = exports.getServiceEndpointPolicyOutput = exports.getServiceEndpointPolicy = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getServiceEndpointPolicy = null;
exports.getServiceEndpointPolicyOutput = null;
utilities.lazyLoad(exports, ["getServiceEndpointPolicy", "getServiceEndpointPolicyOutput"], () => require("./getServiceEndpointPolicy"));
exports.getServiceEndpointPolicyDefinition = null;
exports.getServiceEndpointPolicyDefinitionOutput = null;
utilities.lazyLoad(exports, ["getServiceEndpointPolicyDefinition", "getServiceEndpointPolicyDefinitionOutput"], () => require("./getServiceEndpointPolicyDefinition"));
exports.getVirtualHub = null;
exports.getVirtualHubOutput = null;
utilities.lazyLoad(exports, ["getVirtualHub", "getVirtualHubOutput"], () => require("./getVirtualHub"));
exports.getVirtualWAN = null;
exports.getVirtualWANOutput = null;
utilities.lazyLoad(exports, ["getVirtualWAN", "getVirtualWANOutput"], () => require("./getVirtualWAN"));
exports.getVpnConnection = null;
exports.getVpnConnectionOutput = null;
utilities.lazyLoad(exports, ["getVpnConnection", "getVpnConnectionOutput"], () => require("./getVpnConnection"));
exports.getVpnGateway = null;
exports.getVpnGatewayOutput = null;
utilities.lazyLoad(exports, ["getVpnGateway", "getVpnGatewayOutput"], () => require("./getVpnGateway"));
exports.getVpnSite = null;
exports.getVpnSiteOutput = null;
utilities.lazyLoad(exports, ["getVpnSite", "getVpnSiteOutput"], () => require("./getVpnSite"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.ServiceEndpointPolicy = null;
utilities.lazyLoad(exports, ["ServiceEndpointPolicy"], () => require("./serviceEndpointPolicy"));
exports.ServiceEndpointPolicyDefinition = null;
utilities.lazyLoad(exports, ["ServiceEndpointPolicyDefinition"], () => require("./serviceEndpointPolicyDefinition"));
exports.VirtualHub = null;
utilities.lazyLoad(exports, ["VirtualHub"], () => require("./virtualHub"));
exports.VirtualWAN = null;
utilities.lazyLoad(exports, ["VirtualWAN"], () => require("./virtualWAN"));
exports.VpnConnection = null;
utilities.lazyLoad(exports, ["VpnConnection"], () => require("./vpnConnection"));
exports.VpnGateway = null;
utilities.lazyLoad(exports, ["VpnGateway"], () => require("./vpnGateway"));
exports.VpnSite = null;
utilities.lazyLoad(exports, ["VpnSite"], () => require("./vpnSite"));
// Export enums:
__exportStar(require("../../types/enums/network/v20180701"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20180701:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:network/v20180701:ServiceEndpointPolicy":
                return new exports.ServiceEndpointPolicy(name, undefined, { urn });
            case "azure-native:network/v20180701:ServiceEndpointPolicyDefinition":
                return new exports.ServiceEndpointPolicyDefinition(name, undefined, { urn });
            case "azure-native:network/v20180701:VirtualHub":
                return new exports.VirtualHub(name, undefined, { urn });
            case "azure-native:network/v20180701:VirtualWAN":
                return new exports.VirtualWAN(name, undefined, { urn });
            case "azure-native:network/v20180701:VpnConnection":
                return new exports.VpnConnection(name, undefined, { urn });
            case "azure-native:network/v20180701:VpnGateway":
                return new exports.VpnGateway(name, undefined, { urn });
            case "azure-native:network/v20180701:VpnSite":
                return new exports.VpnSite(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20180701", _module);
