// @pulumi/azure-native/billingbenefits/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241101preview = exports.getDiscountOutput = exports.getDiscount = exports.Discount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Discount = null;
utilities.lazyLoad(exports, ["Discount"], () => require("./discount"));
exports.getDiscount = null;
exports.getDiscountOutput = null;
utilities.lazyLoad(exports, ["getDiscount", "getDiscountOutput"], () => require("./getDiscount"));
// Export enums:
__exportStar(require("../types/enums/billingbenefits"), exports);
// Export sub-modules:
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:billingbenefits:Discount":
                return new exports.Discount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "billingbenefits", _module);
