// @pulumi/azure-native/types/enums/iotoperationsmq/v20231004preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceType = exports.RunMode = exports.ResourceInfoDefinitionMethods = exports.MqttProtocol = exports.MqttBridgeRouteDirection = exports.MqttBridgeRemoteBrokerProtocol = exports.KafkaSaslType = exports.KafkaPartitionStrategy = exports.KafkaMessageCompressionType = exports.KafkaAcks = exports.FabricPathType = exports.ExtendedLocationType = exports.DeltaTableFormatEnum = exports.DataLakeDatabaseFormat = exports.BrokerMemoryProfile = void 0;
exports.BrokerMemoryProfile = {
    /**
     * Tiny memory profile.
     */
    Tiny: "tiny",
    /**
     * Low memory profile.
     */
    Low: "low",
    /**
     * Medium memory profile.
     */
    Medium: "medium",
    /**
     * High memory profile.
     */
    High: "high",
};
exports.DataLakeDatabaseFormat = {
    /**
     * Delta format.
     */
    Delta: "delta",
    /**
     * Parquet format.
     */
    Parquet: "parquet",
};
exports.DeltaTableFormatEnum = {
    /**
     * Bool format
     */
    Boolean: "boolean",
    /**
     * Signed integer 8
     */
    Int8: "int8",
    /**
     * Signed integer 16
     */
    Int16: "int16",
    /**
     * Signed integer 32
     */
    Int32: "int32",
    /**
     * Unsigned integer 8
     */
    UInt8: "uInt8",
    /**
     * Unsigned integer 16
     */
    UInt16: "uInt16",
    /**
     * Unsigned integer 32
     */
    UInt32: "uInt32",
    /**
     * Unsigned integer 64
     */
    UInt64: "uInt64",
    /**
     * Float 16
     */
    Float16: "float16",
    /**
     * Float 32
     */
    Float32: "float32",
    /**
     * Float 64
     */
    Float64: "float64",
    /**
     * Date 32
     */
    Date32: "date32",
    /**
     * Date 64
     */
    Date64: "date64",
    /**
     * Binary data
     */
    Binary: "binary",
    /**
     * UTF8 format
     */
    Utf8: "utf8",
};
exports.ExtendedLocationType = {
    /**
     * CustomLocation type
     */
    CustomLocation: "CustomLocation",
};
exports.FabricPathType = {
    /**
     * Fabric path type is Files.
     */
    Files: "files",
    /**
     * Fabric path type is Tables.
     */
    Tables: "tables",
};
exports.KafkaAcks = {
    /**
     * Kafka acks zero.
     */
    Zero: "zero",
    /**
     * Kafka acks one.
     */
    One: "one",
    /**
     * Kafka acks all.
     */
    All: "all",
};
exports.KafkaMessageCompressionType = {
    /**
     * No Kafka message compression.
     */
    None: "none",
    /**
     * Gzip Kafka message compression.
     */
    Gzip: "gzip",
    /**
     * Snappy Kafka message compression.
     */
    Snappy: "snappy",
    /**
     * Lz4 Kafka message compression.
     */
    Lz4: "lz4",
};
exports.KafkaPartitionStrategy = {
    /**
     * Default partition strategy.
     */
    Default: "default",
    /**
     * Static partition strategy.
     */
    Static: "static",
    /**
     * Topic partition strategy.
     */
    Topic: "topic",
    /**
     * Property partition strategy.
     */
    Property: "property",
};
exports.KafkaSaslType = {
    /**
     * Sasl Plain authentication.
     */
    Plain: "plain",
    /**
     * Sasl ScramSha256 authentication.
     */
    ScramSha256: "scramSha256",
    /**
     * Sasl ScramSha512 authentication.
     */
    ScramSha512: "scramSha512",
};
exports.MqttBridgeRemoteBrokerProtocol = {
    /**
     * MQTT protocol.
     */
    Mqtt: "mqtt",
    /**
     * MQTT over WebSocket protocol.
     */
    WebSocket: "webSocket",
};
exports.MqttBridgeRouteDirection = {
    /**
     * Remote to Local Broker.
     */
    RemoteToLocal: "remote-to-local",
    /**
     * Local to Remote Broker.
     */
    LocalToRemote: "local-to-remote",
};
exports.MqttProtocol = {
    /**
     * Mqttv3
     */
    V3: "v3",
    /**
     * Mqttv5
     */
    V5: "v5",
};
exports.ResourceInfoDefinitionMethods = {
    /**
     * Allowed Connecting to Broker
     */
    Connect: "Connect",
    /**
     * Allowed Publishing to Broker
     */
    Publish: "Publish",
    /**
     * Allowed Subscribing to Broker
     */
    Subscribe: "Subscribe",
};
exports.RunMode = {
    /**
     * Automatically provision Frontend and Backend pods.
     */
    Auto: "auto",
    /**
     * Use Cardinality to set Frontend and Backend pods.
     */
    Distributed: "distributed",
};
exports.ServiceType = {
    /**
     * Cluster IP Service.
     */
    ClusterIp: "clusterIp",
    /**
     * Load Balancer Service.
     */
    LoadBalancer: "loadBalancer",
    /**
     * Node Port Service.
     */
    NodePort: "nodePort",
};
