// @pulumi/azure-native/kubernetes/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241201preview = exports.v20240715preview = exports.v20240701preview = exports.v20240601preview = exports.v20240201preview = exports.v20240101 = exports.v20231101preview = exports.v20221001preview = exports.v20220501preview = exports.v20210401preview = exports.listConnectedClusterUserCredentialsOutput = exports.listConnectedClusterUserCredentials = exports.listConnectedClusterUserCredentialOutput = exports.listConnectedClusterUserCredential = exports.getConnectedClusterOutput = exports.getConnectedCluster = exports.ConnectedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ConnectedCluster = null;
utilities.lazyLoad(exports, ["ConnectedCluster"], () => require("./connectedCluster"));
exports.getConnectedCluster = null;
exports.getConnectedClusterOutput = null;
utilities.lazyLoad(exports, ["getConnectedCluster", "getConnectedClusterOutput"], () => require("./getConnectedCluster"));
exports.listConnectedClusterUserCredential = null;
exports.listConnectedClusterUserCredentialOutput = null;
utilities.lazyLoad(exports, ["listConnectedClusterUserCredential", "listConnectedClusterUserCredentialOutput"], () => require("./listConnectedClusterUserCredential"));
exports.listConnectedClusterUserCredentials = null;
exports.listConnectedClusterUserCredentialsOutput = null;
utilities.lazyLoad(exports, ["listConnectedClusterUserCredentials", "listConnectedClusterUserCredentialsOutput"], () => require("./listConnectedClusterUserCredentials"));
// Export enums:
__exportStar(require("../types/enums/kubernetes"), exports);
// Export sub-modules:
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
const v20240715preview = require("./v20240715preview");
exports.v20240715preview = v20240715preview;
const v20241201preview = require("./v20241201preview");
exports.v20241201preview = v20241201preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetes:ConnectedCluster":
                return new exports.ConnectedCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetes", _module);
