// @pulumi/azure-native/dbformysql/v20220930preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Server = exports.PrivateEndpointConnection = exports.getServerOutput = exports.getServer = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getGetPrivateDnsZoneSuffixExecuteOutput = exports.getGetPrivateDnsZoneSuffixExecute = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getGetPrivateDnsZoneSuffixExecute = null;
exports.getGetPrivateDnsZoneSuffixExecuteOutput = null;
utilities.lazyLoad(exports, ["getGetPrivateDnsZoneSuffixExecute", "getGetPrivateDnsZoneSuffixExecuteOutput"], () => require("./getGetPrivateDnsZoneSuffixExecute"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getServer = null;
exports.getServerOutput = null;
utilities.lazyLoad(exports, ["getServer", "getServerOutput"], () => require("./getServer"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Server = null;
utilities.lazyLoad(exports, ["Server"], () => require("./server"));
// Export enums:
__exportStar(require("../../types/enums/dbformysql/v20220930preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbformysql/v20220930preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:dbformysql/v20220930preview:Server":
                return new exports.Server(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbformysql/v20220930preview", _module);
