// @pulumi/azure-native/automation/v20220808/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Runbook = exports.Python3Package = exports.Python2Package = exports.Module = exports.listKeyByAutomationAccountOutput = exports.listKeyByAutomationAccount = exports.JobSchedule = exports.HybridRunbookWorkerGroup = exports.HybridRunbookWorker = exports.getVariableOutput = exports.getVariable = exports.getSourceControlOutput = exports.getSourceControl = exports.getScheduleOutput = exports.getSchedule = exports.getRunbookOutput = exports.getRunbook = exports.getPython3PackageOutput = exports.getPython3Package = exports.getPython2PackageOutput = exports.getPython2Package = exports.getModuleOutput = exports.getModule = exports.getJobScheduleOutput = exports.getJobSchedule = exports.getHybridRunbookWorkerGroupOutput = exports.getHybridRunbookWorkerGroup = exports.getHybridRunbookWorkerOutput = exports.getHybridRunbookWorker = exports.getDscNodeConfigurationOutput = exports.getDscNodeConfiguration = exports.getDscConfigurationOutput = exports.getDscConfiguration = exports.getCredentialOutput = exports.getCredential = exports.getConnectionTypeOutput = exports.getConnectionType = exports.getConnectionOutput = exports.getConnection = exports.getCertificateOutput = exports.getCertificate = exports.getAutomationAccountOutput = exports.getAutomationAccount = exports.DscNodeConfiguration = exports.DscConfiguration = exports.Credential = exports.ConnectionType = exports.Connection = exports.Certificate = exports.AutomationAccount = void 0;
exports.Variable = exports.SourceControl = exports.Schedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AutomationAccount = null;
utilities.lazyLoad(exports, ["AutomationAccount"], () => require("./automationAccount"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionType = null;
utilities.lazyLoad(exports, ["ConnectionType"], () => require("./connectionType"));
exports.Credential = null;
utilities.lazyLoad(exports, ["Credential"], () => require("./credential"));
exports.DscConfiguration = null;
utilities.lazyLoad(exports, ["DscConfiguration"], () => require("./dscConfiguration"));
exports.DscNodeConfiguration = null;
utilities.lazyLoad(exports, ["DscNodeConfiguration"], () => require("./dscNodeConfiguration"));
exports.getAutomationAccount = null;
exports.getAutomationAccountOutput = null;
utilities.lazyLoad(exports, ["getAutomationAccount", "getAutomationAccountOutput"], () => require("./getAutomationAccount"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.getConnectionType = null;
exports.getConnectionTypeOutput = null;
utilities.lazyLoad(exports, ["getConnectionType", "getConnectionTypeOutput"], () => require("./getConnectionType"));
exports.getCredential = null;
exports.getCredentialOutput = null;
utilities.lazyLoad(exports, ["getCredential", "getCredentialOutput"], () => require("./getCredential"));
exports.getDscConfiguration = null;
exports.getDscConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDscConfiguration", "getDscConfigurationOutput"], () => require("./getDscConfiguration"));
exports.getDscNodeConfiguration = null;
exports.getDscNodeConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDscNodeConfiguration", "getDscNodeConfigurationOutput"], () => require("./getDscNodeConfiguration"));
exports.getHybridRunbookWorker = null;
exports.getHybridRunbookWorkerOutput = null;
utilities.lazyLoad(exports, ["getHybridRunbookWorker", "getHybridRunbookWorkerOutput"], () => require("./getHybridRunbookWorker"));
exports.getHybridRunbookWorkerGroup = null;
exports.getHybridRunbookWorkerGroupOutput = null;
utilities.lazyLoad(exports, ["getHybridRunbookWorkerGroup", "getHybridRunbookWorkerGroupOutput"], () => require("./getHybridRunbookWorkerGroup"));
exports.getJobSchedule = null;
exports.getJobScheduleOutput = null;
utilities.lazyLoad(exports, ["getJobSchedule", "getJobScheduleOutput"], () => require("./getJobSchedule"));
exports.getModule = null;
exports.getModuleOutput = null;
utilities.lazyLoad(exports, ["getModule", "getModuleOutput"], () => require("./getModule"));
exports.getPython2Package = null;
exports.getPython2PackageOutput = null;
utilities.lazyLoad(exports, ["getPython2Package", "getPython2PackageOutput"], () => require("./getPython2Package"));
exports.getPython3Package = null;
exports.getPython3PackageOutput = null;
utilities.lazyLoad(exports, ["getPython3Package", "getPython3PackageOutput"], () => require("./getPython3Package"));
exports.getRunbook = null;
exports.getRunbookOutput = null;
utilities.lazyLoad(exports, ["getRunbook", "getRunbookOutput"], () => require("./getRunbook"));
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.getSourceControl = null;
exports.getSourceControlOutput = null;
utilities.lazyLoad(exports, ["getSourceControl", "getSourceControlOutput"], () => require("./getSourceControl"));
exports.getVariable = null;
exports.getVariableOutput = null;
utilities.lazyLoad(exports, ["getVariable", "getVariableOutput"], () => require("./getVariable"));
exports.HybridRunbookWorker = null;
utilities.lazyLoad(exports, ["HybridRunbookWorker"], () => require("./hybridRunbookWorker"));
exports.HybridRunbookWorkerGroup = null;
utilities.lazyLoad(exports, ["HybridRunbookWorkerGroup"], () => require("./hybridRunbookWorkerGroup"));
exports.JobSchedule = null;
utilities.lazyLoad(exports, ["JobSchedule"], () => require("./jobSchedule"));
exports.listKeyByAutomationAccount = null;
exports.listKeyByAutomationAccountOutput = null;
utilities.lazyLoad(exports, ["listKeyByAutomationAccount", "listKeyByAutomationAccountOutput"], () => require("./listKeyByAutomationAccount"));
exports.Module = null;
utilities.lazyLoad(exports, ["Module"], () => require("./module"));
exports.Python2Package = null;
utilities.lazyLoad(exports, ["Python2Package"], () => require("./python2Package"));
exports.Python3Package = null;
utilities.lazyLoad(exports, ["Python3Package"], () => require("./python3Package"));
exports.Runbook = null;
utilities.lazyLoad(exports, ["Runbook"], () => require("./runbook"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
exports.SourceControl = null;
utilities.lazyLoad(exports, ["SourceControl"], () => require("./sourceControl"));
exports.Variable = null;
utilities.lazyLoad(exports, ["Variable"], () => require("./variable"));
// Export enums:
__exportStar(require("../../types/enums/automation/v20220808"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:automation/v20220808:AutomationAccount":
                return new exports.AutomationAccount(name, undefined, { urn });
            case "azure-native:automation/v20220808:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:automation/v20220808:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "azure-native:automation/v20220808:ConnectionType":
                return new exports.ConnectionType(name, undefined, { urn });
            case "azure-native:automation/v20220808:Credential":
                return new exports.Credential(name, undefined, { urn });
            case "azure-native:automation/v20220808:DscConfiguration":
                return new exports.DscConfiguration(name, undefined, { urn });
            case "azure-native:automation/v20220808:DscNodeConfiguration":
                return new exports.DscNodeConfiguration(name, undefined, { urn });
            case "azure-native:automation/v20220808:HybridRunbookWorker":
                return new exports.HybridRunbookWorker(name, undefined, { urn });
            case "azure-native:automation/v20220808:HybridRunbookWorkerGroup":
                return new exports.HybridRunbookWorkerGroup(name, undefined, { urn });
            case "azure-native:automation/v20220808:JobSchedule":
                return new exports.JobSchedule(name, undefined, { urn });
            case "azure-native:automation/v20220808:Module":
                return new exports.Module(name, undefined, { urn });
            case "azure-native:automation/v20220808:Python2Package":
                return new exports.Python2Package(name, undefined, { urn });
            case "azure-native:automation/v20220808:Python3Package":
                return new exports.Python3Package(name, undefined, { urn });
            case "azure-native:automation/v20220808:Runbook":
                return new exports.Runbook(name, undefined, { urn });
            case "azure-native:automation/v20220808:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            case "azure-native:automation/v20220808:SourceControl":
                return new exports.SourceControl(name, undefined, { urn });
            case "azure-native:automation/v20220808:Variable":
                return new exports.Variable(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "automation/v20220808", _module);
