// @pulumi/azure-native/network/v20200401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateResolverVirtualNetworkLink = exports.OutboundEndpoint = exports.listDnsResolverByVirtualNetworkOutput = exports.listDnsResolverByVirtualNetwork = exports.listDnsForwardingRulesetByVirtualNetworkOutput = exports.listDnsForwardingRulesetByVirtualNetwork = exports.InboundEndpoint = exports.getPrivateResolverVirtualNetworkLinkOutput = exports.getPrivateResolverVirtualNetworkLink = exports.getOutboundEndpointOutput = exports.getOutboundEndpoint = exports.getInboundEndpointOutput = exports.getInboundEndpoint = exports.getDnsForwardingRulesetOutput = exports.getDnsForwardingRuleset = exports.DnsForwardingRuleset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DnsForwardingRuleset = null;
utilities.lazyLoad(exports, ["DnsForwardingRuleset"], () => require("./dnsForwardingRuleset"));
exports.getDnsForwardingRuleset = null;
exports.getDnsForwardingRulesetOutput = null;
utilities.lazyLoad(exports, ["getDnsForwardingRuleset", "getDnsForwardingRulesetOutput"], () => require("./getDnsForwardingRuleset"));
exports.getInboundEndpoint = null;
exports.getInboundEndpointOutput = null;
utilities.lazyLoad(exports, ["getInboundEndpoint", "getInboundEndpointOutput"], () => require("./getInboundEndpoint"));
exports.getOutboundEndpoint = null;
exports.getOutboundEndpointOutput = null;
utilities.lazyLoad(exports, ["getOutboundEndpoint", "getOutboundEndpointOutput"], () => require("./getOutboundEndpoint"));
exports.getPrivateResolverVirtualNetworkLink = null;
exports.getPrivateResolverVirtualNetworkLinkOutput = null;
utilities.lazyLoad(exports, ["getPrivateResolverVirtualNetworkLink", "getPrivateResolverVirtualNetworkLinkOutput"], () => require("./getPrivateResolverVirtualNetworkLink"));
exports.InboundEndpoint = null;
utilities.lazyLoad(exports, ["InboundEndpoint"], () => require("./inboundEndpoint"));
exports.listDnsForwardingRulesetByVirtualNetwork = null;
exports.listDnsForwardingRulesetByVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["listDnsForwardingRulesetByVirtualNetwork", "listDnsForwardingRulesetByVirtualNetworkOutput"], () => require("./listDnsForwardingRulesetByVirtualNetwork"));
exports.listDnsResolverByVirtualNetwork = null;
exports.listDnsResolverByVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["listDnsResolverByVirtualNetwork", "listDnsResolverByVirtualNetworkOutput"], () => require("./listDnsResolverByVirtualNetwork"));
exports.OutboundEndpoint = null;
utilities.lazyLoad(exports, ["OutboundEndpoint"], () => require("./outboundEndpoint"));
exports.PrivateResolverVirtualNetworkLink = null;
utilities.lazyLoad(exports, ["PrivateResolverVirtualNetworkLink"], () => require("./privateResolverVirtualNetworkLink"));
// Export enums:
__exportStar(require("../../types/enums/network/v20200401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20200401preview:DnsForwardingRuleset":
                return new exports.DnsForwardingRuleset(name, undefined, { urn });
            case "azure-native:network/v20200401preview:InboundEndpoint":
                return new exports.InboundEndpoint(name, undefined, { urn });
            case "azure-native:network/v20200401preview:OutboundEndpoint":
                return new exports.OutboundEndpoint(name, undefined, { urn });
            case "azure-native:network/v20200401preview:PrivateResolverVirtualNetworkLink":
                return new exports.PrivateResolverVirtualNetworkLink(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20200401preview", _module);
