// @pulumi/azure-native/operationalinsights/v20151101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.MachineGroup = exports.LinkedService = exports.getWorkspaceSharedKeysOutput = exports.getWorkspaceSharedKeys = exports.getWorkspaceOutput = exports.getWorkspace = exports.getMachineGroupOutput = exports.getMachineGroup = exports.getLinkedServiceOutput = exports.getLinkedService = exports.getDataSourceOutput = exports.getDataSource = exports.DataSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.getDataSource = null;
exports.getDataSourceOutput = null;
utilities.lazyLoad(exports, ["getDataSource", "getDataSourceOutput"], () => require("./getDataSource"));
exports.getLinkedService = null;
exports.getLinkedServiceOutput = null;
utilities.lazyLoad(exports, ["getLinkedService", "getLinkedServiceOutput"], () => require("./getLinkedService"));
exports.getMachineGroup = null;
exports.getMachineGroupOutput = null;
utilities.lazyLoad(exports, ["getMachineGroup", "getMachineGroupOutput"], () => require("./getMachineGroup"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspaceSharedKeys = null;
exports.getWorkspaceSharedKeysOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceSharedKeys", "getWorkspaceSharedKeysOutput"], () => require("./getWorkspaceSharedKeys"));
exports.LinkedService = null;
utilities.lazyLoad(exports, ["LinkedService"], () => require("./linkedService"));
exports.MachineGroup = null;
utilities.lazyLoad(exports, ["MachineGroup"], () => require("./machineGroup"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../../types/enums/operationalinsights/v20151101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:operationalinsights/v20151101preview:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "azure-native:operationalinsights/v20151101preview:LinkedService":
                return new exports.LinkedService(name, undefined, { urn });
            case "azure-native:operationalinsights/v20151101preview:MachineGroup":
                return new exports.MachineGroup(name, undefined, { urn });
            case "azure-native:operationalinsights/v20151101preview:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "operationalinsights/v20151101preview", _module);
