// @pulumi/azure-native/types/enums/containerregistry/v20230701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.WebhookStatus = exports.WebhookAction = exports.TrustPolicyType = exports.TokenStatus = exports.TokenPasswordName = exports.TokenCertificateName = exports.SkuName = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PolicyStatus = exports.NetworkRuleBypassOptions = exports.ExportPolicyStatus = exports.EncryptionStatus = exports.DefaultAction = exports.CredentialName = exports.ConnectionStatus = exports.ActionsRequired = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.ActionsRequired = {
    None: "None",
    Recreate: "Recreate",
};
exports.ConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.CredentialName = {
    Credential1: "Credential1",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EncryptionStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ExportPolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.NetworkRuleBypassOptions = {
    AzureServices: "AzureServices",
    None: "None",
};
exports.PolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.SkuName = {
    Classic: "Classic",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TokenCertificateName = {
    Certificate1: "certificate1",
    Certificate2: "certificate2",
};
exports.TokenPasswordName = {
    Password1: "password1",
    Password2: "password2",
};
exports.TokenStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.TrustPolicyType = {
    Notary: "Notary",
};
exports.WebhookAction = {
    Push: "push",
    Delete: "delete",
    Quarantine: "quarantine",
    Chart_push: "chart_push",
    Chart_delete: "chart_delete",
};
exports.WebhookStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ZoneRedundancy = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
