// @pulumi/azure-native/kubernetesconfiguration/v20230501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceControlConfiguration = exports.getSourceControlConfigurationOutput = exports.getSourceControlConfiguration = exports.getFluxConfigurationOutput = exports.getFluxConfiguration = exports.getExtensionOutput = exports.getExtension = exports.FluxConfiguration = exports.Extension = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.FluxConfiguration = null;
utilities.lazyLoad(exports, ["FluxConfiguration"], () => require("./fluxConfiguration"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getFluxConfiguration = null;
exports.getFluxConfigurationOutput = null;
utilities.lazyLoad(exports, ["getFluxConfiguration", "getFluxConfigurationOutput"], () => require("./getFluxConfiguration"));
exports.getSourceControlConfiguration = null;
exports.getSourceControlConfigurationOutput = null;
utilities.lazyLoad(exports, ["getSourceControlConfiguration", "getSourceControlConfigurationOutput"], () => require("./getSourceControlConfiguration"));
exports.SourceControlConfiguration = null;
utilities.lazyLoad(exports, ["SourceControlConfiguration"], () => require("./sourceControlConfiguration"));
// Export enums:
__exportStar(require("../../types/enums/kubernetesconfiguration/v20230501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetesconfiguration/v20230501:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration/v20230501:FluxConfiguration":
                return new exports.FluxConfiguration(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration/v20230501:SourceControlConfiguration":
                return new exports.SourceControlConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetesconfiguration/v20230501", _module);
