// @pulumi/azure-native/redhatopenshift/v20240812preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenShiftCluster = exports.listOpenShiftClusterCredentialsOutput = exports.listOpenShiftClusterCredentials = exports.listOpenShiftClusterAdminCredentialsOutput = exports.listOpenShiftClusterAdminCredentials = exports.getOpenShiftClusterOutput = exports.getOpenShiftCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getOpenShiftCluster = null;
exports.getOpenShiftClusterOutput = null;
utilities.lazyLoad(exports, ["getOpenShiftCluster", "getOpenShiftClusterOutput"], () => require("./getOpenShiftCluster"));
exports.listOpenShiftClusterAdminCredentials = null;
exports.listOpenShiftClusterAdminCredentialsOutput = null;
utilities.lazyLoad(exports, ["listOpenShiftClusterAdminCredentials", "listOpenShiftClusterAdminCredentialsOutput"], () => require("./listOpenShiftClusterAdminCredentials"));
exports.listOpenShiftClusterCredentials = null;
exports.listOpenShiftClusterCredentialsOutput = null;
utilities.lazyLoad(exports, ["listOpenShiftClusterCredentials", "listOpenShiftClusterCredentialsOutput"], () => require("./listOpenShiftClusterCredentials"));
exports.OpenShiftCluster = null;
utilities.lazyLoad(exports, ["OpenShiftCluster"], () => require("./openShiftCluster"));
// Export enums:
__exportStar(require("../../types/enums/redhatopenshift/v20240812preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:redhatopenshift/v20240812preview:OpenShiftCluster":
                return new exports.OpenShiftCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "redhatopenshift/v20240812preview", _module);
