// @pulumi/azure-native/types/enums/documentdb/v20210401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerVersion = exports.RestoreMode = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.NetworkAclBypass = exports.DefaultConsistencyLevel = exports.DatabaseAccountOfferType = exports.DatabaseAccountKind = exports.CreateMode = exports.ConnectorOffer = exports.BackupStorageRedundancy = exports.BackupPolicyType = void 0;
exports.BackupPolicyType = {
    Periodic: "Periodic",
    Continuous: "Continuous",
};
exports.BackupStorageRedundancy = {
    Geo: "Geo",
    Local: "Local",
    Zone: "Zone",
};
exports.ConnectorOffer = {
    Small: "Small",
};
exports.CreateMode = {
    Default: "Default",
    Restore: "Restore",
};
exports.DatabaseAccountKind = {
    GlobalDocumentDB: "GlobalDocumentDB",
    MongoDB: "MongoDB",
    Parse: "Parse",
};
exports.DatabaseAccountOfferType = {
    Standard: "Standard",
};
exports.DefaultConsistencyLevel = {
    Eventual: "Eventual",
    Session: "Session",
    BoundedStaleness: "BoundedStaleness",
    Strong: "Strong",
    ConsistentPrefix: "ConsistentPrefix",
};
exports.NetworkAclBypass = {
    None: "None",
    AzureServices: "AzureServices",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    None: "None",
};
exports.RestoreMode = {
    PointInTime: "PointInTime",
};
exports.ServerVersion = {
    ServerVersion_3_2: "3.2",
    ServerVersion_3_6: "3.6",
    ServerVersion_4_0: "4.0",
};
