// @pulumi/azure-native/types/enums/storsimple/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeType = exports.VolumeStatus = exports.SslStatus = exports.ScheduleStatus = exports.RecurrenceType = exports.MonitoringStatus = exports.ManagerType = exports.ManagerSkuType = exports.Kind = exports.EncryptionAlgorithm = exports.DayOfWeek = exports.BackupType = exports.v20170601 = void 0;
// Export sub-modules:
const v20170601 = require("./v20170601");
exports.v20170601 = v20170601;
exports.BackupType = {
    LocalSnapshot: "LocalSnapshot",
    CloudSnapshot: "CloudSnapshot",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.EncryptionAlgorithm = {
    None: "None",
    AES256: "AES256",
    RSAES_PKCS1_v_1_5: "RSAES_PKCS1_v_1_5",
};
exports.Kind = {
    Series8000: "Series8000",
};
exports.ManagerSkuType = {
    Standard: "Standard",
};
exports.ManagerType = {
    GardaV1: "GardaV1",
    HelsinkiV1: "HelsinkiV1",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RecurrenceType = {
    Minutes: "Minutes",
    Hourly: "Hourly",
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.ScheduleStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SslStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VolumeStatus = {
    Online: "Online",
    Offline: "Offline",
};
exports.VolumeType = {
    Tiered: "Tiered",
    Archival: "Archival",
    LocallyPinned: "LocallyPinned",
};
