// @pulumi/azure-native/scvmm/v20240601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmmServer = exports.VirtualNetwork = exports.VirtualMachineTemplate = exports.VirtualMachineInstance = exports.InventoryItem = exports.GuestAgent = exports.getVmmServerOutput = exports.getVmmServer = exports.getVirtualNetworkOutput = exports.getVirtualNetwork = exports.getVirtualMachineTemplateOutput = exports.getVirtualMachineTemplate = exports.getVirtualMachineInstanceOutput = exports.getVirtualMachineInstance = exports.getInventoryItemOutput = exports.getInventoryItem = exports.getGuestAgentOutput = exports.getGuestAgent = exports.getCloudOutput = exports.getCloud = exports.getAvailabilitySetOutput = exports.getAvailabilitySet = exports.Cloud = exports.AvailabilitySet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AvailabilitySet = null;
utilities.lazyLoad(exports, ["AvailabilitySet"], () => require("./availabilitySet"));
exports.Cloud = null;
utilities.lazyLoad(exports, ["Cloud"], () => require("./cloud"));
exports.getAvailabilitySet = null;
exports.getAvailabilitySetOutput = null;
utilities.lazyLoad(exports, ["getAvailabilitySet", "getAvailabilitySetOutput"], () => require("./getAvailabilitySet"));
exports.getCloud = null;
exports.getCloudOutput = null;
utilities.lazyLoad(exports, ["getCloud", "getCloudOutput"], () => require("./getCloud"));
exports.getGuestAgent = null;
exports.getGuestAgentOutput = null;
utilities.lazyLoad(exports, ["getGuestAgent", "getGuestAgentOutput"], () => require("./getGuestAgent"));
exports.getInventoryItem = null;
exports.getInventoryItemOutput = null;
utilities.lazyLoad(exports, ["getInventoryItem", "getInventoryItemOutput"], () => require("./getInventoryItem"));
exports.getVirtualMachineInstance = null;
exports.getVirtualMachineInstanceOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineInstance", "getVirtualMachineInstanceOutput"], () => require("./getVirtualMachineInstance"));
exports.getVirtualMachineTemplate = null;
exports.getVirtualMachineTemplateOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineTemplate", "getVirtualMachineTemplateOutput"], () => require("./getVirtualMachineTemplate"));
exports.getVirtualNetwork = null;
exports.getVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetwork", "getVirtualNetworkOutput"], () => require("./getVirtualNetwork"));
exports.getVmmServer = null;
exports.getVmmServerOutput = null;
utilities.lazyLoad(exports, ["getVmmServer", "getVmmServerOutput"], () => require("./getVmmServer"));
exports.GuestAgent = null;
utilities.lazyLoad(exports, ["GuestAgent"], () => require("./guestAgent"));
exports.InventoryItem = null;
utilities.lazyLoad(exports, ["InventoryItem"], () => require("./inventoryItem"));
exports.VirtualMachineInstance = null;
utilities.lazyLoad(exports, ["VirtualMachineInstance"], () => require("./virtualMachineInstance"));
exports.VirtualMachineTemplate = null;
utilities.lazyLoad(exports, ["VirtualMachineTemplate"], () => require("./virtualMachineTemplate"));
exports.VirtualNetwork = null;
utilities.lazyLoad(exports, ["VirtualNetwork"], () => require("./virtualNetwork"));
exports.VmmServer = null;
utilities.lazyLoad(exports, ["VmmServer"], () => require("./vmmServer"));
// Export enums:
__exportStar(require("../../types/enums/scvmm/v20240601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:scvmm/v20240601:AvailabilitySet":
                return new exports.AvailabilitySet(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:Cloud":
                return new exports.Cloud(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:GuestAgent":
                return new exports.GuestAgent(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:InventoryItem":
                return new exports.InventoryItem(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:VirtualMachineInstance":
                return new exports.VirtualMachineInstance(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:VirtualMachineTemplate":
                return new exports.VirtualMachineTemplate(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:VirtualNetwork":
                return new exports.VirtualNetwork(name, undefined, { urn });
            case "azure-native:scvmm/v20240601:VmmServer":
                return new exports.VmmServer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "scvmm/v20240601", _module);
