// @pulumi/azure-native/datashare/synapseWorkspaceSqlPoolTableDataSet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SynapseWorkspaceSqlPoolTableDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Synapse Workspace Sql Pool Table data set.
 *
 * Uses Azure REST API version 2021-08-01. In version 1.x of the Azure Native provider, it used API version 2020-09-01.
 */
class SynapseWorkspaceSqlPoolTableDataSet extends pulumi.CustomResource {
    /**
     * Get an existing SynapseWorkspaceSqlPoolTableDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SynapseWorkspaceSqlPoolTableDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SynapseWorkspaceSqlPoolTableDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SynapseWorkspaceSqlPoolTableDataSet.__pulumiType;
    }
    /**
     * Create a SynapseWorkspaceSqlPoolTableDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            if ((!args || args.synapseWorkspaceSqlPoolTableResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'synapseWorkspaceSqlPoolTableResourceId'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["dataSetName"] = args ? args.dataSetName : undefined;
            resourceInputs["kind"] = "SynapseWorkspaceSqlPoolTable";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["shareName"] = args ? args.shareName : undefined;
            resourceInputs["synapseWorkspaceSqlPoolTableResourceId"] = args ? args.synapseWorkspaceSqlPoolTableResourceId : undefined;
            resourceInputs["dataSetId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataSetId"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["synapseWorkspaceSqlPoolTableResourceId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:datashare/v20181101preview:SynapseWorkspaceSqlPoolTableDataSet" }, { type: "azure-native:datashare/v20191101:SynapseWorkspaceSqlPoolTableDataSet" }, { type: "azure-native:datashare/v20200901:SynapseWorkspaceSqlPoolTableDataSet" }, { type: "azure-native:datashare/v20201001preview:SynapseWorkspaceSqlPoolTableDataSet" }, { type: "azure-native:datashare/v20210801:SynapseWorkspaceSqlPoolTableDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SynapseWorkspaceSqlPoolTableDataSet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SynapseWorkspaceSqlPoolTableDataSet = SynapseWorkspaceSqlPoolTableDataSet;
/** @internal */
SynapseWorkspaceSqlPoolTableDataSet.__pulumiType = 'azure-native:datashare:SynapseWorkspaceSqlPoolTableDataSet';
