// @pulumi/azure-native/types/enums/maintenance/v20230901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Visibility = exports.TagOperators = exports.RebootOptions = exports.MaintenanceScope = void 0;
exports.MaintenanceScope = {
    /**
     * This maintenance scope controls installation of azure platform updates i.e. services on physical nodes hosting customer VMs.
     */
    Host: "Host",
    /**
     * This maintenance scope controls the default update maintenance of the Azure Resource
     */
    Resource: "Resource",
    /**
     * This maintenance scope controls os image installation on VM/VMSS
     */
    OSImage: "OSImage",
    /**
     * This maintenance scope controls extension installation on VM/VMSS
     */
    Extension: "Extension",
    /**
     * This maintenance scope controls installation of windows and linux packages on VM/VMSS
     */
    InGuestPatch: "InGuestPatch",
    /**
     * This maintenance scope controls installation of SQL server platform updates.
     */
    SQLDB: "SQLDB",
    /**
     * This maintenance scope controls installation of SQL managed instance platform update.
     */
    SQLManagedInstance: "SQLManagedInstance",
};
exports.RebootOptions = {
    IfRequired: "IfRequired",
    Never: "Never",
    Always: "Always",
};
exports.TagOperators = {
    All: "All",
    Any: "Any",
};
exports.Visibility = {
    /**
     * Only visible to users with permissions.
     */
    Custom: "Custom",
    /**
     * Visible to all users.
     */
    Public: "Public",
};
