// @pulumi/kubernetes/types/input.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.kubeClientSettingsProvideDefaults = exports.helmReleaseSettingsProvideDefaults = void 0;
const utilities = require("../utilities");
/**
 * helmReleaseSettingsProvideDefaults sets the appropriate defaults for HelmReleaseSettings
 */
function helmReleaseSettingsProvideDefaults(val) {
    return {
        ...val,
        driver: (val.driver) ?? utilities.getEnv("PULUMI_K8S_HELM_DRIVER"),
        pluginsPath: (val.pluginsPath) ?? utilities.getEnv("PULUMI_K8S_HELM_PLUGINS_PATH"),
        registryConfigPath: (val.registryConfigPath) ?? utilities.getEnv("PULUMI_K8S_HELM_REGISTRY_CONFIG_PATH"),
        repositoryCache: (val.repositoryCache) ?? utilities.getEnv("PULUMI_K8S_HELM_REPOSITORY_CACHE"),
        repositoryConfigPath: (val.repositoryConfigPath) ?? utilities.getEnv("PULUMI_K8S_HELM_REPOSITORY_CONFIG_PATH"),
    };
}
exports.helmReleaseSettingsProvideDefaults = helmReleaseSettingsProvideDefaults;
/**
 * kubeClientSettingsProvideDefaults sets the appropriate defaults for KubeClientSettings
 */
function kubeClientSettingsProvideDefaults(val) {
    return {
        ...val,
        burst: (val.burst) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_BURST"),
        qps: (val.qps) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_QPS"),
        timeout: (val.timeout) ?? utilities.getEnvNumber("PULUMI_K8S_CLIENT_TIMEOUT"),
    };
}
exports.kubeClientSettingsProvideDefaults = kubeClientSettingsProvideDefaults;
