// @pulumi/azure-native/types/enums/networkcloud/v20230701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineVirtioInterfaceType = exports.VirtualMachineSchedulingExecution = exports.VirtualMachinePlacementHintType = exports.VirtualMachinePlacementHintPodAffinityScope = exports.VirtualMachineIsolateEmulatorThread = exports.VirtualMachineIPAllocationMethod = exports.VirtualMachineDeviceModelType = exports.VirtualMachineBootMethod = exports.ValidationThresholdType = exports.ValidationThresholdGrouping = exports.OsDiskDeleteOption = exports.OsDiskCreateOption = exports.L3NetworkConfigurationIpamEnabled = exports.KubernetesPluginType = exports.IpAllocationType = exports.HybridAksPluginType = exports.HybridAksIpamEnabled = exports.HugepagesSize = exports.FabricPeeringEnabled = exports.DefaultGateway = exports.ConsoleEnabled = exports.ClusterType = exports.CloudServicesNetworkEnableDefaultEgressEndpoints = exports.BmcKeySetPrivilegeLevel = exports.BgpMultiHop = exports.BfdEnabled = exports.BareMetalMachineKeySetPrivilegeLevel = exports.AgentPoolMode = exports.AdvertiseToFabric = void 0;
exports.AdvertiseToFabric = {
    True: "True",
    False: "False",
};
exports.AgentPoolMode = {
    System: "System",
    User: "User",
    NotApplicable: "NotApplicable",
};
exports.BareMetalMachineKeySetPrivilegeLevel = {
    Standard: "Standard",
    Superuser: "Superuser",
};
exports.BfdEnabled = {
    True: "True",
    False: "False",
};
exports.BgpMultiHop = {
    True: "True",
    False: "False",
};
exports.BmcKeySetPrivilegeLevel = {
    ReadOnly: "ReadOnly",
    Administrator: "Administrator",
};
exports.CloudServicesNetworkEnableDefaultEgressEndpoints = {
    True: "True",
    False: "False",
};
exports.ClusterType = {
    SingleRack: "SingleRack",
    MultiRack: "MultiRack",
};
exports.ConsoleEnabled = {
    True: "True",
    False: "False",
};
exports.DefaultGateway = {
    True: "True",
    False: "False",
};
exports.FabricPeeringEnabled = {
    True: "True",
    False: "False",
};
exports.HugepagesSize = {
    HugepagesSize_2M: "2M",
    HugepagesSize_1G: "1G",
};
exports.HybridAksIpamEnabled = {
    True: "True",
    False: "False",
};
exports.HybridAksPluginType = {
    DPDK: "DPDK",
    SRIOV: "SRIOV",
    OSDevice: "OSDevice",
};
exports.IpAllocationType = {
    IPV4: "IPV4",
    IPV6: "IPV6",
    DualStack: "DualStack",
};
exports.KubernetesPluginType = {
    DPDK: "DPDK",
    SRIOV: "SRIOV",
    OSDevice: "OSDevice",
    MACVLAN: "MACVLAN",
    IPVLAN: "IPVLAN",
};
exports.L3NetworkConfigurationIpamEnabled = {
    True: "True",
    False: "False",
};
exports.OsDiskCreateOption = {
    Ephemeral: "Ephemeral",
};
exports.OsDiskDeleteOption = {
    Delete: "Delete",
};
exports.ValidationThresholdGrouping = {
    PerCluster: "PerCluster",
    PerRack: "PerRack",
};
exports.ValidationThresholdType = {
    CountSuccess: "CountSuccess",
    PercentSuccess: "PercentSuccess",
};
exports.VirtualMachineBootMethod = {
    UEFI: "UEFI",
    BIOS: "BIOS",
};
exports.VirtualMachineDeviceModelType = {
    T1: "T1",
    T2: "T2",
};
exports.VirtualMachineIPAllocationMethod = {
    Dynamic: "Dynamic",
    Static: "Static",
    Disabled: "Disabled",
};
exports.VirtualMachineIsolateEmulatorThread = {
    True: "True",
    False: "False",
};
exports.VirtualMachinePlacementHintPodAffinityScope = {
    Rack: "Rack",
    Machine: "Machine",
};
exports.VirtualMachinePlacementHintType = {
    Affinity: "Affinity",
    AntiAffinity: "AntiAffinity",
};
exports.VirtualMachineSchedulingExecution = {
    Hard: "Hard",
    Soft: "Soft",
};
exports.VirtualMachineVirtioInterfaceType = {
    Modern: "Modern",
    Transitional: "Transitional",
};
