// @pulumi/azure-native/network/v20190201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subnet = exports.NetworkInterface = exports.InterfaceEndpoint = exports.getSubnetOutput = exports.getSubnet = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getInterfaceEndpointOutput = exports.getInterfaceEndpoint = exports.getExpressRouteCircuitPeeringOutput = exports.getExpressRouteCircuitPeering = exports.ExpressRouteCircuitPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ExpressRouteCircuitPeering = null;
utilities.lazyLoad(exports, ["ExpressRouteCircuitPeering"], () => require("./expressRouteCircuitPeering"));
exports.getExpressRouteCircuitPeering = null;
exports.getExpressRouteCircuitPeeringOutput = null;
utilities.lazyLoad(exports, ["getExpressRouteCircuitPeering", "getExpressRouteCircuitPeeringOutput"], () => require("./getExpressRouteCircuitPeering"));
exports.getInterfaceEndpoint = null;
exports.getInterfaceEndpointOutput = null;
utilities.lazyLoad(exports, ["getInterfaceEndpoint", "getInterfaceEndpointOutput"], () => require("./getInterfaceEndpoint"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getSubnet = null;
exports.getSubnetOutput = null;
utilities.lazyLoad(exports, ["getSubnet", "getSubnetOutput"], () => require("./getSubnet"));
exports.InterfaceEndpoint = null;
utilities.lazyLoad(exports, ["InterfaceEndpoint"], () => require("./interfaceEndpoint"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.Subnet = null;
utilities.lazyLoad(exports, ["Subnet"], () => require("./subnet"));
// Export enums:
__exportStar(require("../../types/enums/network/v20190201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20190201:ExpressRouteCircuitPeering":
                return new exports.ExpressRouteCircuitPeering(name, undefined, { urn });
            case "azure-native:network/v20190201:InterfaceEndpoint":
                return new exports.InterfaceEndpoint(name, undefined, { urn });
            case "azure-native:network/v20190201:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:network/v20190201:Subnet":
                return new exports.Subnet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20190201", _module);
