// @pulumi/azure-native/managednetworkfabric/v20230201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutePolicy = exports.NetworkToNetworkInterconnect = exports.NetworkRack = exports.NetworkInterface = exports.NetworkFabricController = exports.NetworkFabric = exports.NetworkDevice = exports.L3IsolationDomain = exports.L2IsolationDomain = exports.IpPrefix = exports.IpExtendedCommunity = exports.IpCommunity = exports.InternalNetwork = exports.getRoutePolicyOutput = exports.getRoutePolicy = exports.getNetworkToNetworkInterconnectOutput = exports.getNetworkToNetworkInterconnect = exports.getNetworkRackOutput = exports.getNetworkRack = exports.getNetworkInterfaceStatusOutput = exports.getNetworkInterfaceStatus = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getNetworkFabricControllerOutput = exports.getNetworkFabricController = exports.getNetworkFabricOutput = exports.getNetworkFabric = exports.getNetworkDeviceStatusOutput = exports.getNetworkDeviceStatus = exports.getNetworkDeviceOutput = exports.getNetworkDevice = exports.getL3IsolationDomainOutput = exports.getL3IsolationDomain = exports.getL2IsolationDomainOutput = exports.getL2IsolationDomain = exports.getIpPrefixOutput = exports.getIpPrefix = exports.getIpExtendedCommunityOutput = exports.getIpExtendedCommunity = exports.getIpCommunityOutput = exports.getIpCommunity = exports.getInternalNetworkOutput = exports.getInternalNetwork = exports.getExternalNetworkOutput = exports.getExternalNetwork = exports.getAccessControlListOutput = exports.getAccessControlList = exports.ExternalNetwork = exports.AccessControlList = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AccessControlList = null;
utilities.lazyLoad(exports, ["AccessControlList"], () => require("./accessControlList"));
exports.ExternalNetwork = null;
utilities.lazyLoad(exports, ["ExternalNetwork"], () => require("./externalNetwork"));
exports.getAccessControlList = null;
exports.getAccessControlListOutput = null;
utilities.lazyLoad(exports, ["getAccessControlList", "getAccessControlListOutput"], () => require("./getAccessControlList"));
exports.getExternalNetwork = null;
exports.getExternalNetworkOutput = null;
utilities.lazyLoad(exports, ["getExternalNetwork", "getExternalNetworkOutput"], () => require("./getExternalNetwork"));
exports.getInternalNetwork = null;
exports.getInternalNetworkOutput = null;
utilities.lazyLoad(exports, ["getInternalNetwork", "getInternalNetworkOutput"], () => require("./getInternalNetwork"));
exports.getIpCommunity = null;
exports.getIpCommunityOutput = null;
utilities.lazyLoad(exports, ["getIpCommunity", "getIpCommunityOutput"], () => require("./getIpCommunity"));
exports.getIpExtendedCommunity = null;
exports.getIpExtendedCommunityOutput = null;
utilities.lazyLoad(exports, ["getIpExtendedCommunity", "getIpExtendedCommunityOutput"], () => require("./getIpExtendedCommunity"));
exports.getIpPrefix = null;
exports.getIpPrefixOutput = null;
utilities.lazyLoad(exports, ["getIpPrefix", "getIpPrefixOutput"], () => require("./getIpPrefix"));
exports.getL2IsolationDomain = null;
exports.getL2IsolationDomainOutput = null;
utilities.lazyLoad(exports, ["getL2IsolationDomain", "getL2IsolationDomainOutput"], () => require("./getL2IsolationDomain"));
exports.getL3IsolationDomain = null;
exports.getL3IsolationDomainOutput = null;
utilities.lazyLoad(exports, ["getL3IsolationDomain", "getL3IsolationDomainOutput"], () => require("./getL3IsolationDomain"));
exports.getNetworkDevice = null;
exports.getNetworkDeviceOutput = null;
utilities.lazyLoad(exports, ["getNetworkDevice", "getNetworkDeviceOutput"], () => require("./getNetworkDevice"));
exports.getNetworkDeviceStatus = null;
exports.getNetworkDeviceStatusOutput = null;
utilities.lazyLoad(exports, ["getNetworkDeviceStatus", "getNetworkDeviceStatusOutput"], () => require("./getNetworkDeviceStatus"));
exports.getNetworkFabric = null;
exports.getNetworkFabricOutput = null;
utilities.lazyLoad(exports, ["getNetworkFabric", "getNetworkFabricOutput"], () => require("./getNetworkFabric"));
exports.getNetworkFabricController = null;
exports.getNetworkFabricControllerOutput = null;
utilities.lazyLoad(exports, ["getNetworkFabricController", "getNetworkFabricControllerOutput"], () => require("./getNetworkFabricController"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getNetworkInterfaceStatus = null;
exports.getNetworkInterfaceStatusOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterfaceStatus", "getNetworkInterfaceStatusOutput"], () => require("./getNetworkInterfaceStatus"));
exports.getNetworkRack = null;
exports.getNetworkRackOutput = null;
utilities.lazyLoad(exports, ["getNetworkRack", "getNetworkRackOutput"], () => require("./getNetworkRack"));
exports.getNetworkToNetworkInterconnect = null;
exports.getNetworkToNetworkInterconnectOutput = null;
utilities.lazyLoad(exports, ["getNetworkToNetworkInterconnect", "getNetworkToNetworkInterconnectOutput"], () => require("./getNetworkToNetworkInterconnect"));
exports.getRoutePolicy = null;
exports.getRoutePolicyOutput = null;
utilities.lazyLoad(exports, ["getRoutePolicy", "getRoutePolicyOutput"], () => require("./getRoutePolicy"));
exports.InternalNetwork = null;
utilities.lazyLoad(exports, ["InternalNetwork"], () => require("./internalNetwork"));
exports.IpCommunity = null;
utilities.lazyLoad(exports, ["IpCommunity"], () => require("./ipCommunity"));
exports.IpExtendedCommunity = null;
utilities.lazyLoad(exports, ["IpExtendedCommunity"], () => require("./ipExtendedCommunity"));
exports.IpPrefix = null;
utilities.lazyLoad(exports, ["IpPrefix"], () => require("./ipPrefix"));
exports.L2IsolationDomain = null;
utilities.lazyLoad(exports, ["L2IsolationDomain"], () => require("./l2isolationDomain"));
exports.L3IsolationDomain = null;
utilities.lazyLoad(exports, ["L3IsolationDomain"], () => require("./l3isolationDomain"));
exports.NetworkDevice = null;
utilities.lazyLoad(exports, ["NetworkDevice"], () => require("./networkDevice"));
exports.NetworkFabric = null;
utilities.lazyLoad(exports, ["NetworkFabric"], () => require("./networkFabric"));
exports.NetworkFabricController = null;
utilities.lazyLoad(exports, ["NetworkFabricController"], () => require("./networkFabricController"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.NetworkRack = null;
utilities.lazyLoad(exports, ["NetworkRack"], () => require("./networkRack"));
exports.NetworkToNetworkInterconnect = null;
utilities.lazyLoad(exports, ["NetworkToNetworkInterconnect"], () => require("./networkToNetworkInterconnect"));
exports.RoutePolicy = null;
utilities.lazyLoad(exports, ["RoutePolicy"], () => require("./routePolicy"));
// Export enums:
__exportStar(require("../../types/enums/managednetworkfabric/v20230201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managednetworkfabric/v20230201preview:AccessControlList":
                return new exports.AccessControlList(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:ExternalNetwork":
                return new exports.ExternalNetwork(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:InternalNetwork":
                return new exports.InternalNetwork(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:IpCommunity":
                return new exports.IpCommunity(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:IpExtendedCommunity":
                return new exports.IpExtendedCommunity(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:IpPrefix":
                return new exports.IpPrefix(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:L2IsolationDomain":
                return new exports.L2IsolationDomain(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:L3IsolationDomain":
                return new exports.L3IsolationDomain(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkDevice":
                return new exports.NetworkDevice(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkFabric":
                return new exports.NetworkFabric(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkFabricController":
                return new exports.NetworkFabricController(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkRack":
                return new exports.NetworkRack(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:NetworkToNetworkInterconnect":
                return new exports.NetworkToNetworkInterconnect(name, undefined, { urn });
            case "azure-native:managednetworkfabric/v20230201preview:RoutePolicy":
                return new exports.RoutePolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managednetworkfabric/v20230201preview", _module);
