// @pulumi/azure-native/types/enums/azurestackhci/v20230601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsServerSubscription = exports.UpdateSummariesPropertiesState = exports.UpdateRunPropertiesState = exports.State = exports.SoftwareAssuranceStatus = exports.SoftwareAssuranceIntent = exports.ServiceName = exports.ManagedServiceIdentityType = exports.DiagnosticLevel = exports.AvailabilityType = void 0;
exports.AvailabilityType = {
    Local: "Local",
    Online: "Online",
    Notify: "Notify",
};
exports.DiagnosticLevel = {
    Off: "Off",
    Basic: "Basic",
    Enhanced: "Enhanced",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.ServiceName = {
    WAC: "WAC",
};
exports.SoftwareAssuranceIntent = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.SoftwareAssuranceStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.State = {
    HasPrerequisite: "HasPrerequisite",
    Obsolete: "Obsolete",
    Ready: "Ready",
    NotApplicableBecauseAnotherUpdateIsInProgress: "NotApplicableBecauseAnotherUpdateIsInProgress",
    Preparing: "Preparing",
    Installing: "Installing",
    Installed: "Installed",
    PreparationFailed: "PreparationFailed",
    InstallationFailed: "InstallationFailed",
    Invalid: "Invalid",
    Recalled: "Recalled",
    Downloading: "Downloading",
    DownloadFailed: "DownloadFailed",
    HealthChecking: "HealthChecking",
    HealthCheckFailed: "HealthCheckFailed",
    ReadyToInstall: "ReadyToInstall",
    ScanInProgress: "ScanInProgress",
    ScanFailed: "ScanFailed",
};
exports.UpdateRunPropertiesState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    InProgress: "InProgress",
    Failed: "Failed",
};
exports.UpdateSummariesPropertiesState = {
    Unknown: "Unknown",
    AppliedSuccessfully: "AppliedSuccessfully",
    UpdateAvailable: "UpdateAvailable",
    UpdateInProgress: "UpdateInProgress",
    UpdateFailed: "UpdateFailed",
    NeedsAttention: "NeedsAttention",
    PreparationInProgress: "PreparationInProgress",
    PreparationFailed: "PreparationFailed",
};
exports.WindowsServerSubscription = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
