// @pulumi/azure-native/types/enums/databricks/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredNsgRules = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.KeySource = exports.EncryptionKeySource = exports.v20250301preview = exports.v20240901preview = exports.v20240501 = exports.v20230915preview = exports.v20230501 = exports.v20230201 = exports.v20220401preview = void 0;
// Export sub-modules:
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20230201 = require("./v20230201");
exports.v20230201 = v20230201;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20230915preview = require("./v20230915preview");
exports.v20230915preview = v20230915preview;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20250301preview = require("./v20250301preview");
exports.v20250301preview = v20250301preview;
exports.EncryptionKeySource = {
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.KeySource = {
    Default: "Default",
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RequiredNsgRules = {
    AllRules: "AllRules",
    NoAzureDatabricksRules: "NoAzureDatabricksRules",
    NoAzureServiceRules: "NoAzureServiceRules",
};
