// @pulumi/azure-native/web/v20160801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSlot = exports.WebAppSiteExtensionSlot = exports.WebAppSiteExtension = exports.WebAppPremierAddOnSlot = exports.WebAppPremierAddOn = exports.WebAppInstanceFunctionSlot = exports.WebAppFunction = exports.WebAppDiagnosticLogsConfigurationSlot = exports.WebAppBackupConfigurationSlot = exports.WebAppBackupConfiguration = exports.WebApp = exports.listWebAppSyncFunctionTriggersSlotOutput = exports.listWebAppSyncFunctionTriggersSlot = exports.listWebAppSyncFunctionTriggersOutput = exports.listWebAppSyncFunctionTriggers = exports.listWebAppPublishingCredentialsSlotOutput = exports.listWebAppPublishingCredentialsSlot = exports.listWebAppPublishingCredentialsOutput = exports.listWebAppPublishingCredentials = exports.listWebAppFunctionSecretsSlotOutput = exports.listWebAppFunctionSecretsSlot = exports.listWebAppFunctionSecretsOutput = exports.listWebAppFunctionSecrets = exports.listWebAppBackupStatusSecretsSlotOutput = exports.listWebAppBackupStatusSecretsSlot = exports.listWebAppBackupStatusSecretsOutput = exports.listWebAppBackupStatusSecrets = exports.listWebAppBackupConfigurationSlotOutput = exports.listWebAppBackupConfigurationSlot = exports.listWebAppBackupConfigurationOutput = exports.listWebAppBackupConfiguration = exports.getWebAppSlotOutput = exports.getWebAppSlot = exports.getWebAppSiteExtensionSlotOutput = exports.getWebAppSiteExtensionSlot = exports.getWebAppSiteExtensionOutput = exports.getWebAppSiteExtension = exports.getWebAppPremierAddOnSlotOutput = exports.getWebAppPremierAddOnSlot = exports.getWebAppPremierAddOnOutput = exports.getWebAppPremierAddOn = exports.getWebAppInstanceFunctionSlotOutput = exports.getWebAppInstanceFunctionSlot = exports.getWebAppFunctionOutput = exports.getWebAppFunction = exports.getWebAppDiagnosticLogsConfigurationSlotOutput = exports.getWebAppDiagnosticLogsConfigurationSlot = exports.getWebAppOutput = exports.getWebApp = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWebApp = null;
exports.getWebAppOutput = null;
utilities.lazyLoad(exports, ["getWebApp", "getWebAppOutput"], () => require("./getWebApp"));
exports.getWebAppDiagnosticLogsConfigurationSlot = null;
exports.getWebAppDiagnosticLogsConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfigurationSlot", "getWebAppDiagnosticLogsConfigurationSlotOutput"], () => require("./getWebAppDiagnosticLogsConfigurationSlot"));
exports.getWebAppFunction = null;
exports.getWebAppFunctionOutput = null;
utilities.lazyLoad(exports, ["getWebAppFunction", "getWebAppFunctionOutput"], () => require("./getWebAppFunction"));
exports.getWebAppInstanceFunctionSlot = null;
exports.getWebAppInstanceFunctionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppInstanceFunctionSlot", "getWebAppInstanceFunctionSlotOutput"], () => require("./getWebAppInstanceFunctionSlot"));
exports.getWebAppPremierAddOn = null;
exports.getWebAppPremierAddOnOutput = null;
utilities.lazyLoad(exports, ["getWebAppPremierAddOn", "getWebAppPremierAddOnOutput"], () => require("./getWebAppPremierAddOn"));
exports.getWebAppPremierAddOnSlot = null;
exports.getWebAppPremierAddOnSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppPremierAddOnSlot", "getWebAppPremierAddOnSlotOutput"], () => require("./getWebAppPremierAddOnSlot"));
exports.getWebAppSiteExtension = null;
exports.getWebAppSiteExtensionOutput = null;
utilities.lazyLoad(exports, ["getWebAppSiteExtension", "getWebAppSiteExtensionOutput"], () => require("./getWebAppSiteExtension"));
exports.getWebAppSiteExtensionSlot = null;
exports.getWebAppSiteExtensionSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSiteExtensionSlot", "getWebAppSiteExtensionSlotOutput"], () => require("./getWebAppSiteExtensionSlot"));
exports.getWebAppSlot = null;
exports.getWebAppSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSlot", "getWebAppSlotOutput"], () => require("./getWebAppSlot"));
exports.listWebAppBackupConfiguration = null;
exports.listWebAppBackupConfigurationOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupConfiguration", "listWebAppBackupConfigurationOutput"], () => require("./listWebAppBackupConfiguration"));
exports.listWebAppBackupConfigurationSlot = null;
exports.listWebAppBackupConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupConfigurationSlot", "listWebAppBackupConfigurationSlotOutput"], () => require("./listWebAppBackupConfigurationSlot"));
exports.listWebAppBackupStatusSecrets = null;
exports.listWebAppBackupStatusSecretsOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecrets", "listWebAppBackupStatusSecretsOutput"], () => require("./listWebAppBackupStatusSecrets"));
exports.listWebAppBackupStatusSecretsSlot = null;
exports.listWebAppBackupStatusSecretsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecretsSlot", "listWebAppBackupStatusSecretsSlotOutput"], () => require("./listWebAppBackupStatusSecretsSlot"));
exports.listWebAppFunctionSecrets = null;
exports.listWebAppFunctionSecretsOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionSecrets", "listWebAppFunctionSecretsOutput"], () => require("./listWebAppFunctionSecrets"));
exports.listWebAppFunctionSecretsSlot = null;
exports.listWebAppFunctionSecretsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppFunctionSecretsSlot", "listWebAppFunctionSecretsSlotOutput"], () => require("./listWebAppFunctionSecretsSlot"));
exports.listWebAppPublishingCredentials = null;
exports.listWebAppPublishingCredentialsOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentials", "listWebAppPublishingCredentialsOutput"], () => require("./listWebAppPublishingCredentials"));
exports.listWebAppPublishingCredentialsSlot = null;
exports.listWebAppPublishingCredentialsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentialsSlot", "listWebAppPublishingCredentialsSlotOutput"], () => require("./listWebAppPublishingCredentialsSlot"));
exports.listWebAppSyncFunctionTriggers = null;
exports.listWebAppSyncFunctionTriggersOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggers", "listWebAppSyncFunctionTriggersOutput"], () => require("./listWebAppSyncFunctionTriggers"));
exports.listWebAppSyncFunctionTriggersSlot = null;
exports.listWebAppSyncFunctionTriggersSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggersSlot", "listWebAppSyncFunctionTriggersSlotOutput"], () => require("./listWebAppSyncFunctionTriggersSlot"));
exports.WebApp = null;
utilities.lazyLoad(exports, ["WebApp"], () => require("./webApp"));
exports.WebAppBackupConfiguration = null;
utilities.lazyLoad(exports, ["WebAppBackupConfiguration"], () => require("./webAppBackupConfiguration"));
exports.WebAppBackupConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppBackupConfigurationSlot"], () => require("./webAppBackupConfigurationSlot"));
exports.WebAppDiagnosticLogsConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfigurationSlot"], () => require("./webAppDiagnosticLogsConfigurationSlot"));
exports.WebAppFunction = null;
utilities.lazyLoad(exports, ["WebAppFunction"], () => require("./webAppFunction"));
exports.WebAppInstanceFunctionSlot = null;
utilities.lazyLoad(exports, ["WebAppInstanceFunctionSlot"], () => require("./webAppInstanceFunctionSlot"));
exports.WebAppPremierAddOn = null;
utilities.lazyLoad(exports, ["WebAppPremierAddOn"], () => require("./webAppPremierAddOn"));
exports.WebAppPremierAddOnSlot = null;
utilities.lazyLoad(exports, ["WebAppPremierAddOnSlot"], () => require("./webAppPremierAddOnSlot"));
exports.WebAppSiteExtension = null;
utilities.lazyLoad(exports, ["WebAppSiteExtension"], () => require("./webAppSiteExtension"));
exports.WebAppSiteExtensionSlot = null;
utilities.lazyLoad(exports, ["WebAppSiteExtensionSlot"], () => require("./webAppSiteExtensionSlot"));
exports.WebAppSlot = null;
utilities.lazyLoad(exports, ["WebAppSlot"], () => require("./webAppSlot"));
// Export enums:
__exportStar(require("../../types/enums/web/v20160801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20160801:WebApp":
                return new exports.WebApp(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppBackupConfiguration":
                return new exports.WebAppBackupConfiguration(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppBackupConfigurationSlot":
                return new exports.WebAppBackupConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppDiagnosticLogsConfigurationSlot":
                return new exports.WebAppDiagnosticLogsConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppFunction":
                return new exports.WebAppFunction(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppInstanceFunctionSlot":
                return new exports.WebAppInstanceFunctionSlot(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppPremierAddOn":
                return new exports.WebAppPremierAddOn(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppPremierAddOnSlot":
                return new exports.WebAppPremierAddOnSlot(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppSiteExtension":
                return new exports.WebAppSiteExtension(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppSiteExtensionSlot":
                return new exports.WebAppSiteExtensionSlot(name, undefined, { urn });
            case "azure-native:web/v20160801:WebAppSlot":
                return new exports.WebAppSlot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20160801", _module);
