// @pulumi/azure-native/web/v20160801/webAppBackupConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppBackupConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * Description of a backup which will be performed.
 */
class WebAppBackupConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing WebAppBackupConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppBackupConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppBackupConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppBackupConfiguration.__pulumiType;
    }
    /**
     * Create a WebAppBackupConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.backupRequestName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backupRequestName'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.storageAccountUrl === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountUrl'");
            }
            resourceInputs["backupRequestName"] = args ? args.backupRequestName : undefined;
            resourceInputs["backupSchedule"] = args ? (args.backupSchedule ? pulumi.output(args.backupSchedule).apply(inputs.web.v20160801.backupScheduleArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["databases"] = args ? args.databases : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageAccountUrl"] = args ? args.storageAccountUrl : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["backupRequestName"] = undefined /*out*/;
            resourceInputs["backupSchedule"] = undefined /*out*/;
            resourceInputs["databases"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["storageAccountUrl"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppBackupConfiguration" }, { type: "azure-native:web/v20180201:WebAppBackupConfiguration" }, { type: "azure-native:web/v20181101:WebAppBackupConfiguration" }, { type: "azure-native:web/v20190801:WebAppBackupConfiguration" }, { type: "azure-native:web/v20200601:WebAppBackupConfiguration" }, { type: "azure-native:web/v20200901:WebAppBackupConfiguration" }, { type: "azure-native:web/v20201001:WebAppBackupConfiguration" }, { type: "azure-native:web/v20201201:WebAppBackupConfiguration" }, { type: "azure-native:web/v20210101:WebAppBackupConfiguration" }, { type: "azure-native:web/v20210115:WebAppBackupConfiguration" }, { type: "azure-native:web/v20210201:WebAppBackupConfiguration" }, { type: "azure-native:web/v20210301:WebAppBackupConfiguration" }, { type: "azure-native:web/v20220301:WebAppBackupConfiguration" }, { type: "azure-native:web/v20220901:WebAppBackupConfiguration" }, { type: "azure-native:web/v20230101:WebAppBackupConfiguration" }, { type: "azure-native:web/v20231201:WebAppBackupConfiguration" }, { type: "azure-native:web/v20240401:WebAppBackupConfiguration" }, { type: "azure-native:web:WebAppBackupConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppBackupConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppBackupConfiguration = WebAppBackupConfiguration;
/** @internal */
WebAppBackupConfiguration.__pulumiType = 'azure-native:web/v20160801:WebAppBackupConfiguration';
