// @pulumi/azure-native/types/enums/scvmm/v20231007/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningAction = exports.LimitCpuForMigration = exports.InventoryType = exports.DynamicMemoryEnabled = exports.CreateDiffDisk = exports.AllocationMethod = void 0;
exports.AllocationMethod = {
    /**
     * Dynamically allocated address.
     */
    Dynamic: "Dynamic",
    /**
     * Statically allocated address.
     */
    Static: "Static",
};
exports.CreateDiffDisk = {
    /**
     * Enable create diff disk.
     */
    True: "true",
    /**
     * Disable create diff disk.
     */
    False: "false",
};
exports.DynamicMemoryEnabled = {
    /**
     * Enable dynamic memory.
     */
    True: "true",
    /**
     * Disable dynamic memory.
     */
    False: "false",
};
exports.InventoryType = {
    /**
     * Cloud inventory type
     */
    Cloud: "Cloud",
    /**
     * VirtualNetwork inventory type
     */
    VirtualNetwork: "VirtualNetwork",
    /**
     * VirtualMachine inventory type
     */
    VirtualMachine: "VirtualMachine",
    /**
     * VirtualMachineTemplate inventory type
     */
    VirtualMachineTemplate: "VirtualMachineTemplate",
};
exports.LimitCpuForMigration = {
    /**
     * Enable limit CPU for migration.
     */
    True: "true",
    /**
     * Disable limit CPU for migration.
     */
    False: "false",
};
exports.ProvisioningAction = {
    /**
     * Install guest agent.
     */
    Install: "install",
    /**
     * Uninstall guest agent.
     */
    Uninstall: "uninstall",
    /**
     * Repair guest agent.
     */
    Repair: "repair",
};
