// @pulumi/azure-native/types/enums/botservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.MsaAppType = exports.Kind = exports.v20230915preview = exports.v20220915 = exports.v20180712 = void 0;
// Export sub-modules:
const v20180712 = require("./v20180712");
exports.v20180712 = v20180712;
const v20220915 = require("./v20220915");
exports.v20220915 = v20220915;
const v20230915preview = require("./v20230915preview");
exports.v20230915preview = v20230915preview;
exports.Kind = {
    Sdk: "sdk",
    Designer: "designer",
    Bot: "bot",
    Function: "function",
    Azurebot: "azurebot",
};
exports.MsaAppType = {
    UserAssignedMSI: "UserAssignedMSI",
    SingleTenant: "SingleTenant",
    MultiTenant: "MultiTenant",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SkuName = {
    F0: "F0",
    S1: "S1",
};
