// @pulumi/azure-native/healthbot/v20240201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listBotSecretsOutput = exports.listBotSecrets = exports.getBotOutput = exports.getBot = exports.Bot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Bot = null;
utilities.lazyLoad(exports, ["Bot"], () => require("./bot"));
exports.getBot = null;
exports.getBotOutput = null;
utilities.lazyLoad(exports, ["getBot", "getBotOutput"], () => require("./getBot"));
exports.listBotSecrets = null;
exports.listBotSecretsOutput = null;
utilities.lazyLoad(exports, ["listBotSecrets", "listBotSecretsOutput"], () => require("./listBotSecrets"));
// Export enums:
__exportStar(require("../../types/enums/healthbot/v20240201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:healthbot/v20240201:Bot":
                return new exports.Bot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "healthbot/v20240201", _module);
