// @pulumi/azure-native/scheduler/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20160301 = exports.JobCollection = exports.Job = exports.getJobCollectionOutput = exports.getJobCollection = exports.getJobOutput = exports.getJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getJobCollection = null;
exports.getJobCollectionOutput = null;
utilities.lazyLoad(exports, ["getJobCollection", "getJobCollectionOutput"], () => require("./getJobCollection"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.JobCollection = null;
utilities.lazyLoad(exports, ["JobCollection"], () => require("./jobCollection"));
// Export enums:
__exportStar(require("../types/enums/scheduler"), exports);
// Export sub-modules:
const v20160301 = require("./v20160301");
exports.v20160301 = v20160301;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:scheduler:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:scheduler:JobCollection":
                return new exports.JobCollection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "scheduler", _module);
