// @pulumi/azure-native/types/enums/databasewatcher/v20230901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetAuthenticationType = exports.ManagedServiceIdentityType = exports.KustoOfferingType = void 0;
exports.KustoOfferingType = {
    /**
     * The Azure Data Explorer cluster Kusto offering.
     */
    Adx: "adx",
    /**
     * The free Azure Data Explorer cluster Kusto offering.
     */
    Free: "free",
    /**
     * The Fabric Real-Time Analytics Kusto offering.
     */
    Fabric: "fabric",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.TargetAuthenticationType = {
    /**
     * The Azure Active Directory authentication.
     */
    Aad: "Aad",
    /**
     * The SQL password authentication.
     */
    Sql: "Sql",
};
