// @pulumi/azure-native/datalakestore/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20161101 = exports.VirtualNetworkRule = exports.TrustedIdProvider = exports.getVirtualNetworkRuleOutput = exports.getVirtualNetworkRule = exports.getTrustedIdProviderOutput = exports.getTrustedIdProvider = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getAccountOutput = exports.getAccount = exports.FirewallRule = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getTrustedIdProvider = null;
exports.getTrustedIdProviderOutput = null;
utilities.lazyLoad(exports, ["getTrustedIdProvider", "getTrustedIdProviderOutput"], () => require("./getTrustedIdProvider"));
exports.getVirtualNetworkRule = null;
exports.getVirtualNetworkRuleOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkRule", "getVirtualNetworkRuleOutput"], () => require("./getVirtualNetworkRule"));
exports.TrustedIdProvider = null;
utilities.lazyLoad(exports, ["TrustedIdProvider"], () => require("./trustedIdProvider"));
exports.VirtualNetworkRule = null;
utilities.lazyLoad(exports, ["VirtualNetworkRule"], () => require("./virtualNetworkRule"));
// Export enums:
__exportStar(require("../types/enums/datalakestore"), exports);
// Export sub-modules:
const v20161101 = require("./v20161101");
exports.v20161101 = v20161101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datalakestore:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:datalakestore:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:datalakestore:TrustedIdProvider":
                return new exports.TrustedIdProvider(name, undefined, { urn });
            case "azure-native:datalakestore:VirtualNetworkRule":
                return new exports.VirtualNetworkRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datalakestore", _module);
