// @pulumi/azure-native/types/enums/quantum/v20231113preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderStatus = exports.ManagedServiceIdentityType = void 0;
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ProviderStatus = {
    /**
     * The provider is successfully provisioned.
     */
    Succeeded: "Succeeded",
    /**
     * The provider is starting provisioning.
     */
    Launching: "Launching",
    /**
     * The provider is updating.
     */
    Updating: "Updating",
    /**
     * The provider is deleting.
     */
    Deleting: "Deleting",
    /**
     * The provider is deleted.
     */
    Deleted: "Deleted",
    /**
     * The provider is failed.
     */
    Failed: "Failed",
};
