// @pulumi/azure-native/containerregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listRegistryCredentialsOutput = exports.listRegistryCredentials = exports.listRegistryBuildSourceUploadUrlOutput = exports.listRegistryBuildSourceUploadUrl = exports.listAgentPoolQueueStatusOutput = exports.listAgentPoolQueueStatus = exports.ImportPipeline = exports.getWebhookCallbackConfigOutput = exports.getWebhookCallbackConfig = exports.getWebhookOutput = exports.getWebhook = exports.getTokenOutput = exports.getToken = exports.getTaskRunOutput = exports.getTaskRun = exports.getTaskOutput = exports.getTask = exports.getScopeMapOutput = exports.getScopeMap = exports.getReplicationOutput = exports.getReplication = exports.getRegistryOutput = exports.getRegistry = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getPipelineRunOutput = exports.getPipelineRun = exports.getImportPipelineOutput = exports.getImportPipeline = exports.getExportPipelineOutput = exports.getExportPipeline = exports.getCredentialSetOutput = exports.getCredentialSet = exports.getConnectedRegistryOutput = exports.getConnectedRegistry = exports.getCacheRuleOutput = exports.getCacheRule = exports.getArchiveVersionOutput = exports.getArchiveVersion = exports.getArchifeOutput = exports.getArchife = exports.getAgentPoolOutput = exports.getAgentPool = exports.ExportPipeline = exports.CredentialSet = exports.ConnectedRegistry = exports.CacheRule = exports.ArchiveVersion = exports.Archife = exports.AgentPool = void 0;
exports.v20241101preview = exports.v20231101preview = exports.v20230801preview = exports.v20230701 = exports.v20230601preview = exports.v20230101preview = exports.v20221201 = exports.v20190601preview = exports.v20190501 = exports.v20190401 = exports.v20180901 = exports.v20170301 = exports.Webhook = exports.Token = exports.TaskRun = exports.Task = exports.ScopeMap = exports.Replication = exports.Registry = exports.PrivateEndpointConnection = exports.PipelineRun = exports.listWebhookEventsOutput = exports.listWebhookEvents = exports.listTaskRunDetailsOutput = exports.listTaskRunDetails = exports.listTaskDetailsOutput = exports.listTaskDetails = exports.listRunLogSasUrlOutput = exports.listRunLogSasUrl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.Archife = null;
utilities.lazyLoad(exports, ["Archife"], () => require("./archife"));
exports.ArchiveVersion = null;
utilities.lazyLoad(exports, ["ArchiveVersion"], () => require("./archiveVersion"));
exports.CacheRule = null;
utilities.lazyLoad(exports, ["CacheRule"], () => require("./cacheRule"));
exports.ConnectedRegistry = null;
utilities.lazyLoad(exports, ["ConnectedRegistry"], () => require("./connectedRegistry"));
exports.CredentialSet = null;
utilities.lazyLoad(exports, ["CredentialSet"], () => require("./credentialSet"));
exports.ExportPipeline = null;
utilities.lazyLoad(exports, ["ExportPipeline"], () => require("./exportPipeline"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getArchife = null;
exports.getArchifeOutput = null;
utilities.lazyLoad(exports, ["getArchife", "getArchifeOutput"], () => require("./getArchife"));
exports.getArchiveVersion = null;
exports.getArchiveVersionOutput = null;
utilities.lazyLoad(exports, ["getArchiveVersion", "getArchiveVersionOutput"], () => require("./getArchiveVersion"));
exports.getCacheRule = null;
exports.getCacheRuleOutput = null;
utilities.lazyLoad(exports, ["getCacheRule", "getCacheRuleOutput"], () => require("./getCacheRule"));
exports.getConnectedRegistry = null;
exports.getConnectedRegistryOutput = null;
utilities.lazyLoad(exports, ["getConnectedRegistry", "getConnectedRegistryOutput"], () => require("./getConnectedRegistry"));
exports.getCredentialSet = null;
exports.getCredentialSetOutput = null;
utilities.lazyLoad(exports, ["getCredentialSet", "getCredentialSetOutput"], () => require("./getCredentialSet"));
exports.getExportPipeline = null;
exports.getExportPipelineOutput = null;
utilities.lazyLoad(exports, ["getExportPipeline", "getExportPipelineOutput"], () => require("./getExportPipeline"));
exports.getImportPipeline = null;
exports.getImportPipelineOutput = null;
utilities.lazyLoad(exports, ["getImportPipeline", "getImportPipelineOutput"], () => require("./getImportPipeline"));
exports.getPipelineRun = null;
exports.getPipelineRunOutput = null;
utilities.lazyLoad(exports, ["getPipelineRun", "getPipelineRunOutput"], () => require("./getPipelineRun"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getRegistry = null;
exports.getRegistryOutput = null;
utilities.lazyLoad(exports, ["getRegistry", "getRegistryOutput"], () => require("./getRegistry"));
exports.getReplication = null;
exports.getReplicationOutput = null;
utilities.lazyLoad(exports, ["getReplication", "getReplicationOutput"], () => require("./getReplication"));
exports.getScopeMap = null;
exports.getScopeMapOutput = null;
utilities.lazyLoad(exports, ["getScopeMap", "getScopeMapOutput"], () => require("./getScopeMap"));
exports.getTask = null;
exports.getTaskOutput = null;
utilities.lazyLoad(exports, ["getTask", "getTaskOutput"], () => require("./getTask"));
exports.getTaskRun = null;
exports.getTaskRunOutput = null;
utilities.lazyLoad(exports, ["getTaskRun", "getTaskRunOutput"], () => require("./getTaskRun"));
exports.getToken = null;
exports.getTokenOutput = null;
utilities.lazyLoad(exports, ["getToken", "getTokenOutput"], () => require("./getToken"));
exports.getWebhook = null;
exports.getWebhookOutput = null;
utilities.lazyLoad(exports, ["getWebhook", "getWebhookOutput"], () => require("./getWebhook"));
exports.getWebhookCallbackConfig = null;
exports.getWebhookCallbackConfigOutput = null;
utilities.lazyLoad(exports, ["getWebhookCallbackConfig", "getWebhookCallbackConfigOutput"], () => require("./getWebhookCallbackConfig"));
exports.ImportPipeline = null;
utilities.lazyLoad(exports, ["ImportPipeline"], () => require("./importPipeline"));
exports.listAgentPoolQueueStatus = null;
exports.listAgentPoolQueueStatusOutput = null;
utilities.lazyLoad(exports, ["listAgentPoolQueueStatus", "listAgentPoolQueueStatusOutput"], () => require("./listAgentPoolQueueStatus"));
exports.listRegistryBuildSourceUploadUrl = null;
exports.listRegistryBuildSourceUploadUrlOutput = null;
utilities.lazyLoad(exports, ["listRegistryBuildSourceUploadUrl", "listRegistryBuildSourceUploadUrlOutput"], () => require("./listRegistryBuildSourceUploadUrl"));
exports.listRegistryCredentials = null;
exports.listRegistryCredentialsOutput = null;
utilities.lazyLoad(exports, ["listRegistryCredentials", "listRegistryCredentialsOutput"], () => require("./listRegistryCredentials"));
exports.listRunLogSasUrl = null;
exports.listRunLogSasUrlOutput = null;
utilities.lazyLoad(exports, ["listRunLogSasUrl", "listRunLogSasUrlOutput"], () => require("./listRunLogSasUrl"));
exports.listTaskDetails = null;
exports.listTaskDetailsOutput = null;
utilities.lazyLoad(exports, ["listTaskDetails", "listTaskDetailsOutput"], () => require("./listTaskDetails"));
exports.listTaskRunDetails = null;
exports.listTaskRunDetailsOutput = null;
utilities.lazyLoad(exports, ["listTaskRunDetails", "listTaskRunDetailsOutput"], () => require("./listTaskRunDetails"));
exports.listWebhookEvents = null;
exports.listWebhookEventsOutput = null;
utilities.lazyLoad(exports, ["listWebhookEvents", "listWebhookEventsOutput"], () => require("./listWebhookEvents"));
exports.PipelineRun = null;
utilities.lazyLoad(exports, ["PipelineRun"], () => require("./pipelineRun"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.Replication = null;
utilities.lazyLoad(exports, ["Replication"], () => require("./replication"));
exports.ScopeMap = null;
utilities.lazyLoad(exports, ["ScopeMap"], () => require("./scopeMap"));
exports.Task = null;
utilities.lazyLoad(exports, ["Task"], () => require("./task"));
exports.TaskRun = null;
utilities.lazyLoad(exports, ["TaskRun"], () => require("./taskRun"));
exports.Token = null;
utilities.lazyLoad(exports, ["Token"], () => require("./token"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
// Export enums:
__exportStar(require("../types/enums/containerregistry"), exports);
// Export sub-modules:
const v20170301 = require("./v20170301");
exports.v20170301 = v20170301;
const v20180901 = require("./v20180901");
exports.v20180901 = v20180901;
const v20190401 = require("./v20190401");
exports.v20190401 = v20190401;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20221201 = require("./v20221201");
exports.v20221201 = v20221201;
const v20230101preview = require("./v20230101preview");
exports.v20230101preview = v20230101preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerregistry:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:containerregistry:Archife":
                return new exports.Archife(name, undefined, { urn });
            case "azure-native:containerregistry:ArchiveVersion":
                return new exports.ArchiveVersion(name, undefined, { urn });
            case "azure-native:containerregistry:CacheRule":
                return new exports.CacheRule(name, undefined, { urn });
            case "azure-native:containerregistry:ConnectedRegistry":
                return new exports.ConnectedRegistry(name, undefined, { urn });
            case "azure-native:containerregistry:CredentialSet":
                return new exports.CredentialSet(name, undefined, { urn });
            case "azure-native:containerregistry:ExportPipeline":
                return new exports.ExportPipeline(name, undefined, { urn });
            case "azure-native:containerregistry:ImportPipeline":
                return new exports.ImportPipeline(name, undefined, { urn });
            case "azure-native:containerregistry:PipelineRun":
                return new exports.PipelineRun(name, undefined, { urn });
            case "azure-native:containerregistry:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:containerregistry:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "azure-native:containerregistry:Replication":
                return new exports.Replication(name, undefined, { urn });
            case "azure-native:containerregistry:ScopeMap":
                return new exports.ScopeMap(name, undefined, { urn });
            case "azure-native:containerregistry:Task":
                return new exports.Task(name, undefined, { urn });
            case "azure-native:containerregistry:TaskRun":
                return new exports.TaskRun(name, undefined, { urn });
            case "azure-native:containerregistry:Token":
                return new exports.Token(name, undefined, { urn });
            case "azure-native:containerregistry:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerregistry", _module);
