// @pulumi/azure-native/apimanagement/v20210401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = exports.getSchemaOutput = exports.getSchema = exports.getApiManagementServiceDomainOwnershipIdentifierOutput = exports.getApiManagementServiceDomainOwnershipIdentifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getApiManagementServiceDomainOwnershipIdentifier = null;
exports.getApiManagementServiceDomainOwnershipIdentifierOutput = null;
utilities.lazyLoad(exports, ["getApiManagementServiceDomainOwnershipIdentifier", "getApiManagementServiceDomainOwnershipIdentifierOutput"], () => require("./getApiManagementServiceDomainOwnershipIdentifier"));
exports.getSchema = null;
exports.getSchemaOutput = null;
utilities.lazyLoad(exports, ["getSchema", "getSchemaOutput"], () => require("./getSchema"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20210401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20210401preview:Schema":
                return new exports.Schema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20210401preview", _module);
