// @pulumi/azure-native/cdn/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Profile = exports.Policy = exports.OriginGroup = exports.Origin = exports.KeyGroup = exports.getTunnelPolicyOutput = exports.getTunnelPolicy = exports.getSecurityPolicyOutput = exports.getSecurityPolicy = exports.getSecretOutput = exports.getSecret = exports.getRuleSetOutput = exports.getRuleSet = exports.getRuleOutput = exports.getRule = exports.getRouteOutput = exports.getRoute = exports.getProfileSupportedOptimizationTypesOutput = exports.getProfileSupportedOptimizationTypes = exports.getProfileOutput = exports.getProfile = exports.getPolicyOutput = exports.getPolicy = exports.getOriginGroupOutput = exports.getOriginGroup = exports.getOriginOutput = exports.getOrigin = exports.getKeyGroupOutput = exports.getKeyGroup = exports.getEndpointOutput = exports.getEndpoint = exports.getCustomDomainOutput = exports.getCustomDomain = exports.getAFDTargetGroupOutput = exports.getAFDTargetGroup = exports.getAFDOriginGroupOutput = exports.getAFDOriginGroup = exports.getAFDOriginOutput = exports.getAFDOrigin = exports.getAFDEndpointOutput = exports.getAFDEndpoint = exports.getAFDCustomDomainOutput = exports.getAFDCustomDomain = exports.Endpoint = exports.CustomDomain = exports.AFDTargetGroup = exports.AFDOriginGroup = exports.AFDOrigin = exports.AFDEndpoint = exports.AFDCustomDomain = void 0;
exports.v20240901 = exports.v20240722preview = exports.v20240601preview = exports.v20240501preview = exports.v20240201 = exports.v20230701preview = exports.v20230501 = exports.v20200901 = exports.TunnelPolicy = exports.SecurityPolicy = exports.Secret = exports.RuleSet = exports.Rule = exports.Route = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AFDCustomDomain = null;
utilities.lazyLoad(exports, ["AFDCustomDomain"], () => require("./afdcustomDomain"));
exports.AFDEndpoint = null;
utilities.lazyLoad(exports, ["AFDEndpoint"], () => require("./afdendpoint"));
exports.AFDOrigin = null;
utilities.lazyLoad(exports, ["AFDOrigin"], () => require("./afdorigin"));
exports.AFDOriginGroup = null;
utilities.lazyLoad(exports, ["AFDOriginGroup"], () => require("./afdoriginGroup"));
exports.AFDTargetGroup = null;
utilities.lazyLoad(exports, ["AFDTargetGroup"], () => require("./afdtargetGroup"));
exports.CustomDomain = null;
utilities.lazyLoad(exports, ["CustomDomain"], () => require("./customDomain"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getAFDCustomDomain = null;
exports.getAFDCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getAFDCustomDomain", "getAFDCustomDomainOutput"], () => require("./getAFDCustomDomain"));
exports.getAFDEndpoint = null;
exports.getAFDEndpointOutput = null;
utilities.lazyLoad(exports, ["getAFDEndpoint", "getAFDEndpointOutput"], () => require("./getAFDEndpoint"));
exports.getAFDOrigin = null;
exports.getAFDOriginOutput = null;
utilities.lazyLoad(exports, ["getAFDOrigin", "getAFDOriginOutput"], () => require("./getAFDOrigin"));
exports.getAFDOriginGroup = null;
exports.getAFDOriginGroupOutput = null;
utilities.lazyLoad(exports, ["getAFDOriginGroup", "getAFDOriginGroupOutput"], () => require("./getAFDOriginGroup"));
exports.getAFDTargetGroup = null;
exports.getAFDTargetGroupOutput = null;
utilities.lazyLoad(exports, ["getAFDTargetGroup", "getAFDTargetGroupOutput"], () => require("./getAFDTargetGroup"));
exports.getCustomDomain = null;
exports.getCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getCustomDomain", "getCustomDomainOutput"], () => require("./getCustomDomain"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getKeyGroup = null;
exports.getKeyGroupOutput = null;
utilities.lazyLoad(exports, ["getKeyGroup", "getKeyGroupOutput"], () => require("./getKeyGroup"));
exports.getOrigin = null;
exports.getOriginOutput = null;
utilities.lazyLoad(exports, ["getOrigin", "getOriginOutput"], () => require("./getOrigin"));
exports.getOriginGroup = null;
exports.getOriginGroupOutput = null;
utilities.lazyLoad(exports, ["getOriginGroup", "getOriginGroupOutput"], () => require("./getOriginGroup"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getProfile = null;
exports.getProfileOutput = null;
utilities.lazyLoad(exports, ["getProfile", "getProfileOutput"], () => require("./getProfile"));
exports.getProfileSupportedOptimizationTypes = null;
exports.getProfileSupportedOptimizationTypesOutput = null;
utilities.lazyLoad(exports, ["getProfileSupportedOptimizationTypes", "getProfileSupportedOptimizationTypesOutput"], () => require("./getProfileSupportedOptimizationTypes"));
exports.getRoute = null;
exports.getRouteOutput = null;
utilities.lazyLoad(exports, ["getRoute", "getRouteOutput"], () => require("./getRoute"));
exports.getRule = null;
exports.getRuleOutput = null;
utilities.lazyLoad(exports, ["getRule", "getRuleOutput"], () => require("./getRule"));
exports.getRuleSet = null;
exports.getRuleSetOutput = null;
utilities.lazyLoad(exports, ["getRuleSet", "getRuleSetOutput"], () => require("./getRuleSet"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSecurityPolicy = null;
exports.getSecurityPolicyOutput = null;
utilities.lazyLoad(exports, ["getSecurityPolicy", "getSecurityPolicyOutput"], () => require("./getSecurityPolicy"));
exports.getTunnelPolicy = null;
exports.getTunnelPolicyOutput = null;
utilities.lazyLoad(exports, ["getTunnelPolicy", "getTunnelPolicyOutput"], () => require("./getTunnelPolicy"));
exports.KeyGroup = null;
utilities.lazyLoad(exports, ["KeyGroup"], () => require("./keyGroup"));
exports.Origin = null;
utilities.lazyLoad(exports, ["Origin"], () => require("./origin"));
exports.OriginGroup = null;
utilities.lazyLoad(exports, ["OriginGroup"], () => require("./originGroup"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.Profile = null;
utilities.lazyLoad(exports, ["Profile"], () => require("./profile"));
exports.Route = null;
utilities.lazyLoad(exports, ["Route"], () => require("./route"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
exports.RuleSet = null;
utilities.lazyLoad(exports, ["RuleSet"], () => require("./ruleSet"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SecurityPolicy = null;
utilities.lazyLoad(exports, ["SecurityPolicy"], () => require("./securityPolicy"));
exports.TunnelPolicy = null;
utilities.lazyLoad(exports, ["TunnelPolicy"], () => require("./tunnelPolicy"));
// Export enums:
__exportStar(require("../types/enums/cdn"), exports);
// Export sub-modules:
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20240201 = require("./v20240201");
exports.v20240201 = v20240201;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240722preview = require("./v20240722preview");
exports.v20240722preview = v20240722preview;
const v20240901 = require("./v20240901");
exports.v20240901 = v20240901;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cdn:AFDCustomDomain":
                return new exports.AFDCustomDomain(name, undefined, { urn });
            case "azure-native:cdn:AFDEndpoint":
                return new exports.AFDEndpoint(name, undefined, { urn });
            case "azure-native:cdn:AFDOrigin":
                return new exports.AFDOrigin(name, undefined, { urn });
            case "azure-native:cdn:AFDOriginGroup":
                return new exports.AFDOriginGroup(name, undefined, { urn });
            case "azure-native:cdn:AFDTargetGroup":
                return new exports.AFDTargetGroup(name, undefined, { urn });
            case "azure-native:cdn:CustomDomain":
                return new exports.CustomDomain(name, undefined, { urn });
            case "azure-native:cdn:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "azure-native:cdn:KeyGroup":
                return new exports.KeyGroup(name, undefined, { urn });
            case "azure-native:cdn:Origin":
                return new exports.Origin(name, undefined, { urn });
            case "azure-native:cdn:OriginGroup":
                return new exports.OriginGroup(name, undefined, { urn });
            case "azure-native:cdn:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:cdn:Profile":
                return new exports.Profile(name, undefined, { urn });
            case "azure-native:cdn:Route":
                return new exports.Route(name, undefined, { urn });
            case "azure-native:cdn:Rule":
                return new exports.Rule(name, undefined, { urn });
            case "azure-native:cdn:RuleSet":
                return new exports.RuleSet(name, undefined, { urn });
            case "azure-native:cdn:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "azure-native:cdn:SecurityPolicy":
                return new exports.SecurityPolicy(name, undefined, { urn });
            case "azure-native:cdn:TunnelPolicy":
                return new exports.TunnelPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cdn", _module);
