// @pulumi/azure-native/security/v20190101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAutomationOutput = exports.getAutomation = exports.getAssessmentsMetadataSubscriptionOutput = exports.getAssessmentsMetadataSubscription = exports.getAlertsSuppressionRuleOutput = exports.getAlertsSuppressionRule = exports.Automation = exports.AssessmentsMetadataSubscription = exports.AlertsSuppressionRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AlertsSuppressionRule = null;
utilities.lazyLoad(exports, ["AlertsSuppressionRule"], () => require("./alertsSuppressionRule"));
exports.AssessmentsMetadataSubscription = null;
utilities.lazyLoad(exports, ["AssessmentsMetadataSubscription"], () => require("./assessmentsMetadataSubscription"));
exports.Automation = null;
utilities.lazyLoad(exports, ["Automation"], () => require("./automation"));
exports.getAlertsSuppressionRule = null;
exports.getAlertsSuppressionRuleOutput = null;
utilities.lazyLoad(exports, ["getAlertsSuppressionRule", "getAlertsSuppressionRuleOutput"], () => require("./getAlertsSuppressionRule"));
exports.getAssessmentsMetadataSubscription = null;
exports.getAssessmentsMetadataSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getAssessmentsMetadataSubscription", "getAssessmentsMetadataSubscriptionOutput"], () => require("./getAssessmentsMetadataSubscription"));
exports.getAutomation = null;
exports.getAutomationOutput = null;
utilities.lazyLoad(exports, ["getAutomation", "getAutomationOutput"], () => require("./getAutomation"));
// Export enums:
__exportStar(require("../../types/enums/security/v20190101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20190101preview:AlertsSuppressionRule":
                return new exports.AlertsSuppressionRule(name, undefined, { urn });
            case "azure-native:security/v20190101preview:AssessmentsMetadataSubscription":
                return new exports.AssessmentsMetadataSubscription(name, undefined, { urn });
            case "azure-native:security/v20190101preview:Automation":
                return new exports.Automation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20190101preview", _module);
