// @pulumi/azure-native/types/enums/eventgrid/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicSpacesConfigurationState = exports.TlsVersion = exports.StaticRoutingEnrichmentType = exports.SkuName = exports.RoutingIdentityType = exports.ResourceProvisioningState = exports.ReadinessState = exports.PublisherType = exports.PublicNetworkAccess = exports.PersistedConnectionStatus = exports.PermissionType = exports.PartnerTopicRoutingMode = exports.PartnerTopicActivationState = exports.PartnerDestinationActivationState = exports.PartnerConfigurationProvisioningState = exports.IpActionType = exports.InputSchemaMappingType = exports.InputSchema = exports.IdentityType = exports.FilterOperatorType = exports.EventSubscriptionIdentityType = exports.EventInputSchema = exports.EventDeliverySchema = exports.EventDefinitionKind = exports.EndpointType = exports.DeliverySchema = exports.DeliveryMode = exports.DeliveryAttributeMappingType = exports.DeadLetterEndPointType = exports.DataResidencyBoundary = exports.ClientState = exports.ClientCertificateValidationScheme = exports.ChannelType = exports.ChannelProvisioningState = exports.AlternativeAuthenticationNameSource = exports.AdvancedFilterOperatorType = exports.v20250215 = exports.v20241215preview = exports.v20240601preview = exports.v20231215preview = exports.v20230601preview = exports.v20220615 = exports.v20211015preview = exports.v20200401preview = void 0;
// Export sub-modules:
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20211015preview = require("./v20211015preview");
exports.v20211015preview = v20211015preview;
const v20220615 = require("./v20220615");
exports.v20220615 = v20220615;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20231215preview = require("./v20231215preview");
exports.v20231215preview = v20231215preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20241215preview = require("./v20241215preview");
exports.v20241215preview = v20241215preview;
const v20250215 = require("./v20250215");
exports.v20250215 = v20250215;
exports.AdvancedFilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
    NumberInRange: "NumberInRange",
    NumberNotInRange: "NumberNotInRange",
    StringNotBeginsWith: "StringNotBeginsWith",
    StringNotEndsWith: "StringNotEndsWith",
    StringNotContains: "StringNotContains",
    IsNullOrUndefined: "IsNullOrUndefined",
    IsNotNull: "IsNotNull",
};
exports.AlternativeAuthenticationNameSource = {
    ClientCertificateSubject: "ClientCertificateSubject",
    ClientCertificateDns: "ClientCertificateDns",
    ClientCertificateUri: "ClientCertificateUri",
    ClientCertificateIp: "ClientCertificateIp",
    ClientCertificateEmail: "ClientCertificateEmail",
};
exports.ChannelProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
    IdleDueToMirroredPartnerTopicDeletion: "IdleDueToMirroredPartnerTopicDeletion",
};
exports.ChannelType = {
    PartnerTopic: "PartnerTopic",
};
exports.ClientCertificateValidationScheme = {
    SubjectMatchesAuthenticationName: "SubjectMatchesAuthenticationName",
    DnsMatchesAuthenticationName: "DnsMatchesAuthenticationName",
    UriMatchesAuthenticationName: "UriMatchesAuthenticationName",
    IpMatchesAuthenticationName: "IpMatchesAuthenticationName",
    EmailMatchesAuthenticationName: "EmailMatchesAuthenticationName",
    ThumbprintMatch: "ThumbprintMatch",
};
exports.ClientState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DataResidencyBoundary = {
    WithinGeopair: "WithinGeopair",
    WithinRegion: "WithinRegion",
};
exports.DeadLetterEndPointType = {
    StorageBlob: "StorageBlob",
};
exports.DeliveryAttributeMappingType = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.DeliveryMode = {
    Queue: "Queue",
};
exports.DeliverySchema = {
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EndpointType = {
    WebHook: "WebHook",
    EventHub: "EventHub",
    StorageQueue: "StorageQueue",
    HybridConnection: "HybridConnection",
    ServiceBusQueue: "ServiceBusQueue",
    ServiceBusTopic: "ServiceBusTopic",
    AzureFunction: "AzureFunction",
};
exports.EventDefinitionKind = {
    Inline: "Inline",
};
exports.EventDeliverySchema = {
    EventGridSchema: "EventGridSchema",
    CustomInputSchema: "CustomInputSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventInputSchema = {
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventSubscriptionIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.FilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
    NumberInRange: "NumberInRange",
    NumberNotInRange: "NumberNotInRange",
    StringNotBeginsWith: "StringNotBeginsWith",
    StringNotEndsWith: "StringNotEndsWith",
    StringNotContains: "StringNotContains",
    IsNullOrUndefined: "IsNullOrUndefined",
    IsNotNull: "IsNotNull",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.InputSchema = {
    EventGridSchema: "EventGridSchema",
    CustomEventSchema: "CustomEventSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.InputSchemaMappingType = {
    Json: "Json",
};
exports.IpActionType = {
    Allow: "Allow",
};
exports.PartnerConfigurationProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.PartnerDestinationActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.PartnerTopicActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
    Deactivated: "Deactivated",
};
exports.PartnerTopicRoutingMode = {
    SourceEventAttribute: "SourceEventAttribute",
    ChannelNameHeader: "ChannelNameHeader",
};
exports.PermissionType = {
    Publisher: "Publisher",
    Subscriber: "Subscriber",
};
exports.PersistedConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PublisherType = {
    Custom: "Custom",
};
exports.ReadinessState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.ResourceProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.RoutingIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.SkuName = {
    Standard: "Standard",
};
exports.StaticRoutingEnrichmentType = {
    String: "String",
};
exports.TlsVersion = {
    TlsVersion_1_0: "1.0",
    TlsVersion_1_1: "1.1",
    TlsVersion_1_2: "1.2",
};
exports.TopicSpacesConfigurationState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
