// @pulumi/azure-native/types/enums/databricks/v20250301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredNsgRules = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.KeySource = exports.InitialType = exports.IdentityType = exports.EnhancedSecurityMonitoringValue = exports.EncryptionKeySource = exports.DefaultStorageFirewall = exports.ComplianceStandard = exports.ComplianceSecurityProfileValue = exports.AutomaticClusterUpdateValue = void 0;
exports.AutomaticClusterUpdateValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ComplianceSecurityProfileValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ComplianceStandard = {
    NONE: "NONE",
    HIPAA: "HIPAA",
    PCI_DSS: "PCI_DSS",
    CYBER_ESSENTIAL_PLUS: "CYBER_ESSENTIAL_PLUS",
    FEDRAMP_HIGH: "FEDRAMP_HIGH",
    CANADA_PROTECTED_B: "CANADA_PROTECTED_B",
    IRAP_PROTECTED: "IRAP_PROTECTED",
    ISMAP: "ISMAP",
    HITRUST: "HITRUST",
};
exports.DefaultStorageFirewall = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.EncryptionKeySource = {
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.EnhancedSecurityMonitoringValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.InitialType = {
    HiveMetastore: "HiveMetastore",
    UnityCatalog: "UnityCatalog",
};
exports.KeySource = {
    Default: "Default",
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RequiredNsgRules = {
    AllRules: "AllRules",
    NoAzureDatabricksRules: "NoAzureDatabricksRules",
    NoAzureServiceRules: "NoAzureServiceRules",
};
