// @pulumi/azure-native/virtualmachineimages/v20220701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineImageTemplate = exports.Trigger = exports.getVirtualMachineImageTemplateOutput = exports.getVirtualMachineImageTemplate = exports.getTriggerOutput = exports.getTrigger = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getTrigger = null;
exports.getTriggerOutput = null;
utilities.lazyLoad(exports, ["getTrigger", "getTriggerOutput"], () => require("./getTrigger"));
exports.getVirtualMachineImageTemplate = null;
exports.getVirtualMachineImageTemplateOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineImageTemplate", "getVirtualMachineImageTemplateOutput"], () => require("./getVirtualMachineImageTemplate"));
exports.Trigger = null;
utilities.lazyLoad(exports, ["Trigger"], () => require("./trigger"));
exports.VirtualMachineImageTemplate = null;
utilities.lazyLoad(exports, ["VirtualMachineImageTemplate"], () => require("./virtualMachineImageTemplate"));
// Export enums:
__exportStar(require("../../types/enums/virtualmachineimages/v20220701"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:virtualmachineimages/v20220701:Trigger":
                return new exports.Trigger(name, undefined, { urn });
            case "azure-native:virtualmachineimages/v20220701:VirtualMachineImageTemplate":
                return new exports.VirtualMachineImageTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "virtualmachineimages/v20220701", _module);
