// @pulumi/azure-native/resources/v20220901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagAtScope = exports.ResourceGroup = exports.Resource = exports.getTagAtScopeOutput = exports.getTagAtScope = exports.getResourceGroupOutput = exports.getResourceGroup = exports.getResourceOutput = exports.getResource = exports.getDeploymentAtTenantScopeOutput = exports.getDeploymentAtTenantScope = exports.getDeploymentAtSubscriptionScopeOutput = exports.getDeploymentAtSubscriptionScope = exports.getDeploymentAtScopeOutput = exports.getDeploymentAtScope = exports.getDeploymentAtManagementGroupScopeOutput = exports.getDeploymentAtManagementGroupScope = exports.getDeploymentOutput = exports.getDeployment = exports.DeploymentAtTenantScope = exports.DeploymentAtSubscriptionScope = exports.DeploymentAtScope = exports.DeploymentAtManagementGroupScope = exports.Deployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.DeploymentAtManagementGroupScope = null;
utilities.lazyLoad(exports, ["DeploymentAtManagementGroupScope"], () => require("./deploymentAtManagementGroupScope"));
exports.DeploymentAtScope = null;
utilities.lazyLoad(exports, ["DeploymentAtScope"], () => require("./deploymentAtScope"));
exports.DeploymentAtSubscriptionScope = null;
utilities.lazyLoad(exports, ["DeploymentAtSubscriptionScope"], () => require("./deploymentAtSubscriptionScope"));
exports.DeploymentAtTenantScope = null;
utilities.lazyLoad(exports, ["DeploymentAtTenantScope"], () => require("./deploymentAtTenantScope"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getDeploymentAtManagementGroupScope = null;
exports.getDeploymentAtManagementGroupScopeOutput = null;
utilities.lazyLoad(exports, ["getDeploymentAtManagementGroupScope", "getDeploymentAtManagementGroupScopeOutput"], () => require("./getDeploymentAtManagementGroupScope"));
exports.getDeploymentAtScope = null;
exports.getDeploymentAtScopeOutput = null;
utilities.lazyLoad(exports, ["getDeploymentAtScope", "getDeploymentAtScopeOutput"], () => require("./getDeploymentAtScope"));
exports.getDeploymentAtSubscriptionScope = null;
exports.getDeploymentAtSubscriptionScopeOutput = null;
utilities.lazyLoad(exports, ["getDeploymentAtSubscriptionScope", "getDeploymentAtSubscriptionScopeOutput"], () => require("./getDeploymentAtSubscriptionScope"));
exports.getDeploymentAtTenantScope = null;
exports.getDeploymentAtTenantScopeOutput = null;
utilities.lazyLoad(exports, ["getDeploymentAtTenantScope", "getDeploymentAtTenantScopeOutput"], () => require("./getDeploymentAtTenantScope"));
exports.getResource = null;
exports.getResourceOutput = null;
utilities.lazyLoad(exports, ["getResource", "getResourceOutput"], () => require("./getResource"));
exports.getResourceGroup = null;
exports.getResourceGroupOutput = null;
utilities.lazyLoad(exports, ["getResourceGroup", "getResourceGroupOutput"], () => require("./getResourceGroup"));
exports.getTagAtScope = null;
exports.getTagAtScopeOutput = null;
utilities.lazyLoad(exports, ["getTagAtScope", "getTagAtScopeOutput"], () => require("./getTagAtScope"));
exports.Resource = null;
utilities.lazyLoad(exports, ["Resource"], () => require("./resource"));
exports.ResourceGroup = null;
utilities.lazyLoad(exports, ["ResourceGroup"], () => require("./resourceGroup"));
exports.TagAtScope = null;
utilities.lazyLoad(exports, ["TagAtScope"], () => require("./tagAtScope"));
// Export enums:
__exportStar(require("../../types/enums/resources/v20220901"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:resources/v20220901:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:resources/v20220901:DeploymentAtManagementGroupScope":
                return new exports.DeploymentAtManagementGroupScope(name, undefined, { urn });
            case "azure-native:resources/v20220901:DeploymentAtScope":
                return new exports.DeploymentAtScope(name, undefined, { urn });
            case "azure-native:resources/v20220901:DeploymentAtSubscriptionScope":
                return new exports.DeploymentAtSubscriptionScope(name, undefined, { urn });
            case "azure-native:resources/v20220901:DeploymentAtTenantScope":
                return new exports.DeploymentAtTenantScope(name, undefined, { urn });
            case "azure-native:resources/v20220901:Resource":
                return new exports.Resource(name, undefined, { urn });
            case "azure-native:resources/v20220901:ResourceGroup":
                return new exports.ResourceGroup(name, undefined, { urn });
            case "azure-native:resources/v20220901:TagAtScope":
                return new exports.TagAtScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "resources/v20220901", _module);
