// @pulumi/azure-native/securityinsights/v20220101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = exports.listSourceControlRepositoriesOutput = exports.listSourceControlRepositories = exports.getWatchlistOutput = exports.getWatchlist = exports.getEntityInsightsOutput = exports.getEntityInsights = exports.getEntityAnalyticsOutput = exports.getEntityAnalytics = exports.getEntitiesGetTimelineOutput = exports.getEntitiesGetTimeline = exports.EntityAnalytics = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.EntityAnalytics = null;
utilities.lazyLoad(exports, ["EntityAnalytics"], () => require("./entityAnalytics"));
exports.getEntitiesGetTimeline = null;
exports.getEntitiesGetTimelineOutput = null;
utilities.lazyLoad(exports, ["getEntitiesGetTimeline", "getEntitiesGetTimelineOutput"], () => require("./getEntitiesGetTimeline"));
exports.getEntityAnalytics = null;
exports.getEntityAnalyticsOutput = null;
utilities.lazyLoad(exports, ["getEntityAnalytics", "getEntityAnalyticsOutput"], () => require("./getEntityAnalytics"));
exports.getEntityInsights = null;
exports.getEntityInsightsOutput = null;
utilities.lazyLoad(exports, ["getEntityInsights", "getEntityInsightsOutput"], () => require("./getEntityInsights"));
exports.getWatchlist = null;
exports.getWatchlistOutput = null;
utilities.lazyLoad(exports, ["getWatchlist", "getWatchlistOutput"], () => require("./getWatchlist"));
exports.listSourceControlRepositories = null;
exports.listSourceControlRepositoriesOutput = null;
utilities.lazyLoad(exports, ["listSourceControlRepositories", "listSourceControlRepositoriesOutput"], () => require("./listSourceControlRepositories"));
exports.Watchlist = null;
utilities.lazyLoad(exports, ["Watchlist"], () => require("./watchlist"));
// Export enums:
__exportStar(require("../../types/enums/securityinsights/v20220101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:securityinsights/v20220101preview:EntityAnalytics":
                return new exports.EntityAnalytics(name, undefined, { urn });
            case "azure-native:securityinsights/v20220101preview:Watchlist":
                return new exports.Watchlist(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "securityinsights/v20220101preview", _module);
