// @pulumi/azure-native/types/enums/digitaltwins/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordPropertyAndItemRemovals = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.IdentityType = exports.EndpointType = exports.DigitalTwinsIdentityType = exports.ConnectionType = exports.AuthenticationType = exports.v20230131 = exports.v20201201 = void 0;
// Export sub-modules:
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20230131 = require("./v20230131");
exports.v20230131 = v20230131;
exports.AuthenticationType = {
    KeyBased: "KeyBased",
    IdentityBased: "IdentityBased",
};
exports.ConnectionType = {
    AzureDataExplorer: "AzureDataExplorer",
};
exports.DigitalTwinsIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.EndpointType = {
    EventHub: "EventHub",
    EventGrid: "EventGrid",
    ServiceBus: "ServiceBus",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RecordPropertyAndItemRemovals = {
    True: "true",
    False: "false",
};
