// @pulumi/gcp/tpu/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2Vm = exports.Node = exports.getV2RuntimeVersionsOutput = exports.getV2RuntimeVersions = exports.getV2AcceleratorTypesOutput = exports.getV2AcceleratorTypes = exports.getTensorflowVersionsOutput = exports.getTensorflowVersions = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getTensorflowVersions = null;
exports.getTensorflowVersionsOutput = null;
utilities.lazyLoad(exports, ["getTensorflowVersions", "getTensorflowVersionsOutput"], () => require("./getTensorflowVersions"));
exports.getV2AcceleratorTypes = null;
exports.getV2AcceleratorTypesOutput = null;
utilities.lazyLoad(exports, ["getV2AcceleratorTypes", "getV2AcceleratorTypesOutput"], () => require("./getV2AcceleratorTypes"));
exports.getV2RuntimeVersions = null;
exports.getV2RuntimeVersionsOutput = null;
utilities.lazyLoad(exports, ["getV2RuntimeVersions", "getV2RuntimeVersionsOutput"], () => require("./getV2RuntimeVersions"));
exports.Node = null;
utilities.lazyLoad(exports, ["Node"], () => require("./node"));
exports.V2Vm = null;
utilities.lazyLoad(exports, ["V2Vm"], () => require("./v2vm"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:tpu/node:Node":
                return new exports.Node(name, undefined, { urn });
            case "gcp:tpu/v2Vm:V2Vm":
                return new exports.V2Vm(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "tpu/node", _module);
pulumi.runtime.registerResourceModule("gcp", "tpu/v2Vm", _module);
