// @pulumi/azure-native/types/enums/network/v20240101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityConfigurationRuleProtocol = exports.SecurityConfigurationRuleDirection = exports.SecurityConfigurationRuleAccess = exports.NetworkProtocol = exports.NetworkIntentPolicyBasedService = exports.ConfigurationType = exports.AdminRuleKind = exports.AddressSpaceAggregationOption = exports.AddressPrefixType = void 0;
exports.AddressPrefixType = {
    IPPrefix: "IPPrefix",
    ServiceTag: "ServiceTag",
    NetworkGroup: "NetworkGroup",
};
exports.AddressSpaceAggregationOption = {
    None: "None",
    Manual: "Manual",
};
exports.AdminRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
exports.ConfigurationType = {
    SecurityAdmin: "SecurityAdmin",
    Connectivity: "Connectivity",
};
exports.NetworkIntentPolicyBasedService = {
    None: "None",
    All: "All",
    AllowRulesOnly: "AllowRulesOnly",
};
exports.NetworkProtocol = {
    Any: "Any",
    TCP: "TCP",
    UDP: "UDP",
    ICMP: "ICMP",
};
exports.SecurityConfigurationRuleAccess = {
    Allow: "Allow",
    Deny: "Deny",
    AlwaysAllow: "AlwaysAllow",
};
exports.SecurityConfigurationRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.SecurityConfigurationRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    Icmp: "Icmp",
    Esp: "Esp",
    Any: "Any",
    Ah: "Ah",
};
