// @pulumi/azure-native/types/enums/search/v20250201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.SharedPrivateLinkResourceStatus = exports.SharedPrivateLinkResourceProvisioningState = exports.SearchSemanticSearch = exports.SearchEncryptionWithCmk = exports.SearchDisabledDataExfiltrationOption = exports.SearchBypass = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.PrivateLinkServiceConnectionProvisioningState = exports.IdentityType = exports.HostingMode = exports.ComputeType = exports.AadAuthFailureMode = void 0;
exports.AadAuthFailureMode = {
    /**
     * Indicates that requests that failed authentication should be presented with an HTTP status code of 403 (Forbidden).
     */
    Http403: "http403",
    /**
     * Indicates that requests that failed authentication should be presented with an HTTP status code of 401 (Unauthorized) and present a Bearer Challenge.
     */
    Http401WithBearerChallenge: "http401WithBearerChallenge",
};
exports.ComputeType = {
    /**
     * Create the service with the default compute.
     */
    Default: "default",
    /**
     * Create the service with Azure Confidential Compute.
     */
    Confidential: "confidential",
};
exports.HostingMode = {
    /**
     * The limit on number of indexes is determined by the default limits for the SKU.
     */
    Default: "default",
    /**
     * Only application for standard3 SKU, where the search service can have up to 1000 indexes.
     */
    HighDensity: "highDensity",
};
exports.IdentityType = {
    /**
     * Indicates that any identity associated with the search service needs to be removed.
     */
    None: "None",
    /**
     * Indicates that system-assigned identity for the search service will be enabled.
     */
    SystemAssigned: "SystemAssigned",
    /**
     * Indicates that one or more user assigned identities will be assigned to the search service.
     */
    UserAssigned: "UserAssigned",
    /**
     * Indicates that system-assigned identity for the search service will be enabled along with the assignment of one or more user assigned identities.
     */
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.PrivateLinkServiceConnectionProvisioningState = {
    /**
     * The private link service connection is in the process of being created along with other resources for it to be fully functional.
     */
    Updating: "Updating",
    /**
     * The private link service connection is in the process of being deleted.
     */
    Deleting: "Deleting",
    /**
     * The private link service connection has failed to be provisioned or deleted.
     */
    Failed: "Failed",
    /**
     * The private link service connection has finished provisioning and is ready for approval.
     */
    Succeeded: "Succeeded",
    /**
     * Provisioning request for the private link service connection resource has been accepted but the process of creation has not commenced yet.
     */
    Incomplete: "Incomplete",
    /**
     * Provisioning request for the private link service connection resource has been canceled.
     */
    Canceled: "Canceled",
};
exports.PrivateLinkServiceConnectionStatus = {
    /**
     * The private endpoint connection has been created and is pending approval.
     */
    Pending: "Pending",
    /**
     * The private endpoint connection is approved and is ready for use.
     */
    Approved: "Approved",
    /**
     * The private endpoint connection has been rejected and cannot be used.
     */
    Rejected: "Rejected",
    /**
     * The private endpoint connection has been removed from the service.
     */
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    /**
     * The search service is accessible from traffic originating from the public internet.
     */
    Enabled: "enabled",
    /**
     * The search service is not accessible from traffic originating from the public internet. Access is only permitted over approved private endpoint connections.
     */
    Disabled: "disabled",
};
exports.SearchBypass = {
    /**
     * Indicates that no origin can bypass the rules defined in the 'ipRules' section. This is the default.
     */
    None: "None",
    /**
     * Indicates that requests originating from the Azure portal can bypass the rules defined in the 'ipRules' section.
     */
    AzurePortal: "AzurePortal",
    /**
     * Indicates that requests originating from Azure trusted services can bypass the rules defined in the 'ipRules' section.
     */
    AzureServices: "AzureServices",
};
exports.SearchDisabledDataExfiltrationOption = {
    /**
     * Indicates that all data exfiltration scenarios are disabled.
     */
    All: "All",
};
exports.SearchEncryptionWithCmk = {
    /**
     * No enforcement of customer-managed key encryption will be made. Only the built-in service-managed encryption is used.
     */
    Disabled: "Disabled",
    /**
     * Search service will be marked as non-compliant if one or more objects aren't encrypted with a customer-managed key.
     */
    Enabled: "Enabled",
    /**
     * Enforcement policy is not explicitly specified, with the behavior being the same as if it were set to 'Disabled'.
     */
    Unspecified: "Unspecified",
};
exports.SearchSemanticSearch = {
    /**
     * Indicates that semantic reranker is disabled for the search service. This is the default.
     */
    Disabled: "disabled",
    /**
     * Enables semantic reranker on a search service and indicates that it is to be used within the limits of the free plan. The free plan would cap the volume of semantic ranking requests and is offered at no extra charge. This is the default for newly provisioned search services.
     */
    Free: "free",
    /**
     * Enables semantic reranker on a search service as a billable feature, with higher throughput and volume of semantically reranked queries.
     */
    Standard: "standard",
};
exports.SharedPrivateLinkResourceProvisioningState = {
    /**
     * The shared private link resource is in the process of being created along with other resources for it to be fully functional.
     */
    Updating: "Updating",
    /**
     * The shared private link resource is in the process of being deleted.
     */
    Deleting: "Deleting",
    /**
     * The shared private link resource has failed to be provisioned or deleted.
     */
    Failed: "Failed",
    /**
     * The shared private link resource has finished provisioning and is ready for approval.
     */
    Succeeded: "Succeeded",
    /**
     * Provisioning request for the shared private link resource has been accepted but the process of creation has not commenced yet.
     */
    Incomplete: "Incomplete",
};
exports.SharedPrivateLinkResourceStatus = {
    /**
     * The shared private link resource has been created and is pending approval.
     */
    Pending: "Pending",
    /**
     * The shared private link resource is approved and is ready for use.
     */
    Approved: "Approved",
    /**
     * The shared private link resource has been rejected and cannot be used.
     */
    Rejected: "Rejected",
    /**
     * The shared private link resource has been removed from the service.
     */
    Disconnected: "Disconnected",
};
exports.SkuName = {
    /**
     * Free tier, with no SLA guarantees and a subset of the features offered on billable tiers.
     */
    Free: "free",
    /**
     * Billable tier for a dedicated service having up to 3 replicas.
     */
    Basic: "basic",
    /**
     * Billable tier for a dedicated service having up to 12 partitions and 12 replicas.
     */
    Standard: "standard",
    /**
     * Similar to 'standard', but with more capacity per search unit.
     */
    Standard2: "standard2",
    /**
     *  The largest Standard offering with up to 12 partitions and 12 replicas (or up to 3 partitions with more indexes if you also set the hostingMode property to 'highDensity').
     */
    Standard3: "standard3",
    /**
     * Billable tier for a dedicated service that supports 1TB per partition, up to 12 partitions.
     */
    StorageOptimizedL1: "storage_optimized_l1",
    /**
     * Billable tier for a dedicated service that supports 2TB per partition, up to 12 partitions.
     */
    StorageOptimizedL2: "storage_optimized_l2",
};
