// @pulumi/azure-native/security/v20190801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotSecuritySolution = exports.getIotSecuritySolutionOutput = exports.getIotSecuritySolution = exports.getDeviceSecurityGroupOutput = exports.getDeviceSecurityGroup = exports.DeviceSecurityGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DeviceSecurityGroup = null;
utilities.lazyLoad(exports, ["DeviceSecurityGroup"], () => require("./deviceSecurityGroup"));
exports.getDeviceSecurityGroup = null;
exports.getDeviceSecurityGroupOutput = null;
utilities.lazyLoad(exports, ["getDeviceSecurityGroup", "getDeviceSecurityGroupOutput"], () => require("./getDeviceSecurityGroup"));
exports.getIotSecuritySolution = null;
exports.getIotSecuritySolutionOutput = null;
utilities.lazyLoad(exports, ["getIotSecuritySolution", "getIotSecuritySolutionOutput"], () => require("./getIotSecuritySolution"));
exports.IotSecuritySolution = null;
utilities.lazyLoad(exports, ["IotSecuritySolution"], () => require("./iotSecuritySolution"));
// Export enums:
__exportStar(require("../../types/enums/security/v20190801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20190801:DeviceSecurityGroup":
                return new exports.DeviceSecurityGroup(name, undefined, { urn });
            case "azure-native:security/v20190801:IotSecuritySolution":
                return new exports.IotSecuritySolution(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20190801", _module);
