// @pulumi/azure-native/network/v20220501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = exports.NetworkWatcher = exports.getPolicyOutput = exports.getPolicy = exports.getNetworkWatcherOutput = exports.getNetworkWatcher = exports.getDdosProtectionPlanOutput = exports.getDdosProtectionPlan = exports.DdosProtectionPlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DdosProtectionPlan = null;
utilities.lazyLoad(exports, ["DdosProtectionPlan"], () => require("./ddosProtectionPlan"));
exports.getDdosProtectionPlan = null;
exports.getDdosProtectionPlanOutput = null;
utilities.lazyLoad(exports, ["getDdosProtectionPlan", "getDdosProtectionPlanOutput"], () => require("./getDdosProtectionPlan"));
exports.getNetworkWatcher = null;
exports.getNetworkWatcherOutput = null;
utilities.lazyLoad(exports, ["getNetworkWatcher", "getNetworkWatcherOutput"], () => require("./getNetworkWatcher"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.NetworkWatcher = null;
utilities.lazyLoad(exports, ["NetworkWatcher"], () => require("./networkWatcher"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
// Export enums:
__exportStar(require("../../types/enums/network/v20220501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20220501:DdosProtectionPlan":
                return new exports.DdosProtectionPlan(name, undefined, { urn });
            case "azure-native:network/v20220501:NetworkWatcher":
                return new exports.NetworkWatcher(name, undefined, { urn });
            case "azure-native:network/v20220501:Policy":
                return new exports.Policy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20220501", _module);
