// @pulumi/azure-native/web/v20160601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listCustomApiWsdlInterfacesOutput = exports.listCustomApiWsdlInterfaces = exports.listConnectionConsentLinksOutput = exports.listConnectionConsentLinks = exports.getCustomApiOutput = exports.getCustomApi = exports.getConnectionGatewayOutput = exports.getConnectionGateway = exports.getConnectionOutput = exports.getConnection = exports.CustomApi = exports.ConnectionGateway = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionGateway = null;
utilities.lazyLoad(exports, ["ConnectionGateway"], () => require("./connectionGateway"));
exports.CustomApi = null;
utilities.lazyLoad(exports, ["CustomApi"], () => require("./customApi"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.getConnectionGateway = null;
exports.getConnectionGatewayOutput = null;
utilities.lazyLoad(exports, ["getConnectionGateway", "getConnectionGatewayOutput"], () => require("./getConnectionGateway"));
exports.getCustomApi = null;
exports.getCustomApiOutput = null;
utilities.lazyLoad(exports, ["getCustomApi", "getCustomApiOutput"], () => require("./getCustomApi"));
exports.listConnectionConsentLinks = null;
exports.listConnectionConsentLinksOutput = null;
utilities.lazyLoad(exports, ["listConnectionConsentLinks", "listConnectionConsentLinksOutput"], () => require("./listConnectionConsentLinks"));
exports.listCustomApiWsdlInterfaces = null;
exports.listCustomApiWsdlInterfacesOutput = null;
utilities.lazyLoad(exports, ["listCustomApiWsdlInterfaces", "listCustomApiWsdlInterfacesOutput"], () => require("./listCustomApiWsdlInterfaces"));
// Export enums:
__exportStar(require("../../types/enums/web/v20160601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20160601:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "azure-native:web/v20160601:ConnectionGateway":
                return new exports.ConnectionGateway(name, undefined, { urn });
            case "azure-native:web/v20160601:CustomApi":
                return new exports.CustomApi(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20160601", _module);
