// @pulumi/azure-native/types/enums/chaos/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetReferenceType = exports.SelectorType = exports.ResourceIdentityType = exports.FilterType = exports.v20250101 = exports.v20241101preview = exports.v20240322preview = exports.v20240101 = exports.v20231101 = exports.v20231027preview = exports.v20230901preview = exports.v20230415preview = void 0;
// Export sub-modules:
const v20230415preview = require("./v20230415preview");
exports.v20230415preview = v20230415preview;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231027preview = require("./v20231027preview");
exports.v20231027preview = v20231027preview;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240322preview = require("./v20240322preview");
exports.v20240322preview = v20240322preview;
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const v20250101 = require("./v20250101");
exports.v20250101 = v20250101;
exports.FilterType = {
    Simple: "Simple",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.SelectorType = {
    List: "List",
    Query: "Query",
};
exports.TargetReferenceType = {
    ChaosTarget: "ChaosTarget",
};
