// @pulumi/azure-native/types/enums/network/v20230701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneType = exports.IpAllocationMethod = exports.ForwardingRuleState = exports.DnsSecurityRuleState = exports.BlockResponseCode = exports.AssociationAccessMode = exports.ActionType = exports.AccessRuleDirection = void 0;
exports.AccessRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.ActionType = {
    Allow: "Allow",
    Alert: "Alert",
    Block: "Block",
};
exports.AssociationAccessMode = {
    Learning: "Learning",
    Enforced: "Enforced",
    Audit: "Audit",
};
exports.BlockResponseCode = {
    SERVFAIL: "SERVFAIL",
};
exports.DnsSecurityRuleState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ForwardingRuleState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.IpAllocationMethod = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.ZoneType = {
    Public: "Public",
    Private: "Private",
};
