// @pulumi/azure-native/networkcloud/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rack = exports.MetricsConfiguration = exports.L3Network = exports.L2Network = exports.KubernetesClusterFeature = exports.KubernetesCluster = exports.getVolumeOutput = exports.getVolume = exports.getVirtualMachineOutput = exports.getVirtualMachine = exports.getTrunkedNetworkOutput = exports.getTrunkedNetwork = exports.getStorageApplianceOutput = exports.getStorageAppliance = exports.getRackOutput = exports.getRack = exports.getMetricsConfigurationOutput = exports.getMetricsConfiguration = exports.getL3NetworkOutput = exports.getL3Network = exports.getL2NetworkOutput = exports.getL2Network = exports.getKubernetesClusterFeatureOutput = exports.getKubernetesClusterFeature = exports.getKubernetesClusterOutput = exports.getKubernetesCluster = exports.getConsoleOutput = exports.getConsole = exports.getClusterManagerOutput = exports.getClusterManager = exports.getClusterOutput = exports.getCluster = exports.getCloudServicesNetworkOutput = exports.getCloudServicesNetwork = exports.getBmcKeySetOutput = exports.getBmcKeySet = exports.getBareMetalMachineKeySetOutput = exports.getBareMetalMachineKeySet = exports.getBareMetalMachineOutput = exports.getBareMetalMachine = exports.getAgentPoolOutput = exports.getAgentPool = exports.Console = exports.ClusterManager = exports.Cluster = exports.CloudServicesNetwork = exports.BmcKeySet = exports.BareMetalMachineKeySet = exports.BareMetalMachine = exports.AgentPool = void 0;
exports.v20250201 = exports.v20241001preview = exports.v20240701 = exports.v20240601preview = exports.v20231001preview = exports.v20230701 = exports.Volume = exports.VirtualMachine = exports.TrunkedNetwork = exports.StorageAppliance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.BareMetalMachine = null;
utilities.lazyLoad(exports, ["BareMetalMachine"], () => require("./bareMetalMachine"));
exports.BareMetalMachineKeySet = null;
utilities.lazyLoad(exports, ["BareMetalMachineKeySet"], () => require("./bareMetalMachineKeySet"));
exports.BmcKeySet = null;
utilities.lazyLoad(exports, ["BmcKeySet"], () => require("./bmcKeySet"));
exports.CloudServicesNetwork = null;
utilities.lazyLoad(exports, ["CloudServicesNetwork"], () => require("./cloudServicesNetwork"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterManager = null;
utilities.lazyLoad(exports, ["ClusterManager"], () => require("./clusterManager"));
exports.Console = null;
utilities.lazyLoad(exports, ["Console"], () => require("./console"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getBareMetalMachine = null;
exports.getBareMetalMachineOutput = null;
utilities.lazyLoad(exports, ["getBareMetalMachine", "getBareMetalMachineOutput"], () => require("./getBareMetalMachine"));
exports.getBareMetalMachineKeySet = null;
exports.getBareMetalMachineKeySetOutput = null;
utilities.lazyLoad(exports, ["getBareMetalMachineKeySet", "getBareMetalMachineKeySetOutput"], () => require("./getBareMetalMachineKeySet"));
exports.getBmcKeySet = null;
exports.getBmcKeySetOutput = null;
utilities.lazyLoad(exports, ["getBmcKeySet", "getBmcKeySetOutput"], () => require("./getBmcKeySet"));
exports.getCloudServicesNetwork = null;
exports.getCloudServicesNetworkOutput = null;
utilities.lazyLoad(exports, ["getCloudServicesNetwork", "getCloudServicesNetworkOutput"], () => require("./getCloudServicesNetwork"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterManager = null;
exports.getClusterManagerOutput = null;
utilities.lazyLoad(exports, ["getClusterManager", "getClusterManagerOutput"], () => require("./getClusterManager"));
exports.getConsole = null;
exports.getConsoleOutput = null;
utilities.lazyLoad(exports, ["getConsole", "getConsoleOutput"], () => require("./getConsole"));
exports.getKubernetesCluster = null;
exports.getKubernetesClusterOutput = null;
utilities.lazyLoad(exports, ["getKubernetesCluster", "getKubernetesClusterOutput"], () => require("./getKubernetesCluster"));
exports.getKubernetesClusterFeature = null;
exports.getKubernetesClusterFeatureOutput = null;
utilities.lazyLoad(exports, ["getKubernetesClusterFeature", "getKubernetesClusterFeatureOutput"], () => require("./getKubernetesClusterFeature"));
exports.getL2Network = null;
exports.getL2NetworkOutput = null;
utilities.lazyLoad(exports, ["getL2Network", "getL2NetworkOutput"], () => require("./getL2Network"));
exports.getL3Network = null;
exports.getL3NetworkOutput = null;
utilities.lazyLoad(exports, ["getL3Network", "getL3NetworkOutput"], () => require("./getL3Network"));
exports.getMetricsConfiguration = null;
exports.getMetricsConfigurationOutput = null;
utilities.lazyLoad(exports, ["getMetricsConfiguration", "getMetricsConfigurationOutput"], () => require("./getMetricsConfiguration"));
exports.getRack = null;
exports.getRackOutput = null;
utilities.lazyLoad(exports, ["getRack", "getRackOutput"], () => require("./getRack"));
exports.getStorageAppliance = null;
exports.getStorageApplianceOutput = null;
utilities.lazyLoad(exports, ["getStorageAppliance", "getStorageApplianceOutput"], () => require("./getStorageAppliance"));
exports.getTrunkedNetwork = null;
exports.getTrunkedNetworkOutput = null;
utilities.lazyLoad(exports, ["getTrunkedNetwork", "getTrunkedNetworkOutput"], () => require("./getTrunkedNetwork"));
exports.getVirtualMachine = null;
exports.getVirtualMachineOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachine", "getVirtualMachineOutput"], () => require("./getVirtualMachine"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.KubernetesCluster = null;
utilities.lazyLoad(exports, ["KubernetesCluster"], () => require("./kubernetesCluster"));
exports.KubernetesClusterFeature = null;
utilities.lazyLoad(exports, ["KubernetesClusterFeature"], () => require("./kubernetesClusterFeature"));
exports.L2Network = null;
utilities.lazyLoad(exports, ["L2Network"], () => require("./l2network"));
exports.L3Network = null;
utilities.lazyLoad(exports, ["L3Network"], () => require("./l3network"));
exports.MetricsConfiguration = null;
utilities.lazyLoad(exports, ["MetricsConfiguration"], () => require("./metricsConfiguration"));
exports.Rack = null;
utilities.lazyLoad(exports, ["Rack"], () => require("./rack"));
exports.StorageAppliance = null;
utilities.lazyLoad(exports, ["StorageAppliance"], () => require("./storageAppliance"));
exports.TrunkedNetwork = null;
utilities.lazyLoad(exports, ["TrunkedNetwork"], () => require("./trunkedNetwork"));
exports.VirtualMachine = null;
utilities.lazyLoad(exports, ["VirtualMachine"], () => require("./virtualMachine"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
// Export enums:
__exportStar(require("../types/enums/networkcloud"), exports);
// Export sub-modules:
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const v20250201 = require("./v20250201");
exports.v20250201 = v20250201;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:networkcloud:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:networkcloud:BareMetalMachine":
                return new exports.BareMetalMachine(name, undefined, { urn });
            case "azure-native:networkcloud:BareMetalMachineKeySet":
                return new exports.BareMetalMachineKeySet(name, undefined, { urn });
            case "azure-native:networkcloud:BmcKeySet":
                return new exports.BmcKeySet(name, undefined, { urn });
            case "azure-native:networkcloud:CloudServicesNetwork":
                return new exports.CloudServicesNetwork(name, undefined, { urn });
            case "azure-native:networkcloud:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:networkcloud:ClusterManager":
                return new exports.ClusterManager(name, undefined, { urn });
            case "azure-native:networkcloud:Console":
                return new exports.Console(name, undefined, { urn });
            case "azure-native:networkcloud:KubernetesCluster":
                return new exports.KubernetesCluster(name, undefined, { urn });
            case "azure-native:networkcloud:KubernetesClusterFeature":
                return new exports.KubernetesClusterFeature(name, undefined, { urn });
            case "azure-native:networkcloud:L2Network":
                return new exports.L2Network(name, undefined, { urn });
            case "azure-native:networkcloud:L3Network":
                return new exports.L3Network(name, undefined, { urn });
            case "azure-native:networkcloud:MetricsConfiguration":
                return new exports.MetricsConfiguration(name, undefined, { urn });
            case "azure-native:networkcloud:Rack":
                return new exports.Rack(name, undefined, { urn });
            case "azure-native:networkcloud:StorageAppliance":
                return new exports.StorageAppliance(name, undefined, { urn });
            case "azure-native:networkcloud:TrunkedNetwork":
                return new exports.TrunkedNetwork(name, undefined, { urn });
            case "azure-native:networkcloud:VirtualMachine":
                return new exports.VirtualMachine(name, undefined, { urn });
            case "azure-native:networkcloud:Volume":
                return new exports.Volume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "networkcloud", _module);
