// @pulumi/azure-native/types/enums/billingbenefits/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.ManagedServiceIdentityType = exports.DiscountEntityType = exports.DiscountAppliedScopeType = exports.v20241101preview = void 0;
// Export sub-modules:
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
exports.DiscountAppliedScopeType = {
    BillingAccount: "BillingAccount",
    BillingProfile: "BillingProfile",
    Customer: "Customer",
};
exports.DiscountEntityType = {
    Primary: "Primary",
    Affiliate: "Affiliate",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
