// @pulumi/azure-native/types/enums/azuresphere/v20220901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePolicy = exports.RegionalDataBoundary = exports.OSFeedType = exports.AllowCrashDumpCollection = void 0;
exports.AllowCrashDumpCollection = {
    /**
     * Crash dump collection enabled
     */
    Enabled: "Enabled",
    /**
     * Crash dump collection disabled
     */
    Disabled: "Disabled",
};
exports.OSFeedType = {
    /**
     * Retail OS feed type.
     */
    Retail: "Retail",
    /**
     * Retail evaluation OS feed type.
     */
    RetailEval: "RetailEval",
};
exports.RegionalDataBoundary = {
    /**
     * No data boundary
     */
    None: "None",
    /**
     * EU data boundary
     */
    EU: "EU",
};
exports.UpdatePolicy = {
    /**
     * Update all policy.
     */
    UpdateAll: "UpdateAll",
    /**
     * No update for 3rd party app policy.
     */
    No3rdPartyAppUpdates: "No3rdPartyAppUpdates",
};
