// @pulumi/azure-native/hybridconnectivity/publicCloudConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicCloudConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Public Cloud Connector
 *
 * Uses Azure REST API version 2024-12-01.
 */
class PublicCloudConnector extends pulumi.CustomResource {
    /**
     * Get an existing PublicCloudConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PublicCloudConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PublicCloudConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PublicCloudConnector.__pulumiType;
    }
    /**
     * Create a PublicCloudConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.awsCloudProfile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'awsCloudProfile'");
            }
            if ((!args || args.hostType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["awsCloudProfile"] = args ? (args.awsCloudProfile ? pulumi.output(args.awsCloudProfile).apply(inputs.hybridconnectivity.awsCloudProfileArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["hostType"] = args ? args.hostType : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["publicCloudConnector"] = args ? args.publicCloudConnector : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["connectorPrimaryIdentifier"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["awsCloudProfile"] = undefined /*out*/;
            resourceInputs["connectorPrimaryIdentifier"] = undefined /*out*/;
            resourceInputs["hostType"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hybridconnectivity/v20241201:PublicCloudConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PublicCloudConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PublicCloudConnector = PublicCloudConnector;
/** @internal */
PublicCloudConnector.__pulumiType = 'azure-native:hybridconnectivity:PublicCloudConnector';
