// @pulumi/azure-native/types/enums/policyinsights/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceDiscoveryMode = exports.ComplianceState = exports.v20241001 = exports.v20220901 = exports.v20211001 = void 0;
// Export sub-modules:
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20220901 = require("./v20220901");
exports.v20220901 = v20220901;
const v20241001 = require("./v20241001");
exports.v20241001 = v20241001;
exports.ComplianceState = {
    /**
     * The resource is in compliance with the policy.
     */
    Compliant: "Compliant",
    /**
     * The resource is not in compliance with the policy.
     */
    NonCompliant: "NonCompliant",
    /**
     * The compliance state of the resource is not known.
     */
    Unknown: "Unknown",
};
exports.ResourceDiscoveryMode = {
    /**
     * Remediate resources that are already known to be non-compliant.
     */
    ExistingNonCompliant: "ExistingNonCompliant",
    /**
     * Re-evaluate the compliance state of resources and then remediate the resources found to be non-compliant.
     */
    ReEvaluateCompliance: "ReEvaluateCompliance",
};
