// @pulumi/azure-native/types/enums/appplatform/v20231101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.Type = exports.StorageType = exports.SessionAffinity = exports.ProbeActionType = exports.ManagedIdentityType = exports.KeyVaultCertificateAutoSync = exports.HTTPSchemeType = exports.GitImplementation = exports.GatewayRouteConfigProtocol = exports.GatewayCertificateVerification = exports.Frequency = exports.DevToolPortalFeatureState = exports.CustomizedAcceleratorType = exports.ConfigurationServiceGeneration = exports.ConfigServerEnabledState = exports.BindingType = exports.BackendProtocol = exports.ApmType = exports.ApiPortalApiTryOutEnabledState = void 0;
exports.ApiPortalApiTryOutEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ApmType = {
    ApplicationInsights: "ApplicationInsights",
    AppDynamics: "AppDynamics",
    Dynatrace: "Dynatrace",
    NewRelic: "NewRelic",
    ElasticAPM: "ElasticAPM",
};
exports.BackendProtocol = {
    GRPC: "GRPC",
    Default: "Default",
};
exports.BindingType = {
    ApplicationInsights: "ApplicationInsights",
    ApacheSkyWalking: "ApacheSkyWalking",
    AppDynamics: "AppDynamics",
    Dynatrace: "Dynatrace",
    NewRelic: "NewRelic",
    ElasticAPM: "ElasticAPM",
    CACertificates: "CACertificates",
};
exports.ConfigServerEnabledState = {
    /**
     * Enable the config server.
     */
    Enabled: "Enabled",
    /**
     * Disable the config server.
     */
    Disabled: "Disabled",
};
exports.ConfigurationServiceGeneration = {
    Gen1: "Gen1",
    Gen2: "Gen2",
};
exports.CustomizedAcceleratorType = {
    Accelerator: "Accelerator",
    Fragment: "Fragment",
};
exports.DevToolPortalFeatureState = {
    /**
     * Enable the plugin in Dev Tool Portal.
     */
    Enabled: "Enabled",
    /**
     * Disable the plugin in Dev Tool Portal.
     */
    Disabled: "Disabled",
};
exports.Frequency = {
    Weekly: "Weekly",
};
exports.GatewayCertificateVerification = {
    /**
     * Enable certificate verification in Spring Cloud Gateway.
     */
    Enabled: "Enabled",
    /**
     * Disable certificate verification in Spring Cloud Gateway.
     */
    Disabled: "Disabled",
};
exports.GatewayRouteConfigProtocol = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.GitImplementation = {
    Go_git: "go-git",
    Libgit2: "libgit2",
};
exports.HTTPSchemeType = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.KeyVaultCertificateAutoSync = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ProbeActionType = {
    HTTPGetAction: "HTTPGetAction",
    TCPSocketAction: "TCPSocketAction",
    ExecAction: "ExecAction",
};
exports.SessionAffinity = {
    Cookie: "Cookie",
    None: "None",
};
exports.StorageType = {
    StorageAccount: "StorageAccount",
};
exports.Type = {
    AzureFileVolume: "AzureFileVolume",
};
exports.WeekDay = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
