// @pulumi/azure-native/types/enums/kubernetes/v20231101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.ProvisioningState = exports.PrivateLinkState = exports.ConnectedClusterKind = exports.AzureHybridBenefit = exports.AutoUpgradeOptions = exports.AuthenticationMethod = void 0;
exports.AuthenticationMethod = {
    Token: "Token",
    AAD: "AAD",
};
exports.AutoUpgradeOptions = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AzureHybridBenefit = {
    True: "True",
    False: "False",
    NotApplicable: "NotApplicable",
};
exports.ConnectedClusterKind = {
    ProvisionedCluster: "ProvisionedCluster",
};
exports.PrivateLinkState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Provisioning: "Provisioning",
    Updating: "Updating",
    Deleting: "Deleting",
    Accepted: "Accepted",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
