// @pulumi/azure-native/authorization/v20190601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicySetDefinitionAtManagementGroup = exports.PolicySetDefinition = exports.PolicyDefinitionAtManagementGroup = exports.PolicyDefinition = exports.PolicyAssignment = exports.getPolicySetDefinitionAtManagementGroupOutput = exports.getPolicySetDefinitionAtManagementGroup = exports.getPolicySetDefinitionOutput = exports.getPolicySetDefinition = exports.getPolicyDefinitionAtManagementGroupOutput = exports.getPolicyDefinitionAtManagementGroup = exports.getPolicyDefinitionOutput = exports.getPolicyDefinition = exports.getPolicyAssignmentOutput = exports.getPolicyAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPolicyAssignment = null;
exports.getPolicyAssignmentOutput = null;
utilities.lazyLoad(exports, ["getPolicyAssignment", "getPolicyAssignmentOutput"], () => require("./getPolicyAssignment"));
exports.getPolicyDefinition = null;
exports.getPolicyDefinitionOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinition", "getPolicyDefinitionOutput"], () => require("./getPolicyDefinition"));
exports.getPolicyDefinitionAtManagementGroup = null;
exports.getPolicyDefinitionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicyDefinitionAtManagementGroup", "getPolicyDefinitionAtManagementGroupOutput"], () => require("./getPolicyDefinitionAtManagementGroup"));
exports.getPolicySetDefinition = null;
exports.getPolicySetDefinitionOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinition", "getPolicySetDefinitionOutput"], () => require("./getPolicySetDefinition"));
exports.getPolicySetDefinitionAtManagementGroup = null;
exports.getPolicySetDefinitionAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getPolicySetDefinitionAtManagementGroup", "getPolicySetDefinitionAtManagementGroupOutput"], () => require("./getPolicySetDefinitionAtManagementGroup"));
exports.PolicyAssignment = null;
utilities.lazyLoad(exports, ["PolicyAssignment"], () => require("./policyAssignment"));
exports.PolicyDefinition = null;
utilities.lazyLoad(exports, ["PolicyDefinition"], () => require("./policyDefinition"));
exports.PolicyDefinitionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicyDefinitionAtManagementGroup"], () => require("./policyDefinitionAtManagementGroup"));
exports.PolicySetDefinition = null;
utilities.lazyLoad(exports, ["PolicySetDefinition"], () => require("./policySetDefinition"));
exports.PolicySetDefinitionAtManagementGroup = null;
utilities.lazyLoad(exports, ["PolicySetDefinitionAtManagementGroup"], () => require("./policySetDefinitionAtManagementGroup"));
// Export enums:
__exportStar(require("../../types/enums/authorization/v20190601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:authorization/v20190601:PolicyAssignment":
                return new exports.PolicyAssignment(name, undefined, { urn });
            case "azure-native:authorization/v20190601:PolicyDefinition":
                return new exports.PolicyDefinition(name, undefined, { urn });
            case "azure-native:authorization/v20190601:PolicyDefinitionAtManagementGroup":
                return new exports.PolicyDefinitionAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization/v20190601:PolicySetDefinition":
                return new exports.PolicySetDefinition(name, undefined, { urn });
            case "azure-native:authorization/v20190601:PolicySetDefinitionAtManagementGroup":
                return new exports.PolicySetDefinitionAtManagementGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "authorization/v20190601", _module);
