// @pulumi/azure-native/compute/v20221101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestorePoint = exports.getRestorePointOutput = exports.getRestorePoint = exports.getLogAnalyticExportThrottledRequestsOutput = exports.getLogAnalyticExportThrottledRequests = exports.getLogAnalyticExportRequestRateByIntervalOutput = exports.getLogAnalyticExportRequestRateByInterval = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLogAnalyticExportRequestRateByInterval = null;
exports.getLogAnalyticExportRequestRateByIntervalOutput = null;
utilities.lazyLoad(exports, ["getLogAnalyticExportRequestRateByInterval", "getLogAnalyticExportRequestRateByIntervalOutput"], () => require("./getLogAnalyticExportRequestRateByInterval"));
exports.getLogAnalyticExportThrottledRequests = null;
exports.getLogAnalyticExportThrottledRequestsOutput = null;
utilities.lazyLoad(exports, ["getLogAnalyticExportThrottledRequests", "getLogAnalyticExportThrottledRequestsOutput"], () => require("./getLogAnalyticExportThrottledRequests"));
exports.getRestorePoint = null;
exports.getRestorePointOutput = null;
utilities.lazyLoad(exports, ["getRestorePoint", "getRestorePointOutput"], () => require("./getRestorePoint"));
exports.RestorePoint = null;
utilities.lazyLoad(exports, ["RestorePoint"], () => require("./restorePoint"));
// Export enums:
__exportStar(require("../../types/enums/compute/v20221101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:compute/v20221101:RestorePoint":
                return new exports.RestorePoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "compute/v20221101", _module);
