// @pulumi/azure-native/servicefabricmesh/v20180901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = exports.SecretValue = exports.Secret = exports.Network = exports.listSecretValueOutput = exports.listSecretValue = exports.getVolumeOutput = exports.getVolume = exports.getSecretValueOutput = exports.getSecretValue = exports.getSecretOutput = exports.getSecret = exports.getNetworkOutput = exports.getNetwork = exports.getGatewayOutput = exports.getGateway = exports.getApplicationOutput = exports.getApplication = exports.Gateway = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getGateway = null;
exports.getGatewayOutput = null;
utilities.lazyLoad(exports, ["getGateway", "getGatewayOutput"], () => require("./getGateway"));
exports.getNetwork = null;
exports.getNetworkOutput = null;
utilities.lazyLoad(exports, ["getNetwork", "getNetworkOutput"], () => require("./getNetwork"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSecretValue = null;
exports.getSecretValueOutput = null;
utilities.lazyLoad(exports, ["getSecretValue", "getSecretValueOutput"], () => require("./getSecretValue"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.listSecretValue = null;
exports.listSecretValueOutput = null;
utilities.lazyLoad(exports, ["listSecretValue", "listSecretValueOutput"], () => require("./listSecretValue"));
exports.Network = null;
utilities.lazyLoad(exports, ["Network"], () => require("./network"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SecretValue = null;
utilities.lazyLoad(exports, ["SecretValue"], () => require("./secretValue"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
// Export enums:
__exportStar(require("../../types/enums/servicefabricmesh/v20180901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicefabricmesh/v20180901preview:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:servicefabricmesh/v20180901preview:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "azure-native:servicefabricmesh/v20180901preview:Network":
                return new exports.Network(name, undefined, { urn });
            case "azure-native:servicefabricmesh/v20180901preview:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "azure-native:servicefabricmesh/v20180901preview:SecretValue":
                return new exports.SecretValue(name, undefined, { urn });
            case "azure-native:servicefabricmesh/v20180901preview:Volume":
                return new exports.Volume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicefabricmesh/v20180901preview", _module);
