// @pulumi/azure-native/migrate/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPrivateEndpointConnectionOperationOutput = exports.getPrivateEndpointConnectionOperation = exports.getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnectionControllerPrivateEndpointConnection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getMoveResourceOutput = exports.getMoveResource = exports.getMoveCollectionOutput = exports.getMoveCollection = exports.getModernizeProjectOutput = exports.getModernizeProject = exports.getMigrateProjectsControllerMigrateProjectOutput = exports.getMigrateProjectsControllerMigrateProject = exports.getMigrateProjectOutput = exports.getMigrateProject = exports.getMigrateAgentOutput = exports.getMigrateAgent = exports.getImportCollectorsOperationOutput = exports.getImportCollectorsOperation = exports.getImportCollectorOutput = exports.getImportCollector = exports.getHypervCollectorsOperationOutput = exports.getHypervCollectorsOperation = exports.getHyperVCollectorOutput = exports.getHyperVCollector = exports.getGroupsOperationOutput = exports.getGroupsOperation = exports.getGroupOutput = exports.getGroup = exports.getBusinessCaseOperationReportDownloadUrlOutput = exports.getBusinessCaseOperationReportDownloadUrl = exports.getBusinessCaseOperationOutput = exports.getBusinessCaseOperation = exports.getAvsAssessmentsOperationOutput = exports.getAvsAssessmentsOperation = exports.getAssessmentsOperationOutput = exports.getAssessmentsOperation = exports.getAssessmentProjectsOperationOutput = exports.getAssessmentProjectsOperation = exports.getAssessmentOutput = exports.getAssessment = exports.getAksAssessmentOperationOutput = exports.getAksAssessmentOperation = exports.BusinessCaseOperation = exports.AvsAssessmentsOperation = exports.AssessmentsOperation = exports.AssessmentProjectsOperation = exports.Assessment = exports.AksAssessmentOperation = void 0;
exports.PrivateEndpointConnectionControllerPrivateEndpointConnection = exports.PrivateEndpointConnection = exports.MoveResource = exports.MoveCollection = exports.ModernizeProject = exports.MigrateProjectsControllerMigrateProject = exports.MigrateProject = exports.MigrateAgent = exports.ImportCollectorsOperation = exports.ImportCollector = exports.HypervCollectorsOperation = exports.HyperVCollector = exports.GroupsOperation = exports.Group = exports.getWorkloadInstanceOutput = exports.getWorkloadInstance = exports.getWorkloadDeploymentSecretConfigurationsOutput = exports.getWorkloadDeploymentSecretConfigurations = exports.getWorkloadDeploymentOutput = exports.getWorkloadDeployment = exports.getWebAppCollectorOperationOutput = exports.getWebAppCollectorOperation = exports.getWebAppAssessmentV2OperationOutput = exports.getWebAppAssessmentV2Operation = exports.getVmwareCollectorsOperationOutput = exports.getVmwareCollectorsOperation = exports.getVMwareCollectorOutput = exports.getVMwareCollector = exports.getSqlCollectorOperationOutput = exports.getSqlCollectorOperation = exports.getSqlAssessmentV2OperationOutput = exports.getSqlAssessmentV2Operation = exports.getSolutionsControllerSolutionOutput = exports.getSolutionsControllerSolution = exports.getSolutionsControllerConfigOutput = exports.getSolutionsControllerConfig = exports.getSolutionConfigOutput = exports.getSolutionConfig = exports.getSolutionOutput = exports.getSolution = exports.getServerCollectorsOperationOutput = exports.getServerCollectorsOperation = exports.getServerCollectorOutput = exports.getServerCollector = exports.getProjectKeysOutput = exports.getProjectKeys = exports.getProjectOutput = exports.getProject = exports.getPrivateEndpointConnectionProxyControllerOutput = exports.getPrivateEndpointConnectionProxyController = void 0;
exports.v20240101preview = exports.v20230909preview = exports.v20230801 = exports.v20230501preview = exports.v20230401preview = exports.v20230315 = exports.v20230101 = exports.v20220801 = exports.v20220501preview = exports.v20200501 = exports.v20191001 = exports.v20180901preview = exports.v20180202 = exports.WorkloadInstance = exports.WorkloadDeployment = exports.WebAppCollectorOperation = exports.WebAppAssessmentV2Operation = exports.VmwareCollectorsOperation = exports.VMwareCollector = exports.SqlCollectorOperation = exports.SqlAssessmentV2Operation = exports.SolutionsControllerSolution = exports.Solution = exports.ServerCollectorsOperation = exports.ServerCollector = exports.Project = exports.PrivateEndpointConnectionProxyController = exports.PrivateEndpointConnectionOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AksAssessmentOperation = null;
utilities.lazyLoad(exports, ["AksAssessmentOperation"], () => require("./aksAssessmentOperation"));
exports.Assessment = null;
utilities.lazyLoad(exports, ["Assessment"], () => require("./assessment"));
exports.AssessmentProjectsOperation = null;
utilities.lazyLoad(exports, ["AssessmentProjectsOperation"], () => require("./assessmentProjectsOperation"));
exports.AssessmentsOperation = null;
utilities.lazyLoad(exports, ["AssessmentsOperation"], () => require("./assessmentsOperation"));
exports.AvsAssessmentsOperation = null;
utilities.lazyLoad(exports, ["AvsAssessmentsOperation"], () => require("./avsAssessmentsOperation"));
exports.BusinessCaseOperation = null;
utilities.lazyLoad(exports, ["BusinessCaseOperation"], () => require("./businessCaseOperation"));
exports.getAksAssessmentOperation = null;
exports.getAksAssessmentOperationOutput = null;
utilities.lazyLoad(exports, ["getAksAssessmentOperation", "getAksAssessmentOperationOutput"], () => require("./getAksAssessmentOperation"));
exports.getAssessment = null;
exports.getAssessmentOutput = null;
utilities.lazyLoad(exports, ["getAssessment", "getAssessmentOutput"], () => require("./getAssessment"));
exports.getAssessmentProjectsOperation = null;
exports.getAssessmentProjectsOperationOutput = null;
utilities.lazyLoad(exports, ["getAssessmentProjectsOperation", "getAssessmentProjectsOperationOutput"], () => require("./getAssessmentProjectsOperation"));
exports.getAssessmentsOperation = null;
exports.getAssessmentsOperationOutput = null;
utilities.lazyLoad(exports, ["getAssessmentsOperation", "getAssessmentsOperationOutput"], () => require("./getAssessmentsOperation"));
exports.getAvsAssessmentsOperation = null;
exports.getAvsAssessmentsOperationOutput = null;
utilities.lazyLoad(exports, ["getAvsAssessmentsOperation", "getAvsAssessmentsOperationOutput"], () => require("./getAvsAssessmentsOperation"));
exports.getBusinessCaseOperation = null;
exports.getBusinessCaseOperationOutput = null;
utilities.lazyLoad(exports, ["getBusinessCaseOperation", "getBusinessCaseOperationOutput"], () => require("./getBusinessCaseOperation"));
exports.getBusinessCaseOperationReportDownloadUrl = null;
exports.getBusinessCaseOperationReportDownloadUrlOutput = null;
utilities.lazyLoad(exports, ["getBusinessCaseOperationReportDownloadUrl", "getBusinessCaseOperationReportDownloadUrlOutput"], () => require("./getBusinessCaseOperationReportDownloadUrl"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getGroupsOperation = null;
exports.getGroupsOperationOutput = null;
utilities.lazyLoad(exports, ["getGroupsOperation", "getGroupsOperationOutput"], () => require("./getGroupsOperation"));
exports.getHyperVCollector = null;
exports.getHyperVCollectorOutput = null;
utilities.lazyLoad(exports, ["getHyperVCollector", "getHyperVCollectorOutput"], () => require("./getHyperVCollector"));
exports.getHypervCollectorsOperation = null;
exports.getHypervCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getHypervCollectorsOperation", "getHypervCollectorsOperationOutput"], () => require("./getHypervCollectorsOperation"));
exports.getImportCollector = null;
exports.getImportCollectorOutput = null;
utilities.lazyLoad(exports, ["getImportCollector", "getImportCollectorOutput"], () => require("./getImportCollector"));
exports.getImportCollectorsOperation = null;
exports.getImportCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getImportCollectorsOperation", "getImportCollectorsOperationOutput"], () => require("./getImportCollectorsOperation"));
exports.getMigrateAgent = null;
exports.getMigrateAgentOutput = null;
utilities.lazyLoad(exports, ["getMigrateAgent", "getMigrateAgentOutput"], () => require("./getMigrateAgent"));
exports.getMigrateProject = null;
exports.getMigrateProjectOutput = null;
utilities.lazyLoad(exports, ["getMigrateProject", "getMigrateProjectOutput"], () => require("./getMigrateProject"));
exports.getMigrateProjectsControllerMigrateProject = null;
exports.getMigrateProjectsControllerMigrateProjectOutput = null;
utilities.lazyLoad(exports, ["getMigrateProjectsControllerMigrateProject", "getMigrateProjectsControllerMigrateProjectOutput"], () => require("./getMigrateProjectsControllerMigrateProject"));
exports.getModernizeProject = null;
exports.getModernizeProjectOutput = null;
utilities.lazyLoad(exports, ["getModernizeProject", "getModernizeProjectOutput"], () => require("./getModernizeProject"));
exports.getMoveCollection = null;
exports.getMoveCollectionOutput = null;
utilities.lazyLoad(exports, ["getMoveCollection", "getMoveCollectionOutput"], () => require("./getMoveCollection"));
exports.getMoveResource = null;
exports.getMoveResourceOutput = null;
utilities.lazyLoad(exports, ["getMoveResource", "getMoveResourceOutput"], () => require("./getMoveResource"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateEndpointConnectionControllerPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionControllerPrivateEndpointConnection", "getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnectionControllerPrivateEndpointConnection"));
exports.getPrivateEndpointConnectionOperation = null;
exports.getPrivateEndpointConnectionOperationOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionOperation", "getPrivateEndpointConnectionOperationOutput"], () => require("./getPrivateEndpointConnectionOperation"));
exports.getPrivateEndpointConnectionProxyController = null;
exports.getPrivateEndpointConnectionProxyControllerOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionProxyController", "getPrivateEndpointConnectionProxyControllerOutput"], () => require("./getPrivateEndpointConnectionProxyController"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getProjectKeys = null;
exports.getProjectKeysOutput = null;
utilities.lazyLoad(exports, ["getProjectKeys", "getProjectKeysOutput"], () => require("./getProjectKeys"));
exports.getServerCollector = null;
exports.getServerCollectorOutput = null;
utilities.lazyLoad(exports, ["getServerCollector", "getServerCollectorOutput"], () => require("./getServerCollector"));
exports.getServerCollectorsOperation = null;
exports.getServerCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getServerCollectorsOperation", "getServerCollectorsOperationOutput"], () => require("./getServerCollectorsOperation"));
exports.getSolution = null;
exports.getSolutionOutput = null;
utilities.lazyLoad(exports, ["getSolution", "getSolutionOutput"], () => require("./getSolution"));
exports.getSolutionConfig = null;
exports.getSolutionConfigOutput = null;
utilities.lazyLoad(exports, ["getSolutionConfig", "getSolutionConfigOutput"], () => require("./getSolutionConfig"));
exports.getSolutionsControllerConfig = null;
exports.getSolutionsControllerConfigOutput = null;
utilities.lazyLoad(exports, ["getSolutionsControllerConfig", "getSolutionsControllerConfigOutput"], () => require("./getSolutionsControllerConfig"));
exports.getSolutionsControllerSolution = null;
exports.getSolutionsControllerSolutionOutput = null;
utilities.lazyLoad(exports, ["getSolutionsControllerSolution", "getSolutionsControllerSolutionOutput"], () => require("./getSolutionsControllerSolution"));
exports.getSqlAssessmentV2Operation = null;
exports.getSqlAssessmentV2OperationOutput = null;
utilities.lazyLoad(exports, ["getSqlAssessmentV2Operation", "getSqlAssessmentV2OperationOutput"], () => require("./getSqlAssessmentV2Operation"));
exports.getSqlCollectorOperation = null;
exports.getSqlCollectorOperationOutput = null;
utilities.lazyLoad(exports, ["getSqlCollectorOperation", "getSqlCollectorOperationOutput"], () => require("./getSqlCollectorOperation"));
exports.getVMwareCollector = null;
exports.getVMwareCollectorOutput = null;
utilities.lazyLoad(exports, ["getVMwareCollector", "getVMwareCollectorOutput"], () => require("./getVMwareCollector"));
exports.getVmwareCollectorsOperation = null;
exports.getVmwareCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getVmwareCollectorsOperation", "getVmwareCollectorsOperationOutput"], () => require("./getVmwareCollectorsOperation"));
exports.getWebAppAssessmentV2Operation = null;
exports.getWebAppAssessmentV2OperationOutput = null;
utilities.lazyLoad(exports, ["getWebAppAssessmentV2Operation", "getWebAppAssessmentV2OperationOutput"], () => require("./getWebAppAssessmentV2Operation"));
exports.getWebAppCollectorOperation = null;
exports.getWebAppCollectorOperationOutput = null;
utilities.lazyLoad(exports, ["getWebAppCollectorOperation", "getWebAppCollectorOperationOutput"], () => require("./getWebAppCollectorOperation"));
exports.getWorkloadDeployment = null;
exports.getWorkloadDeploymentOutput = null;
utilities.lazyLoad(exports, ["getWorkloadDeployment", "getWorkloadDeploymentOutput"], () => require("./getWorkloadDeployment"));
exports.getWorkloadDeploymentSecretConfigurations = null;
exports.getWorkloadDeploymentSecretConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getWorkloadDeploymentSecretConfigurations", "getWorkloadDeploymentSecretConfigurationsOutput"], () => require("./getWorkloadDeploymentSecretConfigurations"));
exports.getWorkloadInstance = null;
exports.getWorkloadInstanceOutput = null;
utilities.lazyLoad(exports, ["getWorkloadInstance", "getWorkloadInstanceOutput"], () => require("./getWorkloadInstance"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupsOperation = null;
utilities.lazyLoad(exports, ["GroupsOperation"], () => require("./groupsOperation"));
exports.HyperVCollector = null;
utilities.lazyLoad(exports, ["HyperVCollector"], () => require("./hyperVCollector"));
exports.HypervCollectorsOperation = null;
utilities.lazyLoad(exports, ["HypervCollectorsOperation"], () => require("./hypervCollectorsOperation"));
exports.ImportCollector = null;
utilities.lazyLoad(exports, ["ImportCollector"], () => require("./importCollector"));
exports.ImportCollectorsOperation = null;
utilities.lazyLoad(exports, ["ImportCollectorsOperation"], () => require("./importCollectorsOperation"));
exports.MigrateAgent = null;
utilities.lazyLoad(exports, ["MigrateAgent"], () => require("./migrateAgent"));
exports.MigrateProject = null;
utilities.lazyLoad(exports, ["MigrateProject"], () => require("./migrateProject"));
exports.MigrateProjectsControllerMigrateProject = null;
utilities.lazyLoad(exports, ["MigrateProjectsControllerMigrateProject"], () => require("./migrateProjectsControllerMigrateProject"));
exports.ModernizeProject = null;
utilities.lazyLoad(exports, ["ModernizeProject"], () => require("./modernizeProject"));
exports.MoveCollection = null;
utilities.lazyLoad(exports, ["MoveCollection"], () => require("./moveCollection"));
exports.MoveResource = null;
utilities.lazyLoad(exports, ["MoveResource"], () => require("./moveResource"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateEndpointConnectionControllerPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionControllerPrivateEndpointConnection"], () => require("./privateEndpointConnectionControllerPrivateEndpointConnection"));
exports.PrivateEndpointConnectionOperation = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionOperation"], () => require("./privateEndpointConnectionOperation"));
exports.PrivateEndpointConnectionProxyController = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionProxyController"], () => require("./privateEndpointConnectionProxyController"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ServerCollector = null;
utilities.lazyLoad(exports, ["ServerCollector"], () => require("./serverCollector"));
exports.ServerCollectorsOperation = null;
utilities.lazyLoad(exports, ["ServerCollectorsOperation"], () => require("./serverCollectorsOperation"));
exports.Solution = null;
utilities.lazyLoad(exports, ["Solution"], () => require("./solution"));
exports.SolutionsControllerSolution = null;
utilities.lazyLoad(exports, ["SolutionsControllerSolution"], () => require("./solutionsControllerSolution"));
exports.SqlAssessmentV2Operation = null;
utilities.lazyLoad(exports, ["SqlAssessmentV2Operation"], () => require("./sqlAssessmentV2Operation"));
exports.SqlCollectorOperation = null;
utilities.lazyLoad(exports, ["SqlCollectorOperation"], () => require("./sqlCollectorOperation"));
exports.VMwareCollector = null;
utilities.lazyLoad(exports, ["VMwareCollector"], () => require("./vmwareCollector"));
exports.VmwareCollectorsOperation = null;
utilities.lazyLoad(exports, ["VmwareCollectorsOperation"], () => require("./vmwareCollectorsOperation"));
exports.WebAppAssessmentV2Operation = null;
utilities.lazyLoad(exports, ["WebAppAssessmentV2Operation"], () => require("./webAppAssessmentV2Operation"));
exports.WebAppCollectorOperation = null;
utilities.lazyLoad(exports, ["WebAppCollectorOperation"], () => require("./webAppCollectorOperation"));
exports.WorkloadDeployment = null;
utilities.lazyLoad(exports, ["WorkloadDeployment"], () => require("./workloadDeployment"));
exports.WorkloadInstance = null;
utilities.lazyLoad(exports, ["WorkloadInstance"], () => require("./workloadInstance"));
// Export enums:
__exportStar(require("../types/enums/migrate"), exports);
// Export sub-modules:
const v20180202 = require("./v20180202");
exports.v20180202 = v20180202;
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20220801 = require("./v20220801");
exports.v20220801 = v20220801;
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
const v20230315 = require("./v20230315");
exports.v20230315 = v20230315;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20230909preview = require("./v20230909preview");
exports.v20230909preview = v20230909preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate:AksAssessmentOperation":
                return new exports.AksAssessmentOperation(name, undefined, { urn });
            case "azure-native:migrate:Assessment":
                return new exports.Assessment(name, undefined, { urn });
            case "azure-native:migrate:AssessmentProjectsOperation":
                return new exports.AssessmentProjectsOperation(name, undefined, { urn });
            case "azure-native:migrate:AssessmentsOperation":
                return new exports.AssessmentsOperation(name, undefined, { urn });
            case "azure-native:migrate:AvsAssessmentsOperation":
                return new exports.AvsAssessmentsOperation(name, undefined, { urn });
            case "azure-native:migrate:BusinessCaseOperation":
                return new exports.BusinessCaseOperation(name, undefined, { urn });
            case "azure-native:migrate:Group":
                return new exports.Group(name, undefined, { urn });
            case "azure-native:migrate:GroupsOperation":
                return new exports.GroupsOperation(name, undefined, { urn });
            case "azure-native:migrate:HyperVCollector":
                return new exports.HyperVCollector(name, undefined, { urn });
            case "azure-native:migrate:HypervCollectorsOperation":
                return new exports.HypervCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate:ImportCollector":
                return new exports.ImportCollector(name, undefined, { urn });
            case "azure-native:migrate:ImportCollectorsOperation":
                return new exports.ImportCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate:MigrateAgent":
                return new exports.MigrateAgent(name, undefined, { urn });
            case "azure-native:migrate:MigrateProject":
                return new exports.MigrateProject(name, undefined, { urn });
            case "azure-native:migrate:MigrateProjectsControllerMigrateProject":
                return new exports.MigrateProjectsControllerMigrateProject(name, undefined, { urn });
            case "azure-native:migrate:ModernizeProject":
                return new exports.ModernizeProject(name, undefined, { urn });
            case "azure-native:migrate:MoveCollection":
                return new exports.MoveCollection(name, undefined, { urn });
            case "azure-native:migrate:MoveResource":
                return new exports.MoveResource(name, undefined, { urn });
            case "azure-native:migrate:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:migrate:PrivateEndpointConnectionControllerPrivateEndpointConnection":
                return new exports.PrivateEndpointConnectionControllerPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:migrate:PrivateEndpointConnectionOperation":
                return new exports.PrivateEndpointConnectionOperation(name, undefined, { urn });
            case "azure-native:migrate:PrivateEndpointConnectionProxyController":
                return new exports.PrivateEndpointConnectionProxyController(name, undefined, { urn });
            case "azure-native:migrate:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:migrate:ServerCollector":
                return new exports.ServerCollector(name, undefined, { urn });
            case "azure-native:migrate:ServerCollectorsOperation":
                return new exports.ServerCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate:Solution":
                return new exports.Solution(name, undefined, { urn });
            case "azure-native:migrate:SolutionsControllerSolution":
                return new exports.SolutionsControllerSolution(name, undefined, { urn });
            case "azure-native:migrate:SqlAssessmentV2Operation":
                return new exports.SqlAssessmentV2Operation(name, undefined, { urn });
            case "azure-native:migrate:SqlCollectorOperation":
                return new exports.SqlCollectorOperation(name, undefined, { urn });
            case "azure-native:migrate:VMwareCollector":
                return new exports.VMwareCollector(name, undefined, { urn });
            case "azure-native:migrate:VmwareCollectorsOperation":
                return new exports.VmwareCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate:WebAppAssessmentV2Operation":
                return new exports.WebAppAssessmentV2Operation(name, undefined, { urn });
            case "azure-native:migrate:WebAppCollectorOperation":
                return new exports.WebAppCollectorOperation(name, undefined, { urn });
            case "azure-native:migrate:WorkloadDeployment":
                return new exports.WorkloadDeployment(name, undefined, { urn });
            case "azure-native:migrate:WorkloadInstance":
                return new exports.WorkloadInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate", _module);
