// @pulumi/azure-native/types/enums/workloads/v20240901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAPVirtualInstanceIdentityType = exports.SAPSoftwareInstallationType = exports.SAPProductType = exports.SAPHighAvailabilityType = exports.SAPEnvironmentType = exports.SAPDeploymentType = exports.SAPDatabaseType = exports.SAPDatabaseScaleMethod = exports.SAPConfigurationType = exports.OSType = exports.NamingPatternType = exports.ManagedResourcesNetworkAccessType = exports.FileShareConfigurationType = exports.DiskSkuName = void 0;
exports.DiskSkuName = {
    /**
     * Standard LRS Disk SKU.
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Premium_LRS Disk SKU.
     */
    Premium_LRS: "Premium_LRS",
    /**
     * StandardSSD_LRS Disk SKU.
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * UltraSSD_LRS Disk SKU.
     */
    UltraSSD_LRS: "UltraSSD_LRS",
    /**
     * Premium_ZRS Disk SKU.
     */
    Premium_ZRS: "Premium_ZRS",
    /**
     * StandardSSD_ZRS Disk SKU.
     */
    StandardSSD_ZRS: "StandardSSD_ZRS",
    /**
     * PremiumV2_LRS Disk SKU.
     */
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.FileShareConfigurationType = {
    /**
     * Skip creating the file share.
     */
    Skip: "Skip",
    /**
     * Fileshare will be created and mounted by service.
     */
    CreateAndMount: "CreateAndMount",
    /**
     * Existing fileshare provided will be mounted by service.
     */
    Mount: "Mount",
};
exports.ManagedResourcesNetworkAccessType = {
    /**
     * Managed resources will be deployed with public network access enabled.
     */
    Public: "Public",
    /**
     * Managed resources will be deployed with public network access disabled.
     */
    Private: "Private",
};
exports.NamingPatternType = {
    /**
     * Full resource names that will be created by service.
     */
    FullResourceName: "FullResourceName",
};
exports.OSType = {
    /**
     * Linux OS Type.
     */
    Linux: "Linux",
    /**
     * Windows OS Type.
     */
    Windows: "Windows",
};
exports.SAPConfigurationType = {
    /**
     * SAP system will be deployed by service. No OS configurations will be done.
     */
    Deployment: "Deployment",
    /**
     * Existing SAP system will be registered.
     */
    Discovery: "Discovery",
    /**
     * SAP system will be deployed by service. OS configurations will be done.
     */
    DeploymentWithOSConfig: "DeploymentWithOSConfig",
};
exports.SAPDatabaseScaleMethod = {
    /**
     * ScaleUp Hana Database deployment type
     */
    ScaleUp: "ScaleUp",
};
exports.SAPDatabaseType = {
    /**
     * HANA Database type of SAP system.
     */
    HANA: "HANA",
    /**
     * DB2 database type of the SAP system.
     */
    DB2: "DB2",
};
exports.SAPDeploymentType = {
    /**
     * SAP Single server deployment type.
     */
    SingleServer: "SingleServer",
    /**
     * SAP Distributed deployment type.
     */
    ThreeTier: "ThreeTier",
};
exports.SAPEnvironmentType = {
    /**
     * Non Production SAP system.
     */
    NonProd: "NonProd",
    /**
     * Production SAP system.
     */
    Prod: "Prod",
};
exports.SAPHighAvailabilityType = {
    /**
     * HA deployment with availability sets.
     */
    AvailabilitySet: "AvailabilitySet",
    /**
     * HA deployment with availability zones.
     */
    AvailabilityZone: "AvailabilityZone",
};
exports.SAPProductType = {
    /**
     * SAP Product ECC.
     */
    ECC: "ECC",
    /**
     * SAP Product S4HANA.
     */
    S4HANA: "S4HANA",
    /**
     * SAP Products other than the ones listed.
     */
    Other: "Other",
};
exports.SAPSoftwareInstallationType = {
    /**
     * SAP Install managed by service.
     */
    ServiceInitiated: "ServiceInitiated",
    /**
     * SAP Install without OS Config.
     */
    SAPInstallWithoutOSConfig: "SAPInstallWithoutOSConfig",
    /**
     * External software installation type.
     */
    External: "External",
};
exports.SAPVirtualInstanceIdentityType = {
    /**
     * No managed identity.
     */
    None: "None",
    /**
     * User assigned managed identity.
     */
    UserAssigned: "UserAssigned",
};
