// @pulumi/azure-native/windowsiot/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20190601 = exports.Service = exports.getServiceOutput = exports.getService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
// Export sub-modules:
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:windowsiot:Service":
                return new exports.Service(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "windowsiot", _module);
