// @pulumi/azure-native/webpubsub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241001preview = exports.v20240801preview = exports.v20240401preview = exports.v20240301 = exports.v20240101preview = exports.v20230801preview = exports.v20230601preview = exports.v20230301preview = exports.v20230201 = exports.v20210901preview = exports.v20210601preview = exports.v20210401preview = exports.WebPubSubSharedPrivateLinkResource = exports.WebPubSubReplica = exports.WebPubSubPrivateEndpointConnection = exports.WebPubSubHub = exports.WebPubSubCustomDomain = exports.WebPubSubCustomCertificate = exports.WebPubSub = exports.listWebPubSubKeysOutput = exports.listWebPubSubKeys = exports.getWebPubSubSharedPrivateLinkResourceOutput = exports.getWebPubSubSharedPrivateLinkResource = exports.getWebPubSubReplicaOutput = exports.getWebPubSubReplica = exports.getWebPubSubPrivateEndpointConnectionOutput = exports.getWebPubSubPrivateEndpointConnection = exports.getWebPubSubHubOutput = exports.getWebPubSubHub = exports.getWebPubSubCustomDomainOutput = exports.getWebPubSubCustomDomain = exports.getWebPubSubCustomCertificateOutput = exports.getWebPubSubCustomCertificate = exports.getWebPubSubOutput = exports.getWebPubSub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getWebPubSub = null;
exports.getWebPubSubOutput = null;
utilities.lazyLoad(exports, ["getWebPubSub", "getWebPubSubOutput"], () => require("./getWebPubSub"));
exports.getWebPubSubCustomCertificate = null;
exports.getWebPubSubCustomCertificateOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubCustomCertificate", "getWebPubSubCustomCertificateOutput"], () => require("./getWebPubSubCustomCertificate"));
exports.getWebPubSubCustomDomain = null;
exports.getWebPubSubCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubCustomDomain", "getWebPubSubCustomDomainOutput"], () => require("./getWebPubSubCustomDomain"));
exports.getWebPubSubHub = null;
exports.getWebPubSubHubOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubHub", "getWebPubSubHubOutput"], () => require("./getWebPubSubHub"));
exports.getWebPubSubPrivateEndpointConnection = null;
exports.getWebPubSubPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubPrivateEndpointConnection", "getWebPubSubPrivateEndpointConnectionOutput"], () => require("./getWebPubSubPrivateEndpointConnection"));
exports.getWebPubSubReplica = null;
exports.getWebPubSubReplicaOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubReplica", "getWebPubSubReplicaOutput"], () => require("./getWebPubSubReplica"));
exports.getWebPubSubSharedPrivateLinkResource = null;
exports.getWebPubSubSharedPrivateLinkResourceOutput = null;
utilities.lazyLoad(exports, ["getWebPubSubSharedPrivateLinkResource", "getWebPubSubSharedPrivateLinkResourceOutput"], () => require("./getWebPubSubSharedPrivateLinkResource"));
exports.listWebPubSubKeys = null;
exports.listWebPubSubKeysOutput = null;
utilities.lazyLoad(exports, ["listWebPubSubKeys", "listWebPubSubKeysOutput"], () => require("./listWebPubSubKeys"));
exports.WebPubSub = null;
utilities.lazyLoad(exports, ["WebPubSub"], () => require("./webPubSub"));
exports.WebPubSubCustomCertificate = null;
utilities.lazyLoad(exports, ["WebPubSubCustomCertificate"], () => require("./webPubSubCustomCertificate"));
exports.WebPubSubCustomDomain = null;
utilities.lazyLoad(exports, ["WebPubSubCustomDomain"], () => require("./webPubSubCustomDomain"));
exports.WebPubSubHub = null;
utilities.lazyLoad(exports, ["WebPubSubHub"], () => require("./webPubSubHub"));
exports.WebPubSubPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["WebPubSubPrivateEndpointConnection"], () => require("./webPubSubPrivateEndpointConnection"));
exports.WebPubSubReplica = null;
utilities.lazyLoad(exports, ["WebPubSubReplica"], () => require("./webPubSubReplica"));
exports.WebPubSubSharedPrivateLinkResource = null;
utilities.lazyLoad(exports, ["WebPubSubSharedPrivateLinkResource"], () => require("./webPubSubSharedPrivateLinkResource"));
// Export enums:
__exportStar(require("../types/enums/webpubsub"), exports);
// Export sub-modules:
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20230201 = require("./v20230201");
exports.v20230201 = v20230201;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:webpubsub:WebPubSub":
                return new exports.WebPubSub(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubCustomCertificate":
                return new exports.WebPubSubCustomCertificate(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubCustomDomain":
                return new exports.WebPubSubCustomDomain(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubHub":
                return new exports.WebPubSubHub(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubPrivateEndpointConnection":
                return new exports.WebPubSubPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubReplica":
                return new exports.WebPubSubReplica(name, undefined, { urn });
            case "azure-native:webpubsub:WebPubSubSharedPrivateLinkResource":
                return new exports.WebPubSubSharedPrivateLinkResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "webpubsub", _module);
