// @pulumi/azure-native/media/v20230101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Track = exports.StreamingPolicy = exports.StreamingLocator = exports.PrivateEndpointConnection = exports.MediaService = exports.listStreamingLocatorPathsOutput = exports.listStreamingLocatorPaths = exports.listStreamingLocatorContentKeysOutput = exports.listStreamingLocatorContentKeys = exports.listMediaServiceEdgePoliciesOutput = exports.listMediaServiceEdgePolicies = exports.listAssetStreamingLocatorsOutput = exports.listAssetStreamingLocators = exports.listAssetContainerSasOutput = exports.listAssetContainerSas = exports.getTrackOutput = exports.getTrack = exports.getStreamingPolicyOutput = exports.getStreamingPolicy = exports.getStreamingLocatorOutput = exports.getStreamingLocator = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getMediaServiceOutput = exports.getMediaService = exports.getContentKeyPolicyPropertiesWithSecretsOutput = exports.getContentKeyPolicyPropertiesWithSecrets = exports.getContentKeyPolicyOutput = exports.getContentKeyPolicy = exports.getAssetFilterOutput = exports.getAssetFilter = exports.getAssetEncryptionKeyOutput = exports.getAssetEncryptionKey = exports.getAssetOutput = exports.getAsset = exports.getAccountFilterOutput = exports.getAccountFilter = exports.ContentKeyPolicy = exports.AssetFilter = exports.Asset = exports.AccountFilter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AccountFilter = null;
utilities.lazyLoad(exports, ["AccountFilter"], () => require("./accountFilter"));
exports.Asset = null;
utilities.lazyLoad(exports, ["Asset"], () => require("./asset"));
exports.AssetFilter = null;
utilities.lazyLoad(exports, ["AssetFilter"], () => require("./assetFilter"));
exports.ContentKeyPolicy = null;
utilities.lazyLoad(exports, ["ContentKeyPolicy"], () => require("./contentKeyPolicy"));
exports.getAccountFilter = null;
exports.getAccountFilterOutput = null;
utilities.lazyLoad(exports, ["getAccountFilter", "getAccountFilterOutput"], () => require("./getAccountFilter"));
exports.getAsset = null;
exports.getAssetOutput = null;
utilities.lazyLoad(exports, ["getAsset", "getAssetOutput"], () => require("./getAsset"));
exports.getAssetEncryptionKey = null;
exports.getAssetEncryptionKeyOutput = null;
utilities.lazyLoad(exports, ["getAssetEncryptionKey", "getAssetEncryptionKeyOutput"], () => require("./getAssetEncryptionKey"));
exports.getAssetFilter = null;
exports.getAssetFilterOutput = null;
utilities.lazyLoad(exports, ["getAssetFilter", "getAssetFilterOutput"], () => require("./getAssetFilter"));
exports.getContentKeyPolicy = null;
exports.getContentKeyPolicyOutput = null;
utilities.lazyLoad(exports, ["getContentKeyPolicy", "getContentKeyPolicyOutput"], () => require("./getContentKeyPolicy"));
exports.getContentKeyPolicyPropertiesWithSecrets = null;
exports.getContentKeyPolicyPropertiesWithSecretsOutput = null;
utilities.lazyLoad(exports, ["getContentKeyPolicyPropertiesWithSecrets", "getContentKeyPolicyPropertiesWithSecretsOutput"], () => require("./getContentKeyPolicyPropertiesWithSecrets"));
exports.getMediaService = null;
exports.getMediaServiceOutput = null;
utilities.lazyLoad(exports, ["getMediaService", "getMediaServiceOutput"], () => require("./getMediaService"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getStreamingLocator = null;
exports.getStreamingLocatorOutput = null;
utilities.lazyLoad(exports, ["getStreamingLocator", "getStreamingLocatorOutput"], () => require("./getStreamingLocator"));
exports.getStreamingPolicy = null;
exports.getStreamingPolicyOutput = null;
utilities.lazyLoad(exports, ["getStreamingPolicy", "getStreamingPolicyOutput"], () => require("./getStreamingPolicy"));
exports.getTrack = null;
exports.getTrackOutput = null;
utilities.lazyLoad(exports, ["getTrack", "getTrackOutput"], () => require("./getTrack"));
exports.listAssetContainerSas = null;
exports.listAssetContainerSasOutput = null;
utilities.lazyLoad(exports, ["listAssetContainerSas", "listAssetContainerSasOutput"], () => require("./listAssetContainerSas"));
exports.listAssetStreamingLocators = null;
exports.listAssetStreamingLocatorsOutput = null;
utilities.lazyLoad(exports, ["listAssetStreamingLocators", "listAssetStreamingLocatorsOutput"], () => require("./listAssetStreamingLocators"));
exports.listMediaServiceEdgePolicies = null;
exports.listMediaServiceEdgePoliciesOutput = null;
utilities.lazyLoad(exports, ["listMediaServiceEdgePolicies", "listMediaServiceEdgePoliciesOutput"], () => require("./listMediaServiceEdgePolicies"));
exports.listStreamingLocatorContentKeys = null;
exports.listStreamingLocatorContentKeysOutput = null;
utilities.lazyLoad(exports, ["listStreamingLocatorContentKeys", "listStreamingLocatorContentKeysOutput"], () => require("./listStreamingLocatorContentKeys"));
exports.listStreamingLocatorPaths = null;
exports.listStreamingLocatorPathsOutput = null;
utilities.lazyLoad(exports, ["listStreamingLocatorPaths", "listStreamingLocatorPathsOutput"], () => require("./listStreamingLocatorPaths"));
exports.MediaService = null;
utilities.lazyLoad(exports, ["MediaService"], () => require("./mediaService"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.StreamingLocator = null;
utilities.lazyLoad(exports, ["StreamingLocator"], () => require("./streamingLocator"));
exports.StreamingPolicy = null;
utilities.lazyLoad(exports, ["StreamingPolicy"], () => require("./streamingPolicy"));
exports.Track = null;
utilities.lazyLoad(exports, ["Track"], () => require("./track"));
// Export enums:
__exportStar(require("../../types/enums/media/v20230101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:media/v20230101:AccountFilter":
                return new exports.AccountFilter(name, undefined, { urn });
            case "azure-native:media/v20230101:Asset":
                return new exports.Asset(name, undefined, { urn });
            case "azure-native:media/v20230101:AssetFilter":
                return new exports.AssetFilter(name, undefined, { urn });
            case "azure-native:media/v20230101:ContentKeyPolicy":
                return new exports.ContentKeyPolicy(name, undefined, { urn });
            case "azure-native:media/v20230101:MediaService":
                return new exports.MediaService(name, undefined, { urn });
            case "azure-native:media/v20230101:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:media/v20230101:StreamingLocator":
                return new exports.StreamingLocator(name, undefined, { urn });
            case "azure-native:media/v20230101:StreamingPolicy":
                return new exports.StreamingPolicy(name, undefined, { urn });
            case "azure-native:media/v20230101:Track":
                return new exports.Track(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "media/v20230101", _module);
