// @pulumi/azure-native/healthcareapis/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250301preview = exports.v20240331 = exports.v20240301 = exports.v20231201 = exports.v20231101 = exports.v20230906 = exports.v20230228 = exports.v20221001preview = exports.WorkspacePrivateEndpointConnection = exports.Workspace = exports.Service = exports.PrivateEndpointConnection = exports.IotConnectorFhirDestination = exports.IotConnector = exports.getWorkspacePrivateEndpointConnectionOutput = exports.getWorkspacePrivateEndpointConnection = exports.getWorkspaceOutput = exports.getWorkspace = exports.getServiceOutput = exports.getService = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getIotConnectorFhirDestinationOutput = exports.getIotConnectorFhirDestination = exports.getIotConnectorOutput = exports.getIotConnector = exports.getFhirServiceOutput = exports.getFhirService = exports.getDicomServiceOutput = exports.getDicomService = exports.getAnalyticsConnectorOutput = exports.getAnalyticsConnector = exports.FhirService = exports.DicomService = exports.AnalyticsConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalyticsConnector = null;
utilities.lazyLoad(exports, ["AnalyticsConnector"], () => require("./analyticsConnector"));
exports.DicomService = null;
utilities.lazyLoad(exports, ["DicomService"], () => require("./dicomService"));
exports.FhirService = null;
utilities.lazyLoad(exports, ["FhirService"], () => require("./fhirService"));
exports.getAnalyticsConnector = null;
exports.getAnalyticsConnectorOutput = null;
utilities.lazyLoad(exports, ["getAnalyticsConnector", "getAnalyticsConnectorOutput"], () => require("./getAnalyticsConnector"));
exports.getDicomService = null;
exports.getDicomServiceOutput = null;
utilities.lazyLoad(exports, ["getDicomService", "getDicomServiceOutput"], () => require("./getDicomService"));
exports.getFhirService = null;
exports.getFhirServiceOutput = null;
utilities.lazyLoad(exports, ["getFhirService", "getFhirServiceOutput"], () => require("./getFhirService"));
exports.getIotConnector = null;
exports.getIotConnectorOutput = null;
utilities.lazyLoad(exports, ["getIotConnector", "getIotConnectorOutput"], () => require("./getIotConnector"));
exports.getIotConnectorFhirDestination = null;
exports.getIotConnectorFhirDestinationOutput = null;
utilities.lazyLoad(exports, ["getIotConnectorFhirDestination", "getIotConnectorFhirDestinationOutput"], () => require("./getIotConnectorFhirDestination"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspacePrivateEndpointConnection = null;
exports.getWorkspacePrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspacePrivateEndpointConnection", "getWorkspacePrivateEndpointConnectionOutput"], () => require("./getWorkspacePrivateEndpointConnection"));
exports.IotConnector = null;
utilities.lazyLoad(exports, ["IotConnector"], () => require("./iotConnector"));
exports.IotConnectorFhirDestination = null;
utilities.lazyLoad(exports, ["IotConnectorFhirDestination"], () => require("./iotConnectorFhirDestination"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspacePrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["WorkspacePrivateEndpointConnection"], () => require("./workspacePrivateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/healthcareapis"), exports);
// Export sub-modules:
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230228 = require("./v20230228");
exports.v20230228 = v20230228;
const v20230906 = require("./v20230906");
exports.v20230906 = v20230906;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240331 = require("./v20240331");
exports.v20240331 = v20240331;
const v20250301preview = require("./v20250301preview");
exports.v20250301preview = v20250301preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:healthcareapis:AnalyticsConnector":
                return new exports.AnalyticsConnector(name, undefined, { urn });
            case "azure-native:healthcareapis:DicomService":
                return new exports.DicomService(name, undefined, { urn });
            case "azure-native:healthcareapis:FhirService":
                return new exports.FhirService(name, undefined, { urn });
            case "azure-native:healthcareapis:IotConnector":
                return new exports.IotConnector(name, undefined, { urn });
            case "azure-native:healthcareapis:IotConnectorFhirDestination":
                return new exports.IotConnectorFhirDestination(name, undefined, { urn });
            case "azure-native:healthcareapis:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:healthcareapis:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:healthcareapis:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "azure-native:healthcareapis:WorkspacePrivateEndpointConnection":
                return new exports.WorkspacePrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "healthcareapis", _module);
