// @pulumi/azure-native/storagecache/v20240701/importJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An import job instance. Follows Azure Resource Manager standards: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md
 */
class ImportJob extends pulumi.CustomResource {
    /**
     * Get an existing ImportJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ImportJob(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ImportJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ImportJob.__pulumiType;
    }
    /**
     * Create a ImportJob resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.amlFilesystemName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'amlFilesystemName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["adminStatus"] = (_a = (args ? args.adminStatus : undefined)) !== null && _a !== void 0 ? _a : "Active";
            resourceInputs["amlFilesystemName"] = args ? args.amlFilesystemName : undefined;
            resourceInputs["conflictResolutionMode"] = (_b = (args ? args.conflictResolutionMode : undefined)) !== null && _b !== void 0 ? _b : "Fail";
            resourceInputs["importJobName"] = args ? args.importJobName : undefined;
            resourceInputs["importPrefixes"] = args ? args.importPrefixes : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maximumErrors"] = (_c = (args ? args.maximumErrors : undefined)) !== null && _c !== void 0 ? _c : 0;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["blobsImportedPerSecond"] = undefined /*out*/;
            resourceInputs["blobsWalkedPerSecond"] = undefined /*out*/;
            resourceInputs["importedDirectories"] = undefined /*out*/;
            resourceInputs["importedFiles"] = undefined /*out*/;
            resourceInputs["importedSymlinks"] = undefined /*out*/;
            resourceInputs["lastCompletionTime"] = undefined /*out*/;
            resourceInputs["lastStartedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["preexistingDirectories"] = undefined /*out*/;
            resourceInputs["preexistingFiles"] = undefined /*out*/;
            resourceInputs["preexistingSymlinks"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["totalBlobsImported"] = undefined /*out*/;
            resourceInputs["totalBlobsWalked"] = undefined /*out*/;
            resourceInputs["totalConflicts"] = undefined /*out*/;
            resourceInputs["totalErrors"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["adminStatus"] = undefined /*out*/;
            resourceInputs["blobsImportedPerSecond"] = undefined /*out*/;
            resourceInputs["blobsWalkedPerSecond"] = undefined /*out*/;
            resourceInputs["conflictResolutionMode"] = undefined /*out*/;
            resourceInputs["importPrefixes"] = undefined /*out*/;
            resourceInputs["importedDirectories"] = undefined /*out*/;
            resourceInputs["importedFiles"] = undefined /*out*/;
            resourceInputs["importedSymlinks"] = undefined /*out*/;
            resourceInputs["lastCompletionTime"] = undefined /*out*/;
            resourceInputs["lastStartedTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maximumErrors"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["preexistingDirectories"] = undefined /*out*/;
            resourceInputs["preexistingFiles"] = undefined /*out*/;
            resourceInputs["preexistingSymlinks"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["totalBlobsImported"] = undefined /*out*/;
            resourceInputs["totalBlobsWalked"] = undefined /*out*/;
            resourceInputs["totalConflicts"] = undefined /*out*/;
            resourceInputs["totalErrors"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storagecache/v20240301:ImportJob" }, { type: "azure-native:storagecache/v20240301:importJob" }, { type: "azure-native:storagecache/v20240701:importJob" }, { type: "azure-native:storagecache:ImportJob" }, { type: "azure-native:storagecache:importJob" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ImportJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ImportJob = ImportJob;
/** @internal */
ImportJob.__pulumiType = 'azure-native:storagecache/v20240701:ImportJob';
