// @pulumi/azure-native/web/v20210115/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppScmAllowedSlot = exports.WebAppScmAllowed = exports.WebAppFtpAllowedSlot = exports.WebAppFtpAllowed = exports.WebAppDiagnosticLogsConfigurationSlot = exports.listSiteIdentifiersAssignedToHostNameOutput = exports.listSiteIdentifiersAssignedToHostName = exports.getWebAppScmAllowedSlotOutput = exports.getWebAppScmAllowedSlot = exports.getWebAppScmAllowedOutput = exports.getWebAppScmAllowed = exports.getWebAppFtpAllowedSlotOutput = exports.getWebAppFtpAllowedSlot = exports.getWebAppFtpAllowedOutput = exports.getWebAppFtpAllowed = exports.getWebAppDiagnosticLogsConfigurationSlotOutput = exports.getWebAppDiagnosticLogsConfigurationSlot = exports.getAppServiceEnvironmentOutput = exports.getAppServiceEnvironment = exports.AppServiceEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AppServiceEnvironment = null;
utilities.lazyLoad(exports, ["AppServiceEnvironment"], () => require("./appServiceEnvironment"));
exports.getAppServiceEnvironment = null;
exports.getAppServiceEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getAppServiceEnvironment", "getAppServiceEnvironmentOutput"], () => require("./getAppServiceEnvironment"));
exports.getWebAppDiagnosticLogsConfigurationSlot = null;
exports.getWebAppDiagnosticLogsConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfigurationSlot", "getWebAppDiagnosticLogsConfigurationSlotOutput"], () => require("./getWebAppDiagnosticLogsConfigurationSlot"));
exports.getWebAppFtpAllowed = null;
exports.getWebAppFtpAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppFtpAllowed", "getWebAppFtpAllowedOutput"], () => require("./getWebAppFtpAllowed"));
exports.getWebAppFtpAllowedSlot = null;
exports.getWebAppFtpAllowedSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppFtpAllowedSlot", "getWebAppFtpAllowedSlotOutput"], () => require("./getWebAppFtpAllowedSlot"));
exports.getWebAppScmAllowed = null;
exports.getWebAppScmAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppScmAllowed", "getWebAppScmAllowedOutput"], () => require("./getWebAppScmAllowed"));
exports.getWebAppScmAllowedSlot = null;
exports.getWebAppScmAllowedSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppScmAllowedSlot", "getWebAppScmAllowedSlotOutput"], () => require("./getWebAppScmAllowedSlot"));
exports.listSiteIdentifiersAssignedToHostName = null;
exports.listSiteIdentifiersAssignedToHostNameOutput = null;
utilities.lazyLoad(exports, ["listSiteIdentifiersAssignedToHostName", "listSiteIdentifiersAssignedToHostNameOutput"], () => require("./listSiteIdentifiersAssignedToHostName"));
exports.WebAppDiagnosticLogsConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfigurationSlot"], () => require("./webAppDiagnosticLogsConfigurationSlot"));
exports.WebAppFtpAllowed = null;
utilities.lazyLoad(exports, ["WebAppFtpAllowed"], () => require("./webAppFtpAllowed"));
exports.WebAppFtpAllowedSlot = null;
utilities.lazyLoad(exports, ["WebAppFtpAllowedSlot"], () => require("./webAppFtpAllowedSlot"));
exports.WebAppScmAllowed = null;
utilities.lazyLoad(exports, ["WebAppScmAllowed"], () => require("./webAppScmAllowed"));
exports.WebAppScmAllowedSlot = null;
utilities.lazyLoad(exports, ["WebAppScmAllowedSlot"], () => require("./webAppScmAllowedSlot"));
// Export enums:
__exportStar(require("../../types/enums/web/v20210115"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20210115:AppServiceEnvironment":
                return new exports.AppServiceEnvironment(name, undefined, { urn });
            case "azure-native:web/v20210115:WebAppDiagnosticLogsConfigurationSlot":
                return new exports.WebAppDiagnosticLogsConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20210115:WebAppFtpAllowed":
                return new exports.WebAppFtpAllowed(name, undefined, { urn });
            case "azure-native:web/v20210115:WebAppFtpAllowedSlot":
                return new exports.WebAppFtpAllowedSlot(name, undefined, { urn });
            case "azure-native:web/v20210115:WebAppScmAllowed":
                return new exports.WebAppScmAllowed(name, undefined, { urn });
            case "azure-native:web/v20210115:WebAppScmAllowedSlot":
                return new exports.WebAppScmAllowedSlot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20210115", _module);
