// @pulumi/azure-native/redhatopenshift/v20230701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncSet = exports.SyncIdentityProvider = exports.Secret = exports.OpenShiftCluster = exports.MachinePool = exports.listOpenShiftClusterCredentialsOutput = exports.listOpenShiftClusterCredentials = exports.listOpenShiftClusterAdminCredentialsOutput = exports.listOpenShiftClusterAdminCredentials = exports.getSyncSetOutput = exports.getSyncSet = exports.getSyncIdentityProviderOutput = exports.getSyncIdentityProvider = exports.getSecretOutput = exports.getSecret = exports.getOpenShiftClusterOutput = exports.getOpenShiftCluster = exports.getMachinePoolOutput = exports.getMachinePool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMachinePool = null;
exports.getMachinePoolOutput = null;
utilities.lazyLoad(exports, ["getMachinePool", "getMachinePoolOutput"], () => require("./getMachinePool"));
exports.getOpenShiftCluster = null;
exports.getOpenShiftClusterOutput = null;
utilities.lazyLoad(exports, ["getOpenShiftCluster", "getOpenShiftClusterOutput"], () => require("./getOpenShiftCluster"));
exports.getSecret = null;
exports.getSecretOutput = null;
utilities.lazyLoad(exports, ["getSecret", "getSecretOutput"], () => require("./getSecret"));
exports.getSyncIdentityProvider = null;
exports.getSyncIdentityProviderOutput = null;
utilities.lazyLoad(exports, ["getSyncIdentityProvider", "getSyncIdentityProviderOutput"], () => require("./getSyncIdentityProvider"));
exports.getSyncSet = null;
exports.getSyncSetOutput = null;
utilities.lazyLoad(exports, ["getSyncSet", "getSyncSetOutput"], () => require("./getSyncSet"));
exports.listOpenShiftClusterAdminCredentials = null;
exports.listOpenShiftClusterAdminCredentialsOutput = null;
utilities.lazyLoad(exports, ["listOpenShiftClusterAdminCredentials", "listOpenShiftClusterAdminCredentialsOutput"], () => require("./listOpenShiftClusterAdminCredentials"));
exports.listOpenShiftClusterCredentials = null;
exports.listOpenShiftClusterCredentialsOutput = null;
utilities.lazyLoad(exports, ["listOpenShiftClusterCredentials", "listOpenShiftClusterCredentialsOutput"], () => require("./listOpenShiftClusterCredentials"));
exports.MachinePool = null;
utilities.lazyLoad(exports, ["MachinePool"], () => require("./machinePool"));
exports.OpenShiftCluster = null;
utilities.lazyLoad(exports, ["OpenShiftCluster"], () => require("./openShiftCluster"));
exports.Secret = null;
utilities.lazyLoad(exports, ["Secret"], () => require("./secret"));
exports.SyncIdentityProvider = null;
utilities.lazyLoad(exports, ["SyncIdentityProvider"], () => require("./syncIdentityProvider"));
exports.SyncSet = null;
utilities.lazyLoad(exports, ["SyncSet"], () => require("./syncSet"));
// Export enums:
__exportStar(require("../../types/enums/redhatopenshift/v20230701preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:redhatopenshift/v20230701preview:MachinePool":
                return new exports.MachinePool(name, undefined, { urn });
            case "azure-native:redhatopenshift/v20230701preview:OpenShiftCluster":
                return new exports.OpenShiftCluster(name, undefined, { urn });
            case "azure-native:redhatopenshift/v20230701preview:Secret":
                return new exports.Secret(name, undefined, { urn });
            case "azure-native:redhatopenshift/v20230701preview:SyncIdentityProvider":
                return new exports.SyncIdentityProvider(name, undefined, { urn });
            case "azure-native:redhatopenshift/v20230701preview:SyncSet":
                return new exports.SyncSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "redhatopenshift/v20230701preview", _module);
