// @pulumi/azure-native/types/enums/peering/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionAddressProvider = exports.Role = exports.Kind = exports.DirectPeeringType = exports.v20221001 = exports.v20210101 = void 0;
// Export sub-modules:
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
exports.DirectPeeringType = {
    Edge: "Edge",
    Transit: "Transit",
    Cdn: "Cdn",
    Internal: "Internal",
    Ix: "Ix",
    IxRs: "IxRs",
    Voice: "Voice",
    EdgeZoneForOperators: "EdgeZoneForOperators",
};
exports.Kind = {
    Direct: "Direct",
    Exchange: "Exchange",
};
exports.Role = {
    Noc: "Noc",
    Policy: "Policy",
    Technical: "Technical",
    Service: "Service",
    Escalation: "Escalation",
    Other: "Other",
};
exports.SessionAddressProvider = {
    Microsoft: "Microsoft",
    Peer: "Peer",
};
