// @pulumi/azure-native/maps/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240701preview = exports.v20240101preview = exports.v20231201preview = exports.v20230801preview = exports.v20230601 = exports.v20211201preview = exports.v20210201 = exports.v20200201preview = exports.v20180501 = exports.PrivateEndpointConnection = exports.PrivateAtlase = exports.listAccountSasOutput = exports.listAccountSas = exports.listAccountKeysOutput = exports.listAccountKeys = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getPrivateAtlaseOutput = exports.getPrivateAtlase = exports.getCreatorOutput = exports.getCreator = exports.getAccountOutput = exports.getAccount = exports.Creator = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.Creator = null;
utilities.lazyLoad(exports, ["Creator"], () => require("./creator"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getCreator = null;
exports.getCreatorOutput = null;
utilities.lazyLoad(exports, ["getCreator", "getCreatorOutput"], () => require("./getCreator"));
exports.getPrivateAtlase = null;
exports.getPrivateAtlaseOutput = null;
utilities.lazyLoad(exports, ["getPrivateAtlase", "getPrivateAtlaseOutput"], () => require("./getPrivateAtlase"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.listAccountKeys = null;
exports.listAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listAccountKeys", "listAccountKeysOutput"], () => require("./listAccountKeys"));
exports.listAccountSas = null;
exports.listAccountSasOutput = null;
utilities.lazyLoad(exports, ["listAccountSas", "listAccountSasOutput"], () => require("./listAccountSas"));
exports.PrivateAtlase = null;
utilities.lazyLoad(exports, ["PrivateAtlase"], () => require("./privateAtlase"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/maps"), exports);
// Export sub-modules:
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20200201preview = require("./v20200201preview");
exports.v20200201preview = v20200201preview;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20230601 = require("./v20230601");
exports.v20230601 = v20230601;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:maps:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:maps:Creator":
                return new exports.Creator(name, undefined, { urn });
            case "azure-native:maps:PrivateAtlase":
                return new exports.PrivateAtlase(name, undefined, { urn });
            case "azure-native:maps:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "maps", _module);
