// @pulumi/azure-native/communication/emailService.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A class representing an EmailService resource.
 *
 * Uses Azure REST API version 2023-03-31. In version 1.x of the Azure Native provider, it used API version 2021-10-01-preview.
 *
 * Other available API versions: 2023-04-01, 2023-04-01-preview, 2023-06-01-preview, 2024-09-01-preview.
 */
class EmailService extends pulumi.CustomResource {
    /**
     * Get an existing EmailService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EmailService(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EmailService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EmailService.__pulumiType;
    }
    /**
     * Create a EmailService resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLocation'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["dataLocation"] = args ? args.dataLocation : undefined;
            resourceInputs["emailServiceName"] = args ? args.emailServiceName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataLocation"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:communication/v20211001preview:EmailService" }, { type: "azure-native:communication/v20220701preview:EmailService" }, { type: "azure-native:communication/v20230301preview:EmailService" }, { type: "azure-native:communication/v20230331:EmailService" }, { type: "azure-native:communication/v20230401:EmailService" }, { type: "azure-native:communication/v20230401preview:EmailService" }, { type: "azure-native:communication/v20230601preview:EmailService" }, { type: "azure-native:communication/v20240901preview:EmailService" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EmailService.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EmailService = EmailService;
/** @internal */
EmailService.__pulumiType = 'azure-native:communication:EmailService';
