// @pulumi/azure-native/communication/suppressionList.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuppressionList = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A class representing a SuppressionList resource.
 *
 * Uses Azure REST API version 2023-06-01-preview.
 *
 * Other available API versions: 2024-09-01-preview.
 */
class SuppressionList extends pulumi.CustomResource {
    /**
     * Get an existing SuppressionList resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SuppressionList(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SuppressionList.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SuppressionList.__pulumiType;
    }
    /**
     * Create a SuppressionList resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.emailServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'emailServiceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["emailServiceName"] = args ? args.emailServiceName : undefined;
            resourceInputs["listName"] = args ? args.listName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["suppressionListName"] = args ? args.suppressionListName : undefined;
            resourceInputs["createdTimeStamp"] = undefined /*out*/;
            resourceInputs["dataLocation"] = undefined /*out*/;
            resourceInputs["lastUpdatedTimeStamp"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdTimeStamp"] = undefined /*out*/;
            resourceInputs["dataLocation"] = undefined /*out*/;
            resourceInputs["lastUpdatedTimeStamp"] = undefined /*out*/;
            resourceInputs["listName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:communication/v20230601preview:SuppressionList" }, { type: "azure-native:communication/v20240901preview:SuppressionList" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SuppressionList.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SuppressionList = SuppressionList;
/** @internal */
SuppressionList.__pulumiType = 'azure-native:communication:SuppressionList';
