// @pulumi/azure-native/desktopvirtualization/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostPool = exports.getHostPoolRegistrationTokenOutput = exports.getHostPoolRegistrationToken = exports.getHostPoolOutput = exports.getHostPool = exports.getApplicationGroupOutput = exports.getApplicationGroup = exports.ApplicationGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ApplicationGroup = null;
utilities.lazyLoad(exports, ["ApplicationGroup"], () => require("./applicationGroup"));
exports.getApplicationGroup = null;
exports.getApplicationGroupOutput = null;
utilities.lazyLoad(exports, ["getApplicationGroup", "getApplicationGroupOutput"], () => require("./getApplicationGroup"));
exports.getHostPool = null;
exports.getHostPoolOutput = null;
utilities.lazyLoad(exports, ["getHostPool", "getHostPoolOutput"], () => require("./getHostPool"));
exports.getHostPoolRegistrationToken = null;
exports.getHostPoolRegistrationTokenOutput = null;
utilities.lazyLoad(exports, ["getHostPoolRegistrationToken", "getHostPoolRegistrationTokenOutput"], () => require("./getHostPoolRegistrationToken"));
exports.HostPool = null;
utilities.lazyLoad(exports, ["HostPool"], () => require("./hostPool"));
// Export enums:
__exportStar(require("../../types/enums/desktopvirtualization/v20220401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:desktopvirtualization/v20220401preview:ApplicationGroup":
                return new exports.ApplicationGroup(name, undefined, { urn });
            case "azure-native:desktopvirtualization/v20220401preview:HostPool":
                return new exports.HostPool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "desktopvirtualization/v20220401preview", _module);
