// @pulumi/azure-native/apimanagement/v20190101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscription = exports.Property = exports.IdentityProvider = exports.getSubscriptionOutput = exports.getSubscription = exports.getPropertyOutput = exports.getProperty = exports.getIdentityProviderOutput = exports.getIdentityProvider = exports.getDiagnosticOutput = exports.getDiagnostic = exports.getApiTagDescriptionOutput = exports.getApiTagDescription = exports.getApiSchemaOutput = exports.getApiSchema = exports.getApiDiagnosticOutput = exports.getApiDiagnostic = exports.Diagnostic = exports.ApiTagDescription = exports.ApiSchema = exports.ApiDiagnostic = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ApiDiagnostic = null;
utilities.lazyLoad(exports, ["ApiDiagnostic"], () => require("./apiDiagnostic"));
exports.ApiSchema = null;
utilities.lazyLoad(exports, ["ApiSchema"], () => require("./apiSchema"));
exports.ApiTagDescription = null;
utilities.lazyLoad(exports, ["ApiTagDescription"], () => require("./apiTagDescription"));
exports.Diagnostic = null;
utilities.lazyLoad(exports, ["Diagnostic"], () => require("./diagnostic"));
exports.getApiDiagnostic = null;
exports.getApiDiagnosticOutput = null;
utilities.lazyLoad(exports, ["getApiDiagnostic", "getApiDiagnosticOutput"], () => require("./getApiDiagnostic"));
exports.getApiSchema = null;
exports.getApiSchemaOutput = null;
utilities.lazyLoad(exports, ["getApiSchema", "getApiSchemaOutput"], () => require("./getApiSchema"));
exports.getApiTagDescription = null;
exports.getApiTagDescriptionOutput = null;
utilities.lazyLoad(exports, ["getApiTagDescription", "getApiTagDescriptionOutput"], () => require("./getApiTagDescription"));
exports.getDiagnostic = null;
exports.getDiagnosticOutput = null;
utilities.lazyLoad(exports, ["getDiagnostic", "getDiagnosticOutput"], () => require("./getDiagnostic"));
exports.getIdentityProvider = null;
exports.getIdentityProviderOutput = null;
utilities.lazyLoad(exports, ["getIdentityProvider", "getIdentityProviderOutput"], () => require("./getIdentityProvider"));
exports.getProperty = null;
exports.getPropertyOutput = null;
utilities.lazyLoad(exports, ["getProperty", "getPropertyOutput"], () => require("./getProperty"));
exports.getSubscription = null;
exports.getSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSubscription", "getSubscriptionOutput"], () => require("./getSubscription"));
exports.IdentityProvider = null;
utilities.lazyLoad(exports, ["IdentityProvider"], () => require("./identityProvider"));
exports.Property = null;
utilities.lazyLoad(exports, ["Property"], () => require("./property"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20190101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20190101:ApiDiagnostic":
                return new exports.ApiDiagnostic(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:ApiSchema":
                return new exports.ApiSchema(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:ApiTagDescription":
                return new exports.ApiTagDescription(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:Diagnostic":
                return new exports.Diagnostic(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:IdentityProvider":
                return new exports.IdentityProvider(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:Property":
                return new exports.Property(name, undefined, { urn });
            case "azure-native:apimanagement/v20190101:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20190101", _module);
