// @pulumi/azure-native/types/enums/connectedcache/v20230501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyRequired = exports.OsType = exports.CycleType = exports.CustomerTransitState = exports.AutoUpdateRingType = void 0;
exports.AutoUpdateRingType = {
    /**
     * customer selection of preview update install mcc on their physical vm
     */
    Preview: "Preview",
    /**
     * customer selection of slow update to install mcc on their physical vm
     */
    Slow: "Slow",
    /**
     * customer selection of fast / auto update to install mcc on their physical vm
     */
    Fast: "Fast",
};
exports.CustomerTransitState = {
    /**
     * do not have transit
     */
    NoTransit: "NoTransit",
    /**
     * transit provider and have own subscribers
     */
    CombinedTransit: "CombinedTransit",
    /**
     * pure transit provider or network service provider
     */
    TransitOnly: "TransitOnly",
};
exports.CycleType = {
    /**
     * customer selection of preview update install mcc on their physical vm
     */
    Preview: "Preview",
    /**
     * customer selection of slow update to install mcc on their physical vm
     */
    Slow: "Slow",
    /**
     * customer selection of fast / auto update to install mcc on their physical vm
     */
    Fast: "Fast",
};
exports.OsType = {
    /**
     * cache node installs on windows operating system
     */
    Windows: "Windows",
    /**
     * cache node installs on Linux Operating system
     */
    Linux: "Linux",
    /**
     * cache node installs on Azure Eflow
     */
    Eflow: "Eflow",
};
exports.ProxyRequired = {
    /**
     * Proxy is not required in setup
     */
    None: "None",
    /**
     * proxy is required in setup
     */
    Required: "Required",
};
