// @pulumi/azure-native/apicenter/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240601preview = exports.v20240315preview = exports.v20240301 = exports.v20230701preview = exports.Workspace = exports.Service = exports.MetadataSchema = exports.getWorkspaceOutput = exports.getWorkspace = exports.getServiceOutput = exports.getService = exports.getMetadataSchemaOutput = exports.getMetadataSchema = exports.getEnvironmentOutput = exports.getEnvironment = exports.getDeploymentOutput = exports.getDeployment = exports.getApiVersionOutput = exports.getApiVersion = exports.getApiSourceOutput = exports.getApiSource = exports.getApiDefinitionOutput = exports.getApiDefinition = exports.getApiOutput = exports.getApi = exports.Environment = exports.Deployment = exports.ApiVersion = exports.ApiSource = exports.ApiDefinition = exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ApiDefinition = null;
utilities.lazyLoad(exports, ["ApiDefinition"], () => require("./apiDefinition"));
exports.ApiSource = null;
utilities.lazyLoad(exports, ["ApiSource"], () => require("./apiSource"));
exports.ApiVersion = null;
utilities.lazyLoad(exports, ["ApiVersion"], () => require("./apiVersion"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getApi = null;
exports.getApiOutput = null;
utilities.lazyLoad(exports, ["getApi", "getApiOutput"], () => require("./getApi"));
exports.getApiDefinition = null;
exports.getApiDefinitionOutput = null;
utilities.lazyLoad(exports, ["getApiDefinition", "getApiDefinitionOutput"], () => require("./getApiDefinition"));
exports.getApiSource = null;
exports.getApiSourceOutput = null;
utilities.lazyLoad(exports, ["getApiSource", "getApiSourceOutput"], () => require("./getApiSource"));
exports.getApiVersion = null;
exports.getApiVersionOutput = null;
utilities.lazyLoad(exports, ["getApiVersion", "getApiVersionOutput"], () => require("./getApiVersion"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getMetadataSchema = null;
exports.getMetadataSchemaOutput = null;
utilities.lazyLoad(exports, ["getMetadataSchema", "getMetadataSchemaOutput"], () => require("./getMetadataSchema"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.MetadataSchema = null;
utilities.lazyLoad(exports, ["MetadataSchema"], () => require("./metadataSchema"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/apicenter"), exports);
// Export sub-modules:
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240315preview = require("./v20240315preview");
exports.v20240315preview = v20240315preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apicenter:Api":
                return new exports.Api(name, undefined, { urn });
            case "azure-native:apicenter:ApiDefinition":
                return new exports.ApiDefinition(name, undefined, { urn });
            case "azure-native:apicenter:ApiSource":
                return new exports.ApiSource(name, undefined, { urn });
            case "azure-native:apicenter:ApiVersion":
                return new exports.ApiVersion(name, undefined, { urn });
            case "azure-native:apicenter:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:apicenter:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "azure-native:apicenter:MetadataSchema":
                return new exports.MetadataSchema(name, undefined, { urn });
            case "azure-native:apicenter:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:apicenter:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apicenter", _module);
