// @pulumi/azure-native/types/enums/storageactions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageTaskOperationName = exports.OnSuccess = exports.OnFailure = exports.ManagedServiceIdentityType = exports.v20230101 = void 0;
// Export sub-modules:
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.OnFailure = {
    Break: "break",
};
exports.OnSuccess = {
    Continue: "continue",
};
exports.StorageTaskOperationName = {
    SetBlobTier: "SetBlobTier",
    SetBlobTags: "SetBlobTags",
    SetBlobImmutabilityPolicy: "SetBlobImmutabilityPolicy",
    SetBlobLegalHold: "SetBlobLegalHold",
    SetBlobExpiry: "SetBlobExpiry",
    DeleteBlob: "DeleteBlob",
    UndeleteBlob: "UndeleteBlob",
};
