// @pulumi/azure-native/datadog/v20231020/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitoredSubscription = exports.Monitor = exports.listMonitorMonitoredResourcesOutput = exports.listMonitorMonitoredResources = exports.listMonitorLinkedResourcesOutput = exports.listMonitorLinkedResources = exports.listMonitorHostsOutput = exports.listMonitorHosts = exports.listMonitorApiKeysOutput = exports.listMonitorApiKeys = exports.getMonitoredSubscriptionOutput = exports.getMonitoredSubscription = exports.getMonitorDefaultKeyOutput = exports.getMonitorDefaultKey = exports.getMonitorOutput = exports.getMonitor = exports.getBillingInfoOutput = exports.getBillingInfo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getBillingInfo = null;
exports.getBillingInfoOutput = null;
utilities.lazyLoad(exports, ["getBillingInfo", "getBillingInfoOutput"], () => require("./getBillingInfo"));
exports.getMonitor = null;
exports.getMonitorOutput = null;
utilities.lazyLoad(exports, ["getMonitor", "getMonitorOutput"], () => require("./getMonitor"));
exports.getMonitorDefaultKey = null;
exports.getMonitorDefaultKeyOutput = null;
utilities.lazyLoad(exports, ["getMonitorDefaultKey", "getMonitorDefaultKeyOutput"], () => require("./getMonitorDefaultKey"));
exports.getMonitoredSubscription = null;
exports.getMonitoredSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getMonitoredSubscription", "getMonitoredSubscriptionOutput"], () => require("./getMonitoredSubscription"));
exports.listMonitorApiKeys = null;
exports.listMonitorApiKeysOutput = null;
utilities.lazyLoad(exports, ["listMonitorApiKeys", "listMonitorApiKeysOutput"], () => require("./listMonitorApiKeys"));
exports.listMonitorHosts = null;
exports.listMonitorHostsOutput = null;
utilities.lazyLoad(exports, ["listMonitorHosts", "listMonitorHostsOutput"], () => require("./listMonitorHosts"));
exports.listMonitorLinkedResources = null;
exports.listMonitorLinkedResourcesOutput = null;
utilities.lazyLoad(exports, ["listMonitorLinkedResources", "listMonitorLinkedResourcesOutput"], () => require("./listMonitorLinkedResources"));
exports.listMonitorMonitoredResources = null;
exports.listMonitorMonitoredResourcesOutput = null;
utilities.lazyLoad(exports, ["listMonitorMonitoredResources", "listMonitorMonitoredResourcesOutput"], () => require("./listMonitorMonitoredResources"));
exports.Monitor = null;
utilities.lazyLoad(exports, ["Monitor"], () => require("./monitor"));
exports.MonitoredSubscription = null;
utilities.lazyLoad(exports, ["MonitoredSubscription"], () => require("./monitoredSubscription"));
// Export enums:
__exportStar(require("../../types/enums/datadog/v20231020"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datadog/v20231020:Monitor":
                return new exports.Monitor(name, undefined, { urn });
            case "azure-native:datadog/v20231020:MonitoredSubscription":
                return new exports.MonitoredSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datadog/v20231020", _module);
