// @pulumi/azure-native/search/v20240301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedPrivateLinkResource = exports.Service = exports.PrivateEndpointConnection = exports.listQueryKeyBySearchServiceOutput = exports.listQueryKeyBySearchService = exports.listAdminKeyOutput = exports.listAdminKey = exports.getSharedPrivateLinkResourceOutput = exports.getSharedPrivateLinkResource = exports.getServiceOutput = exports.getService = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSharedPrivateLinkResource = null;
exports.getSharedPrivateLinkResourceOutput = null;
utilities.lazyLoad(exports, ["getSharedPrivateLinkResource", "getSharedPrivateLinkResourceOutput"], () => require("./getSharedPrivateLinkResource"));
exports.listAdminKey = null;
exports.listAdminKeyOutput = null;
utilities.lazyLoad(exports, ["listAdminKey", "listAdminKeyOutput"], () => require("./listAdminKey"));
exports.listQueryKeyBySearchService = null;
exports.listQueryKeyBySearchServiceOutput = null;
utilities.lazyLoad(exports, ["listQueryKeyBySearchService", "listQueryKeyBySearchServiceOutput"], () => require("./listQueryKeyBySearchService"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.SharedPrivateLinkResource = null;
utilities.lazyLoad(exports, ["SharedPrivateLinkResource"], () => require("./sharedPrivateLinkResource"));
// Export enums:
__exportStar(require("../../types/enums/search/v20240301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:search/v20240301preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:search/v20240301preview:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:search/v20240301preview:SharedPrivateLinkResource":
                return new exports.SharedPrivateLinkResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "search/v20240301preview", _module);
