// @pulumi/azure-native/securityinsights/v20190101preview/ipsyncer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPSyncer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Settings with single toggle.
 */
class IPSyncer extends pulumi.CustomResource {
    /**
     * Get an existing IPSyncer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IPSyncer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IPSyncer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IPSyncer.__pulumiType;
    }
    /**
     * Create a IPSyncer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["kind"] = "IPSyncer";
            resourceInputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["settingsName"] = args ? args.settingsName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isEnabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isEnabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20210301preview:IPSyncer" }, { type: "azure-native:securityinsights/v20210901preview:IPSyncer" }, { type: "azure-native:securityinsights/v20211001preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220101preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220401preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220501preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220601preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220701preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220801preview:IPSyncer" }, { type: "azure-native:securityinsights/v20220901preview:IPSyncer" }, { type: "azure-native:securityinsights/v20221001preview:IPSyncer" }, { type: "azure-native:securityinsights/v20221101preview:IPSyncer" }, { type: "azure-native:securityinsights/v20221201preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230201preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230301preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230401preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230501preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230601preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230701preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230801preview:IPSyncer" }, { type: "azure-native:securityinsights/v20230901preview:IPSyncer" }, { type: "azure-native:securityinsights/v20231001preview:IPSyncer" }, { type: "azure-native:securityinsights/v20231201preview:IPSyncer" }, { type: "azure-native:securityinsights/v20240101preview:IPSyncer" }, { type: "azure-native:securityinsights/v20240401preview:IPSyncer" }, { type: "azure-native:securityinsights/v20241001preview:IPSyncer" }, { type: "azure-native:securityinsights/v20250101preview:IPSyncer" }, { type: "azure-native:securityinsights:IPSyncer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IPSyncer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IPSyncer = IPSyncer;
/** @internal */
IPSyncer.__pulumiType = 'azure-native:securityinsights/v20190101preview:IPSyncer';
