// @pulumi/azure-native/types/enums/confluent/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerConnectorType = exports.Package = exports.DataFormatType = exports.ConnectorType = exports.ConnectorStatus = exports.ConnectorServiceType = exports.ConnectorClass = exports.AuthType = exports.v20240701 = exports.v20240213 = exports.v20230822 = void 0;
// Export sub-modules:
const v20230822 = require("./v20230822");
exports.v20230822 = v20230822;
const v20240213 = require("./v20240213");
exports.v20240213 = v20240213;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
exports.AuthType = {
    SERVICE_ACCOUNT: "SERVICE_ACCOUNT",
    KAFKA_API_KEY: "KAFKA_API_KEY",
};
exports.ConnectorClass = {
    AZUREBLOBSOURCE: "AZUREBLOBSOURCE",
    AZUREBLOBSINK: "AZUREBLOBSINK",
};
exports.ConnectorServiceType = {
    AzureBlobStorageSinkConnector: "AzureBlobStorageSinkConnector",
    AzureBlobStorageSourceConnector: "AzureBlobStorageSourceConnector",
    AzureCosmosDBSinkConnector: "AzureCosmosDBSinkConnector",
    AzureCosmosDBSourceConnector: "AzureCosmosDBSourceConnector",
    AzureSynapseAnalyticsSinkConnector: "AzureSynapseAnalyticsSinkConnector",
};
exports.ConnectorStatus = {
    PROVISIONING: "PROVISIONING",
    RUNNING: "RUNNING",
    PAUSED: "PAUSED",
    FAILED: "FAILED",
};
exports.ConnectorType = {
    SINK: "SINK",
    SOURCE: "SOURCE",
};
exports.DataFormatType = {
    AVRO: "AVRO",
    JSON: "JSON",
    STRING: "STRING",
    BYTES: "BYTES",
    PROTOBUF: "PROTOBUF",
};
exports.Package = {
    ESSENTIALS: "ESSENTIALS",
    ADVANCED: "ADVANCED",
};
exports.PartnerConnectorType = {
    KafkaAzureBlobStorageSource: "KafkaAzureBlobStorageSource",
    KafkaAzureBlobStorageSink: "KafkaAzureBlobStorageSink",
    KafkaAzureCosmosDBSource: "KafkaAzureCosmosDBSource",
    KafkaAzureCosmosDBSink: "KafkaAzureCosmosDBSink",
    KafkaAzureSynapseAnalyticsSink: "KafkaAzureSynapseAnalyticsSink",
};
