// @pulumi/azure-native/costmanagement/v20180801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportByResourceGroupName = exports.ReportByDepartment = exports.ReportByBillingAccount = exports.Report = exports.getReportByResourceGroupNameOutput = exports.getReportByResourceGroupName = exports.getReportByDepartmentOutput = exports.getReportByDepartment = exports.getReportByBillingAccountOutput = exports.getReportByBillingAccount = exports.getReportOutput = exports.getReport = exports.getConnectorOutput = exports.getConnector = exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getReport = null;
exports.getReportOutput = null;
utilities.lazyLoad(exports, ["getReport", "getReportOutput"], () => require("./getReport"));
exports.getReportByBillingAccount = null;
exports.getReportByBillingAccountOutput = null;
utilities.lazyLoad(exports, ["getReportByBillingAccount", "getReportByBillingAccountOutput"], () => require("./getReportByBillingAccount"));
exports.getReportByDepartment = null;
exports.getReportByDepartmentOutput = null;
utilities.lazyLoad(exports, ["getReportByDepartment", "getReportByDepartmentOutput"], () => require("./getReportByDepartment"));
exports.getReportByResourceGroupName = null;
exports.getReportByResourceGroupNameOutput = null;
utilities.lazyLoad(exports, ["getReportByResourceGroupName", "getReportByResourceGroupNameOutput"], () => require("./getReportByResourceGroupName"));
exports.Report = null;
utilities.lazyLoad(exports, ["Report"], () => require("./report"));
exports.ReportByBillingAccount = null;
utilities.lazyLoad(exports, ["ReportByBillingAccount"], () => require("./reportByBillingAccount"));
exports.ReportByDepartment = null;
utilities.lazyLoad(exports, ["ReportByDepartment"], () => require("./reportByDepartment"));
exports.ReportByResourceGroupName = null;
utilities.lazyLoad(exports, ["ReportByResourceGroupName"], () => require("./reportByResourceGroupName"));
// Export enums:
__exportStar(require("../../types/enums/costmanagement/v20180801preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:costmanagement/v20180801preview:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:costmanagement/v20180801preview:Report":
                return new exports.Report(name, undefined, { urn });
            case "azure-native:costmanagement/v20180801preview:ReportByBillingAccount":
                return new exports.ReportByBillingAccount(name, undefined, { urn });
            case "azure-native:costmanagement/v20180801preview:ReportByDepartment":
                return new exports.ReportByDepartment(name, undefined, { urn });
            case "azure-native:costmanagement/v20180801preview:ReportByResourceGroupName":
                return new exports.ReportByResourceGroupName(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "costmanagement/v20180801preview", _module);
