// @pulumi/azure-native/kusto/v20220707/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listDatabasePrincipalsOutput = exports.listDatabasePrincipals = exports.listClusterLanguageExtensionsOutput = exports.listClusterLanguageExtensions = exports.listClusterFollowerDatabasesOutput = exports.listClusterFollowerDatabases = exports.getClusterOutput = exports.getCluster = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.listClusterFollowerDatabases = null;
exports.listClusterFollowerDatabasesOutput = null;
utilities.lazyLoad(exports, ["listClusterFollowerDatabases", "listClusterFollowerDatabasesOutput"], () => require("./listClusterFollowerDatabases"));
exports.listClusterLanguageExtensions = null;
exports.listClusterLanguageExtensionsOutput = null;
utilities.lazyLoad(exports, ["listClusterLanguageExtensions", "listClusterLanguageExtensionsOutput"], () => require("./listClusterLanguageExtensions"));
exports.listDatabasePrincipals = null;
exports.listDatabasePrincipalsOutput = null;
utilities.lazyLoad(exports, ["listDatabasePrincipals", "listDatabasePrincipalsOutput"], () => require("./listDatabasePrincipals"));
// Export enums:
__exportStar(require("../../types/enums/kusto/v20220707"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kusto/v20220707:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kusto/v20220707", _module);
