// @pulumi/azure-native/recoveryservices/v20240101/replicationvCenter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationvCenter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * vCenter definition.
 */
class ReplicationvCenter extends pulumi.CustomResource {
    /**
     * Get an existing ReplicationvCenter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationvCenter(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationvCenter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationvCenter.__pulumiType;
    }
    /**
     * Create a ReplicationvCenter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["vcenterName"] = args ? args.vcenterName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20160810:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20180110:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20180710:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210210:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210301:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210401:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210601:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210701:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20210801:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20211001:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20211101:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20211201:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220101:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220201:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220301:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220401:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220501:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220801:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20220910:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20221001:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20230101:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20230201:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20230401:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20230601:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20230801:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20240201:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20240401:ReplicationvCenter" }, { type: "azure-native:recoveryservices/v20241001:ReplicationvCenter" }, { type: "azure-native:recoveryservices:ReplicationvCenter" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationvCenter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicationvCenter = ReplicationvCenter;
/** @internal */
ReplicationvCenter.__pulumiType = 'azure-native:recoveryservices/v20240101:ReplicationvCenter';
