// @pulumi/azure-native/types/enums/insights/v20240101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeAggregation = exports.Kind = exports.IdentityType = exports.DimensionOperator = exports.CriterionType = exports.ConditionOperator = void 0;
exports.ConditionOperator = {
    Equals: "Equals",
    GreaterThan: "GreaterThan",
    GreaterThanOrEqual: "GreaterThanOrEqual",
    LessThan: "LessThan",
    LessThanOrEqual: "LessThanOrEqual",
    GreaterOrLessThan: "GreaterOrLessThan",
};
exports.CriterionType = {
    StaticThresholdCriterion: "StaticThresholdCriterion",
    DynamicThresholdCriterion: "DynamicThresholdCriterion",
};
exports.DimensionOperator = {
    Include: "Include",
    Exclude: "Exclude",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.Kind = {
    LogAlert: "LogAlert",
    EventLogAlert: "EventLogAlert",
    LogToMetric: "LogToMetric",
};
exports.TimeAggregation = {
    Count: "Count",
    Average: "Average",
    Minimum: "Minimum",
    Maximum: "Maximum",
    Total: "Total",
};
