// @pulumi/azure-native/fabric/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250115preview = exports.v20231101 = exports.getFabricCapacityOutput = exports.getFabricCapacity = exports.FabricCapacity = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.FabricCapacity = null;
utilities.lazyLoad(exports, ["FabricCapacity"], () => require("./fabricCapacity"));
exports.getFabricCapacity = null;
exports.getFabricCapacityOutput = null;
utilities.lazyLoad(exports, ["getFabricCapacity", "getFabricCapacityOutput"], () => require("./getFabricCapacity"));
// Export enums:
__exportStar(require("../types/enums/fabric"), exports);
// Export sub-modules:
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20250115preview = require("./v20250115preview");
exports.v20250115preview = v20250115preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:fabric:FabricCapacity":
                return new exports.FabricCapacity(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "fabric", _module);
