// @pulumi/azure-native/types/enums/azureactivedirectory/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CIAMResourceSKUTier = exports.CIAMResourceSKUName = exports.B2CResourceSKUTier = exports.B2CResourceSKUName = exports.v20230517preview = exports.v20230118preview = exports.v20210401 = exports.v20190101preview = void 0;
// Export sub-modules:
const v20190101preview = require("./v20190101preview");
exports.v20190101preview = v20190101preview;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20230118preview = require("./v20230118preview");
exports.v20230118preview = v20230118preview;
const v20230517preview = require("./v20230517preview");
exports.v20230517preview = v20230517preview;
exports.B2CResourceSKUName = {
    /**
     * Azure AD B2C usage is billed to a linked Azure subscription and uses a monthly active users (MAU) billing model.
     */
    Standard: "Standard",
    /**
     * Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing.
     */
    PremiumP1: "PremiumP1",
    /**
     * Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing.
     */
    PremiumP2: "PremiumP2",
};
exports.B2CResourceSKUTier = {
    /**
     * The SKU tier used for all Azure AD B2C tenants.
     */
    A0: "A0",
};
exports.CIAMResourceSKUName = {
    Standard: "Standard",
    PremiumP1: "PremiumP1",
    PremiumP2: "PremiumP2",
};
exports.CIAMResourceSKUTier = {
    /**
     * The SKU tier used for all Azure AD for customers tenants.
     */
    A0: "A0",
};
