// @pulumi/azure-native/cache/patchSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Response to put/get patch schedules for Redis cache.
 *
 * Uses Azure REST API version 2023-04-01. In version 1.x of the Azure Native provider, it used API version 2020-06-01.
 *
 * Other available API versions: 2023-05-01-preview, 2023-08-01, 2024-03-01, 2024-04-01-preview, 2024-11-01.
 */
class PatchSchedule extends pulumi.CustomResource {
    /**
     * Get an existing PatchSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PatchSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PatchSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PatchSchedule.__pulumiType;
    }
    /**
     * Create a PatchSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scheduleEntries === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleEntries'");
            }
            resourceInputs["default"] = args ? args.default : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scheduleEntries"] = args ? args.scheduleEntries : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scheduleEntries"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cache/v20171001:PatchSchedule" }, { type: "azure-native:cache/v20180301:PatchSchedule" }, { type: "azure-native:cache/v20190701:PatchSchedule" }, { type: "azure-native:cache/v20200601:PatchSchedule" }, { type: "azure-native:cache/v20201201:PatchSchedule" }, { type: "azure-native:cache/v20210601:PatchSchedule" }, { type: "azure-native:cache/v20220501:PatchSchedule" }, { type: "azure-native:cache/v20220601:PatchSchedule" }, { type: "azure-native:cache/v20230401:PatchSchedule" }, { type: "azure-native:cache/v20230501preview:PatchSchedule" }, { type: "azure-native:cache/v20230801:PatchSchedule" }, { type: "azure-native:cache/v20240301:PatchSchedule" }, { type: "azure-native:cache/v20240401preview:PatchSchedule" }, { type: "azure-native:cache/v20241101:PatchSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PatchSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PatchSchedule = PatchSchedule;
/** @internal */
PatchSchedule.__pulumiType = 'azure-native:cache:PatchSchedule';
