// @pulumi/azure-native/types/enums/redhatopenshift/v20240812preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Visibility = exports.ProvisioningState = exports.PreconfiguredNSG = exports.OutboundType = exports.ManagedServiceIdentityType = exports.FipsValidatedModules = exports.EncryptionAtHost = void 0;
exports.EncryptionAtHost = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.FipsValidatedModules = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.OutboundType = {
    Loadbalancer: "Loadbalancer",
    UserDefinedRouting: "UserDefinedRouting",
};
exports.PreconfiguredNSG = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ProvisioningState = {
    AdminUpdating: "AdminUpdating",
    Canceled: "Canceled",
    Creating: "Creating",
    Deleting: "Deleting",
    Failed: "Failed",
    Succeeded: "Succeeded",
    Updating: "Updating",
};
exports.Visibility = {
    Private: "Private",
    Public: "Public",
};
