// @pulumi/azure-native/desktopvirtualization/v20241101preview/scalingPlanPooledSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScalingPlanPooledSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents a ScalingPlanPooledSchedule definition.
 */
class ScalingPlanPooledSchedule extends pulumi.CustomResource {
    /**
     * Get an existing ScalingPlanPooledSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScalingPlanPooledSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScalingPlanPooledSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScalingPlanPooledSchedule.__pulumiType;
    }
    /**
     * Create a ScalingPlanPooledSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.daysOfWeek === undefined) && !opts.urn) {
                throw new Error("Missing required property 'daysOfWeek'");
            }
            if ((!args || args.offPeakStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'offPeakStartTime'");
            }
            if ((!args || args.peakStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peakStartTime'");
            }
            if ((!args || args.rampDownCapacityThresholdPct === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampDownCapacityThresholdPct'");
            }
            if ((!args || args.rampDownStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampDownStartTime'");
            }
            if ((!args || args.rampUpCapacityThresholdPct === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampUpCapacityThresholdPct'");
            }
            if ((!args || args.rampUpStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rampUpStartTime'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scalingPlanName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scalingPlanName'");
            }
            resourceInputs["createDelete"] = args ? args.createDelete : undefined;
            resourceInputs["daysOfWeek"] = args ? args.daysOfWeek : undefined;
            resourceInputs["offPeakLoadBalancingAlgorithm"] = args ? args.offPeakLoadBalancingAlgorithm : undefined;
            resourceInputs["offPeakStartTime"] = args ? args.offPeakStartTime : undefined;
            resourceInputs["peakLoadBalancingAlgorithm"] = args ? args.peakLoadBalancingAlgorithm : undefined;
            resourceInputs["peakStartTime"] = args ? args.peakStartTime : undefined;
            resourceInputs["rampDownCapacityThresholdPct"] = args ? args.rampDownCapacityThresholdPct : undefined;
            resourceInputs["rampDownForceLogoffUsers"] = args ? args.rampDownForceLogoffUsers : undefined;
            resourceInputs["rampDownLoadBalancingAlgorithm"] = args ? args.rampDownLoadBalancingAlgorithm : undefined;
            resourceInputs["rampDownMinimumHostsPct"] = args ? args.rampDownMinimumHostsPct : undefined;
            resourceInputs["rampDownNotificationMessage"] = args ? args.rampDownNotificationMessage : undefined;
            resourceInputs["rampDownStartTime"] = args ? args.rampDownStartTime : undefined;
            resourceInputs["rampDownStopHostsWhen"] = args ? args.rampDownStopHostsWhen : undefined;
            resourceInputs["rampDownWaitTimeMinutes"] = args ? args.rampDownWaitTimeMinutes : undefined;
            resourceInputs["rampUpCapacityThresholdPct"] = args ? args.rampUpCapacityThresholdPct : undefined;
            resourceInputs["rampUpLoadBalancingAlgorithm"] = args ? args.rampUpLoadBalancingAlgorithm : undefined;
            resourceInputs["rampUpMinimumHostsPct"] = args ? args.rampUpMinimumHostsPct : undefined;
            resourceInputs["rampUpStartTime"] = args ? args.rampUpStartTime : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scalingMethod"] = args ? args.scalingMethod : undefined;
            resourceInputs["scalingPlanName"] = args ? args.scalingPlanName : undefined;
            resourceInputs["scalingPlanScheduleName"] = args ? args.scalingPlanScheduleName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createDelete"] = undefined /*out*/;
            resourceInputs["daysOfWeek"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offPeakLoadBalancingAlgorithm"] = undefined /*out*/;
            resourceInputs["offPeakStartTime"] = undefined /*out*/;
            resourceInputs["peakLoadBalancingAlgorithm"] = undefined /*out*/;
            resourceInputs["peakStartTime"] = undefined /*out*/;
            resourceInputs["rampDownCapacityThresholdPct"] = undefined /*out*/;
            resourceInputs["rampDownForceLogoffUsers"] = undefined /*out*/;
            resourceInputs["rampDownLoadBalancingAlgorithm"] = undefined /*out*/;
            resourceInputs["rampDownMinimumHostsPct"] = undefined /*out*/;
            resourceInputs["rampDownNotificationMessage"] = undefined /*out*/;
            resourceInputs["rampDownStartTime"] = undefined /*out*/;
            resourceInputs["rampDownStopHostsWhen"] = undefined /*out*/;
            resourceInputs["rampDownWaitTimeMinutes"] = undefined /*out*/;
            resourceInputs["rampUpCapacityThresholdPct"] = undefined /*out*/;
            resourceInputs["rampUpLoadBalancingAlgorithm"] = undefined /*out*/;
            resourceInputs["rampUpMinimumHostsPct"] = undefined /*out*/;
            resourceInputs["rampUpStartTime"] = undefined /*out*/;
            resourceInputs["scalingMethod"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:desktopvirtualization/v20220401preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20220909:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20221014preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20230905:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20231004preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20231101preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20240116preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20240306preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20240403:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20240408preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization/v20240808preview:ScalingPlanPooledSchedule" }, { type: "azure-native:desktopvirtualization:ScalingPlanPooledSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScalingPlanPooledSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ScalingPlanPooledSchedule = ScalingPlanPooledSchedule;
/** @internal */
ScalingPlanPooledSchedule.__pulumiType = 'azure-native:desktopvirtualization/v20241101preview:ScalingPlanPooledSchedule';
