// @pulumi/azure-native/types/enums/documentdb/v20240701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.PreviewFeature = exports.HighAvailabilityMode = exports.CreateMode = void 0;
exports.CreateMode = {
    /**
     * Create a new mongo cluster.
     */
    Default: "Default",
    /**
     * Create a mongo cluster from a restore point-in-time.
     */
    PointInTimeRestore: "PointInTimeRestore",
    /**
     * Create a replica cluster in distinct geographic region from the source cluster.
     */
    GeoReplica: "GeoReplica",
    /**
     * Create a replica cluster in the same geographic region as the source cluster.
     */
    Replica: "Replica",
};
exports.HighAvailabilityMode = {
    /**
     * High availability mode is disabled. This mode is can see availability impact during faults or maintenance and is not recommended for production.
     */
    Disabled: "Disabled",
    /**
     * High availability mode is enabled, where each server in a shard is placed in the same availability zone.
     */
    SameZone: "SameZone",
    /**
     * High availability mode is enabled and preferences ZoneRedundant if availability zones capacity is available in the region, otherwise falls-back to provisioning with SameZone.
     */
    ZoneRedundantPreferred: "ZoneRedundantPreferred",
};
exports.PreviewFeature = {
    /**
     * Enables geo replicas preview feature. The feature must be set at create-time on new cluster to enable linking a geo-replica cluster to it.
     */
    GeoReplicas: "GeoReplicas",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    /**
     * If set, mongo cluster can be accessed through private and public methods.
     */
    Enabled: "Enabled",
    /**
     * If set, the private endpoints are the exclusive access method.
     */
    Disabled: "Disabled",
};
