// @pulumi/azure-native/types/enums/netapp/v20240901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.SmbNonBrowsable = exports.SmbAccessBasedEnumeration = exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.NetworkFeatures = exports.ManagedServiceIdentityType = exports.KeySource = exports.EndpointType = exports.EncryptionType = exports.EncryptionKeySource = exports.EnableSubvolumes = exports.CoolAccessTieringPolicy = exports.CoolAccessRetrievalPolicy = exports.ChownMode = exports.AvsDataStore = exports.ApplicationType = void 0;
exports.ApplicationType = {
    SAP_HANA: "SAP-HANA",
    ORACLE: "ORACLE",
};
exports.AvsDataStore = {
    /**
     * avsDataStore is enabled
     */
    Enabled: "Enabled",
    /**
     * avsDataStore is disabled
     */
    Disabled: "Disabled",
};
exports.ChownMode = {
    Restricted: "Restricted",
    Unrestricted: "Unrestricted",
};
exports.CoolAccessRetrievalPolicy = {
    Default: "Default",
    OnRead: "OnRead",
    Never: "Never",
};
exports.CoolAccessTieringPolicy = {
    Auto: "Auto",
    SnapshotOnly: "SnapshotOnly",
};
exports.EnableSubvolumes = {
    /**
     * subvolumes are enabled
     */
    Enabled: "Enabled",
    /**
     * subvolumes are not enabled
     */
    Disabled: "Disabled",
};
exports.EncryptionKeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.EncryptionType = {
    /**
     * EncryptionType Single, volumes will use single encryption at rest
     */
    Single: "Single",
    /**
     * EncryptionType Double, volumes will use double encryption at rest
     */
    Double: "Double",
};
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.KeySource = {
    /**
     * Microsoft-managed key encryption
     */
    Microsoft_NetApp: "Microsoft.NetApp",
    /**
     * Customer-managed key encryption
     */
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NetworkFeatures = {
    /**
     * Basic network features.
     */
    Basic: "Basic",
    /**
     * Standard network features.
     */
    Standard: "Standard",
    /**
     * Updating from Basic to Standard network features.
     */
    Basic_Standard: "Basic_Standard",
    /**
     * Updating from Standard to Basic network features.
     */
    Standard_Basic: "Standard_Basic",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
    /**
     * Zone redundant storage service level
     */
    StandardZRS: "StandardZRS",
};
exports.SmbAccessBasedEnumeration = {
    /**
     * smbAccessBasedEnumeration share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbAccessBasedEnumeration share setting is enabled
     */
    Enabled: "Enabled",
};
exports.SmbNonBrowsable = {
    /**
     * smbNonBrowsable share setting is disabled
     */
    Disabled: "Disabled",
    /**
     * smbNonBrowsable share setting is enabled
     */
    Enabled: "Enabled",
};
exports.Type = {
    /**
     * Default user quota
     */
    DefaultUserQuota: "DefaultUserQuota",
    /**
     * Default group quota
     */
    DefaultGroupQuota: "DefaultGroupQuota",
    /**
     * Individual user quota
     */
    IndividualUserQuota: "IndividualUserQuota",
    /**
     * Individual group quota
     */
    IndividualGroupQuota: "IndividualGroupQuota",
};
