// @pulumi/azure-native/types/enums/recoveryservices/v20240930preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardTierStorageRedundancy = exports.SoftDeleteState = exports.SkuName = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.InfrastructureEncryptionState = exports.ImmutabilityState = exports.EnhancedSecurityState = exports.CrossSubscriptionRestoreState = exports.CrossRegionRestore = exports.AlertsState = void 0;
exports.AlertsState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CrossRegionRestore = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CrossSubscriptionRestoreState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    PermanentlyDisabled: "PermanentlyDisabled",
};
exports.EnhancedSecurityState = {
    Invalid: "Invalid",
    Enabled: "Enabled",
    Disabled: "Disabled",
    AlwaysON: "AlwaysON",
};
exports.ImmutabilityState = {
    Disabled: "Disabled",
    Unlocked: "Unlocked",
    Locked: "Locked",
};
exports.InfrastructureEncryptionState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.SkuName = {
    Standard: "Standard",
    RS0: "RS0",
};
exports.SoftDeleteState = {
    Invalid: "Invalid",
    Enabled: "Enabled",
    Disabled: "Disabled",
    AlwaysON: "AlwaysON",
};
exports.StandardTierStorageRedundancy = {
    Invalid: "Invalid",
    LocallyRedundant: "LocallyRedundant",
    GeoRedundant: "GeoRedundant",
    ZoneRedundant: "ZoneRedundant",
};
