// @pulumi/azure-native/types/enums/sql/v20210201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountType = exports.PrincipalType = exports.ManagedServerCreateMode = exports.ManagedInstanceProxyOverride = exports.ManagedInstanceLicenseType = exports.IdentityType = exports.AdministratorType = void 0;
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ManagedInstanceLicenseType = {
    LicenseIncluded: "LicenseIncluded",
    BasePrice: "BasePrice",
};
exports.ManagedInstanceProxyOverride = {
    Proxy: "Proxy",
    Redirect: "Redirect",
    Default: "Default",
};
exports.ManagedServerCreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
};
exports.PrincipalType = {
    User: "User",
    Group: "Group",
    Application: "Application",
};
exports.StorageAccountType = {
    GRS: "GRS",
    LRS: "LRS",
    ZRS: "ZRS",
};
