// @pulumi/azure-native/types/enums/signalrservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpstreamAuthType = exports.SignalRSkuTier = exports.SignalRRequestType = exports.ServiceKind = exports.PrivateLinkServiceConnectionStatus = exports.ManagedIdentityType = exports.FeatureFlags = exports.ACLAction = exports.v20241001preview = exports.v20240801preview = exports.v20240401preview = exports.v20240301 = exports.v20240101preview = exports.v20230801preview = exports.v20230601preview = exports.v20230301preview = exports.v20230201 = void 0;
// Export sub-modules:
const v20230201 = require("./v20230201");
exports.v20230201 = v20230201;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
exports.ACLAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FeatureFlags = {
    ServiceMode: "ServiceMode",
    EnableConnectivityLogs: "EnableConnectivityLogs",
    EnableMessagingLogs: "EnableMessagingLogs",
    EnableLiveTrace: "EnableLiveTrace",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.ServiceKind = {
    SignalR: "SignalR",
    RawWebSockets: "RawWebSockets",
};
exports.SignalRRequestType = {
    ClientConnection: "ClientConnection",
    ServerConnection: "ServerConnection",
    RESTAPI: "RESTAPI",
    Trace: "Trace",
};
exports.SignalRSkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.UpstreamAuthType = {
    None: "None",
    ManagedIdentity: "ManagedIdentity",
};
