// @pulumi/azure-native/networkfunction/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20221101 = exports.v20220801 = exports.v20220501 = exports.getCollectorPolicyOutput = exports.getCollectorPolicy = exports.getAzureTrafficCollectorOutput = exports.getAzureTrafficCollector = exports.CollectorPolicy = exports.AzureTrafficCollector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureTrafficCollector = null;
utilities.lazyLoad(exports, ["AzureTrafficCollector"], () => require("./azureTrafficCollector"));
exports.CollectorPolicy = null;
utilities.lazyLoad(exports, ["CollectorPolicy"], () => require("./collectorPolicy"));
exports.getAzureTrafficCollector = null;
exports.getAzureTrafficCollectorOutput = null;
utilities.lazyLoad(exports, ["getAzureTrafficCollector", "getAzureTrafficCollectorOutput"], () => require("./getAzureTrafficCollector"));
exports.getCollectorPolicy = null;
exports.getCollectorPolicyOutput = null;
utilities.lazyLoad(exports, ["getCollectorPolicy", "getCollectorPolicyOutput"], () => require("./getCollectorPolicy"));
// Export enums:
__exportStar(require("../types/enums/networkfunction"), exports);
// Export sub-modules:
const v20220501 = require("./v20220501");
exports.v20220501 = v20220501;
const v20220801 = require("./v20220801");
exports.v20220801 = v20220801;
const v20221101 = require("./v20221101");
exports.v20221101 = v20221101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:networkfunction:AzureTrafficCollector":
                return new exports.AzureTrafficCollector(name, undefined, { urn });
            case "azure-native:networkfunction:CollectorPolicy":
                return new exports.CollectorPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "networkfunction", _module);
