// @pulumi/azure-native/sql/instanceFailoverGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceFailoverGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An instance failover group.
 *
 * Uses Azure REST API version 2021-11-01. In version 1.x of the Azure Native provider, it used API version 2020-11-01-preview.
 *
 * Other available API versions: 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01, 2023-08-01-preview, 2024-05-01-preview.
 */
class InstanceFailoverGroup extends pulumi.CustomResource {
    /**
     * Get an existing InstanceFailoverGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InstanceFailoverGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InstanceFailoverGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InstanceFailoverGroup.__pulumiType;
    }
    /**
     * Create a InstanceFailoverGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.locationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'locationName'");
            }
            if ((!args || args.managedInstancePairs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managedInstancePairs'");
            }
            if ((!args || args.partnerRegions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'partnerRegions'");
            }
            if ((!args || args.readWriteEndpoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'readWriteEndpoint'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["failoverGroupName"] = args ? args.failoverGroupName : undefined;
            resourceInputs["locationName"] = args ? args.locationName : undefined;
            resourceInputs["managedInstancePairs"] = args ? args.managedInstancePairs : undefined;
            resourceInputs["partnerRegions"] = args ? args.partnerRegions : undefined;
            resourceInputs["readOnlyEndpoint"] = args ? args.readOnlyEndpoint : undefined;
            resourceInputs["readWriteEndpoint"] = args ? args.readWriteEndpoint : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["replicationRole"] = undefined /*out*/;
            resourceInputs["replicationState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["managedInstancePairs"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partnerRegions"] = undefined /*out*/;
            resourceInputs["readOnlyEndpoint"] = undefined /*out*/;
            resourceInputs["readWriteEndpoint"] = undefined /*out*/;
            resourceInputs["replicationRole"] = undefined /*out*/;
            resourceInputs["replicationState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20171001preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20200202preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20200801preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20201101preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20210201preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20210501preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20210801preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20211101:InstanceFailoverGroup" }, { type: "azure-native:sql/v20211101preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20220201preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20220501preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20220801preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20221101preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20230201preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20230501preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20230801:InstanceFailoverGroup" }, { type: "azure-native:sql/v20230801preview:InstanceFailoverGroup" }, { type: "azure-native:sql/v20240501preview:InstanceFailoverGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InstanceFailoverGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InstanceFailoverGroup = InstanceFailoverGroup;
/** @internal */
InstanceFailoverGroup.__pulumiType = 'azure-native:sql:InstanceFailoverGroup';
