// @pulumi/azure-native/sql/ipv6FirewallRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPv6FirewallRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An IPv6 server firewall rule.
 *
 * Uses Azure REST API version 2021-11-01. In version 1.x of the Azure Native provider, it used API version 2021-08-01-preview.
 *
 * Other available API versions: 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01, 2023-08-01-preview, 2024-05-01-preview.
 */
class IPv6FirewallRule extends pulumi.CustomResource {
    /**
     * Get an existing IPv6FirewallRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IPv6FirewallRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IPv6FirewallRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IPv6FirewallRule.__pulumiType;
    }
    /**
     * Create a IPv6FirewallRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            resourceInputs["endIPv6Address"] = args ? args.endIPv6Address : undefined;
            resourceInputs["firewallRuleName"] = args ? args.firewallRuleName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["startIPv6Address"] = args ? args.startIPv6Address : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["endIPv6Address"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["startIPv6Address"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20210801preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20211101:IPv6FirewallRule" }, { type: "azure-native:sql/v20211101preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20220201preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20220501preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20220801preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20221101preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20230201preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20230501preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20230801:IPv6FirewallRule" }, { type: "azure-native:sql/v20230801preview:IPv6FirewallRule" }, { type: "azure-native:sql/v20240501preview:IPv6FirewallRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IPv6FirewallRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IPv6FirewallRule = IPv6FirewallRule;
/** @internal */
IPv6FirewallRule.__pulumiType = 'azure-native:sql:IPv6FirewallRule';
