// @pulumi/azure-native/sql/workloadGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Workload group operations for a data warehouse
 *
 * Uses Azure REST API version 2021-11-01. In version 1.x of the Azure Native provider, it used API version 2020-11-01-preview.
 *
 * Other available API versions: 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01, 2023-08-01-preview, 2024-05-01-preview.
 */
class WorkloadGroup extends pulumi.CustomResource {
    /**
     * Get an existing WorkloadGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkloadGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkloadGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkloadGroup.__pulumiType;
    }
    /**
     * Create a WorkloadGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.maxResourcePercent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'maxResourcePercent'");
            }
            if ((!args || args.minResourcePercent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'minResourcePercent'");
            }
            if ((!args || args.minResourcePercentPerRequest === undefined) && !opts.urn) {
                throw new Error("Missing required property 'minResourcePercentPerRequest'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["importance"] = args ? args.importance : undefined;
            resourceInputs["maxResourcePercent"] = args ? args.maxResourcePercent : undefined;
            resourceInputs["maxResourcePercentPerRequest"] = args ? args.maxResourcePercentPerRequest : undefined;
            resourceInputs["minResourcePercent"] = args ? args.minResourcePercent : undefined;
            resourceInputs["minResourcePercentPerRequest"] = args ? args.minResourcePercentPerRequest : undefined;
            resourceInputs["queryExecutionTimeout"] = args ? args.queryExecutionTimeout : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["workloadGroupName"] = args ? args.workloadGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["importance"] = undefined /*out*/;
            resourceInputs["maxResourcePercent"] = undefined /*out*/;
            resourceInputs["maxResourcePercentPerRequest"] = undefined /*out*/;
            resourceInputs["minResourcePercent"] = undefined /*out*/;
            resourceInputs["minResourcePercentPerRequest"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["queryExecutionTimeout"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20190601preview:WorkloadGroup" }, { type: "azure-native:sql/v20200202preview:WorkloadGroup" }, { type: "azure-native:sql/v20200801preview:WorkloadGroup" }, { type: "azure-native:sql/v20201101preview:WorkloadGroup" }, { type: "azure-native:sql/v20210201preview:WorkloadGroup" }, { type: "azure-native:sql/v20210501preview:WorkloadGroup" }, { type: "azure-native:sql/v20210801preview:WorkloadGroup" }, { type: "azure-native:sql/v20211101:WorkloadGroup" }, { type: "azure-native:sql/v20211101preview:WorkloadGroup" }, { type: "azure-native:sql/v20220201preview:WorkloadGroup" }, { type: "azure-native:sql/v20220501preview:WorkloadGroup" }, { type: "azure-native:sql/v20220801preview:WorkloadGroup" }, { type: "azure-native:sql/v20221101preview:WorkloadGroup" }, { type: "azure-native:sql/v20230201preview:WorkloadGroup" }, { type: "azure-native:sql/v20230501preview:WorkloadGroup" }, { type: "azure-native:sql/v20230801:WorkloadGroup" }, { type: "azure-native:sql/v20230801preview:WorkloadGroup" }, { type: "azure-native:sql/v20240501preview:WorkloadGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkloadGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkloadGroup = WorkloadGroup;
/** @internal */
WorkloadGroup.__pulumiType = 'azure-native:sql:WorkloadGroup';
