// @pulumi/azure-native/apimanagement/v20180601preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagDescription = exports.ProductPolicy = exports.ProductApi = exports.Policy = exports.getTagDescriptionOutput = exports.getTagDescription = exports.getProductPolicyOutput = exports.getProductPolicy = exports.getPolicyOutput = exports.getPolicy = exports.getApiPolicyOutput = exports.getApiPolicy = exports.getApiOperationPolicyOutput = exports.getApiOperationPolicy = exports.getApiOutput = exports.getApi = exports.ApiPolicy = exports.ApiOperationPolicy = exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ApiOperationPolicy = null;
utilities.lazyLoad(exports, ["ApiOperationPolicy"], () => require("./apiOperationPolicy"));
exports.ApiPolicy = null;
utilities.lazyLoad(exports, ["ApiPolicy"], () => require("./apiPolicy"));
exports.getApi = null;
exports.getApiOutput = null;
utilities.lazyLoad(exports, ["getApi", "getApiOutput"], () => require("./getApi"));
exports.getApiOperationPolicy = null;
exports.getApiOperationPolicyOutput = null;
utilities.lazyLoad(exports, ["getApiOperationPolicy", "getApiOperationPolicyOutput"], () => require("./getApiOperationPolicy"));
exports.getApiPolicy = null;
exports.getApiPolicyOutput = null;
utilities.lazyLoad(exports, ["getApiPolicy", "getApiPolicyOutput"], () => require("./getApiPolicy"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getProductPolicy = null;
exports.getProductPolicyOutput = null;
utilities.lazyLoad(exports, ["getProductPolicy", "getProductPolicyOutput"], () => require("./getProductPolicy"));
exports.getTagDescription = null;
exports.getTagDescriptionOutput = null;
utilities.lazyLoad(exports, ["getTagDescription", "getTagDescriptionOutput"], () => require("./getTagDescription"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.ProductApi = null;
utilities.lazyLoad(exports, ["ProductApi"], () => require("./productApi"));
exports.ProductPolicy = null;
utilities.lazyLoad(exports, ["ProductPolicy"], () => require("./productPolicy"));
exports.TagDescription = null;
utilities.lazyLoad(exports, ["TagDescription"], () => require("./tagDescription"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20180601preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20180601preview:Api":
                return new exports.Api(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:ApiOperationPolicy":
                return new exports.ApiOperationPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:ApiPolicy":
                return new exports.ApiPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:ProductApi":
                return new exports.ProductApi(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:ProductPolicy":
                return new exports.ProductPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20180601preview:TagDescription":
                return new exports.TagDescription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20180601preview", _module);
