// @pulumi/azure-native/connectedcache/v20230501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IspCustomer = exports.IspCacheNodesOperation = exports.getIspCustomerOutput = exports.getIspCustomer = exports.getIspCacheNodesOperationCacheNodeInstallDetailsOutput = exports.getIspCacheNodesOperationCacheNodeInstallDetails = exports.getIspCacheNodesOperationBgpCidrsOutput = exports.getIspCacheNodesOperationBgpCidrs = exports.getIspCacheNodesOperationOutput = exports.getIspCacheNodesOperation = exports.getEnterpriseMccCustomerOutput = exports.getEnterpriseMccCustomer = exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput = exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails = exports.getEnterpriseMccCacheNodesOperationOutput = exports.getEnterpriseMccCacheNodesOperation = exports.getEnterpriseCustomerOperationOutput = exports.getEnterpriseCustomerOperation = exports.getCacheNodesOperationOutput = exports.getCacheNodesOperation = exports.EnterpriseMccCustomer = exports.EnterpriseMccCacheNodesOperation = exports.EnterpriseCustomerOperation = exports.CacheNodesOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CacheNodesOperation = null;
utilities.lazyLoad(exports, ["CacheNodesOperation"], () => require("./cacheNodesOperation"));
exports.EnterpriseCustomerOperation = null;
utilities.lazyLoad(exports, ["EnterpriseCustomerOperation"], () => require("./enterpriseCustomerOperation"));
exports.EnterpriseMccCacheNodesOperation = null;
utilities.lazyLoad(exports, ["EnterpriseMccCacheNodesOperation"], () => require("./enterpriseMccCacheNodesOperation"));
exports.EnterpriseMccCustomer = null;
utilities.lazyLoad(exports, ["EnterpriseMccCustomer"], () => require("./enterpriseMccCustomer"));
exports.getCacheNodesOperation = null;
exports.getCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getCacheNodesOperation", "getCacheNodesOperationOutput"], () => require("./getCacheNodesOperation"));
exports.getEnterpriseCustomerOperation = null;
exports.getEnterpriseCustomerOperationOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseCustomerOperation", "getEnterpriseCustomerOperationOutput"], () => require("./getEnterpriseCustomerOperation"));
exports.getEnterpriseMccCacheNodesOperation = null;
exports.getEnterpriseMccCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperation", "getEnterpriseMccCacheNodesOperationOutput"], () => require("./getEnterpriseMccCacheNodesOperation"));
exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails = null;
exports.getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails", "getEnterpriseMccCacheNodesOperationCacheNodeInstallDetailsOutput"], () => require("./getEnterpriseMccCacheNodesOperationCacheNodeInstallDetails"));
exports.getEnterpriseMccCustomer = null;
exports.getEnterpriseMccCustomerOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseMccCustomer", "getEnterpriseMccCustomerOutput"], () => require("./getEnterpriseMccCustomer"));
exports.getIspCacheNodesOperation = null;
exports.getIspCacheNodesOperationOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperation", "getIspCacheNodesOperationOutput"], () => require("./getIspCacheNodesOperation"));
exports.getIspCacheNodesOperationBgpCidrs = null;
exports.getIspCacheNodesOperationBgpCidrsOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationBgpCidrs", "getIspCacheNodesOperationBgpCidrsOutput"], () => require("./getIspCacheNodesOperationBgpCidrs"));
exports.getIspCacheNodesOperationCacheNodeInstallDetails = null;
exports.getIspCacheNodesOperationCacheNodeInstallDetailsOutput = null;
utilities.lazyLoad(exports, ["getIspCacheNodesOperationCacheNodeInstallDetails", "getIspCacheNodesOperationCacheNodeInstallDetailsOutput"], () => require("./getIspCacheNodesOperationCacheNodeInstallDetails"));
exports.getIspCustomer = null;
exports.getIspCustomerOutput = null;
utilities.lazyLoad(exports, ["getIspCustomer", "getIspCustomerOutput"], () => require("./getIspCustomer"));
exports.IspCacheNodesOperation = null;
utilities.lazyLoad(exports, ["IspCacheNodesOperation"], () => require("./ispCacheNodesOperation"));
exports.IspCustomer = null;
utilities.lazyLoad(exports, ["IspCustomer"], () => require("./ispCustomer"));
// Export enums:
__exportStar(require("../../types/enums/connectedcache/v20230501preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:connectedcache/v20230501preview:CacheNodesOperation":
                return new exports.CacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache/v20230501preview:EnterpriseCustomerOperation":
                return new exports.EnterpriseCustomerOperation(name, undefined, { urn });
            case "azure-native:connectedcache/v20230501preview:EnterpriseMccCacheNodesOperation":
                return new exports.EnterpriseMccCacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache/v20230501preview:EnterpriseMccCustomer":
                return new exports.EnterpriseMccCustomer(name, undefined, { urn });
            case "azure-native:connectedcache/v20230501preview:IspCacheNodesOperation":
                return new exports.IspCacheNodesOperation(name, undefined, { urn });
            case "azure-native:connectedcache/v20230501preview:IspCustomer":
                return new exports.IspCustomer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "connectedcache/v20230501preview", _module);
