// @pulumi/azure-native/mysqldiscovery/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240930preview = exports.MySQLSite = exports.MySQLServer = exports.getMySQLSiteOutput = exports.getMySQLSite = exports.getMySQLServerOutput = exports.getMySQLServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getMySQLServer = null;
exports.getMySQLServerOutput = null;
utilities.lazyLoad(exports, ["getMySQLServer", "getMySQLServerOutput"], () => require("./getMySQLServer"));
exports.getMySQLSite = null;
exports.getMySQLSiteOutput = null;
utilities.lazyLoad(exports, ["getMySQLSite", "getMySQLSiteOutput"], () => require("./getMySQLSite"));
exports.MySQLServer = null;
utilities.lazyLoad(exports, ["MySQLServer"], () => require("./mySQLServer"));
exports.MySQLSite = null;
utilities.lazyLoad(exports, ["MySQLSite"], () => require("./mySQLSite"));
// Export enums:
__exportStar(require("../types/enums/mysqldiscovery"), exports);
// Export sub-modules:
const v20240930preview = require("./v20240930preview");
exports.v20240930preview = v20240930preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mysqldiscovery:MySQLServer":
                return new exports.MySQLServer(name, undefined, { urn });
            case "azure-native:mysqldiscovery:MySQLSite":
                return new exports.MySQLSite(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mysqldiscovery", _module);
