// @pulumi/azure-native/machinelearning/v20191001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.listWorkspaceKeysOutput = exports.listWorkspaceKeys = exports.getWorkspaceOutput = exports.getWorkspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.listWorkspaceKeys = null;
exports.listWorkspaceKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceKeys", "listWorkspaceKeysOutput"], () => require("./listWorkspaceKeys"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearning/v20191001:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearning/v20191001", _module);
