// @pulumi/azure-native/types/enums/manufacturingplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.ResourceState = exports.RedundancyState = exports.ManagedServiceIdentityType = exports.v20250301 = void 0;
// Export sub-modules:
const v20250301 = require("./v20250301");
exports.v20250301 = v20250301;
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.RedundancyState = {
    /**
     * Resource is zone redundant
     */
    Zonal: "Zonal",
    /**
     * Resource is not redundant
     */
    None: "None",
};
exports.ResourceState = {
    /**
     * Resource is in active state
     */
    Active: "Active",
    /**
     * Resource is in inactive state
     */
    Inactive: "Inactive",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
