// @pulumi/azure-native/types/enums/mobilenetwork/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficControlPermission = exports.SdfDirection = exports.PreemptionVulnerability = exports.PreemptionCapability = exports.PlatformType = exports.PduSessionType = exports.NaptEnabled = exports.ManagedServiceIdentityType = exports.CreatedByType = exports.CoreNetworkType = exports.BillingSku = void 0;
exports.BillingSku = {
    /**
     * Evaluation package plan
     */
    EvaluationPackage: "EvaluationPackage",
    /**
     * Flagship starter package plan
     */
    FlagshipStarterPackage: "FlagshipStarterPackage",
    /**
     * Edge site 2Gbps plan
     */
    EdgeSite2GBPS: "EdgeSite2GBPS",
    /**
     * Edge site 3Gbps plan
     */
    EdgeSite3GBPS: "EdgeSite3GBPS",
    /**
     * Edge site 4Gbps plan
     */
    EdgeSite4GBPS: "EdgeSite4GBPS",
    /**
     * Medium package plan
     */
    MediumPackage: "MediumPackage",
    /**
     * Large package plan
     */
    LargePackage: "LargePackage",
};
exports.CoreNetworkType = {
    /**
     * 5G core
     */
    CoreNetworkType_5GC: "5GC",
    /**
     * EPC / 4G core
     */
    EPC: "EPC",
};
exports.CreatedByType = {
    User: "User",
    Application: "Application",
    ManagedIdentity: "ManagedIdentity",
    Key: "Key",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NaptEnabled = {
    /**
     * NAPT is enabled
     */
    Enabled: "Enabled",
    /**
     * NAPT is disabled
     */
    Disabled: "Disabled",
};
exports.PduSessionType = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.PlatformType = {
    /**
     * If this option is chosen, you must set one of "azureStackEdgeDevice", "connectedCluster" or "customLocation". If multiple are set then "customLocation" will take precedence over "connectedCluster" which takes precedence over "azureStackEdgeDevice".
     */
    AKS_HCI: "AKS-HCI",
    /**
     * If this option is chosen, you must set one of "connectedCluster" or "customLocation". If multiple are set then "customLocation" will take precedence over "connectedCluster".
     */
    BaseVM: "BaseVM",
};
exports.PreemptionCapability = {
    /**
     * Cannot preempt
     */
    NotPreempt: "NotPreempt",
    /**
     * May preempt
     */
    MayPreempt: "MayPreempt",
};
exports.PreemptionVulnerability = {
    /**
     * Cannot be preempted
     */
    NotPreemptable: "NotPreemptable",
    /**
     * May be preempted
     */
    Preemptable: "Preemptable",
};
exports.SdfDirection = {
    /**
     * Traffic flowing from the UE to the data network.
     */
    Uplink: "Uplink",
    /**
     * Traffic flowing from the data network to the UE.
     */
    Downlink: "Downlink",
    /**
     * Traffic flowing both to and from the UE.
     */
    Bidirectional: "Bidirectional",
};
exports.TrafficControlPermission = {
    /**
     * Traffic matching this rule is allowed to flow.
     */
    Enabled: "Enabled",
    /**
     * Traffic matching this rule is not allowed to flow.
     */
    Blocked: "Blocked",
};
