// @pulumi/azure-native/types/enums/kubernetesconfiguration/v20230501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceKindType = exports.ScopeType = exports.ResourceIdentityType = exports.OperatorType = exports.OperatorScopeType = exports.LevelType = exports.AKSIdentityType = void 0;
exports.AKSIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.LevelType = {
    Error: "Error",
    Warning: "Warning",
    Information: "Information",
};
exports.OperatorScopeType = {
    Cluster: "cluster",
    Namespace: "namespace",
};
exports.OperatorType = {
    Flux: "Flux",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.ScopeType = {
    Cluster: "cluster",
    Namespace: "namespace",
};
exports.SourceKindType = {
    GitRepository: "GitRepository",
    Bucket: "Bucket",
    AzureBlob: "AzureBlob",
};
