// @pulumi/gcp/firebase/appCheckRecaptchaV3Config.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCheckRecaptchaV3Config = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An app's reCAPTCHA V3 configuration object.
 *
 * To get more information about RecaptchaV3Config, see:
 *
 * * [API documentation](https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.recaptchaV3Config)
 * * How-to Guides
 *     * [Official Documentation](https://firebase.google.com/docs/app-check)
 *
 * ## Example Usage
 *
 * ### Firebase App Check Recaptcha V3 Config Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 * import * as time from "@pulumi/time";
 *
 * const _default = new gcp.firebase.WebApp("default", {
 *     project: "my-project-name",
 *     displayName: "Web App for reCAPTCHA V3",
 * });
 * // It takes a while for App Check to recognize the new app
 * // If your app already exists, you don't have to wait 30 seconds.
 * const wait30s = new time.index.Sleep("wait_30s", {createDuration: "30s"}, {
 *     dependsOn: [_default],
 * });
 * const defaultAppCheckRecaptchaV3Config = new gcp.firebase.AppCheckRecaptchaV3Config("default", {
 *     project: "my-project-name",
 *     appId: _default.appId,
 *     siteSecret: "6Lf9YnQpAAAAAC3-MHmdAllTbPwTZxpUw5d34YzX",
 *     tokenTtl: "7200s",
 * }, {
 *     dependsOn: [wait30s],
 * });
 * ```
 *
 * ## Import
 *
 * RecaptchaV3Config can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/apps/{{app_id}}/recaptchaV3Config`
 *
 * * `{{project}}/{{app_id}}`
 *
 * * `{{app_id}}`
 *
 * When using the `pulumi import` command, RecaptchaV3Config can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:firebase/appCheckRecaptchaV3Config:AppCheckRecaptchaV3Config default projects/{{project}}/apps/{{app_id}}/recaptchaV3Config
 * ```
 *
 * ```sh
 * $ pulumi import gcp:firebase/appCheckRecaptchaV3Config:AppCheckRecaptchaV3Config default {{project}}/{{app_id}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:firebase/appCheckRecaptchaV3Config:AppCheckRecaptchaV3Config default {{app_id}}
 * ```
 */
class AppCheckRecaptchaV3Config extends pulumi.CustomResource {
    /**
     * Get an existing AppCheckRecaptchaV3Config resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AppCheckRecaptchaV3Config(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AppCheckRecaptchaV3Config.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AppCheckRecaptchaV3Config.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appId"] = state ? state.appId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["siteSecret"] = state ? state.siteSecret : undefined;
            resourceInputs["siteSecretSet"] = state ? state.siteSecretSet : undefined;
            resourceInputs["tokenTtl"] = state ? state.tokenTtl : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appId'");
            }
            if ((!args || args.siteSecret === undefined) && !opts.urn) {
                throw new Error("Missing required property 'siteSecret'");
            }
            resourceInputs["appId"] = args ? args.appId : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["siteSecret"] = (args === null || args === void 0 ? void 0 : args.siteSecret) ? pulumi.secret(args.siteSecret) : undefined;
            resourceInputs["tokenTtl"] = args ? args.tokenTtl : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["siteSecretSet"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["siteSecret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(AppCheckRecaptchaV3Config.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AppCheckRecaptchaV3Config = AppCheckRecaptchaV3Config;
/** @internal */
AppCheckRecaptchaV3Config.__pulumiType = 'gcp:firebase/appCheckRecaptchaV3Config:AppCheckRecaptchaV3Config';
