// @pulumi/gcp/firebase/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebApp = exports.StorageBucket = exports.Project = exports.HostingVersion = exports.HostingSite = exports.HostingRelease = exports.HostingCustomDomain = exports.HostingChannel = exports.getWebAppConfigOutput = exports.getWebAppConfig = exports.getWebAppOutput = exports.getWebApp = exports.getHostingChannelOutput = exports.getHostingChannel = exports.getAppleAppConfigOutput = exports.getAppleAppConfig = exports.getAppleAppOutput = exports.getAppleApp = exports.getAndroidAppConfigOutput = exports.getAndroidAppConfig = exports.getAndroidAppOutput = exports.getAndroidApp = exports.ExtensionsInstance = exports.DatabaseInstance = exports.AppleApp = exports.AppCheckServiceConfig = exports.AppCheckRecaptchaV3Config = exports.AppCheckRecaptchaEnterpriseConfig = exports.AppCheckPlayIntegrityConfig = exports.AppCheckDeviceCheckConfig = exports.AppCheckDebugToken = exports.AppCheckAppAttestConfig = exports.AndroidApp = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AndroidApp = null;
utilities.lazyLoad(exports, ["AndroidApp"], () => require("./androidApp"));
exports.AppCheckAppAttestConfig = null;
utilities.lazyLoad(exports, ["AppCheckAppAttestConfig"], () => require("./appCheckAppAttestConfig"));
exports.AppCheckDebugToken = null;
utilities.lazyLoad(exports, ["AppCheckDebugToken"], () => require("./appCheckDebugToken"));
exports.AppCheckDeviceCheckConfig = null;
utilities.lazyLoad(exports, ["AppCheckDeviceCheckConfig"], () => require("./appCheckDeviceCheckConfig"));
exports.AppCheckPlayIntegrityConfig = null;
utilities.lazyLoad(exports, ["AppCheckPlayIntegrityConfig"], () => require("./appCheckPlayIntegrityConfig"));
exports.AppCheckRecaptchaEnterpriseConfig = null;
utilities.lazyLoad(exports, ["AppCheckRecaptchaEnterpriseConfig"], () => require("./appCheckRecaptchaEnterpriseConfig"));
exports.AppCheckRecaptchaV3Config = null;
utilities.lazyLoad(exports, ["AppCheckRecaptchaV3Config"], () => require("./appCheckRecaptchaV3Config"));
exports.AppCheckServiceConfig = null;
utilities.lazyLoad(exports, ["AppCheckServiceConfig"], () => require("./appCheckServiceConfig"));
exports.AppleApp = null;
utilities.lazyLoad(exports, ["AppleApp"], () => require("./appleApp"));
exports.DatabaseInstance = null;
utilities.lazyLoad(exports, ["DatabaseInstance"], () => require("./databaseInstance"));
exports.ExtensionsInstance = null;
utilities.lazyLoad(exports, ["ExtensionsInstance"], () => require("./extensionsInstance"));
exports.getAndroidApp = null;
exports.getAndroidAppOutput = null;
utilities.lazyLoad(exports, ["getAndroidApp", "getAndroidAppOutput"], () => require("./getAndroidApp"));
exports.getAndroidAppConfig = null;
exports.getAndroidAppConfigOutput = null;
utilities.lazyLoad(exports, ["getAndroidAppConfig", "getAndroidAppConfigOutput"], () => require("./getAndroidAppConfig"));
exports.getAppleApp = null;
exports.getAppleAppOutput = null;
utilities.lazyLoad(exports, ["getAppleApp", "getAppleAppOutput"], () => require("./getAppleApp"));
exports.getAppleAppConfig = null;
exports.getAppleAppConfigOutput = null;
utilities.lazyLoad(exports, ["getAppleAppConfig", "getAppleAppConfigOutput"], () => require("./getAppleAppConfig"));
exports.getHostingChannel = null;
exports.getHostingChannelOutput = null;
utilities.lazyLoad(exports, ["getHostingChannel", "getHostingChannelOutput"], () => require("./getHostingChannel"));
exports.getWebApp = null;
exports.getWebAppOutput = null;
utilities.lazyLoad(exports, ["getWebApp", "getWebAppOutput"], () => require("./getWebApp"));
exports.getWebAppConfig = null;
exports.getWebAppConfigOutput = null;
utilities.lazyLoad(exports, ["getWebAppConfig", "getWebAppConfigOutput"], () => require("./getWebAppConfig"));
exports.HostingChannel = null;
utilities.lazyLoad(exports, ["HostingChannel"], () => require("./hostingChannel"));
exports.HostingCustomDomain = null;
utilities.lazyLoad(exports, ["HostingCustomDomain"], () => require("./hostingCustomDomain"));
exports.HostingRelease = null;
utilities.lazyLoad(exports, ["HostingRelease"], () => require("./hostingRelease"));
exports.HostingSite = null;
utilities.lazyLoad(exports, ["HostingSite"], () => require("./hostingSite"));
exports.HostingVersion = null;
utilities.lazyLoad(exports, ["HostingVersion"], () => require("./hostingVersion"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.StorageBucket = null;
utilities.lazyLoad(exports, ["StorageBucket"], () => require("./storageBucket"));
exports.WebApp = null;
utilities.lazyLoad(exports, ["WebApp"], () => require("./webApp"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:firebase/androidApp:AndroidApp":
                return new exports.AndroidApp(name, undefined, { urn });
            case "gcp:firebase/appCheckAppAttestConfig:AppCheckAppAttestConfig":
                return new exports.AppCheckAppAttestConfig(name, undefined, { urn });
            case "gcp:firebase/appCheckDebugToken:AppCheckDebugToken":
                return new exports.AppCheckDebugToken(name, undefined, { urn });
            case "gcp:firebase/appCheckDeviceCheckConfig:AppCheckDeviceCheckConfig":
                return new exports.AppCheckDeviceCheckConfig(name, undefined, { urn });
            case "gcp:firebase/appCheckPlayIntegrityConfig:AppCheckPlayIntegrityConfig":
                return new exports.AppCheckPlayIntegrityConfig(name, undefined, { urn });
            case "gcp:firebase/appCheckRecaptchaEnterpriseConfig:AppCheckRecaptchaEnterpriseConfig":
                return new exports.AppCheckRecaptchaEnterpriseConfig(name, undefined, { urn });
            case "gcp:firebase/appCheckRecaptchaV3Config:AppCheckRecaptchaV3Config":
                return new exports.AppCheckRecaptchaV3Config(name, undefined, { urn });
            case "gcp:firebase/appCheckServiceConfig:AppCheckServiceConfig":
                return new exports.AppCheckServiceConfig(name, undefined, { urn });
            case "gcp:firebase/appleApp:AppleApp":
                return new exports.AppleApp(name, undefined, { urn });
            case "gcp:firebase/databaseInstance:DatabaseInstance":
                return new exports.DatabaseInstance(name, undefined, { urn });
            case "gcp:firebase/extensionsInstance:ExtensionsInstance":
                return new exports.ExtensionsInstance(name, undefined, { urn });
            case "gcp:firebase/hostingChannel:HostingChannel":
                return new exports.HostingChannel(name, undefined, { urn });
            case "gcp:firebase/hostingCustomDomain:HostingCustomDomain":
                return new exports.HostingCustomDomain(name, undefined, { urn });
            case "gcp:firebase/hostingRelease:HostingRelease":
                return new exports.HostingRelease(name, undefined, { urn });
            case "gcp:firebase/hostingSite:HostingSite":
                return new exports.HostingSite(name, undefined, { urn });
            case "gcp:firebase/hostingVersion:HostingVersion":
                return new exports.HostingVersion(name, undefined, { urn });
            case "gcp:firebase/project:Project":
                return new exports.Project(name, undefined, { urn });
            case "gcp:firebase/storageBucket:StorageBucket":
                return new exports.StorageBucket(name, undefined, { urn });
            case "gcp:firebase/webApp:WebApp":
                return new exports.WebApp(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "firebase/androidApp", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckAppAttestConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckDebugToken", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckDeviceCheckConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckPlayIntegrityConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckRecaptchaEnterpriseConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckRecaptchaV3Config", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appCheckServiceConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/appleApp", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/databaseInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/extensionsInstance", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/hostingChannel", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/hostingCustomDomain", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/hostingRelease", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/hostingSite", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/hostingVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/project", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/storageBucket", _module);
pulumi.runtime.registerResourceModule("gcp", "firebase/webApp", _module);
