// @pulumi/azure-native/types/enums/databricks/v20230915preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredNsgRules = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.KeySource = exports.InitialType = exports.EnhancedSecurityMonitoringValue = exports.EncryptionKeySource = exports.ComplianceStandard = exports.ComplianceSecurityProfileValue = exports.AutomaticClusterUpdateValue = void 0;
exports.AutomaticClusterUpdateValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ComplianceSecurityProfileValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ComplianceStandard = {
    NONE: "NONE",
    HIPAA: "HIPAA",
    PCI_DSS: "PCI_DSS",
};
exports.EncryptionKeySource = {
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.EnhancedSecurityMonitoringValue = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.InitialType = {
    HiveMetastore: "HiveMetastore",
    UnityCatalog: "UnityCatalog",
};
exports.KeySource = {
    Default: "Default",
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RequiredNsgRules = {
    AllRules: "AllRules",
    NoAzureDatabricksRules: "NoAzureDatabricksRules",
    NoAzureServiceRules: "NoAzureServiceRules",
};
