// @pulumi/azure-native/managedservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20221001 = exports.RegistrationDefinition = exports.RegistrationAssignment = exports.getRegistrationDefinitionOutput = exports.getRegistrationDefinition = exports.getRegistrationAssignmentOutput = exports.getRegistrationAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRegistrationAssignment = null;
exports.getRegistrationAssignmentOutput = null;
utilities.lazyLoad(exports, ["getRegistrationAssignment", "getRegistrationAssignmentOutput"], () => require("./getRegistrationAssignment"));
exports.getRegistrationDefinition = null;
exports.getRegistrationDefinitionOutput = null;
utilities.lazyLoad(exports, ["getRegistrationDefinition", "getRegistrationDefinitionOutput"], () => require("./getRegistrationDefinition"));
exports.RegistrationAssignment = null;
utilities.lazyLoad(exports, ["RegistrationAssignment"], () => require("./registrationAssignment"));
exports.RegistrationDefinition = null;
utilities.lazyLoad(exports, ["RegistrationDefinition"], () => require("./registrationDefinition"));
// Export enums:
__exportStar(require("../types/enums/managedservices"), exports);
// Export sub-modules:
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managedservices:RegistrationAssignment":
                return new exports.RegistrationAssignment(name, undefined, { urn });
            case "azure-native:managedservices:RegistrationDefinition":
                return new exports.RegistrationDefinition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managedservices", _module);
