// @pulumi/azure-native/types/enums/dbforpostgresql/v20230302preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleType = exports.PrivateEndpointServiceConnectionStatus = exports.PrincipalType = exports.PasswordAuth = exports.IdentityType = exports.DataEncryptionType = exports.ActiveDirectoryAuth = void 0;
exports.ActiveDirectoryAuth = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.DataEncryptionType = {
    AzureKeyVault: "AzureKeyVault",
    SystemAssigned: "SystemAssigned",
};
exports.IdentityType = {
    UserAssigned: "UserAssigned",
    SystemAssigned: "SystemAssigned",
};
exports.PasswordAuth = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PrincipalType = {
    User: "user",
    ServicePrincipal: "servicePrincipal",
    Group: "group",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.RoleType = {
    User: "user",
    Admin: "admin",
};
