// @pulumi/azure-native/types/enums/workloads/v20230401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SslPreference = exports.SAPSoftwareInstallationType = exports.SAPProductType = exports.SAPHighAvailabilityType = exports.SAPEnvironmentType = exports.SAPDeploymentType = exports.SAPDatabaseType = exports.SAPDatabaseScaleMethod = exports.SAPConfigurationType = exports.RoutingPreference = exports.OSType = exports.NamingPatternType = exports.ManagedServiceIdentityType = exports.DiskSkuName = exports.ConfigurationType = void 0;
exports.ConfigurationType = {
    Skip: "Skip",
    CreateAndMount: "CreateAndMount",
    Mount: "Mount",
};
exports.DiskSkuName = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    StandardSSD_LRS: "StandardSSD_LRS",
    UltraSSD_LRS: "UltraSSD_LRS",
    Premium_ZRS: "Premium_ZRS",
    StandardSSD_ZRS: "StandardSSD_ZRS",
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    UserAssigned: "UserAssigned",
};
exports.NamingPatternType = {
    FullResourceName: "FullResourceName",
};
exports.OSType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.RoutingPreference = {
    Default: "Default",
    RouteAll: "RouteAll",
};
exports.SAPConfigurationType = {
    Deployment: "Deployment",
    Discovery: "Discovery",
    DeploymentWithOSConfig: "DeploymentWithOSConfig",
};
exports.SAPDatabaseScaleMethod = {
    ScaleUp: "ScaleUp",
};
exports.SAPDatabaseType = {
    HANA: "HANA",
    DB2: "DB2",
};
exports.SAPDeploymentType = {
    SingleServer: "SingleServer",
    ThreeTier: "ThreeTier",
};
exports.SAPEnvironmentType = {
    NonProd: "NonProd",
    Prod: "Prod",
};
exports.SAPHighAvailabilityType = {
    AvailabilitySet: "AvailabilitySet",
    AvailabilityZone: "AvailabilityZone",
};
exports.SAPProductType = {
    ECC: "ECC",
    S4HANA: "S4HANA",
    Other: "Other",
};
exports.SAPSoftwareInstallationType = {
    ServiceInitiated: "ServiceInitiated",
    SAPInstallWithoutOSConfig: "SAPInstallWithoutOSConfig",
    External: "External",
};
exports.SslPreference = {
    Disabled: "Disabled",
    RootCertificate: "RootCertificate",
    ServerCertificate: "ServerCertificate",
};
