// @pulumi/azure-native/elastic/v20241001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagRule = exports.OpenAI = exports.MonitoredSubscription = exports.Monitor = exports.listlistAssociatedTrafficFilterOutput = exports.listlistAssociatedTrafficFilter = exports.listVMHostOutput = exports.listVMHost = exports.listUpgradableVersionDetailsOutput = exports.listUpgradableVersionDetails = exports.listMonitoredResourceOutput = exports.listMonitoredResource = exports.listDeploymentInfoOutput = exports.listDeploymentInfo = exports.listConnectedPartnerResourceOutput = exports.listConnectedPartnerResource = exports.listAllTrafficFilterOutput = exports.listAllTrafficFilter = exports.getTagRuleOutput = exports.getTagRule = exports.getOrganizationElasticToAzureSubscriptionMappingOutput = exports.getOrganizationElasticToAzureSubscriptionMapping = exports.getOrganizationApiKeyOutput = exports.getOrganizationApiKey = exports.getOpenAIStatusOutput = exports.getOpenAIStatus = exports.getOpenAIOutput = exports.getOpenAI = exports.getMonitoredSubscriptionOutput = exports.getMonitoredSubscription = exports.getMonitorOutput = exports.getMonitor = exports.getBillingInfoOutput = exports.getBillingInfo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getBillingInfo = null;
exports.getBillingInfoOutput = null;
utilities.lazyLoad(exports, ["getBillingInfo", "getBillingInfoOutput"], () => require("./getBillingInfo"));
exports.getMonitor = null;
exports.getMonitorOutput = null;
utilities.lazyLoad(exports, ["getMonitor", "getMonitorOutput"], () => require("./getMonitor"));
exports.getMonitoredSubscription = null;
exports.getMonitoredSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getMonitoredSubscription", "getMonitoredSubscriptionOutput"], () => require("./getMonitoredSubscription"));
exports.getOpenAI = null;
exports.getOpenAIOutput = null;
utilities.lazyLoad(exports, ["getOpenAI", "getOpenAIOutput"], () => require("./getOpenAI"));
exports.getOpenAIStatus = null;
exports.getOpenAIStatusOutput = null;
utilities.lazyLoad(exports, ["getOpenAIStatus", "getOpenAIStatusOutput"], () => require("./getOpenAIStatus"));
exports.getOrganizationApiKey = null;
exports.getOrganizationApiKeyOutput = null;
utilities.lazyLoad(exports, ["getOrganizationApiKey", "getOrganizationApiKeyOutput"], () => require("./getOrganizationApiKey"));
exports.getOrganizationElasticToAzureSubscriptionMapping = null;
exports.getOrganizationElasticToAzureSubscriptionMappingOutput = null;
utilities.lazyLoad(exports, ["getOrganizationElasticToAzureSubscriptionMapping", "getOrganizationElasticToAzureSubscriptionMappingOutput"], () => require("./getOrganizationElasticToAzureSubscriptionMapping"));
exports.getTagRule = null;
exports.getTagRuleOutput = null;
utilities.lazyLoad(exports, ["getTagRule", "getTagRuleOutput"], () => require("./getTagRule"));
exports.listAllTrafficFilter = null;
exports.listAllTrafficFilterOutput = null;
utilities.lazyLoad(exports, ["listAllTrafficFilter", "listAllTrafficFilterOutput"], () => require("./listAllTrafficFilter"));
exports.listConnectedPartnerResource = null;
exports.listConnectedPartnerResourceOutput = null;
utilities.lazyLoad(exports, ["listConnectedPartnerResource", "listConnectedPartnerResourceOutput"], () => require("./listConnectedPartnerResource"));
exports.listDeploymentInfo = null;
exports.listDeploymentInfoOutput = null;
utilities.lazyLoad(exports, ["listDeploymentInfo", "listDeploymentInfoOutput"], () => require("./listDeploymentInfo"));
exports.listMonitoredResource = null;
exports.listMonitoredResourceOutput = null;
utilities.lazyLoad(exports, ["listMonitoredResource", "listMonitoredResourceOutput"], () => require("./listMonitoredResource"));
exports.listUpgradableVersionDetails = null;
exports.listUpgradableVersionDetailsOutput = null;
utilities.lazyLoad(exports, ["listUpgradableVersionDetails", "listUpgradableVersionDetailsOutput"], () => require("./listUpgradableVersionDetails"));
exports.listVMHost = null;
exports.listVMHostOutput = null;
utilities.lazyLoad(exports, ["listVMHost", "listVMHostOutput"], () => require("./listVMHost"));
exports.listlistAssociatedTrafficFilter = null;
exports.listlistAssociatedTrafficFilterOutput = null;
utilities.lazyLoad(exports, ["listlistAssociatedTrafficFilter", "listlistAssociatedTrafficFilterOutput"], () => require("./listlistAssociatedTrafficFilter"));
exports.Monitor = null;
utilities.lazyLoad(exports, ["Monitor"], () => require("./monitor"));
exports.MonitoredSubscription = null;
utilities.lazyLoad(exports, ["MonitoredSubscription"], () => require("./monitoredSubscription"));
exports.OpenAI = null;
utilities.lazyLoad(exports, ["OpenAI"], () => require("./openAI"));
exports.TagRule = null;
utilities.lazyLoad(exports, ["TagRule"], () => require("./tagRule"));
// Export enums:
__exportStar(require("../../types/enums/elastic/v20241001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:elastic/v20241001preview:Monitor":
                return new exports.Monitor(name, undefined, { urn });
            case "azure-native:elastic/v20241001preview:MonitoredSubscription":
                return new exports.MonitoredSubscription(name, undefined, { urn });
            case "azure-native:elastic/v20241001preview:OpenAI":
                return new exports.OpenAI(name, undefined, { urn });
            case "azure-native:elastic/v20241001preview:TagRule":
                return new exports.TagRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "elastic/v20241001preview", _module);
