// @pulumi/azure-native/storage/v20230401/queueServiceProperties.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueServiceProperties = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The properties of a storage account’s Queue service.
 */
class QueueServiceProperties extends pulumi.CustomResource {
    /**
     * Get an existing QueueServiceProperties resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new QueueServiceProperties(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of QueueServiceProperties.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QueueServiceProperties.__pulumiType;
    }
    /**
     * Create a QueueServiceProperties resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["cors"] = args ? args.cors : undefined;
            resourceInputs["queueServiceName"] = args ? args.queueServiceName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["cors"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20190601:QueueServiceProperties" }, { type: "azure-native:storage/v20200801preview:QueueServiceProperties" }, { type: "azure-native:storage/v20210101:QueueServiceProperties" }, { type: "azure-native:storage/v20210201:QueueServiceProperties" }, { type: "azure-native:storage/v20210401:QueueServiceProperties" }, { type: "azure-native:storage/v20210601:QueueServiceProperties" }, { type: "azure-native:storage/v20210801:QueueServiceProperties" }, { type: "azure-native:storage/v20210901:QueueServiceProperties" }, { type: "azure-native:storage/v20220501:QueueServiceProperties" }, { type: "azure-native:storage/v20220901:QueueServiceProperties" }, { type: "azure-native:storage/v20230101:QueueServiceProperties" }, { type: "azure-native:storage/v20230501:QueueServiceProperties" }, { type: "azure-native:storage/v20240101:QueueServiceProperties" }, { type: "azure-native:storage:QueueServiceProperties" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(QueueServiceProperties.__pulumiType, name, resourceInputs, opts);
    }
}
exports.QueueServiceProperties = QueueServiceProperties;
/** @internal */
QueueServiceProperties.__pulumiType = 'azure-native:storage/v20230401:QueueServiceProperties';
