// @pulumi/azure-native/portalservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240401preview = exports.v20240401 = exports.getCopilotSettingOutput = exports.getCopilotSetting = exports.CopilotSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CopilotSetting = null;
utilities.lazyLoad(exports, ["CopilotSetting"], () => require("./copilotSetting"));
exports.getCopilotSetting = null;
exports.getCopilotSettingOutput = null;
utilities.lazyLoad(exports, ["getCopilotSetting", "getCopilotSettingOutput"], () => require("./getCopilotSetting"));
// Export sub-modules:
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:portalservices:CopilotSetting":
                return new exports.CopilotSetting(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "portalservices", _module);
