// @pulumi/azure-native/storagemover/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240701 = exports.v20231001 = exports.v20230701preview = exports.v20230301 = exports.StorageMover = exports.Project = exports.JobDefinition = exports.getStorageMoverOutput = exports.getStorageMover = exports.getProjectOutput = exports.getProject = exports.getJobDefinitionOutput = exports.getJobDefinition = exports.getEndpointOutput = exports.getEndpoint = exports.getAgentOutput = exports.getAgent = exports.Endpoint = exports.Agent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Agent = null;
utilities.lazyLoad(exports, ["Agent"], () => require("./agent"));
exports.Endpoint = null;
utilities.lazyLoad(exports, ["Endpoint"], () => require("./endpoint"));
exports.getAgent = null;
exports.getAgentOutput = null;
utilities.lazyLoad(exports, ["getAgent", "getAgentOutput"], () => require("./getAgent"));
exports.getEndpoint = null;
exports.getEndpointOutput = null;
utilities.lazyLoad(exports, ["getEndpoint", "getEndpointOutput"], () => require("./getEndpoint"));
exports.getJobDefinition = null;
exports.getJobDefinitionOutput = null;
utilities.lazyLoad(exports, ["getJobDefinition", "getJobDefinitionOutput"], () => require("./getJobDefinition"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getStorageMover = null;
exports.getStorageMoverOutput = null;
utilities.lazyLoad(exports, ["getStorageMover", "getStorageMoverOutput"], () => require("./getStorageMover"));
exports.JobDefinition = null;
utilities.lazyLoad(exports, ["JobDefinition"], () => require("./jobDefinition"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.StorageMover = null;
utilities.lazyLoad(exports, ["StorageMover"], () => require("./storageMover"));
// Export enums:
__exportStar(require("../types/enums/storagemover"), exports);
// Export sub-modules:
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20231001 = require("./v20231001");
exports.v20231001 = v20231001;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storagemover:Agent":
                return new exports.Agent(name, undefined, { urn });
            case "azure-native:storagemover:Endpoint":
                return new exports.Endpoint(name, undefined, { urn });
            case "azure-native:storagemover:JobDefinition":
                return new exports.JobDefinition(name, undefined, { urn });
            case "azure-native:storagemover:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:storagemover:StorageMover":
                return new exports.StorageMover(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storagemover", _module);
