// @pulumi/azure-native/types/enums/azurestackhci/v20210901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.PrivateIPAllocationMethodEnum = exports.OsTypeEnum = exports.OperatingSystemTypes = exports.NetworkTypeEnum = exports.IpAllocationMethodEnum = exports.IPPoolTypeEnum = exports.HyperVGeneration = exports.ExtendedLocationTypes = exports.DiskFileFormat = exports.CreatedByType = exports.CloudInitDataSource = void 0;
exports.CloudInitDataSource = {
    NoCloud: "NoCloud",
    Azure: "Azure",
};
exports.CreatedByType = {
    User: "User",
    Application: "Application",
    ManagedIdentity: "ManagedIdentity",
    Key: "Key",
};
exports.DiskFileFormat = {
    Vhdx: "vhdx",
    Vhd: "vhd",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.HyperVGeneration = {
    V1: "V1",
    V2: "V2",
};
exports.IPPoolTypeEnum = {
    Vm: "vm",
    Vippool: "vippool",
};
exports.IpAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.NetworkTypeEnum = {
    NAT: "NAT",
    Transparent: "Transparent",
    L2Bridge: "L2Bridge",
    L2Tunnel: "L2Tunnel",
    ICS: "ICS",
    Private: "Private",
    Overlay: "Overlay",
    Internal: "Internal",
    Mirrored: "Mirrored",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.OsTypeEnum = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.PrivateIPAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
