// @pulumi/azure-native/resourcegraph/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240401 = exports.v20221001 = exports.v20210301 = exports.v20200401preview = exports.v20190401 = exports.v20180901preview = exports.GraphQuery = exports.getGraphQueryOutput = exports.getGraphQuery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGraphQuery = null;
exports.getGraphQueryOutput = null;
utilities.lazyLoad(exports, ["getGraphQuery", "getGraphQueryOutput"], () => require("./getGraphQuery"));
exports.GraphQuery = null;
utilities.lazyLoad(exports, ["GraphQuery"], () => require("./graphQuery"));
// Export sub-modules:
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20190401 = require("./v20190401");
exports.v20190401 = v20190401;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:resourcegraph:GraphQuery":
                return new exports.GraphQuery(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "resourcegraph", _module);
