// @pulumi/azure-native/types/enums/azurefleet/v20241101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsVMGuestPatchMode = exports.WindowsVMGuestPatchAutomaticByPlatformRebootSetting = exports.WindowsPatchAssessmentMode = exports.VMCategory = exports.VMAttributeSupport = exports.StorageAccountTypes = exports.SpotAllocationStrategy = exports.SettingNames = exports.SecurityTypes = exports.SecurityEncryptionTypes = exports.RegularPriorityAllocationStrategy = exports.PublicIPAddressSkuTier = exports.PublicIPAddressSkuName = exports.ProtocolTypes = exports.PassName = exports.OperatingSystemTypes = exports.NetworkInterfaceAuxiliarySku = exports.NetworkInterfaceAuxiliaryMode = exports.NetworkApiVersion = exports.Mode = exports.ManagedServiceIdentityType = exports.LocalStorageDiskType = exports.LinuxVMGuestPatchMode = exports.LinuxVMGuestPatchAutomaticByPlatformRebootSetting = exports.LinuxPatchAssessmentMode = exports.IPVersion = exports.EvictionPolicy = exports.DomainNameLabelScopeTypes = exports.DiskDeleteOptionTypes = exports.DiskCreateOptionTypes = exports.DiskControllerTypes = exports.DiffDiskPlacement = exports.DiffDiskOptions = exports.DeleteOptions = exports.CpuManufacturer = exports.ComponentName = exports.CachingTypes = exports.ArchitectureType = exports.AcceleratorType = exports.AcceleratorManufacturer = void 0;
exports.AcceleratorManufacturer = {
    /**
     * AMD GpuType
     */
    AMD: "AMD",
    /**
     * Nvidia GpuType
     */
    Nvidia: "Nvidia",
    /**
     * Xilinx GpuType
     */
    Xilinx: "Xilinx",
};
exports.AcceleratorType = {
    /**
     * GPU Accelerator
     */
    GPU: "GPU",
    /**
     * FPGA Accelerator
     */
    FPGA: "FPGA",
};
exports.ArchitectureType = {
    /**
     * ARM64 Architecture
     */
    ARM64: "ARM64",
    /**
     * X64 Architecture
     */
    X64: "X64",
};
exports.CachingTypes = {
    /**
     * 'None' is default for Standard Storage
     */
    None: "None",
    /**
     * 'ReadOnly' is default for Premium Storage
     */
    ReadOnly: "ReadOnly",
    /**
     * 'ReadWrite' is default for OS Disk
     */
    ReadWrite: "ReadWrite",
};
exports.ComponentName = {
    Microsoft_Windows_Shell_Setup: "Microsoft-Windows-Shell-Setup",
};
exports.CpuManufacturer = {
    /**
     * Intel CPU.
     */
    Intel: "Intel",
    /**
     * AMD CPU.
     */
    AMD: "AMD",
    /**
     * Microsoft CPU.
     */
    Microsoft: "Microsoft",
    /**
     * Ampere CPU.
     */
    Ampere: "Ampere",
};
exports.DeleteOptions = {
    /**
     * Delete Option
     */
    Delete: "Delete",
    /**
     * Detach Option
     */
    Detach: "Detach",
};
exports.DiffDiskOptions = {
    /**
     * Local Option.
     */
    Local: "Local",
};
exports.DiffDiskPlacement = {
    /**
     * CacheDisk option.
     */
    CacheDisk: "CacheDisk",
    /**
     * Resource Disk option.
     */
    ResourceDisk: "ResourceDisk",
    /**
     * NvmeDisk option.
     */
    NvmeDisk: "NvmeDisk",
};
exports.DiskControllerTypes = {
    /**
     * SCSI disk type
     */
    SCSI: "SCSI",
    /**
     * NVMe disk type
     */
    NVMe: "NVMe",
};
exports.DiskCreateOptionTypes = {
    /**
     * This value is used when you are using an image to create the virtual machine.
     * If you are using a platform image, you also use the imageReference element
     * described above. If you are using a marketplace image, you also use the
     * plan element previously described.
     */
    FromImage: "FromImage",
    /**
     * This value is used when creating an empty data disk.
     */
    Empty: "Empty",
    /**
     * This value is used when you are using a specialized disk to create the virtual machine.
     */
    Attach: "Attach",
    /**
     * This value is used to create a data disk from a snapshot or another disk.
     */
    Copy: "Copy",
    /**
     * This value is used to create a data disk from a disk restore point.
     */
    Restore: "Restore",
};
exports.DiskDeleteOptionTypes = {
    /**
     * If this value is used, the managed disk is deleted when VM gets deleted.
     */
    Delete: "Delete",
    /**
     * If this value is used, the managed disk is retained after VM gets deleted.
     */
    Detach: "Detach",
};
exports.DomainNameLabelScopeTypes = {
    /**
     * TenantReuse type
     */
    TenantReuse: "TenantReuse",
    /**
     * SubscriptionReuse type
     */
    SubscriptionReuse: "SubscriptionReuse",
    /**
     * ResourceGroupReuse type
     */
    ResourceGroupReuse: "ResourceGroupReuse",
    /**
     * NoReuse type
     */
    NoReuse: "NoReuse",
};
exports.EvictionPolicy = {
    /**
     * When evicted, the Spot VM will be deleted and the corresponding capacity will be updated to reflect this.
     */
    Delete: "Delete",
    /**
     * When evicted, the Spot VM will be deallocated/stopped
     */
    Deallocate: "Deallocate",
};
exports.IPVersion = {
    /**
     * IPv4 version
     */
    IPv4: "IPv4",
    /**
     * IPv6 version
     */
    IPv6: "IPv6",
};
exports.LinuxPatchAssessmentMode = {
    /**
     * You control the timing of patch assessments on a virtual machine.
     */
    ImageDefault: "ImageDefault",
    /**
     * The platform will trigger periodic patch assessments.The property provisionVMAgent must be true.
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.LinuxVMGuestPatchAutomaticByPlatformRebootSetting = {
    /**
     * Unknown Reboot setting
     */
    Unknown: "Unknown",
    /**
     * IfRequired Reboot setting
     */
    IfRequired: "IfRequired",
    /**
     * Never Reboot setting
     */
    Never: "Never",
    /**
     * Always Reboot setting
     */
    Always: "Always",
};
exports.LinuxVMGuestPatchMode = {
    /**
     * The virtual machine's default patching configuration is used.
     */
    ImageDefault: "ImageDefault",
    /**
     * The virtual machine will be automatically updated by the platform. The property provisionVMAgent must be true.
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.LocalStorageDiskType = {
    /**
     * HDD DiskType.
     */
    HDD: "HDD",
    /**
     * SDD DiskType.
     */
    SSD: "SSD",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.Mode = {
    /**
     * Audit Mode
     */
    Audit: "Audit",
    /**
     * Enforce Mode
     */
    Enforce: "Enforce",
};
exports.NetworkApiVersion = {
    /**
     * Initial version supported. Later versions are supported as well.
     */
    V2020_11_01: "2020-11-01",
};
exports.NetworkInterfaceAuxiliaryMode = {
    /**
     * None Mode
     */
    None: "None",
    /**
     * AcceleratedConnections Mode
     */
    AcceleratedConnections: "AcceleratedConnections",
    /**
     * Floating Mode
     */
    Floating: "Floating",
};
exports.NetworkInterfaceAuxiliarySku = {
    /**
     * no sku
     */
    None: "None",
    /**
     * A1 sku
     */
    A1: "A1",
    /**
     * A2 sku
     */
    A2: "A2",
    /**
     * A4 sku
     */
    A4: "A4",
    /**
     * A8 sku
     */
    A8: "A8",
};
exports.OperatingSystemTypes = {
    /**
     * Windows OS type
     */
    Windows: "Windows",
    /**
     * Linux OS type
     */
    Linux: "Linux",
};
exports.PassName = {
    OobeSystem: "OobeSystem",
};
exports.ProtocolTypes = {
    /**
     * Http protocol
     */
    Http: "Http",
    /**
     * Https protocol
     */
    Https: "Https",
};
exports.PublicIPAddressSkuName = {
    /**
     * Basic sku name
     */
    Basic: "Basic",
    /**
     * Standard sku name
     */
    Standard: "Standard",
};
exports.PublicIPAddressSkuTier = {
    /**
     * Regional sku tier
     */
    Regional: "Regional",
    /**
     * Global sku tier
     */
    Global: "Global",
};
exports.RegularPriorityAllocationStrategy = {
    /**
     * Default. VM sizes distribution will be determined to optimize for price.
     */
    LowestPrice: "LowestPrice",
    /**
     * VM sizes distribution will be determined to optimize for the 'priority' as specified for each vm size.
     */
    Prioritized: "Prioritized",
};
exports.SecurityEncryptionTypes = {
    /**
     * EncryptionType of the managed disk is set to VMGuestStateOnly for encryption
     * of just the VMGuestState blob.
     */
    VMGuestStateOnly: "VMGuestStateOnly",
    /**
     * EncryptionType of the managed disk is set to DiskWithVMGuestState for encryption
     * of the managed disk along with VMGuestState blob.
     */
    DiskWithVMGuestState: "DiskWithVMGuestState",
    /**
     * EncryptionType of the managed disk is set to NonPersistedTPM for not persisting
     * firmware state in the VMGuestState blob.
     */
    NonPersistedTPM: "NonPersistedTPM",
};
exports.SecurityTypes = {
    /**
     * TrustedLaunch security type
     */
    TrustedLaunch: "TrustedLaunch",
    /**
     * ConfidentialVM security type
     */
    ConfidentialVM: "ConfidentialVM",
};
exports.SettingNames = {
    /**
     * AutoLogon setting
     */
    AutoLogon: "AutoLogon",
    /**
     * FirstLogonCommands setting
     */
    FirstLogonCommands: "FirstLogonCommands",
};
exports.SpotAllocationStrategy = {
    /**
     * Default. VM sizes distribution will be determined to optimize for both price and capacity.
     */
    PriceCapacityOptimized: "PriceCapacityOptimized",
    /**
     * VM sizes distribution will be determined to optimize for price. Note: Capacity will still be considered here but will be given much less weight.
     */
    LowestPrice: "LowestPrice",
    /**
     * VM sizes distribution will be determined to optimize for capacity.
     */
    CapacityOptimized: "CapacityOptimized",
};
exports.StorageAccountTypes = {
    /**
     * Standard_LRS option.
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Premium_LRS option.
     */
    Premium_LRS: "Premium_LRS",
    /**
     * StandardSSD_LRS option.
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * UltraSSD_LRS option.
     */
    UltraSSD_LRS: "UltraSSD_LRS",
    /**
     * Premium_ZRS option.
     */
    Premium_ZRS: "Premium_ZRS",
    /**
     * StandardSSD_ZRS option.
     */
    StandardSSD_ZRS: "StandardSSD_ZRS",
    /**
     * PremiumV2_LRS option.
     */
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.VMAttributeSupport = {
    /**
     * All VMSizes having the feature support will be excluded.
     */
    Excluded: "Excluded",
    /**
     *  VMSizes that have the feature support and that do not have the feature support will be used. Included is a union of Excluded and Required.
     */
    Included: "Included",
    /**
     * Only the VMSizes having the feature support will be used.
     */
    Required: "Required",
};
exports.VMCategory = {
    /**
     * General purpose VM sizes provide balanced CPU-to-memory ratio. Ideal for testing and development, small to medium databases, and low to medium traffic web servers.
     */
    GeneralPurpose: "GeneralPurpose",
    /**
     * Compute optimized VM sizes have a high CPU-to-memory ratio. These sizes are good for medium traffic web servers, network appliances, batch processes, and application servers.
     */
    ComputeOptimized: "ComputeOptimized",
    /**
     * Memory optimized VM sizes offer a high memory-to-CPU ratio that is great for relational database servers, medium to large caches, and in-memory analytics.
     */
    MemoryOptimized: "MemoryOptimized",
    /**
     * Storage optimized virtual machine (VM) sizes offer high disk throughput and IO, and are ideal for Big Data, SQL, NoSQL databases, data warehousing, and large transactional databases.
     * Examples include Cassandra, MongoDB, Cloudera, and Redis.
     */
    StorageOptimized: "StorageOptimized",
    /**
     * GPU optimized VM sizes are specialized virtual machines available with single, multiple, or fractional GPUs.
     * These sizes are designed for compute-intensive, graphics-intensive, and visualization workloads.
     */
    GpuAccelerated: "GpuAccelerated",
    /**
     * FPGA optimized VM sizes are specialized virtual machines available with single or multiple FPGA.
     * These sizes are designed for compute-intensive workloads. This article provides information about the number and type of FPGA, vCPUs, data disks, and NICs.
     * Storage throughput and network bandwidth are also included for each size in this grouping.
     */
    FpgaAccelerated: "FpgaAccelerated",
    /**
     * Azure High Performance Compute VMs are optimized for various HPC workloads such as computational fluid dynamics, finite element analysis, frontend and backend EDA,
     * rendering, molecular dynamics, computational geo science, weather simulation, and financial risk analysis.
     */
    HighPerformanceCompute: "HighPerformanceCompute",
};
exports.WindowsPatchAssessmentMode = {
    /**
     * You control the timing of patch assessments on a virtual machine.
     */
    ImageDefault: "ImageDefault",
    /**
     * The platform will trigger periodic patch assessments. The property provisionVMAgent must be true.
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.WindowsVMGuestPatchAutomaticByPlatformRebootSetting = {
    /**
     * Unknown Reboot setting
     */
    Unknown: "Unknown",
    /**
     * IfRequired Reboot setting
     */
    IfRequired: "IfRequired",
    /**
     * Never Reboot setting
     */
    Never: "Never",
    /**
     * Always Reboot setting
     */
    Always: "Always",
};
exports.WindowsVMGuestPatchMode = {
    /**
     * You control the application of patches to a virtual machine.
     * You do this by applying patches manually inside the VM. In this mode,
     * automatic updates are disabled; the property WindowsConfiguration.enableAutomaticUpdates
     * must be false
     */
    Manual: "Manual",
    /**
     * The virtual machine will automatically be updated by the OS.
     * The property WindowsConfiguration.enableAutomaticUpdates must be true.
     */
    AutomaticByOS: "AutomaticByOS",
    /**
     * The virtual machine will automatically updated by the platform. The properties
     * provisionVMAgent and WindowsConfiguration.enableAutomaticUpdates must be true.
     */
    AutomaticByPlatform: "AutomaticByPlatform",
};
