// @pulumi/azure-native/types/enums/avs/v20220501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SslEnum = exports.ScriptExecutionParameterType = exports.ResourceIdentityType = exports.PortMirroringDirectionEnum = exports.PlacementPolicyType = exports.PlacementPolicyState = exports.MountOptionEnum = exports.InternetEnum = exports.EncryptionState = exports.DnsServiceLogLevelEnum = exports.DhcpTypeEnum = exports.AzureHybridBenefitType = exports.AvailabilityStrategy = exports.AffinityType = exports.AffinityStrength = exports.AddonType = void 0;
exports.AddonType = {
    SRM: "SRM",
    VR: "VR",
    HCX: "HCX",
    Arc: "Arc",
};
exports.AffinityStrength = {
    Should: "Should",
    Must: "Must",
};
exports.AffinityType = {
    Affinity: "Affinity",
    AntiAffinity: "AntiAffinity",
};
exports.AvailabilityStrategy = {
    SingleZone: "SingleZone",
    DualZone: "DualZone",
};
exports.AzureHybridBenefitType = {
    SqlHost: "SqlHost",
    None: "None",
};
exports.DhcpTypeEnum = {
    SERVER: "SERVER",
    RELAY: "RELAY",
};
exports.DnsServiceLogLevelEnum = {
    DEBUG: "DEBUG",
    INFO: "INFO",
    WARNING: "WARNING",
    ERROR: "ERROR",
    FATAL: "FATAL",
};
exports.EncryptionState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.InternetEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MountOptionEnum = {
    MOUNT: "MOUNT",
    ATTACH: "ATTACH",
};
exports.PlacementPolicyState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PlacementPolicyType = {
    VmVm: "VmVm",
    VmHost: "VmHost",
};
exports.PortMirroringDirectionEnum = {
    INGRESS: "INGRESS",
    EGRESS: "EGRESS",
    BIDIRECTIONAL: "BIDIRECTIONAL",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.ScriptExecutionParameterType = {
    Value: "Value",
    SecureValue: "SecureValue",
    Credential: "Credential",
};
exports.SslEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
