// @pulumi/azure-native/analysisservices/serverDetails.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerDetails = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Represents an instance of an Analysis Services resource.
 *
 * Uses Azure REST API version 2017-08-01. In version 1.x of the Azure Native provider, it used API version 2017-08-01.
 *
 * Other available API versions: 2017-08-01-beta.
 */
class ServerDetails extends pulumi.CustomResource {
    /**
     * Get an existing ServerDetails resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerDetails(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerDetails.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerDetails.__pulumiType;
    }
    /**
     * Create a ServerDetails resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["asAdministrators"] = args ? args.asAdministrators : undefined;
            resourceInputs["backupBlobContainerUri"] = args ? args.backupBlobContainerUri : undefined;
            resourceInputs["gatewayDetails"] = args ? args.gatewayDetails : undefined;
            resourceInputs["ipV4FirewallSettings"] = args ? args.ipV4FirewallSettings : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managedMode"] = (_a = (args ? args.managedMode : undefined)) !== null && _a !== void 0 ? _a : 1;
            resourceInputs["querypoolConnectionMode"] = (_b = (args ? args.querypoolConnectionMode : undefined)) !== null && _b !== void 0 ? _b : "All";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverMonitorMode"] = (_c = (args ? args.serverMonitorMode : undefined)) !== null && _c !== void 0 ? _c : 1;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["sku"] = args ? (args.sku ? pulumi.output(args.sku).apply(inputs.analysisservices.resourceSkuArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serverFullName"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["asAdministrators"] = undefined /*out*/;
            resourceInputs["backupBlobContainerUri"] = undefined /*out*/;
            resourceInputs["gatewayDetails"] = undefined /*out*/;
            resourceInputs["ipV4FirewallSettings"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedMode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["querypoolConnectionMode"] = undefined /*out*/;
            resourceInputs["serverFullName"] = undefined /*out*/;
            resourceInputs["serverMonitorMode"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:analysisservices/v20160516:ServerDetails" }, { type: "azure-native:analysisservices/v20170714:ServerDetails" }, { type: "azure-native:analysisservices/v20170801:ServerDetails" }, { type: "azure-native:analysisservices/v20170801beta:ServerDetails" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerDetails.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServerDetails = ServerDetails;
/** @internal */
ServerDetails.__pulumiType = 'azure-native:analysisservices:ServerDetails';
