// @pulumi/azure-native/app/daprComponentResiliencyPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaprComponentResiliencyPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Dapr Component Resiliency Policy.
 *
 * Uses Azure REST API version 2023-08-01-preview.
 *
 * Other available API versions: 2023-11-02-preview, 2024-02-02-preview, 2024-08-02-preview, 2024-10-02-preview.
 */
class DaprComponentResiliencyPolicy extends pulumi.CustomResource {
    /**
     * Get an existing DaprComponentResiliencyPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DaprComponentResiliencyPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DaprComponentResiliencyPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DaprComponentResiliencyPolicy.__pulumiType;
    }
    /**
     * Create a DaprComponentResiliencyPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.componentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'componentName'");
            }
            if ((!args || args.environmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["componentName"] = args ? args.componentName : undefined;
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["inboundPolicy"] = args ? args.inboundPolicy : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["outboundPolicy"] = args ? args.outboundPolicy : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["inboundPolicy"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["outboundPolicy"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app/v20230801preview:DaprComponentResiliencyPolicy" }, { type: "azure-native:app/v20231102preview:DaprComponentResiliencyPolicy" }, { type: "azure-native:app/v20240202preview:DaprComponentResiliencyPolicy" }, { type: "azure-native:app/v20240802preview:DaprComponentResiliencyPolicy" }, { type: "azure-native:app/v20241002preview:DaprComponentResiliencyPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DaprComponentResiliencyPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DaprComponentResiliencyPolicy = DaprComponentResiliencyPolicy;
/** @internal */
DaprComponentResiliencyPolicy.__pulumiType = 'azure-native:app:DaprComponentResiliencyPolicy';
