// @pulumi/azure-native/authorization/v20220801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VariableValueAtManagementGroup = exports.VariableValue = exports.VariableAtManagementGroup = exports.Variable = exports.getVariableValueAtManagementGroupOutput = exports.getVariableValueAtManagementGroup = exports.getVariableValueOutput = exports.getVariableValue = exports.getVariableAtManagementGroupOutput = exports.getVariableAtManagementGroup = exports.getVariableOutput = exports.getVariable = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getVariable = null;
exports.getVariableOutput = null;
utilities.lazyLoad(exports, ["getVariable", "getVariableOutput"], () => require("./getVariable"));
exports.getVariableAtManagementGroup = null;
exports.getVariableAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getVariableAtManagementGroup", "getVariableAtManagementGroupOutput"], () => require("./getVariableAtManagementGroup"));
exports.getVariableValue = null;
exports.getVariableValueOutput = null;
utilities.lazyLoad(exports, ["getVariableValue", "getVariableValueOutput"], () => require("./getVariableValue"));
exports.getVariableValueAtManagementGroup = null;
exports.getVariableValueAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getVariableValueAtManagementGroup", "getVariableValueAtManagementGroupOutput"], () => require("./getVariableValueAtManagementGroup"));
exports.Variable = null;
utilities.lazyLoad(exports, ["Variable"], () => require("./variable"));
exports.VariableAtManagementGroup = null;
utilities.lazyLoad(exports, ["VariableAtManagementGroup"], () => require("./variableAtManagementGroup"));
exports.VariableValue = null;
utilities.lazyLoad(exports, ["VariableValue"], () => require("./variableValue"));
exports.VariableValueAtManagementGroup = null;
utilities.lazyLoad(exports, ["VariableValueAtManagementGroup"], () => require("./variableValueAtManagementGroup"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:authorization/v20220801preview:Variable":
                return new exports.Variable(name, undefined, { urn });
            case "azure-native:authorization/v20220801preview:VariableAtManagementGroup":
                return new exports.VariableAtManagementGroup(name, undefined, { urn });
            case "azure-native:authorization/v20220801preview:VariableValue":
                return new exports.VariableValue(name, undefined, { urn });
            case "azure-native:authorization/v20220801preview:VariableValueAtManagementGroup":
                return new exports.VariableValueAtManagementGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "authorization/v20220801preview", _module);
