// @pulumi/azure-native/dependencymap/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250131preview = exports.Map = exports.getMapOutput = exports.getMap = exports.getDiscoverySourceOutput = exports.getDiscoverySource = exports.DiscoverySource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DiscoverySource = null;
utilities.lazyLoad(exports, ["DiscoverySource"], () => require("./discoverySource"));
exports.getDiscoverySource = null;
exports.getDiscoverySourceOutput = null;
utilities.lazyLoad(exports, ["getDiscoverySource", "getDiscoverySourceOutput"], () => require("./getDiscoverySource"));
exports.getMap = null;
exports.getMapOutput = null;
utilities.lazyLoad(exports, ["getMap", "getMapOutput"], () => require("./getMap"));
exports.Map = null;
utilities.lazyLoad(exports, ["Map"], () => require("./map"));
// Export enums:
__exportStar(require("../types/enums/dependencymap"), exports);
// Export sub-modules:
const v20250131preview = require("./v20250131preview");
exports.v20250131preview = v20250131preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dependencymap:DiscoverySource":
                return new exports.DiscoverySource(name, undefined, { urn });
            case "azure-native:dependencymap:Map":
                return new exports.Map(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dependencymap", _module);
