// @pulumi/azure-native/types/enums/datalakeanalytics/v20191101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TierType = exports.FirewallState = exports.FirewallAllowAzureIpsState = exports.AADObjectType = void 0;
exports.AADObjectType = {
    User: "User",
    Group: "Group",
    ServicePrincipal: "ServicePrincipal",
};
exports.FirewallAllowAzureIpsState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FirewallState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TierType = {
    Consumption: "Consumption",
    Commitment_100AUHours: "Commitment_100AUHours",
    Commitment_500AUHours: "Commitment_500AUHours",
    Commitment_1000AUHours: "Commitment_1000AUHours",
    Commitment_5000AUHours: "Commitment_5000AUHours",
    Commitment_10000AUHours: "Commitment_10000AUHours",
    Commitment_50000AUHours: "Commitment_50000AUHours",
    Commitment_100000AUHours: "Commitment_100000AUHours",
    Commitment_500000AUHours: "Commitment_500000AUHours",
};
