// @pulumi/azure-native/security/v20210801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Standard = exports.getStandardOutput = exports.getStandard = exports.getAssignmentOutput = exports.getAssignment = exports.Assignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Assignment = null;
utilities.lazyLoad(exports, ["Assignment"], () => require("./assignment"));
exports.getAssignment = null;
exports.getAssignmentOutput = null;
utilities.lazyLoad(exports, ["getAssignment", "getAssignmentOutput"], () => require("./getAssignment"));
exports.getStandard = null;
exports.getStandardOutput = null;
utilities.lazyLoad(exports, ["getStandard", "getStandardOutput"], () => require("./getStandard"));
exports.Standard = null;
utilities.lazyLoad(exports, ["Standard"], () => require("./standard"));
// Export enums:
__exportStar(require("../../types/enums/security/v20210801preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20210801preview:Assignment":
                return new exports.Assignment(name, undefined, { urn });
            case "azure-native:security/v20210801preview:Standard":
                return new exports.Standard(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20210801preview", _module);
