// @pulumi/azure-native/securityinsights/v20240401preview/ascdataConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ASCDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents ASC (Azure Security Center) data connector.
 */
class ASCDataConnector extends pulumi.CustomResource {
    /**
     * Get an existing ASCDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ASCDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ASCDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ASCDataConnector.__pulumiType;
    }
    /**
     * Create a ASCDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["dataTypes"] = args ? args.dataTypes : undefined;
            resourceInputs["kind"] = "AzureSecurityCenter";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataTypes"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20200101:ASCDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20211001:ASCDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220501preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220601preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220701preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220801:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220801preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20220901preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20221001preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20221101:ASCDataConnector" }, { type: "azure-native:securityinsights/v20221101preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20221201preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230201:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230201preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230301preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230401preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230501preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230601preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230701preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230801preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20230901preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20231001preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20231101:ASCDataConnector" }, { type: "azure-native:securityinsights/v20231201preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20240101preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20240301:ASCDataConnector" }, { type: "azure-native:securityinsights/v20240901:ASCDataConnector" }, { type: "azure-native:securityinsights/v20241001preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20250101preview:ASCDataConnector" }, { type: "azure-native:securityinsights/v20250301:ASCDataConnector" }, { type: "azure-native:securityinsights:ASCDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ASCDataConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ASCDataConnector = ASCDataConnector;
/** @internal */
ASCDataConnector.__pulumiType = 'azure-native:securityinsights/v20240401preview:ASCDataConnector';
