// @pulumi/azure-native/machinelearningservices/v20200901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.listWorkspaceKeysOutput = exports.listWorkspaceKeys = exports.listNotebookKeysOutput = exports.listNotebookKeys = exports.LinkedService = exports.LabelingJob = exports.getWorkspaceOutput = exports.getWorkspace = exports.getLinkedServiceOutput = exports.getLinkedService = exports.getLabelingJobOutput = exports.getLabelingJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLabelingJob = null;
exports.getLabelingJobOutput = null;
utilities.lazyLoad(exports, ["getLabelingJob", "getLabelingJobOutput"], () => require("./getLabelingJob"));
exports.getLinkedService = null;
exports.getLinkedServiceOutput = null;
utilities.lazyLoad(exports, ["getLinkedService", "getLinkedServiceOutput"], () => require("./getLinkedService"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.LabelingJob = null;
utilities.lazyLoad(exports, ["LabelingJob"], () => require("./labelingJob"));
exports.LinkedService = null;
utilities.lazyLoad(exports, ["LinkedService"], () => require("./linkedService"));
exports.listNotebookKeys = null;
exports.listNotebookKeysOutput = null;
utilities.lazyLoad(exports, ["listNotebookKeys", "listNotebookKeysOutput"], () => require("./listNotebookKeys"));
exports.listWorkspaceKeys = null;
exports.listWorkspaceKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceKeys", "listWorkspaceKeysOutput"], () => require("./listWorkspaceKeys"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../../types/enums/machinelearningservices/v20200901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningservices/v20200901preview:LabelingJob":
                return new exports.LabelingJob(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20200901preview:LinkedService":
                return new exports.LinkedService(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20200901preview:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningservices/v20200901preview", _module);
