// @pulumi/azure-native/secretsynccontroller/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240821preview = exports.SecretSync = exports.getSecretSyncOutput = exports.getSecretSync = exports.getAzureKeyVaultSecretProviderClassOutput = exports.getAzureKeyVaultSecretProviderClass = exports.AzureKeyVaultSecretProviderClass = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureKeyVaultSecretProviderClass = null;
utilities.lazyLoad(exports, ["AzureKeyVaultSecretProviderClass"], () => require("./azureKeyVaultSecretProviderClass"));
exports.getAzureKeyVaultSecretProviderClass = null;
exports.getAzureKeyVaultSecretProviderClassOutput = null;
utilities.lazyLoad(exports, ["getAzureKeyVaultSecretProviderClass", "getAzureKeyVaultSecretProviderClassOutput"], () => require("./getAzureKeyVaultSecretProviderClass"));
exports.getSecretSync = null;
exports.getSecretSyncOutput = null;
utilities.lazyLoad(exports, ["getSecretSync", "getSecretSyncOutput"], () => require("./getSecretSync"));
exports.SecretSync = null;
utilities.lazyLoad(exports, ["SecretSync"], () => require("./secretSync"));
// Export enums:
__exportStar(require("../types/enums/secretsynccontroller"), exports);
// Export sub-modules:
const v20240821preview = require("./v20240821preview");
exports.v20240821preview = v20240821preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:secretsynccontroller:AzureKeyVaultSecretProviderClass":
                return new exports.AzureKeyVaultSecretProviderClass(name, undefined, { urn });
            case "azure-native:secretsynccontroller:SecretSync":
                return new exports.SecretSync(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "secretsynccontroller", _module);
