// @pulumi/azure-native/maintenance/v20231001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceConfiguration = exports.getMaintenanceConfigurationOutput = exports.getMaintenanceConfiguration = exports.getConfigurationAssignmentsForSubscriptionOutput = exports.getConfigurationAssignmentsForSubscription = exports.getConfigurationAssignmentsForResourceGroupOutput = exports.getConfigurationAssignmentsForResourceGroup = exports.getConfigurationAssignmentParentOutput = exports.getConfigurationAssignmentParent = exports.getConfigurationAssignmentOutput = exports.getConfigurationAssignment = exports.ConfigurationAssignmentsForSubscription = exports.ConfigurationAssignmentsForResourceGroup = exports.ConfigurationAssignmentParent = exports.ConfigurationAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ConfigurationAssignment = null;
utilities.lazyLoad(exports, ["ConfigurationAssignment"], () => require("./configurationAssignment"));
exports.ConfigurationAssignmentParent = null;
utilities.lazyLoad(exports, ["ConfigurationAssignmentParent"], () => require("./configurationAssignmentParent"));
exports.ConfigurationAssignmentsForResourceGroup = null;
utilities.lazyLoad(exports, ["ConfigurationAssignmentsForResourceGroup"], () => require("./configurationAssignmentsForResourceGroup"));
exports.ConfigurationAssignmentsForSubscription = null;
utilities.lazyLoad(exports, ["ConfigurationAssignmentsForSubscription"], () => require("./configurationAssignmentsForSubscription"));
exports.getConfigurationAssignment = null;
exports.getConfigurationAssignmentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationAssignment", "getConfigurationAssignmentOutput"], () => require("./getConfigurationAssignment"));
exports.getConfigurationAssignmentParent = null;
exports.getConfigurationAssignmentParentOutput = null;
utilities.lazyLoad(exports, ["getConfigurationAssignmentParent", "getConfigurationAssignmentParentOutput"], () => require("./getConfigurationAssignmentParent"));
exports.getConfigurationAssignmentsForResourceGroup = null;
exports.getConfigurationAssignmentsForResourceGroupOutput = null;
utilities.lazyLoad(exports, ["getConfigurationAssignmentsForResourceGroup", "getConfigurationAssignmentsForResourceGroupOutput"], () => require("./getConfigurationAssignmentsForResourceGroup"));
exports.getConfigurationAssignmentsForSubscription = null;
exports.getConfigurationAssignmentsForSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getConfigurationAssignmentsForSubscription", "getConfigurationAssignmentsForSubscriptionOutput"], () => require("./getConfigurationAssignmentsForSubscription"));
exports.getMaintenanceConfiguration = null;
exports.getMaintenanceConfigurationOutput = null;
utilities.lazyLoad(exports, ["getMaintenanceConfiguration", "getMaintenanceConfigurationOutput"], () => require("./getMaintenanceConfiguration"));
exports.MaintenanceConfiguration = null;
utilities.lazyLoad(exports, ["MaintenanceConfiguration"], () => require("./maintenanceConfiguration"));
// Export enums:
__exportStar(require("../../types/enums/maintenance/v20231001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:maintenance/v20231001preview:ConfigurationAssignment":
                return new exports.ConfigurationAssignment(name, undefined, { urn });
            case "azure-native:maintenance/v20231001preview:ConfigurationAssignmentParent":
                return new exports.ConfigurationAssignmentParent(name, undefined, { urn });
            case "azure-native:maintenance/v20231001preview:ConfigurationAssignmentsForResourceGroup":
                return new exports.ConfigurationAssignmentsForResourceGroup(name, undefined, { urn });
            case "azure-native:maintenance/v20231001preview:ConfigurationAssignmentsForSubscription":
                return new exports.ConfigurationAssignmentsForSubscription(name, undefined, { urn });
            case "azure-native:maintenance/v20231001preview:MaintenanceConfiguration":
                return new exports.MaintenanceConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "maintenance/v20231001preview", _module);
