// @pulumi/azure-native/m365securityandcompliance/v20210325preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkServicesForSCCPowershell = exports.PrivateLinkServicesForO365ManagementActivityAPI = exports.PrivateLinkServicesForMIPPolicySync = exports.PrivateLinkServicesForM365SecurityCenter = exports.PrivateLinkServicesForM365ComplianceCenter = exports.PrivateLinkServicesForEDMUpload = exports.PrivateEndpointConnectionsSec = exports.PrivateEndpointConnectionsForSCCPowershell = exports.PrivateEndpointConnectionsForMIPPolicySync = exports.PrivateEndpointConnectionsForEDM = exports.PrivateEndpointConnectionsComp = exports.PrivateEndpointConnectionsAdtAPI = exports.getPrivateLinkServicesForSCCPowershellOutput = exports.getPrivateLinkServicesForSCCPowershell = exports.getPrivateLinkServicesForO365ManagementActivityAPIOutput = exports.getPrivateLinkServicesForO365ManagementActivityAPI = exports.getPrivateLinkServicesForMIPPolicySyncOutput = exports.getPrivateLinkServicesForMIPPolicySync = exports.getPrivateLinkServicesForM365SecurityCenterOutput = exports.getPrivateLinkServicesForM365SecurityCenter = exports.getPrivateLinkServicesForM365ComplianceCenterOutput = exports.getPrivateLinkServicesForM365ComplianceCenter = exports.getPrivateLinkServicesForEDMUploadOutput = exports.getPrivateLinkServicesForEDMUpload = exports.getPrivateEndpointConnectionsSecOutput = exports.getPrivateEndpointConnectionsSec = exports.getPrivateEndpointConnectionsForSCCPowershellOutput = exports.getPrivateEndpointConnectionsForSCCPowershell = exports.getPrivateEndpointConnectionsForMIPPolicySyncOutput = exports.getPrivateEndpointConnectionsForMIPPolicySync = exports.getPrivateEndpointConnectionsForEDMOutput = exports.getPrivateEndpointConnectionsForEDM = exports.getPrivateEndpointConnectionsCompOutput = exports.getPrivateEndpointConnectionsComp = exports.getPrivateEndpointConnectionsAdtAPIOutput = exports.getPrivateEndpointConnectionsAdtAPI = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPrivateEndpointConnectionsAdtAPI = null;
exports.getPrivateEndpointConnectionsAdtAPIOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsAdtAPI", "getPrivateEndpointConnectionsAdtAPIOutput"], () => require("./getPrivateEndpointConnectionsAdtAPI"));
exports.getPrivateEndpointConnectionsComp = null;
exports.getPrivateEndpointConnectionsCompOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsComp", "getPrivateEndpointConnectionsCompOutput"], () => require("./getPrivateEndpointConnectionsComp"));
exports.getPrivateEndpointConnectionsForEDM = null;
exports.getPrivateEndpointConnectionsForEDMOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsForEDM", "getPrivateEndpointConnectionsForEDMOutput"], () => require("./getPrivateEndpointConnectionsForEDM"));
exports.getPrivateEndpointConnectionsForMIPPolicySync = null;
exports.getPrivateEndpointConnectionsForMIPPolicySyncOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsForMIPPolicySync", "getPrivateEndpointConnectionsForMIPPolicySyncOutput"], () => require("./getPrivateEndpointConnectionsForMIPPolicySync"));
exports.getPrivateEndpointConnectionsForSCCPowershell = null;
exports.getPrivateEndpointConnectionsForSCCPowershellOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsForSCCPowershell", "getPrivateEndpointConnectionsForSCCPowershellOutput"], () => require("./getPrivateEndpointConnectionsForSCCPowershell"));
exports.getPrivateEndpointConnectionsSec = null;
exports.getPrivateEndpointConnectionsSecOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionsSec", "getPrivateEndpointConnectionsSecOutput"], () => require("./getPrivateEndpointConnectionsSec"));
exports.getPrivateLinkServicesForEDMUpload = null;
exports.getPrivateLinkServicesForEDMUploadOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForEDMUpload", "getPrivateLinkServicesForEDMUploadOutput"], () => require("./getPrivateLinkServicesForEDMUpload"));
exports.getPrivateLinkServicesForM365ComplianceCenter = null;
exports.getPrivateLinkServicesForM365ComplianceCenterOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForM365ComplianceCenter", "getPrivateLinkServicesForM365ComplianceCenterOutput"], () => require("./getPrivateLinkServicesForM365ComplianceCenter"));
exports.getPrivateLinkServicesForM365SecurityCenter = null;
exports.getPrivateLinkServicesForM365SecurityCenterOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForM365SecurityCenter", "getPrivateLinkServicesForM365SecurityCenterOutput"], () => require("./getPrivateLinkServicesForM365SecurityCenter"));
exports.getPrivateLinkServicesForMIPPolicySync = null;
exports.getPrivateLinkServicesForMIPPolicySyncOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForMIPPolicySync", "getPrivateLinkServicesForMIPPolicySyncOutput"], () => require("./getPrivateLinkServicesForMIPPolicySync"));
exports.getPrivateLinkServicesForO365ManagementActivityAPI = null;
exports.getPrivateLinkServicesForO365ManagementActivityAPIOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForO365ManagementActivityAPI", "getPrivateLinkServicesForO365ManagementActivityAPIOutput"], () => require("./getPrivateLinkServicesForO365ManagementActivityAPI"));
exports.getPrivateLinkServicesForSCCPowershell = null;
exports.getPrivateLinkServicesForSCCPowershellOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkServicesForSCCPowershell", "getPrivateLinkServicesForSCCPowershellOutput"], () => require("./getPrivateLinkServicesForSCCPowershell"));
exports.PrivateEndpointConnectionsAdtAPI = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsAdtAPI"], () => require("./privateEndpointConnectionsAdtAPI"));
exports.PrivateEndpointConnectionsComp = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsComp"], () => require("./privateEndpointConnectionsComp"));
exports.PrivateEndpointConnectionsForEDM = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsForEDM"], () => require("./privateEndpointConnectionsForEDM"));
exports.PrivateEndpointConnectionsForMIPPolicySync = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsForMIPPolicySync"], () => require("./privateEndpointConnectionsForMIPPolicySync"));
exports.PrivateEndpointConnectionsForSCCPowershell = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsForSCCPowershell"], () => require("./privateEndpointConnectionsForSCCPowershell"));
exports.PrivateEndpointConnectionsSec = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionsSec"], () => require("./privateEndpointConnectionsSec"));
exports.PrivateLinkServicesForEDMUpload = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForEDMUpload"], () => require("./privateLinkServicesForEDMUpload"));
exports.PrivateLinkServicesForM365ComplianceCenter = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForM365ComplianceCenter"], () => require("./privateLinkServicesForM365ComplianceCenter"));
exports.PrivateLinkServicesForM365SecurityCenter = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForM365SecurityCenter"], () => require("./privateLinkServicesForM365SecurityCenter"));
exports.PrivateLinkServicesForMIPPolicySync = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForMIPPolicySync"], () => require("./privateLinkServicesForMIPPolicySync"));
exports.PrivateLinkServicesForO365ManagementActivityAPI = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForO365ManagementActivityAPI"], () => require("./privateLinkServicesForO365ManagementActivityAPI"));
exports.PrivateLinkServicesForSCCPowershell = null;
utilities.lazyLoad(exports, ["PrivateLinkServicesForSCCPowershell"], () => require("./privateLinkServicesForSCCPowershell"));
// Export enums:
__exportStar(require("../../types/enums/m365securityandcompliance/v20210325preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsAdtAPI":
                return new exports.PrivateEndpointConnectionsAdtAPI(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsComp":
                return new exports.PrivateEndpointConnectionsComp(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsForEDM":
                return new exports.PrivateEndpointConnectionsForEDM(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsForMIPPolicySync":
                return new exports.PrivateEndpointConnectionsForMIPPolicySync(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsForSCCPowershell":
                return new exports.PrivateEndpointConnectionsForSCCPowershell(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateEndpointConnectionsSec":
                return new exports.PrivateEndpointConnectionsSec(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForEDMUpload":
                return new exports.PrivateLinkServicesForEDMUpload(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForM365ComplianceCenter":
                return new exports.PrivateLinkServicesForM365ComplianceCenter(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForM365SecurityCenter":
                return new exports.PrivateLinkServicesForM365SecurityCenter(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForMIPPolicySync":
                return new exports.PrivateLinkServicesForMIPPolicySync(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForO365ManagementActivityAPI":
                return new exports.PrivateLinkServicesForO365ManagementActivityAPI(name, undefined, { urn });
            case "azure-native:m365securityandcompliance/v20210325preview:PrivateLinkServicesForSCCPowershell":
                return new exports.PrivateLinkServicesForSCCPowershell(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "m365securityandcompliance/v20210325preview", _module);
