// @pulumi/azure-native/apimanagement/v20230301preview/workspaceNotificationRecipientEmail.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceNotificationRecipientEmail = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Recipient Email details.
 */
class WorkspaceNotificationRecipientEmail extends pulumi.CustomResource {
    /**
     * Get an existing WorkspaceNotificationRecipientEmail resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspaceNotificationRecipientEmail(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspaceNotificationRecipientEmail.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceNotificationRecipientEmail.__pulumiType;
    }
    /**
     * Create a WorkspaceNotificationRecipientEmail resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.notificationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'notificationName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            if ((!args || args.workspaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceId'");
            }
            resourceInputs["email"] = args ? args.email : undefined;
            resourceInputs["notificationName"] = args ? args.notificationName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["workspaceId"] = args ? args.workspaceId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["email"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20220901preview:WorkspaceNotificationRecipientEmail" }, { type: "azure-native:apimanagement/v20230501preview:WorkspaceNotificationRecipientEmail" }, { type: "azure-native:apimanagement/v20230901preview:WorkspaceNotificationRecipientEmail" }, { type: "azure-native:apimanagement/v20240501:WorkspaceNotificationRecipientEmail" }, { type: "azure-native:apimanagement/v20240601preview:WorkspaceNotificationRecipientEmail" }, { type: "azure-native:apimanagement:WorkspaceNotificationRecipientEmail" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkspaceNotificationRecipientEmail.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkspaceNotificationRecipientEmail = WorkspaceNotificationRecipientEmail;
/** @internal */
WorkspaceNotificationRecipientEmail.__pulumiType = 'azure-native:apimanagement/v20230301preview:WorkspaceNotificationRecipientEmail';
