// @pulumi/azure-native/types/enums/desktopvirtualization/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SessionHostComponentUpdateType = exports.SSOSecretType = exports.ResourceIdentityType = exports.RegistrationTokenOperation = exports.PreferredAppGroupType = exports.PersonalDesktopAssignmentType = exports.Operation = exports.LoadBalancerType = exports.HostpoolPublicNetworkAccess = exports.HostPoolType = exports.DayOfWeek = exports.ApplicationGroupType = void 0;
exports.ApplicationGroupType = {
    RemoteApp: "RemoteApp",
    Desktop: "Desktop",
};
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.HostPoolType = {
    /**
     * Users will be assigned a SessionHost either by administrators (PersonalDesktopAssignmentType = Direct) or upon connecting to the pool (PersonalDesktopAssignmentType = Automatic). They will always be redirected to their assigned SessionHost.
     */
    Personal: "Personal",
    /**
     * Users get a new (random) SessionHost every time it connects to the HostPool.
     */
    Pooled: "Pooled",
    /**
     * Users assign their own machines, load balancing logic remains the same as Personal. PersonalDesktopAssignmentType must be Direct.
     */
    BYODesktop: "BYODesktop",
};
exports.HostpoolPublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    EnabledForSessionHostsOnly: "EnabledForSessionHostsOnly",
    EnabledForClientsOnly: "EnabledForClientsOnly",
};
exports.LoadBalancerType = {
    BreadthFirst: "BreadthFirst",
    DepthFirst: "DepthFirst",
    Persistent: "Persistent",
};
exports.Operation = {
    /**
     * Start the migration.
     */
    Start: "Start",
    /**
     * Revoke the migration.
     */
    Revoke: "Revoke",
    /**
     * Complete the migration.
     */
    Complete: "Complete",
    /**
     * Hide the hostpool.
     */
    Hide: "Hide",
    /**
     * Unhide the hostpool.
     */
    Unhide: "Unhide",
};
exports.PersonalDesktopAssignmentType = {
    Automatic: "Automatic",
    Direct: "Direct",
};
exports.PreferredAppGroupType = {
    None: "None",
    Desktop: "Desktop",
    RailApplications: "RailApplications",
};
exports.RegistrationTokenOperation = {
    Delete: "Delete",
    None: "None",
    Update: "Update",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SSOSecretType = {
    SharedKey: "SharedKey",
    Certificate: "Certificate",
    SharedKeyInKeyVault: "SharedKeyInKeyVault",
    CertificateInKeyVault: "CertificateInKeyVault",
};
exports.SessionHostComponentUpdateType = {
    /**
     * Agent and other agent side components are delivery schedule is controlled by WVD Infra.
     */
    Default: "Default",
    /**
     * TenantAdmin have opted in for Scheduled Component Update feature.
     */
    Scheduled: "Scheduled",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
