// @pulumi/azure-native/deviceupdate/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230701 = exports.PrivateEndpointConnectionProxy = exports.PrivateEndpointConnection = exports.Instance = exports.getPrivateEndpointConnectionProxyOutput = exports.getPrivateEndpointConnectionProxy = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getInstanceOutput = exports.getInstance = exports.getAccountOutput = exports.getAccount = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateEndpointConnectionProxy = null;
exports.getPrivateEndpointConnectionProxyOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionProxy", "getPrivateEndpointConnectionProxyOutput"], () => require("./getPrivateEndpointConnectionProxy"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateEndpointConnectionProxy = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionProxy"], () => require("./privateEndpointConnectionProxy"));
// Export enums:
__exportStar(require("../types/enums/deviceupdate"), exports);
// Export sub-modules:
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:deviceupdate:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:deviceupdate:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "azure-native:deviceupdate:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:deviceupdate:PrivateEndpointConnectionProxy":
                return new exports.PrivateEndpointConnectionProxy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "deviceupdate", _module);
