// @pulumi/azure-native/types/enums/storagecache/v20240701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsernameSource = exports.StorageTargetType = exports.OperationalStateType = exports.NfsAccessRuleScope = exports.NfsAccessRuleAccess = exports.MaintenanceDayOfWeekType = exports.ImportJobAdminStatus = exports.ConflictResolutionMode = exports.CacheIdentityType = exports.AutoExportStatusType = exports.AutoExportJobAdminStatus = exports.AmlFilesystemSquashMode = exports.AmlFilesystemIdentityType = void 0;
exports.AmlFilesystemIdentityType = {
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.AmlFilesystemSquashMode = {
    None: "None",
    RootOnly: "RootOnly",
    All: "All",
};
exports.AutoExportJobAdminStatus = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.AutoExportStatusType = {
    InProgress: "InProgress",
    Disabling: "Disabling",
    Disabled: "Disabled",
    DisableFailed: "DisableFailed",
    Failed: "Failed",
};
exports.CacheIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ConflictResolutionMode = {
    Fail: "Fail",
    Skip: "Skip",
    OverwriteIfDirty: "OverwriteIfDirty",
    OverwriteAlways: "OverwriteAlways",
};
exports.ImportJobAdminStatus = {
    Active: "Active",
    Cancel: "Cancel",
};
exports.MaintenanceDayOfWeekType = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.NfsAccessRuleAccess = {
    No: "no",
    Ro: "ro",
    Rw: "rw",
};
exports.NfsAccessRuleScope = {
    Default: "default",
    Network: "network",
    Host: "host",
};
exports.OperationalStateType = {
    Ready: "Ready",
    Busy: "Busy",
    Suspended: "Suspended",
    Flushing: "Flushing",
};
exports.StorageTargetType = {
    Nfs3: "nfs3",
    Clfs: "clfs",
    Unknown: "unknown",
    BlobNfs: "blobNfs",
};
exports.UsernameSource = {
    AD: "AD",
    LDAP: "LDAP",
    File: "File",
    None: "None",
};
