// @pulumi/azure-native/logic/v20150801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listIntegrationAccountCallbackUrlOutput = exports.listIntegrationAccountCallbackUrl = exports.IntegrationAccountSchema = exports.IntegrationAccountPartner = exports.IntegrationAccountMap = exports.IntegrationAccountCertificate = exports.IntegrationAccountAgreement = exports.IntegrationAccount = exports.getIntegrationAccountSchemaOutput = exports.getIntegrationAccountSchema = exports.getIntegrationAccountPartnerOutput = exports.getIntegrationAccountPartner = exports.getIntegrationAccountMapOutput = exports.getIntegrationAccountMap = exports.getIntegrationAccountCertificateOutput = exports.getIntegrationAccountCertificate = exports.getIntegrationAccountAgreementOutput = exports.getIntegrationAccountAgreement = exports.getIntegrationAccountOutput = exports.getIntegrationAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getIntegrationAccount = null;
exports.getIntegrationAccountOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccount", "getIntegrationAccountOutput"], () => require("./getIntegrationAccount"));
exports.getIntegrationAccountAgreement = null;
exports.getIntegrationAccountAgreementOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccountAgreement", "getIntegrationAccountAgreementOutput"], () => require("./getIntegrationAccountAgreement"));
exports.getIntegrationAccountCertificate = null;
exports.getIntegrationAccountCertificateOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccountCertificate", "getIntegrationAccountCertificateOutput"], () => require("./getIntegrationAccountCertificate"));
exports.getIntegrationAccountMap = null;
exports.getIntegrationAccountMapOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccountMap", "getIntegrationAccountMapOutput"], () => require("./getIntegrationAccountMap"));
exports.getIntegrationAccountPartner = null;
exports.getIntegrationAccountPartnerOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccountPartner", "getIntegrationAccountPartnerOutput"], () => require("./getIntegrationAccountPartner"));
exports.getIntegrationAccountSchema = null;
exports.getIntegrationAccountSchemaOutput = null;
utilities.lazyLoad(exports, ["getIntegrationAccountSchema", "getIntegrationAccountSchemaOutput"], () => require("./getIntegrationAccountSchema"));
exports.IntegrationAccount = null;
utilities.lazyLoad(exports, ["IntegrationAccount"], () => require("./integrationAccount"));
exports.IntegrationAccountAgreement = null;
utilities.lazyLoad(exports, ["IntegrationAccountAgreement"], () => require("./integrationAccountAgreement"));
exports.IntegrationAccountCertificate = null;
utilities.lazyLoad(exports, ["IntegrationAccountCertificate"], () => require("./integrationAccountCertificate"));
exports.IntegrationAccountMap = null;
utilities.lazyLoad(exports, ["IntegrationAccountMap"], () => require("./integrationAccountMap"));
exports.IntegrationAccountPartner = null;
utilities.lazyLoad(exports, ["IntegrationAccountPartner"], () => require("./integrationAccountPartner"));
exports.IntegrationAccountSchema = null;
utilities.lazyLoad(exports, ["IntegrationAccountSchema"], () => require("./integrationAccountSchema"));
exports.listIntegrationAccountCallbackUrl = null;
exports.listIntegrationAccountCallbackUrlOutput = null;
utilities.lazyLoad(exports, ["listIntegrationAccountCallbackUrl", "listIntegrationAccountCallbackUrlOutput"], () => require("./listIntegrationAccountCallbackUrl"));
// Export enums:
__exportStar(require("../../types/enums/logic/v20150801preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:logic/v20150801preview:IntegrationAccount":
                return new exports.IntegrationAccount(name, undefined, { urn });
            case "azure-native:logic/v20150801preview:IntegrationAccountAgreement":
                return new exports.IntegrationAccountAgreement(name, undefined, { urn });
            case "azure-native:logic/v20150801preview:IntegrationAccountCertificate":
                return new exports.IntegrationAccountCertificate(name, undefined, { urn });
            case "azure-native:logic/v20150801preview:IntegrationAccountMap":
                return new exports.IntegrationAccountMap(name, undefined, { urn });
            case "azure-native:logic/v20150801preview:IntegrationAccountPartner":
                return new exports.IntegrationAccountPartner(name, undefined, { urn });
            case "azure-native:logic/v20150801preview:IntegrationAccountSchema":
                return new exports.IntegrationAccountSchema(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "logic/v20150801preview", _module);
