// @pulumi/azure-native/types/enums/elastic/v20250115preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagAction = exports.Status = exports.ProjectType = exports.Operation = exports.MonitoringStatus = exports.ManagedIdentityTypes = exports.HostingType = exports.ConfigurationType = void 0;
exports.ConfigurationType = {
    GeneralPurpose: "GeneralPurpose",
    Vector: "Vector",
    TimeSeries: "TimeSeries",
    NotApplicable: "NotApplicable",
};
exports.HostingType = {
    Hosted: "Hosted",
    Serverless: "Serverless",
};
exports.ManagedIdentityTypes = {
    SystemAssigned: "SystemAssigned",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Operation = {
    AddBegin: "AddBegin",
    AddComplete: "AddComplete",
    DeleteBegin: "DeleteBegin",
    DeleteComplete: "DeleteComplete",
    Active: "Active",
};
exports.ProjectType = {
    Elasticsearch: "Elasticsearch",
    Observability: "Observability",
    Security: "Security",
    NotApplicable: "NotApplicable",
};
exports.Status = {
    InProgress: "InProgress",
    Active: "Active",
    Failed: "Failed",
    Deleting: "Deleting",
};
exports.TagAction = {
    Include: "Include",
    Exclude: "Exclude",
};
