// @pulumi/azure-native/synapse/v20210601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPoolSensitivityLabel = exports.SqlPool = exports.PrivateLinkHub = exports.PrivateEndpointConnection = exports.listIntegrationRuntimeAuthKeyOutput = exports.listIntegrationRuntimeAuthKey = exports.Key = exports.IpFirewallRule = exports.IntegrationRuntime = exports.getWorkspaceSqlAadAdminOutput = exports.getWorkspaceSqlAadAdmin = exports.getWorkspaceManagedSqlServerVulnerabilityAssessmentOutput = exports.getWorkspaceManagedSqlServerVulnerabilityAssessment = exports.getWorkspaceAadAdminOutput = exports.getWorkspaceAadAdmin = exports.getWorkspaceOutput = exports.getWorkspace = exports.getSqlPoolWorkloadGroupOutput = exports.getSqlPoolWorkloadGroup = exports.getSqlPoolWorkloadClassifierOutput = exports.getSqlPoolWorkloadClassifier = exports.getSqlPoolVulnerabilityAssessmentRuleBaselineOutput = exports.getSqlPoolVulnerabilityAssessmentRuleBaseline = exports.getSqlPoolVulnerabilityAssessmentOutput = exports.getSqlPoolVulnerabilityAssessment = exports.getSqlPoolTransparentDataEncryptionOutput = exports.getSqlPoolTransparentDataEncryption = exports.getSqlPoolSensitivityLabelOutput = exports.getSqlPoolSensitivityLabel = exports.getSqlPoolOutput = exports.getSqlPool = exports.getPrivateLinkHubOutput = exports.getPrivateLinkHub = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getKeyOutput = exports.getKey = exports.getIpFirewallRuleOutput = exports.getIpFirewallRule = exports.getIntegrationRuntimeStatusOutput = exports.getIntegrationRuntimeStatus = exports.getIntegrationRuntimeObjectMetadatumOutput = exports.getIntegrationRuntimeObjectMetadatum = exports.getIntegrationRuntimeConnectionInfoOutput = exports.getIntegrationRuntimeConnectionInfo = exports.getIntegrationRuntimeOutput = exports.getIntegrationRuntime = exports.getBigDataPoolOutput = exports.getBigDataPool = exports.BigDataPool = void 0;
exports.WorkspaceSqlAadAdmin = exports.WorkspaceManagedSqlServerVulnerabilityAssessment = exports.WorkspaceAadAdmin = exports.Workspace = exports.SqlPoolWorkloadGroup = exports.SqlPoolWorkloadClassifier = exports.SqlPoolVulnerabilityAssessmentRuleBaseline = exports.SqlPoolVulnerabilityAssessment = exports.SqlPoolTransparentDataEncryption = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BigDataPool = null;
utilities.lazyLoad(exports, ["BigDataPool"], () => require("./bigDataPool"));
exports.getBigDataPool = null;
exports.getBigDataPoolOutput = null;
utilities.lazyLoad(exports, ["getBigDataPool", "getBigDataPoolOutput"], () => require("./getBigDataPool"));
exports.getIntegrationRuntime = null;
exports.getIntegrationRuntimeOutput = null;
utilities.lazyLoad(exports, ["getIntegrationRuntime", "getIntegrationRuntimeOutput"], () => require("./getIntegrationRuntime"));
exports.getIntegrationRuntimeConnectionInfo = null;
exports.getIntegrationRuntimeConnectionInfoOutput = null;
utilities.lazyLoad(exports, ["getIntegrationRuntimeConnectionInfo", "getIntegrationRuntimeConnectionInfoOutput"], () => require("./getIntegrationRuntimeConnectionInfo"));
exports.getIntegrationRuntimeObjectMetadatum = null;
exports.getIntegrationRuntimeObjectMetadatumOutput = null;
utilities.lazyLoad(exports, ["getIntegrationRuntimeObjectMetadatum", "getIntegrationRuntimeObjectMetadatumOutput"], () => require("./getIntegrationRuntimeObjectMetadatum"));
exports.getIntegrationRuntimeStatus = null;
exports.getIntegrationRuntimeStatusOutput = null;
utilities.lazyLoad(exports, ["getIntegrationRuntimeStatus", "getIntegrationRuntimeStatusOutput"], () => require("./getIntegrationRuntimeStatus"));
exports.getIpFirewallRule = null;
exports.getIpFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getIpFirewallRule", "getIpFirewallRuleOutput"], () => require("./getIpFirewallRule"));
exports.getKey = null;
exports.getKeyOutput = null;
utilities.lazyLoad(exports, ["getKey", "getKeyOutput"], () => require("./getKey"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkHub = null;
exports.getPrivateLinkHubOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkHub", "getPrivateLinkHubOutput"], () => require("./getPrivateLinkHub"));
exports.getSqlPool = null;
exports.getSqlPoolOutput = null;
utilities.lazyLoad(exports, ["getSqlPool", "getSqlPoolOutput"], () => require("./getSqlPool"));
exports.getSqlPoolSensitivityLabel = null;
exports.getSqlPoolSensitivityLabelOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolSensitivityLabel", "getSqlPoolSensitivityLabelOutput"], () => require("./getSqlPoolSensitivityLabel"));
exports.getSqlPoolTransparentDataEncryption = null;
exports.getSqlPoolTransparentDataEncryptionOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolTransparentDataEncryption", "getSqlPoolTransparentDataEncryptionOutput"], () => require("./getSqlPoolTransparentDataEncryption"));
exports.getSqlPoolVulnerabilityAssessment = null;
exports.getSqlPoolVulnerabilityAssessmentOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolVulnerabilityAssessment", "getSqlPoolVulnerabilityAssessmentOutput"], () => require("./getSqlPoolVulnerabilityAssessment"));
exports.getSqlPoolVulnerabilityAssessmentRuleBaseline = null;
exports.getSqlPoolVulnerabilityAssessmentRuleBaselineOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolVulnerabilityAssessmentRuleBaseline", "getSqlPoolVulnerabilityAssessmentRuleBaselineOutput"], () => require("./getSqlPoolVulnerabilityAssessmentRuleBaseline"));
exports.getSqlPoolWorkloadClassifier = null;
exports.getSqlPoolWorkloadClassifierOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolWorkloadClassifier", "getSqlPoolWorkloadClassifierOutput"], () => require("./getSqlPoolWorkloadClassifier"));
exports.getSqlPoolWorkloadGroup = null;
exports.getSqlPoolWorkloadGroupOutput = null;
utilities.lazyLoad(exports, ["getSqlPoolWorkloadGroup", "getSqlPoolWorkloadGroupOutput"], () => require("./getSqlPoolWorkloadGroup"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.getWorkspaceAadAdmin = null;
exports.getWorkspaceAadAdminOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceAadAdmin", "getWorkspaceAadAdminOutput"], () => require("./getWorkspaceAadAdmin"));
exports.getWorkspaceManagedSqlServerVulnerabilityAssessment = null;
exports.getWorkspaceManagedSqlServerVulnerabilityAssessmentOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceManagedSqlServerVulnerabilityAssessment", "getWorkspaceManagedSqlServerVulnerabilityAssessmentOutput"], () => require("./getWorkspaceManagedSqlServerVulnerabilityAssessment"));
exports.getWorkspaceSqlAadAdmin = null;
exports.getWorkspaceSqlAadAdminOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceSqlAadAdmin", "getWorkspaceSqlAadAdminOutput"], () => require("./getWorkspaceSqlAadAdmin"));
exports.IntegrationRuntime = null;
utilities.lazyLoad(exports, ["IntegrationRuntime"], () => require("./integrationRuntime"));
exports.IpFirewallRule = null;
utilities.lazyLoad(exports, ["IpFirewallRule"], () => require("./ipFirewallRule"));
exports.Key = null;
utilities.lazyLoad(exports, ["Key"], () => require("./key"));
exports.listIntegrationRuntimeAuthKey = null;
exports.listIntegrationRuntimeAuthKeyOutput = null;
utilities.lazyLoad(exports, ["listIntegrationRuntimeAuthKey", "listIntegrationRuntimeAuthKeyOutput"], () => require("./listIntegrationRuntimeAuthKey"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkHub = null;
utilities.lazyLoad(exports, ["PrivateLinkHub"], () => require("./privateLinkHub"));
exports.SqlPool = null;
utilities.lazyLoad(exports, ["SqlPool"], () => require("./sqlPool"));
exports.SqlPoolSensitivityLabel = null;
utilities.lazyLoad(exports, ["SqlPoolSensitivityLabel"], () => require("./sqlPoolSensitivityLabel"));
exports.SqlPoolTransparentDataEncryption = null;
utilities.lazyLoad(exports, ["SqlPoolTransparentDataEncryption"], () => require("./sqlPoolTransparentDataEncryption"));
exports.SqlPoolVulnerabilityAssessment = null;
utilities.lazyLoad(exports, ["SqlPoolVulnerabilityAssessment"], () => require("./sqlPoolVulnerabilityAssessment"));
exports.SqlPoolVulnerabilityAssessmentRuleBaseline = null;
utilities.lazyLoad(exports, ["SqlPoolVulnerabilityAssessmentRuleBaseline"], () => require("./sqlPoolVulnerabilityAssessmentRuleBaseline"));
exports.SqlPoolWorkloadClassifier = null;
utilities.lazyLoad(exports, ["SqlPoolWorkloadClassifier"], () => require("./sqlPoolWorkloadClassifier"));
exports.SqlPoolWorkloadGroup = null;
utilities.lazyLoad(exports, ["SqlPoolWorkloadGroup"], () => require("./sqlPoolWorkloadGroup"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
exports.WorkspaceAadAdmin = null;
utilities.lazyLoad(exports, ["WorkspaceAadAdmin"], () => require("./workspaceAadAdmin"));
exports.WorkspaceManagedSqlServerVulnerabilityAssessment = null;
utilities.lazyLoad(exports, ["WorkspaceManagedSqlServerVulnerabilityAssessment"], () => require("./workspaceManagedSqlServerVulnerabilityAssessment"));
exports.WorkspaceSqlAadAdmin = null;
utilities.lazyLoad(exports, ["WorkspaceSqlAadAdmin"], () => require("./workspaceSqlAadAdmin"));
// Export enums:
__exportStar(require("../../types/enums/synapse/v20210601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:synapse/v20210601:BigDataPool":
                return new exports.BigDataPool(name, undefined, { urn });
            case "azure-native:synapse/v20210601:IntegrationRuntime":
                return new exports.IntegrationRuntime(name, undefined, { urn });
            case "azure-native:synapse/v20210601:IpFirewallRule":
                return new exports.IpFirewallRule(name, undefined, { urn });
            case "azure-native:synapse/v20210601:Key":
                return new exports.Key(name, undefined, { urn });
            case "azure-native:synapse/v20210601:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:synapse/v20210601:PrivateLinkHub":
                return new exports.PrivateLinkHub(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPool":
                return new exports.SqlPool(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolSensitivityLabel":
                return new exports.SqlPoolSensitivityLabel(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolTransparentDataEncryption":
                return new exports.SqlPoolTransparentDataEncryption(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolVulnerabilityAssessment":
                return new exports.SqlPoolVulnerabilityAssessment(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolVulnerabilityAssessmentRuleBaseline":
                return new exports.SqlPoolVulnerabilityAssessmentRuleBaseline(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolWorkloadClassifier":
                return new exports.SqlPoolWorkloadClassifier(name, undefined, { urn });
            case "azure-native:synapse/v20210601:SqlPoolWorkloadGroup":
                return new exports.SqlPoolWorkloadGroup(name, undefined, { urn });
            case "azure-native:synapse/v20210601:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            case "azure-native:synapse/v20210601:WorkspaceAadAdmin":
                return new exports.WorkspaceAadAdmin(name, undefined, { urn });
            case "azure-native:synapse/v20210601:WorkspaceManagedSqlServerVulnerabilityAssessment":
                return new exports.WorkspaceManagedSqlServerVulnerabilityAssessment(name, undefined, { urn });
            case "azure-native:synapse/v20210601:WorkspaceSqlAadAdmin":
                return new exports.WorkspaceSqlAadAdmin(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "synapse/v20210601", _module);
