// @pulumi/azure-native/elasticsan/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240601preview = exports.v20240501 = exports.v20230101 = exports.v20221201preview = exports.v20211120preview = exports.VolumeSnapshot = exports.VolumeGroup = exports.Volume = exports.PrivateEndpointConnection = exports.getVolumeSnapshotOutput = exports.getVolumeSnapshot = exports.getVolumeGroupOutput = exports.getVolumeGroup = exports.getVolumeOutput = exports.getVolume = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getElasticSanOutput = exports.getElasticSan = exports.ElasticSan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ElasticSan = null;
utilities.lazyLoad(exports, ["ElasticSan"], () => require("./elasticSan"));
exports.getElasticSan = null;
exports.getElasticSanOutput = null;
utilities.lazyLoad(exports, ["getElasticSan", "getElasticSanOutput"], () => require("./getElasticSan"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.getVolumeGroup = null;
exports.getVolumeGroupOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroup", "getVolumeGroupOutput"], () => require("./getVolumeGroup"));
exports.getVolumeSnapshot = null;
exports.getVolumeSnapshotOutput = null;
utilities.lazyLoad(exports, ["getVolumeSnapshot", "getVolumeSnapshotOutput"], () => require("./getVolumeSnapshot"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
exports.VolumeGroup = null;
utilities.lazyLoad(exports, ["VolumeGroup"], () => require("./volumeGroup"));
exports.VolumeSnapshot = null;
utilities.lazyLoad(exports, ["VolumeSnapshot"], () => require("./volumeSnapshot"));
// Export enums:
__exportStar(require("../types/enums/elasticsan"), exports);
// Export sub-modules:
const v20211120preview = require("./v20211120preview");
exports.v20211120preview = v20211120preview;
const v20221201preview = require("./v20221201preview");
exports.v20221201preview = v20221201preview;
const v20230101 = require("./v20230101");
exports.v20230101 = v20230101;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:elasticsan:ElasticSan":
                return new exports.ElasticSan(name, undefined, { urn });
            case "azure-native:elasticsan:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:elasticsan:Volume":
                return new exports.Volume(name, undefined, { urn });
            case "azure-native:elasticsan:VolumeGroup":
                return new exports.VolumeGroup(name, undefined, { urn });
            case "azure-native:elasticsan:VolumeSnapshot":
                return new exports.VolumeSnapshot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "elasticsan", _module);
