// @pulumi/azure-native/types/enums/scvmm/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningAction = exports.LimitCpuForMigration = exports.InventoryType = exports.IdentityType = exports.DynamicMemoryEnabled = exports.CreateDiffDisk = exports.AllocationMethod = exports.v20240601 = exports.v20231007 = exports.v20230401preview = exports.v20220521preview = void 0;
// Export sub-modules:
const v20220521preview = require("./v20220521preview");
exports.v20220521preview = v20220521preview;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20231007 = require("./v20231007");
exports.v20231007 = v20231007;
const v20240601 = require("./v20240601");
exports.v20240601 = v20240601;
exports.AllocationMethod = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.CreateDiffDisk = {
    False: "false",
    True: "true",
};
exports.DynamicMemoryEnabled = {
    False: "false",
    True: "true",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.InventoryType = {
    Cloud: "Cloud",
    VirtualNetwork: "VirtualNetwork",
    VirtualMachineTemplate: "VirtualMachineTemplate",
    VirtualMachine: "VirtualMachine",
};
exports.LimitCpuForMigration = {
    False: "false",
    True: "true",
};
exports.ProvisioningAction = {
    Install: "install",
    Uninstall: "uninstall",
    Repair: "repair",
};
