// @pulumi/azure-native/dbforpostgresql/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualEndpoint = exports.Server = exports.PrivateEndpointConnection = exports.Migration = exports.getVirtualEndpointOutput = exports.getVirtualEndpoint = exports.getServerOutput = exports.getServer = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getMigrationOutput = exports.getMigration = exports.getGetPrivateDnsZoneSuffixExecuteOutput = exports.getGetPrivateDnsZoneSuffixExecute = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getDatabaseOutput = exports.getDatabase = exports.getConfigurationOutput = exports.getConfiguration = exports.getBackupOutput = exports.getBackup = exports.getAdministratorOutput = exports.getAdministrator = exports.FirewallRule = exports.Database = exports.Configuration = exports.Backup = exports.Administrator = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Administrator = null;
utilities.lazyLoad(exports, ["Administrator"], () => require("./administrator"));
exports.Backup = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getAdministrator = null;
exports.getAdministratorOutput = null;
utilities.lazyLoad(exports, ["getAdministrator", "getAdministratorOutput"], () => require("./getAdministrator"));
exports.getBackup = null;
exports.getBackupOutput = null;
utilities.lazyLoad(exports, ["getBackup", "getBackupOutput"], () => require("./getBackup"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getGetPrivateDnsZoneSuffixExecute = null;
exports.getGetPrivateDnsZoneSuffixExecuteOutput = null;
utilities.lazyLoad(exports, ["getGetPrivateDnsZoneSuffixExecute", "getGetPrivateDnsZoneSuffixExecuteOutput"], () => require("./getGetPrivateDnsZoneSuffixExecute"));
exports.getMigration = null;
exports.getMigrationOutput = null;
utilities.lazyLoad(exports, ["getMigration", "getMigrationOutput"], () => require("./getMigration"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getServer = null;
exports.getServerOutput = null;
utilities.lazyLoad(exports, ["getServer", "getServerOutput"], () => require("./getServer"));
exports.getVirtualEndpoint = null;
exports.getVirtualEndpointOutput = null;
utilities.lazyLoad(exports, ["getVirtualEndpoint", "getVirtualEndpointOutput"], () => require("./getVirtualEndpoint"));
exports.Migration = null;
utilities.lazyLoad(exports, ["Migration"], () => require("./migration"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Server = null;
utilities.lazyLoad(exports, ["Server"], () => require("./server"));
exports.VirtualEndpoint = null;
utilities.lazyLoad(exports, ["VirtualEndpoint"], () => require("./virtualEndpoint"));
// Export enums:
__exportStar(require("../../types/enums/dbforpostgresql/v20241101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbforpostgresql/v20241101preview:Administrator":
                return new exports.Administrator(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:Database":
                return new exports.Database(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:Migration":
                return new exports.Migration(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:Server":
                return new exports.Server(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20241101preview:VirtualEndpoint":
                return new exports.VirtualEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbforpostgresql/v20241101preview", _module);
