// @pulumi/azure-native/kusto/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReadOnlyFollowingDatabase = exports.PrivateEndpointConnection = exports.ManagedPrivateEndpoint = exports.listDatabasePrincipalsOutput = exports.listDatabasePrincipals = exports.listClusterLanguageExtensionsOutput = exports.listClusterLanguageExtensions = exports.listClusterFollowerDatabasesOutput = exports.listClusterFollowerDatabases = exports.listClusterCalloutPoliciesOutput = exports.listClusterCalloutPolicies = exports.IotHubDataConnection = exports.getScriptOutput = exports.getScript = exports.getSandboxCustomImageOutput = exports.getSandboxCustomImage = exports.getReadWriteDatabaseOutput = exports.getReadWriteDatabase = exports.getReadOnlyFollowingDatabaseOutput = exports.getReadOnlyFollowingDatabase = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getManagedPrivateEndpointOutput = exports.getManagedPrivateEndpoint = exports.getIotHubDataConnectionOutput = exports.getIotHubDataConnection = exports.getEventHubDataConnectionOutput = exports.getEventHubDataConnection = exports.getEventHubConnectionOutput = exports.getEventHubConnection = exports.getEventGridDataConnectionOutput = exports.getEventGridDataConnection = exports.getDatabasePrincipalAssignmentOutput = exports.getDatabasePrincipalAssignment = exports.getCosmosDbDataConnectionOutput = exports.getCosmosDbDataConnection = exports.getClusterPrincipalAssignmentOutput = exports.getClusterPrincipalAssignment = exports.getClusterOutput = exports.getCluster = exports.getAttachedDatabaseConfigurationOutput = exports.getAttachedDatabaseConfiguration = exports.EventHubDataConnection = exports.EventHubConnection = exports.EventGridDataConnection = exports.DatabasePrincipalAssignment = exports.CosmosDbDataConnection = exports.ClusterPrincipalAssignment = exports.Cluster = exports.AttachedDatabaseConfiguration = void 0;
exports.v20240413 = exports.v20230815 = exports.v20230502 = exports.v20221229 = exports.v20220707 = exports.v20210827 = exports.v20200215 = exports.v20191109 = exports.v20190907 = exports.v20190515 = exports.v20180907preview = exports.Script = exports.SandboxCustomImage = exports.ReadWriteDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AttachedDatabaseConfiguration = null;
utilities.lazyLoad(exports, ["AttachedDatabaseConfiguration"], () => require("./attachedDatabaseConfiguration"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterPrincipalAssignment = null;
utilities.lazyLoad(exports, ["ClusterPrincipalAssignment"], () => require("./clusterPrincipalAssignment"));
exports.CosmosDbDataConnection = null;
utilities.lazyLoad(exports, ["CosmosDbDataConnection"], () => require("./cosmosDbDataConnection"));
exports.DatabasePrincipalAssignment = null;
utilities.lazyLoad(exports, ["DatabasePrincipalAssignment"], () => require("./databasePrincipalAssignment"));
exports.EventGridDataConnection = null;
utilities.lazyLoad(exports, ["EventGridDataConnection"], () => require("./eventGridDataConnection"));
exports.EventHubConnection = null;
utilities.lazyLoad(exports, ["EventHubConnection"], () => require("./eventHubConnection"));
exports.EventHubDataConnection = null;
utilities.lazyLoad(exports, ["EventHubDataConnection"], () => require("./eventHubDataConnection"));
exports.getAttachedDatabaseConfiguration = null;
exports.getAttachedDatabaseConfigurationOutput = null;
utilities.lazyLoad(exports, ["getAttachedDatabaseConfiguration", "getAttachedDatabaseConfigurationOutput"], () => require("./getAttachedDatabaseConfiguration"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterPrincipalAssignment = null;
exports.getClusterPrincipalAssignmentOutput = null;
utilities.lazyLoad(exports, ["getClusterPrincipalAssignment", "getClusterPrincipalAssignmentOutput"], () => require("./getClusterPrincipalAssignment"));
exports.getCosmosDbDataConnection = null;
exports.getCosmosDbDataConnectionOutput = null;
utilities.lazyLoad(exports, ["getCosmosDbDataConnection", "getCosmosDbDataConnectionOutput"], () => require("./getCosmosDbDataConnection"));
exports.getDatabasePrincipalAssignment = null;
exports.getDatabasePrincipalAssignmentOutput = null;
utilities.lazyLoad(exports, ["getDatabasePrincipalAssignment", "getDatabasePrincipalAssignmentOutput"], () => require("./getDatabasePrincipalAssignment"));
exports.getEventGridDataConnection = null;
exports.getEventGridDataConnectionOutput = null;
utilities.lazyLoad(exports, ["getEventGridDataConnection", "getEventGridDataConnectionOutput"], () => require("./getEventGridDataConnection"));
exports.getEventHubConnection = null;
exports.getEventHubConnectionOutput = null;
utilities.lazyLoad(exports, ["getEventHubConnection", "getEventHubConnectionOutput"], () => require("./getEventHubConnection"));
exports.getEventHubDataConnection = null;
exports.getEventHubDataConnectionOutput = null;
utilities.lazyLoad(exports, ["getEventHubDataConnection", "getEventHubDataConnectionOutput"], () => require("./getEventHubDataConnection"));
exports.getIotHubDataConnection = null;
exports.getIotHubDataConnectionOutput = null;
utilities.lazyLoad(exports, ["getIotHubDataConnection", "getIotHubDataConnectionOutput"], () => require("./getIotHubDataConnection"));
exports.getManagedPrivateEndpoint = null;
exports.getManagedPrivateEndpointOutput = null;
utilities.lazyLoad(exports, ["getManagedPrivateEndpoint", "getManagedPrivateEndpointOutput"], () => require("./getManagedPrivateEndpoint"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getReadOnlyFollowingDatabase = null;
exports.getReadOnlyFollowingDatabaseOutput = null;
utilities.lazyLoad(exports, ["getReadOnlyFollowingDatabase", "getReadOnlyFollowingDatabaseOutput"], () => require("./getReadOnlyFollowingDatabase"));
exports.getReadWriteDatabase = null;
exports.getReadWriteDatabaseOutput = null;
utilities.lazyLoad(exports, ["getReadWriteDatabase", "getReadWriteDatabaseOutput"], () => require("./getReadWriteDatabase"));
exports.getSandboxCustomImage = null;
exports.getSandboxCustomImageOutput = null;
utilities.lazyLoad(exports, ["getSandboxCustomImage", "getSandboxCustomImageOutput"], () => require("./getSandboxCustomImage"));
exports.getScript = null;
exports.getScriptOutput = null;
utilities.lazyLoad(exports, ["getScript", "getScriptOutput"], () => require("./getScript"));
exports.IotHubDataConnection = null;
utilities.lazyLoad(exports, ["IotHubDataConnection"], () => require("./iotHubDataConnection"));
exports.listClusterCalloutPolicies = null;
exports.listClusterCalloutPoliciesOutput = null;
utilities.lazyLoad(exports, ["listClusterCalloutPolicies", "listClusterCalloutPoliciesOutput"], () => require("./listClusterCalloutPolicies"));
exports.listClusterFollowerDatabases = null;
exports.listClusterFollowerDatabasesOutput = null;
utilities.lazyLoad(exports, ["listClusterFollowerDatabases", "listClusterFollowerDatabasesOutput"], () => require("./listClusterFollowerDatabases"));
exports.listClusterLanguageExtensions = null;
exports.listClusterLanguageExtensionsOutput = null;
utilities.lazyLoad(exports, ["listClusterLanguageExtensions", "listClusterLanguageExtensionsOutput"], () => require("./listClusterLanguageExtensions"));
exports.listDatabasePrincipals = null;
exports.listDatabasePrincipalsOutput = null;
utilities.lazyLoad(exports, ["listDatabasePrincipals", "listDatabasePrincipalsOutput"], () => require("./listDatabasePrincipals"));
exports.ManagedPrivateEndpoint = null;
utilities.lazyLoad(exports, ["ManagedPrivateEndpoint"], () => require("./managedPrivateEndpoint"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.ReadOnlyFollowingDatabase = null;
utilities.lazyLoad(exports, ["ReadOnlyFollowingDatabase"], () => require("./readOnlyFollowingDatabase"));
exports.ReadWriteDatabase = null;
utilities.lazyLoad(exports, ["ReadWriteDatabase"], () => require("./readWriteDatabase"));
exports.SandboxCustomImage = null;
utilities.lazyLoad(exports, ["SandboxCustomImage"], () => require("./sandboxCustomImage"));
exports.Script = null;
utilities.lazyLoad(exports, ["Script"], () => require("./script"));
// Export enums:
__exportStar(require("../types/enums/kusto"), exports);
// Export sub-modules:
const v20180907preview = require("./v20180907preview");
exports.v20180907preview = v20180907preview;
const v20190515 = require("./v20190515");
exports.v20190515 = v20190515;
const v20190907 = require("./v20190907");
exports.v20190907 = v20190907;
const v20191109 = require("./v20191109");
exports.v20191109 = v20191109;
const v20200215 = require("./v20200215");
exports.v20200215 = v20200215;
const v20210827 = require("./v20210827");
exports.v20210827 = v20210827;
const v20220707 = require("./v20220707");
exports.v20220707 = v20220707;
const v20221229 = require("./v20221229");
exports.v20221229 = v20221229;
const v20230502 = require("./v20230502");
exports.v20230502 = v20230502;
const v20230815 = require("./v20230815");
exports.v20230815 = v20230815;
const v20240413 = require("./v20240413");
exports.v20240413 = v20240413;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kusto:AttachedDatabaseConfiguration":
                return new exports.AttachedDatabaseConfiguration(name, undefined, { urn });
            case "azure-native:kusto:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:kusto:ClusterPrincipalAssignment":
                return new exports.ClusterPrincipalAssignment(name, undefined, { urn });
            case "azure-native:kusto:CosmosDbDataConnection":
                return new exports.CosmosDbDataConnection(name, undefined, { urn });
            case "azure-native:kusto:DatabasePrincipalAssignment":
                return new exports.DatabasePrincipalAssignment(name, undefined, { urn });
            case "azure-native:kusto:EventGridDataConnection":
                return new exports.EventGridDataConnection(name, undefined, { urn });
            case "azure-native:kusto:EventHubConnection":
                return new exports.EventHubConnection(name, undefined, { urn });
            case "azure-native:kusto:EventHubDataConnection":
                return new exports.EventHubDataConnection(name, undefined, { urn });
            case "azure-native:kusto:IotHubDataConnection":
                return new exports.IotHubDataConnection(name, undefined, { urn });
            case "azure-native:kusto:ManagedPrivateEndpoint":
                return new exports.ManagedPrivateEndpoint(name, undefined, { urn });
            case "azure-native:kusto:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:kusto:ReadOnlyFollowingDatabase":
                return new exports.ReadOnlyFollowingDatabase(name, undefined, { urn });
            case "azure-native:kusto:ReadWriteDatabase":
                return new exports.ReadWriteDatabase(name, undefined, { urn });
            case "azure-native:kusto:SandboxCustomImage":
                return new exports.SandboxCustomImage(name, undefined, { urn });
            case "azure-native:kusto:Script":
                return new exports.Script(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kusto", _module);
