// @pulumi/azure-native/dashboard/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241001 = exports.v20231001preview = exports.v20230901 = exports.v20221001preview = exports.v20220801 = exports.v20210901preview = exports.PrivateEndpointConnection = exports.ManagedPrivateEndpoint = exports.IntegrationFabric = exports.Grafana = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getManagedPrivateEndpointOutput = exports.getManagedPrivateEndpoint = exports.getIntegrationFabricOutput = exports.getIntegrationFabric = exports.getGrafanaOutput = exports.getGrafana = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGrafana = null;
exports.getGrafanaOutput = null;
utilities.lazyLoad(exports, ["getGrafana", "getGrafanaOutput"], () => require("./getGrafana"));
exports.getIntegrationFabric = null;
exports.getIntegrationFabricOutput = null;
utilities.lazyLoad(exports, ["getIntegrationFabric", "getIntegrationFabricOutput"], () => require("./getIntegrationFabric"));
exports.getManagedPrivateEndpoint = null;
exports.getManagedPrivateEndpointOutput = null;
utilities.lazyLoad(exports, ["getManagedPrivateEndpoint", "getManagedPrivateEndpointOutput"], () => require("./getManagedPrivateEndpoint"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.Grafana = null;
utilities.lazyLoad(exports, ["Grafana"], () => require("./grafana"));
exports.IntegrationFabric = null;
utilities.lazyLoad(exports, ["IntegrationFabric"], () => require("./integrationFabric"));
exports.ManagedPrivateEndpoint = null;
utilities.lazyLoad(exports, ["ManagedPrivateEndpoint"], () => require("./managedPrivateEndpoint"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
// Export enums:
__exportStar(require("../types/enums/dashboard"), exports);
// Export sub-modules:
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20220801 = require("./v20220801");
exports.v20220801 = v20220801;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20241001 = require("./v20241001");
exports.v20241001 = v20241001;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dashboard:Grafana":
                return new exports.Grafana(name, undefined, { urn });
            case "azure-native:dashboard:IntegrationFabric":
                return new exports.IntegrationFabric(name, undefined, { urn });
            case "azure-native:dashboard:ManagedPrivateEndpoint":
                return new exports.ManagedPrivateEndpoint(name, undefined, { urn });
            case "azure-native:dashboard:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dashboard", _module);
