// @pulumi/azure-native/web/v20150801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listConnectionKeysOutput = exports.listConnectionKeys = exports.listConnectionConsentLinksOutput = exports.listConnectionConsentLinks = exports.getConnectionOutput = exports.getConnection = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.listConnectionConsentLinks = null;
exports.listConnectionConsentLinksOutput = null;
utilities.lazyLoad(exports, ["listConnectionConsentLinks", "listConnectionConsentLinksOutput"], () => require("./listConnectionConsentLinks"));
exports.listConnectionKeys = null;
exports.listConnectionKeysOutput = null;
utilities.lazyLoad(exports, ["listConnectionKeys", "listConnectionKeysOutput"], () => require("./listConnectionKeys"));
// Export enums:
__exportStar(require("../../types/enums/web/v20150801preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20150801preview:Connection":
                return new exports.Connection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20150801preview", _module);
