// @pulumi/azure-native/guestconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240405 = exports.v20220125 = exports.GuestConfigurationHCRPAssignment = exports.GuestConfigurationConnectedVMwarevSphereAssignment = exports.GuestConfigurationAssignmentsVMSS = exports.GuestConfigurationAssignment = exports.getGuestConfigurationHCRPAssignmentOutput = exports.getGuestConfigurationHCRPAssignment = exports.getGuestConfigurationConnectedVMwarevSphereAssignmentOutput = exports.getGuestConfigurationConnectedVMwarevSphereAssignment = exports.getGuestConfigurationAssignmentsVMSSOutput = exports.getGuestConfigurationAssignmentsVMSS = exports.getGuestConfigurationAssignmentOutput = exports.getGuestConfigurationAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGuestConfigurationAssignment = null;
exports.getGuestConfigurationAssignmentOutput = null;
utilities.lazyLoad(exports, ["getGuestConfigurationAssignment", "getGuestConfigurationAssignmentOutput"], () => require("./getGuestConfigurationAssignment"));
exports.getGuestConfigurationAssignmentsVMSS = null;
exports.getGuestConfigurationAssignmentsVMSSOutput = null;
utilities.lazyLoad(exports, ["getGuestConfigurationAssignmentsVMSS", "getGuestConfigurationAssignmentsVMSSOutput"], () => require("./getGuestConfigurationAssignmentsVMSS"));
exports.getGuestConfigurationConnectedVMwarevSphereAssignment = null;
exports.getGuestConfigurationConnectedVMwarevSphereAssignmentOutput = null;
utilities.lazyLoad(exports, ["getGuestConfigurationConnectedVMwarevSphereAssignment", "getGuestConfigurationConnectedVMwarevSphereAssignmentOutput"], () => require("./getGuestConfigurationConnectedVMwarevSphereAssignment"));
exports.getGuestConfigurationHCRPAssignment = null;
exports.getGuestConfigurationHCRPAssignmentOutput = null;
utilities.lazyLoad(exports, ["getGuestConfigurationHCRPAssignment", "getGuestConfigurationHCRPAssignmentOutput"], () => require("./getGuestConfigurationHCRPAssignment"));
exports.GuestConfigurationAssignment = null;
utilities.lazyLoad(exports, ["GuestConfigurationAssignment"], () => require("./guestConfigurationAssignment"));
exports.GuestConfigurationAssignmentsVMSS = null;
utilities.lazyLoad(exports, ["GuestConfigurationAssignmentsVMSS"], () => require("./guestConfigurationAssignmentsVMSS"));
exports.GuestConfigurationConnectedVMwarevSphereAssignment = null;
utilities.lazyLoad(exports, ["GuestConfigurationConnectedVMwarevSphereAssignment"], () => require("./guestConfigurationConnectedVMwarevSphereAssignment"));
exports.GuestConfigurationHCRPAssignment = null;
utilities.lazyLoad(exports, ["GuestConfigurationHCRPAssignment"], () => require("./guestConfigurationHCRPAssignment"));
// Export enums:
__exportStar(require("../types/enums/guestconfiguration"), exports);
// Export sub-modules:
const v20220125 = require("./v20220125");
exports.v20220125 = v20220125;
const v20240405 = require("./v20240405");
exports.v20240405 = v20240405;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:guestconfiguration:GuestConfigurationAssignment":
                return new exports.GuestConfigurationAssignment(name, undefined, { urn });
            case "azure-native:guestconfiguration:GuestConfigurationAssignmentsVMSS":
                return new exports.GuestConfigurationAssignmentsVMSS(name, undefined, { urn });
            case "azure-native:guestconfiguration:GuestConfigurationConnectedVMwarevSphereAssignment":
                return new exports.GuestConfigurationConnectedVMwarevSphereAssignment(name, undefined, { urn });
            case "azure-native:guestconfiguration:GuestConfigurationHCRPAssignment":
                return new exports.GuestConfigurationHCRPAssignment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "guestconfiguration", _module);
