// @pulumi/azure-native/types/enums/deviceregistry/v20241101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicRetainType = exports.EventObservabilityMode = exports.DataPointObservabilityMode = exports.AuthenticationMethod = void 0;
exports.AuthenticationMethod = {
    /**
     * The user authentication method is anonymous.
     */
    Anonymous: "Anonymous",
    /**
     * The user authentication method is an x509 certificate.
     */
    Certificate: "Certificate",
    /**
     * The user authentication method is a username and password.
     */
    UsernamePassword: "UsernamePassword",
};
exports.DataPointObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as counter to OpenTelemetry.
     */
    Counter: "Counter",
    /**
     * Map as gauge to OpenTelemetry.
     */
    Gauge: "Gauge",
    /**
     * Map as histogram to OpenTelemetry.
     */
    Histogram: "Histogram",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.EventObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.TopicRetainType = {
    /**
     * Retain the messages.
     */
    Keep: "Keep",
    /**
     * Never retain messages.
     */
    Never: "Never",
};
