// @pulumi/azure-native/types/enums/media/v20220701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoSyncMode = exports.TrackAttribute = exports.StretchMode = exports.Rotation = exports.Priority = exports.OnErrorType = exports.InterleaveOutput = exports.InsightsType = exports.H265VideoProfile = exports.H265Complexity = exports.H264VideoProfile = exports.H264RateControlMode = exports.H264Complexity = exports.FaceRedactorMode = exports.EntropyMode = exports.EncoderNamedPreset = exports.DeinterlaceParity = exports.DeinterlaceMode = exports.Complexity = exports.ChannelMapping = exports.BlurType = exports.AudioAnalysisMode = exports.AttributeFilter = exports.AnalysisResolution = exports.AacAudioProfile = void 0;
exports.AacAudioProfile = {
    /**
     * Specifies that the output audio is to be encoded into AAC Low Complexity profile (AAC-LC).
     */
    AacLc: "AacLc",
    /**
     * Specifies that the output audio is to be encoded into HE-AAC v1 profile.
     */
    HeAacV1: "HeAacV1",
    /**
     * Specifies that the output audio is to be encoded into HE-AAC v2 profile.
     */
    HeAacV2: "HeAacV2",
};
exports.AnalysisResolution = {
    SourceResolution: "SourceResolution",
    StandardDefinition: "StandardDefinition",
};
exports.AttributeFilter = {
    /**
     * All tracks will be included.
     */
    All: "All",
    /**
     * The first track will be included when the attribute is sorted in descending order.  Generally used to select the largest bitrate.
     */
    Top: "Top",
    /**
     * The first track will be included when the attribute is sorted in ascending order.  Generally used to select the smallest bitrate.
     */
    Bottom: "Bottom",
    /**
     * Any tracks that have an attribute equal to the value given will be included.
     */
    ValueEquals: "ValueEquals",
};
exports.AudioAnalysisMode = {
    /**
     * Performs all operations included in the Basic mode, additionally performing language detection and speaker diarization.
     */
    Standard: "Standard",
    /**
     * This mode performs speech-to-text transcription and generation of a VTT subtitle/caption file. The output of this mode includes an Insights JSON file including only the keywords, transcription,and timing information. Automatic language detection and speaker diarization are not included in this mode.
     */
    Basic: "Basic",
};
exports.BlurType = {
    /**
     * Box: debug filter, bounding box only
     */
    Box: "Box",
    /**
     * Low: box-car blur filter
     */
    Low: "Low",
    /**
     * Med: Gaussian blur filter
     */
    Med: "Med",
    /**
     * High: Confuse blur filter
     */
    High: "High",
    /**
     * Black: Black out filter
     */
    Black: "Black",
};
exports.ChannelMapping = {
    /**
     * The Front Left Channel.
     */
    FrontLeft: "FrontLeft",
    /**
     * The Front Right Channel.
     */
    FrontRight: "FrontRight",
    /**
     * The Center Channel.
     */
    Center: "Center",
    /**
     * Low Frequency Effects Channel.  Sometimes referred to as the subwoofer.
     */
    LowFrequencyEffects: "LowFrequencyEffects",
    /**
     * The Back Left Channel.  Sometimes referred to as the Left Surround Channel.
     */
    BackLeft: "BackLeft",
    /**
     * The Back Right Channel.  Sometimes referred to as the Right Surround Channel.
     */
    BackRight: "BackRight",
    /**
     * The Left Stereo channel.  Sometimes referred to as Down Mix Left.
     */
    StereoLeft: "StereoLeft",
    /**
     * The Right Stereo channel.  Sometimes referred to as Down Mix Right.
     */
    StereoRight: "StereoRight",
};
exports.Complexity = {
    /**
     * Configures the encoder to use settings optimized for faster encoding. Quality is sacrificed to decrease encoding time.
     */
    Speed: "Speed",
    /**
     * Configures the encoder to use settings that achieve a balance between speed and quality.
     */
    Balanced: "Balanced",
    /**
     * Configures the encoder to use settings optimized to produce higher quality output at the expense of slower overall encode time.
     */
    Quality: "Quality",
};
exports.DeinterlaceMode = {
    /**
     * Disables de-interlacing of the source video.
     */
    Off: "Off",
    /**
     * Apply automatic pixel adaptive de-interlacing on each frame in the input video.
     */
    AutoPixelAdaptive: "AutoPixelAdaptive",
};
exports.DeinterlaceParity = {
    /**
     * Automatically detect the order of fields
     */
    Auto: "Auto",
    /**
     * Apply top field first processing of input video.
     */
    TopFieldFirst: "TopFieldFirst",
    /**
     * Apply bottom field first processing of input video.
     */
    BottomFieldFirst: "BottomFieldFirst",
};
exports.EncoderNamedPreset = {
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 2200 kbps and a picture height of 480 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H264SingleBitrateSD: "H264SingleBitrateSD",
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 4500 kbps and a picture height of 720 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H264SingleBitrate720p: "H264SingleBitrate720p",
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 6750 kbps and a picture height of 1080 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H264SingleBitrate1080p: "H264SingleBitrate1080p",
    /**
     * Produces a set of GOP aligned MP4 files with H.264 video and stereo AAC audio. Auto-generates a bitrate ladder based on the input resolution, bitrate and frame rate. The auto-generated preset will never exceed the input resolution. For example, if the input is 720p, output will remain 720p at best.
     */
    AdaptiveStreaming: "AdaptiveStreaming",
    /**
     * Produces a single MP4 file containing only AAC stereo audio encoded at 192 kbps.
     */
    AACGoodQualityAudio: "AACGoodQualityAudio",
    /**
     * Produces a single MP4 file containing only DD(Digital Dolby) stereo audio encoded at 192 kbps.
     */
    DDGoodQualityAudio: "DDGoodQualityAudio",
    /**
     * Exposes an experimental preset for content-aware encoding. Given any input content, the service attempts to automatically determine the optimal number of layers, appropriate bitrate and resolution settings for delivery by adaptive streaming. The underlying algorithms will continue to evolve over time. The output will contain MP4 files with video and audio interleaved.
     */
    ContentAwareEncodingExperimental: "ContentAwareEncodingExperimental",
    /**
     * Produces a set of GOP-aligned MP4s by using content-aware encoding. Given any input content, the service performs an initial lightweight analysis of the input content, and uses the results to determine the optimal number of layers, appropriate bitrate and resolution settings for delivery by adaptive streaming. This preset is particularly effective for low and medium complexity videos, where the output files will be at lower bitrates but at a quality that still delivers a good experience to viewers. The output will contain MP4 files with video and audio interleaved.
     */
    ContentAwareEncoding: "ContentAwareEncoding",
    /**
     * Copy all video and audio streams from the input asset as non-interleaved video and audio output files. This preset can be used to clip an existing asset or convert a group of key frame (GOP) aligned MP4 files as an asset that can be streamed.
     */
    CopyAllBitrateNonInterleaved: "CopyAllBitrateNonInterleaved",
    /**
     * Produces a set of 8 GOP-aligned MP4 files, ranging from 6000 kbps to 400 kbps, and stereo AAC audio. Resolution starts at 1080p and goes down to 180p.
     */
    H264MultipleBitrate1080p: "H264MultipleBitrate1080p",
    /**
     * Produces a set of 6 GOP-aligned MP4 files, ranging from 3400 kbps to 400 kbps, and stereo AAC audio. Resolution starts at 720p and goes down to 180p.
     */
    H264MultipleBitrate720p: "H264MultipleBitrate720p",
    /**
     * Produces a set of 5 GOP-aligned MP4 files, ranging from 1900kbps to 400 kbps, and stereo AAC audio. Resolution starts at 480p and goes down to 240p.
     */
    H264MultipleBitrateSD: "H264MultipleBitrateSD",
    /**
     * Produces a set of GOP-aligned MP4s by using content-aware encoding. Given any input content, the service performs an initial lightweight analysis of the input content, and uses the results to determine the optimal number of layers, appropriate bitrate and resolution settings for delivery by adaptive streaming. This preset is particularly effective for low and medium complexity videos, where the output files will be at lower bitrates but at a quality that still delivers a good experience to viewers. The output will contain MP4 files with video and audio interleaved.
     */
    H265ContentAwareEncoding: "H265ContentAwareEncoding",
    /**
     * Produces a set of GOP aligned MP4 files with H.265 video and stereo AAC audio. Auto-generates a bitrate ladder based on the input resolution, bitrate and frame rate. The auto-generated preset will never exceed the input resolution. For example, if the input is 720p, output will remain 720p at best.
     */
    H265AdaptiveStreaming: "H265AdaptiveStreaming",
    /**
     * Produces an MP4 file where the video is encoded with H.265 codec at 1800 kbps and a picture height of 720 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H265SingleBitrate720p: "H265SingleBitrate720p",
    /**
     * Produces an MP4 file where the video is encoded with H.265 codec at 3500 kbps and a picture height of 1080 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H265SingleBitrate1080p: "H265SingleBitrate1080p",
    /**
     * Produces an MP4 file where the video is encoded with H.265 codec at 9500 kbps and a picture height of 2160 pixels, and the stereo audio is encoded with AAC-LC codec at 128 kbps.
     */
    H265SingleBitrate4K: "H265SingleBitrate4K",
};
exports.EntropyMode = {
    /**
     * Context Adaptive Binary Arithmetic Coder (CABAC) entropy encoding.
     */
    Cabac: "Cabac",
    /**
     * Context Adaptive Variable Length Coder (CAVLC) entropy encoding.
     */
    Cavlc: "Cavlc",
};
exports.FaceRedactorMode = {
    /**
     * Analyze mode detects faces and outputs a metadata file with the results. Allows editing of the metadata file before faces are blurred with Redact mode.
     */
    Analyze: "Analyze",
    /**
     * Redact mode consumes the metadata file from Analyze mode and redacts the faces found.
     */
    Redact: "Redact",
    /**
     * Combined mode does the Analyze and Redact steps in one pass when editing the analyzed faces is not desired.
     */
    Combined: "Combined",
};
exports.H264Complexity = {
    /**
     * Tells the encoder to use settings that are optimized for faster encoding. Quality is sacrificed to decrease encoding time.
     */
    Speed: "Speed",
    /**
     * Tells the encoder to use settings that achieve a balance between speed and quality.
     */
    Balanced: "Balanced",
    /**
     * Tells the encoder to use settings that are optimized to produce higher quality output at the expense of slower overall encode time.
     */
    Quality: "Quality",
};
exports.H264RateControlMode = {
    /**
     * Average Bitrate (ABR) mode that hits the target bitrate: Default mode.
     */
    ABR: "ABR",
    /**
     * Constant Bitrate (CBR) mode that tightens bitrate variations around target bitrate.
     */
    CBR: "CBR",
    /**
     * Constant Rate Factor (CRF) mode that targets at constant subjective quality.
     */
    CRF: "CRF",
};
exports.H264VideoProfile = {
    /**
     * Tells the encoder to automatically determine the appropriate H.264 profile.
     */
    Auto: "Auto",
    /**
     * Baseline profile
     */
    Baseline: "Baseline",
    /**
     * Main profile
     */
    Main: "Main",
    /**
     * High profile.
     */
    High: "High",
    /**
     * High 4:2:2 profile.
     */
    High422: "High422",
    /**
     * High 4:4:4 predictive profile.
     */
    High444: "High444",
};
exports.H265Complexity = {
    /**
     * Tells the encoder to use settings that are optimized for faster encoding. Quality is sacrificed to decrease encoding time.
     */
    Speed: "Speed",
    /**
     * Tells the encoder to use settings that achieve a balance between speed and quality.
     */
    Balanced: "Balanced",
    /**
     * Tells the encoder to use settings that are optimized to produce higher quality output at the expense of slower overall encode time.
     */
    Quality: "Quality",
};
exports.H265VideoProfile = {
    /**
     * Tells the encoder to automatically determine the appropriate H.265 profile.
     */
    Auto: "Auto",
    /**
     * Main profile (https://x265.readthedocs.io/en/default/cli.html?highlight=profile#profile-level-tier)
     */
    Main: "Main",
    /**
     * Main 10 profile (https://en.wikipedia.org/wiki/High_Efficiency_Video_Coding#Main_10)
     */
    Main10: "Main10",
};
exports.InsightsType = {
    /**
     * Generate audio only insights. Ignore video even if present. Fails if no audio is present.
     */
    AudioInsightsOnly: "AudioInsightsOnly",
    /**
     * Generate video only insights. Ignore audio if present. Fails if no video is present.
     */
    VideoInsightsOnly: "VideoInsightsOnly",
    /**
     * Generate both audio and video insights. Fails if either audio or video Insights fail.
     */
    AllInsights: "AllInsights",
};
exports.InterleaveOutput = {
    /**
     * The output is video-only or audio-only.
     */
    NonInterleavedOutput: "NonInterleavedOutput",
    /**
     * The output includes both audio and video.
     */
    InterleavedOutput: "InterleavedOutput",
};
exports.OnErrorType = {
    /**
     * Tells the service that if this TransformOutput fails, then any other incomplete TransformOutputs can be stopped.
     */
    StopProcessingJob: "StopProcessingJob",
    /**
     * Tells the service that if this TransformOutput fails, then allow any other TransformOutput to continue.
     */
    ContinueJob: "ContinueJob",
};
exports.Priority = {
    /**
     * Used for TransformOutputs that can be generated after Normal and High priority TransformOutputs.
     */
    Low: "Low",
    /**
     * Used for TransformOutputs that can be generated at Normal priority.
     */
    Normal: "Normal",
    /**
     * Used for TransformOutputs that should take precedence over others.
     */
    High: "High",
};
exports.Rotation = {
    /**
     * Automatically detect and rotate as needed.
     */
    Auto: "Auto",
    /**
     * Do not rotate the video.  If the output format supports it, any metadata about rotation is kept intact.
     */
    None: "None",
    /**
     * Do not rotate the video but remove any metadata about the rotation.
     */
    Rotate0: "Rotate0",
    /**
     * Rotate 90 degrees clockwise.
     */
    Rotate90: "Rotate90",
    /**
     * Rotate 180 degrees clockwise.
     */
    Rotate180: "Rotate180",
    /**
     * Rotate 270 degrees clockwise.
     */
    Rotate270: "Rotate270",
};
exports.StretchMode = {
    /**
     * Strictly respect the output resolution without considering the pixel aspect ratio or display aspect ratio of the input video.
     */
    None: "None",
    /**
     * Override the output resolution, and change it to match the display aspect ratio of the input, without padding. For example, if the input is 1920x1080 and the encoding preset asks for 1280x1280, then the value in the preset is overridden, and the output will be at 1280x720, which maintains the input aspect ratio of 16:9.
     */
    AutoSize: "AutoSize",
    /**
     * Pad the output (with either letterbox or pillar box) to honor the output resolution, while ensuring that the active video region in the output has the same aspect ratio as the input. For example, if the input is 1920x1080 and the encoding preset asks for 1280x1280, then the output will be at 1280x1280, which contains an inner rectangle of 1280x720 at aspect ratio of 16:9, and pillar box regions 280 pixels wide at the left and right.
     */
    AutoFit: "AutoFit",
};
exports.TrackAttribute = {
    /**
     * The bitrate of the track.
     */
    Bitrate: "Bitrate",
    /**
     * The language of the track.
     */
    Language: "Language",
};
exports.VideoSyncMode = {
    /**
     * This is the default method. Chooses between Cfr and Vfr depending on muxer capabilities. For output format MP4, the default mode is Cfr.
     */
    Auto: "Auto",
    /**
     * The presentation timestamps on frames are passed through from the input file to the output file writer. Recommended when the input source has variable frame rate, and are attempting to produce multiple layers for adaptive streaming in the output which have aligned GOP boundaries. Note: if two or more frames in the input have duplicate timestamps, then the output will also have the same behavior
     */
    Passthrough: "Passthrough",
    /**
     * Input frames will be repeated and/or dropped as needed to achieve exactly the requested constant frame rate. Recommended when the output frame rate is explicitly set at a specified value
     */
    Cfr: "Cfr",
    /**
     * Similar to the Passthrough mode, but if the input has frames that have duplicate timestamps, then only one frame is passed through to the output, and others are dropped. Recommended when the number of output frames is expected to be equal to the number of input frames. For example, the output is used to calculate a quality metric like PSNR against the input
     */
    Vfr: "Vfr",
};
