// @pulumi/azure-native/logic/v20150201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowAccessKey = exports.Workflow = exports.listWorkflowAccessKeySecretKeysOutput = exports.listWorkflowAccessKeySecretKeys = exports.getWorkflowAccessKeyOutput = exports.getWorkflowAccessKey = exports.getWorkflowOutput = exports.getWorkflow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWorkflow = null;
exports.getWorkflowOutput = null;
utilities.lazyLoad(exports, ["getWorkflow", "getWorkflowOutput"], () => require("./getWorkflow"));
exports.getWorkflowAccessKey = null;
exports.getWorkflowAccessKeyOutput = null;
utilities.lazyLoad(exports, ["getWorkflowAccessKey", "getWorkflowAccessKeyOutput"], () => require("./getWorkflowAccessKey"));
exports.listWorkflowAccessKeySecretKeys = null;
exports.listWorkflowAccessKeySecretKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkflowAccessKeySecretKeys", "listWorkflowAccessKeySecretKeysOutput"], () => require("./listWorkflowAccessKeySecretKeys"));
exports.Workflow = null;
utilities.lazyLoad(exports, ["Workflow"], () => require("./workflow"));
exports.WorkflowAccessKey = null;
utilities.lazyLoad(exports, ["WorkflowAccessKey"], () => require("./workflowAccessKey"));
// Export enums:
__exportStar(require("../../types/enums/logic/v20150201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:logic/v20150201preview:Workflow":
                return new exports.Workflow(name, undefined, { urn });
            case "azure-native:logic/v20150201preview:WorkflowAccessKey":
                return new exports.WorkflowAccessKey(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "logic/v20150201preview", _module);
