// @pulumi/azure-native/network/v20210201preview/networkGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The network group resource
 */
class NetworkGroup extends pulumi.CustomResource {
    /**
     * Get an existing NetworkGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkGroup.__pulumiType;
    }
    /**
     * Create a NetworkGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkManagerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["conditionalMembership"] = args ? args.conditionalMembership : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["groupMembers"] = args ? args.groupMembers : undefined;
            resourceInputs["memberType"] = args ? args.memberType : undefined;
            resourceInputs["networkGroupName"] = args ? args.networkGroupName : undefined;
            resourceInputs["networkManagerName"] = args ? args.networkManagerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["conditionalMembership"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["groupMembers"] = undefined /*out*/;
            resourceInputs["memberType"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20210501preview:NetworkGroup" }, { type: "azure-native:network/v20220101:NetworkGroup" }, { type: "azure-native:network/v20220201preview:NetworkGroup" }, { type: "azure-native:network/v20220401preview:NetworkGroup" }, { type: "azure-native:network/v20220501:NetworkGroup" }, { type: "azure-native:network/v20220701:NetworkGroup" }, { type: "azure-native:network/v20220901:NetworkGroup" }, { type: "azure-native:network/v20221101:NetworkGroup" }, { type: "azure-native:network/v20230201:NetworkGroup" }, { type: "azure-native:network/v20230401:NetworkGroup" }, { type: "azure-native:network/v20230501:NetworkGroup" }, { type: "azure-native:network/v20230601:NetworkGroup" }, { type: "azure-native:network/v20230901:NetworkGroup" }, { type: "azure-native:network/v20231101:NetworkGroup" }, { type: "azure-native:network/v20240101:NetworkGroup" }, { type: "azure-native:network/v20240301:NetworkGroup" }, { type: "azure-native:network/v20240501:NetworkGroup" }, { type: "azure-native:network:NetworkGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkGroup = NetworkGroup;
/** @internal */
NetworkGroup.__pulumiType = 'azure-native:network/v20210201preview:NetworkGroup';
