// @pulumi/azure-native/notificationhubs/v20170401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationHubAuthorizationRule = exports.NotificationHub = exports.NamespaceAuthorizationRule = exports.Namespace = exports.listNotificationHubKeysOutput = exports.listNotificationHubKeys = exports.listNamespaceKeysOutput = exports.listNamespaceKeys = exports.getNotificationHubPnsCredentialsOutput = exports.getNotificationHubPnsCredentials = exports.getNotificationHubAuthorizationRuleOutput = exports.getNotificationHubAuthorizationRule = exports.getNotificationHubOutput = exports.getNotificationHub = exports.getNamespaceAuthorizationRuleOutput = exports.getNamespaceAuthorizationRule = exports.getNamespaceOutput = exports.getNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getNamespaceAuthorizationRule = null;
exports.getNamespaceAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getNamespaceAuthorizationRule", "getNamespaceAuthorizationRuleOutput"], () => require("./getNamespaceAuthorizationRule"));
exports.getNotificationHub = null;
exports.getNotificationHubOutput = null;
utilities.lazyLoad(exports, ["getNotificationHub", "getNotificationHubOutput"], () => require("./getNotificationHub"));
exports.getNotificationHubAuthorizationRule = null;
exports.getNotificationHubAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getNotificationHubAuthorizationRule", "getNotificationHubAuthorizationRuleOutput"], () => require("./getNotificationHubAuthorizationRule"));
exports.getNotificationHubPnsCredentials = null;
exports.getNotificationHubPnsCredentialsOutput = null;
utilities.lazyLoad(exports, ["getNotificationHubPnsCredentials", "getNotificationHubPnsCredentialsOutput"], () => require("./getNotificationHubPnsCredentials"));
exports.listNamespaceKeys = null;
exports.listNamespaceKeysOutput = null;
utilities.lazyLoad(exports, ["listNamespaceKeys", "listNamespaceKeysOutput"], () => require("./listNamespaceKeys"));
exports.listNotificationHubKeys = null;
exports.listNotificationHubKeysOutput = null;
utilities.lazyLoad(exports, ["listNotificationHubKeys", "listNotificationHubKeysOutput"], () => require("./listNotificationHubKeys"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.NamespaceAuthorizationRule = null;
utilities.lazyLoad(exports, ["NamespaceAuthorizationRule"], () => require("./namespaceAuthorizationRule"));
exports.NotificationHub = null;
utilities.lazyLoad(exports, ["NotificationHub"], () => require("./notificationHub"));
exports.NotificationHubAuthorizationRule = null;
utilities.lazyLoad(exports, ["NotificationHubAuthorizationRule"], () => require("./notificationHubAuthorizationRule"));
// Export enums:
__exportStar(require("../../types/enums/notificationhubs/v20170401"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:notificationhubs/v20170401:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure-native:notificationhubs/v20170401:NamespaceAuthorizationRule":
                return new exports.NamespaceAuthorizationRule(name, undefined, { urn });
            case "azure-native:notificationhubs/v20170401:NotificationHub":
                return new exports.NotificationHub(name, undefined, { urn });
            case "azure-native:notificationhubs/v20170401:NotificationHubAuthorizationRule":
                return new exports.NotificationHubAuthorizationRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "notificationhubs/v20170401", _module);
