// @pulumi/azure-native/types/enums/network/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRuleKind = exports.TrafficViewEnrollmentStatus = exports.TrafficRoutingMethod = exports.SecurityConfigurationRuleProtocol = exports.SecurityConfigurationRuleDirection = exports.ProfileStatus = exports.ProfileMonitorStatus = exports.MonitorProtocol = exports.EndpointStatus = exports.EndpointMonitorStatus = exports.DeleteExistingNSGs = exports.AlwaysServe = exports.AllowedEndpointRecordType = exports.AddressPrefixType = void 0;
exports.AddressPrefixType = {
    IPPrefix: "IPPrefix",
    ServiceTag: "ServiceTag",
};
exports.AllowedEndpointRecordType = {
    DomainName: "DomainName",
    IPv4Address: "IPv4Address",
    IPv6Address: "IPv6Address",
    Any: "Any",
};
exports.AlwaysServe = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DeleteExistingNSGs = {
    False: "False",
    True: "True",
};
exports.EndpointMonitorStatus = {
    CheckingEndpoint: "CheckingEndpoint",
    Online: "Online",
    Degraded: "Degraded",
    Disabled: "Disabled",
    Inactive: "Inactive",
    Stopped: "Stopped",
};
exports.EndpointStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MonitorProtocol = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    TCP: "TCP",
};
exports.ProfileMonitorStatus = {
    CheckingEndpoints: "CheckingEndpoints",
    Online: "Online",
    Degraded: "Degraded",
    Disabled: "Disabled",
    Inactive: "Inactive",
};
exports.ProfileStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SecurityConfigurationRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.SecurityConfigurationRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    Icmp: "Icmp",
    Esp: "Esp",
    Any: "Any",
    Ah: "Ah",
};
exports.TrafficRoutingMethod = {
    Performance: "Performance",
    Priority: "Priority",
    Weighted: "Weighted",
    Geographic: "Geographic",
    MultiValue: "MultiValue",
    Subnet: "Subnet",
};
exports.TrafficViewEnrollmentStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.UserRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
