// @pulumi/azure-native/types/enums/securityinsights/v20230501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Version = exports.RepoType = exports.EntityTimelineKind = exports.DeploymentState = exports.DeploymentResult = exports.DeploymentFetchStatus = exports.ContentType = void 0;
exports.ContentType = {
    AnalyticsRule: "AnalyticsRule",
    Workbook: "Workbook",
};
exports.DeploymentFetchStatus = {
    Success: "Success",
    Unauthorized: "Unauthorized",
    NotFound: "NotFound",
};
exports.DeploymentResult = {
    Success: "Success",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.DeploymentState = {
    In_Progress: "In_Progress",
    Completed: "Completed",
    Queued: "Queued",
    Canceling: "Canceling",
};
exports.EntityTimelineKind = {
    /**
     * activity
     */
    Activity: "Activity",
    /**
     * bookmarks
     */
    Bookmark: "Bookmark",
    /**
     * security alerts
     */
    SecurityAlert: "SecurityAlert",
    /**
     * anomaly
     */
    Anomaly: "Anomaly",
};
exports.RepoType = {
    Github: "Github",
    DevOps: "DevOps",
};
exports.Version = {
    V1: "V1",
    V2: "V2",
};
