// @pulumi/gcp/diagflow/cxPage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CxPage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Dialogflow CX conversation (session) can be described and visualized as a state machine. The states of a CX session are represented by pages.
 *
 * To get more information about Page, see:
 *
 * * [API documentation](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.flows.pages)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/dialogflow/cx/docs)
 *
 * ## Example Usage
 *
 * ### Dialogflowcx Page Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const agent = new gcp.diagflow.CxAgent("agent", {
 *     displayName: "dialogflowcx-agent",
 *     location: "global",
 *     defaultLanguageCode: "en",
 *     supportedLanguageCodes: [
 *         "fr",
 *         "de",
 *         "es",
 *     ],
 *     timeZone: "America/New_York",
 *     description: "Example description.",
 *     avatarUri: "https://cloud.google.com/_static/images/cloud/icons/favicons/onecloud/super_cloud.png",
 *     enableStackdriverLogging: true,
 *     enableSpellCorrection: true,
 *     speechToTextSettings: {
 *         enableSpeechAdaptation: true,
 *     },
 * });
 * const myPage2 = new gcp.diagflow.CxPage("my_page2", {
 *     parent: agent.startFlow,
 *     displayName: "MyPage2",
 * });
 * const myWebhook = new gcp.diagflow.CxWebhook("my_webhook", {
 *     parent: agent.id,
 *     displayName: "MyWebhook",
 *     genericWebService: {
 *         uri: "https://example.com",
 *     },
 * });
 * const basicPage = new gcp.diagflow.CxPage("basic_page", {
 *     parent: agent.startFlow,
 *     displayName: "MyPage",
 *     entryFulfillment: {
 *         messages: [
 *             {
 *                 channel: "some-channel",
 *                 text: {
 *                     texts: ["Welcome to page"],
 *                 },
 *             },
 *             {
 *                 payload: "        {\"some-key\": \"some-value\", \"other-key\": [\"other-value\"]}\n",
 *             },
 *             {
 *                 conversationSuccess: {
 *                     metadata: "          {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                 },
 *             },
 *             {
 *                 outputAudioText: {
 *                     text: "some output text",
 *                 },
 *             },
 *             {
 *                 outputAudioText: {
 *                     ssml: "          <speak>Some example <say-as interpret-as=\"characters\">SSML XML</say-as></speak>\n",
 *                 },
 *             },
 *             {
 *                 liveAgentHandoff: {
 *                     metadata: "          {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                 },
 *             },
 *             {
 *                 playAudio: {
 *                     audioUri: "http://example.com/some-audio-file.mp3",
 *                 },
 *             },
 *             {
 *                 telephonyTransferCall: {
 *                     phoneNumber: "1-234-567-8901",
 *                 },
 *             },
 *         ],
 *         setParameterActions: [
 *             {
 *                 parameter: "some-param",
 *                 value: "123.45",
 *             },
 *             {
 *                 parameter: "another-param",
 *                 value: JSON.stringify("abc"),
 *             },
 *             {
 *                 parameter: "other-param",
 *                 value: JSON.stringify(["foo"]),
 *             },
 *         ],
 *         conditionalCases: [{
 *             cases: JSON.stringify([
 *                 {
 *                     condition: "$sys.func.RAND() < 0.5",
 *                     caseContent: [
 *                         {
 *                             message: {
 *                                 text: {
 *                                     text: ["First case"],
 *                                 },
 *                             },
 *                         },
 *                         {
 *                             additionalCases: {
 *                                 cases: [{
 *                                     condition: "$sys.func.RAND() < 0.2",
 *                                     caseContent: [{
 *                                         message: {
 *                                             text: {
 *                                                 text: ["Nested case"],
 *                                             },
 *                                         },
 *                                     }],
 *                                 }],
 *                             },
 *                         },
 *                     ],
 *                 },
 *                 {
 *                     caseContent: [{
 *                         message: {
 *                             text: {
 *                                 text: ["Final case"],
 *                             },
 *                         },
 *                     }],
 *                 },
 *             ]),
 *         }],
 *     },
 *     eventHandlers: [{
 *         event: "some-event",
 *         triggerFulfillment: {
 *             returnPartialResponses: true,
 *             messages: [
 *                 {
 *                     channel: "some-channel",
 *                     text: {
 *                         texts: ["Some text"],
 *                     },
 *                 },
 *                 {
 *                     payload: "          {\"some-key\": \"some-value\", \"other-key\": [\"other-value\"]}\n",
 *                 },
 *                 {
 *                     conversationSuccess: {
 *                         metadata: "            {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                     },
 *                 },
 *                 {
 *                     outputAudioText: {
 *                         text: "some output text",
 *                     },
 *                 },
 *                 {
 *                     outputAudioText: {
 *                         ssml: "            <speak>Some example <say-as interpret-as=\"characters\">SSML XML</say-as></speak>\n",
 *                     },
 *                 },
 *                 {
 *                     liveAgentHandoff: {
 *                         metadata: "            {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                     },
 *                 },
 *                 {
 *                     playAudio: {
 *                         audioUri: "http://example.com/some-audio-file.mp3",
 *                     },
 *                 },
 *                 {
 *                     telephonyTransferCall: {
 *                         phoneNumber: "1-234-567-8901",
 *                     },
 *                 },
 *             ],
 *             setParameterActions: [
 *                 {
 *                     parameter: "some-param",
 *                     value: "123.45",
 *                 },
 *                 {
 *                     parameter: "another-param",
 *                     value: JSON.stringify("abc"),
 *                 },
 *                 {
 *                     parameter: "other-param",
 *                     value: JSON.stringify(["foo"]),
 *                 },
 *             ],
 *             conditionalCases: [{
 *                 cases: JSON.stringify([
 *                     {
 *                         condition: "$sys.func.RAND() < 0.5",
 *                         caseContent: [
 *                             {
 *                                 message: {
 *                                     text: {
 *                                         text: ["First case"],
 *                                     },
 *                                 },
 *                             },
 *                             {
 *                                 additionalCases: {
 *                                     cases: [{
 *                                         condition: "$sys.func.RAND() < 0.2",
 *                                         caseContent: [{
 *                                             message: {
 *                                                 text: {
 *                                                     text: ["Nested case"],
 *                                                 },
 *                                             },
 *                                         }],
 *                                     }],
 *                                 },
 *                             },
 *                         ],
 *                     },
 *                     {
 *                         caseContent: [{
 *                             message: {
 *                                 text: {
 *                                     text: ["Final case"],
 *                                 },
 *                             },
 *                         }],
 *                     },
 *                 ]),
 *             }],
 *         },
 *     }],
 *     form: {
 *         parameters: [{
 *             displayName: "param1",
 *             entityType: "projects/-/locations/-/agents/-/entityTypes/sys.date",
 *             defaultValue: JSON.stringify("2000-01-01"),
 *             fillBehavior: {
 *                 initialPromptFulfillment: {
 *                     messages: [
 *                         {
 *                             channel: "some-channel",
 *                             text: {
 *                                 texts: ["Please provide param1"],
 *                             },
 *                         },
 *                         {
 *                             payload: "              {\"some-key\": \"some-value\", \"other-key\": [\"other-value\"]}\n",
 *                         },
 *                         {
 *                             conversationSuccess: {
 *                                 metadata: "                {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                             },
 *                         },
 *                         {
 *                             outputAudioText: {
 *                                 text: "some output text",
 *                             },
 *                         },
 *                         {
 *                             outputAudioText: {
 *                                 ssml: "                <speak>Some example <say-as interpret-as=\"characters\">SSML XML</say-as></speak>\n",
 *                             },
 *                         },
 *                         {
 *                             liveAgentHandoff: {
 *                                 metadata: "                {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                             },
 *                         },
 *                         {
 *                             playAudio: {
 *                                 audioUri: "http://example.com/some-audio-file.mp3",
 *                             },
 *                         },
 *                         {
 *                             telephonyTransferCall: {
 *                                 phoneNumber: "1-234-567-8901",
 *                             },
 *                         },
 *                     ],
 *                     setParameterActions: [
 *                         {
 *                             parameter: "some-param",
 *                             value: "123.45",
 *                         },
 *                         {
 *                             parameter: "another-param",
 *                             value: JSON.stringify("abc"),
 *                         },
 *                         {
 *                             parameter: "other-param",
 *                             value: JSON.stringify(["foo"]),
 *                         },
 *                     ],
 *                     conditionalCases: [{
 *                         cases: JSON.stringify([
 *                             {
 *                                 condition: "$sys.func.RAND() < 0.5",
 *                                 caseContent: [
 *                                     {
 *                                         message: {
 *                                             text: {
 *                                                 text: ["First case"],
 *                                             },
 *                                         },
 *                                     },
 *                                     {
 *                                         additionalCases: {
 *                                             cases: [{
 *                                                 condition: "$sys.func.RAND() < 0.2",
 *                                                 caseContent: [{
 *                                                     message: {
 *                                                         text: {
 *                                                             text: ["Nested case"],
 *                                                         },
 *                                                     },
 *                                                 }],
 *                                             }],
 *                                         },
 *                                     },
 *                                 ],
 *                             },
 *                             {
 *                                 caseContent: [{
 *                                     message: {
 *                                         text: {
 *                                             text: ["Final case"],
 *                                         },
 *                                     },
 *                                 }],
 *                             },
 *                         ]),
 *                     }],
 *                 },
 *                 repromptEventHandlers: [
 *                     {
 *                         event: "sys.no-match-1",
 *                         triggerFulfillment: {
 *                             returnPartialResponses: true,
 *                             webhook: myWebhook.id,
 *                             tag: "some-tag",
 *                             messages: [
 *                                 {
 *                                     channel: "some-channel",
 *                                     text: {
 *                                         texts: ["Please provide param1"],
 *                                     },
 *                                 },
 *                                 {
 *                                     payload: "                {\"some-key\": \"some-value\", \"other-key\": [\"other-value\"]}\n",
 *                                 },
 *                                 {
 *                                     conversationSuccess: {
 *                                         metadata: "                  {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                                     },
 *                                 },
 *                                 {
 *                                     outputAudioText: {
 *                                         text: "some output text",
 *                                     },
 *                                 },
 *                                 {
 *                                     outputAudioText: {
 *                                         ssml: "                  <speak>Some example <say-as interpret-as=\"characters\">SSML XML</say-as></speak>\n",
 *                                     },
 *                                 },
 *                                 {
 *                                     liveAgentHandoff: {
 *                                         metadata: "                  {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                                     },
 *                                 },
 *                                 {
 *                                     playAudio: {
 *                                         audioUri: "http://example.com/some-audio-file.mp3",
 *                                     },
 *                                 },
 *                                 {
 *                                     telephonyTransferCall: {
 *                                         phoneNumber: "1-234-567-8901",
 *                                     },
 *                                 },
 *                             ],
 *                             setParameterActions: [
 *                                 {
 *                                     parameter: "some-param",
 *                                     value: "123.45",
 *                                 },
 *                                 {
 *                                     parameter: "another-param",
 *                                     value: JSON.stringify("abc"),
 *                                 },
 *                                 {
 *                                     parameter: "other-param",
 *                                     value: JSON.stringify(["foo"]),
 *                                 },
 *                             ],
 *                             conditionalCases: [{
 *                                 cases: JSON.stringify([
 *                                     {
 *                                         condition: "$sys.func.RAND() < 0.5",
 *                                         caseContent: [
 *                                             {
 *                                                 message: {
 *                                                     text: {
 *                                                         text: ["First case"],
 *                                                     },
 *                                                 },
 *                                             },
 *                                             {
 *                                                 additionalCases: {
 *                                                     cases: [{
 *                                                         condition: "$sys.func.RAND() < 0.2",
 *                                                         caseContent: [{
 *                                                             message: {
 *                                                                 text: {
 *                                                                     text: ["Nested case"],
 *                                                                 },
 *                                                             },
 *                                                         }],
 *                                                     }],
 *                                                 },
 *                                             },
 *                                         ],
 *                                     },
 *                                     {
 *                                         caseContent: [{
 *                                             message: {
 *                                                 text: {
 *                                                     text: ["Final case"],
 *                                                 },
 *                                             },
 *                                         }],
 *                                     },
 *                                 ]),
 *                             }],
 *                         },
 *                     },
 *                     {
 *                         event: "sys.no-match-2",
 *                         targetFlow: agent.startFlow,
 *                     },
 *                     {
 *                         event: "sys.no-match-3",
 *                         targetPage: myPage2.id,
 *                     },
 *                 ],
 *             },
 *             required: true,
 *             redact: true,
 *             advancedSettings: {
 *                 dtmfSettings: {
 *                     enabled: true,
 *                     maxDigits: 1,
 *                     finishDigit: "#",
 *                 },
 *             },
 *         }],
 *     },
 *     transitionRoutes: [{
 *         condition: "$page.params.status = 'FINAL'",
 *         triggerFulfillment: {
 *             messages: [
 *                 {
 *                     channel: "some-channel",
 *                     text: {
 *                         texts: ["information completed, navigating to page 2"],
 *                     },
 *                 },
 *                 {
 *                     payload: "          {\"some-key\": \"some-value\", \"other-key\": [\"other-value\"]}\n",
 *                 },
 *                 {
 *                     conversationSuccess: {
 *                         metadata: "            {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                     },
 *                 },
 *                 {
 *                     outputAudioText: {
 *                         text: "some output text",
 *                     },
 *                 },
 *                 {
 *                     outputAudioText: {
 *                         ssml: "            <speak>Some example <say-as interpret-as=\"characters\">SSML XML</say-as></speak>\n",
 *                     },
 *                 },
 *                 {
 *                     liveAgentHandoff: {
 *                         metadata: "            {\"some-metadata-key\": \"some-value\", \"other-metadata-key\": 1234}\n",
 *                     },
 *                 },
 *                 {
 *                     playAudio: {
 *                         audioUri: "http://example.com/some-audio-file.mp3",
 *                     },
 *                 },
 *                 {
 *                     telephonyTransferCall: {
 *                         phoneNumber: "1-234-567-8901",
 *                     },
 *                 },
 *             ],
 *             setParameterActions: [
 *                 {
 *                     parameter: "some-param",
 *                     value: "123.45",
 *                 },
 *                 {
 *                     parameter: "another-param",
 *                     value: JSON.stringify("abc"),
 *                 },
 *                 {
 *                     parameter: "other-param",
 *                     value: JSON.stringify(["foo"]),
 *                 },
 *             ],
 *             conditionalCases: [{
 *                 cases: JSON.stringify([
 *                     {
 *                         condition: "$sys.func.RAND() < 0.5",
 *                         caseContent: [
 *                             {
 *                                 message: {
 *                                     text: {
 *                                         text: ["First case"],
 *                                     },
 *                                 },
 *                             },
 *                             {
 *                                 additionalCases: {
 *                                     cases: [{
 *                                         condition: "$sys.func.RAND() < 0.2",
 *                                         caseContent: [{
 *                                             message: {
 *                                                 text: {
 *                                                     text: ["Nested case"],
 *                                                 },
 *                                             },
 *                                         }],
 *                                     }],
 *                                 },
 *                             },
 *                         ],
 *                     },
 *                     {
 *                         caseContent: [{
 *                             message: {
 *                                 text: {
 *                                     text: ["Final case"],
 *                                 },
 *                             },
 *                         }],
 *                     },
 *                 ]),
 *             }],
 *         },
 *         targetPage: myPage2.id,
 *     }],
 *     advancedSettings: {
 *         dtmfSettings: {
 *             enabled: true,
 *             maxDigits: 1,
 *             finishDigit: "#",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Page can be imported using any of these accepted formats:
 *
 * * `{{parent}}/pages/{{name}}`
 *
 * * `{{parent}}/{{name}}`
 *
 * When using the `pulumi import` command, Page can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:diagflow/cxPage:CxPage default {{parent}}/pages/{{name}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:diagflow/cxPage:CxPage default {{parent}}/{{name}}
 * ```
 */
class CxPage extends pulumi.CustomResource {
    /**
     * Get an existing CxPage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CxPage(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CxPage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CxPage.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["advancedSettings"] = state ? state.advancedSettings : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["entryFulfillment"] = state ? state.entryFulfillment : undefined;
            resourceInputs["eventHandlers"] = state ? state.eventHandlers : undefined;
            resourceInputs["form"] = state ? state.form : undefined;
            resourceInputs["languageCode"] = state ? state.languageCode : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["parent"] = state ? state.parent : undefined;
            resourceInputs["transitionRouteGroups"] = state ? state.transitionRouteGroups : undefined;
            resourceInputs["transitionRoutes"] = state ? state.transitionRoutes : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            resourceInputs["advancedSettings"] = args ? args.advancedSettings : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["entryFulfillment"] = args ? args.entryFulfillment : undefined;
            resourceInputs["eventHandlers"] = args ? args.eventHandlers : undefined;
            resourceInputs["form"] = args ? args.form : undefined;
            resourceInputs["languageCode"] = args ? args.languageCode : undefined;
            resourceInputs["parent"] = args ? args.parent : undefined;
            resourceInputs["transitionRouteGroups"] = args ? args.transitionRouteGroups : undefined;
            resourceInputs["transitionRoutes"] = args ? args.transitionRoutes : undefined;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CxPage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CxPage = CxPage;
/** @internal */
CxPage.__pulumiType = 'gcp:diagflow/cxPage:CxPage';
