// @pulumi/azure-native/servicefabric/v20221001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedCluster = exports.getmanagedAzResiliencyStatusOutput = exports.getmanagedAzResiliencyStatus = exports.getManagedClusterOutput = exports.getManagedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getManagedCluster = null;
exports.getManagedClusterOutput = null;
utilities.lazyLoad(exports, ["getManagedCluster", "getManagedClusterOutput"], () => require("./getManagedCluster"));
exports.getmanagedAzResiliencyStatus = null;
exports.getmanagedAzResiliencyStatusOutput = null;
utilities.lazyLoad(exports, ["getmanagedAzResiliencyStatus", "getmanagedAzResiliencyStatusOutput"], () => require("./getmanagedAzResiliencyStatus"));
exports.ManagedCluster = null;
utilities.lazyLoad(exports, ["ManagedCluster"], () => require("./managedCluster"));
// Export enums:
__exportStar(require("../../types/enums/servicefabric/v20221001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicefabric/v20221001preview:ManagedCluster":
                return new exports.ManagedCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicefabric/v20221001preview", _module);
