// @pulumi/azure-native/costmanagement/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20190301preview = exports.v20180801preview = exports.ViewByScope = exports.View = exports.TagInheritanceSetting = exports.Setting = exports.ScheduledActionByScope = exports.ScheduledAction = exports.ReportByResourceGroupName = exports.ReportByDepartment = exports.ReportByBillingAccount = exports.Report = exports.MarkupRule = exports.getViewByScopeOutput = exports.getViewByScope = exports.getViewOutput = exports.getView = exports.getTagInheritanceSettingOutput = exports.getTagInheritanceSetting = exports.getSettingOutput = exports.getSetting = exports.getScheduledActionByScopeOutput = exports.getScheduledActionByScope = exports.getScheduledActionOutput = exports.getScheduledAction = exports.getReportByResourceGroupNameOutput = exports.getReportByResourceGroupName = exports.getReportByDepartmentOutput = exports.getReportByDepartment = exports.getReportByBillingAccountOutput = exports.getReportByBillingAccount = exports.getReportOutput = exports.getReport = exports.getMarkupRuleOutput = exports.getMarkupRule = exports.getExportOutput = exports.getExport = exports.getCostAllocationRuleOutput = exports.getCostAllocationRule = exports.getConnectorOutput = exports.getConnector = exports.getCloudConnectorOutput = exports.getCloudConnector = exports.getBudgetOutput = exports.getBudget = exports.Export = exports.CostAllocationRule = exports.Connector = exports.CloudConnector = exports.Budget = void 0;
exports.v20241001preview = exports.v20240801 = exports.v20231101 = exports.v20230901 = exports.v20230801 = exports.v20230701preview = exports.v20230401preview = exports.v20230301 = exports.v20221005preview = exports.v20221001 = exports.v20200601 = exports.v20200301preview = exports.v20191101 = exports.v20191001 = exports.v20190401preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Budget = null;
utilities.lazyLoad(exports, ["Budget"], () => require("./budget"));
exports.CloudConnector = null;
utilities.lazyLoad(exports, ["CloudConnector"], () => require("./cloudConnector"));
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.CostAllocationRule = null;
utilities.lazyLoad(exports, ["CostAllocationRule"], () => require("./costAllocationRule"));
exports.Export = null;
utilities.lazyLoad(exports, ["Export"], () => require("./export"));
exports.getBudget = null;
exports.getBudgetOutput = null;
utilities.lazyLoad(exports, ["getBudget", "getBudgetOutput"], () => require("./getBudget"));
exports.getCloudConnector = null;
exports.getCloudConnectorOutput = null;
utilities.lazyLoad(exports, ["getCloudConnector", "getCloudConnectorOutput"], () => require("./getCloudConnector"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getCostAllocationRule = null;
exports.getCostAllocationRuleOutput = null;
utilities.lazyLoad(exports, ["getCostAllocationRule", "getCostAllocationRuleOutput"], () => require("./getCostAllocationRule"));
exports.getExport = null;
exports.getExportOutput = null;
utilities.lazyLoad(exports, ["getExport", "getExportOutput"], () => require("./getExport"));
exports.getMarkupRule = null;
exports.getMarkupRuleOutput = null;
utilities.lazyLoad(exports, ["getMarkupRule", "getMarkupRuleOutput"], () => require("./getMarkupRule"));
exports.getReport = null;
exports.getReportOutput = null;
utilities.lazyLoad(exports, ["getReport", "getReportOutput"], () => require("./getReport"));
exports.getReportByBillingAccount = null;
exports.getReportByBillingAccountOutput = null;
utilities.lazyLoad(exports, ["getReportByBillingAccount", "getReportByBillingAccountOutput"], () => require("./getReportByBillingAccount"));
exports.getReportByDepartment = null;
exports.getReportByDepartmentOutput = null;
utilities.lazyLoad(exports, ["getReportByDepartment", "getReportByDepartmentOutput"], () => require("./getReportByDepartment"));
exports.getReportByResourceGroupName = null;
exports.getReportByResourceGroupNameOutput = null;
utilities.lazyLoad(exports, ["getReportByResourceGroupName", "getReportByResourceGroupNameOutput"], () => require("./getReportByResourceGroupName"));
exports.getScheduledAction = null;
exports.getScheduledActionOutput = null;
utilities.lazyLoad(exports, ["getScheduledAction", "getScheduledActionOutput"], () => require("./getScheduledAction"));
exports.getScheduledActionByScope = null;
exports.getScheduledActionByScopeOutput = null;
utilities.lazyLoad(exports, ["getScheduledActionByScope", "getScheduledActionByScopeOutput"], () => require("./getScheduledActionByScope"));
exports.getSetting = null;
exports.getSettingOutput = null;
utilities.lazyLoad(exports, ["getSetting", "getSettingOutput"], () => require("./getSetting"));
exports.getTagInheritanceSetting = null;
exports.getTagInheritanceSettingOutput = null;
utilities.lazyLoad(exports, ["getTagInheritanceSetting", "getTagInheritanceSettingOutput"], () => require("./getTagInheritanceSetting"));
exports.getView = null;
exports.getViewOutput = null;
utilities.lazyLoad(exports, ["getView", "getViewOutput"], () => require("./getView"));
exports.getViewByScope = null;
exports.getViewByScopeOutput = null;
utilities.lazyLoad(exports, ["getViewByScope", "getViewByScopeOutput"], () => require("./getViewByScope"));
exports.MarkupRule = null;
utilities.lazyLoad(exports, ["MarkupRule"], () => require("./markupRule"));
exports.Report = null;
utilities.lazyLoad(exports, ["Report"], () => require("./report"));
exports.ReportByBillingAccount = null;
utilities.lazyLoad(exports, ["ReportByBillingAccount"], () => require("./reportByBillingAccount"));
exports.ReportByDepartment = null;
utilities.lazyLoad(exports, ["ReportByDepartment"], () => require("./reportByDepartment"));
exports.ReportByResourceGroupName = null;
utilities.lazyLoad(exports, ["ReportByResourceGroupName"], () => require("./reportByResourceGroupName"));
exports.ScheduledAction = null;
utilities.lazyLoad(exports, ["ScheduledAction"], () => require("./scheduledAction"));
exports.ScheduledActionByScope = null;
utilities.lazyLoad(exports, ["ScheduledActionByScope"], () => require("./scheduledActionByScope"));
exports.Setting = null;
utilities.lazyLoad(exports, ["Setting"], () => require("./setting"));
exports.TagInheritanceSetting = null;
utilities.lazyLoad(exports, ["TagInheritanceSetting"], () => require("./tagInheritanceSetting"));
exports.View = null;
utilities.lazyLoad(exports, ["View"], () => require("./view"));
exports.ViewByScope = null;
utilities.lazyLoad(exports, ["ViewByScope"], () => require("./viewByScope"));
// Export enums:
__exportStar(require("../types/enums/costmanagement"), exports);
// Export sub-modules:
const v20180801preview = require("./v20180801preview");
exports.v20180801preview = v20180801preview;
const v20190301preview = require("./v20190301preview");
exports.v20190301preview = v20190301preview;
const v20190401preview = require("./v20190401preview");
exports.v20190401preview = v20190401preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const v20221005preview = require("./v20221005preview");
exports.v20221005preview = v20221005preview;
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240801 = require("./v20240801");
exports.v20240801 = v20240801;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:costmanagement:Budget":
                return new exports.Budget(name, undefined, { urn });
            case "azure-native:costmanagement:CloudConnector":
                return new exports.CloudConnector(name, undefined, { urn });
            case "azure-native:costmanagement:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:costmanagement:CostAllocationRule":
                return new exports.CostAllocationRule(name, undefined, { urn });
            case "azure-native:costmanagement:Export":
                return new exports.Export(name, undefined, { urn });
            case "azure-native:costmanagement:MarkupRule":
                return new exports.MarkupRule(name, undefined, { urn });
            case "azure-native:costmanagement:Report":
                return new exports.Report(name, undefined, { urn });
            case "azure-native:costmanagement:ReportByBillingAccount":
                return new exports.ReportByBillingAccount(name, undefined, { urn });
            case "azure-native:costmanagement:ReportByDepartment":
                return new exports.ReportByDepartment(name, undefined, { urn });
            case "azure-native:costmanagement:ReportByResourceGroupName":
                return new exports.ReportByResourceGroupName(name, undefined, { urn });
            case "azure-native:costmanagement:ScheduledAction":
                return new exports.ScheduledAction(name, undefined, { urn });
            case "azure-native:costmanagement:ScheduledActionByScope":
                return new exports.ScheduledActionByScope(name, undefined, { urn });
            case "azure-native:costmanagement:Setting":
                return new exports.Setting(name, undefined, { urn });
            case "azure-native:costmanagement:TagInheritanceSetting":
                return new exports.TagInheritanceSetting(name, undefined, { urn });
            case "azure-native:costmanagement:View":
                return new exports.View(name, undefined, { urn });
            case "azure-native:costmanagement:ViewByScope":
                return new exports.ViewByScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "costmanagement", _module);
