// @pulumi/azure-native/types/enums/deviceregistry/v20240901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicRetainType = exports.SystemAssignedServiceIdentityType = exports.SchemaType = exports.Format = exports.EventObservabilityMode = exports.DataPointObservabilityMode = exports.AuthenticationMethod = void 0;
exports.AuthenticationMethod = {
    /**
     * The user authentication method is anonymous.
     */
    Anonymous: "Anonymous",
    /**
     * The user authentication method is an x509 certificate.
     */
    Certificate: "Certificate",
    /**
     * The user authentication method is a username and password.
     */
    UsernamePassword: "UsernamePassword",
};
exports.DataPointObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as counter to OpenTelemetry.
     */
    Counter: "Counter",
    /**
     * Map as gauge to OpenTelemetry.
     */
    Gauge: "Gauge",
    /**
     * Map as histogram to OpenTelemetry.
     */
    Histogram: "Histogram",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.EventObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "None",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "Log",
};
exports.Format = {
    /**
     * JSON Schema version draft 7 format
     */
    JsonSchema_draft7: "JsonSchema/draft-07",
    /**
     * Delta format
     */
    Delta_1_0: "Delta/1.0",
};
exports.SchemaType = {
    /**
     * Message Schema schema type
     */
    MessageSchema: "MessageSchema",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.TopicRetainType = {
    /**
     * Retain the messages.
     */
    Keep: "Keep",
    /**
     * Never retain messages.
     */
    Never: "Never",
};
