// @pulumi/azure-native/sql/v20140401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransparentDataEncryption = exports.ServerCommunicationLink = exports.ServerAzureADAdministrator = exports.ServerAdvisor = exports.Server = exports.getTransparentDataEncryptionOutput = exports.getTransparentDataEncryption = exports.getServerCommunicationLinkOutput = exports.getServerCommunicationLink = exports.getServerAzureADAdministratorOutput = exports.getServerAzureADAdministrator = exports.getServerAdvisorOutput = exports.getServerAdvisor = exports.getServerOutput = exports.getServer = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getElasticPoolOutput = exports.getElasticPool = exports.getDisasterRecoveryConfigurationOutput = exports.getDisasterRecoveryConfiguration = exports.getDatabaseThreatDetectionPolicyOutput = exports.getDatabaseThreatDetectionPolicy = exports.getDatabaseAdvisorOutput = exports.getDatabaseAdvisor = exports.getDatabaseOutput = exports.getDatabase = exports.FirewallRule = exports.ElasticPool = exports.DisasterRecoveryConfiguration = exports.DatabaseThreatDetectionPolicy = exports.DatabaseAdvisor = exports.Database = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.DatabaseAdvisor = null;
utilities.lazyLoad(exports, ["DatabaseAdvisor"], () => require("./databaseAdvisor"));
exports.DatabaseThreatDetectionPolicy = null;
utilities.lazyLoad(exports, ["DatabaseThreatDetectionPolicy"], () => require("./databaseThreatDetectionPolicy"));
exports.DisasterRecoveryConfiguration = null;
utilities.lazyLoad(exports, ["DisasterRecoveryConfiguration"], () => require("./disasterRecoveryConfiguration"));
exports.ElasticPool = null;
utilities.lazyLoad(exports, ["ElasticPool"], () => require("./elasticPool"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getDatabaseAdvisor = null;
exports.getDatabaseAdvisorOutput = null;
utilities.lazyLoad(exports, ["getDatabaseAdvisor", "getDatabaseAdvisorOutput"], () => require("./getDatabaseAdvisor"));
exports.getDatabaseThreatDetectionPolicy = null;
exports.getDatabaseThreatDetectionPolicyOutput = null;
utilities.lazyLoad(exports, ["getDatabaseThreatDetectionPolicy", "getDatabaseThreatDetectionPolicyOutput"], () => require("./getDatabaseThreatDetectionPolicy"));
exports.getDisasterRecoveryConfiguration = null;
exports.getDisasterRecoveryConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDisasterRecoveryConfiguration", "getDisasterRecoveryConfigurationOutput"], () => require("./getDisasterRecoveryConfiguration"));
exports.getElasticPool = null;
exports.getElasticPoolOutput = null;
utilities.lazyLoad(exports, ["getElasticPool", "getElasticPoolOutput"], () => require("./getElasticPool"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getServer = null;
exports.getServerOutput = null;
utilities.lazyLoad(exports, ["getServer", "getServerOutput"], () => require("./getServer"));
exports.getServerAdvisor = null;
exports.getServerAdvisorOutput = null;
utilities.lazyLoad(exports, ["getServerAdvisor", "getServerAdvisorOutput"], () => require("./getServerAdvisor"));
exports.getServerAzureADAdministrator = null;
exports.getServerAzureADAdministratorOutput = null;
utilities.lazyLoad(exports, ["getServerAzureADAdministrator", "getServerAzureADAdministratorOutput"], () => require("./getServerAzureADAdministrator"));
exports.getServerCommunicationLink = null;
exports.getServerCommunicationLinkOutput = null;
utilities.lazyLoad(exports, ["getServerCommunicationLink", "getServerCommunicationLinkOutput"], () => require("./getServerCommunicationLink"));
exports.getTransparentDataEncryption = null;
exports.getTransparentDataEncryptionOutput = null;
utilities.lazyLoad(exports, ["getTransparentDataEncryption", "getTransparentDataEncryptionOutput"], () => require("./getTransparentDataEncryption"));
exports.Server = null;
utilities.lazyLoad(exports, ["Server"], () => require("./server"));
exports.ServerAdvisor = null;
utilities.lazyLoad(exports, ["ServerAdvisor"], () => require("./serverAdvisor"));
exports.ServerAzureADAdministrator = null;
utilities.lazyLoad(exports, ["ServerAzureADAdministrator"], () => require("./serverAzureADAdministrator"));
exports.ServerCommunicationLink = null;
utilities.lazyLoad(exports, ["ServerCommunicationLink"], () => require("./serverCommunicationLink"));
exports.TransparentDataEncryption = null;
utilities.lazyLoad(exports, ["TransparentDataEncryption"], () => require("./transparentDataEncryption"));
// Export enums:
__exportStar(require("../../types/enums/sql/v20140401"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:sql/v20140401:Database":
                return new exports.Database(name, undefined, { urn });
            case "azure-native:sql/v20140401:DatabaseAdvisor":
                return new exports.DatabaseAdvisor(name, undefined, { urn });
            case "azure-native:sql/v20140401:DatabaseThreatDetectionPolicy":
                return new exports.DatabaseThreatDetectionPolicy(name, undefined, { urn });
            case "azure-native:sql/v20140401:DisasterRecoveryConfiguration":
                return new exports.DisasterRecoveryConfiguration(name, undefined, { urn });
            case "azure-native:sql/v20140401:ElasticPool":
                return new exports.ElasticPool(name, undefined, { urn });
            case "azure-native:sql/v20140401:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:sql/v20140401:Server":
                return new exports.Server(name, undefined, { urn });
            case "azure-native:sql/v20140401:ServerAdvisor":
                return new exports.ServerAdvisor(name, undefined, { urn });
            case "azure-native:sql/v20140401:ServerAzureADAdministrator":
                return new exports.ServerAzureADAdministrator(name, undefined, { urn });
            case "azure-native:sql/v20140401:ServerCommunicationLink":
                return new exports.ServerCommunicationLink(name, undefined, { urn });
            case "azure-native:sql/v20140401:TransparentDataEncryption":
                return new exports.TransparentDataEncryption(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "sql/v20140401", _module);
