// @pulumi/azure-native/iotoperationsdataprocessor/v20231004preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pipeline = exports.Instance = exports.getPipelineOutput = exports.getPipeline = exports.getInstanceOutput = exports.getInstance = exports.getDatasetOutput = exports.getDataset = exports.Dataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.getDataset = null;
exports.getDatasetOutput = null;
utilities.lazyLoad(exports, ["getDataset", "getDatasetOutput"], () => require("./getDataset"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:iotoperationsdataprocessor/v20231004preview:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "azure-native:iotoperationsdataprocessor/v20231004preview:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "azure-native:iotoperationsdataprocessor/v20231004preview:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "iotoperationsdataprocessor/v20231004preview", _module);
