// @pulumi/azure-native/powerbi/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20200601 = exports.v20160129 = exports.WorkspaceCollection = exports.PrivateEndpointConnection = exports.PowerBIResource = exports.listWorkspaceCollectionAccessKeysOutput = exports.listWorkspaceCollectionAccessKeys = exports.getWorkspaceCollectionOutput = exports.getWorkspaceCollection = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getWorkspaceCollection = null;
exports.getWorkspaceCollectionOutput = null;
utilities.lazyLoad(exports, ["getWorkspaceCollection", "getWorkspaceCollectionOutput"], () => require("./getWorkspaceCollection"));
exports.listWorkspaceCollectionAccessKeys = null;
exports.listWorkspaceCollectionAccessKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceCollectionAccessKeys", "listWorkspaceCollectionAccessKeysOutput"], () => require("./listWorkspaceCollectionAccessKeys"));
exports.PowerBIResource = null;
utilities.lazyLoad(exports, ["PowerBIResource"], () => require("./powerBIResource"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.WorkspaceCollection = null;
utilities.lazyLoad(exports, ["WorkspaceCollection"], () => require("./workspaceCollection"));
// Export enums:
__exportStar(require("../types/enums/powerbi"), exports);
// Export sub-modules:
const v20160129 = require("./v20160129");
exports.v20160129 = v20160129;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:powerbi:PowerBIResource":
                return new exports.PowerBIResource(name, undefined, { urn });
            case "azure-native:powerbi:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:powerbi:WorkspaceCollection":
                return new exports.WorkspaceCollection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "powerbi", _module);
