// @pulumi/azure-native/types/enums/costmanagement/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingsKind = exports.ScheduledActionStatus = exports.ScheduledActionKind = exports.ScheduleFrequency = exports.RuleStatus = exports.ReportType = exports.ReportTimeframeType = exports.ReportGranularityType = exports.ReportConfigSortingType = exports.ReportColumnType = exports.RecurrenceType = exports.QueryColumnType = exports.PivotTypeType = exports.OperatorType = exports.MetricType = exports.KpiTypeType = exports.GranularityType = exports.FunctionType = exports.Frequency = exports.FormatType = exports.FileFormat = exports.ExportType = exports.DaysOfWeek = exports.CultureCode = exports.CostAllocationResourceType = exports.CostAllocationPolicyType = exports.ConnectorStatus = exports.ConnectorBillingModel = exports.ChartType = exports.CategoryType = exports.BudgetOperatorType = exports.BudgetNotificationOperatorType = exports.AccumulatedType = exports.v20241001preview = exports.v20240801 = exports.v20231101 = exports.v20230901 = exports.v20230801 = exports.v20230701preview = exports.v20230401preview = exports.v20230301 = exports.v20221005preview = exports.v20221001 = exports.v20200601 = exports.v20200301preview = exports.v20191101 = exports.v20191001 = exports.v20190401preview = exports.v20190301preview = exports.v20180801preview = void 0;
exports.WeeksOfMonth = exports.TimeframeType = exports.TimeGrainType = exports.ThresholdType = exports.StatusType = void 0;
// Export sub-modules:
const v20180801preview = require("./v20180801preview");
exports.v20180801preview = v20180801preview;
const v20190301preview = require("./v20190301preview");
exports.v20190301preview = v20190301preview;
const v20190401preview = require("./v20190401preview");
exports.v20190401preview = v20190401preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const v20221005preview = require("./v20221005preview");
exports.v20221005preview = v20221005preview;
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240801 = require("./v20240801");
exports.v20240801 = v20240801;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
exports.AccumulatedType = {
    True: "true",
    False: "false",
};
exports.BudgetNotificationOperatorType = {
    /**
     * Notification will be triggered if the evaluated cost is the same as threshold value. Note: It’s not recommended to use this OperatorType as there’s low chance of cost being exactly the same as threshold value, leading to missing of your alert. This OperatorType will be deprecated in future.
     *
     *  Supported for CategoryType(s): Cost.
     */
    EqualTo: "EqualTo",
    /**
     * Notification will be triggered if the evaluated cost is greater than the threshold value. Note: This is the recommended OperatorType while configuring Budget Alert.
     *
     *  Supported for CategoryType(s): Cost.
     */
    GreaterThan: "GreaterThan",
    /**
     * Notification will be triggered if the evaluated cost is greater than or equal to the threshold value.
     *
     *  Supported for CategoryType(s): Cost.
     */
    GreaterThanOrEqualTo: "GreaterThanOrEqualTo",
    /**
     * Notification will be triggered if any Reservations in the scope of the Reservation Utilization Alert Rule have a utilization less than the threshold percentage.
     *
     *  Supported for CategoryType(s): ReservationUtilization.
     */
    LessThan: "LessThan",
};
exports.BudgetOperatorType = {
    In: "In",
};
exports.CategoryType = {
    /**
     * A Budget that evaluates monetary cost of Azure resources against an amount, and alerts based on a configured notification threshold.
     */
    Cost: "Cost",
    /**
     * An Alert Rule that evaluates the utilization percentage of Azure Reservations, and alerts based on a configured notification threshold.
     */
    ReservationUtilization: "ReservationUtilization",
};
exports.ChartType = {
    Area: "Area",
    Line: "Line",
    StackedColumn: "StackedColumn",
    GroupedColumn: "GroupedColumn",
    Table: "Table",
};
exports.ConnectorBillingModel = {
    Trial: "trial",
    AutoUpgrade: "autoUpgrade",
    Premium: "premium",
    Expired: "expired",
};
exports.ConnectorStatus = {
    Active: "active",
    Error: "error",
    Suspended: "suspended",
};
exports.CostAllocationPolicyType = {
    FixedProportion: "FixedProportion",
};
exports.CostAllocationResourceType = {
    /**
     * Indicates an Azure dimension such as a subscription id or resource group name is being used for allocation.
     */
    Dimension: "Dimension",
    /**
     * Allocates cost based on Azure Tag key value pairs.
     */
    Tag: "Tag",
};
exports.CultureCode = {
    En_us: "en-us",
    Ja_jp: "ja-jp",
    Zh_cn: "zh-cn",
    De_de: "de-de",
    Es_es: "es-es",
    Fr_fr: "fr-fr",
    It_it: "it-it",
    Ko_kr: "ko-kr",
    Pt_br: "pt-br",
    Ru_ru: "ru-ru",
    Zh_tw: "zh-tw",
    Cs_cz: "cs-cz",
    Pl_pl: "pl-pl",
    Tr_tr: "tr-tr",
    Da_dk: "da-dk",
    En_gb: "en-gb",
    Hu_hu: "hu-hu",
    Nb_no: "nb-no",
    Nl_nl: "nl-nl",
    Pt_pt: "pt-pt",
    Sv_se: "sv-se",
};
exports.DaysOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.ExportType = {
    Usage: "Usage",
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
};
exports.FileFormat = {
    Csv: "Csv",
};
exports.FormatType = {
    Csv: "Csv",
};
exports.Frequency = {
    /**
     * After the threshold breaches and an Alert is fired, no further alerts will be sent until the next calendar day.
     */
    Daily: "Daily",
    /**
     * After the threshold breaches and an Alert is fired, no further alerts will be sent for 7 calendar days.
     */
    Weekly: "Weekly",
    /**
     * After the threshold breaches and an Alert is fired, no further alerts will be sent for 30 calendar days.
     */
    Monthly: "Monthly",
};
exports.FunctionType = {
    Sum: "Sum",
};
exports.GranularityType = {
    Daily: "Daily",
    Hourly: "Hourly",
};
exports.KpiTypeType = {
    Forecast: "Forecast",
    Budget: "Budget",
};
exports.MetricType = {
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
    AHUB: "AHUB",
};
exports.OperatorType = {
    In: "In",
    Contains: "Contains",
};
exports.PivotTypeType = {
    Dimension: "Dimension",
    TagKey: "TagKey",
};
exports.QueryColumnType = {
    /**
     * The tag associated with the cost data.
     */
    TagKey: "TagKey",
    /**
     * The dimension of cost data.
     */
    Dimension: "Dimension",
};
exports.RecurrenceType = {
    Daily: "Daily",
    Weekly: "Weekly",
    Monthly: "Monthly",
    Annually: "Annually",
};
exports.ReportColumnType = {
    Tag: "Tag",
    Dimension: "Dimension",
};
exports.ReportConfigSortingType = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.ReportGranularityType = {
    Daily: "Daily",
    Monthly: "Monthly",
};
exports.ReportTimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    YearToDate: "YearToDate",
    Custom: "Custom",
};
exports.ReportType = {
    Usage: "Usage",
};
exports.RuleStatus = {
    /**
     * Rule is saved but not used to allocate costs.
     */
    NotActive: "NotActive",
    /**
     * Rule is saved and impacting cost allocation.
     */
    Active: "Active",
    /**
     * Rule is saved and cost allocation is being updated. Readonly value that cannot be submitted in a put request.
     */
    Processing: "Processing",
};
exports.ScheduleFrequency = {
    /**
     * Cost analysis data will be emailed every day.
     */
    Daily: "Daily",
    /**
     * Cost analysis data will be emailed every week.
     */
    Weekly: "Weekly",
    /**
     * Cost analysis data will be emailed every month.
     */
    Monthly: "Monthly",
};
exports.ScheduledActionKind = {
    /**
     * Cost analysis data will be emailed.
     */
    Email: "Email",
    /**
     * Cost anomaly information will be emailed. Available only on subscription scope at daily frequency. If no anomaly is detected on the resource, an email won't be sent.
     */
    InsightAlert: "InsightAlert",
};
exports.ScheduledActionStatus = {
    /**
     * Scheduled action is saved but will not be run.
     */
    Disabled: "Disabled",
    /**
     * Scheduled action is saved and will be run.
     */
    Enabled: "Enabled",
    /**
     * Scheduled action is expired.
     */
    Expired: "Expired",
};
exports.SettingsKind = {
    Taginheritance: "taginheritance",
};
exports.StatusType = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.ThresholdType = {
    /**
     * Actual costs budget alerts notify when the actual accrued cost exceeds the allocated budget.
     */
    Actual: "Actual",
    /**
     * Forecasted costs budget alerts provide advanced notification that your spending trends are likely to exceed your allocated budget, as it relies on forecasted cost predictions.
     */
    Forecasted: "Forecasted",
};
exports.TimeGrainType = {
    /**
     * The budget will track costs in the current calendar month against the amount.
     *
     *  Supported for CategoryType: Cost only.
     */
    Monthly: "Monthly",
    /**
     * The budget will track costs in the current calendar quarter against the amount.
     *
     *  Supported for CategoryType: Cost only.
     */
    Quarterly: "Quarterly",
    /**
     * The budget will track costs in the current calendar year against the amount.
     *
     *  Supported for CategoryType: Cost only.
     */
    Annually: "Annually",
    /**
     * The budget will track costs in the current billing month against the amount.
     *
     *  Supported for CategoryType: Cost and Web Direct customers only.
     */
    BillingMonth: "BillingMonth",
    /**
     * The budget will track costs in the current billing quarter against the amount.
     *
     *  Supported for CategoryType: Cost and Web Direct customers only.
     */
    BillingQuarter: "BillingQuarter",
    /**
     * The budget will track costs in the current billing year against the amount.
     *
     *  Supported for CategoryType: Cost and Web Direct customers only.
     */
    BillingAnnual: "BillingAnnual",
    /**
     * The Reservation Utilization Alert Rule will evaluate reservations based on their 7-Day utilization percentage.
     *
     *  Supported for CategoryType: ReservationUtilization only.
     */
    Last7Days: "Last7Days",
    /**
     * The Reservation Utilization Alert Rule will evaluate reservations based on their 30-Day utilization percentage.
     *
     *  Supported for CategoryType: ReservationUtilization only.
     */
    Last30Days: "Last30Days",
};
exports.TimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    Custom: "Custom",
};
exports.WeeksOfMonth = {
    First: "First",
    Second: "Second",
    Third: "Third",
    Fourth: "Fourth",
    Last: "Last",
};
