// @pulumi/azure-native/dbformysql/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241001preview = exports.v20240601preview = exports.v20240201preview = exports.v20231230 = exports.v20231201preview = exports.v20231001preview = exports.v20230630 = exports.v20230601preview = exports.v20220930preview = exports.v20220101 = exports.v20211201preview = exports.v20210501preview = exports.v20210501 = exports.v20200701privatepreview = exports.v20200701preview = exports.v20180601privatepreview = exports.v20171201 = exports.Server = exports.PrivateEndpointConnection = exports.getServerOutput = exports.getServer = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getGetPrivateDnsZoneSuffixExecuteOutput = exports.getGetPrivateDnsZoneSuffixExecute = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getDatabaseOutput = exports.getDatabase = exports.getConfigurationOutput = exports.getConfiguration = exports.getAzureADAdministratorOutput = exports.getAzureADAdministrator = exports.FirewallRule = exports.Database = exports.Configuration = exports.AzureADAdministrator = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureADAdministrator = null;
utilities.lazyLoad(exports, ["AzureADAdministrator"], () => require("./azureADAdministrator"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getAzureADAdministrator = null;
exports.getAzureADAdministratorOutput = null;
utilities.lazyLoad(exports, ["getAzureADAdministrator", "getAzureADAdministratorOutput"], () => require("./getAzureADAdministrator"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getGetPrivateDnsZoneSuffixExecute = null;
exports.getGetPrivateDnsZoneSuffixExecuteOutput = null;
utilities.lazyLoad(exports, ["getGetPrivateDnsZoneSuffixExecute", "getGetPrivateDnsZoneSuffixExecuteOutput"], () => require("./getGetPrivateDnsZoneSuffixExecute"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getServer = null;
exports.getServerOutput = null;
utilities.lazyLoad(exports, ["getServer", "getServerOutput"], () => require("./getServer"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Server = null;
utilities.lazyLoad(exports, ["Server"], () => require("./server"));
// Export enums:
__exportStar(require("../types/enums/dbformysql"), exports);
// Export sub-modules:
const v20171201 = require("./v20171201");
exports.v20171201 = v20171201;
const v20180601privatepreview = require("./v20180601privatepreview");
exports.v20180601privatepreview = v20180601privatepreview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20200701privatepreview = require("./v20200701privatepreview");
exports.v20200701privatepreview = v20200701privatepreview;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220930preview = require("./v20220930preview");
exports.v20220930preview = v20220930preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20230630 = require("./v20230630");
exports.v20230630 = v20230630;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20231230 = require("./v20231230");
exports.v20231230 = v20231230;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbformysql:AzureADAdministrator":
                return new exports.AzureADAdministrator(name, undefined, { urn });
            case "azure-native:dbformysql:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            case "azure-native:dbformysql:Database":
                return new exports.Database(name, undefined, { urn });
            case "azure-native:dbformysql:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:dbformysql:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:dbformysql:Server":
                return new exports.Server(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbformysql", _module);
