// @pulumi/azure-native/cache/v20240901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedisEnterprise = exports.listDatabaseKeysOutput = exports.listDatabaseKeys = exports.getRedisEnterpriseOutput = exports.getRedisEnterprise = exports.getEnterprisePrivateEndpointConnectionOutput = exports.getEnterprisePrivateEndpointConnection = exports.getDatabaseOutput = exports.getDatabase = exports.getAccessPolicyAssignmentOutput = exports.getAccessPolicyAssignment = exports.EnterprisePrivateEndpointConnection = exports.Database = exports.AccessPolicyAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AccessPolicyAssignment = null;
utilities.lazyLoad(exports, ["AccessPolicyAssignment"], () => require("./accessPolicyAssignment"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.EnterprisePrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["EnterprisePrivateEndpointConnection"], () => require("./enterprisePrivateEndpointConnection"));
exports.getAccessPolicyAssignment = null;
exports.getAccessPolicyAssignmentOutput = null;
utilities.lazyLoad(exports, ["getAccessPolicyAssignment", "getAccessPolicyAssignmentOutput"], () => require("./getAccessPolicyAssignment"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getEnterprisePrivateEndpointConnection = null;
exports.getEnterprisePrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getEnterprisePrivateEndpointConnection", "getEnterprisePrivateEndpointConnectionOutput"], () => require("./getEnterprisePrivateEndpointConnection"));
exports.getRedisEnterprise = null;
exports.getRedisEnterpriseOutput = null;
utilities.lazyLoad(exports, ["getRedisEnterprise", "getRedisEnterpriseOutput"], () => require("./getRedisEnterprise"));
exports.listDatabaseKeys = null;
exports.listDatabaseKeysOutput = null;
utilities.lazyLoad(exports, ["listDatabaseKeys", "listDatabaseKeysOutput"], () => require("./listDatabaseKeys"));
exports.RedisEnterprise = null;
utilities.lazyLoad(exports, ["RedisEnterprise"], () => require("./redisEnterprise"));
// Export enums:
__exportStar(require("../../types/enums/cache/v20240901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cache/v20240901preview:AccessPolicyAssignment":
                return new exports.AccessPolicyAssignment(name, undefined, { urn });
            case "azure-native:cache/v20240901preview:Database":
                return new exports.Database(name, undefined, { urn });
            case "azure-native:cache/v20240901preview:EnterprisePrivateEndpointConnection":
                return new exports.EnterprisePrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:cache/v20240901preview:RedisEnterprise":
                return new exports.RedisEnterprise(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cache/v20240901preview", _module);
