// @pulumi/azure-native/types/enums/dbforpostgresql/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualEndpointType = exports.TriggerCutoverEnum = exports.StorageType = exports.StorageAutoGrow = exports.StartDataMigrationEnum = exports.SslMode = exports.SourceType = exports.SkuTier = exports.ServerVersion = exports.ServerPublicNetworkAccessState = exports.ReplicationRole = exports.ReplicationPromoteOption = exports.ReadReplicaPromoteMode = exports.PrivateEndpointServiceConnectionStatus = exports.PrincipalType = exports.PasswordAuthEnum = exports.OverwriteDbsInTargetEnum = exports.MigrationOption = exports.MigrationMode = exports.MigrateRolesEnum = exports.LogicalReplicationOnSourceDbEnum = exports.KeyStatusEnum = exports.IdentityType = exports.HighAvailabilityMode = exports.GeoRedundantBackupEnum = exports.CreateMode = exports.CancelEnum = exports.AzureManagedDiskPerformanceTiers = exports.ArmServerKeyType = exports.ActiveDirectoryAuthEnum = void 0;
exports.ActiveDirectoryAuthEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ArmServerKeyType = {
    SystemManaged: "SystemManaged",
    AzureKeyVault: "AzureKeyVault",
};
exports.AzureManagedDiskPerformanceTiers = {
    P1: "P1",
    P2: "P2",
    P3: "P3",
    P4: "P4",
    P6: "P6",
    P10: "P10",
    P15: "P15",
    P20: "P20",
    P30: "P30",
    P40: "P40",
    P50: "P50",
    P60: "P60",
    P70: "P70",
    P80: "P80",
};
exports.CancelEnum = {
    True: "True",
    False: "False",
};
exports.CreateMode = {
    Default: "Default",
    Create: "Create",
    Update: "Update",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
    ReviveDropped: "ReviveDropped",
};
exports.GeoRedundantBackupEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
    SameZone: "SameZone",
};
exports.IdentityType = {
    None: "None",
    UserAssigned: "UserAssigned",
    SystemAssigned: "SystemAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.KeyStatusEnum = {
    Valid: "Valid",
    Invalid: "Invalid",
};
exports.LogicalReplicationOnSourceDbEnum = {
    True: "True",
    False: "False",
};
exports.MigrateRolesEnum = {
    True: "True",
    False: "False",
};
exports.MigrationMode = {
    Offline: "Offline",
    Online: "Online",
};
exports.MigrationOption = {
    Validate: "Validate",
    Migrate: "Migrate",
    ValidateAndMigrate: "ValidateAndMigrate",
};
exports.OverwriteDbsInTargetEnum = {
    True: "True",
    False: "False",
};
exports.PasswordAuthEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PrincipalType = {
    Unknown: "Unknown",
    User: "User",
    Group: "Group",
    ServicePrincipal: "ServicePrincipal",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ReadReplicaPromoteMode = {
    Standalone: "standalone",
    Switchover: "switchover",
};
exports.ReplicationPromoteOption = {
    Planned: "planned",
    Forced: "forced",
};
exports.ReplicationRole = {
    None: "None",
    Primary: "Primary",
    AsyncReplica: "AsyncReplica",
    GeoAsyncReplica: "GeoAsyncReplica",
};
exports.ServerPublicNetworkAccessState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServerVersion = {
    ServerVersion_17: "17",
    ServerVersion_16: "16",
    ServerVersion_15: "15",
    ServerVersion_14: "14",
    ServerVersion_13: "13",
    ServerVersion_12: "12",
    ServerVersion_11: "11",
};
exports.SkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SourceType = {
    OnPremises: "OnPremises",
    AWS: "AWS",
    GCP: "GCP",
    AzureVM: "AzureVM",
    PostgreSQLSingleServer: "PostgreSQLSingleServer",
    AWS_RDS: "AWS_RDS",
    AWS_AURORA: "AWS_AURORA",
    AWS_EC2: "AWS_EC2",
    GCP_CloudSQL: "GCP_CloudSQL",
    GCP_AlloyDB: "GCP_AlloyDB",
    GCP_Compute: "GCP_Compute",
    EDB: "EDB",
    EDB_Oracle_Server: "EDB_Oracle_Server",
    EDB_PostgreSQL: "EDB_PostgreSQL",
    PostgreSQLFlexibleServer: "PostgreSQLFlexibleServer",
    PostgreSQLCosmosDB: "PostgreSQLCosmosDB",
    Huawei_RDS: "Huawei_RDS",
    Huawei_Compute: "Huawei_Compute",
    Heroku_PostgreSQL: "Heroku_PostgreSQL",
    Crunchy_PostgreSQL: "Crunchy_PostgreSQL",
    ApsaraDB_RDS: "ApsaraDB_RDS",
    Digital_Ocean_Droplets: "Digital_Ocean_Droplets",
    Digital_Ocean_PostgreSQL: "Digital_Ocean_PostgreSQL",
    Supabase_PostgreSQL: "Supabase_PostgreSQL",
};
exports.SslMode = {
    Prefer: "Prefer",
    Require: "Require",
    VerifyCA: "VerifyCA",
    VerifyFull: "VerifyFull",
};
exports.StartDataMigrationEnum = {
    True: "True",
    False: "False",
};
exports.StorageAutoGrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageType = {
    Premium_LRS: "Premium_LRS",
    PremiumV2_LRS: "PremiumV2_LRS",
    UltraSSD_LRS: "UltraSSD_LRS",
};
exports.TriggerCutoverEnum = {
    True: "True",
    False: "False",
};
exports.VirtualEndpointType = {
    ReadWrite: "ReadWrite",
};
