// @pulumi/azure-native/apimanagement/authorizationAccessPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizationAccessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Authorization access policy contract.
 *
 * Uses Azure REST API version 2022-08-01.
 *
 * Other available API versions: 2022-09-01-preview, 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-05-01, 2024-06-01-preview.
 */
class AuthorizationAccessPolicy extends pulumi.CustomResource {
    /**
     * Get an existing AuthorizationAccessPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AuthorizationAccessPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AuthorizationAccessPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AuthorizationAccessPolicy.__pulumiType;
    }
    /**
     * Create a AuthorizationAccessPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authorizationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizationId'");
            }
            if ((!args || args.authorizationProviderId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizationProviderId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["authorizationAccessPolicyId"] = args ? args.authorizationAccessPolicyId : undefined;
            resourceInputs["authorizationId"] = args ? args.authorizationId : undefined;
            resourceInputs["authorizationProviderId"] = args ? args.authorizationProviderId : undefined;
            resourceInputs["objectId"] = args ? args.objectId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20220401preview:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20220801:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20220901preview:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20230301preview:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20230501preview:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20230901preview:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20240501:AuthorizationAccessPolicy" }, { type: "azure-native:apimanagement/v20240601preview:AuthorizationAccessPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AuthorizationAccessPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AuthorizationAccessPolicy = AuthorizationAccessPolicy;
/** @internal */
AuthorizationAccessPolicy.__pulumiType = 'azure-native:apimanagement:AuthorizationAccessPolicy';
