// @pulumi/azure-native/edgeorder/v20220501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderItem = exports.listProductsAndConfigurationsOutput = exports.listProductsAndConfigurations = exports.listProductsAndConfigurationProductFamiliesOutput = exports.listProductsAndConfigurationProductFamilies = exports.getOrderItemOutput = exports.getOrderItem = exports.getAddressOutput = exports.getAddress = exports.Address = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Address = null;
utilities.lazyLoad(exports, ["Address"], () => require("./address"));
exports.getAddress = null;
exports.getAddressOutput = null;
utilities.lazyLoad(exports, ["getAddress", "getAddressOutput"], () => require("./getAddress"));
exports.getOrderItem = null;
exports.getOrderItemOutput = null;
utilities.lazyLoad(exports, ["getOrderItem", "getOrderItemOutput"], () => require("./getOrderItem"));
exports.listProductsAndConfigurationProductFamilies = null;
exports.listProductsAndConfigurationProductFamiliesOutput = null;
utilities.lazyLoad(exports, ["listProductsAndConfigurationProductFamilies", "listProductsAndConfigurationProductFamiliesOutput"], () => require("./listProductsAndConfigurationProductFamilies"));
exports.listProductsAndConfigurations = null;
exports.listProductsAndConfigurationsOutput = null;
utilities.lazyLoad(exports, ["listProductsAndConfigurations", "listProductsAndConfigurationsOutput"], () => require("./listProductsAndConfigurations"));
exports.OrderItem = null;
utilities.lazyLoad(exports, ["OrderItem"], () => require("./orderItem"));
// Export enums:
__exportStar(require("../../types/enums/edgeorder/v20220501preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:edgeorder/v20220501preview:Address":
                return new exports.Address(name, undefined, { urn });
            case "azure-native:edgeorder/v20220501preview:OrderItem":
                return new exports.OrderItem(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "edgeorder/v20220501preview", _module);
