// @pulumi/azure-native/migrate/v20230401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppCollectorOperation = exports.WebAppAssessmentV2Operation = exports.VmwareCollectorsOperation = exports.SqlCollectorOperation = exports.SqlAssessmentV2Operation = exports.ServerCollectorsOperation = exports.PrivateEndpointConnectionOperation = exports.ImportCollectorsOperation = exports.HypervCollectorsOperation = exports.GroupsOperation = exports.getWebAppCollectorOperationOutput = exports.getWebAppCollectorOperation = exports.getWebAppAssessmentV2OperationOutput = exports.getWebAppAssessmentV2Operation = exports.getVmwareCollectorsOperationOutput = exports.getVmwareCollectorsOperation = exports.getSqlCollectorOperationOutput = exports.getSqlCollectorOperation = exports.getSqlAssessmentV2OperationOutput = exports.getSqlAssessmentV2Operation = exports.getServerCollectorsOperationOutput = exports.getServerCollectorsOperation = exports.getPrivateEndpointConnectionOperationOutput = exports.getPrivateEndpointConnectionOperation = exports.getImportCollectorsOperationOutput = exports.getImportCollectorsOperation = exports.getHypervCollectorsOperationOutput = exports.getHypervCollectorsOperation = exports.getGroupsOperationOutput = exports.getGroupsOperation = exports.getBusinessCaseOperationReportDownloadUrlOutput = exports.getBusinessCaseOperationReportDownloadUrl = exports.getBusinessCaseOperationOutput = exports.getBusinessCaseOperation = exports.getAvsAssessmentsOperationOutput = exports.getAvsAssessmentsOperation = exports.getAssessmentsOperationOutput = exports.getAssessmentsOperation = exports.getAssessmentProjectsOperationOutput = exports.getAssessmentProjectsOperation = exports.getAksAssessmentOperationOutput = exports.getAksAssessmentOperation = exports.BusinessCaseOperation = exports.AvsAssessmentsOperation = exports.AssessmentsOperation = exports.AssessmentProjectsOperation = exports.AksAssessmentOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AksAssessmentOperation = null;
utilities.lazyLoad(exports, ["AksAssessmentOperation"], () => require("./aksAssessmentOperation"));
exports.AssessmentProjectsOperation = null;
utilities.lazyLoad(exports, ["AssessmentProjectsOperation"], () => require("./assessmentProjectsOperation"));
exports.AssessmentsOperation = null;
utilities.lazyLoad(exports, ["AssessmentsOperation"], () => require("./assessmentsOperation"));
exports.AvsAssessmentsOperation = null;
utilities.lazyLoad(exports, ["AvsAssessmentsOperation"], () => require("./avsAssessmentsOperation"));
exports.BusinessCaseOperation = null;
utilities.lazyLoad(exports, ["BusinessCaseOperation"], () => require("./businessCaseOperation"));
exports.getAksAssessmentOperation = null;
exports.getAksAssessmentOperationOutput = null;
utilities.lazyLoad(exports, ["getAksAssessmentOperation", "getAksAssessmentOperationOutput"], () => require("./getAksAssessmentOperation"));
exports.getAssessmentProjectsOperation = null;
exports.getAssessmentProjectsOperationOutput = null;
utilities.lazyLoad(exports, ["getAssessmentProjectsOperation", "getAssessmentProjectsOperationOutput"], () => require("./getAssessmentProjectsOperation"));
exports.getAssessmentsOperation = null;
exports.getAssessmentsOperationOutput = null;
utilities.lazyLoad(exports, ["getAssessmentsOperation", "getAssessmentsOperationOutput"], () => require("./getAssessmentsOperation"));
exports.getAvsAssessmentsOperation = null;
exports.getAvsAssessmentsOperationOutput = null;
utilities.lazyLoad(exports, ["getAvsAssessmentsOperation", "getAvsAssessmentsOperationOutput"], () => require("./getAvsAssessmentsOperation"));
exports.getBusinessCaseOperation = null;
exports.getBusinessCaseOperationOutput = null;
utilities.lazyLoad(exports, ["getBusinessCaseOperation", "getBusinessCaseOperationOutput"], () => require("./getBusinessCaseOperation"));
exports.getBusinessCaseOperationReportDownloadUrl = null;
exports.getBusinessCaseOperationReportDownloadUrlOutput = null;
utilities.lazyLoad(exports, ["getBusinessCaseOperationReportDownloadUrl", "getBusinessCaseOperationReportDownloadUrlOutput"], () => require("./getBusinessCaseOperationReportDownloadUrl"));
exports.getGroupsOperation = null;
exports.getGroupsOperationOutput = null;
utilities.lazyLoad(exports, ["getGroupsOperation", "getGroupsOperationOutput"], () => require("./getGroupsOperation"));
exports.getHypervCollectorsOperation = null;
exports.getHypervCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getHypervCollectorsOperation", "getHypervCollectorsOperationOutput"], () => require("./getHypervCollectorsOperation"));
exports.getImportCollectorsOperation = null;
exports.getImportCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getImportCollectorsOperation", "getImportCollectorsOperationOutput"], () => require("./getImportCollectorsOperation"));
exports.getPrivateEndpointConnectionOperation = null;
exports.getPrivateEndpointConnectionOperationOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionOperation", "getPrivateEndpointConnectionOperationOutput"], () => require("./getPrivateEndpointConnectionOperation"));
exports.getServerCollectorsOperation = null;
exports.getServerCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getServerCollectorsOperation", "getServerCollectorsOperationOutput"], () => require("./getServerCollectorsOperation"));
exports.getSqlAssessmentV2Operation = null;
exports.getSqlAssessmentV2OperationOutput = null;
utilities.lazyLoad(exports, ["getSqlAssessmentV2Operation", "getSqlAssessmentV2OperationOutput"], () => require("./getSqlAssessmentV2Operation"));
exports.getSqlCollectorOperation = null;
exports.getSqlCollectorOperationOutput = null;
utilities.lazyLoad(exports, ["getSqlCollectorOperation", "getSqlCollectorOperationOutput"], () => require("./getSqlCollectorOperation"));
exports.getVmwareCollectorsOperation = null;
exports.getVmwareCollectorsOperationOutput = null;
utilities.lazyLoad(exports, ["getVmwareCollectorsOperation", "getVmwareCollectorsOperationOutput"], () => require("./getVmwareCollectorsOperation"));
exports.getWebAppAssessmentV2Operation = null;
exports.getWebAppAssessmentV2OperationOutput = null;
utilities.lazyLoad(exports, ["getWebAppAssessmentV2Operation", "getWebAppAssessmentV2OperationOutput"], () => require("./getWebAppAssessmentV2Operation"));
exports.getWebAppCollectorOperation = null;
exports.getWebAppCollectorOperationOutput = null;
utilities.lazyLoad(exports, ["getWebAppCollectorOperation", "getWebAppCollectorOperationOutput"], () => require("./getWebAppCollectorOperation"));
exports.GroupsOperation = null;
utilities.lazyLoad(exports, ["GroupsOperation"], () => require("./groupsOperation"));
exports.HypervCollectorsOperation = null;
utilities.lazyLoad(exports, ["HypervCollectorsOperation"], () => require("./hypervCollectorsOperation"));
exports.ImportCollectorsOperation = null;
utilities.lazyLoad(exports, ["ImportCollectorsOperation"], () => require("./importCollectorsOperation"));
exports.PrivateEndpointConnectionOperation = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionOperation"], () => require("./privateEndpointConnectionOperation"));
exports.ServerCollectorsOperation = null;
utilities.lazyLoad(exports, ["ServerCollectorsOperation"], () => require("./serverCollectorsOperation"));
exports.SqlAssessmentV2Operation = null;
utilities.lazyLoad(exports, ["SqlAssessmentV2Operation"], () => require("./sqlAssessmentV2Operation"));
exports.SqlCollectorOperation = null;
utilities.lazyLoad(exports, ["SqlCollectorOperation"], () => require("./sqlCollectorOperation"));
exports.VmwareCollectorsOperation = null;
utilities.lazyLoad(exports, ["VmwareCollectorsOperation"], () => require("./vmwareCollectorsOperation"));
exports.WebAppAssessmentV2Operation = null;
utilities.lazyLoad(exports, ["WebAppAssessmentV2Operation"], () => require("./webAppAssessmentV2Operation"));
exports.WebAppCollectorOperation = null;
utilities.lazyLoad(exports, ["WebAppCollectorOperation"], () => require("./webAppCollectorOperation"));
// Export enums:
__exportStar(require("../../types/enums/migrate/v20230401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate/v20230401preview:AksAssessmentOperation":
                return new exports.AksAssessmentOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:AssessmentProjectsOperation":
                return new exports.AssessmentProjectsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:AssessmentsOperation":
                return new exports.AssessmentsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:AvsAssessmentsOperation":
                return new exports.AvsAssessmentsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:BusinessCaseOperation":
                return new exports.BusinessCaseOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:GroupsOperation":
                return new exports.GroupsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:HypervCollectorsOperation":
                return new exports.HypervCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:ImportCollectorsOperation":
                return new exports.ImportCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:PrivateEndpointConnectionOperation":
                return new exports.PrivateEndpointConnectionOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:ServerCollectorsOperation":
                return new exports.ServerCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:SqlAssessmentV2Operation":
                return new exports.SqlAssessmentV2Operation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:SqlCollectorOperation":
                return new exports.SqlCollectorOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:VmwareCollectorsOperation":
                return new exports.VmwareCollectorsOperation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:WebAppAssessmentV2Operation":
                return new exports.WebAppAssessmentV2Operation(name, undefined, { urn });
            case "azure-native:migrate/v20230401preview:WebAppCollectorOperation":
                return new exports.WebAppCollectorOperation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate/v20230401preview", _module);
