// @pulumi/azure-native/types/enums/agricultureplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.ManagedServiceIdentityType = exports.AuthCredentialsKind = exports.v20240601preview = void 0;
// Export sub-modules:
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
exports.AuthCredentialsKind = {
    /**
     * OAuth Client Credential type
     */
    OAuthClientCredentials: "OAuthClientCredentials",
    /**
     * API Key Auth Credential type
     */
    ApiKeyAuthCredentials: "ApiKeyAuthCredentials",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
