// @pulumi/azure-native/devhub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250301preview = exports.v20240801preview = exports.v20240501preview = exports.v20230801 = exports.v20221011preview = exports.v20220401preview = exports.Workflow = exports.IacProfile = exports.getWorkflowOutput = exports.getWorkflow = exports.getIacProfileOutput = exports.getIacProfile = exports.getGitHubOAuthOutput = exports.getGitHubOAuth = exports.getADOOAuthInfoOutput = exports.getADOOAuthInfo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getADOOAuthInfo = null;
exports.getADOOAuthInfoOutput = null;
utilities.lazyLoad(exports, ["getADOOAuthInfo", "getADOOAuthInfoOutput"], () => require("./getADOOAuthInfo"));
exports.getGitHubOAuth = null;
exports.getGitHubOAuthOutput = null;
utilities.lazyLoad(exports, ["getGitHubOAuth", "getGitHubOAuthOutput"], () => require("./getGitHubOAuth"));
exports.getIacProfile = null;
exports.getIacProfileOutput = null;
utilities.lazyLoad(exports, ["getIacProfile", "getIacProfileOutput"], () => require("./getIacProfile"));
exports.getWorkflow = null;
exports.getWorkflowOutput = null;
utilities.lazyLoad(exports, ["getWorkflow", "getWorkflowOutput"], () => require("./getWorkflow"));
exports.IacProfile = null;
utilities.lazyLoad(exports, ["IacProfile"], () => require("./iacProfile"));
exports.Workflow = null;
utilities.lazyLoad(exports, ["Workflow"], () => require("./workflow"));
// Export enums:
__exportStar(require("../types/enums/devhub"), exports);
// Export sub-modules:
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20221011preview = require("./v20221011preview");
exports.v20221011preview = v20221011preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20250301preview = require("./v20250301preview");
exports.v20250301preview = v20250301preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:devhub:IacProfile":
                return new exports.IacProfile(name, undefined, { urn });
            case "azure-native:devhub:Workflow":
                return new exports.Workflow(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "devhub", _module);
