// @pulumi/azure-native/mobilenetwork/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240401 = exports.v20240201 = exports.v20230901 = exports.v20230601 = exports.v20221101 = exports.v20220401preview = exports.v20220301preview = exports.Slice = exports.Site = exports.SimPolicy = exports.SimGroup = exports.Sim = exports.Service = exports.PacketCoreDataPlane = exports.PacketCoreControlPlane = exports.PacketCapture = exports.MobileNetwork = exports.listMobileNetworkSimIdsOutput = exports.listMobileNetworkSimIds = exports.listMobileNetworkSimGroupsOutput = exports.listMobileNetworkSimGroups = exports.getSliceOutput = exports.getSlice = exports.getSiteOutput = exports.getSite = exports.getSimPolicyOutput = exports.getSimPolicy = exports.getSimGroupOutput = exports.getSimGroup = exports.getSimOutput = exports.getSim = exports.getServiceOutput = exports.getService = exports.getPacketCoreDataPlaneOutput = exports.getPacketCoreDataPlane = exports.getPacketCoreControlPlaneOutput = exports.getPacketCoreControlPlane = exports.getPacketCaptureOutput = exports.getPacketCapture = exports.getMobileNetworkOutput = exports.getMobileNetwork = exports.getDiagnosticsPackageOutput = exports.getDiagnosticsPackage = exports.getDataNetworkOutput = exports.getDataNetwork = exports.getAttachedDataNetworkOutput = exports.getAttachedDataNetwork = exports.DiagnosticsPackage = exports.DataNetwork = exports.AttachedDataNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AttachedDataNetwork = null;
utilities.lazyLoad(exports, ["AttachedDataNetwork"], () => require("./attachedDataNetwork"));
exports.DataNetwork = null;
utilities.lazyLoad(exports, ["DataNetwork"], () => require("./dataNetwork"));
exports.DiagnosticsPackage = null;
utilities.lazyLoad(exports, ["DiagnosticsPackage"], () => require("./diagnosticsPackage"));
exports.getAttachedDataNetwork = null;
exports.getAttachedDataNetworkOutput = null;
utilities.lazyLoad(exports, ["getAttachedDataNetwork", "getAttachedDataNetworkOutput"], () => require("./getAttachedDataNetwork"));
exports.getDataNetwork = null;
exports.getDataNetworkOutput = null;
utilities.lazyLoad(exports, ["getDataNetwork", "getDataNetworkOutput"], () => require("./getDataNetwork"));
exports.getDiagnosticsPackage = null;
exports.getDiagnosticsPackageOutput = null;
utilities.lazyLoad(exports, ["getDiagnosticsPackage", "getDiagnosticsPackageOutput"], () => require("./getDiagnosticsPackage"));
exports.getMobileNetwork = null;
exports.getMobileNetworkOutput = null;
utilities.lazyLoad(exports, ["getMobileNetwork", "getMobileNetworkOutput"], () => require("./getMobileNetwork"));
exports.getPacketCapture = null;
exports.getPacketCaptureOutput = null;
utilities.lazyLoad(exports, ["getPacketCapture", "getPacketCaptureOutput"], () => require("./getPacketCapture"));
exports.getPacketCoreControlPlane = null;
exports.getPacketCoreControlPlaneOutput = null;
utilities.lazyLoad(exports, ["getPacketCoreControlPlane", "getPacketCoreControlPlaneOutput"], () => require("./getPacketCoreControlPlane"));
exports.getPacketCoreDataPlane = null;
exports.getPacketCoreDataPlaneOutput = null;
utilities.lazyLoad(exports, ["getPacketCoreDataPlane", "getPacketCoreDataPlaneOutput"], () => require("./getPacketCoreDataPlane"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSim = null;
exports.getSimOutput = null;
utilities.lazyLoad(exports, ["getSim", "getSimOutput"], () => require("./getSim"));
exports.getSimGroup = null;
exports.getSimGroupOutput = null;
utilities.lazyLoad(exports, ["getSimGroup", "getSimGroupOutput"], () => require("./getSimGroup"));
exports.getSimPolicy = null;
exports.getSimPolicyOutput = null;
utilities.lazyLoad(exports, ["getSimPolicy", "getSimPolicyOutput"], () => require("./getSimPolicy"));
exports.getSite = null;
exports.getSiteOutput = null;
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
exports.getSlice = null;
exports.getSliceOutput = null;
utilities.lazyLoad(exports, ["getSlice", "getSliceOutput"], () => require("./getSlice"));
exports.listMobileNetworkSimGroups = null;
exports.listMobileNetworkSimGroupsOutput = null;
utilities.lazyLoad(exports, ["listMobileNetworkSimGroups", "listMobileNetworkSimGroupsOutput"], () => require("./listMobileNetworkSimGroups"));
exports.listMobileNetworkSimIds = null;
exports.listMobileNetworkSimIdsOutput = null;
utilities.lazyLoad(exports, ["listMobileNetworkSimIds", "listMobileNetworkSimIdsOutput"], () => require("./listMobileNetworkSimIds"));
exports.MobileNetwork = null;
utilities.lazyLoad(exports, ["MobileNetwork"], () => require("./mobileNetwork"));
exports.PacketCapture = null;
utilities.lazyLoad(exports, ["PacketCapture"], () => require("./packetCapture"));
exports.PacketCoreControlPlane = null;
utilities.lazyLoad(exports, ["PacketCoreControlPlane"], () => require("./packetCoreControlPlane"));
exports.PacketCoreDataPlane = null;
utilities.lazyLoad(exports, ["PacketCoreDataPlane"], () => require("./packetCoreDataPlane"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.Sim = null;
utilities.lazyLoad(exports, ["Sim"], () => require("./sim"));
exports.SimGroup = null;
utilities.lazyLoad(exports, ["SimGroup"], () => require("./simGroup"));
exports.SimPolicy = null;
utilities.lazyLoad(exports, ["SimPolicy"], () => require("./simPolicy"));
exports.Site = null;
utilities.lazyLoad(exports, ["Site"], () => require("./site"));
exports.Slice = null;
utilities.lazyLoad(exports, ["Slice"], () => require("./slice"));
// Export enums:
__exportStar(require("../types/enums/mobilenetwork"), exports);
// Export sub-modules:
const v20220301preview = require("./v20220301preview");
exports.v20220301preview = v20220301preview;
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20221101 = require("./v20221101");
exports.v20221101 = v20221101;
const v20230601 = require("./v20230601");
exports.v20230601 = v20230601;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20240201 = require("./v20240201");
exports.v20240201 = v20240201;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mobilenetwork:AttachedDataNetwork":
                return new exports.AttachedDataNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork:DataNetwork":
                return new exports.DataNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork:DiagnosticsPackage":
                return new exports.DiagnosticsPackage(name, undefined, { urn });
            case "azure-native:mobilenetwork:MobileNetwork":
                return new exports.MobileNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork:PacketCapture":
                return new exports.PacketCapture(name, undefined, { urn });
            case "azure-native:mobilenetwork:PacketCoreControlPlane":
                return new exports.PacketCoreControlPlane(name, undefined, { urn });
            case "azure-native:mobilenetwork:PacketCoreDataPlane":
                return new exports.PacketCoreDataPlane(name, undefined, { urn });
            case "azure-native:mobilenetwork:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:mobilenetwork:Sim":
                return new exports.Sim(name, undefined, { urn });
            case "azure-native:mobilenetwork:SimGroup":
                return new exports.SimGroup(name, undefined, { urn });
            case "azure-native:mobilenetwork:SimPolicy":
                return new exports.SimPolicy(name, undefined, { urn });
            case "azure-native:mobilenetwork:Site":
                return new exports.Site(name, undefined, { urn });
            case "azure-native:mobilenetwork:Slice":
                return new exports.Slice(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mobilenetwork", _module);
