// @pulumi/azure-native/resources/v20220801preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDeploymentStackAtSubscriptionOutput = exports.getDeploymentStackAtSubscription = exports.getDeploymentStackAtResourceGroupOutput = exports.getDeploymentStackAtResourceGroup = exports.getDeploymentStackAtManagementGroupOutput = exports.getDeploymentStackAtManagementGroup = exports.DeploymentStackAtSubscription = exports.DeploymentStackAtResourceGroup = exports.DeploymentStackAtManagementGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DeploymentStackAtManagementGroup = null;
utilities.lazyLoad(exports, ["DeploymentStackAtManagementGroup"], () => require("./deploymentStackAtManagementGroup"));
exports.DeploymentStackAtResourceGroup = null;
utilities.lazyLoad(exports, ["DeploymentStackAtResourceGroup"], () => require("./deploymentStackAtResourceGroup"));
exports.DeploymentStackAtSubscription = null;
utilities.lazyLoad(exports, ["DeploymentStackAtSubscription"], () => require("./deploymentStackAtSubscription"));
exports.getDeploymentStackAtManagementGroup = null;
exports.getDeploymentStackAtManagementGroupOutput = null;
utilities.lazyLoad(exports, ["getDeploymentStackAtManagementGroup", "getDeploymentStackAtManagementGroupOutput"], () => require("./getDeploymentStackAtManagementGroup"));
exports.getDeploymentStackAtResourceGroup = null;
exports.getDeploymentStackAtResourceGroupOutput = null;
utilities.lazyLoad(exports, ["getDeploymentStackAtResourceGroup", "getDeploymentStackAtResourceGroupOutput"], () => require("./getDeploymentStackAtResourceGroup"));
exports.getDeploymentStackAtSubscription = null;
exports.getDeploymentStackAtSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getDeploymentStackAtSubscription", "getDeploymentStackAtSubscriptionOutput"], () => require("./getDeploymentStackAtSubscription"));
// Export enums:
__exportStar(require("../../types/enums/resources/v20220801preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:resources/v20220801preview:DeploymentStackAtManagementGroup":
                return new exports.DeploymentStackAtManagementGroup(name, undefined, { urn });
            case "azure-native:resources/v20220801preview:DeploymentStackAtResourceGroup":
                return new exports.DeploymentStackAtResourceGroup(name, undefined, { urn });
            case "azure-native:resources/v20220801preview:DeploymentStackAtSubscription":
                return new exports.DeploymentStackAtSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "resources/v20220801preview", _module);
