// @pulumi/azure-native/network/v20190801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getVirtualNetworkGatewayVpnProfilePackageUrlOutput = exports.getVirtualNetworkGatewayVpnProfilePackageUrl = exports.getVirtualNetworkGatewayLearnedRoutesOutput = exports.getVirtualNetworkGatewayLearnedRoutes = exports.getVirtualNetworkGatewayConnectionOutput = exports.getVirtualNetworkGatewayConnection = exports.getVirtualNetworkGatewayBgpPeerStatusOutput = exports.getVirtualNetworkGatewayBgpPeerStatus = exports.getVirtualNetworkGatewayAdvertisedRoutesOutput = exports.getVirtualNetworkGatewayAdvertisedRoutes = exports.getVirtualNetworkGatewayOutput = exports.getVirtualNetworkGateway = exports.getVirtualNetworkOutput = exports.getVirtualNetwork = exports.getSubnetOutput = exports.getSubnet = exports.getRouteFilterOutput = exports.getRouteFilter = exports.getPublicIPPrefixOutput = exports.getPublicIPPrefix = exports.getPublicIPAddressOutput = exports.getPublicIPAddress = exports.getPrivateLinkServiceOutput = exports.getPrivateLinkService = exports.getNetworkSecurityGroupOutput = exports.getNetworkSecurityGroup = exports.getNetworkProfileOutput = exports.getNetworkProfile = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getNatGatewayOutput = exports.getNatGateway = exports.getLocalNetworkGatewayOutput = exports.getLocalNetworkGateway = exports.getLoadBalancerOutput = exports.getLoadBalancer = exports.getExpressRoutePortOutput = exports.getExpressRoutePort = exports.getExpressRouteCrossConnectionPeeringOutput = exports.getExpressRouteCrossConnectionPeering = exports.getExpressRouteCircuitPeeringOutput = exports.getExpressRouteCircuitPeering = exports.getApplicationGatewayBackendHealthOnDemandOutput = exports.getApplicationGatewayBackendHealthOnDemand = exports.getApplicationGatewayOutput = exports.getApplicationGateway = exports.ExpressRoutePort = exports.ExpressRouteCrossConnectionPeering = exports.ExpressRouteCircuitPeering = exports.ApplicationGateway = void 0;
exports.VirtualNetworkGatewayConnection = exports.VirtualNetworkGateway = exports.VirtualNetwork = exports.Subnet = exports.RouteFilter = exports.PublicIPPrefix = exports.PublicIPAddress = exports.PrivateLinkService = exports.NetworkSecurityGroup = exports.NetworkProfile = exports.NetworkInterface = exports.NatGateway = exports.LocalNetworkGateway = exports.LoadBalancer = exports.getVirtualNetworkGatewayVpnclientIpsecParametersOutput = exports.getVirtualNetworkGatewayVpnclientIpsecParameters = exports.getVirtualNetworkGatewayVpnclientConnectionHealthOutput = exports.getVirtualNetworkGatewayVpnclientConnectionHealth = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ApplicationGateway = null;
utilities.lazyLoad(exports, ["ApplicationGateway"], () => require("./applicationGateway"));
exports.ExpressRouteCircuitPeering = null;
utilities.lazyLoad(exports, ["ExpressRouteCircuitPeering"], () => require("./expressRouteCircuitPeering"));
exports.ExpressRouteCrossConnectionPeering = null;
utilities.lazyLoad(exports, ["ExpressRouteCrossConnectionPeering"], () => require("./expressRouteCrossConnectionPeering"));
exports.ExpressRoutePort = null;
utilities.lazyLoad(exports, ["ExpressRoutePort"], () => require("./expressRoutePort"));
exports.getApplicationGateway = null;
exports.getApplicationGatewayOutput = null;
utilities.lazyLoad(exports, ["getApplicationGateway", "getApplicationGatewayOutput"], () => require("./getApplicationGateway"));
exports.getApplicationGatewayBackendHealthOnDemand = null;
exports.getApplicationGatewayBackendHealthOnDemandOutput = null;
utilities.lazyLoad(exports, ["getApplicationGatewayBackendHealthOnDemand", "getApplicationGatewayBackendHealthOnDemandOutput"], () => require("./getApplicationGatewayBackendHealthOnDemand"));
exports.getExpressRouteCircuitPeering = null;
exports.getExpressRouteCircuitPeeringOutput = null;
utilities.lazyLoad(exports, ["getExpressRouteCircuitPeering", "getExpressRouteCircuitPeeringOutput"], () => require("./getExpressRouteCircuitPeering"));
exports.getExpressRouteCrossConnectionPeering = null;
exports.getExpressRouteCrossConnectionPeeringOutput = null;
utilities.lazyLoad(exports, ["getExpressRouteCrossConnectionPeering", "getExpressRouteCrossConnectionPeeringOutput"], () => require("./getExpressRouteCrossConnectionPeering"));
exports.getExpressRoutePort = null;
exports.getExpressRoutePortOutput = null;
utilities.lazyLoad(exports, ["getExpressRoutePort", "getExpressRoutePortOutput"], () => require("./getExpressRoutePort"));
exports.getLoadBalancer = null;
exports.getLoadBalancerOutput = null;
utilities.lazyLoad(exports, ["getLoadBalancer", "getLoadBalancerOutput"], () => require("./getLoadBalancer"));
exports.getLocalNetworkGateway = null;
exports.getLocalNetworkGatewayOutput = null;
utilities.lazyLoad(exports, ["getLocalNetworkGateway", "getLocalNetworkGatewayOutput"], () => require("./getLocalNetworkGateway"));
exports.getNatGateway = null;
exports.getNatGatewayOutput = null;
utilities.lazyLoad(exports, ["getNatGateway", "getNatGatewayOutput"], () => require("./getNatGateway"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getNetworkProfile = null;
exports.getNetworkProfileOutput = null;
utilities.lazyLoad(exports, ["getNetworkProfile", "getNetworkProfileOutput"], () => require("./getNetworkProfile"));
exports.getNetworkSecurityGroup = null;
exports.getNetworkSecurityGroupOutput = null;
utilities.lazyLoad(exports, ["getNetworkSecurityGroup", "getNetworkSecurityGroupOutput"], () => require("./getNetworkSecurityGroup"));
exports.getPrivateLinkService = null;
exports.getPrivateLinkServiceOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkService", "getPrivateLinkServiceOutput"], () => require("./getPrivateLinkService"));
exports.getPublicIPAddress = null;
exports.getPublicIPAddressOutput = null;
utilities.lazyLoad(exports, ["getPublicIPAddress", "getPublicIPAddressOutput"], () => require("./getPublicIPAddress"));
exports.getPublicIPPrefix = null;
exports.getPublicIPPrefixOutput = null;
utilities.lazyLoad(exports, ["getPublicIPPrefix", "getPublicIPPrefixOutput"], () => require("./getPublicIPPrefix"));
exports.getRouteFilter = null;
exports.getRouteFilterOutput = null;
utilities.lazyLoad(exports, ["getRouteFilter", "getRouteFilterOutput"], () => require("./getRouteFilter"));
exports.getSubnet = null;
exports.getSubnetOutput = null;
utilities.lazyLoad(exports, ["getSubnet", "getSubnetOutput"], () => require("./getSubnet"));
exports.getVirtualNetwork = null;
exports.getVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetwork", "getVirtualNetworkOutput"], () => require("./getVirtualNetwork"));
exports.getVirtualNetworkGateway = null;
exports.getVirtualNetworkGatewayOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGateway", "getVirtualNetworkGatewayOutput"], () => require("./getVirtualNetworkGateway"));
exports.getVirtualNetworkGatewayAdvertisedRoutes = null;
exports.getVirtualNetworkGatewayAdvertisedRoutesOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayAdvertisedRoutes", "getVirtualNetworkGatewayAdvertisedRoutesOutput"], () => require("./getVirtualNetworkGatewayAdvertisedRoutes"));
exports.getVirtualNetworkGatewayBgpPeerStatus = null;
exports.getVirtualNetworkGatewayBgpPeerStatusOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayBgpPeerStatus", "getVirtualNetworkGatewayBgpPeerStatusOutput"], () => require("./getVirtualNetworkGatewayBgpPeerStatus"));
exports.getVirtualNetworkGatewayConnection = null;
exports.getVirtualNetworkGatewayConnectionOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayConnection", "getVirtualNetworkGatewayConnectionOutput"], () => require("./getVirtualNetworkGatewayConnection"));
exports.getVirtualNetworkGatewayLearnedRoutes = null;
exports.getVirtualNetworkGatewayLearnedRoutesOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayLearnedRoutes", "getVirtualNetworkGatewayLearnedRoutesOutput"], () => require("./getVirtualNetworkGatewayLearnedRoutes"));
exports.getVirtualNetworkGatewayVpnProfilePackageUrl = null;
exports.getVirtualNetworkGatewayVpnProfilePackageUrlOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayVpnProfilePackageUrl", "getVirtualNetworkGatewayVpnProfilePackageUrlOutput"], () => require("./getVirtualNetworkGatewayVpnProfilePackageUrl"));
exports.getVirtualNetworkGatewayVpnclientConnectionHealth = null;
exports.getVirtualNetworkGatewayVpnclientConnectionHealthOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayVpnclientConnectionHealth", "getVirtualNetworkGatewayVpnclientConnectionHealthOutput"], () => require("./getVirtualNetworkGatewayVpnclientConnectionHealth"));
exports.getVirtualNetworkGatewayVpnclientIpsecParameters = null;
exports.getVirtualNetworkGatewayVpnclientIpsecParametersOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkGatewayVpnclientIpsecParameters", "getVirtualNetworkGatewayVpnclientIpsecParametersOutput"], () => require("./getVirtualNetworkGatewayVpnclientIpsecParameters"));
exports.LoadBalancer = null;
utilities.lazyLoad(exports, ["LoadBalancer"], () => require("./loadBalancer"));
exports.LocalNetworkGateway = null;
utilities.lazyLoad(exports, ["LocalNetworkGateway"], () => require("./localNetworkGateway"));
exports.NatGateway = null;
utilities.lazyLoad(exports, ["NatGateway"], () => require("./natGateway"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.NetworkProfile = null;
utilities.lazyLoad(exports, ["NetworkProfile"], () => require("./networkProfile"));
exports.NetworkSecurityGroup = null;
utilities.lazyLoad(exports, ["NetworkSecurityGroup"], () => require("./networkSecurityGroup"));
exports.PrivateLinkService = null;
utilities.lazyLoad(exports, ["PrivateLinkService"], () => require("./privateLinkService"));
exports.PublicIPAddress = null;
utilities.lazyLoad(exports, ["PublicIPAddress"], () => require("./publicIPAddress"));
exports.PublicIPPrefix = null;
utilities.lazyLoad(exports, ["PublicIPPrefix"], () => require("./publicIPPrefix"));
exports.RouteFilter = null;
utilities.lazyLoad(exports, ["RouteFilter"], () => require("./routeFilter"));
exports.Subnet = null;
utilities.lazyLoad(exports, ["Subnet"], () => require("./subnet"));
exports.VirtualNetwork = null;
utilities.lazyLoad(exports, ["VirtualNetwork"], () => require("./virtualNetwork"));
exports.VirtualNetworkGateway = null;
utilities.lazyLoad(exports, ["VirtualNetworkGateway"], () => require("./virtualNetworkGateway"));
exports.VirtualNetworkGatewayConnection = null;
utilities.lazyLoad(exports, ["VirtualNetworkGatewayConnection"], () => require("./virtualNetworkGatewayConnection"));
// Export enums:
__exportStar(require("../../types/enums/network/v20190801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20190801:ApplicationGateway":
                return new exports.ApplicationGateway(name, undefined, { urn });
            case "azure-native:network/v20190801:ExpressRouteCircuitPeering":
                return new exports.ExpressRouteCircuitPeering(name, undefined, { urn });
            case "azure-native:network/v20190801:ExpressRouteCrossConnectionPeering":
                return new exports.ExpressRouteCrossConnectionPeering(name, undefined, { urn });
            case "azure-native:network/v20190801:ExpressRoutePort":
                return new exports.ExpressRoutePort(name, undefined, { urn });
            case "azure-native:network/v20190801:LoadBalancer":
                return new exports.LoadBalancer(name, undefined, { urn });
            case "azure-native:network/v20190801:LocalNetworkGateway":
                return new exports.LocalNetworkGateway(name, undefined, { urn });
            case "azure-native:network/v20190801:NatGateway":
                return new exports.NatGateway(name, undefined, { urn });
            case "azure-native:network/v20190801:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:network/v20190801:NetworkProfile":
                return new exports.NetworkProfile(name, undefined, { urn });
            case "azure-native:network/v20190801:NetworkSecurityGroup":
                return new exports.NetworkSecurityGroup(name, undefined, { urn });
            case "azure-native:network/v20190801:PrivateLinkService":
                return new exports.PrivateLinkService(name, undefined, { urn });
            case "azure-native:network/v20190801:PublicIPAddress":
                return new exports.PublicIPAddress(name, undefined, { urn });
            case "azure-native:network/v20190801:PublicIPPrefix":
                return new exports.PublicIPPrefix(name, undefined, { urn });
            case "azure-native:network/v20190801:RouteFilter":
                return new exports.RouteFilter(name, undefined, { urn });
            case "azure-native:network/v20190801:Subnet":
                return new exports.Subnet(name, undefined, { urn });
            case "azure-native:network/v20190801:VirtualNetwork":
                return new exports.VirtualNetwork(name, undefined, { urn });
            case "azure-native:network/v20190801:VirtualNetworkGateway":
                return new exports.VirtualNetworkGateway(name, undefined, { urn });
            case "azure-native:network/v20190801:VirtualNetworkGatewayConnection":
                return new exports.VirtualNetworkGatewayConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20190801", _module);
