// @pulumi/azure-native/aad/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20221201 = exports.OuContainer = exports.getOuContainerOutput = exports.getOuContainer = exports.getDomainServiceOutput = exports.getDomainService = exports.DomainService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DomainService = null;
utilities.lazyLoad(exports, ["DomainService"], () => require("./domainService"));
exports.getDomainService = null;
exports.getDomainServiceOutput = null;
utilities.lazyLoad(exports, ["getDomainService", "getDomainServiceOutput"], () => require("./getDomainService"));
exports.getOuContainer = null;
exports.getOuContainerOutput = null;
utilities.lazyLoad(exports, ["getOuContainer", "getOuContainerOutput"], () => require("./getOuContainer"));
exports.OuContainer = null;
utilities.lazyLoad(exports, ["OuContainer"], () => require("./ouContainer"));
// Export enums:
__exportStar(require("../types/enums/aad"), exports);
// Export sub-modules:
const v20221201 = require("./v20221201");
exports.v20221201 = v20221201;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:aad:DomainService":
                return new exports.DomainService(name, undefined, { urn });
            case "azure-native:aad:OuContainer":
                return new exports.OuContainer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "aad", _module);
