// @pulumi/azure-native/types/enums/attestation/v20210601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TpmAttestationAuthenticationType = exports.PublicNetworkAccessType = exports.PrivateEndpointServiceConnectionStatus = void 0;
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccessType = {
    /**
     * Enables public network connectivity to the Attestation Provider REST APIs.
     */
    Enabled: "Enabled",
    /**
     * Disables public network connectivity to the Attestation Provider REST APIs.
     */
    Disabled: "Disabled",
};
exports.TpmAttestationAuthenticationType = {
    /**
     * Enables the requirement of authentication for TPM Attestation REST APIs.
     */
    Enabled: "Enabled",
    /**
     * Disables the requirement of authentication for TPM Attestation REST APIs.
     */
    Disabled: "Disabled",
};
