// @pulumi/gcp/dataform/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryWorkflowConfig = exports.RepositoryReleaseConfig = exports.RepositoryIamPolicy = exports.RepositoryIamMember = exports.RepositoryIamBinding = exports.Repository = exports.getRepositoryIamPolicyOutput = exports.getRepositoryIamPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRepositoryIamPolicy = null;
exports.getRepositoryIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getRepositoryIamPolicy", "getRepositoryIamPolicyOutput"], () => require("./getRepositoryIamPolicy"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
exports.RepositoryIamBinding = null;
utilities.lazyLoad(exports, ["RepositoryIamBinding"], () => require("./repositoryIamBinding"));
exports.RepositoryIamMember = null;
utilities.lazyLoad(exports, ["RepositoryIamMember"], () => require("./repositoryIamMember"));
exports.RepositoryIamPolicy = null;
utilities.lazyLoad(exports, ["RepositoryIamPolicy"], () => require("./repositoryIamPolicy"));
exports.RepositoryReleaseConfig = null;
utilities.lazyLoad(exports, ["RepositoryReleaseConfig"], () => require("./repositoryReleaseConfig"));
exports.RepositoryWorkflowConfig = null;
utilities.lazyLoad(exports, ["RepositoryWorkflowConfig"], () => require("./repositoryWorkflowConfig"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:dataform/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            case "gcp:dataform/repositoryIamBinding:RepositoryIamBinding":
                return new exports.RepositoryIamBinding(name, undefined, { urn });
            case "gcp:dataform/repositoryIamMember:RepositoryIamMember":
                return new exports.RepositoryIamMember(name, undefined, { urn });
            case "gcp:dataform/repositoryIamPolicy:RepositoryIamPolicy":
                return new exports.RepositoryIamPolicy(name, undefined, { urn });
            case "gcp:dataform/repositoryReleaseConfig:RepositoryReleaseConfig":
                return new exports.RepositoryReleaseConfig(name, undefined, { urn });
            case "gcp:dataform/repositoryWorkflowConfig:RepositoryWorkflowConfig":
                return new exports.RepositoryWorkflowConfig(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "dataform/repository", _module);
pulumi.runtime.registerResourceModule("gcp", "dataform/repositoryIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "dataform/repositoryIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "dataform/repositoryIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "dataform/repositoryReleaseConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "dataform/repositoryWorkflowConfig", _module);
