// @pulumi/azure-native/network/v20200601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkLink = exports.VirtualHub = exports.Subnet = exports.PrivateZone = exports.PrivateRecordSet = exports.PacketCapture = exports.getVirtualNetworkLinkOutput = exports.getVirtualNetworkLink = exports.getVirtualHubOutput = exports.getVirtualHub = exports.getSubnetOutput = exports.getSubnet = exports.getPrivateZoneOutput = exports.getPrivateZone = exports.getPrivateRecordSetOutput = exports.getPrivateRecordSet = exports.getPacketCaptureOutput = exports.getPacketCapture = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPacketCapture = null;
exports.getPacketCaptureOutput = null;
utilities.lazyLoad(exports, ["getPacketCapture", "getPacketCaptureOutput"], () => require("./getPacketCapture"));
exports.getPrivateRecordSet = null;
exports.getPrivateRecordSetOutput = null;
utilities.lazyLoad(exports, ["getPrivateRecordSet", "getPrivateRecordSetOutput"], () => require("./getPrivateRecordSet"));
exports.getPrivateZone = null;
exports.getPrivateZoneOutput = null;
utilities.lazyLoad(exports, ["getPrivateZone", "getPrivateZoneOutput"], () => require("./getPrivateZone"));
exports.getSubnet = null;
exports.getSubnetOutput = null;
utilities.lazyLoad(exports, ["getSubnet", "getSubnetOutput"], () => require("./getSubnet"));
exports.getVirtualHub = null;
exports.getVirtualHubOutput = null;
utilities.lazyLoad(exports, ["getVirtualHub", "getVirtualHubOutput"], () => require("./getVirtualHub"));
exports.getVirtualNetworkLink = null;
exports.getVirtualNetworkLinkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkLink", "getVirtualNetworkLinkOutput"], () => require("./getVirtualNetworkLink"));
exports.PacketCapture = null;
utilities.lazyLoad(exports, ["PacketCapture"], () => require("./packetCapture"));
exports.PrivateRecordSet = null;
utilities.lazyLoad(exports, ["PrivateRecordSet"], () => require("./privateRecordSet"));
exports.PrivateZone = null;
utilities.lazyLoad(exports, ["PrivateZone"], () => require("./privateZone"));
exports.Subnet = null;
utilities.lazyLoad(exports, ["Subnet"], () => require("./subnet"));
exports.VirtualHub = null;
utilities.lazyLoad(exports, ["VirtualHub"], () => require("./virtualHub"));
exports.VirtualNetworkLink = null;
utilities.lazyLoad(exports, ["VirtualNetworkLink"], () => require("./virtualNetworkLink"));
// Export enums:
__exportStar(require("../../types/enums/network/v20200601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:network/v20200601:PacketCapture":
                return new exports.PacketCapture(name, undefined, { urn });
            case "azure-native:network/v20200601:PrivateRecordSet":
                return new exports.PrivateRecordSet(name, undefined, { urn });
            case "azure-native:network/v20200601:PrivateZone":
                return new exports.PrivateZone(name, undefined, { urn });
            case "azure-native:network/v20200601:Subnet":
                return new exports.Subnet(name, undefined, { urn });
            case "azure-native:network/v20200601:VirtualHub":
                return new exports.VirtualHub(name, undefined, { urn });
            case "azure-native:network/v20200601:VirtualNetworkLink":
                return new exports.VirtualNetworkLink(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "network/v20200601", _module);
