// @pulumi/azure-native/azurestackhci/v20221215preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetwork = exports.VirtualMachine = exports.VirtualHardDisk = exports.UpdateSummary = exports.UpdateRun = exports.Update = exports.StorageContainer = exports.NetworkInterface = exports.MarketplaceGalleryImage = exports.MachineExtension = exports.HybridIdentityMetadatum = exports.GuestAgent = exports.getVirtualNetworkOutput = exports.getVirtualNetwork = exports.getVirtualMachineOutput = exports.getVirtualMachine = exports.getVirtualHardDiskOutput = exports.getVirtualHardDisk = exports.getUpdateSummaryOutput = exports.getUpdateSummary = exports.getUpdateRunOutput = exports.getUpdateRun = exports.getUpdateOutput = exports.getUpdate = exports.getStorageContainerOutput = exports.getStorageContainer = exports.getNetworkInterfaceOutput = exports.getNetworkInterface = exports.getMarketplaceGalleryImageOutput = exports.getMarketplaceGalleryImage = exports.getMachineExtensionOutput = exports.getMachineExtension = exports.getHybridIdentityMetadatumOutput = exports.getHybridIdentityMetadatum = exports.getGuestAgentOutput = exports.getGuestAgent = exports.getGalleryImageOutput = exports.getGalleryImage = exports.getExtensionOutput = exports.getExtension = exports.getClusterOutput = exports.getCluster = exports.getArcSettingOutput = exports.getArcSetting = exports.GalleryImage = exports.Extension = exports.Cluster = exports.ArcSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ArcSetting = null;
utilities.lazyLoad(exports, ["ArcSetting"], () => require("./arcSetting"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.getArcSetting = null;
exports.getArcSettingOutput = null;
utilities.lazyLoad(exports, ["getArcSetting", "getArcSettingOutput"], () => require("./getArcSetting"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGuestAgent = null;
exports.getGuestAgentOutput = null;
utilities.lazyLoad(exports, ["getGuestAgent", "getGuestAgentOutput"], () => require("./getGuestAgent"));
exports.getHybridIdentityMetadatum = null;
exports.getHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getHybridIdentityMetadatum", "getHybridIdentityMetadatumOutput"], () => require("./getHybridIdentityMetadatum"));
exports.getMachineExtension = null;
exports.getMachineExtensionOutput = null;
utilities.lazyLoad(exports, ["getMachineExtension", "getMachineExtensionOutput"], () => require("./getMachineExtension"));
exports.getMarketplaceGalleryImage = null;
exports.getMarketplaceGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getMarketplaceGalleryImage", "getMarketplaceGalleryImageOutput"], () => require("./getMarketplaceGalleryImage"));
exports.getNetworkInterface = null;
exports.getNetworkInterfaceOutput = null;
utilities.lazyLoad(exports, ["getNetworkInterface", "getNetworkInterfaceOutput"], () => require("./getNetworkInterface"));
exports.getStorageContainer = null;
exports.getStorageContainerOutput = null;
utilities.lazyLoad(exports, ["getStorageContainer", "getStorageContainerOutput"], () => require("./getStorageContainer"));
exports.getUpdate = null;
exports.getUpdateOutput = null;
utilities.lazyLoad(exports, ["getUpdate", "getUpdateOutput"], () => require("./getUpdate"));
exports.getUpdateRun = null;
exports.getUpdateRunOutput = null;
utilities.lazyLoad(exports, ["getUpdateRun", "getUpdateRunOutput"], () => require("./getUpdateRun"));
exports.getUpdateSummary = null;
exports.getUpdateSummaryOutput = null;
utilities.lazyLoad(exports, ["getUpdateSummary", "getUpdateSummaryOutput"], () => require("./getUpdateSummary"));
exports.getVirtualHardDisk = null;
exports.getVirtualHardDiskOutput = null;
utilities.lazyLoad(exports, ["getVirtualHardDisk", "getVirtualHardDiskOutput"], () => require("./getVirtualHardDisk"));
exports.getVirtualMachine = null;
exports.getVirtualMachineOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachine", "getVirtualMachineOutput"], () => require("./getVirtualMachine"));
exports.getVirtualNetwork = null;
exports.getVirtualNetworkOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetwork", "getVirtualNetworkOutput"], () => require("./getVirtualNetwork"));
exports.GuestAgent = null;
utilities.lazyLoad(exports, ["GuestAgent"], () => require("./guestAgent"));
exports.HybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["HybridIdentityMetadatum"], () => require("./hybridIdentityMetadatum"));
exports.MachineExtension = null;
utilities.lazyLoad(exports, ["MachineExtension"], () => require("./machineExtension"));
exports.MarketplaceGalleryImage = null;
utilities.lazyLoad(exports, ["MarketplaceGalleryImage"], () => require("./marketplaceGalleryImage"));
exports.NetworkInterface = null;
utilities.lazyLoad(exports, ["NetworkInterface"], () => require("./networkInterface"));
exports.StorageContainer = null;
utilities.lazyLoad(exports, ["StorageContainer"], () => require("./storageContainer"));
exports.Update = null;
utilities.lazyLoad(exports, ["Update"], () => require("./update"));
exports.UpdateRun = null;
utilities.lazyLoad(exports, ["UpdateRun"], () => require("./updateRun"));
exports.UpdateSummary = null;
utilities.lazyLoad(exports, ["UpdateSummary"], () => require("./updateSummary"));
exports.VirtualHardDisk = null;
utilities.lazyLoad(exports, ["VirtualHardDisk"], () => require("./virtualHardDisk"));
exports.VirtualMachine = null;
utilities.lazyLoad(exports, ["VirtualMachine"], () => require("./virtualMachine"));
exports.VirtualNetwork = null;
utilities.lazyLoad(exports, ["VirtualNetwork"], () => require("./virtualNetwork"));
// Export enums:
__exportStar(require("../../types/enums/azurestackhci/v20221215preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestackhci/v20221215preview:ArcSetting":
                return new exports.ArcSetting(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:GuestAgent":
                return new exports.GuestAgent(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:HybridIdentityMetadatum":
                return new exports.HybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:MachineExtension":
                return new exports.MachineExtension(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:MarketplaceGalleryImage":
                return new exports.MarketplaceGalleryImage(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:NetworkInterface":
                return new exports.NetworkInterface(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:StorageContainer":
                return new exports.StorageContainer(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:Update":
                return new exports.Update(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:UpdateRun":
                return new exports.UpdateRun(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:UpdateSummary":
                return new exports.UpdateSummary(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:VirtualHardDisk":
                return new exports.VirtualHardDisk(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:VirtualMachine":
                return new exports.VirtualMachine(name, undefined, { urn });
            case "azure-native:azurestackhci/v20221215preview:VirtualNetwork":
                return new exports.VirtualNetwork(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestackhci/v20221215preview", _module);
