// @pulumi/azure-native/securityinsights/v20230201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SentinelOnboardingState = exports.OfficeDataConnector = exports.Metadata = exports.MDATPDataConnector = exports.MCASDataConnector = exports.IncidentRelation = exports.IncidentComment = exports.Incident = exports.getWatchlistItemOutput = exports.getWatchlistItem = exports.getWatchlistOutput = exports.getWatchlist = exports.getThreatIntelligenceIndicatorOutput = exports.getThreatIntelligenceIndicator = exports.getTIDataConnectorOutput = exports.getTIDataConnector = exports.getSentinelOnboardingStateOutput = exports.getSentinelOnboardingState = exports.getOfficeDataConnectorOutput = exports.getOfficeDataConnector = exports.getMetadataOutput = exports.getMetadata = exports.getMDATPDataConnectorOutput = exports.getMDATPDataConnector = exports.getMCASDataConnectorOutput = exports.getMCASDataConnector = exports.getIncidentRelationOutput = exports.getIncidentRelation = exports.getIncidentCommentOutput = exports.getIncidentComment = exports.getIncidentOutput = exports.getIncident = exports.getBookmarkOutput = exports.getBookmark = exports.getAwsCloudTrailDataConnectorOutput = exports.getAwsCloudTrailDataConnector = exports.getAutomationRuleOutput = exports.getAutomationRule = exports.getActionOutput = exports.getAction = exports.getAATPDataConnectorOutput = exports.getAATPDataConnector = exports.getAADDataConnectorOutput = exports.getAADDataConnector = exports.Bookmark = exports.AwsCloudTrailDataConnector = exports.AutomationRule = exports.Action = exports.AATPDataConnector = exports.AADDataConnector = void 0;
exports.WatchlistItem = exports.Watchlist = exports.TIDataConnector = exports.ThreatIntelligenceIndicator = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AADDataConnector = null;
utilities.lazyLoad(exports, ["AADDataConnector"], () => require("./aaddataConnector"));
exports.AATPDataConnector = null;
utilities.lazyLoad(exports, ["AATPDataConnector"], () => require("./aatpdataConnector"));
exports.Action = null;
utilities.lazyLoad(exports, ["Action"], () => require("./action"));
exports.AutomationRule = null;
utilities.lazyLoad(exports, ["AutomationRule"], () => require("./automationRule"));
exports.AwsCloudTrailDataConnector = null;
utilities.lazyLoad(exports, ["AwsCloudTrailDataConnector"], () => require("./awsCloudTrailDataConnector"));
exports.Bookmark = null;
utilities.lazyLoad(exports, ["Bookmark"], () => require("./bookmark"));
exports.getAADDataConnector = null;
exports.getAADDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAADDataConnector", "getAADDataConnectorOutput"], () => require("./getAADDataConnector"));
exports.getAATPDataConnector = null;
exports.getAATPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAATPDataConnector", "getAATPDataConnectorOutput"], () => require("./getAATPDataConnector"));
exports.getAction = null;
exports.getActionOutput = null;
utilities.lazyLoad(exports, ["getAction", "getActionOutput"], () => require("./getAction"));
exports.getAutomationRule = null;
exports.getAutomationRuleOutput = null;
utilities.lazyLoad(exports, ["getAutomationRule", "getAutomationRuleOutput"], () => require("./getAutomationRule"));
exports.getAwsCloudTrailDataConnector = null;
exports.getAwsCloudTrailDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAwsCloudTrailDataConnector", "getAwsCloudTrailDataConnectorOutput"], () => require("./getAwsCloudTrailDataConnector"));
exports.getBookmark = null;
exports.getBookmarkOutput = null;
utilities.lazyLoad(exports, ["getBookmark", "getBookmarkOutput"], () => require("./getBookmark"));
exports.getIncident = null;
exports.getIncidentOutput = null;
utilities.lazyLoad(exports, ["getIncident", "getIncidentOutput"], () => require("./getIncident"));
exports.getIncidentComment = null;
exports.getIncidentCommentOutput = null;
utilities.lazyLoad(exports, ["getIncidentComment", "getIncidentCommentOutput"], () => require("./getIncidentComment"));
exports.getIncidentRelation = null;
exports.getIncidentRelationOutput = null;
utilities.lazyLoad(exports, ["getIncidentRelation", "getIncidentRelationOutput"], () => require("./getIncidentRelation"));
exports.getMCASDataConnector = null;
exports.getMCASDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMCASDataConnector", "getMCASDataConnectorOutput"], () => require("./getMCASDataConnector"));
exports.getMDATPDataConnector = null;
exports.getMDATPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMDATPDataConnector", "getMDATPDataConnectorOutput"], () => require("./getMDATPDataConnector"));
exports.getMetadata = null;
exports.getMetadataOutput = null;
utilities.lazyLoad(exports, ["getMetadata", "getMetadataOutput"], () => require("./getMetadata"));
exports.getOfficeDataConnector = null;
exports.getOfficeDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getOfficeDataConnector", "getOfficeDataConnectorOutput"], () => require("./getOfficeDataConnector"));
exports.getSentinelOnboardingState = null;
exports.getSentinelOnboardingStateOutput = null;
utilities.lazyLoad(exports, ["getSentinelOnboardingState", "getSentinelOnboardingStateOutput"], () => require("./getSentinelOnboardingState"));
exports.getTIDataConnector = null;
exports.getTIDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getTIDataConnector", "getTIDataConnectorOutput"], () => require("./getTIDataConnector"));
exports.getThreatIntelligenceIndicator = null;
exports.getThreatIntelligenceIndicatorOutput = null;
utilities.lazyLoad(exports, ["getThreatIntelligenceIndicator", "getThreatIntelligenceIndicatorOutput"], () => require("./getThreatIntelligenceIndicator"));
exports.getWatchlist = null;
exports.getWatchlistOutput = null;
utilities.lazyLoad(exports, ["getWatchlist", "getWatchlistOutput"], () => require("./getWatchlist"));
exports.getWatchlistItem = null;
exports.getWatchlistItemOutput = null;
utilities.lazyLoad(exports, ["getWatchlistItem", "getWatchlistItemOutput"], () => require("./getWatchlistItem"));
exports.Incident = null;
utilities.lazyLoad(exports, ["Incident"], () => require("./incident"));
exports.IncidentComment = null;
utilities.lazyLoad(exports, ["IncidentComment"], () => require("./incidentComment"));
exports.IncidentRelation = null;
utilities.lazyLoad(exports, ["IncidentRelation"], () => require("./incidentRelation"));
exports.MCASDataConnector = null;
utilities.lazyLoad(exports, ["MCASDataConnector"], () => require("./mcasdataConnector"));
exports.MDATPDataConnector = null;
utilities.lazyLoad(exports, ["MDATPDataConnector"], () => require("./mdatpdataConnector"));
exports.Metadata = null;
utilities.lazyLoad(exports, ["Metadata"], () => require("./metadata"));
exports.OfficeDataConnector = null;
utilities.lazyLoad(exports, ["OfficeDataConnector"], () => require("./officeDataConnector"));
exports.SentinelOnboardingState = null;
utilities.lazyLoad(exports, ["SentinelOnboardingState"], () => require("./sentinelOnboardingState"));
exports.ThreatIntelligenceIndicator = null;
utilities.lazyLoad(exports, ["ThreatIntelligenceIndicator"], () => require("./threatIntelligenceIndicator"));
exports.TIDataConnector = null;
utilities.lazyLoad(exports, ["TIDataConnector"], () => require("./tidataConnector"));
exports.Watchlist = null;
utilities.lazyLoad(exports, ["Watchlist"], () => require("./watchlist"));
exports.WatchlistItem = null;
utilities.lazyLoad(exports, ["WatchlistItem"], () => require("./watchlistItem"));
// Export enums:
__exportStar(require("../../types/enums/securityinsights/v20230201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:securityinsights/v20230201:AADDataConnector":
                return new exports.AADDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:AATPDataConnector":
                return new exports.AATPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:Action":
                return new exports.Action(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:AutomationRule":
                return new exports.AutomationRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:AwsCloudTrailDataConnector":
                return new exports.AwsCloudTrailDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:Bookmark":
                return new exports.Bookmark(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:Incident":
                return new exports.Incident(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:IncidentComment":
                return new exports.IncidentComment(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:IncidentRelation":
                return new exports.IncidentRelation(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:MCASDataConnector":
                return new exports.MCASDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:MDATPDataConnector":
                return new exports.MDATPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:Metadata":
                return new exports.Metadata(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:OfficeDataConnector":
                return new exports.OfficeDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:SentinelOnboardingState":
                return new exports.SentinelOnboardingState(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:TIDataConnector":
                return new exports.TIDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:ThreatIntelligenceIndicator":
                return new exports.ThreatIntelligenceIndicator(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:Watchlist":
                return new exports.Watchlist(name, undefined, { urn });
            case "azure-native:securityinsights/v20230201:WatchlistItem":
                return new exports.WatchlistItem(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "securityinsights/v20230201", _module);
