// @pulumi/azure-native/datamigration/v20220330preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Task = exports.SqlMigrationService = exports.ServiceTask = exports.Service = exports.Project = exports.listSqlMigrationServiceMonitoringDataOutput = exports.listSqlMigrationServiceMonitoringData = exports.listSqlMigrationServiceAuthKeysOutput = exports.listSqlMigrationServiceAuthKeys = exports.getTaskOutput = exports.getTask = exports.getSqlMigrationServiceOutput = exports.getSqlMigrationService = exports.getServiceTaskOutput = exports.getServiceTask = exports.getServiceOutput = exports.getService = exports.getProjectOutput = exports.getProject = exports.getFileOutput = exports.getFile = exports.getDatabaseMigrationsSqlDbOutput = exports.getDatabaseMigrationsSqlDb = exports.File = exports.DatabaseMigrationsSqlDb = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DatabaseMigrationsSqlDb = null;
utilities.lazyLoad(exports, ["DatabaseMigrationsSqlDb"], () => require("./databaseMigrationsSqlDb"));
exports.File = null;
utilities.lazyLoad(exports, ["File"], () => require("./file"));
exports.getDatabaseMigrationsSqlDb = null;
exports.getDatabaseMigrationsSqlDbOutput = null;
utilities.lazyLoad(exports, ["getDatabaseMigrationsSqlDb", "getDatabaseMigrationsSqlDbOutput"], () => require("./getDatabaseMigrationsSqlDb"));
exports.getFile = null;
exports.getFileOutput = null;
utilities.lazyLoad(exports, ["getFile", "getFileOutput"], () => require("./getFile"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceTask = null;
exports.getServiceTaskOutput = null;
utilities.lazyLoad(exports, ["getServiceTask", "getServiceTaskOutput"], () => require("./getServiceTask"));
exports.getSqlMigrationService = null;
exports.getSqlMigrationServiceOutput = null;
utilities.lazyLoad(exports, ["getSqlMigrationService", "getSqlMigrationServiceOutput"], () => require("./getSqlMigrationService"));
exports.getTask = null;
exports.getTaskOutput = null;
utilities.lazyLoad(exports, ["getTask", "getTaskOutput"], () => require("./getTask"));
exports.listSqlMigrationServiceAuthKeys = null;
exports.listSqlMigrationServiceAuthKeysOutput = null;
utilities.lazyLoad(exports, ["listSqlMigrationServiceAuthKeys", "listSqlMigrationServiceAuthKeysOutput"], () => require("./listSqlMigrationServiceAuthKeys"));
exports.listSqlMigrationServiceMonitoringData = null;
exports.listSqlMigrationServiceMonitoringDataOutput = null;
utilities.lazyLoad(exports, ["listSqlMigrationServiceMonitoringData", "listSqlMigrationServiceMonitoringDataOutput"], () => require("./listSqlMigrationServiceMonitoringData"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceTask = null;
utilities.lazyLoad(exports, ["ServiceTask"], () => require("./serviceTask"));
exports.SqlMigrationService = null;
utilities.lazyLoad(exports, ["SqlMigrationService"], () => require("./sqlMigrationService"));
exports.Task = null;
utilities.lazyLoad(exports, ["Task"], () => require("./task"));
// Export enums:
__exportStar(require("../../types/enums/datamigration/v20220330preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datamigration/v20220330preview:DatabaseMigrationsSqlDb":
                return new exports.DatabaseMigrationsSqlDb(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:File":
                return new exports.File(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:ServiceTask":
                return new exports.ServiceTask(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:SqlMigrationService":
                return new exports.SqlMigrationService(name, undefined, { urn });
            case "azure-native:datamigration/v20220330preview:Task":
                return new exports.Task(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datamigration/v20220330preview", _module);
