// @pulumi/azure-native/compute/galleryInVMAccessControlProfileVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GalleryInVMAccessControlProfileVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Specifies information about the gallery inVMAccessControlProfile version that you want to create or update.
 *
 * Uses Azure REST API version 2024-03-03.
 */
class GalleryInVMAccessControlProfileVersion extends pulumi.CustomResource {
    /**
     * Get an existing GalleryInVMAccessControlProfileVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GalleryInVMAccessControlProfileVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GalleryInVMAccessControlProfileVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GalleryInVMAccessControlProfileVersion.__pulumiType;
    }
    /**
     * Create a GalleryInVMAccessControlProfileVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.defaultAccess === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultAccess'");
            }
            if ((!args || args.galleryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'galleryName'");
            }
            if ((!args || args.inVMAccessControlProfileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inVMAccessControlProfileName'");
            }
            if ((!args || args.mode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mode'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["defaultAccess"] = args ? args.defaultAccess : undefined;
            resourceInputs["excludeFromLatest"] = args ? args.excludeFromLatest : undefined;
            resourceInputs["galleryName"] = args ? args.galleryName : undefined;
            resourceInputs["inVMAccessControlProfileName"] = args ? args.inVMAccessControlProfileName : undefined;
            resourceInputs["inVMAccessControlProfileVersionName"] = args ? args.inVMAccessControlProfileVersionName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mode"] = args ? args.mode : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rules"] = args ? args.rules : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetLocations"] = args ? args.targetLocations : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publishedDate"] = undefined /*out*/;
            resourceInputs["replicationStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["defaultAccess"] = undefined /*out*/;
            resourceInputs["excludeFromLatest"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publishedDate"] = undefined /*out*/;
            resourceInputs["replicationStatus"] = undefined /*out*/;
            resourceInputs["rules"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetLocations"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:compute/v20240303:GalleryInVMAccessControlProfileVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GalleryInVMAccessControlProfileVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GalleryInVMAccessControlProfileVersion = GalleryInVMAccessControlProfileVersion;
/** @internal */
GalleryInVMAccessControlProfileVersion.__pulumiType = 'azure-native:compute:GalleryInVMAccessControlProfileVersion';
