// @pulumi/azure-native/aadiam/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20200701preview = exports.v20200301preview = exports.v20200301 = exports.v20170401preview = exports.v20170401 = exports.PrivateLinkForAzureAd = exports.PrivateEndpointConnection = exports.getPrivateLinkForAzureAdOutput = exports.getPrivateLinkForAzureAd = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getDiagnosticSettingOutput = exports.getDiagnosticSetting = exports.getAzureADMetricOutput = exports.getAzureADMetric = exports.DiagnosticSetting = exports.AzureADMetric = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureADMetric = null;
utilities.lazyLoad(exports, ["AzureADMetric"], () => require("./azureADMetric"));
exports.DiagnosticSetting = null;
utilities.lazyLoad(exports, ["DiagnosticSetting"], () => require("./diagnosticSetting"));
exports.getAzureADMetric = null;
exports.getAzureADMetricOutput = null;
utilities.lazyLoad(exports, ["getAzureADMetric", "getAzureADMetricOutput"], () => require("./getAzureADMetric"));
exports.getDiagnosticSetting = null;
exports.getDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getDiagnosticSetting", "getDiagnosticSettingOutput"], () => require("./getDiagnosticSetting"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkForAzureAd = null;
exports.getPrivateLinkForAzureAdOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkForAzureAd", "getPrivateLinkForAzureAdOutput"], () => require("./getPrivateLinkForAzureAd"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkForAzureAd = null;
utilities.lazyLoad(exports, ["PrivateLinkForAzureAd"], () => require("./privateLinkForAzureAd"));
// Export enums:
__exportStar(require("../types/enums/aadiam"), exports);
// Export sub-modules:
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20170401preview = require("./v20170401preview");
exports.v20170401preview = v20170401preview;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:aadiam:AzureADMetric":
                return new exports.AzureADMetric(name, undefined, { urn });
            case "azure-native:aadiam:DiagnosticSetting":
                return new exports.DiagnosticSetting(name, undefined, { urn });
            case "azure-native:aadiam:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:aadiam:PrivateLinkForAzureAd":
                return new exports.PrivateLinkForAzureAd(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "aadiam", _module);
