// @pulumi/azure-native/types/enums/voiceservices/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLinePurpose = exports.TeamsCodecs = exports.ManagedServiceIdentityType = exports.E911Type = exports.Connectivity = exports.CommunicationsPlatform = exports.AutoGeneratedDomainNameLabelScope = exports.v20230901 = exports.v20230403 = exports.v20221201preview = void 0;
// Export sub-modules:
const v20221201preview = require("./v20221201preview");
exports.v20221201preview = v20221201preview;
const v20230403 = require("./v20230403");
exports.v20230403 = v20230403;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
exports.AutoGeneratedDomainNameLabelScope = {
    TenantReuse: "TenantReuse",
    SubscriptionReuse: "SubscriptionReuse",
    ResourceGroupReuse: "ResourceGroupReuse",
    NoReuse: "NoReuse",
};
exports.CommunicationsPlatform = {
    OperatorConnect: "OperatorConnect",
    TeamsPhoneMobile: "TeamsPhoneMobile",
};
exports.Connectivity = {
    /**
     * This deployment connects to the operator network using a Public IP address, e.g. when using MAPS
     */
    PublicAddress: "PublicAddress",
};
exports.E911Type = {
    /**
     * Emergency calls are not handled different from other calls
     */
    Standard: "Standard",
    /**
     * Emergency calls are routed directly to the ESRP
     */
    DirectToEsrp: "DirectToEsrp",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.TeamsCodecs = {
    PCMA: "PCMA",
    PCMU: "PCMU",
    G722: "G722",
    G722_2: "G722_2",
    SILK_8: "SILK_8",
    SILK_16: "SILK_16",
};
exports.TestLinePurpose = {
    Manual: "Manual",
    Automated: "Automated",
};
