// @pulumi/azure-native/machinelearningservices/v20241001preview/connectionRaiBlocklistItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionRaiBlocklistItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class ConnectionRaiBlocklistItem extends pulumi.CustomResource {
    /**
     * Get an existing ConnectionRaiBlocklistItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConnectionRaiBlocklistItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConnectionRaiBlocklistItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectionRaiBlocklistItem.__pulumiType;
    }
    /**
     * Create a ConnectionRaiBlocklistItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectionName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.raiBlocklistName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'raiBlocklistName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["connectionName"] = args ? args.connectionName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["proxyApiVersion"] = args ? args.proxyApiVersion : undefined;
            resourceInputs["raiBlocklistItemName"] = args ? args.raiBlocklistItemName : undefined;
            resourceInputs["raiBlocklistName"] = args ? args.raiBlocklistName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20240401preview:ConnectionRaiBlocklistItem" }, { type: "azure-native:machinelearningservices/v20240701preview:ConnectionRaiBlocklistItem" }, { type: "azure-native:machinelearningservices/v20250101preview:ConnectionRaiBlocklistItem" }, { type: "azure-native:machinelearningservices:ConnectionRaiBlocklistItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConnectionRaiBlocklistItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConnectionRaiBlocklistItem = ConnectionRaiBlocklistItem;
/** @internal */
ConnectionRaiBlocklistItem.__pulumiType = 'azure-native:machinelearningservices/v20241001preview:ConnectionRaiBlocklistItem';
