// @pulumi/azure-native/automation/v20230515preview/hybridRunbookWorkerGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HybridRunbookWorkerGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of hybrid runbook worker group.
 */
class HybridRunbookWorkerGroup extends pulumi.CustomResource {
    /**
     * Get an existing HybridRunbookWorkerGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HybridRunbookWorkerGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HybridRunbookWorkerGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HybridRunbookWorkerGroup.__pulumiType;
    }
    /**
     * Create a HybridRunbookWorkerGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.automationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'automationAccountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["automationAccountName"] = args ? args.automationAccountName : undefined;
            resourceInputs["credential"] = args ? args.credential : undefined;
            resourceInputs["hybridRunbookWorkerGroupName"] = args ? args.hybridRunbookWorkerGroupName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["credential"] = undefined /*out*/;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:automation/v20210622:HybridRunbookWorkerGroup" }, { type: "azure-native:automation/v20220222:HybridRunbookWorkerGroup" }, { type: "azure-native:automation/v20220808:HybridRunbookWorkerGroup" }, { type: "azure-native:automation/v20231101:HybridRunbookWorkerGroup" }, { type: "azure-native:automation/v20241023:HybridRunbookWorkerGroup" }, { type: "azure-native:automation:HybridRunbookWorkerGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HybridRunbookWorkerGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HybridRunbookWorkerGroup = HybridRunbookWorkerGroup;
/** @internal */
HybridRunbookWorkerGroup.__pulumiType = 'azure-native:automation/v20230515preview:HybridRunbookWorkerGroup';
