// @pulumi/azure-native/types/enums/timeseriesinsights/v20210630preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageLimitExceededBehavior = exports.SkuName = exports.ReferenceDataKeyPropertyType = exports.PropertyType = exports.LocalTimestampFormat = exports.IngressStartAtType = exports.EventSourceKind = exports.EnvironmentKind = exports.DataStringComparisonBehavior = exports.AccessPolicyRole = void 0;
exports.AccessPolicyRole = {
    Reader: "Reader",
    Contributor: "Contributor",
};
exports.DataStringComparisonBehavior = {
    Ordinal: "Ordinal",
    OrdinalIgnoreCase: "OrdinalIgnoreCase",
};
exports.EnvironmentKind = {
    Gen1: "Gen1",
    Gen2: "Gen2",
};
exports.EventSourceKind = {
    Microsoft_EventHub: "Microsoft.EventHub",
    Microsoft_IoTHub: "Microsoft.IoTHub",
};
exports.IngressStartAtType = {
    EarliestAvailable: "EarliestAvailable",
    EventSourceCreationTime: "EventSourceCreationTime",
    CustomEnqueuedTime: "CustomEnqueuedTime",
};
exports.LocalTimestampFormat = {
    Embedded: "Embedded",
};
exports.PropertyType = {
    String: "String",
};
exports.ReferenceDataKeyPropertyType = {
    String: "String",
    Double: "Double",
    Bool: "Bool",
    DateTime: "DateTime",
};
exports.SkuName = {
    S1: "S1",
    S2: "S2",
    P1: "P1",
    L1: "L1",
};
exports.StorageLimitExceededBehavior = {
    PurgeOldData: "PurgeOldData",
    PauseIngress: "PauseIngress",
};
