// @pulumi/azure-native/types/enums/billing/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningTenantState = exports.InvoiceSectionStateReasonCode = exports.InvoiceSectionState = exports.BillingManagementTenantState = exports.v20240401 = void 0;
// Export sub-modules:
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
exports.BillingManagementTenantState = {
    Other: "Other",
    NotAllowed: "NotAllowed",
    Active: "Active",
    Revoked: "Revoked",
};
exports.InvoiceSectionState = {
    Other: "Other",
    Active: "Active",
    Deleted: "Deleted",
    Disabled: "Disabled",
    UnderReview: "UnderReview",
    Warned: "Warned",
    Restricted: "Restricted",
};
exports.InvoiceSectionStateReasonCode = {
    Other: "Other",
    PastDue: "PastDue",
    UnusualActivity: "UnusualActivity",
    SpendingLimitReached: "SpendingLimitReached",
    SpendingLimitExpired: "SpendingLimitExpired",
};
exports.ProvisioningTenantState = {
    Other: "Other",
    NotRequested: "NotRequested",
    Active: "Active",
    Pending: "Pending",
    BillingRequestExpired: "BillingRequestExpired",
    BillingRequestDeclined: "BillingRequestDeclined",
    Revoked: "Revoked",
};
