// @pulumi/azure-native/containerservice/v20240502preview/snapshot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Snapshot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A node pool snapshot resource.
 */
class Snapshot extends pulumi.CustomResource {
    /**
     * Get an existing Snapshot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Snapshot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Snapshot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Snapshot.__pulumiType;
    }
    /**
     * Create a Snapshot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["creationData"] = args ? args.creationData : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["snapshotType"] = args ? args.snapshotType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["enableFIPS"] = undefined /*out*/;
            resourceInputs["kubernetesVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeImageVersion"] = undefined /*out*/;
            resourceInputs["osSku"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmSize"] = undefined /*out*/;
        }
        else {
            resourceInputs["creationData"] = undefined /*out*/;
            resourceInputs["enableFIPS"] = undefined /*out*/;
            resourceInputs["kubernetesVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeImageVersion"] = undefined /*out*/;
            resourceInputs["osSku"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["snapshotType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmSize"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerservice/v20210801:Snapshot" }, { type: "azure-native:containerservice/v20210901:Snapshot" }, { type: "azure-native:containerservice/v20211001:Snapshot" }, { type: "azure-native:containerservice/v20211101preview:Snapshot" }, { type: "azure-native:containerservice/v20220101:Snapshot" }, { type: "azure-native:containerservice/v20220102preview:Snapshot" }, { type: "azure-native:containerservice/v20220201:Snapshot" }, { type: "azure-native:containerservice/v20220202preview:Snapshot" }, { type: "azure-native:containerservice/v20220301:Snapshot" }, { type: "azure-native:containerservice/v20220302preview:Snapshot" }, { type: "azure-native:containerservice/v20220401:Snapshot" }, { type: "azure-native:containerservice/v20220402preview:Snapshot" }, { type: "azure-native:containerservice/v20220502preview:Snapshot" }, { type: "azure-native:containerservice/v20220601:Snapshot" }, { type: "azure-native:containerservice/v20220602preview:Snapshot" }, { type: "azure-native:containerservice/v20220701:Snapshot" }, { type: "azure-native:containerservice/v20220702preview:Snapshot" }, { type: "azure-native:containerservice/v20220802preview:Snapshot" }, { type: "azure-native:containerservice/v20220803preview:Snapshot" }, { type: "azure-native:containerservice/v20220901:Snapshot" }, { type: "azure-native:containerservice/v20220902preview:Snapshot" }, { type: "azure-native:containerservice/v20221002preview:Snapshot" }, { type: "azure-native:containerservice/v20221101:Snapshot" }, { type: "azure-native:containerservice/v20221102preview:Snapshot" }, { type: "azure-native:containerservice/v20230101:Snapshot" }, { type: "azure-native:containerservice/v20230102preview:Snapshot" }, { type: "azure-native:containerservice/v20230201:Snapshot" }, { type: "azure-native:containerservice/v20230202preview:Snapshot" }, { type: "azure-native:containerservice/v20230301:Snapshot" }, { type: "azure-native:containerservice/v20230302preview:Snapshot" }, { type: "azure-native:containerservice/v20230401:Snapshot" }, { type: "azure-native:containerservice/v20230402preview:Snapshot" }, { type: "azure-native:containerservice/v20230501:Snapshot" }, { type: "azure-native:containerservice/v20230502preview:Snapshot" }, { type: "azure-native:containerservice/v20230601:Snapshot" }, { type: "azure-native:containerservice/v20230602preview:Snapshot" }, { type: "azure-native:containerservice/v20230701:Snapshot" }, { type: "azure-native:containerservice/v20230702preview:Snapshot" }, { type: "azure-native:containerservice/v20230801:Snapshot" }, { type: "azure-native:containerservice/v20230802preview:Snapshot" }, { type: "azure-native:containerservice/v20230901:Snapshot" }, { type: "azure-native:containerservice/v20230902preview:Snapshot" }, { type: "azure-native:containerservice/v20231001:Snapshot" }, { type: "azure-native:containerservice/v20231002preview:Snapshot" }, { type: "azure-native:containerservice/v20231101:Snapshot" }, { type: "azure-native:containerservice/v20231102preview:Snapshot" }, { type: "azure-native:containerservice/v20240101:Snapshot" }, { type: "azure-native:containerservice/v20240102preview:Snapshot" }, { type: "azure-native:containerservice/v20240201:Snapshot" }, { type: "azure-native:containerservice/v20240202preview:Snapshot" }, { type: "azure-native:containerservice/v20240302preview:Snapshot" }, { type: "azure-native:containerservice/v20240402preview:Snapshot" }, { type: "azure-native:containerservice/v20240501:Snapshot" }, { type: "azure-native:containerservice/v20240602preview:Snapshot" }, { type: "azure-native:containerservice/v20240701:Snapshot" }, { type: "azure-native:containerservice/v20240702preview:Snapshot" }, { type: "azure-native:containerservice/v20240801:Snapshot" }, { type: "azure-native:containerservice/v20240901:Snapshot" }, { type: "azure-native:containerservice/v20240902preview:Snapshot" }, { type: "azure-native:containerservice/v20241001:Snapshot" }, { type: "azure-native:containerservice/v20241002preview:Snapshot" }, { type: "azure-native:containerservice/v20250101:Snapshot" }, { type: "azure-native:containerservice:Snapshot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Snapshot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Snapshot = Snapshot;
/** @internal */
Snapshot.__pulumiType = 'azure-native:containerservice/v20240502preview:Snapshot';
