// @pulumi/azure-native/machinelearningservices/v20220101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = exports.PrivateEndpointConnection = exports.listWorkspaceStorageAccountKeysOutput = exports.listWorkspaceStorageAccountKeys = exports.listWorkspaceNotebookKeysOutput = exports.listWorkspaceNotebookKeys = exports.listWorkspaceNotebookAccessTokenOutput = exports.listWorkspaceNotebookAccessToken = exports.listWorkspaceKeysOutput = exports.listWorkspaceKeys = exports.listComputeNodesOutput = exports.listComputeNodes = exports.listComputeKeysOutput = exports.listComputeKeys = exports.getWorkspaceOutput = exports.getWorkspace = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getComputeOutput = exports.getCompute = exports.Compute = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Compute = null;
utilities.lazyLoad(exports, ["Compute"], () => require("./compute"));
exports.getCompute = null;
exports.getComputeOutput = null;
utilities.lazyLoad(exports, ["getCompute", "getComputeOutput"], () => require("./getCompute"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.listComputeKeys = null;
exports.listComputeKeysOutput = null;
utilities.lazyLoad(exports, ["listComputeKeys", "listComputeKeysOutput"], () => require("./listComputeKeys"));
exports.listComputeNodes = null;
exports.listComputeNodesOutput = null;
utilities.lazyLoad(exports, ["listComputeNodes", "listComputeNodesOutput"], () => require("./listComputeNodes"));
exports.listWorkspaceKeys = null;
exports.listWorkspaceKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceKeys", "listWorkspaceKeysOutput"], () => require("./listWorkspaceKeys"));
exports.listWorkspaceNotebookAccessToken = null;
exports.listWorkspaceNotebookAccessTokenOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceNotebookAccessToken", "listWorkspaceNotebookAccessTokenOutput"], () => require("./listWorkspaceNotebookAccessToken"));
exports.listWorkspaceNotebookKeys = null;
exports.listWorkspaceNotebookKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceNotebookKeys", "listWorkspaceNotebookKeysOutput"], () => require("./listWorkspaceNotebookKeys"));
exports.listWorkspaceStorageAccountKeys = null;
exports.listWorkspaceStorageAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listWorkspaceStorageAccountKeys", "listWorkspaceStorageAccountKeysOutput"], () => require("./listWorkspaceStorageAccountKeys"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../../types/enums/machinelearningservices/v20220101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningservices/v20220101preview:Compute":
                return new exports.Compute(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220101preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:machinelearningservices/v20220101preview:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningservices/v20220101preview", _module);
