// @pulumi/azure-native/types/enums/datareplication/v20240901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VMNicSelection = exports.ReplicationVaultType = exports.PrivateEndpointConnectionStatus = exports.ManagedServiceIdentityType = void 0;
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateEndpointConnectionStatus = {
    /**
     * Approved Status.
     */
    Approved: "Approved",
    /**
     * Disconnected Status.
     */
    Disconnected: "Disconnected",
    /**
     * Pending Status.
     */
    Pending: "Pending",
    /**
     * Rejected Status.
     */
    Rejected: "Rejected",
};
exports.ReplicationVaultType = {
    /**
     * Disaster recovery vault.
     */
    DisasterRecovery: "DisasterRecovery",
    /**
     * Migrate vault.
     */
    Migrate: "Migrate",
};
exports.VMNicSelection = {
    /**
     * Not Selected.
     */
    NotSelected: "NotSelected",
    /**
     * Selected by user.
     */
    SelectedByUser: "SelectedByUser",
    /**
     * Default selection by ASR.
     */
    SelectedByDefault: "SelectedByDefault",
    /**
     * NIC configuration overridden by user. Differs from SelectedByUser in the sense that the legacy SelectedByUser is used both for explicit modification by user and implicit approval of user if the settings are used for TFO/FO. SelectedByUserOverride implies user overriding at least one of the configurations.
     */
    SelectedByUserOverride: "SelectedByUserOverride",
};
