// @pulumi/azure-native/types/enums/resources/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScriptType = exports.ResourceIdentityType = exports.OnErrorDeploymentType = exports.ManagedServiceIdentityType = exports.ExtendedLocationType = exports.ExpressionEvaluationOptionsScopeType = exports.DeploymentStacksDeleteDetachEnum = exports.DeploymentMode = exports.DenySettingsMode = exports.CleanupOptions = exports.v20241101 = exports.v20240701 = exports.v20240301 = exports.v20230801 = exports.v20230701 = exports.v20220901 = exports.v20220801preview = exports.v20201001 = exports.v20191001preview = exports.v20190601preview = void 0;
// Export sub-modules:
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20220801preview = require("./v20220801preview");
exports.v20220801preview = v20220801preview;
const v20220901 = require("./v20220901");
exports.v20220901 = v20220901;
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240701 = require("./v20240701");
exports.v20240701 = v20240701;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
exports.CleanupOptions = {
    Always: "Always",
    OnSuccess: "OnSuccess",
    OnExpiration: "OnExpiration",
};
exports.DenySettingsMode = {
    /**
     * Authorized users are able to read and modify the resources, but cannot delete.
     */
    DenyDelete: "denyDelete",
    /**
     * Authorized users can only read from a resource, but cannot modify or delete it.
     */
    DenyWriteAndDelete: "denyWriteAndDelete",
    /**
     * No denyAssignments have been applied.
     */
    None: "none",
};
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.DeploymentStacksDeleteDetachEnum = {
    Delete: "delete",
    Detach: "detach",
};
exports.ExpressionEvaluationOptionsScopeType = {
    NotSpecified: "NotSpecified",
    Outer: "Outer",
    Inner: "Inner",
};
exports.ExtendedLocationType = {
    EdgeZone: "EdgeZone",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.OnErrorDeploymentType = {
    LastSuccessful: "LastSuccessful",
    SpecificDeployment: "SpecificDeployment",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ScriptType = {
    AzurePowerShell: "AzurePowerShell",
    AzureCLI: "AzureCLI",
};
