// @pulumi/azure-native/types/enums/devcenter/v20250201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkType = exports.StopOnNoConnectEnableStatus = exports.StopOnDisconnectEnableStatus = exports.SkuTier = exports.SingleSignOnStatus = exports.ScheduledType = exports.ScheduledFrequency = exports.ScheduleEnableStatus = exports.PoolDevBoxDefinitionType = exports.PolicyAction = exports.MicrosoftHostedNetworkEnableStatus = exports.ManagedServiceIdentityType = exports.LocalAdminStatus = exports.LicenseType = exports.InstallAzureMonitorAgentEnableStatus = exports.IdentityType = exports.HibernateSupport = exports.EnvironmentTypeEnableStatus = exports.DomainJoinType = exports.DevCenterResourceType = exports.CatalogSyncType = exports.CatalogItemType = exports.CatalogItemSyncEnableStatus = void 0;
exports.CatalogItemSyncEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CatalogItemType = {
    EnvironmentDefinition: "EnvironmentDefinition",
    ImageDefinition: "ImageDefinition",
};
exports.CatalogSyncType = {
    Manual: "Manual",
    Scheduled: "Scheduled",
};
exports.DevCenterResourceType = {
    Images: "Images",
    AttachedNetworks: "AttachedNetworks",
    Skus: "Skus",
};
exports.DomainJoinType = {
    HybridAzureADJoin: "HybridAzureADJoin",
    AzureADJoin: "AzureADJoin",
    None: "None",
};
exports.EnvironmentTypeEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HibernateSupport = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.IdentityType = {
    SystemAssignedIdentity: "systemAssignedIdentity",
    UserAssignedIdentity: "userAssignedIdentity",
    DelegatedResourceIdentity: "delegatedResourceIdentity",
};
exports.InstallAzureMonitorAgentEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.LicenseType = {
    Windows_Client: "Windows_Client",
};
exports.LocalAdminStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.MicrosoftHostedNetworkEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PolicyAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.PoolDevBoxDefinitionType = {
    Reference: "Reference",
    Value: "Value",
};
exports.ScheduleEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ScheduledFrequency = {
    Daily: "Daily",
};
exports.ScheduledType = {
    StopDevBox: "StopDevBox",
};
exports.SingleSignOnStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.StopOnDisconnectEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StopOnNoConnectEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VirtualNetworkType = {
    Managed: "Managed",
    Unmanaged: "Unmanaged",
};
