// @pulumi/azure-native/types/enums/cloudngfw/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageType = exports.StateEnum = exports.ScopeType = exports.ProtocolType = exports.NetworkType = exports.MarketplaceSubscriptionStatus = exports.ManagedIdentityType = exports.EnabledDNSType = exports.EgressNat = exports.DefaultMode = exports.DecryptionRuleTypeEnum = exports.DNSProxy = exports.BooleanEnum = exports.BillingCycle = exports.ActionEnum = exports.v20250206preview = exports.v20240207preview = exports.v20240119preview = exports.v20231010preview = exports.v20230901preview = exports.v20230901 = exports.v20220829preview = exports.v20220829 = void 0;
// Export sub-modules:
const v20220829 = require("./v20220829");
exports.v20220829 = v20220829;
const v20220829preview = require("./v20220829preview");
exports.v20220829preview = v20220829preview;
const v20230901 = require("./v20230901");
exports.v20230901 = v20230901;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231010preview = require("./v20231010preview");
exports.v20231010preview = v20231010preview;
const v20240119preview = require("./v20240119preview");
exports.v20240119preview = v20240119preview;
const v20240207preview = require("./v20240207preview");
exports.v20240207preview = v20240207preview;
const v20250206preview = require("./v20250206preview");
exports.v20250206preview = v20250206preview;
exports.ActionEnum = {
    Allow: "Allow",
    DenySilent: "DenySilent",
    DenyResetServer: "DenyResetServer",
    DenyResetBoth: "DenyResetBoth",
};
exports.BillingCycle = {
    WEEKLY: "WEEKLY",
    MONTHLY: "MONTHLY",
};
exports.BooleanEnum = {
    TRUE: "TRUE",
    FALSE: "FALSE",
};
exports.DNSProxy = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.DecryptionRuleTypeEnum = {
    SSLOutboundInspection: "SSLOutboundInspection",
    SSLInboundInspection: "SSLInboundInspection",
    None: "None",
};
exports.DefaultMode = {
    IPS: "IPS",
    FIREWALL: "FIREWALL",
    NONE: "NONE",
};
exports.EgressNat = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.EnabledDNSType = {
    CUSTOM: "CUSTOM",
    AZURE: "AZURE",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAndUserAssigned: "SystemAssigned,UserAssigned",
};
exports.MarketplaceSubscriptionStatus = {
    PendingFulfillmentStart: "PendingFulfillmentStart",
    Subscribed: "Subscribed",
    Suspended: "Suspended",
    Unsubscribed: "Unsubscribed",
    NotStarted: "NotStarted",
    FulfillmentRequested: "FulfillmentRequested",
};
exports.NetworkType = {
    VNET: "VNET",
    VWAN: "VWAN",
};
exports.ProtocolType = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.ScopeType = {
    LOCAL: "LOCAL",
    GLOBAL: "GLOBAL",
};
exports.StateEnum = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.UsageType = {
    PAYG: "PAYG",
    COMMITTED: "COMMITTED",
};
