// @pulumi/azure-native/appplatform/app.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.App = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * App resource payload
 *
 * Uses Azure REST API version 2023-05-01-preview. In version 1.x of the Azure Native provider, it used API version 2020-07-01.
 *
 * Other available API versions: 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01, 2024-01-01-preview, 2024-05-01-preview.
 */
class App extends pulumi.CustomResource {
    /**
     * Get an existing App resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new App(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of App.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === App.__pulumiType;
    }
    /**
     * Create a App resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["appName"] = args ? args.appName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(inputs.appplatform.appResourcePropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:appplatform/v20200701:App" }, { type: "azure-native:appplatform/v20201101preview:App" }, { type: "azure-native:appplatform/v20210601preview:App" }, { type: "azure-native:appplatform/v20210901preview:App" }, { type: "azure-native:appplatform/v20220101preview:App" }, { type: "azure-native:appplatform/v20220301preview:App" }, { type: "azure-native:appplatform/v20220401:App" }, { type: "azure-native:appplatform/v20220501preview:App" }, { type: "azure-native:appplatform/v20220901preview:App" }, { type: "azure-native:appplatform/v20221101preview:App" }, { type: "azure-native:appplatform/v20221201:App" }, { type: "azure-native:appplatform/v20230101preview:App" }, { type: "azure-native:appplatform/v20230301preview:App" }, { type: "azure-native:appplatform/v20230501preview:App" }, { type: "azure-native:appplatform/v20230701preview:App" }, { type: "azure-native:appplatform/v20230901preview:App" }, { type: "azure-native:appplatform/v20231101preview:App" }, { type: "azure-native:appplatform/v20231201:App" }, { type: "azure-native:appplatform/v20240101preview:App" }, { type: "azure-native:appplatform/v20240501preview:App" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(App.__pulumiType, name, resourceInputs, opts);
    }
}
exports.App = App;
/** @internal */
App.__pulumiType = 'azure-native:appplatform:App';
