// @pulumi/azure-native/appplatform/buildServiceBuild.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuildServiceBuild = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Build resource payload
 *
 * Uses Azure REST API version 2023-05-01-preview.
 *
 * Other available API versions: 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01, 2024-01-01-preview, 2024-05-01-preview.
 */
class BuildServiceBuild extends pulumi.CustomResource {
    /**
     * Get an existing BuildServiceBuild resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BuildServiceBuild(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BuildServiceBuild.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BuildServiceBuild.__pulumiType;
    }
    /**
     * Create a BuildServiceBuild resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.buildServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'buildServiceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["buildName"] = args ? args.buildName : undefined;
            resourceInputs["buildServiceName"] = args ? args.buildServiceName : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(inputs.appplatform.buildPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:appplatform/v20230301preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20230501preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20230701preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20230901preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20231101preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20231201:BuildServiceBuild" }, { type: "azure-native:appplatform/v20240101preview:BuildServiceBuild" }, { type: "azure-native:appplatform/v20240501preview:BuildServiceBuild" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BuildServiceBuild.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BuildServiceBuild = BuildServiceBuild;
/** @internal */
BuildServiceBuild.__pulumiType = 'azure-native:appplatform:BuildServiceBuild';
