// @pulumi/azure-native/appplatform/devToolPortal.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DevToolPortal = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Dev Tool Portal resource
 *
 * Uses Azure REST API version 2023-05-01-preview.
 *
 * Other available API versions: 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01, 2024-01-01-preview, 2024-05-01-preview.
 */
class DevToolPortal extends pulumi.CustomResource {
    /**
     * Get an existing DevToolPortal resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DevToolPortal(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DevToolPortal.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DevToolPortal.__pulumiType;
    }
    /**
     * Create a DevToolPortal resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["devToolPortalName"] = args ? args.devToolPortalName : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(inputs.appplatform.devToolPortalPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:appplatform/v20221101preview:DevToolPortal" }, { type: "azure-native:appplatform/v20230101preview:DevToolPortal" }, { type: "azure-native:appplatform/v20230301preview:DevToolPortal" }, { type: "azure-native:appplatform/v20230501preview:DevToolPortal" }, { type: "azure-native:appplatform/v20230701preview:DevToolPortal" }, { type: "azure-native:appplatform/v20230901preview:DevToolPortal" }, { type: "azure-native:appplatform/v20231101preview:DevToolPortal" }, { type: "azure-native:appplatform/v20231201:DevToolPortal" }, { type: "azure-native:appplatform/v20240101preview:DevToolPortal" }, { type: "azure-native:appplatform/v20240501preview:DevToolPortal" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DevToolPortal.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DevToolPortal = DevToolPortal;
/** @internal */
DevToolPortal.__pulumiType = 'azure-native:appplatform:DevToolPortal';
