// @pulumi/azure-native/appplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBuildpackBindingOutput = exports.getBuildpackBinding = exports.getBuildServiceResourceUploadUrlOutput = exports.getBuildServiceResourceUploadUrl = exports.getBuildServiceBuilderOutput = exports.getBuildServiceBuilder = exports.getBuildServiceBuildResultLogOutput = exports.getBuildServiceBuildResultLog = exports.getBuildServiceBuildOutput = exports.getBuildServiceBuild = exports.getBuildServiceAgentPoolOutput = exports.getBuildServiceAgentPool = exports.getBindingOutput = exports.getBinding = exports.getApplicationLiveViewOutput = exports.getApplicationLiveView = exports.getApplicationAcceleratorOutput = exports.getApplicationAccelerator = exports.getAppResourceUploadUrlOutput = exports.getAppResourceUploadUrl = exports.getAppOutput = exports.getApp = exports.getApmOutput = exports.getApm = exports.getApiPortalCustomDomainOutput = exports.getApiPortalCustomDomain = exports.getApiPortalOutput = exports.getApiPortal = exports.GatewayRouteConfig = exports.GatewayCustomDomain = exports.Gateway = exports.DevToolPortal = exports.Deployment = exports.CustomizedAccelerator = exports.CustomDomain = exports.ContainerRegistry = exports.ConfigurationService = exports.ConfigServer = exports.Certificate = exports.BuildpackBinding = exports.BuildServiceBuilder = exports.BuildServiceBuild = exports.BuildServiceAgentPool = exports.Binding = exports.ApplicationLiveView = exports.ApplicationAccelerator = exports.App = exports.Apm = exports.ApiPortalCustomDomain = exports.ApiPortal = void 0;
exports.MonitoringSetting = exports.listServiceTestKeysOutput = exports.listServiceTestKeys = exports.listServiceGloballyEnabledApmsOutput = exports.listServiceGloballyEnabledApms = exports.listJobExecutionEnvSecretsOutput = exports.listJobExecutionEnvSecrets = exports.listJobEnvSecretsOutput = exports.listJobEnvSecrets = exports.listBuildServiceBuilderDeploymentsOutput = exports.listBuildServiceBuilderDeployments = exports.listApmSecretKeysOutput = exports.listApmSecretKeys = exports.Job = exports.getStorageOutput = exports.getStorage = exports.getServiceRegistryOutput = exports.getServiceRegistry = exports.getServiceOutput = exports.getService = exports.getMonitoringSettingOutput = exports.getMonitoringSetting = exports.getJobOutput = exports.getJob = exports.getGatewayRouteConfigOutput = exports.getGatewayRouteConfig = exports.getGatewayCustomDomainOutput = exports.getGatewayCustomDomain = exports.getGatewayOutput = exports.getGateway = exports.getDevToolPortalOutput = exports.getDevToolPortal = exports.getDeploymentRemoteDebuggingConfigOutput = exports.getDeploymentRemoteDebuggingConfig = exports.getDeploymentLogFileUrlOutput = exports.getDeploymentLogFileUrl = exports.getDeploymentOutput = exports.getDeployment = exports.getCustomizedAcceleratorOutput = exports.getCustomizedAccelerator = exports.getCustomDomainOutput = exports.getCustomDomain = exports.getContainerRegistryOutput = exports.getContainerRegistry = exports.getConfigurationServiceOutput = exports.getConfigurationService = exports.getConfigServerOutput = exports.getConfigServer = exports.getCertificateOutput = exports.getCertificate = void 0;
exports.v20240501preview = exports.v20240101preview = exports.v20231201 = exports.v20231101preview = exports.v20230901preview = exports.v20230701preview = exports.v20230501preview = exports.v20210601preview = exports.Storage = exports.ServiceRegistry = exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ApiPortal = null;
utilities.lazyLoad(exports, ["ApiPortal"], () => require("./apiPortal"));
exports.ApiPortalCustomDomain = null;
utilities.lazyLoad(exports, ["ApiPortalCustomDomain"], () => require("./apiPortalCustomDomain"));
exports.Apm = null;
utilities.lazyLoad(exports, ["Apm"], () => require("./apm"));
exports.App = null;
utilities.lazyLoad(exports, ["App"], () => require("./app"));
exports.ApplicationAccelerator = null;
utilities.lazyLoad(exports, ["ApplicationAccelerator"], () => require("./applicationAccelerator"));
exports.ApplicationLiveView = null;
utilities.lazyLoad(exports, ["ApplicationLiveView"], () => require("./applicationLiveView"));
exports.Binding = null;
utilities.lazyLoad(exports, ["Binding"], () => require("./binding"));
exports.BuildServiceAgentPool = null;
utilities.lazyLoad(exports, ["BuildServiceAgentPool"], () => require("./buildServiceAgentPool"));
exports.BuildServiceBuild = null;
utilities.lazyLoad(exports, ["BuildServiceBuild"], () => require("./buildServiceBuild"));
exports.BuildServiceBuilder = null;
utilities.lazyLoad(exports, ["BuildServiceBuilder"], () => require("./buildServiceBuilder"));
exports.BuildpackBinding = null;
utilities.lazyLoad(exports, ["BuildpackBinding"], () => require("./buildpackBinding"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.ConfigServer = null;
utilities.lazyLoad(exports, ["ConfigServer"], () => require("./configServer"));
exports.ConfigurationService = null;
utilities.lazyLoad(exports, ["ConfigurationService"], () => require("./configurationService"));
exports.ContainerRegistry = null;
utilities.lazyLoad(exports, ["ContainerRegistry"], () => require("./containerRegistry"));
exports.CustomDomain = null;
utilities.lazyLoad(exports, ["CustomDomain"], () => require("./customDomain"));
exports.CustomizedAccelerator = null;
utilities.lazyLoad(exports, ["CustomizedAccelerator"], () => require("./customizedAccelerator"));
exports.Deployment = null;
utilities.lazyLoad(exports, ["Deployment"], () => require("./deployment"));
exports.DevToolPortal = null;
utilities.lazyLoad(exports, ["DevToolPortal"], () => require("./devToolPortal"));
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.GatewayCustomDomain = null;
utilities.lazyLoad(exports, ["GatewayCustomDomain"], () => require("./gatewayCustomDomain"));
exports.GatewayRouteConfig = null;
utilities.lazyLoad(exports, ["GatewayRouteConfig"], () => require("./gatewayRouteConfig"));
exports.getApiPortal = null;
exports.getApiPortalOutput = null;
utilities.lazyLoad(exports, ["getApiPortal", "getApiPortalOutput"], () => require("./getApiPortal"));
exports.getApiPortalCustomDomain = null;
exports.getApiPortalCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getApiPortalCustomDomain", "getApiPortalCustomDomainOutput"], () => require("./getApiPortalCustomDomain"));
exports.getApm = null;
exports.getApmOutput = null;
utilities.lazyLoad(exports, ["getApm", "getApmOutput"], () => require("./getApm"));
exports.getApp = null;
exports.getAppOutput = null;
utilities.lazyLoad(exports, ["getApp", "getAppOutput"], () => require("./getApp"));
exports.getAppResourceUploadUrl = null;
exports.getAppResourceUploadUrlOutput = null;
utilities.lazyLoad(exports, ["getAppResourceUploadUrl", "getAppResourceUploadUrlOutput"], () => require("./getAppResourceUploadUrl"));
exports.getApplicationAccelerator = null;
exports.getApplicationAcceleratorOutput = null;
utilities.lazyLoad(exports, ["getApplicationAccelerator", "getApplicationAcceleratorOutput"], () => require("./getApplicationAccelerator"));
exports.getApplicationLiveView = null;
exports.getApplicationLiveViewOutput = null;
utilities.lazyLoad(exports, ["getApplicationLiveView", "getApplicationLiveViewOutput"], () => require("./getApplicationLiveView"));
exports.getBinding = null;
exports.getBindingOutput = null;
utilities.lazyLoad(exports, ["getBinding", "getBindingOutput"], () => require("./getBinding"));
exports.getBuildServiceAgentPool = null;
exports.getBuildServiceAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getBuildServiceAgentPool", "getBuildServiceAgentPoolOutput"], () => require("./getBuildServiceAgentPool"));
exports.getBuildServiceBuild = null;
exports.getBuildServiceBuildOutput = null;
utilities.lazyLoad(exports, ["getBuildServiceBuild", "getBuildServiceBuildOutput"], () => require("./getBuildServiceBuild"));
exports.getBuildServiceBuildResultLog = null;
exports.getBuildServiceBuildResultLogOutput = null;
utilities.lazyLoad(exports, ["getBuildServiceBuildResultLog", "getBuildServiceBuildResultLogOutput"], () => require("./getBuildServiceBuildResultLog"));
exports.getBuildServiceBuilder = null;
exports.getBuildServiceBuilderOutput = null;
utilities.lazyLoad(exports, ["getBuildServiceBuilder", "getBuildServiceBuilderOutput"], () => require("./getBuildServiceBuilder"));
exports.getBuildServiceResourceUploadUrl = null;
exports.getBuildServiceResourceUploadUrlOutput = null;
utilities.lazyLoad(exports, ["getBuildServiceResourceUploadUrl", "getBuildServiceResourceUploadUrlOutput"], () => require("./getBuildServiceResourceUploadUrl"));
exports.getBuildpackBinding = null;
exports.getBuildpackBindingOutput = null;
utilities.lazyLoad(exports, ["getBuildpackBinding", "getBuildpackBindingOutput"], () => require("./getBuildpackBinding"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getConfigServer = null;
exports.getConfigServerOutput = null;
utilities.lazyLoad(exports, ["getConfigServer", "getConfigServerOutput"], () => require("./getConfigServer"));
exports.getConfigurationService = null;
exports.getConfigurationServiceOutput = null;
utilities.lazyLoad(exports, ["getConfigurationService", "getConfigurationServiceOutput"], () => require("./getConfigurationService"));
exports.getContainerRegistry = null;
exports.getContainerRegistryOutput = null;
utilities.lazyLoad(exports, ["getContainerRegistry", "getContainerRegistryOutput"], () => require("./getContainerRegistry"));
exports.getCustomDomain = null;
exports.getCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getCustomDomain", "getCustomDomainOutput"], () => require("./getCustomDomain"));
exports.getCustomizedAccelerator = null;
exports.getCustomizedAcceleratorOutput = null;
utilities.lazyLoad(exports, ["getCustomizedAccelerator", "getCustomizedAcceleratorOutput"], () => require("./getCustomizedAccelerator"));
exports.getDeployment = null;
exports.getDeploymentOutput = null;
utilities.lazyLoad(exports, ["getDeployment", "getDeploymentOutput"], () => require("./getDeployment"));
exports.getDeploymentLogFileUrl = null;
exports.getDeploymentLogFileUrlOutput = null;
utilities.lazyLoad(exports, ["getDeploymentLogFileUrl", "getDeploymentLogFileUrlOutput"], () => require("./getDeploymentLogFileUrl"));
exports.getDeploymentRemoteDebuggingConfig = null;
exports.getDeploymentRemoteDebuggingConfigOutput = null;
utilities.lazyLoad(exports, ["getDeploymentRemoteDebuggingConfig", "getDeploymentRemoteDebuggingConfigOutput"], () => require("./getDeploymentRemoteDebuggingConfig"));
exports.getDevToolPortal = null;
exports.getDevToolPortalOutput = null;
utilities.lazyLoad(exports, ["getDevToolPortal", "getDevToolPortalOutput"], () => require("./getDevToolPortal"));
exports.getGateway = null;
exports.getGatewayOutput = null;
utilities.lazyLoad(exports, ["getGateway", "getGatewayOutput"], () => require("./getGateway"));
exports.getGatewayCustomDomain = null;
exports.getGatewayCustomDomainOutput = null;
utilities.lazyLoad(exports, ["getGatewayCustomDomain", "getGatewayCustomDomainOutput"], () => require("./getGatewayCustomDomain"));
exports.getGatewayRouteConfig = null;
exports.getGatewayRouteConfigOutput = null;
utilities.lazyLoad(exports, ["getGatewayRouteConfig", "getGatewayRouteConfigOutput"], () => require("./getGatewayRouteConfig"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getMonitoringSetting = null;
exports.getMonitoringSettingOutput = null;
utilities.lazyLoad(exports, ["getMonitoringSetting", "getMonitoringSettingOutput"], () => require("./getMonitoringSetting"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getServiceRegistry = null;
exports.getServiceRegistryOutput = null;
utilities.lazyLoad(exports, ["getServiceRegistry", "getServiceRegistryOutput"], () => require("./getServiceRegistry"));
exports.getStorage = null;
exports.getStorageOutput = null;
utilities.lazyLoad(exports, ["getStorage", "getStorageOutput"], () => require("./getStorage"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.listApmSecretKeys = null;
exports.listApmSecretKeysOutput = null;
utilities.lazyLoad(exports, ["listApmSecretKeys", "listApmSecretKeysOutput"], () => require("./listApmSecretKeys"));
exports.listBuildServiceBuilderDeployments = null;
exports.listBuildServiceBuilderDeploymentsOutput = null;
utilities.lazyLoad(exports, ["listBuildServiceBuilderDeployments", "listBuildServiceBuilderDeploymentsOutput"], () => require("./listBuildServiceBuilderDeployments"));
exports.listJobEnvSecrets = null;
exports.listJobEnvSecretsOutput = null;
utilities.lazyLoad(exports, ["listJobEnvSecrets", "listJobEnvSecretsOutput"], () => require("./listJobEnvSecrets"));
exports.listJobExecutionEnvSecrets = null;
exports.listJobExecutionEnvSecretsOutput = null;
utilities.lazyLoad(exports, ["listJobExecutionEnvSecrets", "listJobExecutionEnvSecretsOutput"], () => require("./listJobExecutionEnvSecrets"));
exports.listServiceGloballyEnabledApms = null;
exports.listServiceGloballyEnabledApmsOutput = null;
utilities.lazyLoad(exports, ["listServiceGloballyEnabledApms", "listServiceGloballyEnabledApmsOutput"], () => require("./listServiceGloballyEnabledApms"));
exports.listServiceTestKeys = null;
exports.listServiceTestKeysOutput = null;
utilities.lazyLoad(exports, ["listServiceTestKeys", "listServiceTestKeysOutput"], () => require("./listServiceTestKeys"));
exports.MonitoringSetting = null;
utilities.lazyLoad(exports, ["MonitoringSetting"], () => require("./monitoringSetting"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.ServiceRegistry = null;
utilities.lazyLoad(exports, ["ServiceRegistry"], () => require("./serviceRegistry"));
exports.Storage = null;
utilities.lazyLoad(exports, ["Storage"], () => require("./storage"));
// Export enums:
__exportStar(require("../types/enums/appplatform"), exports);
// Export sub-modules:
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const v20240101preview = require("./v20240101preview");
exports.v20240101preview = v20240101preview;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:appplatform:ApiPortal":
                return new exports.ApiPortal(name, undefined, { urn });
            case "azure-native:appplatform:ApiPortalCustomDomain":
                return new exports.ApiPortalCustomDomain(name, undefined, { urn });
            case "azure-native:appplatform:Apm":
                return new exports.Apm(name, undefined, { urn });
            case "azure-native:appplatform:App":
                return new exports.App(name, undefined, { urn });
            case "azure-native:appplatform:ApplicationAccelerator":
                return new exports.ApplicationAccelerator(name, undefined, { urn });
            case "azure-native:appplatform:ApplicationLiveView":
                return new exports.ApplicationLiveView(name, undefined, { urn });
            case "azure-native:appplatform:Binding":
                return new exports.Binding(name, undefined, { urn });
            case "azure-native:appplatform:BuildServiceAgentPool":
                return new exports.BuildServiceAgentPool(name, undefined, { urn });
            case "azure-native:appplatform:BuildServiceBuild":
                return new exports.BuildServiceBuild(name, undefined, { urn });
            case "azure-native:appplatform:BuildServiceBuilder":
                return new exports.BuildServiceBuilder(name, undefined, { urn });
            case "azure-native:appplatform:BuildpackBinding":
                return new exports.BuildpackBinding(name, undefined, { urn });
            case "azure-native:appplatform:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:appplatform:ConfigServer":
                return new exports.ConfigServer(name, undefined, { urn });
            case "azure-native:appplatform:ConfigurationService":
                return new exports.ConfigurationService(name, undefined, { urn });
            case "azure-native:appplatform:ContainerRegistry":
                return new exports.ContainerRegistry(name, undefined, { urn });
            case "azure-native:appplatform:CustomDomain":
                return new exports.CustomDomain(name, undefined, { urn });
            case "azure-native:appplatform:CustomizedAccelerator":
                return new exports.CustomizedAccelerator(name, undefined, { urn });
            case "azure-native:appplatform:Deployment":
                return new exports.Deployment(name, undefined, { urn });
            case "azure-native:appplatform:DevToolPortal":
                return new exports.DevToolPortal(name, undefined, { urn });
            case "azure-native:appplatform:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "azure-native:appplatform:GatewayCustomDomain":
                return new exports.GatewayCustomDomain(name, undefined, { urn });
            case "azure-native:appplatform:GatewayRouteConfig":
                return new exports.GatewayRouteConfig(name, undefined, { urn });
            case "azure-native:appplatform:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:appplatform:MonitoringSetting":
                return new exports.MonitoringSetting(name, undefined, { urn });
            case "azure-native:appplatform:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:appplatform:ServiceRegistry":
                return new exports.ServiceRegistry(name, undefined, { urn });
            case "azure-native:appplatform:Storage":
                return new exports.Storage(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "appplatform", _module);
