// @pulumi/azure-native/confluent/v20240213/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = exports.listOrganizationRegionsOutput = exports.listOrganizationRegions = exports.listAccessUsersOutput = exports.listAccessUsers = exports.listAccessServiceAccountsOutput = exports.listAccessServiceAccounts = exports.listAccessRoleBindingsOutput = exports.listAccessRoleBindings = exports.listAccessRoleBindingNameOutput = exports.listAccessRoleBindingName = exports.listAccessInvitationsOutput = exports.listAccessInvitations = exports.listAccessEnvironmentsOutput = exports.listAccessEnvironments = exports.listAccessClustersOutput = exports.listAccessClusters = exports.getOrganizationOutput = exports.getOrganization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getOrganization = null;
exports.getOrganizationOutput = null;
utilities.lazyLoad(exports, ["getOrganization", "getOrganizationOutput"], () => require("./getOrganization"));
exports.listAccessClusters = null;
exports.listAccessClustersOutput = null;
utilities.lazyLoad(exports, ["listAccessClusters", "listAccessClustersOutput"], () => require("./listAccessClusters"));
exports.listAccessEnvironments = null;
exports.listAccessEnvironmentsOutput = null;
utilities.lazyLoad(exports, ["listAccessEnvironments", "listAccessEnvironmentsOutput"], () => require("./listAccessEnvironments"));
exports.listAccessInvitations = null;
exports.listAccessInvitationsOutput = null;
utilities.lazyLoad(exports, ["listAccessInvitations", "listAccessInvitationsOutput"], () => require("./listAccessInvitations"));
exports.listAccessRoleBindingName = null;
exports.listAccessRoleBindingNameOutput = null;
utilities.lazyLoad(exports, ["listAccessRoleBindingName", "listAccessRoleBindingNameOutput"], () => require("./listAccessRoleBindingName"));
exports.listAccessRoleBindings = null;
exports.listAccessRoleBindingsOutput = null;
utilities.lazyLoad(exports, ["listAccessRoleBindings", "listAccessRoleBindingsOutput"], () => require("./listAccessRoleBindings"));
exports.listAccessServiceAccounts = null;
exports.listAccessServiceAccountsOutput = null;
utilities.lazyLoad(exports, ["listAccessServiceAccounts", "listAccessServiceAccountsOutput"], () => require("./listAccessServiceAccounts"));
exports.listAccessUsers = null;
exports.listAccessUsersOutput = null;
utilities.lazyLoad(exports, ["listAccessUsers", "listAccessUsersOutput"], () => require("./listAccessUsers"));
exports.listOrganizationRegions = null;
exports.listOrganizationRegionsOutput = null;
utilities.lazyLoad(exports, ["listOrganizationRegions", "listOrganizationRegionsOutput"], () => require("./listOrganizationRegions"));
exports.Organization = null;
utilities.lazyLoad(exports, ["Organization"], () => require("./organization"));
// Export enums:
__exportStar(require("../../types/enums/confluent/v20240213"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:confluent/v20240213:Organization":
                return new exports.Organization(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "confluent/v20240213", _module);
