// @pulumi/azure-native/iotfirmwaredefense/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250401preview = exports.v20240110 = exports.v20230208preview = exports.Workspace = exports.getWorkspaceOutput = exports.getWorkspace = exports.getFirmwareOutput = exports.getFirmware = exports.Firmware = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Firmware = null;
utilities.lazyLoad(exports, ["Firmware"], () => require("./firmware"));
exports.getFirmware = null;
exports.getFirmwareOutput = null;
utilities.lazyLoad(exports, ["getFirmware", "getFirmwareOutput"], () => require("./getFirmware"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export enums:
__exportStar(require("../types/enums/iotfirmwaredefense"), exports);
// Export sub-modules:
const v20230208preview = require("./v20230208preview");
exports.v20230208preview = v20230208preview;
const v20240110 = require("./v20240110");
exports.v20240110 = v20240110;
const v20250401preview = require("./v20250401preview");
exports.v20250401preview = v20250401preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:iotfirmwaredefense:Firmware":
                return new exports.Firmware(name, undefined, { urn });
            case "azure-native:iotfirmwaredefense:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "iotfirmwaredefense", _module);
