// @pulumi/azure-native/cdn/v20240601preview/afdcustomDomain.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AFDCustomDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Friendly domain name mapping to the endpoint hostname that the customer provides for branding purposes, e.g. www.contoso.com.
 */
class AFDCustomDomain extends pulumi.CustomResource {
    /**
     * Get an existing AFDCustomDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AFDCustomDomain(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AFDCustomDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AFDCustomDomain.__pulumiType;
    }
    /**
     * Create a AFDCustomDomain resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hostName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostName'");
            }
            if ((!args || args.profileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["azureDnsZone"] = args ? args.azureDnsZone : undefined;
            resourceInputs["customDomainName"] = args ? args.customDomainName : undefined;
            resourceInputs["extendedProperties"] = args ? args.extendedProperties : undefined;
            resourceInputs["hostName"] = args ? args.hostName : undefined;
            resourceInputs["mtlsSettings"] = args ? args.mtlsSettings : undefined;
            resourceInputs["preValidatedCustomDomainResourceId"] = args ? args.preValidatedCustomDomainResourceId : undefined;
            resourceInputs["profileName"] = args ? args.profileName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tlsSettings"] = args ? args.tlsSettings : undefined;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["domainValidationState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["referencedRoutePaths"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["validationProperties"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureDnsZone"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["domainValidationState"] = undefined /*out*/;
            resourceInputs["extendedProperties"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["mtlsSettings"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["preValidatedCustomDomainResourceId"] = undefined /*out*/;
            resourceInputs["profileName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["referencedRoutePaths"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tlsSettings"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["validationProperties"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cdn/v20200901:AFDCustomDomain" }, { type: "azure-native:cdn/v20210601:AFDCustomDomain" }, { type: "azure-native:cdn/v20220501preview:AFDCustomDomain" }, { type: "azure-native:cdn/v20221101preview:AFDCustomDomain" }, { type: "azure-native:cdn/v20230501:AFDCustomDomain" }, { type: "azure-native:cdn/v20230701preview:AFDCustomDomain" }, { type: "azure-native:cdn/v20240201:AFDCustomDomain" }, { type: "azure-native:cdn/v20240501preview:AFDCustomDomain" }, { type: "azure-native:cdn/v20240901:AFDCustomDomain" }, { type: "azure-native:cdn:AFDCustomDomain" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AFDCustomDomain.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AFDCustomDomain = AFDCustomDomain;
/** @internal */
AFDCustomDomain.__pulumiType = 'azure-native:cdn/v20240601preview:AFDCustomDomain';
