// @pulumi/azure-native/types/enums/sovereign/v20250227preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceType = exports.ResourceCreationOptions = exports.ManagedServiceIdentityType = exports.ManagedIdentityResourceType = exports.FirewallCreationOptions = void 0;
exports.FirewallCreationOptions = {
    /**
     * Select 'None' value to skip Firewall deployment.
     */
    None: "None",
    /**
     * Select 'Standard' value to deploy the Firewall Standard SKU.
     */
    Standard: "Standard",
    /**
     * Select 'Premium' value to deploy the Firewall Premium SKU.
     */
    Premium: "Premium",
};
exports.ManagedIdentityResourceType = {
    /**
     * System assigned managed identity.
     */
    SystemAssigned: "SystemAssigned",
    /**
     * User assigned managed identity.
     */
    UserAssigned: "UserAssigned",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ResourceCreationOptions = {
    /**
     * Value to create a new resource.
     */
    Yes: "Yes",
    /**
     * Value to not create a new resource.
     */
    No: "No",
    /**
     * Value to use an existing resource.
     */
    UseExisting: "UseExisting",
};
exports.ResourceType = {
    /**
     * Value when customer wants to provide a custom naming convention for Azure Firewall.
     */
    AzureFirewalls: "azureFirewalls",
    /**
     * Value when customer wants to provide a custom naming convention for Log Analytics workspace.
     */
    Workspaces: "workspaces",
    /**
     * Value when customer wants to provide a custom naming convention for automation account.
     */
    AutomationAccounts: "automationAccounts",
    /**
     * Value when customer wants to provide a custom naming convention for dashboard.
     */
    Dashboards: "dashboards",
    /**
     * Value when customer wants to provide a custom naming convention for managed identity.
     */
    UserAssignedIdentities: "userAssignedIdentities",
    /**
     * Value when customer wants to provide a custom naming convention for Bastion.
     */
    BastionHosts: "bastionHosts",
    /**
     * Value when customer wants to provide a custom naming convention for DDoS protection plan.
     */
    DdosProtectionPlans: "ddosProtectionPlans",
    /**
     * Value when customer wants to provide a custom naming convention for Bastion NSG.
     */
    NetworkSecurityGroups: "networkSecurityGroups",
    /**
     * Value when customer wants to provide a custom naming convention for virtual network.
     */
    VirtualNetworks: "virtualNetworks",
    /**
     * Value when customer wants to provide a custom naming convention for route table.
     */
    RouteTables: "routeTables",
};
