// @pulumi/azure-native/addons/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20180301 = exports.SupportPlanType = exports.getSupportPlanTypeOutput = exports.getSupportPlanType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSupportPlanType = null;
exports.getSupportPlanTypeOutput = null;
utilities.lazyLoad(exports, ["getSupportPlanType", "getSupportPlanTypeOutput"], () => require("./getSupportPlanType"));
exports.SupportPlanType = null;
utilities.lazyLoad(exports, ["SupportPlanType"], () => require("./supportPlanType"));
// Export sub-modules:
const v20180301 = require("./v20180301");
exports.v20180301 = v20180301;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:addons:SupportPlanType":
                return new exports.SupportPlanType(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "addons", _module);
