// @pulumi/azure-native/chaos/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250101 = exports.v20241101preview = exports.v20240322preview = exports.v20240101 = exports.v20231101 = exports.v20231027preview = exports.v20230901preview = exports.v20230415preview = exports.Target = exports.PrivateAccess = exports.getTargetOutput = exports.getTarget = exports.getPrivateAccessOutput = exports.getPrivateAccess = exports.getExperimentExecutionDetailsOutput = exports.getExperimentExecutionDetails = exports.getExperimentOutput = exports.getExperiment = exports.getCapabilityOutput = exports.getCapability = exports.Experiment = exports.Capability = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Capability = null;
utilities.lazyLoad(exports, ["Capability"], () => require("./capability"));
exports.Experiment = null;
utilities.lazyLoad(exports, ["Experiment"], () => require("./experiment"));
exports.getCapability = null;
exports.getCapabilityOutput = null;
utilities.lazyLoad(exports, ["getCapability", "getCapabilityOutput"], () => require("./getCapability"));
exports.getExperiment = null;
exports.getExperimentOutput = null;
utilities.lazyLoad(exports, ["getExperiment", "getExperimentOutput"], () => require("./getExperiment"));
exports.getExperimentExecutionDetails = null;
exports.getExperimentExecutionDetailsOutput = null;
utilities.lazyLoad(exports, ["getExperimentExecutionDetails", "getExperimentExecutionDetailsOutput"], () => require("./getExperimentExecutionDetails"));
exports.getPrivateAccess = null;
exports.getPrivateAccessOutput = null;
utilities.lazyLoad(exports, ["getPrivateAccess", "getPrivateAccessOutput"], () => require("./getPrivateAccess"));
exports.getTarget = null;
exports.getTargetOutput = null;
utilities.lazyLoad(exports, ["getTarget", "getTargetOutput"], () => require("./getTarget"));
exports.PrivateAccess = null;
utilities.lazyLoad(exports, ["PrivateAccess"], () => require("./privateAccess"));
exports.Target = null;
utilities.lazyLoad(exports, ["Target"], () => require("./target"));
// Export enums:
__exportStar(require("../types/enums/chaos"), exports);
// Export sub-modules:
const v20230415preview = require("./v20230415preview");
exports.v20230415preview = v20230415preview;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231027preview = require("./v20231027preview");
exports.v20231027preview = v20231027preview;
const v20231101 = require("./v20231101");
exports.v20231101 = v20231101;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240322preview = require("./v20240322preview");
exports.v20240322preview = v20240322preview;
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const v20250101 = require("./v20250101");
exports.v20250101 = v20250101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:chaos:Capability":
                return new exports.Capability(name, undefined, { urn });
            case "azure-native:chaos:Experiment":
                return new exports.Experiment(name, undefined, { urn });
            case "azure-native:chaos:PrivateAccess":
                return new exports.PrivateAccess(name, undefined, { urn });
            case "azure-native:chaos:Target":
                return new exports.Target(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "chaos", _module);
