// @pulumi/azure-native/deviceregistry/asset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Asset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Asset definition.
 *
 * Uses Azure REST API version 2023-11-01-preview.
 *
 * Other available API versions: 2024-09-01-preview, 2024-11-01.
 */
class Asset extends pulumi.CustomResource {
    /**
     * Get an existing Asset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Asset(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Asset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Asset.__pulumiType;
    }
    /**
     * Create a Asset resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.assetEndpointProfileUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assetEndpointProfileUri'");
            }
            if ((!args || args.extendedLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assetEndpointProfileUri"] = args ? args.assetEndpointProfileUri : undefined;
            resourceInputs["assetName"] = args ? args.assetName : undefined;
            resourceInputs["assetType"] = args ? args.assetType : undefined;
            resourceInputs["attributes"] = args ? args.attributes : undefined;
            resourceInputs["dataPoints"] = args ? args.dataPoints : undefined;
            resourceInputs["defaultDataPointsConfiguration"] = args ? args.defaultDataPointsConfiguration : undefined;
            resourceInputs["defaultEventsConfiguration"] = args ? args.defaultEventsConfiguration : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["documentationUri"] = args ? args.documentationUri : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["events"] = args ? args.events : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["externalAssetId"] = args ? args.externalAssetId : undefined;
            resourceInputs["hardwareRevision"] = args ? args.hardwareRevision : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["manufacturer"] = args ? args.manufacturer : undefined;
            resourceInputs["manufacturerUri"] = args ? args.manufacturerUri : undefined;
            resourceInputs["model"] = args ? args.model : undefined;
            resourceInputs["productCode"] = args ? args.productCode : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serialNumber"] = args ? args.serialNumber : undefined;
            resourceInputs["softwareRevision"] = args ? args.softwareRevision : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uuid"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        else {
            resourceInputs["assetEndpointProfileUri"] = undefined /*out*/;
            resourceInputs["assetType"] = undefined /*out*/;
            resourceInputs["attributes"] = undefined /*out*/;
            resourceInputs["dataPoints"] = undefined /*out*/;
            resourceInputs["defaultDataPointsConfiguration"] = undefined /*out*/;
            resourceInputs["defaultEventsConfiguration"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["documentationUri"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["events"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["externalAssetId"] = undefined /*out*/;
            resourceInputs["hardwareRevision"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["manufacturer"] = undefined /*out*/;
            resourceInputs["manufacturerUri"] = undefined /*out*/;
            resourceInputs["model"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["productCode"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serialNumber"] = undefined /*out*/;
            resourceInputs["softwareRevision"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uuid"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deviceregistry/v20231101preview:Asset" }, { type: "azure-native:deviceregistry/v20240901preview:Asset" }, { type: "azure-native:deviceregistry/v20241101:Asset" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Asset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Asset = Asset;
/** @internal */
Asset.__pulumiType = 'azure-native:deviceregistry:Asset';
