// @pulumi/azure-native/deviceregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20241101 = exports.v20240901preview = exports.v20231101preview = exports.SchemaVersion = exports.SchemaRegistry = exports.Schema = exports.getSchemaVersionOutput = exports.getSchemaVersion = exports.getSchemaRegistryOutput = exports.getSchemaRegistry = exports.getSchemaOutput = exports.getSchema = exports.getDiscoveredAssetEndpointProfileOutput = exports.getDiscoveredAssetEndpointProfile = exports.getDiscoveredAssetOutput = exports.getDiscoveredAsset = exports.getAssetEndpointProfileOutput = exports.getAssetEndpointProfile = exports.getAssetOutput = exports.getAsset = exports.DiscoveredAssetEndpointProfile = exports.DiscoveredAsset = exports.AssetEndpointProfile = exports.Asset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Asset = null;
utilities.lazyLoad(exports, ["Asset"], () => require("./asset"));
exports.AssetEndpointProfile = null;
utilities.lazyLoad(exports, ["AssetEndpointProfile"], () => require("./assetEndpointProfile"));
exports.DiscoveredAsset = null;
utilities.lazyLoad(exports, ["DiscoveredAsset"], () => require("./discoveredAsset"));
exports.DiscoveredAssetEndpointProfile = null;
utilities.lazyLoad(exports, ["DiscoveredAssetEndpointProfile"], () => require("./discoveredAssetEndpointProfile"));
exports.getAsset = null;
exports.getAssetOutput = null;
utilities.lazyLoad(exports, ["getAsset", "getAssetOutput"], () => require("./getAsset"));
exports.getAssetEndpointProfile = null;
exports.getAssetEndpointProfileOutput = null;
utilities.lazyLoad(exports, ["getAssetEndpointProfile", "getAssetEndpointProfileOutput"], () => require("./getAssetEndpointProfile"));
exports.getDiscoveredAsset = null;
exports.getDiscoveredAssetOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredAsset", "getDiscoveredAssetOutput"], () => require("./getDiscoveredAsset"));
exports.getDiscoveredAssetEndpointProfile = null;
exports.getDiscoveredAssetEndpointProfileOutput = null;
utilities.lazyLoad(exports, ["getDiscoveredAssetEndpointProfile", "getDiscoveredAssetEndpointProfileOutput"], () => require("./getDiscoveredAssetEndpointProfile"));
exports.getSchema = null;
exports.getSchemaOutput = null;
utilities.lazyLoad(exports, ["getSchema", "getSchemaOutput"], () => require("./getSchema"));
exports.getSchemaRegistry = null;
exports.getSchemaRegistryOutput = null;
utilities.lazyLoad(exports, ["getSchemaRegistry", "getSchemaRegistryOutput"], () => require("./getSchemaRegistry"));
exports.getSchemaVersion = null;
exports.getSchemaVersionOutput = null;
utilities.lazyLoad(exports, ["getSchemaVersion", "getSchemaVersionOutput"], () => require("./getSchemaVersion"));
exports.Schema = null;
utilities.lazyLoad(exports, ["Schema"], () => require("./schema"));
exports.SchemaRegistry = null;
utilities.lazyLoad(exports, ["SchemaRegistry"], () => require("./schemaRegistry"));
exports.SchemaVersion = null;
utilities.lazyLoad(exports, ["SchemaVersion"], () => require("./schemaVersion"));
// Export enums:
__exportStar(require("../types/enums/deviceregistry"), exports);
// Export sub-modules:
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:deviceregistry:Asset":
                return new exports.Asset(name, undefined, { urn });
            case "azure-native:deviceregistry:AssetEndpointProfile":
                return new exports.AssetEndpointProfile(name, undefined, { urn });
            case "azure-native:deviceregistry:DiscoveredAsset":
                return new exports.DiscoveredAsset(name, undefined, { urn });
            case "azure-native:deviceregistry:DiscoveredAssetEndpointProfile":
                return new exports.DiscoveredAssetEndpointProfile(name, undefined, { urn });
            case "azure-native:deviceregistry:Schema":
                return new exports.Schema(name, undefined, { urn });
            case "azure-native:deviceregistry:SchemaRegistry":
                return new exports.SchemaRegistry(name, undefined, { urn });
            case "azure-native:deviceregistry:SchemaVersion":
                return new exports.SchemaVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "deviceregistry", _module);
