// @pulumi/azure-native/types/enums/orbital/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Protocol = exports.Polarization = exports.Direction = exports.Capability = exports.AutoTrackingConfiguration = exports.v20240301preview = exports.v20240301 = exports.v20221101 = exports.v20220301 = void 0;
// Export sub-modules:
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
const v20221101 = require("./v20221101");
exports.v20221101 = v20221101;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const v20240301preview = require("./v20240301preview");
exports.v20240301preview = v20240301preview;
exports.AutoTrackingConfiguration = {
    Disabled: "disabled",
    XBand: "xBand",
    SBand: "sBand",
};
exports.Capability = {
    EarthObservation: "EarthObservation",
    Communication: "Communication",
};
exports.Direction = {
    Uplink: "Uplink",
    Downlink: "Downlink",
};
exports.Polarization = {
    RHCP: "RHCP",
    LHCP: "LHCP",
    LinearVertical: "linearVertical",
    LinearHorizontal: "linearHorizontal",
};
exports.Protocol = {
    TCP: "TCP",
    UDP: "UDP",
};
