// @pulumi/azure-native/servicelinker/v20221101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listLinkerConfigurationsOutput = exports.listLinkerConfigurations = exports.LinkerDryrun = exports.Linker = exports.getLinkerDryrunOutput = exports.getLinkerDryrun = exports.getLinkerOutput = exports.getLinker = exports.getConnectorDryrunOutput = exports.getConnectorDryrun = exports.getConnectorOutput = exports.getConnector = exports.ConnectorDryrun = exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.ConnectorDryrun = null;
utilities.lazyLoad(exports, ["ConnectorDryrun"], () => require("./connectorDryrun"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getConnectorDryrun = null;
exports.getConnectorDryrunOutput = null;
utilities.lazyLoad(exports, ["getConnectorDryrun", "getConnectorDryrunOutput"], () => require("./getConnectorDryrun"));
exports.getLinker = null;
exports.getLinkerOutput = null;
utilities.lazyLoad(exports, ["getLinker", "getLinkerOutput"], () => require("./getLinker"));
exports.getLinkerDryrun = null;
exports.getLinkerDryrunOutput = null;
utilities.lazyLoad(exports, ["getLinkerDryrun", "getLinkerDryrunOutput"], () => require("./getLinkerDryrun"));
exports.Linker = null;
utilities.lazyLoad(exports, ["Linker"], () => require("./linker"));
exports.LinkerDryrun = null;
utilities.lazyLoad(exports, ["LinkerDryrun"], () => require("./linkerDryrun"));
exports.listLinkerConfigurations = null;
exports.listLinkerConfigurationsOutput = null;
utilities.lazyLoad(exports, ["listLinkerConfigurations", "listLinkerConfigurationsOutput"], () => require("./listLinkerConfigurations"));
// Export enums:
__exportStar(require("../../types/enums/servicelinker/v20221101preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicelinker/v20221101preview:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:servicelinker/v20221101preview:ConnectorDryrun":
                return new exports.ConnectorDryrun(name, undefined, { urn });
            case "azure-native:servicelinker/v20221101preview:Linker":
                return new exports.Linker(name, undefined, { urn });
            case "azure-native:servicelinker/v20221101preview:LinkerDryrun":
                return new exports.LinkerDryrun(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicelinker/v20221101preview", _module);
