// @pulumi/azure-native/types/enums/hdinsight/v20230601preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleDay = exports.ScaleActionType = exports.KeyVaultObjectType = exports.ContentEncoding = exports.ComparisonOperator = exports.AutoscaleType = void 0;
exports.AutoscaleType = {
    ScheduleBased: "ScheduleBased",
    LoadBased: "LoadBased",
};
exports.ComparisonOperator = {
    GreaterThan: "greaterThan",
    GreaterThanOrEqual: "greaterThanOrEqual",
    LessThan: "lessThan",
    LessThanOrEqual: "lessThanOrEqual",
};
exports.ContentEncoding = {
    Base64: "Base64",
    None: "None",
};
exports.KeyVaultObjectType = {
    Key: "Key",
    Secret: "Secret",
    Certificate: "Certificate",
};
exports.ScaleActionType = {
    Scaleup: "scaleup",
    Scaledown: "scaledown",
};
exports.ScheduleDay = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
