// @pulumi/azure-native/webpubsub/v20240101preview/webPubSubReplica.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebPubSubReplica = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A class represent a replica resource.
 */
class WebPubSubReplica extends pulumi.CustomResource {
    /**
     * Get an existing WebPubSubReplica resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebPubSubReplica(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebPubSubReplica.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebPubSubReplica.__pulumiType;
    }
    /**
     * Create a WebPubSubReplica resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["regionEndpointEnabled"] = (_a = (args ? args.regionEndpointEnabled : undefined)) !== null && _a !== void 0 ? _a : "Enabled";
            resourceInputs["replicaName"] = args ? args.replicaName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["resourceStopped"] = (_b = (args ? args.resourceStopped : undefined)) !== null && _b !== void 0 ? _b : "false";
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["regionEndpointEnabled"] = undefined /*out*/;
            resourceInputs["resourceStopped"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:webpubsub/v20230301preview:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20230601preview:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20230801preview:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20240301:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20240401preview:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20240801preview:WebPubSubReplica" }, { type: "azure-native:webpubsub/v20241001preview:WebPubSubReplica" }, { type: "azure-native:webpubsub:WebPubSubReplica" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebPubSubReplica.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebPubSubReplica = WebPubSubReplica;
/** @internal */
WebPubSubReplica.__pulumiType = 'azure-native:webpubsub/v20240101preview:WebPubSubReplica';
