// @pulumi/azure-native/streamanalytics/v20200301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpoint = exports.listClusterStreamingJobsOutput = exports.listClusterStreamingJobs = exports.getPrivateEndpointOutput = exports.getPrivateEndpoint = exports.getClusterOutput = exports.getCluster = exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getPrivateEndpoint = null;
exports.getPrivateEndpointOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpoint", "getPrivateEndpointOutput"], () => require("./getPrivateEndpoint"));
exports.listClusterStreamingJobs = null;
exports.listClusterStreamingJobsOutput = null;
utilities.lazyLoad(exports, ["listClusterStreamingJobs", "listClusterStreamingJobsOutput"], () => require("./listClusterStreamingJobs"));
exports.PrivateEndpoint = null;
utilities.lazyLoad(exports, ["PrivateEndpoint"], () => require("./privateEndpoint"));
// Export enums:
__exportStar(require("../../types/enums/streamanalytics/v20200301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:streamanalytics/v20200301preview:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:streamanalytics/v20200301preview:PrivateEndpoint":
                return new exports.PrivateEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "streamanalytics/v20200301preview", _module);
