// @pulumi/azure-native/cognitiveservices/v20240601preview/commitmentPlanAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentPlanAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The commitment plan association.
 */
class CommitmentPlanAssociation extends pulumi.CustomResource {
    /**
     * Get an existing CommitmentPlanAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CommitmentPlanAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CommitmentPlanAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CommitmentPlanAssociation.__pulumiType;
    }
    /**
     * Create a CommitmentPlanAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.commitmentPlanName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'commitmentPlanName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["commitmentPlanAssociationName"] = args ? args.commitmentPlanAssociationName : undefined;
            resourceInputs["commitmentPlanName"] = args ? args.commitmentPlanName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cognitiveservices/v20221201:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices/v20230501:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices/v20231001preview:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices/v20240401preview:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices/v20241001:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices/v20250401preview:CommitmentPlanAssociation" }, { type: "azure-native:cognitiveservices:CommitmentPlanAssociation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CommitmentPlanAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CommitmentPlanAssociation = CommitmentPlanAssociation;
/** @internal */
CommitmentPlanAssociation.__pulumiType = 'azure-native:cognitiveservices/v20240601preview:CommitmentPlanAssociation';
