// @pulumi/azure-native/analysisservices/v20170801beta/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerDetails = exports.listServerGatewayStatusOutput = exports.listServerGatewayStatus = exports.getServerDetailsOutput = exports.getServerDetails = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getServerDetails = null;
exports.getServerDetailsOutput = null;
utilities.lazyLoad(exports, ["getServerDetails", "getServerDetailsOutput"], () => require("./getServerDetails"));
exports.listServerGatewayStatus = null;
exports.listServerGatewayStatusOutput = null;
utilities.lazyLoad(exports, ["listServerGatewayStatus", "listServerGatewayStatusOutput"], () => require("./listServerGatewayStatus"));
exports.ServerDetails = null;
utilities.lazyLoad(exports, ["ServerDetails"], () => require("./serverDetails"));
// Export enums:
__exportStar(require("../../types/enums/analysisservices/v20170801beta"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:analysisservices/v20170801beta:ServerDetails":
                return new exports.ServerDetails(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "analysisservices/v20170801beta", _module);
