// @pulumi/azure-native/operationalinsights/v20190901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryPack = exports.Query = exports.getQueryPackOutput = exports.getQueryPack = exports.getQueryOutput = exports.getQuery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getQuery = null;
exports.getQueryOutput = null;
utilities.lazyLoad(exports, ["getQuery", "getQueryOutput"], () => require("./getQuery"));
exports.getQueryPack = null;
exports.getQueryPackOutput = null;
utilities.lazyLoad(exports, ["getQueryPack", "getQueryPackOutput"], () => require("./getQueryPack"));
exports.Query = null;
utilities.lazyLoad(exports, ["Query"], () => require("./query"));
exports.QueryPack = null;
utilities.lazyLoad(exports, ["QueryPack"], () => require("./queryPack"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:operationalinsights/v20190901preview:Query":
                return new exports.Query(name, undefined, { urn });
            case "azure-native:operationalinsights/v20190901preview:QueryPack":
                return new exports.QueryPack(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "operationalinsights/v20190901preview", _module);
