// @pulumi/azure-native/types/enums/deviceregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAuthenticationMode = exports.TopicRetainType = exports.SystemAssignedServiceIdentityType = exports.SchemaType = exports.Format = exports.EventsObservabilityMode = exports.DataPointsObservabilityMode = exports.AuthenticationMethod = exports.v20241101 = exports.v20240901preview = exports.v20231101preview = void 0;
// Export sub-modules:
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
exports.AuthenticationMethod = {
    /**
     * The user authentication method is anonymous.
     */
    Anonymous: "Anonymous",
    /**
     * The user authentication method is an x509 certificate.
     */
    Certificate: "Certificate",
    /**
     * The user authentication method is a username and password.
     */
    UsernamePassword: "UsernamePassword",
};
exports.DataPointsObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "none",
    /**
     * Map as counter to OpenTelemetry.
     */
    Counter: "counter",
    /**
     * Map as gauge to OpenTelemetry.
     */
    Gauge: "gauge",
    /**
     * Map as histogram to OpenTelemetry.
     */
    Histogram: "histogram",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "log",
};
exports.EventsObservabilityMode = {
    /**
     * No mapping to OpenTelemetry.
     */
    None: "none",
    /**
     * Map as log to OpenTelemetry.
     */
    Log: "log",
};
exports.Format = {
    /**
     * JSON Schema version draft 7 format
     */
    JsonSchema_draft7: "JsonSchema/draft-07",
    /**
     * Delta format
     */
    Delta_1_0: "Delta/1.0",
};
exports.SchemaType = {
    /**
     * Message Schema schema type
     */
    MessageSchema: "MessageSchema",
};
exports.SystemAssignedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.TopicRetainType = {
    /**
     * Retain the messages.
     */
    Keep: "Keep",
    /**
     * Never retain messages.
     */
    Never: "Never",
};
exports.UserAuthenticationMode = {
    /**
     * The user authentication method is anonymous.
     */
    Anonymous: "Anonymous",
    /**
     * The user authentication method is an x509 certificate.
     */
    Certificate: "Certificate",
    /**
     * The user authentication method is a username and password.
     */
    UsernamePassword: "UsernamePassword",
};
