// @pulumi/azure-native/containerservice/v20230815preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRun = exports.listFleetCredentialsOutput = exports.listFleetCredentials = exports.getUpdateRunOutput = exports.getUpdateRun = exports.getFleetUpdateStrategyOutput = exports.getFleetUpdateStrategy = exports.getFleetMemberOutput = exports.getFleetMember = exports.getFleetOutput = exports.getFleet = exports.FleetUpdateStrategy = exports.FleetMember = exports.Fleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.FleetMember = null;
utilities.lazyLoad(exports, ["FleetMember"], () => require("./fleetMember"));
exports.FleetUpdateStrategy = null;
utilities.lazyLoad(exports, ["FleetUpdateStrategy"], () => require("./fleetUpdateStrategy"));
exports.getFleet = null;
exports.getFleetOutput = null;
utilities.lazyLoad(exports, ["getFleet", "getFleetOutput"], () => require("./getFleet"));
exports.getFleetMember = null;
exports.getFleetMemberOutput = null;
utilities.lazyLoad(exports, ["getFleetMember", "getFleetMemberOutput"], () => require("./getFleetMember"));
exports.getFleetUpdateStrategy = null;
exports.getFleetUpdateStrategyOutput = null;
utilities.lazyLoad(exports, ["getFleetUpdateStrategy", "getFleetUpdateStrategyOutput"], () => require("./getFleetUpdateStrategy"));
exports.getUpdateRun = null;
exports.getUpdateRunOutput = null;
utilities.lazyLoad(exports, ["getUpdateRun", "getUpdateRunOutput"], () => require("./getUpdateRun"));
exports.listFleetCredentials = null;
exports.listFleetCredentialsOutput = null;
utilities.lazyLoad(exports, ["listFleetCredentials", "listFleetCredentialsOutput"], () => require("./listFleetCredentials"));
exports.UpdateRun = null;
utilities.lazyLoad(exports, ["UpdateRun"], () => require("./updateRun"));
// Export enums:
__exportStar(require("../../types/enums/containerservice/v20230815preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerservice/v20230815preview:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "azure-native:containerservice/v20230815preview:FleetMember":
                return new exports.FleetMember(name, undefined, { urn });
            case "azure-native:containerservice/v20230815preview:FleetUpdateStrategy":
                return new exports.FleetUpdateStrategy(name, undefined, { urn });
            case "azure-native:containerservice/v20230815preview:UpdateRun":
                return new exports.UpdateRun(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerservice/v20230815preview", _module);
