// @pulumi/azure-native/batch/v20240201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pool = exports.listBatchAccountKeysOutput = exports.listBatchAccountKeys = exports.getPoolOutput = exports.getPool = exports.getBatchAccountOutput = exports.getBatchAccount = exports.getApplicationPackageOutput = exports.getApplicationPackage = exports.getApplicationOutput = exports.getApplication = exports.BatchAccount = exports.ApplicationPackage = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationPackage = null;
utilities.lazyLoad(exports, ["ApplicationPackage"], () => require("./applicationPackage"));
exports.BatchAccount = null;
utilities.lazyLoad(exports, ["BatchAccount"], () => require("./batchAccount"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getApplicationPackage = null;
exports.getApplicationPackageOutput = null;
utilities.lazyLoad(exports, ["getApplicationPackage", "getApplicationPackageOutput"], () => require("./getApplicationPackage"));
exports.getBatchAccount = null;
exports.getBatchAccountOutput = null;
utilities.lazyLoad(exports, ["getBatchAccount", "getBatchAccountOutput"], () => require("./getBatchAccount"));
exports.getPool = null;
exports.getPoolOutput = null;
utilities.lazyLoad(exports, ["getPool", "getPoolOutput"], () => require("./getPool"));
exports.listBatchAccountKeys = null;
exports.listBatchAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listBatchAccountKeys", "listBatchAccountKeysOutput"], () => require("./listBatchAccountKeys"));
exports.Pool = null;
utilities.lazyLoad(exports, ["Pool"], () => require("./pool"));
// Export enums:
__exportStar(require("../../types/enums/batch/v20240201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:batch/v20240201:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:batch/v20240201:ApplicationPackage":
                return new exports.ApplicationPackage(name, undefined, { urn });
            case "azure-native:batch/v20240201:BatchAccount":
                return new exports.BatchAccount(name, undefined, { urn });
            case "azure-native:batch/v20240201:Pool":
                return new exports.Pool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "batch/v20240201", _module);
