// @pulumi/azure-native/hybridcompute/v20200815preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkScopedResource = exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.MachineExtension = exports.Machine = exports.getPrivateLinkScopedResourceOutput = exports.getPrivateLinkScopedResource = exports.getPrivateLinkScopeOutput = exports.getPrivateLinkScope = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getMachineExtensionOutput = exports.getMachineExtension = exports.getMachineOutput = exports.getMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMachine = null;
exports.getMachineOutput = null;
utilities.lazyLoad(exports, ["getMachine", "getMachineOutput"], () => require("./getMachine"));
exports.getMachineExtension = null;
exports.getMachineExtensionOutput = null;
utilities.lazyLoad(exports, ["getMachineExtension", "getMachineExtensionOutput"], () => require("./getMachineExtension"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkScope = null;
exports.getPrivateLinkScopeOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScope", "getPrivateLinkScopeOutput"], () => require("./getPrivateLinkScope"));
exports.getPrivateLinkScopedResource = null;
exports.getPrivateLinkScopedResourceOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScopedResource", "getPrivateLinkScopedResourceOutput"], () => require("./getPrivateLinkScopedResource"));
exports.Machine = null;
utilities.lazyLoad(exports, ["Machine"], () => require("./machine"));
exports.MachineExtension = null;
utilities.lazyLoad(exports, ["MachineExtension"], () => require("./machineExtension"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkScope = null;
utilities.lazyLoad(exports, ["PrivateLinkScope"], () => require("./privateLinkScope"));
exports.PrivateLinkScopedResource = null;
utilities.lazyLoad(exports, ["PrivateLinkScopedResource"], () => require("./privateLinkScopedResource"));
// Export enums:
__exportStar(require("../../types/enums/hybridcompute/v20200815preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcompute/v20200815preview:Machine":
                return new exports.Machine(name, undefined, { urn });
            case "azure-native:hybridcompute/v20200815preview:MachineExtension":
                return new exports.MachineExtension(name, undefined, { urn });
            case "azure-native:hybridcompute/v20200815preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:hybridcompute/v20200815preview:PrivateLinkScope":
                return new exports.PrivateLinkScope(name, undefined, { urn });
            case "azure-native:hybridcompute/v20200815preview:PrivateLinkScopedResource":
                return new exports.PrivateLinkScopedResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcompute/v20200815preview", _module);
