// @pulumi/azure-native/security/v20240515preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listGitLabSubgroupOutput = exports.listGitLabSubgroup = exports.listGitLabGroupAvailableOutput = exports.listGitLabGroupAvailable = exports.listGitHubOwnerAvailableOutput = exports.listGitHubOwnerAvailable = exports.listAzureDevOpsOrgAvailableOutput = exports.listAzureDevOpsOrgAvailable = exports.getDevOpsPolicyAssignmentOutput = exports.getDevOpsPolicyAssignment = exports.getDevOpsConfigurationOutput = exports.getDevOpsConfiguration = exports.DevOpsPolicyAssignment = exports.DevOpsConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DevOpsConfiguration = null;
utilities.lazyLoad(exports, ["DevOpsConfiguration"], () => require("./devOpsConfiguration"));
exports.DevOpsPolicyAssignment = null;
utilities.lazyLoad(exports, ["DevOpsPolicyAssignment"], () => require("./devOpsPolicyAssignment"));
exports.getDevOpsConfiguration = null;
exports.getDevOpsConfigurationOutput = null;
utilities.lazyLoad(exports, ["getDevOpsConfiguration", "getDevOpsConfigurationOutput"], () => require("./getDevOpsConfiguration"));
exports.getDevOpsPolicyAssignment = null;
exports.getDevOpsPolicyAssignmentOutput = null;
utilities.lazyLoad(exports, ["getDevOpsPolicyAssignment", "getDevOpsPolicyAssignmentOutput"], () => require("./getDevOpsPolicyAssignment"));
exports.listAzureDevOpsOrgAvailable = null;
exports.listAzureDevOpsOrgAvailableOutput = null;
utilities.lazyLoad(exports, ["listAzureDevOpsOrgAvailable", "listAzureDevOpsOrgAvailableOutput"], () => require("./listAzureDevOpsOrgAvailable"));
exports.listGitHubOwnerAvailable = null;
exports.listGitHubOwnerAvailableOutput = null;
utilities.lazyLoad(exports, ["listGitHubOwnerAvailable", "listGitHubOwnerAvailableOutput"], () => require("./listGitHubOwnerAvailable"));
exports.listGitLabGroupAvailable = null;
exports.listGitLabGroupAvailableOutput = null;
utilities.lazyLoad(exports, ["listGitLabGroupAvailable", "listGitLabGroupAvailableOutput"], () => require("./listGitLabGroupAvailable"));
exports.listGitLabSubgroup = null;
exports.listGitLabSubgroupOutput = null;
utilities.lazyLoad(exports, ["listGitLabSubgroup", "listGitLabSubgroupOutput"], () => require("./listGitLabSubgroup"));
// Export enums:
__exportStar(require("../../types/enums/security/v20240515preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:security/v20240515preview:DevOpsConfiguration":
                return new exports.DevOpsConfiguration(name, undefined, { urn });
            case "azure-native:security/v20240515preview:DevOpsPolicyAssignment":
                return new exports.DevOpsPolicyAssignment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "security/v20240515preview", _module);
