// @pulumi/azure-native/media/v20221101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingEndpoint = exports.LiveOutput = exports.LiveEvent = exports.getStreamingEndpointOutput = exports.getStreamingEndpoint = exports.getLiveOutputOutput = exports.getLiveOutput = exports.getLiveEventTrackIngestHeartbeatsOutput = exports.getLiveEventTrackIngestHeartbeats = exports.getLiveEventStreamEventsOutput = exports.getLiveEventStreamEvents = exports.getLiveEventStatusOutput = exports.getLiveEventStatus = exports.getLiveEventOutput = exports.getLiveEvent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLiveEvent = null;
exports.getLiveEventOutput = null;
utilities.lazyLoad(exports, ["getLiveEvent", "getLiveEventOutput"], () => require("./getLiveEvent"));
exports.getLiveEventStatus = null;
exports.getLiveEventStatusOutput = null;
utilities.lazyLoad(exports, ["getLiveEventStatus", "getLiveEventStatusOutput"], () => require("./getLiveEventStatus"));
exports.getLiveEventStreamEvents = null;
exports.getLiveEventStreamEventsOutput = null;
utilities.lazyLoad(exports, ["getLiveEventStreamEvents", "getLiveEventStreamEventsOutput"], () => require("./getLiveEventStreamEvents"));
exports.getLiveEventTrackIngestHeartbeats = null;
exports.getLiveEventTrackIngestHeartbeatsOutput = null;
utilities.lazyLoad(exports, ["getLiveEventTrackIngestHeartbeats", "getLiveEventTrackIngestHeartbeatsOutput"], () => require("./getLiveEventTrackIngestHeartbeats"));
exports.getLiveOutput = null;
exports.getLiveOutputOutput = null;
utilities.lazyLoad(exports, ["getLiveOutput", "getLiveOutputOutput"], () => require("./getLiveOutput"));
exports.getStreamingEndpoint = null;
exports.getStreamingEndpointOutput = null;
utilities.lazyLoad(exports, ["getStreamingEndpoint", "getStreamingEndpointOutput"], () => require("./getStreamingEndpoint"));
exports.LiveEvent = null;
utilities.lazyLoad(exports, ["LiveEvent"], () => require("./liveEvent"));
exports.LiveOutput = null;
utilities.lazyLoad(exports, ["LiveOutput"], () => require("./liveOutput"));
exports.StreamingEndpoint = null;
utilities.lazyLoad(exports, ["StreamingEndpoint"], () => require("./streamingEndpoint"));
// Export enums:
__exportStar(require("../../types/enums/media/v20221101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:media/v20221101:LiveEvent":
                return new exports.LiveEvent(name, undefined, { urn });
            case "azure-native:media/v20221101:LiveOutput":
                return new exports.LiveOutput(name, undefined, { urn });
            case "azure-native:media/v20221101:StreamingEndpoint":
                return new exports.StreamingEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "media/v20221101", _module);
