// @pulumi/azure-native/iotoperationsmq/v20231004preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MqttBridgeTopicMap = exports.MqttBridgeConnector = exports.Mq = exports.KafkaConnectorTopicMap = exports.KafkaConnector = exports.getMqttBridgeTopicMapOutput = exports.getMqttBridgeTopicMap = exports.getMqttBridgeConnectorOutput = exports.getMqttBridgeConnector = exports.getMqOutput = exports.getMq = exports.getKafkaConnectorTopicMapOutput = exports.getKafkaConnectorTopicMap = exports.getKafkaConnectorOutput = exports.getKafkaConnector = exports.getDiagnosticServiceOutput = exports.getDiagnosticService = exports.getDataLakeConnectorTopicMapOutput = exports.getDataLakeConnectorTopicMap = exports.getDataLakeConnectorOutput = exports.getDataLakeConnector = exports.getBrokerListenerOutput = exports.getBrokerListener = exports.getBrokerAuthorizationOutput = exports.getBrokerAuthorization = exports.getBrokerAuthenticationOutput = exports.getBrokerAuthentication = exports.getBrokerOutput = exports.getBroker = exports.DiagnosticService = exports.DataLakeConnectorTopicMap = exports.DataLakeConnector = exports.BrokerListener = exports.BrokerAuthorization = exports.BrokerAuthentication = exports.Broker = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Broker = null;
utilities.lazyLoad(exports, ["Broker"], () => require("./broker"));
exports.BrokerAuthentication = null;
utilities.lazyLoad(exports, ["BrokerAuthentication"], () => require("./brokerAuthentication"));
exports.BrokerAuthorization = null;
utilities.lazyLoad(exports, ["BrokerAuthorization"], () => require("./brokerAuthorization"));
exports.BrokerListener = null;
utilities.lazyLoad(exports, ["BrokerListener"], () => require("./brokerListener"));
exports.DataLakeConnector = null;
utilities.lazyLoad(exports, ["DataLakeConnector"], () => require("./dataLakeConnector"));
exports.DataLakeConnectorTopicMap = null;
utilities.lazyLoad(exports, ["DataLakeConnectorTopicMap"], () => require("./dataLakeConnectorTopicMap"));
exports.DiagnosticService = null;
utilities.lazyLoad(exports, ["DiagnosticService"], () => require("./diagnosticService"));
exports.getBroker = null;
exports.getBrokerOutput = null;
utilities.lazyLoad(exports, ["getBroker", "getBrokerOutput"], () => require("./getBroker"));
exports.getBrokerAuthentication = null;
exports.getBrokerAuthenticationOutput = null;
utilities.lazyLoad(exports, ["getBrokerAuthentication", "getBrokerAuthenticationOutput"], () => require("./getBrokerAuthentication"));
exports.getBrokerAuthorization = null;
exports.getBrokerAuthorizationOutput = null;
utilities.lazyLoad(exports, ["getBrokerAuthorization", "getBrokerAuthorizationOutput"], () => require("./getBrokerAuthorization"));
exports.getBrokerListener = null;
exports.getBrokerListenerOutput = null;
utilities.lazyLoad(exports, ["getBrokerListener", "getBrokerListenerOutput"], () => require("./getBrokerListener"));
exports.getDataLakeConnector = null;
exports.getDataLakeConnectorOutput = null;
utilities.lazyLoad(exports, ["getDataLakeConnector", "getDataLakeConnectorOutput"], () => require("./getDataLakeConnector"));
exports.getDataLakeConnectorTopicMap = null;
exports.getDataLakeConnectorTopicMapOutput = null;
utilities.lazyLoad(exports, ["getDataLakeConnectorTopicMap", "getDataLakeConnectorTopicMapOutput"], () => require("./getDataLakeConnectorTopicMap"));
exports.getDiagnosticService = null;
exports.getDiagnosticServiceOutput = null;
utilities.lazyLoad(exports, ["getDiagnosticService", "getDiagnosticServiceOutput"], () => require("./getDiagnosticService"));
exports.getKafkaConnector = null;
exports.getKafkaConnectorOutput = null;
utilities.lazyLoad(exports, ["getKafkaConnector", "getKafkaConnectorOutput"], () => require("./getKafkaConnector"));
exports.getKafkaConnectorTopicMap = null;
exports.getKafkaConnectorTopicMapOutput = null;
utilities.lazyLoad(exports, ["getKafkaConnectorTopicMap", "getKafkaConnectorTopicMapOutput"], () => require("./getKafkaConnectorTopicMap"));
exports.getMq = null;
exports.getMqOutput = null;
utilities.lazyLoad(exports, ["getMq", "getMqOutput"], () => require("./getMq"));
exports.getMqttBridgeConnector = null;
exports.getMqttBridgeConnectorOutput = null;
utilities.lazyLoad(exports, ["getMqttBridgeConnector", "getMqttBridgeConnectorOutput"], () => require("./getMqttBridgeConnector"));
exports.getMqttBridgeTopicMap = null;
exports.getMqttBridgeTopicMapOutput = null;
utilities.lazyLoad(exports, ["getMqttBridgeTopicMap", "getMqttBridgeTopicMapOutput"], () => require("./getMqttBridgeTopicMap"));
exports.KafkaConnector = null;
utilities.lazyLoad(exports, ["KafkaConnector"], () => require("./kafkaConnector"));
exports.KafkaConnectorTopicMap = null;
utilities.lazyLoad(exports, ["KafkaConnectorTopicMap"], () => require("./kafkaConnectorTopicMap"));
exports.Mq = null;
utilities.lazyLoad(exports, ["Mq"], () => require("./mq"));
exports.MqttBridgeConnector = null;
utilities.lazyLoad(exports, ["MqttBridgeConnector"], () => require("./mqttBridgeConnector"));
exports.MqttBridgeTopicMap = null;
utilities.lazyLoad(exports, ["MqttBridgeTopicMap"], () => require("./mqttBridgeTopicMap"));
// Export enums:
__exportStar(require("../../types/enums/iotoperationsmq/v20231004preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:iotoperationsmq/v20231004preview:Broker":
                return new exports.Broker(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:BrokerAuthentication":
                return new exports.BrokerAuthentication(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:BrokerAuthorization":
                return new exports.BrokerAuthorization(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:BrokerListener":
                return new exports.BrokerListener(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:DataLakeConnector":
                return new exports.DataLakeConnector(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:DataLakeConnectorTopicMap":
                return new exports.DataLakeConnectorTopicMap(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:DiagnosticService":
                return new exports.DiagnosticService(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:KafkaConnector":
                return new exports.KafkaConnector(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:KafkaConnectorTopicMap":
                return new exports.KafkaConnectorTopicMap(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:Mq":
                return new exports.Mq(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:MqttBridgeConnector":
                return new exports.MqttBridgeConnector(name, undefined, { urn });
            case "azure-native:iotoperationsmq/v20231004preview:MqttBridgeTopicMap":
                return new exports.MqttBridgeTopicMap(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "iotoperationsmq/v20231004preview", _module);
