// @pulumi/azure-native/types/enums/cloudngfw/v20230901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageType = exports.StateEnum = exports.ScopeType = exports.ProtocolType = exports.NetworkType = exports.MarketplaceSubscriptionStatus = exports.ManagedIdentityType = exports.EnabledDNSType = exports.EgressNat = exports.DefaultMode = exports.DecryptionRuleTypeEnum = exports.DNSProxy = exports.BooleanEnum = exports.BillingCycle = exports.ActionEnum = void 0;
exports.ActionEnum = {
    Allow: "Allow",
    DenySilent: "DenySilent",
    DenyResetServer: "DenyResetServer",
    DenyResetBoth: "DenyResetBoth",
};
exports.BillingCycle = {
    WEEKLY: "WEEKLY",
    MONTHLY: "MONTHLY",
};
exports.BooleanEnum = {
    TRUE: "TRUE",
    FALSE: "FALSE",
};
exports.DNSProxy = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.DecryptionRuleTypeEnum = {
    SSLOutboundInspection: "SSLOutboundInspection",
    SSLInboundInspection: "SSLInboundInspection",
    None: "None",
};
exports.DefaultMode = {
    IPS: "IPS",
    FIREWALL: "FIREWALL",
    NONE: "NONE",
};
exports.EgressNat = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.EnabledDNSType = {
    CUSTOM: "CUSTOM",
    AZURE: "AZURE",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAndUserAssigned: "SystemAssigned,UserAssigned",
};
exports.MarketplaceSubscriptionStatus = {
    PendingFulfillmentStart: "PendingFulfillmentStart",
    Subscribed: "Subscribed",
    Suspended: "Suspended",
    Unsubscribed: "Unsubscribed",
    NotStarted: "NotStarted",
    FulfillmentRequested: "FulfillmentRequested",
};
exports.NetworkType = {
    VNET: "VNET",
    VWAN: "VWAN",
};
exports.ProtocolType = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.ScopeType = {
    LOCAL: "LOCAL",
    GLOBAL: "GLOBAL",
};
exports.StateEnum = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.UsageType = {
    PAYG: "PAYG",
    COMMITTED: "COMMITTED",
};
