// @pulumi/azure-native/types/enums/machinelearningservices/v20240101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenerationSafetyQualityMetric = exports.ForecastingPrimaryMetrics = exports.ForecastingModels = exports.ForecastHorizonMode = exports.FineTuningTaskType = exports.FeaturizationMode = exports.FeatureLags = exports.FeatureImportanceMode = exports.FeatureDataType = exports.FeatureAttributionMetric = exports.EnvironmentVariableType = exports.EndpointServiceConnectionStatus = exports.EndpointComputeType = exports.EndpointAuthMode = exports.EncryptionStatus = exports.EmailNotificationEnableType = exports.EgressPublicNetworkAccessType = exports.EarlyTerminationPolicyType = exports.DistributionType = exports.DeploymentModelVersionUpgradeOption = exports.DatastoreType = exports.DataType = exports.DataImportSourceType = exports.DataCollectionMode = exports.CredentialsType = exports.ContainerType = exports.ConnectionCategory = exports.ConnectionAuthType = exports.ComputeWeekDay = exports.ComputeType = exports.ComputeTriggerType = exports.ComputeRecurrenceFrequency = exports.ComputePowerAction = exports.ComputeInstanceAuthorizationType = exports.ClusterPurpose = exports.ClassificationPrimaryMetrics = exports.ClassificationMultilabelPrimaryMetrics = exports.ClassificationModels = exports.ClassificationModelPerformanceMetric = exports.CategoricalPredictionDriftMetric = exports.CategoricalDataQualityMetric = exports.CategoricalDataDriftMetric = exports.BlockedTransformers = exports.BatchOutputAction = exports.BatchLoggingLevel = exports.BatchDeploymentConfigurationType = exports.AutoRebuildSetting = exports.AutoDeleteCondition = exports.AuthMode = exports.ApplicationSharingPolicy = void 0;
exports.PrivateEndpointConnectionProvisioningState = exports.OutputDeliveryMode = exports.OsType = exports.OperatingSystemType = exports.OneLakeArtifactType = exports.ObjectDetectionPrimaryMetrics = exports.NumericalPredictionDriftMetric = exports.NumericalDataQualityMetric = exports.NumericalDataDriftMetric = exports.NodesValueType = exports.NlpLearningRateScheduler = exports.NCrossValidationsMode = exports.MultiSelect = exports.MonitoringSignalType = exports.MonitoringNotificationType = exports.MonitoringModelType = exports.MonitoringInputDataType = exports.MonitoringFeatureFilterType = exports.MonitoringFeatureDataType = exports.MonitorComputeType = exports.MonitorComputeIdentityType = exports.ModelTaskType = exports.ModelSize = exports.ModelProvider = exports.MlflowAutologger = exports.MediaType = exports.MaterializationStoreType = exports.ManagedServiceIdentityType = exports.ManagedNetworkStatus = exports.MLFlowAutologgerState = exports.MLAssistConfigurationType = exports.LogVerbosity = exports.LogValidationLoss = exports.LogTrainingMetrics = exports.LoadBalancerType = exports.LearningRateScheduler = exports.JobType = exports.JobTier = exports.JobOutputType = exports.JobLimitsType = exports.JobInputType = exports.IsolationMode = exports.InstanceSegmentationPrimaryMetrics = exports.InputDeliveryMode = exports.IncrementalDataRefresh = exports.ImageType = exports.ImageAnnotationType = exports.IdentityConfigurationType = exports.Goal = exports.GenerationTokenUsageMetric = void 0;
exports.WeekDay = exports.WebhookType = exports.VolumeDefinitionType = exports.VmPriority = exports.ValidationMetricType = exports.UseStl = exports.TriggerType = exports.TrainingMode = exports.TextAnnotationType = exports.TaskType = exports.TargetRollingWindowSizeMode = exports.TargetLagsMode = exports.TargetAggregationFunction = exports.StochasticOptimizer = exports.StackMetaLearnerType = exports.SslConfigStatus = exports.SshPublicAccess = exports.SparkJobEntryType = exports.SkuTier = exports.ShortSeriesHandlingConfiguration = exports.ServiceDataAccessAuthIdentity = exports.ServerlessInferenceEndpointAuthMode = exports.SecretsType = exports.SeasonalityMode = exports.ScheduleStatus = exports.ScheduleProvisioningState = exports.ScheduleActionType = exports.ScaleType = exports.SamplingAlgorithmType = exports.RuleType = exports.RuleStatus = exports.RuleCategory = exports.RuleAction = exports.RollingRateType = exports.RemoteLoginPortPublicAccess = exports.RegressionPrimaryMetrics = exports.RegressionModels = exports.RegressionModelPerformanceMetric = exports.ReferenceType = exports.RecurrenceFrequency = exports.RandomSamplingAlgorithmRule = exports.PublicNetworkAccessType = exports.Protocol = exports.ProtectionLevel = void 0;
exports.ApplicationSharingPolicy = {
    Personal: "Personal",
    Shared: "Shared",
};
exports.AuthMode = {
    AAD: "AAD",
};
exports.AutoDeleteCondition = {
    CreatedGreaterThan: "CreatedGreaterThan",
    LastAccessedGreaterThan: "LastAccessedGreaterThan",
};
exports.AutoRebuildSetting = {
    Disabled: "Disabled",
    OnBaseImageUpdate: "OnBaseImageUpdate",
};
exports.BatchDeploymentConfigurationType = {
    Model: "Model",
    PipelineComponent: "PipelineComponent",
};
exports.BatchLoggingLevel = {
    Info: "Info",
    Warning: "Warning",
    Debug: "Debug",
};
exports.BatchOutputAction = {
    SummaryOnly: "SummaryOnly",
    AppendRow: "AppendRow",
};
exports.BlockedTransformers = {
    /**
     * Target encoding for text data.
     */
    TextTargetEncoder: "TextTargetEncoder",
    /**
     * Ohe hot encoding creates a binary feature transformation.
     */
    OneHotEncoder: "OneHotEncoder",
    /**
     * Target encoding for categorical data.
     */
    CatTargetEncoder: "CatTargetEncoder",
    /**
     * Tf-Idf stands for, term-frequency times inverse document-frequency. This is a common term weighting scheme for identifying information from documents.
     */
    TfIdf: "TfIdf",
    /**
     * Weight of Evidence encoding is a technique used to encode categorical variables. It uses the natural log of the P(1)/P(0) to create weights.
     */
    WoETargetEncoder: "WoETargetEncoder",
    /**
     * Label encoder converts labels/categorical variables in a numerical form.
     */
    LabelEncoder: "LabelEncoder",
    /**
     * Word embedding helps represents words or phrases as a vector, or a series of numbers.
     */
    WordEmbedding: "WordEmbedding",
    /**
     * Naive Bayes is a classified that is used for classification of discrete features that are categorically distributed.
     */
    NaiveBayes: "NaiveBayes",
    /**
     * Count Vectorizer converts a collection of text documents to a matrix of token counts.
     */
    CountVectorizer: "CountVectorizer",
    /**
     * Hashing One Hot Encoder can turn categorical variables into a limited number of new features. This is often used for high-cardinality categorical features.
     */
    HashOneHotEncoder: "HashOneHotEncoder",
};
exports.CategoricalDataDriftMetric = {
    /**
     * The Jensen Shannon Distance (JSD) metric.
     */
    JensenShannonDistance: "JensenShannonDistance",
    /**
     * The Population Stability Index (PSI) metric.
     */
    PopulationStabilityIndex: "PopulationStabilityIndex",
    /**
     * The Pearsons Chi Squared Test metric.
     */
    PearsonsChiSquaredTest: "PearsonsChiSquaredTest",
};
exports.CategoricalDataQualityMetric = {
    /**
     * Calculates the rate of null values.
     */
    NullValueRate: "NullValueRate",
    /**
     * Calculates the rate of data type errors.
     */
    DataTypeErrorRate: "DataTypeErrorRate",
    /**
     * Calculates the rate values are out of bounds.
     */
    OutOfBoundsRate: "OutOfBoundsRate",
};
exports.CategoricalPredictionDriftMetric = {
    /**
     * The Jensen Shannon Distance (JSD) metric.
     */
    JensenShannonDistance: "JensenShannonDistance",
    /**
     * The Population Stability Index (PSI) metric.
     */
    PopulationStabilityIndex: "PopulationStabilityIndex",
    /**
     * The Pearsons Chi Squared Test metric.
     */
    PearsonsChiSquaredTest: "PearsonsChiSquaredTest",
};
exports.ClassificationModelPerformanceMetric = {
    /**
     * Calculates the accuracy of the model predictions.
     */
    Accuracy: "Accuracy",
    /**
     * Calculates the precision of the model predictions.
     */
    Precision: "Precision",
    /**
     * Calculates the recall of the model predictions.
     */
    Recall: "Recall",
};
exports.ClassificationModels = {
    /**
     * Logistic regression is a fundamental classification technique.
     * It belongs to the group of linear classifiers and is somewhat similar to polynomial and linear regression.
     * Logistic regression is fast and relatively uncomplicated, and it's convenient for you to interpret the results.
     * Although it's essentially a method for binary classification, it can also be applied to multiclass problems.
     */
    LogisticRegression: "LogisticRegression",
    /**
     * SGD: Stochastic gradient descent is an optimization algorithm often used in machine learning applications
     * to find the model parameters that correspond to the best fit between predicted and actual outputs.
     */
    SGD: "SGD",
    /**
     * The multinomial Naive Bayes classifier is suitable for classification with discrete features (e.g., word counts for text classification).
     * The multinomial distribution normally requires integer feature counts. However, in practice, fractional counts such as tf-idf may also work.
     */
    MultinomialNaiveBayes: "MultinomialNaiveBayes",
    /**
     * Naive Bayes classifier for multivariate Bernoulli models.
     */
    BernoulliNaiveBayes: "BernoulliNaiveBayes",
    /**
     * A support vector machine (SVM) is a supervised machine learning model that uses classification algorithms for two-group classification problems.
     * After giving an SVM model sets of labeled training data for each category, they're able to categorize new text.
     */
    SVM: "SVM",
    /**
     * A support vector machine (SVM) is a supervised machine learning model that uses classification algorithms for two-group classification problems.
     * After giving an SVM model sets of labeled training data for each category, they're able to categorize new text.
     * Linear SVM performs best when input data is linear, i.e., data can be easily classified by drawing the straight line between classified values on a plotted graph.
     */
    LinearSVM: "LinearSVM",
    /**
     * K-nearest neighbors (KNN) algorithm uses 'feature similarity' to predict the values of new datapoints
     * which further means that the new data point will be assigned a value based on how closely it matches the points in the training set.
     */
    KNN: "KNN",
    /**
     * Decision Trees are a non-parametric supervised learning method used for both classification and regression tasks.
     * The goal is to create a model that predicts the value of a target variable by learning simple decision rules inferred from the data features.
     */
    DecisionTree: "DecisionTree",
    /**
     * Random forest is a supervised learning algorithm.
     * The "forest" it builds, is an ensemble of decision trees, usually trained with the bagging method.
     * The general idea of the bagging method is that a combination of learning models increases the overall result.
     */
    RandomForest: "RandomForest",
    /**
     * Extreme Trees is an ensemble machine learning algorithm that combines the predictions from many decision trees. It is related to the widely used random forest algorithm.
     */
    ExtremeRandomTrees: "ExtremeRandomTrees",
    /**
     * LightGBM is a gradient boosting framework that uses tree based learning algorithms.
     */
    LightGBM: "LightGBM",
    /**
     * The technique of transiting week learners into a strong learner is called Boosting. The gradient boosting algorithm process works on this theory of execution.
     */
    GradientBoosting: "GradientBoosting",
    /**
     * XGBoost: Extreme Gradient Boosting Algorithm. This algorithm is used for structured data where target column values can be divided into distinct class values.
     */
    XGBoostClassifier: "XGBoostClassifier",
};
exports.ClassificationMultilabelPrimaryMetrics = {
    /**
     * AUC is the Area under the curve.
     * This metric represents arithmetic mean of the score for each class,
     * weighted by the number of true instances in each class.
     */
    AUCWeighted: "AUCWeighted",
    /**
     * Accuracy is the ratio of predictions that exactly match the true class labels.
     */
    Accuracy: "Accuracy",
    /**
     * Normalized macro recall is recall macro-averaged and normalized, so that random
     * performance has a score of 0, and perfect performance has a score of 1.
     */
    NormMacroRecall: "NormMacroRecall",
    /**
     * The arithmetic mean of the average precision score for each class, weighted by
     * the number of true instances in each class.
     */
    AveragePrecisionScoreWeighted: "AveragePrecisionScoreWeighted",
    /**
     * The arithmetic mean of precision for each class, weighted by number of true instances in each class.
     */
    PrecisionScoreWeighted: "PrecisionScoreWeighted",
    /**
     * Intersection Over Union. Intersection of predictions divided by union of predictions.
     */
    IOU: "IOU",
};
exports.ClassificationPrimaryMetrics = {
    /**
     * AUC is the Area under the curve.
     * This metric represents arithmetic mean of the score for each class,
     * weighted by the number of true instances in each class.
     */
    AUCWeighted: "AUCWeighted",
    /**
     * Accuracy is the ratio of predictions that exactly match the true class labels.
     */
    Accuracy: "Accuracy",
    /**
     * Normalized macro recall is recall macro-averaged and normalized, so that random
     * performance has a score of 0, and perfect performance has a score of 1.
     */
    NormMacroRecall: "NormMacroRecall",
    /**
     * The arithmetic mean of the average precision score for each class, weighted by
     * the number of true instances in each class.
     */
    AveragePrecisionScoreWeighted: "AveragePrecisionScoreWeighted",
    /**
     * The arithmetic mean of precision for each class, weighted by number of true instances in each class.
     */
    PrecisionScoreWeighted: "PrecisionScoreWeighted",
};
exports.ClusterPurpose = {
    FastProd: "FastProd",
    DenseProd: "DenseProd",
    DevTest: "DevTest",
};
exports.ComputeInstanceAuthorizationType = {
    Personal: "personal",
};
exports.ComputePowerAction = {
    Start: "Start",
    Stop: "Stop",
};
exports.ComputeRecurrenceFrequency = {
    /**
     * Minute frequency
     */
    Minute: "Minute",
    /**
     * Hour frequency
     */
    Hour: "Hour",
    /**
     * Day frequency
     */
    Day: "Day",
    /**
     * Week frequency
     */
    Week: "Week",
    /**
     * Month frequency
     */
    Month: "Month",
};
exports.ComputeTriggerType = {
    Recurrence: "Recurrence",
    Cron: "Cron",
};
exports.ComputeType = {
    AKS: "AKS",
    Kubernetes: "Kubernetes",
    AmlCompute: "AmlCompute",
    ComputeInstance: "ComputeInstance",
    DataFactory: "DataFactory",
    VirtualMachine: "VirtualMachine",
    HDInsight: "HDInsight",
    Databricks: "Databricks",
    DataLakeAnalytics: "DataLakeAnalytics",
    SynapseSpark: "SynapseSpark",
};
exports.ComputeWeekDay = {
    /**
     * Monday weekday
     */
    Monday: "Monday",
    /**
     * Tuesday weekday
     */
    Tuesday: "Tuesday",
    /**
     * Wednesday weekday
     */
    Wednesday: "Wednesday",
    /**
     * Thursday weekday
     */
    Thursday: "Thursday",
    /**
     * Friday weekday
     */
    Friday: "Friday",
    /**
     * Saturday weekday
     */
    Saturday: "Saturday",
    /**
     * Sunday weekday
     */
    Sunday: "Sunday",
};
exports.ConnectionAuthType = {
    PAT: "PAT",
    ManagedIdentity: "ManagedIdentity",
    UsernamePassword: "UsernamePassword",
    None: "None",
    SAS: "SAS",
    AccountKey: "AccountKey",
    ServicePrincipal: "ServicePrincipal",
    AccessKey: "AccessKey",
    ApiKey: "ApiKey",
    CustomKeys: "CustomKeys",
    OAuth2: "OAuth2",
    AAD: "AAD",
};
exports.ConnectionCategory = {
    PythonFeed: "PythonFeed",
    ContainerRegistry: "ContainerRegistry",
    Git: "Git",
    S3: "S3",
    Snowflake: "Snowflake",
    AzureSqlDb: "AzureSqlDb",
    AzureSynapseAnalytics: "AzureSynapseAnalytics",
    AzureMySqlDb: "AzureMySqlDb",
    AzurePostgresDb: "AzurePostgresDb",
    ADLSGen2: "ADLSGen2",
    Redis: "Redis",
    ApiKey: "ApiKey",
    AzureOpenAI: "AzureOpenAI",
    CognitiveSearch: "CognitiveSearch",
    CognitiveService: "CognitiveService",
    CustomKeys: "CustomKeys",
    AzureBlob: "AzureBlob",
    AzureOneLake: "AzureOneLake",
    CosmosDb: "CosmosDb",
    CosmosDbMongoDbApi: "CosmosDbMongoDbApi",
    AzureDataExplorer: "AzureDataExplorer",
    AzureMariaDb: "AzureMariaDb",
    AzureDatabricksDeltaLake: "AzureDatabricksDeltaLake",
    AzureSqlMi: "AzureSqlMi",
    AzureTableStorage: "AzureTableStorage",
    AmazonRdsForOracle: "AmazonRdsForOracle",
    AmazonRdsForSqlServer: "AmazonRdsForSqlServer",
    AmazonRedshift: "AmazonRedshift",
    Db2: "Db2",
    Drill: "Drill",
    GoogleBigQuery: "GoogleBigQuery",
    Greenplum: "Greenplum",
    Hbase: "Hbase",
    Hive: "Hive",
    Impala: "Impala",
    Informix: "Informix",
    MariaDb: "MariaDb",
    MicrosoftAccess: "MicrosoftAccess",
    MySql: "MySql",
    Netezza: "Netezza",
    Oracle: "Oracle",
    Phoenix: "Phoenix",
    PostgreSql: "PostgreSql",
    Presto: "Presto",
    SapOpenHub: "SapOpenHub",
    SapBw: "SapBw",
    SapHana: "SapHana",
    SapTable: "SapTable",
    Spark: "Spark",
    SqlServer: "SqlServer",
    Sybase: "Sybase",
    Teradata: "Teradata",
    Vertica: "Vertica",
    Cassandra: "Cassandra",
    Couchbase: "Couchbase",
    MongoDbV2: "MongoDbV2",
    MongoDbAtlas: "MongoDbAtlas",
    AmazonS3Compatible: "AmazonS3Compatible",
    FileServer: "FileServer",
    FtpServer: "FtpServer",
    GoogleCloudStorage: "GoogleCloudStorage",
    Hdfs: "Hdfs",
    OracleCloudStorage: "OracleCloudStorage",
    Sftp: "Sftp",
    GenericHttp: "GenericHttp",
    ODataRest: "ODataRest",
    Odbc: "Odbc",
    GenericRest: "GenericRest",
    AmazonMws: "AmazonMws",
    Concur: "Concur",
    Dynamics: "Dynamics",
    DynamicsAx: "DynamicsAx",
    DynamicsCrm: "DynamicsCrm",
    GoogleAdWords: "GoogleAdWords",
    Hubspot: "Hubspot",
    Jira: "Jira",
    Magento: "Magento",
    Marketo: "Marketo",
    Office365: "Office365",
    Eloqua: "Eloqua",
    Responsys: "Responsys",
    OracleServiceCloud: "OracleServiceCloud",
    PayPal: "PayPal",
    QuickBooks: "QuickBooks",
    Salesforce: "Salesforce",
    SalesforceServiceCloud: "SalesforceServiceCloud",
    SalesforceMarketingCloud: "SalesforceMarketingCloud",
    SapCloudForCustomer: "SapCloudForCustomer",
    SapEcc: "SapEcc",
    ServiceNow: "ServiceNow",
    SharePointOnlineList: "SharePointOnlineList",
    Shopify: "Shopify",
    Square: "Square",
    WebTable: "WebTable",
    Xero: "Xero",
    Zoho: "Zoho",
    GenericContainerRegistry: "GenericContainerRegistry",
};
exports.ContainerType = {
    /**
     * The container used to download models and score script.
     */
    StorageInitializer: "StorageInitializer",
    /**
     * The container used to serve user's request.
     */
    InferenceServer: "InferenceServer",
    /**
     * The container used to collect payload and custom logging when mdc is enabled.
     */
    ModelDataCollector: "ModelDataCollector",
};
exports.CredentialsType = {
    AccountKey: "AccountKey",
    Certificate: "Certificate",
    None: "None",
    Sas: "Sas",
    ServicePrincipal: "ServicePrincipal",
    KerberosKeytab: "KerberosKeytab",
    KerberosPassword: "KerberosPassword",
};
exports.DataCollectionMode = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DataImportSourceType = {
    Database: "database",
    File_system: "file_system",
};
exports.DataType = {
    Uri_file: "uri_file",
    Uri_folder: "uri_folder",
    Mltable: "mltable",
};
exports.DatastoreType = {
    AzureBlob: "AzureBlob",
    AzureDataLakeGen1: "AzureDataLakeGen1",
    AzureDataLakeGen2: "AzureDataLakeGen2",
    AzureFile: "AzureFile",
    Hdfs: "Hdfs",
    OneLake: "OneLake",
};
exports.DeploymentModelVersionUpgradeOption = {
    OnceNewDefaultVersionAvailable: "OnceNewDefaultVersionAvailable",
    OnceCurrentVersionExpired: "OnceCurrentVersionExpired",
    NoAutoUpgrade: "NoAutoUpgrade",
};
exports.DistributionType = {
    PyTorch: "PyTorch",
    TensorFlow: "TensorFlow",
    Mpi: "Mpi",
    Ray: "Ray",
};
exports.EarlyTerminationPolicyType = {
    Bandit: "Bandit",
    MedianStopping: "MedianStopping",
    TruncationSelection: "TruncationSelection",
};
exports.EgressPublicNetworkAccessType = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EmailNotificationEnableType = {
    JobCompleted: "JobCompleted",
    JobFailed: "JobFailed",
    JobCancelled: "JobCancelled",
};
exports.EncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EndpointAuthMode = {
    AMLToken: "AMLToken",
    Key: "Key",
    AADToken: "AADToken",
};
exports.EndpointComputeType = {
    Managed: "Managed",
    Kubernetes: "Kubernetes",
    AzureMLCompute: "AzureMLCompute",
};
exports.EndpointServiceConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
    Timeout: "Timeout",
};
exports.EnvironmentVariableType = {
    Local: "local",
};
exports.FeatureAttributionMetric = {
    /**
     * The Normalized Discounted Cumulative Gain metric.
     */
    NormalizedDiscountedCumulativeGain: "NormalizedDiscountedCumulativeGain",
};
exports.FeatureDataType = {
    String: "String",
    Integer: "Integer",
    Long: "Long",
    Float: "Float",
    Double: "Double",
    Binary: "Binary",
    Datetime: "Datetime",
    Boolean: "Boolean",
};
exports.FeatureImportanceMode = {
    /**
     * Disables computing feature importance within a signal.
     */
    Disabled: "Disabled",
    /**
     * Enables computing feature importance within a signal.
     */
    Enabled: "Enabled",
};
exports.FeatureLags = {
    /**
     * No feature lags generated.
     */
    None: "None",
    /**
     * System auto-generates feature lags.
     */
    Auto: "Auto",
};
exports.FeaturizationMode = {
    /**
     * Auto mode, system performs featurization without any custom featurization inputs.
     */
    Auto: "Auto",
    /**
     * Custom featurization.
     */
    Custom: "Custom",
    /**
     * Featurization off. 'Forecasting' task cannot use this value.
     */
    Off: "Off",
};
exports.FineTuningTaskType = {
    ChatCompletion: "ChatCompletion",
    TextCompletion: "TextCompletion",
    TextClassification: "TextClassification",
    QuestionAnswering: "QuestionAnswering",
    TextSummarization: "TextSummarization",
    TokenClassification: "TokenClassification",
    TextTranslation: "TextTranslation",
    ImageClassification: "ImageClassification",
    ImageInstanceSegmentation: "ImageInstanceSegmentation",
    ImageObjectDetection: "ImageObjectDetection",
    VideoMultiObjectTracking: "VideoMultiObjectTracking",
};
exports.ForecastHorizonMode = {
    /**
     * Forecast horizon to be determined automatically.
     */
    Auto: "Auto",
    /**
     * Use the custom forecast horizon.
     */
    Custom: "Custom",
};
exports.ForecastingModels = {
    /**
     * Auto-Autoregressive Integrated Moving Average (ARIMA) model uses time-series data and statistical analysis to interpret the data and make future predictions.
     * This model aims to explain data by using time series data on its past values and uses linear regression to make predictions.
     */
    AutoArima: "AutoArima",
    /**
     * Prophet is a procedure for forecasting time series data based on an additive model where non-linear trends are fit with yearly, weekly, and daily seasonality, plus holiday effects.
     * It works best with time series that have strong seasonal effects and several seasons of historical data. Prophet is robust to missing data and shifts in the trend, and typically handles outliers well.
     */
    Prophet: "Prophet",
    /**
     * The Naive forecasting model makes predictions by carrying forward the latest target value for each time-series in the training data.
     */
    Naive: "Naive",
    /**
     * The Seasonal Naive forecasting model makes predictions by carrying forward the latest season of target values for each time-series in the training data.
     */
    SeasonalNaive: "SeasonalNaive",
    /**
     * The Average forecasting model makes predictions by carrying forward the average of the target values for each time-series in the training data.
     */
    Average: "Average",
    /**
     * The Seasonal Average forecasting model makes predictions by carrying forward the average value of the latest season of data for each time-series in the training data.
     */
    SeasonalAverage: "SeasonalAverage",
    /**
     * Exponential smoothing is a time series forecasting method for univariate data that can be extended to support data with a systematic trend or seasonal component.
     */
    ExponentialSmoothing: "ExponentialSmoothing",
    /**
     * An Autoregressive Integrated Moving Average with Explanatory Variable (ARIMAX) model can be viewed as a multiple regression model with one or more autoregressive (AR) terms and/or one or more moving average (MA) terms.
     * This method is suitable for forecasting when data is stationary/non stationary, and multivariate with any type of data pattern, i.e., level/trend /seasonality/cyclicity.
     */
    Arimax: "Arimax",
    /**
     * TCNForecaster: Temporal Convolutional Networks Forecaster. //TODO: Ask forecasting team for brief intro.
     */
    TCNForecaster: "TCNForecaster",
    /**
     * Elastic net is a popular type of regularized linear regression that combines two popular penalties, specifically the L1 and L2 penalty functions.
     */
    ElasticNet: "ElasticNet",
    /**
     * The technique of transiting week learners into a strong learner is called Boosting. The gradient boosting algorithm process works on this theory of execution.
     */
    GradientBoosting: "GradientBoosting",
    /**
     * Decision Trees are a non-parametric supervised learning method used for both classification and regression tasks.
     * The goal is to create a model that predicts the value of a target variable by learning simple decision rules inferred from the data features.
     */
    DecisionTree: "DecisionTree",
    /**
     * K-nearest neighbors (KNN) algorithm uses 'feature similarity' to predict the values of new datapoints
     * which further means that the new data point will be assigned a value based on how closely it matches the points in the training set.
     */
    KNN: "KNN",
    /**
     * Lasso model fit with Least Angle Regression a.k.a. Lars. It is a Linear Model trained with an L1 prior as regularizer.
     */
    LassoLars: "LassoLars",
    /**
     * SGD: Stochastic gradient descent is an optimization algorithm often used in machine learning applications
     * to find the model parameters that correspond to the best fit between predicted and actual outputs.
     * It's an inexact but powerful technique.
     */
    SGD: "SGD",
    /**
     * Random forest is a supervised learning algorithm.
     * The "forest" it builds, is an ensemble of decision trees, usually trained with the bagging method.
     * The general idea of the bagging method is that a combination of learning models increases the overall result.
     */
    RandomForest: "RandomForest",
    /**
     * Extreme Trees is an ensemble machine learning algorithm that combines the predictions from many decision trees. It is related to the widely used random forest algorithm.
     */
    ExtremeRandomTrees: "ExtremeRandomTrees",
    /**
     * LightGBM is a gradient boosting framework that uses tree based learning algorithms.
     */
    LightGBM: "LightGBM",
    /**
     * XGBoostRegressor: Extreme Gradient Boosting Regressor is a supervised machine learning model using ensemble of base learners.
     */
    XGBoostRegressor: "XGBoostRegressor",
};
exports.ForecastingPrimaryMetrics = {
    /**
     * The Spearman's rank coefficient of correlation is a non-parametric measure of rank correlation.
     */
    SpearmanCorrelation: "SpearmanCorrelation",
    /**
     * The Normalized Root Mean Squared Error (NRMSE) the RMSE facilitates the comparison between models with different scales.
     */
    NormalizedRootMeanSquaredError: "NormalizedRootMeanSquaredError",
    /**
     * The R2 score is one of the performance evaluation measures for forecasting-based machine learning models.
     */
    R2Score: "R2Score",
    /**
     * The Normalized Mean Absolute Error (NMAE) is a validation metric to compare the Mean Absolute Error (MAE) of (time) series with different scales.
     */
    NormalizedMeanAbsoluteError: "NormalizedMeanAbsoluteError",
};
exports.GenerationSafetyQualityMetric = {
    AcceptableGroundednessScorePerInstance: "AcceptableGroundednessScorePerInstance",
    AggregatedGroundednessPassRate: "AggregatedGroundednessPassRate",
    AcceptableCoherenceScorePerInstance: "AcceptableCoherenceScorePerInstance",
    AggregatedCoherencePassRate: "AggregatedCoherencePassRate",
    AcceptableFluencyScorePerInstance: "AcceptableFluencyScorePerInstance",
    AggregatedFluencyPassRate: "AggregatedFluencyPassRate",
    AcceptableSimilarityScorePerInstance: "AcceptableSimilarityScorePerInstance",
    AggregatedSimilarityPassRate: "AggregatedSimilarityPassRate",
    AcceptableRelevanceScorePerInstance: "AcceptableRelevanceScorePerInstance",
    AggregatedRelevancePassRate: "AggregatedRelevancePassRate",
};
exports.GenerationTokenUsageMetric = {
    TotalTokenCount: "TotalTokenCount",
    TotalTokenCountPerGroup: "TotalTokenCountPerGroup",
};
exports.Goal = {
    Minimize: "Minimize",
    Maximize: "Maximize",
};
exports.IdentityConfigurationType = {
    Managed: "Managed",
    AMLToken: "AMLToken",
    UserIdentity: "UserIdentity",
};
exports.ImageAnnotationType = {
    Classification: "Classification",
    BoundingBox: "BoundingBox",
    InstanceSegmentation: "InstanceSegmentation",
};
exports.ImageType = {
    Docker: "docker",
    Azureml: "azureml",
};
exports.IncrementalDataRefresh = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.InputDeliveryMode = {
    ReadOnlyMount: "ReadOnlyMount",
    ReadWriteMount: "ReadWriteMount",
    Download: "Download",
    Direct: "Direct",
    EvalMount: "EvalMount",
    EvalDownload: "EvalDownload",
};
exports.InstanceSegmentationPrimaryMetrics = {
    /**
     * Mean Average Precision (MAP) is the average of AP (Average Precision).
     * AP is calculated for each class and averaged to get the MAP.
     */
    MeanAveragePrecision: "MeanAveragePrecision",
};
exports.IsolationMode = {
    Disabled: "Disabled",
    AllowInternetOutbound: "AllowInternetOutbound",
    AllowOnlyApprovedOutbound: "AllowOnlyApprovedOutbound",
};
exports.JobInputType = {
    Literal: "literal",
    Uri_file: "uri_file",
    Uri_folder: "uri_folder",
    Mltable: "mltable",
    Custom_model: "custom_model",
    Mlflow_model: "mlflow_model",
    Triton_model: "triton_model",
};
exports.JobLimitsType = {
    Command: "Command",
    Sweep: "Sweep",
};
exports.JobOutputType = {
    Uri_file: "uri_file",
    Uri_folder: "uri_folder",
    Mltable: "mltable",
    Custom_model: "custom_model",
    Mlflow_model: "mlflow_model",
    Triton_model: "triton_model",
};
exports.JobTier = {
    Null: "Null",
    Spot: "Spot",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.JobType = {
    AutoML: "AutoML",
    Command: "Command",
    Labeling: "Labeling",
    Sweep: "Sweep",
    Pipeline: "Pipeline",
    Spark: "Spark",
    FineTuning: "FineTuning",
};
exports.LearningRateScheduler = {
    /**
     * No learning rate scheduler selected.
     */
    None: "None",
    /**
     * Cosine Annealing With Warmup.
     */
    WarmupCosine: "WarmupCosine",
    /**
     * Step learning rate scheduler.
     */
    Step: "Step",
};
exports.LoadBalancerType = {
    PublicIp: "PublicIp",
    InternalLoadBalancer: "InternalLoadBalancer",
};
exports.LogTrainingMetrics = {
    /**
     * Enable compute and log training metrics.
     */
    Enable: "Enable",
    /**
     * Disable compute and log training metrics.
     */
    Disable: "Disable",
};
exports.LogValidationLoss = {
    /**
     * Enable compute and log validation metrics.
     */
    Enable: "Enable",
    /**
     * Disable compute and log validation metrics.
     */
    Disable: "Disable",
};
exports.LogVerbosity = {
    /**
     * No logs emitted.
     */
    NotSet: "NotSet",
    /**
     * Debug and above log statements logged.
     */
    Debug: "Debug",
    /**
     * Info and above log statements logged.
     */
    Info: "Info",
    /**
     * Warning and above log statements logged.
     */
    Warning: "Warning",
    /**
     * Error and above log statements logged.
     */
    Error: "Error",
    /**
     * Only critical statements logged.
     */
    Critical: "Critical",
};
exports.MLAssistConfigurationType = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MLFlowAutologgerState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ManagedNetworkStatus = {
    Inactive: "Inactive",
    Active: "Active",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.MaterializationStoreType = {
    None: "None",
    Online: "Online",
    Offline: "Offline",
    OnlineAndOffline: "OnlineAndOffline",
};
exports.MediaType = {
    Image: "Image",
    Text: "Text",
};
exports.MlflowAutologger = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ModelProvider = {
    /**
     * Fine tuning using Azure Open AI model.
     */
    AzureOpenAI: "AzureOpenAI",
    /**
     * Fine tuning using custom model.
     */
    Custom: "Custom",
};
exports.ModelSize = {
    /**
     * No value selected.
     */
    None: "None",
    /**
     * Small size.
     */
    Small: "Small",
    /**
     * Medium size.
     */
    Medium: "Medium",
    /**
     * Large size.
     */
    Large: "Large",
    /**
     * Extra large size.
     */
    ExtraLarge: "ExtraLarge",
};
exports.ModelTaskType = {
    Classification: "Classification",
    Regression: "Regression",
    QuestionAnswering: "QuestionAnswering",
};
exports.MonitorComputeIdentityType = {
    /**
     * Authenticates through user's AML token.
     */
    AmlToken: "AmlToken",
    /**
     * Authenticates through a user-provided managed identity.
     */
    ManagedIdentity: "ManagedIdentity",
};
exports.MonitorComputeType = {
    /**
     * Serverless Spark compute.
     */
    ServerlessSpark: "ServerlessSpark",
};
exports.MonitoringFeatureDataType = {
    /**
     * Used for features of numerical data type.
     */
    Numerical: "Numerical",
    /**
     * Used for features of categorical data type.
     */
    Categorical: "Categorical",
};
exports.MonitoringFeatureFilterType = {
    /**
     * Includes all features.
     */
    AllFeatures: "AllFeatures",
    /**
     * Only includes the top contributing features, measured by feature attribution.
     */
    TopNByAttribution: "TopNByAttribution",
    /**
     * Includes a user-defined subset of features.
     */
    FeatureSubset: "FeatureSubset",
};
exports.MonitoringInputDataType = {
    /**
     * An input data with a fixed window size.
     */
    Static: "Static",
    /**
     * An input data which rolls relatively to the monitor's current run time.
     */
    Rolling: "Rolling",
    /**
     * An input data with tabular format which doesn't require preprocessing.
     */
    Fixed: "Fixed",
};
exports.MonitoringModelType = {
    /**
     * A model trained for classification tasks.
     */
    Classification: "Classification",
    /**
     * A model trained for regressions tasks.
     */
    Regression: "Regression",
};
exports.MonitoringNotificationType = {
    /**
     * Enables email notifications through AML notifications.
     */
    AmlNotification: "AmlNotification",
    /**
     * Enables notifications through Azure Monitor by posting metrics to the workspace's Azure Monitor instance.
     */
    AzureMonitor: "AzureMonitor",
};
exports.MonitoringSignalType = {
    /**
     * Tracks model input data distribution change, comparing against training data or past production data.
     */
    DataDrift: "DataDrift",
    /**
     * Tracks prediction result data distribution change, comparing against validation/test label data or past production data.
     */
    PredictionDrift: "PredictionDrift",
    /**
     * Tracks model input data integrity.
     */
    DataQuality: "DataQuality",
    /**
     * Tracks feature importance change in production, comparing against feature importance at training time.
     */
    FeatureAttributionDrift: "FeatureAttributionDrift",
    /**
     * Tracks a custom signal provided by users.
     */
    Custom: "Custom",
    /**
     * Tracks model performance based on ground truth data.
     */
    ModelPerformance: "ModelPerformance",
    /**
     * Tracks the safety and quality of generated content.
     */
    GenerationSafetyQuality: "GenerationSafetyQuality",
    /**
     * Tracks the token usage of generative endpoints.
     */
    GenerationTokenStatistics: "GenerationTokenStatistics",
};
exports.MultiSelect = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.NCrossValidationsMode = {
    /**
     * Determine N-Cross validations value automatically. Supported only for 'Forecasting' AutoML task.
     */
    Auto: "Auto",
    /**
     * Use custom N-Cross validations value.
     */
    Custom: "Custom",
};
exports.NlpLearningRateScheduler = {
    /**
     * No learning rate schedule.
     */
    None: "None",
    /**
     * Linear warmup and decay.
     */
    Linear: "Linear",
    /**
     * Linear warmup then cosine decay.
     */
    Cosine: "Cosine",
    /**
     * Linear warmup, cosine decay, then restart to initial LR.
     */
    CosineWithRestarts: "CosineWithRestarts",
    /**
     * Increase linearly then polynomially decay.
     */
    Polynomial: "Polynomial",
    /**
     * Constant learning rate.
     */
    Constant: "Constant",
    /**
     * Linear warmup followed by constant value.
     */
    ConstantWithWarmup: "ConstantWithWarmup",
};
exports.NodesValueType = {
    All: "All",
    Custom: "Custom",
};
exports.NumericalDataDriftMetric = {
    /**
     * The Jensen Shannon Distance (JSD) metric.
     */
    JensenShannonDistance: "JensenShannonDistance",
    /**
     * The Population Stability Index (PSI) metric.
     */
    PopulationStabilityIndex: "PopulationStabilityIndex",
    /**
     * The Normalized Wasserstein Distance metric.
     */
    NormalizedWassersteinDistance: "NormalizedWassersteinDistance",
    /**
     * The Two Sample Kolmogorov-Smirnov Test (two-sample K–S) metric.
     */
    TwoSampleKolmogorovSmirnovTest: "TwoSampleKolmogorovSmirnovTest",
};
exports.NumericalDataQualityMetric = {
    /**
     * Calculates the rate of null values.
     */
    NullValueRate: "NullValueRate",
    /**
     * Calculates the rate of data type errors.
     */
    DataTypeErrorRate: "DataTypeErrorRate",
    /**
     * Calculates the rate values are out of bounds.
     */
    OutOfBoundsRate: "OutOfBoundsRate",
};
exports.NumericalPredictionDriftMetric = {
    /**
     * The Jensen Shannon Distance (JSD) metric.
     */
    JensenShannonDistance: "JensenShannonDistance",
    /**
     * The Population Stability Index (PSI) metric.
     */
    PopulationStabilityIndex: "PopulationStabilityIndex",
    /**
     * The Normalized Wasserstein Distance metric.
     */
    NormalizedWassersteinDistance: "NormalizedWassersteinDistance",
    /**
     * The Two Sample Kolmogorov-Smirnov Test (two-sample K–S) metric.
     */
    TwoSampleKolmogorovSmirnovTest: "TwoSampleKolmogorovSmirnovTest",
};
exports.ObjectDetectionPrimaryMetrics = {
    /**
     * Mean Average Precision (MAP) is the average of AP (Average Precision).
     * AP is calculated for each class and averaged to get the MAP.
     */
    MeanAveragePrecision: "MeanAveragePrecision",
};
exports.OneLakeArtifactType = {
    LakeHouse: "LakeHouse",
};
exports.OperatingSystemType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OsType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OutputDeliveryMode = {
    ReadWriteMount: "ReadWriteMount",
    Upload: "Upload",
    Direct: "Direct",
};
exports.PrivateEndpointConnectionProvisioningState = {
    Succeeded: "Succeeded",
    Creating: "Creating",
    Deleting: "Deleting",
    Failed: "Failed",
};
exports.ProtectionLevel = {
    /**
     * All means Intellectual Property is fully protected.
     */
    All: "All",
    /**
     * None means it is not an Intellectual Property.
     */
    None: "None",
};
exports.Protocol = {
    Tcp: "tcp",
    Udp: "udp",
    Http: "http",
};
exports.PublicNetworkAccessType = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RandomSamplingAlgorithmRule = {
    Random: "Random",
    Sobol: "Sobol",
};
exports.RecurrenceFrequency = {
    /**
     * Minute frequency
     */
    Minute: "Minute",
    /**
     * Hour frequency
     */
    Hour: "Hour",
    /**
     * Day frequency
     */
    Day: "Day",
    /**
     * Week frequency
     */
    Week: "Week",
    /**
     * Month frequency
     */
    Month: "Month",
};
exports.ReferenceType = {
    Id: "Id",
    DataPath: "DataPath",
    OutputPath: "OutputPath",
};
exports.RegressionModelPerformanceMetric = {
    /**
     * The Mean Absolute Error (MAE) metric.
     */
    MeanAbsoluteError: "MeanAbsoluteError",
    /**
     * The Root Mean Squared Error (RMSE) metric.
     */
    RootMeanSquaredError: "RootMeanSquaredError",
    /**
     * The Mean Squared Error (MSE) metric.
     */
    MeanSquaredError: "MeanSquaredError",
};
exports.RegressionModels = {
    /**
     * Elastic net is a popular type of regularized linear regression that combines two popular penalties, specifically the L1 and L2 penalty functions.
     */
    ElasticNet: "ElasticNet",
    /**
     * The technique of transiting week learners into a strong learner is called Boosting. The gradient boosting algorithm process works on this theory of execution.
     */
    GradientBoosting: "GradientBoosting",
    /**
     * Decision Trees are a non-parametric supervised learning method used for both classification and regression tasks.
     * The goal is to create a model that predicts the value of a target variable by learning simple decision rules inferred from the data features.
     */
    DecisionTree: "DecisionTree",
    /**
     * K-nearest neighbors (KNN) algorithm uses 'feature similarity' to predict the values of new datapoints
     * which further means that the new data point will be assigned a value based on how closely it matches the points in the training set.
     */
    KNN: "KNN",
    /**
     * Lasso model fit with Least Angle Regression a.k.a. Lars. It is a Linear Model trained with an L1 prior as regularizer.
     */
    LassoLars: "LassoLars",
    /**
     * SGD: Stochastic gradient descent is an optimization algorithm often used in machine learning applications
     * to find the model parameters that correspond to the best fit between predicted and actual outputs.
     * It's an inexact but powerful technique.
     */
    SGD: "SGD",
    /**
     * Random forest is a supervised learning algorithm.
     * The "forest" it builds, is an ensemble of decision trees, usually trained with the bagging method.
     * The general idea of the bagging method is that a combination of learning models increases the overall result.
     */
    RandomForest: "RandomForest",
    /**
     * Extreme Trees is an ensemble machine learning algorithm that combines the predictions from many decision trees. It is related to the widely used random forest algorithm.
     */
    ExtremeRandomTrees: "ExtremeRandomTrees",
    /**
     * LightGBM is a gradient boosting framework that uses tree based learning algorithms.
     */
    LightGBM: "LightGBM",
    /**
     * XGBoostRegressor: Extreme Gradient Boosting Regressor is a supervised machine learning model using ensemble of base learners.
     */
    XGBoostRegressor: "XGBoostRegressor",
};
exports.RegressionPrimaryMetrics = {
    /**
     * The Spearman's rank coefficient of correlation is a nonparametric measure of rank correlation.
     */
    SpearmanCorrelation: "SpearmanCorrelation",
    /**
     * The Normalized Root Mean Squared Error (NRMSE) the RMSE facilitates the comparison between models with different scales.
     */
    NormalizedRootMeanSquaredError: "NormalizedRootMeanSquaredError",
    /**
     * The R2 score is one of the performance evaluation measures for forecasting-based machine learning models.
     */
    R2Score: "R2Score",
    /**
     * The Normalized Mean Absolute Error (NMAE) is a validation metric to compare the Mean Absolute Error (MAE) of (time) series with different scales.
     */
    NormalizedMeanAbsoluteError: "NormalizedMeanAbsoluteError",
};
exports.RemoteLoginPortPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    NotSpecified: "NotSpecified",
};
exports.RollingRateType = {
    Year: "Year",
    Month: "Month",
    Day: "Day",
    Hour: "Hour",
    Minute: "Minute",
};
exports.RuleAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.RuleCategory = {
    Required: "Required",
    Recommended: "Recommended",
    UserDefined: "UserDefined",
};
exports.RuleStatus = {
    Inactive: "Inactive",
    Active: "Active",
};
exports.RuleType = {
    FQDN: "FQDN",
    PrivateEndpoint: "PrivateEndpoint",
    ServiceTag: "ServiceTag",
};
exports.SamplingAlgorithmType = {
    Grid: "Grid",
    Random: "Random",
    Bayesian: "Bayesian",
};
exports.ScaleType = {
    Default: "Default",
    TargetUtilization: "TargetUtilization",
};
exports.ScheduleActionType = {
    CreateJob: "CreateJob",
    InvokeBatchEndpoint: "InvokeBatchEndpoint",
    ImportData: "ImportData",
    CreateMonitor: "CreateMonitor",
};
exports.ScheduleProvisioningState = {
    Completed: "Completed",
    Provisioning: "Provisioning",
    Failed: "Failed",
};
exports.ScheduleStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SeasonalityMode = {
    /**
     * Seasonality to be determined automatically.
     */
    Auto: "Auto",
    /**
     * Use the custom seasonality value.
     */
    Custom: "Custom",
};
exports.SecretsType = {
    AccountKey: "AccountKey",
    Certificate: "Certificate",
    Sas: "Sas",
    ServicePrincipal: "ServicePrincipal",
    KerberosPassword: "KerberosPassword",
    KerberosKeytab: "KerberosKeytab",
};
exports.ServerlessInferenceEndpointAuthMode = {
    Key: "Key",
    AAD: "AAD",
};
exports.ServiceDataAccessAuthIdentity = {
    /**
     * Do not use any identity for service data access.
     */
    None: "None",
    /**
     * Use the system assigned managed identity of the Workspace to authenticate service data access.
     */
    WorkspaceSystemAssignedIdentity: "WorkspaceSystemAssignedIdentity",
    /**
     * Use the user assigned managed identity of the Workspace to authenticate service data access.
     */
    WorkspaceUserAssignedIdentity: "WorkspaceUserAssignedIdentity",
};
exports.ShortSeriesHandlingConfiguration = {
    /**
     * Represents no/null value.
     */
    None: "None",
    /**
     * Short series will be padded if there are no long series, otherwise short series will be dropped.
     */
    Auto: "Auto",
    /**
     * All the short series will be padded.
     */
    Pad: "Pad",
    /**
     * All the short series will be dropped.
     */
    Drop: "Drop",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SparkJobEntryType = {
    SparkJobPythonEntry: "SparkJobPythonEntry",
    SparkJobScalaEntry: "SparkJobScalaEntry",
};
exports.SshPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SslConfigStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
    Auto: "Auto",
};
exports.StackMetaLearnerType = {
    None: "None",
    /**
     * Default meta-learners are LogisticRegression for classification tasks.
     */
    LogisticRegression: "LogisticRegression",
    /**
     * Default meta-learners are LogisticRegression for classification task when CV is on.
     */
    LogisticRegressionCV: "LogisticRegressionCV",
    LightGBMClassifier: "LightGBMClassifier",
    /**
     * Default meta-learners are LogisticRegression for regression task.
     */
    ElasticNet: "ElasticNet",
    /**
     * Default meta-learners are LogisticRegression for regression task when CV is on.
     */
    ElasticNetCV: "ElasticNetCV",
    LightGBMRegressor: "LightGBMRegressor",
    LinearRegression: "LinearRegression",
};
exports.StochasticOptimizer = {
    /**
     * No optimizer selected.
     */
    None: "None",
    /**
     * Stochastic Gradient Descent optimizer.
     */
    Sgd: "Sgd",
    /**
     * Adam is algorithm the optimizes stochastic objective functions based on adaptive estimates of moments
     */
    Adam: "Adam",
    /**
     * AdamW is a variant of the optimizer Adam that has an improved implementation of weight decay.
     */
    Adamw: "Adamw",
};
exports.TargetAggregationFunction = {
    /**
     * Represent no value set.
     */
    None: "None",
    Sum: "Sum",
    Max: "Max",
    Min: "Min",
    Mean: "Mean",
};
exports.TargetLagsMode = {
    /**
     * Target lags to be determined automatically.
     */
    Auto: "Auto",
    /**
     * Use the custom target lags.
     */
    Custom: "Custom",
};
exports.TargetRollingWindowSizeMode = {
    /**
     * Determine rolling windows size automatically.
     */
    Auto: "Auto",
    /**
     * Use the specified rolling window size.
     */
    Custom: "Custom",
};
exports.TaskType = {
    /**
     * Classification in machine learning and statistics is a supervised learning approach in which
     * the computer program learns from the data given to it and make new observations or classifications.
     */
    Classification: "Classification",
    /**
     * Regression means to predict the value using the input data. Regression models are used to predict a continuous value.
     */
    Regression: "Regression",
    /**
     * Forecasting is a special kind of regression task that deals with time-series data and creates forecasting model
     * that can be used to predict the near future values based on the inputs.
     */
    Forecasting: "Forecasting",
    /**
     * Image Classification. Multi-class image classification is used when an image is classified with only a single label
     * from a set of classes - e.g. each image is classified as either an image of a 'cat' or a 'dog' or a 'duck'.
     */
    ImageClassification: "ImageClassification",
    /**
     * Image Classification Multilabel. Multi-label image classification is used when an image could have one or more labels
     * from a set of labels - e.g. an image could be labeled with both 'cat' and 'dog'.
     */
    ImageClassificationMultilabel: "ImageClassificationMultilabel",
    /**
     * Image Object Detection. Object detection is used to identify objects in an image and locate each object with a
     * bounding box e.g. locate all dogs and cats in an image and draw a bounding box around each.
     */
    ImageObjectDetection: "ImageObjectDetection",
    /**
     * Image Instance Segmentation. Instance segmentation is used to identify objects in an image at the pixel level,
     * drawing a polygon around each object in the image.
     */
    ImageInstanceSegmentation: "ImageInstanceSegmentation",
    /**
     * Text classification (also known as text tagging or text categorization) is the process of sorting texts into categories.
     * Categories are mutually exclusive.
     */
    TextClassification: "TextClassification",
    /**
     * Multilabel classification task assigns each sample to a group (zero or more) of target labels.
     */
    TextClassificationMultilabel: "TextClassificationMultilabel",
    /**
     * Text Named Entity Recognition a.k.a. TextNER.
     * Named Entity Recognition (NER) is the ability to take free-form text and identify the occurrences of entities such as people, locations, organizations, and more.
     */
    TextNER: "TextNER",
};
exports.TextAnnotationType = {
    Classification: "Classification",
    NamedEntityRecognition: "NamedEntityRecognition",
};
exports.TrainingMode = {
    /**
     * Auto mode
     */
    Auto: "Auto",
    /**
     * Distributed training mode
     */
    Distributed: "Distributed",
    /**
     * Non distributed training mode
     */
    NonDistributed: "NonDistributed",
};
exports.TriggerType = {
    Recurrence: "Recurrence",
    Cron: "Cron",
};
exports.UseStl = {
    /**
     * No stl decomposition.
     */
    None: "None",
    Season: "Season",
    SeasonTrend: "SeasonTrend",
};
exports.ValidationMetricType = {
    /**
     * No metric.
     */
    None: "None",
    /**
     * Coco metric.
     */
    Coco: "Coco",
    /**
     * Voc metric.
     */
    Voc: "Voc",
    /**
     * CocoVoc metric.
     */
    CocoVoc: "CocoVoc",
};
exports.VmPriority = {
    Dedicated: "Dedicated",
    LowPriority: "LowPriority",
};
exports.VolumeDefinitionType = {
    Bind: "bind",
    Volume: "volume",
    Tmpfs: "tmpfs",
    Npipe: "npipe",
};
exports.WebhookType = {
    AzureDevOps: "AzureDevOps",
};
exports.WeekDay = {
    /**
     * Monday weekday
     */
    Monday: "Monday",
    /**
     * Tuesday weekday
     */
    Tuesday: "Tuesday",
    /**
     * Wednesday weekday
     */
    Wednesday: "Wednesday",
    /**
     * Thursday weekday
     */
    Thursday: "Thursday",
    /**
     * Friday weekday
     */
    Friday: "Friday",
    /**
     * Saturday weekday
     */
    Saturday: "Saturday",
    /**
     * Sunday weekday
     */
    Sunday: "Sunday",
};
