// @pulumi/azure-native/containerstorage/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230701preview = exports.Volume = exports.Snapshot = exports.Pool = exports.getVolumeOutput = exports.getVolume = exports.getSnapshotOutput = exports.getSnapshot = exports.getPoolOutput = exports.getPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getPool = null;
exports.getPoolOutput = null;
utilities.lazyLoad(exports, ["getPool", "getPoolOutput"], () => require("./getPool"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.Pool = null;
utilities.lazyLoad(exports, ["Pool"], () => require("./pool"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
// Export enums:
__exportStar(require("../types/enums/containerstorage"), exports);
// Export sub-modules:
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerstorage:Pool":
                return new exports.Pool(name, undefined, { urn });
            case "azure-native:containerstorage:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "azure-native:containerstorage:Volume":
                return new exports.Volume(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerstorage", _module);
