// @pulumi/azure-native/types/enums/servicefabric/v20241101preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZonalUpdateMode = exports.VmssExtensionSetupOrder = exports.VmSetupAction = exports.SkuName = exports.ServiceScalingTriggerKind = exports.ServiceScalingMechanismKind = exports.ServicePlacementPolicyType = exports.ServicePackageActivationMode = exports.ServiceLoadMetricWeight = exports.ServiceKind = exports.ServiceCorrelationScheme = exports.SecurityType = exports.RollingUpgradeMode = exports.PublicIPAddressVersion = exports.Protocol = exports.ProbeProtocol = exports.PrivateLinkServiceNetworkPolicies = exports.PrivateIPAddressVersion = exports.PrivateEndpointNetworkPolicies = exports.PartitionScheme = exports.NsgProtocol = exports.MoveCost = exports.ManagedIdentityType = exports.ManagedClusterAddOnFeature = exports.IPAddressType = exports.FailureAction = exports.EvictionPolicyType = exports.DiskType = exports.Direction = exports.ClusterUpgradeMode = exports.ClusterUpgradeCadence = exports.AutoGeneratedDomainNameLabelScope = exports.Access = void 0;
exports.Access = {
    /**
     * The network traffic is allowed.
     */
    Allow: "allow",
    /**
     * The network traffic is denied.
     */
    Deny: "deny",
};
exports.AutoGeneratedDomainNameLabelScope = {
    /**
     * TenantReuse allows for the same hash to be created if the resource is created in the same Tenant with the same resource name.
     */
    TenantReuse: "TenantReuse",
    /**
     * SubscriptionReuse allows for the same hash to be created if the resource is created in the same Subscription with the same resource name.
     */
    SubscriptionReuse: "SubscriptionReuse",
    /**
     * ResourceGroupReuse allows for the same hash to be created if the resource is created in the same Resource Group with the same resource name.
     */
    ResourceGroupReuse: "ResourceGroupReuse",
    /**
     * NoReuse will create a new hash regardless of the Subscription, Resource Group, Tenant and Resource name.
     */
    NoReuse: "NoReuse",
};
exports.ClusterUpgradeCadence = {
    /**
     * Cluster upgrade starts immediately after a new version is rolled out. Recommended for Test/Dev clusters.
     */
    Wave0: "Wave0",
    /**
     * Cluster upgrade starts 7 days after a new version is rolled out. Recommended for Pre-prod clusters.
     */
    Wave1: "Wave1",
    /**
     * Cluster upgrade starts 14 days after a new version is rolled out. Recommended for Production clusters.
     */
    Wave2: "Wave2",
};
exports.ClusterUpgradeMode = {
    /**
     * The cluster will be automatically upgraded to the latest Service Fabric runtime version, **clusterUpgradeCadence** will determine when the upgrade starts after the new version becomes available.
     */
    Automatic: "Automatic",
    /**
     * The cluster will not be automatically upgraded to the latest Service Fabric runtime version. The cluster is upgraded by setting the **clusterCodeVersion** property in the cluster resource.
     */
    Manual: "Manual",
};
exports.Direction = {
    /**
     * Inbound direction.
     */
    Inbound: "inbound",
    /**
     * Outbound direction.
     */
    Outbound: "outbound",
};
exports.DiskType = {
    /**
     * Standard HDD locally redundant storage. Best for backup, non-critical, and infrequent access.
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Standard SSD locally redundant storage. Best for web servers, lightly used enterprise applications and dev/test.
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * Premium SSD locally redundant storage. Best for production and performance sensitive workloads.
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Premium SSD V2 locally redundant storage. Best for production and performance sensitive workloads that consistently require low latency and high IOPS and throughput.
     */
    PremiumV2_LRS: "PremiumV2_LRS",
    /**
     * Standard SSD zone redundant storage. Best for web servers, lightly used enterprise applications and dev/test that need storage resiliency against zone failures.
     */
    StandardSSD_ZRS: "StandardSSD_ZRS",
    /**
     * Premium SSD zone redundant storage. Best for production workloads that need storage resiliency against zone failures.
     */
    Premium_ZRS: "Premium_ZRS",
};
exports.EvictionPolicyType = {
    /**
     * Eviction policy will be Delete for SPOT vms.
     */
    Delete: "Delete",
    /**
     * Eviction policy will be Deallocate for SPOT vms.
     */
    Deallocate: "Deallocate",
};
exports.FailureAction = {
    /**
     * Indicates that a rollback of the upgrade will be performed by Service Fabric if the upgrade fails.
     */
    Rollback: "Rollback",
    /**
     * Indicates that a manual repair will need to be performed by the administrator if the upgrade fails. Service Fabric will not proceed to the next upgrade domain automatically.
     */
    Manual: "Manual",
};
exports.IPAddressType = {
    /**
     * IPv4 address type.
     */
    IPv4: "IPv4",
    /**
     * IPv6 address type.
     */
    IPv6: "IPv6",
};
exports.ManagedClusterAddOnFeature = {
    /**
     * Dns service
     */
    DnsService: "DnsService",
    /**
     * Backup and restore service
     */
    BackupRestoreService: "BackupRestoreService",
    /**
     * Resource monitor service
     */
    ResourceMonitorService: "ResourceMonitorService",
};
exports.ManagedIdentityType = {
    /**
     * Indicates that no identity is associated with the resource.
     */
    None: "None",
    /**
     * Indicates that system assigned identity is associated with the resource.
     */
    SystemAssigned: "SystemAssigned",
    /**
     * Indicates that user assigned identity is associated with the resource.
     */
    UserAssigned: "UserAssigned",
    /**
     * Indicates that both system assigned and user assigned identity are associated with the resource.
     */
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.MoveCost = {
    /**
     * Zero move cost. This value is zero.
     */
    Zero: "Zero",
    /**
     * Specifies the move cost of the service as Low. The value is 1.
     */
    Low: "Low",
    /**
     * Specifies the move cost of the service as Medium. The value is 2.
     */
    Medium: "Medium",
    /**
     * Specifies the move cost of the service as High. The value is 3.
     */
    High: "High",
};
exports.NsgProtocol = {
    /**
     * Protocol applies to HTTP.
     */
    Http: "http",
    /**
     * Protocol applies to HTTPS.
     */
    Https: "https",
    /**
     * Protocol applies to TCP.
     */
    Tcp: "tcp",
    /**
     * Protocol applies to UDP.
     */
    Udp: "udp",
    /**
     * Protocol applies to ICMP.
     */
    Icmp: "icmp",
    /**
     * Protocol applies to AH.
     */
    Ah: "ah",
    /**
     * Protocol applies to ESP.
     */
    Esp: "esp",
};
exports.PartitionScheme = {
    /**
     * Indicates that the partition is based on string names, and is a SingletonPartitionScheme object, The value is 0.
     */
    Singleton: "Singleton",
    /**
     * Indicates that the partition is based on Int64 key ranges, and is a UniformInt64RangePartitionScheme object. The value is 1.
     */
    UniformInt64Range: "UniformInt64Range",
    /**
     * Indicates that the partition is based on string names, and is a NamedPartitionScheme object. The value is 2.
     */
    Named: "Named",
};
exports.PrivateEndpointNetworkPolicies = {
    /**
     * Enable apply network policies on private end point in the subnet.
     */
    Enabled: "enabled",
    /**
     * Disable apply network policies on private end point in the subnet.
     */
    Disabled: "disabled",
};
exports.PrivateIPAddressVersion = {
    /**
     * The IP configuration's private IP is IPv4.
     */
    IPv4: "IPv4",
    /**
     * The IP configuration's private IP is IPv6.
     */
    IPv6: "IPv6",
};
exports.PrivateLinkServiceNetworkPolicies = {
    /**
     * Enable apply network policies on private link service in the subnet.
     */
    Enabled: "enabled",
    /**
     * Disable apply network policies on private link service in the subnet.
     */
    Disabled: "disabled",
};
exports.ProbeProtocol = {
    /**
     * Probe protocol is TCP.
     */
    Tcp: "tcp",
    /**
     * Probe protocol is HTTP.
     */
    Http: "http",
    /**
     * Probe protocol is HTTPS.
     */
    Https: "https",
};
exports.Protocol = {
    /**
     * Transport protocol is TCP.
     */
    Tcp: "tcp",
    /**
     * Transport protocol is UDP.
     */
    Udp: "udp",
};
exports.PublicIPAddressVersion = {
    /**
     * The IP configuration's public IP is IPv4.
     */
    IPv4: "IPv4",
    /**
     * The IP configuration's public IP is IPv6.
     */
    IPv6: "IPv6",
};
exports.RollingUpgradeMode = {
    /**
     * The upgrade will stop after completing each upgrade domain and automatically monitor health before proceeding. The value is 0.
     */
    Monitored: "Monitored",
    /**
     * The upgrade will proceed automatically without performing any health monitoring. The value is 1.
     */
    UnmonitoredAuto: "UnmonitoredAuto",
};
exports.SecurityType = {
    /**
     * Trusted Launch is a security type that secures generation 2 virtual machines.
     */
    TrustedLaunch: "TrustedLaunch",
    /**
     * Standard is the default security type for all machines.
     */
    Standard: "Standard",
};
exports.ServiceCorrelationScheme = {
    /**
     * Aligned affinity ensures that the primaries of the partitions of the affinitized services are collocated on the same nodes. This is the default and is the same as selecting the Affinity scheme. The value is 0.
     */
    AlignedAffinity: "AlignedAffinity",
    /**
     * Non-Aligned affinity guarantees that all replicas of each service will be placed on the same nodes. Unlike Aligned Affinity, this does not guarantee that replicas of particular role will be collocated. The value is 1.
     */
    NonAlignedAffinity: "NonAlignedAffinity",
};
exports.ServiceKind = {
    /**
     * Does not use Service Fabric to make its state highly available or reliable. The value is 0.
     */
    Stateless: "Stateless",
    /**
     * Uses Service Fabric to make its state or part of its state highly available and reliable. The value is 1.
     */
    Stateful: "Stateful",
};
exports.ServiceLoadMetricWeight = {
    /**
     * Disables resource balancing for this metric. This value is zero.
     */
    Zero: "Zero",
    /**
     * Specifies the metric weight of the service load as Low. The value is 1.
     */
    Low: "Low",
    /**
     * Specifies the metric weight of the service load as Medium. The value is 2.
     */
    Medium: "Medium",
    /**
     * Specifies the metric weight of the service load as High. The value is 3.
     */
    High: "High",
};
exports.ServicePackageActivationMode = {
    /**
     * Indicates the application package activation mode will use shared process.
     */
    SharedProcess: "SharedProcess",
    /**
     * Indicates the application package activation mode will use exclusive process.
     */
    ExclusiveProcess: "ExclusiveProcess",
};
exports.ServicePlacementPolicyType = {
    /**
     * Indicates that the ServicePlacementPolicyDescription is of type ServicePlacementInvalidDomainPolicyDescription, which indicates that a particular fault or upgrade domain cannot be used for placement of this service. The value is 0.
     */
    InvalidDomain: "InvalidDomain",
    /**
     * Indicates that the ServicePlacementPolicyDescription is of type ServicePlacementRequireDomainDistributionPolicyDescription indicating that the replicas of the service must be placed in a specific domain. The value is 1.
     */
    RequiredDomain: "RequiredDomain",
    /**
     * Indicates that the ServicePlacementPolicyDescription is of type ServicePlacementPreferPrimaryDomainPolicyDescription, which indicates that if possible the Primary replica for the partitions of the service should be located in a particular domain as an optimization. The value is 2.
     */
    PreferredPrimaryDomain: "PreferredPrimaryDomain",
    /**
     * Indicates that the ServicePlacementPolicyDescription is of type ServicePlacementRequireDomainDistributionPolicyDescription, indicating that the system will disallow placement of any two replicas from the same partition in the same domain at any time. The value is 3.
     */
    RequiredDomainDistribution: "RequiredDomainDistribution",
    /**
     * Indicates that the ServicePlacementPolicyDescription is of type ServicePlacementNonPartiallyPlaceServicePolicyDescription, which indicates that if possible all replicas of a particular partition of the service should be placed atomically. The value is 4.
     */
    NonPartiallyPlaceService: "NonPartiallyPlaceService",
};
exports.ServiceScalingMechanismKind = {
    /**
     * Represents a scaling mechanism for adding or removing instances of stateless service partition. The value is 0.
     */
    ScalePartitionInstanceCount: "ScalePartitionInstanceCount",
    /**
     * Represents a scaling mechanism for adding or removing named partitions of a stateless service. The value is 1.
     */
    AddRemoveIncrementalNamedPartition: "AddRemoveIncrementalNamedPartition",
};
exports.ServiceScalingTriggerKind = {
    /**
     * Represents a scaling trigger related to an average load of a metric/resource of a partition. The value is 0.
     */
    AveragePartitionLoadTrigger: "AveragePartitionLoadTrigger",
    /**
     * Represents a scaling policy related to an average load of a metric/resource of a service. The value is 1.
     */
    AverageServiceLoadTrigger: "AverageServiceLoadTrigger",
};
exports.SkuName = {
    /**
     * Basic requires a minimum of 3 nodes and allows only 1 node type.
     */
    Basic: "Basic",
    /**
     * Requires a minimum of 5 nodes and allows 1 or more node type.
     */
    Standard: "Standard",
};
exports.VmSetupAction = {
    /**
     * Enable windows containers feature.
     */
    EnableContainers: "EnableContainers",
    /**
     * Enables windows HyperV feature.
     */
    EnableHyperV: "EnableHyperV",
};
exports.VmssExtensionSetupOrder = {
    /**
     * Indicates that the vm extension should run before the service fabric runtime starts.
     */
    BeforeSFRuntime: "BeforeSFRuntime",
};
exports.ZonalUpdateMode = {
    /**
     * The cluster will use 5 upgrade domains for Cross Az Node types.
     */
    Standard: "Standard",
    /**
     * The cluster will use a maximum of 3 upgrade domains per zone instead of 5 for Cross Az Node types for faster deployments.
     */
    Fast: "Fast",
};
