// @pulumi/azure-native/web/v20160301/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.listSiteIdentifiersAssignedToHostNameOutput = exports.listSiteIdentifiersAssignedToHostName = exports.getCertificateOutput = exports.getCertificate = exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.listSiteIdentifiersAssignedToHostName = null;
exports.listSiteIdentifiersAssignedToHostNameOutput = null;
utilities.lazyLoad(exports, ["listSiteIdentifiersAssignedToHostName", "listSiteIdentifiersAssignedToHostNameOutput"], () => require("./listSiteIdentifiersAssignedToHostName"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20160301:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20160301", _module);
