// @pulumi/azure-native/devcenter/v20240601preview/pool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A pool of Virtual Machines.
 */
class Pool extends pulumi.CustomResource {
    /**
     * Get an existing Pool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Pool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Pool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Pool.__pulumiType;
    }
    /**
     * Create a Pool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.devBoxDefinitionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'devBoxDefinitionName'");
            }
            if ((!args || args.licenseType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'licenseType'");
            }
            if ((!args || args.localAdministrator === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localAdministrator'");
            }
            if ((!args || args.networkConnectionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkConnectionName'");
            }
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["devBoxDefinition"] = args ? args.devBoxDefinition : undefined;
            resourceInputs["devBoxDefinitionName"] = args ? args.devBoxDefinitionName : undefined;
            resourceInputs["devBoxDefinitionType"] = args ? args.devBoxDefinitionType : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["licenseType"] = args ? args.licenseType : undefined;
            resourceInputs["localAdministrator"] = args ? args.localAdministrator : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managedVirtualNetworkRegions"] = args ? args.managedVirtualNetworkRegions : undefined;
            resourceInputs["networkConnectionName"] = args ? args.networkConnectionName : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["singleSignOnStatus"] = args ? args.singleSignOnStatus : undefined;
            resourceInputs["stopOnDisconnect"] = args ? args.stopOnDisconnect : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualNetworkType"] = args ? args.virtualNetworkType : undefined;
            resourceInputs["devBoxCount"] = undefined /*out*/;
            resourceInputs["healthStatus"] = undefined /*out*/;
            resourceInputs["healthStatusDetails"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["devBoxCount"] = undefined /*out*/;
            resourceInputs["devBoxDefinition"] = undefined /*out*/;
            resourceInputs["devBoxDefinitionName"] = undefined /*out*/;
            resourceInputs["devBoxDefinitionType"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["healthStatus"] = undefined /*out*/;
            resourceInputs["healthStatusDetails"] = undefined /*out*/;
            resourceInputs["licenseType"] = undefined /*out*/;
            resourceInputs["localAdministrator"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedVirtualNetworkRegions"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkConnectionName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["singleSignOnStatus"] = undefined /*out*/;
            resourceInputs["stopOnDisconnect"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworkType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devcenter/v20220801preview:Pool" }, { type: "azure-native:devcenter/v20220901preview:Pool" }, { type: "azure-native:devcenter/v20221012preview:Pool" }, { type: "azure-native:devcenter/v20221111preview:Pool" }, { type: "azure-native:devcenter/v20230101preview:Pool" }, { type: "azure-native:devcenter/v20230401:Pool" }, { type: "azure-native:devcenter/v20230801preview:Pool" }, { type: "azure-native:devcenter/v20231001preview:Pool" }, { type: "azure-native:devcenter/v20240201:Pool" }, { type: "azure-native:devcenter/v20240501preview:Pool" }, { type: "azure-native:devcenter/v20240701preview:Pool" }, { type: "azure-native:devcenter/v20240801preview:Pool" }, { type: "azure-native:devcenter/v20241001preview:Pool" }, { type: "azure-native:devcenter/v20250201:Pool" }, { type: "azure-native:devcenter:Pool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Pool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Pool = Pool;
/** @internal */
Pool.__pulumiType = 'azure-native:devcenter/v20240601preview:Pool';
