// @pulumi/azure-native/resources/v20191001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAzurePowerShellScriptOutput = exports.getAzurePowerShellScript = exports.getAzureCliScriptOutput = exports.getAzureCliScript = exports.AzurePowerShellScript = exports.AzureCliScript = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AzureCliScript = null;
utilities.lazyLoad(exports, ["AzureCliScript"], () => require("./azureCliScript"));
exports.AzurePowerShellScript = null;
utilities.lazyLoad(exports, ["AzurePowerShellScript"], () => require("./azurePowerShellScript"));
exports.getAzureCliScript = null;
exports.getAzureCliScriptOutput = null;
utilities.lazyLoad(exports, ["getAzureCliScript", "getAzureCliScriptOutput"], () => require("./getAzureCliScript"));
exports.getAzurePowerShellScript = null;
exports.getAzurePowerShellScriptOutput = null;
utilities.lazyLoad(exports, ["getAzurePowerShellScript", "getAzurePowerShellScriptOutput"], () => require("./getAzurePowerShellScript"));
// Export enums:
__exportStar(require("../../types/enums/resources/v20191001preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:resources/v20191001preview:AzureCliScript":
                return new exports.AzureCliScript(name, undefined, { urn });
            case "azure-native:resources/v20191001preview:AzurePowerShellScript":
                return new exports.AzurePowerShellScript(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "resources/v20191001preview", _module);
