// @pulumi/azure-native/powerbidedicated/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210101 = exports.getCapacityDetailsOutput = exports.getCapacityDetails = exports.getAutoScaleVCoreOutput = exports.getAutoScaleVCore = exports.CapacityDetails = exports.AutoScaleVCore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AutoScaleVCore = null;
utilities.lazyLoad(exports, ["AutoScaleVCore"], () => require("./autoScaleVCore"));
exports.CapacityDetails = null;
utilities.lazyLoad(exports, ["CapacityDetails"], () => require("./capacityDetails"));
exports.getAutoScaleVCore = null;
exports.getAutoScaleVCoreOutput = null;
utilities.lazyLoad(exports, ["getAutoScaleVCore", "getAutoScaleVCoreOutput"], () => require("./getAutoScaleVCore"));
exports.getCapacityDetails = null;
exports.getCapacityDetailsOutput = null;
utilities.lazyLoad(exports, ["getCapacityDetails", "getCapacityDetailsOutput"], () => require("./getCapacityDetails"));
// Export enums:
__exportStar(require("../types/enums/powerbidedicated"), exports);
// Export sub-modules:
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:powerbidedicated:AutoScaleVCore":
                return new exports.AutoScaleVCore(name, undefined, { urn });
            case "azure-native:powerbidedicated:CapacityDetails":
                return new exports.CapacityDetails(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "powerbidedicated", _module);
