// @pulumi/azure-native/advisor/v20230901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Suppression = exports.getSuppressionOutput = exports.getSuppression = exports.getAssessmentOutput = exports.getAssessment = exports.Assessment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Assessment = null;
utilities.lazyLoad(exports, ["Assessment"], () => require("./assessment"));
exports.getAssessment = null;
exports.getAssessmentOutput = null;
utilities.lazyLoad(exports, ["getAssessment", "getAssessmentOutput"], () => require("./getAssessment"));
exports.getSuppression = null;
exports.getSuppressionOutput = null;
utilities.lazyLoad(exports, ["getSuppression", "getSuppressionOutput"], () => require("./getSuppression"));
exports.Suppression = null;
utilities.lazyLoad(exports, ["Suppression"], () => require("./suppression"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:advisor/v20230901preview:Assessment":
                return new exports.Assessment(name, undefined, { urn });
            case "azure-native:advisor/v20230901preview:Suppression":
                return new exports.Suppression(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "advisor/v20230901preview", _module);
