// @pulumi/azure-native/managednetworkfabric/v20230615/externalNetwork.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../../types/input");
const utilities = require("../../utilities");
/**
 * Defines the External Network resource.
 */
class ExternalNetwork extends pulumi.CustomResource {
    /**
     * Get an existing ExternalNetwork resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExternalNetwork(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExternalNetwork.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExternalNetwork.__pulumiType;
    }
    /**
     * Create a ExternalNetwork resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.l3IsolationDomainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'l3IsolationDomainName'");
            }
            if ((!args || args.peeringOption === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringOption'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["annotation"] = args ? args.annotation : undefined;
            resourceInputs["exportRoutePolicy"] = args ? args.exportRoutePolicy : undefined;
            resourceInputs["exportRoutePolicyId"] = args ? args.exportRoutePolicyId : undefined;
            resourceInputs["externalNetworkName"] = args ? args.externalNetworkName : undefined;
            resourceInputs["importRoutePolicy"] = args ? args.importRoutePolicy : undefined;
            resourceInputs["importRoutePolicyId"] = args ? args.importRoutePolicyId : undefined;
            resourceInputs["l3IsolationDomainName"] = args ? args.l3IsolationDomainName : undefined;
            resourceInputs["networkToNetworkInterconnectId"] = args ? args.networkToNetworkInterconnectId : undefined;
            resourceInputs["optionAProperties"] = args ? (args.optionAProperties ? pulumi.output(args.optionAProperties).apply(inputs.managednetworkfabric.v20230615.externalNetworkPropertiesOptionAPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["optionBProperties"] = args ? args.optionBProperties : undefined;
            resourceInputs["peeringOption"] = args ? args.peeringOption : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["annotation"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["exportRoutePolicy"] = undefined /*out*/;
            resourceInputs["exportRoutePolicyId"] = undefined /*out*/;
            resourceInputs["importRoutePolicy"] = undefined /*out*/;
            resourceInputs["importRoutePolicyId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkToNetworkInterconnectId"] = undefined /*out*/;
            resourceInputs["optionAProperties"] = undefined /*out*/;
            resourceInputs["optionBProperties"] = undefined /*out*/;
            resourceInputs["peeringOption"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:managednetworkfabric/v20230201preview:ExternalNetwork" }, { type: "azure-native:managednetworkfabric:ExternalNetwork" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExternalNetwork.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExternalNetwork = ExternalNetwork;
/** @internal */
ExternalNetwork.__pulumiType = 'azure-native:managednetworkfabric/v20230615:ExternalNetwork';
