// @pulumi/azure-native/migrate/v20220801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MoveResource = exports.MoveCollection = exports.getMoveResourceOutput = exports.getMoveResource = exports.getMoveCollectionOutput = exports.getMoveCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMoveCollection = null;
exports.getMoveCollectionOutput = null;
utilities.lazyLoad(exports, ["getMoveCollection", "getMoveCollectionOutput"], () => require("./getMoveCollection"));
exports.getMoveResource = null;
exports.getMoveResourceOutput = null;
utilities.lazyLoad(exports, ["getMoveResource", "getMoveResourceOutput"], () => require("./getMoveResource"));
exports.MoveCollection = null;
utilities.lazyLoad(exports, ["MoveCollection"], () => require("./moveCollection"));
exports.MoveResource = null;
utilities.lazyLoad(exports, ["MoveResource"], () => require("./moveResource"));
// Export enums:
__exportStar(require("../../types/enums/migrate/v20220801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate/v20220801:MoveCollection":
                return new exports.MoveCollection(name, undefined, { urn });
            case "azure-native:migrate/v20220801:MoveResource":
                return new exports.MoveResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate/v20220801", _module);
