// @pulumi/azure-native/types/enums/webpubsub/v20210401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebPubSubSkuTier = exports.WebPubSubRequestType = exports.UpstreamAuthType = exports.ManagedIdentityType = exports.FeatureFlags = exports.ACLAction = void 0;
exports.ACLAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FeatureFlags = {
    EnableConnectivityLogs: "EnableConnectivityLogs",
    EnableMessagingLogs: "EnableMessagingLogs",
    EnableLiveTrace: "EnableLiveTrace",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.UpstreamAuthType = {
    None: "None",
    ManagedIdentity: "ManagedIdentity",
};
exports.WebPubSubRequestType = {
    ClientConnection: "ClientConnection",
    ServerConnection: "ServerConnection",
    RESTAPI: "RESTAPI",
    Trace: "Trace",
};
exports.WebPubSubSkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
