// @pulumi/azure-native/types/enums/purview/v20240401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantEndpointState = exports.PublicNetworkAccess = exports.PrivateEndpointConnectionStatus = exports.ManagedIdentityType = exports.ManagedEventHubState = exports.KafkaConfigurationIdentityType = exports.EventStreamingType = exports.EventStreamingState = exports.EventHubType = exports.AccountSkuName = void 0;
exports.AccountSkuName = {
    Standard: "Standard",
    Free: "Free",
};
exports.EventHubType = {
    Notification: "Notification",
    Hook: "Hook",
};
exports.EventStreamingState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.EventStreamingType = {
    None: "None",
    Managed: "Managed",
    Azure: "Azure",
};
exports.KafkaConfigurationIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.ManagedEventHubState = {
    NotSpecified: "NotSpecified",
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.PrivateEndpointConnectionStatus = {
    Unknown: "Unknown",
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    NotSpecified: "NotSpecified",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TenantEndpointState = {
    NotSpecified: "NotSpecified",
    Disabled: "Disabled",
    Enabled: "Enabled",
};
