// @pulumi/azure-native/scom/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230707preview = exports.MonitoredResource = exports.ManagedGateway = exports.Instance = exports.getMonitoredResourceOutput = exports.getMonitoredResource = exports.getManagedGatewayOutput = exports.getManagedGateway = exports.getInstanceOutput = exports.getInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.getManagedGateway = null;
exports.getManagedGatewayOutput = null;
utilities.lazyLoad(exports, ["getManagedGateway", "getManagedGatewayOutput"], () => require("./getManagedGateway"));
exports.getMonitoredResource = null;
exports.getMonitoredResourceOutput = null;
utilities.lazyLoad(exports, ["getMonitoredResource", "getMonitoredResourceOutput"], () => require("./getMonitoredResource"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.ManagedGateway = null;
utilities.lazyLoad(exports, ["ManagedGateway"], () => require("./managedGateway"));
exports.MonitoredResource = null;
utilities.lazyLoad(exports, ["MonitoredResource"], () => require("./monitoredResource"));
// Export enums:
__exportStar(require("../types/enums/scom"), exports);
// Export sub-modules:
const v20230707preview = require("./v20230707preview");
exports.v20230707preview = v20230707preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:scom:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "azure-native:scom:ManagedGateway":
                return new exports.ManagedGateway(name, undefined, { urn });
            case "azure-native:scom:MonitoredResource":
                return new exports.MonitoredResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "scom", _module);
