// @pulumi/kubernetes/events/v1beta1/eventPatch.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventPatch = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Patch resources are used to modify existing Kubernetes resources by using
 * Server-Side Apply updates. The name of the resource must be specified, but all other properties are optional. More than
 * one patch may be applied to the same resource, and a random FieldManager name will be used for each Patch resource.
 * Conflicts will result in an error by default, but can be forced using the "pulumi.com/patchForce" annotation. See the
 * [Server-Side Apply Docs](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/managing-resources-with-server-side-apply/) for
 * additional information about using Server-Side Apply to manage Kubernetes resources with Pulumi.
 * Event is a report of an event somewhere in the cluster. It generally denotes some state change in the system.
 */
class EventPatch extends pulumi.CustomResource {
    /**
     * Get an existing EventPatch resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EventPatch(name, undefined, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EventPatch.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventPatch.__pulumiType;
    }
    /**
     * Create a EventPatch resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["action"] = args?.action;
            resourceInputs["apiVersion"] = "events.k8s.io/v1beta1";
            resourceInputs["deprecatedCount"] = args?.deprecatedCount;
            resourceInputs["deprecatedFirstTimestamp"] = args?.deprecatedFirstTimestamp;
            resourceInputs["deprecatedLastTimestamp"] = args?.deprecatedLastTimestamp;
            resourceInputs["deprecatedSource"] = args?.deprecatedSource;
            resourceInputs["eventTime"] = args?.eventTime;
            resourceInputs["kind"] = "Event";
            resourceInputs["metadata"] = args?.metadata;
            resourceInputs["note"] = args?.note;
            resourceInputs["reason"] = args?.reason;
            resourceInputs["regarding"] = args?.regarding;
            resourceInputs["related"] = args?.related;
            resourceInputs["reportingController"] = args?.reportingController;
            resourceInputs["reportingInstance"] = args?.reportingInstance;
            resourceInputs["series"] = args?.series;
            resourceInputs["type"] = args?.type;
        }
        else {
            resourceInputs["action"] = undefined /*out*/;
            resourceInputs["apiVersion"] = undefined /*out*/;
            resourceInputs["deprecatedCount"] = undefined /*out*/;
            resourceInputs["deprecatedFirstTimestamp"] = undefined /*out*/;
            resourceInputs["deprecatedLastTimestamp"] = undefined /*out*/;
            resourceInputs["deprecatedSource"] = undefined /*out*/;
            resourceInputs["eventTime"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["note"] = undefined /*out*/;
            resourceInputs["reason"] = undefined /*out*/;
            resourceInputs["regarding"] = undefined /*out*/;
            resourceInputs["related"] = undefined /*out*/;
            resourceInputs["reportingController"] = undefined /*out*/;
            resourceInputs["reportingInstance"] = undefined /*out*/;
            resourceInputs["series"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "kubernetes:core/v1:EventPatch" }, { type: "kubernetes:events.k8s.io/v1:EventPatch" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EventPatch.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EventPatch = EventPatch;
/** @internal */
EventPatch.__pulumiType = 'kubernetes:events.k8s.io/v1beta1:EventPatch';
