// @pulumi/azure-native/mobilenetwork/v20220301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sim = exports.PacketCoreControlPlane = exports.getSimOutput = exports.getSim = exports.getPacketCoreControlPlaneOutput = exports.getPacketCoreControlPlane = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getPacketCoreControlPlane = null;
exports.getPacketCoreControlPlaneOutput = null;
utilities.lazyLoad(exports, ["getPacketCoreControlPlane", "getPacketCoreControlPlaneOutput"], () => require("./getPacketCoreControlPlane"));
exports.getSim = null;
exports.getSimOutput = null;
utilities.lazyLoad(exports, ["getSim", "getSimOutput"], () => require("./getSim"));
exports.PacketCoreControlPlane = null;
utilities.lazyLoad(exports, ["PacketCoreControlPlane"], () => require("./packetCoreControlPlane"));
exports.Sim = null;
utilities.lazyLoad(exports, ["Sim"], () => require("./sim"));
// Export enums:
__exportStar(require("../../types/enums/mobilenetwork/v20220301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mobilenetwork/v20220301preview:PacketCoreControlPlane":
                return new exports.PacketCoreControlPlane(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220301preview:Sim":
                return new exports.Sim(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mobilenetwork/v20220301preview", _module);
