// @pulumi/azure-native/servicebus/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicAuthorizationRule = exports.Topic = exports.Subscription = exports.Rule = exports.QueueAuthorizationRule = exports.Queue = exports.PrivateEndpointConnection = exports.NamespaceVirtualNetworkRule = exports.NamespaceNetworkRuleSet = exports.NamespaceIpFilterRule = exports.NamespaceAuthorizationRule = exports.Namespace = exports.MigrationConfig = exports.listTopicKeysOutput = exports.listTopicKeys = exports.listQueueKeysOutput = exports.listQueueKeys = exports.listNamespaceKeysOutput = exports.listNamespaceKeys = exports.listDisasterRecoveryConfigKeysOutput = exports.listDisasterRecoveryConfigKeys = exports.getTopicAuthorizationRuleOutput = exports.getTopicAuthorizationRule = exports.getTopicOutput = exports.getTopic = exports.getSubscriptionOutput = exports.getSubscription = exports.getRuleOutput = exports.getRule = exports.getQueueAuthorizationRuleOutput = exports.getQueueAuthorizationRule = exports.getQueueOutput = exports.getQueue = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNamespaceVirtualNetworkRuleOutput = exports.getNamespaceVirtualNetworkRule = exports.getNamespaceNetworkRuleSetOutput = exports.getNamespaceNetworkRuleSet = exports.getNamespaceIpFilterRuleOutput = exports.getNamespaceIpFilterRule = exports.getNamespaceAuthorizationRuleOutput = exports.getNamespaceAuthorizationRule = exports.getNamespaceOutput = exports.getNamespace = exports.getMigrationConfigOutput = exports.getMigrationConfig = exports.getDisasterRecoveryConfigOutput = exports.getDisasterRecoveryConfig = exports.DisasterRecoveryConfig = void 0;
exports.v20240101 = exports.v20230101preview = exports.v20221001preview = exports.v20220101preview = exports.v20180101preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DisasterRecoveryConfig = null;
utilities.lazyLoad(exports, ["DisasterRecoveryConfig"], () => require("./disasterRecoveryConfig"));
exports.getDisasterRecoveryConfig = null;
exports.getDisasterRecoveryConfigOutput = null;
utilities.lazyLoad(exports, ["getDisasterRecoveryConfig", "getDisasterRecoveryConfigOutput"], () => require("./getDisasterRecoveryConfig"));
exports.getMigrationConfig = null;
exports.getMigrationConfigOutput = null;
utilities.lazyLoad(exports, ["getMigrationConfig", "getMigrationConfigOutput"], () => require("./getMigrationConfig"));
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getNamespaceAuthorizationRule = null;
exports.getNamespaceAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getNamespaceAuthorizationRule", "getNamespaceAuthorizationRuleOutput"], () => require("./getNamespaceAuthorizationRule"));
exports.getNamespaceIpFilterRule = null;
exports.getNamespaceIpFilterRuleOutput = null;
utilities.lazyLoad(exports, ["getNamespaceIpFilterRule", "getNamespaceIpFilterRuleOutput"], () => require("./getNamespaceIpFilterRule"));
exports.getNamespaceNetworkRuleSet = null;
exports.getNamespaceNetworkRuleSetOutput = null;
utilities.lazyLoad(exports, ["getNamespaceNetworkRuleSet", "getNamespaceNetworkRuleSetOutput"], () => require("./getNamespaceNetworkRuleSet"));
exports.getNamespaceVirtualNetworkRule = null;
exports.getNamespaceVirtualNetworkRuleOutput = null;
utilities.lazyLoad(exports, ["getNamespaceVirtualNetworkRule", "getNamespaceVirtualNetworkRuleOutput"], () => require("./getNamespaceVirtualNetworkRule"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getQueue = null;
exports.getQueueOutput = null;
utilities.lazyLoad(exports, ["getQueue", "getQueueOutput"], () => require("./getQueue"));
exports.getQueueAuthorizationRule = null;
exports.getQueueAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getQueueAuthorizationRule", "getQueueAuthorizationRuleOutput"], () => require("./getQueueAuthorizationRule"));
exports.getRule = null;
exports.getRuleOutput = null;
utilities.lazyLoad(exports, ["getRule", "getRuleOutput"], () => require("./getRule"));
exports.getSubscription = null;
exports.getSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSubscription", "getSubscriptionOutput"], () => require("./getSubscription"));
exports.getTopic = null;
exports.getTopicOutput = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
exports.getTopicAuthorizationRule = null;
exports.getTopicAuthorizationRuleOutput = null;
utilities.lazyLoad(exports, ["getTopicAuthorizationRule", "getTopicAuthorizationRuleOutput"], () => require("./getTopicAuthorizationRule"));
exports.listDisasterRecoveryConfigKeys = null;
exports.listDisasterRecoveryConfigKeysOutput = null;
utilities.lazyLoad(exports, ["listDisasterRecoveryConfigKeys", "listDisasterRecoveryConfigKeysOutput"], () => require("./listDisasterRecoveryConfigKeys"));
exports.listNamespaceKeys = null;
exports.listNamespaceKeysOutput = null;
utilities.lazyLoad(exports, ["listNamespaceKeys", "listNamespaceKeysOutput"], () => require("./listNamespaceKeys"));
exports.listQueueKeys = null;
exports.listQueueKeysOutput = null;
utilities.lazyLoad(exports, ["listQueueKeys", "listQueueKeysOutput"], () => require("./listQueueKeys"));
exports.listTopicKeys = null;
exports.listTopicKeysOutput = null;
utilities.lazyLoad(exports, ["listTopicKeys", "listTopicKeysOutput"], () => require("./listTopicKeys"));
exports.MigrationConfig = null;
utilities.lazyLoad(exports, ["MigrationConfig"], () => require("./migrationConfig"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.NamespaceAuthorizationRule = null;
utilities.lazyLoad(exports, ["NamespaceAuthorizationRule"], () => require("./namespaceAuthorizationRule"));
exports.NamespaceIpFilterRule = null;
utilities.lazyLoad(exports, ["NamespaceIpFilterRule"], () => require("./namespaceIpFilterRule"));
exports.NamespaceNetworkRuleSet = null;
utilities.lazyLoad(exports, ["NamespaceNetworkRuleSet"], () => require("./namespaceNetworkRuleSet"));
exports.NamespaceVirtualNetworkRule = null;
utilities.lazyLoad(exports, ["NamespaceVirtualNetworkRule"], () => require("./namespaceVirtualNetworkRule"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueAuthorizationRule = null;
utilities.lazyLoad(exports, ["QueueAuthorizationRule"], () => require("./queueAuthorizationRule"));
exports.Rule = null;
utilities.lazyLoad(exports, ["Rule"], () => require("./rule"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
exports.Topic = null;
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
exports.TopicAuthorizationRule = null;
utilities.lazyLoad(exports, ["TopicAuthorizationRule"], () => require("./topicAuthorizationRule"));
// Export enums:
__exportStar(require("../types/enums/servicebus"), exports);
// Export sub-modules:
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20230101preview = require("./v20230101preview");
exports.v20230101preview = v20230101preview;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicebus:DisasterRecoveryConfig":
                return new exports.DisasterRecoveryConfig(name, undefined, { urn });
            case "azure-native:servicebus:MigrationConfig":
                return new exports.MigrationConfig(name, undefined, { urn });
            case "azure-native:servicebus:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceAuthorizationRule":
                return new exports.NamespaceAuthorizationRule(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceIpFilterRule":
                return new exports.NamespaceIpFilterRule(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceNetworkRuleSet":
                return new exports.NamespaceNetworkRuleSet(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceVirtualNetworkRule":
                return new exports.NamespaceVirtualNetworkRule(name, undefined, { urn });
            case "azure-native:servicebus:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:servicebus:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "azure-native:servicebus:QueueAuthorizationRule":
                return new exports.QueueAuthorizationRule(name, undefined, { urn });
            case "azure-native:servicebus:Rule":
                return new exports.Rule(name, undefined, { urn });
            case "azure-native:servicebus:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            case "azure-native:servicebus:Topic":
                return new exports.Topic(name, undefined, { urn });
            case "azure-native:servicebus:TopicAuthorizationRule":
                return new exports.TopicAuthorizationRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicebus", _module);
