// @pulumi/azure-native/types/enums/dbformysql/v20241001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageRedundancyEnum = exports.ServerVersion = exports.ServerSkuTier = exports.ReplicationRole = exports.PatchStrategy = exports.ManagedServiceIdentityType = exports.ImportSourceStorageType = exports.HighAvailabilityMode = exports.EnableStatusEnum = exports.DataEncryptionType = exports.CreateMode = void 0;
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    Replica: "Replica",
    GeoRestore: "GeoRestore",
};
exports.DataEncryptionType = {
    AzureKeyVault: "AzureKeyVault",
    SystemManaged: "SystemManaged",
};
exports.EnableStatusEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
    SameZone: "SameZone",
};
exports.ImportSourceStorageType = {
    AzureBlob: "AzureBlob",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.PatchStrategy = {
    Regular: "Regular",
    VirtualCanary: "VirtualCanary",
};
exports.ReplicationRole = {
    None: "None",
    Source: "Source",
    Replica: "Replica",
};
exports.ServerSkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.ServerVersion = {
    ServerVersion_5_7: "5.7",
    ServerVersion_8_0_21: "8.0.21",
};
exports.StorageRedundancyEnum = {
    LocalRedundancy: "LocalRedundancy",
    ZoneRedundancy: "ZoneRedundancy",
};
