// @pulumi/azure-native/web/v20181101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSlot = exports.WebAppDomainOwnershipIdentifierSlot = exports.WebAppDomainOwnershipIdentifier = exports.WebAppDiagnosticLogsConfigurationSlot = exports.WebApp = exports.listWebAppSyncFunctionTriggersSlotOutput = exports.listWebAppSyncFunctionTriggersSlot = exports.listWebAppSyncFunctionTriggersOutput = exports.listWebAppSyncFunctionTriggers = exports.listWebAppPublishingCredentialsSlotOutput = exports.listWebAppPublishingCredentialsSlot = exports.listWebAppPublishingCredentialsOutput = exports.listWebAppPublishingCredentials = exports.listWebAppHybridConnectionKeysSlotOutput = exports.listWebAppHybridConnectionKeysSlot = exports.listWebAppHybridConnectionKeysOutput = exports.listWebAppHybridConnectionKeys = exports.listWebAppBackupStatusSecretsSlotOutput = exports.listWebAppBackupStatusSecretsSlot = exports.listWebAppBackupStatusSecretsOutput = exports.listWebAppBackupStatusSecrets = exports.getWebAppSlotOutput = exports.getWebAppSlot = exports.getWebAppDomainOwnershipIdentifierSlotOutput = exports.getWebAppDomainOwnershipIdentifierSlot = exports.getWebAppDomainOwnershipIdentifierOutput = exports.getWebAppDomainOwnershipIdentifier = exports.getWebAppDiagnosticLogsConfigurationSlotOutput = exports.getWebAppDiagnosticLogsConfigurationSlot = exports.getWebAppOutput = exports.getWebApp = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWebApp = null;
exports.getWebAppOutput = null;
utilities.lazyLoad(exports, ["getWebApp", "getWebAppOutput"], () => require("./getWebApp"));
exports.getWebAppDiagnosticLogsConfigurationSlot = null;
exports.getWebAppDiagnosticLogsConfigurationSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDiagnosticLogsConfigurationSlot", "getWebAppDiagnosticLogsConfigurationSlotOutput"], () => require("./getWebAppDiagnosticLogsConfigurationSlot"));
exports.getWebAppDomainOwnershipIdentifier = null;
exports.getWebAppDomainOwnershipIdentifierOutput = null;
utilities.lazyLoad(exports, ["getWebAppDomainOwnershipIdentifier", "getWebAppDomainOwnershipIdentifierOutput"], () => require("./getWebAppDomainOwnershipIdentifier"));
exports.getWebAppDomainOwnershipIdentifierSlot = null;
exports.getWebAppDomainOwnershipIdentifierSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppDomainOwnershipIdentifierSlot", "getWebAppDomainOwnershipIdentifierSlotOutput"], () => require("./getWebAppDomainOwnershipIdentifierSlot"));
exports.getWebAppSlot = null;
exports.getWebAppSlotOutput = null;
utilities.lazyLoad(exports, ["getWebAppSlot", "getWebAppSlotOutput"], () => require("./getWebAppSlot"));
exports.listWebAppBackupStatusSecrets = null;
exports.listWebAppBackupStatusSecretsOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecrets", "listWebAppBackupStatusSecretsOutput"], () => require("./listWebAppBackupStatusSecrets"));
exports.listWebAppBackupStatusSecretsSlot = null;
exports.listWebAppBackupStatusSecretsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppBackupStatusSecretsSlot", "listWebAppBackupStatusSecretsSlotOutput"], () => require("./listWebAppBackupStatusSecretsSlot"));
exports.listWebAppHybridConnectionKeys = null;
exports.listWebAppHybridConnectionKeysOutput = null;
utilities.lazyLoad(exports, ["listWebAppHybridConnectionKeys", "listWebAppHybridConnectionKeysOutput"], () => require("./listWebAppHybridConnectionKeys"));
exports.listWebAppHybridConnectionKeysSlot = null;
exports.listWebAppHybridConnectionKeysSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppHybridConnectionKeysSlot", "listWebAppHybridConnectionKeysSlotOutput"], () => require("./listWebAppHybridConnectionKeysSlot"));
exports.listWebAppPublishingCredentials = null;
exports.listWebAppPublishingCredentialsOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentials", "listWebAppPublishingCredentialsOutput"], () => require("./listWebAppPublishingCredentials"));
exports.listWebAppPublishingCredentialsSlot = null;
exports.listWebAppPublishingCredentialsSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppPublishingCredentialsSlot", "listWebAppPublishingCredentialsSlotOutput"], () => require("./listWebAppPublishingCredentialsSlot"));
exports.listWebAppSyncFunctionTriggers = null;
exports.listWebAppSyncFunctionTriggersOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggers", "listWebAppSyncFunctionTriggersOutput"], () => require("./listWebAppSyncFunctionTriggers"));
exports.listWebAppSyncFunctionTriggersSlot = null;
exports.listWebAppSyncFunctionTriggersSlotOutput = null;
utilities.lazyLoad(exports, ["listWebAppSyncFunctionTriggersSlot", "listWebAppSyncFunctionTriggersSlotOutput"], () => require("./listWebAppSyncFunctionTriggersSlot"));
exports.WebApp = null;
utilities.lazyLoad(exports, ["WebApp"], () => require("./webApp"));
exports.WebAppDiagnosticLogsConfigurationSlot = null;
utilities.lazyLoad(exports, ["WebAppDiagnosticLogsConfigurationSlot"], () => require("./webAppDiagnosticLogsConfigurationSlot"));
exports.WebAppDomainOwnershipIdentifier = null;
utilities.lazyLoad(exports, ["WebAppDomainOwnershipIdentifier"], () => require("./webAppDomainOwnershipIdentifier"));
exports.WebAppDomainOwnershipIdentifierSlot = null;
utilities.lazyLoad(exports, ["WebAppDomainOwnershipIdentifierSlot"], () => require("./webAppDomainOwnershipIdentifierSlot"));
exports.WebAppSlot = null;
utilities.lazyLoad(exports, ["WebAppSlot"], () => require("./webAppSlot"));
// Export enums:
__exportStar(require("../../types/enums/web/v20181101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20181101:WebApp":
                return new exports.WebApp(name, undefined, { urn });
            case "azure-native:web/v20181101:WebAppDiagnosticLogsConfigurationSlot":
                return new exports.WebAppDiagnosticLogsConfigurationSlot(name, undefined, { urn });
            case "azure-native:web/v20181101:WebAppDomainOwnershipIdentifier":
                return new exports.WebAppDomainOwnershipIdentifier(name, undefined, { urn });
            case "azure-native:web/v20181101:WebAppDomainOwnershipIdentifierSlot":
                return new exports.WebAppDomainOwnershipIdentifierSlot(name, undefined, { urn });
            case "azure-native:web/v20181101:WebAppSlot":
                return new exports.WebAppSlot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20181101", _module);
