// @pulumi/azure-native/compute/v20240303/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getGalleryInVMAccessControlProfileVersionOutput = exports.getGalleryInVMAccessControlProfileVersion = exports.getGalleryInVMAccessControlProfileOutput = exports.getGalleryInVMAccessControlProfile = exports.getGalleryImageVersionOutput = exports.getGalleryImageVersion = exports.getGalleryImageOutput = exports.getGalleryImage = exports.getGalleryApplicationVersionOutput = exports.getGalleryApplicationVersion = exports.getGalleryApplicationOutput = exports.getGalleryApplication = exports.getGalleryOutput = exports.getGallery = exports.GalleryInVMAccessControlProfileVersion = exports.GalleryInVMAccessControlProfile = exports.GalleryImageVersion = exports.GalleryImage = exports.GalleryApplicationVersion = exports.GalleryApplication = exports.Gallery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Gallery = null;
utilities.lazyLoad(exports, ["Gallery"], () => require("./gallery"));
exports.GalleryApplication = null;
utilities.lazyLoad(exports, ["GalleryApplication"], () => require("./galleryApplication"));
exports.GalleryApplicationVersion = null;
utilities.lazyLoad(exports, ["GalleryApplicationVersion"], () => require("./galleryApplicationVersion"));
exports.GalleryImage = null;
utilities.lazyLoad(exports, ["GalleryImage"], () => require("./galleryImage"));
exports.GalleryImageVersion = null;
utilities.lazyLoad(exports, ["GalleryImageVersion"], () => require("./galleryImageVersion"));
exports.GalleryInVMAccessControlProfile = null;
utilities.lazyLoad(exports, ["GalleryInVMAccessControlProfile"], () => require("./galleryInVMAccessControlProfile"));
exports.GalleryInVMAccessControlProfileVersion = null;
utilities.lazyLoad(exports, ["GalleryInVMAccessControlProfileVersion"], () => require("./galleryInVMAccessControlProfileVersion"));
exports.getGallery = null;
exports.getGalleryOutput = null;
utilities.lazyLoad(exports, ["getGallery", "getGalleryOutput"], () => require("./getGallery"));
exports.getGalleryApplication = null;
exports.getGalleryApplicationOutput = null;
utilities.lazyLoad(exports, ["getGalleryApplication", "getGalleryApplicationOutput"], () => require("./getGalleryApplication"));
exports.getGalleryApplicationVersion = null;
exports.getGalleryApplicationVersionOutput = null;
utilities.lazyLoad(exports, ["getGalleryApplicationVersion", "getGalleryApplicationVersionOutput"], () => require("./getGalleryApplicationVersion"));
exports.getGalleryImage = null;
exports.getGalleryImageOutput = null;
utilities.lazyLoad(exports, ["getGalleryImage", "getGalleryImageOutput"], () => require("./getGalleryImage"));
exports.getGalleryImageVersion = null;
exports.getGalleryImageVersionOutput = null;
utilities.lazyLoad(exports, ["getGalleryImageVersion", "getGalleryImageVersionOutput"], () => require("./getGalleryImageVersion"));
exports.getGalleryInVMAccessControlProfile = null;
exports.getGalleryInVMAccessControlProfileOutput = null;
utilities.lazyLoad(exports, ["getGalleryInVMAccessControlProfile", "getGalleryInVMAccessControlProfileOutput"], () => require("./getGalleryInVMAccessControlProfile"));
exports.getGalleryInVMAccessControlProfileVersion = null;
exports.getGalleryInVMAccessControlProfileVersionOutput = null;
utilities.lazyLoad(exports, ["getGalleryInVMAccessControlProfileVersion", "getGalleryInVMAccessControlProfileVersionOutput"], () => require("./getGalleryInVMAccessControlProfileVersion"));
// Export enums:
__exportStar(require("../../types/enums/compute/v20240303"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:compute/v20240303:Gallery":
                return new exports.Gallery(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryApplication":
                return new exports.GalleryApplication(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryApplicationVersion":
                return new exports.GalleryApplicationVersion(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryImage":
                return new exports.GalleryImage(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryImageVersion":
                return new exports.GalleryImageVersion(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryInVMAccessControlProfile":
                return new exports.GalleryInVMAccessControlProfile(name, undefined, { urn });
            case "azure-native:compute/v20240303:GalleryInVMAccessControlProfileVersion":
                return new exports.GalleryInVMAccessControlProfileVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "compute/v20240303", _module);
