// @pulumi/gcp/networksecurity/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlList = exports.TlsInspectionPolicy = exports.ServerTlsPolicy = exports.SecurityProfileGroup = exports.SecurityProfile = exports.getAddressGroupIamPolicyOutput = exports.getAddressGroupIamPolicy = exports.GatewaySecurityPolicyRule = exports.GatewaySecurityPolicy = exports.FirewallEndpointAssociation = exports.FirewallEndpoint = exports.ClientTlsPolicy = exports.AuthorizationPolicy = exports.AddressGroupIamPolicy = exports.AddressGroupIamMember = exports.AddressGroupIamBinding = exports.AddressGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AddressGroup = null;
utilities.lazyLoad(exports, ["AddressGroup"], () => require("./addressGroup"));
exports.AddressGroupIamBinding = null;
utilities.lazyLoad(exports, ["AddressGroupIamBinding"], () => require("./addressGroupIamBinding"));
exports.AddressGroupIamMember = null;
utilities.lazyLoad(exports, ["AddressGroupIamMember"], () => require("./addressGroupIamMember"));
exports.AddressGroupIamPolicy = null;
utilities.lazyLoad(exports, ["AddressGroupIamPolicy"], () => require("./addressGroupIamPolicy"));
exports.AuthorizationPolicy = null;
utilities.lazyLoad(exports, ["AuthorizationPolicy"], () => require("./authorizationPolicy"));
exports.ClientTlsPolicy = null;
utilities.lazyLoad(exports, ["ClientTlsPolicy"], () => require("./clientTlsPolicy"));
exports.FirewallEndpoint = null;
utilities.lazyLoad(exports, ["FirewallEndpoint"], () => require("./firewallEndpoint"));
exports.FirewallEndpointAssociation = null;
utilities.lazyLoad(exports, ["FirewallEndpointAssociation"], () => require("./firewallEndpointAssociation"));
exports.GatewaySecurityPolicy = null;
utilities.lazyLoad(exports, ["GatewaySecurityPolicy"], () => require("./gatewaySecurityPolicy"));
exports.GatewaySecurityPolicyRule = null;
utilities.lazyLoad(exports, ["GatewaySecurityPolicyRule"], () => require("./gatewaySecurityPolicyRule"));
exports.getAddressGroupIamPolicy = null;
exports.getAddressGroupIamPolicyOutput = null;
utilities.lazyLoad(exports, ["getAddressGroupIamPolicy", "getAddressGroupIamPolicyOutput"], () => require("./getAddressGroupIamPolicy"));
exports.SecurityProfile = null;
utilities.lazyLoad(exports, ["SecurityProfile"], () => require("./securityProfile"));
exports.SecurityProfileGroup = null;
utilities.lazyLoad(exports, ["SecurityProfileGroup"], () => require("./securityProfileGroup"));
exports.ServerTlsPolicy = null;
utilities.lazyLoad(exports, ["ServerTlsPolicy"], () => require("./serverTlsPolicy"));
exports.TlsInspectionPolicy = null;
utilities.lazyLoad(exports, ["TlsInspectionPolicy"], () => require("./tlsInspectionPolicy"));
exports.UrlList = null;
utilities.lazyLoad(exports, ["UrlList"], () => require("./urlList"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:networksecurity/addressGroup:AddressGroup":
                return new exports.AddressGroup(name, undefined, { urn });
            case "gcp:networksecurity/addressGroupIamBinding:AddressGroupIamBinding":
                return new exports.AddressGroupIamBinding(name, undefined, { urn });
            case "gcp:networksecurity/addressGroupIamMember:AddressGroupIamMember":
                return new exports.AddressGroupIamMember(name, undefined, { urn });
            case "gcp:networksecurity/addressGroupIamPolicy:AddressGroupIamPolicy":
                return new exports.AddressGroupIamPolicy(name, undefined, { urn });
            case "gcp:networksecurity/authorizationPolicy:AuthorizationPolicy":
                return new exports.AuthorizationPolicy(name, undefined, { urn });
            case "gcp:networksecurity/clientTlsPolicy:ClientTlsPolicy":
                return new exports.ClientTlsPolicy(name, undefined, { urn });
            case "gcp:networksecurity/firewallEndpoint:FirewallEndpoint":
                return new exports.FirewallEndpoint(name, undefined, { urn });
            case "gcp:networksecurity/firewallEndpointAssociation:FirewallEndpointAssociation":
                return new exports.FirewallEndpointAssociation(name, undefined, { urn });
            case "gcp:networksecurity/gatewaySecurityPolicy:GatewaySecurityPolicy":
                return new exports.GatewaySecurityPolicy(name, undefined, { urn });
            case "gcp:networksecurity/gatewaySecurityPolicyRule:GatewaySecurityPolicyRule":
                return new exports.GatewaySecurityPolicyRule(name, undefined, { urn });
            case "gcp:networksecurity/securityProfile:SecurityProfile":
                return new exports.SecurityProfile(name, undefined, { urn });
            case "gcp:networksecurity/securityProfileGroup:SecurityProfileGroup":
                return new exports.SecurityProfileGroup(name, undefined, { urn });
            case "gcp:networksecurity/serverTlsPolicy:ServerTlsPolicy":
                return new exports.ServerTlsPolicy(name, undefined, { urn });
            case "gcp:networksecurity/tlsInspectionPolicy:TlsInspectionPolicy":
                return new exports.TlsInspectionPolicy(name, undefined, { urn });
            case "gcp:networksecurity/urlList:UrlList":
                return new exports.UrlList(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "networksecurity/addressGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/addressGroupIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/addressGroupIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/addressGroupIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/authorizationPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/clientTlsPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/firewallEndpoint", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/firewallEndpointAssociation", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/gatewaySecurityPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/gatewaySecurityPolicyRule", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/securityProfile", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/securityProfileGroup", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/serverTlsPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/tlsInspectionPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "networksecurity/urlList", _module);
