// @pulumi/azure-native/databoxedge/bandwidthSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BandwidthSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The bandwidth schedule details.
 *
 * Uses Azure REST API version 2022-03-01. In version 1.x of the Azure Native provider, it used API version 2020-12-01.
 *
 * Other available API versions: 2023-01-01-preview, 2023-07-01, 2023-12-01.
 */
class BandwidthSchedule extends pulumi.CustomResource {
    /**
     * Get an existing BandwidthSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BandwidthSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BandwidthSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BandwidthSchedule.__pulumiType;
    }
    /**
     * Create a BandwidthSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.days === undefined) && !opts.urn) {
                throw new Error("Missing required property 'days'");
            }
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.rateInMbps === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rateInMbps'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.start === undefined) && !opts.urn) {
                throw new Error("Missing required property 'start'");
            }
            if ((!args || args.stop === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stop'");
            }
            resourceInputs["days"] = args ? args.days : undefined;
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["rateInMbps"] = args ? args.rateInMbps : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["start"] = args ? args.start : undefined;
            resourceInputs["stop"] = args ? args.stop : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["days"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["rateInMbps"] = undefined /*out*/;
            resourceInputs["start"] = undefined /*out*/;
            resourceInputs["stop"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge/v20190301:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20190701:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20190801:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20200501preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20200901:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20200901preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20201201:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20210201:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20210201preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20210601:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20210601preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20220301:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20220401preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20221201preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20230101preview:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20230701:BandwidthSchedule" }, { type: "azure-native:databoxedge/v20231201:BandwidthSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BandwidthSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BandwidthSchedule = BandwidthSchedule;
/** @internal */
BandwidthSchedule.__pulumiType = 'azure-native:databoxedge:BandwidthSchedule';
