// @pulumi/azure-native/databoxedge/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order = exports.MonitoringConfig = exports.MECRole = exports.listOrderDCAccessCodeOutput = exports.listOrderDCAccessCode = exports.KubernetesRole = exports.IoTRole = exports.IoTAddon = exports.getUserOutput = exports.getUser = exports.getStorageAccountCredentialOutput = exports.getStorageAccountCredential = exports.getStorageAccountOutput = exports.getStorageAccount = exports.getShareOutput = exports.getShare = exports.getPeriodicTimerEventTriggerOutput = exports.getPeriodicTimerEventTrigger = exports.getOrderOutput = exports.getOrder = exports.getMonitoringConfigOutput = exports.getMonitoringConfig = exports.getMECRoleOutput = exports.getMECRole = exports.getKubernetesRoleOutput = exports.getKubernetesRole = exports.getIoTRoleOutput = exports.getIoTRole = exports.getIoTAddonOutput = exports.getIoTAddon = exports.getFileEventTriggerOutput = exports.getFileEventTrigger = exports.getDeviceExtendedInformationOutput = exports.getDeviceExtendedInformation = exports.getDeviceOutput = exports.getDevice = exports.getContainerOutput = exports.getContainer = exports.getCloudEdgeManagementRoleOutput = exports.getCloudEdgeManagementRole = exports.getBandwidthScheduleOutput = exports.getBandwidthSchedule = exports.getArcAddonOutput = exports.getArcAddon = exports.FileEventTrigger = exports.Device = exports.Container = exports.CloudEdgeManagementRole = exports.BandwidthSchedule = exports.ArcAddon = void 0;
exports.v20231201 = exports.v20230701 = exports.v20230101preview = exports.v20220401preview = exports.v20220301 = exports.v20210201preview = exports.v20210201 = exports.User = exports.StorageAccountCredential = exports.StorageAccount = exports.Share = exports.PeriodicTimerEventTrigger = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ArcAddon = null;
utilities.lazyLoad(exports, ["ArcAddon"], () => require("./arcAddon"));
exports.BandwidthSchedule = null;
utilities.lazyLoad(exports, ["BandwidthSchedule"], () => require("./bandwidthSchedule"));
exports.CloudEdgeManagementRole = null;
utilities.lazyLoad(exports, ["CloudEdgeManagementRole"], () => require("./cloudEdgeManagementRole"));
exports.Container = null;
utilities.lazyLoad(exports, ["Container"], () => require("./container"));
exports.Device = null;
utilities.lazyLoad(exports, ["Device"], () => require("./device"));
exports.FileEventTrigger = null;
utilities.lazyLoad(exports, ["FileEventTrigger"], () => require("./fileEventTrigger"));
exports.getArcAddon = null;
exports.getArcAddonOutput = null;
utilities.lazyLoad(exports, ["getArcAddon", "getArcAddonOutput"], () => require("./getArcAddon"));
exports.getBandwidthSchedule = null;
exports.getBandwidthScheduleOutput = null;
utilities.lazyLoad(exports, ["getBandwidthSchedule", "getBandwidthScheduleOutput"], () => require("./getBandwidthSchedule"));
exports.getCloudEdgeManagementRole = null;
exports.getCloudEdgeManagementRoleOutput = null;
utilities.lazyLoad(exports, ["getCloudEdgeManagementRole", "getCloudEdgeManagementRoleOutput"], () => require("./getCloudEdgeManagementRole"));
exports.getContainer = null;
exports.getContainerOutput = null;
utilities.lazyLoad(exports, ["getContainer", "getContainerOutput"], () => require("./getContainer"));
exports.getDevice = null;
exports.getDeviceOutput = null;
utilities.lazyLoad(exports, ["getDevice", "getDeviceOutput"], () => require("./getDevice"));
exports.getDeviceExtendedInformation = null;
exports.getDeviceExtendedInformationOutput = null;
utilities.lazyLoad(exports, ["getDeviceExtendedInformation", "getDeviceExtendedInformationOutput"], () => require("./getDeviceExtendedInformation"));
exports.getFileEventTrigger = null;
exports.getFileEventTriggerOutput = null;
utilities.lazyLoad(exports, ["getFileEventTrigger", "getFileEventTriggerOutput"], () => require("./getFileEventTrigger"));
exports.getIoTAddon = null;
exports.getIoTAddonOutput = null;
utilities.lazyLoad(exports, ["getIoTAddon", "getIoTAddonOutput"], () => require("./getIoTAddon"));
exports.getIoTRole = null;
exports.getIoTRoleOutput = null;
utilities.lazyLoad(exports, ["getIoTRole", "getIoTRoleOutput"], () => require("./getIoTRole"));
exports.getKubernetesRole = null;
exports.getKubernetesRoleOutput = null;
utilities.lazyLoad(exports, ["getKubernetesRole", "getKubernetesRoleOutput"], () => require("./getKubernetesRole"));
exports.getMECRole = null;
exports.getMECRoleOutput = null;
utilities.lazyLoad(exports, ["getMECRole", "getMECRoleOutput"], () => require("./getMECRole"));
exports.getMonitoringConfig = null;
exports.getMonitoringConfigOutput = null;
utilities.lazyLoad(exports, ["getMonitoringConfig", "getMonitoringConfigOutput"], () => require("./getMonitoringConfig"));
exports.getOrder = null;
exports.getOrderOutput = null;
utilities.lazyLoad(exports, ["getOrder", "getOrderOutput"], () => require("./getOrder"));
exports.getPeriodicTimerEventTrigger = null;
exports.getPeriodicTimerEventTriggerOutput = null;
utilities.lazyLoad(exports, ["getPeriodicTimerEventTrigger", "getPeriodicTimerEventTriggerOutput"], () => require("./getPeriodicTimerEventTrigger"));
exports.getShare = null;
exports.getShareOutput = null;
utilities.lazyLoad(exports, ["getShare", "getShareOutput"], () => require("./getShare"));
exports.getStorageAccount = null;
exports.getStorageAccountOutput = null;
utilities.lazyLoad(exports, ["getStorageAccount", "getStorageAccountOutput"], () => require("./getStorageAccount"));
exports.getStorageAccountCredential = null;
exports.getStorageAccountCredentialOutput = null;
utilities.lazyLoad(exports, ["getStorageAccountCredential", "getStorageAccountCredentialOutput"], () => require("./getStorageAccountCredential"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.IoTAddon = null;
utilities.lazyLoad(exports, ["IoTAddon"], () => require("./ioTAddon"));
exports.IoTRole = null;
utilities.lazyLoad(exports, ["IoTRole"], () => require("./ioTRole"));
exports.KubernetesRole = null;
utilities.lazyLoad(exports, ["KubernetesRole"], () => require("./kubernetesRole"));
exports.listOrderDCAccessCode = null;
exports.listOrderDCAccessCodeOutput = null;
utilities.lazyLoad(exports, ["listOrderDCAccessCode", "listOrderDCAccessCodeOutput"], () => require("./listOrderDCAccessCode"));
exports.MECRole = null;
utilities.lazyLoad(exports, ["MECRole"], () => require("./mecrole"));
exports.MonitoringConfig = null;
utilities.lazyLoad(exports, ["MonitoringConfig"], () => require("./monitoringConfig"));
exports.Order = null;
utilities.lazyLoad(exports, ["Order"], () => require("./order"));
exports.PeriodicTimerEventTrigger = null;
utilities.lazyLoad(exports, ["PeriodicTimerEventTrigger"], () => require("./periodicTimerEventTrigger"));
exports.Share = null;
utilities.lazyLoad(exports, ["Share"], () => require("./share"));
exports.StorageAccount = null;
utilities.lazyLoad(exports, ["StorageAccount"], () => require("./storageAccount"));
exports.StorageAccountCredential = null;
utilities.lazyLoad(exports, ["StorageAccountCredential"], () => require("./storageAccountCredential"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../types/enums/databoxedge"), exports);
// Export sub-modules:
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20230101preview = require("./v20230101preview");
exports.v20230101preview = v20230101preview;
const v20230701 = require("./v20230701");
exports.v20230701 = v20230701;
const v20231201 = require("./v20231201");
exports.v20231201 = v20231201;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:databoxedge:ArcAddon":
                return new exports.ArcAddon(name, undefined, { urn });
            case "azure-native:databoxedge:BandwidthSchedule":
                return new exports.BandwidthSchedule(name, undefined, { urn });
            case "azure-native:databoxedge:CloudEdgeManagementRole":
                return new exports.CloudEdgeManagementRole(name, undefined, { urn });
            case "azure-native:databoxedge:Container":
                return new exports.Container(name, undefined, { urn });
            case "azure-native:databoxedge:Device":
                return new exports.Device(name, undefined, { urn });
            case "azure-native:databoxedge:FileEventTrigger":
                return new exports.FileEventTrigger(name, undefined, { urn });
            case "azure-native:databoxedge:IoTAddon":
                return new exports.IoTAddon(name, undefined, { urn });
            case "azure-native:databoxedge:IoTRole":
                return new exports.IoTRole(name, undefined, { urn });
            case "azure-native:databoxedge:KubernetesRole":
                return new exports.KubernetesRole(name, undefined, { urn });
            case "azure-native:databoxedge:MECRole":
                return new exports.MECRole(name, undefined, { urn });
            case "azure-native:databoxedge:MonitoringConfig":
                return new exports.MonitoringConfig(name, undefined, { urn });
            case "azure-native:databoxedge:Order":
                return new exports.Order(name, undefined, { urn });
            case "azure-native:databoxedge:PeriodicTimerEventTrigger":
                return new exports.PeriodicTimerEventTrigger(name, undefined, { urn });
            case "azure-native:databoxedge:Share":
                return new exports.Share(name, undefined, { urn });
            case "azure-native:databoxedge:StorageAccount":
                return new exports.StorageAccount(name, undefined, { urn });
            case "azure-native:databoxedge:StorageAccountCredential":
                return new exports.StorageAccountCredential(name, undefined, { urn });
            case "azure-native:databoxedge:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "databoxedge", _module);
