// @pulumi/azure-native/quota/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250315preview = exports.v20250301 = exports.v20241218preview = exports.v20241015preview = exports.v20230601preview = exports.GroupQuotaSubscription = exports.GroupQuota = exports.getGroupQuotaSubscriptionOutput = exports.getGroupQuotaSubscription = exports.getGroupQuotaOutput = exports.getGroupQuota = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroupQuota = null;
exports.getGroupQuotaOutput = null;
utilities.lazyLoad(exports, ["getGroupQuota", "getGroupQuotaOutput"], () => require("./getGroupQuota"));
exports.getGroupQuotaSubscription = null;
exports.getGroupQuotaSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getGroupQuotaSubscription", "getGroupQuotaSubscriptionOutput"], () => require("./getGroupQuotaSubscription"));
exports.GroupQuota = null;
utilities.lazyLoad(exports, ["GroupQuota"], () => require("./groupQuota"));
exports.GroupQuotaSubscription = null;
utilities.lazyLoad(exports, ["GroupQuotaSubscription"], () => require("./groupQuotaSubscription"));
// Export enums:
__exportStar(require("../types/enums/quota"), exports);
// Export sub-modules:
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20241015preview = require("./v20241015preview");
exports.v20241015preview = v20241015preview;
const v20241218preview = require("./v20241218preview");
exports.v20241218preview = v20241218preview;
const v20250301 = require("./v20250301");
exports.v20250301 = v20250301;
const v20250315preview = require("./v20250315preview");
exports.v20250315preview = v20250315preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:quota:GroupQuota":
                return new exports.GroupQuota(name, undefined, { urn });
            case "azure-native:quota:GroupQuotaSubscription":
                return new exports.GroupQuotaSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "quota", _module);
