// @pulumi/azure-native/types/enums/dataprotection/v20230101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekNumber = exports.ValidationType = exports.StorageSettingTypes = exports.StorageSettingStoreTypes = exports.SoftDeleteState = exports.SecretStoreType = exports.Month = exports.ImmutabilityState = exports.DayOfWeek = exports.DataStoreTypes = exports.CrossSubscriptionRestoreState = exports.AlertsState = exports.AbsoluteMarker = void 0;
exports.AbsoluteMarker = {
    AllBackup: "AllBackup",
    FirstOfDay: "FirstOfDay",
    FirstOfMonth: "FirstOfMonth",
    FirstOfWeek: "FirstOfWeek",
    FirstOfYear: "FirstOfYear",
};
exports.AlertsState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CrossSubscriptionRestoreState = {
    Disabled: "Disabled",
    PermanentlyDisabled: "PermanentlyDisabled",
    Enabled: "Enabled",
};
exports.DataStoreTypes = {
    OperationalStore: "OperationalStore",
    VaultStore: "VaultStore",
    ArchiveStore: "ArchiveStore",
};
exports.DayOfWeek = {
    Friday: "Friday",
    Monday: "Monday",
    Saturday: "Saturday",
    Sunday: "Sunday",
    Thursday: "Thursday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
};
exports.ImmutabilityState = {
    Disabled: "Disabled",
    Unlocked: "Unlocked",
    Locked: "Locked",
};
exports.Month = {
    April: "April",
    August: "August",
    December: "December",
    February: "February",
    January: "January",
    July: "July",
    June: "June",
    March: "March",
    May: "May",
    November: "November",
    October: "October",
    September: "September",
};
exports.SecretStoreType = {
    Invalid: "Invalid",
    AzureKeyVault: "AzureKeyVault",
};
exports.SoftDeleteState = {
    /**
     * Soft Delete is turned off for the BackupVault
     */
    Off: "Off",
    /**
     * Soft Delete is enabled for the BackupVault but can be turned off
     */
    On: "On",
    /**
     * Soft Delete is permanently enabled for the BackupVault and the setting cannot be changed
     */
    AlwaysOn: "AlwaysOn",
};
exports.StorageSettingStoreTypes = {
    ArchiveStore: "ArchiveStore",
    OperationalStore: "OperationalStore",
    VaultStore: "VaultStore",
};
exports.StorageSettingTypes = {
    GeoRedundant: "GeoRedundant",
    LocallyRedundant: "LocallyRedundant",
    ZoneRedundant: "ZoneRedundant",
};
exports.ValidationType = {
    ShallowValidation: "ShallowValidation",
    DeepValidation: "DeepValidation",
};
exports.WeekNumber = {
    First: "First",
    Fourth: "Fourth",
    Last: "Last",
    Second: "Second",
    Third: "Third",
};
