// @pulumi/azure-native/managednetwork/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20190601preview = exports.ScopeAssignment = exports.ManagedNetworkPeeringPolicy = exports.ManagedNetworkGroup = exports.ManagedNetwork = exports.getScopeAssignmentOutput = exports.getScopeAssignment = exports.getManagedNetworkPeeringPolicyOutput = exports.getManagedNetworkPeeringPolicy = exports.getManagedNetworkGroupOutput = exports.getManagedNetworkGroup = exports.getManagedNetworkOutput = exports.getManagedNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getManagedNetwork = null;
exports.getManagedNetworkOutput = null;
utilities.lazyLoad(exports, ["getManagedNetwork", "getManagedNetworkOutput"], () => require("./getManagedNetwork"));
exports.getManagedNetworkGroup = null;
exports.getManagedNetworkGroupOutput = null;
utilities.lazyLoad(exports, ["getManagedNetworkGroup", "getManagedNetworkGroupOutput"], () => require("./getManagedNetworkGroup"));
exports.getManagedNetworkPeeringPolicy = null;
exports.getManagedNetworkPeeringPolicyOutput = null;
utilities.lazyLoad(exports, ["getManagedNetworkPeeringPolicy", "getManagedNetworkPeeringPolicyOutput"], () => require("./getManagedNetworkPeeringPolicy"));
exports.getScopeAssignment = null;
exports.getScopeAssignmentOutput = null;
utilities.lazyLoad(exports, ["getScopeAssignment", "getScopeAssignmentOutput"], () => require("./getScopeAssignment"));
exports.ManagedNetwork = null;
utilities.lazyLoad(exports, ["ManagedNetwork"], () => require("./managedNetwork"));
exports.ManagedNetworkGroup = null;
utilities.lazyLoad(exports, ["ManagedNetworkGroup"], () => require("./managedNetworkGroup"));
exports.ManagedNetworkPeeringPolicy = null;
utilities.lazyLoad(exports, ["ManagedNetworkPeeringPolicy"], () => require("./managedNetworkPeeringPolicy"));
exports.ScopeAssignment = null;
utilities.lazyLoad(exports, ["ScopeAssignment"], () => require("./scopeAssignment"));
// Export enums:
__exportStar(require("../types/enums/managednetwork"), exports);
// Export sub-modules:
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managednetwork:ManagedNetwork":
                return new exports.ManagedNetwork(name, undefined, { urn });
            case "azure-native:managednetwork:ManagedNetworkGroup":
                return new exports.ManagedNetworkGroup(name, undefined, { urn });
            case "azure-native:managednetwork:ManagedNetworkPeeringPolicy":
                return new exports.ManagedNetworkPeeringPolicy(name, undefined, { urn });
            case "azure-native:managednetwork:ScopeAssignment":
                return new exports.ScopeAssignment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managednetwork", _module);
