// @pulumi/azure-native/web/v20190801/appServiceEnvironment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppServiceEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * App Service Environment ARM resource.
 */
class AppServiceEnvironment extends pulumi.CustomResource {
    /**
     * Get an existing AppServiceEnvironment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AppServiceEnvironment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AppServiceEnvironment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AppServiceEnvironment.__pulumiType;
    }
    /**
     * Create a AppServiceEnvironment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.virtualNetwork === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualNetwork'");
            }
            if ((!args || args.workerPools === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workerPools'");
            }
            resourceInputs["apiManagementAccountId"] = args ? args.apiManagementAccountId : undefined;
            resourceInputs["clusterSettings"] = args ? args.clusterSettings : undefined;
            resourceInputs["dnsSuffix"] = args ? args.dnsSuffix : undefined;
            resourceInputs["dynamicCacheEnabled"] = args ? args.dynamicCacheEnabled : undefined;
            resourceInputs["frontEndScaleFactor"] = args ? args.frontEndScaleFactor : undefined;
            resourceInputs["hasLinuxWorkers"] = args ? args.hasLinuxWorkers : undefined;
            resourceInputs["internalLoadBalancingMode"] = args ? args.internalLoadBalancingMode : undefined;
            resourceInputs["ipsslAddressCount"] = args ? args.ipsslAddressCount : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["multiRoleCount"] = args ? args.multiRoleCount : undefined;
            resourceInputs["multiSize"] = args ? args.multiSize : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkAccessControlList"] = args ? args.networkAccessControlList : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sslCertKeyVaultId"] = args ? args.sslCertKeyVaultId : undefined;
            resourceInputs["sslCertKeyVaultSecretName"] = args ? args.sslCertKeyVaultSecretName : undefined;
            resourceInputs["suspended"] = args ? args.suspended : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["userWhitelistedIpRanges"] = args ? args.userWhitelistedIpRanges : undefined;
            resourceInputs["virtualNetwork"] = args ? args.virtualNetwork : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vnetResourceGroupName"] = args ? args.vnetResourceGroupName : undefined;
            resourceInputs["vnetSubnetName"] = args ? args.vnetSubnetName : undefined;
            resourceInputs["workerPools"] = args ? args.workerPools : undefined;
            resourceInputs["allowedMultiSizes"] = undefined /*out*/;
            resourceInputs["allowedWorkerSizes"] = undefined /*out*/;
            resourceInputs["databaseEdition"] = undefined /*out*/;
            resourceInputs["databaseServiceObjective"] = undefined /*out*/;
            resourceInputs["defaultFrontEndScaleFactor"] = undefined /*out*/;
            resourceInputs["environmentCapacities"] = undefined /*out*/;
            resourceInputs["environmentIsHealthy"] = undefined /*out*/;
            resourceInputs["environmentStatus"] = undefined /*out*/;
            resourceInputs["lastAction"] = undefined /*out*/;
            resourceInputs["lastActionResult"] = undefined /*out*/;
            resourceInputs["maximumNumberOfMachines"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upgradeDomains"] = undefined /*out*/;
            resourceInputs["vipMappings"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowedMultiSizes"] = undefined /*out*/;
            resourceInputs["allowedWorkerSizes"] = undefined /*out*/;
            resourceInputs["apiManagementAccountId"] = undefined /*out*/;
            resourceInputs["clusterSettings"] = undefined /*out*/;
            resourceInputs["databaseEdition"] = undefined /*out*/;
            resourceInputs["databaseServiceObjective"] = undefined /*out*/;
            resourceInputs["defaultFrontEndScaleFactor"] = undefined /*out*/;
            resourceInputs["dnsSuffix"] = undefined /*out*/;
            resourceInputs["dynamicCacheEnabled"] = undefined /*out*/;
            resourceInputs["environmentCapacities"] = undefined /*out*/;
            resourceInputs["environmentIsHealthy"] = undefined /*out*/;
            resourceInputs["environmentStatus"] = undefined /*out*/;
            resourceInputs["frontEndScaleFactor"] = undefined /*out*/;
            resourceInputs["hasLinuxWorkers"] = undefined /*out*/;
            resourceInputs["internalLoadBalancingMode"] = undefined /*out*/;
            resourceInputs["ipsslAddressCount"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastAction"] = undefined /*out*/;
            resourceInputs["lastActionResult"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maximumNumberOfMachines"] = undefined /*out*/;
            resourceInputs["multiRoleCount"] = undefined /*out*/;
            resourceInputs["multiSize"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkAccessControlList"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["sslCertKeyVaultId"] = undefined /*out*/;
            resourceInputs["sslCertKeyVaultSecretName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
            resourceInputs["suspended"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upgradeDomains"] = undefined /*out*/;
            resourceInputs["userWhitelistedIpRanges"] = undefined /*out*/;
            resourceInputs["vipMappings"] = undefined /*out*/;
            resourceInputs["virtualNetwork"] = undefined /*out*/;
            resourceInputs["vnetName"] = undefined /*out*/;
            resourceInputs["vnetResourceGroupName"] = undefined /*out*/;
            resourceInputs["vnetSubnetName"] = undefined /*out*/;
            resourceInputs["workerPools"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:AppServiceEnvironment" }, { type: "azure-native:web/v20160901:AppServiceEnvironment" }, { type: "azure-native:web/v20180201:AppServiceEnvironment" }, { type: "azure-native:web/v20200601:AppServiceEnvironment" }, { type: "azure-native:web/v20200901:AppServiceEnvironment" }, { type: "azure-native:web/v20201001:AppServiceEnvironment" }, { type: "azure-native:web/v20201201:AppServiceEnvironment" }, { type: "azure-native:web/v20210101:AppServiceEnvironment" }, { type: "azure-native:web/v20210115:AppServiceEnvironment" }, { type: "azure-native:web/v20210201:AppServiceEnvironment" }, { type: "azure-native:web/v20210301:AppServiceEnvironment" }, { type: "azure-native:web/v20220301:AppServiceEnvironment" }, { type: "azure-native:web/v20220901:AppServiceEnvironment" }, { type: "azure-native:web/v20230101:AppServiceEnvironment" }, { type: "azure-native:web/v20231201:AppServiceEnvironment" }, { type: "azure-native:web/v20240401:AppServiceEnvironment" }, { type: "azure-native:web:AppServiceEnvironment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AppServiceEnvironment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AppServiceEnvironment = AppServiceEnvironment;
/** @internal */
AppServiceEnvironment.__pulumiType = 'azure-native:web/v20190801:AppServiceEnvironment';
