// @pulumi/azure-native/types/enums/voiceservices/v20230901/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLinePurpose = exports.TeamsCodecs = exports.SkuTier = exports.ManagedServiceIdentityType = exports.E911Type = exports.Connectivity = exports.CommunicationsPlatform = exports.AutoGeneratedDomainNameLabelScope = exports.ApiBridgeActivationState = void 0;
exports.ApiBridgeActivationState = {
    /**
     * API Bridge is enabled
     */
    Enabled: "enabled",
    /**
     * API Bridge is disabled
     */
    Disabled: "disabled",
};
exports.AutoGeneratedDomainNameLabelScope = {
    /**
     * Generated domain name label depends on resource name and tenant ID.
     */
    TenantReuse: "TenantReuse",
    /**
     * Generated domain name label depends on resource name, tenant ID and subscription ID.
     */
    SubscriptionReuse: "SubscriptionReuse",
    /**
     * Generated domain name label depends on resource name, tenant ID, subscription ID and resource group name.
     */
    ResourceGroupReuse: "ResourceGroupReuse",
    /**
     * Generated domain name label is always unique.
     */
    NoReuse: "NoReuse",
};
exports.CommunicationsPlatform = {
    /**
     * Operator Connect
     */
    OperatorConnect: "OperatorConnect",
    /**
     * Teams Phone Mobile
     */
    TeamsPhoneMobile: "TeamsPhoneMobile",
    /**
     * Teams Direct Routing
     */
    TeamsDirectRouting: "TeamsDirectRouting",
};
exports.Connectivity = {
    /**
     * This deployment connects to the operator network using a Public IP address, e.g. when using MAPS
     */
    PublicAddress: "PublicAddress",
};
exports.E911Type = {
    /**
     * Emergency calls are not handled different from other calls
     */
    Standard: "Standard",
    /**
     * Emergency calls are routed directly to the ESRP
     */
    DirectToEsrp: "DirectToEsrp",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TeamsCodecs = {
    /**
     * Pulse code modulation(PCM) U-law narrowband audio codec(G.711u)
     */
    PCMA: "PCMA",
    /**
     * Pulse code modulation(PCM) U-law narrowband audio codec(G.711u)
     */
    PCMU: "PCMU",
    /**
     * G.722 wideband audio codec
     */
    G722: "G722",
    /**
     * G.722.2 wideband audio codec
     */
    G722_2: "G722_2",
    /**
     * SILK/8000 narrowband audio codec
     */
    SILK_8: "SILK_8",
    /**
     * SILK/16000 wideband audio codec
     */
    SILK_16: "SILK_16",
};
exports.TestLinePurpose = {
    /**
     * The test line is used for manual testing
     */
    Manual: "Manual",
    /**
     * The test line is used for automated testing
     */
    Automated: "Automated",
};
