// @pulumi/azure-native/types/enums/devcenter/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopOnDisconnectEnableStatus = exports.SkuTier = exports.ScheduledType = exports.ScheduledFrequency = exports.ScheduleEnableStatus = exports.PlanMemberType = exports.ManagedServiceIdentityType = exports.LocalAdminStatus = exports.LicenseType = exports.HibernateSupport = exports.EnvironmentTypeEnableStatus = exports.DomainJoinType = exports.DevboxDisksEncryptionEnableStatus = exports.CatalogSyncType = exports.v20250201 = exports.v20241001preview = exports.v20240801preview = exports.v20240701preview = exports.v20240601preview = exports.v20240501preview = exports.v20240201 = exports.v20231001preview = exports.v20230801preview = exports.v20230401 = exports.v20221111preview = void 0;
// Export sub-modules:
const v20221111preview = require("./v20221111preview");
exports.v20221111preview = v20221111preview;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20240201 = require("./v20240201");
exports.v20240201 = v20240201;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const v20250201 = require("./v20250201");
exports.v20250201 = v20250201;
exports.CatalogSyncType = {
    Manual: "Manual",
    Scheduled: "Scheduled",
};
exports.DevboxDisksEncryptionEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DomainJoinType = {
    HybridAzureADJoin: "HybridAzureADJoin",
    AzureADJoin: "AzureADJoin",
};
exports.EnvironmentTypeEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HibernateSupport = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.LicenseType = {
    Windows_Client: "Windows_Client",
};
exports.LocalAdminStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.PlanMemberType = {
    User: "User",
    Group: "Group",
};
exports.ScheduleEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ScheduledFrequency = {
    Daily: "Daily",
};
exports.ScheduledType = {
    StopDevBox: "StopDevBox",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.StopOnDisconnectEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
