// @pulumi/azure-native/types/enums/devcenter/v20230401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopOnDisconnectEnableStatus = exports.SkuTier = exports.ScheduledType = exports.ScheduledFrequency = exports.ScheduleEnableStatus = exports.ManagedServiceIdentityType = exports.LocalAdminStatus = exports.LicenseType = exports.HibernateSupport = exports.EnvironmentTypeEnableStatus = exports.DomainJoinType = void 0;
exports.DomainJoinType = {
    HybridAzureADJoin: "HybridAzureADJoin",
    AzureADJoin: "AzureADJoin",
};
exports.EnvironmentTypeEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HibernateSupport = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.LicenseType = {
    Windows_Client: "Windows_Client",
};
exports.LocalAdminStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.ScheduleEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ScheduledFrequency = {
    Daily: "Daily",
};
exports.ScheduledType = {
    StopDevBox: "StopDevBox",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.StopOnDisconnectEnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
