// @pulumi/azure-native/networkcloud/v20240601preview/kubernetesClusterFeature.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesClusterFeature = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class KubernetesClusterFeature extends pulumi.CustomResource {
    /**
     * Get an existing KubernetesClusterFeature resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KubernetesClusterFeature(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KubernetesClusterFeature.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KubernetesClusterFeature.__pulumiType;
    }
    /**
     * Create a KubernetesClusterFeature resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kubernetesClusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesClusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["featureName"] = args ? args.featureName : undefined;
            resourceInputs["kubernetesClusterName"] = args ? args.kubernetesClusterName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["availabilityLifecycle"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["required"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        else {
            resourceInputs["availabilityLifecycle"] = undefined /*out*/;
            resourceInputs["detailedStatus"] = undefined /*out*/;
            resourceInputs["detailedStatusMessage"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["options"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["required"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:networkcloud/v20240701:KubernetesClusterFeature" }, { type: "azure-native:networkcloud/v20241001preview:KubernetesClusterFeature" }, { type: "azure-native:networkcloud/v20250201:KubernetesClusterFeature" }, { type: "azure-native:networkcloud:KubernetesClusterFeature" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KubernetesClusterFeature.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KubernetesClusterFeature = KubernetesClusterFeature;
/** @internal */
KubernetesClusterFeature.__pulumiType = 'azure-native:networkcloud/v20240601preview:KubernetesClusterFeature';
