// @pulumi/azure-native/insights/v20230311/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDataCollectionRuleAssociationOutput = exports.getDataCollectionRuleAssociation = exports.getDataCollectionRuleOutput = exports.getDataCollectionRule = exports.getDataCollectionEndpointOutput = exports.getDataCollectionEndpoint = exports.DataCollectionRuleAssociation = exports.DataCollectionRule = exports.DataCollectionEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DataCollectionEndpoint = null;
utilities.lazyLoad(exports, ["DataCollectionEndpoint"], () => require("./dataCollectionEndpoint"));
exports.DataCollectionRule = null;
utilities.lazyLoad(exports, ["DataCollectionRule"], () => require("./dataCollectionRule"));
exports.DataCollectionRuleAssociation = null;
utilities.lazyLoad(exports, ["DataCollectionRuleAssociation"], () => require("./dataCollectionRuleAssociation"));
exports.getDataCollectionEndpoint = null;
exports.getDataCollectionEndpointOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionEndpoint", "getDataCollectionEndpointOutput"], () => require("./getDataCollectionEndpoint"));
exports.getDataCollectionRule = null;
exports.getDataCollectionRuleOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionRule", "getDataCollectionRuleOutput"], () => require("./getDataCollectionRule"));
exports.getDataCollectionRuleAssociation = null;
exports.getDataCollectionRuleAssociationOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionRuleAssociation", "getDataCollectionRuleAssociationOutput"], () => require("./getDataCollectionRuleAssociation"));
// Export enums:
__exportStar(require("../../types/enums/insights/v20230311"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights/v20230311:DataCollectionEndpoint":
                return new exports.DataCollectionEndpoint(name, undefined, { urn });
            case "azure-native:insights/v20230311:DataCollectionRule":
                return new exports.DataCollectionRule(name, undefined, { urn });
            case "azure-native:insights/v20230311:DataCollectionRuleAssociation":
                return new exports.DataCollectionRuleAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights/v20230311", _module);
