// @pulumi/azure-native/quota/v20241218preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupQuotaSubscription = exports.GroupQuota = exports.getGroupQuotaSubscriptionOutput = exports.getGroupQuotaSubscription = exports.getGroupQuotaOutput = exports.getGroupQuota = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getGroupQuota = null;
exports.getGroupQuotaOutput = null;
utilities.lazyLoad(exports, ["getGroupQuota", "getGroupQuotaOutput"], () => require("./getGroupQuota"));
exports.getGroupQuotaSubscription = null;
exports.getGroupQuotaSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getGroupQuotaSubscription", "getGroupQuotaSubscriptionOutput"], () => require("./getGroupQuotaSubscription"));
exports.GroupQuota = null;
utilities.lazyLoad(exports, ["GroupQuota"], () => require("./groupQuota"));
exports.GroupQuotaSubscription = null;
utilities.lazyLoad(exports, ["GroupQuotaSubscription"], () => require("./groupQuotaSubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:quota/v20241218preview:GroupQuota":
                return new exports.GroupQuota(name, undefined, { urn });
            case "azure-native:quota/v20241218preview:GroupQuotaSubscription":
                return new exports.GroupQuotaSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "quota/v20241218preview", _module);
