// @pulumi/azure-native/vmwarecloudsimple/v20190401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachine = exports.getVirtualMachineOutput = exports.getVirtualMachine = exports.getDedicatedCloudServiceOutput = exports.getDedicatedCloudService = exports.getDedicatedCloudNodeOutput = exports.getDedicatedCloudNode = exports.DedicatedCloudService = exports.DedicatedCloudNode = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.DedicatedCloudNode = null;
utilities.lazyLoad(exports, ["DedicatedCloudNode"], () => require("./dedicatedCloudNode"));
exports.DedicatedCloudService = null;
utilities.lazyLoad(exports, ["DedicatedCloudService"], () => require("./dedicatedCloudService"));
exports.getDedicatedCloudNode = null;
exports.getDedicatedCloudNodeOutput = null;
utilities.lazyLoad(exports, ["getDedicatedCloudNode", "getDedicatedCloudNodeOutput"], () => require("./getDedicatedCloudNode"));
exports.getDedicatedCloudService = null;
exports.getDedicatedCloudServiceOutput = null;
utilities.lazyLoad(exports, ["getDedicatedCloudService", "getDedicatedCloudServiceOutput"], () => require("./getDedicatedCloudService"));
exports.getVirtualMachine = null;
exports.getVirtualMachineOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachine", "getVirtualMachineOutput"], () => require("./getVirtualMachine"));
exports.VirtualMachine = null;
utilities.lazyLoad(exports, ["VirtualMachine"], () => require("./virtualMachine"));
// Export enums:
__exportStar(require("../../types/enums/vmwarecloudsimple/v20190401"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:vmwarecloudsimple/v20190401:DedicatedCloudNode":
                return new exports.DedicatedCloudNode(name, undefined, { urn });
            case "azure-native:vmwarecloudsimple/v20190401:DedicatedCloudService":
                return new exports.DedicatedCloudService(name, undefined, { urn });
            case "azure-native:vmwarecloudsimple/v20190401:VirtualMachine":
                return new exports.VirtualMachine(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "vmwarecloudsimple/v20190401", _module);
