// @pulumi/azure-native/cdn/v20240722preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getEdgeActionVersionCodeOutput = exports.getEdgeActionVersionCode = exports.getEdgeActionVersionOutput = exports.getEdgeActionVersion = exports.getEdgeActionExecutionFilterOutput = exports.getEdgeActionExecutionFilter = exports.getEdgeActionOutput = exports.getEdgeAction = exports.EdgeActionVersion = exports.EdgeActionExecutionFilter = exports.EdgeAction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.EdgeAction = null;
utilities.lazyLoad(exports, ["EdgeAction"], () => require("./edgeAction"));
exports.EdgeActionExecutionFilter = null;
utilities.lazyLoad(exports, ["EdgeActionExecutionFilter"], () => require("./edgeActionExecutionFilter"));
exports.EdgeActionVersion = null;
utilities.lazyLoad(exports, ["EdgeActionVersion"], () => require("./edgeActionVersion"));
exports.getEdgeAction = null;
exports.getEdgeActionOutput = null;
utilities.lazyLoad(exports, ["getEdgeAction", "getEdgeActionOutput"], () => require("./getEdgeAction"));
exports.getEdgeActionExecutionFilter = null;
exports.getEdgeActionExecutionFilterOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionExecutionFilter", "getEdgeActionExecutionFilterOutput"], () => require("./getEdgeActionExecutionFilter"));
exports.getEdgeActionVersion = null;
exports.getEdgeActionVersionOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionVersion", "getEdgeActionVersionOutput"], () => require("./getEdgeActionVersion"));
exports.getEdgeActionVersionCode = null;
exports.getEdgeActionVersionCodeOutput = null;
utilities.lazyLoad(exports, ["getEdgeActionVersionCode", "getEdgeActionVersionCodeOutput"], () => require("./getEdgeActionVersionCode"));
// Export enums:
__exportStar(require("../../types/enums/cdn/v20240722preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cdn/v20240722preview:EdgeAction":
                return new exports.EdgeAction(name, undefined, { urn });
            case "azure-native:cdn/v20240722preview:EdgeActionExecutionFilter":
                return new exports.EdgeActionExecutionFilter(name, undefined, { urn });
            case "azure-native:cdn/v20240722preview:EdgeActionVersion":
                return new exports.EdgeActionVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cdn/v20240722preview", _module);
