// @pulumi/azure-native/apimanagement/v20220801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getApiOperationPolicyOutput = exports.getApiOperationPolicy = exports.getApiOperationOutput = exports.getApiOperation = exports.getApiManagementServiceSsoTokenOutput = exports.getApiManagementServiceSsoToken = exports.getApiManagementServiceDomainOwnershipIdentifierOutput = exports.getApiManagementServiceDomainOwnershipIdentifier = exports.getApiManagementServiceOutput = exports.getApiManagementService = exports.getApiIssueCommentOutput = exports.getApiIssueComment = exports.getApiIssueAttachmentOutput = exports.getApiIssueAttachment = exports.getApiIssueOutput = exports.getApiIssue = exports.getApiDiagnosticOutput = exports.getApiDiagnostic = exports.getApiOutput = exports.getApi = exports.GatewayHostnameConfiguration = exports.GatewayCertificateAuthority = exports.GatewayApiEntityTag = exports.Gateway = exports.EmailTemplate = exports.Documentation = exports.Diagnostic = exports.ContentType = exports.ContentItem = exports.Certificate = exports.Cache = exports.Backend = exports.AuthorizationServer = exports.AuthorizationProvider = exports.AuthorizationAccessPolicy = exports.Authorization = exports.ApiWiki = exports.ApiVersionSet = exports.ApiTagDescription = exports.ApiSchema = exports.ApiRelease = exports.ApiPolicy = exports.ApiOperationPolicy = exports.ApiOperation = exports.ApiManagementService = exports.ApiIssueComment = exports.ApiIssueAttachment = exports.ApiIssue = exports.ApiDiagnostic = exports.Api = void 0;
exports.getGraphQLApiResolverPolicyOutput = exports.getGraphQLApiResolverPolicy = exports.getGraphQLApiResolverOutput = exports.getGraphQLApiResolver = exports.getGlobalSchemaOutput = exports.getGlobalSchema = exports.getGatewayHostnameConfigurationOutput = exports.getGatewayHostnameConfiguration = exports.getGatewayCertificateAuthorityOutput = exports.getGatewayCertificateAuthority = exports.getGatewayOutput = exports.getGateway = exports.getEmailTemplateOutput = exports.getEmailTemplate = exports.getDocumentationOutput = exports.getDocumentation = exports.getDiagnosticOutput = exports.getDiagnostic = exports.getContentTypeOutput = exports.getContentType = exports.getContentItemOutput = exports.getContentItem = exports.getCertificateOutput = exports.getCertificate = exports.getCacheOutput = exports.getCache = exports.getBackendOutput = exports.getBackend = exports.getAuthorizationServerOutput = exports.getAuthorizationServer = exports.getAuthorizationProviderOutput = exports.getAuthorizationProvider = exports.getAuthorizationLoginLinkPostOutput = exports.getAuthorizationLoginLinkPost = exports.getAuthorizationAccessPolicyOutput = exports.getAuthorizationAccessPolicy = exports.getAuthorizationOutput = exports.getAuthorization = exports.getApiWikiOutput = exports.getApiWiki = exports.getApiVersionSetOutput = exports.getApiVersionSet = exports.getApiTagDescriptionOutput = exports.getApiTagDescription = exports.getApiSchemaOutput = exports.getApiSchema = exports.getApiReleaseOutput = exports.getApiRelease = exports.getApiPolicyOutput = exports.getApiPolicy = void 0;
exports.listNamedValueOutput = exports.listNamedValue = exports.listIdentityProviderSecretsOutput = exports.listIdentityProviderSecrets = exports.listGatewayKeysOutput = exports.listGatewayKeys = exports.listAuthorizationServerSecretsOutput = exports.listAuthorizationServerSecrets = exports.IdentityProvider = exports.GroupUser = exports.Group = exports.GraphQLApiResolverPolicy = exports.GraphQLApiResolver = exports.GlobalSchema = exports.getUserSharedAccessTokenOutput = exports.getUserSharedAccessToken = exports.getUserOutput = exports.getUser = exports.getTagByProductOutput = exports.getTagByProduct = exports.getTagByOperationOutput = exports.getTagByOperation = exports.getTagByApiOutput = exports.getTagByApi = exports.getTagOutput = exports.getTag = exports.getSubscriptionOutput = exports.getSubscription = exports.getProductWikiOutput = exports.getProductWiki = exports.getProductPolicyOutput = exports.getProductPolicy = exports.getProductOutput = exports.getProduct = exports.getPrivateEndpointConnectionByNameOutput = exports.getPrivateEndpointConnectionByName = exports.getPolicyFragmentOutput = exports.getPolicyFragment = exports.getPolicyOutput = exports.getPolicy = exports.getOpenIdConnectProviderOutput = exports.getOpenIdConnectProvider = exports.getNamedValueOutput = exports.getNamedValue = exports.getLoggerOutput = exports.getLogger = exports.getIdentityProviderOutput = exports.getIdentityProvider = exports.getGroupOutput = exports.getGroup = void 0;
exports.User = exports.TagByProduct = exports.TagByOperation = exports.TagByApi = exports.Tag = exports.Subscription = exports.ProductWiki = exports.ProductPolicy = exports.ProductGroup = exports.ProductApi = exports.Product = exports.PrivateEndpointConnectionByName = exports.PolicyFragment = exports.Policy = exports.OpenIdConnectProvider = exports.NotificationRecipientUser = exports.NotificationRecipientEmail = exports.NamedValue = exports.Logger = exports.listTenantAccessSecretsOutput = exports.listTenantAccessSecrets = exports.listSubscriptionSecretsOutput = exports.listSubscriptionSecrets = exports.listPolicyFragmentReferencesOutput = exports.listPolicyFragmentReferences = exports.listOpenIdConnectProviderSecretsOutput = exports.listOpenIdConnectProviderSecrets = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ApiDiagnostic = null;
utilities.lazyLoad(exports, ["ApiDiagnostic"], () => require("./apiDiagnostic"));
exports.ApiIssue = null;
utilities.lazyLoad(exports, ["ApiIssue"], () => require("./apiIssue"));
exports.ApiIssueAttachment = null;
utilities.lazyLoad(exports, ["ApiIssueAttachment"], () => require("./apiIssueAttachment"));
exports.ApiIssueComment = null;
utilities.lazyLoad(exports, ["ApiIssueComment"], () => require("./apiIssueComment"));
exports.ApiManagementService = null;
utilities.lazyLoad(exports, ["ApiManagementService"], () => require("./apiManagementService"));
exports.ApiOperation = null;
utilities.lazyLoad(exports, ["ApiOperation"], () => require("./apiOperation"));
exports.ApiOperationPolicy = null;
utilities.lazyLoad(exports, ["ApiOperationPolicy"], () => require("./apiOperationPolicy"));
exports.ApiPolicy = null;
utilities.lazyLoad(exports, ["ApiPolicy"], () => require("./apiPolicy"));
exports.ApiRelease = null;
utilities.lazyLoad(exports, ["ApiRelease"], () => require("./apiRelease"));
exports.ApiSchema = null;
utilities.lazyLoad(exports, ["ApiSchema"], () => require("./apiSchema"));
exports.ApiTagDescription = null;
utilities.lazyLoad(exports, ["ApiTagDescription"], () => require("./apiTagDescription"));
exports.ApiVersionSet = null;
utilities.lazyLoad(exports, ["ApiVersionSet"], () => require("./apiVersionSet"));
exports.ApiWiki = null;
utilities.lazyLoad(exports, ["ApiWiki"], () => require("./apiWiki"));
exports.Authorization = null;
utilities.lazyLoad(exports, ["Authorization"], () => require("./authorization"));
exports.AuthorizationAccessPolicy = null;
utilities.lazyLoad(exports, ["AuthorizationAccessPolicy"], () => require("./authorizationAccessPolicy"));
exports.AuthorizationProvider = null;
utilities.lazyLoad(exports, ["AuthorizationProvider"], () => require("./authorizationProvider"));
exports.AuthorizationServer = null;
utilities.lazyLoad(exports, ["AuthorizationServer"], () => require("./authorizationServer"));
exports.Backend = null;
utilities.lazyLoad(exports, ["Backend"], () => require("./backend"));
exports.Cache = null;
utilities.lazyLoad(exports, ["Cache"], () => require("./cache"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.ContentItem = null;
utilities.lazyLoad(exports, ["ContentItem"], () => require("./contentItem"));
exports.ContentType = null;
utilities.lazyLoad(exports, ["ContentType"], () => require("./contentType"));
exports.Diagnostic = null;
utilities.lazyLoad(exports, ["Diagnostic"], () => require("./diagnostic"));
exports.Documentation = null;
utilities.lazyLoad(exports, ["Documentation"], () => require("./documentation"));
exports.EmailTemplate = null;
utilities.lazyLoad(exports, ["EmailTemplate"], () => require("./emailTemplate"));
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.GatewayApiEntityTag = null;
utilities.lazyLoad(exports, ["GatewayApiEntityTag"], () => require("./gatewayApiEntityTag"));
exports.GatewayCertificateAuthority = null;
utilities.lazyLoad(exports, ["GatewayCertificateAuthority"], () => require("./gatewayCertificateAuthority"));
exports.GatewayHostnameConfiguration = null;
utilities.lazyLoad(exports, ["GatewayHostnameConfiguration"], () => require("./gatewayHostnameConfiguration"));
exports.getApi = null;
exports.getApiOutput = null;
utilities.lazyLoad(exports, ["getApi", "getApiOutput"], () => require("./getApi"));
exports.getApiDiagnostic = null;
exports.getApiDiagnosticOutput = null;
utilities.lazyLoad(exports, ["getApiDiagnostic", "getApiDiagnosticOutput"], () => require("./getApiDiagnostic"));
exports.getApiIssue = null;
exports.getApiIssueOutput = null;
utilities.lazyLoad(exports, ["getApiIssue", "getApiIssueOutput"], () => require("./getApiIssue"));
exports.getApiIssueAttachment = null;
exports.getApiIssueAttachmentOutput = null;
utilities.lazyLoad(exports, ["getApiIssueAttachment", "getApiIssueAttachmentOutput"], () => require("./getApiIssueAttachment"));
exports.getApiIssueComment = null;
exports.getApiIssueCommentOutput = null;
utilities.lazyLoad(exports, ["getApiIssueComment", "getApiIssueCommentOutput"], () => require("./getApiIssueComment"));
exports.getApiManagementService = null;
exports.getApiManagementServiceOutput = null;
utilities.lazyLoad(exports, ["getApiManagementService", "getApiManagementServiceOutput"], () => require("./getApiManagementService"));
exports.getApiManagementServiceDomainOwnershipIdentifier = null;
exports.getApiManagementServiceDomainOwnershipIdentifierOutput = null;
utilities.lazyLoad(exports, ["getApiManagementServiceDomainOwnershipIdentifier", "getApiManagementServiceDomainOwnershipIdentifierOutput"], () => require("./getApiManagementServiceDomainOwnershipIdentifier"));
exports.getApiManagementServiceSsoToken = null;
exports.getApiManagementServiceSsoTokenOutput = null;
utilities.lazyLoad(exports, ["getApiManagementServiceSsoToken", "getApiManagementServiceSsoTokenOutput"], () => require("./getApiManagementServiceSsoToken"));
exports.getApiOperation = null;
exports.getApiOperationOutput = null;
utilities.lazyLoad(exports, ["getApiOperation", "getApiOperationOutput"], () => require("./getApiOperation"));
exports.getApiOperationPolicy = null;
exports.getApiOperationPolicyOutput = null;
utilities.lazyLoad(exports, ["getApiOperationPolicy", "getApiOperationPolicyOutput"], () => require("./getApiOperationPolicy"));
exports.getApiPolicy = null;
exports.getApiPolicyOutput = null;
utilities.lazyLoad(exports, ["getApiPolicy", "getApiPolicyOutput"], () => require("./getApiPolicy"));
exports.getApiRelease = null;
exports.getApiReleaseOutput = null;
utilities.lazyLoad(exports, ["getApiRelease", "getApiReleaseOutput"], () => require("./getApiRelease"));
exports.getApiSchema = null;
exports.getApiSchemaOutput = null;
utilities.lazyLoad(exports, ["getApiSchema", "getApiSchemaOutput"], () => require("./getApiSchema"));
exports.getApiTagDescription = null;
exports.getApiTagDescriptionOutput = null;
utilities.lazyLoad(exports, ["getApiTagDescription", "getApiTagDescriptionOutput"], () => require("./getApiTagDescription"));
exports.getApiVersionSet = null;
exports.getApiVersionSetOutput = null;
utilities.lazyLoad(exports, ["getApiVersionSet", "getApiVersionSetOutput"], () => require("./getApiVersionSet"));
exports.getApiWiki = null;
exports.getApiWikiOutput = null;
utilities.lazyLoad(exports, ["getApiWiki", "getApiWikiOutput"], () => require("./getApiWiki"));
exports.getAuthorization = null;
exports.getAuthorizationOutput = null;
utilities.lazyLoad(exports, ["getAuthorization", "getAuthorizationOutput"], () => require("./getAuthorization"));
exports.getAuthorizationAccessPolicy = null;
exports.getAuthorizationAccessPolicyOutput = null;
utilities.lazyLoad(exports, ["getAuthorizationAccessPolicy", "getAuthorizationAccessPolicyOutput"], () => require("./getAuthorizationAccessPolicy"));
exports.getAuthorizationLoginLinkPost = null;
exports.getAuthorizationLoginLinkPostOutput = null;
utilities.lazyLoad(exports, ["getAuthorizationLoginLinkPost", "getAuthorizationLoginLinkPostOutput"], () => require("./getAuthorizationLoginLinkPost"));
exports.getAuthorizationProvider = null;
exports.getAuthorizationProviderOutput = null;
utilities.lazyLoad(exports, ["getAuthorizationProvider", "getAuthorizationProviderOutput"], () => require("./getAuthorizationProvider"));
exports.getAuthorizationServer = null;
exports.getAuthorizationServerOutput = null;
utilities.lazyLoad(exports, ["getAuthorizationServer", "getAuthorizationServerOutput"], () => require("./getAuthorizationServer"));
exports.getBackend = null;
exports.getBackendOutput = null;
utilities.lazyLoad(exports, ["getBackend", "getBackendOutput"], () => require("./getBackend"));
exports.getCache = null;
exports.getCacheOutput = null;
utilities.lazyLoad(exports, ["getCache", "getCacheOutput"], () => require("./getCache"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getContentItem = null;
exports.getContentItemOutput = null;
utilities.lazyLoad(exports, ["getContentItem", "getContentItemOutput"], () => require("./getContentItem"));
exports.getContentType = null;
exports.getContentTypeOutput = null;
utilities.lazyLoad(exports, ["getContentType", "getContentTypeOutput"], () => require("./getContentType"));
exports.getDiagnostic = null;
exports.getDiagnosticOutput = null;
utilities.lazyLoad(exports, ["getDiagnostic", "getDiagnosticOutput"], () => require("./getDiagnostic"));
exports.getDocumentation = null;
exports.getDocumentationOutput = null;
utilities.lazyLoad(exports, ["getDocumentation", "getDocumentationOutput"], () => require("./getDocumentation"));
exports.getEmailTemplate = null;
exports.getEmailTemplateOutput = null;
utilities.lazyLoad(exports, ["getEmailTemplate", "getEmailTemplateOutput"], () => require("./getEmailTemplate"));
exports.getGateway = null;
exports.getGatewayOutput = null;
utilities.lazyLoad(exports, ["getGateway", "getGatewayOutput"], () => require("./getGateway"));
exports.getGatewayCertificateAuthority = null;
exports.getGatewayCertificateAuthorityOutput = null;
utilities.lazyLoad(exports, ["getGatewayCertificateAuthority", "getGatewayCertificateAuthorityOutput"], () => require("./getGatewayCertificateAuthority"));
exports.getGatewayHostnameConfiguration = null;
exports.getGatewayHostnameConfigurationOutput = null;
utilities.lazyLoad(exports, ["getGatewayHostnameConfiguration", "getGatewayHostnameConfigurationOutput"], () => require("./getGatewayHostnameConfiguration"));
exports.getGlobalSchema = null;
exports.getGlobalSchemaOutput = null;
utilities.lazyLoad(exports, ["getGlobalSchema", "getGlobalSchemaOutput"], () => require("./getGlobalSchema"));
exports.getGraphQLApiResolver = null;
exports.getGraphQLApiResolverOutput = null;
utilities.lazyLoad(exports, ["getGraphQLApiResolver", "getGraphQLApiResolverOutput"], () => require("./getGraphQLApiResolver"));
exports.getGraphQLApiResolverPolicy = null;
exports.getGraphQLApiResolverPolicyOutput = null;
utilities.lazyLoad(exports, ["getGraphQLApiResolverPolicy", "getGraphQLApiResolverPolicyOutput"], () => require("./getGraphQLApiResolverPolicy"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getIdentityProvider = null;
exports.getIdentityProviderOutput = null;
utilities.lazyLoad(exports, ["getIdentityProvider", "getIdentityProviderOutput"], () => require("./getIdentityProvider"));
exports.getLogger = null;
exports.getLoggerOutput = null;
utilities.lazyLoad(exports, ["getLogger", "getLoggerOutput"], () => require("./getLogger"));
exports.getNamedValue = null;
exports.getNamedValueOutput = null;
utilities.lazyLoad(exports, ["getNamedValue", "getNamedValueOutput"], () => require("./getNamedValue"));
exports.getOpenIdConnectProvider = null;
exports.getOpenIdConnectProviderOutput = null;
utilities.lazyLoad(exports, ["getOpenIdConnectProvider", "getOpenIdConnectProviderOutput"], () => require("./getOpenIdConnectProvider"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getPolicyFragment = null;
exports.getPolicyFragmentOutput = null;
utilities.lazyLoad(exports, ["getPolicyFragment", "getPolicyFragmentOutput"], () => require("./getPolicyFragment"));
exports.getPrivateEndpointConnectionByName = null;
exports.getPrivateEndpointConnectionByNameOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionByName", "getPrivateEndpointConnectionByNameOutput"], () => require("./getPrivateEndpointConnectionByName"));
exports.getProduct = null;
exports.getProductOutput = null;
utilities.lazyLoad(exports, ["getProduct", "getProductOutput"], () => require("./getProduct"));
exports.getProductPolicy = null;
exports.getProductPolicyOutput = null;
utilities.lazyLoad(exports, ["getProductPolicy", "getProductPolicyOutput"], () => require("./getProductPolicy"));
exports.getProductWiki = null;
exports.getProductWikiOutput = null;
utilities.lazyLoad(exports, ["getProductWiki", "getProductWikiOutput"], () => require("./getProductWiki"));
exports.getSubscription = null;
exports.getSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSubscription", "getSubscriptionOutput"], () => require("./getSubscription"));
exports.getTag = null;
exports.getTagOutput = null;
utilities.lazyLoad(exports, ["getTag", "getTagOutput"], () => require("./getTag"));
exports.getTagByApi = null;
exports.getTagByApiOutput = null;
utilities.lazyLoad(exports, ["getTagByApi", "getTagByApiOutput"], () => require("./getTagByApi"));
exports.getTagByOperation = null;
exports.getTagByOperationOutput = null;
utilities.lazyLoad(exports, ["getTagByOperation", "getTagByOperationOutput"], () => require("./getTagByOperation"));
exports.getTagByProduct = null;
exports.getTagByProductOutput = null;
utilities.lazyLoad(exports, ["getTagByProduct", "getTagByProductOutput"], () => require("./getTagByProduct"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.getUserSharedAccessToken = null;
exports.getUserSharedAccessTokenOutput = null;
utilities.lazyLoad(exports, ["getUserSharedAccessToken", "getUserSharedAccessTokenOutput"], () => require("./getUserSharedAccessToken"));
exports.GlobalSchema = null;
utilities.lazyLoad(exports, ["GlobalSchema"], () => require("./globalSchema"));
exports.GraphQLApiResolver = null;
utilities.lazyLoad(exports, ["GraphQLApiResolver"], () => require("./graphQLApiResolver"));
exports.GraphQLApiResolverPolicy = null;
utilities.lazyLoad(exports, ["GraphQLApiResolverPolicy"], () => require("./graphQLApiResolverPolicy"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.GroupUser = null;
utilities.lazyLoad(exports, ["GroupUser"], () => require("./groupUser"));
exports.IdentityProvider = null;
utilities.lazyLoad(exports, ["IdentityProvider"], () => require("./identityProvider"));
exports.listAuthorizationServerSecrets = null;
exports.listAuthorizationServerSecretsOutput = null;
utilities.lazyLoad(exports, ["listAuthorizationServerSecrets", "listAuthorizationServerSecretsOutput"], () => require("./listAuthorizationServerSecrets"));
exports.listGatewayKeys = null;
exports.listGatewayKeysOutput = null;
utilities.lazyLoad(exports, ["listGatewayKeys", "listGatewayKeysOutput"], () => require("./listGatewayKeys"));
exports.listIdentityProviderSecrets = null;
exports.listIdentityProviderSecretsOutput = null;
utilities.lazyLoad(exports, ["listIdentityProviderSecrets", "listIdentityProviderSecretsOutput"], () => require("./listIdentityProviderSecrets"));
exports.listNamedValue = null;
exports.listNamedValueOutput = null;
utilities.lazyLoad(exports, ["listNamedValue", "listNamedValueOutput"], () => require("./listNamedValue"));
exports.listOpenIdConnectProviderSecrets = null;
exports.listOpenIdConnectProviderSecretsOutput = null;
utilities.lazyLoad(exports, ["listOpenIdConnectProviderSecrets", "listOpenIdConnectProviderSecretsOutput"], () => require("./listOpenIdConnectProviderSecrets"));
exports.listPolicyFragmentReferences = null;
exports.listPolicyFragmentReferencesOutput = null;
utilities.lazyLoad(exports, ["listPolicyFragmentReferences", "listPolicyFragmentReferencesOutput"], () => require("./listPolicyFragmentReferences"));
exports.listSubscriptionSecrets = null;
exports.listSubscriptionSecretsOutput = null;
utilities.lazyLoad(exports, ["listSubscriptionSecrets", "listSubscriptionSecretsOutput"], () => require("./listSubscriptionSecrets"));
exports.listTenantAccessSecrets = null;
exports.listTenantAccessSecretsOutput = null;
utilities.lazyLoad(exports, ["listTenantAccessSecrets", "listTenantAccessSecretsOutput"], () => require("./listTenantAccessSecrets"));
exports.Logger = null;
utilities.lazyLoad(exports, ["Logger"], () => require("./logger"));
exports.NamedValue = null;
utilities.lazyLoad(exports, ["NamedValue"], () => require("./namedValue"));
exports.NotificationRecipientEmail = null;
utilities.lazyLoad(exports, ["NotificationRecipientEmail"], () => require("./notificationRecipientEmail"));
exports.NotificationRecipientUser = null;
utilities.lazyLoad(exports, ["NotificationRecipientUser"], () => require("./notificationRecipientUser"));
exports.OpenIdConnectProvider = null;
utilities.lazyLoad(exports, ["OpenIdConnectProvider"], () => require("./openIdConnectProvider"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.PolicyFragment = null;
utilities.lazyLoad(exports, ["PolicyFragment"], () => require("./policyFragment"));
exports.PrivateEndpointConnectionByName = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionByName"], () => require("./privateEndpointConnectionByName"));
exports.Product = null;
utilities.lazyLoad(exports, ["Product"], () => require("./product"));
exports.ProductApi = null;
utilities.lazyLoad(exports, ["ProductApi"], () => require("./productApi"));
exports.ProductGroup = null;
utilities.lazyLoad(exports, ["ProductGroup"], () => require("./productGroup"));
exports.ProductPolicy = null;
utilities.lazyLoad(exports, ["ProductPolicy"], () => require("./productPolicy"));
exports.ProductWiki = null;
utilities.lazyLoad(exports, ["ProductWiki"], () => require("./productWiki"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
exports.Tag = null;
utilities.lazyLoad(exports, ["Tag"], () => require("./tag"));
exports.TagByApi = null;
utilities.lazyLoad(exports, ["TagByApi"], () => require("./tagByApi"));
exports.TagByOperation = null;
utilities.lazyLoad(exports, ["TagByOperation"], () => require("./tagByOperation"));
exports.TagByProduct = null;
utilities.lazyLoad(exports, ["TagByProduct"], () => require("./tagByProduct"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20220801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20220801:Api":
                return new exports.Api(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiDiagnostic":
                return new exports.ApiDiagnostic(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiIssue":
                return new exports.ApiIssue(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiIssueAttachment":
                return new exports.ApiIssueAttachment(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiIssueComment":
                return new exports.ApiIssueComment(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiManagementService":
                return new exports.ApiManagementService(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiOperation":
                return new exports.ApiOperation(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiOperationPolicy":
                return new exports.ApiOperationPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiPolicy":
                return new exports.ApiPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiRelease":
                return new exports.ApiRelease(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiSchema":
                return new exports.ApiSchema(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiTagDescription":
                return new exports.ApiTagDescription(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiVersionSet":
                return new exports.ApiVersionSet(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ApiWiki":
                return new exports.ApiWiki(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Authorization":
                return new exports.Authorization(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:AuthorizationAccessPolicy":
                return new exports.AuthorizationAccessPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:AuthorizationProvider":
                return new exports.AuthorizationProvider(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:AuthorizationServer":
                return new exports.AuthorizationServer(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Backend":
                return new exports.Backend(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Cache":
                return new exports.Cache(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ContentItem":
                return new exports.ContentItem(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ContentType":
                return new exports.ContentType(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Diagnostic":
                return new exports.Diagnostic(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Documentation":
                return new exports.Documentation(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:EmailTemplate":
                return new exports.EmailTemplate(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GatewayApiEntityTag":
                return new exports.GatewayApiEntityTag(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GatewayCertificateAuthority":
                return new exports.GatewayCertificateAuthority(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GatewayHostnameConfiguration":
                return new exports.GatewayHostnameConfiguration(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GlobalSchema":
                return new exports.GlobalSchema(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GraphQLApiResolver":
                return new exports.GraphQLApiResolver(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GraphQLApiResolverPolicy":
                return new exports.GraphQLApiResolverPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Group":
                return new exports.Group(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:GroupUser":
                return new exports.GroupUser(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:IdentityProvider":
                return new exports.IdentityProvider(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Logger":
                return new exports.Logger(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:NamedValue":
                return new exports.NamedValue(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:NotificationRecipientEmail":
                return new exports.NotificationRecipientEmail(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:NotificationRecipientUser":
                return new exports.NotificationRecipientUser(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:OpenIdConnectProvider":
                return new exports.OpenIdConnectProvider(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:PolicyFragment":
                return new exports.PolicyFragment(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:PrivateEndpointConnectionByName":
                return new exports.PrivateEndpointConnectionByName(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Product":
                return new exports.Product(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ProductApi":
                return new exports.ProductApi(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ProductGroup":
                return new exports.ProductGroup(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ProductPolicy":
                return new exports.ProductPolicy(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:ProductWiki":
                return new exports.ProductWiki(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:Tag":
                return new exports.Tag(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:TagByApi":
                return new exports.TagByApi(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:TagByOperation":
                return new exports.TagByOperation(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:TagByProduct":
                return new exports.TagByProduct(name, undefined, { urn });
            case "azure-native:apimanagement/v20220801:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20220801", _module);
