// @pulumi/azure-native/containerregistry/v20190601preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskRun = exports.Task = exports.listTaskRunDetailsOutput = exports.listTaskRunDetails = exports.listTaskDetailsOutput = exports.listTaskDetails = exports.listRunLogSasUrlOutput = exports.listRunLogSasUrl = exports.listRegistryBuildSourceUploadUrlOutput = exports.listRegistryBuildSourceUploadUrl = exports.listAgentPoolQueueStatusOutput = exports.listAgentPoolQueueStatus = exports.getTaskRunOutput = exports.getTaskRun = exports.getTaskOutput = exports.getTask = exports.getAgentPoolOutput = exports.getAgentPool = exports.AgentPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getTask = null;
exports.getTaskOutput = null;
utilities.lazyLoad(exports, ["getTask", "getTaskOutput"], () => require("./getTask"));
exports.getTaskRun = null;
exports.getTaskRunOutput = null;
utilities.lazyLoad(exports, ["getTaskRun", "getTaskRunOutput"], () => require("./getTaskRun"));
exports.listAgentPoolQueueStatus = null;
exports.listAgentPoolQueueStatusOutput = null;
utilities.lazyLoad(exports, ["listAgentPoolQueueStatus", "listAgentPoolQueueStatusOutput"], () => require("./listAgentPoolQueueStatus"));
exports.listRegistryBuildSourceUploadUrl = null;
exports.listRegistryBuildSourceUploadUrlOutput = null;
utilities.lazyLoad(exports, ["listRegistryBuildSourceUploadUrl", "listRegistryBuildSourceUploadUrlOutput"], () => require("./listRegistryBuildSourceUploadUrl"));
exports.listRunLogSasUrl = null;
exports.listRunLogSasUrlOutput = null;
utilities.lazyLoad(exports, ["listRunLogSasUrl", "listRunLogSasUrlOutput"], () => require("./listRunLogSasUrl"));
exports.listTaskDetails = null;
exports.listTaskDetailsOutput = null;
utilities.lazyLoad(exports, ["listTaskDetails", "listTaskDetailsOutput"], () => require("./listTaskDetails"));
exports.listTaskRunDetails = null;
exports.listTaskRunDetailsOutput = null;
utilities.lazyLoad(exports, ["listTaskRunDetails", "listTaskRunDetailsOutput"], () => require("./listTaskRunDetails"));
exports.Task = null;
utilities.lazyLoad(exports, ["Task"], () => require("./task"));
exports.TaskRun = null;
utilities.lazyLoad(exports, ["TaskRun"], () => require("./taskRun"));
// Export enums:
__exportStar(require("../../types/enums/containerregistry/v20190601preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerregistry/v20190601preview:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:containerregistry/v20190601preview:Task":
                return new exports.Task(name, undefined, { urn });
            case "azure-native:containerregistry/v20190601preview:TaskRun":
                return new exports.TaskRun(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerregistry/v20190601preview", _module);
