// @pulumi/azure-native/types/enums/azurestackhci/v20240101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsServerSubscription = exports.VmSizeEnum = exports.UpdateSummariesPropertiesState = exports.UpdateRunPropertiesState = exports.State = exports.SoftwareAssuranceIntent = exports.ServiceName = exports.SecurityTypes = exports.ResourceIdentityType = exports.ProvisioningAction = exports.OperatingSystemTypes = exports.ManagedServiceIdentityType = exports.IpAllocationMethodEnum = exports.IPPoolTypeEnum = exports.HyperVGeneration = exports.ExtendedLocationTypes = exports.DiskFileFormat = exports.DiagnosticLevel = exports.DeploymentMode = exports.ComplianceAssignmentType = exports.CloudInitDataSource = exports.AvailabilityType = void 0;
exports.AvailabilityType = {
    Local: "Local",
    Online: "Online",
    Notify: "Notify",
};
exports.CloudInitDataSource = {
    NoCloud: "NoCloud",
    Azure: "Azure",
};
exports.ComplianceAssignmentType = {
    /**
     * Report on the state of the machine, but don't make changes.
     */
    Audit: "Audit",
    /**
     * Applied to the machine. If it drifts, the local service inside the machine makes a correction at the next evaluation.
     */
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
};
exports.DeploymentMode = {
    /**
     * Validate deployment settings for cluster.
     */
    Validate: "Validate",
    /**
     * Deploy cluster using deployment settings.
     */
    Deploy: "Deploy",
};
exports.DiagnosticLevel = {
    Off: "Off",
    Basic: "Basic",
    Enhanced: "Enhanced",
};
exports.DiskFileFormat = {
    Vhdx: "vhdx",
    Vhd: "vhd",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.HyperVGeneration = {
    V1: "V1",
    V2: "V2",
};
exports.IPPoolTypeEnum = {
    Vm: "vm",
    Vippool: "vippool",
};
exports.IpAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.OperatingSystemTypes = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.ProvisioningAction = {
    Install: "install",
    Uninstall: "uninstall",
    Repair: "repair",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SecurityTypes = {
    TrustedLaunch: "TrustedLaunch",
    ConfidentialVM: "ConfidentialVM",
};
exports.ServiceName = {
    WAC: "WAC",
};
exports.SoftwareAssuranceIntent = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.State = {
    HasPrerequisite: "HasPrerequisite",
    Obsolete: "Obsolete",
    Ready: "Ready",
    NotApplicableBecauseAnotherUpdateIsInProgress: "NotApplicableBecauseAnotherUpdateIsInProgress",
    Preparing: "Preparing",
    Installing: "Installing",
    Installed: "Installed",
    PreparationFailed: "PreparationFailed",
    InstallationFailed: "InstallationFailed",
    Invalid: "Invalid",
    Recalled: "Recalled",
    Downloading: "Downloading",
    DownloadFailed: "DownloadFailed",
    HealthChecking: "HealthChecking",
    HealthCheckFailed: "HealthCheckFailed",
    ReadyToInstall: "ReadyToInstall",
    ScanInProgress: "ScanInProgress",
    ScanFailed: "ScanFailed",
    AdditionalContentRequired: "AdditionalContentRequired",
};
exports.UpdateRunPropertiesState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    InProgress: "InProgress",
    Failed: "Failed",
};
exports.UpdateSummariesPropertiesState = {
    Unknown: "Unknown",
    AppliedSuccessfully: "AppliedSuccessfully",
    UpdateAvailable: "UpdateAvailable",
    UpdateInProgress: "UpdateInProgress",
    UpdateFailed: "UpdateFailed",
    NeedsAttention: "NeedsAttention",
    PreparationInProgress: "PreparationInProgress",
    PreparationFailed: "PreparationFailed",
};
exports.VmSizeEnum = {
    Default: "Default",
    Standard_A2_v2: "Standard_A2_v2",
    Standard_A4_v2: "Standard_A4_v2",
    Standard_D2s_v3: "Standard_D2s_v3",
    Standard_D4s_v3: "Standard_D4s_v3",
    Standard_D8s_v3: "Standard_D8s_v3",
    Standard_D16s_v3: "Standard_D16s_v3",
    Standard_D32s_v3: "Standard_D32s_v3",
    Standard_DS2_v2: "Standard_DS2_v2",
    Standard_DS3_v2: "Standard_DS3_v2",
    Standard_DS4_v2: "Standard_DS4_v2",
    Standard_DS5_v2: "Standard_DS5_v2",
    Standard_DS13_v2: "Standard_DS13_v2",
    Standard_K8S_v1: "Standard_K8S_v1",
    Standard_K8S2_v1: "Standard_K8S2_v1",
    Standard_K8S3_v1: "Standard_K8S3_v1",
    Standard_K8S4_v1: "Standard_K8S4_v1",
    Standard_NK6: "Standard_NK6",
    Standard_NK12: "Standard_NK12",
    Standard_NV6: "Standard_NV6",
    Standard_NV12: "Standard_NV12",
    Standard_K8S5_v1: "Standard_K8S5_v1",
    Custom: "Custom",
};
exports.WindowsServerSubscription = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
