// @pulumi/azure-native/devcenter/v20250201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listSkusByProjectOutput = exports.listSkusByProject = exports.getScheduleOutput = exports.getSchedule = exports.getProjectPolicyOutput = exports.getProjectPolicy = exports.getProjectInheritedSettingsOutput = exports.getProjectInheritedSettings = exports.getProjectEnvironmentTypeOutput = exports.getProjectEnvironmentType = exports.getProjectCatalogSyncErrorDetailsOutput = exports.getProjectCatalogSyncErrorDetails = exports.getProjectCatalogImageDefinitionErrorDetailsOutput = exports.getProjectCatalogImageDefinitionErrorDetails = exports.getProjectCatalogImageDefinitionBuildDetailsOutput = exports.getProjectCatalogImageDefinitionBuildDetails = exports.getProjectCatalogEnvironmentDefinitionErrorDetailsOutput = exports.getProjectCatalogEnvironmentDefinitionErrorDetails = exports.getProjectCatalogOutput = exports.getProjectCatalog = exports.getProjectOutput = exports.getProject = exports.getPoolOutput = exports.getPool = exports.getNetworkConnectionOutput = exports.getNetworkConnection = exports.getGalleryOutput = exports.getGallery = exports.getEnvironmentTypeOutput = exports.getEnvironmentType = exports.getEnvironmentDefinitionErrorDetailsOutput = exports.getEnvironmentDefinitionErrorDetails = exports.getDevCenterOutput = exports.getDevCenter = exports.getDevBoxDefinitionOutput = exports.getDevBoxDefinition = exports.getCustomizationTaskErrorDetailsOutput = exports.getCustomizationTaskErrorDetails = exports.getCatalogSyncErrorDetailsOutput = exports.getCatalogSyncErrorDetails = exports.getCatalogOutput = exports.getCatalog = exports.getAttachedNetworkByDevCenterOutput = exports.getAttachedNetworkByDevCenter = exports.Gallery = exports.EnvironmentType = exports.DevCenter = exports.DevBoxDefinition = exports.Catalog = exports.AttachedNetworkByDevCenter = void 0;
exports.Schedule = exports.ProjectPolicy = exports.ProjectEnvironmentType = exports.ProjectCatalog = exports.Project = exports.Pool = exports.NetworkConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AttachedNetworkByDevCenter = null;
utilities.lazyLoad(exports, ["AttachedNetworkByDevCenter"], () => require("./attachedNetworkByDevCenter"));
exports.Catalog = null;
utilities.lazyLoad(exports, ["Catalog"], () => require("./catalog"));
exports.DevBoxDefinition = null;
utilities.lazyLoad(exports, ["DevBoxDefinition"], () => require("./devBoxDefinition"));
exports.DevCenter = null;
utilities.lazyLoad(exports, ["DevCenter"], () => require("./devCenter"));
exports.EnvironmentType = null;
utilities.lazyLoad(exports, ["EnvironmentType"], () => require("./environmentType"));
exports.Gallery = null;
utilities.lazyLoad(exports, ["Gallery"], () => require("./gallery"));
exports.getAttachedNetworkByDevCenter = null;
exports.getAttachedNetworkByDevCenterOutput = null;
utilities.lazyLoad(exports, ["getAttachedNetworkByDevCenter", "getAttachedNetworkByDevCenterOutput"], () => require("./getAttachedNetworkByDevCenter"));
exports.getCatalog = null;
exports.getCatalogOutput = null;
utilities.lazyLoad(exports, ["getCatalog", "getCatalogOutput"], () => require("./getCatalog"));
exports.getCatalogSyncErrorDetails = null;
exports.getCatalogSyncErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getCatalogSyncErrorDetails", "getCatalogSyncErrorDetailsOutput"], () => require("./getCatalogSyncErrorDetails"));
exports.getCustomizationTaskErrorDetails = null;
exports.getCustomizationTaskErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getCustomizationTaskErrorDetails", "getCustomizationTaskErrorDetailsOutput"], () => require("./getCustomizationTaskErrorDetails"));
exports.getDevBoxDefinition = null;
exports.getDevBoxDefinitionOutput = null;
utilities.lazyLoad(exports, ["getDevBoxDefinition", "getDevBoxDefinitionOutput"], () => require("./getDevBoxDefinition"));
exports.getDevCenter = null;
exports.getDevCenterOutput = null;
utilities.lazyLoad(exports, ["getDevCenter", "getDevCenterOutput"], () => require("./getDevCenter"));
exports.getEnvironmentDefinitionErrorDetails = null;
exports.getEnvironmentDefinitionErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentDefinitionErrorDetails", "getEnvironmentDefinitionErrorDetailsOutput"], () => require("./getEnvironmentDefinitionErrorDetails"));
exports.getEnvironmentType = null;
exports.getEnvironmentTypeOutput = null;
utilities.lazyLoad(exports, ["getEnvironmentType", "getEnvironmentTypeOutput"], () => require("./getEnvironmentType"));
exports.getGallery = null;
exports.getGalleryOutput = null;
utilities.lazyLoad(exports, ["getGallery", "getGalleryOutput"], () => require("./getGallery"));
exports.getNetworkConnection = null;
exports.getNetworkConnectionOutput = null;
utilities.lazyLoad(exports, ["getNetworkConnection", "getNetworkConnectionOutput"], () => require("./getNetworkConnection"));
exports.getPool = null;
exports.getPoolOutput = null;
utilities.lazyLoad(exports, ["getPool", "getPoolOutput"], () => require("./getPool"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getProjectCatalog = null;
exports.getProjectCatalogOutput = null;
utilities.lazyLoad(exports, ["getProjectCatalog", "getProjectCatalogOutput"], () => require("./getProjectCatalog"));
exports.getProjectCatalogEnvironmentDefinitionErrorDetails = null;
exports.getProjectCatalogEnvironmentDefinitionErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getProjectCatalogEnvironmentDefinitionErrorDetails", "getProjectCatalogEnvironmentDefinitionErrorDetailsOutput"], () => require("./getProjectCatalogEnvironmentDefinitionErrorDetails"));
exports.getProjectCatalogImageDefinitionBuildDetails = null;
exports.getProjectCatalogImageDefinitionBuildDetailsOutput = null;
utilities.lazyLoad(exports, ["getProjectCatalogImageDefinitionBuildDetails", "getProjectCatalogImageDefinitionBuildDetailsOutput"], () => require("./getProjectCatalogImageDefinitionBuildDetails"));
exports.getProjectCatalogImageDefinitionErrorDetails = null;
exports.getProjectCatalogImageDefinitionErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getProjectCatalogImageDefinitionErrorDetails", "getProjectCatalogImageDefinitionErrorDetailsOutput"], () => require("./getProjectCatalogImageDefinitionErrorDetails"));
exports.getProjectCatalogSyncErrorDetails = null;
exports.getProjectCatalogSyncErrorDetailsOutput = null;
utilities.lazyLoad(exports, ["getProjectCatalogSyncErrorDetails", "getProjectCatalogSyncErrorDetailsOutput"], () => require("./getProjectCatalogSyncErrorDetails"));
exports.getProjectEnvironmentType = null;
exports.getProjectEnvironmentTypeOutput = null;
utilities.lazyLoad(exports, ["getProjectEnvironmentType", "getProjectEnvironmentTypeOutput"], () => require("./getProjectEnvironmentType"));
exports.getProjectInheritedSettings = null;
exports.getProjectInheritedSettingsOutput = null;
utilities.lazyLoad(exports, ["getProjectInheritedSettings", "getProjectInheritedSettingsOutput"], () => require("./getProjectInheritedSettings"));
exports.getProjectPolicy = null;
exports.getProjectPolicyOutput = null;
utilities.lazyLoad(exports, ["getProjectPolicy", "getProjectPolicyOutput"], () => require("./getProjectPolicy"));
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.listSkusByProject = null;
exports.listSkusByProjectOutput = null;
utilities.lazyLoad(exports, ["listSkusByProject", "listSkusByProjectOutput"], () => require("./listSkusByProject"));
exports.NetworkConnection = null;
utilities.lazyLoad(exports, ["NetworkConnection"], () => require("./networkConnection"));
exports.Pool = null;
utilities.lazyLoad(exports, ["Pool"], () => require("./pool"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.ProjectCatalog = null;
utilities.lazyLoad(exports, ["ProjectCatalog"], () => require("./projectCatalog"));
exports.ProjectEnvironmentType = null;
utilities.lazyLoad(exports, ["ProjectEnvironmentType"], () => require("./projectEnvironmentType"));
exports.ProjectPolicy = null;
utilities.lazyLoad(exports, ["ProjectPolicy"], () => require("./projectPolicy"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
// Export enums:
__exportStar(require("../../types/enums/devcenter/v20250201"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:devcenter/v20250201:AttachedNetworkByDevCenter":
                return new exports.AttachedNetworkByDevCenter(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:Catalog":
                return new exports.Catalog(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:DevBoxDefinition":
                return new exports.DevBoxDefinition(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:DevCenter":
                return new exports.DevCenter(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:EnvironmentType":
                return new exports.EnvironmentType(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:Gallery":
                return new exports.Gallery(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:NetworkConnection":
                return new exports.NetworkConnection(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:Pool":
                return new exports.Pool(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:ProjectCatalog":
                return new exports.ProjectCatalog(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:ProjectEnvironmentType":
                return new exports.ProjectEnvironmentType(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:ProjectPolicy":
                return new exports.ProjectPolicy(name, undefined, { urn });
            case "azure-native:devcenter/v20250201:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "devcenter/v20250201", _module);
