// @pulumi/gcp/dataplex/aspectType.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AspectType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Aspect Type is a template for creating Aspects.
 *
 * ## Example Usage
 *
 * ### Dataplex Aspect Type Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const testAspectTypeBasic = new gcp.dataplex.AspectType("test_aspect_type_basic", {
 *     aspectTypeId: "aspect-type-basic",
 *     project: "my-project-name",
 *     location: "us-central1",
 *     metadataTemplate: `{
 *   "name": "tf-test-template",
 *   "type": "record",
 *   "recordFields": [
 *     {
 *       "name": "type",
 *       "type": "enum",
 *       "annotations": {
 *         "displayName": "Type",
 *         "description": "Specifies the type of view represented by the entry."
 *       },
 *       "index": 1,
 *       "constraints": {
 *         "required": true
 *       },
 *       "enumValues": [
 *         {
 *           "name": "VIEW",
 *           "index": 1
 *         }
 *       ]
 *     }
 *   ]
 * }
 * `,
 * });
 * ```
 * ### Dataplex Aspect Type Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const testAspectTypeFull = new gcp.dataplex.AspectType("test_aspect_type_full", {
 *     aspectTypeId: "aspect-type-full",
 *     project: "my-project-name",
 *     location: "us-central1",
 *     labels: {
 *         tag: "test-tf",
 *     },
 *     displayName: "terraform aspect type",
 *     description: "aspect type created by Terraform",
 *     metadataTemplate: `{
 *   "type": "record",
 *   "name": "Schema",
 *   "recordFields": [
 *     {
 *       "name": "fields",
 *       "type": "array",
 *       "index": 1,
 *       "arrayItems": {
 *         "name": "field",
 *         "type": "record",
 *         "typeId": "field",
 *         "recordFields": [
 *           {
 *             "name": "name",
 *             "type": "string",
 *             "index": 1,
 *             "constraints": {
 *               "required": true
 *             }
 *           },
 *           {
 *             "name": "description",
 *             "type": "string",
 *             "index": 2
 *           },
 *           {
 *             "name": "dataType",
 *             "type": "string",
 *             "index": 3,
 *             "constraints": {
 *               "required": true
 *             }
 *           },
 *           {
 *             "name": "metadataType",
 *             "type": "enum",
 *             "index": 4,
 *             "constraints": {
 *               "required": true
 *             },
 *             "enumValues": [
 *               {
 *                 "name": "BOOLEAN",
 *                 "index": 1
 *               },
 *               {
 *                 "name": "NUMBER",
 *                 "index": 2
 *               },
 *               {
 *                 "name": "STRING",
 *                 "index": 3
 *               },
 *               {
 *                 "name": "BYTES",
 *                 "index": 4
 *               },
 *               {
 *                 "name": "DATETIME",
 *                 "index": 5
 *               },
 *               {
 *                 "name": "TIMESTAMP",
 *                 "index": 6
 *               },
 *               {
 *                 "name": "GEOSPATIAL",
 *                 "index": 7
 *               },
 *               {
 *                 "name": "STRUCT",
 *                 "index": 8
 *               },
 *               {
 *                 "name": "OTHER",
 *                 "index": 100
 *               }
 *             ]
 *           },
 *           {
 *             "name": "mode",
 *             "type": "enum",
 *             "index": 5,
 *             "enumValues": [
 *               {
 *                 "name": "NULLABLE",
 *                 "index": 1
 *               },
 *               {
 *                 "name": "REPEATED",
 *                 "index": 2
 *               },
 *               {
 *                 "name": "REQUIRED",
 *                 "index": 3
 *               }
 *             ]
 *           },
 *           {
 *             "name": "defaultValue",
 *             "type": "string",
 *             "index": 6
 *           },
 *           {
 *             "name": "annotations",
 *             "type": "map",
 *             "index": 7,
 *             "mapItems": {
 *               "name": "label",
 *               "type": "string"
 *             }
 *           },
 *           {
 *             "name": "fields",
 *             "type": "array",
 *             "index": 20,
 *             "arrayItems": {
 *               "name": "field",
 *               "type": "record",
 *               "typeRef": "field"
 *             }
 *           }
 *         ]
 *       }
 *     }
 *   ]
 * }
 * `,
 * });
 * ```
 *
 * ## Import
 *
 * AspectType can be imported using any of these accepted formats:
 *
 * * `projects/{{project}}/locations/{{location}}/aspectTypes/{{aspect_type_id}}`
 *
 * * `{{project}}/{{location}}/{{aspect_type_id}}`
 *
 * * `{{location}}/{{aspect_type_id}}`
 *
 * When using the `pulumi import` command, AspectType can be imported using one of the formats above. For example:
 *
 * ```sh
 * $ pulumi import gcp:dataplex/aspectType:AspectType default projects/{{project}}/locations/{{location}}/aspectTypes/{{aspect_type_id}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:dataplex/aspectType:AspectType default {{project}}/{{location}}/{{aspect_type_id}}
 * ```
 *
 * ```sh
 * $ pulumi import gcp:dataplex/aspectType:AspectType default {{location}}/{{aspect_type_id}}
 * ```
 */
class AspectType extends pulumi.CustomResource {
    /**
     * Get an existing AspectType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AspectType(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AspectType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AspectType.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aspectTypeId"] = state ? state.aspectTypeId : undefined;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["effectiveLabels"] = state ? state.effectiveLabels : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["metadataTemplate"] = state ? state.metadataTemplate : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["pulumiLabels"] = state ? state.pulumiLabels : undefined;
            resourceInputs["transferStatus"] = state ? state.transferStatus : undefined;
            resourceInputs["uid"] = state ? state.uid : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["aspectTypeId"] = args ? args.aspectTypeId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["metadataTemplate"] = args ? args.metadataTemplate : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["effectiveLabels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pulumiLabels"] = undefined /*out*/;
            resourceInputs["transferStatus"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["effectiveLabels", "pulumiLabels"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(AspectType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AspectType = AspectType;
/** @internal */
AspectType.__pulumiType = 'gcp:dataplex/aspectType:AspectType';
