// @pulumi/azure-native/containerinstance/v20240501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getContainerGroupProfileOutput = exports.getContainerGroupProfile = exports.getContainerGroupOutput = exports.getContainerGroup = exports.ContainerGroupProfile = exports.ContainerGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ContainerGroup = null;
utilities.lazyLoad(exports, ["ContainerGroup"], () => require("./containerGroup"));
exports.ContainerGroupProfile = null;
utilities.lazyLoad(exports, ["ContainerGroupProfile"], () => require("./containerGroupProfile"));
exports.getContainerGroup = null;
exports.getContainerGroupOutput = null;
utilities.lazyLoad(exports, ["getContainerGroup", "getContainerGroupOutput"], () => require("./getContainerGroup"));
exports.getContainerGroupProfile = null;
exports.getContainerGroupProfileOutput = null;
utilities.lazyLoad(exports, ["getContainerGroupProfile", "getContainerGroupProfileOutput"], () => require("./getContainerGroupProfile"));
// Export enums:
__exportStar(require("../../types/enums/containerinstance/v20240501preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerinstance/v20240501preview:ContainerGroup":
                return new exports.ContainerGroup(name, undefined, { urn });
            case "azure-native:containerinstance/v20240501preview:ContainerGroupProfile":
                return new exports.ContainerGroupProfile(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerinstance/v20240501preview", _module);
