// @pulumi/azure-native/dbforpostgresql/v20201005privatepreview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerGroup = exports.getServerGroupOutput = exports.getServerGroup = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.FirewallRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getServerGroup = null;
exports.getServerGroupOutput = null;
utilities.lazyLoad(exports, ["getServerGroup", "getServerGroupOutput"], () => require("./getServerGroup"));
exports.ServerGroup = null;
utilities.lazyLoad(exports, ["ServerGroup"], () => require("./serverGroup"));
// Export enums:
__exportStar(require("../../types/enums/dbforpostgresql/v20201005privatepreview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbforpostgresql/v20201005privatepreview:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:dbforpostgresql/v20201005privatepreview:ServerGroup":
                return new exports.ServerGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbforpostgresql/v20201005privatepreview", _module);
