// @pulumi/azure-native/appcomplianceautomation/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240627 = exports.v20221116preview = exports.Webhook = exports.ScopingConfiguration = exports.Report = exports.listProviderActionInUseStorageAccountsOutput = exports.listProviderActionInUseStorageAccounts = exports.getWebhookOutput = exports.getWebhook = exports.getScopingConfigurationOutput = exports.getScopingConfiguration = exports.getReportScopingQuestionsOutput = exports.getReportScopingQuestions = exports.getReportOutput = exports.getReport = exports.getProviderActionOverviewStatusOutput = exports.getProviderActionOverviewStatus = exports.getProviderActionCollectionCountOutput = exports.getProviderActionCollectionCount = exports.getEvidenceOutput = exports.getEvidence = exports.Evidence = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Evidence = null;
utilities.lazyLoad(exports, ["Evidence"], () => require("./evidence"));
exports.getEvidence = null;
exports.getEvidenceOutput = null;
utilities.lazyLoad(exports, ["getEvidence", "getEvidenceOutput"], () => require("./getEvidence"));
exports.getProviderActionCollectionCount = null;
exports.getProviderActionCollectionCountOutput = null;
utilities.lazyLoad(exports, ["getProviderActionCollectionCount", "getProviderActionCollectionCountOutput"], () => require("./getProviderActionCollectionCount"));
exports.getProviderActionOverviewStatus = null;
exports.getProviderActionOverviewStatusOutput = null;
utilities.lazyLoad(exports, ["getProviderActionOverviewStatus", "getProviderActionOverviewStatusOutput"], () => require("./getProviderActionOverviewStatus"));
exports.getReport = null;
exports.getReportOutput = null;
utilities.lazyLoad(exports, ["getReport", "getReportOutput"], () => require("./getReport"));
exports.getReportScopingQuestions = null;
exports.getReportScopingQuestionsOutput = null;
utilities.lazyLoad(exports, ["getReportScopingQuestions", "getReportScopingQuestionsOutput"], () => require("./getReportScopingQuestions"));
exports.getScopingConfiguration = null;
exports.getScopingConfigurationOutput = null;
utilities.lazyLoad(exports, ["getScopingConfiguration", "getScopingConfigurationOutput"], () => require("./getScopingConfiguration"));
exports.getWebhook = null;
exports.getWebhookOutput = null;
utilities.lazyLoad(exports, ["getWebhook", "getWebhookOutput"], () => require("./getWebhook"));
exports.listProviderActionInUseStorageAccounts = null;
exports.listProviderActionInUseStorageAccountsOutput = null;
utilities.lazyLoad(exports, ["listProviderActionInUseStorageAccounts", "listProviderActionInUseStorageAccountsOutput"], () => require("./listProviderActionInUseStorageAccounts"));
exports.Report = null;
utilities.lazyLoad(exports, ["Report"], () => require("./report"));
exports.ScopingConfiguration = null;
utilities.lazyLoad(exports, ["ScopingConfiguration"], () => require("./scopingConfiguration"));
exports.Webhook = null;
utilities.lazyLoad(exports, ["Webhook"], () => require("./webhook"));
// Export enums:
__exportStar(require("../types/enums/appcomplianceautomation"), exports);
// Export sub-modules:
const v20221116preview = require("./v20221116preview");
exports.v20221116preview = v20221116preview;
const v20240627 = require("./v20240627");
exports.v20240627 = v20240627;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:appcomplianceautomation:Evidence":
                return new exports.Evidence(name, undefined, { urn });
            case "azure-native:appcomplianceautomation:Report":
                return new exports.Report(name, undefined, { urn });
            case "azure-native:appcomplianceautomation:ScopingConfiguration":
                return new exports.ScopingConfiguration(name, undefined, { urn });
            case "azure-native:appcomplianceautomation:Webhook":
                return new exports.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "appcomplianceautomation", _module);
