// @pulumi/azure-native/appcomplianceautomation/webhook.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A class represent an AppComplianceAutomation webhook resource.
 *
 * Uses Azure REST API version 2024-06-27.
 */
class Webhook extends pulumi.CustomResource {
    /**
     * Get an existing Webhook resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Webhook(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Webhook.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Webhook.__pulumiType;
    }
    /**
     * Create a Webhook resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.reportName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reportName'");
            }
            resourceInputs["contentType"] = args ? args.contentType : undefined;
            resourceInputs["enableSslVerification"] = args ? args.enableSslVerification : undefined;
            resourceInputs["events"] = args ? args.events : undefined;
            resourceInputs["payloadUrl"] = args ? args.payloadUrl : undefined;
            resourceInputs["reportName"] = args ? args.reportName : undefined;
            resourceInputs["sendAllEvents"] = args ? args.sendAllEvents : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["updateWebhookKey"] = args ? args.updateWebhookKey : undefined;
            resourceInputs["webhookKey"] = args ? args.webhookKey : undefined;
            resourceInputs["webhookName"] = args ? args.webhookName : undefined;
            resourceInputs["deliveryStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["webhookId"] = undefined /*out*/;
            resourceInputs["webhookKeyEnabled"] = undefined /*out*/;
        }
        else {
            resourceInputs["contentType"] = undefined /*out*/;
            resourceInputs["deliveryStatus"] = undefined /*out*/;
            resourceInputs["enableSslVerification"] = undefined /*out*/;
            resourceInputs["events"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["payloadUrl"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sendAllEvents"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updateWebhookKey"] = undefined /*out*/;
            resourceInputs["webhookId"] = undefined /*out*/;
            resourceInputs["webhookKey"] = undefined /*out*/;
            resourceInputs["webhookKeyEnabled"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:appcomplianceautomation/v20240627:Webhook" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Webhook.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Webhook = Webhook;
/** @internal */
Webhook.__pulumiType = 'azure-native:appcomplianceautomation:Webhook';
