// @pulumi/azure-native/contoso/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211101 = exports.v20211001preview = exports.getEmployeeOutput = exports.getEmployee = exports.Employee = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Employee = null;
utilities.lazyLoad(exports, ["Employee"], () => require("./employee"));
exports.getEmployee = null;
exports.getEmployeeOutput = null;
utilities.lazyLoad(exports, ["getEmployee", "getEmployeeOutput"], () => require("./getEmployee"));
// Export sub-modules:
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:contoso:Employee":
                return new exports.Employee(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "contoso", _module);
