// @pulumi/azure-native/types/enums/sqlvirtualmachine/v20231001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmIdentityType = exports.StorageWorkloadType = exports.SqlWorkloadType = exports.SqlVmGroupImageSku = exports.SqlServerLicenseType = exports.SqlManagementMode = exports.SqlImageSku = exports.Role = exports.ReadableSecondary = exports.LeastPrivilegeMode = exports.IdentityType = exports.FullBackupFrequencyType = exports.Failover = exports.DiskConfigurationType = exports.DayOfWeek = exports.ConnectivityType = exports.Commit = exports.ClusterSubnetType = exports.BackupScheduleType = exports.AutoBackupDaysOfWeek = exports.AssessmentDayOfWeek = exports.AdditionalVmPatch = void 0;
exports.AdditionalVmPatch = {
    NotSet: "NotSet",
    MicrosoftUpdate: "MicrosoftUpdate",
};
exports.AssessmentDayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.AutoBackupDaysOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.BackupScheduleType = {
    Manual: "Manual",
    Automated: "Automated",
};
exports.ClusterSubnetType = {
    SingleSubnet: "SingleSubnet",
    MultiSubnet: "MultiSubnet",
};
exports.Commit = {
    Synchronous_Commit: "Synchronous_Commit",
    Asynchronous_Commit: "Asynchronous_Commit",
};
exports.ConnectivityType = {
    LOCAL: "LOCAL",
    PRIVATE: "PRIVATE",
    PUBLIC: "PUBLIC",
};
exports.DayOfWeek = {
    Everyday: "Everyday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DiskConfigurationType = {
    NEW: "NEW",
    EXTEND: "EXTEND",
    ADD: "ADD",
};
exports.Failover = {
    Automatic: "Automatic",
    Manual: "Manual",
};
exports.FullBackupFrequencyType = {
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.LeastPrivilegeMode = {
    Enabled: "Enabled",
    NotSet: "NotSet",
};
exports.ReadableSecondary = {
    No: "No",
    All: "All",
    Read_Only: "Read_Only",
};
exports.Role = {
    Primary: "Primary",
    Secondary: "Secondary",
};
exports.SqlImageSku = {
    Developer: "Developer",
    Express: "Express",
    Standard: "Standard",
    Enterprise: "Enterprise",
    Web: "Web",
};
exports.SqlManagementMode = {
    Full: "Full",
    LightWeight: "LightWeight",
    NoAgent: "NoAgent",
};
exports.SqlServerLicenseType = {
    PAYG: "PAYG",
    AHUB: "AHUB",
    DR: "DR",
};
exports.SqlVmGroupImageSku = {
    Developer: "Developer",
    Enterprise: "Enterprise",
};
exports.SqlWorkloadType = {
    GENERAL: "GENERAL",
    OLTP: "OLTP",
    DW: "DW",
};
exports.StorageWorkloadType = {
    GENERAL: "GENERAL",
    OLTP: "OLTP",
    DW: "DW",
};
exports.VmIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
