// @pulumi/azure-native/videoanalyzer/v20210501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoAnalyzer = exports.Video = exports.listVideoStreamingTokenOutput = exports.listVideoStreamingToken = exports.getVideoAnalyzerOutput = exports.getVideoAnalyzer = exports.getVideoOutput = exports.getVideo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getVideo = null;
exports.getVideoOutput = null;
utilities.lazyLoad(exports, ["getVideo", "getVideoOutput"], () => require("./getVideo"));
exports.getVideoAnalyzer = null;
exports.getVideoAnalyzerOutput = null;
utilities.lazyLoad(exports, ["getVideoAnalyzer", "getVideoAnalyzerOutput"], () => require("./getVideoAnalyzer"));
exports.listVideoStreamingToken = null;
exports.listVideoStreamingTokenOutput = null;
utilities.lazyLoad(exports, ["listVideoStreamingToken", "listVideoStreamingTokenOutput"], () => require("./listVideoStreamingToken"));
exports.Video = null;
utilities.lazyLoad(exports, ["Video"], () => require("./video"));
exports.VideoAnalyzer = null;
utilities.lazyLoad(exports, ["VideoAnalyzer"], () => require("./videoAnalyzer"));
// Export enums:
__exportStar(require("../../types/enums/videoanalyzer/v20210501preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:videoanalyzer/v20210501preview:Video":
                return new exports.Video(name, undefined, { urn });
            case "azure-native:videoanalyzer/v20210501preview:VideoAnalyzer":
                return new exports.VideoAnalyzer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "videoanalyzer/v20210501preview", _module);
