// @pulumi/azure-native/eventgrid/v20220615/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSystemTopicOutput = exports.getSystemTopic = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getPartnerTopicEventSubscriptionFullUrlOutput = exports.getPartnerTopicEventSubscriptionFullUrl = exports.getPartnerTopicEventSubscriptionDeliveryAttributesOutput = exports.getPartnerTopicEventSubscriptionDeliveryAttributes = exports.getPartnerTopicEventSubscriptionOutput = exports.getPartnerTopicEventSubscription = exports.getPartnerTopicOutput = exports.getPartnerTopic = exports.getPartnerRegistrationOutput = exports.getPartnerRegistration = exports.getPartnerNamespaceOutput = exports.getPartnerNamespace = exports.getPartnerConfigurationOutput = exports.getPartnerConfiguration = exports.getEventSubscriptionFullUrlOutput = exports.getEventSubscriptionFullUrl = exports.getEventSubscriptionDeliveryAttributesOutput = exports.getEventSubscriptionDeliveryAttributes = exports.getEventSubscriptionOutput = exports.getEventSubscription = exports.getDomainTopicEventSubscriptionFullUrlOutput = exports.getDomainTopicEventSubscriptionFullUrl = exports.getDomainTopicEventSubscriptionDeliveryAttributesOutput = exports.getDomainTopicEventSubscriptionDeliveryAttributes = exports.getDomainTopicEventSubscriptionOutput = exports.getDomainTopicEventSubscription = exports.getDomainTopicOutput = exports.getDomainTopic = exports.getDomainEventSubscriptionFullUrlOutput = exports.getDomainEventSubscriptionFullUrl = exports.getDomainEventSubscriptionDeliveryAttributesOutput = exports.getDomainEventSubscriptionDeliveryAttributes = exports.getDomainEventSubscriptionOutput = exports.getDomainEventSubscription = exports.getDomainOutput = exports.getDomain = exports.getChannelFullUrlOutput = exports.getChannelFullUrl = exports.getChannelOutput = exports.getChannel = exports.EventSubscription = exports.DomainTopicEventSubscription = exports.DomainTopic = exports.DomainEventSubscription = exports.Domain = exports.Channel = void 0;
exports.TopicEventSubscription = exports.Topic = exports.SystemTopicEventSubscription = exports.SystemTopic = exports.PrivateEndpointConnection = exports.PartnerTopicEventSubscription = exports.PartnerTopic = exports.PartnerRegistration = exports.PartnerNamespace = exports.PartnerConfiguration = exports.listTopicSharedAccessKeysOutput = exports.listTopicSharedAccessKeys = exports.listPartnerNamespaceSharedAccessKeysOutput = exports.listPartnerNamespaceSharedAccessKeys = exports.listDomainSharedAccessKeysOutput = exports.listDomainSharedAccessKeys = exports.getTopicEventSubscriptionFullUrlOutput = exports.getTopicEventSubscriptionFullUrl = exports.getTopicEventSubscriptionDeliveryAttributesOutput = exports.getTopicEventSubscriptionDeliveryAttributes = exports.getTopicEventSubscriptionOutput = exports.getTopicEventSubscription = exports.getTopicOutput = exports.getTopic = exports.getSystemTopicEventSubscriptionFullUrlOutput = exports.getSystemTopicEventSubscriptionFullUrl = exports.getSystemTopicEventSubscriptionDeliveryAttributesOutput = exports.getSystemTopicEventSubscriptionDeliveryAttributes = exports.getSystemTopicEventSubscriptionOutput = exports.getSystemTopicEventSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.Domain = null;
utilities.lazyLoad(exports, ["Domain"], () => require("./domain"));
exports.DomainEventSubscription = null;
utilities.lazyLoad(exports, ["DomainEventSubscription"], () => require("./domainEventSubscription"));
exports.DomainTopic = null;
utilities.lazyLoad(exports, ["DomainTopic"], () => require("./domainTopic"));
exports.DomainTopicEventSubscription = null;
utilities.lazyLoad(exports, ["DomainTopicEventSubscription"], () => require("./domainTopicEventSubscription"));
exports.EventSubscription = null;
utilities.lazyLoad(exports, ["EventSubscription"], () => require("./eventSubscription"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.getChannelFullUrl = null;
exports.getChannelFullUrlOutput = null;
utilities.lazyLoad(exports, ["getChannelFullUrl", "getChannelFullUrlOutput"], () => require("./getChannelFullUrl"));
exports.getDomain = null;
exports.getDomainOutput = null;
utilities.lazyLoad(exports, ["getDomain", "getDomainOutput"], () => require("./getDomain"));
exports.getDomainEventSubscription = null;
exports.getDomainEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getDomainEventSubscription", "getDomainEventSubscriptionOutput"], () => require("./getDomainEventSubscription"));
exports.getDomainEventSubscriptionDeliveryAttributes = null;
exports.getDomainEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getDomainEventSubscriptionDeliveryAttributes", "getDomainEventSubscriptionDeliveryAttributesOutput"], () => require("./getDomainEventSubscriptionDeliveryAttributes"));
exports.getDomainEventSubscriptionFullUrl = null;
exports.getDomainEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getDomainEventSubscriptionFullUrl", "getDomainEventSubscriptionFullUrlOutput"], () => require("./getDomainEventSubscriptionFullUrl"));
exports.getDomainTopic = null;
exports.getDomainTopicOutput = null;
utilities.lazyLoad(exports, ["getDomainTopic", "getDomainTopicOutput"], () => require("./getDomainTopic"));
exports.getDomainTopicEventSubscription = null;
exports.getDomainTopicEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getDomainTopicEventSubscription", "getDomainTopicEventSubscriptionOutput"], () => require("./getDomainTopicEventSubscription"));
exports.getDomainTopicEventSubscriptionDeliveryAttributes = null;
exports.getDomainTopicEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getDomainTopicEventSubscriptionDeliveryAttributes", "getDomainTopicEventSubscriptionDeliveryAttributesOutput"], () => require("./getDomainTopicEventSubscriptionDeliveryAttributes"));
exports.getDomainTopicEventSubscriptionFullUrl = null;
exports.getDomainTopicEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getDomainTopicEventSubscriptionFullUrl", "getDomainTopicEventSubscriptionFullUrlOutput"], () => require("./getDomainTopicEventSubscriptionFullUrl"));
exports.getEventSubscription = null;
exports.getEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getEventSubscription", "getEventSubscriptionOutput"], () => require("./getEventSubscription"));
exports.getEventSubscriptionDeliveryAttributes = null;
exports.getEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getEventSubscriptionDeliveryAttributes", "getEventSubscriptionDeliveryAttributesOutput"], () => require("./getEventSubscriptionDeliveryAttributes"));
exports.getEventSubscriptionFullUrl = null;
exports.getEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getEventSubscriptionFullUrl", "getEventSubscriptionFullUrlOutput"], () => require("./getEventSubscriptionFullUrl"));
exports.getPartnerConfiguration = null;
exports.getPartnerConfigurationOutput = null;
utilities.lazyLoad(exports, ["getPartnerConfiguration", "getPartnerConfigurationOutput"], () => require("./getPartnerConfiguration"));
exports.getPartnerNamespace = null;
exports.getPartnerNamespaceOutput = null;
utilities.lazyLoad(exports, ["getPartnerNamespace", "getPartnerNamespaceOutput"], () => require("./getPartnerNamespace"));
exports.getPartnerRegistration = null;
exports.getPartnerRegistrationOutput = null;
utilities.lazyLoad(exports, ["getPartnerRegistration", "getPartnerRegistrationOutput"], () => require("./getPartnerRegistration"));
exports.getPartnerTopic = null;
exports.getPartnerTopicOutput = null;
utilities.lazyLoad(exports, ["getPartnerTopic", "getPartnerTopicOutput"], () => require("./getPartnerTopic"));
exports.getPartnerTopicEventSubscription = null;
exports.getPartnerTopicEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getPartnerTopicEventSubscription", "getPartnerTopicEventSubscriptionOutput"], () => require("./getPartnerTopicEventSubscription"));
exports.getPartnerTopicEventSubscriptionDeliveryAttributes = null;
exports.getPartnerTopicEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getPartnerTopicEventSubscriptionDeliveryAttributes", "getPartnerTopicEventSubscriptionDeliveryAttributesOutput"], () => require("./getPartnerTopicEventSubscriptionDeliveryAttributes"));
exports.getPartnerTopicEventSubscriptionFullUrl = null;
exports.getPartnerTopicEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getPartnerTopicEventSubscriptionFullUrl", "getPartnerTopicEventSubscriptionFullUrlOutput"], () => require("./getPartnerTopicEventSubscriptionFullUrl"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSystemTopic = null;
exports.getSystemTopicOutput = null;
utilities.lazyLoad(exports, ["getSystemTopic", "getSystemTopicOutput"], () => require("./getSystemTopic"));
exports.getSystemTopicEventSubscription = null;
exports.getSystemTopicEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSystemTopicEventSubscription", "getSystemTopicEventSubscriptionOutput"], () => require("./getSystemTopicEventSubscription"));
exports.getSystemTopicEventSubscriptionDeliveryAttributes = null;
exports.getSystemTopicEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getSystemTopicEventSubscriptionDeliveryAttributes", "getSystemTopicEventSubscriptionDeliveryAttributesOutput"], () => require("./getSystemTopicEventSubscriptionDeliveryAttributes"));
exports.getSystemTopicEventSubscriptionFullUrl = null;
exports.getSystemTopicEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getSystemTopicEventSubscriptionFullUrl", "getSystemTopicEventSubscriptionFullUrlOutput"], () => require("./getSystemTopicEventSubscriptionFullUrl"));
exports.getTopic = null;
exports.getTopicOutput = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
exports.getTopicEventSubscription = null;
exports.getTopicEventSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getTopicEventSubscription", "getTopicEventSubscriptionOutput"], () => require("./getTopicEventSubscription"));
exports.getTopicEventSubscriptionDeliveryAttributes = null;
exports.getTopicEventSubscriptionDeliveryAttributesOutput = null;
utilities.lazyLoad(exports, ["getTopicEventSubscriptionDeliveryAttributes", "getTopicEventSubscriptionDeliveryAttributesOutput"], () => require("./getTopicEventSubscriptionDeliveryAttributes"));
exports.getTopicEventSubscriptionFullUrl = null;
exports.getTopicEventSubscriptionFullUrlOutput = null;
utilities.lazyLoad(exports, ["getTopicEventSubscriptionFullUrl", "getTopicEventSubscriptionFullUrlOutput"], () => require("./getTopicEventSubscriptionFullUrl"));
exports.listDomainSharedAccessKeys = null;
exports.listDomainSharedAccessKeysOutput = null;
utilities.lazyLoad(exports, ["listDomainSharedAccessKeys", "listDomainSharedAccessKeysOutput"], () => require("./listDomainSharedAccessKeys"));
exports.listPartnerNamespaceSharedAccessKeys = null;
exports.listPartnerNamespaceSharedAccessKeysOutput = null;
utilities.lazyLoad(exports, ["listPartnerNamespaceSharedAccessKeys", "listPartnerNamespaceSharedAccessKeysOutput"], () => require("./listPartnerNamespaceSharedAccessKeys"));
exports.listTopicSharedAccessKeys = null;
exports.listTopicSharedAccessKeysOutput = null;
utilities.lazyLoad(exports, ["listTopicSharedAccessKeys", "listTopicSharedAccessKeysOutput"], () => require("./listTopicSharedAccessKeys"));
exports.PartnerConfiguration = null;
utilities.lazyLoad(exports, ["PartnerConfiguration"], () => require("./partnerConfiguration"));
exports.PartnerNamespace = null;
utilities.lazyLoad(exports, ["PartnerNamespace"], () => require("./partnerNamespace"));
exports.PartnerRegistration = null;
utilities.lazyLoad(exports, ["PartnerRegistration"], () => require("./partnerRegistration"));
exports.PartnerTopic = null;
utilities.lazyLoad(exports, ["PartnerTopic"], () => require("./partnerTopic"));
exports.PartnerTopicEventSubscription = null;
utilities.lazyLoad(exports, ["PartnerTopicEventSubscription"], () => require("./partnerTopicEventSubscription"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.SystemTopic = null;
utilities.lazyLoad(exports, ["SystemTopic"], () => require("./systemTopic"));
exports.SystemTopicEventSubscription = null;
utilities.lazyLoad(exports, ["SystemTopicEventSubscription"], () => require("./systemTopicEventSubscription"));
exports.Topic = null;
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
exports.TopicEventSubscription = null;
utilities.lazyLoad(exports, ["TopicEventSubscription"], () => require("./topicEventSubscription"));
// Export enums:
__exportStar(require("../../types/enums/eventgrid/v20220615"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:eventgrid/v20220615:Channel":
                return new exports.Channel(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:Domain":
                return new exports.Domain(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:DomainEventSubscription":
                return new exports.DomainEventSubscription(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:DomainTopic":
                return new exports.DomainTopic(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:DomainTopicEventSubscription":
                return new exports.DomainTopicEventSubscription(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:EventSubscription":
                return new exports.EventSubscription(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PartnerConfiguration":
                return new exports.PartnerConfiguration(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PartnerNamespace":
                return new exports.PartnerNamespace(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PartnerRegistration":
                return new exports.PartnerRegistration(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PartnerTopic":
                return new exports.PartnerTopic(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PartnerTopicEventSubscription":
                return new exports.PartnerTopicEventSubscription(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:SystemTopic":
                return new exports.SystemTopic(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:SystemTopicEventSubscription":
                return new exports.SystemTopicEventSubscription(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:Topic":
                return new exports.Topic(name, undefined, { urn });
            case "azure-native:eventgrid/v20220615:TopicEventSubscription":
                return new exports.TopicEventSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "eventgrid/v20220615", _module);
