// @pulumi/azure-native/types/enums/relay/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.Relaytype = exports.PublicNetworkAccess = exports.PrivateLinkConnectionStatus = exports.EndPointProvisioningState = exports.AccessRights = exports.v20240101 = exports.v20211101 = exports.v20180101preview = exports.v20170401 = void 0;
// Export sub-modules:
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.EndPointProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.PrivateLinkConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    SecuredByPerimeter: "SecuredByPerimeter",
};
exports.Relaytype = {
    NetTcp: "NetTcp",
    Http: "Http",
};
exports.SkuName = {
    Standard: "Standard",
};
exports.SkuTier = {
    Standard: "Standard",
};
