// @pulumi/azure-native/openenergyplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220404preview = exports.listEnergyServicePartitionsOutput = exports.listEnergyServicePartitions = exports.getEnergyServiceOutput = exports.getEnergyService = exports.EnergyService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EnergyService = null;
utilities.lazyLoad(exports, ["EnergyService"], () => require("./energyService"));
exports.getEnergyService = null;
exports.getEnergyServiceOutput = null;
utilities.lazyLoad(exports, ["getEnergyService", "getEnergyServiceOutput"], () => require("./getEnergyService"));
exports.listEnergyServicePartitions = null;
exports.listEnergyServicePartitionsOutput = null;
utilities.lazyLoad(exports, ["listEnergyServicePartitions", "listEnergyServicePartitionsOutput"], () => require("./listEnergyServicePartitions"));
// Export sub-modules:
const v20220404preview = require("./v20220404preview");
exports.v20220404preview = v20220404preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:openenergyplatform:EnergyService":
                return new exports.EnergyService(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "openenergyplatform", _module);
