// @pulumi/azure-native/types/enums/operationalinsights/v20221001/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceSkuNameEnum = exports.TablePlanEnum = exports.PublicNetworkAccessType = exports.ManagedServiceIdentityType = exports.IdentityType = exports.ColumnTypeEnum = exports.ColumnDataTypeHintEnum = exports.ClusterSkuNameEnum = exports.BillingType = void 0;
exports.BillingType = {
    Cluster: "Cluster",
    Workspaces: "Workspaces",
};
exports.ClusterSkuNameEnum = {
    CapacityReservation: "CapacityReservation",
};
exports.ColumnDataTypeHintEnum = {
    /**
     * A string that matches the pattern of a URI, for example, scheme://username:password@host:1234/this/is/a/path?k1=v1&k2=v2#fragment
     */
    Uri: "uri",
    /**
     * A standard 128-bit GUID following the standard shape, xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
     */
    Guid: "guid",
    /**
     * An Azure Resource Model (ARM) path: /subscriptions/{...}/resourceGroups/{...}/providers/Microsoft.{...}/{...}/{...}/{...}...
     */
    ArmPath: "armPath",
    /**
     * A standard V4/V6 ip address following the standard shape, x.x.x.x/y:y:y:y:y:y:y:y
     */
    Ip: "ip",
};
exports.ColumnTypeEnum = {
    String: "string",
    Int: "int",
    Long: "long",
    Real: "real",
    Boolean: "boolean",
    DateTime: "dateTime",
    Guid: "guid",
    Dynamic: "dynamic",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PublicNetworkAccessType = {
    /**
     * Enables connectivity to Log Analytics through public DNS.
     */
    Enabled: "Enabled",
    /**
     * Disables public connectivity to Log Analytics through public DNS.
     */
    Disabled: "Disabled",
};
exports.TablePlanEnum = {
    /**
     * Logs  that are adjusted to support high volume low value verbose logs.
     */
    Basic: "Basic",
    /**
     * Logs  that allow monitoring and analytics.
     */
    Analytics: "Analytics",
};
exports.WorkspaceSkuNameEnum = {
    Free: "Free",
    Standard: "Standard",
    Premium: "Premium",
    PerNode: "PerNode",
    PerGB2018: "PerGB2018",
    Standalone: "Standalone",
    CapacityReservation: "CapacityReservation",
    LACluster: "LACluster",
};
