// @pulumi/azure-native/types/enums/edgeorder/v20220501preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransportShipmentTypes = exports.SupportedFilterTypes = exports.OrderMode = exports.OrderItemType = exports.NotificationStageName = exports.DoubleEncryptionStatus = exports.ChildConfigurationType = exports.AddressType = void 0;
exports.AddressType = {
    /**
     * Address type not known.
     */
    None: "None",
    /**
     * Residential Address.
     */
    Residential: "Residential",
    /**
     * Commercial Address.
     */
    Commercial: "Commercial",
};
exports.ChildConfigurationType = {
    /**
     * Child configuration is a device configuration.
     */
    DeviceConfiguration: "DeviceConfiguration",
    /**
     * Child configuration is an additional configuration.
     */
    AdditionalConfiguration: "AdditionalConfiguration",
};
exports.DoubleEncryptionStatus = {
    /**
     * Double encryption is disabled.
     */
    Disabled: "Disabled",
    /**
     * Double encryption is enabled.
     */
    Enabled: "Enabled",
};
exports.NotificationStageName = {
    /**
     * Notification at order item shipped from microsoft datacenter.
     */
    Shipped: "Shipped",
    /**
     * Notification at order item delivered to customer.
     */
    Delivered: "Delivered",
};
exports.OrderItemType = {
    /**
     * Purchase OrderItem.
     */
    Purchase: "Purchase",
    /**
     * Rental OrderItem.
     */
    Rental: "Rental",
};
exports.OrderMode = {
    /**
     * Default Order mode.
     */
    Default: "Default",
    /**
     * Mode in which the Order will not be fulfilled.
     */
    DoNotFulfill: "DoNotFulfill",
};
exports.SupportedFilterTypes = {
    /**
     * Ship to country.
     */
    ShipToCountries: "ShipToCountries",
    /**
     * Double encryption status.
     */
    DoubleEncryptionStatus: "DoubleEncryptionStatus",
};
exports.TransportShipmentTypes = {
    /**
     * Shipment Logistics is handled by the customer.
     */
    CustomerManaged: "CustomerManaged",
    /**
     * Shipment Logistics is handled by Microsoft.
     */
    MicrosoftManaged: "MicrosoftManaged",
};
