// @pulumi/azure-native/kubernetesconfiguration/v20220402preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.getPrivateLinkScopeOutput = exports.getPrivateLinkScope = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getExtensionOutput = exports.getExtension = exports.Extension = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkScope = null;
exports.getPrivateLinkScopeOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScope", "getPrivateLinkScopeOutput"], () => require("./getPrivateLinkScope"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkScope = null;
utilities.lazyLoad(exports, ["PrivateLinkScope"], () => require("./privateLinkScope"));
// Export enums:
__exportStar(require("../../types/enums/kubernetesconfiguration/v20220402preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetesconfiguration/v20220402preview:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration/v20220402preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration/v20220402preview:PrivateLinkScope":
                return new exports.PrivateLinkScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetesconfiguration/v20220402preview", _module);
