// @pulumi/azure-native/types/enums/purview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.Status = exports.PublicNetworkAccess = exports.ManagedResourcesPublicNetworkAccess = exports.ManagedEventHubState = exports.EventStreamingType = exports.EventStreamingState = exports.EventHubType = exports.CredentialsType = exports.v20240401preview = exports.v20230501preview = exports.v20211201 = exports.v20210701 = exports.v20201201preview = void 0;
// Export sub-modules:
const v20201201preview = require("./v20201201preview");
exports.v20201201preview = v20201201preview;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20211201 = require("./v20211201");
exports.v20211201 = v20211201;
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const v20240401preview = require("./v20240401preview");
exports.v20240401preview = v20240401preview;
exports.CredentialsType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.EventHubType = {
    Notification: "Notification",
    Hook: "Hook",
};
exports.EventStreamingState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.EventStreamingType = {
    None: "None",
    Managed: "Managed",
    Azure: "Azure",
};
exports.ManagedEventHubState = {
    NotSpecified: "NotSpecified",
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ManagedResourcesPublicNetworkAccess = {
    NotSpecified: "NotSpecified",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PublicNetworkAccess = {
    NotSpecified: "NotSpecified",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Status = {
    Unknown: "Unknown",
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.Type = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
