// @pulumi/azure-native/types/enums/confluent/v20240701/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SaaSOfferStatus = exports.PartnerConnectorType = exports.Package = exports.DataFormatType = exports.ConnectorType = exports.ConnectorStatus = exports.ConnectorServiceType = exports.ConnectorClass = exports.AuthType = void 0;
exports.AuthType = {
    SERVICE_ACCOUNT: "SERVICE_ACCOUNT",
    KAFKA_API_KEY: "KAFKA_API_KEY",
};
exports.ConnectorClass = {
    AZUREBLOBSOURCE: "AZUREBLOBSOURCE",
    AZUREBLOBSINK: "AZUREBLOBSINK",
};
exports.ConnectorServiceType = {
    AzureBlobStorageSinkConnector: "AzureBlobStorageSinkConnector",
    AzureBlobStorageSourceConnector: "AzureBlobStorageSourceConnector",
    AzureCosmosDBSinkConnector: "AzureCosmosDBSinkConnector",
    AzureCosmosDBSourceConnector: "AzureCosmosDBSourceConnector",
    AzureSynapseAnalyticsSinkConnector: "AzureSynapseAnalyticsSinkConnector",
};
exports.ConnectorStatus = {
    PROVISIONING: "PROVISIONING",
    RUNNING: "RUNNING",
    PAUSED: "PAUSED",
    FAILED: "FAILED",
};
exports.ConnectorType = {
    SINK: "SINK",
    SOURCE: "SOURCE",
};
exports.DataFormatType = {
    AVRO: "AVRO",
    JSON: "JSON",
    STRING: "STRING",
    BYTES: "BYTES",
    PROTOBUF: "PROTOBUF",
};
exports.Package = {
    ESSENTIALS: "ESSENTIALS",
    ADVANCED: "ADVANCED",
};
exports.PartnerConnectorType = {
    KafkaAzureBlobStorageSource: "KafkaAzureBlobStorageSource",
    KafkaAzureBlobStorageSink: "KafkaAzureBlobStorageSink",
    KafkaAzureCosmosDBSource: "KafkaAzureCosmosDBSource",
    KafkaAzureCosmosDBSink: "KafkaAzureCosmosDBSink",
    KafkaAzureSynapseAnalyticsSink: "KafkaAzureSynapseAnalyticsSink",
};
exports.SaaSOfferStatus = {
    Started: "Started",
    PendingFulfillmentStart: "PendingFulfillmentStart",
    InProgress: "InProgress",
    Subscribed: "Subscribed",
    Suspended: "Suspended",
    Reinstated: "Reinstated",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Unsubscribed: "Unsubscribed",
    Updating: "Updating",
};
