// @pulumi/azure-native/customproviders/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20180901preview = exports.getCustomResourceProviderOutput = exports.getCustomResourceProvider = exports.getAssociationOutput = exports.getAssociation = exports.CustomResourceProvider = exports.Association = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Association = null;
utilities.lazyLoad(exports, ["Association"], () => require("./association"));
exports.CustomResourceProvider = null;
utilities.lazyLoad(exports, ["CustomResourceProvider"], () => require("./customResourceProvider"));
exports.getAssociation = null;
exports.getAssociationOutput = null;
utilities.lazyLoad(exports, ["getAssociation", "getAssociationOutput"], () => require("./getAssociation"));
exports.getCustomResourceProvider = null;
exports.getCustomResourceProviderOutput = null;
utilities.lazyLoad(exports, ["getCustomResourceProvider", "getCustomResourceProviderOutput"], () => require("./getCustomResourceProvider"));
// Export enums:
__exportStar(require("../types/enums/customproviders"), exports);
// Export sub-modules:
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:customproviders:Association":
                return new exports.Association(name, undefined, { urn });
            case "azure-native:customproviders:CustomResourceProvider":
                return new exports.CustomResourceProvider(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "customproviders", _module);
