// @pulumi/azure-native/mobilenetwork/v20220401preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Slice = exports.Site = exports.SimPolicy = exports.SimGroup = exports.Sim = exports.Service = exports.PacketCoreDataPlane = exports.PacketCoreControlPlane = exports.MobileNetwork = exports.listMobileNetworkSimIdsOutput = exports.listMobileNetworkSimIds = exports.getSliceOutput = exports.getSlice = exports.getSiteOutput = exports.getSite = exports.getSimPolicyOutput = exports.getSimPolicy = exports.getSimGroupOutput = exports.getSimGroup = exports.getSimOutput = exports.getSim = exports.getServiceOutput = exports.getService = exports.getPacketCoreDataPlaneOutput = exports.getPacketCoreDataPlane = exports.getPacketCoreControlPlaneOutput = exports.getPacketCoreControlPlane = exports.getMobileNetworkOutput = exports.getMobileNetwork = exports.getDataNetworkOutput = exports.getDataNetwork = exports.getAttachedDataNetworkOutput = exports.getAttachedDataNetwork = exports.DataNetwork = exports.AttachedDataNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AttachedDataNetwork = null;
utilities.lazyLoad(exports, ["AttachedDataNetwork"], () => require("./attachedDataNetwork"));
exports.DataNetwork = null;
utilities.lazyLoad(exports, ["DataNetwork"], () => require("./dataNetwork"));
exports.getAttachedDataNetwork = null;
exports.getAttachedDataNetworkOutput = null;
utilities.lazyLoad(exports, ["getAttachedDataNetwork", "getAttachedDataNetworkOutput"], () => require("./getAttachedDataNetwork"));
exports.getDataNetwork = null;
exports.getDataNetworkOutput = null;
utilities.lazyLoad(exports, ["getDataNetwork", "getDataNetworkOutput"], () => require("./getDataNetwork"));
exports.getMobileNetwork = null;
exports.getMobileNetworkOutput = null;
utilities.lazyLoad(exports, ["getMobileNetwork", "getMobileNetworkOutput"], () => require("./getMobileNetwork"));
exports.getPacketCoreControlPlane = null;
exports.getPacketCoreControlPlaneOutput = null;
utilities.lazyLoad(exports, ["getPacketCoreControlPlane", "getPacketCoreControlPlaneOutput"], () => require("./getPacketCoreControlPlane"));
exports.getPacketCoreDataPlane = null;
exports.getPacketCoreDataPlaneOutput = null;
utilities.lazyLoad(exports, ["getPacketCoreDataPlane", "getPacketCoreDataPlaneOutput"], () => require("./getPacketCoreDataPlane"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.getSim = null;
exports.getSimOutput = null;
utilities.lazyLoad(exports, ["getSim", "getSimOutput"], () => require("./getSim"));
exports.getSimGroup = null;
exports.getSimGroupOutput = null;
utilities.lazyLoad(exports, ["getSimGroup", "getSimGroupOutput"], () => require("./getSimGroup"));
exports.getSimPolicy = null;
exports.getSimPolicyOutput = null;
utilities.lazyLoad(exports, ["getSimPolicy", "getSimPolicyOutput"], () => require("./getSimPolicy"));
exports.getSite = null;
exports.getSiteOutput = null;
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
exports.getSlice = null;
exports.getSliceOutput = null;
utilities.lazyLoad(exports, ["getSlice", "getSliceOutput"], () => require("./getSlice"));
exports.listMobileNetworkSimIds = null;
exports.listMobileNetworkSimIdsOutput = null;
utilities.lazyLoad(exports, ["listMobileNetworkSimIds", "listMobileNetworkSimIdsOutput"], () => require("./listMobileNetworkSimIds"));
exports.MobileNetwork = null;
utilities.lazyLoad(exports, ["MobileNetwork"], () => require("./mobileNetwork"));
exports.PacketCoreControlPlane = null;
utilities.lazyLoad(exports, ["PacketCoreControlPlane"], () => require("./packetCoreControlPlane"));
exports.PacketCoreDataPlane = null;
utilities.lazyLoad(exports, ["PacketCoreDataPlane"], () => require("./packetCoreDataPlane"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
exports.Sim = null;
utilities.lazyLoad(exports, ["Sim"], () => require("./sim"));
exports.SimGroup = null;
utilities.lazyLoad(exports, ["SimGroup"], () => require("./simGroup"));
exports.SimPolicy = null;
utilities.lazyLoad(exports, ["SimPolicy"], () => require("./simPolicy"));
exports.Site = null;
utilities.lazyLoad(exports, ["Site"], () => require("./site"));
exports.Slice = null;
utilities.lazyLoad(exports, ["Slice"], () => require("./slice"));
// Export enums:
__exportStar(require("../../types/enums/mobilenetwork/v20220401preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mobilenetwork/v20220401preview:AttachedDataNetwork":
                return new exports.AttachedDataNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:DataNetwork":
                return new exports.DataNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:MobileNetwork":
                return new exports.MobileNetwork(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:PacketCoreControlPlane":
                return new exports.PacketCoreControlPlane(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:PacketCoreDataPlane":
                return new exports.PacketCoreDataPlane(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:Service":
                return new exports.Service(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:Sim":
                return new exports.Sim(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:SimGroup":
                return new exports.SimGroup(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:SimPolicy":
                return new exports.SimPolicy(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:Site":
                return new exports.Site(name, undefined, { urn });
            case "azure-native:mobilenetwork/v20220401preview:Slice":
                return new exports.Slice(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mobilenetwork/v20220401preview", _module);
