// @pulumi/azure-native/types/enums/resources/v20241101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationLevel = exports.ResourceIdentityType = exports.OnErrorDeploymentType = exports.ExtendedLocationType = exports.ExpressionEvaluationOptionsScopeType = exports.DeploymentMode = void 0;
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.ExpressionEvaluationOptionsScopeType = {
    NotSpecified: "NotSpecified",
    Outer: "Outer",
    Inner: "Inner",
};
exports.ExtendedLocationType = {
    EdgeZone: "EdgeZone",
};
exports.OnErrorDeploymentType = {
    LastSuccessful: "LastSuccessful",
    SpecificDeployment: "SpecificDeployment",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ValidationLevel = {
    /**
     * Static analysis of the template is performed.
     */
    Template: "Template",
    /**
     * Static analysis of the template is performed and resource declarations are sent to resource providers for semantic validation. Validates that the caller has RBAC write permissions on each resource.
     */
    Provider: "Provider",
    /**
     * Static analysis of the template is performed and resource declarations are sent to resource providers for semantic validation. Skips validating that the caller has RBAC write permissions on each resource.
     */
    ProviderNoRbac: "ProviderNoRbac",
};
