// @pulumi/azure-native/types/enums/recommendationsservice/v20220301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrincipalType = exports.ModelingSize = exports.ModelingFrequency = exports.ModelingFeatures = exports.ManagedServiceIdentityType = exports.AccountConfiguration = void 0;
exports.AccountConfiguration = {
    Free: "Free",
    Capacity: "Capacity",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.ModelingFeatures = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.ModelingFrequency = {
    Low: "Low",
    Medium: "Medium",
    High: "High",
};
exports.ModelingSize = {
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
};
exports.PrincipalType = {
    Application: "Application",
    User: "User",
};
