// @pulumi/azure-native/types/enums/databasefleetmanager/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.PrincipalType = exports.IdentityType = exports.DatabaseCreateMode = exports.v20250201preview = void 0;
// Export sub-modules:
const v20250201preview = require("./v20250201preview");
exports.v20250201preview = v20250201preview;
exports.DatabaseCreateMode = {
    /**
     * Create a database.
     */
    Default: "Default",
    /**
     * Copy the source database (source database name must be specified).
     */
    Copy: "Copy",
    /**
     * Create a database by restoring source database from a point in time (source database name and restore from time must be specified).
     */
    PointInTimeRestore: "PointInTimeRestore",
};
exports.IdentityType = {
    /**
     * No identity.
     */
    None: "None",
    /**
     * User assigned identity.
     */
    UserAssigned: "UserAssigned",
};
exports.PrincipalType = {
    /**
     * Application principal type.
     */
    Application: "Application",
    /**
     * User principal type.
     */
    User: "User",
};
exports.ZoneRedundancy = {
    /**
     * Zone redundancy enabled.
     */
    Enabled: "Enabled",
    /**
     * Zone redundancy disabled.
     */
    Disabled: "Disabled",
};
