// @pulumi/azure-native/netapp/v20240901preview/backupVault.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupVault = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Backup Vault information
 */
class BackupVault extends pulumi.CustomResource {
    /**
     * Get an existing BackupVault resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BackupVault(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupVault.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupVault.__pulumiType;
    }
    /**
     * Create a BackupVault resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["backupVaultName"] = args ? args.backupVaultName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20221101preview:BackupVault" }, { type: "azure-native:netapp/v20230501preview:BackupVault" }, { type: "azure-native:netapp/v20230701preview:BackupVault" }, { type: "azure-native:netapp/v20231101:BackupVault" }, { type: "azure-native:netapp/v20231101preview:BackupVault" }, { type: "azure-native:netapp/v20240101:BackupVault" }, { type: "azure-native:netapp/v20240301:BackupVault" }, { type: "azure-native:netapp/v20240301preview:BackupVault" }, { type: "azure-native:netapp/v20240501:BackupVault" }, { type: "azure-native:netapp/v20240501preview:BackupVault" }, { type: "azure-native:netapp/v20240701:BackupVault" }, { type: "azure-native:netapp/v20240701preview:BackupVault" }, { type: "azure-native:netapp/v20240901:BackupVault" }, { type: "azure-native:netapp:BackupVault" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BackupVault.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BackupVault = BackupVault;
/** @internal */
BackupVault.__pulumiType = 'azure-native:netapp/v20240901preview:BackupVault';
