// @pulumi/azure-native/netapp/v20240901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeQuotaRule = exports.VolumeGroup = exports.Volume = exports.Subvolume = exports.SnapshotPolicy = exports.Snapshot = exports.Pool = exports.listVolumeReplicationsOutput = exports.listVolumeReplications = exports.listVolumeQuotaReportOutput = exports.listVolumeQuotaReport = exports.getVolumeQuotaRuleOutput = exports.getVolumeQuotaRule = exports.getVolumeGroupIdForLdapUserOutput = exports.getVolumeGroupIdForLdapUser = exports.getVolumeGroupOutput = exports.getVolumeGroup = exports.getVolumeOutput = exports.getVolume = exports.getSubvolumeMetadataOutput = exports.getSubvolumeMetadata = exports.getSubvolumeOutput = exports.getSubvolume = exports.getSnapshotPolicyOutput = exports.getSnapshotPolicy = exports.getSnapshotOutput = exports.getSnapshot = exports.getPoolOutput = exports.getPool = exports.getBackupVaultOutput = exports.getBackupVault = exports.getBackupPolicyOutput = exports.getBackupPolicy = exports.getBackupOutput = exports.getBackup = exports.getAccountKeyVaultStatusOutput = exports.getAccountKeyVaultStatus = exports.getAccountOutput = exports.getAccount = exports.BackupVault = exports.BackupPolicy = exports.Backup = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.Backup = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
exports.BackupPolicy = null;
utilities.lazyLoad(exports, ["BackupPolicy"], () => require("./backupPolicy"));
exports.BackupVault = null;
utilities.lazyLoad(exports, ["BackupVault"], () => require("./backupVault"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getAccountKeyVaultStatus = null;
exports.getAccountKeyVaultStatusOutput = null;
utilities.lazyLoad(exports, ["getAccountKeyVaultStatus", "getAccountKeyVaultStatusOutput"], () => require("./getAccountKeyVaultStatus"));
exports.getBackup = null;
exports.getBackupOutput = null;
utilities.lazyLoad(exports, ["getBackup", "getBackupOutput"], () => require("./getBackup"));
exports.getBackupPolicy = null;
exports.getBackupPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackupPolicy", "getBackupPolicyOutput"], () => require("./getBackupPolicy"));
exports.getBackupVault = null;
exports.getBackupVaultOutput = null;
utilities.lazyLoad(exports, ["getBackupVault", "getBackupVaultOutput"], () => require("./getBackupVault"));
exports.getPool = null;
exports.getPoolOutput = null;
utilities.lazyLoad(exports, ["getPool", "getPoolOutput"], () => require("./getPool"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getSnapshotPolicy = null;
exports.getSnapshotPolicyOutput = null;
utilities.lazyLoad(exports, ["getSnapshotPolicy", "getSnapshotPolicyOutput"], () => require("./getSnapshotPolicy"));
exports.getSubvolume = null;
exports.getSubvolumeOutput = null;
utilities.lazyLoad(exports, ["getSubvolume", "getSubvolumeOutput"], () => require("./getSubvolume"));
exports.getSubvolumeMetadata = null;
exports.getSubvolumeMetadataOutput = null;
utilities.lazyLoad(exports, ["getSubvolumeMetadata", "getSubvolumeMetadataOutput"], () => require("./getSubvolumeMetadata"));
exports.getVolume = null;
exports.getVolumeOutput = null;
utilities.lazyLoad(exports, ["getVolume", "getVolumeOutput"], () => require("./getVolume"));
exports.getVolumeGroup = null;
exports.getVolumeGroupOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroup", "getVolumeGroupOutput"], () => require("./getVolumeGroup"));
exports.getVolumeGroupIdForLdapUser = null;
exports.getVolumeGroupIdForLdapUserOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroupIdForLdapUser", "getVolumeGroupIdForLdapUserOutput"], () => require("./getVolumeGroupIdForLdapUser"));
exports.getVolumeQuotaRule = null;
exports.getVolumeQuotaRuleOutput = null;
utilities.lazyLoad(exports, ["getVolumeQuotaRule", "getVolumeQuotaRuleOutput"], () => require("./getVolumeQuotaRule"));
exports.listVolumeQuotaReport = null;
exports.listVolumeQuotaReportOutput = null;
utilities.lazyLoad(exports, ["listVolumeQuotaReport", "listVolumeQuotaReportOutput"], () => require("./listVolumeQuotaReport"));
exports.listVolumeReplications = null;
exports.listVolumeReplicationsOutput = null;
utilities.lazyLoad(exports, ["listVolumeReplications", "listVolumeReplicationsOutput"], () => require("./listVolumeReplications"));
exports.Pool = null;
utilities.lazyLoad(exports, ["Pool"], () => require("./pool"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.SnapshotPolicy = null;
utilities.lazyLoad(exports, ["SnapshotPolicy"], () => require("./snapshotPolicy"));
exports.Subvolume = null;
utilities.lazyLoad(exports, ["Subvolume"], () => require("./subvolume"));
exports.Volume = null;
utilities.lazyLoad(exports, ["Volume"], () => require("./volume"));
exports.VolumeGroup = null;
utilities.lazyLoad(exports, ["VolumeGroup"], () => require("./volumeGroup"));
exports.VolumeQuotaRule = null;
utilities.lazyLoad(exports, ["VolumeQuotaRule"], () => require("./volumeQuotaRule"));
// Export enums:
__exportStar(require("../../types/enums/netapp/v20240901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:netapp/v20240901preview:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:BackupPolicy":
                return new exports.BackupPolicy(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:BackupVault":
                return new exports.BackupVault(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:Pool":
                return new exports.Pool(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:SnapshotPolicy":
                return new exports.SnapshotPolicy(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:Subvolume":
                return new exports.Subvolume(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:Volume":
                return new exports.Volume(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:VolumeGroup":
                return new exports.VolumeGroup(name, undefined, { urn });
            case "azure-native:netapp/v20240901preview:VolumeQuotaRule":
                return new exports.VolumeQuotaRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "netapp/v20240901preview", _module);
