// @pulumi/azure-native/types/enums/datalakestore/v20161101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustedIdProviderState = exports.TierType = exports.FirewallState = exports.FirewallAllowAzureIpsState = exports.EncryptionState = exports.EncryptionIdentityType = exports.EncryptionConfigType = void 0;
exports.EncryptionConfigType = {
    UserManaged: "UserManaged",
    ServiceManaged: "ServiceManaged",
};
exports.EncryptionIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.EncryptionState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FirewallAllowAzureIpsState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FirewallState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TierType = {
    Consumption: "Consumption",
    Commitment_1TB: "Commitment_1TB",
    Commitment_10TB: "Commitment_10TB",
    Commitment_100TB: "Commitment_100TB",
    Commitment_500TB: "Commitment_500TB",
    Commitment_1PB: "Commitment_1PB",
    Commitment_5PB: "Commitment_5PB",
};
exports.TrustedIdProviderState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
