// @pulumi/azure-native/migrate/v20200501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnectionControllerPrivateEndpointConnection = exports.MigrateProjectsControllerMigrateProject = exports.getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnectionControllerPrivateEndpointConnection = exports.getMigrateProjectsControllerMigrateProjectOutput = exports.getMigrateProjectsControllerMigrateProject = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMigrateProjectsControllerMigrateProject = null;
exports.getMigrateProjectsControllerMigrateProjectOutput = null;
utilities.lazyLoad(exports, ["getMigrateProjectsControllerMigrateProject", "getMigrateProjectsControllerMigrateProjectOutput"], () => require("./getMigrateProjectsControllerMigrateProject"));
exports.getPrivateEndpointConnectionControllerPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnectionControllerPrivateEndpointConnection", "getPrivateEndpointConnectionControllerPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnectionControllerPrivateEndpointConnection"));
exports.MigrateProjectsControllerMigrateProject = null;
utilities.lazyLoad(exports, ["MigrateProjectsControllerMigrateProject"], () => require("./migrateProjectsControllerMigrateProject"));
exports.PrivateEndpointConnectionControllerPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnectionControllerPrivateEndpointConnection"], () => require("./privateEndpointConnectionControllerPrivateEndpointConnection"));
// Export enums:
__exportStar(require("../../types/enums/migrate/v20200501"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate/v20200501:MigrateProjectsControllerMigrateProject":
                return new exports.MigrateProjectsControllerMigrateProject(name, undefined, { urn });
            case "azure-native:migrate/v20200501:PrivateEndpointConnectionControllerPrivateEndpointConnection":
                return new exports.PrivateEndpointConnectionControllerPrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate/v20200501", _module);
