// @pulumi/azure-native/appconfiguration/v20230901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Replica = exports.PrivateEndpointConnection = exports.listConfigurationStoreKeysOutput = exports.listConfigurationStoreKeys = exports.KeyValue = exports.getReplicaOutput = exports.getReplica = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getKeyValueOutput = exports.getKeyValue = exports.getConfigurationStoreOutput = exports.getConfigurationStore = exports.ConfigurationStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ConfigurationStore = null;
utilities.lazyLoad(exports, ["ConfigurationStore"], () => require("./configurationStore"));
exports.getConfigurationStore = null;
exports.getConfigurationStoreOutput = null;
utilities.lazyLoad(exports, ["getConfigurationStore", "getConfigurationStoreOutput"], () => require("./getConfigurationStore"));
exports.getKeyValue = null;
exports.getKeyValueOutput = null;
utilities.lazyLoad(exports, ["getKeyValue", "getKeyValueOutput"], () => require("./getKeyValue"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getReplica = null;
exports.getReplicaOutput = null;
utilities.lazyLoad(exports, ["getReplica", "getReplicaOutput"], () => require("./getReplica"));
exports.KeyValue = null;
utilities.lazyLoad(exports, ["KeyValue"], () => require("./keyValue"));
exports.listConfigurationStoreKeys = null;
exports.listConfigurationStoreKeysOutput = null;
utilities.lazyLoad(exports, ["listConfigurationStoreKeys", "listConfigurationStoreKeysOutput"], () => require("./listConfigurationStoreKeys"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Replica = null;
utilities.lazyLoad(exports, ["Replica"], () => require("./replica"));
// Export enums:
__exportStar(require("../../types/enums/appconfiguration/v20230901preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:appconfiguration/v20230901preview:ConfigurationStore":
                return new exports.ConfigurationStore(name, undefined, { urn });
            case "azure-native:appconfiguration/v20230901preview:KeyValue":
                return new exports.KeyValue(name, undefined, { urn });
            case "azure-native:appconfiguration/v20230901preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:appconfiguration/v20230901preview:Replica":
                return new exports.Replica(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "appconfiguration/v20230901preview", _module);
