// @pulumi/azure-native/mixedreality/v20250101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteRenderingAccount = exports.listRemoteRenderingAccountKeysOutput = exports.listRemoteRenderingAccountKeys = exports.getRemoteRenderingAccountOutput = exports.getRemoteRenderingAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getRemoteRenderingAccount = null;
exports.getRemoteRenderingAccountOutput = null;
utilities.lazyLoad(exports, ["getRemoteRenderingAccount", "getRemoteRenderingAccountOutput"], () => require("./getRemoteRenderingAccount"));
exports.listRemoteRenderingAccountKeys = null;
exports.listRemoteRenderingAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listRemoteRenderingAccountKeys", "listRemoteRenderingAccountKeysOutput"], () => require("./listRemoteRenderingAccountKeys"));
exports.RemoteRenderingAccount = null;
utilities.lazyLoad(exports, ["RemoteRenderingAccount"], () => require("./remoteRenderingAccount"));
// Export enums:
__exportStar(require("../../types/enums/mixedreality/v20250101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mixedreality/v20250101:RemoteRenderingAccount":
                return new exports.RemoteRenderingAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mixedreality/v20250101", _module);
