// @pulumi/azure-native/types/enums/kubernetes/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.ProvisioningState = exports.PrivateLinkState = exports.AuthenticationMethod = exports.v20241201preview = exports.v20240715preview = exports.v20240701preview = exports.v20240601preview = exports.v20240201preview = exports.v20240101 = exports.v20231101preview = exports.v20221001preview = exports.v20220501preview = exports.v20210401preview = void 0;
// Export sub-modules:
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20221001preview = require("./v20221001preview");
exports.v20221001preview = v20221001preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240601preview = require("./v20240601preview");
exports.v20240601preview = v20240601preview;
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
const v20240715preview = require("./v20240715preview");
exports.v20240715preview = v20240715preview;
const v20241201preview = require("./v20241201preview");
exports.v20241201preview = v20241201preview;
exports.AuthenticationMethod = {
    Token: "Token",
    AAD: "AAD",
};
exports.PrivateLinkState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Provisioning: "Provisioning",
    Updating: "Updating",
    Deleting: "Deleting",
    Accepted: "Accepted",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
