// @pulumi/azure-native/labservices/v20220801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.Schedule = exports.LabPlan = exports.Lab = exports.getUserOutput = exports.getUser = exports.getScheduleOutput = exports.getSchedule = exports.getLabPlanOutput = exports.getLabPlan = exports.getLabOutput = exports.getLab = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLab = null;
exports.getLabOutput = null;
utilities.lazyLoad(exports, ["getLab", "getLabOutput"], () => require("./getLab"));
exports.getLabPlan = null;
exports.getLabPlanOutput = null;
utilities.lazyLoad(exports, ["getLabPlan", "getLabPlanOutput"], () => require("./getLabPlan"));
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.Lab = null;
utilities.lazyLoad(exports, ["Lab"], () => require("./lab"));
exports.LabPlan = null;
utilities.lazyLoad(exports, ["LabPlan"], () => require("./labPlan"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../../types/enums/labservices/v20220801"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:labservices/v20220801:Lab":
                return new exports.Lab(name, undefined, { urn });
            case "azure-native:labservices/v20220801:LabPlan":
                return new exports.LabPlan(name, undefined, { urn });
            case "azure-native:labservices/v20220801:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            case "azure-native:labservices/v20220801:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "labservices/v20220801", _module);
