// @pulumi/azure-native/types/enums/azurestackhci/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftwareAssuranceStatus = exports.SoftwareAssuranceIntent = exports.SecurityTypes = exports.SecurityRuleProtocol = exports.SecurityRuleDirection = exports.SecurityRuleAccess = exports.ResourceIdentityType = exports.RemoteSupportType = exports.RemoteSupportAccessLevel = exports.ProvisioningAction = exports.PrivateIPAllocationMethodEnum = exports.OsTypeEnum = exports.OperatingSystemTypes = exports.NetworkTypeEnum = exports.ManagedServiceIdentityType = exports.IpAllocationMethodEnum = exports.IPPoolTypeEnum = exports.HyperVGeneration = exports.HciEdgeDeviceJobType = exports.ExtendedLocationTypes = exports.EdgeDeviceKind = exports.DiskFileFormat = exports.DiagnosticLevel = exports.DeploymentMode = exports.ComplianceAssignmentType = exports.CloudInitDataSource = exports.AvailabilityType = exports.v20250401preview = exports.v20250201preview = exports.v20241201preview = exports.v20241001preview = exports.v20240901preview = exports.v20240801preview = exports.v20240715preview = exports.v20240501preview = exports.v20240401 = exports.v20240215preview = exports.v20240201preview = exports.v20240101 = exports.v20231101preview = exports.v20230901preview = exports.v20230801preview = exports.v20230801 = exports.v20230701preview = exports.v20230601 = exports.v20230301 = exports.v20221215preview = exports.v20220901 = exports.v20220101 = exports.v20210901preview = void 0;
exports.WindowsServerSubscription = exports.VmSizeEnum = exports.UpdateSummariesPropertiesState = exports.UpdateRunPropertiesState = exports.State = void 0;
// Export sub-modules:
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220901 = require("./v20220901");
exports.v20220901 = v20220901;
const v20221215preview = require("./v20221215preview");
exports.v20221215preview = v20221215preview;
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20230601 = require("./v20230601");
exports.v20230601 = v20230601;
const v20230701preview = require("./v20230701preview");
exports.v20230701preview = v20230701preview;
const v20230801 = require("./v20230801");
exports.v20230801 = v20230801;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20230901preview = require("./v20230901preview");
exports.v20230901preview = v20230901preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const v20240101 = require("./v20240101");
exports.v20240101 = v20240101;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240215preview = require("./v20240215preview");
exports.v20240215preview = v20240215preview;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const v20240501preview = require("./v20240501preview");
exports.v20240501preview = v20240501preview;
const v20240715preview = require("./v20240715preview");
exports.v20240715preview = v20240715preview;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const v20240901preview = require("./v20240901preview");
exports.v20240901preview = v20240901preview;
const v20241001preview = require("./v20241001preview");
exports.v20241001preview = v20241001preview;
const v20241201preview = require("./v20241201preview");
exports.v20241201preview = v20241201preview;
const v20250201preview = require("./v20250201preview");
exports.v20250201preview = v20250201preview;
const v20250401preview = require("./v20250401preview");
exports.v20250401preview = v20250401preview;
exports.AvailabilityType = {
    Local: "Local",
    Online: "Online",
    Notify: "Notify",
};
exports.CloudInitDataSource = {
    NoCloud: "NoCloud",
    Azure: "Azure",
};
exports.ComplianceAssignmentType = {
    /**
     * Report on the state of the machine, but don't make changes.
     */
    Audit: "Audit",
    /**
     * Applied to the machine. If it drifts, the local service inside the machine makes a correction at the next evaluation.
     */
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
};
exports.DeploymentMode = {
    /**
     * Validate ECE action deployment for a cluster.
     */
    Validate: "Validate",
    /**
     * Deploy ECE action deployment for a cluster.
     */
    Deploy: "Deploy",
};
exports.DiagnosticLevel = {
    Off: "Off",
    Basic: "Basic",
    Enhanced: "Enhanced",
};
exports.DiskFileFormat = {
    Vhdx: "vhdx",
    Vhd: "vhd",
};
exports.EdgeDeviceKind = {
    /**
     * Arc-enabled edge device with HCI OS.
     */
    HCI: "HCI",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.HciEdgeDeviceJobType = {
    /**
     * Job to collect logs from the device.
     */
    CollectLog: "CollectLog",
    /**
     * Job to provide remote support to the device.
     */
    RemoteSupport: "RemoteSupport",
};
exports.HyperVGeneration = {
    V1: "V1",
    V2: "V2",
};
exports.IPPoolTypeEnum = {
    Vm: "vm",
    Vippool: "vippool",
};
exports.IpAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.NetworkTypeEnum = {
    NAT: "NAT",
    Transparent: "Transparent",
    L2Bridge: "L2Bridge",
    L2Tunnel: "L2Tunnel",
    ICS: "ICS",
    Private: "Private",
    Overlay: "Overlay",
    Internal: "Internal",
    Mirrored: "Mirrored",
};
exports.OperatingSystemTypes = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OsTypeEnum = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.PrivateIPAllocationMethodEnum = {
    Dynamic: "Dynamic",
    Static: "Static",
};
exports.ProvisioningAction = {
    Install: "install",
    Uninstall: "uninstall",
    Repair: "repair",
};
exports.RemoteSupportAccessLevel = {
    /**
     * No remote support access is granted.
     */
    None: "None",
    /**
     * Access is limited to diagnostics information only.
     */
    Diagnostics: "Diagnostics",
    /**
     * Access includes diagnostics information and the ability to perform repairs.
     */
    DiagnosticsAndRepair: "DiagnosticsAndRepair",
};
exports.RemoteSupportType = {
    /**
     * Enables remote support for the edge device.
     */
    Enable: "Enable",
    /**
     * Revokes previously granted remote support access for the edge device.
     */
    Revoke: "Revoke",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SecurityRuleAccess = {
    /**
     * Network traffic is allowed
     */
    Allow: "Allow",
    /**
     * Network traffic is denied
     */
    Deny: "Deny",
};
exports.SecurityRuleDirection = {
    /**
     * Rule is evaluated on incoming traffic
     */
    Inbound: "Inbound",
    /**
     * Rule is evaluated on outgoing traffic
     */
    Outbound: "Outbound",
};
exports.SecurityRuleProtocol = {
    /**
     * Transmission Control Protocol
     */
    Tcp: "Tcp",
    /**
     * User Datagram Protocol
     */
    Udp: "Udp",
    /**
     * Internet Control Message Protocol
     */
    Icmp: "Icmp",
    /**
     * Wildcard rule for all protocols
     */
    Asterisk: "*",
};
exports.SecurityTypes = {
    TrustedLaunch: "TrustedLaunch",
    ConfidentialVM: "ConfidentialVM",
};
exports.SoftwareAssuranceIntent = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.SoftwareAssuranceStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.State = {
    HasPrerequisite: "HasPrerequisite",
    Obsolete: "Obsolete",
    Ready: "Ready",
    NotApplicableBecauseAnotherUpdateIsInProgress: "NotApplicableBecauseAnotherUpdateIsInProgress",
    Preparing: "Preparing",
    Installing: "Installing",
    Installed: "Installed",
    PreparationFailed: "PreparationFailed",
    InstallationFailed: "InstallationFailed",
    Invalid: "Invalid",
    Recalled: "Recalled",
    Downloading: "Downloading",
    DownloadFailed: "DownloadFailed",
    HealthChecking: "HealthChecking",
    HealthCheckFailed: "HealthCheckFailed",
    ReadyToInstall: "ReadyToInstall",
    ScanInProgress: "ScanInProgress",
    ScanFailed: "ScanFailed",
};
exports.UpdateRunPropertiesState = {
    Unknown: "Unknown",
    Succeeded: "Succeeded",
    InProgress: "InProgress",
    Failed: "Failed",
};
exports.UpdateSummariesPropertiesState = {
    Unknown: "Unknown",
    AppliedSuccessfully: "AppliedSuccessfully",
    UpdateAvailable: "UpdateAvailable",
    UpdateInProgress: "UpdateInProgress",
    UpdateFailed: "UpdateFailed",
    NeedsAttention: "NeedsAttention",
    PreparationInProgress: "PreparationInProgress",
    PreparationFailed: "PreparationFailed",
};
exports.VmSizeEnum = {
    Default: "Default",
    Standard_A2_v2: "Standard_A2_v2",
    Standard_A4_v2: "Standard_A4_v2",
    Standard_D2s_v3: "Standard_D2s_v3",
    Standard_D4s_v3: "Standard_D4s_v3",
    Standard_D8s_v3: "Standard_D8s_v3",
    Standard_D16s_v3: "Standard_D16s_v3",
    Standard_D32s_v3: "Standard_D32s_v3",
    Standard_DS2_v2: "Standard_DS2_v2",
    Standard_DS3_v2: "Standard_DS3_v2",
    Standard_DS4_v2: "Standard_DS4_v2",
    Standard_DS5_v2: "Standard_DS5_v2",
    Standard_DS13_v2: "Standard_DS13_v2",
    Standard_K8S_v1: "Standard_K8S_v1",
    Standard_K8S2_v1: "Standard_K8S2_v1",
    Standard_K8S3_v1: "Standard_K8S3_v1",
    Standard_K8S4_v1: "Standard_K8S4_v1",
    Standard_NK6: "Standard_NK6",
    Standard_NK12: "Standard_NK12",
    Standard_NV6: "Standard_NV6",
    Standard_NV12: "Standard_NV12",
    Standard_K8S5_v1: "Standard_K8S5_v1",
    Custom: "Custom",
};
exports.WindowsServerSubscription = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
