// @pulumi/azure-native/apimanagement/v20161010/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUserSharedAccessTokenOutput = exports.getUserSharedAccessToken = exports.getApiManagementServiceSsoTokenOutput = exports.getApiManagementServiceSsoToken = exports.getApiManagementServiceOutput = exports.getApiManagementService = exports.ApiManagementService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.ApiManagementService = null;
utilities.lazyLoad(exports, ["ApiManagementService"], () => require("./apiManagementService"));
exports.getApiManagementService = null;
exports.getApiManagementServiceOutput = null;
utilities.lazyLoad(exports, ["getApiManagementService", "getApiManagementServiceOutput"], () => require("./getApiManagementService"));
exports.getApiManagementServiceSsoToken = null;
exports.getApiManagementServiceSsoTokenOutput = null;
utilities.lazyLoad(exports, ["getApiManagementServiceSsoToken", "getApiManagementServiceSsoTokenOutput"], () => require("./getApiManagementServiceSsoToken"));
exports.getUserSharedAccessToken = null;
exports.getUserSharedAccessTokenOutput = null;
utilities.lazyLoad(exports, ["getUserSharedAccessToken", "getUserSharedAccessTokenOutput"], () => require("./getUserSharedAccessToken"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20161010"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20161010:ApiManagementService":
                return new exports.ApiManagementService(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20161010", _module);
