// @pulumi/azure-native/workloads/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SapLandscapeMonitor = exports.SapInstance = exports.SapDiscoverySite = exports.ProviderInstance = exports.Monitor = exports.getServerInstanceOutput = exports.getServerInstance = exports.getSapVirtualInstanceInvokeSizingRecommendationsOutput = exports.getSapVirtualInstanceInvokeSizingRecommendations = exports.getSapVirtualInstanceInvokeSapSupportedSkuOutput = exports.getSapVirtualInstanceInvokeSapSupportedSku = exports.getSapVirtualInstanceInvokeDiskConfigurationsOutput = exports.getSapVirtualInstanceInvokeDiskConfigurations = exports.getSapVirtualInstanceInvokeAvailabilityZoneDetailsOutput = exports.getSapVirtualInstanceInvokeAvailabilityZoneDetails = exports.getSapLandscapeMonitorOutput = exports.getSapLandscapeMonitor = exports.getSapInstanceOutput = exports.getSapInstance = exports.getSapDiscoverySiteOutput = exports.getSapDiscoverySite = exports.getSAPVirtualInstanceOutput = exports.getSAPVirtualInstance = exports.getSAPSupportedSkuOutput = exports.getSAPSupportedSku = exports.getSAPSizingRecommendationsOutput = exports.getSAPSizingRecommendations = exports.getSAPDiskConfigurationsOutput = exports.getSAPDiskConfigurations = exports.getSAPDatabaseInstanceOutput = exports.getSAPDatabaseInstance = exports.getSAPCentralInstanceOutput = exports.getSAPCentralInstance = exports.getSAPAvailabilityZoneDetailsOutput = exports.getSAPAvailabilityZoneDetails = exports.getSAPApplicationServerInstanceOutput = exports.getSAPApplicationServerInstance = exports.getProviderInstanceOutput = exports.getProviderInstance = exports.getMonitorOutput = exports.getMonitor = exports.getConnectorOutput = exports.getConnector = exports.getAlertOutput = exports.getAlert = exports.getACSSBackupConnectionOutput = exports.getACSSBackupConnection = exports.Connector = exports.Alert = exports.ACSSBackupConnection = void 0;
exports.v20240901 = exports.v20240201preview = exports.v20231201preview = exports.v20231001preview = exports.v20230401 = exports.v20221101preview = exports.v20211201preview = exports.ServerInstance = exports.SAPVirtualInstance = exports.SAPDatabaseInstance = exports.SAPCentralInstance = exports.SAPApplicationServerInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ACSSBackupConnection = null;
utilities.lazyLoad(exports, ["ACSSBackupConnection"], () => require("./acssbackupConnection"));
exports.Alert = null;
utilities.lazyLoad(exports, ["Alert"], () => require("./alert"));
exports.Connector = null;
utilities.lazyLoad(exports, ["Connector"], () => require("./connector"));
exports.getACSSBackupConnection = null;
exports.getACSSBackupConnectionOutput = null;
utilities.lazyLoad(exports, ["getACSSBackupConnection", "getACSSBackupConnectionOutput"], () => require("./getACSSBackupConnection"));
exports.getAlert = null;
exports.getAlertOutput = null;
utilities.lazyLoad(exports, ["getAlert", "getAlertOutput"], () => require("./getAlert"));
exports.getConnector = null;
exports.getConnectorOutput = null;
utilities.lazyLoad(exports, ["getConnector", "getConnectorOutput"], () => require("./getConnector"));
exports.getMonitor = null;
exports.getMonitorOutput = null;
utilities.lazyLoad(exports, ["getMonitor", "getMonitorOutput"], () => require("./getMonitor"));
exports.getProviderInstance = null;
exports.getProviderInstanceOutput = null;
utilities.lazyLoad(exports, ["getProviderInstance", "getProviderInstanceOutput"], () => require("./getProviderInstance"));
exports.getSAPApplicationServerInstance = null;
exports.getSAPApplicationServerInstanceOutput = null;
utilities.lazyLoad(exports, ["getSAPApplicationServerInstance", "getSAPApplicationServerInstanceOutput"], () => require("./getSAPApplicationServerInstance"));
exports.getSAPAvailabilityZoneDetails = null;
exports.getSAPAvailabilityZoneDetailsOutput = null;
utilities.lazyLoad(exports, ["getSAPAvailabilityZoneDetails", "getSAPAvailabilityZoneDetailsOutput"], () => require("./getSAPAvailabilityZoneDetails"));
exports.getSAPCentralInstance = null;
exports.getSAPCentralInstanceOutput = null;
utilities.lazyLoad(exports, ["getSAPCentralInstance", "getSAPCentralInstanceOutput"], () => require("./getSAPCentralInstance"));
exports.getSAPDatabaseInstance = null;
exports.getSAPDatabaseInstanceOutput = null;
utilities.lazyLoad(exports, ["getSAPDatabaseInstance", "getSAPDatabaseInstanceOutput"], () => require("./getSAPDatabaseInstance"));
exports.getSAPDiskConfigurations = null;
exports.getSAPDiskConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getSAPDiskConfigurations", "getSAPDiskConfigurationsOutput"], () => require("./getSAPDiskConfigurations"));
exports.getSAPSizingRecommendations = null;
exports.getSAPSizingRecommendationsOutput = null;
utilities.lazyLoad(exports, ["getSAPSizingRecommendations", "getSAPSizingRecommendationsOutput"], () => require("./getSAPSizingRecommendations"));
exports.getSAPSupportedSku = null;
exports.getSAPSupportedSkuOutput = null;
utilities.lazyLoad(exports, ["getSAPSupportedSku", "getSAPSupportedSkuOutput"], () => require("./getSAPSupportedSku"));
exports.getSAPVirtualInstance = null;
exports.getSAPVirtualInstanceOutput = null;
utilities.lazyLoad(exports, ["getSAPVirtualInstance", "getSAPVirtualInstanceOutput"], () => require("./getSAPVirtualInstance"));
exports.getSapDiscoverySite = null;
exports.getSapDiscoverySiteOutput = null;
utilities.lazyLoad(exports, ["getSapDiscoverySite", "getSapDiscoverySiteOutput"], () => require("./getSapDiscoverySite"));
exports.getSapInstance = null;
exports.getSapInstanceOutput = null;
utilities.lazyLoad(exports, ["getSapInstance", "getSapInstanceOutput"], () => require("./getSapInstance"));
exports.getSapLandscapeMonitor = null;
exports.getSapLandscapeMonitorOutput = null;
utilities.lazyLoad(exports, ["getSapLandscapeMonitor", "getSapLandscapeMonitorOutput"], () => require("./getSapLandscapeMonitor"));
exports.getSapVirtualInstanceInvokeAvailabilityZoneDetails = null;
exports.getSapVirtualInstanceInvokeAvailabilityZoneDetailsOutput = null;
utilities.lazyLoad(exports, ["getSapVirtualInstanceInvokeAvailabilityZoneDetails", "getSapVirtualInstanceInvokeAvailabilityZoneDetailsOutput"], () => require("./getSapVirtualInstanceInvokeAvailabilityZoneDetails"));
exports.getSapVirtualInstanceInvokeDiskConfigurations = null;
exports.getSapVirtualInstanceInvokeDiskConfigurationsOutput = null;
utilities.lazyLoad(exports, ["getSapVirtualInstanceInvokeDiskConfigurations", "getSapVirtualInstanceInvokeDiskConfigurationsOutput"], () => require("./getSapVirtualInstanceInvokeDiskConfigurations"));
exports.getSapVirtualInstanceInvokeSapSupportedSku = null;
exports.getSapVirtualInstanceInvokeSapSupportedSkuOutput = null;
utilities.lazyLoad(exports, ["getSapVirtualInstanceInvokeSapSupportedSku", "getSapVirtualInstanceInvokeSapSupportedSkuOutput"], () => require("./getSapVirtualInstanceInvokeSapSupportedSku"));
exports.getSapVirtualInstanceInvokeSizingRecommendations = null;
exports.getSapVirtualInstanceInvokeSizingRecommendationsOutput = null;
utilities.lazyLoad(exports, ["getSapVirtualInstanceInvokeSizingRecommendations", "getSapVirtualInstanceInvokeSizingRecommendationsOutput"], () => require("./getSapVirtualInstanceInvokeSizingRecommendations"));
exports.getServerInstance = null;
exports.getServerInstanceOutput = null;
utilities.lazyLoad(exports, ["getServerInstance", "getServerInstanceOutput"], () => require("./getServerInstance"));
exports.Monitor = null;
utilities.lazyLoad(exports, ["Monitor"], () => require("./monitor"));
exports.ProviderInstance = null;
utilities.lazyLoad(exports, ["ProviderInstance"], () => require("./providerInstance"));
exports.SapDiscoverySite = null;
utilities.lazyLoad(exports, ["SapDiscoverySite"], () => require("./sapDiscoverySite"));
exports.SapInstance = null;
utilities.lazyLoad(exports, ["SapInstance"], () => require("./sapInstance"));
exports.SapLandscapeMonitor = null;
utilities.lazyLoad(exports, ["SapLandscapeMonitor"], () => require("./sapLandscapeMonitor"));
exports.SAPApplicationServerInstance = null;
utilities.lazyLoad(exports, ["SAPApplicationServerInstance"], () => require("./sapapplicationServerInstance"));
exports.SAPCentralInstance = null;
utilities.lazyLoad(exports, ["SAPCentralInstance"], () => require("./sapcentralInstance"));
exports.SAPDatabaseInstance = null;
utilities.lazyLoad(exports, ["SAPDatabaseInstance"], () => require("./sapdatabaseInstance"));
exports.SAPVirtualInstance = null;
utilities.lazyLoad(exports, ["SAPVirtualInstance"], () => require("./sapvirtualInstance"));
exports.ServerInstance = null;
utilities.lazyLoad(exports, ["ServerInstance"], () => require("./serverInstance"));
// Export enums:
__exportStar(require("../types/enums/workloads"), exports);
// Export sub-modules:
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20221101preview = require("./v20221101preview");
exports.v20221101preview = v20221101preview;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20231001preview = require("./v20231001preview");
exports.v20231001preview = v20231001preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20240201preview = require("./v20240201preview");
exports.v20240201preview = v20240201preview;
const v20240901 = require("./v20240901");
exports.v20240901 = v20240901;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:workloads:ACSSBackupConnection":
                return new exports.ACSSBackupConnection(name, undefined, { urn });
            case "azure-native:workloads:Alert":
                return new exports.Alert(name, undefined, { urn });
            case "azure-native:workloads:Connector":
                return new exports.Connector(name, undefined, { urn });
            case "azure-native:workloads:Monitor":
                return new exports.Monitor(name, undefined, { urn });
            case "azure-native:workloads:ProviderInstance":
                return new exports.ProviderInstance(name, undefined, { urn });
            case "azure-native:workloads:SAPApplicationServerInstance":
                return new exports.SAPApplicationServerInstance(name, undefined, { urn });
            case "azure-native:workloads:SAPCentralInstance":
                return new exports.SAPCentralInstance(name, undefined, { urn });
            case "azure-native:workloads:SAPDatabaseInstance":
                return new exports.SAPDatabaseInstance(name, undefined, { urn });
            case "azure-native:workloads:SAPVirtualInstance":
                return new exports.SAPVirtualInstance(name, undefined, { urn });
            case "azure-native:workloads:SapDiscoverySite":
                return new exports.SapDiscoverySite(name, undefined, { urn });
            case "azure-native:workloads:SapInstance":
                return new exports.SapInstance(name, undefined, { urn });
            case "azure-native:workloads:SapLandscapeMonitor":
                return new exports.SapLandscapeMonitor(name, undefined, { urn });
            case "azure-native:workloads:ServerInstance":
                return new exports.ServerInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "workloads", _module);
