// @pulumi/azure-native/types/enums/portal/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShellType = exports.ProvisioningState = exports.OsType = exports.FontStyle = exports.FontSize = exports.DashboardPartMetadataType = exports.v20250401preview = exports.v20221201preview = exports.v20200901preview = exports.v20181001 = void 0;
// Export sub-modules:
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
const v20221201preview = require("./v20221201preview");
exports.v20221201preview = v20221201preview;
const v20250401preview = require("./v20250401preview");
exports.v20250401preview = v20250401preview;
exports.DashboardPartMetadataType = {
    /**
     * The markdown part type.
     */
    Markdown: "Extension/HubsExtension/PartType/MarkdownPart",
};
exports.FontSize = {
    NotSpecified: "NotSpecified",
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
};
exports.FontStyle = {
    NotSpecified: "NotSpecified",
    Monospace: "Monospace",
    Courier: "Courier",
};
exports.OsType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ProvisioningState = {
    NotSpecified: "NotSpecified",
    Accepted: "Accepted",
    Pending: "Pending",
    Updating: "Updating",
    Creating: "Creating",
    Repairing: "Repairing",
    Failed: "Failed",
    Canceled: "Canceled",
    Succeeded: "Succeeded",
};
exports.ShellType = {
    Bash: "bash",
    Pwsh: "pwsh",
    Powershell: "powershell",
};
