// @pulumi/azure-native/verifiedid/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240126preview = exports.getAuthorityOutput = exports.getAuthority = exports.Authority = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Authority = null;
utilities.lazyLoad(exports, ["Authority"], () => require("./authority"));
exports.getAuthority = null;
exports.getAuthorityOutput = null;
utilities.lazyLoad(exports, ["getAuthority", "getAuthorityOutput"], () => require("./getAuthority"));
// Export sub-modules:
const v20240126preview = require("./v20240126preview");
exports.v20240126preview = v20240126preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:verifiedid:Authority":
                return new exports.Authority(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "verifiedid", _module);
