// @pulumi/azure-native/voiceservices/v20221201preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLine = exports.getTestLineOutput = exports.getTestLine = exports.getContactOutput = exports.getContact = exports.Contact = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Contact = null;
utilities.lazyLoad(exports, ["Contact"], () => require("./contact"));
exports.getContact = null;
exports.getContactOutput = null;
utilities.lazyLoad(exports, ["getContact", "getContactOutput"], () => require("./getContact"));
exports.getTestLine = null;
exports.getTestLineOutput = null;
utilities.lazyLoad(exports, ["getTestLine", "getTestLineOutput"], () => require("./getTestLine"));
exports.TestLine = null;
utilities.lazyLoad(exports, ["TestLine"], () => require("./testLine"));
// Export enums:
__exportStar(require("../../types/enums/voiceservices/v20221201preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:voiceservices/v20221201preview:Contact":
                return new exports.Contact(name, undefined, { urn });
            case "azure-native:voiceservices/v20221201preview:TestLine":
                return new exports.TestLine(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "voiceservices/v20221201preview", _module);
