// @pulumi/azure-native/subscription/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240801preview = exports.v20211001 = exports.v20200901 = exports.SubscriptionTarDirectory = exports.getSubscriptionTarDirectoryOutput = exports.getSubscriptionTarDirectory = exports.getAliasOutput = exports.getAlias = exports.Alias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Alias = null;
utilities.lazyLoad(exports, ["Alias"], () => require("./alias"));
exports.getAlias = null;
exports.getAliasOutput = null;
utilities.lazyLoad(exports, ["getAlias", "getAliasOutput"], () => require("./getAlias"));
exports.getSubscriptionTarDirectory = null;
exports.getSubscriptionTarDirectoryOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionTarDirectory", "getSubscriptionTarDirectoryOutput"], () => require("./getSubscriptionTarDirectory"));
exports.SubscriptionTarDirectory = null;
utilities.lazyLoad(exports, ["SubscriptionTarDirectory"], () => require("./subscriptionTarDirectory"));
// Export enums:
__exportStar(require("../types/enums/subscription"), exports);
// Export sub-modules:
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20240801preview = require("./v20240801preview");
exports.v20240801preview = v20240801preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:subscription:Alias":
                return new exports.Alias(name, undefined, { urn });
            case "azure-native:subscription:SubscriptionTarDirectory":
                return new exports.SubscriptionTarDirectory(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "subscription", _module);
