// @pulumi/azure-native/azurestackhci/v20210901preview/galleryimageRetrieve.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GalleryimageRetrieve = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The gallery image resource definition.
 */
class GalleryimageRetrieve extends pulumi.CustomResource {
    /**
     * Get an existing GalleryimageRetrieve resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GalleryimageRetrieve(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GalleryimageRetrieve.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GalleryimageRetrieve.__pulumiType;
    }
    /**
     * Create a GalleryimageRetrieve resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["cloudInitDataSource"] = args ? args.cloudInitDataSource : undefined;
            resourceInputs["containerName"] = args ? args.containerName : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["galleryimagesName"] = args ? args.galleryimagesName : undefined;
            resourceInputs["hyperVGeneration"] = args ? args.hyperVGeneration : undefined;
            resourceInputs["identifier"] = args ? args.identifier : undefined;
            resourceInputs["imagePath"] = args ? args.imagePath : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["osType"] = args ? args.osType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["cloudInitDataSource"] = undefined /*out*/;
            resourceInputs["containerName"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["hyperVGeneration"] = undefined /*out*/;
            resourceInputs["identifier"] = undefined /*out*/;
            resourceInputs["imagePath"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20210701preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20210701preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20210901preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20221215preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20221215preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20230701preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20230701preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20230901preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20230901preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240101:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240101:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240201preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240201preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240501preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240501preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240715preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240715preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240801preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20240801preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20241001preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20241001preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20250201preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20250201preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20250401preview:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci/v20250401preview:galleryimageRetrieve" }, { type: "azure-native:azurestackhci:GalleryimageRetrieve" }, { type: "azure-native:azurestackhci:galleryimageRetrieve" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GalleryimageRetrieve.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GalleryimageRetrieve = GalleryimageRetrieve;
/** @internal */
GalleryimageRetrieve.__pulumiType = 'azure-native:azurestackhci/v20210901preview:GalleryimageRetrieve';
