// @pulumi/azure-native/azurestackhci/v20210901preview/networkinterfaceRetrieve.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkinterfaceRetrieve = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The network interface resource definition.
 */
class NetworkinterfaceRetrieve extends pulumi.CustomResource {
    /**
     * Get an existing NetworkinterfaceRetrieve resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkinterfaceRetrieve(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkinterfaceRetrieve.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkinterfaceRetrieve.__pulumiType;
    }
    /**
     * Create a NetworkinterfaceRetrieve resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["dnsSettings"] = args ? args.dnsSettings : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["ipConfigurations"] = args ? args.ipConfigurations : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["macAddress"] = args ? args.macAddress : undefined;
            resourceInputs["networkinterfacesName"] = args ? args.networkinterfacesName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dnsSettings"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["ipConfigurations"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["macAddress"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20210701preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20210701preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20210901preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20221215preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20221215preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20230701preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20230701preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20230901preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20230901preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240101:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240101:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240201preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240201preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240501preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240501preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240715preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240715preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240801preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20240801preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20241001preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20241001preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20250201preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20250201preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20250401preview:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci/v20250401preview:networkinterfaceRetrieve" }, { type: "azure-native:azurestackhci:NetworkinterfaceRetrieve" }, { type: "azure-native:azurestackhci:networkinterfaceRetrieve" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkinterfaceRetrieve.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkinterfaceRetrieve = NetworkinterfaceRetrieve;
/** @internal */
NetworkinterfaceRetrieve.__pulumiType = 'azure-native:azurestackhci/v20210901preview:NetworkinterfaceRetrieve';
