// @pulumi/azure-native/maps/v20180501/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAccountKeysOutput = exports.listAccountKeys = exports.getAccountOutput = exports.getAccount = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.listAccountKeys = null;
exports.listAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listAccountKeys", "listAccountKeysOutput"], () => require("./listAccountKeys"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:maps/v20180501:Account":
                return new exports.Account(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "maps/v20180501", _module);
