// @pulumi/azure-native/azurestackhci/v20240901preview/securitySetting.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecuritySetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Security settings proxy resource
 */
class SecuritySetting extends pulumi.CustomResource {
    /**
     * Get an existing SecuritySetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecuritySetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecuritySetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecuritySetting.__pulumiType;
    }
    /**
     * Create a SecuritySetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["securedCoreComplianceAssignment"] = (_a = (args ? args.securedCoreComplianceAssignment : undefined)) !== null && _a !== void 0 ? _a : "Audit";
            resourceInputs["securitySettingsName"] = args ? args.securitySettingsName : undefined;
            resourceInputs["smbEncryptionForIntraClusterTrafficComplianceAssignment"] = (_b = (args ? args.smbEncryptionForIntraClusterTrafficComplianceAssignment : undefined)) !== null && _b !== void 0 ? _b : "Audit";
            resourceInputs["wdacComplianceAssignment"] = (_c = (args ? args.wdacComplianceAssignment : undefined)) !== null && _c !== void 0 ? _c : "Audit";
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["securityComplianceStatus"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["securedCoreComplianceAssignment"] = undefined /*out*/;
            resourceInputs["securityComplianceStatus"] = undefined /*out*/;
            resourceInputs["smbEncryptionForIntraClusterTrafficComplianceAssignment"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["wdacComplianceAssignment"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20231101preview:SecuritySetting" }, { type: "azure-native:azurestackhci/v20240101:SecuritySetting" }, { type: "azure-native:azurestackhci/v20240215preview:SecuritySetting" }, { type: "azure-native:azurestackhci/v20240401:SecuritySetting" }, { type: "azure-native:azurestackhci/v20241201preview:SecuritySetting" }, { type: "azure-native:azurestackhci:SecuritySetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SecuritySetting.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecuritySetting = SecuritySetting;
/** @internal */
SecuritySetting.__pulumiType = 'azure-native:azurestackhci/v20240901preview:SecuritySetting';
