// @pulumi/azure-native/apimanagement/v20170301/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.ProductApi = exports.Logger = exports.GroupUser = exports.getUserSharedAccessTokenOutput = exports.getUserSharedAccessToken = exports.getUserOutput = exports.getUser = exports.getLoggerOutput = exports.getLogger = exports.getApiManagementServiceSsoTokenOutput = exports.getApiManagementServiceSsoToken = exports.getApiManagementServiceOutput = exports.getApiManagementService = exports.getApiOutput = exports.getApi = exports.DiagnosticLogger = exports.ApiManagementService = exports.ApiDiagnosticLogger = exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ApiDiagnosticLogger = null;
utilities.lazyLoad(exports, ["ApiDiagnosticLogger"], () => require("./apiDiagnosticLogger"));
exports.ApiManagementService = null;
utilities.lazyLoad(exports, ["ApiManagementService"], () => require("./apiManagementService"));
exports.DiagnosticLogger = null;
utilities.lazyLoad(exports, ["DiagnosticLogger"], () => require("./diagnosticLogger"));
exports.getApi = null;
exports.getApiOutput = null;
utilities.lazyLoad(exports, ["getApi", "getApiOutput"], () => require("./getApi"));
exports.getApiManagementService = null;
exports.getApiManagementServiceOutput = null;
utilities.lazyLoad(exports, ["getApiManagementService", "getApiManagementServiceOutput"], () => require("./getApiManagementService"));
exports.getApiManagementServiceSsoToken = null;
exports.getApiManagementServiceSsoTokenOutput = null;
utilities.lazyLoad(exports, ["getApiManagementServiceSsoToken", "getApiManagementServiceSsoTokenOutput"], () => require("./getApiManagementServiceSsoToken"));
exports.getLogger = null;
exports.getLoggerOutput = null;
utilities.lazyLoad(exports, ["getLogger", "getLoggerOutput"], () => require("./getLogger"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.getUserSharedAccessToken = null;
exports.getUserSharedAccessTokenOutput = null;
utilities.lazyLoad(exports, ["getUserSharedAccessToken", "getUserSharedAccessTokenOutput"], () => require("./getUserSharedAccessToken"));
exports.GroupUser = null;
utilities.lazyLoad(exports, ["GroupUser"], () => require("./groupUser"));
exports.Logger = null;
utilities.lazyLoad(exports, ["Logger"], () => require("./logger"));
exports.ProductApi = null;
utilities.lazyLoad(exports, ["ProductApi"], () => require("./productApi"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
// Export enums:
__exportStar(require("../../types/enums/apimanagement/v20170301"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:apimanagement/v20170301:Api":
                return new exports.Api(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:ApiDiagnosticLogger":
                return new exports.ApiDiagnosticLogger(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:ApiManagementService":
                return new exports.ApiManagementService(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:DiagnosticLogger":
                return new exports.DiagnosticLogger(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:GroupUser":
                return new exports.GroupUser(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:Logger":
                return new exports.Logger(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:ProductApi":
                return new exports.ProductApi(name, undefined, { urn });
            case "azure-native:apimanagement/v20170301:User":
                return new exports.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "apimanagement/v20170301", _module);
