// @pulumi/azure-native/logic/v20180701preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workflow = exports.listWorkflowVersionTriggerCallbackUrlOutput = exports.listWorkflowVersionTriggerCallbackUrl = exports.listWorkflowTriggerCallbackUrlOutput = exports.listWorkflowTriggerCallbackUrl = exports.listWorkflowRunActionRepetitionExpressionTracesOutput = exports.listWorkflowRunActionRepetitionExpressionTraces = exports.listWorkflowRunActionExpressionTracesOutput = exports.listWorkflowRunActionExpressionTraces = exports.listWorkflowCallbackUrlOutput = exports.listWorkflowCallbackUrl = exports.getWorkflowOutput = exports.getWorkflow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWorkflow = null;
exports.getWorkflowOutput = null;
utilities.lazyLoad(exports, ["getWorkflow", "getWorkflowOutput"], () => require("./getWorkflow"));
exports.listWorkflowCallbackUrl = null;
exports.listWorkflowCallbackUrlOutput = null;
utilities.lazyLoad(exports, ["listWorkflowCallbackUrl", "listWorkflowCallbackUrlOutput"], () => require("./listWorkflowCallbackUrl"));
exports.listWorkflowRunActionExpressionTraces = null;
exports.listWorkflowRunActionExpressionTracesOutput = null;
utilities.lazyLoad(exports, ["listWorkflowRunActionExpressionTraces", "listWorkflowRunActionExpressionTracesOutput"], () => require("./listWorkflowRunActionExpressionTraces"));
exports.listWorkflowRunActionRepetitionExpressionTraces = null;
exports.listWorkflowRunActionRepetitionExpressionTracesOutput = null;
utilities.lazyLoad(exports, ["listWorkflowRunActionRepetitionExpressionTraces", "listWorkflowRunActionRepetitionExpressionTracesOutput"], () => require("./listWorkflowRunActionRepetitionExpressionTraces"));
exports.listWorkflowTriggerCallbackUrl = null;
exports.listWorkflowTriggerCallbackUrlOutput = null;
utilities.lazyLoad(exports, ["listWorkflowTriggerCallbackUrl", "listWorkflowTriggerCallbackUrlOutput"], () => require("./listWorkflowTriggerCallbackUrl"));
exports.listWorkflowVersionTriggerCallbackUrl = null;
exports.listWorkflowVersionTriggerCallbackUrlOutput = null;
utilities.lazyLoad(exports, ["listWorkflowVersionTriggerCallbackUrl", "listWorkflowVersionTriggerCallbackUrlOutput"], () => require("./listWorkflowVersionTriggerCallbackUrl"));
exports.Workflow = null;
utilities.lazyLoad(exports, ["Workflow"], () => require("./workflow"));
// Export enums:
__exportStar(require("../../types/enums/logic/v20180701preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:logic/v20180701preview:Workflow":
                return new exports.Workflow(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "logic/v20180701preview", _module);
