// @pulumi/azure-native/datareplication/v20210216preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vault = exports.ReplicationExtension = exports.ProtectedItem = exports.Policy = exports.getVaultOutput = exports.getVault = exports.getReplicationExtensionOutput = exports.getReplicationExtension = exports.getProtectedItemOutput = exports.getProtectedItem = exports.getPolicyOutput = exports.getPolicy = exports.getFabricOutput = exports.getFabric = exports.getDraOutput = exports.getDra = exports.Fabric = exports.Dra = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Dra = null;
utilities.lazyLoad(exports, ["Dra"], () => require("./dra"));
exports.Fabric = null;
utilities.lazyLoad(exports, ["Fabric"], () => require("./fabric"));
exports.getDra = null;
exports.getDraOutput = null;
utilities.lazyLoad(exports, ["getDra", "getDraOutput"], () => require("./getDra"));
exports.getFabric = null;
exports.getFabricOutput = null;
utilities.lazyLoad(exports, ["getFabric", "getFabricOutput"], () => require("./getFabric"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getProtectedItem = null;
exports.getProtectedItemOutput = null;
utilities.lazyLoad(exports, ["getProtectedItem", "getProtectedItemOutput"], () => require("./getProtectedItem"));
exports.getReplicationExtension = null;
exports.getReplicationExtensionOutput = null;
utilities.lazyLoad(exports, ["getReplicationExtension", "getReplicationExtensionOutput"], () => require("./getReplicationExtension"));
exports.getVault = null;
exports.getVaultOutput = null;
utilities.lazyLoad(exports, ["getVault", "getVaultOutput"], () => require("./getVault"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.ProtectedItem = null;
utilities.lazyLoad(exports, ["ProtectedItem"], () => require("./protectedItem"));
exports.ReplicationExtension = null;
utilities.lazyLoad(exports, ["ReplicationExtension"], () => require("./replicationExtension"));
exports.Vault = null;
utilities.lazyLoad(exports, ["Vault"], () => require("./vault"));
// Export enums:
__exportStar(require("../../types/enums/datareplication/v20210216preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datareplication/v20210216preview:Dra":
                return new exports.Dra(name, undefined, { urn });
            case "azure-native:datareplication/v20210216preview:Fabric":
                return new exports.Fabric(name, undefined, { urn });
            case "azure-native:datareplication/v20210216preview:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:datareplication/v20210216preview:ProtectedItem":
                return new exports.ProtectedItem(name, undefined, { urn });
            case "azure-native:datareplication/v20210216preview:ReplicationExtension":
                return new exports.ReplicationExtension(name, undefined, { urn });
            case "azure-native:datareplication/v20210216preview:Vault":
                return new exports.Vault(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datareplication/v20210216preview", _module);
