// @pulumi/azure-native/azurestack/v20220601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registration = exports.listProductsOutput = exports.listProducts = exports.listProductDetailsOutput = exports.listProductDetails = exports.getRegistrationActivationKeyOutput = exports.getRegistrationActivationKey = exports.getRegistrationOutput = exports.getRegistration = exports.getProductsOutput = exports.getProducts = exports.getProductOutput = exports.getProduct = exports.getCustomerSubscriptionOutput = exports.getCustomerSubscription = exports.CustomerSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CustomerSubscription = null;
utilities.lazyLoad(exports, ["CustomerSubscription"], () => require("./customerSubscription"));
exports.getCustomerSubscription = null;
exports.getCustomerSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getCustomerSubscription", "getCustomerSubscriptionOutput"], () => require("./getCustomerSubscription"));
exports.getProduct = null;
exports.getProductOutput = null;
utilities.lazyLoad(exports, ["getProduct", "getProductOutput"], () => require("./getProduct"));
exports.getProducts = null;
exports.getProductsOutput = null;
utilities.lazyLoad(exports, ["getProducts", "getProductsOutput"], () => require("./getProducts"));
exports.getRegistration = null;
exports.getRegistrationOutput = null;
utilities.lazyLoad(exports, ["getRegistration", "getRegistrationOutput"], () => require("./getRegistration"));
exports.getRegistrationActivationKey = null;
exports.getRegistrationActivationKeyOutput = null;
utilities.lazyLoad(exports, ["getRegistrationActivationKey", "getRegistrationActivationKeyOutput"], () => require("./getRegistrationActivationKey"));
exports.listProductDetails = null;
exports.listProductDetailsOutput = null;
utilities.lazyLoad(exports, ["listProductDetails", "listProductDetailsOutput"], () => require("./listProductDetails"));
exports.listProducts = null;
exports.listProductsOutput = null;
utilities.lazyLoad(exports, ["listProducts", "listProductsOutput"], () => require("./listProducts"));
exports.Registration = null;
utilities.lazyLoad(exports, ["Registration"], () => require("./registration"));
// Export enums:
__exportStar(require("../../types/enums/azurestack/v20220601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestack/v20220601:CustomerSubscription":
                return new exports.CustomerSubscription(name, undefined, { urn });
            case "azure-native:azurestack/v20220601:Registration":
                return new exports.Registration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestack/v20220601", _module);
