// @pulumi/azure-native/dbforpostgresql/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20221108 = exports.v20220501preview = exports.v20220308privatepreview = exports.v20220308preview = exports.v20220120preview = exports.v20210615privatepreview = exports.v20210601 = exports.v20210410privatepreview = exports.v20210331privatepreview = exports.v20201005privatepreview = exports.v20200214privatepreview = exports.v20200214preview = exports.v20180601privatepreview = exports.v20171201preview = exports.v20171201 = exports.VirtualEndpoint = exports.Server = exports.Role = exports.PrivateEndpointConnection = exports.Migration = exports.getVirtualEndpointOutput = exports.getVirtualEndpoint = exports.getServerOutput = exports.getServer = exports.getRoleOutput = exports.getRole = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getMigrationOutput = exports.getMigration = exports.getGetPrivateDnsZoneSuffixExecuteOutput = exports.getGetPrivateDnsZoneSuffixExecute = exports.getFirewallRuleOutput = exports.getFirewallRule = exports.getDatabaseOutput = exports.getDatabase = exports.getConfigurationOutput = exports.getConfiguration = exports.getClusterOutput = exports.getCluster = exports.getBackupOutput = exports.getBackup = exports.getAdministratorOutput = exports.getAdministrator = exports.FirewallRule = exports.Database = exports.Configuration = exports.Cluster = exports.Backup = exports.Administrator = void 0;
exports.v20241101preview = exports.v20240801 = exports.v20240301preview = exports.v20231201preview = exports.v20230601preview = exports.v20230302preview = exports.v20230301preview = exports.v20221201 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Administrator = null;
utilities.lazyLoad(exports, ["Administrator"], () => require("./administrator"));
exports.Backup = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.getAdministrator = null;
exports.getAdministratorOutput = null;
utilities.lazyLoad(exports, ["getAdministrator", "getAdministratorOutput"], () => require("./getAdministrator"));
exports.getBackup = null;
exports.getBackupOutput = null;
utilities.lazyLoad(exports, ["getBackup", "getBackupOutput"], () => require("./getBackup"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getDatabase = null;
exports.getDatabaseOutput = null;
utilities.lazyLoad(exports, ["getDatabase", "getDatabaseOutput"], () => require("./getDatabase"));
exports.getFirewallRule = null;
exports.getFirewallRuleOutput = null;
utilities.lazyLoad(exports, ["getFirewallRule", "getFirewallRuleOutput"], () => require("./getFirewallRule"));
exports.getGetPrivateDnsZoneSuffixExecute = null;
exports.getGetPrivateDnsZoneSuffixExecuteOutput = null;
utilities.lazyLoad(exports, ["getGetPrivateDnsZoneSuffixExecute", "getGetPrivateDnsZoneSuffixExecuteOutput"], () => require("./getGetPrivateDnsZoneSuffixExecute"));
exports.getMigration = null;
exports.getMigrationOutput = null;
utilities.lazyLoad(exports, ["getMigration", "getMigrationOutput"], () => require("./getMigration"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getRole = null;
exports.getRoleOutput = null;
utilities.lazyLoad(exports, ["getRole", "getRoleOutput"], () => require("./getRole"));
exports.getServer = null;
exports.getServerOutput = null;
utilities.lazyLoad(exports, ["getServer", "getServerOutput"], () => require("./getServer"));
exports.getVirtualEndpoint = null;
exports.getVirtualEndpointOutput = null;
utilities.lazyLoad(exports, ["getVirtualEndpoint", "getVirtualEndpointOutput"], () => require("./getVirtualEndpoint"));
exports.Migration = null;
utilities.lazyLoad(exports, ["Migration"], () => require("./migration"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Role = null;
utilities.lazyLoad(exports, ["Role"], () => require("./role"));
exports.Server = null;
utilities.lazyLoad(exports, ["Server"], () => require("./server"));
exports.VirtualEndpoint = null;
utilities.lazyLoad(exports, ["VirtualEndpoint"], () => require("./virtualEndpoint"));
// Export enums:
__exportStar(require("../types/enums/dbforpostgresql"), exports);
// Export sub-modules:
const v20171201 = require("./v20171201");
exports.v20171201 = v20171201;
const v20171201preview = require("./v20171201preview");
exports.v20171201preview = v20171201preview;
const v20180601privatepreview = require("./v20180601privatepreview");
exports.v20180601privatepreview = v20180601privatepreview;
const v20200214preview = require("./v20200214preview");
exports.v20200214preview = v20200214preview;
const v20200214privatepreview = require("./v20200214privatepreview");
exports.v20200214privatepreview = v20200214privatepreview;
const v20201005privatepreview = require("./v20201005privatepreview");
exports.v20201005privatepreview = v20201005privatepreview;
const v20210331privatepreview = require("./v20210331privatepreview");
exports.v20210331privatepreview = v20210331privatepreview;
const v20210410privatepreview = require("./v20210410privatepreview");
exports.v20210410privatepreview = v20210410privatepreview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210615privatepreview = require("./v20210615privatepreview");
exports.v20210615privatepreview = v20210615privatepreview;
const v20220120preview = require("./v20220120preview");
exports.v20220120preview = v20220120preview;
const v20220308preview = require("./v20220308preview");
exports.v20220308preview = v20220308preview;
const v20220308privatepreview = require("./v20220308privatepreview");
exports.v20220308privatepreview = v20220308privatepreview;
const v20220501preview = require("./v20220501preview");
exports.v20220501preview = v20220501preview;
const v20221108 = require("./v20221108");
exports.v20221108 = v20221108;
const v20221201 = require("./v20221201");
exports.v20221201 = v20221201;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230302preview = require("./v20230302preview");
exports.v20230302preview = v20230302preview;
const v20230601preview = require("./v20230601preview");
exports.v20230601preview = v20230601preview;
const v20231201preview = require("./v20231201preview");
exports.v20231201preview = v20231201preview;
const v20240301preview = require("./v20240301preview");
exports.v20240301preview = v20240301preview;
const v20240801 = require("./v20240801");
exports.v20240801 = v20240801;
const v20241101preview = require("./v20241101preview");
exports.v20241101preview = v20241101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbforpostgresql:Administrator":
                return new exports.Administrator(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Database":
                return new exports.Database(name, undefined, { urn });
            case "azure-native:dbforpostgresql:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Migration":
                return new exports.Migration(name, undefined, { urn });
            case "azure-native:dbforpostgresql:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Role":
                return new exports.Role(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Server":
                return new exports.Server(name, undefined, { urn });
            case "azure-native:dbforpostgresql:VirtualEndpoint":
                return new exports.VirtualEndpoint(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbforpostgresql", _module);
