// @pulumi/azure-native/types/enums/codesigning/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.ProfileType = exports.v20240930preview = exports.v20240205preview = void 0;
// Export sub-modules:
const v20240205preview = require("./v20240205preview");
exports.v20240205preview = v20240205preview;
const v20240930preview = require("./v20240930preview");
exports.v20240930preview = v20240930preview;
exports.ProfileType = {
    /**
     * Used for signing files which are distributed publicly.
     */
    PublicTrust: "PublicTrust",
    /**
     * Used for signing files which are distributed internally within organization or group boundary.
     */
    PrivateTrust: "PrivateTrust",
    /**
     * Used for signing CI policy files.
     */
    PrivateTrustCIPolicy: "PrivateTrustCIPolicy",
    /**
     * Used for signing files which are run in secure vbs enclave.
     */
    VBSEnclave: "VBSEnclave",
    /**
     * Used for signing files for testing purpose.
     */
    PublicTrustTest: "PublicTrustTest",
};
exports.SkuName = {
    /**
     * Basic sku.
     */
    Basic: "Basic",
    /**
     * Premium sku.
     */
    Premium: "Premium",
};
