// @pulumi/azure-native/engagementfabric/v20180901preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAccountKeysOutput = exports.listAccountKeys = exports.listAccountChannelTypesOutput = exports.listAccountChannelTypes = exports.getChannelOutput = exports.getChannel = exports.getAccountOutput = exports.getAccount = exports.Channel = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.listAccountChannelTypes = null;
exports.listAccountChannelTypesOutput = null;
utilities.lazyLoad(exports, ["listAccountChannelTypes", "listAccountChannelTypesOutput"], () => require("./listAccountChannelTypes"));
exports.listAccountKeys = null;
exports.listAccountKeysOutput = null;
utilities.lazyLoad(exports, ["listAccountKeys", "listAccountKeysOutput"], () => require("./listAccountKeys"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:engagementfabric/v20180901preview:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:engagementfabric/v20180901preview:Channel":
                return new exports.Channel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "engagementfabric/v20180901preview", _module);
