// @pulumi/azure-native/managednetworkfabric/networkToNetworkInterconnect.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkToNetworkInterconnect = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * The NetworkToNetworkInterconnect resource definition.
 *
 * Uses Azure REST API version 2023-02-01-preview. In version 1.x of the Azure Native provider, it used API version 2023-02-01-preview.
 *
 * Other available API versions: 2023-06-15.
 */
class NetworkToNetworkInterconnect extends pulumi.CustomResource {
    /**
     * Get an existing NetworkToNetworkInterconnect resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkToNetworkInterconnect(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkToNetworkInterconnect.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkToNetworkInterconnect.__pulumiType;
    }
    /**
     * Create a NetworkToNetworkInterconnect resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.isManagementType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'isManagementType'");
            }
            if ((!args || args.networkFabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkFabricName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.useOptionB === undefined) && !opts.urn) {
                throw new Error("Missing required property 'useOptionB'");
            }
            resourceInputs["isManagementType"] = args ? args.isManagementType : undefined;
            resourceInputs["layer2Configuration"] = args ? (args.layer2Configuration ? pulumi.output(args.layer2Configuration).apply(inputs.managednetworkfabric.layer2ConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["layer3Configuration"] = args ? args.layer3Configuration : undefined;
            resourceInputs["networkFabricName"] = args ? args.networkFabricName : undefined;
            resourceInputs["networkToNetworkInterconnectName"] = args ? args.networkToNetworkInterconnectName : undefined;
            resourceInputs["nniType"] = (_a = (args ? args.nniType : undefined)) !== null && _a !== void 0 ? _a : "CE";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["useOptionB"] = args ? args.useOptionB : undefined;
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["isManagementType"] = undefined /*out*/;
            resourceInputs["layer2Configuration"] = undefined /*out*/;
            resourceInputs["layer3Configuration"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nniType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useOptionB"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:managednetworkfabric/v20230201preview:NetworkToNetworkInterconnect" }, { type: "azure-native:managednetworkfabric/v20230615:NetworkToNetworkInterconnect" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkToNetworkInterconnect.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkToNetworkInterconnect = NetworkToNetworkInterconnect;
/** @internal */
NetworkToNetworkInterconnect.__pulumiType = 'azure-native:managednetworkfabric:NetworkToNetworkInterconnect';
