// @pulumi/azure-native/botservice/v20180712/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listBotConnectionServiceProvidersOutput = exports.listBotConnectionServiceProviders = exports.getEnterpriseChannelOutput = exports.getEnterpriseChannel = exports.EnterpriseChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.EnterpriseChannel = null;
utilities.lazyLoad(exports, ["EnterpriseChannel"], () => require("./enterpriseChannel"));
exports.getEnterpriseChannel = null;
exports.getEnterpriseChannelOutput = null;
utilities.lazyLoad(exports, ["getEnterpriseChannel", "getEnterpriseChannelOutput"], () => require("./getEnterpriseChannel"));
exports.listBotConnectionServiceProviders = null;
exports.listBotConnectionServiceProvidersOutput = null;
utilities.lazyLoad(exports, ["listBotConnectionServiceProviders", "listBotConnectionServiceProvidersOutput"], () => require("./listBotConnectionServiceProviders"));
// Export enums:
__exportStar(require("../../types/enums/botservice/v20180712"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:botservice/v20180712:EnterpriseChannel":
                return new exports.EnterpriseChannel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "botservice/v20180712", _module);
