// @pulumi/azure-native/programmableconnectivity/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240115preview = exports.OperatorApiConnection = exports.getOperatorApiConnectionOutput = exports.getOperatorApiConnection = exports.getGatewayOutput = exports.getGateway = exports.Gateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.getGateway = null;
exports.getGatewayOutput = null;
utilities.lazyLoad(exports, ["getGateway", "getGatewayOutput"], () => require("./getGateway"));
exports.getOperatorApiConnection = null;
exports.getOperatorApiConnectionOutput = null;
utilities.lazyLoad(exports, ["getOperatorApiConnection", "getOperatorApiConnectionOutput"], () => require("./getOperatorApiConnection"));
exports.OperatorApiConnection = null;
utilities.lazyLoad(exports, ["OperatorApiConnection"], () => require("./operatorApiConnection"));
// Export enums:
__exportStar(require("../types/enums/programmableconnectivity"), exports);
// Export sub-modules:
const v20240115preview = require("./v20240115preview");
exports.v20240115preview = v20240115preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:programmableconnectivity:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "azure-native:programmableconnectivity:OperatorApiConnection":
                return new exports.OperatorApiConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "programmableconnectivity", _module);
