// @pulumi/azure-native/costmanagement/v20221005preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewByScope = exports.View = exports.TagInheritanceSetting = exports.MarkupRule = exports.getViewByScopeOutput = exports.getViewByScope = exports.getViewOutput = exports.getView = exports.getTagInheritanceSettingOutput = exports.getTagInheritanceSetting = exports.getMarkupRuleOutput = exports.getMarkupRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getMarkupRule = null;
exports.getMarkupRuleOutput = null;
utilities.lazyLoad(exports, ["getMarkupRule", "getMarkupRuleOutput"], () => require("./getMarkupRule"));
exports.getTagInheritanceSetting = null;
exports.getTagInheritanceSettingOutput = null;
utilities.lazyLoad(exports, ["getTagInheritanceSetting", "getTagInheritanceSettingOutput"], () => require("./getTagInheritanceSetting"));
exports.getView = null;
exports.getViewOutput = null;
utilities.lazyLoad(exports, ["getView", "getViewOutput"], () => require("./getView"));
exports.getViewByScope = null;
exports.getViewByScopeOutput = null;
utilities.lazyLoad(exports, ["getViewByScope", "getViewByScopeOutput"], () => require("./getViewByScope"));
exports.MarkupRule = null;
utilities.lazyLoad(exports, ["MarkupRule"], () => require("./markupRule"));
exports.TagInheritanceSetting = null;
utilities.lazyLoad(exports, ["TagInheritanceSetting"], () => require("./tagInheritanceSetting"));
exports.View = null;
utilities.lazyLoad(exports, ["View"], () => require("./view"));
exports.ViewByScope = null;
utilities.lazyLoad(exports, ["ViewByScope"], () => require("./viewByScope"));
// Export enums:
__exportStar(require("../../types/enums/costmanagement/v20221005preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:costmanagement/v20221005preview:MarkupRule":
                return new exports.MarkupRule(name, undefined, { urn });
            case "azure-native:costmanagement/v20221005preview:TagInheritanceSetting":
                return new exports.TagInheritanceSetting(name, undefined, { urn });
            case "azure-native:costmanagement/v20221005preview:View":
                return new exports.View(name, undefined, { urn });
            case "azure-native:costmanagement/v20221005preview:ViewByScope":
                return new exports.ViewByScope(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "costmanagement/v20221005preview", _module);
