// @pulumi/azure-native/web/staticSitePrivateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticSitePrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Remote Private Endpoint Connection ARM resource.
 *
 * Uses Azure REST API version 2022-09-01. In version 1.x of the Azure Native provider, it used API version 2020-12-01.
 *
 * Other available API versions: 2023-01-01, 2023-12-01, 2024-04-01.
 */
class StaticSitePrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing StaticSitePrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StaticSitePrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StaticSitePrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StaticSitePrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a StaticSitePrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["ipAddresses"] = args ? args.ipAddresses : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["ipAddresses"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20201201:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20210101:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20210115:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20210201:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20210301:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20220301:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20220901:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20230101:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20231201:StaticSitePrivateEndpointConnection" }, { type: "azure-native:web/v20240401:StaticSitePrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StaticSitePrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StaticSitePrivateEndpointConnection = StaticSitePrivateEndpointConnection;
/** @internal */
StaticSitePrivateEndpointConnection.__pulumiType = 'azure-native:web:StaticSitePrivateEndpointConnection';
