// @pulumi/azure-native/web/webAppMetadata.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppMetadata = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * String dictionary resource.
 *
 * Uses Azure REST API version 2022-09-01. In version 1.x of the Azure Native provider, it used API version 2020-12-01.
 *
 * Other available API versions: 2020-10-01, 2023-01-01, 2023-12-01, 2024-04-01.
 */
class WebAppMetadata extends pulumi.CustomResource {
    /**
     * Get an existing WebAppMetadata resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppMetadata(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppMetadata.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppMetadata.__pulumiType;
    }
    /**
     * Create a WebAppMetadata resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppMetadata" }, { type: "azure-native:web/v20160801:WebAppMetadata" }, { type: "azure-native:web/v20180201:WebAppMetadata" }, { type: "azure-native:web/v20181101:WebAppMetadata" }, { type: "azure-native:web/v20190801:WebAppMetadata" }, { type: "azure-native:web/v20200601:WebAppMetadata" }, { type: "azure-native:web/v20200901:WebAppMetadata" }, { type: "azure-native:web/v20201001:WebAppMetadata" }, { type: "azure-native:web/v20201201:WebAppMetadata" }, { type: "azure-native:web/v20210101:WebAppMetadata" }, { type: "azure-native:web/v20210115:WebAppMetadata" }, { type: "azure-native:web/v20210201:WebAppMetadata" }, { type: "azure-native:web/v20210301:WebAppMetadata" }, { type: "azure-native:web/v20220301:WebAppMetadata" }, { type: "azure-native:web/v20220901:WebAppMetadata" }, { type: "azure-native:web/v20230101:WebAppMetadata" }, { type: "azure-native:web/v20231201:WebAppMetadata" }, { type: "azure-native:web/v20240401:WebAppMetadata" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppMetadata.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppMetadata = WebAppMetadata;
/** @internal */
WebAppMetadata.__pulumiType = 'azure-native:web:WebAppMetadata';
