// @pulumi/azure-native/web/webAppHybridConnectionSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppHybridConnectionSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Hybrid Connection contract. This is used to configure a Hybrid Connection.
 *
 * Uses Azure REST API version 2022-09-01. In version 1.x of the Azure Native provider, it used API version 2020-12-01.
 *
 * Other available API versions: 2020-10-01, 2023-01-01, 2023-12-01, 2024-04-01.
 */
class WebAppHybridConnectionSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppHybridConnectionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppHybridConnectionSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppHybridConnectionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppHybridConnectionSlot.__pulumiType;
    }
    /**
     * Create a WebAppHybridConnectionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.namespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'namespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["hostname"] = args ? args.hostname : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["namespaceName"] = args ? args.namespaceName : undefined;
            resourceInputs["port"] = args ? args.port : undefined;
            resourceInputs["relayArmUri"] = args ? args.relayArmUri : undefined;
            resourceInputs["relayName"] = args ? args.relayName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sendKeyName"] = args ? args.sendKeyName : undefined;
            resourceInputs["sendKeyValue"] = args ? args.sendKeyValue : undefined;
            resourceInputs["serviceBusNamespace"] = args ? args.serviceBusNamespace : undefined;
            resourceInputs["serviceBusSuffix"] = args ? args.serviceBusSuffix : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["hostname"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["relayArmUri"] = undefined /*out*/;
            resourceInputs["relayName"] = undefined /*out*/;
            resourceInputs["sendKeyName"] = undefined /*out*/;
            resourceInputs["sendKeyValue"] = undefined /*out*/;
            resourceInputs["serviceBusNamespace"] = undefined /*out*/;
            resourceInputs["serviceBusSuffix"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20160801:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20180201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20181101:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20190801:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20200601:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20200901:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20201001:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20201201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210101:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210115:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210301:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20220301:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20220901:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20230101:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20231201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20240401:WebAppHybridConnectionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppHybridConnectionSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppHybridConnectionSlot = WebAppHybridConnectionSlot;
/** @internal */
WebAppHybridConnectionSlot.__pulumiType = 'azure-native:web:WebAppHybridConnectionSlot';
