// @pulumi/azure-native/billing/v20191001preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.listBillingAccountInvoiceSectionsByCreateSubscriptionPermissionOutput = exports.listBillingAccountInvoiceSectionsByCreateSubscriptionPermission = exports.getBillingRoleAssignmentByEnrollmentAccountOutput = exports.getBillingRoleAssignmentByEnrollmentAccount = exports.getBillingRoleAssignmentByDepartmentOutput = exports.getBillingRoleAssignmentByDepartment = exports.getBillingRoleAssignmentByBillingAccountOutput = exports.getBillingRoleAssignmentByBillingAccount = exports.BillingRoleAssignmentByEnrollmentAccount = exports.BillingRoleAssignmentByDepartment = exports.BillingRoleAssignmentByBillingAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.BillingRoleAssignmentByBillingAccount = null;
utilities.lazyLoad(exports, ["BillingRoleAssignmentByBillingAccount"], () => require("./billingRoleAssignmentByBillingAccount"));
exports.BillingRoleAssignmentByDepartment = null;
utilities.lazyLoad(exports, ["BillingRoleAssignmentByDepartment"], () => require("./billingRoleAssignmentByDepartment"));
exports.BillingRoleAssignmentByEnrollmentAccount = null;
utilities.lazyLoad(exports, ["BillingRoleAssignmentByEnrollmentAccount"], () => require("./billingRoleAssignmentByEnrollmentAccount"));
exports.getBillingRoleAssignmentByBillingAccount = null;
exports.getBillingRoleAssignmentByBillingAccountOutput = null;
utilities.lazyLoad(exports, ["getBillingRoleAssignmentByBillingAccount", "getBillingRoleAssignmentByBillingAccountOutput"], () => require("./getBillingRoleAssignmentByBillingAccount"));
exports.getBillingRoleAssignmentByDepartment = null;
exports.getBillingRoleAssignmentByDepartmentOutput = null;
utilities.lazyLoad(exports, ["getBillingRoleAssignmentByDepartment", "getBillingRoleAssignmentByDepartmentOutput"], () => require("./getBillingRoleAssignmentByDepartment"));
exports.getBillingRoleAssignmentByEnrollmentAccount = null;
exports.getBillingRoleAssignmentByEnrollmentAccountOutput = null;
utilities.lazyLoad(exports, ["getBillingRoleAssignmentByEnrollmentAccount", "getBillingRoleAssignmentByEnrollmentAccountOutput"], () => require("./getBillingRoleAssignmentByEnrollmentAccount"));
exports.listBillingAccountInvoiceSectionsByCreateSubscriptionPermission = null;
exports.listBillingAccountInvoiceSectionsByCreateSubscriptionPermissionOutput = null;
utilities.lazyLoad(exports, ["listBillingAccountInvoiceSectionsByCreateSubscriptionPermission", "listBillingAccountInvoiceSectionsByCreateSubscriptionPermissionOutput"], () => require("./listBillingAccountInvoiceSectionsByCreateSubscriptionPermission"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:billing/v20191001preview:BillingRoleAssignmentByBillingAccount":
                return new exports.BillingRoleAssignmentByBillingAccount(name, undefined, { urn });
            case "azure-native:billing/v20191001preview:BillingRoleAssignmentByDepartment":
                return new exports.BillingRoleAssignmentByDepartment(name, undefined, { urn });
            case "azure-native:billing/v20191001preview:BillingRoleAssignmentByEnrollmentAccount":
                return new exports.BillingRoleAssignmentByEnrollmentAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "billing/v20191001preview", _module);
