// @pulumi/azure-native/servicefabric/v20210601/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = exports.listClusterUpgradableVersionsOutput = exports.listClusterUpgradableVersions = exports.getServiceOutput = exports.getService = exports.getClusterOutput = exports.getCluster = exports.getApplicationTypeVersionOutput = exports.getApplicationTypeVersion = exports.getApplicationTypeOutput = exports.getApplicationType = exports.getApplicationOutput = exports.getApplication = exports.Cluster = exports.ApplicationTypeVersion = exports.ApplicationType = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationType = null;
utilities.lazyLoad(exports, ["ApplicationType"], () => require("./applicationType"));
exports.ApplicationTypeVersion = null;
utilities.lazyLoad(exports, ["ApplicationTypeVersion"], () => require("./applicationTypeVersion"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getApplicationType = null;
exports.getApplicationTypeOutput = null;
utilities.lazyLoad(exports, ["getApplicationType", "getApplicationTypeOutput"], () => require("./getApplicationType"));
exports.getApplicationTypeVersion = null;
exports.getApplicationTypeVersionOutput = null;
utilities.lazyLoad(exports, ["getApplicationTypeVersion", "getApplicationTypeVersionOutput"], () => require("./getApplicationTypeVersion"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getService = null;
exports.getServiceOutput = null;
utilities.lazyLoad(exports, ["getService", "getServiceOutput"], () => require("./getService"));
exports.listClusterUpgradableVersions = null;
exports.listClusterUpgradableVersionsOutput = null;
utilities.lazyLoad(exports, ["listClusterUpgradableVersions", "listClusterUpgradableVersionsOutput"], () => require("./listClusterUpgradableVersions"));
exports.Service = null;
utilities.lazyLoad(exports, ["Service"], () => require("./service"));
// Export enums:
__exportStar(require("../../types/enums/servicefabric/v20210601"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicefabric/v20210601:Application":
                return new exports.Application(name, undefined, { urn });
            case "azure-native:servicefabric/v20210601:ApplicationType":
                return new exports.ApplicationType(name, undefined, { urn });
            case "azure-native:servicefabric/v20210601:ApplicationTypeVersion":
                return new exports.ApplicationTypeVersion(name, undefined, { urn });
            case "azure-native:servicefabric/v20210601:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "azure-native:servicefabric/v20210601:Service":
                return new exports.Service(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicefabric/v20210601", _module);
