// @pulumi/azure-native/migrate/v20191001/hyperVCollector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HyperVCollector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class HyperVCollector extends pulumi.CustomResource {
    /**
     * Get an existing HyperVCollector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HyperVCollector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HyperVCollector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HyperVCollector.__pulumiType;
    }
    /**
     * Create a HyperVCollector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["hyperVCollectorName"] = args ? args.hyperVCollectorName : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20230315:HyperVCollector" }, { type: "azure-native:migrate/v20230401preview:HyperVCollector" }, { type: "azure-native:migrate/v20230501preview:HyperVCollector" }, { type: "azure-native:migrate/v20230909preview:HyperVCollector" }, { type: "azure-native:migrate/v20240101preview:HyperVCollector" }, { type: "azure-native:migrate:HyperVCollector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HyperVCollector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HyperVCollector = HyperVCollector;
/** @internal */
HyperVCollector.__pulumiType = 'azure-native:migrate/v20191001:HyperVCollector';
