// @pulumi/azure-native/migrate/v20191001/importCollector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportCollector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class ImportCollector extends pulumi.CustomResource {
    /**
     * Get an existing ImportCollector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ImportCollector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ImportCollector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ImportCollector.__pulumiType;
    }
    /**
     * Create a ImportCollector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["importCollectorName"] = args ? args.importCollectorName : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20230315:ImportCollector" }, { type: "azure-native:migrate/v20230401preview:ImportCollector" }, { type: "azure-native:migrate/v20230501preview:ImportCollector" }, { type: "azure-native:migrate/v20230909preview:ImportCollector" }, { type: "azure-native:migrate/v20240101preview:ImportCollector" }, { type: "azure-native:migrate:ImportCollector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ImportCollector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ImportCollector = ImportCollector;
/** @internal */
ImportCollector.__pulumiType = 'azure-native:migrate/v20191001:ImportCollector';
