// @pulumi/azure-native/types/enums/network/v20200401/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.FirewallPolicyRuleType = exports.FirewallPolicyRuleConditionType = exports.FirewallPolicyRuleConditionNetworkProtocol = exports.FirewallPolicyRuleConditionApplicationProtocolType = exports.FirewallPolicyNatRuleActionType = exports.FirewallPolicyFilterRuleActionType = exports.AzureFirewallThreatIntelMode = exports.AzureFirewallSkuTier = exports.AzureFirewallSkuName = exports.AzureFirewallRCActionType = exports.AzureFirewallNetworkRuleProtocol = exports.AzureFirewallNatRCActionType = exports.AzureFirewallApplicationRuleProtocolType = void 0;
exports.AzureFirewallApplicationRuleProtocolType = {
    Http: "Http",
    Https: "Https",
    Mssql: "Mssql",
};
exports.AzureFirewallNatRCActionType = {
    Snat: "Snat",
    Dnat: "Dnat",
};
exports.AzureFirewallNetworkRuleProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
    ICMP: "ICMP",
};
exports.AzureFirewallRCActionType = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.AzureFirewallSkuName = {
    AZFW_VNet: "AZFW_VNet",
    AZFW_Hub: "AZFW_Hub",
};
exports.AzureFirewallSkuTier = {
    Standard: "Standard",
    Premium: "Premium",
};
exports.AzureFirewallThreatIntelMode = {
    Alert: "Alert",
    Deny: "Deny",
    Off: "Off",
};
exports.FirewallPolicyFilterRuleActionType = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FirewallPolicyNatRuleActionType = {
    DNAT: "DNAT",
};
exports.FirewallPolicyRuleConditionApplicationProtocolType = {
    Http: "Http",
    Https: "Https",
};
exports.FirewallPolicyRuleConditionNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
    Any: "Any",
    ICMP: "ICMP",
};
exports.FirewallPolicyRuleConditionType = {
    ApplicationRuleCondition: "ApplicationRuleCondition",
    NetworkRuleCondition: "NetworkRuleCondition",
    NatRuleCondition: "NatRuleCondition",
};
exports.FirewallPolicyRuleType = {
    FirewallPolicyNatRule: "FirewallPolicyNatRule",
    FirewallPolicyFilterRule: "FirewallPolicyFilterRule",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
