// @pulumi/azure-native/types/enums/security/v20240801/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardSupportedCloud = exports.SeverityEnum = exports.SecurityIssue = exports.RecommendationSupportedClouds = exports.ExemptionCategory = exports.Effect = exports.AttestationComplianceState = void 0;
exports.AttestationComplianceState = {
    Unknown: "unknown",
    Compliant: "compliant",
    NonCompliant: "nonCompliant",
};
exports.Effect = {
    Audit: "Audit",
    Exempt: "Exempt",
    Attest: "Attest",
};
exports.ExemptionCategory = {
    Waiver: "waiver",
    Mitigated: "mitigated",
};
exports.RecommendationSupportedClouds = {
    Azure: "Azure",
    AWS: "AWS",
    GCP: "GCP",
};
exports.SecurityIssue = {
    Vulnerability: "Vulnerability",
    ExcessivePermissions: "ExcessivePermissions",
    AnonymousAccess: "AnonymousAccess",
    NetworkExposure: "NetworkExposure",
    TrafficEncryption: "TrafficEncryption",
    BestPractices: "BestPractices",
};
exports.SeverityEnum = {
    High: "High",
    Medium: "Medium",
    Low: "Low",
};
exports.StandardSupportedCloud = {
    Azure: "Azure",
    AWS: "AWS",
    GCP: "GCP",
};
