// @pulumi/azure-native/hardwaresecuritymodules/v20231210preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCloudHsmClusterPrivateEndpointConnectionOutput = exports.getCloudHsmClusterPrivateEndpointConnection = exports.getCloudHsmClusterOutput = exports.getCloudHsmCluster = exports.CloudHsmClusterPrivateEndpointConnection = exports.CloudHsmCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.CloudHsmCluster = null;
utilities.lazyLoad(exports, ["CloudHsmCluster"], () => require("./cloudHsmCluster"));
exports.CloudHsmClusterPrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["CloudHsmClusterPrivateEndpointConnection"], () => require("./cloudHsmClusterPrivateEndpointConnection"));
exports.getCloudHsmCluster = null;
exports.getCloudHsmClusterOutput = null;
utilities.lazyLoad(exports, ["getCloudHsmCluster", "getCloudHsmClusterOutput"], () => require("./getCloudHsmCluster"));
exports.getCloudHsmClusterPrivateEndpointConnection = null;
exports.getCloudHsmClusterPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getCloudHsmClusterPrivateEndpointConnection", "getCloudHsmClusterPrivateEndpointConnectionOutput"], () => require("./getCloudHsmClusterPrivateEndpointConnection"));
// Export enums:
__exportStar(require("../../types/enums/hardwaresecuritymodules/v20231210preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hardwaresecuritymodules/v20231210preview:CloudHsmCluster":
                return new exports.CloudHsmCluster(name, undefined, { urn });
            case "azure-native:hardwaresecuritymodules/v20231210preview:CloudHsmClusterPrivateEndpointConnection":
                return new exports.CloudHsmClusterPrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hardwaresecuritymodules/v20231210preview", _module);
