// @pulumi/azure-native/compute/v20211101/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineScaleSetVMExtension = exports.VirtualMachineScaleSetExtension = exports.VirtualMachineExtension = exports.RestorePoint = exports.getVirtualMachineScaleSetVMExtensionOutput = exports.getVirtualMachineScaleSetVMExtension = exports.getVirtualMachineScaleSetExtensionOutput = exports.getVirtualMachineScaleSetExtension = exports.getVirtualMachineExtensionOutput = exports.getVirtualMachineExtension = exports.getRestorePointOutput = exports.getRestorePoint = exports.getLogAnalyticExportThrottledRequestsOutput = exports.getLogAnalyticExportThrottledRequests = exports.getLogAnalyticExportRequestRateByIntervalOutput = exports.getLogAnalyticExportRequestRateByInterval = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getLogAnalyticExportRequestRateByInterval = null;
exports.getLogAnalyticExportRequestRateByIntervalOutput = null;
utilities.lazyLoad(exports, ["getLogAnalyticExportRequestRateByInterval", "getLogAnalyticExportRequestRateByIntervalOutput"], () => require("./getLogAnalyticExportRequestRateByInterval"));
exports.getLogAnalyticExportThrottledRequests = null;
exports.getLogAnalyticExportThrottledRequestsOutput = null;
utilities.lazyLoad(exports, ["getLogAnalyticExportThrottledRequests", "getLogAnalyticExportThrottledRequestsOutput"], () => require("./getLogAnalyticExportThrottledRequests"));
exports.getRestorePoint = null;
exports.getRestorePointOutput = null;
utilities.lazyLoad(exports, ["getRestorePoint", "getRestorePointOutput"], () => require("./getRestorePoint"));
exports.getVirtualMachineExtension = null;
exports.getVirtualMachineExtensionOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineExtension", "getVirtualMachineExtensionOutput"], () => require("./getVirtualMachineExtension"));
exports.getVirtualMachineScaleSetExtension = null;
exports.getVirtualMachineScaleSetExtensionOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineScaleSetExtension", "getVirtualMachineScaleSetExtensionOutput"], () => require("./getVirtualMachineScaleSetExtension"));
exports.getVirtualMachineScaleSetVMExtension = null;
exports.getVirtualMachineScaleSetVMExtensionOutput = null;
utilities.lazyLoad(exports, ["getVirtualMachineScaleSetVMExtension", "getVirtualMachineScaleSetVMExtensionOutput"], () => require("./getVirtualMachineScaleSetVMExtension"));
exports.RestorePoint = null;
utilities.lazyLoad(exports, ["RestorePoint"], () => require("./restorePoint"));
exports.VirtualMachineExtension = null;
utilities.lazyLoad(exports, ["VirtualMachineExtension"], () => require("./virtualMachineExtension"));
exports.VirtualMachineScaleSetExtension = null;
utilities.lazyLoad(exports, ["VirtualMachineScaleSetExtension"], () => require("./virtualMachineScaleSetExtension"));
exports.VirtualMachineScaleSetVMExtension = null;
utilities.lazyLoad(exports, ["VirtualMachineScaleSetVMExtension"], () => require("./virtualMachineScaleSetVMExtension"));
// Export enums:
__exportStar(require("../../types/enums/compute/v20211101"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:compute/v20211101:RestorePoint":
                return new exports.RestorePoint(name, undefined, { urn });
            case "azure-native:compute/v20211101:VirtualMachineExtension":
                return new exports.VirtualMachineExtension(name, undefined, { urn });
            case "azure-native:compute/v20211101:VirtualMachineScaleSetExtension":
                return new exports.VirtualMachineScaleSetExtension(name, undefined, { urn });
            case "azure-native:compute/v20211101:VirtualMachineScaleSetVMExtension":
                return new exports.VirtualMachineScaleSetVMExtension(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "compute/v20211101", _module);
