// @pulumi/azure-native/hybridcontainerservice/v20231115preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VMSkus = exports.VirtualNetworkRetrieve = exports.ProvisionedClusterInstance = exports.listprovisionedClusterInstanceUserKubeconfigOutput = exports.listprovisionedClusterInstanceUserKubeconfig = exports.listprovisionedClusterInstanceAdminKubeconfigOutput = exports.listprovisionedClusterInstanceAdminKubeconfig = exports.KubernetesVersions = exports.HybridIdentityMetadatum = exports.getVirtualNetworkRetrieveOutput = exports.getVirtualNetworkRetrieve = exports.getVMSkusOutput = exports.getVMSkus = exports.getProvisionedClusterInstanceOutput = exports.getProvisionedClusterInstance = exports.getKubernetesVersionsOutput = exports.getKubernetesVersions = exports.getHybridIdentityMetadatumOutput = exports.getHybridIdentityMetadatum = exports.getAgentPoolOutput = exports.getAgentPool = exports.AgentPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getHybridIdentityMetadatum = null;
exports.getHybridIdentityMetadatumOutput = null;
utilities.lazyLoad(exports, ["getHybridIdentityMetadatum", "getHybridIdentityMetadatumOutput"], () => require("./getHybridIdentityMetadatum"));
exports.getKubernetesVersions = null;
exports.getKubernetesVersionsOutput = null;
utilities.lazyLoad(exports, ["getKubernetesVersions", "getKubernetesVersionsOutput"], () => require("./getKubernetesVersions"));
exports.getProvisionedClusterInstance = null;
exports.getProvisionedClusterInstanceOutput = null;
utilities.lazyLoad(exports, ["getProvisionedClusterInstance", "getProvisionedClusterInstanceOutput"], () => require("./getProvisionedClusterInstance"));
exports.getVMSkus = null;
exports.getVMSkusOutput = null;
utilities.lazyLoad(exports, ["getVMSkus", "getVMSkusOutput"], () => require("./getVMSkus"));
exports.getVirtualNetworkRetrieve = null;
exports.getVirtualNetworkRetrieveOutput = null;
utilities.lazyLoad(exports, ["getVirtualNetworkRetrieve", "getVirtualNetworkRetrieveOutput"], () => require("./getVirtualNetworkRetrieve"));
exports.HybridIdentityMetadatum = null;
utilities.lazyLoad(exports, ["HybridIdentityMetadatum"], () => require("./hybridIdentityMetadatum"));
exports.KubernetesVersions = null;
utilities.lazyLoad(exports, ["KubernetesVersions"], () => require("./kubernetesVersions"));
exports.listprovisionedClusterInstanceAdminKubeconfig = null;
exports.listprovisionedClusterInstanceAdminKubeconfigOutput = null;
utilities.lazyLoad(exports, ["listprovisionedClusterInstanceAdminKubeconfig", "listprovisionedClusterInstanceAdminKubeconfigOutput"], () => require("./listprovisionedClusterInstanceAdminKubeconfig"));
exports.listprovisionedClusterInstanceUserKubeconfig = null;
exports.listprovisionedClusterInstanceUserKubeconfigOutput = null;
utilities.lazyLoad(exports, ["listprovisionedClusterInstanceUserKubeconfig", "listprovisionedClusterInstanceUserKubeconfigOutput"], () => require("./listprovisionedClusterInstanceUserKubeconfig"));
exports.ProvisionedClusterInstance = null;
utilities.lazyLoad(exports, ["ProvisionedClusterInstance"], () => require("./provisionedClusterInstance"));
exports.VirtualNetworkRetrieve = null;
utilities.lazyLoad(exports, ["VirtualNetworkRetrieve"], () => require("./virtualNetworkRetrieve"));
exports.VMSkus = null;
utilities.lazyLoad(exports, ["VMSkus"], () => require("./vmskus"));
// Export enums:
__exportStar(require("../../types/enums/hybridcontainerservice/v20231115preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcontainerservice/v20231115preview:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20231115preview:HybridIdentityMetadatum":
                return new exports.HybridIdentityMetadatum(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20231115preview:KubernetesVersions":
                return new exports.KubernetesVersions(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20231115preview:ProvisionedClusterInstance":
                return new exports.ProvisionedClusterInstance(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20231115preview:VMSkus":
                return new exports.VMSkus(name, undefined, { urn });
            case "azure-native:hybridcontainerservice/v20231115preview:VirtualNetworkRetrieve":
                return new exports.VirtualNetworkRetrieve(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcontainerservice/v20231115preview", _module);
