// @pulumi/azure-native/manufacturingplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250301 = exports.ManufacturingDataService = exports.listManufacturingDataServiceAvailableVersionsOutput = exports.listManufacturingDataServiceAvailableVersions = exports.getManufacturingDataServiceOutput = exports.getManufacturingDataService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getManufacturingDataService = null;
exports.getManufacturingDataServiceOutput = null;
utilities.lazyLoad(exports, ["getManufacturingDataService", "getManufacturingDataServiceOutput"], () => require("./getManufacturingDataService"));
exports.listManufacturingDataServiceAvailableVersions = null;
exports.listManufacturingDataServiceAvailableVersionsOutput = null;
utilities.lazyLoad(exports, ["listManufacturingDataServiceAvailableVersions", "listManufacturingDataServiceAvailableVersionsOutput"], () => require("./listManufacturingDataServiceAvailableVersions"));
exports.ManufacturingDataService = null;
utilities.lazyLoad(exports, ["ManufacturingDataService"], () => require("./manufacturingDataService"));
// Export enums:
__exportStar(require("../types/enums/manufacturingplatform"), exports);
// Export sub-modules:
const v20250301 = require("./v20250301");
exports.v20250301 = v20250301;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:manufacturingplatform:ManufacturingDataService":
                return new exports.ManufacturingDataService(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "manufacturingplatform", _module);
