// @pulumi/azure-native/securityinsights/v20210301preview/codelessUiDataConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodelessUiDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents Codeless UI data connector.
 */
class CodelessUiDataConnector extends pulumi.CustomResource {
    /**
     * Get an existing CodelessUiDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CodelessUiDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CodelessUiDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CodelessUiDataConnector.__pulumiType;
    }
    /**
     * Create a CodelessUiDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["connectorUiConfig"] = args ? args.connectorUiConfig : undefined;
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["kind"] = "GenericUI";
            resourceInputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["connectorUiConfig"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20200101:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20211001:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220501preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220601preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220701preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220801:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220801preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20220901preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20221001preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20221101:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20221101preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20221201preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230201:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230201preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230301preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230401preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230501preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230601preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230701preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230801preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20230901preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20231001preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20231101:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20231201preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20240101preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20240301:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20240401preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20240901:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20241001preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20250101preview:CodelessUiDataConnector" }, { type: "azure-native:securityinsights/v20250301:CodelessUiDataConnector" }, { type: "azure-native:securityinsights:CodelessUiDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CodelessUiDataConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CodelessUiDataConnector = CodelessUiDataConnector;
/** @internal */
CodelessUiDataConnector.__pulumiType = 'azure-native:securityinsights/v20210301preview:CodelessUiDataConnector';
