// @pulumi/azure-native/securityinsights/v20210301preview/incidentComment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentComment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents an incident comment
 */
class IncidentComment extends pulumi.CustomResource {
    /**
     * Get an existing IncidentComment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IncidentComment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IncidentComment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IncidentComment.__pulumiType;
    }
    /**
     * Create a IncidentComment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.incidentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'incidentId'");
            }
            if ((!args || args.message === undefined) && !opts.urn) {
                throw new Error("Missing required property 'message'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["incidentCommentId"] = args ? args.incidentCommentId : undefined;
            resourceInputs["incidentId"] = args ? args.incidentId : undefined;
            resourceInputs["message"] = args ? args.message : undefined;
            resourceInputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeUtc"] = undefined /*out*/;
            resourceInputs["message"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:IncidentComment" }, { type: "azure-native:securityinsights/v20210401:IncidentComment" }, { type: "azure-native:securityinsights/v20210901preview:IncidentComment" }, { type: "azure-native:securityinsights/v20211001:IncidentComment" }, { type: "azure-native:securityinsights/v20211001preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220101preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220401preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220501preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220601preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220701preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220801:IncidentComment" }, { type: "azure-native:securityinsights/v20220801preview:IncidentComment" }, { type: "azure-native:securityinsights/v20220901preview:IncidentComment" }, { type: "azure-native:securityinsights/v20221001preview:IncidentComment" }, { type: "azure-native:securityinsights/v20221101:IncidentComment" }, { type: "azure-native:securityinsights/v20221101preview:IncidentComment" }, { type: "azure-native:securityinsights/v20221201preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230201:IncidentComment" }, { type: "azure-native:securityinsights/v20230201preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230301preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230401preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230501preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230601preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230701preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230801preview:IncidentComment" }, { type: "azure-native:securityinsights/v20230901preview:IncidentComment" }, { type: "azure-native:securityinsights/v20231001preview:IncidentComment" }, { type: "azure-native:securityinsights/v20231101:IncidentComment" }, { type: "azure-native:securityinsights/v20231201preview:IncidentComment" }, { type: "azure-native:securityinsights/v20240101preview:IncidentComment" }, { type: "azure-native:securityinsights/v20240301:IncidentComment" }, { type: "azure-native:securityinsights/v20240401preview:IncidentComment" }, { type: "azure-native:securityinsights/v20240901:IncidentComment" }, { type: "azure-native:securityinsights/v20241001preview:IncidentComment" }, { type: "azure-native:securityinsights/v20250101preview:IncidentComment" }, { type: "azure-native:securityinsights/v20250301:IncidentComment" }, { type: "azure-native:securityinsights:IncidentComment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IncidentComment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IncidentComment = IncidentComment;
/** @internal */
IncidentComment.__pulumiType = 'azure-native:securityinsights/v20210301preview:IncidentComment';
