// @pulumi/azure-native/securityinsights/v20210301preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMSTIDataConnectorOutput = exports.getMSTIDataConnector = exports.getMLBehaviorAnalyticsAlertRuleOutput = exports.getMLBehaviorAnalyticsAlertRule = exports.getMDATPDataConnectorOutput = exports.getMDATPDataConnector = exports.getMCASDataConnectorOutput = exports.getMCASDataConnector = exports.getIncidentRelationOutput = exports.getIncidentRelation = exports.getIncidentCommentOutput = exports.getIncidentComment = exports.getIncidentOutput = exports.getIncident = exports.getFusionAlertRuleOutput = exports.getFusionAlertRule = exports.getEyesOnOutput = exports.getEyesOn = exports.getEntityAnalyticsOutput = exports.getEntityAnalytics = exports.getDynamics365DataConnectorOutput = exports.getDynamics365DataConnector = exports.getCodelessUiDataConnectorOutput = exports.getCodelessUiDataConnector = exports.getAwsCloudTrailDataConnectorOutput = exports.getAwsCloudTrailDataConnector = exports.getAnomaliesOutput = exports.getAnomalies = exports.getActivityCustomEntityQueryOutput = exports.getActivityCustomEntityQuery = exports.getActionOutput = exports.getAction = exports.getASCDataConnectorOutput = exports.getASCDataConnector = exports.getAATPDataConnectorOutput = exports.getAATPDataConnector = exports.getAADDataConnectorOutput = exports.getAADDataConnector = exports.FusionAlertRule = exports.EyesOn = exports.EntityAnalytics = exports.Dynamics365DataConnector = exports.CodelessUiDataConnector = exports.AwsCloudTrailDataConnector = exports.ASCDataConnector = exports.Anomalies = exports.ActivityCustomEntityQuery = exports.Action = exports.AATPDataConnector = exports.AADDataConnector = void 0;
exports.Watchlist = exports.Ueba = exports.TIDataConnector = exports.TiTaxiiDataConnector = exports.ThreatIntelligenceAlertRule = exports.SourceControl = exports.SentinelOnboardingState = exports.ScheduledAlertRule = exports.OfficeDataConnector = exports.OfficeATPDataConnector = exports.MTPDataConnector = exports.MSTIDataConnector = exports.MLBehaviorAnalyticsAlertRule = exports.MicrosoftSecurityIncidentCreationAlertRule = exports.Metadata = exports.MDATPDataConnector = exports.MCASDataConnector = exports.listSourceControlRepositoriesOutput = exports.listSourceControlRepositories = exports.IncidentRelation = exports.IncidentComment = exports.Incident = exports.getWatchlistOutput = exports.getWatchlist = exports.getUebaOutput = exports.getUeba = exports.getTiTaxiiDataConnectorOutput = exports.getTiTaxiiDataConnector = exports.getThreatIntelligenceAlertRuleOutput = exports.getThreatIntelligenceAlertRule = exports.getTIDataConnectorOutput = exports.getTIDataConnector = exports.getSourceControlOutput = exports.getSourceControl = exports.getSentinelOnboardingStateOutput = exports.getSentinelOnboardingState = exports.getScheduledAlertRuleOutput = exports.getScheduledAlertRule = exports.getOfficeDataConnectorOutput = exports.getOfficeDataConnector = exports.getOfficeATPDataConnectorOutput = exports.getOfficeATPDataConnector = exports.getMicrosoftSecurityIncidentCreationAlertRuleOutput = exports.getMicrosoftSecurityIncidentCreationAlertRule = exports.getMetadataOutput = exports.getMetadata = exports.getMTPDataConnectorOutput = exports.getMTPDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AADDataConnector = null;
utilities.lazyLoad(exports, ["AADDataConnector"], () => require("./aaddataConnector"));
exports.AATPDataConnector = null;
utilities.lazyLoad(exports, ["AATPDataConnector"], () => require("./aatpdataConnector"));
exports.Action = null;
utilities.lazyLoad(exports, ["Action"], () => require("./action"));
exports.ActivityCustomEntityQuery = null;
utilities.lazyLoad(exports, ["ActivityCustomEntityQuery"], () => require("./activityCustomEntityQuery"));
exports.Anomalies = null;
utilities.lazyLoad(exports, ["Anomalies"], () => require("./anomalies"));
exports.ASCDataConnector = null;
utilities.lazyLoad(exports, ["ASCDataConnector"], () => require("./ascdataConnector"));
exports.AwsCloudTrailDataConnector = null;
utilities.lazyLoad(exports, ["AwsCloudTrailDataConnector"], () => require("./awsCloudTrailDataConnector"));
exports.CodelessUiDataConnector = null;
utilities.lazyLoad(exports, ["CodelessUiDataConnector"], () => require("./codelessUiDataConnector"));
exports.Dynamics365DataConnector = null;
utilities.lazyLoad(exports, ["Dynamics365DataConnector"], () => require("./dynamics365DataConnector"));
exports.EntityAnalytics = null;
utilities.lazyLoad(exports, ["EntityAnalytics"], () => require("./entityAnalytics"));
exports.EyesOn = null;
utilities.lazyLoad(exports, ["EyesOn"], () => require("./eyesOn"));
exports.FusionAlertRule = null;
utilities.lazyLoad(exports, ["FusionAlertRule"], () => require("./fusionAlertRule"));
exports.getAADDataConnector = null;
exports.getAADDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAADDataConnector", "getAADDataConnectorOutput"], () => require("./getAADDataConnector"));
exports.getAATPDataConnector = null;
exports.getAATPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAATPDataConnector", "getAATPDataConnectorOutput"], () => require("./getAATPDataConnector"));
exports.getASCDataConnector = null;
exports.getASCDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getASCDataConnector", "getASCDataConnectorOutput"], () => require("./getASCDataConnector"));
exports.getAction = null;
exports.getActionOutput = null;
utilities.lazyLoad(exports, ["getAction", "getActionOutput"], () => require("./getAction"));
exports.getActivityCustomEntityQuery = null;
exports.getActivityCustomEntityQueryOutput = null;
utilities.lazyLoad(exports, ["getActivityCustomEntityQuery", "getActivityCustomEntityQueryOutput"], () => require("./getActivityCustomEntityQuery"));
exports.getAnomalies = null;
exports.getAnomaliesOutput = null;
utilities.lazyLoad(exports, ["getAnomalies", "getAnomaliesOutput"], () => require("./getAnomalies"));
exports.getAwsCloudTrailDataConnector = null;
exports.getAwsCloudTrailDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getAwsCloudTrailDataConnector", "getAwsCloudTrailDataConnectorOutput"], () => require("./getAwsCloudTrailDataConnector"));
exports.getCodelessUiDataConnector = null;
exports.getCodelessUiDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getCodelessUiDataConnector", "getCodelessUiDataConnectorOutput"], () => require("./getCodelessUiDataConnector"));
exports.getDynamics365DataConnector = null;
exports.getDynamics365DataConnectorOutput = null;
utilities.lazyLoad(exports, ["getDynamics365DataConnector", "getDynamics365DataConnectorOutput"], () => require("./getDynamics365DataConnector"));
exports.getEntityAnalytics = null;
exports.getEntityAnalyticsOutput = null;
utilities.lazyLoad(exports, ["getEntityAnalytics", "getEntityAnalyticsOutput"], () => require("./getEntityAnalytics"));
exports.getEyesOn = null;
exports.getEyesOnOutput = null;
utilities.lazyLoad(exports, ["getEyesOn", "getEyesOnOutput"], () => require("./getEyesOn"));
exports.getFusionAlertRule = null;
exports.getFusionAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getFusionAlertRule", "getFusionAlertRuleOutput"], () => require("./getFusionAlertRule"));
exports.getIncident = null;
exports.getIncidentOutput = null;
utilities.lazyLoad(exports, ["getIncident", "getIncidentOutput"], () => require("./getIncident"));
exports.getIncidentComment = null;
exports.getIncidentCommentOutput = null;
utilities.lazyLoad(exports, ["getIncidentComment", "getIncidentCommentOutput"], () => require("./getIncidentComment"));
exports.getIncidentRelation = null;
exports.getIncidentRelationOutput = null;
utilities.lazyLoad(exports, ["getIncidentRelation", "getIncidentRelationOutput"], () => require("./getIncidentRelation"));
exports.getMCASDataConnector = null;
exports.getMCASDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMCASDataConnector", "getMCASDataConnectorOutput"], () => require("./getMCASDataConnector"));
exports.getMDATPDataConnector = null;
exports.getMDATPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMDATPDataConnector", "getMDATPDataConnectorOutput"], () => require("./getMDATPDataConnector"));
exports.getMLBehaviorAnalyticsAlertRule = null;
exports.getMLBehaviorAnalyticsAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getMLBehaviorAnalyticsAlertRule", "getMLBehaviorAnalyticsAlertRuleOutput"], () => require("./getMLBehaviorAnalyticsAlertRule"));
exports.getMSTIDataConnector = null;
exports.getMSTIDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMSTIDataConnector", "getMSTIDataConnectorOutput"], () => require("./getMSTIDataConnector"));
exports.getMTPDataConnector = null;
exports.getMTPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getMTPDataConnector", "getMTPDataConnectorOutput"], () => require("./getMTPDataConnector"));
exports.getMetadata = null;
exports.getMetadataOutput = null;
utilities.lazyLoad(exports, ["getMetadata", "getMetadataOutput"], () => require("./getMetadata"));
exports.getMicrosoftSecurityIncidentCreationAlertRule = null;
exports.getMicrosoftSecurityIncidentCreationAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getMicrosoftSecurityIncidentCreationAlertRule", "getMicrosoftSecurityIncidentCreationAlertRuleOutput"], () => require("./getMicrosoftSecurityIncidentCreationAlertRule"));
exports.getOfficeATPDataConnector = null;
exports.getOfficeATPDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getOfficeATPDataConnector", "getOfficeATPDataConnectorOutput"], () => require("./getOfficeATPDataConnector"));
exports.getOfficeDataConnector = null;
exports.getOfficeDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getOfficeDataConnector", "getOfficeDataConnectorOutput"], () => require("./getOfficeDataConnector"));
exports.getScheduledAlertRule = null;
exports.getScheduledAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getScheduledAlertRule", "getScheduledAlertRuleOutput"], () => require("./getScheduledAlertRule"));
exports.getSentinelOnboardingState = null;
exports.getSentinelOnboardingStateOutput = null;
utilities.lazyLoad(exports, ["getSentinelOnboardingState", "getSentinelOnboardingStateOutput"], () => require("./getSentinelOnboardingState"));
exports.getSourceControl = null;
exports.getSourceControlOutput = null;
utilities.lazyLoad(exports, ["getSourceControl", "getSourceControlOutput"], () => require("./getSourceControl"));
exports.getTIDataConnector = null;
exports.getTIDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getTIDataConnector", "getTIDataConnectorOutput"], () => require("./getTIDataConnector"));
exports.getThreatIntelligenceAlertRule = null;
exports.getThreatIntelligenceAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getThreatIntelligenceAlertRule", "getThreatIntelligenceAlertRuleOutput"], () => require("./getThreatIntelligenceAlertRule"));
exports.getTiTaxiiDataConnector = null;
exports.getTiTaxiiDataConnectorOutput = null;
utilities.lazyLoad(exports, ["getTiTaxiiDataConnector", "getTiTaxiiDataConnectorOutput"], () => require("./getTiTaxiiDataConnector"));
exports.getUeba = null;
exports.getUebaOutput = null;
utilities.lazyLoad(exports, ["getUeba", "getUebaOutput"], () => require("./getUeba"));
exports.getWatchlist = null;
exports.getWatchlistOutput = null;
utilities.lazyLoad(exports, ["getWatchlist", "getWatchlistOutput"], () => require("./getWatchlist"));
exports.Incident = null;
utilities.lazyLoad(exports, ["Incident"], () => require("./incident"));
exports.IncidentComment = null;
utilities.lazyLoad(exports, ["IncidentComment"], () => require("./incidentComment"));
exports.IncidentRelation = null;
utilities.lazyLoad(exports, ["IncidentRelation"], () => require("./incidentRelation"));
exports.listSourceControlRepositories = null;
exports.listSourceControlRepositoriesOutput = null;
utilities.lazyLoad(exports, ["listSourceControlRepositories", "listSourceControlRepositoriesOutput"], () => require("./listSourceControlRepositories"));
exports.MCASDataConnector = null;
utilities.lazyLoad(exports, ["MCASDataConnector"], () => require("./mcasdataConnector"));
exports.MDATPDataConnector = null;
utilities.lazyLoad(exports, ["MDATPDataConnector"], () => require("./mdatpdataConnector"));
exports.Metadata = null;
utilities.lazyLoad(exports, ["Metadata"], () => require("./metadata"));
exports.MicrosoftSecurityIncidentCreationAlertRule = null;
utilities.lazyLoad(exports, ["MicrosoftSecurityIncidentCreationAlertRule"], () => require("./microsoftSecurityIncidentCreationAlertRule"));
exports.MLBehaviorAnalyticsAlertRule = null;
utilities.lazyLoad(exports, ["MLBehaviorAnalyticsAlertRule"], () => require("./mlbehaviorAnalyticsAlertRule"));
exports.MSTIDataConnector = null;
utilities.lazyLoad(exports, ["MSTIDataConnector"], () => require("./mstidataConnector"));
exports.MTPDataConnector = null;
utilities.lazyLoad(exports, ["MTPDataConnector"], () => require("./mtpdataConnector"));
exports.OfficeATPDataConnector = null;
utilities.lazyLoad(exports, ["OfficeATPDataConnector"], () => require("./officeATPDataConnector"));
exports.OfficeDataConnector = null;
utilities.lazyLoad(exports, ["OfficeDataConnector"], () => require("./officeDataConnector"));
exports.ScheduledAlertRule = null;
utilities.lazyLoad(exports, ["ScheduledAlertRule"], () => require("./scheduledAlertRule"));
exports.SentinelOnboardingState = null;
utilities.lazyLoad(exports, ["SentinelOnboardingState"], () => require("./sentinelOnboardingState"));
exports.SourceControl = null;
utilities.lazyLoad(exports, ["SourceControl"], () => require("./sourceControl"));
exports.ThreatIntelligenceAlertRule = null;
utilities.lazyLoad(exports, ["ThreatIntelligenceAlertRule"], () => require("./threatIntelligenceAlertRule"));
exports.TiTaxiiDataConnector = null;
utilities.lazyLoad(exports, ["TiTaxiiDataConnector"], () => require("./tiTaxiiDataConnector"));
exports.TIDataConnector = null;
utilities.lazyLoad(exports, ["TIDataConnector"], () => require("./tidataConnector"));
exports.Ueba = null;
utilities.lazyLoad(exports, ["Ueba"], () => require("./ueba"));
exports.Watchlist = null;
utilities.lazyLoad(exports, ["Watchlist"], () => require("./watchlist"));
// Export enums:
__exportStar(require("../../types/enums/securityinsights/v20210301preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:securityinsights/v20210301preview:AADDataConnector":
                return new exports.AADDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:AATPDataConnector":
                return new exports.AATPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:ASCDataConnector":
                return new exports.ASCDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Action":
                return new exports.Action(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:ActivityCustomEntityQuery":
                return new exports.ActivityCustomEntityQuery(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Anomalies":
                return new exports.Anomalies(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:AwsCloudTrailDataConnector":
                return new exports.AwsCloudTrailDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:CodelessUiDataConnector":
                return new exports.CodelessUiDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Dynamics365DataConnector":
                return new exports.Dynamics365DataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:EntityAnalytics":
                return new exports.EntityAnalytics(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:EyesOn":
                return new exports.EyesOn(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:FusionAlertRule":
                return new exports.FusionAlertRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Incident":
                return new exports.Incident(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:IncidentComment":
                return new exports.IncidentComment(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:IncidentRelation":
                return new exports.IncidentRelation(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MCASDataConnector":
                return new exports.MCASDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MDATPDataConnector":
                return new exports.MDATPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MLBehaviorAnalyticsAlertRule":
                return new exports.MLBehaviorAnalyticsAlertRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MSTIDataConnector":
                return new exports.MSTIDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MTPDataConnector":
                return new exports.MTPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Metadata":
                return new exports.Metadata(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:MicrosoftSecurityIncidentCreationAlertRule":
                return new exports.MicrosoftSecurityIncidentCreationAlertRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:OfficeATPDataConnector":
                return new exports.OfficeATPDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:OfficeDataConnector":
                return new exports.OfficeDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:ScheduledAlertRule":
                return new exports.ScheduledAlertRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:SentinelOnboardingState":
                return new exports.SentinelOnboardingState(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:SourceControl":
                return new exports.SourceControl(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:TIDataConnector":
                return new exports.TIDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:ThreatIntelligenceAlertRule":
                return new exports.ThreatIntelligenceAlertRule(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:TiTaxiiDataConnector":
                return new exports.TiTaxiiDataConnector(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Ueba":
                return new exports.Ueba(name, undefined, { urn });
            case "azure-native:securityinsights/v20210301preview:Watchlist":
                return new exports.Watchlist(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "securityinsights/v20210301preview", _module);
