// @pulumi/azure-native/containerservice/v20230902preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustedAccessRoleBinding = exports.Snapshot = exports.PrivateEndpointConnection = exports.ManagedClusterSnapshot = exports.ManagedCluster = exports.MaintenanceConfiguration = exports.listManagedClusterUserCredentialsOutput = exports.listManagedClusterUserCredentials = exports.listManagedClusterMonitoringUserCredentialsOutput = exports.listManagedClusterMonitoringUserCredentials = exports.listManagedClusterAdminCredentialsOutput = exports.listManagedClusterAdminCredentials = exports.getTrustedAccessRoleBindingOutput = exports.getTrustedAccessRoleBinding = exports.getSnapshotOutput = exports.getSnapshot = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getManagedClusterSnapshotOutput = exports.getManagedClusterSnapshot = exports.getManagedClusterOutput = exports.getManagedCluster = exports.getMaintenanceConfigurationOutput = exports.getMaintenanceConfiguration = exports.getAgentPoolOutput = exports.getAgentPool = exports.AgentPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getMaintenanceConfiguration = null;
exports.getMaintenanceConfigurationOutput = null;
utilities.lazyLoad(exports, ["getMaintenanceConfiguration", "getMaintenanceConfigurationOutput"], () => require("./getMaintenanceConfiguration"));
exports.getManagedCluster = null;
exports.getManagedClusterOutput = null;
utilities.lazyLoad(exports, ["getManagedCluster", "getManagedClusterOutput"], () => require("./getManagedCluster"));
exports.getManagedClusterSnapshot = null;
exports.getManagedClusterSnapshotOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterSnapshot", "getManagedClusterSnapshotOutput"], () => require("./getManagedClusterSnapshot"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getTrustedAccessRoleBinding = null;
exports.getTrustedAccessRoleBindingOutput = null;
utilities.lazyLoad(exports, ["getTrustedAccessRoleBinding", "getTrustedAccessRoleBindingOutput"], () => require("./getTrustedAccessRoleBinding"));
exports.listManagedClusterAdminCredentials = null;
exports.listManagedClusterAdminCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterAdminCredentials", "listManagedClusterAdminCredentialsOutput"], () => require("./listManagedClusterAdminCredentials"));
exports.listManagedClusterMonitoringUserCredentials = null;
exports.listManagedClusterMonitoringUserCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterMonitoringUserCredentials", "listManagedClusterMonitoringUserCredentialsOutput"], () => require("./listManagedClusterMonitoringUserCredentials"));
exports.listManagedClusterUserCredentials = null;
exports.listManagedClusterUserCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterUserCredentials", "listManagedClusterUserCredentialsOutput"], () => require("./listManagedClusterUserCredentials"));
exports.MaintenanceConfiguration = null;
utilities.lazyLoad(exports, ["MaintenanceConfiguration"], () => require("./maintenanceConfiguration"));
exports.ManagedCluster = null;
utilities.lazyLoad(exports, ["ManagedCluster"], () => require("./managedCluster"));
exports.ManagedClusterSnapshot = null;
utilities.lazyLoad(exports, ["ManagedClusterSnapshot"], () => require("./managedClusterSnapshot"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.TrustedAccessRoleBinding = null;
utilities.lazyLoad(exports, ["TrustedAccessRoleBinding"], () => require("./trustedAccessRoleBinding"));
// Export enums:
__exportStar(require("../../types/enums/containerservice/v20230902preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerservice/v20230902preview:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:MaintenanceConfiguration":
                return new exports.MaintenanceConfiguration(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:ManagedCluster":
                return new exports.ManagedCluster(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:ManagedClusterSnapshot":
                return new exports.ManagedClusterSnapshot(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "azure-native:containerservice/v20230902preview:TrustedAccessRoleBinding":
                return new exports.TrustedAccessRoleBinding(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerservice/v20230902preview", _module);
