// @pulumi/azure-native/iotoperations/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20250401 = exports.v20241101 = exports.v20240915preview = exports.v20240815preview = exports.v20240701preview = exports.Instance = exports.getInstanceOutput = exports.getInstance = exports.getDataFlowProfileOutput = exports.getDataFlowProfile = exports.getDataFlowEndpointOutput = exports.getDataFlowEndpoint = exports.getDataFlowOutput = exports.getDataFlow = exports.getBrokerListenerOutput = exports.getBrokerListener = exports.getBrokerAuthorizationOutput = exports.getBrokerAuthorization = exports.getBrokerAuthenticationOutput = exports.getBrokerAuthentication = exports.getBrokerOutput = exports.getBroker = exports.DataFlowProfile = exports.DataFlowEndpoint = exports.DataFlow = exports.BrokerListener = exports.BrokerAuthorization = exports.BrokerAuthentication = exports.Broker = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Broker = null;
utilities.lazyLoad(exports, ["Broker"], () => require("./broker"));
exports.BrokerAuthentication = null;
utilities.lazyLoad(exports, ["BrokerAuthentication"], () => require("./brokerAuthentication"));
exports.BrokerAuthorization = null;
utilities.lazyLoad(exports, ["BrokerAuthorization"], () => require("./brokerAuthorization"));
exports.BrokerListener = null;
utilities.lazyLoad(exports, ["BrokerListener"], () => require("./brokerListener"));
exports.DataFlow = null;
utilities.lazyLoad(exports, ["DataFlow"], () => require("./dataFlow"));
exports.DataFlowEndpoint = null;
utilities.lazyLoad(exports, ["DataFlowEndpoint"], () => require("./dataFlowEndpoint"));
exports.DataFlowProfile = null;
utilities.lazyLoad(exports, ["DataFlowProfile"], () => require("./dataFlowProfile"));
exports.getBroker = null;
exports.getBrokerOutput = null;
utilities.lazyLoad(exports, ["getBroker", "getBrokerOutput"], () => require("./getBroker"));
exports.getBrokerAuthentication = null;
exports.getBrokerAuthenticationOutput = null;
utilities.lazyLoad(exports, ["getBrokerAuthentication", "getBrokerAuthenticationOutput"], () => require("./getBrokerAuthentication"));
exports.getBrokerAuthorization = null;
exports.getBrokerAuthorizationOutput = null;
utilities.lazyLoad(exports, ["getBrokerAuthorization", "getBrokerAuthorizationOutput"], () => require("./getBrokerAuthorization"));
exports.getBrokerListener = null;
exports.getBrokerListenerOutput = null;
utilities.lazyLoad(exports, ["getBrokerListener", "getBrokerListenerOutput"], () => require("./getBrokerListener"));
exports.getDataFlow = null;
exports.getDataFlowOutput = null;
utilities.lazyLoad(exports, ["getDataFlow", "getDataFlowOutput"], () => require("./getDataFlow"));
exports.getDataFlowEndpoint = null;
exports.getDataFlowEndpointOutput = null;
utilities.lazyLoad(exports, ["getDataFlowEndpoint", "getDataFlowEndpointOutput"], () => require("./getDataFlowEndpoint"));
exports.getDataFlowProfile = null;
exports.getDataFlowProfileOutput = null;
utilities.lazyLoad(exports, ["getDataFlowProfile", "getDataFlowProfileOutput"], () => require("./getDataFlowProfile"));
exports.getInstance = null;
exports.getInstanceOutput = null;
utilities.lazyLoad(exports, ["getInstance", "getInstanceOutput"], () => require("./getInstance"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
// Export enums:
__exportStar(require("../types/enums/iotoperations"), exports);
// Export sub-modules:
const v20240701preview = require("./v20240701preview");
exports.v20240701preview = v20240701preview;
const v20240815preview = require("./v20240815preview");
exports.v20240815preview = v20240815preview;
const v20240915preview = require("./v20240915preview");
exports.v20240915preview = v20240915preview;
const v20241101 = require("./v20241101");
exports.v20241101 = v20241101;
const v20250401 = require("./v20250401");
exports.v20250401 = v20250401;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:iotoperations:Broker":
                return new exports.Broker(name, undefined, { urn });
            case "azure-native:iotoperations:BrokerAuthentication":
                return new exports.BrokerAuthentication(name, undefined, { urn });
            case "azure-native:iotoperations:BrokerAuthorization":
                return new exports.BrokerAuthorization(name, undefined, { urn });
            case "azure-native:iotoperations:BrokerListener":
                return new exports.BrokerListener(name, undefined, { urn });
            case "azure-native:iotoperations:DataFlow":
                return new exports.DataFlow(name, undefined, { urn });
            case "azure-native:iotoperations:DataFlowEndpoint":
                return new exports.DataFlowEndpoint(name, undefined, { urn });
            case "azure-native:iotoperations:DataFlowProfile":
                return new exports.DataFlowProfile(name, undefined, { urn });
            case "azure-native:iotoperations:Instance":
                return new exports.Instance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "iotoperations", _module);
