// @pulumi/azure-native/network/expressRoutePortAuthorization.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRoutePortAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ExpressRoutePort Authorization resource definition.
 *
 * Uses Azure REST API version 2023-02-01. In version 1.x of the Azure Native provider, it used API version 2022-01-01.
 *
 * Other available API versions: 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-05-01.
 */
class ExpressRoutePortAuthorization extends pulumi.CustomResource {
    /**
     * Get an existing ExpressRoutePortAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExpressRoutePortAuthorization(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRoutePortAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRoutePortAuthorization.__pulumiType;
    }
    /**
     * Create a ExpressRoutePortAuthorization resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.expressRoutePortName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'expressRoutePortName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authorizationName"] = args ? args.authorizationName : undefined;
            resourceInputs["expressRoutePortName"] = args ? args.expressRoutePortName : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["authorizationKey"] = undefined /*out*/;
            resourceInputs["authorizationUseStatus"] = undefined /*out*/;
            resourceInputs["circuitResourceUri"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["authorizationKey"] = undefined /*out*/;
            resourceInputs["authorizationUseStatus"] = undefined /*out*/;
            resourceInputs["circuitResourceUri"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20210801:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20220101:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20220501:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20220701:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20220901:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20221101:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20230201:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20230401:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20230501:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20230601:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20230901:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20231101:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20240101:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20240301:ExpressRoutePortAuthorization" }, { type: "azure-native:network/v20240501:ExpressRoutePortAuthorization" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExpressRoutePortAuthorization.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExpressRoutePortAuthorization = ExpressRoutePortAuthorization;
/** @internal */
ExpressRoutePortAuthorization.__pulumiType = 'azure-native:network:ExpressRoutePortAuthorization';
