// @pulumi/azure-native/network/firewallPolicyRuleCollectionGroupDraft.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallPolicyRuleCollectionGroupDraft = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Rule Collection Group resource.
 *
 * Uses Azure REST API version 2023-11-01.
 *
 * Other available API versions: 2024-01-01, 2024-03-01, 2024-05-01.
 */
class FirewallPolicyRuleCollectionGroupDraft extends pulumi.CustomResource {
    /**
     * Get an existing FirewallPolicyRuleCollectionGroupDraft resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FirewallPolicyRuleCollectionGroupDraft(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FirewallPolicyRuleCollectionGroupDraft.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FirewallPolicyRuleCollectionGroupDraft.__pulumiType;
    }
    /**
     * Create a FirewallPolicyRuleCollectionGroupDraft resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.firewallPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'firewallPolicyName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.ruleCollectionGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ruleCollectionGroupName'");
            }
            resourceInputs["firewallPolicyName"] = args ? args.firewallPolicyName : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["ruleCollectionGroupName"] = args ? args.ruleCollectionGroupName : undefined;
            resourceInputs["ruleCollections"] = args ? args.ruleCollections : undefined;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["ruleCollections"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20231101:FirewallPolicyRuleCollectionGroupDraft" }, { type: "azure-native:network/v20240101:FirewallPolicyRuleCollectionGroupDraft" }, { type: "azure-native:network/v20240301:FirewallPolicyRuleCollectionGroupDraft" }, { type: "azure-native:network/v20240501:FirewallPolicyRuleCollectionGroupDraft" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FirewallPolicyRuleCollectionGroupDraft.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FirewallPolicyRuleCollectionGroupDraft = FirewallPolicyRuleCollectionGroupDraft;
/** @internal */
FirewallPolicyRuleCollectionGroupDraft.__pulumiType = 'azure-native:network:FirewallPolicyRuleCollectionGroupDraft';
