// @pulumi/azure-native/testbase/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20231101preview = exports.v20220401preview = exports.TestBaseAccount = exports.Package = exports.ImageDefinition = exports.getTestResultVideoDownloadURLOutput = exports.getTestResultVideoDownloadURL = exports.getTestResultDownloadURLOutput = exports.getTestResultDownloadURL = exports.getTestResultConsoleLogDownloadURLOutput = exports.getTestResultConsoleLogDownloadURL = exports.getTestBaseAccountFileUploadUrlOutput = exports.getTestBaseAccountFileUploadUrl = exports.getTestBaseAccountOutput = exports.getTestBaseAccount = exports.getPackageDownloadURLOutput = exports.getPackageDownloadURL = exports.getPackageOutput = exports.getPackage = exports.getImageDefinitionOutput = exports.getImageDefinition = exports.getFavoriteProcessOutput = exports.getFavoriteProcess = exports.getDraftPackagePathOutput = exports.getDraftPackagePath = exports.getDraftPackageOutput = exports.getDraftPackage = exports.getCustomerEventOutput = exports.getCustomerEvent = exports.getCustomImageOutput = exports.getCustomImage = exports.getCredentialOutput = exports.getCredential = exports.getBillingHubServiceUsageOutput = exports.getBillingHubServiceUsage = exports.getBillingHubServiceFreeHourBalanceOutput = exports.getBillingHubServiceFreeHourBalance = exports.getActionRequestOutput = exports.getActionRequest = exports.FavoriteProcess = exports.DraftPackage = exports.CustomerEvent = exports.CustomImage = exports.Credential = exports.ActionRequest = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ActionRequest = null;
utilities.lazyLoad(exports, ["ActionRequest"], () => require("./actionRequest"));
exports.Credential = null;
utilities.lazyLoad(exports, ["Credential"], () => require("./credential"));
exports.CustomImage = null;
utilities.lazyLoad(exports, ["CustomImage"], () => require("./customImage"));
exports.CustomerEvent = null;
utilities.lazyLoad(exports, ["CustomerEvent"], () => require("./customerEvent"));
exports.DraftPackage = null;
utilities.lazyLoad(exports, ["DraftPackage"], () => require("./draftPackage"));
exports.FavoriteProcess = null;
utilities.lazyLoad(exports, ["FavoriteProcess"], () => require("./favoriteProcess"));
exports.getActionRequest = null;
exports.getActionRequestOutput = null;
utilities.lazyLoad(exports, ["getActionRequest", "getActionRequestOutput"], () => require("./getActionRequest"));
exports.getBillingHubServiceFreeHourBalance = null;
exports.getBillingHubServiceFreeHourBalanceOutput = null;
utilities.lazyLoad(exports, ["getBillingHubServiceFreeHourBalance", "getBillingHubServiceFreeHourBalanceOutput"], () => require("./getBillingHubServiceFreeHourBalance"));
exports.getBillingHubServiceUsage = null;
exports.getBillingHubServiceUsageOutput = null;
utilities.lazyLoad(exports, ["getBillingHubServiceUsage", "getBillingHubServiceUsageOutput"], () => require("./getBillingHubServiceUsage"));
exports.getCredential = null;
exports.getCredentialOutput = null;
utilities.lazyLoad(exports, ["getCredential", "getCredentialOutput"], () => require("./getCredential"));
exports.getCustomImage = null;
exports.getCustomImageOutput = null;
utilities.lazyLoad(exports, ["getCustomImage", "getCustomImageOutput"], () => require("./getCustomImage"));
exports.getCustomerEvent = null;
exports.getCustomerEventOutput = null;
utilities.lazyLoad(exports, ["getCustomerEvent", "getCustomerEventOutput"], () => require("./getCustomerEvent"));
exports.getDraftPackage = null;
exports.getDraftPackageOutput = null;
utilities.lazyLoad(exports, ["getDraftPackage", "getDraftPackageOutput"], () => require("./getDraftPackage"));
exports.getDraftPackagePath = null;
exports.getDraftPackagePathOutput = null;
utilities.lazyLoad(exports, ["getDraftPackagePath", "getDraftPackagePathOutput"], () => require("./getDraftPackagePath"));
exports.getFavoriteProcess = null;
exports.getFavoriteProcessOutput = null;
utilities.lazyLoad(exports, ["getFavoriteProcess", "getFavoriteProcessOutput"], () => require("./getFavoriteProcess"));
exports.getImageDefinition = null;
exports.getImageDefinitionOutput = null;
utilities.lazyLoad(exports, ["getImageDefinition", "getImageDefinitionOutput"], () => require("./getImageDefinition"));
exports.getPackage = null;
exports.getPackageOutput = null;
utilities.lazyLoad(exports, ["getPackage", "getPackageOutput"], () => require("./getPackage"));
exports.getPackageDownloadURL = null;
exports.getPackageDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getPackageDownloadURL", "getPackageDownloadURLOutput"], () => require("./getPackageDownloadURL"));
exports.getTestBaseAccount = null;
exports.getTestBaseAccountOutput = null;
utilities.lazyLoad(exports, ["getTestBaseAccount", "getTestBaseAccountOutput"], () => require("./getTestBaseAccount"));
exports.getTestBaseAccountFileUploadUrl = null;
exports.getTestBaseAccountFileUploadUrlOutput = null;
utilities.lazyLoad(exports, ["getTestBaseAccountFileUploadUrl", "getTestBaseAccountFileUploadUrlOutput"], () => require("./getTestBaseAccountFileUploadUrl"));
exports.getTestResultConsoleLogDownloadURL = null;
exports.getTestResultConsoleLogDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultConsoleLogDownloadURL", "getTestResultConsoleLogDownloadURLOutput"], () => require("./getTestResultConsoleLogDownloadURL"));
exports.getTestResultDownloadURL = null;
exports.getTestResultDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultDownloadURL", "getTestResultDownloadURLOutput"], () => require("./getTestResultDownloadURL"));
exports.getTestResultVideoDownloadURL = null;
exports.getTestResultVideoDownloadURLOutput = null;
utilities.lazyLoad(exports, ["getTestResultVideoDownloadURL", "getTestResultVideoDownloadURLOutput"], () => require("./getTestResultVideoDownloadURL"));
exports.ImageDefinition = null;
utilities.lazyLoad(exports, ["ImageDefinition"], () => require("./imageDefinition"));
exports.Package = null;
utilities.lazyLoad(exports, ["Package"], () => require("./package"));
exports.TestBaseAccount = null;
utilities.lazyLoad(exports, ["TestBaseAccount"], () => require("./testBaseAccount"));
// Export enums:
__exportStar(require("../types/enums/testbase"), exports);
// Export sub-modules:
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:testbase:ActionRequest":
                return new exports.ActionRequest(name, undefined, { urn });
            case "azure-native:testbase:Credential":
                return new exports.Credential(name, undefined, { urn });
            case "azure-native:testbase:CustomImage":
                return new exports.CustomImage(name, undefined, { urn });
            case "azure-native:testbase:CustomerEvent":
                return new exports.CustomerEvent(name, undefined, { urn });
            case "azure-native:testbase:DraftPackage":
                return new exports.DraftPackage(name, undefined, { urn });
            case "azure-native:testbase:FavoriteProcess":
                return new exports.FavoriteProcess(name, undefined, { urn });
            case "azure-native:testbase:ImageDefinition":
                return new exports.ImageDefinition(name, undefined, { urn });
            case "azure-native:testbase:Package":
                return new exports.Package(name, undefined, { urn });
            case "azure-native:testbase:TestBaseAccount":
                return new exports.TestBaseAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "testbase", _module);
