// @pulumi/azure-native/securityinsights/v20231001preview/codelessApiPollingDataConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodelessApiPollingDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents Codeless API Polling data connector.
 */
class CodelessApiPollingDataConnector extends pulumi.CustomResource {
    /**
     * Get an existing CodelessApiPollingDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CodelessApiPollingDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CodelessApiPollingDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CodelessApiPollingDataConnector.__pulumiType;
    }
    /**
     * Create a CodelessApiPollingDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["connectorUiConfig"] = args ? args.connectorUiConfig : undefined;
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["kind"] = "APIPolling";
            resourceInputs["pollingConfig"] = args ? args.pollingConfig : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["connectorUiConfig"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pollingConfig"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20200101:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20211001:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220501preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220601preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220701preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220801:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220801preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20220901preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20221001preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20221101:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20221101preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20221201preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230201:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230201preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230301preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230401preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230501preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230601preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230701preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230801preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20230901preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20231101:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20231201preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20240101preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20240301:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20240401preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20240901:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20241001preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20250101preview:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights/v20250301:CodelessApiPollingDataConnector" }, { type: "azure-native:securityinsights:CodelessApiPollingDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CodelessApiPollingDataConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CodelessApiPollingDataConnector = CodelessApiPollingDataConnector;
/** @internal */
CodelessApiPollingDataConnector.__pulumiType = 'azure-native:securityinsights/v20231001preview:CodelessApiPollingDataConnector';
