// @pulumi/azure-native/securityandcompliance/privateLinkServicesForO365ManagementActivityAPI.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkServicesForO365ManagementActivityAPI = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The description of the service.
 *
 * Uses Azure REST API version 2021-03-08. In version 1.x of the Azure Native provider, it used API version 2021-03-08.
 */
class PrivateLinkServicesForO365ManagementActivityAPI extends pulumi.CustomResource {
    /**
     * Get an existing PrivateLinkServicesForO365ManagementActivityAPI resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateLinkServicesForO365ManagementActivityAPI(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateLinkServicesForO365ManagementActivityAPI.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateLinkServicesForO365ManagementActivityAPI.__pulumiType;
    }
    /**
     * Create a PrivateLinkServicesForO365ManagementActivityAPI resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityandcompliance/v20210111:PrivateLinkServicesForO365ManagementActivityAPI" }, { type: "azure-native:securityandcompliance/v20210111:privateLinkServicesForO365ManagementActivityAPI" }, { type: "azure-native:securityandcompliance/v20210308:PrivateLinkServicesForO365ManagementActivityAPI" }, { type: "azure-native:securityandcompliance/v20210308:privateLinkServicesForO365ManagementActivityAPI" }, { type: "azure-native:securityandcompliance:privateLinkServicesForO365ManagementActivityAPI" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateLinkServicesForO365ManagementActivityAPI.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateLinkServicesForO365ManagementActivityAPI = PrivateLinkServicesForO365ManagementActivityAPI;
/** @internal */
PrivateLinkServicesForO365ManagementActivityAPI.__pulumiType = 'azure-native:securityandcompliance:PrivateLinkServicesForO365ManagementActivityAPI';
